<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Coot-Cryo-EM-v9.md</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__html"><h1 id="coot-cryo-em-practical"><em>Coot</em> Cryo-EM Practical</h1>
<div> By Ana Casañal &amp; Paul Emsley</div>
<p>Aim: we will fit a domain of the Cleavage and Polyadenylation Factor</p>
<h2 id="setting-up">1: Setting Up</h2>
<h3 id="fetch-the-files">1.1 Fetch the Files</h3>
<p>Using a web browser, download the bundle for EMD-3908</p>
<ul>
<li><a href="http://www.ebi.ac.uk/pdbe/entry/emdb/EMD-3908">http://www.ebi.ac.uk/pdbe/entry/emdb/EMD-3908</a></li>
</ul>
<p>Move that tar file here and extract it.</p>
<p>Let’s download the sequence of a fragment of the structure:</p>
<ul>
<li><a href="https://www2.mrc-lmb.cam.ac.uk/personal/pemsley/coot/files/CPF-X-domain.seq">https://www2.mrc-lmb.cam.ac.uk/personal/pemsley/coot/files/CPF-X-domain.seq</a></li>
</ul>
<p>Move that sequence file to the current directory (for easy access).</p>
<p>Start <em>Coot</em>:</p>
<p><code>$ coot --map EMD-3908/map/emd_3908.map</code></p>
<h3 id="map-manipulation">1.2 Map Manipulation</h3>
<p>Let’s see more of the map</p>
<ul>
<li><strong>Edit</strong> → <strong>Map Parameters</strong> → <strong>Map Radius</strong> → <code>70</code></li>
<li><strong>OK</strong></li>
</ul>
<p>Let’s use smoother maps</p>
<ul>
<li><strong>Calculate</strong> → <strong>Modules</strong> → <strong>Cryo-EM</strong></li>
<li><strong>Cryo-EM</strong> → <strong>Multi-sharpen…</strong></li>
<li>Use 10 levels to 200</li>
<li><strong>OK</strong></li>
<li><em>{wait}</em></li>
<li>On reading the new mtz file:<br>
– In Amplitudes: choose FoutBlur_20.0<br>
– <strong>OK</strong></li>
<li>Re-read the mtz file, but this time:<br>
– In Amplitudes: choose FoutBlur_200.0<br>
– Use <strong>Expert Mode?</strong> to cut the high resolution limit to 4.0 Å</li>
</ul>
<p>Compare these maps and then delete the 1st (mrc) map.</p>
<p>Choose the FoutBlur_200.0 map for fitting (at the moment):</p>
<ul>
<li><strong>Map</strong> → {Select the FoutBlur_200.0} → <strong>OK</strong></li>
</ul>
<p>Change the contour Step for the new Maps:</p>
<p><strong>Display Manager</strong> → <strong>Properties</strong> → <strong>Change by rmsd</strong> <code>0.33</code> → <strong>OK</strong></p>
<p>As a rule of thumb, a good contour level is 5.5 rmsd, but for the blurred map we should use about 0.03 V (9.4 rmsd).</p>
<h3 id="get-the-homolog">1.3 Get the Homolog:</h3>
<ul>
<li><strong>File</strong> → <strong>Fetch Model using Accession Code…</strong> <code>6f9n</code></li>
</ul>
<p>Move back to the middle of the map with Undo Navigation:</p>
<ul>
<li>“U”</li>
</ul>
<p>Unless you’ve moved the view around, you should be at the centre of the map: (112, 112, 112). If the centre has moved, you might need a bit of manual assistance:</p>
<ul>
<li>Pan the view so that the middle of the map is at the middle of the screen</li>
<li>Now move the homolog to the centre of the map:</li>
<li><strong>Calculate</strong> → <strong>Move Molecule Here</strong></li>
<li>Check “<strong>Allow Big Molecules to Move</strong>”</li>
<li>Choose the Atom Selection/Fragment molecule</li>
<li><strong>Move It</strong></li>
</ul>
<h2 id="jiggle">2: Jiggle</h2>
<ul>
<li><strong>Morph</strong> → <strong>Jiggle Fit This Molecule</strong></li>
</ul>
<p>It should roughly fit now. If it doesn’t, try jiggling again once, twice or perhaps several times (either that or you didn’t follow the instructions 😄).</p>
<p>In this case, you should be looking for a fit score of over 1000.</p>
<h2 id="extract-our-fragment">3: Extract Our Fragment</h2>
<p>Extract the worst-fitting (WD40) domain:</p>
<ul>
<li>Using <strong>Jones’ Rainbow</strong>, find the domain start and end residues numbers</li>
<li>Let’s imagine that you think that they are 517 and 1011:</li>
<li><strong>Edit</strong> → <strong>Copy Fragment</strong> → [Use Atom Selection:] <code>//A/517-1011</code> → <strong>OK</strong></li>
</ul>
<p>Let’s work on this fragment:</p>
<ul>
<li><strong>Display Manager</strong> → <strong>Last Only</strong></li>
</ul>
<p>Let’s delete the sidechains from the atom selection molecule (number 4 usually):</p>
<ul>
<li><strong>Calculate</strong> → <strong>Modelling</strong> → <strong>Delete Side-chains for Active Chain</strong></li>
</ul>
<p>For the most recent model, use</p>
<ul>
<li><strong>C-alphas/Backbone + Ligands</strong></li>
<li><strong>OK</strong> [Dismiss the Display Manager]</li>
</ul>
<h2 id="setup-refinement">4: Setup Refinement</h2>
<p>We need to adjust the weighting of the map to the model:</p>
<ul>
<li><strong>R/RC</strong> → <strong>Estimate</strong> → <strong>OK</strong></li>
</ul>
<p>Let’s add some local distance restraints:</p>
<ul>
<li><strong>Calculate</strong> → <strong>Modules</strong> → <strong>ProSMART</strong></li>
<li><em>Usually 5.0 is a good</em></li>
<li><strong>ProSMART</strong> → <strong>Generate All Molecule Self Restraints 5.0</strong></li>
<li>Review them, then undisplay them:</li>
<li><strong>ProSMART</strong> → <strong>Undisplay Extra Distance Restraints</strong></li>
<li><strong>Refine</strong> → <strong>Set Geman-McClure alpha 0.01</strong></li>
</ul>
<h2 id="refinement">5: Refinement</h2>
<ul>
<li><strong>Refine</strong> → <strong>Chain Refine</strong></li>
<li>{wait and watch, you can turn the view if you wish}</li>
</ul>
<p>When the refinement dialog says “Success,” examine the model, being careful not to inadvertently pull on an atom. Maybe you will see that there is a domain that doesn’t fit, if so, yank on the worst fitting CA and pull it to where you think it should go.</p>
<ul>
<li>Double-clicking on an atom will release the pull restraint</li>
</ul>
<p>When you are happy, dismiss the Refinement dialog:</p>
<ul>
<li>OK</li>
</ul>
<h3 id="redo">5.1 Redo</h3>
<p>It can be non-trivial to decide what needs to move and how to move it. It is worth undoing your modifications and refining again for practice.</p>
<p>This time perhaps without drawing the restraints:</p>
<ul>
<li>Undo</li>
<li><code>set_draw_moving_atom_restraints(0)</code></li>
<li><strong>Refine</strong> → <strong>Chain Refine</strong></li>
<li><em>yank as needed</em></li>
<li><strong>OK</strong></li>
</ul>
<p>… or with different cut-off for the Geman-McClure restraints, or a different alpha for the Geman-McClure restraints, or a different weight for the map. Or a different blur for the map. You can delete the current extra restraints with <strong>ProSMART</strong> → <strong>Delete All Extra Restraints</strong>.</p>
<ul>
<li>Test, play, refine, yank until satisfied.</li>
</ul>
<p>Reset Geman-McClure alpha to 1.</p>
<h2 id="review-and-trim">6: Review and Trim</h2>
<p>Upon review, you will notice that there are parts of the model that<br>
don’t fit the map. Try yanking them around with Tandem Refine. Other<br>
parts of the model don’t have density - so delete the residue range - this will help the alignment we are about to do.</p>
<p>Maybe the density fit validation dialog will be useful? You will need<br>
to reset the weight:</p>
<ul>
<li><strong>Edit</strong> → <strong>Settings</strong> → <strong>Set Density Fit Graph Weight</strong> <code>0.3</code> → <strong>OK</strong></li>
<li><strong>Validate</strong> → <strong>Density Fit Analysis</strong></li>
</ul>
<h2 id="mutate">7: Mutate</h2>
<ul>
<li><strong>Calculate</strong> → <strong>Scripting</strong> → <strong>Python</strong></li>
<li><code>print_sequence(4)</code>  Note: the molecule number that was a result of “Copy Fragment”</li>
<li>Open a web browser to:
<ul>
<li><a href="https://www.ebi.ac.uk/Tools/msa/muscle/">https://www.ebi.ac.uk/Tools/msa/muscle/</a></li>
</ul>
</li>
<li>Paste the sequence from <code>CPF-X-domain.seq</code></li>
<li>Paste the sequence output from the print_sequence() above</li>
<li><em>Note: the order of these is important</em></li>
<li>Change the Output to “Pearson/FASTA”</li>
<li>Submit</li>
<li>Save the alignment to <code>a1-align.fasta</code> (say)</li>
<li>In the Python Scripting window:</li>
<li><code>associate_pir_alignment_from_file(4, "A", "a1-align.fasta")</code> Note: use the molecule number that was a result of “Copy Fragment”</li>
<li>You can tell that it worked because there is alignment info in the terminal</li>
<li><code>apply_pir_alignment(4, "A")</code></li>
<li><code>simple_fill_partial_residues(4)</code></li>
<li><em>{wait}</em></li>
<li><strong>Refine</strong> → <strong>Chain Refine</strong></li>
<li><strong>OK</strong></li>
</ul>
<p>(Note: this interface will change in the near future)</p>
<h2 id="check--edit">8: Check &amp; Edit</h2>
<p>Go through the structure residue by residue looking for things to fix</p>
<ul>
<li>Missing side-chains</li>
<li>Missing loops</li>
<li>Loops with too many residues in the model</li>
<li>Bad rotamers</li>
<li>Bad peptides</li>
<li>Clashes</li>
<li>Use: <strong>Validate</strong> → <strong>Alignment vs. PIR…</strong> to help you adjust the residue numbers</li>
</ul>
<p>There will be places where you need to close (or open) a loop by renumbering residues.</p>
<h2 id="now-back-to-the-demo-box">9: Now Back To the Demo Box</h2>
<ul>
<li><strong>Calculate</strong> → <strong>Run Script</strong> → <code>demo-box-of-buttons-madrid.scm</code></li>
<li><strong>Open EM map (Baton Building)</strong></li>
<li>Use Multi-sharpen (as above, Section 1.2) to make a map blurred by 10 A² based on the map that was just read</li>
<li>Select that map for building</li>
<li>Build into it with baton building:</li>
<li><strong>Calculate</strong> → <strong>Other Modelling Tools</strong> → <strong>C-alpha Baton Mode…</strong></li>
<li>Skeletonize the blurred map</li>
<li>Adjust the Skeletonization Level:</li>
<li><strong>Edit</strong> → <strong>Skeleton Parameters</strong>… → <strong>Skeleton Level</strong>: → <code>0.08</code></li>
<li>Add CA positions at the arrow tip - Happy building!
<ul>
<li>Truth be told, this takes some skill, but is useful for <em>de novo</em> model-building when nothing else does the job</li>
</ul>
</li>
<li>When you have built (say) 40 CAs:</li>
<li><strong>CA Zone → Mainchain</strong></li>
<li><em>Which direction is the correct direction? How can you tell?</em></li>
</ul>
<h2 id="done">11: Done</h2>
<p>Apart from the alignment in Section 8, you can, with a bit of practice, comfortably do both of these tutorials in less than half an hour. You may have some experience with <em>Coot</em> and now have got this far, so try to refine (with your favourite macromolecular refinement program) the domain that you have just modelled. Use validation and fix up some issues. You can use <em>Coot</em>’s validation analysis:</p>
<ul>
<li>Calculate → Scripting → Scheme</li>
<li>Use something like:</li>
<li><code>(validation-outliers-dialog 4 1)</code></li>
<li>Where 4 is the model molecule number and 1 is the map molecule number</li>
<li><em>{wait}</em></li>
<li>Click on the baddies</li>
<li>Note that CG torsion analysis and density correlation analysis are only useful on the very best cryo-EM maps</li>
</ul>
<p>After that, you can try another domain (I suggest the “B” chain).</p>
</div>
</body>

</html>
