<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<!-- This Manual is for Coot version 0.7 and beyond

Copyright (C) 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012 Paul Emsley
 -->
<!-- Created by GNU Texinfo 5.1, http://www.gnu.org/software/texinfo/ -->
<head>
<title>The Coot User Manual</title>

<meta name="description" content="The Coot User Manual">
<meta name="keywords" content="The Coot User Manual">
<meta name="resource-type" content="document">
<meta name="distribution" content="global">
<meta name="Generator" content="makeinfo">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link href="#Top" rel="start" title="Top">
<link href="#Concept-Index" rel="index" title="Concept Index">
<link href="#SEC_Contents" rel="contents" title="Table of Contents">
<link href="dir.html#Top" rel="up" title="(dir)">
<style type="text/css">
<!--
a.summary-letter {text-decoration: none}
blockquote.smallquotation {font-size: smaller}
div.display {margin-left: 3.2em}
div.example {margin-left: 3.2em}
div.indentedblock {margin-left: 3.2em}
div.lisp {margin-left: 3.2em}
div.smalldisplay {margin-left: 3.2em}
div.smallexample {margin-left: 3.2em}
div.smallindentedblock {margin-left: 3.2em; font-size: smaller}
div.smalllisp {margin-left: 3.2em}
kbd {font-style:oblique}
pre.display {font-family: inherit}
pre.format {font-family: inherit}
pre.menu-comment {font-family: serif}
pre.menu-preformatted {font-family: serif}
pre.smalldisplay {font-family: inherit; font-size: smaller}
pre.smallexample {font-size: smaller}
pre.smallformat {font-family: inherit; font-size: smaller}
pre.smalllisp {font-size: smaller}
span.nocodebreak {white-space:nowrap}
span.nolinebreak {white-space:nowrap}
span.roman {font-family:serif; font-weight:normal}
span.sansserif {font-family:sans-serif; font-weight:normal}
ul.no-bullet {list-style: none}
-->
</style>


</head>

<body lang="en" bgcolor="#FFFFFF" text="#000000" link="#0000FF" vlink="#800080" alink="#FF0000">
<h1 class="settitle" align="center">The Coot User Manual</h1>











     
<a name="SEC_Contents"></a>
<h2 class="contents-heading">Table of Contents</h2>

<div class="contents">

<ul class="no-bullet">
  <li><a name="toc-Introduction-1" href="#Introduction">1 Introduction</a>
  <ul class="no-bullet">
    <li><a name="toc-Citing-Coot-and-Friends" href="#Citing-Coot">1.1 Citing Coot and Friends</a></li>
    <li><a name="toc-What-is-Coot_003f-1" href="#What-is-Coot_003f">1.2 What is Coot?</a></li>
    <li><a name="toc-What-Coot-is-Not-1" href="#What-Coot-is-Not">1.3 What Coot is Not</a></li>
    <li><a name="toc-Hardware-Requirements-1" href="#Hardware-Requirements">1.4 Hardware Requirements</a>
    <ul class="no-bullet">
      <li><a name="toc-Mouse" href="#Mouse">1.4.1 Mouse</a></li>
    </ul></li>
    <li><a name="toc-Environment-Variables-1" href="#Environment-Variables">1.5 Environment Variables</a></li>
    <li><a name="toc-Command-Line-Arguments-1" href="#Command-Line-Arguments">1.6 Command Line Arguments</a></li>
    <li><a name="toc-Web-Page-1" href="#Web-Page">1.7 Web Page</a></li>
    <li><a name="toc-Crash-1" href="#Crash">1.8 Crash</a></li>
  </ul></li>
  <li><a name="toc-Mousing-and-Keyboarding-1" href="#Mousing-and-Keyboarding">2 Mousing and Keyboarding</a>
  <ul class="no-bullet">
    <li><a name="toc-Next-Residue-1" href="#Next-Residue">2.1 Next Residue</a></li>
    <li><a name="toc-Keyboard-Contouring-1" href="#Keyboard-Contouring">2.2 Keyboard Contouring</a></li>
    <li><a name="toc-Mouse-Z-Translation-and-Clipping" href="#Mouse-Z-Translation">2.3 Mouse Z Translation and Clipping</a></li>
    <li><a name="toc-Keyboard-Translation" href="#Keyboard-Z-Translation">2.4 Keyboard Translation</a></li>
    <li><a name="toc-Keyboard-Zoom-and-Clip-1" href="#Keyboard-Zoom-and-Clip">2.5 Keyboard Zoom and Clip</a></li>
    <li><a name="toc-Scrollwheel-1" href="#Scrollwheel">2.6 Scrollwheel</a></li>
    <li><a name="toc-Selecting-Atoms-1" href="#Selecting-Atoms">2.7 Selecting Atoms</a></li>
    <li><a name="toc-Virtual-Trackball-1" href="#Virtual-Trackball">2.8 Virtual Trackball</a></li>
    <li><a name="toc-More-on-Zooming" href="#more-on-zooming">2.9 More on Zooming</a></li>
  </ul></li>
  <li><a name="toc-General-Features-1" href="#General-Features">3 General Features</a>
  <ul class="no-bullet">
    <li><a name="toc-Version-number-1" href="#Version-number">3.1 Version number</a></li>
    <li><a name="toc-Antialiasing-1" href="#Antialiasing">3.2 Antialiasing</a></li>
    <li><a name="toc-Molecule-Number-1" href="#Molecule-Number">3.3 Molecule Number</a></li>
    <li><a name="toc-Display-Issues-1" href="#Display-Issues">3.4 Display Issues</a>
    <ul class="no-bullet">
      <li><a name="toc-Stereo" href="#Stereo">3.4.1 Stereo</a></li>
      <li><a name="toc-Pick-Cursor" href="#Pick-Cursor">3.4.2 Pick Cursor</a></li>
      <li><a name="toc-Origin-Marker" href="#Origin-Marker">3.4.3 Origin Marker</a></li>
    </ul></li>
    <li><a name="toc-Screenshot-1" href="#Screenshot">3.5 Screenshot</a></li>
    <li><a name="toc-Raster3D-output" href="#Raster3D">3.6 Raster3D output</a></li>
    <li><a name="toc-Display-Manager-1" href="#Display-Manager">3.7 Display Manager</a></li>
    <li><a name="toc-The-Modelling-Toolbar-1" href="#The-Modelling-Toolbar">3.8 The Modelling Toolbar</a></li>
    <li><a name="toc-The-file-selector-1" href="#The-file-selector">3.9 The file selector</a>
    <ul class="no-bullet">
      <li><a name="toc-File_002dname-Filtering" href="#File_002dname-Filtering">3.9.1 File-name Filtering</a></li>
      <li><a name="toc-Filename-Sorting" href="#Filename-Sorting">3.9.2 Filename Sorting</a></li>
      <li><a name="toc-Save-Coordinates-Directory" href="#Save-Coordinates-Directory">3.9.3 Save Coordinates Directory</a></li>
    </ul></li>
    <li><a name="toc-Scripting-1" href="#Scripting">3.10 Scripting</a>
    <ul class="no-bullet">
      <li><a name="toc-Python-1" href="#Python">3.10.1 Python</a>
      <ul class="no-bullet">
        <li><a name="toc-Python-Commands" href="#Python-Commands">3.10.1.1 Python Commands</a></li>
      </ul></li>
      <li><a name="toc-Scheme-1" href="#Scheme">3.10.2 Scheme</a></li>
      <li><a name="toc-Coot-State-1" href="#Coot-State">3.10.3 Coot State</a></li>
      <li><a name="toc-Key-Binding-1" href="#Key-Binding">3.10.4 Key Binding</a></li>
      <li><a name="toc-User_002dDefined-Functions-1" href="#User_002dDefined-Functions">3.10.5 User-Defined Functions</a></li>
    </ul></li>
    <li><a name="toc-Backups-and-Undo-1" href="#Backups-and-Undo">3.11 Backups and Undo</a>
    <ul class="no-bullet">
      <li><a name="toc-Redo" href="#Redo">3.11.1 Redo</a></li>
      <li><a name="toc-Restoring-from-Backup" href="#Restoring-from-Backup">3.11.2 Restoring from Backup</a></li>
    </ul></li>
    <li><a name="toc-View-Matrix-1" href="#View-Matrix">3.12 View Matrix</a></li>
    <li><a name="toc-Space-Group-and-Symmetry-1" href="#Space-Group-and-Symmetry">3.13 Space Group and Symmetry</a></li>
    <li><a name="toc-Recentring-View-1" href="#Recentring-View">3.14 Recentring View</a></li>
    <li><a name="toc-Views-1" href="#Views">3.15 Views</a></li>
    <li><a name="toc-Clipping-Manipulation-1" href="#Clipping-Manipulation">3.16 Clipping Manipulation</a></li>
    <li><a name="toc-Background-colour-1" href="#Background-colour">3.17 Background colour</a></li>
    <li><a name="toc-Unit-Cell-1" href="#Unit-Cell">3.18 Unit Cell</a></li>
    <li><a name="toc-Rotation-Centre-Pointer-1" href="#Rotation-Centre-Pointer">3.19 Rotation Centre Pointer</a></li>
    <li><a name="toc-Orientation-Axes-1" href="#Orientation-Axes">3.20 Orientation Axes</a></li>
    <li><a name="toc-Pointer-Distances-1" href="#Pointer-Distances">3.21 Pointer Distances</a></li>
    <li><a name="toc-Crosshairs-1" href="#Crosshairs">3.22 Crosshairs</a></li>
    <li><a name="toc-3D-Annotations-1" href="#g_t3D-Annotations">3.23 3D Annotations</a></li>
    <li><a name="toc-Frame-Rate-1" href="#Frame-Rate">3.24 Frame Rate</a></li>
    <li><a name="toc-Program-Output-1" href="#Program-Output">3.25 Program Output</a></li>
  </ul></li>
  <li><a name="toc-Coordinate_002dRelated-Features-1" href="#Coordinate_002dRelated-Features">4 Coordinate-Related Features</a>
  <ul class="no-bullet">
    <li><a name="toc-Reading-coordinates-1" href="#Reading-coordinates">4.1 Reading coordinates</a>
    <ul class="no-bullet">
      <li><a name="toc-A-Note-on-Space-Groups-Names" href="#A-Note-on-Space-Groups-Names">4.1.1 A Note on Space Groups Names</a></li>
      <li><a name="toc-Read-multiple-coordinate-files" href="#Read-multiple-coordinate-files">4.1.2 Read multiple coordinate files</a></li>
      <li><a name="toc-SHELX-_002eins_002f_002eres-files" href="#SHELX-_002eins_002f_002eres-files">4.1.3 SHELX .ins/.res files</a></li>
    </ul></li>
    <li><a name="toc-Atom-Info-1" href="#Atom-Info">4.2 Atom Info</a></li>
    <li><a name="toc-Atom-Labeling-1" href="#Atom-Labeling">4.3 Atom Labeling</a></li>
    <li><a name="toc-Atom-Colouring-1" href="#Atom-Colouring">4.4 Atom Colouring</a></li>
    <li><a name="toc-Bond-Parameters-1" href="#Bond-Parameters">4.5 Bond Parameters</a>
    <ul class="no-bullet">
      <li><a name="toc-Bond-Thickness" href="#Bond-Thickness">4.5.1 Bond Thickness</a></li>
      <li><a name="toc-Display-Hydrogens" href="#Display-Hydrogens">4.5.2 Display Hydrogens</a></li>
      <li><a name="toc-NCS-Ghosts-Coordinates" href="#NCS-Ghosts-Coordinates">4.5.3 NCS Ghosts Coordinates</a></li>
      <li><a name="toc-NCS-Maps" href="#NCS-Maps">4.5.4 NCS Maps</a></li>
      <li><a name="toc-Using-Strict-NCS" href="#Using-Strict-NCS">4.5.5 Using Strict NCS</a></li>
    </ul></li>
    <li><a name="toc-Download-coordinates-1" href="#Download-coordinates">4.6 Download coordinates</a></li>
    <li><a name="toc-Get-Coordinates-and-Map-from-EDS-1" href="#Get-Coordinates-and-Map-from-EDS">4.7 Get Coordinates and Map from EDS</a></li>
    <li><a name="toc-Save-Coordinates-1" href="#Save-coordinates">4.8 Save Coordinates</a></li>
    <li><a name="toc-Setting-the-Space-Group-1" href="#Setting-the-Space-Group">4.9 Setting the Space Group</a></li>
    <li><a name="toc-Anisotropic-Atoms-1" href="#Anisotropic-Atoms">4.10 Anisotropic Atoms</a></li>
    <li><a name="toc-Symmetry-1" href="#Symmetry">4.11 Symmetry</a>
    <ul class="no-bullet">
      <li><a name="toc-Missing-symmetry" href="#Missing-symmetry">4.11.1 Missing symmetry</a></li>
    </ul></li>
    <li><a name="toc-Sequence-View-1" href="#Sequence-View">4.12 Sequence View</a></li>
    <li><a name="toc-Print-Sequence-1" href="#Print-Sequence">4.13 Print Sequence</a></li>
    <li><a name="toc-Environment-Distances-1" href="#Environment-Distances">4.14 Environment Distances</a></li>
    <li><a name="toc-Distances-and-Angles-1" href="#Distances-and-Angles">4.15 Distances and Angles</a></li>
    <li><a name="toc-Zero-Occupancy-Marker-1" href="#Zero-Occupancy-Marker">4.16 Zero Occupancy Marker</a></li>
    <li><a name="toc-Atomic-Dots-1" href="#Atomic-Dots">4.17 Atomic Dots</a></li>
    <li><a name="toc-Ball-and-Stick-Representation-1" href="#Ball-and-Stick-Representation">4.18 Ball and Stick Representation</a></li>
    <li><a name="toc-Mean_002c-Median-Temperature-Factors" href="#Mean-and-Median-Temperature-Factors">4.19 Mean, Median Temperature Factors</a></li>
    <li><a name="toc-Secondary-Structure-Matching-_0028SSM_0029-1" href="#Secondary-Structure-Matching-_0028SSM_0029">4.20 Secondary Structure Matching (SSM)</a></li>
    <li><a name="toc-Least_002dSquares-Fitting-1" href="#Least_002dSquares-Fitting">4.21 Least-Squares Fitting</a></li>
    <li><a name="toc-Ligand-Overlaying-1" href="#Ligand-Overlaying">4.22 Ligand Overlaying</a></li>
    <li><a name="toc-Writing-PDB-files-1" href="#Writing-PDB-files">4.23 Writing PDB files</a></li>
  </ul></li>
  <li><a name="toc-Modelling-and-Building-1" href="#Modelling-and-Building">5 Modelling and Building</a>
  <ul class="no-bullet">
    <li><a name="toc-Regularization-and-Real-Space-Refinement-1" href="#Regularization-and-Real-Space-Refinement">5.1 Regularization and Real Space Refinement</a>
    <ul class="no-bullet">
      <li><a name="toc-Dictionary-1" href="#Dictionary">5.1.1 Dictionary</a></li>
      <li><a name="toc-Sphere-Refinement-1" href="#Sphere-Refinement">5.1.2 Sphere Refinement</a></li>
      <li><a name="toc-Refining-Specific-Residues-1" href="#Refining-Specific-Residues">5.1.3 Refining Specific Residues</a></li>
      <li><a name="toc-Refining-Carbohydrates-1" href="#Refining-Carbohydrates">5.1.4 Refining Carbohydrates</a></li>
      <li><a name="toc-Planar-Peptide-Restraints-1" href="#Planar-Peptide-Restraints">5.1.5 Planar Peptide Restraints</a></li>
      <li><a name="toc-The-UNK-residue-type-1" href="#The-UNK-residue-type">5.1.6 The UNK residue type</a></li>
      <li><a name="toc-Moving-Zero-Occupancy-Atoms-1" href="#Moving-Zero-Occupancy-Atoms">5.1.7 Moving Zero Occupancy Atoms</a></li>
    </ul></li>
    <li><a name="toc-Changing-the-Map-for-Building_002fRefinement-1" href="#Changing-the-Map-for-Building_002fRefinement">5.2 Changing the Map for Building/Refinement</a></li>
    <li><a name="toc-Rotate_002fTranslate-Zone-1" href="#Rotate_002fTranslate-Zone">5.3 Rotate/Translate Zone</a></li>
    <li><a name="toc-Rigid-Body-Refinement-1" href="#Rigid-Body-Refinement">5.4 Rigid Body Refinement</a></li>
    <li><a name="toc-Simplex-Refinement-1" href="#Simplex-Refinement">5.5 Simplex Refinement</a></li>
    <li><a name="toc-Post_002dmanipulation_002dhook-1" href="#Post_002dmanipulation_002dhook">5.6 Post-manipulation-hook</a></li>
    <li><a name="toc-Baton-Building-1" href="#Baton-Building">5.7 Baton Building</a>
    <ul class="no-bullet">
      <li><a name="toc-Undo" href="#Undo">5.7.1 Undo</a></li>
      <li><a name="toc-Missing-Skeleton" href="#Missing-Skeleton">5.7.2 Missing Skeleton</a></li>
      <li><a name="toc-Building-Backwards" href="#Building-Backwards">5.7.3 Building Backwards</a></li>
    </ul></li>
    <li><a name="toc-Reversing-Direction-of-Fragment-1" href="#Reversing-Direction-of-Fragment">5.8 Reversing Direction of Fragment</a></li>
    <li><a name="toc-C_005calpha-_002d_003e-Mainchain" href="#C_002dalpha-_002d_003e-Mainchain">5.9 C<em>\alpha</em> -&gt; Mainchain</a></li>
    <li><a name="toc-Backbone-Torsion-Angles-1" href="#Backbone-Torsion-Angles">5.10 Backbone Torsion Angles</a></li>
    <li><a name="toc-Docking-Sidechains-1" href="#Docking-Sidechains">5.11 Docking Sidechains</a></li>
    <li><a name="toc-Rotamers-1" href="#Rotamers">5.12 Rotamers</a>
    <ul class="no-bullet">
      <li><a name="toc-Auto-Fit-Rotamer" href="#Auto-Fit-Rotamer">5.12.1 Auto Fit Rotamer</a>
      <ul class="no-bullet">
        <li><a name="toc-Backrub-Rotamers" href="#Backrub-Rotamers">5.12.1.1 Backrub Rotamers</a></li>
      </ul></li>
      <li><a name="toc-De_002dclashing-residues" href="#De_002dclashing-residues">5.12.2 De-clashing residues</a></li>
    </ul></li>
    <li><a name="toc-Editing-chi-Angles" href="#Editing-Chi-Angles">5.13 Editing chi Angles</a></li>
    <li><a name="toc-Torsion-General-1" href="#Torsion-General">5.14 Torsion General</a>
    <ul class="no-bullet">
      <li><a name="toc-Ligand-Torsion-angles" href="#Ligand-Torsion-angles">5.14.1 Ligand Torsion angles</a></li>
    </ul></li>
    <li><a name="toc-Pep_002dflip-1" href="#Pep_002dflip">5.15 Pep-flip</a></li>
    <li><a name="toc-Add-Alternate-Conformation" href="#Adding-Alternative-Conformations">5.16 Add Alternate Conformation</a></li>
    <li><a name="toc-Mutation-1" href="#Mutation">5.17 Mutation</a>
    <ul class="no-bullet">
      <li><a name="toc-Mutating-DNA_002fRNA" href="#Mutating-DNA_002fRNA">5.17.1 Mutating DNA/RNA</a></li>
      <li><a name="toc-Multiple-mutations" href="#Multiple-mutations">5.17.2 Multiple mutations</a></li>
      <li><a name="toc-Mutating-to-a-Non_002dStandard-Residue" href="#Mutating-to-a-Non_002dStandard-Residue">5.17.3 Mutating to a Non-Standard Residue</a></li>
      <li><a name="toc-Mutate-and-Autofit" href="#Mutate-and-Autofit">5.17.4 Mutate and Autofit</a></li>
      <li><a name="toc-Renumbering" href="#Renumbering">5.17.5 Renumbering</a></li>
    </ul></li>
    <li><a name="toc-Importing-Lignds_002fMonomers" href="#Importing-Ligands_002fMonomers">5.18 Importing Lignds/Monomers</a></li>
    <li><a name="toc-Ligand-from-SMILES-strings-1" href="#Ligand-from-SMILES-strings">5.19 Ligand from SMILES strings</a></li>
    <li><a name="toc-Find-Ligands-1" href="#Find-Ligands">5.20 Find Ligands</a>
    <ul class="no-bullet">
      <li><a name="toc-Flexible-Ligands" href="#Flexible-Ligands">5.20.1 Flexible Ligands</a></li>
      <li><a name="toc-Adding-Ligands-to-Model" href="#Adding-Ligands-to-Model">5.20.2 Adding Ligands to Model</a></li>
    </ul></li>
    <li><a name="toc-Flip-Ligand-1" href="#Flip-Ligand">5.21 Flip Ligand</a></li>
    <li><a name="toc-Find-Waters-1" href="#Find-Waters">5.22 Find Waters</a>
    <ul class="no-bullet">
      <li><a name="toc-Refinement-Failure" href="#Refinement-Failure">5.22.1 Refinement Failure</a></li>
      <li><a name="toc-Blobs" href="#Blobs">5.22.2 Blobs</a></li>
    </ul></li>
    <li><a name="toc-Add-Terminal-Residue-1" href="#Add-Terminal-Residue">5.23 Add Terminal Residue</a></li>
    <li><a name="toc-Add-OXT-Atom-to-Residue-1" href="#Add-OXT-Atom-to-Residue">5.24 Add OXT Atom to Residue</a></li>
    <li><a name="toc-Add-Atom-at-Pointer-1" href="#Add-Atom-at-Pointer">5.25 Add Atom at Pointer</a></li>
    <li><a name="toc-Place-Helix-1" href="#Place-Helix">5.26 Place Helix</a></li>
    <li><a name="toc-Building-Ideal-DNA-and-RNA-1" href="#Building-Ideal-DNA-and-RNA">5.27 Building Ideal DNA and RNA</a></li>
    <li><a name="toc-Merge-Molecules-1" href="#Merge-Molecules">5.28 Merge Molecules</a></li>
    <li><a name="toc-Temperature-Factor-for-New-Atoms-1" href="#Temperature-Factor-for-New-Atoms">5.29 Temperature Factor for New Atoms</a></li>
    <li><a name="toc-Applying-NCS-Edits-1" href="#Applying-NCS-Edits">5.30 Applying NCS Edits</a></li>
    <li><a name="toc-Running-Refmac-1" href="#Running-Refmac">5.31 Running Refmac</a></li>
    <li><a name="toc-Running-SHELXL-1" href="#Running-SHELXL">5.32 Running SHELXL</a></li>
    <li><a name="toc-Clear-Pending-Picks-1" href="#Clear-Pending-Picks">5.33 Clear Pending Picks</a></li>
    <li><a name="toc-Delete-1" href="#Delete">5.34 Delete</a></li>
    <li><a name="toc-Sequence-Assignment-1" href="#Sequence-Assignment">5.35 Sequence Assignment</a></li>
    <li><a name="toc-Building-Links-and-Loops-1" href="#Building-Links-and-Loops">5.36 Building Links and Loops</a></li>
    <li><a name="toc-Fill-Partial-Residues-1" href="#Fill-Partial-Residues">5.37 Fill Partial Residues</a></li>
    <li><a name="toc-Changing-Chain-IDs-1" href="#Changing-Chain-IDs">5.38 Changing Chain IDs</a></li>
    <li><a name="toc-Setting-Occupancies-1" href="#Setting-Occupancies">5.39 Setting Occupancies</a></li>
    <li><a name="toc-Fix-Nomenclature-Errors-1" href="#Fix-Nomenclature-Errors">5.40 Fix Nomenclature Errors</a></li>
    <li><a name="toc-Rotamer-Fix-Whole-Protein-1" href="#Rotamer-Fix-Whole-Protein">5.41 Rotamer Fix Whole Protein</a></li>
    <li><a name="toc-Refine-All-Waters-1" href="#Refine-All-Waters">5.42 Refine All Waters</a></li>
    <li><a name="toc-Moving-Molecules_002fLigands-1" href="#Moving-Molecules_002fLigands">5.43 Moving Molecules/Ligands</a></li>
    <li><a name="toc-Modifying-the-Labels-on-the-Model_002fFit_002fRefine-dialog-1" href="#Modifying-the-Labels-on-the-Model_002fFit_002fRefine-dialog">5.44 Modifying the Labels on the Model/Fit/Refine dialog</a></li>
  </ul></li>
  <li><a name="toc-Map_002dRelated-Features-1" href="#Map_002dRelated-Features">6 Map-Related Features</a>
  <ul class="no-bullet">
    <li><a name="toc-Maps-in-General-1" href="#Maps-in-General">6.1 Maps in General</a>
    <ul class="no-bullet">
      <li><a name="toc-Map-Reading-Bug" href="#Map-Reading-Bug">6.1.1 Map Reading Bug</a></li>
    </ul></li>
    <li><a name="toc-Create-a-Map-1" href="#Create-a-Map">6.2 Create a Map</a>
    <ul class="no-bullet">
      <li><a name="toc-Auto_002dread-MTZ-file" href="#Auto_002dread-MTZ-file">6.2.1 Auto-read MTZ file</a></li>
      <li><a name="toc-Reading-CIF-data" href="#Reading-CIF-data">6.2.2 Reading CIF data</a></li>
      <li><a name="toc-Reading-PHS-data" href="#Reading-PHS-data">6.2.3 Reading PHS data</a></li>
    </ul></li>
    <li><a name="toc-Map-Contouring-1" href="#Map-Contouring">6.3 Map Contouring</a></li>
    <li><a name="toc-Map-Extent-1" href="#Map-Extent">6.4 Map Extent</a></li>
    <li><a name="toc-Map-Contour-_0060_0060Scrolling_0027_0027-Limits-1" href="#Map-Contour-_0060_0060Scrolling_0027_0027-Limits">6.5 Map Contour &ldquo;Scrolling&rdquo; Limits</a></li>
    <li><a name="toc-Map-Line-Width-1" href="#Map-Line-Width">6.6 Map Line Width</a></li>
    <li><a name="toc-_0060_0060Dynamic_0027_0027-Map-colouring" href="#Map-colouring">6.7 &ldquo;Dynamic&rdquo; Map colouring</a></li>
    <li><a name="toc-Difference-Map-Colouring-1" href="#Difference-Map-Colouring">6.8 Difference Map Colouring</a></li>
    <li><a name="toc-Make-a-Difference-Map-1" href="#Make-a-Difference-Map">6.9 Make a Difference Map</a></li>
    <li><a name="toc-Make-an-Averaged-Map-1" href="#Make-an-Averaged-Map">6.10 Make an Averaged Map</a></li>
    <li><a name="toc-Map-Sampling-1" href="#Map-Sampling">6.11 Map Sampling</a></li>
    <li><a name="toc-Dragged-Map-1" href="#Dragged-Map">6.12 Dragged Map</a></li>
    <li><a name="toc-Dynamic-Map-Sampling-and-Display-Size-1" href="#Dynamic-Map-Sampling-and-Display-Size">6.13 Dynamic Map Sampling and Display Size</a></li>
    <li><a name="toc-Skeletonization-1" href="#Skeletonization">6.14 Skeletonization</a></li>
    <li><a name="toc-Map-Sharpening-1" href="#Map-Sharpening">6.15 Map Sharpening</a></li>
    <li><a name="toc-Pattersons-1" href="#Pattersons">6.16 Pattersons</a></li>
    <li><a name="toc-Map-Re_002dInterpolation-1" href="#Map-Re_002dInterpolation">6.17 Map Re-Interpolation</a></li>
    <li><a name="toc-Masks-1" href="#masks">6.18 Masks</a>
    <ul class="no-bullet">
      <li><a name="toc-Example" href="#Example">6.18.1 Example</a></li>
    </ul></li>
    <li><a name="toc-Trimming" href="#Trimming-Atoms">6.19 Trimming</a></li>
    <li><a name="toc-Map-Transformation-1" href="#Map-Transformation">6.20 Map Transformation</a></li>
    <li><a name="toc-Export-Map-1" href="#Export-Map">6.21 Export Map</a></li>
  </ul></li>
  <li><a name="toc-Validation-1" href="#Validation">7 Validation</a>
  <ul class="no-bullet">
    <li><a name="toc-Ramachandran-Plots-1" href="#Ramachandran-Plots">7.1 Ramachandran Plots</a></li>
    <li><a name="toc-Geometry-Analysis-1" href="#Geometry-Analysis">7.2 Geometry Analysis</a></li>
    <li><a name="toc-Chiral-Volumes-1" href="#Chiral-Volumes">7.3 Chiral Volumes</a>
    <ul class="no-bullet">
      <li><a name="toc-Fixing-Chiral-Volume-Errors" href="#Fixing-Chiral-Volume-Errors">7.3.1 Fixing Chiral Volume Errors</a></li>
    </ul></li>
    <li><a name="toc-Blobs_003a-a_002ek_002ea_002e-Unmodelled-density" href="#sec_005fblobs">7.4 Blobs: a.k.a. Unmodelled density</a></li>
    <li><a name="toc-Difference-Map-Peaks-1" href="#Difference-Map-Peaks">7.5 Difference Map Peaks</a></li>
    <li><a name="toc-Check-Waters-by-Difference-Map-1" href="#Check-Waters-by-Difference-Map">7.6 Check Waters by Difference Map</a></li>
    <li><a name="toc-Molprobity-Tools-Interface-1" href="#Molprobity-Tools-Interface">7.7 Molprobity Tools Interface</a></li>
    <li><a name="toc-GLN-and-ASN-B_002dfactor-Outliers-1" href="#GLN-and-ASN-B_002dfactor-Outliers">7.8 GLN and ASN B-factor Outliers</a></li>
    <li><a name="toc-Validation-Graphs-1" href="#Validation-Graphs">7.9 Validation Graphs</a>
    <ul class="no-bullet">
      <li><a name="toc-Residue-Density-Fit" href="#Residue-Density-Fit">7.9.1 Residue Density Fit</a></li>
      <li><a name="toc-Rotamer-Analysis" href="#Rotamer-Analysis">7.9.2 Rotamer Analysis</a></li>
      <li><a name="toc-Temperature-Factor-Variance" href="#Temperature-Factor-Variance">7.9.3 Temperature Factor Variance</a></li>
      <li><a name="toc-Peptide-Omega-Angle-Distortion" href="#Peptide-Omega-Angle-Distortion">7.9.4 Peptide Omega Angle Distortion</a></li>
    </ul></li>
  </ul></li>
  <li><a name="toc-Representation-1" href="#Representation">8 Representation</a>
  <ul class="no-bullet">
    <li><a name="toc-Surfaces-1" href="#Surfaces">8.1 Surfaces</a></li>
  </ul></li>
  <li><a name="toc-Hints-and-Usage-Tips-1" href="#Hints-and-Usage-Tips">9 Hints and Usage Tips</a>
  <ul class="no-bullet">
    <li><a name="toc-Documentation-1" href="#Documentation">9.1 Documentation</a></li>
    <li><a name="toc-Low-Resolution-1" href="#Low-Resolution">9.2 Low Resolution</a></li>
    <li><a name="toc-Coot-Droppings-1" href="#Coot-Droppings">9.3 Coot Droppings</a></li>
    <li><a name="toc-Clearing-Backups-1" href="#Clearing-Backups">9.4 Clearing Backups</a></li>
    <li><a name="toc-Getting-out-of-_0060_0060Translate_0027_0027-Mode-1" href="#Getting-out-of-_0060_0060Translate_0027_0027-Mode">9.5 Getting out of &ldquo;Translate&rdquo; Mode</a></li>
    <li><a name="toc-Getting-out-of-_0060_0060Continuous-Rotation_0027_0027-Mode-1" href="#Getting-out-of-_0060_0060Continuous-Rotation_0027_0027-Mode">9.6 Getting out of &ldquo;Continuous Rotation&rdquo; Mode</a></li>
    <li><a name="toc-Getting-out-of-_0060_0060Label-Atom-Only_0027_0027-Mode" href="#Label-Atom-Only-Mode">9.7 Getting out of &ldquo;Label Atom Only&rdquo; Mode</a></li>
    <li><a name="toc-Button-Labels-1" href="#Button-Labels">9.8 Button Labels</a></li>
    <li><a name="toc-Picking" href="#sec_005fpicking">9.9 Picking</a></li>
    <li><a name="toc-Resizing-View-1" href="#Resizing-View">9.10 Resizing View</a></li>
    <li><a name="toc-Scroll_002dwheel-1" href="#Scroll_002dwheel">9.11 Scroll-wheel</a></li>
    <li><a name="toc-Slow-Computer-Configuration-1" href="#Slow-Computer-Configuration">9.12 Slow Computer Configuration</a></li>
  </ul></li>
  <li><a name="toc-Other-Programs-1" href="#Other-Programs">10 Other Programs</a>
  <ul class="no-bullet">
    <li><a name="toc-findligand-1" href="#findligand">10.1 findligand</a></li>
  </ul></li>
  <li><a name="toc-Scripting-Functions-1" href="#Scripting-Functions">11 Scripting Functions</a>
  <ul class="no-bullet">
    <li><a name="toc-Startup-Functions-1" href="#Startup-Functions">11.1 Startup Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dprefer_002dpython-1" href="#set_002dprefer_002dpython">11.1.1 set-prefer-python</a></li>
      <li><a name="toc-prefer_002dpython-1" href="#prefer_002dpython">11.1.2 prefer-python</a></li>
    </ul></li>
    <li><a name="toc-File-System-Functions-1" href="#File-System-Functions">11.2 File System Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-make_002ddirectory_002dmaybe-1" href="#make_002ddirectory_002dmaybe">11.2.1 make-directory-maybe</a></li>
      <li><a name="toc-set_002dshow_002dpaths_002din_002ddisplay_002dmanager-1" href="#set_002dshow_002dpaths_002din_002ddisplay_002dmanager">11.2.2 set-show-paths-in-display-manager</a></li>
      <li><a name="toc-show_002dpaths_002din_002ddisplay_002dmanager_002dstate-1" href="#show_002dpaths_002din_002ddisplay_002dmanager_002dstate">11.2.3 show-paths-in-display-manager-state</a></li>
      <li><a name="toc-add_002dcoordinates_002dglob_002dextension-1" href="#add_002dcoordinates_002dglob_002dextension">11.2.4 add-coordinates-glob-extension</a></li>
      <li><a name="toc-add_002ddata_002dglob_002dextension-1" href="#add_002ddata_002dglob_002dextension">11.2.5 add-data-glob-extension</a></li>
      <li><a name="toc-add_002ddictionary_002dglob_002dextension-1" href="#add_002ddictionary_002dglob_002dextension">11.2.6 add-dictionary-glob-extension</a></li>
      <li><a name="toc-add_002dmap_002dglob_002dextension-1" href="#add_002dmap_002dglob_002dextension">11.2.7 add-map-glob-extension</a></li>
      <li><a name="toc-remove_002dcoordinates_002dglob_002dextension-1" href="#remove_002dcoordinates_002dglob_002dextension">11.2.8 remove-coordinates-glob-extension</a></li>
      <li><a name="toc-remove_002ddata_002dglob_002dextension-1" href="#remove_002ddata_002dglob_002dextension">11.2.9 remove-data-glob-extension</a></li>
      <li><a name="toc-remove_002ddictionary_002dglob_002dextension-1" href="#remove_002ddictionary_002dglob_002dextension">11.2.10 remove-dictionary-glob-extension</a></li>
      <li><a name="toc-remove_002dmap_002dglob_002dextension-1" href="#remove_002dmap_002dglob_002dextension">11.2.11 remove-map-glob-extension</a></li>
      <li><a name="toc-set_002dsticky_002dsort_002dby_002ddate-1" href="#set_002dsticky_002dsort_002dby_002ddate">11.2.12 set-sticky-sort-by-date</a></li>
      <li><a name="toc-unset_002dsticky_002dsort_002dby_002ddate-1" href="#unset_002dsticky_002dsort_002dby_002ddate">11.2.13 unset-sticky-sort-by-date</a></li>
      <li><a name="toc-set_002dfilter_002dfileselection_002dfilenames-1" href="#set_002dfilter_002dfileselection_002dfilenames">11.2.14 set-filter-fileselection-filenames</a></li>
      <li><a name="toc-filter_002dfileselection_002dfilenames_002dstate-1" href="#filter_002dfileselection_002dfilenames_002dstate">11.2.15 filter-fileselection-filenames-state</a></li>
      <li><a name="toc-file_002dtype_002dcoords-1" href="#file_002dtype_002dcoords">11.2.16 file-type-coords</a></li>
      <li><a name="toc-open_002dcoords_002ddialog-1" href="#open_002dcoords_002ddialog">11.2.17 open-coords-dialog</a></li>
      <li><a name="toc-set_002dfile_002dchooser_002dselector-1" href="#set_002dfile_002dchooser_002dselector">11.2.18 set-file-chooser-selector</a></li>
    </ul></li>
    <li><a name="toc-Widget-Utilities-1" href="#Widget-Utilities">11.3 Widget Utilities</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dmain_002dwindow_002dtitle-1" href="#set_002dmain_002dwindow_002dtitle">11.3.1 set-main-window-title</a></li>
    </ul></li>
    <li><a name="toc-MTZ-and-data-handling-utilities-1" href="#MTZ-and-data-handling-utilities">11.4 MTZ and data handling utilities</a>
    <ul class="no-bullet">
      <li><a name="toc-manage_002dcolumn_002dselector-1" href="#manage_002dcolumn_002dselector">11.4.1 manage-column-selector</a></li>
    </ul></li>
    <li><a name="toc-Molecule-Info-Functions-1" href="#Molecule-Info-Functions">11.5 Molecule Info Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-chain_002dn_002dresidues-1" href="#chain_002dn_002dresidues">11.5.1 chain-n-residues</a></li>
      <li><a name="toc-molecule_002dcentre_002dinternal-1" href="#molecule_002dcentre_002dinternal">11.5.2 molecule-centre-internal</a></li>
      <li><a name="toc-seqnum_002dfrom_002dserial_002dnumber-1" href="#seqnum_002dfrom_002dserial_002dnumber">11.5.3 seqnum-from-serial-number</a></li>
      <li><a name="toc-insertion_002dcode_002dfrom_002dserial_002dnumber-1" href="#insertion_002dcode_002dfrom_002dserial_002dnumber">11.5.4 insertion-code-from-serial-number</a></li>
      <li><a name="toc-chain_002did_002dscm-1" href="#chain_002did_002dscm">11.5.5 chain-id-scm</a></li>
      <li><a name="toc-n_002dmodels-1" href="#n_002dmodels">11.5.6 n-models</a></li>
      <li><a name="toc-n_002dchains-1" href="#n_002dchains">11.5.7 n-chains</a></li>
      <li><a name="toc-is_002dsolvent_002dchain_002dp-1" href="#is_002dsolvent_002dchain_002dp">11.5.8 is-solvent-chain-p</a></li>
      <li><a name="toc-is_002dprotein_002dchain_002dp-1" href="#is_002dprotein_002dchain_002dp">11.5.9 is-protein-chain-p</a></li>
      <li><a name="toc-is_002dnucleotide_002dchain_002dp-1" href="#is_002dnucleotide_002dchain_002dp">11.5.10 is-nucleotide-chain-p</a></li>
      <li><a name="toc-n_002dresidues-1" href="#n_002dresidues">11.5.11 n-residues</a></li>
      <li><a name="toc-n_002datoms-1" href="#n_002datoms">11.5.12 n-atoms</a></li>
      <li><a name="toc-remarks_002dscm-1" href="#remarks_002dscm">11.5.13 remarks-scm</a></li>
      <li><a name="toc-sort_002dchains-1" href="#sort_002dchains">11.5.14 sort-chains</a></li>
      <li><a name="toc-sort_002dresidues-1" href="#sort_002dresidues">11.5.15 sort-residues</a></li>
      <li><a name="toc-remarks_002ddialog-1" href="#remarks_002ddialog">11.5.16 remarks-dialog</a></li>
      <li><a name="toc-print_002dheader_002dsecondary_002dstructure_002dinfo-1" href="#print_002dheader_002dsecondary_002dstructure_002dinfo">11.5.17 print-header-secondary-structure-info</a></li>
      <li><a name="toc-add_002dheader_002dsecondary_002dstructure_002dinfo-1" href="#add_002dheader_002dsecondary_002dstructure_002dinfo">11.5.18 add-header-secondary-structure-info</a></li>
      <li><a name="toc-copy_002dmolecule-1" href="#copy_002dmolecule">11.5.19 copy-molecule</a></li>
      <li><a name="toc-add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule-1" href="#add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule">11.5.20 add-ligand-delete-residue-copy-molecule</a></li>
      <li><a name="toc-exchange_002dchain_002dids_002dfor_002dseg_002dids-1" href="#exchange_002dchain_002dids_002dfor_002dseg_002dids">11.5.21 exchange-chain-ids-for-seg-ids</a></li>
      <li><a name="toc-show_002dremarks_002dbrowswer-1" href="#show_002dremarks_002dbrowswer">11.5.22 show-remarks-browswer</a></li>
    </ul></li>
    <li><a name="toc-Library-and-Utility-Functions-1" href="#Library-and-Utility-Functions">11.6 Library and Utility Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-git_002drevision_002dcount-1" href="#git_002drevision_002dcount">11.6.1 git-revision-count</a></li>
      <li><a name="toc-svn_002drevision-1" href="#svn_002drevision">11.6.2 svn-revision</a></li>
      <li><a name="toc-molecule_002dname-1" href="#molecule_002dname">11.6.3 molecule-name</a></li>
      <li><a name="toc-molecule_002dname_002dstub_002dscm-1" href="#molecule_002dname_002dstub_002dscm">11.6.4 molecule-name-stub-scm</a></li>
      <li><a name="toc-molecule_002dname_002dstub_002dpy-1" href="#molecule_002dname_002dstub_002dpy">11.6.5 molecule-name-stub-py</a></li>
      <li><a name="toc-set_002dmolecule_002dname-1" href="#set_002dmolecule_002dname">11.6.6 set-molecule-name</a></li>
      <li><a name="toc-coot_002dreal_002dexit-1" href="#coot_002dreal_002dexit">11.6.7 coot-real-exit</a></li>
      <li><a name="toc-coot_002dno_002dstate_002dreal_002dexit-1" href="#coot_002dno_002dstate_002dreal_002dexit">11.6.8 coot-no-state-real-exit</a></li>
      <li><a name="toc-coot_002dclear_002dbackup_002dor_002dreal_002dexit-1" href="#coot_002dclear_002dbackup_002dor_002dreal_002dexit">11.6.9 coot-clear-backup-or-real-exit</a></li>
      <li><a name="toc-coot_002dsave_002dstate_002dand_002dexit-1" href="#coot_002dsave_002dstate_002dand_002dexit">11.6.10 coot-save-state-and-exit</a></li>
      <li><a name="toc-run_002dclear_002dbackups-1" href="#run_002dclear_002dbackups">11.6.11 run-clear-backups</a></li>
      <li><a name="toc-first_002dcoords_002dimol-1" href="#first_002dcoords_002dimol">11.6.12 first-coords-imol</a></li>
      <li><a name="toc-first_002dsmall_002dcoords_002dimol-1" href="#first_002dsmall_002dcoords_002dimol">11.6.13 first-small-coords-imol</a></li>
      <li><a name="toc-first_002dunsaved_002dcoords_002dimol-1" href="#first_002dunsaved_002dcoords_002dimol">11.6.14 first-unsaved-coords-imol</a></li>
      <li><a name="toc-mmcif_002dsfs_002dto_002dmtz-1" href="#mmcif_002dsfs_002dto_002dmtz">11.6.15 mmcif-sfs-to-mtz</a></li>
    </ul></li>
    <li><a name="toc-Graphics-Utility-Functions-1" href="#Graphics-Utility-Functions">11.7 Graphics Utility Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002ddo_002danti_002daliasing-1" href="#set_002ddo_002danti_002daliasing">11.7.1 set-do-anti-aliasing</a></li>
      <li><a name="toc-do_002danti_002daliasing_002dstate-1" href="#do_002danti_002daliasing_002dstate">11.7.2 do-anti-aliasing-state</a></li>
      <li><a name="toc-set_002ddo_002dGL_002dlighting-1" href="#set_002ddo_002dGL_002dlighting">11.7.3 set-do-GL-lighting</a></li>
      <li><a name="toc-do_002dGL_002dlighting_002dstate-1" href="#do_002dGL_002dlighting_002dstate">11.7.4 do-GL-lighting-state</a></li>
      <li><a name="toc-use_002dgraphics_002dinterface_002dstate-1" href="#use_002dgraphics_002dinterface_002dstate">11.7.5 use-graphics-interface-state</a></li>
      <li><a name="toc-python_002dat_002dprompt_002dat_002dstartup_002dstate-1" href="#python_002dat_002dprompt_002dat_002dstartup_002dstate">11.7.6 python-at-prompt-at-startup-state</a></li>
      <li><a name="toc-start_002dgraphics_002dinterface-1" href="#start_002dgraphics_002dinterface">11.7.7 start-graphics-interface</a></li>
      <li><a name="toc-reset_002dview-1" href="#reset_002dview">11.7.8 reset-view</a></li>
      <li><a name="toc-graphics_002dn_002dmolecules-1" href="#graphics_002dn_002dmolecules">11.7.9 graphics-n-molecules</a></li>
      <li><a name="toc-toggle_002didle_002dspin_002dfunction-1" href="#toggle_002didle_002dspin_002dfunction">11.7.10 toggle-idle-spin-function</a></li>
      <li><a name="toc-toggle_002didle_002drock_002dfunction-1" href="#toggle_002didle_002drock_002dfunction">11.7.11 toggle-idle-rock-function</a></li>
      <li><a name="toc-set_002drocking_002dfactors-1" href="#set_002drocking_002dfactors">11.7.12 set-rocking-factors</a></li>
      <li><a name="toc-set_002didle_002dfunction_002drotate_002dangle-1" href="#set_002didle_002dfunction_002drotate_002dangle">11.7.13 set-idle-function-rotate-angle</a></li>
      <li><a name="toc-idle_002dfunction_002drotate_002dangle-1" href="#idle_002dfunction_002drotate_002dangle">11.7.14 idle-function-rotate-angle</a></li>
      <li><a name="toc-handle_002dread_002ddraw_002dmolecule-1" href="#handle_002dread_002ddraw_002dmolecule">11.7.15 handle-read-draw-molecule</a></li>
      <li><a name="toc-make_002dupdating_002dmodel_002dmolecule-1" href="#make_002dupdating_002dmodel_002dmolecule">11.7.16 make-updating-model-molecule</a></li>
      <li><a name="toc-allow_002dduplicate_002dsequence_002dnumbers-1" href="#allow_002dduplicate_002dsequence_002dnumbers">11.7.17 allow-duplicate-sequence-numbers</a></li>
      <li><a name="toc-set_002dconvert_002dto_002dv2_002datom_002dnames-1" href="#set_002dconvert_002dto_002dv2_002datom_002dnames">11.7.18 set-convert-to-v2-atom-names</a></li>
      <li><a name="toc-handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre-1" href="#handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre">11.7.19 handle-read-draw-molecule-with-recentre</a></li>
      <li><a name="toc-handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere-1" href="#handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere">11.7.20 handle-read-draw-molecule-and-move-molecule-here</a></li>
      <li><a name="toc-read_002dpdb-1" href="#read_002dpdb">11.7.21 read-pdb</a></li>
      <li><a name="toc-assign_002dhetatms-1" href="#assign_002dhetatms">11.7.22 assign-hetatms</a></li>
      <li><a name="toc-hetify_002dresidue-1" href="#hetify_002dresidue">11.7.23 hetify-residue</a></li>
      <li><a name="toc-residue_002dhas_002dhetatms-1" href="#residue_002dhas_002dhetatms">11.7.24 residue-has-hetatms</a></li>
      <li><a name="toc-het_002dgroup_002dresidues_002dscm-1" href="#het_002dgroup_002dresidues_002dscm">11.7.25 het-group-residues-scm</a></li>
      <li><a name="toc-het_002dgroup_002dresidues_002dpy-1" href="#het_002dgroup_002dresidues_002dpy">11.7.26 het-group-residues-py</a></li>
      <li><a name="toc-het_002dgroup_002dn_002datoms-1" href="#het_002dgroup_002dn_002datoms">11.7.27 het-group-n-atoms</a></li>
      <li><a name="toc-replace_002dfragment-1" href="#replace_002dfragment">11.7.28 replace-fragment</a></li>
      <li><a name="toc-copy_002dresidue_002drange-1" href="#copy_002dresidue_002drange">11.7.29 copy-residue-range</a></li>
      <li><a name="toc-replace_002dresidues_002dfrom_002dmol_002dscm-1" href="#replace_002dresidues_002dfrom_002dmol_002dscm">11.7.30 replace-residues-from-mol-scm</a></li>
      <li><a name="toc-clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile-1" href="#clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile">11.7.31 clear-and-update-model-molecule-from-file</a></li>
      <li><a name="toc-screendump_002dimage-1" href="#screendump_002dimage">11.7.32 screendump-image</a></li>
      <li><a name="toc-check_002dfor_002ddark_002dblue_002ddensity-1" href="#check_002dfor_002ddark_002dblue_002ddensity">11.7.33 check-for-dark-blue-density</a></li>
      <li><a name="toc-set_002ddraw_002dsolid_002ddensity_002dsurface-1" href="#set_002ddraw_002dsolid_002ddensity_002dsurface">11.7.34 set-draw-solid-density-surface</a></li>
      <li><a name="toc-set_002ddraw_002dmap_002dstandard_002dlines-1" href="#set_002ddraw_002dmap_002dstandard_002dlines">11.7.35 set-draw-map-standard-lines</a></li>
      <li><a name="toc-set_002dsolid_002ddensity_002dsurface_002dopacity-1" href="#set_002dsolid_002ddensity_002dsurface_002dopacity">11.7.36 set-solid-density-surface-opacity</a></li>
      <li><a name="toc-set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface-1" href="#set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface">11.7.37 set-flat-shading-for-solid-density-surface</a></li>
    </ul></li>
    <li><a name="toc-Interface-Preferences-1" href="#Interface-Preferences">11.8 Interface Preferences</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dscroll_002dby_002dwheel_002dmouse-1" href="#set_002dscroll_002dby_002dwheel_002dmouse">11.8.1 set-scroll-by-wheel-mouse</a></li>
      <li><a name="toc-scroll_002dby_002dwheel_002dmouse_002dstate-1" href="#scroll_002dby_002dwheel_002dmouse_002dstate">11.8.2 scroll-by-wheel-mouse-state</a></li>
      <li><a name="toc-set_002dauto_002drecontour_002dmap-1" href="#set_002dauto_002drecontour_002dmap">11.8.3 set-auto-recontour-map</a></li>
      <li><a name="toc-get_002dauto_002drecontour_002dmap-1" href="#get_002dauto_002drecontour_002dmap">11.8.4 get-auto-recontour-map</a></li>
      <li><a name="toc-set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap-1" href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap">11.8.5 set-default-initial-contour-level-for-map</a></li>
      <li><a name="toc-set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap-1" href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap">11.8.6 set-default-initial-contour-level-for-difference-map</a></li>
      <li><a name="toc-print_002dview_002dmatrix-1" href="#print_002dview_002dmatrix">11.8.7 print-view-matrix</a></li>
      <li><a name="toc-get_002dview_002dquaternion_002dinternal-1" href="#get_002dview_002dquaternion_002dinternal">11.8.8 get-view-quaternion-internal</a></li>
      <li><a name="toc-set_002dview_002dquaternion-1" href="#set_002dview_002dquaternion">11.8.9 set-view-quaternion</a></li>
      <li><a name="toc-apply_002dncs_002dto_002dview_002dorientation-1" href="#apply_002dncs_002dto_002dview_002dorientation">11.8.10 apply-ncs-to-view-orientation</a></li>
      <li><a name="toc-apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre-1" href="#apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre">11.8.11 apply-ncs-to-view-orientation-and-screen-centre</a></li>
      <li><a name="toc-set_002dfps_002dflag-1" href="#set_002dfps_002dflag">11.8.12 set-fps-flag</a></li>
      <li><a name="toc-get_002dfps_002dflag-1" href="#get_002dfps_002dflag">11.8.13 get-fps-flag</a></li>
      <li><a name="toc-set_002dshow_002dorigin_002dmarker-1" href="#set_002dshow_002dorigin_002dmarker">11.8.14 set-show-origin-marker</a></li>
      <li><a name="toc-show_002dorigin_002dmarker_002dstate-1" href="#show_002dorigin_002dmarker_002dstate">11.8.15 show-origin-marker-state</a></li>
      <li><a name="toc-hide_002dmodelling_002dtoolbar-1" href="#hide_002dmodelling_002dtoolbar">11.8.16 hide-modelling-toolbar</a></li>
      <li><a name="toc-show_002dmodelling_002dtoolbar-1" href="#show_002dmodelling_002dtoolbar">11.8.17 show-modelling-toolbar</a></li>
      <li><a name="toc-hide_002dmain_002dtoolbar-1" href="#hide_002dmain_002dtoolbar">11.8.18 hide-main-toolbar</a></li>
      <li><a name="toc-show_002dmain_002dtoolbar-1" href="#show_002dmain_002dtoolbar">11.8.19 show-main-toolbar</a></li>
      <li><a name="toc-show_002dmodel_002dtoolbar_002dall_002dicons-1" href="#show_002dmodel_002dtoolbar_002dall_002dicons">11.8.20 show-model-toolbar-all-icons</a></li>
      <li><a name="toc-show_002dmodel_002dtoolbar_002dmain_002dicons-1" href="#show_002dmodel_002dtoolbar_002dmain_002dicons">11.8.21 show-model-toolbar-main-icons</a></li>
      <li><a name="toc-reattach_002dmodelling_002dtoolbar-1" href="#reattach_002dmodelling_002dtoolbar">11.8.22 reattach-modelling-toolbar</a></li>
      <li><a name="toc-set_002dmodel_002dtoolbar_002ddocked_002dposition-1" href="#set_002dmodel_002dtoolbar_002ddocked_002dposition">11.8.23 set-model-toolbar-docked-position</a></li>
      <li><a name="toc-suck_002dmodel_002dfit_002ddialog-1" href="#suck_002dmodel_002dfit_002ddialog">11.8.24 suck-model-fit-dialog</a></li>
      <li><a name="toc-add_002dstatus_002dbar_002dtext-1" href="#add_002dstatus_002dbar_002dtext">11.8.25 add-status-bar-text</a></li>
      <li><a name="toc-set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop-1" href="#set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop">11.8.26 set-model-fit-refine-dialog-stays-on-top</a></li>
      <li><a name="toc-model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate-1" href="#model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate">11.8.27 model-fit-refine-dialog-stays-on-top-state</a></li>
      <li><a name="toc-accept_002dreject_002ddialog_002ddocked_002dstate-1" href="#accept_002dreject_002ddialog_002ddocked_002dstate">11.8.28 accept-reject-dialog-docked-state</a></li>
      <li><a name="toc-set_002daccept_002dreject_002ddialog_002ddocked_002dshow-1" href="#set_002daccept_002dreject_002ddialog_002ddocked_002dshow">11.8.29 set-accept-reject-dialog-docked-show</a></li>
      <li><a name="toc-accept_002dreject_002ddialog_002ddocked_002dshow_002dstate-1" href="#accept_002dreject_002ddialog_002ddocked_002dshow_002dstate">11.8.30 accept-reject-dialog-docked-show-state</a></li>
    </ul></li>
    <li><a name="toc-Mouse-Buttons-1" href="#Mouse-Buttons">11.9 Mouse Buttons</a>
    <ul class="no-bullet">
      <li><a name="toc-quanta_002dbuttons-1" href="#quanta_002dbuttons">11.9.1 quanta-buttons</a></li>
      <li><a name="toc-quanta_002dlike_002dzoom-1" href="#quanta_002dlike_002dzoom">11.9.2 quanta-like-zoom</a></li>
      <li><a name="toc-set_002dcontrol_002dkey_002dfor_002drotate-1" href="#set_002dcontrol_002dkey_002dfor_002drotate">11.9.3 set-control-key-for-rotate</a></li>
      <li><a name="toc-control_002dkey_002dfor_002drotate_002dstate-1" href="#control_002dkey_002dfor_002drotate_002dstate">11.9.4 control-key-for-rotate-state</a></li>
      <li><a name="toc-blob_002dunder_002dpointer_002dto_002dscreen_002dcentre-1" href="#blob_002dunder_002dpointer_002dto_002dscreen_002dcentre">11.9.5 blob-under-pointer-to-screen-centre</a></li>
      <li><a name="toc-select_002datom_002dunder_002dpointer_002dscm-1" href="#select_002datom_002dunder_002dpointer_002dscm">11.9.6 select-atom-under-pointer-scm</a></li>
      <li><a name="toc-select_002datom_002dunder_002dpointer_002dpy-1" href="#select_002datom_002dunder_002dpointer_002dpy">11.9.7 select-atom-under-pointer-py</a></li>
    </ul></li>
    <li><a name="toc-Cursor-Function-1" href="#Cursor-Function">11.10 Cursor Function</a>
    <ul class="no-bullet">
      <li><a name="toc-normal_002dcursor-1" href="#normal_002dcursor">11.10.1 normal-cursor</a></li>
      <li><a name="toc-fleur_002dcursor-1" href="#fleur_002dcursor">11.10.2 fleur-cursor</a></li>
      <li><a name="toc-pick_002dcursor_002dmaybe-1" href="#pick_002dcursor_002dmaybe">11.10.3 pick-cursor-maybe</a></li>
      <li><a name="toc-rotate_002dcursor-1" href="#rotate_002dcursor">11.10.4 rotate-cursor</a></li>
      <li><a name="toc-set_002dpick_002dcursor_002dindex-1" href="#set_002dpick_002dcursor_002dindex">11.10.5 set-pick-cursor-index</a></li>
    </ul></li>
    <li><a name="toc-Model_002fFit_002fRefine-Functions-1" href="#Model_002fFit_002fRefine-Functions">11.11 Model/Fit/Refine Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-post_002dmodel_002dfit_002drefine_002ddialog-1" href="#post_002dmodel_002dfit_002drefine_002ddialog">11.11.1 post-model-fit-refine-dialog</a></li>
      <li><a name="toc-unset_002dmodel_002dfit_002drefine_002ddialog-1" href="#unset_002dmodel_002dfit_002drefine_002ddialog">11.11.2 unset-model-fit-refine-dialog</a></li>
      <li><a name="toc-unset_002drefine_002dparams_002ddialog-1" href="#unset_002drefine_002dparams_002ddialog">11.11.3 unset-refine-params-dialog</a></li>
      <li><a name="toc-show_002dselect_002dmap_002ddialog-1" href="#show_002dselect_002dmap_002ddialog">11.11.4 show-select-map-dialog</a></li>
      <li><a name="toc-set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel-1" href="#set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel">11.11.5 set-model-fit-refine-rotate-translate-zone-label</a></li>
      <li><a name="toc-set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel-1" href="#set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel">11.11.6 set-model-fit-refine-place-atom-at-pointer-label</a></li>
      <li><a name="toc-post_002dother_002dmodelling_002dtools_002ddialog-1" href="#post_002dother_002dmodelling_002dtools_002ddialog">11.11.7 post-other-modelling-tools-dialog</a></li>
      <li><a name="toc-set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy-1" href="#set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy">11.11.8 set-refinement-move-atoms-with-zero-occupancy</a></li>
      <li><a name="toc-refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate-1" href="#refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate">11.11.9 refinement-move-atoms-with-zero-occupancy-state</a></li>
    </ul></li>
    <li><a name="toc-Backup-Functions-1" href="#Backup-Functions">11.12 Backup Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-make_002dbackup-1" href="#make_002dbackup">11.12.1 make-backup</a></li>
      <li><a name="toc-turn_002doff_002dbackup-1" href="#turn_002doff_002dbackup">11.12.2 turn-off-backup</a></li>
      <li><a name="toc-turn_002don_002dbackup-1" href="#turn_002don_002dbackup">11.12.3 turn-on-backup</a></li>
      <li><a name="toc-backup_002dstate-1" href="#backup_002dstate">11.12.4 backup-state</a></li>
      <li><a name="toc-apply_002dundo-1" href="#apply_002dundo">11.12.5 apply-undo</a></li>
      <li><a name="toc-apply_002dredo-1" href="#apply_002dredo">11.12.6 apply-redo</a></li>
      <li><a name="toc-set_002dhave_002dunsaved_002dchanges-1" href="#set_002dhave_002dunsaved_002dchanges">11.12.7 set-have-unsaved-changes</a></li>
      <li><a name="toc-have_002dunsaved_002dchanges_002dp-1" href="#have_002dunsaved_002dchanges_002dp">11.12.8 have-unsaved-changes-p</a></li>
      <li><a name="toc-set_002dundo_002dmolecule-1" href="#set_002dundo_002dmolecule">11.12.9 set-undo-molecule</a></li>
      <li><a name="toc-show_002dset_002dundo_002dmolecule_002dchooser-1" href="#show_002dset_002dundo_002dmolecule_002dchooser">11.12.10 show-set-undo-molecule-chooser</a></li>
      <li><a name="toc-set_002dunpathed_002dbackup_002dfile_002dnames-1" href="#set_002dunpathed_002dbackup_002dfile_002dnames">11.12.11 set-unpathed-backup-file-names</a></li>
      <li><a name="toc-unpathed_002dbackup_002dfile_002dnames_002dstate-1" href="#unpathed_002dbackup_002dfile_002dnames_002dstate">11.12.12 unpathed-backup-file-names-state</a></li>
      <li><a name="toc-set_002ddecoloned_002dbackup_002dfile_002dnames-1" href="#set_002ddecoloned_002dbackup_002dfile_002dnames">11.12.13 set-decoloned-backup-file-names</a></li>
      <li><a name="toc-decoloned_002dbackup_002dfile_002dnames_002dstate-1" href="#decoloned_002dbackup_002dfile_002dnames_002dstate">11.12.14 decoloned-backup-file-names-state</a></li>
      <li><a name="toc-backup_002dcompress_002dfiles_002dstate-1" href="#backup_002dcompress_002dfiles_002dstate">11.12.15 backup-compress-files-state</a></li>
      <li><a name="toc-set_002dbackup_002dcompress_002dfiles-1" href="#set_002dbackup_002dcompress_002dfiles">11.12.16 set-backup-compress-files</a></li>
    </ul></li>
    <li><a name="toc-Recover-Session-Function-1" href="#Recover-Session-Function">11.13 Recover Session Function</a>
    <ul class="no-bullet">
      <li><a name="toc-recover_002dsession-1" href="#recover_002dsession">11.13.1 recover-session</a></li>
    </ul></li>
    <li><a name="toc-Map-Functions-1" href="#Map-Functions">11.14 Map Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-calc_002dphases_002dgeneric-1" href="#calc_002dphases_002dgeneric">11.14.1 calc-phases-generic</a></li>
      <li><a name="toc-map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases-1" href="#map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases">11.14.2 map-from-mtz-by-refmac-calc-phases</a></li>
      <li><a name="toc-map_002dfrom_002dmtz_002dby_002dcalc_002dphases-1" href="#map_002dfrom_002dmtz_002dby_002dcalc_002dphases">11.14.3 map-from-mtz-by-calc-phases</a></li>
      <li><a name="toc-sfcalc_002dgenmap-1" href="#sfcalc_002dgenmap">11.14.4 sfcalc-genmap</a></li>
      <li><a name="toc-set_002dauto_002dupdating_002dsfcalc_002dgenmap-1" href="#set_002dauto_002dupdating_002dsfcalc_002dgenmap">11.14.5 set-auto-updating-sfcalc-genmap</a></li>
      <li><a name="toc-set_002dscroll_002dwheel_002dmap-1" href="#set_002dscroll_002dwheel_002dmap">11.14.6 set-scroll-wheel-map</a></li>
      <li><a name="toc-set_002dscrollable_002dmap-1" href="#set_002dscrollable_002dmap">11.14.7 set-scrollable-map</a></li>
      <li><a name="toc-scroll_002dwheel_002dmap-1" href="#scroll_002dwheel_002dmap">11.14.8 scroll-wheel-map</a></li>
      <li><a name="toc-save_002dprevious_002dmap_002dcolour-1" href="#save_002dprevious_002dmap_002dcolour">11.14.9 save-previous-map-colour</a></li>
      <li><a name="toc-restore_002dprevious_002dmap_002dcolour-1" href="#restore_002dprevious_002dmap_002dcolour">11.14.10 restore-previous-map-colour</a></li>
      <li><a name="toc-set_002dactive_002dmap_002ddrag_002dflag-1" href="#set_002dactive_002dmap_002ddrag_002dflag">11.14.11 set-active-map-drag-flag</a></li>
      <li><a name="toc-get_002dactive_002dmap_002ddrag_002dflag-1" href="#get_002dactive_002dmap_002ddrag_002dflag">11.14.12 get-active-map-drag-flag</a></li>
      <li><a name="toc-set_002dlast_002dmap_002dcolour-1" href="#set_002dlast_002dmap_002dcolour">11.14.13 set-last-map-colour</a></li>
      <li><a name="toc-set_002dmap_002dcolour-1" href="#set_002dmap_002dcolour">11.14.14 set-map-colour</a></li>
      <li><a name="toc-set_002dcontour_002dlevel_002dabsolute-1" href="#set_002dcontour_002dlevel_002dabsolute">11.14.15 set-contour-level-absolute</a></li>
      <li><a name="toc-set_002dcontour_002dlevel_002din_002dsigma-1" href="#set_002dcontour_002dlevel_002din_002dsigma">11.14.16 set-contour-level-in-sigma</a></li>
      <li><a name="toc-get_002dcontour_002dlevel_002dabsolute-1" href="#get_002dcontour_002dlevel_002dabsolute">11.14.17 get-contour-level-absolute</a></li>
      <li><a name="toc-get_002dcontour_002dlevel_002din_002dsigma-1" href="#get_002dcontour_002dlevel_002din_002dsigma">11.14.18 get-contour-level-in-sigma</a></li>
      <li><a name="toc-set_002dlast_002dmap_002dsigma_002dstep-1" href="#set_002dlast_002dmap_002dsigma_002dstep">11.14.19 set-last-map-sigma-step</a></li>
      <li><a name="toc-set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol-1" href="#set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol">11.14.20 set-contour-by-sigma-step-by-mol</a></li>
      <li><a name="toc-data_002dresolution-1" href="#data_002dresolution">11.14.21 data-resolution</a></li>
      <li><a name="toc-model_002dresolution-1" href="#model_002dresolution">11.14.22 model-resolution</a></li>
      <li><a name="toc-export_002dmap-1" href="#export_002dmap">11.14.23 export-map</a></li>
      <li><a name="toc-export_002dmap_002dfragment-1" href="#export_002dmap_002dfragment">11.14.24 export-map-fragment</a></li>
      <li><a name="toc-export_002dmap_002dfragment_002dwith_002dorigin_002dshift-1" href="#export_002dmap_002dfragment_002dwith_002dorigin_002dshift">11.14.25 export-map-fragment-with-origin-shift</a></li>
      <li><a name="toc-export_002dmap_002dfragment_002dto_002dplain_002dfile-1" href="#export_002dmap_002dfragment_002dto_002dplain_002dfile">11.14.26 export-map-fragment-to-plain-file</a></li>
      <li><a name="toc-difference_002dmap-1" href="#difference_002dmap">11.14.27 difference-map</a></li>
      <li><a name="toc-reinterp_002dmap-1" href="#reinterp_002dmap">11.14.28 reinterp-map</a></li>
      <li><a name="toc-smooth_002dmap-1" href="#smooth_002dmap">11.14.29 smooth-map</a></li>
      <li><a name="toc-average_002dmap_002dscm-1" href="#average_002dmap_002dscm">11.14.30 average-map-scm</a></li>
      <li><a name="toc-average_002dmap_002dpy-1" href="#average_002dmap_002dpy">11.14.31 average-map-py</a></li>
    </ul></li>
    <li><a name="toc-Density-Increment-1" href="#Density-Increment">11.15 Density Increment</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002diso_002dlevel_002dincrement-1" href="#set_002diso_002dlevel_002dincrement">11.15.1 set-iso-level-increment</a></li>
      <li><a name="toc-set_002ddiff_002dmap_002diso_002dlevel_002dincrement-1" href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement">11.15.2 set-diff-map-iso-level-increment</a></li>
      <li><a name="toc-get_002ddiff_002dmap_002diso_002dlevel_002dincrement-1" href="#get_002ddiff_002dmap_002diso_002dlevel_002dincrement">11.15.3 get-diff-map-iso-level-increment</a></li>
      <li><a name="toc-set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext-1" href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext">11.15.4 set-diff-map-iso-level-increment-from-text</a></li>
      <li><a name="toc-set_002dmap_002dsampling_002drate_002dtext-1" href="#set_002dmap_002dsampling_002drate_002dtext">11.15.5 set-map-sampling-rate-text</a></li>
      <li><a name="toc-set_002dmap_002dsampling_002drate-1" href="#set_002dmap_002dsampling_002drate">11.15.6 set-map-sampling-rate</a></li>
      <li><a name="toc-get_002dmap_002dsampling_002drate-1" href="#get_002dmap_002dsampling_002drate">11.15.7 get-map-sampling-rate</a></li>
      <li><a name="toc-change_002dcontour_002dlevel-1" href="#change_002dcontour_002dlevel">11.15.8 change-contour-level</a></li>
      <li><a name="toc-set_002dlast_002dmap_002dcontour_002dlevel-1" href="#set_002dlast_002dmap_002dcontour_002dlevel">11.15.9 set-last-map-contour-level</a></li>
      <li><a name="toc-set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma-1" href="#set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma">11.15.10 set-last-map-contour-level-by-sigma</a></li>
      <li><a name="toc-set_002dstop_002dscroll_002ddiff_002dmap-1" href="#set_002dstop_002dscroll_002ddiff_002dmap">11.15.11 set-stop-scroll-diff-map</a></li>
      <li><a name="toc-set_002dstop_002dscroll_002diso_002dmap-1" href="#set_002dstop_002dscroll_002diso_002dmap">11.15.12 set-stop-scroll-iso-map</a></li>
      <li><a name="toc-set_002dstop_002dscroll_002diso_002dmap_002dlevel-1" href="#set_002dstop_002dscroll_002diso_002dmap_002dlevel">11.15.13 set-stop-scroll-iso-map-level</a></li>
      <li><a name="toc-set_002dstop_002dscroll_002ddiff_002dmap_002dlevel-1" href="#set_002dstop_002dscroll_002ddiff_002dmap_002dlevel">11.15.14 set-stop-scroll-diff-map-level</a></li>
      <li><a name="toc-set_002dresidue_002ddensity_002dfit_002dscale_002dfactor-1" href="#set_002dresidue_002ddensity_002dfit_002dscale_002dfactor">11.15.15 set-residue-density-fit-scale-factor</a></li>
    </ul></li>
    <li><a name="toc-Density-Functions-1" href="#Density-Functions">11.16 Density Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dmap_002dline_002dwidth-1" href="#set_002dmap_002dline_002dwidth">11.16.1 set-map-line-width</a></li>
      <li><a name="toc-map_002dline_002dwidth_002dstate-1" href="#map_002dline_002dwidth_002dstate">11.16.2 map-line-width-state</a></li>
      <li><a name="toc-make_002dand_002ddraw_002dmap-1" href="#make_002dand_002ddraw_002dmap">11.16.3 make-and-draw-map</a></li>
      <li><a name="toc-make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams-1" href="#make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams">11.16.4 make-and-draw-map-with-refmac-params</a></li>
      <li><a name="toc-make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams-1" href="#make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams">11.16.5 make-and-draw-map-with-reso-with-refmac-params</a></li>
      <li><a name="toc-make_002dupdating_002dmap-1" href="#make_002dupdating_002dmap">11.16.6 make-updating-map</a></li>
      <li><a name="toc-valid_002dlabels-1" href="#valid_002dlabels">11.16.7 valid-labels</a></li>
      <li><a name="toc-mtz_002dfile_002dhas_002dphases_002dp-1" href="#mtz_002dfile_002dhas_002dphases_002dp">11.16.8 mtz-file-has-phases-p</a></li>
      <li><a name="toc-is_002dmtz_002dfile_002dp-1" href="#is_002dmtz_002dfile_002dp">11.16.9 is-mtz-file-p</a></li>
      <li><a name="toc-cns_002dfile_002dhas_002dphases_002dp-1" href="#cns_002dfile_002dhas_002dphases_002dp">11.16.10 cns-file-has-phases-p</a></li>
      <li><a name="toc-auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate-1" href="#auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate">11.16.11 auto-read-do-difference-map-too-state</a></li>
      <li><a name="toc-set_002dauto_002dread_002dcolumn_002dlabels-1" href="#set_002dauto_002dread_002dcolumn_002dlabels">11.16.12 set-auto-read-column-labels</a></li>
      <li><a name="toc-set_002dmap_002dradius-1" href="#set_002dmap_002dradius">11.16.13 set-map-radius</a></li>
      <li><a name="toc-set_002dmap_002dradius_002dem-1" href="#set_002dmap_002dradius_002dem">11.16.14 set-map-radius-em</a></li>
      <li><a name="toc-set_002ddensity_002dsize-1" href="#set_002ddensity_002dsize">11.16.15 set-density-size</a></li>
      <li><a name="toc-set_002ddisplay_002dintro_002dstring-1" href="#set_002ddisplay_002dintro_002dstring">11.16.16 set-display-intro-string</a></li>
      <li><a name="toc-get_002dmap_002dradius-1" href="#get_002dmap_002dradius">11.16.17 get-map-radius</a></li>
      <li><a name="toc-set_002desoteric_002ddepth_002dcue-1" href="#set_002desoteric_002ddepth_002dcue">11.16.18 set-esoteric-depth-cue</a></li>
      <li><a name="toc-esoteric_002ddepth_002dcue_002dstate-1" href="#esoteric_002ddepth_002dcue_002dstate">11.16.19 esoteric-depth-cue-state</a></li>
      <li><a name="toc-set_002dswap_002ddifference_002dmap_002dcolours-1" href="#set_002dswap_002ddifference_002dmap_002dcolours">11.16.20 set-swap-difference-map-colours</a></li>
      <li><a name="toc-set_002dmap_002dis_002ddifference_002dmap-1" href="#set_002dmap_002dis_002ddifference_002dmap">11.16.21 set-map-is-difference-map</a></li>
      <li><a name="toc-map_002dis_002ddifference_002dmap-1" href="#map_002dis_002ddifference_002dmap">11.16.22 map-is-difference-map</a></li>
      <li><a name="toc-another_002dlevel-1" href="#another_002dlevel">11.16.23 another-level</a></li>
      <li><a name="toc-another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber-1" href="#another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber">11.16.24 another-level-from-map-molecule-number</a></li>
      <li><a name="toc-residue_002ddensity_002dfit_002dscale_002dfactor-1" href="#residue_002ddensity_002dfit_002dscale_002dfactor">11.16.25 residue-density-fit-scale-factor</a></li>
      <li><a name="toc-density_002dat_002dpoint-1" href="#density_002dat_002dpoint">11.16.26 density-at-point</a></li>
    </ul></li>
    <li><a name="toc-Parameters-from-map-1" href="#Parameters-from-map">11.17 Parameters from map</a>
    <ul class="no-bullet">
      <li><a name="toc-mtz_002dhklin_002dfor_002dmap-1" href="#mtz_002dhklin_002dfor_002dmap">11.17.1 mtz-hklin-for-map</a></li>
      <li><a name="toc-mtz_002dfp_002dfor_002dmap-1" href="#mtz_002dfp_002dfor_002dmap">11.17.2 mtz-fp-for-map</a></li>
      <li><a name="toc-mtz_002dphi_002dfor_002dmap-1" href="#mtz_002dphi_002dfor_002dmap">11.17.3 mtz-phi-for-map</a></li>
      <li><a name="toc-mtz_002dweight_002dfor_002dmap-1" href="#mtz_002dweight_002dfor_002dmap">11.17.4 mtz-weight-for-map</a></li>
      <li><a name="toc-mtz_002duse_002dweight_002dfor_002dmap-1" href="#mtz_002duse_002dweight_002dfor_002dmap">11.17.5 mtz-use-weight-for-map</a></li>
      <li><a name="toc-map_002dparameters_002dscm-1" href="#map_002dparameters_002dscm">11.17.6 map-parameters-scm</a></li>
      <li><a name="toc-cell_002dscm-1" href="#cell_002dscm">11.17.7 cell-scm</a></li>
      <li><a name="toc-map_002dparameters_002dpy-1" href="#map_002dparameters_002dpy">11.17.8 map-parameters-py</a></li>
      <li><a name="toc-cell_002dpy-1" href="#cell_002dpy">11.17.9 cell-py</a></li>
    </ul></li>
    <li><a name="toc-PDB-Functions-1" href="#PDB-Functions">11.18 PDB Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-write_002dpdb_002dfile-1" href="#write_002dpdb_002dfile">11.18.1 write-pdb-file</a></li>
      <li><a name="toc-write_002dcif_002dfile-1" href="#write_002dcif_002dfile">11.18.2 write-cif-file</a></li>
      <li><a name="toc-write_002dresidue_002drange_002dto_002dpdb_002dfile-1" href="#write_002dresidue_002drange_002dto_002dpdb_002dfile">11.18.3 write-residue-range-to-pdb-file</a></li>
      <li><a name="toc-quick_002dsave-1" href="#quick_002dsave">11.18.4 quick-save</a></li>
    </ul></li>
    <li><a name="toc-Info-Dialog-1" href="#Info-Dialog">11.19 Info Dialog</a>
    <ul class="no-bullet">
      <li><a name="toc-info_002ddialog-1" href="#info_002ddialog">11.19.1 info-dialog</a></li>
      <li><a name="toc-info_002ddialog_002dand_002dtext-1" href="#info_002ddialog_002dand_002dtext">11.19.2 info-dialog-and-text</a></li>
      <li><a name="toc-info_002ddialog_002dwith_002dmarkup-1" href="#info_002ddialog_002dwith_002dmarkup">11.19.3 info-dialog-with-markup</a></li>
    </ul></li>
    <li><a name="toc-Refmac-Functions-1" href="#Refmac-Functions">11.20 Refmac Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002drefmac_002dcounter-1" href="#set_002drefmac_002dcounter">11.20.1 set-refmac-counter</a></li>
      <li><a name="toc-swap_002dmap_002dcolours-1" href="#swap_002dmap_002dcolours">11.20.2 swap-map-colours</a></li>
      <li><a name="toc-set_002dkeep_002dmap_002dcolour_002dafter_002drefmac-1" href="#set_002dkeep_002dmap_002dcolour_002dafter_002drefmac">11.20.3 set-keep-map-colour-after-refmac</a></li>
      <li><a name="toc-keep_002dmap_002dcolour_002dafter_002drefmac_002dstate-1" href="#keep_002dmap_002dcolour_002dafter_002drefmac_002dstate">11.20.4 keep-map-colour-after-refmac-state</a></li>
    </ul></li>
    <li><a name="toc-Symmetry-Functions-1" href="#Symmetry-Functions">11.21 Symmetry Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dsymmetry_002dsize-1" href="#set_002dsymmetry_002dsize">11.21.1 set-symmetry-size</a></li>
      <li><a name="toc-get_002dshow_002dsymmetry-1" href="#get_002dshow_002dsymmetry">11.21.2 get-show-symmetry</a></li>
      <li><a name="toc-set_002dshow_002dsymmetry_002dmaster-1" href="#set_002dshow_002dsymmetry_002dmaster">11.21.3 set-show-symmetry-master</a></li>
      <li><a name="toc-set_002dshow_002dsymmetry_002dmolecule-1" href="#set_002dshow_002dsymmetry_002dmolecule">11.21.4 set-show-symmetry-molecule</a></li>
      <li><a name="toc-symmetry_002das_002dcalphas-1" href="#symmetry_002das_002dcalphas">11.21.5 symmetry-as-calphas</a></li>
      <li><a name="toc-get_002dsymmetry_002das_002dcalphas_002dstate-1" href="#get_002dsymmetry_002das_002dcalphas_002dstate">11.21.6 get-symmetry-as-calphas-state</a></li>
      <li><a name="toc-set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap-1" href="#set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap">11.21.7 set-symmetry-molecule-rotate-colour-map</a></li>
      <li><a name="toc-symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate-1" href="#symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate">11.21.8 symmetry-molecule-rotate-colour-map-state</a></li>
      <li><a name="toc-set_002dsymmetry_002dcolour_002dby_002dsymop-1" href="#set_002dsymmetry_002dcolour_002dby_002dsymop">11.21.9 set-symmetry-colour-by-symop</a></li>
      <li><a name="toc-set_002dsymmetry_002dwhole_002dchain-1" href="#set_002dsymmetry_002dwhole_002dchain">11.21.10 set-symmetry-whole-chain</a></li>
      <li><a name="toc-set_002dsymmetry_002datom_002dlabels_002dexpanded-1" href="#set_002dsymmetry_002datom_002dlabels_002dexpanded">11.21.11 set-symmetry-atom-labels-expanded</a></li>
      <li><a name="toc-has_002dunit_002dcell_002dstate-1" href="#has_002dunit_002dcell_002dstate">11.21.12 has-unit-cell-state</a></li>
      <li><a name="toc-undo_002dsymmetry_002dview-1" href="#undo_002dsymmetry_002dview">11.21.13 undo-symmetry-view</a></li>
      <li><a name="toc-first_002dmolecule_002dwith_002dsymmetry_002ddisplayed-1" href="#first_002dmolecule_002dwith_002dsymmetry_002ddisplayed">11.21.14 first-molecule-with-symmetry-displayed</a></li>
      <li><a name="toc-save_002dsymmetry_002dcoords-1" href="#save_002dsymmetry_002dcoords">11.21.15 save-symmetry-coords</a></li>
      <li><a name="toc-new_002dmolecule_002dby_002dsymmetry-1" href="#new_002dmolecule_002dby_002dsymmetry">11.21.16 new-molecule-by-symmetry</a></li>
      <li><a name="toc-new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection-1" href="#new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection">11.21.17 new-molecule-by-symmetry-with-atom-selection</a></li>
      <li><a name="toc-new_002dmolecule_002dby_002dsymop-1" href="#new_002dmolecule_002dby_002dsymop">11.21.18 new-molecule-by-symop</a></li>
      <li><a name="toc-n_002dsymops-1" href="#n_002dsymops">11.21.19 n-symops</a></li>
      <li><a name="toc-origin_002dpre_002dshift_002dscm-1" href="#origin_002dpre_002dshift_002dscm">11.21.20 origin-pre-shift-scm</a></li>
      <li><a name="toc-origin_002dpre_002dshift_002dpy-1" href="#origin_002dpre_002dshift_002dpy">11.21.21 origin-pre-shift-py</a></li>
      <li><a name="toc-set_002dspace_002dgroup-1" href="#set_002dspace_002dgroup">11.21.22 set-space-group</a></li>
      <li><a name="toc-set_002dunit_002dcell_002dand_002dspace_002dgroup-1" href="#set_002dunit_002dcell_002dand_002dspace_002dgroup">11.21.23 set-unit-cell-and-space-group</a></li>
      <li><a name="toc-set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule-1" href="#set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule">11.21.24 set-unit-cell-and-space-group-using-molecule</a></li>
      <li><a name="toc-set_002dsymmetry_002dshift_002dsearch_002dsize-1" href="#set_002dsymmetry_002dshift_002dsearch_002dsize">11.21.25 set-symmetry-shift-search-size</a></li>
    </ul></li>
    <li><a name="toc-History-Functions-1" href="#History-Functions">11.22 History Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-print_002dall_002dhistory_002din_002dscheme-1" href="#print_002dall_002dhistory_002din_002dscheme">11.22.1 print-all-history-in-scheme</a></li>
      <li><a name="toc-print_002dall_002dhistory_002din_002dpython-1" href="#print_002dall_002dhistory_002din_002dpython">11.22.2 print-all-history-in-python</a></li>
      <li><a name="toc-set_002dconsole_002ddisplay_002dcommands_002dstate-1" href="#set_002dconsole_002ddisplay_002dcommands_002dstate">11.22.3 set-console-display-commands-state</a></li>
      <li><a name="toc-set_002dconsole_002ddisplay_002dcommands_002dhilights-1" href="#set_002dconsole_002ddisplay_002dcommands_002dhilights">11.22.4 set-console-display-commands-hilights</a></li>
    </ul></li>
    <li><a name="toc-State-Functions-1" href="#State-Functions">11.23 State Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-save_002dstate-1" href="#save_002dstate">11.23.1 save-state</a></li>
      <li><a name="toc-save_002dstate_002dfile-1" href="#save_002dstate_002dfile">11.23.2 save-state-file</a></li>
      <li><a name="toc-save_002dstate_002dfile_002dpy-1" href="#save_002dstate_002dfile_002dpy">11.23.3 save-state-file-py</a></li>
      <li><a name="toc-set_002dsave_002dstate_002dfile_002dname-1" href="#set_002dsave_002dstate_002dfile_002dname">11.23.4 set-save-state-file-name</a></li>
      <li><a name="toc-save_002dstate_002dfile_002dname_002dscm-1" href="#save_002dstate_002dfile_002dname_002dscm">11.23.5 save-state-file-name-scm</a></li>
      <li><a name="toc-save_002dstate_002dfile_002dname_002dpy-1" href="#save_002dstate_002dfile_002dname_002dpy">11.23.6 save-state-file-name-py</a></li>
      <li><a name="toc-set_002drun_002dstate_002dfile_002dstatus-1" href="#set_002drun_002dstate_002dfile_002dstatus">11.23.7 set-run-state-file-status</a></li>
      <li><a name="toc-run_002dstate_002dfile-1" href="#run_002dstate_002dfile">11.23.8 run-state-file</a></li>
      <li><a name="toc-run_002dstate_002dfile_002dmaybe-1" href="#run_002dstate_002dfile_002dmaybe">11.23.9 run-state-file-maybe</a></li>
    </ul></li>
    <li><a name="toc-The-Virtual-Trackball-1" href="#The-Virtual-Trackball">11.24 The Virtual Trackball</a>
    <ul class="no-bullet">
      <li><a name="toc-vt_002dsurface-1" href="#vt_002dsurface">11.24.1 vt-surface</a></li>
      <li><a name="toc-vt_002dsurface_002dstatus-1" href="#vt_002dsurface_002dstatus">11.24.2 vt-surface-status</a></li>
    </ul></li>
    <li><a name="toc-Clipping-Functions-1" href="#Clipping-Functions">11.25 Clipping Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dclipping_002dback-1" href="#set_002dclipping_002dback">11.25.1 set-clipping-back</a></li>
      <li><a name="toc-set_002dclipping_002dfront-1" href="#set_002dclipping_002dfront">11.25.2 set-clipping-front</a></li>
    </ul></li>
    <li><a name="toc-Unit-Cell-interface-1" href="#Unit-Cell-interface">11.26 Unit Cell interface</a>
    <ul class="no-bullet">
      <li><a name="toc-get_002dshow_002dunit_002dcell-1" href="#get_002dshow_002dunit_002dcell">11.26.1 get-show-unit-cell</a></li>
      <li><a name="toc-set_002dshow_002dunit_002dcells_002dall-1" href="#set_002dshow_002dunit_002dcells_002dall">11.26.2 set-show-unit-cells-all</a></li>
      <li><a name="toc-set_002dshow_002dunit_002dcell-1" href="#set_002dshow_002dunit_002dcell">11.26.3 set-show-unit-cell</a></li>
    </ul></li>
    <li><a name="toc-Colour-1" href="#Colour">11.27 Colour</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb-1" href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb">11.27.1 set-colour-map-rotation-on-read-pdb</a></li>
      <li><a name="toc-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag-1" href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag">11.27.2 set-colour-map-rotation-on-read-pdb-flag</a></li>
      <li><a name="toc-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag-1" href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag">11.27.3 set-colour-map-rotation-on-read-pdb-c-only-flag</a></li>
      <li><a name="toc-set_002dcolour_002dby_002dchain-1" href="#set_002dcolour_002dby_002dchain">11.27.4 set-colour-by-chain</a></li>
      <li><a name="toc-set_002dcolour_002dby_002dchain_002dgoodsell_002dmode-1" href="#set_002dcolour_002dby_002dchain_002dgoodsell_002dmode">11.27.5 set-colour-by-chain-goodsell-mode</a></li>
      <li><a name="toc-set_002dcolour_002dby_002dmolecule-1" href="#set_002dcolour_002dby_002dmolecule">11.27.6 set-colour-by-molecule</a></li>
      <li><a name="toc-set_002dsymmetry_002dcolour-1" href="#set_002dsymmetry_002dcolour">11.27.7 set-symmetry-colour</a></li>
    </ul></li>
    <li><a name="toc-Map-colour-1" href="#Map-colour">11.28 Map colour</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dcolour_002dmap_002drotation_002dfor_002dmap-1" href="#set_002dcolour_002dmap_002drotation_002dfor_002dmap">11.28.1 set-colour-map-rotation-for-map</a></li>
      <li><a name="toc-set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation-1" href="#set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation">11.28.2 set-molecule-bonds-colour-map-rotation</a></li>
      <li><a name="toc-get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation-1" href="#get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation">11.28.3 get-molecule-bonds-colour-map-rotation</a></li>
    </ul></li>
    <li><a name="toc-Anisotropic-Atoms-Interface-1" href="#Anisotropic-Atoms-Interface">11.29 Anisotropic Atoms Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-get_002dlimit_002daniso-1" href="#get_002dlimit_002daniso">11.29.1 get-limit-aniso</a></li>
      <li><a name="toc-get_002dshow_002dlimit_002daniso-1" href="#get_002dshow_002dlimit_002daniso">11.29.2 get-show-limit-aniso</a></li>
      <li><a name="toc-get_002dshow_002daniso-1" href="#get_002dshow_002daniso">11.29.3 get-show-aniso</a></li>
      <li><a name="toc-set_002dlimit_002daniso-1" href="#set_002dlimit_002daniso">11.29.4 set-limit-aniso</a></li>
      <li><a name="toc-set_002dshow_002daniso-1" href="#set_002dshow_002daniso">11.29.5 set-show-aniso</a></li>
      <li><a name="toc-set_002daniso_002dprobability-1" href="#set_002daniso_002dprobability">11.29.6 set-aniso-probability</a></li>
      <li><a name="toc-get_002daniso_002dprobability-1" href="#get_002daniso_002dprobability">11.29.7 get-aniso-probability</a></li>
    </ul></li>
    <li><a name="toc-Display-Functions-1" href="#Display-Functions">11.30 Display Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dgraphics_002dwindow_002dsize-1" href="#set_002dgraphics_002dwindow_002dsize">11.30.1 set-graphics-window-size</a></li>
      <li><a name="toc-set_002dgraphics_002dwindow_002dposition-1" href="#set_002dgraphics_002dwindow_002dposition">11.30.2 set-graphics-window-position</a></li>
      <li><a name="toc-store_002dgraphics_002dwindow_002dposition-1" href="#store_002dgraphics_002dwindow_002dposition">11.30.3 store-graphics-window-position</a></li>
      <li><a name="toc-graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences-1" href="#graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences">11.30.4 graphics-window-size-and-position-to-preferences</a></li>
      <li><a name="toc-graphics_002ddraw-1" href="#graphics_002ddraw">11.30.5 graphics-draw</a></li>
      <li><a name="toc-zalman_002dstereo_002dmode-1" href="#zalman_002dstereo_002dmode">11.30.6 zalman-stereo-mode</a></li>
      <li><a name="toc-hardware_002dstereo_002dmode-1" href="#hardware_002dstereo_002dmode">11.30.7 hardware-stereo-mode</a></li>
      <li><a name="toc-stereo_002dmode_002dstate-1" href="#stereo_002dmode_002dstate">11.30.8 stereo-mode-state</a></li>
      <li><a name="toc-mono_002dmode-1" href="#mono_002dmode">11.30.9 mono-mode</a></li>
      <li><a name="toc-side_002dby_002dside_002dstereo_002dmode-1" href="#side_002dby_002dside_002dstereo_002dmode">11.30.10 side-by-side-stereo-mode</a></li>
      <li><a name="toc-set_002dhardware_002dstereo_002dangle_002dfactor-1" href="#set_002dhardware_002dstereo_002dangle_002dfactor">11.30.11 set-hardware-stereo-angle-factor</a></li>
      <li><a name="toc-hardware_002dstereo_002dangle_002dfactor_002dstate-1" href="#hardware_002dstereo_002dangle_002dfactor_002dstate">11.30.12 hardware-stereo-angle-factor-state</a></li>
      <li><a name="toc-set_002dmodel_002dfit_002drefine_002ddialog_002dposition-1" href="#set_002dmodel_002dfit_002drefine_002ddialog_002dposition">11.30.13 set-model-fit-refine-dialog-position</a></li>
      <li><a name="toc-set_002ddisplay_002dcontrol_002ddialog_002dposition-1" href="#set_002ddisplay_002dcontrol_002ddialog_002dposition">11.30.14 set-display-control-dialog-position</a></li>
      <li><a name="toc-set_002dgo_002dto_002datom_002dwindow_002dposition-1" href="#set_002dgo_002dto_002datom_002dwindow_002dposition">11.30.15 set-go-to-atom-window-position</a></li>
      <li><a name="toc-set_002ddelete_002ddialog_002dposition-1" href="#set_002ddelete_002ddialog_002dposition">11.30.16 set-delete-dialog-position</a></li>
      <li><a name="toc-set_002drotate_002dtranslate_002ddialog_002dposition-1" href="#set_002drotate_002dtranslate_002ddialog_002dposition">11.30.17 set-rotate-translate-dialog-position</a></li>
      <li><a name="toc-set_002daccept_002dreject_002ddialog_002dposition-1" href="#set_002daccept_002dreject_002ddialog_002dposition">11.30.18 set-accept-reject-dialog-position</a></li>
      <li><a name="toc-set_002dramachandran_002dplot_002ddialog_002dposition-1" href="#set_002dramachandran_002dplot_002ddialog_002dposition">11.30.19 set-ramachandran-plot-dialog-position</a></li>
      <li><a name="toc-set_002dedit_002dchi_002dangles_002ddialog_002dposition-1" href="#set_002dedit_002dchi_002dangles_002ddialog_002dposition">11.30.20 set-edit-chi-angles-dialog-position</a></li>
      <li><a name="toc-set_002drotamer_002dselection_002ddialog_002dposition-1" href="#set_002drotamer_002dselection_002ddialog_002dposition">11.30.21 set-rotamer-selection-dialog-position</a></li>
    </ul></li>
    <li><a name="toc-Smooth-Scrolling-1" href="#Smooth-Scrolling">11.31 Smooth Scrolling</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dsmooth_002dscroll_002dflag-1" href="#set_002dsmooth_002dscroll_002dflag">11.31.1 set-smooth-scroll-flag</a></li>
      <li><a name="toc-get_002dsmooth_002dscroll-1" href="#get_002dsmooth_002dscroll">11.31.2 get-smooth-scroll</a></li>
      <li><a name="toc-set_002dsmooth_002dscroll_002dsteps-1" href="#set_002dsmooth_002dscroll_002dsteps">11.31.3 set-smooth-scroll-steps</a></li>
      <li><a name="toc-set_002dsmooth_002dscroll_002dlimit-1" href="#set_002dsmooth_002dscroll_002dlimit">11.31.4 set-smooth-scroll-limit</a></li>
    </ul></li>
    <li><a name="toc-Font-Parameters-1" href="#Font-Parameters">11.32 Font Parameters</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dfont_002dsize-1" href="#set_002dfont_002dsize">11.32.1 set-font-size</a></li>
      <li><a name="toc-get_002dfont_002dsize-1" href="#get_002dfont_002dsize">11.32.2 get-font-size</a></li>
      <li><a name="toc-set_002dfont_002dcolour-1" href="#set_002dfont_002dcolour">11.32.3 set-font-colour</a></li>
      <li><a name="toc-set_002duse_002dstroke_002dcharacters-1" href="#set_002duse_002dstroke_002dcharacters">11.32.4 set-use-stroke-characters</a></li>
    </ul></li>
    <li><a name="toc-Rotation-Centre-1" href="#Rotation-Centre">11.33 Rotation Centre</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002drotation_002dcentre_002dsize-1" href="#set_002drotation_002dcentre_002dsize">11.33.1 set-rotation-centre-size</a></li>
      <li><a name="toc-recentre_002don_002dread_002dpdb-1" href="#recentre_002don_002dread_002dpdb">11.33.2 recentre-on-read-pdb</a></li>
      <li><a name="toc-set_002drecentre_002don_002dread_002dpdb-1" href="#set_002drecentre_002don_002dread_002dpdb">11.33.3 set-recentre-on-read-pdb</a></li>
      <li><a name="toc-set_002drotation_002dcentre-1" href="#set_002drotation_002dcentre">11.33.4 set-rotation-centre</a></li>
      <li><a name="toc-go_002dto_002dligand-1" href="#go_002dto_002dligand">11.33.5 go-to-ligand</a></li>
      <li><a name="toc-set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit-1" href="#set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit">11.33.6 set-go-to-ligand-n-atoms-limit</a></li>
      <li><a name="toc-set_002dreorienting_002dnext_002dresidue_002dmode-1" href="#set_002dreorienting_002dnext_002dresidue_002dmode">11.33.7 set-reorienting-next-residue-mode</a></li>
    </ul></li>
    <li><a name="toc-Atom-Selection-Utilities-1" href="#Atom-Selection-Utilities">11.34 Atom Selection Utilities</a>
    <ul class="no-bullet">
      <li><a name="toc-median_002dtemperature_002dfactor-1" href="#median_002dtemperature_002dfactor">11.34.1 median-temperature-factor</a></li>
      <li><a name="toc-average_002dtemperature_002dfactor-1" href="#average_002dtemperature_002dfactor">11.34.2 average-temperature-factor</a></li>
      <li><a name="toc-standard_002ddeviation_002dtemperature_002dfactor-1" href="#standard_002ddeviation_002dtemperature_002dfactor">11.34.3 standard-deviation-temperature-factor</a></li>
      <li><a name="toc-clear_002dpending_002dpicks-1" href="#clear_002dpending_002dpicks">11.34.4 clear-pending-picks</a></li>
      <li><a name="toc-set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms-1" href="#set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms">11.34.5 set-default-temperature-factor-for-new-atoms</a></li>
      <li><a name="toc-default_002dnew_002datoms_002db_002dfactor-1" href="#default_002dnew_002datoms_002db_002dfactor">11.34.6 default-new-atoms-b-factor</a></li>
      <li><a name="toc-set_002dreset_002db_002dfactor_002dmoved_002datoms-1" href="#set_002dreset_002db_002dfactor_002dmoved_002datoms">11.34.7 set-reset-b-factor-moved-atoms</a></li>
      <li><a name="toc-get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate-1" href="#get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate">11.34.8 get-reset-b-factor-moved-atoms-state</a></li>
      <li><a name="toc-set_002datom_002dattribute-1" href="#set_002datom_002dattribute">11.34.9 set-atom-attribute</a></li>
      <li><a name="toc-set_002datom_002dstring_002dattribute-1" href="#set_002datom_002dstring_002dattribute">11.34.10 set-atom-string-attribute</a></li>
      <li><a name="toc-set_002datom_002dattributes-1" href="#set_002datom_002dattributes">11.34.11 set-atom-attributes</a></li>
      <li><a name="toc-set_002dresidue_002dname-1" href="#set_002dresidue_002dname">11.34.12 set-residue-name</a></li>
    </ul></li>
    <li><a name="toc-Skeletonization-Interface-1" href="#Skeletonization-Interface">11.35 Skeletonization Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-skeletonize_002dmap-1" href="#skeletonize_002dmap">11.35.1 skeletonize-map</a></li>
      <li><a name="toc-unskeletonize_002dmap-1" href="#unskeletonize_002dmap">11.35.2 unskeletonize-map</a></li>
      <li><a name="toc-set_002dmax_002dskeleton_002dsearch_002ddepth-1" href="#set_002dmax_002dskeleton_002dsearch_002ddepth">11.35.3 set-max-skeleton-search-depth</a></li>
      <li><a name="toc-set_002dskeleton_002dbox_002dsize-1" href="#set_002dskeleton_002dbox_002dsize">11.35.4 set-skeleton-box-size</a></li>
    </ul></li>
    <li><a name="toc-Save-Coordinates-2" href="#Save-Coordinates">11.36 Save Coordinates</a>
    <ul class="no-bullet">
      <li><a name="toc-save_002dcoordinates-1" href="#save_002dcoordinates">11.36.1 save-coordinates</a></li>
      <li><a name="toc-set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory-1" href="#set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory">11.36.2 set-save-coordinates-in-original-directory</a></li>
    </ul></li>
    <li><a name="toc-Read-Phases-File-Functions-1" href="#Read-Phases-File-Functions">11.37 Read Phases File Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap-1" href="#read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap">11.37.1 read-phs-and-coords-and-make-map</a></li>
      <li><a name="toc-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol-1" href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol">11.37.2 read-phs-and-make-map-using-cell-symm-from-previous-mol</a></li>
      <li><a name="toc-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol-1" href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol">11.37.3 read-phs-and-make-map-using-cell-symm-from-mol</a></li>
      <li><a name="toc-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm-1" href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm">11.37.4 read-phs-and-make-map-using-cell-symm</a></li>
      <li><a name="toc-read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits-1" href="#read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits">11.37.5 read-phs-and-make-map-with-reso-limits</a></li>
    </ul></li>
    <li><a name="toc-Graphics-Move-1" href="#Graphics-Move">11.38 Graphics Move</a>
    <ul class="no-bullet">
      <li><a name="toc-undo_002dlast_002dmove-1" href="#undo_002dlast_002dmove">11.38.1 undo-last-move</a></li>
      <li><a name="toc-translate_002dmolecule_002dby-1" href="#translate_002dmolecule_002dby">11.38.2 translate-molecule-by</a></li>
      <li><a name="toc-transform_002dmolecule_002dby-1" href="#transform_002dmolecule_002dby">11.38.3 transform-molecule-by</a></li>
      <li><a name="toc-transform_002dzone-1" href="#transform_002dzone">11.38.4 transform-zone</a></li>
    </ul></li>
    <li><a name="toc-Go-To-Atom-Widget-Functions-1" href="#Go-To-Atom-Widget-Functions">11.39 Go To Atom Widget Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-post_002dgo_002dto_002datom_002dwindow-1" href="#post_002dgo_002dto_002datom_002dwindow">11.39.1 post-go-to-atom-window</a></li>
      <li><a name="toc-go_002dto_002datom_002dmolecule_002dnumber-1" href="#go_002dto_002datom_002dmolecule_002dnumber">11.39.2 go-to-atom-molecule-number</a></li>
      <li><a name="toc-go_002dto_002datom_002dchain_002did-1" href="#go_002dto_002datom_002dchain_002did">11.39.3 go-to-atom-chain-id</a></li>
      <li><a name="toc-go_002dto_002datom_002datom_002dname-1" href="#go_002dto_002datom_002datom_002dname">11.39.4 go-to-atom-atom-name</a></li>
      <li><a name="toc-go_002dto_002datom_002dresidue_002dnumber-1" href="#go_002dto_002datom_002dresidue_002dnumber">11.39.5 go-to-atom-residue-number</a></li>
      <li><a name="toc-go_002dto_002datom_002dins_002dcode-1" href="#go_002dto_002datom_002dins_002dcode">11.39.6 go-to-atom-ins-code</a></li>
      <li><a name="toc-go_002dto_002datom_002dalt_002dconf-1" href="#go_002dto_002datom_002dalt_002dconf">11.39.7 go-to-atom-alt-conf</a></li>
      <li><a name="toc-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname-1" href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname">11.39.8 set-go-to-atom-chain-residue-atom-name</a></li>
      <li><a name="toc-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull-1" href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull">11.39.9 set-go-to-atom-chain-residue-atom-name-full</a></li>
      <li><a name="toc-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw-1" href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw">11.39.10 set-go-to-atom-chain-residue-atom-name-no-redraw</a></li>
      <li><a name="toc-update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition-1" href="#update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition">11.39.11 update-go-to-atom-from-current-position</a></li>
      <li><a name="toc-atom_002dspec_002dto_002datom_002dindex-1" href="#atom_002dspec_002dto_002datom_002dindex">11.39.12 atom-spec-to-atom-index</a></li>
      <li><a name="toc-full_002datom_002dspec_002dto_002datom_002dindex-1" href="#full_002datom_002dspec_002dto_002datom_002dindex">11.39.13 full-atom-spec-to-atom-index</a></li>
      <li><a name="toc-update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol-1" href="#update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol">11.39.14 update-go-to-atom-window-on-changed-mol</a></li>
      <li><a name="toc-update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol-1" href="#update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol">11.39.15 update-go-to-atom-window-on-new-mol</a></li>
      <li><a name="toc-set_002dgo_002dto_002datom_002dmolecule-1" href="#set_002dgo_002dto_002datom_002dmolecule">11.39.16 set-go-to-atom-molecule</a></li>
    </ul></li>
    <li><a name="toc-Map-and-Molecule-Control-1" href="#Map-and-Molecule-Control">11.40 Map and Molecule Control</a>
    <ul class="no-bullet">
      <li><a name="toc-post_002ddisplay_002dcontrol_002dwindow-1" href="#post_002ddisplay_002dcontrol_002dwindow">11.40.1 post-display-control-window</a></li>
      <li><a name="toc-set_002dmap_002ddisplayed-1" href="#set_002dmap_002ddisplayed">11.40.2 set-map-displayed</a></li>
      <li><a name="toc-set_002dmol_002ddisplayed-1" href="#set_002dmol_002ddisplayed">11.40.3 set-mol-displayed</a></li>
      <li><a name="toc-set_002ddisplay_002donly_002dmodel_002dmol-1" href="#set_002ddisplay_002donly_002dmodel_002dmol">11.40.4 set-display-only-model-mol</a></li>
      <li><a name="toc-set_002dmol_002dactive-1" href="#set_002dmol_002dactive">11.40.5 set-mol-active</a></li>
      <li><a name="toc-display_002dmaps_002dscm-1" href="#display_002dmaps_002dscm">11.40.6 display-maps-scm</a></li>
      <li><a name="toc-mol_002dis_002ddisplayed-1" href="#mol_002dis_002ddisplayed">11.40.7 mol-is-displayed</a></li>
      <li><a name="toc-mol_002dis_002dactive-1" href="#mol_002dis_002dactive">11.40.8 mol-is-active</a></li>
      <li><a name="toc-map_002dis_002ddisplayed-1" href="#map_002dis_002ddisplayed">11.40.9 map-is-displayed</a></li>
      <li><a name="toc-set_002dall_002dmaps_002ddisplayed-1" href="#set_002dall_002dmaps_002ddisplayed">11.40.10 set-all-maps-displayed</a></li>
      <li><a name="toc-set_002dall_002dmodels_002ddisplayed_002dand_002dactive-1" href="#set_002dall_002dmodels_002ddisplayed_002dand_002dactive">11.40.11 set-all-models-displayed-and-active</a></li>
      <li><a name="toc-set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed-1" href="#set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed">11.40.12 set-only-last-model-molecule-displayed</a></li>
      <li><a name="toc-display_002donly_002dactive-1" href="#display_002donly_002dactive">11.40.13 display-only-active</a></li>
      <li><a name="toc-space_002dgroup_002dscm-1" href="#space_002dgroup_002dscm">11.40.14 space-group-scm</a></li>
      <li><a name="toc-show_002dspacegroup-1" href="#show_002dspacegroup">11.40.15 show-spacegroup</a></li>
      <li><a name="toc-symmetry_002doperators_002dscm-1" href="#symmetry_002doperators_002dscm">11.40.16 symmetry-operators-scm</a></li>
      <li><a name="toc-symmetry_002doperators_002dpy-1" href="#symmetry_002doperators_002dpy">11.40.17 symmetry-operators-py</a></li>
    </ul></li>
    <li><a name="toc-Align-and-Mutate-1" href="#Align-and-Mutate">11.41 Align and Mutate</a>
    <ul class="no-bullet">
      <li><a name="toc-align_002dand_002dmutate-1" href="#align_002dand_002dmutate">11.41.1 align-and-mutate</a></li>
      <li><a name="toc-set_002dalignment_002dgap_002dand_002dspace_002dpenalty-1" href="#set_002dalignment_002dgap_002dand_002dspace_002dpenalty">11.41.2 set-alignment-gap-and-space-penalty</a></li>
      <li><a name="toc-nearest_002dresidue_002dby_002dsequence_002dscm-1" href="#nearest_002dresidue_002dby_002dsequence_002dscm">11.41.3 nearest-residue-by-sequence-scm</a></li>
      <li><a name="toc-nearest_002dresidue_002dby_002dsequence_002dpy-1" href="#nearest_002dresidue_002dby_002dsequence_002dpy">11.41.4 nearest-residue-by-sequence-py</a></li>
    </ul></li>
    <li><a name="toc-Renumber-Residue-Range-1" href="#Renumber-Residue-Range">11.42 Renumber Residue Range</a>
    <ul class="no-bullet">
      <li><a name="toc-renumber_002dresidue_002drange-1" href="#renumber_002dresidue_002drange">11.42.1 renumber-residue-range</a></li>
      <li><a name="toc-change_002dresidue_002dnumber-1" href="#change_002dresidue_002dnumber">11.42.2 change-residue-number</a></li>
    </ul></li>
    <li><a name="toc-Scripting-Interface-1" href="#Scripting-Interface">11.43 Scripting Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-probe_002davailable_002dp-1" href="#probe_002davailable_002dp">11.43.1 probe-available-p</a></li>
      <li><a name="toc-post_002dscripting_002dwindow-1" href="#post_002dscripting_002dwindow">11.43.2 post-scripting-window</a></li>
      <li><a name="toc-post_002dscheme_002dscripting_002dwindow-1" href="#post_002dscheme_002dscripting_002dwindow">11.43.3 post-scheme-scripting-window</a></li>
      <li><a name="toc-post_002dpython_002dscripting_002dwindow-1" href="#post_002dpython_002dscripting_002dwindow">11.43.4 post-python-scripting-window</a></li>
    </ul></li>
    <li><a name="toc-Monomer-1" href="#Monomer">11.44 Monomer</a>
    <ul class="no-bullet">
      <li><a name="toc-run_002dscript-1" href="#run_002dscript">11.44.1 run-script</a></li>
      <li><a name="toc-run_002dguile_002dscript-1" href="#run_002dguile_002dscript">11.44.2 run-guile-script</a></li>
      <li><a name="toc-run_002dpython_002dscript-1" href="#run_002dpython_002dscript">11.44.3 run-python-script</a></li>
      <li><a name="toc-import_002dpython_002dmodule-1" href="#import_002dpython_002dmodule">11.44.4 import-python-module</a></li>
      <li><a name="toc-matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm-1" href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm">11.44.5 matching-compound-names-from-dictionary-scm</a></li>
      <li><a name="toc-comp_002did_002dto_002dname_002dscm-1" href="#comp_002did_002dto_002dname_002dscm">11.44.6 comp-id-to-name-scm</a></li>
      <li><a name="toc-auto_002dload_002ddictionary-1" href="#auto_002dload_002ddictionary">11.44.7 auto-load-dictionary</a></li>
      <li><a name="toc-reload_002ddictionary-1" href="#reload_002ddictionary">11.44.8 reload-dictionary</a></li>
      <li><a name="toc-add_002dnon_002dauto_002dload_002dresidue_002dname-1" href="#add_002dnon_002dauto_002dload_002dresidue_002dname">11.44.9 add-non-auto-load-residue-name</a></li>
      <li><a name="toc-remove_002dnon_002dauto_002dload_002dresidue_002dname-1" href="#remove_002dnon_002dauto_002dload_002dresidue_002dname">11.44.10 remove-non-auto-load-residue-name</a></li>
      <li><a name="toc-matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy-1" href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy">11.44.11 matching-compound-names-from-dictionary-py</a></li>
      <li><a name="toc-comp_002did_002dto_002dname_002dpy-1" href="#comp_002did_002dto_002dname_002dpy">11.44.12 comp-id-to-name-py</a></li>
    </ul></li>
    <li><a name="toc-Regularization-and-Refinement-1" href="#Regularization-and-Refinement">11.45 Regularization and Refinement</a>
    <ul class="no-bullet">
      <li><a name="toc-add_002dplanar_002dpeptide_002drestraints-1" href="#add_002dplanar_002dpeptide_002drestraints">11.45.1 add-planar-peptide-restraints</a></li>
      <li><a name="toc-remove_002dplanar_002dpeptide_002drestraints-1" href="#remove_002dplanar_002dpeptide_002drestraints">11.45.2 remove-planar-peptide-restraints</a></li>
      <li><a name="toc-make_002dtight_002dplanar_002dpeptide_002drestraints-1" href="#make_002dtight_002dplanar_002dpeptide_002drestraints">11.45.3 make-tight-planar-peptide-restraints</a></li>
      <li><a name="toc-set_002duse_002dtrans_002dpeptide_002drestraints-1" href="#set_002duse_002dtrans_002dpeptide_002drestraints">11.45.4 set-use-trans-peptide-restraints</a></li>
      <li><a name="toc-add_002domega_002dtorsion_002drestriants-1" href="#add_002domega_002dtorsion_002drestriants">11.45.5 add-omega-torsion-restriants</a></li>
      <li><a name="toc-remove_002domega_002dtorsion_002drestriants-1" href="#remove_002domega_002dtorsion_002drestriants">11.45.6 remove-omega-torsion-restriants</a></li>
      <li><a name="toc-set_002dauto_002dh_002dbond_002drestraints-1" href="#set_002dauto_002dh_002dbond_002drestraints">11.45.7 set-auto-h-bond-restraints</a></li>
      <li><a name="toc-set_002drefinement_002dimmediate_002dreplacement-1" href="#set_002drefinement_002dimmediate_002dreplacement">11.45.8 set-refinement-immediate-replacement</a></li>
      <li><a name="toc-refinement_002dimmediate_002dreplacement_002dstate-1" href="#refinement_002dimmediate_002dreplacement_002dstate">11.45.9 refinement-immediate-replacement-state</a></li>
      <li><a name="toc-set_002dresidue_002dselection_002dflash_002dframes_002dnumber-1" href="#set_002dresidue_002dselection_002dflash_002dframes_002dnumber">11.45.10 set-residue-selection-flash-frames-number</a></li>
      <li><a name="toc-accept_002dmoving_002datoms-1" href="#accept_002dmoving_002datoms">11.45.11 accept-moving-atoms</a></li>
      <li><a name="toc-accept_002dregularizement-1" href="#accept_002dregularizement">11.45.12 accept-regularizement</a></li>
      <li><a name="toc-refinement_002dalready_002dongoing_002dp-1" href="#refinement_002dalready_002dongoing_002dp">11.45.13 refinement-already-ongoing-p</a></li>
      <li><a name="toc-refine_002dresidues_002dscm-1" href="#refine_002dresidues_002dscm">11.45.14 refine-residues-scm</a></li>
      <li><a name="toc-refine_002dresidues_002dpy-1" href="#refine_002dresidues_002dpy">11.45.15 refine-residues-py</a></li>
      <li><a name="toc-shiftfield_002db_002dfactor_002drefinement-1" href="#shiftfield_002db_002dfactor_002drefinement">11.45.16 shiftfield-b-factor-refinement</a></li>
      <li><a name="toc-shiftfield_002dxyz_002dfactor_002drefinement-1" href="#shiftfield_002dxyz_002dfactor_002drefinement">11.45.17 shiftfield-xyz-factor-refinement</a></li>
      <li><a name="toc-set_002drefine_002dwith_002dtorsion_002drestraints-1" href="#set_002drefine_002dwith_002dtorsion_002drestraints">11.45.18 set-refine-with-torsion-restraints</a></li>
      <li><a name="toc-refine_002dwith_002dtorsion_002drestraints_002dstate-1" href="#refine_002dwith_002dtorsion_002drestraints_002dstate">11.45.19 refine-with-torsion-restraints-state</a></li>
      <li><a name="toc-set_002dmatrix-1" href="#set_002dmatrix">11.45.20 set-matrix</a></li>
      <li><a name="toc-matrix_002dstate-1" href="#matrix_002dstate">11.45.21 matrix-state</a></li>
      <li><a name="toc-get_002dmap_002dweight-1" href="#get_002dmap_002dweight">11.45.22 get-map-weight</a></li>
      <li><a name="toc-set_002drefine_002dauto_002drange_002dstep-1" href="#set_002drefine_002dauto_002drange_002dstep">11.45.23 set-refine-auto-range-step</a></li>
      <li><a name="toc-set_002drefine_002dmax_002dresidues-1" href="#set_002drefine_002dmax_002dresidues">11.45.24 set-refine-max-residues</a></li>
      <li><a name="toc-refine_002dzone_002datom_002dindex_002ddefine-1" href="#refine_002dzone_002datom_002dindex_002ddefine">11.45.25 refine-zone-atom-index-define</a></li>
      <li><a name="toc-refine_002dzone-1" href="#refine_002dzone">11.45.26 refine-zone</a></li>
      <li><a name="toc-repeat_002drefine_002dzone-1" href="#repeat_002drefine_002dzone">11.45.27 repeat-refine-zone</a></li>
      <li><a name="toc-refine_002dauto_002drange-1" href="#refine_002dauto_002drange">11.45.28 refine-auto-range</a></li>
      <li><a name="toc-regularize_002dzone-1" href="#regularize_002dzone">11.45.29 regularize-zone</a></li>
      <li><a name="toc-set_002ddragged_002drefinement_002dsteps_002dper_002dframe-1" href="#set_002ddragged_002drefinement_002dsteps_002dper_002dframe">11.45.30 set-dragged-refinement-steps-per-frame</a></li>
      <li><a name="toc-dragged_002drefinement_002dsteps_002dper_002dframe-1" href="#dragged_002drefinement_002dsteps_002dper_002dframe">11.45.31 dragged-refinement-steps-per-frame</a></li>
      <li><a name="toc-set_002drefinement_002drefine_002dper_002dframe-1" href="#set_002drefinement_002drefine_002dper_002dframe">11.45.32 set-refinement-refine-per-frame</a></li>
      <li><a name="toc-refinement_002drefine_002dper_002dframe_002dstate-1" href="#refinement_002drefine_002dper_002dframe_002dstate">11.45.33 refinement-refine-per-frame-state</a></li>
      <li><a name="toc-set_002drefinement_002ddrag_002delasticity-1" href="#set_002drefinement_002ddrag_002delasticity">11.45.34 set-refinement-drag-elasticity</a></li>
      <li><a name="toc-set_002drefine_002dramachandran_002dangles-1" href="#set_002drefine_002dramachandran_002dangles">11.45.35 set-refine-ramachandran-angles</a></li>
      <li><a name="toc-set_002drefine_002dramachandran_002drestraints_002dtype-1" href="#set_002drefine_002dramachandran_002drestraints_002dtype">11.45.36 set-refine-ramachandran-restraints-type</a></li>
      <li><a name="toc-set_002drefine_002dramachandran_002drestraints_002dweight-1" href="#set_002drefine_002dramachandran_002drestraints_002dweight">11.45.37 set-refine-ramachandran-restraints-weight</a></li>
      <li><a name="toc-set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement-1" href="#set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement">11.45.38 set-fix-chiral-volumes-before-refinement</a></li>
      <li><a name="toc-check_002dchiral_002dvolumes-1" href="#check_002dchiral_002dvolumes">11.45.39 check-chiral-volumes</a></li>
      <li><a name="toc-set_002dshow_002dchiral_002dvolume_002derrors_002ddialog-1" href="#set_002dshow_002dchiral_002dvolume_002derrors_002ddialog">11.45.40 set-show-chiral-volume-errors-dialog</a></li>
      <li><a name="toc-set_002dsecondary_002dstructure_002drestraints_002dtype-1" href="#set_002dsecondary_002dstructure_002drestraints_002dtype">11.45.41 set-secondary-structure-restraints-type</a></li>
      <li><a name="toc-secondary_002dstructure_002drestraints_002dtype-1" href="#secondary_002dstructure_002drestraints_002dtype">11.45.42 secondary-structure-restraints-type</a></li>
      <li><a name="toc-imol_002drefinement_002dmap-1" href="#imol_002drefinement_002dmap">11.45.43 imol-refinement-map</a></li>
      <li><a name="toc-set_002dimol_002drefinement_002dmap-1" href="#set_002dimol_002drefinement_002dmap">11.45.44 set-imol-refinement-map</a></li>
      <li><a name="toc-does_002dresidue_002dexist_002dp-1" href="#does_002dresidue_002dexist_002dp">11.45.45 does-residue-exist-p</a></li>
      <li><a name="toc-delete_002drestraints-1" href="#delete_002drestraints">11.45.46 delete-restraints</a></li>
      <li><a name="toc-add_002dextra_002dbond_002drestraint-1" href="#add_002dextra_002dbond_002drestraint">11.45.47 add-extra-bond-restraint</a></li>
      <li><a name="toc-add_002dextra_002dgeman_002dmcclure_002drestraint-1" href="#add_002dextra_002dgeman_002dmcclure_002drestraint">11.45.48 add-extra-geman-mcclure-restraint</a></li>
      <li><a name="toc-delete_002dall_002dextra_002drestraints-1" href="#delete_002dall_002dextra_002drestraints">11.45.49 delete-all-extra-restraints</a></li>
      <li><a name="toc-delete_002dextra_002drestraints_002dfor_002dresidue-1" href="#delete_002dextra_002drestraints_002dfor_002dresidue">11.45.50 delete-extra-restraints-for-residue</a></li>
      <li><a name="toc-set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits-1" href="#set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits">11.45.51 set-extra-restraints-prosmart-sigma-limits</a></li>
      <li><a name="toc-generate_002dlocal_002dself_002drestraints-1" href="#generate_002dlocal_002dself_002drestraints">11.45.52 generate-local-self-restraints</a></li>
      <li><a name="toc-generate_002dself_002drestraints-1" href="#generate_002dself_002drestraints">11.45.53 generate-self-restraints</a></li>
      <li><a name="toc-generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm-1" href="#generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm">11.45.54 generate-local-self-restraints-by-residues-scm</a></li>
      <li><a name="toc-write_002dinterpolated_002dextra_002drestraints-1" href="#write_002dinterpolated_002dextra_002drestraints">11.45.55 write-interpolated-extra-restraints</a></li>
      <li><a name="toc-write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints-1" href="#write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints">11.45.56 write-interpolated-models-and-extra-restraints</a></li>
      <li><a name="toc-set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions-1" href="#set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions">11.45.57 set-use-only-extra-torsion-restraints-for-torsions</a></li>
      <li><a name="toc-use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate-1" href="#use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate">11.45.58 use-only-extra-torsion-restraints-for-torsions-state</a></li>
      <li><a name="toc-show_002drestraints_002deditor-1" href="#show_002drestraints_002deditor">11.45.59 show-restraints-editor</a></li>
      <li><a name="toc-show_002drestraints_002deditor_002dby_002dindex-1" href="#show_002drestraints_002deditor_002dby_002dindex">11.45.60 show-restraints-editor-by-index</a></li>
      <li><a name="toc-write_002drestraints_002dcif_002ddictionary-1" href="#write_002drestraints_002dcif_002ddictionary">11.45.61 write-restraints-cif-dictionary</a></li>
    </ul></li>
    <li><a name="toc-Simplex-Refinement-Interface-1" href="#Simplex-Refinement-Interface">11.46 Simplex Refinement Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex-1" href="#fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex">11.46.1 fit-residue-range-to-map-by-simplex</a></li>
      <li><a name="toc-score_002dresidue_002drange_002dfit_002dto_002dmap-1" href="#score_002dresidue_002drange_002dfit_002dto_002dmap">11.46.2 score-residue-range-fit-to-map</a></li>
    </ul></li>
    <li><a name="toc-Nomenclature-Errors-1" href="#Nomenclature-Errors">11.47 Nomenclature Errors</a>
    <ul class="no-bullet">
      <li><a name="toc-fix_002dnomenclature_002derrors-1" href="#fix_002dnomenclature_002derrors">11.47.1 fix-nomenclature-errors</a></li>
      <li><a name="toc-set_002dnomenclature_002derrors_002don_002dread-1" href="#set_002dnomenclature_002derrors_002don_002dread">11.47.2 set-nomenclature-errors-on-read</a></li>
    </ul></li>
    <li><a name="toc-Atom-Info-Interface-1" href="#Atom-Info-Interface">11.48 Atom Info  Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-output_002datom_002dinfo_002das_002dtext-1" href="#output_002datom_002dinfo_002das_002dtext">11.48.1 output-atom-info-as-text</a></li>
    </ul></li>
    <li><a name="toc-Residue-Info-1" href="#Residue-Info">11.49 Residue Info</a>
    <ul class="no-bullet">
      <li><a name="toc-residue_002dinfo_002ddialog-1" href="#residue_002dinfo_002ddialog">11.49.1 residue-info-dialog</a></li>
    </ul></li>
    <li><a name="toc-Residue-Environment-Functions-1" href="#Residue-Environment-Functions">11.50 Residue Environment Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dshow_002denvironment_002ddistances-1" href="#set_002dshow_002denvironment_002ddistances">11.50.1 set-show-environment-distances</a></li>
      <li><a name="toc-set_002dshow_002denvironment_002ddistances_002dbumps-1" href="#set_002dshow_002denvironment_002ddistances_002dbumps">11.50.2 set-show-environment-distances-bumps</a></li>
      <li><a name="toc-set_002dshow_002denvironment_002ddistances_002dh_002dbonds-1" href="#set_002dshow_002denvironment_002ddistances_002dh_002dbonds">11.50.3 set-show-environment-distances-h-bonds</a></li>
      <li><a name="toc-show_002denvironment_002ddistances_002dstate-1" href="#show_002denvironment_002ddistances_002dstate">11.50.4 show-environment-distances-state</a></li>
      <li><a name="toc-set_002denvironment_002ddistances_002ddistance_002dlimits-1" href="#set_002denvironment_002ddistances_002ddistance_002dlimits">11.50.5 set-environment-distances-distance-limits</a></li>
      <li><a name="toc-set_002dshow_002denvironment_002ddistances_002das_002dsolid-1" href="#set_002dshow_002denvironment_002ddistances_002das_002dsolid">11.50.6 set-show-environment-distances-as-solid</a></li>
      <li><a name="toc-set_002denvironment_002ddistances_002dlabel_002datom-1" href="#set_002denvironment_002ddistances_002dlabel_002datom">11.50.7 set-environment-distances-label-atom</a></li>
      <li><a name="toc-label_002dneighbours-1" href="#label_002dneighbours">11.50.8 label-neighbours</a></li>
      <li><a name="toc-label_002datoms_002din_002dresidue-1" href="#label_002datoms_002din_002dresidue">11.50.9 label-atoms-in-residue</a></li>
      <li><a name="toc-add_002dgeometry_002ddistance-1" href="#add_002dgeometry_002ddistance">11.50.10 add-geometry-distance</a></li>
    </ul></li>
    <li><a name="toc-Pointer-Position-Function-1" href="#Pointer-Position-Function">11.51 Pointer Position Function</a>
    <ul class="no-bullet">
      <li><a name="toc-get_002dpointer_002dposition_002dfrac_002dpy-1" href="#get_002dpointer_002dposition_002dfrac_002dpy">11.51.1 get-pointer-position-frac-py</a></li>
    </ul></li>
    <li><a name="toc-Pointer-Functions-1" href="#Pointer-Functions">11.52 Pointer Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dshow_002dpointer_002ddistances-1" href="#set_002dshow_002dpointer_002ddistances">11.52.1 set-show-pointer-distances</a></li>
      <li><a name="toc-show_002dpointer_002ddistances_002dstate-1" href="#show_002dpointer_002ddistances_002dstate">11.52.2 show-pointer-distances-state</a></li>
    </ul></li>
    <li><a name="toc-Zoom-Functions-1" href="#Zoom-Functions">11.53 Zoom Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-scale_002dzoom-1" href="#scale_002dzoom">11.53.1 scale-zoom</a></li>
      <li><a name="toc-zoom_002dfactor-1" href="#zoom_002dfactor">11.53.2 zoom-factor</a></li>
      <li><a name="toc-set_002dsmooth_002dscroll_002ddo_002dzoom-1" href="#set_002dsmooth_002dscroll_002ddo_002dzoom">11.53.3 set-smooth-scroll-do-zoom</a></li>
      <li><a name="toc-smooth_002dscroll_002ddo_002dzoom-1" href="#smooth_002dscroll_002ddo_002dzoom">11.53.4 smooth-scroll-do-zoom</a></li>
    </ul></li>
    <li><a name="toc-CNS-Data-Functions-1" href="#CNS-Data-Functions">11.54 CNS Data Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-handle_002dcns_002ddata_002dfile-1" href="#handle_002dcns_002ddata_002dfile">11.54.1 handle-cns-data-file</a></li>
      <li><a name="toc-handle_002dcns_002ddata_002dfile_002dwith_002dcell-1" href="#handle_002dcns_002ddata_002dfile_002dwith_002dcell">11.54.2 handle-cns-data-file-with-cell</a></li>
    </ul></li>
    <li><a name="toc-mmCIF-Functions-1" href="#mmCIF-Functions">11.55 mmCIF Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-open_002dcif_002ddictionary_002dfile_002dselector_002ddialog-1" href="#open_002dcif_002ddictionary_002dfile_002dselector_002ddialog">11.55.1 open-cif-dictionary-file-selector-dialog</a></li>
      <li><a name="toc-non_002dstandard_002dresidue_002dnames_002dscm-1" href="#non_002dstandard_002dresidue_002dnames_002dscm">11.55.2 non-standard-residue-names-scm</a></li>
      <li><a name="toc-non_002dstandard_002dresidue_002dnames_002dpy-1" href="#non_002dstandard_002dresidue_002dnames_002dpy">11.55.3 non-standard-residue-names-py</a></li>
    </ul></li>
    <li><a name="toc-SHELXL-Functions-1" href="#SHELXL-Functions">11.56 SHELXL Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-read_002dshelx_002dins_002dfile-1" href="#read_002dshelx_002dins_002dfile">11.56.1 read-shelx-ins-file</a></li>
      <li><a name="toc-write_002dshelx_002dins_002dfile-1" href="#write_002dshelx_002dins_002dfile">11.56.2 write-shelx-ins-file</a></li>
    </ul></li>
    <li><a name="toc-Validation-Functions-1" href="#Validation-Functions">11.57 Validation Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-difference_002dmap_002dpeaks-1" href="#difference_002dmap_002dpeaks">11.57.1 difference-map-peaks</a></li>
      <li><a name="toc-gln_002dasn_002db_002dfactor_002doutliers-1" href="#gln_002dasn_002db_002dfactor_002doutliers">11.57.2 gln-asn-b-factor-outliers</a></li>
      <li><a name="toc-map_002dpeaks_002dpy-1" href="#map_002dpeaks_002dpy">11.57.3 map-peaks-py</a></li>
      <li><a name="toc-map_002dpeaks_002dscm-1" href="#map_002dpeaks_002dscm">11.57.4 map-peaks-scm</a></li>
      <li><a name="toc-set_002dtorsion_002dscm-1" href="#set_002dtorsion_002dscm">11.57.5 set-torsion-scm</a></li>
      <li><a name="toc-multi_002dresidue_002dtorsion_002dscm-1" href="#multi_002dresidue_002dtorsion_002dscm">11.57.6 multi-residue-torsion-scm</a></li>
      <li><a name="toc-set_002dtorsion_002dpy-1" href="#set_002dtorsion_002dpy">11.57.7 set-torsion-py</a></li>
      <li><a name="toc-multi_002dresidue_002dtorsion_002dpy-1" href="#multi_002dresidue_002dtorsion_002dpy">11.57.8 multi-residue-torsion-py</a></li>
    </ul></li>
    <li><a name="toc-Ramachandran-Plot-Functions-1" href="#Ramachandran-Plot-Functions">11.58 Ramachandran Plot Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-do_002dramachandran_002dplot-1" href="#do_002dramachandran_002dplot">11.58.1 do-ramachandran-plot</a></li>
      <li><a name="toc-set_002dkleywegt_002dplot_002dn_002ddiffs-1" href="#set_002dkleywegt_002dplot_002dn_002ddiffs">11.58.2 set-kleywegt-plot-n-diffs</a></li>
      <li><a name="toc-set_002dramachandran_002dplot_002dcontour_002dlevels-1" href="#set_002dramachandran_002dplot_002dcontour_002dlevels">11.58.3 set-ramachandran-plot-contour-levels</a></li>
      <li><a name="toc-set_002dramachandran_002dplot_002dbackground_002dblock_002dsize-1" href="#set_002dramachandran_002dplot_002dbackground_002dblock_002dsize">11.58.4 set-ramachandran-plot-background-block-size</a></li>
      <li><a name="toc-set_002dramachandran_002dpsi_002daxis_002dmode-1" href="#set_002dramachandran_002dpsi_002daxis_002dmode">11.58.5 set-ramachandran-psi-axis-mode</a></li>
      <li><a name="toc-ramachandran_002dplot_002ddifferences-1" href="#ramachandran_002dplot_002ddifferences">11.58.6 ramachandran-plot-differences</a></li>
      <li><a name="toc-ramachandran_002dplot_002ddifferences_002dby_002dchain-1" href="#ramachandran_002dplot_002ddifferences_002dby_002dchain">11.58.7 ramachandran-plot-differences-by-chain</a></li>
    </ul></li>
    <li><a name="toc-Sequence-View-Interface-1" href="#Sequence-View-Interface">11.59 Sequence View Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-do_002dsequence_002dview-1" href="#do_002dsequence_002dview">11.59.1 do-sequence-view</a></li>
    </ul></li>
    <li><a name="toc-Atom-Labelling-1" href="#Atom-Labelling">11.60 Atom Labelling</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dbrief_002datom_002dlabels-1" href="#set_002dbrief_002datom_002dlabels">11.60.1 set-brief-atom-labels</a></li>
      <li><a name="toc-brief_002datom_002dlabels_002dstate-1" href="#brief_002datom_002dlabels_002dstate">11.60.2 brief-atom-labels-state</a></li>
      <li><a name="toc-set_002dseg_002dids_002din_002datom_002dlabels-1" href="#set_002dseg_002dids_002din_002datom_002dlabels">11.60.3 set-seg-ids-in-atom-labels</a></li>
    </ul></li>
    <li><a name="toc-Screen-Rotation-1" href="#Screen-Rotation">11.61 Screen Rotation</a>
    <ul class="no-bullet">
      <li><a name="toc-rotate_002dy_002dscene-1" href="#rotate_002dy_002dscene">11.61.1 rotate-y-scene</a></li>
      <li><a name="toc-rotate_002dx_002dscene-1" href="#rotate_002dx_002dscene">11.61.2 rotate-x-scene</a></li>
      <li><a name="toc-rotate_002dz_002dscene-1" href="#rotate_002dz_002dscene">11.61.3 rotate-z-scene</a></li>
      <li><a name="toc-spin_002dzoom_002dtrans-1" href="#spin_002dzoom_002dtrans">11.61.4 spin-zoom-trans</a></li>
    </ul></li>
    <li><a name="toc-Screen-Translation-1" href="#Screen-Translation">11.62 Screen Translation</a>
    <ul class="no-bullet">
      <li><a name="toc-translate_002dscene_002dx-1" href="#translate_002dscene_002dx">11.62.1 translate-scene-x</a></li>
      <li><a name="toc-translate_002dscene_002dy-1" href="#translate_002dscene_002dy">11.62.2 translate-scene-y</a></li>
      <li><a name="toc-translate_002dscene_002dz-1" href="#translate_002dscene_002dz">11.62.3 translate-scene-z</a></li>
    </ul></li>
    <li><a name="toc-Views-Interface-1" href="#Views-Interface">11.63 Views Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-add_002dview_002dhere-1" href="#add_002dview_002dhere">11.63.1 add-view-here</a></li>
      <li><a name="toc-add_002dview_002draw-1" href="#add_002dview_002draw">11.63.2 add-view-raw</a></li>
      <li><a name="toc-remove_002dnamed_002dview-1" href="#remove_002dnamed_002dview">11.63.3 remove-named-view</a></li>
      <li><a name="toc-remove_002dview-1" href="#remove_002dview">11.63.4 remove-view</a></li>
      <li><a name="toc-add_002dview_002ddescription-1" href="#add_002dview_002ddescription">11.63.5 add-view-description</a></li>
      <li><a name="toc-add_002daction_002dview-1" href="#add_002daction_002dview">11.63.6 add-action-view</a></li>
      <li><a name="toc-insert_002daction_002dview_002dafter_002dview-1" href="#insert_002daction_002dview_002dafter_002dview">11.63.7 insert-action-view-after-view</a></li>
      <li><a name="toc-save_002dviews-1" href="#save_002dviews">11.63.8 save-views</a></li>
      <li><a name="toc-view_002dname-1" href="#view_002dname">11.63.9 view-name</a></li>
      <li><a name="toc-view_002dname_002dpy-1" href="#view_002dname_002dpy">11.63.10 view-name-py</a></li>
      <li><a name="toc-clear_002dall_002dviews-1" href="#clear_002dall_002dviews">11.63.11 clear-all-views</a></li>
    </ul></li>
    <li><a name="toc-Background-Colour-1" href="#Background-Colour">11.64 Background Colour</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dbackground_002dcolour-1" href="#set_002dbackground_002dcolour">11.64.1 set-background-colour</a></li>
      <li><a name="toc-redraw_002dbackground-1" href="#redraw_002dbackground">11.64.2 redraw-background</a></li>
      <li><a name="toc-background_002dis_002dblack_002dp-1" href="#background_002dis_002dblack_002dp">11.64.3 background-is-black-p</a></li>
    </ul></li>
    <li><a name="toc-Ligand-Fitting-Functions-1" href="#Ligand-Fitting-Functions">11.65 Ligand Fitting Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dligand_002dacceptable_002dfit_002dfraction-1" href="#set_002dligand_002dacceptable_002dfit_002dfraction">11.65.1 set-ligand-acceptable-fit-fraction</a></li>
      <li><a name="toc-set_002dligand_002dcluster_002dsigma_002dlevel-1" href="#set_002dligand_002dcluster_002dsigma_002dlevel">11.65.2 set-ligand-cluster-sigma-level</a></li>
      <li><a name="toc-set_002dligand_002dflexible_002dligand_002dn_002dsamples-1" href="#set_002dligand_002dflexible_002dligand_002dn_002dsamples">11.65.3 set-ligand-flexible-ligand-n-samples</a></li>
      <li><a name="toc-set_002dfind_002dligand_002dn_002dtop_002dligands-1" href="#set_002dfind_002dligand_002dn_002dtop_002dligands">11.65.4 set-find-ligand-n-top-ligands</a></li>
      <li><a name="toc-set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster-1" href="#set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster">11.65.5 set-find-ligand-multi-solutions-per-cluster</a></li>
      <li><a name="toc-set_002dfind_002dligand_002dmask_002dwaters-1" href="#set_002dfind_002dligand_002dmask_002dwaters">11.65.6 set-find-ligand-mask-waters</a></li>
      <li><a name="toc-set_002dligand_002dsearch_002dprotein_002dmolecule-1" href="#set_002dligand_002dsearch_002dprotein_002dmolecule">11.65.7 set-ligand-search-protein-molecule</a></li>
      <li><a name="toc-set_002dligand_002dsearch_002dmap_002dmolecule-1" href="#set_002dligand_002dsearch_002dmap_002dmolecule">11.65.8 set-ligand-search-map-molecule</a></li>
      <li><a name="toc-add_002dligand_002dsearch_002dligand_002dmolecule-1" href="#add_002dligand_002dsearch_002dligand_002dmolecule">11.65.9 add-ligand-search-ligand-molecule</a></li>
      <li><a name="toc-add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule-1" href="#add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule">11.65.10 add-ligand-search-wiggly-ligand-molecule</a></li>
      <li><a name="toc-set_002dfind_002dligand_002dhere_002dcluster-1" href="#set_002dfind_002dligand_002dhere_002dcluster">11.65.11 set-find-ligand-here-cluster</a></li>
      <li><a name="toc-ligand_002dexpert-1" href="#ligand_002dexpert">11.65.12 ligand-expert</a></li>
      <li><a name="toc-do_002dfind_002dligands_002ddialog-1" href="#do_002dfind_002dligands_002ddialog">11.65.13 do-find-ligands-dialog</a></li>
      <li><a name="toc-overlap_002dligands-1" href="#overlap_002dligands">11.65.14 overlap-ligands</a></li>
      <li><a name="toc-match_002dligand_002datom_002dnames-1" href="#match_002dligand_002datom_002dnames">11.65.15 match-ligand-atom-names</a></li>
      <li><a name="toc-match_002dligand_002datom_002dnames_002dto_002dcomp_002did-1" href="#match_002dligand_002datom_002dnames_002dto_002dcomp_002did">11.65.16 match-ligand-atom-names-to-comp-id</a></li>
      <li><a name="toc-flip_002dligand-1" href="#flip_002dligand">11.65.17 flip-ligand</a></li>
    </ul></li>
    <li><a name="toc-Water-Fitting-Functions-1" href="#Water-Fitting-Functions">11.66 Water Fitting Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-wrapped_002dcreate_002dfind_002dwaters_002ddialog-1" href="#wrapped_002dcreate_002dfind_002dwaters_002ddialog">11.66.1 wrapped-create-find-waters-dialog</a></li>
      <li><a name="toc-renumber_002dwaters-1" href="#renumber_002dwaters">11.66.2 renumber-waters</a></li>
      <li><a name="toc-execute_002dfind_002dwaters_002dreal-1" href="#execute_002dfind_002dwaters_002dreal">11.66.3 execute-find-waters-real</a></li>
      <li><a name="toc-move_002dwaters_002dto_002daround_002dprotein-1" href="#move_002dwaters_002dto_002daround_002dprotein">11.66.4 move-waters-to-around-protein</a></li>
      <li><a name="toc-move_002dhetgroups_002dto_002daround_002dprotein-1" href="#move_002dhetgroups_002dto_002daround_002dprotein">11.66.5 move-hetgroups-to-around-protein</a></li>
      <li><a name="toc-max_002dwater_002ddistance-1" href="#max_002dwater_002ddistance">11.66.6 max-water-distance</a></li>
      <li><a name="toc-set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit-1" href="#set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit">11.66.7 set-water-check-spherical-variance-limit</a></li>
      <li><a name="toc-set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits-1" href="#set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits">11.66.8 set-ligand-water-to-protein-distance-limits</a></li>
      <li><a name="toc-set_002dligand_002dwater_002dn_002dcycles-1" href="#set_002dligand_002dwater_002dn_002dcycles">11.66.9 set-ligand-water-n-cycles</a></li>
      <li><a name="toc-execute_002dfind_002dblobs-1" href="#execute_002dfind_002dblobs">11.66.10 execute-find-blobs</a></li>
      <li><a name="toc-split_002dwater-1" href="#split_002dwater">11.66.11 split-water</a></li>
    </ul></li>
    <li><a name="toc-Bond-Representation-1" href="#Bond-Representation">11.67 Bond Representation</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002ddefault_002dbond_002dthickness-1" href="#set_002ddefault_002dbond_002dthickness">11.67.1 set-default-bond-thickness</a></li>
      <li><a name="toc-set_002dbond_002dthickness-1" href="#set_002dbond_002dthickness">11.67.2 set-bond-thickness</a></li>
      <li><a name="toc-set_002dbond_002dthickness_002dintermediate_002datoms-1" href="#set_002dbond_002dthickness_002dintermediate_002datoms">11.67.3 set-bond-thickness-intermediate-atoms</a></li>
      <li><a name="toc-set_002duse_002dvariable_002dbond_002dthickness-1" href="#set_002duse_002dvariable_002dbond_002dthickness">11.67.4 set-use-variable-bond-thickness</a></li>
      <li><a name="toc-set_002dbond_002dcolour_002drotation_002dfor_002dmolecule-1" href="#set_002dbond_002dcolour_002drotation_002dfor_002dmolecule">11.67.5 set-bond-colour-rotation-for-molecule</a></li>
      <li><a name="toc-set_002ddraw_002dstick_002dmode_002datoms_002ddefault-1" href="#set_002ddraw_002dstick_002dmode_002datoms_002ddefault">11.67.6 set-draw-stick-mode-atoms-default</a></li>
      <li><a name="toc-get_002dbond_002dcolour_002drotation_002dfor_002dmolecule-1" href="#get_002dbond_002dcolour_002drotation_002dfor_002dmolecule">11.67.7 get-bond-colour-rotation-for-molecule</a></li>
      <li><a name="toc-set_002ddefault_002drepresentation_002dtype-1" href="#set_002ddefault_002drepresentation_002dtype">11.67.8 set-default-representation-type</a></li>
      <li><a name="toc-get_002ddefault_002dbond_002dthickness-1" href="#get_002ddefault_002dbond_002dthickness">11.67.9 get-default-bond-thickness</a></li>
      <li><a name="toc-set_002ddraw_002dzero_002docc_002dmarkers-1" href="#set_002ddraw_002dzero_002docc_002dmarkers">11.67.10 set-draw-zero-occ-markers</a></li>
      <li><a name="toc-set_002ddraw_002dcis_002dpeptide_002dmarkups-1" href="#set_002ddraw_002dcis_002dpeptide_002dmarkups">11.67.11 set-draw-cis-peptide-markups</a></li>
      <li><a name="toc-set_002ddraw_002dhydrogens-1" href="#set_002ddraw_002dhydrogens">11.67.12 set-draw-hydrogens</a></li>
      <li><a name="toc-draw_002dhydrogens_002dstate-1" href="#draw_002dhydrogens_002dstate">11.67.13 draw-hydrogens-state</a></li>
      <li><a name="toc-set_002ddraw_002dstick_002dmode_002datoms-1" href="#set_002ddraw_002dstick_002dmode_002datoms">11.67.14 set-draw-stick-mode-atoms</a></li>
      <li><a name="toc-set_002ddraw_002dmissing_002dresidues_002dloops-1" href="#set_002ddraw_002dmissing_002dresidues_002dloops">11.67.15 set-draw-missing-residues-loops</a></li>
      <li><a name="toc-graphics_002dto_002dca_002drepresentation-1" href="#graphics_002dto_002dca_002drepresentation">11.67.16 graphics-to-ca-representation</a></li>
      <li><a name="toc-graphics_002dto_002dca_002dplus_002dligands_002drepresentation-1" href="#graphics_002dto_002dca_002dplus_002dligands_002drepresentation">11.67.17 graphics-to-ca-plus-ligands-representation</a></li>
      <li><a name="toc-graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation-1" href="#graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation">11.67.18 graphics-to-ca-plus-ligands-and-sidechains-representation</a></li>
      <li><a name="toc-graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation-1" href="#graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation">11.67.19 graphics-to-bonds-no-waters-representation</a></li>
      <li><a name="toc-graphics_002dto_002dbonds_002drepresentation-1" href="#graphics_002dto_002dbonds_002drepresentation">11.67.20 graphics-to-bonds-representation</a></li>
      <li><a name="toc-graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation-1" href="#graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation">11.67.21 graphics-to-ca-plus-ligands-sec-struct-representation</a></li>
      <li><a name="toc-graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation-1" href="#graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation">11.67.22 graphics-to-sec-struct-bonds-representation</a></li>
      <li><a name="toc-graphics_002dto_002drainbow_002drepresentation-1" href="#graphics_002dto_002drainbow_002drepresentation">11.67.23 graphics-to-rainbow-representation</a></li>
      <li><a name="toc-graphics_002dto_002db_002dfactor_002drepresentation-1" href="#graphics_002dto_002db_002dfactor_002drepresentation">11.67.24 graphics-to-b-factor-representation</a></li>
      <li><a name="toc-graphics_002dto_002db_002dfactor_002dcas_002drepresentation-1" href="#graphics_002dto_002db_002dfactor_002dcas_002drepresentation">11.67.25 graphics-to-b-factor-cas-representation</a></li>
      <li><a name="toc-graphics_002dto_002doccupancy_002drepresentation-1" href="#graphics_002dto_002doccupancy_002drepresentation">11.67.26 graphics-to-occupancy-representation</a></li>
      <li><a name="toc-graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation-1" href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation">11.67.27 graphics-to-user-defined-atom-colours-representation</a></li>
      <li><a name="toc-graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation-1" href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation">11.67.28 graphics-to-user-defined-atom-colours-all-atoms-representation</a></li>
      <li><a name="toc-graphics_002dmolecule_002dbond_002dtype-1" href="#graphics_002dmolecule_002dbond_002dtype">11.67.29 graphics-molecule-bond-type</a></li>
      <li><a name="toc-set_002db_002dfactor_002dbonds_002dscale_002dfactor-1" href="#set_002db_002dfactor_002dbonds_002dscale_002dfactor">11.67.30 set-b-factor-bonds-scale-factor</a></li>
      <li><a name="toc-change_002dmodel_002dmolecule_002drepresentation_002dmode-1" href="#change_002dmodel_002dmolecule_002drepresentation_002dmode">11.67.31 change-model-molecule-representation-mode</a></li>
      <li><a name="toc-set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule-1" href="#set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule">11.67.32 set-use-grey-carbons-for-molecule</a></li>
      <li><a name="toc-set_002dgrey_002dcarbon_002dcolour-1" href="#set_002dgrey_002dcarbon_002dcolour">11.67.33 set-grey-carbon-colour</a></li>
      <li><a name="toc-make_002dball_002dand_002dstick-1" href="#make_002dball_002dand_002dstick">11.67.34 make-ball-and-stick</a></li>
      <li><a name="toc-clear_002dball_002dand_002dstick-1" href="#clear_002dball_002dand_002dstick">11.67.35 clear-ball-and-stick</a></li>
      <li><a name="toc-additional_002drepresentation_002dby_002dstring-1" href="#additional_002drepresentation_002dby_002dstring">11.67.36 additional-representation-by-string</a></li>
      <li><a name="toc-additional_002drepresentation_002dby_002dattributes-1" href="#additional_002drepresentation_002dby_002dattributes">11.67.37 additional-representation-by-attributes</a></li>
    </ul></li>
    <li><a name="toc-Dots-Representation-1" href="#Dots-Representation">11.68 Dots Representation</a>
    <ul class="no-bullet">
      <li><a name="toc-dots-1" href="#dots">11.68.1 dots</a></li>
      <li><a name="toc-set_002ddots_002dcolour-1" href="#set_002ddots_002dcolour">11.68.2 set-dots-colour</a></li>
      <li><a name="toc-unset_002ddots_002dcolour-1" href="#unset_002ddots_002dcolour">11.68.3 unset-dots-colour</a></li>
      <li><a name="toc-clear_002ddots-1" href="#clear_002ddots">11.68.4 clear-dots</a></li>
      <li><a name="toc-clear_002ddots_002dby_002dname-1" href="#clear_002ddots_002dby_002dname">11.68.5 clear-dots-by-name</a></li>
      <li><a name="toc-n_002ddots_002dsets-1" href="#n_002ddots_002dsets">11.68.6 n-dots-sets</a></li>
    </ul></li>
    <li><a name="toc-Pep_002dflip-Interface-1" href="#Pep_002dflip-Interface">11.69 Pep-flip Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-pepflip-1" href="#pepflip">11.69.1 pepflip</a></li>
    </ul></li>
    <li><a name="toc-Rigid-Body-Refinement-Interface-1" href="#Rigid-Body-Refinement-Interface">11.70 Rigid Body Refinement Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-rigid_002dbody_002drefine_002dzone-1" href="#rigid_002dbody_002drefine_002dzone">11.70.1 rigid-body-refine-zone</a></li>
      <li><a name="toc-rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm-1" href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm">11.70.2 rigid-body-refine-by-residue-ranges-scm</a></li>
      <li><a name="toc-rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy-1" href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy">11.70.3 rigid-body-refine-by-residue-ranges-py</a></li>
      <li><a name="toc-set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction-1" href="#set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction">11.70.4 set-rigid-body-fit-acceptable-fit-fraction</a></li>
    </ul></li>
    <li><a name="toc-Add-Terminal-Residue-Functions-1" href="#Add-Terminal-Residue-Functions">11.71 Add Terminal Residue Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition-1" href="#set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition">11.71.1 set-add-terminal-residue-immediate-addition</a></li>
      <li><a name="toc-add_002dterminal_002dresidue-1" href="#add_002dterminal_002dresidue">11.71.2 add-terminal-residue</a></li>
      <li><a name="toc-add_002dnucleotide-1" href="#add_002dnucleotide">11.71.3 add-nucleotide</a></li>
      <li><a name="toc-add_002dterminal_002dresidue_002dusing_002dphi_002dpsi-1" href="#add_002dterminal_002dresidue_002dusing_002dphi_002dpsi">11.71.4 add-terminal-residue-using-phi-psi</a></li>
      <li><a name="toc-set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype-1" href="#set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype">11.71.5 set-add-terminal-residue-default-residue-type</a></li>
      <li><a name="toc-set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine-1" href="#set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine">11.71.6 set-add-terminal-residue-do-post-refine</a></li>
      <li><a name="toc-add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate-1" href="#add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate">11.71.7 add-terminal-residue-do-post-refine-state</a></li>
    </ul></li>
    <li><a name="toc-Delete-Residues-1" href="#Delete-Residues">11.72 Delete Residues</a>
    <ul class="no-bullet">
      <li><a name="toc-delete_002dresidue_002drange-1" href="#delete_002dresidue_002drange">11.72.1 delete-residue-range</a></li>
      <li><a name="toc-delete_002dresidue-1" href="#delete_002dresidue">11.72.2 delete-residue</a></li>
      <li><a name="toc-delete_002dresidue_002dwith_002dfull_002dspec-1" href="#delete_002dresidue_002dwith_002dfull_002dspec">11.72.3 delete-residue-with-full-spec</a></li>
      <li><a name="toc-delete_002dresidues_002dscm-1" href="#delete_002dresidues_002dscm">11.72.4 delete-residues-scm</a></li>
      <li><a name="toc-delete_002dresidues_002dpy-1" href="#delete_002dresidues_002dpy">11.72.5 delete-residues-py</a></li>
      <li><a name="toc-delete_002dresidue_002dhydrogens-1" href="#delete_002dresidue_002dhydrogens">11.72.6 delete-residue-hydrogens</a></li>
      <li><a name="toc-delete_002datom-1" href="#delete_002datom">11.72.7 delete-atom</a></li>
      <li><a name="toc-delete_002dresidue_002dsidechain-1" href="#delete_002dresidue_002dsidechain">11.72.8 delete-residue-sidechain</a></li>
      <li><a name="toc-delete_002dhydrogen_002datoms-1" href="#delete_002dhydrogen_002datoms">11.72.9 delete-hydrogen-atoms</a></li>
      <li><a name="toc-delete_002dhydrogens-1" href="#delete_002dhydrogens">11.72.10 delete-hydrogens</a></li>
      <li><a name="toc-delete_002dwaters-1" href="#delete_002dwaters">11.72.11 delete-waters</a></li>
      <li><a name="toc-delete_002dchain-1" href="#delete_002dchain">11.72.12 delete-chain</a></li>
      <li><a name="toc-delete_002dsidechains_002dfor_002dchain-1" href="#delete_002dsidechains_002dfor_002dchain">11.72.13 delete-sidechains-for-chain</a></li>
    </ul></li>
    <li><a name="toc-Mainchain-Building-Functions-1" href="#Mainchain-Building-Functions">11.73 Mainchain Building Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-db_002dmainchain-1" href="#db_002dmainchain">11.73.1 db-mainchain</a></li>
      <li><a name="toc-db_002dmainchains_002dfragment-1" href="#db_002dmainchains_002dfragment">11.73.2 db-mainchains-fragment</a></li>
    </ul></li>
    <li><a name="toc-Close-Molecule-Functions-1" href="#Close-Molecule-Functions">11.74 Close Molecule Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-close_002dmolecule-1" href="#close_002dmolecule">11.74.1 close-molecule</a></li>
    </ul></li>
    <li><a name="toc-Rotamer-Functions-1" href="#Rotamer-Functions">11.75 Rotamer Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002drotamer_002dsearch_002dmode-1" href="#set_002drotamer_002dsearch_002dmode">11.75.1 set-rotamer-search-mode</a></li>
      <li><a name="toc-set_002drotamer_002dlowest_002dprobability-1" href="#set_002drotamer_002dlowest_002dprobability">11.75.2 set-rotamer-lowest-probability</a></li>
      <li><a name="toc-set_002drotamer_002dcheck_002dclashes-1" href="#set_002drotamer_002dcheck_002dclashes">11.75.3 set-rotamer-check-clashes</a></li>
      <li><a name="toc-auto_002dfit_002dbest_002drotamer-1" href="#auto_002dfit_002dbest_002drotamer">11.75.4 auto-fit-best-rotamer</a></li>
      <li><a name="toc-set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag-1" href="#set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag">11.75.5 set-auto-fit-best-rotamer-clash-flag</a></li>
      <li><a name="toc-n_002drotamers-1" href="#n_002drotamers">11.75.6 n-rotamers</a></li>
      <li><a name="toc-set_002dresidue_002dto_002drotamer_002dnumber-1" href="#set_002dresidue_002dto_002drotamer_002dnumber">11.75.7 set-residue-to-rotamer-number</a></li>
      <li><a name="toc-set_002dresidue_002dto_002drotamer_002dname-1" href="#set_002dresidue_002dto_002drotamer_002dname">11.75.8 set-residue-to-rotamer-name</a></li>
      <li><a name="toc-fill_002dpartial_002dresidues-1" href="#fill_002dpartial_002dresidues">11.75.9 fill-partial-residues</a></li>
      <li><a name="toc-simple_002dfill_002dpartial_002dresidues-1" href="#simple_002dfill_002dpartial_002dresidues">11.75.10 simple-fill-partial-residues</a></li>
      <li><a name="toc-rotamer_002dgraphs-1" href="#rotamer_002dgraphs">11.75.11 rotamer-graphs</a></li>
      <li><a name="toc-rotamer_002dgraphs_002dpy-1" href="#rotamer_002dgraphs_002dpy">11.75.12 rotamer-graphs-py</a></li>
    </ul></li>
    <li><a name="toc-180-Flip-Side-chain-1" href="#g_t180-Flip-Side-chain">11.76 180 Flip Side chain</a>
    <ul class="no-bullet">
      <li><a name="toc-do_002d180_002ddegree_002dside_002dchain_002dflip-1" href="#do_002d180_002ddegree_002dside_002dchain_002dflip">11.76.1 do-180-degree-side-chain-flip</a></li>
    </ul></li>
    <li><a name="toc-Mutate-Functions-1" href="#Mutate-Functions">11.77 Mutate Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-setup_002dmutate_002dauto_002dfit-1" href="#setup_002dmutate_002dauto_002dfit">11.77.1 setup-mutate-auto-fit</a></li>
      <li><a name="toc-mutate_002dactive_002dresidue-1" href="#mutate_002dactive_002dresidue">11.77.2 mutate-active-residue</a></li>
      <li><a name="toc-mutate-1" href="#mutate">11.77.3 mutate</a></li>
      <li><a name="toc-mutate_002dbase-1" href="#mutate_002dbase">11.77.4 mutate-base</a></li>
      <li><a name="toc-set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine-1" href="#set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine">11.77.5 set-mutate-auto-fit-do-post-refine</a></li>
      <li><a name="toc-mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate-1" href="#mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate">11.77.6 mutate-auto-fit-do-post-refine-state</a></li>
      <li><a name="toc-set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine-1" href="#set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine">11.77.7 set-rotamer-auto-fit-do-post-refine</a></li>
      <li><a name="toc-rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate-1" href="#rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate">11.77.8 rotamer-auto-fit-do-post-refine-state</a></li>
      <li><a name="toc-mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber-1" href="#mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber">11.77.9 mutate-single-residue-by-serial-number</a></li>
      <li><a name="toc-set_002dresidue_002dtype_002dchooser_002dstub_002dstate-1" href="#set_002dresidue_002dtype_002dchooser_002dstub_002dstate">11.77.10 set-residue-type-chooser-stub-state</a></li>
    </ul></li>
    <li><a name="toc-Alternative-Conformation-1" href="#Alternative-Conformation">11.78 Alternative Conformation</a>
    <ul class="no-bullet">
      <li><a name="toc-add_002dalt_002dconf_002dscm-1" href="#add_002dalt_002dconf_002dscm">11.78.1 add-alt-conf-scm</a></li>
      <li><a name="toc-add_002dalt_002dconf_002dpy-1" href="#add_002dalt_002dconf_002dpy">11.78.2 add-alt-conf-py</a></li>
    </ul></li>
    <li><a name="toc-Pointer-Atom-Functions-1" href="#Pointer-Atom-Functions">11.79 Pointer Atom Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-create_002dpointer_002datom_002dmolecule_002dmaybe-1" href="#create_002dpointer_002datom_002dmolecule_002dmaybe">11.79.1 create-pointer-atom-molecule-maybe</a></li>
      <li><a name="toc-pointer_002datom_002dmolecule-1" href="#pointer_002datom_002dmolecule">11.79.2 pointer-atom-molecule</a></li>
    </ul></li>
    <li><a name="toc-Baton-Build-Interface-Functions-1" href="#Baton-Build-Interface-Functions">11.80 Baton Build Interface Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dbaton_002dmode-1" href="#set_002dbaton_002dmode">11.80.1 set-baton-mode</a></li>
      <li><a name="toc-try_002dset_002ddraw_002dbaton-1" href="#try_002dset_002ddraw_002dbaton">11.80.2 try-set-draw-baton</a></li>
      <li><a name="toc-accept_002dbaton_002dposition-1" href="#accept_002dbaton_002dposition">11.80.3 accept-baton-position</a></li>
      <li><a name="toc-baton_002dtip_002dtry_002danother-1" href="#baton_002dtip_002dtry_002danother">11.80.4 baton-tip-try-another</a></li>
      <li><a name="toc-baton_002dtip_002dprevious-1" href="#baton_002dtip_002dprevious">11.80.5 baton-tip-previous</a></li>
      <li><a name="toc-shorten_002dbaton-1" href="#shorten_002dbaton">11.80.6 shorten-baton</a></li>
      <li><a name="toc-lengthen_002dbaton-1" href="#lengthen_002dbaton">11.80.7 lengthen-baton</a></li>
      <li><a name="toc-baton_002dbuild_002ddelete_002dlast_002dresidue-1" href="#baton_002dbuild_002ddelete_002dlast_002dresidue">11.80.8 baton-build-delete-last-residue</a></li>
      <li><a name="toc-set_002dbaton_002dbuild_002dparams-1" href="#set_002dbaton_002dbuild_002dparams">11.80.9 set-baton-build-params</a></li>
    </ul></li>
    <li><a name="toc-Crosshairs-Interface-1" href="#Crosshairs-Interface">11.81 Crosshairs  Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002ddraw_002dcrosshairs-1" href="#set_002ddraw_002dcrosshairs">11.81.1 set-draw-crosshairs</a></li>
    </ul></li>
    <li><a name="toc-Edit-Chi-Angles-1" href="#Edit-Chi-Angles">11.82 Edit Chi Angles</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dfind_002dhydrogen_002dtorsions-1" href="#set_002dfind_002dhydrogen_002dtorsions">11.82.1 set-find-hydrogen-torsions</a></li>
      <li><a name="toc-edit_002dchi_002dangles-1" href="#edit_002dchi_002dangles">11.82.2 edit-chi-angles</a></li>
      <li><a name="toc-setup_002dtorsion_002dgeneral-1" href="#setup_002dtorsion_002dgeneral">11.82.3 setup-torsion-general</a></li>
    </ul></li>
    <li><a name="toc-Masks-2" href="#Masks">11.83 Masks</a>
    <ul class="no-bullet">
      <li><a name="toc-mask_002dmap_002dby_002dmolecule-1" href="#mask_002dmap_002dby_002dmolecule">11.83.1 mask-map-by-molecule</a></li>
      <li><a name="toc-mask_002dmap_002dby_002datom_002dselection-1" href="#mask_002dmap_002dby_002datom_002dselection">11.83.2 mask-map-by-atom-selection</a></li>
      <li><a name="toc-make_002dmasked_002dmaps_002dsplit_002dby_002dchain-1" href="#make_002dmasked_002dmaps_002dsplit_002dby_002dchain">11.83.3 make-masked-maps-split-by-chain</a></li>
      <li><a name="toc-set_002dmap_002dmask_002datom_002dradius-1" href="#set_002dmap_002dmask_002datom_002dradius">11.83.4 set-map-mask-atom-radius</a></li>
      <li><a name="toc-map_002dmask_002datom_002dradius-1" href="#map_002dmask_002datom_002dradius">11.83.5 map-mask-atom-radius</a></li>
    </ul></li>
    <li><a name="toc-check-Waters-Interface-1" href="#check-Waters-Interface">11.84 check Waters Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-delete_002dchecked_002dwaters_002dbaddies-1" href="#delete_002dchecked_002dwaters_002dbaddies">11.84.1 delete-checked-waters-baddies</a></li>
      <li><a name="toc-highly_002dcoordinated_002dwaters_002dscm-1" href="#highly_002dcoordinated_002dwaters_002dscm">11.84.2 highly-coordinated-waters-scm</a></li>
      <li><a name="toc-highly_002dcoordinated_002dwaters_002dpy-1" href="#highly_002dcoordinated_002dwaters_002dpy">11.84.3 highly-coordinated-waters-py</a></li>
    </ul></li>
    <li><a name="toc-Least_002dSquares-matching-1" href="#Least_002dSquares-matching">11.85 Least-Squares matching</a>
    <ul class="no-bullet">
      <li><a name="toc-apply_002dlsq_002dmatches-1" href="#apply_002dlsq_002dmatches">11.85.1 apply-lsq-matches</a></li>
    </ul></li>
    <li><a name="toc-Trim-1" href="#Trim">11.86 Trim</a>
    <ul class="no-bullet">
      <li><a name="toc-trim_002dmolecule_002dby_002dmap-1" href="#trim_002dmolecule_002dby_002dmap">11.86.1 trim-molecule-by-map</a></li>
    </ul></li>
    <li><a name="toc-External-Ray_002dTracing-1" href="#External-Ray_002dTracing">11.87 External Ray-Tracing</a>
    <ul class="no-bullet">
      <li><a name="toc-raster3d-1" href="#raster3d">11.87.1 raster3d</a></li>
      <li><a name="toc-set_002draster3d_002dbond_002dthickness-1" href="#set_002draster3d_002dbond_002dthickness">11.87.2 set-raster3d-bond-thickness</a></li>
      <li><a name="toc-set_002draster3d_002datom_002dradius-1" href="#set_002draster3d_002datom_002dradius">11.87.3 set-raster3d-atom-radius</a></li>
      <li><a name="toc-set_002draster3d_002ddensity_002dthickness-1" href="#set_002draster3d_002ddensity_002dthickness">11.87.4 set-raster3d-density-thickness</a></li>
      <li><a name="toc-set_002drenderer_002dshow_002datoms-1" href="#set_002drenderer_002dshow_002datoms">11.87.5 set-renderer-show-atoms</a></li>
      <li><a name="toc-set_002draster3d_002dbone_002dthickness-1" href="#set_002draster3d_002dbone_002dthickness">11.87.6 set-raster3d-bone-thickness</a></li>
      <li><a name="toc-set_002draster3d_002dshadows_002denabled-1" href="#set_002draster3d_002dshadows_002denabled">11.87.7 set-raster3d-shadows-enabled</a></li>
      <li><a name="toc-set_002draster3d_002dwater_002dsphere-1" href="#set_002draster3d_002dwater_002dsphere">11.87.8 set-raster3d-water-sphere</a></li>
      <li><a name="toc-set_002draster3d_002dfont_002dsize-1" href="#set_002draster3d_002dfont_002dsize">11.87.9 set-raster3d-font-size</a></li>
      <li><a name="toc-raster_002dscreen_002dshot-1" href="#raster_002dscreen_002dshot">11.87.10 raster-screen-shot</a></li>
    </ul></li>
    <li><a name="toc-Superposition-_0028SSM_0029-1" href="#Superposition-_0028SSM_0029">11.88 Superposition (SSM)</a>
    <ul class="no-bullet">
      <li><a name="toc-superpose-1" href="#superpose">11.88.1 superpose</a></li>
      <li><a name="toc-superpose_002dwith_002dchain_002dselection-1" href="#superpose_002dwith_002dchain_002dselection">11.88.2 superpose-with-chain-selection</a></li>
      <li><a name="toc-superpose_002dwith_002datom_002dselection-1" href="#superpose_002dwith_002datom_002dselection">11.88.3 superpose-with-atom-selection</a></li>
    </ul></li>
    <li><a name="toc-NCS-1" href="#NCS">11.89 NCS</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002ddraw_002dncs_002dghosts-1" href="#set_002ddraw_002dncs_002dghosts">11.89.1 set-draw-ncs-ghosts</a></li>
      <li><a name="toc-draw_002dncs_002dghosts_002dstate-1" href="#draw_002dncs_002dghosts_002dstate">11.89.2 draw-ncs-ghosts-state</a></li>
      <li><a name="toc-set_002dncs_002dghost_002dbond_002dthickness-1" href="#set_002dncs_002dghost_002dbond_002dthickness">11.89.3 set-ncs-ghost-bond-thickness</a></li>
      <li><a name="toc-ncs_002dupdate_002dghosts-1" href="#ncs_002dupdate_002dghosts">11.89.4 ncs-update-ghosts</a></li>
      <li><a name="toc-make_002ddynamically_002dtransformed_002dncs_002dmaps-1" href="#make_002ddynamically_002dtransformed_002dncs_002dmaps">11.89.5 make-dynamically-transformed-ncs-maps</a></li>
      <li><a name="toc-add_002dncs_002dmatrix-1" href="#add_002dncs_002dmatrix">11.89.6 add-ncs-matrix</a></li>
      <li><a name="toc-add_002dstrict_002dncs_002dmatrix-1" href="#add_002dstrict_002dncs_002dmatrix">11.89.7 add-strict-ncs-matrix</a></li>
      <li><a name="toc-show_002dstrict_002dncs_002dstate-1" href="#show_002dstrict_002dncs_002dstate">11.89.8 show-strict-ncs-state</a></li>
      <li><a name="toc-set_002dshow_002dstrict_002dncs-1" href="#set_002dshow_002dstrict_002dncs">11.89.9 set-show-strict-ncs</a></li>
      <li><a name="toc-set_002dncs_002dhomology_002dlevel-1" href="#set_002dncs_002dhomology_002dlevel">11.89.10 set-ncs-homology-level</a></li>
      <li><a name="toc-copy_002dchain-1" href="#copy_002dchain">11.89.11 copy-chain</a></li>
      <li><a name="toc-copy_002dfrom_002dncs_002dmaster_002dto_002dothers-1" href="#copy_002dfrom_002dncs_002dmaster_002dto_002dothers">11.89.12 copy-from-ncs-master-to-others</a></li>
      <li><a name="toc-copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers-1" href="#copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers">11.89.13 copy-residue-range-from-ncs-master-to-others</a></li>
      <li><a name="toc-ncs_002dmaster_002dchains_002dscm-1" href="#ncs_002dmaster_002dchains_002dscm">11.89.14 ncs-master-chains-scm</a></li>
      <li><a name="toc-copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm-1" href="#copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm">11.89.15 copy-from-ncs-master-to-chains-scm</a></li>
      <li><a name="toc-ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain-1" href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain">11.89.16 ncs-control-change-ncs-master-to-chain</a></li>
      <li><a name="toc-ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did-1" href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did">11.89.17 ncs-control-change-ncs-master-to-chain-id</a></li>
      <li><a name="toc-ncs_002dcontrol_002ddisplay_002dchain-1" href="#ncs_002dcontrol_002ddisplay_002dchain">11.89.18 ncs-control-display-chain</a></li>
      <li><a name="toc-ncs_002dchain_002dids_002dscm-1" href="#ncs_002dchain_002dids_002dscm">11.89.19 ncs-chain-ids-scm</a></li>
      <li><a name="toc-ncs_002dchain_002dids_002dpy-1" href="#ncs_002dchain_002dids_002dpy">11.89.20 ncs-chain-ids-py</a></li>
      <li><a name="toc-ncs_002dghosts_002dscm-1" href="#ncs_002dghosts_002dscm">11.89.21 ncs-ghosts-scm</a></li>
      <li><a name="toc-ncs_002dghosts_002dpy-1" href="#ncs_002dghosts_002dpy">11.89.22 ncs-ghosts-py</a></li>
    </ul></li>
    <li><a name="toc-Helices-and-Strands-1" href="#Helices-and-Strands">11.90 Helices and Strands</a>
    <ul class="no-bullet">
      <li><a name="toc-place_002dhelix_002dhere-1" href="#place_002dhelix_002dhere">11.90.1 place-helix-here</a></li>
      <li><a name="toc-place_002dstrand_002dhere-1" href="#place_002dstrand_002dhere">11.90.2 place-strand-here</a></li>
      <li><a name="toc-place_002dstrand_002dhere_002ddialog-1" href="#place_002dstrand_002dhere_002ddialog">11.90.3 place-strand-here-dialog</a></li>
      <li><a name="toc-find_002dhelices-1" href="#find_002dhelices">11.90.4 find-helices</a></li>
      <li><a name="toc-find_002dstrands-1" href="#find_002dstrands">11.90.5 find-strands</a></li>
      <li><a name="toc-find_002dsecondary_002dstructure-1" href="#find_002dsecondary_002dstructure">11.90.6 find-secondary-structure</a></li>
      <li><a name="toc-find_002dsecondary_002dstructure_002dlocal-1" href="#find_002dsecondary_002dstructure_002dlocal">11.90.7 find-secondary-structure-local</a></li>
    </ul></li>
    <li><a name="toc-Nucleotides-1" href="#Nucleotides">11.91 Nucleotides</a>
    <ul class="no-bullet">
      <li><a name="toc-find_002dnucleic_002dacids_002dlocal-1" href="#find_002dnucleic_002dacids_002dlocal">11.91.1 find-nucleic-acids-local</a></li>
    </ul></li>
    <li><a name="toc-New-Molecule-by-Section-Interface-1" href="#New-Molecule-by-Section-Interface">11.92 New Molecule by Section Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-new_002dmolecule_002dby_002dresidue_002dtype_002dselection-1" href="#new_002dmolecule_002dby_002dresidue_002dtype_002dselection">11.92.1 new-molecule-by-residue-type-selection</a></li>
      <li><a name="toc-new_002dmolecule_002dby_002datom_002dselection-1" href="#new_002dmolecule_002dby_002datom_002dselection">11.92.2 new-molecule-by-atom-selection</a></li>
      <li><a name="toc-new_002dmolecule_002dby_002dsphere_002dselection-1" href="#new_002dmolecule_002dby_002dsphere_002dselection">11.92.3 new-molecule-by-sphere-selection</a></li>
      <li><a name="toc-new_002dmolecule_002dby_002dresidue_002dspecs_002dpy-1" href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dpy">11.92.4 new-molecule-by-residue-specs-py</a></li>
      <li><a name="toc-new_002dmolecule_002dby_002dresidue_002dspecs_002dscm-1" href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dscm">11.92.5 new-molecule-by-residue-specs-scm</a></li>
    </ul></li>
    <li><a name="toc-RNA_002fDNA-1" href="#RNA_002fDNA">11.93 RNA/DNA</a>
    <ul class="no-bullet">
      <li><a name="toc-ideal_002dnucleic_002dacid-1" href="#ideal_002dnucleic_002dacid">11.93.1 ideal-nucleic-acid</a></li>
      <li><a name="toc-pucker_002dinfo_002dscm-1" href="#pucker_002dinfo_002dscm">11.93.2 pucker-info-scm</a></li>
      <li><a name="toc-pucker_002dinfo_002dpy-1" href="#pucker_002dinfo_002dpy">11.93.3 pucker-info-py</a></li>
      <li><a name="toc-watson_002dcrick_002dpair-1" href="#watson_002dcrick_002dpair">11.93.4 watson-crick-pair</a></li>
      <li><a name="toc-watson_002dcrick_002dpair_002dfor_002dresidue_002drange-1" href="#watson_002dcrick_002dpair_002dfor_002dresidue_002drange">11.93.5 watson-crick-pair-for-residue-range</a></li>
    </ul></li>
    <li><a name="toc-Sequence-_0028Assignment_0029-1" href="#Sequence-_0028Assignment_0029">11.94 Sequence (Assignment)</a>
    <ul class="no-bullet">
      <li><a name="toc-print_002dsequence_002dchain-1" href="#print_002dsequence_002dchain">11.94.1 print-sequence-chain</a></li>
      <li><a name="toc-print_002dsequence_002dchain_002dgeneral-1" href="#print_002dsequence_002dchain_002dgeneral">11.94.2 print-sequence-chain-general</a></li>
      <li><a name="toc-assign_002dfasta_002dsequence-1" href="#assign_002dfasta_002dsequence">11.94.3 assign-fasta-sequence</a></li>
      <li><a name="toc-assign_002dpir_002dsequence-1" href="#assign_002dpir_002dsequence">11.94.4 assign-pir-sequence</a></li>
      <li><a name="toc-assign_002dsequence_002dfrom_002dfile-1" href="#assign_002dsequence_002dfrom_002dfile">11.94.5 assign-sequence-from-file</a></li>
      <li><a name="toc-assign_002dsequence_002dfrom_002dstring-1" href="#assign_002dsequence_002dfrom_002dstring">11.94.6 assign-sequence-from-string</a></li>
      <li><a name="toc-delete_002dall_002dsequences_002dfrom_002dmolecule-1" href="#delete_002dall_002dsequences_002dfrom_002dmolecule">11.94.7 delete-all-sequences-from-molecule</a></li>
      <li><a name="toc-delete_002dsequence_002dby_002dchain_002did-1" href="#delete_002dsequence_002dby_002dchain_002did">11.94.8 delete-sequence-by-chain-id</a></li>
      <li><a name="toc-sequence_002dinfo-1" href="#sequence_002dinfo">11.94.9 sequence-info</a></li>
      <li><a name="toc-alignment_002dmismatches_002dscm-1" href="#alignment_002dmismatches_002dscm">11.94.10 alignment-mismatches-scm</a></li>
      <li><a name="toc-sequence_002dinfo_002dpy-1" href="#sequence_002dinfo_002dpy">11.94.11 sequence-info-py</a></li>
      <li><a name="toc-alignment_002dmismatches_002dpy-1" href="#alignment_002dmismatches_002dpy">11.94.12 alignment-mismatches-py</a></li>
    </ul></li>
    <li><a name="toc-Surface-Interface-1" href="#Surface-Interface">11.95 Surface Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-do_002dsurface-1" href="#do_002dsurface">11.95.1 do-surface</a></li>
      <li><a name="toc-do_002dclipped_002dsurface_002dscm-1" href="#do_002dclipped_002dsurface_002dscm">11.95.2 do-clipped-surface-scm</a></li>
      <li><a name="toc-do_002dclipped_002dsurface_002dpy-1" href="#do_002dclipped_002dsurface_002dpy">11.95.3 do-clipped-surface-py</a></li>
      <li><a name="toc-set_002dtransparent_002delectrostatic_002dsurface-1" href="#set_002dtransparent_002delectrostatic_002dsurface">11.95.4 set-transparent-electrostatic-surface</a></li>
      <li><a name="toc-get_002delectrostatic_002dsurface_002dopacity-1" href="#get_002delectrostatic_002dsurface_002dopacity">11.95.5 get-electrostatic-surface-opacity</a></li>
    </ul></li>
    <li><a name="toc-FFFearing-1" href="#FFFearing">11.96 FFFearing</a>
    <ul class="no-bullet">
      <li><a name="toc-fffear_002dsearch-1" href="#fffear_002dsearch">11.96.1 fffear-search</a></li>
      <li><a name="toc-set_002dfffear_002dangular_002dresolution-1" href="#set_002dfffear_002dangular_002dresolution">11.96.2 set-fffear-angular-resolution</a></li>
      <li><a name="toc-fffear_002dangular_002dresolution-1" href="#fffear_002dangular_002dresolution">11.96.3 fffear-angular-resolution</a></li>
    </ul></li>
    <li><a name="toc-Remote-Control-1" href="#Remote-Control">11.97 Remote Control</a>
    <ul class="no-bullet">
      <li><a name="toc-make_002dsocket_002dlistener_002dmaybe-1" href="#make_002dsocket_002dlistener_002dmaybe">11.97.1 make-socket-listener-maybe</a></li>
      <li><a name="toc-set_002dsocket_002dstring_002dwaiting-1" href="#set_002dsocket_002dstring_002dwaiting">11.97.2 set-socket-string-waiting</a></li>
      <li><a name="toc-set_002dsocket_002dpython_002dstring_002dwaiting-1" href="#set_002dsocket_002dpython_002dstring_002dwaiting">11.97.3 set-socket-python-string-waiting</a></li>
    </ul></li>
    <li><a name="toc-Display-Lists-for-Maps-1" href="#Display-Lists-for-Maps">11.98 Display Lists for Maps</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002ddisplay_002dlists_002dfor_002dmaps-1" href="#set_002ddisplay_002dlists_002dfor_002dmaps">11.98.1 set-display-lists-for-maps</a></li>
      <li><a name="toc-display_002dlists_002dfor_002dmaps_002dstate-1" href="#display_002dlists_002dfor_002dmaps_002dstate">11.98.2 display-lists-for-maps-state</a></li>
    </ul></li>
    <li><a name="toc-Browser-Interface-1" href="#Browser-Interface">11.99 Browser Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-browser_002durl-1" href="#browser_002durl">11.99.1 browser-url</a></li>
      <li><a name="toc-set_002dbrowser_002dinterface-1" href="#set_002dbrowser_002dinterface">11.99.2 set-browser-interface</a></li>
      <li><a name="toc-handle_002donline_002dcoot_002dsearch_002drequest-1" href="#handle_002donline_002dcoot_002dsearch_002drequest">11.99.3 handle-online-coot-search-request</a></li>
    </ul></li>
    <li><a name="toc-Molprobity-Interface-1" href="#Molprobity-Interface">11.100 Molprobity Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-handle_002dread_002ddraw_002dprobe_002ddots-1" href="#handle_002dread_002ddraw_002dprobe_002ddots">11.100.1 handle-read-draw-probe-dots</a></li>
      <li><a name="toc-handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted-1" href="#handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted">11.100.2 handle-read-draw-probe-dots-unformatted</a></li>
      <li><a name="toc-set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis-1" href="#set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis">11.100.3 set-do-probe-dots-on-rotamers-and-chis</a></li>
      <li><a name="toc-do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate-1" href="#do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate">11.100.4 do-probe-dots-on-rotamers-and-chis-state</a></li>
      <li><a name="toc-set_002ddo_002dprobe_002ddots_002dpost_002drefine-1" href="#set_002ddo_002dprobe_002ddots_002dpost_002drefine">11.100.5 set-do-probe-dots-post-refine</a></li>
      <li><a name="toc-do_002dprobe_002ddots_002dpost_002drefine_002dstate-1" href="#do_002dprobe_002ddots_002dpost_002drefine_002dstate">11.100.6 do-probe-dots-post-refine-state</a></li>
      <li><a name="toc-unmangle_002dhydrogen_002dname-1" href="#unmangle_002dhydrogen_002dname">11.100.7 unmangle-hydrogen-name</a></li>
      <li><a name="toc-set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius-1" href="#set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius">11.100.8 set-interactive-probe-dots-molprobity-radius</a></li>
      <li><a name="toc-interactive_002dprobe_002ddots_002dmolprobity_002dradius-1" href="#interactive_002dprobe_002ddots_002dmolprobity_002dradius">11.100.9 interactive-probe-dots-molprobity-radius</a></li>
      <li><a name="toc-user_002dmods_002dscm-1" href="#user_002dmods_002dscm">11.100.10 user-mods-scm</a></li>
      <li><a name="toc-user_002dmods_002dpy-1" href="#user_002dmods_002dpy">11.100.11 user-mods-py</a></li>
    </ul></li>
    <li><a name="toc-Map-Sharpening-Interface-1" href="#Map-Sharpening-Interface">11.101 Map Sharpening Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-sharpen-1" href="#sharpen">11.101.1 sharpen</a></li>
      <li><a name="toc-set_002dmap_002dsharpening_002dscale_002dlimit-1" href="#set_002dmap_002dsharpening_002dscale_002dlimit">11.101.2 set-map-sharpening-scale-limit</a></li>
    </ul></li>
    <li><a name="toc-Intermediate-Atom-Manipulation-Interface-1" href="#Intermediate-Atom-Manipulation-Interface">11.102 Intermediate Atom Manipulation Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-add_002dtarget_002dposition_002drestraint_002dfor_002dintermediate_002datom_002dpy-1" href="#add_002dtarget_002dposition_002drestraint_002dfor_002dintermediate_002datom_002dpy">11.102.1 add-target-position-restraint-for-intermediate-atom-py</a></li>
    </ul></li>
    <li><a name="toc-Marking-Fixed-Atom-Interface-1" href="#Marking-Fixed-Atom-Interface">11.103 Marking Fixed Atom Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-clear_002dall_002dfixed_002datoms-1" href="#clear_002dall_002dfixed_002datoms">11.103.1 clear-all-fixed-atoms</a></li>
    </ul></li>
    <li><a name="toc-Partial-Charges-1" href="#Partial-Charges">11.104 Partial Charges</a>
    <ul class="no-bullet">
      <li><a name="toc-show_002dpartial_002dcharge_002dinfo-1" href="#show_002dpartial_002dcharge_002dinfo">11.104.1 show-partial-charge-info</a></li>
    </ul></li>
    <li><a name="toc-EM-interface-1" href="#EM-interface">11.105 EM interface</a>
    <ul class="no-bullet">
      <li><a name="toc-scale_002dcell-1" href="#scale_002dcell">11.105.1 scale-cell</a></li>
    </ul></li>
    <li><a name="toc-CCP4mg-Interface-1" href="#CCP4mg-Interface">11.106 CCP4mg Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-ccp4i_002dprojects_002dscm-1" href="#ccp4i_002dprojects_002dscm">11.106.1 ccp4i-projects-scm</a></li>
      <li><a name="toc-ccp4i_002dprojects_002dpy-1" href="#ccp4i_002dprojects_002dpy">11.106.2 ccp4i-projects-py</a></li>
      <li><a name="toc-set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs-1" href="#set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs">11.106.3 set-add-ccp4i-projects-to-file-dialogs</a></li>
      <li><a name="toc-write_002dccp4mg_002dpicture_002ddescription-1" href="#write_002dccp4mg_002dpicture_002ddescription">11.106.4 write-ccp4mg-picture-description</a></li>
      <li><a name="toc-get_002datom_002dcolour_002dfrom_002dmol_002dno-1" href="#get_002datom_002dcolour_002dfrom_002dmol_002dno">11.106.5 get-atom-colour-from-mol-no</a></li>
    </ul></li>
    <li><a name="toc-Dipoles-1" href="#Dipoles">11.107 Dipoles</a>
    <ul class="no-bullet">
      <li><a name="toc-add_002ddipole_002dfor_002dresidues_002dscm-1" href="#add_002ddipole_002dfor_002dresidues_002dscm">11.107.1 add-dipole-for-residues-scm</a></li>
      <li><a name="toc-add_002ddipole_002dscm-1" href="#add_002ddipole_002dscm">11.107.2 add-dipole-scm</a></li>
      <li><a name="toc-add_002ddipole_002dpy-1" href="#add_002ddipole_002dpy">11.107.3 add-dipole-py</a></li>
      <li><a name="toc-add_002ddipole_002dfor_002dresidues_002dpy-1" href="#add_002ddipole_002dfor_002dresidues_002dpy">11.107.4 add-dipole-for-residues-py</a></li>
    </ul></li>
    <li><a name="toc-Aux-functions-1" href="#Aux-functions">11.108 Aux functions</a>
    <ul class="no-bullet">
      <li><a name="toc-laplacian-1" href="#laplacian">11.108.1 laplacian</a></li>
    </ul></li>
    <li><a name="toc-SMILES-1" href="#SMILES">11.109 SMILES</a>
    <ul class="no-bullet">
      <li><a name="toc-do_002dsmiles_002dgui-1" href="#do_002dsmiles_002dgui">11.109.1 do-smiles-gui</a></li>
    </ul></li>
    <li><a name="toc-PHENIX-Support-1" href="#PHENIX-Support">11.110 PHENIX Support</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dbutton_002dlabel_002dfor_002dexternal_002drefinement-1" href="#set_002dbutton_002dlabel_002dfor_002dexternal_002drefinement">11.110.1 set-button-label-for-external-refinement</a></li>
    </ul></li>
    <li><a name="toc-Graphics-Text-1" href="#Graphics-Text">11.111 Graphics Text</a>
    <ul class="no-bullet">
      <li><a name="toc-place_002dtext-1" href="#place_002dtext">11.111.1 place-text</a></li>
      <li><a name="toc-remove_002dtext-1" href="#remove_002dtext">11.111.2 remove-text</a></li>
      <li><a name="toc-text_002dindex_002dnear_002dposition-1" href="#text_002dindex_002dnear_002dposition">11.111.3 text-index-near-position</a></li>
    </ul></li>
    <li><a name="toc-PISA-Interaction-1" href="#PISA-Interaction">11.112 PISA Interaction</a>
    <ul class="no-bullet">
      <li><a name="toc-pisa_002dinteraction-1" href="#pisa_002dinteraction">11.112.1 pisa-interaction</a></li>
    </ul></li>
    <li><a name="toc-Jiggle-Fit-1" href="#Jiggle-Fit">11.113 Jiggle Fit</a>
    <ul class="no-bullet">
      <li><a name="toc-fit_002dto_002dmap_002dby_002drandom_002djiggle-1" href="#fit_002dto_002dmap_002dby_002drandom_002djiggle">11.113.1 fit-to-map-by-random-jiggle</a></li>
      <li><a name="toc-fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle-1" href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle">11.113.2 fit-molecule-to-map-by-random-jiggle</a></li>
      <li><a name="toc-fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur-1" href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur">11.113.3 fit-molecule-to-map-by-random-jiggle-and-blur</a></li>
      <li><a name="toc-fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle-1" href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle">11.113.4 fit-chain-to-map-by-random-jiggle</a></li>
      <li><a name="toc-fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur-1" href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur">11.113.5 fit-chain-to-map-by-random-jiggle-and-blur</a></li>
    </ul></li>
    <li><a name="toc-SBase-interface-1" href="#SBase-interface">11.114 SBase interface</a>
    <ul class="no-bullet">
      <li><a name="toc-matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm-1" href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm">11.114.1 matching-compound-names-from-sbase-scm</a></li>
      <li><a name="toc-matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy-1" href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy">11.114.2 matching-compound-names-from-sbase-py</a></li>
      <li><a name="toc-get_002dccp4srs_002dmonomer_002dand_002ddictionary-1" href="#get_002dccp4srs_002dmonomer_002dand_002ddictionary">11.114.3 get-ccp4srs-monomer-and-dictionary</a></li>
      <li><a name="toc-get_002dsbase_002dmonomer-1" href="#get_002dsbase_002dmonomer">11.114.4 get-sbase-monomer</a></li>
    </ul></li>
    <li><a name="toc-FLE_002dView-1" href="#FLE_002dView">11.115 FLE-View</a>
    <ul class="no-bullet">
      <li><a name="toc-fle_002dview_002dset_002dwater_002ddist_002dmax-1" href="#fle_002dview_002dset_002dwater_002ddist_002dmax">11.115.1 fle-view-set-water-dist-max</a></li>
      <li><a name="toc-fle_002dview_002dset_002dh_002dbond_002ddist_002dmax-1" href="#fle_002dview_002dset_002dh_002dbond_002ddist_002dmax">11.115.2 fle-view-set-h-bond-dist-max</a></li>
    </ul></li>
    <li><a name="toc-LSQ_002dimprove-1" href="#LSQ_002dimprove">11.116 LSQ-improve</a>
    <ul class="no-bullet">
      <li><a name="toc-lsq_002dimprove-1" href="#lsq_002dimprove">11.116.1 lsq-improve</a></li>
    </ul></li>
    <li><a name="toc-single_002dmodel-view-1" href="#single_002dmodel-view">11.117 single-model view</a>
    <ul class="no-bullet">
      <li><a name="toc-single_002dmodel_002dview_002dmodel_002dnumber-1" href="#single_002dmodel_002dview_002dmodel_002dnumber">11.117.1 single-model-view-model-number</a></li>
      <li><a name="toc-single_002dmodel_002dview_002dthis_002dmodel_002dnumber-1" href="#single_002dmodel_002dview_002dthis_002dmodel_002dnumber">11.117.2 single-model-view-this-model-number</a></li>
      <li><a name="toc-single_002dmodel_002dview_002dnext_002dmodel_002dnumber-1" href="#single_002dmodel_002dview_002dnext_002dmodel_002dnumber">11.117.3 single-model-view-next-model-number</a></li>
      <li><a name="toc-single_002dmodel_002dview_002dprev_002dmodel_002dnumber-1" href="#single_002dmodel_002dview_002dprev_002dmodel_002dnumber">11.117.4 single-model-view-prev-model-number</a></li>
    </ul></li>
    <li><a name="toc-graphics-2D-ligand-view-1" href="#graphics-2D-ligand-view">11.118 graphics 2D ligand view</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dshow_002dgraphics_002dligand_002dview-1" href="#set_002dshow_002dgraphics_002dligand_002dview">11.118.1 set-show-graphics-ligand-view</a></li>
    </ul></li>
  </ul></li>
  <li><a name="toc-More-Scripting-Functions-1" href="#More-Scripting-Functions">12 More Scripting Functions</a>
  <ul class="no-bullet">
    <li><a name="toc-More-Symmetry-Functions-1" href="#More-Symmetry-Functions">12.1 More Symmetry Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-get_002dsymmetry-1" href="#get_002dsymmetry">12.1.1 get-symmetry</a></li>
      <li><a name="toc-clashes_002dwith_002dsymmetry-1" href="#clashes_002dwith_002dsymmetry">12.1.2 clashes-with-symmetry</a></li>
    </ul></li>
    <li><a name="toc-Extra-Map-Functions-1" href="#Extra-Map-Functions">12.2 Extra Map Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-auto_002dread_002dmake_002dand_002ddraw_002dmaps-1" href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps">12.2.1 auto-read-make-and-draw-maps</a></li>
      <li><a name="toc-auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz-1" href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz">12.2.2 auto-read-make-and-draw-maps-from-mtz</a></li>
      <li><a name="toc-sharpen_002dblur_002dmap-1" href="#sharpen_002dblur_002dmap">12.2.3 sharpen-blur-map</a></li>
      <li><a name="toc-sharpen_002dblur_002dmap_002dwith_002dresampling-1" href="#sharpen_002dblur_002dmap_002dwith_002dresampling">12.2.4 sharpen-blur-map-with-resampling</a></li>
      <li><a name="toc-multi_002dsharpen_002dblur_002dmap_002dscm-1" href="#multi_002dsharpen_002dblur_002dmap_002dscm">12.2.5 multi-sharpen-blur-map-scm</a></li>
      <li><a name="toc-multi_002dsharpen_002dblur_002dmap_002dpy-1" href="#multi_002dsharpen_002dblur_002dmap_002dpy">12.2.6 multi-sharpen-blur-map-py</a></li>
      <li><a name="toc-flip_002dhand-1" href="#flip_002dhand">12.2.7 flip-hand</a></li>
      <li><a name="toc-go_002dto_002dmap_002dmolecule_002dcentre-1" href="#go_002dto_002dmap_002dmolecule_002dcentre">12.2.8 go-to-map-molecule-centre</a></li>
      <li><a name="toc-b_002dfactor_002dfrom_002dmap-1" href="#b_002dfactor_002dfrom_002dmap">12.2.9 b-factor-from-map</a></li>
      <li><a name="toc-map_002dcolour_002dcomponents-1" href="#map_002dcolour_002dcomponents">12.2.10 map-colour-components</a></li>
      <li><a name="toc-map_002dcolour_002dcomponents_002dpy-1" href="#map_002dcolour_002dcomponents_002dpy">12.2.11 map-colour-components-py</a></li>
      <li><a name="toc-handle_002dread_002dccp4_002dmap-1" href="#handle_002dread_002dccp4_002dmap">12.2.12 handle-read-ccp4-map</a></li>
    </ul></li>
    <li><a name="toc-Multi_002dResidue-Torsion-1" href="#Multi_002dResidue-Torsion">12.3 Multi-Residue Torsion</a>
    <ul class="no-bullet">
      <li><a name="toc-multi_002dresidue_002dtorsion_002dfit_002dscm-1" href="#multi_002dresidue_002dtorsion_002dfit_002dscm">12.3.1 multi-residue-torsion-fit-scm</a></li>
      <li><a name="toc-multi_002dresidue_002dtorsion_002dfit_002dpy-1" href="#multi_002dresidue_002dtorsion_002dfit_002dpy">12.3.2 multi-residue-torsion-fit-py</a></li>
    </ul></li>
    <li><a name="toc-Merge-Fragments-1" href="#Merge-Fragments">12.4 Merge Fragments</a>
    <ul class="no-bullet">
      <li><a name="toc-merge_002dfragments-1" href="#merge_002dfragments">12.4.1 merge-fragments</a></li>
    </ul></li>
    <li><a name="toc-Execute-Refmac-1" href="#Execute-Refmac">12.5 Execute Refmac</a>
    <ul class="no-bullet">
      <li><a name="toc-execute_002drefmac_002dreal-1" href="#execute_002drefmac_002dreal">12.5.1 execute-refmac-real</a></li>
      <li><a name="toc-refmac_002dname-1" href="#refmac_002dname">12.5.2 refmac-name</a></li>
    </ul></li>
    <li><a name="toc-Dictionary-Functions-1" href="#Dictionary-Functions">12.6 Dictionary Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-dictionaries_002dread-1" href="#dictionaries_002dread">12.6.1 dictionaries-read</a></li>
    </ul></li>
    <li><a name="toc-Restraints-Interface-1" href="#Restraints-Interface">12.7 Restraints Interface</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dmonomer_002drestraints-1" href="#set_002dmonomer_002drestraints">12.7.1 set-monomer-restraints</a></li>
    </ul></li>
    <li><a name="toc-Atom-Information-functions-1" href="#Atom-Information-functions">12.8 Atom Information functions</a>
    <ul class="no-bullet">
      <li><a name="toc-resname_002dfrom_002dserial_002dnumber-1" href="#resname_002dfrom_002dserial_002dnumber">12.8.1 resname-from-serial-number</a></li>
      <li><a name="toc-residue_002dname-1" href="#residue_002dname">12.8.2 residue-name</a></li>
      <li><a name="toc-serial_002dnumber_002dfrom_002dresidue_002dspecs-1" href="#serial_002dnumber_002dfrom_002dresidue_002dspecs">12.8.3 serial-number-from-residue-specs</a></li>
      <li><a name="toc-residue_002dinfo-1" href="#residue_002dinfo">12.8.4 residue-info</a></li>
      <li><a name="toc-chain_002dfragments_002dscm-1" href="#chain_002dfragments_002dscm">12.8.5 chain-fragments-scm</a></li>
      <li><a name="toc-add_002dmolecule-1" href="#add_002dmolecule">12.8.6 add-molecule</a></li>
      <li><a name="toc-clear_002dand_002dupdate_002dmolecule-1" href="#clear_002dand_002dupdate_002dmolecule">12.8.7 clear-and-update-molecule</a></li>
      <li><a name="toc-active_002dresidue-1" href="#active_002dresidue">12.8.8 active-residue</a></li>
      <li><a name="toc-closest_002datom_002dsimple_002dscm-1" href="#closest_002datom_002dsimple_002dscm">12.8.9 closest-atom-simple-scm</a></li>
      <li><a name="toc-closest_002datom-1" href="#closest_002datom">12.8.10 closest-atom</a></li>
      <li><a name="toc-closest_002datom_002draw_002dscm-1" href="#closest_002datom_002draw_002dscm">12.8.11 closest-atom-raw-scm</a></li>
      <li><a name="toc-residues_002dnear_002dresidue-1" href="#residues_002dnear_002dresidue">12.8.12 residues-near-residue</a></li>
      <li><a name="toc-residues_002dnear_002dresidues_002dscm-1" href="#residues_002dnear_002dresidues_002dscm">12.8.13 residues-near-residues-scm</a></li>
      <li><a name="toc-residues_002dnear_002dposition_002dscm-1" href="#residues_002dnear_002dposition_002dscm">12.8.14 residues-near-position-scm</a></li>
      <li><a name="toc-label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm-1" href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm">12.8.15 label-closest-atoms-in-neighbour-residues-scm</a></li>
      <li><a name="toc-hydrogenate_002dregion-1" href="#hydrogenate_002dregion">12.8.16 hydrogenate-region</a></li>
      <li><a name="toc-add_002dhydrogens_002dfrom_002dfile-1" href="#add_002dhydrogens_002dfrom_002dfile">12.8.17 add-hydrogens-from-file</a></li>
      <li><a name="toc-atom_002dinfo_002dstring_002dpy-1" href="#atom_002dinfo_002dstring_002dpy">12.8.18 atom-info-string-py</a></li>
      <li><a name="toc-molecule_002dto_002dpdb_002dstring_002dpy-1" href="#molecule_002dto_002dpdb_002dstring_002dpy">12.8.19 molecule-to-pdb-string-py</a></li>
      <li><a name="toc-residue_002dinfo_002dpy-1" href="#residue_002dinfo_002dpy">12.8.20 residue-info-py</a></li>
    </ul></li>
    <li><a name="toc-Using-S_002dexpression-molecules-1" href="#Using-S_002dexpression-molecules">12.9 Using S-expression molecules</a>
    <ul class="no-bullet">
      <li><a name="toc-active_002dresidue_002dpy-1" href="#active_002dresidue_002dpy">12.9.1 active-residue-py</a></li>
      <li><a name="toc-closest_002datom_002dsimple_002dpy-1" href="#closest_002datom_002dsimple_002dpy">12.9.2 closest-atom-simple-py</a></li>
      <li><a name="toc-closest_002datom_002dpy-1" href="#closest_002datom_002dpy">12.9.3 closest-atom-py</a></li>
      <li><a name="toc-closest_002datom_002draw_002dpy-1" href="#closest_002datom_002draw_002dpy">12.9.4 closest-atom-raw-py</a></li>
      <li><a name="toc-residues_002dnear_002dposition_002dpy-1" href="#residues_002dnear_002dposition_002dpy">12.9.5 residues-near-position-py</a></li>
      <li><a name="toc-label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy-1" href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy">12.9.6 label-closest-atoms-in-neighbour-residues-py</a></li>
      <li><a name="toc-get_002dbonds_002drepresentation-1" href="#get_002dbonds_002drepresentation">12.9.7 get-bonds-representation</a></li>
      <li><a name="toc-get_002ddictionary_002dradii-1" href="#get_002ddictionary_002dradii">12.9.8 get-dictionary-radii</a></li>
      <li><a name="toc-get_002denvironment_002ddistances_002drepresentation_002dpy-1" href="#get_002denvironment_002ddistances_002drepresentation_002dpy">12.9.9 get-environment-distances-representation-py</a></li>
      <li><a name="toc-get_002dintermediate_002datoms_002dbonds_002drepresentation-1" href="#get_002dintermediate_002datoms_002dbonds_002drepresentation">12.9.10 get-intermediate-atoms-bonds-representation</a></li>
      <li><a name="toc-get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate-1" href="#get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate">12.9.11 get-continue-updating-refinement-atoms-state</a></li>
    </ul></li>
    <li><a name="toc-Refinement-with-specs-1" href="#Refinement-with-specs">12.10 Refinement with specs</a>
    <ul class="no-bullet">
      <li><a name="toc-all_002dresidues_002dwith_002dserial_002dnumbers_002dscm-1" href="#all_002dresidues_002dwith_002dserial_002dnumbers_002dscm">12.10.1 all-residues-with-serial-numbers-scm</a></li>
      <li><a name="toc-regularize_002dresidues-1" href="#regularize_002dresidues">12.10.2 regularize-residues</a></li>
      <li><a name="toc-mtz_002dfile_002dname-1" href="#mtz_002dfile_002dname">12.10.3 mtz-file-name</a></li>
      <li><a name="toc-refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm-1" href="#refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm">12.10.4 refine-zone-with-full-residue-spec-scm</a></li>
    </ul></li>
    <li><a name="toc-Water-Chain-Functions-1" href="#Water-Chain-Functions">12.11 Water Chain Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-water_002dchain_002dfrom_002dshelx_002dins_002dscm-1" href="#water_002dchain_002dfrom_002dshelx_002dins_002dscm">12.11.1 water-chain-from-shelx-ins-scm</a></li>
      <li><a name="toc-water_002dchain_002dscm-1" href="#water_002dchain_002dscm">12.11.2 water-chain-scm</a></li>
      <li><a name="toc-water_002dchain_002dpy-1" href="#water_002dchain_002dpy">12.11.3 water-chain-py</a></li>
    </ul></li>
    <li><a name="toc-Glyco-Tools-1" href="#Glyco-Tools">12.12 Glyco Tools</a>
    <ul class="no-bullet">
      <li><a name="toc-print_002dglyco_002dtree-1" href="#print_002dglyco_002dtree">12.12.1 print-glyco-tree</a></li>
    </ul></li>
    <li><a name="toc-Spin-Search-Functions-1" href="#Spin-Search-Functions">12.13 Spin Search Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-spin_002dsearch-1" href="#spin_002dsearch">12.13.1 spin-search</a></li>
      <li><a name="toc-spin_002dN_002dscm-1" href="#spin_002dN_002dscm">12.13.2 spin-N-scm</a></li>
      <li><a name="toc-CG_002dspin_002dsearch_002dscm-1" href="#CG_002dspin_002dsearch_002dscm">12.13.3 CG-spin-search-scm</a></li>
      <li><a name="toc-spin_002dsearch_002dpy-1" href="#spin_002dsearch_002dpy">12.13.4 spin-search-py</a></li>
      <li><a name="toc-spin_002dN_002dpy-1" href="#spin_002dN_002dpy">12.13.5 spin-N-py</a></li>
      <li><a name="toc-CG_002dspin_002dsearch_002dpy-1" href="#CG_002dspin_002dsearch_002dpy">12.13.6 CG-spin-search-py</a></li>
    </ul></li>
    <li><a name="toc-Rotamer-Scoring-1" href="#Rotamer-Scoring">12.14 Rotamer Scoring</a>
    <ul class="no-bullet">
      <li><a name="toc-score_002drotamers_002dscm-1" href="#score_002drotamers_002dscm">12.14.1 score-rotamers-scm</a></li>
    </ul></li>
    <li><a name="toc-protein_002ddb-1" href="#protein_002ddb">12.15 protein-db</a>
    <ul class="no-bullet">
      <li><a name="toc-protein_002ddb_002dloops-1" href="#protein_002ddb_002dloops">12.15.1 protein-db-loops</a></li>
    </ul></li>
    <li><a name="toc-Coot_0027s-Hole-implementation-1" href="#Coot_0027s-Hole-implementation">12.16 Coot&rsquo;s Hole implementation</a>
    <ul class="no-bullet">
      <li><a name="toc-hole-1" href="#hole">12.16.1 hole</a></li>
      <li><a name="toc-make_002dlink-1" href="#make_002dlink">12.16.2 make-link</a></li>
    </ul></li>
    <li><a name="toc-Drag-and-Drop-Functions-1" href="#Drag-and-Drop-Functions">12.17 Drag and Drop Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-handle_002ddrag_002dand_002ddrop_002dstring-1" href="#handle_002ddrag_002dand_002ddrop_002dstring">12.17.1 handle-drag-and-drop-string</a></li>
    </ul></li>
    <li><a name="toc-Map-Contouring-Functions-1" href="#Map-Contouring-Functions">12.18 Map Contouring Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-map_002dcontours-1" href="#map_002dcontours">12.18.1 map-contours</a></li>
    </ul></li>
    <li><a name="toc-Map-to-Model-Correlation-1" href="#Map-to-Model-Correlation">12.19 Map to Model Correlation</a>
    <ul class="no-bullet">
      <li><a name="toc-set_002dmap_002dcorrelation_002datom_002dradius-1" href="#set_002dmap_002dcorrelation_002datom_002dradius">12.19.1 set-map-correlation-atom-radius</a></li>
      <li><a name="toc-map_002dto_002dmodel_002dcorrelation_002dscm-1" href="#map_002dto_002dmodel_002dcorrelation_002dscm">12.19.2 map-to-model-correlation-scm</a></li>
      <li><a name="toc-map_002dto_002dmodel_002dcorrelation-1" href="#map_002dto_002dmodel_002dcorrelation">12.19.3 map-to-model-correlation</a></li>
      <li><a name="toc-map_002dto_002dmodel_002dcorrelation_002dstats-1" href="#map_002dto_002dmodel_002dcorrelation_002dstats">12.19.4 map-to-model-correlation-stats</a></li>
      <li><a name="toc-map_002dto_002dmodel_002dcorrelation_002dper_002dresidue-1" href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue">12.19.5 map-to-model-correlation-per-residue</a></li>
      <li><a name="toc-map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue-1" href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue">12.19.6 map-to-model-correlation-stats-per-residue</a></li>
      <li><a name="toc-map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm-1" href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm">12.19.7 map-to-model-correlation-per-residue-scm</a></li>
      <li><a name="toc-map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm-1" href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm">12.19.8 map-to-model-correlation-stats-per-residue-scm</a></li>
      <li><a name="toc-qq_002dplot_002dmap_002dand_002dmodel_002dscm-1" href="#qq_002dplot_002dmap_002dand_002dmodel_002dscm">12.19.9 qq-plot-map-and-model-scm</a></li>
      <li><a name="toc-density_002dscore_002dresidue-1" href="#density_002dscore_002dresidue">12.19.10 density-score-residue</a></li>
      <li><a name="toc-map_002dmean_002dscm-1" href="#map_002dmean_002dscm">12.19.11 map-mean-scm</a></li>
      <li><a name="toc-map_002dstatistics_002dscm-1" href="#map_002dstatistics_002dscm">12.19.12 map-statistics-scm</a></li>
      <li><a name="toc-map_002dmean_002dpy-1" href="#map_002dmean_002dpy">12.19.13 map-mean-py</a></li>
    </ul></li>
    <li><a name="toc-Get-Sequence-1" href="#Get-Sequence">12.20 Get Sequence</a>
    <ul class="no-bullet">
      <li><a name="toc-get_002dsequence_002das_002dfasta_002dfor_002dchain-1" href="#get_002dsequence_002das_002dfasta_002dfor_002dchain">12.20.1 get-sequence-as-fasta-for-chain</a></li>
      <li><a name="toc-write_002dsequence-1" href="#write_002dsequence">12.20.2 write-sequence</a></li>
    </ul></li>
  </ul></li>
  <li><a name="toc-More-Refinement-Scripting-Functions-1" href="#More-Refinement-Scripting-Functions">13 More Refinement Scripting Functions</a>
  <ul class="no-bullet">
    <li><a name="toc-More-Refinement-Functions-1" href="#More-Refinement-Functions">13.1 More Refinement Functions</a>
    <ul class="no-bullet">
      <li><a name="toc-use_002dunimodal_002dring_002dtorsion_002drestraints-1" href="#use_002dunimodal_002dring_002dtorsion_002drestraints">13.1.1 use-unimodal-ring-torsion-restraints</a></li>
      <li><a name="toc-use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue-1" href="#use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue">13.1.2 use-unimodal-ring-torsion-restraints-for-residue</a></li>
      <li><a name="toc-set_002drefinement_002dgeman_002dmcclure_002dalpha-1" href="#set_002drefinement_002dgeman_002dmcclure_002dalpha">13.1.3 set-refinement-geman-mcclure-alpha</a></li>
      <li><a name="toc-get_002drefinement_002dgeman_002dmcclure_002dalpha-1" href="#get_002drefinement_002dgeman_002dmcclure_002dalpha">13.1.4 get-refinement-geman-mcclure-alpha</a></li>
      <li><a name="toc-set_002drefinement_002dlennard_002djones_002depsilon-1" href="#set_002drefinement_002dlennard_002djones_002depsilon">13.1.5 set-refinement-lennard-jones-epsilon</a></li>
      <li><a name="toc-set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor-1" href="#set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor">13.1.6 set-log-cosh-target-distance-scale-factor</a></li>
      <li><a name="toc-crankshaft_002dpeptide_002drotation_002doptimization_002dscm-1" href="#crankshaft_002dpeptide_002drotation_002doptimization_002dscm">13.1.7 crankshaft-peptide-rotation-optimization-scm</a></li>
      <li><a name="toc-crankshaft_002dpeptide_002drotation_002doptimization_002dpy-1" href="#crankshaft_002dpeptide_002drotation_002doptimization_002dpy">13.1.8 crankshaft-peptide-rotation-optimization-py</a></li>
    </ul></li>
  </ul></li>
  <li><a name="toc-Scheme-Scripting-Functions-1" href="#Scheme-Scripting-Functions">14 Scheme Scripting Functions</a>
  <ul class="no-bullet">
    <li><a name="toc-shelx-1" href="#shelx">14.1 shelx</a></li>
    <li><a name="toc-run_002dmogul-1" href="#run_002dmogul">14.2 run-mogul</a></li>
    <li><a name="toc-remote_002dcontrol-1" href="#remote_002dcontrol">14.3 remote-control</a></li>
    <li><a name="toc-tips-1" href="#tips">14.4 tips</a></li>
    <li><a name="toc-refmac-1" href="#refmac">14.5 refmac</a></li>
    <li><a name="toc-refmac_002dproblems-1" href="#refmac_002dproblems">14.6 refmac-problems</a></li>
    <li><a name="toc-redefine_002dfunctions-1" href="#redefine_002dfunctions">14.7 redefine-functions</a></li>
    <li><a name="toc-tips_002dgui-1" href="#tips_002dgui">14.8 tips-gui</a></li>
    <li><a name="toc-raster3d_002dfrom_002dscheme-1" href="#raster3d_002dfrom_002dscheme">14.9 raster3d-from-scheme</a></li>
    <li><a name="toc-quat_002dconvert-1" href="#quat_002dconvert">14.10 quat-convert</a></li>
    <li><a name="toc-gui_002dprosmart-1" href="#gui_002dprosmart">14.11 gui-prosmart</a></li>
    <li><a name="toc-mutate_002dfrom_002dscheme-1" href="#mutate_002dfrom_002dscheme">14.12 mutate-from-scheme</a></li>
    <li><a name="toc-parse_002dpisa_002dxml-1" href="#parse_002dpisa_002dxml">14.13 parse-pisa-xml</a></li>
    <li><a name="toc-add_002dlinked_002dcho-1" href="#add_002dlinked_002dcho">14.14 add-linked-cho</a></li>
    <li><a name="toc-ncs-1" href="#ncs">14.15 ncs</a></li>
    <li><a name="toc-acedrg_002dlink-1" href="#acedrg_002dlink">14.16 acedrg-link</a></li>
    <li><a name="toc-ligand_002dcheck-1" href="#ligand_002dcheck">14.17 ligand-check</a></li>
    <li><a name="toc-what_002dcheck-1" href="#what_002dcheck">14.18 what-check</a></li>
    <li><a name="toc-user_002ddefine_002drestraints-1" href="#user_002ddefine_002drestraints">14.19 user-define-restraints</a></li>
    <li><a name="toc-jligand_002dgui-1" href="#jligand_002dgui">14.20 jligand-gui</a></li>
    <li><a name="toc-gui_002dligand_002dsliders-1" href="#gui_002dligand_002dsliders">14.21 gui-ligand-sliders</a></li>
    <li><a name="toc-snarf_002dcoot_002ddocs-1" href="#snarf_002dcoot_002ddocs">14.22 snarf-coot-docs</a></li>
    <li><a name="toc-mutate_002din_002dscheme-1" href="#mutate_002din_002dscheme">14.23 mutate-in-scheme</a></li>
    <li><a name="toc-entry_002bdo_002dbutton-1" href="#entry_002bdo_002dbutton">14.24 entry+do-button</a></li>
    <li><a name="toc-dictionary_002dgenerators-1" href="#dictionary_002dgenerators">14.25 dictionary-generators</a></li>
    <li><a name="toc-extra_002dtop_002dlevel-1" href="#extra_002dtop_002dlevel">14.26 extra-top-level</a></li>
    <li><a name="toc-coot_002dlsq-1" href="#coot_002dlsq">14.27 coot-lsq</a></li>
    <li><a name="toc-get_002debi-1" href="#get_002debi">14.28 get-ebi</a></li>
    <li><a name="toc-coot_002dgui-1" href="#coot_002dgui">14.29 coot-gui</a></li>
    <li><a name="toc-hello-1" href="#hello">14.30 hello</a></li>
    <li><a name="toc-coot_002dcrash_002dcatcher-1" href="#coot_002dcrash_002dcatcher">14.31 coot-crash-catcher</a></li>
    <li><a name="toc-coot_002dutils-1" href="#coot_002dutils">14.32 coot-utils</a></li>
    <li><a name="toc-contact_002dscore_002disolated_002dligand-1" href="#contact_002dscore_002disolated_002dligand">14.33 contact-score-isolated-ligand</a></li>
    <li><a name="toc-libcheck-1" href="#libcheck">14.34 libcheck</a></li>
    <li><a name="toc-cns2coot-1" href="#cns2coot">14.35 cns2coot</a></li>
    <li><a name="toc-clear_002dbackup-1" href="#clear_002dbackup">14.36 clear-backup</a></li>
    <li><a name="toc-generic_002dobjects-1" href="#generic_002dobjects">14.37 generic-objects</a></li>
    <li><a name="toc-exercise_002dscm_002dmol-1" href="#exercise_002dscm_002dmol">14.38 exercise-scm-mol</a></li>
    <li><a name="toc-check_002dfor_002dupdates-1" href="#check_002dfor_002dupdates">14.39 check-for-updates</a></li>
    <li><a name="toc-brute_002dlsqman-1" href="#brute_002dlsqman">14.40 brute-lsqman</a></li>
    <li><a name="toc-fitting-1" href="#fitting">14.41 fitting</a></li>
    <li><a name="toc-group_002dsettings-1" href="#group_002dsettings">14.42 group-settings</a></li>
    <li><a name="toc-background_002ddemo-1" href="#background_002ddemo">14.43 background-demo</a></li>
    <li><a name="toc-americanisms-1" href="#americanisms">14.44 americanisms</a></li>
    <li><a name="toc-filter-1" href="#filter">14.45 filter</a></li>
    <li><a name="toc-a_002drapper_002dgui-1" href="#a_002drapper_002dgui">14.46 a-rapper-gui</a></li>
    <li><a name="toc-fascinating_002dthings-1" href="#fascinating_002dthings">14.47 fascinating-things</a></li>
    <li><a name="toc-get_002drecent_002dpdbe-1" href="#get_002drecent_002dpdbe">14.48 get-recent-pdbe</a></li>
    <li><a name="toc-3d_002dgenerator_002dimport-1" href="#g_t3d_002dgenerator_002dimport">14.49 3d-generator-import</a></li>
    <li><a name="toc-coot-1" href="#coot">14.50 coot</a></li>
    <li><a name="toc-jligand-1" href="#jligand">14.51 jligand</a></li>
  </ul></li>
  <li><a name="toc-Concept-Index-1" href="#Concept-Index">Concept Index</a></li>
  <li><a name="toc-Function-Index-1" href="#Function-Index">Function Index</a></li>
</ul>
</div>




<a name="Top"></a>
<div class="header">
<p>
Next: <a href="#Introduction" accesskey="n" rel="next">Introduction</a>, Up: <a href="dir.html#Top" accesskey="u" rel="up">(dir)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Coot-User-Manual"></a>
<h1 class="top">Coot User Manual</h1>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Introduction" accesskey="1">Introduction</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mousing-and-Keyboarding" accesskey="2">Mousing and Keyboarding</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#General-Features" accesskey="3">General Features</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Coordinate_002dRelated-Features" accesskey="4">Coordinate-Related Features</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Modelling-and-Building" accesskey="5">Modelling and Building</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map_002dRelated-Features" accesskey="6">Map-Related Features</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Validation" accesskey="7">Validation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Representation" accesskey="8">Representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Hints-and-Usage-Tips" accesskey="9">Hints and Usage Tips</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Other-Programs">Other Programs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scripting-Functions">Scripting Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#More-Scripting-Functions">More Scripting Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#More-Refinement-Scripting-Functions">More Refinement Scripting Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scheme-Scripting-Functions">Scheme Scripting Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Concept-Index">Concept Index</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">Concept index.
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Function-Index">Function Index</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">Function index.
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
</pre></th></tr><tr><th colspan="3" align="left" valign="top"><pre class="menu-comment"> &mdash; The Detailed Node Listing &mdash;

Introduction

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Citing-Coot">Citing Coot</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#What-is-Coot_003f">What is Coot?</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#What-Coot-is-Not">What Coot is Not</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Hardware-Requirements">Hardware Requirements</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Environment-Variables">Environment Variables</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Command-Line-Arguments">Command Line Arguments</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Web-Page">Web Page</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Crash">Crash</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Mousing and Keyboarding

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Next-Residue">Next Residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Keyboard-Contouring">Keyboard Contouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mouse-Z-Translation">Mouse Z Translation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Keyboard-Z-Translation">Keyboard Z Translation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Keyboard-Zoom-and-Clip">Keyboard Zoom and Clip</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scrollwheel">Scrollwheel</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Selecting-Atoms">Selecting Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Virtual-Trackball">Virtual Trackball</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#more-on-zooming">more on zooming</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
General Features

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Version-number">Version number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Screenshot">Screenshot</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Raster3D">Raster3D</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Display-Manager">Display Manager</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#The-Modelling-Toolbar">The Modelling Toolbar</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#The-file-selector">The file selector</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scripting">Scripting</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Backups-and-Undo">Backups and Undo</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#View-Matrix">View Matrix</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Space-Group-and-Symmetry">Space Group and Symmetry</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Recentring-View">Recentring View</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Views">Views</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Clipping-Manipulation">Clipping Manipulation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Background-colour">Background colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Unit-Cell">Unit Cell</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotation-Centre-Pointer">Rotation Centre Pointer</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Orientation-Axes">Orientation Axes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pointer-Distances">Pointer Distances</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Crosshairs">Crosshairs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#g_t3D-Annotations">3D Annotations</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Frame-Rate">Frame Rate</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Program-Output">Program Output</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Scripting

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Python">Python</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scheme">Scheme</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Coot-State">Coot State</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Key-Binding">Key Binding</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#User_002dDefined-Functions">User-Defined Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Python


Backups and Undo


Coordinate-Related Features

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Reading-coordinates">Reading coordinates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Info">Atom Info</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Labeling">Atom Labeling</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Colouring">Atom Colouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Bond-Parameters">Bond Parameters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Download-coordinates">Download coordinates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Get-Coordinates-and-Map-from-EDS">Get Coordinates and Map from EDS</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Save-coordinates">Save coordinates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Setting-the-Space-Group">Setting the Space Group</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Anisotropic-Atoms">Anisotropic Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Symmetry">Symmetry</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Sequence-View">Sequence View</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Print-Sequence">Print Sequence</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Environment-Distances">Environment Distances</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Distances-and-Angles">Distances and Angles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Zero-Occupancy-Marker">Zero Occupancy Marker</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atomic-Dots">Atomic Dots</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Ball-and-Stick-Representation">Ball and Stick Representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mean-and-Median-Temperature-Factors">Mean and Median Temperature Factors</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Secondary-Structure-Matching-_0028SSM_0029">Secondary Structure Matching (SSM)</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Least_002dSquares-Fitting">Least-Squares Fitting</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Ligand-Overlaying">Ligand Overlaying</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Writing-PDB-files">Writing PDB files</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Modelling and Building

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Changing-the-Map-for-Building_002fRefinement">Changing the Map for Building/Refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotate_002fTranslate-Zone">Rotate/Translate Zone</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rigid-Body-Refinement">Rigid Body Refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Simplex-Refinement">Simplex Refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Post_002dmanipulation_002dhook">Post-manipulation-hook</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Baton-Building">Baton Building</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Reversing-Direction-of-Fragment">Reversing Direction of Fragment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#C_002dalpha-_002d_003e-Mainchain">C-alpha -&gt; Mainchain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Backbone-Torsion-Angles">Backbone Torsion Angles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Docking-Sidechains">Docking Sidechains</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotamers">Rotamers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Editing-Chi-Angles">Editing Chi Angles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Torsion-General">Torsion General</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pep_002dflip">Pep-flip</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Adding-Alternative-Conformations">Adding Alternative Conformations</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mutation">Mutation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Importing-Ligands_002fMonomers">Importing Ligands/Monomers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Ligand-from-SMILES-strings">Ligand from SMILES strings</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Find-Ligands">Find Ligands</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Flip-Ligand">Flip Ligand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Find-Waters">Find Waters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Place-Helix">Place Helix</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Building-Ideal-DNA-and-RNA">Building Ideal DNA and RNA</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Merge-Molecules">Merge Molecules</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Temperature-Factor-for-New-Atoms">Temperature Factor for New Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Applying-NCS-Edits">Applying NCS Edits</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Running-Refmac">Running Refmac</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Running-SHELXL">Running SHELXL</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Clear-Pending-Picks">Clear Pending Picks</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Delete">Delete</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Sequence-Assignment">Sequence Assignment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Building-Links-and-Loops">Building Links and Loops</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Fill-Partial-Residues">Fill Partial Residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Changing-Chain-IDs">Changing Chain IDs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Setting-Occupancies">Setting Occupancies</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Fix-Nomenclature-Errors">Fix Nomenclature Errors</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotamer-Fix-Whole-Protein">Rotamer Fix Whole Protein</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Refine-All-Waters">Refine All Waters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Moving-Molecules_002fLigands">Moving Molecules/Ligands</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Modifying-the-Labels-on-the-Model_002fFit_002fRefine-dialog">Modifying the Labels on the Model/Fit/Refine dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Regularization and Real Space Refinement 

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Dictionary">Dictionary</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Sphere-Refinement">Sphere Refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Refining-Carbohydrates">Refining Carbohydrates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Planar-Peptide-Restraints">Planar Peptide Restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#The-UNK-residue-type">The UNK residue type</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Moving-Zero-Occupancy-Atoms">Moving Zero Occupancy Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Map-Related Features 

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Maps-in-General">Maps in General</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Create-a-Map">Create a Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Contouring">Map Contouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Extent">Map Extent</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Line-Width">Map Line Width</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-colouring">Map colouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Difference-Map-Colouring">Difference Map Colouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Sampling">Map Sampling</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Make-a-Difference-Map">Make a Difference Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Make-an-Averaged-Map">Make an Averaged Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Dragged-Map">Dragged Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Dynamic-Map-Sampling-and-Display-Size">Dynamic Map Sampling and Display Size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Skeletonization">Skeletonization</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Sharpening">Map Sharpening</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pattersons">Pattersons</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Re_002dInterpolation">Map Re-Interpolation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#masks">masks</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Trimming-Atoms">Trimming Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Transformation">Map Transformation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Export-Map">Export Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Validation 

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Ramachandran-Plots">Ramachandran Plots</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Chiral-Volumes">Chiral Volumes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sec_005fblobs">sec_blobs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Check-Waters-by-Difference-Map">Check Waters by Difference Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Molprobity-Tools-Interface">Molprobity Tools Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#GLN-and-ASN-B_002dfactor-Outliers">GLN and ASN B-factor Outliers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Validation-Graphs">Validation Graphs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Hints and Usage Tips

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#Documentation">Documentation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Low-Resolution">Low Resolution</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Coot-Droppings">Coot Droppings</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Clearing-Backups">Clearing Backups</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Getting-out-of-_0060_0060Translate_0027_0027-Mode">Getting out of ``Translate'' Mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Getting-out-of-_0060_0060Continuous-Rotation_0027_0027-Mode">Getting out of ``Continuous Rotation'' Mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Label-Atom-Only-Mode">Label Atom Only Mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Button-Labels">Button Labels</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sec_005fpicking">sec_picking</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Resizing-View">Resizing View</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scroll_002dwheel">Scroll-wheel</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Slow-Computer-Configuration">Slow Computer Configuration</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
Other Programs

</pre></th></tr><tr><td align="left" valign="top">&bull; <a href="#findligand">findligand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment">
</pre></th></tr></table>

<hr>
<a name="Introduction"></a>
<div class="header">
<p>
Next: <a href="#Mousing-and-Keyboarding" accesskey="n" rel="next">Mousing and Keyboarding</a>, Previous: <a href="#Top" accesskey="p" rel="previous">Top</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Introduction-1"></a>
<h2 class="chapter">1 Introduction</h2>
     
<p>This document is the Coot User Manual, giving an overview of the
interactive features.  Other documentation includes the
<a name="index-reference-manual"></a>
Coot Reference Manual and the Coot
Tutorial.  These documents should be distributed with
the source code.
</p>
<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Citing-Coot" accesskey="1">Citing Coot</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#What-is-Coot_003f" accesskey="2">What is Coot?</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#What-Coot-is-Not" accesskey="3">What Coot is Not</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Hardware-Requirements" accesskey="4">Hardware Requirements</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Environment-Variables" accesskey="5">Environment Variables</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Command-Line-Arguments" accesskey="6">Command Line Arguments</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Web-Page" accesskey="7">Web Page</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Crash" accesskey="8">Crash</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Citing-Coot"></a>
<div class="header">
<p>
Next: <a href="#What-is-Coot_003f" accesskey="n" rel="next">What is Coot?</a>, Up: <a href="#Introduction" accesskey="u" rel="up">Introduction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Citing-Coot-and-Friends"></a>
<h3 class="section">1.1 Citing Coot and Friends</h3>
<a name="index-citing-coot"></a>


<p>If have found this software to be useful, you are requested
(if appropriate) to cite:
</p>

<blockquote>
<p>&quot;Features and Development of Coot&quot;
   P Emsley, B Lohkamp, W Scott, and K Cowtan
   Acta Cryst. (2010).  D66, 486-501
   <em>Acta Crystallographica Section D-Biological Crystallography</em>
   <strong>66</strong>: 486-501
</p></blockquote>


<p>The reference for the REFMAC5 Dictionary is:
</p>
<blockquote>
<p>REFMAC5 dictionary: &quot;Organization of Prior Chemical Knowledge and
   Guidelines for its Use&quot; Vagin AA, Steiner RA, Lebedev AA, Potterton L,
   McNicholas S Long F, Murshudov GN <em>Acta Crystallographica Section
   D-Biological Crystallography</em> <strong>60</strong>: 2184-2195 Part 12 Sp. Iss. 1
   DEC 2004&quot;
</p></blockquote>


<p>If using &quot;SSM Superposition&quot;, please cite:
</p>
<blockquote>
<p>&quot;Secondary-structure matching (SSM), a new tool for fast
   protein structure alignment in three dimensions&quot;
   Krissinel E, Henrick K
   <em>Acta Crystallographica Section D-Biological Crystallography</em>
   <strong>60</strong>: 2256-2268 Part 12 Sp. Iss. 1 DEC 2004
</p></blockquote>



<p>The reference for the the Electron Density Server is: 
</p>
<blockquote>
<p>GJ Kleywegt, MR Harris, JY Zou, TC Taylor, A W&auml;hlby,  TA Jones (2004),
&quot;The Uppsala Electron-Density Server&quot;, 
<em>Acta Crystallographica Section D-Biological Crystallography</em> <strong>60</strong>, 2240-2249.
</p>
<p>Please also cite the primary literature for the received structures.
</p>
</blockquote>



<hr>
<a name="What-is-Coot_003f"></a>
<div class="header">
<p>
Next: <a href="#What-Coot-is-Not" accesskey="n" rel="next">What Coot is Not</a>, Previous: <a href="#Citing-Coot" accesskey="p" rel="previous">Citing Coot</a>, Up: <a href="#Introduction" accesskey="u" rel="up">Introduction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="What-is-Coot_003f-1"></a>
<h3 class="section">1.2 What is Coot?</h3>

<p>Coot is a molecular graphics application. Its primary focus is
crystallographic macromolecular model-building and manipulation rather
than representation <em>i.e.</em> more like Frodo than Rasmol.
Having said that, Coot can work with small molecule (SHELXL) and electron
microscopy data, be used for homology modelling, make passably pretty
pictures and display NMR structures.
</p>
<p>Coot is Free Software.  You can give it away. If you don&rsquo;t like the way
it behaves, you can fix it yourself.
</p>
<hr>
<a name="What-Coot-is-Not"></a>
<div class="header">
<p>
Next: <a href="#Hardware-Requirements" accesskey="n" rel="next">Hardware Requirements</a>, Previous: <a href="#What-is-Coot_003f" accesskey="p" rel="previous">What is Coot?</a>, Up: <a href="#Introduction" accesskey="u" rel="up">Introduction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="What-Coot-is-Not-1"></a>
<h3 class="section">1.3 What Coot is Not</h3>

<p>Coot is not:
</p>
<ul>
<li> CCP4&rsquo;s official Molecular Graphics program <a name="DOCF1" href="#FOOT1"><sup>1</sup></a>

</li><li> a program to do refinement <a name="DOCF2" href="#FOOT2"><sup>2</sup></a>

</li><li> a protein crystallographic suite<a name="DOCF3" href="#FOOT3"><sup>3</sup></a>.

</li></ul>

<hr>
<a name="Hardware-Requirements"></a>
<div class="header">
<p>
Next: <a href="#Environment-Variables" accesskey="n" rel="next">Environment Variables</a>, Previous: <a href="#What-Coot-is-Not" accesskey="p" rel="previous">What Coot is Not</a>, Up: <a href="#Introduction" accesskey="u" rel="up">Introduction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Hardware-Requirements-1"></a>
<h3 class="section">1.4 Hardware Requirements</h3>

<p>The code is designed to be portable to any Unix-like operating system.
Coot certainly runs on RedHat Linux of various sorts, Fedora, Ubuntu,
Debian, SuSe Linux and MacOS X.  There is also a Window port (called
WinCoot).
</p>

<p>If you want to port to some other operating system, you are welcome
<a name="DOCF4" href="#FOOT4"><sup>4</sup></a>.  Note that your task will be eased by using GNU GCC to compile
the programs components.
</p>
<a name="Mouse"></a>
<h4 class="subsection">1.4.1 Mouse</h4>
<a name="index-mouse"></a>
<p>Coot works best with a 3-button mouse and works better if
it has a scroll-wheel too (see Chapter 2 for more details) <a name="DOCF5" href="#FOOT5"><sup>5</sup></a>.
</p>
<hr>
<a name="Environment-Variables"></a>
<div class="header">
<p>
Next: <a href="#Command-Line-Arguments" accesskey="n" rel="next">Command Line Arguments</a>, Previous: <a href="#Hardware-Requirements" accesskey="p" rel="previous">Hardware Requirements</a>, Up: <a href="#Introduction" accesskey="u" rel="up">Introduction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Environment-Variables-1"></a>
<h3 class="section">1.5 Environment Variables</h3>
<p>Coot responds to several environment variables that modify its
behaviour.
</p>
<ul>
<li> <code>COOT_STANDARD_RESIDUES</code> 
  The filename of the pdb file
  containing the standard amino acid residues in &ldquo;standard
  conformation&rdquo; <a name="DOCF6" href="#FOOT6"><sup>6</sup></a>
</li><li> <code>COOT_SCHEME_DIR</code> 
  The directory containing standard (part of the distribution) scheme
  files 
</li><li> <code>COOT_SCHEME_EXTRAS_DIR</code> 
  A &rsquo;:&rsquo;-separated list of directories containing bespoke scheme files.
  This variable is not set by default.  If you set it, Coot will test
  each &rsquo;:&rsquo;-separated string that it points to a directory, and if it does, 
  Coot will load all the <tt>.scm</tt> files in that directory.
</li><li> <code>COOT_PYTHON_EXTRAS_DIR</code> 
  A &rsquo;:&rsquo;-separated list of directories containing bespoke python files.
  This variable is not set by default.  If you set it, Coot will test
  each &rsquo;:&rsquo;-separated string that it points to a directory, and if it does, 
  Coot will load all the <tt>.py</tt> files in that directory.
</li><li> <code>COOT_REF_STRUCTS</code> 
  The directory containing a set of high resolution pdb files used as 
<a name="index-reference-structures"></a>
  reference structures to build backbone atoms from
  C<em>\alpha</em> positions
</li><li> <code>COOT_REF_SEC_STRUCTS</code> 
  The directory containing a set of high-quality structures to be used
  as templates for fitting beta strands.  If this is not set, then the
  directory COOT_REF_SEC_STRUCTS will be used to find the reference pdb
  files.  
</li><li> <code>COOT_REFMAC_LIB_DIR</code> 
<a name="index-dictionary_002c-cif"></a>
  Refmac&rsquo;s CIF directory containing the monomers and link descriptions.
  In the future this may simply be the same directory in which refmac
  looks to find the library dictionary.
</li><li> <code>COOT_SBASE_DIR</code> 
<a name="index-dictionary_002c-ligands"></a>
  The directory to find the SBASE dictionary (often comes with CCP4).
</li><li> <code>COOT_RESOURCES_DIR</code> 
  The directory that contains the
  splash screen image and the GTK+ application resources.
</li><li> <code>COOT_BACKUP_DIR</code> 
  The directory to which backup are
  written (if it exists as a directory).  If it is not, then backups
  are written to the current directory (the directory in which coot
  was started).
</li></ul>
  
<p>And of course extension language environment variables are used too:
</p>
<ul>
<li> <code>PYTHONPATH</code> (for python modules)
</li><li> <code>GUILE_LOAD_PATH</code> (for guile modules)
</li></ul>

<p>Normally, these environment variables will be set correctly in the coot
shell script.
</p>
<hr>
<a name="Command-Line-Arguments"></a>
<div class="header">
<p>
Next: <a href="#Web-Page" accesskey="n" rel="next">Web Page</a>, Previous: <a href="#Environment-Variables" accesskey="p" rel="previous">Environment Variables</a>, Up: <a href="#Introduction" accesskey="u" rel="up">Introduction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Command-Line-Arguments-1"></a>
<h3 class="section">1.6 Command Line Arguments</h3>
<a name="index-command-line-arguments"></a>
<p>Rather that using the GUI to read in information, you can use the
following command line arguments:
</p><ul>
<li> <code>--c <em>cmd</em></code> to run a command <code><em>cmd</em></code> on start up
</li><li> <code>--script <em>filename</em></code> to run a script on start up (but see Section <a href="#Scripting">Scripting</a>)
</li><li> <code>--no-state-script</code> don&rsquo;t run the <tt>0-coot.state.scm</tt> script on start up.  
   Don&rsquo;t save a state script on exit either.
</li><li> <code>--pdb <em>filename</em></code>  for pdb/coordinates file
</li><li> <code>--coords <em>filename</em></code>  for SHELX .ins/.res and CIF files
</li><li> <code>--data <em>filename</em></code> for mtz, phs or mmCIF data file
</li><li> <code>--auto <em>filename</em></code> for auto-reading mtz files (mtz file has the default labels FWT, PHWT)
</li><li> <code>--map <em>filename</em></code>  for a map (currently CCP4-format only)
</li><li> <code>--dictionary <em>filename</em></code> read in a cif monomer dictionary 
</li><li> <code>--help</code> print command line options
</li><li> <code>--stereo</code> start up in hardware stereo mode
</li><li> <code>--version</code> print the version of coot and exit
</li><li> <code>--code <em>accession-code</em></code> on starting Coot, get the pdb file 
   and mtz file (if it exists) from the EDS
</li><li> <code>--no-guano</code> don&rsquo;t leave &ldquo;Coot droppings&rdquo; i.e. don&rsquo;t write state
    and history files on exit
</li><li> <code>--side-by-side</code> start in side-by-side stereo mode
</li><li> <code>--update-self</code> command-line mode to update the coot to the latest 
    pre-release on the server
</li><li> <code>--python</code> an argument with no parameters - used to tell Coot that 
    the -c arguments should be process as python (rather than as scheme).
</li><li> <code>--small-screen</code> start with smaller icons and font to fit on small
    screen displays
</li><li> <code>--zalman-stereo</code> start in Zalman stereo mode
</li></ul>

<p>So, for example, one might use:
</p><ul>
<li> <code>coot --pdb post-refinement.pdb --auto refmac-2.mtz --dictionary lig.cif</code>
</li></ul>

<hr>
<a name="Web-Page"></a>
<div class="header">
<p>
Next: <a href="#Crash" accesskey="n" rel="next">Crash</a>, Previous: <a href="#Command-Line-Arguments" accesskey="p" rel="previous">Command Line Arguments</a>, Up: <a href="#Introduction" accesskey="u" rel="up">Introduction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Web-Page-1"></a>
<h3 class="section">1.7 Web Page</h3>
<a name="index-web-page"></a>
<p>Coot has a web page:
</p>
<ul>
<li> <a href="http://www2.mrc-lmb.cam.ac.uk/personal/pemsley/coot/">http://www2.mrc-lmb.cam.ac.uk/personal/pemsley/coot/</a>
</li></ul>

<p>There you can read more about the CCP4 molecular graphics project in
general and other projects which are important for Coot <a name="DOCF7" href="#FOOT7"><sup>7</sup></a>.
</p>
<hr>
<a name="Crash"></a>
<div class="header">
<p>
Previous: <a href="#Web-Page" accesskey="p" rel="previous">Web Page</a>, Up: <a href="#Introduction" accesskey="u" rel="up">Introduction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Crash-1"></a>
<h3 class="section">1.8 Crash</h3>
<a name="index-crash"></a>
<p>Coot might crash on you - it shouldn&rsquo;t.  
</p>
<a name="index-recover-session"></a>
<a name="index-crash-recovery"></a>
<p>Whenever Coot manipulates the model, it saves a backup pdb file.
There are backup files in
the directory <tt>coot-backup</tt> <a name="DOCF8" href="#FOOT8"><sup>8</sup></a>. You can recover the session (until the last
edit) by reading in the pdb file that you started with last time and
then use <code>File -&gt; Recover Session&hellip;</code>.
</p>
<p>I would like to know about coot crashing <a name="DOCF9" href="#FOOT9"><sup>9</sup></a> so that I can fix it as soon as possible. If you want your
problem fixed, this involves some work on your part sadly.
</p>
<p>First please make sure that you are using the most recent version of
coot.  I will often need to know as much as possible about what you did
to cause the bug.  If you can reproduce the bug and send me the files
that are needed to cause it, I can almost certainly fix it <a name="DOCF10" href="#FOOT10"><sup>10</sup></a> - especially if you
<a name="index-debugger"></a>
<a name="index-gdb"></a>
use the debugger (gdb) and send a backtrace
too<a name="DOCF11" href="#FOOT11"><sup>11</sup></a>.  Note that you may have to source the
  contents of <tt>bin/coot</tt> so that the libraries are can be found when
  the executable dynamically links.
</p>

<hr>
<a name="Mousing-and-Keyboarding"></a>
<div class="header">
<p>
Next: <a href="#General-Features" accesskey="n" rel="next">General Features</a>, Previous: <a href="#Introduction" accesskey="p" rel="previous">Introduction</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Mousing-and-Keyboarding-1"></a>
<h2 class="chapter">2 Mousing and Keyboarding</h2>
<a name="index-mouse-buttons"></a>
<p>How do we move around and select things?
</p>

<dl compact="compact">
<dt><code>Left-mouse Drag</code></dt>
<dd><p>Rotate view
</p></dd>
<dt><code>Ctrl Left-Mouse Drag</code></dt>
<dd><p>Translates view
</p></dd>
<dt><code>Shift Left-Mouse</code></dt>
<dd><p>Label Atom
</p></dd>
<dt><code>Right-Mouse Drag</code></dt>
<dd><p>Zoom in and out</p></dd>
<dt><code>Ctrl Shift Right-Mouse Drag</code></dt>
<dd><p>Rotate View around Screen Z axis
</p></dd>
<dt><code>Middle-mouse</code></dt>
<dd><p>Centre on atom
</p></dd>
<dt><code>Scroll-wheel Forward</code></dt>
<dd><p>Increase map contour level
</p></dd>
<dt><code>Scroll-wheel Backward</code></dt>
<dd><p>Decrease map contour level
</p></dd>
</dl>

<p>See also Chapter <a href="#Hints-and-Usage-Tips">Hints and Usage Tips</a> for more help.
</p>
<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Next-Residue" accesskey="1">Next Residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Keyboard-Contouring" accesskey="2">Keyboard Contouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mouse-Z-Translation" accesskey="3">Mouse Z Translation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Keyboard-Z-Translation" accesskey="4">Keyboard Z Translation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Keyboard-Zoom-and-Clip" accesskey="5">Keyboard Zoom and Clip</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scrollwheel" accesskey="6">Scrollwheel</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Selecting-Atoms" accesskey="7">Selecting Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Virtual-Trackball" accesskey="8">Virtual Trackball</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#more-on-zooming" accesskey="9">more on zooming</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Next-Residue"></a>
<div class="header">
<p>
Next: <a href="#Keyboard-Contouring" accesskey="n" rel="next">Keyboard Contouring</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Next-Residue-1"></a>
<h3 class="section">2.1 Next Residue</h3>

<dl compact="compact">
<dt><code>``Space''</code></dt>
<dd><p>Next Residue
</p></dd>
<dt><code>``Shift'' ``Space''</code></dt>
<dd><p>Previous Residue
</p></dd>
</dl>

<p>See also &ldquo;Recentring View&rdquo; (Section <a href="#Recentring-View">Recentring View</a>).
</p>
<hr>
<a name="Keyboard-Contouring"></a>
<div class="header">
<p>
Next: <a href="#Mouse-Z-Translation" accesskey="n" rel="next">Mouse Z Translation</a>, Previous: <a href="#Next-Residue" accesskey="p" rel="previous">Next Residue</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Keyboard-Contouring-1"></a>
<h3 class="section">2.2 Keyboard Contouring</h3>

<p>Use <tt class="key">+</tt> or <tt class="key">-</tt> on the keyboard if you don&rsquo;t have a
scroll-wheel.
</p>

<hr>
<a name="Mouse-Z-Translation"></a>
<div class="header">
<p>
Next: <a href="#Keyboard-Z-Translation" accesskey="n" rel="next">Keyboard Z Translation</a>, Previous: <a href="#Keyboard-Contouring" accesskey="p" rel="previous">Keyboard Contouring</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Mouse-Z-Translation-and-Clipping"></a>
<h3 class="section">2.3 Mouse Z Translation and Clipping</h3>
<a name="index-translation_002c-mouse"></a>

<p>Here we can change the clipping and Translate in Screen Z
</p>
<dl compact="compact">
<dt><code>Ctrl Right-Mouse Drag Up/Down</code></dt>
<dd><p>changes the slab (clipping planes)
</p></dd>
<dt><code>Ctrl Right-Mouse Drag Left/Right</code></dt>
<dd><p>translates the view in screen Z
</p></dd>
</dl>

<hr>
<a name="Keyboard-Z-Translation"></a>
<div class="header">
<p>
Next: <a href="#Keyboard-Zoom-and-Clip" accesskey="n" rel="next">Keyboard Zoom and Clip</a>, Previous: <a href="#Mouse-Z-Translation" accesskey="p" rel="previous">Mouse Z Translation</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Keyboard-Translation"></a>
<h3 class="section">2.4 Keyboard Translation</h3>
<a name="index-translation_002c-keyboard"></a>

<dl compact="compact">
<dt><code>Keypad 3</code></dt>
<dd><p>Push View (+Z translation)
</p></dd>
<dt><code>Keypad .</code></dt>
<dd><p>Pull View (-Z translation)
</p></dd>
</dl>


<hr>
<a name="Keyboard-Zoom-and-Clip"></a>
<div class="header">
<p>
Next: <a href="#Scrollwheel" accesskey="n" rel="next">Scrollwheel</a>, Previous: <a href="#Keyboard-Z-Translation" accesskey="p" rel="previous">Keyboard Z Translation</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Keyboard-Zoom-and-Clip-1"></a>
<h3 class="section">2.5 Keyboard Zoom and Clip</h3>

<dl compact="compact">
<dt><code>N</code></dt>
<dd><p>Zoom out  
</p></dd>
<dt><code>M</code></dt>
<dd><p>Zoom in    
</p></dd>
<dt><code>D</code></dt>
<dd><p>Slim clip  
</p></dd>
<dt><code>F</code></dt>
<dd><p>Fatten clip
</p></dd>
</dl>

<hr>
<a name="Scrollwheel"></a>
<div class="header">
<p>
Next: <a href="#Selecting-Atoms" accesskey="n" rel="next">Selecting Atoms</a>, Previous: <a href="#Keyboard-Zoom-and-Clip" accesskey="p" rel="previous">Keyboard Zoom and Clip</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Scrollwheel-1"></a>
<h3 class="section">2.6 Scrollwheel</h3>
<p>When there is no map, using the scroll-wheel has no effect.  If there
is exactly one map displayed, 
<a name="index-contouring_002c-map"></a>
<a name="index-scroll_002c-map-contour-change-by"></a>
<a name="index-map-scrolling"></a>
<a name="index-scroll_002dwheel"></a>
the scroll-wheel
will change the contour level of that map.  If there are two or more
maps, the map for which the contour level is changed can be set using either 
<code>HID -&gt; Scrollwheel -&gt; Attach scroll-wheel
  to which map?</code> and selecting a map number or clicking the &quot;Scroll&quot;
  radio button for the map in the Display Manager.
</p>
<p>You can turn off the map contour level changing by the scroll wheel using:
</p>
<p><code>(set-scroll-by-wheel-mouse 0)</code>
</p>
<p>(the default is 1 [on]).
</p>
<hr>
<a name="Selecting-Atoms"></a>
<div class="header">
<p>
Next: <a href="#Virtual-Trackball" accesskey="n" rel="next">Virtual Trackball</a>, Previous: <a href="#Scrollwheel" accesskey="p" rel="previous">Scrollwheel</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Selecting-Atoms-1"></a>
<h3 class="section">2.7 Selecting Atoms</h3>
<p>Several Coot functions require the selecting of atoms to specify a
residue range (for example: Regularize, Refine (Section
<a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a>) or Rigid Body Fit Zone (Section
<a href="#Rigid-Body-Refinement">Rigid Body Refinement</a>)).  Select atoms with the Left-mouse.
See also Picking (Section <a href="#sec_005fpicking">sec_picking</a>).
</p>
<p>Use the scripting function
<code>(quanta-buttons)</code> to make the mouse
functions more like other molecular graphics programs to which you may
be more accustomed <a name="DOCF12" href="#FOOT12"><sup>12</sup></a>.
</p>
<hr>
<a name="Virtual-Trackball"></a>
<div class="header">
<p>
Next: <a href="#more-on-zooming" accesskey="n" rel="next">more on zooming</a>, Previous: <a href="#Selecting-Atoms" accesskey="p" rel="previous">Selecting Atoms</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Virtual-Trackball-1"></a>
<h3 class="section">2.8 Virtual Trackball</h3>
<a name="index-trackball_002c-virtual"></a>
<p>You may not completely like the way the
molecule is moved by the mouse movement <a name="DOCF13" href="#FOOT13"><sup>13</sup></a>.  To change this, try:
<code>HID -&gt; Virtual Trackball -&gt; Flat</code>.  To
do this from the scripting interface: <code>(vt-surface
  1)</code> <a name="DOCF14" href="#FOOT14"><sup>14</sup></a>.
</p>
<p>If you <em>do</em> want <code>screen-z rotation</code>
<a name="index-z_002drotation"></a>
screen-z rotation, you can either use Shift Right-Mouse Drag or set
the Virtual Trackball to Spherical Surface mode and move the mouse
along the bottom edge of the screen.
</p>

<hr>
<a name="more-on-zooming"></a>
<div class="header">
<p>
Previous: <a href="#Virtual-Trackball" accesskey="p" rel="previous">Virtual Trackball</a>, Up: <a href="#Mousing-and-Keyboarding" accesskey="u" rel="up">Mousing and Keyboarding</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="More-on-Zooming"></a>
<h3 class="section">2.9 More on Zooming</h3>
<p>The function <code>(quanta-like-zoom)</code> adds the ability to zoom the
view using just Shift + Mouse movement <a name="DOCF15" href="#FOOT15"><sup>15</sup></a>.
</p>
<p>There is also a Zoom slider
<a name="index-zoom_002c-slider"></a>
(<code>Draw -&gt; Zoom</code>) for those without a right-mouse button.
</p>

<hr>
<a name="General-Features"></a>
<div class="header">
<p>
Next: <a href="#Coordinate_002dRelated-Features" accesskey="n" rel="next">Coordinate-Related Features</a>, Previous: <a href="#Mousing-and-Keyboarding" accesskey="p" rel="previous">Mousing and Keyboarding</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="General-Features-1"></a>
<h2 class="chapter">3 General Features</h2>

<p>The map-fitting and model-building tools can be accessed by using
<code>Calculate -&gt; Model/Fit/Refine&hellip;</code>.  Many functions have
tooltips <a name="DOCF16" href="#FOOT16"><sup>16</sup></a>
<a name="index-tooltips"></a>
describing the particular features and are documented in Chapter
<a href="#Modelling-and-Building">Modelling and Building</a>.
</p>
<dl compact="compact">
<dt><code>F5:</code></dt>
<dd><p>posts the Model/Fit/Refine dialog
</p></dd>
<dt><code>F6:</code></dt>
<dd><p>posts the Go To Atom Window
</p></dd>
<dt><code>F7:</code></dt>
<dd><p>posts the Display Control Window
</p></dd>
</dl>


<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Version-number" accesskey="1">Version number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Antialiasing" accesskey="2">Antialiasing</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Molecule-Number" accesskey="3">Molecule Number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Display-Issues" accesskey="4">Display Issues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Screenshot" accesskey="5">Screenshot</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Raster3D" accesskey="6">Raster3D</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Display-Manager" accesskey="7">Display Manager</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#The-Modelling-Toolbar" accesskey="8">The Modelling Toolbar</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#The-file-selector" accesskey="9">The file selector</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scripting">Scripting</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Backups-and-Undo">Backups and Undo</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#View-Matrix">View Matrix</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Space-Group-and-Symmetry">Space Group and Symmetry</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Recentring-View">Recentring View</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Views">Views</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Clipping-Manipulation">Clipping Manipulation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Background-colour">Background colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Unit-Cell">Unit Cell</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotation-Centre-Pointer">Rotation Centre Pointer</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Orientation-Axes">Orientation Axes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pointer-Distances">Pointer Distances</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Crosshairs">Crosshairs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#g_t3D-Annotations">3D Annotations</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Frame-Rate">Frame Rate</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Program-Output">Program Output</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Version-number"></a>
<div class="header">
<p>
Next: <a href="#Antialiasing" accesskey="n" rel="next">Antialiasing</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Version-number-1"></a>
<h3 class="section">3.1 Version number</h3>
<a name="index-version-number"></a>

<p>The version number of Coot can be found at the top of the &ldquo;About&rdquo;
window (<code>Help -&gt; About</code>).
</p>
<p>This will return the version of coot:
</p>
<p><code>$ coot --version</code>
</p>
<p>There is also a script function to return the version of coot:
</p>
<p><code>(coot-version)</code>
</p>
<hr>
<a name="Antialiasing"></a>
<div class="header">
<p>
Next: <a href="#Molecule-Number" accesskey="n" rel="next">Molecule Number</a>, Previous: <a href="#Version-number" accesskey="p" rel="previous">Version number</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Antialiasing-1"></a>
<h3 class="section">3.2 Antialiasing</h3>

<p>The built-in antialiasing (for what it&rsquo;s worth) can be enabled using:
</p>
<p><code>(set-do-anti-aliasing 1)</code>
</p>
<p>The default is <code>0</code> (off).
</p>
<p>This can also be activated using <code>Edit Preferences -&gt; Others -&gt;
Antialiasing -&gt; Yes</code>.
</p>
<p>If you have an nVidia graphics card, external antialiasing can be
actived setting the environment variable <tt>__GL_FSAA_MODE</tt>.  For me a
setting of <tt>5</tt> works nicely and gives a better image than using
Coot&rsquo;s built-in antialiasing.
</p>

<p>Also for nVidia graphics card users, there is the application
nvidia-settings: 
</p>
<p><code>Antialiasing Setting -&gt; Override Application Settings</code> and
slide the slider to the right. On restarting Coot, it should be in
antialias mode <a name="DOCF17" href="#FOOT17"><sup>17</sup></a>.
</p>
<hr>
<a name="Molecule-Number"></a>
<div class="header">
<p>
Next: <a href="#Display-Issues" accesskey="n" rel="next">Display Issues</a>, Previous: <a href="#Antialiasing" accesskey="p" rel="previous">Antialiasing</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Molecule-Number-1"></a>
<h3 class="section">3.3 Molecule Number</h3>
<a name="index-molecule-number"></a>
<p>Coot is based on the concept of molecules.  Maps and coordinates are
different representations of molecules.  The access to the molecule is
<em>via</em> the <em>molecule number</em>.  It is often important therefore to
know the molecule number of a particular molecule.
</p>
<p>The Molecule Number of a molecule can be found by clicking on an atom of
that molecule (if it has coordinates of course). The first number in
brackets in the resulting text in the status bar and console is the
Molecule Number.  The Molecule Number can also be found in Display
Control window (Section <a href="#Display-Manager">Display Manager</a>).  It is also displayed on
the left-hand side of the molecule name in the option menus of the
&ldquo;Save Coordinates&rdquo; and &ldquo;Go To Atom&rdquo; windows.
</p>
<hr>
<a name="Display-Issues"></a>
<div class="header">
<p>
Next: <a href="#Screenshot" accesskey="n" rel="next">Screenshot</a>, Previous: <a href="#Molecule-Number" accesskey="p" rel="previous">Molecule Number</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Display-Issues-1"></a>
<h3 class="section">3.4 Display Issues</h3>
<p>The &ldquo;graphics&rdquo; window is drawn using 
<a name="index-OpenGL"></a>
OpenGL.  It is considerably smoother (i.e. more frames/sec) when using a
3D accelerated X server.
</p>
<p>The view is orthographic (<em>i.e.</em> the back is the same size as the
front).  The default clipping is about right for viewing coordinate
data, but is often a little too &ldquo;thick&rdquo; for viewing electron
density.  It is easily changed (see Section <a href="#Clipping-Manipulation">Clipping Manipulation</a>).
</p>
<p>Depth-cueing
<a name="index-depth_002dcueing"></a>
is linear and fixed on.
</p>
<p>The graphics window can be resized, but it has a minimum size of
400x400 pixels.
</p>
<a name="Stereo"></a>
<h4 class="subsection">3.4.1 Stereo</h4>
<p>Hardware Stereo is an option for Coot (<code>Draw -&gt; Stereo&hellip; -&gt;
Hardware Stereo -&gt; OK</code>), side-by-side stereo is not an option.
</p>
<p>The angle between the stereo pairs (the stereo separation) can be
changed to suit your personal tastes using:
</p>
<p><code>(set-hardware-stereo-angle-factor <em>angle-factor</em>)</code>
</p>
<p>where <code><em>angle-factor</em></code> would typically be between 1.0 and 2.0
</p>
<a name="Pick-Cursor"></a>
<h4 class="subsection">3.4.2 Pick Cursor</h4>
<p>When asked to pick a residue or atom, the cursor changes from the normal
arrow shape to a &quot;pick&quot; cursor.  Sometimes it is difficult to see the
default pick cursor, so you can change it using the function
</p>
<p><code>(set-pick-cursor-index <em>i</em>)</code>
</p>
<p>where <code><em>i</em></code> is an integer less than 256.  The cursors can be
viewed using an external X program:
</p>
<p><code>xfd -fn cursor</code>
</p>

<a name="Origin-Marker"></a>
<h4 class="subsection">3.4.3 Origin Marker</h4>

<p>A yellow box
<a name="index-yellow-box"></a>
 called the &ldquo;origin
marker&rdquo;
<a name="index-origin-marker"></a>
 marks the origin.  It can be removed using:
</p>
<p><code>(set-show-origin-marker 0)</code>
</p>
<p>Its state can be queried like this:
</p>
<p><code>(show-origin-marker-state)</code>
</p>
<p>which returns an number (0 if it is not displayed, 1 if it is).
</p>
<hr>
<a name="Screenshot"></a>
<div class="header">
<p>
Next: <a href="#Raster3D" accesskey="n" rel="next">Raster3D</a>, Previous: <a href="#Display-Issues" accesskey="p" rel="previous">Display Issues</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Screenshot-1"></a>
<h3 class="section">3.5 Screenshot</h3>
<a name="index-screenshot_002e"></a>

<p>A simple screenshot (image dump) can be made using <code>Draw -&gt;
Screenshot -&gt; Simple&hellip;</code>.  Note that in side by side stereo mode you
only get the left-hand image.
</p>

<hr>
<a name="Raster3D"></a>
<div class="header">
<p>
Next: <a href="#Display-Manager" accesskey="n" rel="next">Display Manager</a>, Previous: <a href="#Screenshot" accesskey="p" rel="previous">Screenshot</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Raster3D-output"></a>
<h3 class="section">3.6 Raster3D output</h3>
<p>Output suitable for use by Raster3D&rsquo;s
<a name="index-Raster3D"></a>
&ldquo;render&rdquo;
<a name="index-render"></a>
can be generated using the scripting function
</p>
<p><code>(raster3d <em>file-name</em>)</code>
</p>
<p>where <code><em>file-name</em></code> is such as <code>&quot;test.r3d&quot;</code>
<a name="DOCF18" href="#FOOT18"><sup>18</sup></a>.
</p>
<p>There is a keyboard key to generate this file, run &ldquo;render&rdquo; and
display the image: Function key F8.
</p>
<p>You can also use the function
</p>
<p><code>(render-image)</code>
</p>
<p>which will create a file <samp>coot.r3d</samp>, from which &ldquo;render&rdquo; produces
<samp>coot.png</samp>. This png file is displayed using ImageMagick&rsquo;s display
program (by default).  Use something like:
</p>
<p><code>(set! coot-png-display-program &quot;gqview&quot;)</code>
</p>
<p>to change that to different display program (&quot;gqview&quot; in this case).
</p>
<p><code>(set! coot-png-display-program &quot;open&quot;)</code> 
</p>
<p>would use Preview (by default) on Macintosh.
</p>
<p>To change the widths of the bonds and density &ldquo;lines&rdquo; use (for example):
</p>
<p><code>(set-raster3d-bond-thickness 0.1)</code>
</p>
<p>and 
</p>
<p><code>(set-raster3d-density-thickness 0.01)</code>
</p>
<p>Similarly for bones:
</p>
<p><code>(set-raster3d-bone-thickness 0.05)</code>
</p>

<p>To turn off the representations of the atoms (spheres):
</p>
<p><code>(set-renderer-show-atoms 0)</code>
</p>

<hr>
<a name="Display-Manager"></a>
<div class="header">
<p>
Next: <a href="#The-Modelling-Toolbar" accesskey="n" rel="next">The Modelling Toolbar</a>, Previous: <a href="#Raster3D" accesskey="p" rel="previous">Raster3D</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Display-Manager-1"></a>
<h3 class="section">3.7 Display Manager</h3>
<a name="index-Display-Manager"></a>
<p>This is also known as &ldquo;Map and molecule
(coordinates) display control&rdquo;.  Here you can select which maps and
molecules you can see and how they are drawn <a name="DOCF19" href="#FOOT19"><sup>19</sup></a>.  The &ldquo;Display&rdquo; and &ldquo;Active&rdquo; are toggle buttons, either
depressed (active) or undepressed (inactive).  The &ldquo;Display&rdquo; buttons
control whether a molecule (or map) is drawn and the &ldquo;Active&rdquo; button
controls if the molecule is clickable <a name="DOCF20" href="#FOOT20"><sup>20</sup></a>
(<em>i.e.</em> if the molecule&rsquo;s atoms can be labeled).
</p>
<a name="index-Scroll"></a>
<p>The &quot;Scroll&quot; radio buttons sets which map is has its contour level
changed by scrolling the mouse scroll wheel.
</p>
<p>By default, the path names of the files are not displayed in the
Display Manager.  To turn them on:
</p>
<p><code>(set-show-paths-in-display-manager 1)</code>
</p>
<a name="index-colour-by-chain"></a>
<a name="index-atom-colouring"></a>
<p>If you pull across the
horizontal scrollbar in a Molecule view, you will see the &ldquo;Render
as&rdquo; menu.  You can use this to change between normal &ldquo;Bonds (Colour
by Atom)&rdquo;,&ldquo;Bonds (Colour by Chain)&rdquo; and &ldquo;C<em>\alpha</em>&rdquo;
representation
<a name="index-C_005calpha-representation"></a>
There is also
available &ldquo;No Waters&rdquo; and &ldquo;C<em>\alpha</em> + ligands&rdquo; representations.
</p>

<hr>
<a name="The-Modelling-Toolbar"></a>
<div class="header">
<p>
Next: <a href="#The-file-selector" accesskey="n" rel="next">The file selector</a>, Previous: <a href="#Display-Manager" accesskey="p" rel="previous">Display Manager</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="The-Modelling-Toolbar-1"></a>
<h3 class="section">3.8 The Modelling Toolbar</h3>
<a name="index-modelling-toolbar"></a>

<p>You might not want to have the right-hand-side vertical toolbar that
contains icons for some modelling operations <a name="DOCF21" href="#FOOT21"><sup>21</sup></a> displayed:
</p>
<p><code>(hide-modelling-toolbar)</code>
</p>
<p>to bring it back again:
</p>
<p><code>(show-modelling-toolbar)</code>
</p>


<hr>
<a name="The-file-selector"></a>
<div class="header">
<p>
Next: <a href="#Scripting" accesskey="n" rel="next">Scripting</a>, Previous: <a href="#The-Modelling-Toolbar" accesskey="p" rel="previous">The Modelling Toolbar</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="The-file-selector-1"></a>
<h3 class="section">3.9 The file selector</h3>
<a name="File_002dname-Filtering"></a>
<h4 class="subsection">3.9.1 File-name Filtering</h4>
<a name="index-file_002dname-filtering"></a>
<p>The &ldquo;Filter&rdquo; button in the fileselection
filters the filenames according to extension.  For coordinates files
the extensions are &ldquo;.pdb&rdquo; &ldquo;.brk&rdquo; &ldquo;.mmcif&rdquo; and others.  For data:
&ldquo;.mtz&rdquo;, &ldquo;.hkl&rdquo;, &ldquo;.phs&rdquo;, &ldquo;.cif&rdquo; and for (CCP4) maps &ldquo;.ext&rdquo;,
&ldquo;.msk&rdquo; and &ldquo;.map&rdquo;.  If you want to add to the extensions, the
following functions are available:
</p>
<ul>
<li> <code>(add-coordinates-glob-extension <em>extension</em>)</code>
</li><li> <code>(add-data-glob-extension <em>extension</em>)</code>
</li><li> <code>(add-map-glob-extension <em>extension</em>)</code>
</li><li> <code>(add-dictionary-glob-extension <em>extension</em>)</code>
</li></ul>
<p>where <code><em>extension</em></code> is something like: <code>&quot;.mycif&quot;</code>.
</p>
<p>If you want the fileselection to be filtered without having to use the
&quot;Filter&quot; button, use the scripting function
</p>
<p><code>(set-filter-fileselection-filenames 1)</code>
</p>
<a name="Filename-Sorting"></a>
<h4 class="subsection">3.9.2 Filename Sorting</h4>
<p>If you like your files initially sorted by date (rather than
lexicographically, which is the default) use:
</p>
<p><code>(set-sticky-sort-by-date)</code>
</p>
<a name="Save-Coordinates-Directory"></a>
<h4 class="subsection">3.9.3 Save Coordinates Directory</h4>
<a name="index-save-coordinates-directory"></a>
<a name="index-directory-for-save-coordinates"></a>

<p>Some people prefer that the fileselection for saving coordinates starts
in the original directory (rather than the directory from which they
last imported coordinates).  This option is for them:
</p>
<p><code>(set-save-coordinates-in-original-directory 1)</code>
</p>

<hr>
<a name="Scripting"></a>
<div class="header">
<p>
Next: <a href="#Backups-and-Undo" accesskey="n" rel="next">Backups and Undo</a>, Previous: <a href="#The-file-selector" accesskey="p" rel="previous">The file selector</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Scripting-1"></a>
<h3 class="section">3.10 Scripting</h3>
<a name="index-scripting"></a>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Python" accesskey="1">Python</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scheme" accesskey="2">Scheme</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Coot-State" accesskey="3">Coot State</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Key-Binding" accesskey="4">Key Binding</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#User_002dDefined-Functions" accesskey="5">User-Defined Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<p>There is an compile-time option of adding a script interpreter.
Currently the options are python and guile.  It seems possible that in
future you will be able to use both in the same executable.  The binary
distribution of Coot are linked with guile, others with python.
</p>
<p>Hundreds of commands are made available for use in scripting by using
SWIG, some of which are documented here. Other functions documented less
well, but descriptions for them can be found at the end of this manual.
</p>
<p>Commands described throughout this manual (such as <code>(vt-surface
  1))</code> can be evaluated
<a name="index-executing-commands"></a>
directly by Coot by using the &ldquo;Scripting Window&rdquo; (<code>Calculate
-&gt; Scripting&hellip;</code>). Note that you type the commands in the upper
entry widget and the command gets echoed (in red) and the return value
and any output is displayed in the text widget lower (green).  The typed
command should be terminated with a carriage return <a name="DOCF22" href="#FOOT22"><sup>22</sup></a>.  Files <a name="DOCF23" href="#FOOT23"><sup>23</sup></a> can be evaluated (executed)
using <code>Calculate -&gt; Run Script&hellip;</code>.  
</p>
<p>Note that in scheme (the usual scripting language of Coot), the
parentheses are important.
</p>
<p>To execute a script file from the command line use the <code>--script
<em>filename</em></code> arguments 
(except when also using the command line
argument <code>--no-graphics</code>, in which case you should use <code>-s
<em>filename</em></code>).
</p>
<p>After you have used the scripting window, you may have noticed that you
can no longer kill Coot by using Ctrl-C in the console.  To recover this
ability:
</p>
<p><code>(exit)</code>
</p>
<p>in the scripting window.
</p>
<hr>
<a name="Python"></a>
<div class="header">
<p>
Next: <a href="#Scheme" accesskey="n" rel="next">Scheme</a>, Up: <a href="#Scripting" accesskey="u" rel="up">Scripting</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Python-1"></a>
<h4 class="subsection">3.10.1 Python</h4>
<a name="index-python"></a>

<table class="menu" border="0" cellspacing="0">
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment"> * Python Commands
</pre></th></tr></table>

<p>Coot has an (optional) embedded python interpreter.
Thus the full power of python is available to you.  Coot will look for
an initialization script 
<a name="index-startup-settings-_0028python_0029"></a>
<a name="index-_002ecoot_002epy"></a>
(<code>$HOME/.coot.py</code>) and
will execute it if found.  This file should contain python commands
that set your personal preferences.
</p>

<a name="Python-Commands"></a>
<h4 class="subsubsection">3.10.1.1 Python Commands</h4>
<p>The scripting functions described in this manual are formatted
suitable for use with guile, <em>i.e.</em>:
</p>
<p><code>(<em>function</em> <em>arg1</em> <em>arg2&hellip;</em>)</code>
</p>
<p>If you are using Python instead: the format needs to be changed to:
</p>
<p><code><em>function</em>(<em>arg1</em>,<em>arg2&hellip;</em>)</code>
</p>
<p>Note that dashes in guile function names become underscores for
python, so that (for example) <code>(raster-screen-shot)</code> becomes
<code>raster_screen_shot()</code>.
</p>

<hr>
<a name="Scheme"></a>
<div class="header">
<p>
Next: <a href="#Coot-State" accesskey="n" rel="next">Coot State</a>, Previous: <a href="#Python" accesskey="p" rel="previous">Python</a>, Up: <a href="#Scripting" accesskey="u" rel="up">Scripting</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Scheme-1"></a>
<h4 class="subsection">3.10.2 Scheme</h4>
<a name="index-guile"></a>
<a name="index-scheme"></a>
<a name="index-initialization-file"></a>
<p>The scheme interpreter is made available by embedding
guile.  The initialization script used by this interpreter is
<a name="index-startup-settings-_0028scheme_0029"></a>
<a name="index-_002ecoot"></a>
<code>$HOME/.coot</code>.  This file should contain scheme commands that
set your personal preferences.
</p>
<hr>
<a name="Coot-State"></a>
<div class="header">
<p>
Next: <a href="#Key-Binding" accesskey="n" rel="next">Key Binding</a>, Previous: <a href="#Scheme" accesskey="p" rel="previous">Scheme</a>, Up: <a href="#Scripting" accesskey="u" rel="up">Scripting</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Coot-State-1"></a>
<h4 class="subsection">3.10.3 Coot State</h4>
<p>The &ldquo;state&rdquo;
<a name="index-state"></a>
of Coot is saved on Exit and written to a
file called <code>0-coot.state.scm</code> (scheme)
<code>0-coot.state.py</code> (python).   This
state file contains information about the screen centre, the
clipping, colour map rotation size, the symmetry radius, and other
molecule related parameters such as filename, column labels,
coordinate filename <em>etc.</em>.
</p>
<p>Use <code>Calculate -&gt; Run Script&hellip;</code> to use this file
to re-create the loaded maps and models that you had when you finished
using Coot <a name="DOCF24" href="#FOOT24"><sup>24</sup></a> last time.
A state file can be saved at any time using <code>(save-state)</code>
which saves to file <code>0-coot.state.scm</code> or
<code>(save-state-filename &quot;thing.scm&quot;)</code> which saves to file
<code>thing.scm</code>.
</p>
<p>When Coot starts it can optionally run the commands in
<code>0-coot.state.scm</code>.  
</p>
<p>Use <code>(set-run-state-file-status i)</code>
to change the behaviour: <code>i</code> is <code>0</code> to never run this
state file at 
<a name="index-startup-dialog-_0028state_0029"></a>
startup, <code>i</code> is
<code>1</code> to get a dialog option (this is the default) and <code>i</code>
is <code>2</code> to run the commands without question.
</p>
<hr>
<a name="Key-Binding"></a>
<div class="header">
<p>
Next: <a href="#User_002dDefined-Functions" accesskey="n" rel="next">User-Defined Functions</a>, Previous: <a href="#Coot-State" accesskey="p" rel="previous">Coot State</a>, Up: <a href="#Scripting" accesskey="u" rel="up">Scripting</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Key-Binding-1"></a>
<h4 class="subsection">3.10.4 Key Binding</h4>

<p>&ldquo;Power users&rdquo; of Coot might like to write their own functions and bind
that function to a keyboard key.  How do they do that?
</p>
<p>By using the add-key-binding function:
</p>
<p><code>(add-key-binding <em>function-name key function</em>)</code>
</p>
<p>where <code><em>key</em></code> is a quoted string (note that upper case and
lower case keys are distinguished - activate get upper case key binding
you need to chord the shift key <a name="DOCF25" href="#FOOT25"><sup>25</sup></a>).
</p>
<p>for example:
</p>
<p><code>(add-key-binding &quot;Refine Active Residue with Auto-accept&quot; &quot;x&quot; refine-active-residue)</code>
</p> 
<p>Have a look at the key bindings section on the Coot wiki for several
more examples.
</p>
<hr>
<a name="User_002dDefined-Functions"></a>
<div class="header">
<p>
Previous: <a href="#Key-Binding" accesskey="p" rel="previous">Key Binding</a>, Up: <a href="#Scripting" accesskey="u" rel="up">Scripting</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="User_002dDefined-Functions-1"></a>
<h4 class="subsection">3.10.5 User-Defined Functions</h4>

<p>&ldquo;Power users&rdquo; of Coot might also like to write their own functions
that occur after picking an atom (or a number of atoms)
</p>
<p><code>(user-defined-click <em>n_clicks</em> <em>udfunc</em>)</code>
</p>
<p>define a function <code><em>func</em></code> which runs after the user has
made <code><em>n_clicked</em></code> atom picks.  <code><em>func</em></code> is called
with a list of atom specifiers - the first member of which is the
molecule number.
</p>

<hr>
<a name="Backups-and-Undo"></a>
<div class="header">
<p>
Next: <a href="#View-Matrix" accesskey="n" rel="next">View Matrix</a>, Previous: <a href="#Scripting" accesskey="p" rel="previous">Scripting</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Backups-and-Undo-1"></a>
<h3 class="section">3.11 Backups and Undo</h3>
<a name="index-backups"></a>
<a name="index-undo"></a>

<table class="menu" border="0" cellspacing="0">
<tr><th colspan="3" align="left" valign="top"><pre class="menu-comment"> * Redo::
 * Restoring from Backup::
</pre></th></tr></table>

<p>By default, each time a modification is made to a model, the old
coordinates are written out <a name="DOCF26" href="#FOOT26"><sup>26</sup></a>.  The backups are kept in a
backup directory and are tagged with the date and the history number
(lower numbers are more ancient <a name="DOCF27" href="#FOOT27"><sup>27</sup></a>).  The &ldquo;Undo&rdquo; function discards
the current molecule and loads itself from the most recent backup
coordinates.  Thus you do not have to remember to &ldquo;Save Changes&rdquo; -
coot will do it for you <a name="DOCF28" href="#FOOT28"><sup>28</sup></a>.
</p>
<p>If you have made changes to more than one molecule, Coot will pop-up a
dialog box in which you should set the &ldquo;Undo Molecule&rdquo; <em>i.e.</em>
the molecule to which the Undo operations will apply.  Further Undo
operations will continue to apply to this molecule until there are
none left.  If another Undo is requested Coot checks to see if there
are other molecules that can be undone, if there is exactly one, then
that molecule becomes the &ldquo;Undo Molecule&rdquo;, if there are more than
one, then another Undo selection dialog will be displayed.
</p>
<p>You can set the undo molecule using the scripting function:
</p>
<p><code>(set-undo-molecule <em>imol</em>)</code>
</p>
<p>If for reasons of strange system<a name="DOCF29" href="#FOOT29"><sup>29</sup></a>
requirements you want to remove the path components of the backup file
name you can do so using:
</p>
<p><code>(set-unpathed-backup-file-names 1)</code>
</p>
<a name="Redo"></a>
<h4 class="subsection">3.11.1 Redo</h4>
<a name="index-redo"></a>
<p>The &ldquo;undone&rdquo; modifications can be re-done using this
button.  This is not available immediately after a
modification <a name="DOCF30" href="#FOOT30"><sup>30</sup></a>.
</p>
<a name="Restoring-from-Backup"></a>
<h4 class="subsection">3.11.2 Restoring from Backup</h4>
<a name="index-restore-after-crash"></a>
<p>There may be certain
circumstances <a name="DOCF31" href="#FOOT31"><sup>31</sup></a> in which you
wish to restore from a backup but can&rsquo;t get it by the &ldquo;Undo&rdquo;
mechanism described above.  In that case, start coot as normal and
then open the (typically most recent) coordinates file in the
directory <code>coot-backup</code> (or the directory pointed to the
environment variable <code>COOT_BACKUP_DIR</code> if it was set) .
This file should contain your most recent edits.  In such a case, it
is sensible for neatness purposes to immediately save the coordinates
(probably to the current directory) so that you are not modifying a
file in the backup directory.
</p>
<p>See also Section <a href="#Crash">Crash</a>.
</p>
<hr>
<a name="View-Matrix"></a>
<div class="header">
<p>
Next: <a href="#Space-Group-and-Symmetry" accesskey="n" rel="next">Space Group and Symmetry</a>, Previous: <a href="#Backups-and-Undo" accesskey="p" rel="previous">Backups and Undo</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="View-Matrix-1"></a>
<h3 class="section">3.12 View Matrix</h3>
<a name="index-view-matrix"></a>
<p>It is sometimes useful to use this to orient the
view and export this orientation to other programs.  The orientation
matrix of the view can be displayed (in the console) using:
</p>
<p><code>(view-matrix)</code>
</p>
<p>Also, the internal representation of the view can be returned and set
using:
</p>
<p><code>(view-quaternion)</code> to return a 4-element list
</p>
<p><code>(set-view-quaternion i j k l)</code> which sets the view quaternion.
</p>
<p>So the usage of these functions would be something like:
</p>
<pre class="verbatim">(let ((v (view-quaternion)))
   ;; manipulate v here, maybe
   (apply set-view-quaternion v))
</pre>
<hr>
<a name="Space-Group-and-Symmetry"></a>
<div class="header">
<p>
Next: <a href="#Recentring-View" accesskey="n" rel="next">Recentring View</a>, Previous: <a href="#View-Matrix" accesskey="p" rel="previous">View Matrix</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Space-Group-and-Symmetry-1"></a>
<h3 class="section">3.13 Space Group and Symmetry</h3>
<p>Occasionally you may want to know the space group of a particular
molecule.  Interactively (for maps) you can see it using the Map
Properties button in the Molecule Display Control dialog.
</p>
<p>There is a scripting interface function that returns the space group
for a given molecule  <a name="DOCF32" href="#FOOT32"><sup>32</sup></a>:
</p>
<p><code>(show-spacegroup <em>imol</em>)</code>
</p>
<p>You can force a space group onto a molecule using the following:
</p>
<p><code>(set-space-group <em>imol</em> <em>space-group</em>)</code>
</p>
<p>where <code><em>space-group</em></code> is one of the standard CCP4 space group
names (<em>e.g.</em> <code><tt>&quot;P 21 21 21&quot;</tt></code>).
</p>
<a name="index-space-group-operators"></a>
<a name="index-symmetry-operators"></a>
<p>To show the symmetry operators of a particular molecule use:
<code>(get-symmetry <em>imol</em>)</code>
which will return a list of strings.
</p>
<p>Sometimes molecular replacement solutions (for example) create models
with chains non-optimally placed relative to each other - a
symmetry-related copy would be more apealling (but would be equivalent,
crystalographically). For example, to move the B chain to a
symmetry-related position:
</p>
<p>Centre on an atom in the symmetry-related B chain (where you want the B chain to be)
</p>
<p><code>Extensions -&gt; Modelling -&gt; Symm Shift Reference Chain Here.</code>
</p>
<hr>
<a name="Recentring-View"></a>
<div class="header">
<p>
Next: <a href="#Views" accesskey="n" rel="next">Views</a>, Previous: <a href="#Space-Group-and-Symmetry" accesskey="p" rel="previous">Space Group and Symmetry</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Recentring-View-1"></a>
<h3 class="section">3.14 Recentring View</h3>
<a name="index-recentring-view"></a>
<ul>
<li> Use Control + left-mouse to drag around the view
</li><li> or
</li><li> middle-mouse over an atom.  In this case, you will often see
  &ldquo;slide-recentring&rdquo;, the graphics smoothly changes between the
  current centre and the newly selected centre.
</li><li> or
</li><li> Use <code>Draw -&gt; Go To Atom&hellip;</code> to select an atom
  using the keyboard.  Note that you can subsequently use &ldquo;Space&rdquo; in
  the &ldquo;graphics&rdquo; window (OpenGL canvas) to recentre on the next
  C<em>\alpha</em>.
</li><li> or
</li><li> <a name="index-set_002drotation_002dcentre"></a>
<a name="index-rotation-centre"></a>
To centre on an arbitrary position (x,y,z), use the scripting function
   <code>(set-rotation-centre x y z)</code>.
</li><li> or
</li><li> Use the keyboard: [Ctrl G] then key in a residue number and (optionally) a chainid 
      and press Return
</li></ul>

<p>If you don&rsquo;t want smooth recentring (sliding)
<a name="index-sliding"></a>
<code>Edit -&gt; Preferences -&gt; Smooth Recentring -&gt; Off</code>.  You
can also use this dialog to speed it up a bit (by decreasing the
number of steps instead of turning it off).
</p>
<hr>
<a name="Views"></a>
<div class="header">
<p>
Next: <a href="#Clipping-Manipulation" accesskey="n" rel="next">Clipping Manipulation</a>, Previous: <a href="#Recentring-View" accesskey="p" rel="previous">Recentring View</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Views-1"></a>
<h3 class="section">3.15 Views</h3>

<p>Coot has a views interface (you might call them &rdquo;scenes&ldquo;) that
define a particular orientation, zoom and view centre.  Coot and
linearly interpolate between the views.  The animation play back speed
can be set with the &rdquo;Views Play Speed&ldquo; menu item - default is a
speed of 10.
</p>
<p>The views interface can be found under the Extensions menu item.
</p>
<hr>
<a name="Clipping-Manipulation"></a>
<div class="header">
<p>
Next: <a href="#Background-colour" accesskey="n" rel="next">Background colour</a>, Previous: <a href="#Views" accesskey="p" rel="previous">Views</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Clipping-Manipulation-1"></a>
<h3 class="section">3.16 Clipping Manipulation</h3>
<a name="index-clipping"></a>
<p>The clipping planes (a.k.a. &ldquo;slab&rdquo;
<a name="index-slab"></a>
) can
be adjusted using <code>Edit -&gt; Clipping</code> and adjusting
the slider.  There is only one parameter to change and it affects both
the front and the back clipping planes <a name="DOCF33" href="#FOOT33"><sup>33</sup></a>.
The clipping can also be changed using keyboard &ldquo;D&rdquo; and &ldquo;F&rdquo;.  
</p>
<p>It can also be changed with Ctrl + Right-mouse drag up and down.
Likewise the screen-Z can be changed with Ctrl + Right-mouse left and
right <a name="DOCF34" href="#FOOT34"><sup>34</sup></a>.
</p>

<p>One can &ldquo;push&rdquo; and &ldquo;pull&rdquo; the view in the screen-Z direction using
keypad 3 and keypad &ldquo;.&rdquo; (see Section <a href="#Keyboard-Z-Translation">Keyboard Z Translation</a>).
</p>

<hr>
<a name="Background-colour"></a>
<div class="header">
<p>
Next: <a href="#Unit-Cell" accesskey="n" rel="next">Unit Cell</a>, Previous: <a href="#Clipping-Manipulation" accesskey="p" rel="previous">Clipping Manipulation</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Background-colour-1"></a>
<h3 class="section">3.17 Background colour</h3>
<a name="index-background-colour"></a>
<p>The background colour can be set either using a GUI dialog
(<code>Edit$ -&gt; Background Colour</code>) or the function
<code>(set-background-colour 0.00 0.00 0.00)</code>, where the arguments
are 3 numbers between 0.0 and 1.0, which respectively represent the
red, green and blue components of the background colour.  The default
is (0.0, 0.0, 0.0) (black).
</p>
<hr>
<a name="Unit-Cell"></a>
<div class="header">
<p>
Next: <a href="#Rotation-Centre-Pointer" accesskey="n" rel="next">Rotation Centre Pointer</a>, Previous: <a href="#Background-colour" accesskey="p" rel="previous">Background colour</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Unit-Cell-1"></a>
<h3 class="section">3.18 Unit Cell</h3>
<a name="index-unit-cell"></a>
<p>If coordinates have symmetry available then unit
cells can be drawn for molecules (<code>Draw -&gt; Cell &amp;
  Symmetry -&gt; Show Unit Cell?</code>).
</p>
<hr>
<a name="Rotation-Centre-Pointer"></a>
<div class="header">
<p>
Next: <a href="#Orientation-Axes" accesskey="n" rel="next">Orientation Axes</a>, Previous: <a href="#Unit-Cell" accesskey="p" rel="previous">Unit Cell</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rotation-Centre-Pointer-1"></a>
<h3 class="section">3.19 Rotation Centre Pointer</h3>
<a name="index-rotation-centre-pointer"></a>
<p>There is a pink pointer
<a name="index-pink-pointer"></a>
at the centre of the screen that marks the rotation centre.
The size of the pointer can be changed using <code>Edit
  -&gt; Pink Pointer Size&hellip;</code> or using scripting commands:
<code>(set-rotation-centre-size 0.3)</code>.
</p>
<hr>
<a name="Orientation-Axes"></a>
<div class="header">
<p>
Next: <a href="#Pointer-Distances" accesskey="n" rel="next">Pointer Distances</a>, Previous: <a href="#Rotation-Centre-Pointer" accesskey="p" rel="previous">Rotation Centre Pointer</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Orientation-Axes-1"></a>
<h3 class="section">3.20 Orientation Axes</h3>
<a name="index-Orientation-Axes"></a>

<p>The green axes showing the orientation of the molecule are displayed
by default.  To remove them use the scripting function;
</p>
<p><code>(set-draw-axes 0)</code>
</p>

<hr>
<a name="Pointer-Distances"></a>
<div class="header">
<p>
Next: <a href="#Crosshairs" accesskey="n" rel="next">Crosshairs</a>, Previous: <a href="#Orientation-Axes" accesskey="p" rel="previous">Orientation Axes</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Pointer-Distances-1"></a>
<h3 class="section">3.21 Pointer Distances</h3>
<p>The Rotation Centre Pointer is sometimes called simply &ldquo;Pointer&rdquo;.
One can find distances to the pointer from any active set of atoms
using &ldquo;Pointer Distances&rdquo; (under Measures).  If you move the Pointer
(<em>e.g.</em> by centering on an atom) and want to update the distances
to it, you have to toggle off and on the &ldquo;Show Pointer Distances&rdquo; on
the Pointer Distances dialog.
</p>
<hr>
<a name="Crosshairs"></a>
<div class="header">
<p>
Next: <a href="#g_t3D-Annotations" accesskey="n" rel="next">3D Annotations</a>, Previous: <a href="#Pointer-Distances" accesskey="p" rel="previous">Pointer Distances</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Crosshairs-1"></a>
<h3 class="section">3.22 Crosshairs</h3>
<a name="index-crosshairs"></a>
<p>Crosshairs can be drawn at the centre of the screen, using either the
<tt class="key">C</tt> key<a name="DOCF35" href="#FOOT35"><sup>35</sup></a> in graphics
window or <code>Draw -&gt; Crosshairs&hellip;</code>.  The ticks are at
1.54&Aring;, 2.7&Aring; and 3.8&Aring;.
</p>
<hr>
<a name="g_t3D-Annotations"></a>
<div class="header">
<p>
Next: <a href="#Frame-Rate" accesskey="n" rel="next">Frame Rate</a>, Previous: <a href="#Crosshairs" accesskey="p" rel="previous">Crosshairs</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="g_t3D-Annotations-1"></a>
<h3 class="section">3.23 3D Annotations</h3>
<a name="index-annotations"></a>
<p>Positions in 3D space can be annotated with 3D text.  The mechanism to
do this can be found under Extensions -&gt; Representations -&gt; 3D
Annotations.  3D Annotations can be saved to and loaded from a file.
</p>
<hr>
<a name="Frame-Rate"></a>
<div class="header">
<p>
Next: <a href="#Program-Output" accesskey="n" rel="next">Program Output</a>, Previous: <a href="#g_t3D-Annotations" accesskey="p" rel="previous">3D Annotations</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Frame-Rate-1"></a>
<h3 class="section">3.24 Frame Rate</h3>
<a name="index-frame-rate"></a>
<p>Sometimes, you might as yourself &ldquo;how fast is the computer?&rdquo; 
<a name="DOCF36" href="#FOOT36"><sup>36</sup></a>.  Using <code>Calculate -&gt;
Frames/Sec</code> you can see how fast the molecule is rotating, giving an
indication of graphics performance.  It is often better to use a map
that is more realistic and stop the picture whizzing round.  The output
is written to the status bar and the console, you need to give it a few
seconds to &ldquo;settle down&rdquo;.  It is best not to have other widgets
overlaying the GL canvas as you do this.
</p>
<p>The contouring elapsed time <a name="DOCF37" href="#FOOT37"><sup>37</sup></a> gives an indication of CPU performance.
</p>
<hr>
<a name="Program-Output"></a>
<div class="header">
<p>
Previous: <a href="#Frame-Rate" accesskey="p" rel="previous">Frame Rate</a>, Up: <a href="#General-Features" accesskey="u" rel="up">General Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Program-Output-1"></a>
<h3 class="section">3.25 Program Output</h3>
<a name="index-output"></a>
<p>Due to its &ldquo;in development&rdquo; nature (at the moment),
Coot produces a lot of &ldquo;console&rdquo; <a name="DOCF38" href="#FOOT38"><sup>38</sup></a> output - much of it debugging or
&ldquo;informational&rdquo;.  This will go away in due course.  You are advised
to run Coot so that you can see the console and the graphics window at
the same time, since feedback from atom clicking (for example) is
often written there rather than displayed in the graphics window.
</p>
<ul>
<li> Output that starts &ldquo;ERROR...&rdquo; is a programming problem (and
  ideally, you should never see it).
</li><li> Output that starts &ldquo;WARNING...&rdquo; means that something probably
unintended happened due to the unexpected nature of your input or
file(s).
</li><li> Output that starts &ldquo;DEBUG...&rdquo; has (obviously enough) been
  added to aid debugging.  Most of them should have been cleaned up
  before release, but as Coot is constantly being developed, a few may
  slip through.  Just ignore them.
</li></ul>


<hr>
<a name="Coordinate_002dRelated-Features"></a>
<div class="header">
<p>
Next: <a href="#Modelling-and-Building" accesskey="n" rel="next">Modelling and Building</a>, Previous: <a href="#General-Features" accesskey="p" rel="previous">General Features</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Coordinate_002dRelated-Features-1"></a>
<h2 class="chapter">4 Coordinate-Related Features</h2>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Reading-coordinates" accesskey="1">Reading coordinates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Info" accesskey="2">Atom Info</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Labeling" accesskey="3">Atom Labeling</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Colouring" accesskey="4">Atom Colouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Bond-Parameters" accesskey="5">Bond Parameters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Download-coordinates" accesskey="6">Download coordinates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Get-Coordinates-and-Map-from-EDS" accesskey="7">Get Coordinates and Map from EDS</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Save-coordinates" accesskey="8">Save coordinates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Setting-the-Space-Group" accesskey="9">Setting the Space Group</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Anisotropic-Atoms">Anisotropic Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Symmetry">Symmetry</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Sequence-View">Sequence View</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Print-Sequence">Print Sequence</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Environment-Distances">Environment Distances</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Distances-and-Angles">Distances and Angles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Zero-Occupancy-Marker">Zero Occupancy Marker</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atomic-Dots">Atomic Dots</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Ball-and-Stick-Representation">Ball and Stick Representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mean-and-Median-Temperature-Factors">Mean and Median Temperature Factors</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Secondary-Structure-Matching-_0028SSM_0029">Secondary Structure Matching (SSM)</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Least_002dSquares-Fitting">Least-Squares Fitting</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Ligand-Overlaying">Ligand Overlaying</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Writing-PDB-files">Writing PDB files</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Reading-coordinates"></a>
<div class="header">
<p>
Next: <a href="#Atom-Info" accesskey="n" rel="next">Atom Info</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Reading-coordinates-1"></a>
<h3 class="section">4.1 Reading coordinates</h3>
<p>The format
<a name="index-coordinates-format"></a>
of coordinates that can be read by coot is either PDB or mmCIF.  To read
coordinates, choose <code>File -&gt; Read Coordinates</code> from the
menu-bar.  Immediately after the coordinates have been read, the view is
(by default) recentred to the centre of this new molecule and the
molecule is displayed.  The recentring of the view after the coordinates
have been read can be turned off by unclicking the &quot;Recentre?&quot; 
radio-button.
</p>
<p>To disable the recentring of the view on reading a coordinates file via
scripting, use: <code>(set-recentre-on-read-pdb 0)</code>.  However, when
reading a coordinates file from a script it is just as good (if not
better) to use <code>(handle-read-draw-molecule-with-recentre
<em>filename</em> 0)</code> - the additional <code>0</code> means &ldquo;don&rsquo;t recentre&rdquo;.
And that affects just the reading of <code><em>filename</em></code> and not
subsequent files.
</p>
<p>By default coot does not allow reading coordinates with duplicated
sequence numbers.  To enable the reading of files with duplicated
sequence numbers use the function:
</p>
<p><code>(allow-duplicate-sequence-numbers)</code>
</p>
<p>Coot can read MDL mol files.
</p>
<a name="A-Note-on-Space-Groups-Names"></a>
<h4 class="subsection">4.1.1 A Note on Space Groups Names</h4>
<a name="index-space-group-names"></a>
<p>Coot uses the space group on the &ldquo;CRYST1&rdquo; line of the pdb file.  The
space group should be one of the xHM symbols listed (for example) in the
CCP4 dictionary file <samp>syminfo.lib</samp>.  So, for example, <tt>&quot;R 3 2 :H&quot;</tt>
should be used in preference to <tt>&quot;H32&quot;</tt>.
</p>
<a name="Read-multiple-coordinate-files"></a>
<h4 class="subsection">4.1.2 Read multiple coordinate files</h4>
<a name="index-reading-multiple-pdb-files"></a>
<a name="index-multiple-coordinates-files"></a>
<p>The reading multiple files using the GUI is not available (at the
moment).  However the following scripting functions are available:
</p>
<p><code>(read-pdb-all)</code>
</p>
<p>which reads all the &ldquo;*.pdb&rdquo; files in the current directory
</p>
<p><code>(multi-read-pdb <em>glob-pattern</em> <em>dir</em>)</code>
</p>
<p>which reads all the files matching <code><em>glob-pattern</em></code> in
directory <code><em>dir</em></code>.  Typical usage of this might be:
</p>
<p><code>(multi-read-pdb &quot;a*.pdb&quot; &quot;.&quot;)</code>
</p>
<p>Alternatively you can specify the files to be opened on the command
line when you start coot (see Section
<a href="#Command-Line-Arguments">Command Line Arguments</a>).
</p>
<a name="SHELX-_002eins_002f_002eres-files"></a>
<h4 class="subsection">4.1.3 SHELX .ins/.res files</h4>
<a name="index-SHELX-_002eins"></a>

<p>SHELX &quot;.res&quot; (and &quot;.ins&quot; of course) files can be read into Coot, either
using the GUI <code>File -&gt; Open Coordinates&hellip;</code> or by the
scripting function:
</p>
<p><code>(read-shelx-ins-file <em>file-name</em>)</code>
</p>
<p>where <code><em>file-name</em></code> is quoted, such as <code>&quot;thox.ins&quot;</code>.
</p>
<p>Although Coot should be able to read any SHELX &quot;.res&quot; file, it may
currently have trouble displaying the bonds for centro-symmetric
structures.
</p>
<p>ShelxL atoms with negative <code>PART</code> numbers are given alternative
configuration identifiers in lower case.
</p>
<p>To write a SHELX &quot;.ins&quot; file:
</p>
<p><code>(write-shelx-ins-file <em>imol</em> <em>file-name</em>)</code>
</p>
<p>where <code><em>imol</em></code> is the number of the molecule you wish to
export.
</p>
<p>This will be a rudimentary file if the coordinates were initially from a
&quot;PDB&quot; file, but will contain substantial SHELX commands if the
coordinates were initially generated from a SHELX <tt>ins</tt> file.  
</p>


<hr>
<a name="Atom-Info"></a>
<div class="header">
<p>
Next: <a href="#Atom-Labeling" accesskey="n" rel="next">Atom Labeling</a>, Previous: <a href="#Reading-coordinates" accesskey="p" rel="previous">Reading coordinates</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Atom-Info-1"></a>
<h3 class="section">4.2 Atom Info</h3>
<a name="index-atom-info"></a>
<a name="index-residue-info"></a>
<p>Information about about a particular atom is displayed in the text
console when you click using middle-mouse.  Information for all the
atoms in a residue is available using <code>Info -&gt; Residue
Info&hellip;</code>.
</p>
<a name="index-edit-B_002dfactors"></a>
<a name="index-edit-occupancy"></a>
<p>The temperature factors
and occupancy of the atoms in a residue can be set by using
<code>Edit -&gt; Residue Info&hellip;</code>.
</p>
<hr>
<a name="Atom-Labeling"></a>
<div class="header">
<p>
Next: <a href="#Atom-Colouring" accesskey="n" rel="next">Atom Colouring</a>, Previous: <a href="#Atom-Info" accesskey="p" rel="previous">Atom Info</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Atom-Labeling-1"></a>
<h3 class="section">4.3 Atom Labeling</h3>
<a name="index-atom-labeling"></a>
<p>Use Shift + left-mouse to label atom.  Do the same to toggle off the
label.  The font size is changeable using <code>Edit -&gt;
  Font Size&hellip;</code>.  The newly centred atom is labelled by default.
To turn this off use:
</p>
<p><code>(set-label-on-recentre-flag 0)</code>
</p>
<a name="index-atom-label_002c-brief"></a>
<p>Some people prefer to have atom labels that
are shorter, without the slashes and residue name:
</p>
<p><code>(set-brief-atom-labels 1)</code>
</p>
<p>To change the atom label colour, use:
</p>
<p><code>(set-font-colour 0.9 0.9 0.9)</code>
</p>

<hr>
<a name="Atom-Colouring"></a>
<div class="header">
<p>
Next: <a href="#Bond-Parameters" accesskey="n" rel="next">Bond Parameters</a>, Previous: <a href="#Atom-Labeling" accesskey="p" rel="previous">Atom Labeling</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Atom-Colouring-1"></a>
<h3 class="section">4.4 Atom Colouring</h3>
<p>The atom colouring 
<a name="index-colouring_002c-atoms"></a>
<a name="index-atom-colouring-1"></a>
system in coot is unsophisticated. Typically, atoms are coloured by
element: carbons are yellow, oxygens red, nitrogens blue, hydrogens
white and everything else green (see Section <a href="#Display-Manager">Display Manager</a>
for colour by chain).  However, it is useful to be able to distinguish
different molecules by colour, so by default coot rotates the colour
map of the atoms (<em>i.e.</em> changes the H value in the
HSV <a name="DOCF39" href="#FOOT39"><sup>39</sup></a>  colour system).  The
amount of the rotation depends on the molecule number and a
user-settable parameter:
</p><ul>
<li> <code>(set-colour-map-rotation-on-read-pdb 30)</code>.
</li></ul>

<p>The default value is 31<em>^\circ</em>.
</p>
<p>Also one is able to select only the Carbon atoms to change colour in
this manner: <code>(set-colour-map-rotation-on-read-pdb-c-only-flag
  1)</code>.
</p>
<p>The colour map rotation can be set individually for each molecule by
using the GUI: <code>Edit -&gt; Bond Colours<small class="enddots">...</small></code>.
</p>
<hr>
<a name="Bond-Parameters"></a>
<div class="header">
<p>
Next: <a href="#Download-coordinates" accesskey="n" rel="next">Download coordinates</a>, Previous: <a href="#Atom-Colouring" accesskey="p" rel="previous">Atom Colouring</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Bond-Parameters-1"></a>
<h3 class="section">4.5 Bond Parameters</h3>
<p>The various bond parameters can be set using the GUI dialog
<code>Draw -&gt; Bond Parameters</code> or <em>via</em> scripting
functions.
</p>
<p>The represention style of the molecule that has the active residue (if any) 
can be changed using the scroll wheel with Ctrl and Shift.
</p>
<a name="Bond-Thickness"></a>
<h4 class="subsection">4.5.1 Bond Thickness</h4>
<a name="index-bond-thickness"></a>
<a name="index-width_002c-bonds"></a>
<p>The thickness (width) of
bonds of individual molecules can be changed.  This can be done via the
<code>Bond Parameters</code> dialog or the scripting interface:
</p>
<p><code>(set-bond-thickness <em>thickness imol</em>)</code>
</p>
<p>where <code>imol</code> is the molecule number. 
</p>
<p>The default thickness is 3 pixels. The bond thickness also applies to
the symmetry atoms of the molecule. The default bond thickness for new
molecules can be set using:
</p>
<p><code>(set-default-bond-thickness <em>thick</em>)</code>
</p>
<p>where <code><em>thick</em></code> is an integer.
</p>
<p>There is no means to change the
bond thickness of a residue selection within a molecule.
</p>
<a name="Display-Hydrogens"></a>
<h4 class="subsection">4.5.2 Display Hydrogens</h4>
<a name="index-hydrogens"></a>
<p>Initially, hydrogens are displayed.  They can be
undisplayed using 
</p>
<p><code>(set-draw-hydrogens <em>mol-no</em> 0)</code> <a name="DOCF40" href="#FOOT40"><sup>40</sup></a>
</p>
<p>where <code><em>mol-no</em></code> is the molecule number.
</p>
<p>There is a GUI to control this too, under &ldquo;Edit -&gt; Bond Parameters&rdquo;.
</p>
<a name="NCS-Ghosts-Coordinates"></a>
<h4 class="subsection">4.5.3 NCS Ghosts Coordinates</h4>
<a name="index-NCS"></a>
<p>It is occasionally useful when analysing non-crystallographically
related molecules to have &ldquo;images&rdquo; of the other related molecules
appear matched onto the current coordinates.  It is important to
understand that these ghosts are for displaying differences of
NCS-related molecules by structure superposition, not displaying
neighbouring NCS related molecules.  As you read in coordinates in
Coot, they are checked for NCS relationships and clicking on &ldquo;Edit -&gt;
Bond Parameters -&gt; Show NCS Ghosts&rdquo; -&gt; &ldquo;Yes&rdquo; -&gt; &ldquo;Apply&rdquo; will
create &ldquo;ghost&rdquo; copies of them over the reference chain <a name="DOCF41" href="#FOOT41"><sup>41</sup></a>.
</p>
<p>Sometimes SSM does not provide a good (or even useful) matrix.  In that
case, we can specify the residue range ourselves and let the LSQ
algorithm provide the matrix.  A gui dialog for this operation can be
found under <code>Extensions -&gt; NCS -&gt; NCS Ghosts by Residue
Range&hellip;</code>.
</p>
<p>The scripting function is used like this:
</p>
<p><code>(manual-ncs-ghosts <em>imol resno-start resno-end ncs-chain-ids</em>)</code>
</p>
<p>Typical usage: <tt>(manual-ncs-ghosts 0 1 10 (list &quot;A&quot; &quot;B&quot; &quot;C&quot;))</tt>
</p>
<p>note that in <code><em>ncs-chain-ids</em></code>, the NCS master/reference
chain-id goes first.
</p>
<a name="NCS-Maps"></a>
<h4 class="subsection">4.5.4 NCS Maps</h4>
<p>Coot can use the relative transformations of the NCS-related molecules
in a coordinates molecule to transform maps. Use <code>Calculate</code>
-&gt; <code>NCS Maps&hellip;</code> to do this (note the NCS maps
only make sense in the region of the reference chain (see above).
</p>
<p>Note also that the internal representation of the map is not
transformed.  If you try to export a NCS overlay map you will get an
untransformed map.  A transformed map only makes sense around a given
point (and when using transformed maps in Coot, this reference point is
changed on the fly, thus allowing map transformations on the fly).
[This applies to NCS overlap maps, NCS averaged maps are transformed].
</p>
<a name="index-NCS-averaging"></a>
<p>This will also create an NCS averaged
map <a name="DOCF42" href="#FOOT42"><sup>42</sup></a>.
</p>
<a name="Using-Strict-NCS"></a>
<h4 class="subsection">4.5.5 Using Strict NCS</h4>
<p>Coot can use a set of strict NCS matrices to specify NCS which means
that NCS-related molecules can appear like convention symmetry-related
molecules.
</p>
<p><code>(add-strict-ncs-matrix <em>imol ncs-chain-id ncs-target-chain-id m11 m12 m13 m21 m22 m23 m31 m32 m33 t1 t2 t3</em>)</code>
</p>
<p>where <code><em>ncs-chain-id</em></code> might be &quot;B&quot;, &quot;C&quot; &quot;D&quot; (etc.) and
<code><em>ncs-target-chain-id</em></code> is &quot;A&quot;, i.e. the B, C, D molecules are
NCS copies of the A chain.
</p>
<p>for icosahedral symmetry the translation components <code><em>t1</em></code>, 
<code><em>t2</em></code>, <code><em>t3</em></code> will be 0.
</p>
<p>You need to turn on symmetry for molecule <code><em>imol</em></code> and set the
displayed symmetry object type to &quot;Display Near Chains&quot;.
</p>

<hr>
<a name="Download-coordinates"></a>
<div class="header">
<p>
Next: <a href="#Get-Coordinates-and-Map-from-EDS" accesskey="n" rel="next">Get Coordinates and Map from EDS</a>, Previous: <a href="#Bond-Parameters" accesskey="p" rel="previous">Bond Parameters</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Download-coordinates-1"></a>
<h3 class="section">4.6 Download coordinates</h3>
<p>Coot provides the possibility to download coordinates from an
<a name="index-OCA"></a>
OCA <a name="DOCF43" href="#FOOT43"><sup>43</sup></a>.
<a name="index-goose"></a>
(<em>e.g.</em> EBI) server <a name="DOCF44" href="#FOOT44"><sup>44</sup></a> (<code>File
  -&gt; Get PDB Using Code&hellip;</code>). A pop-up entry box is
displayed into which you can type a PDB accession code.  Coot will
then connect to the web server and transfer the file.  Coot blocks as
it does this (which is not ideal) but on a semi-decent internet
connection, it&rsquo;s not too bad.  The downloaded coordinates are saved
into a directory called <samp>coot-download</samp>.
</p>
<p>It is also possible to download mmCIF data and generate a map.  This
currently requires a properly formatted database structure factors
mmCIF file <a name="DOCF45" href="#FOOT45"><sup>45</sup></a>.
</p>
<hr>
<a name="Get-Coordinates-and-Map-from-EDS"></a>
<div class="header">
<p>
Next: <a href="#Save-coordinates" accesskey="n" rel="next">Save coordinates</a>, Previous: <a href="#Download-coordinates" accesskey="p" rel="previous">Download coordinates</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Get-Coordinates-and-Map-from-EDS-1"></a>
<h3 class="section">4.7 Get Coordinates and Map from EDS</h3>
<a name="index-Electron-Density-Server"></a>

<p>Using this function we have the ability to download coordinates and view
the map from structures in the Electron Density Server (EDS) at Uppsala
University.  This is a much more robust and faster way to see maps from
deposited structures.  This function can be found under the File menu
item.
</p>
<p>This feature was added with the assistance of Gerard Kleywegt.  If you
use the EDS, please cite GJ Kleywegt, MR Harris, JY Zou, TC Taylor, A
W&auml;hlby &amp; TA Jones (2004), <i>&quot;The Uppsala Electron-Density Server&quot;</i>, <i>Acta
Cryst.</i> <b>D</b>60, 2240-2249.
</p>
<hr>
<a name="Save-coordinates"></a>
<div class="header">
<p>
Next: <a href="#Setting-the-Space-Group" accesskey="n" rel="next">Setting the Space Group</a>, Previous: <a href="#Get-Coordinates-and-Map-from-EDS" accesskey="p" rel="previous">Get Coordinates and Map from EDS</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Save-Coordinates-1"></a>
<h3 class="section">4.8 Save Coordinates</h3>
<p>On selecting from the menus <code>File -&gt; Save
  Coordinates&hellip;</code> you are first presented with a list of molecules
which have coordinates.  As well as the molecule number, there is the
molecule name - very frequently the name of the file that was read in
to generate the coordinates in coot initially.  However, this is only
a <em>molecule</em> name and should not be confused with the filename to
which the coordinates are saved.  The coordinates <em>filename</em> can
be selected using the <code>Select Filename&hellip;</code> button.
</p>
<p>If your filename ends in <code>.cif</code>, <code>.mmcif</code> or
<code>.mmCIF</code> then an mmCIF file will be written (not a &ldquo;PDB&rdquo;
file).
</p>
<hr>
<a name="Setting-the-Space-Group"></a>
<div class="header">
<p>
Next: <a href="#Anisotropic-Atoms" accesskey="n" rel="next">Anisotropic Atoms</a>, Previous: <a href="#Save-coordinates" accesskey="p" rel="previous">Save coordinates</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Setting-the-Space-Group-1"></a>
<h3 class="section">4.9 Setting the Space Group</h3>
<a name="index-setting-space-group"></a>

<p>If for some reason, the pdb file that you read does not have a space
group, or has the wrong space group, then you can set it using the
following function:
</p>
<p><code>(set-space-group <em>imol</em> <em>symbol</em>)</code>
</p>
<p>e.g.:
</p>
<p><code>(set-space-group 0 &quot;P 41 21 2&quot;)</code>
</p>
<hr>
<a name="Anisotropic-Atoms"></a>
<div class="header">
<p>
Next: <a href="#Symmetry" accesskey="n" rel="next">Symmetry</a>, Previous: <a href="#Setting-the-Space-Group" accesskey="p" rel="previous">Setting the Space Group</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Anisotropic-Atoms-1"></a>
<h3 class="section">4.10 Anisotropic Atoms</h3>
<a name="index-anisotropic-atoms"></a>
<p>By default anisotropic atom information is
not represented <a name="DOCF46" href="#FOOT46"><sup>46</sup></a>.  To turn them on,
use <code>Draw -&gt; Anisotropic Atoms -&gt; Show
  Anisotropic Atoms?  -&gt; Yes</code>, or the command:
<code>(set-show-aniso 1)</code>.
</p>
<p>You cannot currently display thermal ellipsoids <a name="DOCF47" href="#FOOT47"><sup>47</sup></a> for
isotropic atoms.
</p>
<hr>
<a name="Symmetry"></a>
<div class="header">
<p>
Next: <a href="#Sequence-View" accesskey="n" rel="next">Sequence View</a>, Previous: <a href="#Anisotropic-Atoms" accesskey="p" rel="previous">Anisotropic Atoms</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Symmetry-1"></a>
<h3 class="section">4.11 Symmetry</h3>
<a name="index-symmetry"></a>
<p>Coordinates symmetry is &ldquo;dynamic&rdquo;.  Symmetry atoms
can be labeled <a name="DOCF48" href="#FOOT48"><sup>48</sup></a>.  
</p>
<p>Every time you recentre, the symmetry coordinates are updated.  The
information shown contains the atom information and the symmetry
operation number and translations needed to generate the atom in that
position.  To show the symmetry operator as a string (rather than a
(1-based) index into the list of symmetry operators (as is the default))
Use <code>Draw -&gt; Show Symmetry&gt; -&gt; Expanded Symmetry Atom Labels</code>.
Coot generates symmetry related atoms by moving the current set close to
the origin by a translation, performing the symmetry expansion around
the origin and moving the the symmetry coordinates back by applying the
inverse of the origin translation.  The origin translation is also
displayed in curly braces, e.g. &ldquo;{1 -1 0}&rdquo;.
</p>
<p>By default symmetry atoms are not displayed.  
</p>
<p>If you want coot to display symmetry coordinates without having to use
the gui, add to your <samp>~/.coot</samp> the following:
</p>
<p><code>(set-show-symmetry-master 1)</code>
</p>

<p>The symmetry can be represented as C<em>\alpha</em>s.
<a name="index-C_005calpha-symmetry"></a>
This along with representation of the molecule as
C<em>\alpha</em>s (Section <a href="#Display-Manager">Display Manager</a>) allow the production of
a packing diagram.
<a name="index-packing-diagram"></a>
</p>
<a name="Missing-symmetry"></a>
<h4 class="subsection">4.11.1 Missing symmetry</h4>
<a name="index-missing-symmetry"></a>

<p>Sometimes (rarely) coot misses symmetry-related molecules that should be
displayed.  In that case you need to expand the shift search (the default is 1):
</p>
<p><code>(set-symmetry-shift-search-size 2)</code>
</p>
<p>This is a hack, until the symmetry search algorithm is improved.
</p>       

<hr>
<a name="Sequence-View"></a>
<div class="header">
<p>
Next: <a href="#Print-Sequence" accesskey="n" rel="next">Print Sequence</a>, Previous: <a href="#Symmetry" accesskey="p" rel="previous">Symmetry</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Sequence-View-1"></a>
<h3 class="section">4.12 Sequence View</h3>
<a name="index-sequence-view"></a>
<p>The protein is represented by one letter codes
and coloured according to secondary structure.  These one letter codes
are active - if you click on them, they will change the centre of the
graphics window - in much the same way as clicking on a residue in the
Ramachandran plot.
</p>
<hr>
<a name="Print-Sequence"></a>
<div class="header">
<p>
Next: <a href="#Environment-Distances" accesskey="n" rel="next">Environment Distances</a>, Previous: <a href="#Sequence-View" accesskey="p" rel="previous">Sequence View</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Print-Sequence-1"></a>
<h3 class="section">4.13 Print Sequence</h3>
<a name="index-Print-Sequence"></a>
<p>The single letter code (of the <code><em>imol</em></code>th molecule) is written
out to the console in FASTA format.  Use can use this to cut and paste
into other applications:
</p>
<p><code>(print-sequence <em>imol</em>)</code>
</p>
<hr>
<a name="Environment-Distances"></a>
<div class="header">
<p>
Next: <a href="#Distances-and-Angles" accesskey="n" rel="next">Distances and Angles</a>, Previous: <a href="#Print-Sequence" accesskey="p" rel="previous">Print Sequence</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Environment-Distances-1"></a>
<h3 class="section">4.14 Environment Distances</h3>
<p>Environment distances are turned on using <code>Info -&gt; Environment
Distances&hellip;</code>.  Contacts to other residues are shown and to
symmetry-related atoms if symmetry is being displayed.  The contacts
are coloured by atom type <a name="DOCF49" href="#FOOT49"><sup>49</sup></a>.
</p>
<hr>
<a name="Distances-and-Angles"></a>
<div class="header">
<p>
Next: <a href="#Zero-Occupancy-Marker" accesskey="n" rel="next">Zero Occupancy Marker</a>, Previous: <a href="#Environment-Distances" accesskey="p" rel="previous">Environment Distances</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Distances-and-Angles-1"></a>
<h3 class="section">4.15 Distances and Angles</h3>
<p>The distance between atoms can be found using <code>Info
  -&gt; Distance</code> <a name="DOCF50" href="#FOOT50"><sup>50</sup></a>.  The result is displayed graphically, and written to the
console.
</p>
<hr>
<a name="Zero-Occupancy-Marker"></a>
<div class="header">
<p>
Next: <a href="#Atomic-Dots" accesskey="n" rel="next">Atomic Dots</a>, Previous: <a href="#Distances-and-Angles" accesskey="p" rel="previous">Distances and Angles</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Zero-Occupancy-Marker-1"></a>
<h3 class="section">4.16 Zero Occupancy Marker</h3>
<a name="index-zero-occupancy"></a>
<p>Atoms of zero occupancy are marked with a grey
spot. To turn off these markers, use:
</p>
<p><code>(set-draw-zero-occ-markers 0)</code>
</p>
<p>Use an argument of 1 to turn them on.
</p>
<hr>
<a name="Atomic-Dots"></a>
<div class="header">
<p>
Next: <a href="#Ball-and-Stick-Representation" accesskey="n" rel="next">Ball and Stick Representation</a>, Previous: <a href="#Zero-Occupancy-Marker" accesskey="p" rel="previous">Zero Occupancy Marker</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Atomic-Dots-1"></a>
<h3 class="section">4.17 Atomic Dots</h3>
<a name="index-atomic-dots"></a>

<p>You can draw dots round arbitrary atom selections
</p>
<p><code>(dots <em>imol atom-selection dot-density radius</em>)</code>
The function returns a handle.
</p>
<p><em>e.g.</em> put a sphere of dots around all atoms of the 0th molecule
(it might be a set of heavy atom coordinates) at the default dot density
and radius:
</p>
<p><code>(dots 0 &quot;/1&quot;  &quot;heavy-atom-sites&quot; 1 1)</code>
</p>
<p>You can&rsquo;t change the colour of the dots.  
</p>
<p>There is no internal mechanism to change the radius according to atom
type.  With some cleverness you might be able to call this function
several times and change the radius according to the atom selection.
</p>
<p>There is a function to clear up the dots for a particular molecule
<code><em>imol</em></code> and dots set identifier <code><em>dots-handle</em></code>
</p>
<p><code>(clear-dots <em>imol dots-handle</em>)</code>
</p>
<p>There is a function to return how many dots sets there are for a 
particular molecule <code><em>imol</em></code>:
</p>
<p><code>(n-dots-set <em>imol</em>)</code>
</p>
<hr>
<a name="Ball-and-Stick-Representation"></a>
<div class="header">
<p>
Next: <a href="#Mean-and-Median-Temperature-Factors" accesskey="n" rel="next">Mean and Median Temperature Factors</a>, Previous: <a href="#Atomic-Dots" accesskey="p" rel="previous">Atomic Dots</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Ball-and-Stick-Representation-1"></a>
<h3 class="section">4.18 Ball and Stick Representation</h3>
<a name="index-ball-and-stick"></a>

<p>Fragments of the molecule can be rendered as a &ldquo;ball and stick&rdquo; molecule:
</p>
<p><code>(make-ball-and-stick <em>imol atom-selection bond-thickness sphere-size draw-spheres-flag</em>)</code>
</p>
<p>e.g. 
<code>(make-ball-and-stick 0 &quot;/1/A/10-20&quot; 0.3 0.4 1)</code>
</p>
<p>The ball-and-stick representation can be cleared using:
</p>
<p><code>(clear-ball-and-stick <em>imol</em>)</code>
</p>


<hr>
<a name="Mean-and-Median-Temperature-Factors"></a>
<div class="header">
<p>
Next: <a href="#Secondary-Structure-Matching-_0028SSM_0029" accesskey="n" rel="next">Secondary Structure Matching (SSM)</a>, Previous: <a href="#Ball-and-Stick-Representation" accesskey="p" rel="previous">Ball and Stick Representation</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Mean_002c-Median-Temperature-Factors"></a>
<h3 class="section">4.19 Mean, Median Temperature Factors</h3>
<p>Coot can be used to calculate the 
<a name="index-mean-B_002dfactor"></a>
mean (average)
and
<a name="index-median-B_002dfactor"></a>
median temperatures factors:
</p>
<p><code>(average-temperature-factor <em>imol</em>)</code>
</p>
<p><code>(median-temperature-factor <em>imol</em>)</code>
</p>
<p><em>-1</em> is returned if there was a problem <a name="DOCF51" href="#FOOT51"><sup>51</sup></a>.
</p>
<hr>
<a name="Secondary-Structure-Matching-_0028SSM_0029"></a>
<div class="header">
<p>
Next: <a href="#Least_002dSquares-Fitting" accesskey="n" rel="next">Least-Squares Fitting</a>, Previous: <a href="#Mean-and-Median-Temperature-Factors" accesskey="p" rel="previous">Mean and Median Temperature Factors</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Secondary-Structure-Matching-_0028SSM_0029-1"></a>
<h3 class="section">4.20 Secondary Structure Matching (SSM)</h3>
<a name="index-Secondary-Structure-Matching-_0028SSM_0029"></a>
<a name="index-superposition"></a>

<p>The excellent SSM alogrithm<a name="DOCF52" href="#FOOT52"><sup>52</sup></a> of Eugene Krissinel is available in Coot.  The GUI interface
is straight-forward and can be found under <code>Calculate -&gt; SSM
Superpose</code>.  You can specify the specific chains that you wish to match
using the &quot;Use Specific Chain&quot; check-button.
</p>
<p>There is a scripting level function which gives even finer control:
</p>
<p><code>(superpose-with-atom-selection <em>imol1 imol2
mmdb-atom-selection-string-1 mmdb-atom-selection-string-2
move-copy-flag</em> )</code>
</p>
<p>the <code><em>move-copy-flag</em></code> should be 1 if you want to apply the
transformation to a copy of <code><em>imol2</em></code> (rather than
<code><em>imol2</em></code> itself). Otherwise, <code><em>move-copy-flag</em></code>
should be 0.
</p>
<p>mmdb atom selection strings (Coordinate-IDs) are explained in detail in
the mmdb manual.
</p>
<p>Briefly, the string should be formed in this manner:
</p>
<p><code>/mdl/chn/seq(res).ic/atm[elm]:aloc</code>
</p>
<p><em>e.g.</em> <code>&quot;/1/A/12-130/CA&quot;</code>
</p>
<p>&lt;p&gt;&lt;a href=&quot;http://www.ebi.ac.uk/~keb/cldoc/object/cl_obj_surf.html#CoordinateID&quot;&gt;The mmdb manual CoordinateID description&lt;/a&gt;.&lt;/p&gt;
</p>

<hr>
<a name="Least_002dSquares-Fitting"></a>
<div class="header">
<p>
Next: <a href="#Ligand-Overlaying" accesskey="n" rel="next">Ligand Overlaying</a>, Previous: <a href="#Secondary-Structure-Matching-_0028SSM_0029" accesskey="p" rel="previous">Secondary Structure Matching (SSM)</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Least_002dSquares-Fitting-1"></a>
<h3 class="section">4.21 Least-Squares Fitting</h3>
<a name="index-Least-Squares-Fitting"></a>

<p>There is a simple GUI for this <code>Calculate -&gt; LSQ Superpose&hellip;</code>
</p>
<p>The scripting interface to LSQ fitting is as follows:
</p>
<p><code>(simple-lsq-match <em>ref-start-resno ref-end-resno ref-chain-id imol-ref
           mov-start-resno mov-end-resno mov-chain-id imol-mov
           match-type</em>)</code>
</p>
<p>where:
</p><ul>
<li> <code><em>ref-start-resno</em></code> is the starting residue number
  of the reference molecule
</li><li> <code><em>ref-end-resno</em></code> is the last residue number
  of the reference molecule
</li><li> <code><em>mov-start-resno</em></code> is the starting residue number
  of the moving molecule
</li><li> <code><em>mov-end-resno</em></code> is the last residue number
  of the moving molecule
</li><li> <code><em>match-type</em></code> is one of <code>'CA</code>,
  <code>'main</code>, or <code>'all</code>.
</li></ul>

<p><em>e.g.</em>: 
<code>(simple-lsq-match 940 950 &quot;A&quot; 0 940 950 &quot;A&quot; 1 'main)</code>
</p>
<p>More sophisticated (match molecule number 1 chain &ldquo;B&rdquo; on to molecule
number 0 chain &ldquo;A&rdquo;):
</p>
<p><code>(define match1 (list 840 850 &quot;A&quot; 440 450 &quot;B&quot; 'all))</code>
</p>
<p><code>(define match2 (list 940 950 &quot;A&quot; 540 550 &quot;B&quot; 'main))</code>
</p>
<p><code>(clear-lsq-matches)</code>
</p>
<p><code>(set-match-element match1)</code>
</p>
<p><code>(set-match-element match2)</code>
</p>
<p><code>(apply-lsq-matches 0 1)</code> ; match molecule number 1 onto molecule number 0.
</p>

<hr>
<a name="Ligand-Overlaying"></a>
<div class="header">
<p>
Next: <a href="#Writing-PDB-files" accesskey="n" rel="next">Writing PDB files</a>, Previous: <a href="#Least_002dSquares-Fitting" accesskey="p" rel="previous">Least-Squares Fitting</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Ligand-Overlaying-1"></a>
<h3 class="section">4.22 Ligand Overlaying</h3>
<a name="index-ligand_002c-overlay"></a>
<a name="index-overlaying-ligands"></a>

<p>The scripting function 
</p>
<p><code>(overlap-ligands <em>imol-ligand imol-ref chain-id-ref resno-ref</em>)</code>
</p>
<p>returns a rotation+translation operator which can be applied to other
molecules (and maps).  Here, <code><em>imol-ligand</em></code> is the molecule
number of the ligand (which is presumed to be a a molecule on its own -
Coot simply takes the first residue that it finds). <code><em>imol-ref
chain-id-ref resno-ref</em></code> collectively describe the target position for
the moving <code><em>imol-ligand</em></code> molecule.
</p>
<p>The convenience function 
</p>
<p><code>(overlay-my-ligands <em>imol-mov chain-id-mov resno-mov imol-ref chain-id-ref resno-ref</em>)</code> 
</p>
<p>wraps <code>overlap-ligands</code>.  
</p>
<p>The GUI for the function can be found under 
</p>
<p><code>Extensions -&gt; Modelling -&gt; Supperpose Ligands&hellip;</code>
</p>
<hr>
<a name="Writing-PDB-files"></a>
<div class="header">
<p>
Previous: <a href="#Ligand-Overlaying" accesskey="p" rel="previous">Ligand Overlaying</a>, Up: <a href="#Coordinate_002dRelated-Features" accesskey="u" rel="up">Coordinate-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Writing-PDB-files-1"></a>
<h3 class="section">4.23 Writing PDB files</h3>
<a name="index-writing-PDBs"></a>

<p>As well as the GUI option <code>File -&gt; Save Coordinates&hellip;</code> there
is a scripting options available:
</p>
<p><code>(write-pdb-file <em>imol pdb-file-name</em>)</code>
</p>
<p>which writes the <code><em>imol</em></code>th coordinates molecule to
<code><em>filename</em></code>.
</p>
<p>To write a specific residue range: 
</p>
<p><code>(write-residue-range-to-pdb-file <em>imol chain-id start-resno
endresno pdb-file-name</em>)</code>
</p>

<hr>
<a name="Modelling-and-Building"></a>
<div class="header">
<p>
Next: <a href="#Map_002dRelated-Features" accesskey="n" rel="next">Map-Related Features</a>, Previous: <a href="#Coordinate_002dRelated-Features" accesskey="p" rel="previous">Coordinate-Related Features</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Modelling-and-Building-1"></a>
<h2 class="chapter">5 Modelling and Building</h2>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Regularization-and-Real-Space-Refinement" accesskey="1">Regularization and Real Space Refinement </a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Changing-the-Map-for-Building_002fRefinement" accesskey="2">Changing the Map for Building/Refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotate_002fTranslate-Zone" accesskey="3">Rotate/Translate Zone</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rigid-Body-Refinement" accesskey="4">Rigid Body Refinement  </a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Simplex-Refinement" accesskey="5">Simplex Refinement </a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Post_002dmanipulation_002dhook" accesskey="6">Post-manipulation-hook</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Baton-Building" accesskey="7">Baton Building</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Reversing-Direction-of-Fragment" accesskey="8">Reversing Direction of Fragment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#C_002dalpha-_002d_003e-Mainchain" accesskey="9">C-alpha -&gt; Mainchain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Backbone-Torsion-Angles">Backbone Torsion Angles </a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Docking-Sidechains">Docking Sidechains</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotamers">Rotamers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Editing-Chi-Angles">Editing Chi Angles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Torsion-General">Torsion General</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pep_002dflip">Pep-flip</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Adding-Alternative-Conformations">Adding Alternative Conformations</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mutation">Mutation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Importing-Ligands_002fMonomers">Importing Ligands/Monomers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Ligand-from-SMILES-strings">Ligand from SMILES strings</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Find-Ligands">Find Ligands</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Flip-Ligand">Flip Ligand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Find-Waters">Find Waters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Add-Terminal-Residue">Add Terminal Residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Add-OXT-Atom-to-Residue">Add OXT Atom to Residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Add-Atom-at-Pointer">Add Atom at Pointer</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Place-Helix">Place Helix</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Building-Ideal-DNA-and-RNA">Building Ideal DNA and RNA</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Merge-Molecules">Merge Molecules</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Temperature-Factor-for-New-Atoms">Temperature Factor for New Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Applying-NCS-Edits">Applying NCS Edits</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Running-Refmac">Running Refmac</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Running-SHELXL">Running SHELXL</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Clear-Pending-Picks">Clear Pending Picks</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Delete">Delete</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Sequence-Assignment">Sequence Assignment </a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Building-Links-and-Loops">Building Links and Loops</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Fill-Partial-Residues">Fill Partial Residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Changing-Chain-IDs">Changing Chain IDs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Setting-Occupancies">Setting Occupancies</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Fix-Nomenclature-Errors">Fix Nomenclature Errors</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotamer-Fix-Whole-Protein">Rotamer Fix Whole Protein</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Refine-All-Waters">Refine All Waters </a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Moving-Molecules_002fLigands">Moving Molecules/Ligands</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Modifying-the-Labels-on-the-Model_002fFit_002fRefine-dialog">Modifying the Labels on the Model/Fit/Refine dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<p>The functions described in this chapter manipulate, extend or build
molecules and can be found under <code>Calculate -&gt;
  Model/Fit/Refine&hellip;</code>.  When activated, the dialog &quot;stays on top&quot;
  of the main graphics window <a name="DOCF53" href="#FOOT53"><sup>53</sup></a>.  Some people think that this is not always desirable, so
  this behaviour can be undone using:
</p>
<p><code>(set-model-fit-refine-dialog-stays-on-top 0)</code>
</p>

<hr>
<a name="Regularization-and-Real-Space-Refinement"></a>
<div class="header">
<p>
Next: <a href="#Changing-the-Map-for-Building_002fRefinement" accesskey="n" rel="next">Changing the Map for Building/Refinement</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Regularization-and-Real-Space-Refinement-1"></a>
<h3 class="section">5.1 Regularization and Real Space Refinement</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Dictionary" accesskey="1">Dictionary</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Sphere-Refinement" accesskey="2">Sphere Refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Refining-Specific-Residues" accesskey="3">Refining Specific Residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Refining-Carbohydrates" accesskey="4">Refining Carbohydrates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Planar-Peptide-Restraints" accesskey="5">Planar Peptide Restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#The-UNK-residue-type" accesskey="6">The UNK residue type</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Moving-Zero-Occupancy-Atoms" accesskey="7">Moving Zero Occupancy Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<p>Coot will read the geometry restraints for
refmac and use them in fragment (zone) idealization - this is called
&ldquo;Regularization&rdquo;.
<a name="index-regularization"></a>
  The geometrical restraints
are, by default, bonds, angles, planes
<a name="index-planes"></a>
 and non-bonded
contacts.  You can additionally use torsion restraints
<a name="index-torsion-restraints"></a>
by <code>Calculate -&gt; Model/Fit/Refine&hellip; -&gt; Refine/Regularize Control
  -&gt; Use Torsion Restraints</code>.  Truth to tell, this has not been
  successful in my hands (sadly).
</p>
<p>&ldquo;RS (Real Space) Refinement&rdquo;
<a name="index-refinement"></a>
 (after Diamond,
1971 <a name="DOCF54" href="#FOOT54"><sup>54</sup></a>) in Coot is the use of the map in addition to geometry terms to
improve the positions of the atoms.  Select &ldquo;Regularize&rdquo; from the
&ldquo;Model/Fit/Refine&rdquo; dialog and click on 2 atoms to define the zone
(you can of course click on the same atom twice if you only want to
regularize one residue).  Coot then regularizes the residue range.  At
the end Coot, displays the intermediate atoms in white and also
displays a dialog, in which you can accept or reject this
regularization.  In the console are displayed the <em>\chi^2</em>
values of the various geometrical restraints for the zone before 
and after the regularization.  Usually the <em>\chi^2</em> values are 
considerably decreased - structure idealization such as this should 
drive the <em>\chi^2</em> values toward zero.
</p>
<p>The use of &ldquo;Refinement&rdquo; is similar - with the addition of using a
map.  The map used to refine the structure is set by using the
&ldquo;Refine/Regularize Control&rdquo; dialog.  If you have read/created only
one map into Coot, then that map will be used (there is no need to set
it explicitly).
</p>

<p>Use, for example, 
<a name="index-set_002dmatrix"></a>
<a name="index-weight_002c-real-space-refinement"></a>
<a name="index-refinement-weight"></a>
<code>(set-matrix 20.0)</code>
</p>

<p>to change the weight of the map gradients to geometric gradients.  The
higher the number the more weight that is given to the map terms
<a name="DOCF55" href="#FOOT55"><sup>55</sup></a>.  The
default is 60.0.  This will be needed for maps generated from data not
on (or close to) the absolute scale or maps that have been scaled (for
example so that the sigma level has been scaled to 1.0).
</p>
<p>For both &ldquo;Regularize Zone&rdquo; and &ldquo;Refine Zone&rdquo; one is able to use a
single click to
<a name="index-single-click-refine"></a>
<a name="index-refine-single-click"></a>
refine a residue range.  Pressing <tt class="key">A</tt> on the keyboard while
selecting an atom in a residue will automatically create a residue
range with that residue in the middle.  By default the zone is
extended one residue either size of the central residue.  This can be
changed to 2 either side using <code>(set-refine-auto-range-step
  2)</code>.
</p>
<p>Intermediate (white) atoms can be moved around with the mouse (click
and drag with left-mouse, by default).  
Refinement will proceed from the
new atom positions when the mouse button is released.  It is possible
to create incorrect atom nomenclature and/or chiral volumes in this
manner - so some care must be taken.  Press the <tt class="key">A</tt> key as you
left-mouse click to move atoms more &ldquo;locally&rdquo; (rather than a linear
shear) and <tt class="key">Ctrl</tt> key as you left-mouse click to move just one atom.
</p>
<p>In more up to date versions, Coot will display colour patches (something
like a traffic light system) 
<a name="index-traffic-lights"></a>
representing the chi squared values of each
of types of geometric feature refined.  Typically &ldquo;5 greens&rdquo; is the
thing to aim for, the colour changes occurring at chi squared values 2, 5
and 8 (8 being the most red).
</p>
<p>To prevent the unintentional refinement of a large number of residues,
there is a &ldquo;heuristic fencepost&rdquo; of 20 residues.  A selection of
than 20 residues will not be regularized or refined.  The limit can be
changed using the scripting function: <em>e.g.</em>
<code>(set-refine-max-residues 30)</code>.
</p>
<hr>
<a name="Dictionary"></a>
<div class="header">
<p>
Next: <a href="#Sphere-Refinement" accesskey="n" rel="next">Sphere Refinement</a>, Up: <a href="#Regularization-and-Real-Space-Refinement" accesskey="u" rel="up">Regularization and Real Space Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Dictionary-1"></a>
<h4 class="subsection">5.1.1 Dictionary</h4>
<a name="index-cif-dictionary"></a>
<a name="index-mmCIF-dictionary"></a>

<p>The geometry description for residues, monomers and links used by Coot
are in the standard mmCIF format.  Because this format alows multiple
comp_ids (residue types) to be described within a cif loop, it is hard
to tell when a dictionary entry needs to be overwritten when reading a
new file.  Therefore Coot makes this extra constraint: that the
&ldquo;chem_comp&rdquo; loop should appear first in the comp list data item - if
this is the case, then Coot can overwrite an old restraint table for a
particular comp_id/residue-type when a new one is read.
</p>
<p>By default, 
the geometry dictionary entries for only the standard
residues are read in at the start  <a name="DOCF56" href="#FOOT56"><sup>56</sup></a>.  It may be that your particular ligand is not amongst
these.  To interactively add a dictionary entry use <code>File
  -&gt; Import CIF Dictionary</code>.
</p>
<p>There is a selector in the cif dictionary file chooser that allow you to
select the molecule to which molecule refers. Each of the indididual
molecules can be specifically selected. &ldquo;All&ldquo; means that the dictionary
refers to all loaded molecules and &ldquo;Auto&ldquo; means that the dictionary will
be applied to all model molecules if the comp_id/residue name is not on the
non-auto-load list and will choose the latest model molecule if the comp_id/residue
name is on the non-auto-load list. By default the non-auto-load list
consists of INH, LIG, DRG, XXX, and the series LG0-9.
</p>

<p>Alternatively, you can use
the function:
</p>
<p><code>(read-cif-dictionary <em>filename</em>)</code>
</p>
<p>and add this to your <code>.coot</code> file (this may be the preferred
method if you want to read the file on more than one occasion).  
</p>
<p>Note: the dictionary also provides the description of the ligand&rsquo;s
torsions.
</p>
<hr>
<a name="Sphere-Refinement"></a>
<div class="header">
<p>
Next: <a href="#Refining-Specific-Residues" accesskey="n" rel="next">Refining Specific Residues</a>, Previous: <a href="#Dictionary" accesskey="p" rel="previous">Dictionary</a>, Up: <a href="#Regularization-and-Real-Space-Refinement" accesskey="u" rel="up">Regularization and Real Space Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Sphere-Refinement-1"></a>
<h4 class="subsection">5.1.2 Sphere Refinement</h4>
<a name="index-sphere-refinement"></a>

<p>Sphere refinement selects residues within a certain distance of the
residue at the centre of the screen and includes them for real space
refinement.  In this way, one can select residues that are not in a
linear range.  This technique is useful for refining disulfide bonds and
glycosidic linkages.
</p>
<p>To enable sphere refinement, Right-mouse click in the right hand side of
the horizontal toolbutton menu, Manage buttons -&gt; [Tick] Sphere Refine
-&gt; Apply.  You will need a python-enabled Coot to do this.
</p>
<p>The following adds a key binding (Shift-R) that refines resides that are
within 3.5&Aring; of the residue at the centre of the screen:
</p>
<pre class="verbatim">        
(define *sphere-refine-radius* 3.5)

(add-key-binding &quot;Refine residues in a sphere&quot; &quot;R&quot;
    (lambda ()
     (using-active-atom
      
      (let* ((rc-spec (list aa-chain-id aa-res-no aa-ins-code))
	     (ls (residues-near-residue aa-imol rc-spec *sphere-refine-radius*)))
	(refine-residues aa-imol (cons rc-spec ls))))))


</pre>

<hr>
<a name="Refining-Specific-Residues"></a>
<div class="header">
<p>
Next: <a href="#Refining-Carbohydrates" accesskey="n" rel="next">Refining Carbohydrates</a>, Previous: <a href="#Sphere-Refinement" accesskey="p" rel="previous">Sphere Refinement</a>, Up: <a href="#Regularization-and-Real-Space-Refinement" accesskey="u" rel="up">Regularization and Real Space Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Refining-Specific-Residues-1"></a>
<h4 class="subsection">5.1.3 Refining Specific Residues</h4>
<a name="index-refining-residues"></a>

<p>You can specify the residues that you want to refine without using a
linear or sphere selection usine <code>refine-residues</code>.   For example:
</p>
<p><code>(refine-residues 0 '((&quot;L&quot; 501 &quot;&quot;) (&quot;L&quot; 503 &quot;&quot;)))</code>
</p>
<p>will refine residues A501 and A503 (and residue A502 (if it exists)
will be an anchoring residue - used in optimizing the link geometry of
the atoms in A501 and A503).
</p>
<hr>
<a name="Refining-Carbohydrates"></a>
<div class="header">
<p>
Next: <a href="#Planar-Peptide-Restraints" accesskey="n" rel="next">Planar Peptide Restraints</a>, Previous: <a href="#Refining-Specific-Residues" accesskey="p" rel="previous">Refining Specific Residues</a>, Up: <a href="#Regularization-and-Real-Space-Refinement" accesskey="u" rel="up">Regularization and Real Space Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Refining-Carbohydrates-1"></a>
<h4 class="subsection">5.1.4 Refining Carbohydrates</h4>
<a name="index-carbohydrates"></a>

<p>Refining carbohydrates monomers should be as straightforward as refining
a protein residue.  Coot will look in the dictionary for the 3-letter
code for the particular residue type, if it does not find it, Coot will
try to search for dictionary files using &ldquo;-b-D&rdquo; or &ldquo;-a-L&rdquo; extensions.
</p>
<p>When refining a group of carbohydrates, the situation needs a bit more
explanation. For each residue pair with tandem residue numbers specified
in the refinement range selection, Coot checks if these residue types
are are furanose or pyranose in the dictionary, and if the are both one
or the other, then it tries to see if there are any of the 11 link types
(BETA1-4, BETA2-3, ALPHA1-2 and so on) specified in the dictionary.  It
does this by a distance check of the potentially bonding atoms.  If the
distance is less than 3.0&Aring;, then a glycosidic bond is made and used
in the refinement.
</p>
<p>Bonds between protein and carbohydrate and branched carbohydrates can be
refined using &ldquo;Sphere Refinement&rdquo;.
</p>
<p>Instead of using a sphere to make a residue selection, you can specify
the residues directly using <code>refine-residues</code>, for example:
</p>
<p><code>(refine-residues 0 '((&quot;L&quot; 501 &quot;&quot;) (&quot;L&quot; 503 &quot;&quot;)))</code>
</p>
<p>LINK and LINKR cards are not yet used to determine the geometry of the
restraints.
</p>


<hr>
<a name="Planar-Peptide-Restraints"></a>
<div class="header">
<p>
Next: <a href="#The-UNK-residue-type" accesskey="n" rel="next">The UNK residue type</a>, Previous: <a href="#Refining-Carbohydrates" accesskey="p" rel="previous">Refining Carbohydrates</a>, Up: <a href="#Regularization-and-Real-Space-Refinement" accesskey="u" rel="up">Regularization and Real Space Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Planar-Peptide-Restraints-1"></a>
<h4 class="subsection">5.1.5 Planar Peptide Restraints</h4>
<a name="index-planar-peptide-restraints"></a>
<a name="index-peptide-restraints_002c-planar"></a>

<p>By default, Coot uses a 5 atom (CA-1, C-1, O-1, N-2, CA-2) planar
peptide restraints.  These restraints should help in low resolution
fitting (the main-chains becomes less distorted), reduce accidental
cis-peptides and may help &ldquo;clean up&rdquo; Ramachandran plots.
</p>
<p><code>(add-planar-peptide-restraints)</code>
</p>
<p>And similarly they can be removed:
</p>
<p><code>(remove-planar-peptide-restraints)</code>
</p>
<p>There is also a GUI to add and remove these restraints in
<code>Extensions -&gt; Refine&hellip; -&gt; Peptide Restraints&hellip;</code> 
</p>



<hr>
<a name="The-UNK-residue-type"></a>
<div class="header">
<p>
Next: <a href="#Moving-Zero-Occupancy-Atoms" accesskey="n" rel="next">Moving Zero Occupancy Atoms</a>, Previous: <a href="#Planar-Peptide-Restraints" accesskey="p" rel="previous">Planar Peptide Restraints</a>, Up: <a href="#Regularization-and-Real-Space-Refinement" accesskey="u" rel="up">Regularization and Real Space Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="The-UNK-residue-type-1"></a>
<h4 class="subsection">5.1.6 The UNK residue type</h4>
<a name="index-UNK-residue"></a>

<p>The <code>UNK</code> residue type is a special residue type to Coot.  It has
been added for use with Buccaneer. Don&rsquo;t give you ligand (or anything
else) the 3-letter-code <code>UNK</code> or confusion will result
<a name="DOCF57" href="#FOOT57"><sup>57</sup></a>.
</p>
<hr>
<a name="Moving-Zero-Occupancy-Atoms"></a>
<div class="header">
<p>
Previous: <a href="#The-UNK-residue-type" accesskey="p" rel="previous">The UNK residue type</a>, Up: <a href="#Regularization-and-Real-Space-Refinement" accesskey="u" rel="up">Regularization and Real Space Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Moving-Zero-Occupancy-Atoms-1"></a>
<h4 class="subsection">5.1.7 Moving Zero Occupancy Atoms</h4>
<a name="index-Moving-Zero-Occupancy-Atoms"></a>

<p>By default, atoms with zero occupancy are moved when refining and
regularizing.  This can sometimes be inconvenient.  To turn of the
movement of atoms with zero occupancy when refining and regularizing:
</p>
<p><code>(set-refinement-move-atoms-with-zero-occupancy 0)</code>
</p>

<hr>
<a name="Changing-the-Map-for-Building_002fRefinement"></a>
<div class="header">
<p>
Next: <a href="#Rotate_002fTranslate-Zone" accesskey="n" rel="next">Rotate/Translate Zone</a>, Previous: <a href="#Regularization-and-Real-Space-Refinement" accesskey="p" rel="previous">Regularization and Real Space Refinement</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Changing-the-Map-for-Building_002fRefinement-1"></a>
<h3 class="section">5.2 Changing the Map for Building/Refinement</h3>
<a name="index-changing-the-Refinement-Map"></a>
<a name="index-map-changing-_0028for-refinement_002fbuilding_0029"></a>

<p>You can change the map that is used for the fitting and refinement tools
using the <code>Select Map<small class="enddots">...</small></code> button on the Model/Fit/Refine
dialog.
</p>

<hr>
<a name="Rotate_002fTranslate-Zone"></a>
<div class="header">
<p>
Next: <a href="#Rigid-Body-Refinement" accesskey="n" rel="next">Rigid Body Refinement</a>, Previous: <a href="#Changing-the-Map-for-Building_002fRefinement" accesskey="p" rel="previous">Changing the Map for Building/Refinement</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rotate_002fTranslate-Zone-1"></a>
<h3 class="section">5.3 Rotate/Translate Zone</h3>
<a name="index-rotate_002ftranslate_002c-manual"></a>

<p>&ldquo;Rotate/Translate Zone&rdquo; from the &ldquo;Model/Fit/Refine&rdquo; menu allows
manual movement of a zone.  After pressing the &ldquo;Rotate/Translate Zone&rdquo;
button, select two atoms in the graphics canvas to define a residue
range <a name="DOCF58" href="#FOOT58"><sup>58</sup></a>, the
second atom that you click will be the local rotation centre for the
zone.  The atoms selected in the moving fragment have the same
alternate conformation code as the first atom you click.  To actuate a
transformation, click and drag horizontally across the relevant button
in the newly-created &ldquo;Rotation &amp; Translation&rdquo; dialog. The axis
system of the rotations and translations are the screen coordinates.
Alternatively  <a name="DOCF59" href="#FOOT59"><sup>59</sup></a>, you can
click using left-mouse on an atom in the fragment and drag the
fragment around. Use Control Left-mouse to move just one atom, rather
than the whole fragment.  If you click Control Left-mouse whilst <em>not</em> 
over an atom then you can rotate the fragment using mouse drag. 
Click &ldquo;OK&rdquo; (or press Return) when the transformation is complete.
</p>
<p>To change the rotation point to the centre of the intermediate atoms
(rather than the second clicked atom), use the setting:
</p>
<p><code>(set-rotate-translate-zone-rotates-about-zone-centre 1)</code>
</p>
<hr>
<a name="Rigid-Body-Refinement"></a>
<div class="header">
<p>
Next: <a href="#Simplex-Refinement" accesskey="n" rel="next">Simplex Refinement</a>, Previous: <a href="#Rotate_002fTranslate-Zone" accesskey="p" rel="previous">Rotate/Translate Zone</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rigid-Body-Refinement-1"></a>
<h3 class="section">5.4 Rigid Body Refinement</h3>
<a name="index-refinement_002c-rigid-body"></a>
<a name="index-rigid-body-fit"></a>
<p>&ldquo;Rigid Body Fit Zone&rdquo; from the
&ldquo;Model/Fit/Refine&rdquo; dialog provides rigid body refinement.  The
selection is zone-based <a name="DOCF60" href="#FOOT60"><sup>60</sup></a>.
So to refine just one residue, click on one atom twice.
</p>
<p>Sometimes no results are displayed after Rigid Body Fit Zone.  This is
because the final model positions had too many final atom positions in
negative density.  If you want to over-rule the default fraction of
atoms in the zone that have an acceptable fit (0.75), to be (say)
0.25:
</p>
<p><code>(set-rigid-body-fit-acceptable-fit-fraction 0.25)</code>
</p>
<hr>
<a name="Simplex-Refinement"></a>
<div class="header">
<p>
Next: <a href="#Post_002dmanipulation_002dhook" accesskey="n" rel="next">Post-manipulation-hook</a>, Previous: <a href="#Rigid-Body-Refinement" accesskey="p" rel="previous">Rigid Body Refinement</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Simplex-Refinement-1"></a>
<h3 class="section">5.5 Simplex Refinement</h3>
<a name="index-simplex-refinement"></a>
<a name="index-refinement_002c-simplex"></a>

<p>Rigid body refinement via Nelder-Mead Simplex minimization is available
in Coot.  Simplex refinement has a larger radius of convergence and thus
is useful in a position where simple rigid body refinement finds the
wrong minimum. However the Simplex algorithm is much slower.  Simplex
refinement for a residue range <code><em>start-resno</em></code> to
<code><em>end-resno</em></code> (inclusive) in chain <code><em>chain-id</em></code> can
be accessed as follows:
</p>
<p><code>(fit-residue-range-to-map-by-simplex <em>start-resno end-resno alt-loc 
        chain-id imol imol-for-map</em>)</code>
</p>
<p>There is currently no GUI interface to Simplex refinement.
</p>
<hr>
<a name="Post_002dmanipulation_002dhook"></a>
<div class="header">
<p>
Next: <a href="#Baton-Building" accesskey="n" rel="next">Baton Building</a>, Previous: <a href="#Simplex-Refinement" accesskey="p" rel="previous">Simplex Refinement</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Post_002dmanipulation_002dhook-1"></a>
<h3 class="section">5.6 Post-manipulation-hook</h3>
<a name="index-post_002dmanipulation_002dhook"></a>

<p>If you wanted automatically run a function after a model has been
manipulated then you can do so using by creating a function that takes
2 arguments, such as:
</p>
<p><code>(post-manipulation-hook <em>imol manipulation-mode</em>)</code>
</p>
<p><code><em>manipulation-mode</em></code> is one of <code><em>(DELETED)</em></code>,
<code><em>(MUTATED)</em></code> or <code><em>(MOVINGATOMS)</em></code>.
</p>
<p>And of course <code><em>imol</em></code> is the model number of the maniplated
molecule.
</p>
<p>(It would of course be far more useful if this function was also
passed a list of residues - that is something for the future).
</p>
<hr>
<a name="Baton-Building"></a>
<div class="header">
<p>
Next: <a href="#Reversing-Direction-of-Fragment" accesskey="n" rel="next">Reversing Direction of Fragment</a>, Previous: <a href="#Post_002dmanipulation_002dhook" accesskey="p" rel="previous">Post-manipulation-hook</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Baton-Building-1"></a>
<h3 class="section">5.7 Baton Building</h3>
<a name="index-baton-build"></a>

<p>Baton build is most useful if a skeleton is already calculated and
displayed (see Section <a href="#Skeletonization">Skeletonization</a>).  When three or more atoms
have been built in a chain, Coot will use a prior probability
distribution for the next position based on the position of the previous
three.  The analysis is similar to that of Oldfield &amp; Hubbard (1994)
<a name="DOCF61" href="#FOOT61"><sup>61</sup></a>, however it is based on a
more recent and considerably larger database.
</p>
<p>Little crosses are drawn representing directions in which is is
possible that the chain goes, and a baton is drawn from the current
point to one of these new positions.  If you don&rsquo;t like this
particular direction <a name="DOCF62" href="#FOOT62"><sup>62</sup></a>, use <code>Try Another</code>.  The
list of directions is scored according to the above criterion and
sorted so that the most likely is at the top of the list and displayed
first as the baton direction.
</p>
<p>When starting baton building, be sure to be about 3.8&Aring; from the
position of the first-placed C<em>\alpha</em>, this is because the next
C<em>\alpha</em> is placed at the end of the baton, the baton root being at
the centre of the screen.  So, when trying to baton-build a chain
starting at residue 1, centre the screen at about the position of
residue 2.
</p>
<p>It seems like a good idea to increase the map sampling to 2 or even 2.5
(before reading in your mtz file) [a grid sampling of about 0.5&Aring;
seems reasonable] when trying to baton-build a low resolution map.  You
can set the map sampling using <code>Edit -&gt; Map Parameters -&gt; Map
Sampling</code>.
</p>

<a name="index-baton-mode"></a>
<p>Occasionally, every point is not where you want to
position the next atom.  In that case you can either shorten or
lengthen the baton, or position it yourself using the mouse.  Use
&ldquo;b&rdquo; on the keyboard to swap to baton mode for the
mouse <a name="DOCF63" href="#FOOT63"><sup>63</sup></a>.
</p>
<p>Baton-built atoms are placed into a molecule called &ldquo;Baton Atom&rdquo; and
it is often sensible to save the coordinates of this molecule before
quitting coot.
</p>
<p>If you try to trace a high resolution map (1.5&Aring;  or better) you will
need to increase the skeleton search depth from the default (10), for
example:
</p>
<p><code>(set-max-skeleton-search-depth 20)</code>
</p>
<p>Alternatively, you could generate a new map using data
to a more moderate resolution (2&Aring;), the map may be easier to
interpret at that resolution anyhow <a name="DOCF64" href="#FOOT64"><sup>64</sup></a>.
</p>
<p>The guide positions are updated every time the &ldquo;Accept&rdquo; button is
clicked.  The molecule name for these atoms is &ldquo;Baton Build Guide Points&rdquo;
and is is not usually necessary to keep them.
</p>
<a name="Undo"></a>
<h4 class="subsection">5.7.1 Undo</h4>
<p>There is also an &ldquo;Undo&rdquo; button for baton-building.  Pressing this
will delete the most recently placed C<em>\alpha</em> and the guide points
will be recalculated for the previous position.  The number of
&ldquo;Undo&rdquo;s is unlimited.  Note that you should use the &ldquo;Undo&rdquo; button
in the Baton Build dialog, not the one in the &ldquo;Model/Fit/Refine&rdquo;
dialog (Section <a href="#Backups-and-Undo">Backups and Undo</a>).
</p>
<a name="Missing-Skeleton"></a>
<h4 class="subsection">5.7.2 Missing Skeleton</h4>
<a name="index-skeleton_002c-missing"></a>
<p>Sometimes (especially at loops) you can see
the direction in which the chain should go, but there is no skeleton
(see Section <a href="#Skeletonization">Skeletonization</a>) is displayed (and consequently no
guide points) in that direction. In that case, &ldquo;Undo&rdquo; the previous
atom and decrease the skeletonization level (<code>Edit
  -&gt; Skeleton Parameters -&gt; Skeletonization
  Level</code>).  Accept the atom (in the same place as last time) and now
when the new guide points are displayed, there should be an option to
build in a new direction.
</p>
<a name="Building-Backwards"></a>
<h4 class="subsection">5.7.3 Building Backwards</h4>
<p>The following scenario is not uncommon: you find a nice stretch of
density and start baton building in it.  After a while you come to a
point where you stop (dismissing the baton build dialog).  You want to
go back to where you started and build the other way.  How do you do
that?
</p>
<ul>
<li> Use the command: 

<p><code>(set-baton-build-params <em>start-resno</em>
    <em>chain-id</em> &quot;backwards&quot;)</code>
</p>
<p>where <code><em>start-resno</em></code> would typically
    be 0 <a name="DOCF65" href="#FOOT65"><sup>65</sup></a> and <code><em>chain-id</em></code> would be
    <code>&quot;&quot;</code> (default).
</p></li><li> Recentre the graphics window on the first atom of the just-build
  fragment
</li><li> Select &ldquo;Ca Baton Mode&rdquo; and select a baton direction
  that goes in the &ldquo;opposite&rdquo; direction to what is typically residue
  2.  This is slightly awkward because the initial baton atoms build
  in the &ldquo;opposite&rdquo; direction are not dependent on the first few
  atoms of the previously build fragment.
</li></ul>

<hr>
<a name="Reversing-Direction-of-Fragment"></a>
<div class="header">
<p>
Next: <a href="#C_002dalpha-_002d_003e-Mainchain" accesskey="n" rel="next">C-alpha -&gt; Mainchain</a>, Previous: <a href="#Baton-Building" accesskey="p" rel="previous">Baton Building</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Reversing-Direction-of-Fragment-1"></a>
<h3 class="section">5.8 Reversing Direction of Fragment</h3>
<a name="index-reverse-direction"></a>
<a name="index-fragment-direction-change"></a>

<p>After you&rsquo;ve build a fragment, sometimes you might want to change the
direction of that fragment (this function changes an already existing
fragment, as opposed to Backwards Building which sets up Baton Building
to place new points in reverse order).
</p>
<p>The fragment is defined as a contiguous set of residues numbers.  So
that you should be sure that other partial fragments which have the same
chain id and that are not connected to this fragment have residue
numbers that are not contiguous with the fragment you are trying to
reverse.  
</p>



<hr>
<a name="C_002dalpha-_002d_003e-Mainchain"></a>
<div class="header">
<p>
Next: <a href="#Backbone-Torsion-Angles" accesskey="n" rel="next">Backbone Torsion Angles</a>, Previous: <a href="#Reversing-Direction-of-Fragment" accesskey="p" rel="previous">Reversing Direction of Fragment</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="C_005calpha-_002d_003e-Mainchain"></a>
<h3 class="section">5.9 C<em>\alpha</em> -&gt; Mainchain</h3>
<a name="index-mainchain"></a>
<p>Mainchain can be generated using a set of C<em>\alpha</em>s
as guide-points (such as those from Baton-building) along the line of
Esnouf <a name="DOCF66" href="#FOOT66"><sup>66</sup></a> or Jones and
coworkers <a name="DOCF67" href="#FOOT67"><sup>67</sup></a>.  Briefly, 6-residue
fragments of are generated from a list of high-quality <a name="DOCF68" href="#FOOT68"><sup>68</sup></a> structures. The C<em>\alpha</em> atoms of these fragments
are matched against overlapping sets of the guide-point C<em>\alpha</em>s.
The resulting matches are merged to provide positions for the
mainchain (and C<em>\beta</em>) atoms.  This procedure works well for
helices and strands, but less well <a name="DOCF69" href="#FOOT69"><sup>69</sup></a> for less common structural features.
</p>
<p>This function is also available from the scripting interface:
</p>
<p><code>(db-mainchain <em>imol chain-id resno-start resno-end direction</em>)</code>
</p>    
<p>where direction is either <code>&quot;backwards&quot;</code> or <code>&quot;forwards&quot;</code>. 
</p>
<p>Recall that the <code><em>chain-id</em></code> needs to be quoted, <em>i.e.</em>
use <code>&quot;A&quot;</code> not <code>A</code>.  Note that <code><em>chain-id</em></code> is
<code>&quot;&quot;</code> when the C<em>\alpha</em>s have been built with Baton Mode in
Coot.
</p>


<hr>
<a name="Backbone-Torsion-Angles"></a>
<div class="header">
<p>
Next: <a href="#Docking-Sidechains" accesskey="n" rel="next">Docking Sidechains</a>, Previous: <a href="#C_002dalpha-_002d_003e-Mainchain" accesskey="p" rel="previous">C-alpha -&gt; Mainchain</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Backbone-Torsion-Angles-1"></a>
<h3 class="section">5.10 Backbone Torsion Angles</h3>
<a name="index-mainchain-torsions"></a>
<p>It is possible to edit the backbone <em>\phi</em> and <em>\psi</em> angles
indirectly using an option in the Model/Fit/Refine&rsquo;s dialog: &ldquo;Edit
Backbone Torsions..&rdquo;. When clicked and an atom of a peptide is
selected, this produces a new dialog that offers &ldquo;Rotate Peptide&rdquo;
which changes this residues <em>\psi</em> and &ldquo;Rotate Carbonyl&rdquo; which
changes <em>\phi</em>.  Click and drag across the button <a name="DOCF70" href="#FOOT70"><sup>70</sup></a> to rotate
the moving atoms in the graphics window.  You should know, of course,
that making these modifications alter the <em>\phi/\psi</em> angles of
more than one residue.
</p>

<hr>
<a name="Docking-Sidechains"></a>
<div class="header">
<p>
Next: <a href="#Rotamers" accesskey="n" rel="next">Rotamers</a>, Previous: <a href="#Backbone-Torsion-Angles" accesskey="p" rel="previous">Backbone Torsion Angles</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Docking-Sidechains-1"></a>
<h3 class="section">5.11 Docking Sidechains</h3>
<a name="index-docking-sidechains"></a>
<a name="index-sidechains"></a>

<p>Docking sidechains means adding sidechains to a model or fragment that
has currently only poly-Ala, where the sequence assignment is unknown.
The algorithm is basically the same as in Cowtan&rsquo;s Buccaneer, but with
some corners cut to make things (more or less) interactive.  The
algorithm uses the shape of the density around the C-beta position to
estimate the probability of each sidechain type at that position.
</p>
<p>The function is accessed via the <code>Extensions -&gt; Dock Sequence</code>
menu item.  First, a sequence should be assigned from a PIR file to a
particular chain-id and model number.  Secondly <code>Extensions -&gt;
Dock Sequence -&gt; Dock Sequence on this fragment&hellip;</code>.  Choose the
model to build on and then <code>Dock Sequence!</code> If all goes well,
the model will be updated with mutated residues and undergo rotamer
seach for each of the new residues.  If the sequence alignment is not
sufficiently clear, then you will get a dialog suggesting that you
extend or improve the fragment.
</p>
<hr>
<a name="Rotamers"></a>
<div class="header">
<p>
Next: <a href="#Editing-Chi-Angles" accesskey="n" rel="next">Editing Chi Angles</a>, Previous: <a href="#Docking-Sidechains" accesskey="p" rel="previous">Docking Sidechains</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rotamers-1"></a>
<h3 class="section">5.12 Rotamers</h3>
<a name="index-rotamers"></a>

<p>The rotamers are generated <a name="DOCF71" href="#FOOT71"><sup>71</sup></a> from the
backbone independent sidechain library of the Richardsons group
<a name="DOCF72" href="#FOOT72"><sup>72</sup></a>.
</p>
<p>The m, t and p stand for &ldquo;minus (-60)&rdquo;, &ldquo;trans (180)&rdquo; and &ldquo;plus
(+60)&rdquo;.  There is one letter per <em>\chi</em> angle.
</p>
<p>Use keyboard <tt class="key">.</tt> and <tt class="key">,</tt> to cycle round the rotamers.
</p>
<a name="Auto-Fit-Rotamer"></a>
<h4 class="subsection">5.12.1 Auto Fit Rotamer</h4>
<a name="index-auto_002dfit-rotamer"></a>
<p>&ldquo;Auto Fit Rotamer&rdquo; will try to fit the
rotamer to the electron density.  Each rotamer is generated, rigid
body refined and scored according to the fit to the map.  Fitting the
second conformation of a dual conformation in this way will often fail
- the algorithm will pick the best fit to the density - ignoring the
position of the other atoms.
</p>
<p>The algorithm doesn&rsquo;t know if the other atoms in the structure are in
sensible positions.  If they are, then it is sensible not to put this
residue too close to them, if they are not then there should be no
restriction from the other atoms as to the position of this residue -
the default is &ldquo;are sensible&rdquo;, which means that the algorithm is
prevented from finding solutions that are too close to the atoms of
other residues. <code>(set-rotamer-check-clashes 0)</code> will stop this.
</p>

<p>There is a scripting interface to auto-fitting rotamers:
</p>
<p><code>(auto-fit-best-rotamer <em>resno alt-loc ins-code chain-id
imol-coords
imol-map clash-flag lowest-rotamer-probability</em>)</code>
</p>
<p>where:
</p>
<p><code><em>resno</em></code> is the residue number
</p>
<p><code><em>alt-loc</em></code> is the alternate/alternative location symbol
(<em>e.g.</em> <code>&quot;A&quot;</code> or <code>&quot;B&quot;</code>, but most often <code>&quot;&quot;</code>)
</p>
<p><code><em>ins-code</em></code> is the insertion code (usually <code>&quot;&quot;</code>)
</p>
<p><code><em>imol-coords</em></code> is the molecule number of the coordinates molecule
</p>
<p><code><em>imol-map</em></code> is the molecule number of the map to which
you wish to fit the side chains
</p>
<p><code><em>clash-flag</em></code> should the positions of other residues be
included in the scoring of the rotamers (<em>i.e.</em> clashing with other
other atoms gets marked as bad/unlikely)
</p>
<p><code><em>lowest-rotamer-probability</em></code>: some rotamers of some side
chains are so unlikely that they shouldn&rsquo;t be considered - typically
0.01 (1%).
</p>

<p>You can change the auto-fit rotamer fitting algorithms using
</p>
<p><code>(set-rotamer-search-mode <em>mode</em>)</code>
</p> 
<p>where <code><em>mode</em></code> is one of <code>(ROTAMERSEARCHAUTOMATIC)</code>,
<code>(ROTAMERSEARCHLOWRES)</code> (<em>i.e.</em> &quot;Backrub Rotamers&quot;
(<em>vide infra</em>)) or <code>(ROTAMERSEARCHHIGHRES)</code> (the
conventional/high-resolution method using rigid-body fitting).
</p>
<p>By default, the auto-fit rotamer method is <code>(ROTAMERSEARCHAUTOMATIC)</code>.
</p>
<a name="Backrub-Rotamers"></a>
<h4 class="subsubsection">5.12.1.1 Backrub Rotamers</h4>
<a name="index-backrub-rotamers"></a>
<p>By default, Auto Fit Rotamer will switch to &ldquo;Backrub Rotamer&rdquo;
<a name="DOCF73" href="#FOOT73"><sup>73</sup></a>
mode when fitting against a map of worse than 2.7&Aring;.  This
search mode moves the some atoms of the mainchain of the
neighbouring residues.  After rotation of the central residue and 
neighbouring atoms around the &ldquo;backrub vector&rdquo;, the 
individual peptides are back-rotated (along the peptide axis) so that the
carbonyl oxygen are placed as near as possible to their original position.  
The Ramachandran plot is not used in this fitting algorithm.
</p>

<a name="De_002dclashing-residues"></a>
<h4 class="subsection">5.12.2 De-clashing residues</h4>
<p>Sometimes you don&rsquo;t have a map <a name="DOCF74" href="#FOOT74"><sup>74</sup></a> but nevertheless there are
clashing residues
<a name="index-clashing-residues"></a>
 <a name="DOCF75" href="#FOOT75"><sup>75</sup></a> (for example after mutation of a
residue range) and you need to rotate side-chains to a non-clashing
rotamer.  There is a scripting interface:
</p>
<p><code>(de-clash <em>imol chain-id start-resno end-resno</em>)</code>
</p>
<p><code><em>start-resno</em></code> is the residue number of the first residue
you wish to de-clash
</p>
<p><code><em>end-resno</em></code> is the residue number of the last residue
you wish to de-clash
</p>
<p><code><em>imol</em></code> is the molecule number of the coordinates molecule
</p>
<p>This interface will not change residues with insertion codes or
alternate conformation.  The
<code><em>lowest-rotamer-probability</em></code> is set to 0.01.
</p>
<hr>
<a name="Editing-Chi-Angles"></a>
<div class="header">
<p>
Next: <a href="#Torsion-General" accesskey="n" rel="next">Torsion General</a>, Previous: <a href="#Rotamers" accesskey="p" rel="previous">Rotamers</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Editing-chi-Angles"></a>
<h3 class="section">5.13 Editing chi Angles</h3>
<a name="index-edit-_005cchi-angles"></a>
<a name="index-chi-angles"></a>
<p>Instead of using Rotamers, one can instead
change the <em>\chi</em> angles 
<a name="index-torsions"></a>
(often called &ldquo;torsions&rdquo;)
&ldquo;by hand&rdquo; (using &ldquo;Edit Chi Angles&rdquo; from the &ldquo;Model/Fit/Refine&rdquo;
dialog). To edit a residue&rsquo;s <em>\chi_1</em> press &ldquo;1&rdquo;: to edit 
<em>\chi_2</em>,
&ldquo;2&rdquo;: <em>\chi_3</em> &ldquo;3&rdquo; and <em>\chi_4</em> &ldquo;4&rdquo;.  
Use left-mouse click and
drag to change the <em>\chi</em> value.  Use keyboard &ldquo;0&rdquo; <a name="DOCF76" href="#FOOT76"><sup>76</sup></a> to go back to ordinary view mode at any time during the
editing.  Alternatively, one can use the &ldquo;View Rotation Mode&rdquo; or use
the <tt class="key">Ctrl</tt> key when moving the mouse in the graphics window.  Use the
Accept/Reject dialog when you have finished editing the <em>\chi</em> 
angles.
</p>
<p>For non-standard residues, the clicked atom defines the base of the atom
tree, which defines the &ldquo;head&rdquo; of the molecule (it&rsquo;s the &ldquo;tail&rdquo;
(twigs/leaves) that wags).  To emphasise, then: it matters on which atom
you click!
</p>

<p>By default torsions for hydrogen atoms are turned off.  To turn them on:
</p>
<p><code>(set-find-hydrogen-torsions 1)</code>
</p>
<p>To edit the rotatable bonds of a ligand using this tool, you will need
to have read in the mmCIF dictionary beforehand.
</p>
<hr>
<a name="Torsion-General"></a>
<div class="header">
<p>
Next: <a href="#Pep_002dflip" accesskey="n" rel="next">Pep-flip</a>, Previous: <a href="#Editing-Chi-Angles" accesskey="p" rel="previous">Editing Chi Angles</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Torsion-General-1"></a>
<h3 class="section">5.14 Torsion General</h3>
<a name="index-torsion-general"></a>

<p>You need to click on the torsion-general button, then click 4 atoms
   that describe the torsion - the first atom will be the base (non
   moving) part of the atom tree, on clicking the 4th atom a dialog
   will pop up with a &quot;Reverse&quot; button.  Move this dialog out of
   the way and then left mouse click and drag in the main window will
   rotate the &quot;top&quot; part of the residue round the clicked atoms 2 and
   3.  When you are happy, click &quot;Accept&quot;.
</p>
<p>If you are torsion generaling a residue that has an alt conf, then
   the atoms of residue that are moved are those that have the same alt
   conf as the 4th clicked atom (or have an blank alt conf).
</p>

<a name="Ligand-Torsion-angles"></a>
<h4 class="subsection">5.14.1 Ligand Torsion angles</h4>
<a name="index-torsion-angles_002c-ligand"></a>
<a name="index-ligand-torsion-angles"></a>
<p>For ligands, you will need to read the mmCIF file that contains a
description of the ligand&rsquo;s geometry (see Section <a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a>).  By default, torsions that move hydrogens
are not included.  Only 9 torsion angles are available from the keyboard
torsion angle selection.
</p>
<hr>
<a name="Pep_002dflip"></a>
<div class="header">
<p>
Next: <a href="#Adding-Alternative-Conformations" accesskey="n" rel="next">Adding Alternative Conformations</a>, Previous: <a href="#Torsion-General" accesskey="p" rel="previous">Torsion General</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Pep_002dflip-1"></a>
<h3 class="section">5.15 Pep-flip</h3>
<a name="index-pepflip"></a>
<a name="index-flip-peptide"></a>
<p>Coot uses the same pepflip scheme
as is used in <code>O</code> (<em>i.e.</em> the C, N and O atoms are rotated
180<em>^o</em> round a line joining the C<em>\alpha</em> atoms of the residues
involved in the peptide).  Flip the peptide again to return the atoms
to their previous position.
</p>
<hr>
<a name="Adding-Alternative-Conformations"></a>
<div class="header">
<p>
Next: <a href="#Mutation" accesskey="n" rel="next">Mutation</a>, Previous: <a href="#Pep_002dflip" accesskey="p" rel="previous">Pep-flip</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Add-Alternate-Conformation"></a>
<h3 class="section">5.16 Add Alternate Conformation</h3>
<a name="index-alternate-conformation"></a>

<p>The allows the addition alternate (dual,
<a name="index-dual-conformations"></a>
triple <em>etc.</em>)  conformations to the picked residue.  By default,
this provides a choice of rotamer (Section <a href="#Rotamers">Rotamers</a>).  If
there are not the correct main chain atoms a rotamer choice cannot be
provided, and Coot falls back to providing intermediate atoms.
</p>
<p>The default occupancy for new atoms is 0.5.  This can be changed by
using use slider on the rotamer selection window or by using the
scripting function:
</p>
<p><code>(set-add-alt-conf-new-atoms-occupancy 0.4)</code>
</p>
<p>The remaining occupancy of the atoms (after the new occupancy has been
added) is split amongst the atoms that existed in the residue before
the split.  It is important therefore that the residues atoms have
sane occupancies before adding an alternative conformation.
</p>


<p>The default Split Type is to split the whole residue.  If you want the
default to be to split a residue after (and including) the CA, then add
to your <samp>.coot</samp> file:
</p>
<p><code>(set-add-alt-conf-split-type-number 0)</code>
</p>
<hr>
<a name="Mutation"></a>
<div class="header">
<p>
Next: <a href="#Importing-Ligands_002fMonomers" accesskey="n" rel="next">Importing Ligands/Monomers</a>, Previous: <a href="#Adding-Alternative-Conformations" accesskey="p" rel="previous">Adding Alternative Conformations</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Mutation-1"></a>
<h3 class="section">5.17 Mutation</h3>
<a name="index-mutate"></a>
<p>Mutations are available on a 1-by-1 basis using the
graphics.  After selecting &ldquo;Mutate&hellip;&rdquo; from the
&ldquo;Model/Fit/Refine&rdquo; dialog, click on an atom in the graphics.  A
&ldquo;Residue Type&rdquo; window will now appear.  Select the new residue type
you wish and the residue in the graphics is updated to the new residue
type <a name="DOCF77" href="#FOOT77"><sup>77</sup></a>.  The
initial position of the new rotamer is the <em>a priori</em> most
likely rotamer. Note that in interactive mode, such as this, a residue
type match <a name="DOCF78" href="#FOOT78"><sup>78</sup></a> will not stop
the mutation action occurring.
</p>
<a name="Mutating-DNA_002fRNA"></a>
<h4 class="subsection">5.17.1 Mutating DNA/RNA</h4>
<a name="index-mutating-RNA"></a>
<a name="index-mutating-DNA"></a>
<a name="index-RNA_002c-mutating"></a>
<a name="index-DNA_002c-mutating"></a>
<p>Mutation of DNA or RNA can be performed using &ldquo;Simple Mutate&rdquo; from the
Model/Fit/Refine dialog.  Residues need to be named &quot;Ad&quot;, &quot;Gr&quot;, &quot;Ur&quot;
etc.  
</p>
<a name="Multiple-mutations"></a>
<h4 class="subsection">5.17.2 Multiple mutations</h4>
<p>This dialog can be found under <code>Calculate -&gt; Mutate
  Residue Range</code>.  A residue range can be assigned a sequence and
optionally fitted to the map.  This is useful converting a poly-ALA
model to the correct sequence <a name="DOCF79" href="#FOOT79"><sup>79</sup></a>.
</p>
<p>Multiple mutations
<a name="index-multi_002dmutate"></a>
 are also supported <em>via</em>
the scripting interface.  Unlike the single residue mutation function,
a residue type match <em>will</em> prevent a modification of the
residue <a name="DOCF80" href="#FOOT80"><sup>80</sup></a>.
Two functions are provided: To mutate a whole chain, use
<code>(mutate-chain <em>imol</em> <em>chain-id sequence</em>)</code> where:
</p>
<p><code><em>chain-id</em></code> is the chain identifier of the chain that you wish
to mutate (<em>e.g.</em> <code>&quot;A&quot;</code>) and 
</p>
<p><code><em>imol</em></code> is molecule number.  
</p>
<p><code><em>sequence</em></code> is a list of single-letter residue codes,
such as <code>&quot;GYRESDF&quot;</code> (this should be a straight string with no
additional spaces or carriage returns).
</p>
<p>Note that the number of residues in the sequence chain and those in
the chain of the protein must match exactly (<em>i.e.</em> the whole of
  the chain is mutated (except residues that have a matching residue
  type).)
</p>
<p>To mutate a residue range, use 
</p>
<ul>
<li> <code>(mutate-residue-range <em>imol</em> <em>chain-id</em>
  <em>start-res-no</em> <em>stop-res-no sequence</em>)</code>
</li></ul>

<p>where
</p>
<p><code><em>start-res-no</em></code> is the starting residue for mutation
</p>
<p><code><em>stop-res-no</em></code> is the last residue for mutation, <em>i.e.</em>
using values of 2 and 3 for <code><em>start-res-no</em></code> and
<code><em>stop-res-no</em></code> respectively will mutate 2 residues.
</p>
<p>Again, the length of the sequence must correspond to the residue range
length.  Note also that this is a protein sequence - not nucleic acid.
</p>
<p>For mutation of nucleic acids, use: 
</p>
<p><code>(mutate-nucleotide-range <em>imol chain-id resno-start
      resno-end sequence</em>)</code>
</p>
<a name="Mutating-to-a-Non_002dStandard-Residue"></a>
<h4 class="subsection">5.17.3 Mutating to a Non-Standard Residue</h4>

<p>Sometimes one might like to model post-translational or other such
modifications.  How is that done, if the new residue type is not one of
the standard residue types?
</p> 
<p>There is a scripting function: 
</p>
<p><code>(mutate-by-overlap <em>imol chain-id resno new-three-letter-code</em>)</code>
</p>
<p>This imports a model residue for the new residue type and overlays it on
to the given residue by using graph-matching to determine the equivalent
atoms.
</p>
<p>The GUI for this can be found under <code>Extensions -&gt; Modelling
-&gt; Replace Residue...</code> (for this to work, you need to be centred on
the residue you wish to replace).
</p>
<p>Note that if you are replacing are conventional protein residue with a
modified form (<em>e.g.</em> replacing a TYR with a phoso-tyrosine or a
LYS with an acetyl-lysine) you will need to make sure that the group
of the resulting restraints is an <code>L-peptide</code> (use <code>Edit
-&gt; Restraints</code> to check and modify the restraints group.  Likewise for
modified RNA/DNA nucleotides, you need to specify the group as
<tt>RNA</tt> or <tt>DNA</tt> as appropriate.
</p>

<a name="Mutate-and-Autofit"></a>
<h4 class="subsection">5.17.4 Mutate and Autofit</h4>
<p>The function combines Mutation and Auto Fit Rotamer and is the easiest
way to make a mutation and then fit to the map.  You can currently only
&ldquo;Mutate and Autofit&rdquo; protein residues (<em>i.e.</em> things with a
rotamer dictionary.
</p>
<a name="Renumbering"></a>
<h4 class="subsection">5.17.5 Renumbering</h4>
<a name="index-renumbering-residues"></a>
<p>Renumbering is straightforward using the
renumber dialog available under <code>Calculate -&gt;
  Renumber Residue Range&hellip;</code>.  There is also a scripting interface:
</p>
<p><code>(renumber-residue-range <em>imol chain-id start-res-no
    last-resno offset</em>)</code>
</p>

<hr>
<a name="Importing-Ligands_002fMonomers"></a>
<div class="header">
<p>
Next: <a href="#Ligand-from-SMILES-strings" accesskey="n" rel="next">Ligand from SMILES strings</a>, Previous: <a href="#Mutation" accesskey="p" rel="previous">Mutation</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Importing-Lignds_002fMonomers"></a>
<h3 class="section">5.18 Importing Lignds/Monomers</h3>
<a name="index-monomers"></a>
<a name="index-ligands"></a>

<p>You can import monomers (often ligands) using <code>File -&gt; Get
Monomer&hellip;</code><a name="DOCF81" href="#FOOT81"><sup>81</sup></a> by providing
the 3-letter code of your monomer/ligand.  The resulting molecule will
be moved so that it placed at the current screen centre.
</p>
<p>Typically, when you are happy about the placement of the ligand, you&rsquo;d
then use <code>Merge Molecules</code> to add the ligand/monomer to the main
set of coordinates.
</p>
<p>This procedure creates a pdb file <samp>monomer-XXX.pdb</samp> and a
dictionary file <samp>libcheck_XXX.cif</samp> in the directory in which Coot
was started.
</p>
<p>A future invocation of Get Monomer uses these file so that the monomer
appears quickly <a name="DOCF82" href="#FOOT82"><sup>82</sup></a>.
</p>
<hr>
<a name="Ligand-from-SMILES-strings"></a>
<div class="header">
<p>
Next: <a href="#Find-Ligands" accesskey="n" rel="next">Find Ligands</a>, Previous: <a href="#Importing-Ligands_002fMonomers" accesskey="p" rel="previous">Importing Ligands/Monomers</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Ligand-from-SMILES-strings-1"></a>
<h3 class="section">5.19 Ligand from SMILES strings</h3>
<a name="index-SMILES-strings"></a>

<p>Similarly, you can generate ligands using <code>File -&gt; SMILES...</code>
and providing a SMILES string and a code for the residue name (this is
your name for the residue type and a dictionary will be generated for
the monomer of this type). This function is also a wrapper to
LIBCHECK.
</p>

<hr>
<a name="Find-Ligands"></a>
<div class="header">
<p>
Next: <a href="#Flip-Ligand" accesskey="n" rel="next">Flip Ligand</a>, Previous: <a href="#Ligand-from-SMILES-strings" accesskey="p" rel="previous">Ligand from SMILES strings</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Find-Ligands-1"></a>
<h3 class="section">5.20 Find Ligands</h3>
<a name="index-ligands-1"></a>
<p>You are offered a selection of maps to search (you can only choose one
at a time) and a selection of molecules that act as a mask to this
map.  Finally you must choose which ligand types you are going to
search for in this map <a name="DOCF83" href="#FOOT83"><sup>83</sup></a>.  Only molecules with less than 400 atoms are suggested
as potential ligands.
</p>
<p>If you do not have any molecules with less that 400 atoms loaded in
Coot, you will get the message:
</p>
<p>&quot;Error: you must have at least one ligand to search for!&quot;
</p>
<p>New ligands are placed where the
map density is and protein (mask) atoms are <em>not</em>).  The masked
map is searched for clusters using a default cut-off of 1.0<em>\sigma</em>.
In weak density this cut-off may be too high and in such a case the
cut-off value can be changed using something such as:
</p>
<p><code>(set-ligand-cluster-sigma-level 0.8)</code>
</p>
<p>However, if the map to be searched for ligands is a difference map, a
cluster level of 2.0 or 3.0 would probably be more appropriate (less
likely to generate spurious sites).
</p>
<p>Each ligand is fitted with rigid body refinement to each potential
ligand site in the map and the best one for each site selected and
written out as a pdb file.  The clusters are sorted by size, the
biggest one first (with an index of 0).  The output placed ligands
files have a prefix &ldquo;best-overall&rdquo; and are tagged by the cluster
index and residue type of the best fit ligand in that site.
</p>
<p>By default, the top 10 sites are tested for ligands - to increase this
use:
</p>
<p><code>(set-ligand-n-top-ligands 20)</code>
</p>
<a name="Flexible-Ligands"></a>
<h4 class="subsection">5.20.1 Flexible Ligands</h4>
<a name="index-ligands_002c-flexible"></a>
<p>If the &ldquo;Flexible?&rdquo; checkbutton is activated, coot will generate a
number of variable conformations (default 100) by rotating around the
rotatable bonds (torsions).  Each of these conformations will be fitted
to each of the potential ligand sites in the map and the best one will
be selected (again, if it passes the fitting criteria above).
</p>
<p>Before you search for flexible ligands you must have read the mmCIF
dictionary for that particular ligand residue type (<code>File
  -&gt; Import CIF dictionary</code>).
<a name="index-dictionary_002c-cif-1"></a>
</p>
<p>Use:
</p>
<p><code>(set-ligand-flexible-ligand-n-samples <em>n-samples</em>)</code>
</p>
<p>where <code><em>n-samples</em></code> is the number of samples of flexibility
made for each ligand.  Generally speaking, The more the number of
rotatable bonds, the bigger this number should be.
</p>
<p>By default the options to change these values are not in the GUI.  To
enable these GUI options, use the scripting function:
</p>
<p><code>(ligand-expert)</code>
</p>
<a name="Adding-Ligands-to-Model"></a>
<h4 class="subsection">5.20.2 Adding Ligands to Model</h4>
<p>After successful ligand searching, one may well want to add that
displayed ligand to the current model (the coordinates set that
provided the map mask).  To do so, use Merge Molecules (Section
<a href="#Merge-Molecules">Merge Molecules</a>).
</p>

<hr>
<a name="Flip-Ligand"></a>
<div class="header">
<p>
Next: <a href="#Find-Waters" accesskey="n" rel="next">Find Waters</a>, Previous: <a href="#Find-Ligands" accesskey="p" rel="previous">Find Ligands</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Flip-Ligand-1"></a>
<h3 class="section">5.21 Flip Ligand</h3>
<a name="index-ligand-orientation"></a>

<p>Sometimes a ligand is placed more or less in the correct position, but
the orientation is wrong - or at least you might want to explore other
possible orientation.  To do that easily a function has been provided:
</p>
<p><code>(flip-ligand <em>imol chain-id residue-number</em>)</code>
</p>
<p>This will flip the orientation of the residue around the Eigen vector
corresponding to the largest Eigen value, exploring 4 possible
orientations.
</p>
<p>This function has been further wrapped to provide flipping for the active
residue: 
</p>
<p><code>(flip-active-ligand)</code>
</p>
<p>This function can easily be bound to a key.
</p>

<hr>
<a name="Find-Waters"></a>
<div class="header">
<p>
Next: <a href="#Add-Terminal-Residue" accesskey="n" rel="next">Add Terminal Residue</a>, Previous: <a href="#Flip-Ligand" accesskey="p" rel="previous">Flip Ligand</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Find-Waters-1"></a>
<h3 class="section">5.22 Find Waters</h3>
<a name="index-waters_002c-finding"></a>
<p>As with finding ligands, you are given a choice of maps, protein
(masking) atoms. A final selection has to be made for the cut-off level,
note that this value is the number of standard deviation of the density
of the map <em>before</em> the map has been masked.  The default sigma
level (water positions must have density above this level) is set for a
&ldquo;2Fo-Fc&rdquo;-style map.  If you want to use a difference map, you must
change the sigma level (typically to 3 sigma) otherwise you run the risk
of fitting waters to difference map noise peaks.
</p>
<p>Then the map is masked by the masking atoms and a search is
made of features in the map about the electron density cut-off value.
Waters are added if the feature is approximately water-sized and can
make sensible hydrogen bonds to the protein atoms.  The new waters are
optionally created in a new molecule called &ldquo;Waters&rdquo;.
</p>
<p>You have control over several parameters used in the water finding:
</p>
<p><code>(set-write-peaksearched-waters)</code> 
</p>
<p>which writes <code>ligand-waters-peaksearch-results.pdb</code>, which
contains the water peaks (from the clusters) without any filtering and
<code>ligand-waters.pdb</code> which are a disk copy filtered waters that
have been either added to the molecule or from which a new molecule
has been created.
</p>
<p><code>(set-ligand-water-to-protein-distance-limits <em>min-d max-d</em>)</code> sets
the minimum and maximum allowable distances between new waters and the
masking molecule (usually the protein).  Defaults are 2.4 and 3.2&Aring;.
</p>
<p><code>(set-ligand-water-spherical-variance-limit <em>varlim</em>)</code> sets the
upper limit for the density variance around water atoms. The default
is 0.12.
</p>
<p>The map that is marked by the protein and is searched to find the
waters is written out in CCP4 format as <code>&quot;masked-for-waters.map&quot;</code>.
</p>
<a name="Refinement-Failure"></a>
<h4 class="subsection">5.22.1 Refinement Failure</h4>

<p>Sometimes as a result of water fitting, you may see something like:
</p>
<pre class="verbatim"> WARNING:: refinement failure
           start pos: xyz = (      17.1,     34.76,     60.42)
           final pos: xyz = (     17.19,     34.61,     60.59)
</pre> 
<p>When Coot finds a blob, it does a crude positioning of an atom at the
centre of the grid points.  It then proceeds to move to the peak of the
blob by a series of translations.  There are a certain number of
cycles, and if it doesn&rsquo;t reach convergence by the end of those cycles
then you get the error message.
</p>
<p>Often when you go to the position indicated, you can see why Coot had a
problem in the refinement.
</p>
<a name="Blobs"></a>
<h4 class="subsection">5.22.2 Blobs</h4>
<p>After a water search, Coot will create a blobs dialog (see Section
<a href="#sec_005fblobs">sec_blobs</a>).
</p>
<hr>
<a name="Add-Terminal-Residue"></a>
<div class="header">
<p>
Next: <a href="#Add-OXT-Atom-to-Residue" accesskey="n" rel="next">Add OXT Atom to Residue</a>, Previous: <a href="#Find-Waters" accesskey="p" rel="previous">Find Waters</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Add-Terminal-Residue-1"></a>
<h3 class="section">5.23 Add Terminal Residue</h3>
<a name="index-terminal-residue"></a>
<p>This creates a new residue at the C or N terminal extension of the
residue clicked by fitting to the map.  <em>\phi,\psi</em> angle pairs
are selected at random based on the Ramachandran plot probability (for
a generic residue) and fitted to the density.  By default there are
100 trials.  It is possible that a wrong position will be selected for
the terminal residue and if so, you can reject this fit and try again
with Fit Terminal Residue <a name="DOCF84" href="#FOOT84"><sup>84</sup></a>. Each of the trial
positions are scored according to their fit to the map <a name="DOCF85" href="#FOOT85"><sup>85</sup></a> and the best one
selected.  It is probably a good idea to run &ldquo;Refine Zone&rdquo; on these
new residues.
</p>
<p>If you use the Extensions (<code>Dock Sequence... -&gt; Associate
Sequence with Model</code>) to apply a PIR sequence file to a model then
<code>Add Terminal Residue</code> will use the sequence alignment to
determine the residue type of the added residue.
</p>
<p>Sometimes, particularly with low resolution maps, the added terminal
residue will wander off to somewhere inappropriate.  This can be
addressed in a number of ways:
</p>

<ol>
<li> <code>(set-terminal-residue-do-rigid-body-refine 0)</code> will disable
rigid body fitting of the terminal residue fragment for
each trial residue position (the default is 1 (on)) - this may help if
the search does not provide good results.

</li><li> to anneal the newly added residue back to the clicked residue (no
matter where it ended up being positioned):
<code>(set-add-terminal-residue-do-post-refine 1)</code>

</li><li> <code>(set-add-terminal-residue-n-phi-psi-trials 200)</code> will change the
number of trials (default is 100).  This is useful if you think that
Coot needs to search harder to find a good solution to the positioning
of the next residue.

</li></ol>

<hr>
<a name="Add-OXT-Atom-to-Residue"></a>
<div class="header">
<p>
Next: <a href="#Add-Atom-at-Pointer" accesskey="n" rel="next">Add Atom at Pointer</a>, Previous: <a href="#Add-Terminal-Residue" accesskey="p" rel="previous">Add Terminal Residue</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Add-OXT-Atom-to-Residue-1"></a>
<h3 class="section">5.24 Add OXT Atom to Residue</h3>

<a name="index-terminal-oxygen"></a>
<a name="index-OXT-atom"></a>
<p>At the
C-terminus
<a name="index-C_002dterminus"></a>
of a chain of amino-acid residues, there
is a &ldquo;modification&rdquo; so that the C-O becomes a carbonyl, <em>i.e.</em>
an extra (terminal) oxygen (OXT) needs to be added.  This atom is
added so that it is in the plane of the C<em>\alpha</em>, C and O atoms of
the residue.
</p>
<p>Scripting usage:
</p>
<p><code>(add-OXT-to-residue imol residue-number insertion-code
  chain-id)</code> <a name="DOCF86" href="#FOOT86"><sup>86</sup></a>, 
</p>
<p>where <code>insertion-code</code> is typically <code>&quot;&quot;</code>.  
</p>
<p>Note, in order to place OXT, the N, CA, C and O atoms must be present
in the residue - if (for example) the existing carbonyl oxygen atom is
called &ldquo;OE1&rdquo; then this function will not work.
</p>
<hr>
<a name="Add-Atom-at-Pointer"></a>
<div class="header">
<p>
Next: <a href="#Place-Helix" accesskey="n" rel="next">Place Helix</a>, Previous: <a href="#Add-OXT-Atom-to-Residue" accesskey="p" rel="previous">Add OXT Atom to Residue</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Add-Atom-at-Pointer-1"></a>
<h3 class="section">5.25 Add Atom at Pointer</h3>
<p>By default, &ldquo;Add Atom At Pointer&rdquo; will pop-up a dialog from which
you can choose the atom type you wish to insert <a name="DOCF87" href="#FOOT87"><sup>87</sup></a>.  Using
<code>(set-pointer-atom-is-dummy 1)</code> you can by-pass this dialog and
immediately create a dummy atom at the pointer position.  Use an
argument of <code>0</code> to revert to using the atom type selection
pop-up on a button press.
</p>
<p>The atoms are added to a new molecule called &ldquo;Pointer Atoms&rdquo;.  They
should be saved and merged with your coordinates outside of Coot.
</p>
<hr>
<a name="Place-Helix"></a>
<div class="header">
<p>
Next: <a href="#Building-Ideal-DNA-and-RNA" accesskey="n" rel="next">Building Ideal DNA and RNA</a>, Previous: <a href="#Add-Atom-at-Pointer" accesskey="p" rel="previous">Add Atom at Pointer</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Place-Helix-1"></a>
<h3 class="section">5.26 Place Helix</h3>
<a name="index-helix-placement"></a>

<p>The idea is to place a helix more or less &ldquo;here&rdquo; (the screen centre)
by fitting to the electron density map.  The algorithm is
straightforward.  First we move to the local centre of density, then
examine the density for characteristic directions and fit ideal helices
(of length 20 residues) to these directions.  The helix is then extended
if possible (by checking the fit to the map of residues added in ideal
helix conformation) and chopped back if not.  If the fit is successful,
the helix is created in a new molecule called &ldquo;Helix&rdquo;.  If the fit is
not successful, there is instead a message added to the status bar.  You
can build the majority of a helical protein in a few minutes using this
method (you will of course have to assemble the helices and assign
residue numbers and sequence later).
</p>
<p>This is available as a scripting function <code>(place-helix-here)</code> and
in the GUI (in the &ldquo;Other Modelling Tools&rdquo; dialog).
</p>
<hr>
<a name="Building-Ideal-DNA-and-RNA"></a>
<div class="header">
<p>
Next: <a href="#Merge-Molecules" accesskey="n" rel="next">Merge Molecules</a>, Previous: <a href="#Place-Helix" accesskey="p" rel="previous">Place Helix</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Building-Ideal-DNA-and-RNA-1"></a>
<h3 class="section">5.27 Building Ideal DNA and RNA</h3>
<a name="index-DNA_002c-ideal"></a>
<a name="index-RNA_002c-ideal"></a>
<a name="index-polynucleotides"></a>

<p>The interface to building ideal polynucleotides can be found by
pressing the &ldquo;Ideal RNA/DNA&hellip;&rdquo; button on the &ldquo;Other Modelling
Tools&rdquo; dialog.
</p>
<p>For a given sequence, a choice of DNA or RNA, A or B form, single or
double stranded is presented.  
</p>
<p>The interface may not gracefully handle uracils in DNA, thymines in
RNA or B form RNA.
</p>
<p>The ideal B-form DNA is somewhat under-wound, needing 11 base-pairs to
repeat (instead of the expected 10.5).  There is no easy fix for this
currently.
</p>


<hr>
<a name="Merge-Molecules"></a>
<div class="header">
<p>
Next: <a href="#Temperature-Factor-for-New-Atoms" accesskey="n" rel="next">Temperature Factor for New Atoms</a>, Previous: <a href="#Building-Ideal-DNA-and-RNA" accesskey="p" rel="previous">Building Ideal DNA and RNA</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Merge-Molecules-1"></a>
<h3 class="section">5.28 Merge Molecules</h3>
<a name="index-merge-molecules"></a>
<p>The dialog for this opperation can be found under &ldquo;Calculate&rdquo; in the
main menubar.  This is typically used to add molecule fragments or
residues that are in one molecule to the &ldquo;working&rdquo; coordinates
<a name="DOCF88" href="#FOOT88"><sup>88</sup></a>.
</p>
<p>The scripting interface is used like this
</p>
<p><code>(merge-molecules <em>molecule-list</em> <em>target-molecule</em>)</code>
</p>
<p><em>e.g.</em>
</p>
<p><code>(merge-molecules (list 1 2 ) 0)</code>
</p>
<p>merges molecules 1 and 2 into molecule 0.
</p>
<hr>
<a name="Temperature-Factor-for-New-Atoms"></a>
<div class="header">
<p>
Next: <a href="#Applying-NCS-Edits" accesskey="n" rel="next">Applying NCS Edits</a>, Previous: <a href="#Merge-Molecules" accesskey="p" rel="previous">Merge Molecules</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Temperature-Factor-for-New-Atoms-1"></a>
<h3 class="section">5.29 Temperature Factor for New Atoms</h3>
<a name="index-B_002dFactor_002c-New-Atoms"></a>

<p>The default temperature factor for new atoms is 30.0.  This can be
changed by the following
</p>
<p><code>(set-default-temperature-factor-for-new-atoms 50.0)</code>
</p>
<hr>
<a name="Applying-NCS-Edits"></a>
<div class="header">
<p>
Next: <a href="#Running-Refmac" accesskey="n" rel="next">Running Refmac</a>, Previous: <a href="#Temperature-Factor-for-New-Atoms" accesskey="p" rel="previous">Temperature Factor for New Atoms</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Applying-NCS-Edits-1"></a>
<h3 class="section">5.30 Applying NCS Edits</h3>
<a name="index-NCS-edits"></a>

<p>Let&rsquo;s imagine that you have 3-fold NCS.  You have molecule &ldquo;A&rdquo; as your
master molecule and you make edits to that molecule.  Now you want to
apply the edits that you made to &ldquo;A&rdquo; (the NCS master chain ID) to the
&ldquo;B&rdquo; and &ldquo;C&rdquo; molecules (i.e. you want the &ldquo;B&rdquo; and &ldquo;C&rdquo; molecules
to be rotated/translated versions of the &ldquo;A&rdquo; molecule).  How is that
done?
</p>
<p>There are now guis to NCS command to help you out (under Extensions).
However, for completeness here are the scripting versions:
</p>
<p><code>(copy-from-ncs-master-to-others <em>imol</em> <em>master-chain-id</em>)</code>
</p>
<p>If you have only a range of residues, rather than a whole chain to replace:
</p>
<p><code>(copy-residue-range-from-ncs-master-to-others <em>imol</em> 
        <em>master-chain-id start-resno end-resno</em>)</code>
</p>
<p>e.g.
</p>
<p><code>(copy-residue-range-from-ncs-master-to-others 0 &quot;A&quot; 1 5)</code>
</p>
<p>If you want to copy a residue range to a specific chain, or specific
list of chains (rather than all NCS peer chains) then make a list of
the chain-ids that you wish replaced:
</p>
<p><code>(copy-residue-range-from-ncs-master-to-chains 0 &quot;A&quot; 1 5 (list &quot;C&quot;))</code>
</p>
<p>in this case, just the residues in the &quot;C&quot; chain is replaced.
</p>

<hr>
<a name="Running-Refmac"></a>
<div class="header">
<p>
Next: <a href="#Running-SHELXL" accesskey="n" rel="next">Running SHELXL</a>, Previous: <a href="#Applying-NCS-Edits" accesskey="p" rel="previous">Applying NCS Edits</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Running-Refmac-1"></a>
<h3 class="section">5.31 Running Refmac</h3>
<a name="index-refmac"></a>
<a name="index-running-refmac"></a>

<p>Use the &ldquo;Run Refmac<small class="enddots">...</small>&rdquo; button to select the dataset and the
coordinates on which you would like to run Refmac.  Note that here Coot
only allows the use of datasets which has Refmac parameters set as the
MTZ file was read. By default, Coot displays the new coordinates and the
new map generated from refmac&rsquo;s output MTZ file.  Optionally, you can
also display the difference map.
</p>
<p>You can add extra parameters 
<a name="index-refmac-parameters"></a>
(data lines) to
refmac&rsquo;s input by storing them in a file called
<code>refmac-extra-params</code> in the directory in which you started
coot.
</p>
<p>You can also provide extra/replacement parameters for refmac by setting
the variable <code>refmac-extra-params</code> to a list of strings, for
example:
</p>
<p><code>(set! refmac-extra-params (list &quot;REFINE MATRIX 0.1&quot; &quot;MAKE HYDROGENS NO&quot;))</code>
</p>

<p>Coot &ldquo;blocks&rdquo; <a name="DOCF89" href="#FOOT89"><sup>89</sup></a> until Refmac has terminated <a name="DOCF90" href="#FOOT90"><sup>90</sup></a>.
</p>
<p>The default refmac executable
<a name="index-refmac_002c-default"></a>
<a name="index-default-refmac-executable"></a>
is <code>refmac5</code> it is presumed to be in the
path.  If you don&rsquo;t want this, it can be overridden using a
re-definition either at the scripting interface or in one&rsquo;s
<code>~/.coot</code> file <em>e.g.</em>:
</p><ul>
<li> <code>(define refmac-exec &quot;/e/refmac-new/bin/refmac5.6.3&quot;)</code>
</li></ul>

<a name="index-refmac-map-colour"></a>
<p>After running refmac several times, you may
find that you prefer if the new map that refmac creates (after refmac
refinement) is the same colour as the previous one (from before this
refmac refinement).  If so, use:
</p>
<p><code>(set-keep-map-colour-after-refmac 1)</code>
</p>
<p>which will swap the colours of then new and old refmac map so that the
post-refmac map has the same colour as the pre-refmac map and the
pre-refmac map is coloured with a different colour.
</p>
<hr>
<a name="Running-SHELXL"></a>
<div class="header">
<p>
Next: <a href="#Clear-Pending-Picks" accesskey="n" rel="next">Clear Pending Picks</a>, Previous: <a href="#Running-Refmac" accesskey="p" rel="previous">Running Refmac</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Running-SHELXL-1"></a>
<h3 class="section">5.32 Running SHELXL</h3>
<a name="index-SHELXL"></a>
<a name="index-running-SHELXL"></a>

<p>Coot can read shelx <tt>.res</tt> files and write <tt>.ins</tt> files, and thus
one can refine using SHELXL in a convenient manner using the function 
</p>
<p><code>(shelxl-refine <em>imol</em> . <em>hkl-file-name</em>)</code>
</p>
<p>(the <code><em>hkl-file-name</em></code> is an optional argument)
</p>
<p><em>e.g.</em>
</p>
<p><code>(shelxl-refine 0)</code>
</p>
<p>or 
</p>
<p><code>(shelxl-refine 0 &quot;insulin.hkl&quot;)</code>
</p>
<p>In the former case, coot will presume that there is a SHELX <tt>hkl</tt> file
corresponding to the <tt>res</tt> file that you read in; if there is not coot
will print a warning and not try to run shelxl.  In the latter case, you
can specify the location of the <tt>hkl</tt> file.
</p>
<p>After shelxl has finished, coot will automatically read in the resulting
<tt>res</tt> coordinates, the <tt>fcf</tt> file, convert the data to <tt>mmCIF</tt>
format and read that, which generates a <em>\sigma_A</em> map and a
difference map.  
</p>
<p>Coot creates a time stamped <tt>ins</tt> file and a time-stamped sym-link to
the <tt>hkl</tt> file in the <code>coot-shelxl</code> directory.
</p>

<p>Please note that
the output <tt>ins</tt> file will not be particularly useful (and thus shelxl
will fail) if the input file was not in SHELX <tt>ins</tt> format.
</p>
<p>There is a GUI for this operation under the &ldquo;Extensions&rdquo; menu item.
</p>
<hr>
<a name="Clear-Pending-Picks"></a>
<div class="header">
<p>
Next: <a href="#Delete" accesskey="n" rel="next">Delete</a>, Previous: <a href="#Running-SHELXL" accesskey="p" rel="previous">Running SHELXL</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Clear-Pending-Picks-1"></a>
<h3 class="section">5.33 Clear Pending Picks</h3>
<a name="index-Clear-Pending-Picks"></a>
<a name="index-atom-picking"></a>
<p>Sometimes one can click
on a button <a name="DOCF91" href="#FOOT91"><sup>91</sup></a> unintentionally. This
button is there for such a case.  It clears the expectation of an
atom pick.  This works not only for modelling functions, but also
geometry functions (such as Distance and Angle).
</p>
<hr>
<a name="Delete"></a>
<div class="header">
<p>
Next: <a href="#Sequence-Assignment" accesskey="n" rel="next">Sequence Assignment</a>, Previous: <a href="#Clear-Pending-Picks" accesskey="p" rel="previous">Clear Pending Picks</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Delete-1"></a>
<h3 class="section">5.34 Delete</h3>
<a name="index-delete"></a>

<p>Single atoms or residues can be deleted from the molecule using
&ldquo;Delete&hellip;&rdquo; from the &ldquo;Model/Fit/Refine&rdquo;dialog.  Pressing this
button results in a new dialog, with the options of &ldquo;Residue&rdquo; (the
default), &ldquo;Atom&rdquo; and &ldquo;Hydrogen Atoms&rdquo;.  Now click on an atom in the
graphics - the deleted object will be the whole residue of the atom if
&ldquo;Residue&rdquo; was selected and just that atom if &ldquo;Atom&rdquo; was selected.
Note that if a residue has an alternative conformation, then &ldquo;Delete
Residue&rdquo; will delete only the conformation that matches that
alternative conformation specifier of the clicked atom.
</p>
<p>Only waters are deletable if the &quot;Water&quot; check button is active and
waters are not deletable if the &quot;Residue/Monomer&quot; check button is
active.  This is to reduce mis-clicking.
</p>
<p>To rotate the view when in &ldquo;Delete Mode&rdquo;, use Ctrl left-mouse.
</p>
<p>If you want to delete multiple items you can use check the &ldquo;Keep Delete
Active&rdquo; check-button on this dialog This will will keep the dialog
open, ready for deletion of next item.
</p>

<p>An atom can be delete using the scripting
</p>
<p><code>(delete-atom <em>imol chain-id residue-no ins-code atom-name alt-conf</em>)</code>
</p>
<p>Residues can be deleted using the scripting
</p>
<p><code>(delete-residue <em>imol chain-id residue-no ins-code</em>)</code>
</p>
<p>Residue ranges can be deleted using the scripting
</p>
<p><code>(delete-residue-range <em>imol chain-id residue-no-start residue-no-end</em>)</code>
</p>
<p>Chains can be deleted using the scripting
</p>
<p><code>(delete-chain <em>imol chain-id</em>)</code>
</p>
<p>Sidechains can be deleted from a region:
</p>
<p><code>(delete-side-chain-range <em>imol chain-id residue-no-start residue-no-end</em>)</code>
</p>
<p>or for a chain
</p>
<p><code>(delete-sidechains-for-chain <em>imol chain-id</em>)</code>
</p>
<hr>
<a name="Sequence-Assignment"></a>
<div class="header">
<p>
Next: <a href="#Building-Links-and-Loops" accesskey="n" rel="next">Building Links and Loops</a>, Previous: <a href="#Delete" accesskey="p" rel="previous">Delete</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Sequence-Assignment-1"></a>
<h3 class="section">5.35 Sequence Assignment</h3>
<p>You can assign a (FASTA format) sequence to a molecule using:
</p>
<p><code>(assign-fasta-sequence imol chain-id fasta-seq)</code>
</p>
<p>This function has been provided as a precursor to functions that will
(as automatically as possible) mutate your current coordinates to one
that has the desired sequence. It will be used in automatic side-chain
assignment (at some stage in the future).
</p>
<hr>
<a name="Building-Links-and-Loops"></a>
<div class="header">
<p>
Next: <a href="#Fill-Partial-Residues" accesskey="n" rel="next">Fill Partial Residues</a>, Previous: <a href="#Sequence-Assignment" accesskey="p" rel="previous">Sequence Assignment</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Building-Links-and-Loops-1"></a>
<h3 class="section">5.36 Building Links and Loops</h3>

<p>Coot can build missing linking regions or loops
<a name="DOCF92" href="#FOOT92"><sup>92</sup></a>. The function can be found under  <code>Calculate -&gt; Fit Loop -&gt; Fit Loop by Rama Search</code> 
or the scripting function:
</p>
<p><code>(fit-gap <em>imol</em> <em>chain-id</em> <em>start-resno</em> <em>stop-resno</em>)</code>
</p>
<p>and 
</p>
<p><code>(fit-gap <em>imol</em> <em>chain-id</em> <em>start-resno</em> <em>stop-resno</em> <em>sequence</em>)</code>
</p>
<p>the second form will also mutate and try to rotamer fit the provided sequence.
</p>
<p>Example usage: let&rsquo;s say for molecule number 0 in chain <code>&quot;A&quot;</code>
we have residues up to 56 and then a gap after which we have residues
62 and beyond:
</p>
<p><code>(fit-gap 0 &quot;A&quot; 57 61 &quot;TYPWS&quot;)</code>
</p>
<hr>
<a name="Fill-Partial-Residues"></a>
<div class="header">
<p>
Next: <a href="#Changing-Chain-IDs" accesskey="n" rel="next">Changing Chain IDs</a>, Previous: <a href="#Building-Links-and-Loops" accesskey="p" rel="previous">Building Links and Loops</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Fill-Partial-Residues-1"></a>
<h3 class="section">5.37 Fill Partial Residues</h3>
<a name="index-fill-partial-residues"></a>

<p>After molecular replacement, the residues of your protein could well
have the correct sequence but be chopped back to CG or CB atoms. There
is a function to fill such partially-filled residues:
</p>
<p><code>(fill-partial-residues <em>imol</em>)</code>
</p>
<p>This identifies residues with missing atoms, then fills them and does a
rotamer fit and real-space refinement.
</p>
<p>If you want to fill the side chain of just one residue
</p>
<p><code>(fill-partial-residue <em>imol chain-id res-no ins-code</em>)</code>
</p>
<p>this does a auto-fit-best-rotamer and a refinement on the resulting
side-chain position.
</p>

<hr>
<a name="Changing-Chain-IDs"></a>
<div class="header">
<p>
Next: <a href="#Setting-Occupancies" accesskey="n" rel="next">Setting Occupancies</a>, Previous: <a href="#Fill-Partial-Residues" accesskey="p" rel="previous">Fill Partial Residues</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Changing-Chain-IDs-1"></a>
<h3 class="section">5.38 Changing Chain IDs</h3>
<a name="index-chain-id"></a>
<a name="index-changing-chain-ids"></a>

<p>You can change the chain ids of chains using <code>Calculate -&gt;
Change Chain IDs&hellip;</code>.  Coot will block an attempt to change the
whole of a chain and the target chain id already exists in the
molecule.  
</p>
<p>If you use the &quot;Residue Range&quot; option then you can insert residues
with non-conflicting residue number into pre-existing chains.
</p>

<hr>
<a name="Setting-Occupancies"></a>
<div class="header">
<p>
Next: <a href="#Fix-Nomenclature-Errors" accesskey="n" rel="next">Fix Nomenclature Errors</a>, Previous: <a href="#Changing-Chain-IDs" accesskey="p" rel="previous">Changing Chain IDs</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Setting-Occupancies-1"></a>
<h3 class="section">5.39 Setting Occupancies</h3>
<p>As well as the editing &ldquo;Residue Info&rdquo; to change occupancies of
individual atoms, one can use a scripting function to change
occupancies of a whole residue range:
</p>
<ul>
<li> <code>(zero-occupancy-residue-range <em>imol chain-id 
resno-start resno-last</em>)</code>
</li></ul>

<p>example usage:
</p>
<p><code>(zero-occupancy-residue-range 0 &quot;A&quot; 23 28)</code>
</p>
<p>This is often useful to zero out a questionable loop before submitting
for refinement.  After refinement (with refmac) there should be
relatively unbiased density in the resulting 2Fo-Fc-style and
difference maps.
</p>
<p>Similarly there is a function to reverse this operation:
</p>
<ul>
<li> <code>(fill-occupancy-residue-range <em>imol chain-id 
      resno-start resno-last</em>)</code>
</li></ul>


<hr>
<a name="Fix-Nomenclature-Errors"></a>
<div class="header">
<p>
Next: <a href="#Rotamer-Fix-Whole-Protein" accesskey="n" rel="next">Rotamer Fix Whole Protein</a>, Previous: <a href="#Setting-Occupancies" accesskey="p" rel="previous">Setting Occupancies</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Fix-Nomenclature-Errors-1"></a>
<h3 class="section">5.40 Fix Nomenclature Errors</h3>

<p>Currently this is available only in scripting form:
</p>
<p><code>(fix-nomenclature-errors <em>imol</em>)</code>
</p>
<p>This will fix atoms nomenclature problems in molecule number
<code><em>imol</em></code> according to the same criteria as WATCHECK
<a name="DOCF93" href="#FOOT93"><sup>93</sup></a>
<em>e.g.</em> Chi-2 for Phe, Tyr, Asp, and Glu should be between -90 and
90 degrees.  Note that Val and Leu nomenclature errors are also corrected.
</p>

<hr>
<a name="Rotamer-Fix-Whole-Protein"></a>
<div class="header">
<p>
Next: <a href="#Refine-All-Waters" accesskey="n" rel="next">Refine All Waters</a>, Previous: <a href="#Fix-Nomenclature-Errors" accesskey="p" rel="previous">Fix Nomenclature Errors</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rotamer-Fix-Whole-Protein-1"></a>
<h3 class="section">5.41 Rotamer Fix Whole Protein</h3>
       
<p>There is an experimental scripting function
</p>
<p><code>(fit-protein <em>imol</em>)</code>
</p>
<p>which does a auto-fit rotamer and Real Space Refinement for each
residue.  The graphics follow the refinement.
</p>

<hr>
<a name="Refine-All-Waters"></a>
<div class="header">
<p>
Next: <a href="#Moving-Molecules_002fLigands" accesskey="n" rel="next">Moving Molecules/Ligands</a>, Previous: <a href="#Rotamer-Fix-Whole-Protein" accesskey="p" rel="previous">Rotamer Fix Whole Protein</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Refine-All-Waters-1"></a>
<h3 class="section">5.42 Refine All Waters</h3>

<p>All the waters in a model can be refined (that is, moved to the local
density peak) using
</p>
<p><code>(fit-waters <em>imol</em>)</code>
</p>
<p>This is a non-interactive function (the waters are moved without user
intervention).
</p>

<hr>
<a name="Moving-Molecules_002fLigands"></a>
<div class="header">
<p>
Next: <a href="#Modifying-the-Labels-on-the-Model_002fFit_002fRefine-dialog" accesskey="n" rel="next">Modifying the Labels on the Model/Fit/Refine dialog</a>, Previous: <a href="#Refine-All-Waters" accesskey="p" rel="previous">Refine All Waters</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Moving-Molecules_002fLigands-1"></a>
<h3 class="section">5.43 Moving Molecules/Ligands</h3>
<a name="index-moving-molecules"></a>
<a name="index-moving-ligands"></a>

<p>Often you want to move a ligand (or some such) from wherever it was read
in to the position of interest in your molecule (<em>i.e.</em> the current view
centre).  There is a GUI to do this: <code>Calculate -&gt; Move Molecule
Here</code>.
</p>
<p>There are scripting functions available for this sort of thing:
</p>
<p><code>(molecule-centre <em>imol</em>)</code> 
</p>
<p>will tell you the molecule centre 
<a name="index-molecule-centre"></a>
 of the
<code><em>imol</em></code>th molecule.
</p>
<p><code>(translate-molecule-by <code>imol x-shift y-shift z-shift</code>)</code>
</p>
<p>will translate all the atoms in molecule <code><em>imol</em></code> by the
given amount (in &Aring;ngstr&ouml;ms).
<a name="index-translate-molecule"></a>
</p>
<p><code>(move-molecule-to-screen-centre <em>imol</em>)</code>
</p>
<p>will move the <code><em>imol</em></code>th molecule to the current centre of
the screen (sometimes useful for imported ligands).  Note that this
moves the atoms of the molecule - not just the view of the molecule.
</p>
<hr>
<a name="Modifying-the-Labels-on-the-Model_002fFit_002fRefine-dialog"></a>
<div class="header">
<p>
Previous: <a href="#Moving-Molecules_002fLigands" accesskey="p" rel="previous">Moving Molecules/Ligands</a>, Up: <a href="#Modelling-and-Building" accesskey="u" rel="up">Modelling and Building</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Modifying-the-Labels-on-the-Model_002fFit_002fRefine-dialog-1"></a>
<h3 class="section">5.44 Modifying the Labels on the Model/Fit/Refine dialog</h3>
<a name="index-modified-labels"></a>

<p>If you don&rsquo;t like the labels &quot;Rotate/Translate Zone&quot; or &quot;Place Atom at
Pointer&quot; and rather they said something else, you can change the button
names using:
</p>
<p><code>(set-model-fit-refine-rotate-translate-zone-label &quot;Move Zone&quot;)</code>
</p>
<p>and
</p>
<p><code>(set-model-fit-refine-place-atom-at-pointer &quot;Add Atom&quot;)</code>
</p>


<hr>
<a name="Map_002dRelated-Features"></a>
<div class="header">
<p>
Next: <a href="#Validation" accesskey="n" rel="next">Validation</a>, Previous: <a href="#Modelling-and-Building" accesskey="p" rel="previous">Modelling and Building</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map_002dRelated-Features-1"></a>
<h2 class="chapter">6 Map-Related Features</h2>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Maps-in-General" accesskey="1">Maps in General</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Create-a-Map" accesskey="2">Create a Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Contouring" accesskey="3">Map Contouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Extent" accesskey="4">Map Extent</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Contour-_0060_0060Scrolling_0027_0027-Limits" accesskey="5">Map Contour ``Scrolling'' Limits</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Line-Width" accesskey="6">Map Line Width</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-colouring" accesskey="7">Map colouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Difference-Map-Colouring" accesskey="8">Difference Map Colouring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Make-a-Difference-Map" accesskey="9">Make a Difference Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Make-an-Averaged-Map">Make an Averaged Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Sampling">Map Sampling</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Dragged-Map">Dragged Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Dynamic-Map-Sampling-and-Display-Size">Dynamic Map Sampling and Display Size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Skeletonization">Skeletonization</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Sharpening">Map Sharpening</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pattersons">Pattersons</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Re_002dInterpolation">Map Re-Interpolation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#masks">masks</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Trimming-Atoms">Trimming Atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Transformation">Map Transformation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Export-Map">Export Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Maps-in-General"></a>
<div class="header">
<p>
Next: <a href="#Create-a-Map" accesskey="n" rel="next">Create a Map</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Maps-in-General-1"></a>
<h3 class="section">6.1 Maps in General</h3>
<p>Maps are &ldquo;infinite,&rdquo; not limited to pre-calculated volume (the
&ldquo;Everywhere You Click - There Is Electron Density&rdquo;
(EYC-TIED)
<a name="index-EYC_002dTIED"></a>
paradigm) symmetry-related electron
density is generated automatically. Maps are easily re-contoured.
Simply use the scroll wheel on you mouse to alter the contour level
(or -/+ on the keyboard).
<a name="index-change-contour-level"></a>
</p> 
<p>Maps follow the molecule.  As you recentre or move about the crystal,
the map quickly follows.  If your computer is not up to re-contouring
all the maps for every frame, then use <code>Draw -&gt;
  Dragged Map&hellip;</code> to turn off this feature.
</p>

<a name="Map-Reading-Bug"></a>
<h4 class="subsection">6.1.1 Map Reading Bug</h4>
<p>Unfortunately, there is a bug in map-reading.  If the map is not a
bona-fide CCP4 map <a name="DOCF94" href="#FOOT94"><sup>94</sup></a>, then coot will crash.  Sorry.  A fix is in the
works but &ldquo;it&rsquo;s complicated&rdquo;.  That&rsquo;s why maps are limited to the
extension &quot;.ext&quot; and &quot;.map&quot;, to make it less likely a non-CCP4 map is
read.
</p>
<hr>
<a name="Create-a-Map"></a>
<div class="header">
<p>
Next: <a href="#Map-Contouring" accesskey="n" rel="next">Map Contouring</a>, Previous: <a href="#Maps-in-General" accesskey="p" rel="previous">Maps in General</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Create-a-Map-1"></a>
<h3 class="section">6.2 Create a Map</h3>
<p>From MTZ, mmCIF and .phs
<a name="index-PHS-data-format"></a>
data use <code>File -&gt; Open MTZ, CIF or phs&hellip;</code>. You can then
choose the MTZ columns for the Fourier synthesis.  The button &ldquo;Expert
mode&rdquo; also adds to the options any anomalous columns you may have in
the MTZ file (a -90 degree phase shift will be applied).  It also
provides the option to apply resolution limits.
</p>
<p>From a CCP4
map use <code>File -&gt; Read Map</code>.  After being
generated/read, the map is immediately contoured and centred on the
current rotation centre.
</p>

<a name="Auto_002dread-MTZ-file"></a>
<h4 class="subsection">6.2.1 Auto-read MTZ file</h4>
<p>This function allows Coot to read an MTZ file and make a map directly
(without going through the column selection procedure).  The default
column labels for auto-reading are &quot;FWT&quot; and &quot;PHWT&quot; for the 2Fo-Fc-style
map, &quot;DELFWT&quot; and &quot;PHDELWT&quot; for the difference map.  You can change the
column labels that Coot uses for auto-reading - here is an example of
how to do that:
</p>
<p><code>(set-auto-read-column-labels &quot;2FOFCWT&quot; &quot;PHIWT&quot; 0)</code>
<code>(set-auto-read-column-labels &quot;FOFCWT&quot; &quot;DELPHIWT&quot; 1)</code>
</p>
<p>By default the difference map is created in auto-reading the MTZ file.
If you don&rsquo;t want a difference map, you can use the function:
</p>
<p><code>(set-auto-read-do-difference-map-too 0)</code>
</p>
<a name="Reading-CIF-data"></a>
<h4 class="subsection">6.2.2 Reading CIF data</h4>
<p>There are several maps that can be generated from CIF files that
contain observed Fs, calculated Fs and calculated phases:
</p>
<ul>
<li> <code>(read-cif-data-with-phases-fo-alpha-calc
    <em>cif-file-name</em>)</code> Calculate an atom map using <em>F_obs</em> and
    <em>\alpha_calc</em>
</li><li> <code>(read-cif-data-with-phases-2fo-fc <em>cif-file-name</em>)</code>
 Calculate an atom map using <em>F_obs</em>, <em>F_calc</em> and
  <em>\alpha_calc</em>
</li><li> <code>(read-cif-data-with-phases-fo-fc <em>cif-file-name</em>)</code>
 Calculate an difference map using <em>F_obs</em>, <em>F_calc</em> and
  <em>\alpha_calc</em>.
</li></ul>

<a name="Reading-PHS-data"></a>
<h4 class="subsection">6.2.3 Reading PHS data</h4>
<a name="index-PHS-data"></a>

<p>There are 2 ways to read data by scripting:
</p>
<p><code>(read-phs-and-make-map-using-cell-symm <em>phs-file-name space-group-name a b c alpha beta gamma</em>)</code>
</p>
<p><code>(read-pdb-and-make-map-with-reso-limits <em>imol-previous phs-file-name reso-limit-low reso-limit-high</em>)</code>
</p>
<p>The first specifies the cell explicitly, and <code><em>alpha, beta</em></code>
and <code><em>gamma</em></code> are specified in degrees. 
</p>
<p>The second form allows the specification of resolution limits and takes
the cell and symmetry from a previous molecule (typically a pdb file).
</p>
<hr>
<a name="Map-Contouring"></a>
<div class="header">
<p>
Next: <a href="#Map-Extent" accesskey="n" rel="next">Map Extent</a>, Previous: <a href="#Create-a-Map" accesskey="p" rel="previous">Create a Map</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Contouring-1"></a>
<h3 class="section">6.3 Map Contouring</h3>
<a name="index-contouring_002c-map-1"></a>
<p>Maps can be re-contoured using the middle-mouse
scroll-wheel (buttons 4 and 5 in X Window System(TM)
terminology).  Scrolling the mouse wheel will change the map contour
level and the map it redrawn.  If you have several maps displayed then
the map that has its contour level changed can be set using
<code>HID -&gt; Scrollwheel -&gt; Attach scroll-wheel
  to which map?</code>.  If there is only one map displayed, then that is
the map that has its contour level changed (no matter what the
scroll-wheel is attached to in the menu).  The level of the electron
density is displayed in the top right hand corner of the OpenGL canvas.
</p>
<p>Use keyboard <tt class="key">+</tt> or <tt class="key">-</tt> to change the contour level if you don&rsquo;t
have a scroll-wheel <a name="DOCF95" href="#FOOT95"><sup>95</sup></a>.
</p>
<p>If you are creating your map from an MTZ file, you can choose to click
on the &ldquo;is difference map&rdquo;
<a name="index-difference-map"></a>
button on the Column
Label selection widget (after a data set filename has been selected)
then this map will be displayed in 2 colours corresponding to + and -
the map contour level.
</p>
<p>If you read in a map and it is a difference map then there is
a checkbutton to tell Coot that.
</p>
<p>If you want to tell Coot that a map is a difference
map
<a name="index-difference-map-colours"></a>
after it has been read, use:
</p>
<p><code>(set-map-is-difference-map <em>imol</em>)</code>
</p>
<p>where <code><em>imol</em></code> is the molecule number.
</p>
<p>By default the change of the contour level is determined from the sigma
of the map.  You can change this in the map properties dialog or by
using the scripting function:
</p>
<p><code>(set-contour-by-sigma-step-by-mol <em>step on/off? imol</em>)</code>
</p>
<p>where 
</p>
<p><code><em>step</em></code> is the difference in sigma from one level to the next
(typically 0.2)
</p>
<p><code><em>on/off?</em></code> is either 0 (sigma stepping off) or 1 (sigma stepping on)
</p>
<p>By default the map radius <a name="DOCF96" href="#FOOT96"><sup>96</sup></a> is 10&Aring;.
The default increment to the electron density depends on whether or
not this is a difference map (0.05 <em>e^-/\AA^3</em> for a &ldquo;2Fo-Fc&rdquo;
style map and 0.005 <em>e^-/\AA^3</em> for a difference map).  You can
change these using <code>Edit -&gt; Map Parameters</code> or by
using the &ldquo;Properties&rdquo; button of a particular map in the Display
Control (Display Manager) window.
</p>
<hr>
<a name="Map-Extent"></a>
<div class="header">
<p>
Next: <a href="#Map-Contour-_0060_0060Scrolling_0027_0027-Limits" accesskey="n" rel="next">Map Contour ``Scrolling'' Limits</a>, Previous: <a href="#Map-Contouring" accesskey="p" rel="previous">Map Contouring</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Extent-1"></a>
<h3 class="section">6.4 Map Extent</h3>
<a name="index-map-extent"></a>
<a name="index-map-box-radius"></a>

<p>The extent of the map can be set using the GUI (<code>Edit -&gt; Map
Parameters -&gt; Map Radius</code>) or by using the scripting function,
<em>e.g.</em>:
</p>
<p><code>(set-map-radius 13.2)</code>
</p>
<hr>
<a name="Map-Contour-_0060_0060Scrolling_0027_0027-Limits"></a>
<div class="header">
<p>
Next: <a href="#Map-Line-Width" accesskey="n" rel="next">Map Line Width</a>, Previous: <a href="#Map-Extent" accesskey="p" rel="previous">Map Extent</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Contour-_0060_0060Scrolling_0027_0027-Limits-1"></a>
<h3 class="section">6.5 Map Contour &ldquo;Scrolling&rdquo; Limits</h3>
<p>Usually one doesn&rsquo;t want to look at 
<a name="index-negative-contour-levels"></a>
negative contour levels of a map<a name="DOCF97" href="#FOOT97"><sup>97</sup></a>, so
Coot has by default a limit that stops the contour level going beyond
(less than) 0.  To remove the limit:
</p>
<p><code>(set-stop-scroll-iso-map 0)</code> for a 2Fo-Fc style map
</p>
<p><code>(set-stop-scroll-diff-map 0)</code> for a difference map
</p>
<p>To set the limits to negative (<em>e.g.</em> -0.6) levels:
</p>
<p><code>(set-stop-scroll-iso-map-level -0.6)</code>
</p>
<p>and similarly: 
</p>
<p><code>(set-stop-scroll-diff-map-level -0.6)</code>
</p>
<p>where the level is specified in <em>e^-/\AA^3</em>.
</p>
<hr>
<a name="Map-Line-Width"></a>
<div class="header">
<p>
Next: <a href="#Map-colouring" accesskey="n" rel="next">Map colouring</a>, Previous: <a href="#Map-Contour-_0060_0060Scrolling_0027_0027-Limits" accesskey="p" rel="previous">Map Contour ``Scrolling'' Limits</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Line-Width-1"></a>
<h3 class="section">6.6 Map Line Width</h3>
<a name="index-map-line-width"></a>
<a name="index-density-line-thickness"></a>
<a name="index-thickness-of-density-lines"></a>
<p>The width of the lines that describe the density can
be changed like this:
</p>
<p><code>(set-map-line-width 2)</code>
</p>
<p>The default line width is 1.
</p>
<hr>
<a name="Map-colouring"></a>
<div class="header">
<p>
Next: <a href="#Difference-Map-Colouring" accesskey="n" rel="next">Difference Map Colouring</a>, Previous: <a href="#Map-Line-Width" accesskey="p" rel="previous">Map Line Width</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="g_t_0060_0060Dynamic_0027_0027-Map-colouring"></a>
<h3 class="section">6.7 &ldquo;Dynamic&rdquo; Map colouring</h3>
<a name="index-colouring_002c-map"></a>
<p>By default, maps get coloured according to
their molecule number.  The starting colour (<em>i.e.</em> for molecule
0) is blue.  The colour of a map can be changed by <code>Edit
  -&gt; Map Colour..</code>. The map colour gets updated as you
change the value in the colour selector <a name="DOCF98" href="#FOOT98"><sup>98</sup></a>.  Use &ldquo;OK&rdquo; to fix that colour.
</p>
<p>As subsequent maps are read, they are coloured by rotation round a
colour wheel.  The default colour map step is 31 degrees.  You can
change this using:
</p>
<p><code>(set-colour-map-rotation-for-map <em>step</em>)</code>
</p>
<hr>
<a name="Difference-Map-Colouring"></a>
<div class="header">
<p>
Next: <a href="#Make-a-Difference-Map" accesskey="n" rel="next">Make a Difference Map</a>, Previous: <a href="#Map-colouring" accesskey="p" rel="previous">Map colouring</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Difference-Map-Colouring-1"></a>
<h3 class="section">6.8 Difference Map Colouring</h3>
<p>For some strange reason, some crystallographers <a name="DOCF99" href="#FOOT99"><sup>99</sup></a> like to have their difference maps coloured with red
as positive and green as negative, this option is for them:
</p>
<p><code>(set-swap-difference-map-colours 1)</code>
</p>
<p>This option will allow the &ldquo;blue is positive, red is negative&rdquo; colour
scheme on &ldquo;Edit -&gt; Map Colour&rdquo;.
</p>

<hr>
<a name="Make-a-Difference-Map"></a>
<div class="header">
<p>
Next: <a href="#Make-an-Averaged-Map" accesskey="n" rel="next">Make an Averaged Map</a>, Previous: <a href="#Difference-Map-Colouring" accesskey="p" rel="previous">Difference Map Colouring</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Make-a-Difference-Map-1"></a>
<h3 class="section">6.9 Make a Difference Map</h3>
<a name="index-difference-map-1"></a>

<p>Using the &ldquo;Make a Difference Map&rdquo; function in the Extensions menu, one
can make a difference from two arbitrary maps.  The maps need not be on
the same griding, or in the same space group even.  The resulting map
will be on the same griding and space group as the &ldquo;Reference&rdquo; map.
</p>
<hr>
<a name="Make-an-Averaged-Map"></a>
<div class="header">
<p>
Next: <a href="#Map-Sampling" accesskey="n" rel="next">Map Sampling</a>, Previous: <a href="#Make-a-Difference-Map" accesskey="p" rel="previous">Make a Difference Map</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Make-an-Averaged-Map-1"></a>
<h3 class="section">6.10 Make an Averaged Map</h3>
<a name="index-average-map"></a>

<p>There is a scripting interface to the generation of map averages.  As
above, the maps need not be on the same grid or in the same space group.
The resulting map will have the same gridding and space group as the
first map in the list.  Typical usage:
</p>
<p><code>(average-map '((1 1.0) (2 1.0))))</code>
</p>
<p>The argument to <code>(average-map</code> is a list of lists, each list
element is a list of the map number and a weighting factor (1.0 in this
case).
</p>

<hr>
<a name="Map-Sampling"></a>
<div class="header">
<p>
Next: <a href="#Dragged-Map" accesskey="n" rel="next">Dragged Map</a>, Previous: <a href="#Make-an-Averaged-Map" accesskey="p" rel="previous">Make an Averaged Map</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Sampling-1"></a>
<h3 class="section">6.11 Map Sampling</h3>
<p>By default, the Shannon sampling factor is the conventional 1.5.  Use
larger values (<code>Edit -&gt; Map Parameters -&gt;
  Sampling Rate</code>) for smoother maps <a name="DOCF100" href="#FOOT100"><sup>100</sup></a>.
</p>
<p>This value can be set by the scripting command
</p>
<p><code>(set-map-sampling-rate 2.5)</code>
</p>
<hr>
<a name="Dragged-Map"></a>
<div class="header">
<p>
Next: <a href="#Dynamic-Map-Sampling-and-Display-Size" accesskey="n" rel="next">Dynamic Map Sampling and Display Size</a>, Previous: <a href="#Map-Sampling" accesskey="p" rel="previous">Map Sampling</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Dragged-Map-1"></a>
<h3 class="section">6.12 Dragged Map</h3>
<p>By default, the map is re-contoured at every frame during a drag (Ctrl
Left-mouse).  Sometimes this can be annoyingly slow and jerky so it is
possible to turn it off: <code>Draw -&gt; Dragged Map
  -&gt; No</code>.
</p>
<p>To change this by scripting:
</p>
<p><code>(set-active-map-drag-flag 0)</code>
</p>
<hr>
<a name="Dynamic-Map-Sampling-and-Display-Size"></a>
<div class="header">
<p>
Next: <a href="#Skeletonization" accesskey="n" rel="next">Skeletonization</a>, Previous: <a href="#Dragged-Map" accesskey="p" rel="previous">Dragged Map</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Dynamic-Map-Sampling-and-Display-Size-1"></a>
<h3 class="section">6.13 Dynamic Map Sampling and Display Size</h3>
<p>If activated (<code>Edit -&gt; Map Parameters -&gt;
  Dynamic Map Sampling</code>) the map will be re-sampled on a more coarse grid
when the view is zoomed out.  If &ldquo;Display Size&rdquo; is also activated,
the box of electron density will be increased in size also.  In this
way, you can see electron density for
<a name="index-big-maps"></a>
big maps (many
unit cells) and the graphics still remain rotatable.
</p>
<p>If you want to have these functions active for all maps, add the
following to your initialization file <a href="#Scheme">Scheme</a>:
</p>
<p><code>(set-dynamic-map-sampling-on)</code>
<code>(set-dynamic-map-size-display-on)</code>
</p>
<hr>
<a name="Skeletonization"></a>
<div class="header">
<p>
Next: <a href="#Map-Sharpening" accesskey="n" rel="next">Map Sharpening</a>, Previous: <a href="#Dynamic-Map-Sampling-and-Display-Size" accesskey="p" rel="previous">Dynamic Map Sampling and Display Size</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Skeletonization-1"></a>
<h3 class="section">6.14 Skeletonization</h3>
<a name="index-skeletonization"></a>
<a name="index-bones"></a>
<p>The skeleton (also known as
&ldquo;Bones&rdquo; <a name="DOCF101" href="#FOOT101"><sup>101</sup></a>) can be displayed for any map.  A map can be skeletonized using
<code>Calculate -&gt; Map Skeleton&hellip;</code>.  Use the option
menu to choose the map and click &ldquo;On&rdquo; then &ldquo;OK&rdquo; to the generate
the map (the skeleton is off by default).
</p>
<p>The level of the skeleton can be changed by using <code>Edit
  -&gt; Skeleton Parameters&hellip; -&gt;
  Skeletonization Level&hellip;</code> and corresponds to the electron density
level in the map.  By default this value is 1.2 map standard
deviations.  The amount of map can be changed using <code>Edit
  -&gt; Skeleton Parameters&hellip; -&gt; Skeleton Box
  Radius&hellip;</code><a name="DOCF102" href="#FOOT102"><sup>102</sup></a>.  The units are in 
&Aring;ngstr&ouml;ms, with 40 as the default value.
</p>
<p>The skeleton is often recalculated as the screen centre changes - but
not always since it can be an irritatingly slow calculation.
<a name="index-skeleton-updating"></a>
If you want to force a regeneration of
the displayed skeleton, simply centre on an atom (using the middle
mouse button) or press the <tt class="key">S</tt> key.
</p>
<hr>
<a name="Map-Sharpening"></a>
<div class="header">
<p>
Next: <a href="#Pattersons" accesskey="n" rel="next">Pattersons</a>, Previous: <a href="#Skeletonization" accesskey="p" rel="previous">Skeletonization</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Sharpening-1"></a>
<h3 class="section">6.15 Map Sharpening</h3>
<a name="index-sharpening_002c-map"></a>

<p>It can be educational (even useful at lower resolutions)
to sharpen or blur a map.  This can be achieved with the sharpening
tool <code>Calculate -&gt; Map Sharpening&hellip;</code>.  By default, the
maximum and minimum sharpness is +/- <em>200&Aring;^2</em>, this can be
changed (in this case to 300) using:
</p>
<p><code>(set-map-sharpening-scale-limit 300)</code>
</p>
<p>This currently only works on maps created by reading an MTZ (or other)
reflection data file.
</p>
<hr>
<a name="Pattersons"></a>
<div class="header">
<p>
Next: <a href="#Map-Re_002dInterpolation" accesskey="n" rel="next">Map Re-Interpolation</a>, Previous: <a href="#Map-Sharpening" accesskey="p" rel="previous">Map Sharpening</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Pattersons-1"></a>
<h3 class="section">6.16 Pattersons</h3>
<a name="index-Patterson"></a>

<p>Pattersons can be generated using the <code>make-and-draw-patterson</code>
function.  Example usage:
</p>
<p><code>(make-and-draw-patterson <em>mtz-file-name f-col sig-f-col</em>)</code>
</p>
<p>where <code><em>use-weights-flag</em></code> is either <code>0</code> or <code>1</code>.
</p>
<p><em>e.g.</em>
</p>
<p><code>(make-and-draw-patterson &quot;native.mtz&quot; &quot;FP_nat&quot; &quot;SIGFP_nat&quot;)</code>
</p>
<hr>
<a name="Map-Re_002dInterpolation"></a>
<div class="header">
<p>
Next: <a href="#masks" accesskey="n" rel="next">masks</a>, Previous: <a href="#Pattersons" accesskey="p" rel="previous">Pattersons</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Re_002dInterpolation-1"></a>
<h3 class="section">6.17 Map Re-Interpolation</h3>
<a name="index-map-re_002dInterpolation"></a>

<p>Maps can be re-interpolated to match a reference map.
</p>
<p><code>(reinterp-map <em>map-no reference-map-no</em>)</code>
</p>
<p>will create a copy of  <code><em>map-no</em></code> in the same cell, spacegroup and grid spacing as  
the <code><em>reference-map-no</em></code> map.
</p>

<hr>
<a name="masks"></a>
<div class="header">
<p>
Next: <a href="#Trimming-Atoms" accesskey="n" rel="next">Trimming Atoms</a>, Previous: <a href="#Map-Re_002dInterpolation" accesskey="p" rel="previous">Map Re-Interpolation</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Masks-1"></a>
<h3 class="section">6.18 Masks</h3>
<a name="index-masks"></a>
<p>A map can be masked by a set of coordinates. Use the
scripting function: 
</p>
<p><code>(mask-map-by-molecule <em>imol-map</em> <em>imol-model</em> <em>invert-mask?</em>)</code>
</p>
<p>If <code><em>invert-mask?</em></code> is 0, this will create a new map that has
density only where there are no (close) coordinates.  If
<code><em>invert-mask?</em></code> is 1 then the map density values will be set
to zero everywhere <em>except</em> close to the atoms of molecule number
<em>imol-model</em>.
</p>
<p>The radius of the mask around each atom is 2.0&Aring; by default.  You can
change this using:
</p>
<p><code>(set-map-mask-atom-radius <em>radius</em>)</code>
</p>
<p>There is a GUI interface to Map Masking under the Extensions menu. 
</p>
<a name="Example"></a>
<h4 class="subsection">6.18.1 Example</h4>
<p>If one wanted to show just the density around a ligand:
</p>
<ol>
<li> Make a pdb file the contains just the ligand and read it in to
  Coot - let&rsquo;s say it is molecule 1 and the ligand is residue 3 of
  chain &ldquo;L&rdquo;.
</li><li> Get a map that covers the ligand (<em>e.g.</em> from refmac).
  Let&rsquo;s say this map is molecule number 2.
</li><li> Mask the map:

<p><code>(mask-map-by-molecule 2 1 1)</code>
</p>
<p>This creates a new map.  Turn the other maps off, leaving only the
masked map.
</p>
</li></ol>

<p>To get a nice rendered image, press F8 (see Section <a href="#Raster3D">Raster3D</a>).
</p>
<hr>
<a name="Trimming-Atoms"></a>
<div class="header">
<p>
Next: <a href="#Map-Transformation" accesskey="n" rel="next">Map Transformation</a>, Previous: <a href="#masks" accesskey="p" rel="previous">masks</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Trimming"></a>
<h3 class="section">6.19 Trimming</h3>
<a name="index-trimming-atoms"></a>
<p>If you want to remove all the atoms <a name="DOCF103" href="#FOOT103"><sup>103</sup></a> that lie &ldquo;outside the map&rdquo; (<em>i.e.</em> in low density) you can use
</p>
<p><code>(trim-molecule-by-map <em>imol-coords imol-map density-level delete/zero-occ?</em>)</code>
</p>
<p>where <code><em>delete/zero-occ?</em></code> is <code>0</code> to remove the atoms and
<code>1</code> to set their occupancy to zero.
</p>
<p>There is a GUI interface for this feature under the &ldquo;Extensions&rdquo; menu
item.
</p>
<hr>
<a name="Map-Transformation"></a>
<div class="header">
<p>
Next: <a href="#Export-Map" accesskey="n" rel="next">Export Map</a>, Previous: <a href="#Trimming-Atoms" accesskey="p" rel="previous">Trimming Atoms</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Transformation-1"></a>
<h3 class="section">6.20 Map Transformation</h3>
<a name="index-map-transformation"></a>

<p>If you want to transform a map, you can do it thusly:
</p>
<p><code>(transform-map <em>imol</em> <em>rotation-matrix</em> <em>trans</em> <em>point radius</em>)</code>
</p>
<p>where:
</p>
<p><code><em>rotation-matrix</em></code> is a 9-membered list of numbers for an orthogonal rotation matrix.
</p>
<p><code><em>trans</em></code> is a 3-membered list of numbers (distances in &Aring;ngst&ouml;ms).
</p>
<p><code><em>point</em></code> is a 3-membered list of numbers (centre point in &Aring;ngst&ouml;ms).
</p>
<p><code><em>radius</em></code> is a single number (also in &Aring;ngst&ouml;ms).
</p>

<p>This applies the rotation <code><em>rotation-matrix</em></code> and a
translation <code><em>trans</em></code> to a map fragment, so that when the
transformation is applied the centre of the new map is at
<code><em>point</em></code>.  
</p>
<p>Example usage:
</p>
<p><code>(transform-map 2 '(1 0 0 0 1 0 0 0 1) '(0 0 1) (rotation-centre) 10)</code>
</p>
<p>which transforms map number 2 by a translation of 1&Aring; along the Z
axis, centred at the screen centre for 10&Aring; around that centre.
</p>
<p>Here&rsquo;s a more real-world example:
</p>
<p>Let&rsquo;s say we want to tranform the density over the &ldquo;B&rdquo; molecule to a
position over the &ldquo;A&rdquo; molecule. First we do a LSQ transformation to
get the rotation and translation that moves the &ldquo;B&rdquo; coordinates over
the &ldquo;A&rdquo; coordinates:
</p>
<p>In the terminal output we get:
</p>
<pre class="verbatim">|    0.9707,    0.2351,   0.05033|
|  -0.04676,      0.39,   -0.9196|
|   -0.2358,    0.8903,    0.3896|
(    -33.34,     21.14,     18.82)
</pre>
<p>The centre of the &ldquo;A&rdquo; molecule is at (58.456, 5.65, 11.108).  So we do:
</p>
<p><code>(transform-map 3 (list 0.9707 0.2351 0.05033 -0.04676 0.39 -0.9196
-0.2358 0.8903 0.3896) (list -33.34 21.14 18.82) (list 58.456 5.65
11.108) 8)</code>
</p>
<p>Which creates a map over the middle of the &ldquo;A&rdquo; molecule.  Note that
using a too high <code><em>radius</em></code> can cause overlap problems, so try
with a small <code><em>radius</em></code> (<em>e.g.</em> 5.0) if the resulting map
looks problematic.
</p>
<p>Alternatively, instead of typing the whole matrix, you can use a
coordinates least-squares fit to generate the matrix for you.
<code>(transform-map-using-lsq-matrix)</code> does just that.
</p>
<p>Heres how to use it:
</p>
<p><code>(transform-map-using-lsq-matrix <em>imol-ref ref-chain
ref-resno-start ref-resno-end imol-mov mov-chain mov-resno-start
mov-resno-end imol-map about-pt radius</em>)</code>
</p>

<p>Hopefully the arguments are self explanatory (<code><em>ref</em></code> refers
to the reference molecule, of course and <code><em>about-pt</em></code> is a
3-number list such as is returned by <code>(rotation-centre)</code>).
</p>
<p>We can now export that map, if we want.
</p>

<hr>
<a name="Export-Map"></a>
<div class="header">
<p>
Previous: <a href="#Map-Transformation" accesskey="p" rel="previous">Map Transformation</a>, Up: <a href="#Map_002dRelated-Features" accesskey="u" rel="up">Map-Related Features</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Export-Map-1"></a>
<h3 class="section">6.21 Export Map</h3>
<a name="index-write-map"></a>
<a name="index-export-map"></a>

<p>You can write out a map from Coot (e.g. one from NCS averaging, or
masking or general transformation) using the export map function:
</p>
<p><code>(export-map <em>imol</em> <em>filename</em>)</code>
</p>
<p><em>e.g.</em>
</p>
<p><code>(export-map 4 &quot;ncs-averaged.map&quot;)</code>
</p>
<hr>
<a name="Validation"></a>
<div class="header">
<p>
Next: <a href="#Representation" accesskey="n" rel="next">Representation</a>, Previous: <a href="#Map_002dRelated-Features" accesskey="p" rel="previous">Map-Related Features</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Validation-1"></a>
<h2 class="chapter">7 Validation</h2>

<p>The validation functions are still being added to from time to time.  In
future there will be more functions, particularly those that will
interface to other programs.
</p>
<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Ramachandran-Plots" accesskey="1">Ramachandran Plots</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Geometry-Analysis" accesskey="2">Geometry Analysis</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Chiral-Volumes" accesskey="3">Chiral Volumes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sec_005fblobs" accesskey="4">sec_blobs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Difference-Map-Peaks" accesskey="5">Difference Map Peaks</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Check-Waters-by-Difference-Map" accesskey="6">Check Waters by Difference Map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Molprobity-Tools-Interface" accesskey="7">Molprobity Tools Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#GLN-and-ASN-B_002dfactor-Outliers" accesskey="8">GLN and ASN B-factor Outliers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Validation-Graphs" accesskey="9">Validation Graphs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Ramachandran-Plots"></a>
<div class="header">
<p>
Next: <a href="#Geometry-Analysis" accesskey="n" rel="next">Geometry Analysis</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Ramachandran-Plots-1"></a>
<h3 class="section">7.1 Ramachandran Plots</h3>
<a name="index-Ramachandran-plot"></a>

<p>Ramachandran plots are &ldquo;dynamic&rdquo;.  When you edit the molecule
(<em>i.e.</em> move the coordinates of some of atoms) the Ramachandran
plot gets updated to reflect those changes.  Also the underlying
<em>\phi/\psi</em> probability density changes according to the selected
residue type (<em>i.e.</em> the residue under the mouse in the plot).
There are 3 different residue types: GLY, PRO, and not-GLY-or-PRO
<a name="DOCF104" href="#FOOT104"><sup>104</sup></a>.
</p>
<p>When you mouse over a representation of a residue (a little square or
triangle <a name="DOCF105" href="#FOOT105"><sup>105</sup></a>) the residue label pops up.  The
residue is &ldquo;active&rdquo; <em>i.e.</em> it can be clicked.  The &ldquo;graphics&rdquo;
view changes so that the C<em>\alpha</em> of the selected residue is centred.
In the Ramachandran plot window, the current residue is highlighted by
a green square.
</p>
<p>The underlying distributions are taken from the Richardson&rsquo;s Top500
structures <a href="http://kinemage.biochem.duke.edu/databases/top500.php">http://kinemage.biochem.duke.edu/databases/top500.php</a>.
</p>
<p>The probability levels for acceptable (yellow) and preferred (red) are
0.2% and 2% respectively.
</p>
<p>You can change the contour levels:
</p>
<p><code>(set-ramachandran-plot-contour-levels 0.025 0.003)</code>
</p>
<p>You can change the &ldquo;blocksize&rdquo; (the default is 10 degrees) of the
contours using
</p>
<p><code>(set-ramachandran-plot-background-block-size 5)</code>
</p>
<p>These comes into effect when a new plot is created (it doesn&rsquo;t change
plots currently displayed).
</p>

<hr>
<a name="Geometry-Analysis"></a>
<div class="header">
<p>
Next: <a href="#Chiral-Volumes" accesskey="n" rel="next">Chiral Volumes</a>, Previous: <a href="#Ramachandran-Plots" accesskey="p" rel="previous">Ramachandran Plots</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Geometry-Analysis-1"></a>
<h3 class="section">7.2 Geometry Analysis</h3>

<p>A restraints-based geometry analysis of the molecule.  The distortion
is weighted by atom occupancy. The distortion of the geometry due to
links is shared between the contributing residues.
</p>
<p>Note that only the first model of a multi-model molecule is analysed.
</p>

<hr>
<a name="Chiral-Volumes"></a>
<div class="header">
<p>
Next: <a href="#sec_005fblobs" accesskey="n" rel="next">sec_blobs</a>, Previous: <a href="#Geometry-Analysis" accesskey="p" rel="previous">Geometry Analysis</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Chiral-Volumes-1"></a>
<h3 class="section">7.3 Chiral Volumes</h3>
<p>The dictionary is used to identify the chiral atoms of each of the
model&rsquo;s residues.  A clickable list is created of atoms whose chiral
volume in the model is of a different sign to that in the dictionary.
</p>
<p>During refinement and regularization, Coot will pop-up dialogs warning
about chiral volume errors - if you have them.  This can be annoying
<a name="DOCF106" href="#FOOT106"><sup>106</sup></a>. You can inhibit this
dialog like this:
</p>
<p><code>(set-show-chiral-volume-errors-dialog 0)</code>
</p>

<a name="Fixing-Chiral-Volume-Errors"></a>
<h4 class="subsection">7.3.1 Fixing Chiral Volume Errors</h4>

<p>There are two obvious ways:
</p>
<p>1) mutate and auto-fit rotamer (mutate it to the residue type that it is)
</p>
<p>2) RS Refine the residue and invert the chiral centre by pulling an
atom. Usually you can pull the CA to the other side of the plane made
by the chiral neighbouring atoms (using ctrl left-click).  Sometimes
giving the CB a good old tweak is the easier way.
</p>
<p>Inverting the CB of THR is easier, just move the OG so that the plane
of the neighbours is on the other side of the CB (again with ctrl
left-click).
</p>

<hr>
<a name="sec_005fblobs"></a>
<div class="header">
<p>
Next: <a href="#Difference-Map-Peaks" accesskey="n" rel="next">Difference Map Peaks</a>, Previous: <a href="#Chiral-Volumes" accesskey="p" rel="previous">Chiral Volumes</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Blobs_003a-a_002ek_002ea_002e-Unmodelled-density"></a>
<h3 class="section">7.4 Blobs: a.k.a. Unmodelled density</h3>
<p>This is an interface to the Blobs
<a name="index-blobs"></a>
<a name="index-unmodelled-density"></a>
dialog.  A map and a set of coordinates that model the
protein are required.
</p>
<p>A blob is region of relatively high residual election density that
cannot be explained by a simple water.
<a name="index-unexplained-density"></a>
So, for example, sulfates, ligands, mis-placed sidechains or unbuilt
terminal residues might appear as blobs.  The blobs are in order, the
biggest  <a name="DOCF107" href="#FOOT107"><sup>107</sup></a> at the top.
</p>
<hr>
<a name="Difference-Map-Peaks"></a>
<div class="header">
<p>
Next: <a href="#Check-Waters-by-Difference-Map" accesskey="n" rel="next">Check Waters by Difference Map</a>, Previous: <a href="#sec_005fblobs" accesskey="p" rel="previous">sec_blobs</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Difference-Map-Peaks-1"></a>
<h3 class="section">7.5 Difference Map Peaks</h3>

<p>This is one of the fastest ways to validate a model and its data
(presuming that the difference map comes from a post-refinement
mFo-DFc map).  It highlights regions where the model and the data do
not agree.
</p>
<p>Lesser peaks within a certain distance (by default, 2.0&Aring;) of a
large peak are not shown.  This cuts down on the number of times one
is navigated to a particular region because of ripple or other noise
peaks around a central peak. 
</p>
<p>This value can be queried:
</p>
<p><code>(difference-map-peaks-max-closeness)</code>
</p>
<p>and adjusted:
</p>
<p><code>(set-difference-map-peaks-max-closeness 0.1)</code>
</p>

<hr>
<a name="Check-Waters-by-Difference-Map"></a>
<div class="header">
<p>
Next: <a href="#Molprobity-Tools-Interface" accesskey="n" rel="next">Molprobity Tools Interface</a>, Previous: <a href="#Difference-Map-Peaks" accesskey="p" rel="previous">Difference Map Peaks</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Check-Waters-by-Difference-Map-1"></a>
<h3 class="section">7.6 Check Waters by Difference Map</h3>

<p>Sometimes waters can be misplaced - taking the place of sidechains or
ligands or crystallization agents such as phosphate for
example <a name="DOCF108" href="#FOOT108"><sup>108</sup></a>.  In such cases the variance
of the difference map can be used to identify these problems.
</p>
<p>This function is also useful to check anomalous maps.  Often waters
are placed in density that is really a something else, perhaps a
cation, anion, sulphate or a ligand.  If such an atom diffracts
anomalously this can be identified and corrected.
</p>
<p>By default the waters with a map variance greater than 3.5<em>\sigma</em>
are listed.  One can be more rigorous by using a lower cut-off:
</p>
<p><code>(set-check-waters-by-difference-map-sigma-level 3.0)</code>
</p>
<p>The scripting interface is: 
</p>
<p><code>(check-waters-by-difference-map <em>imol-coords</em>
  <em>imol-diff-map</em>)</code>
</p>
<p>where <code><em>imol-coords</em></code> is the molecule number of the
coordinates that contain the waters to be checked
</p>
<p><code><em>imol-diff-map</em></code> is the molecule number of the difference
map (it must be a difference map, not an &ldquo;ordinary&rdquo; map).  This
difference map must have been calculated using the waters. So there is
no point in doing this check immediately after &ldquo;Find Waters&rdquo;.  You
will need to run Refmac or some other refinement first first
<a name="DOCF109" href="#FOOT109"><sup>109</sup></a>.
</p>
<hr>
<a name="Molprobity-Tools-Interface"></a>
<div class="header">
<p>
Next: <a href="#GLN-and-ASN-B_002dfactor-Outliers" accesskey="n" rel="next">GLN and ASN B-factor Outliers</a>, Previous: <a href="#Check-Waters-by-Difference-Map" accesskey="p" rel="previous">Check Waters by Difference Map</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Molprobity-Tools-Interface-1"></a>
<h3 class="section">7.7 Molprobity Tools Interface</h3>
<a name="index-Molprobity-Tools"></a>
<a name="index-probe"></a>
<a name="index-reduce"></a>

<p>The molprobity tools <samp>probe</samp> and <samp>reduce</samp> have been
interfaced into Coot (currently, the interface is not as slick as it
might be).  However, the tools are useful and can be used in the
following way:
</p>
<p>first we need to tell Coot where to find the relevant executables
(typically you would add the following lines to you <samp>~/.coot</samp>
file):
</p>
<p><code>(define *probe-command* &quot;/path/to/probe/executable&quot;)</code>
</p>
<p><code>(define *reduce-command* &quot;/path/to/reduce/executable&quot;)</code>
</p> 
<p>now the probe hydrogens and probe dots can be generated using
<code>Validate -&gt; Probe Clashes</code> (or in the Scripting Window):
</p>
<p><code>(probe <em>imol</em>)</code>
</p>
<p>where <code><em>imol</em></code> is the molecule number of coordinates to be
probed.  A new molecule with Hydrogens is created (by <samp>reduce</samp>) and
read in.
</p>
<p>By default Coot creates a new molecule for the molecule that now has
hydrogens.  To change this:
</p>
<p><code>(set! reduce-molecule-updates-current #t)</code>
</p>
<p>and that, as you can guess, replaces, rather than adds to the
&ldquo;probed&rdquo; molecule.
</p>
<p>This gives a &quot;static&quot; view of the molecule&rsquo;s interactions.
</p>
<p>To get a dynamic view (which is currently only enabled on rotating chi
angles) add these to your <samp>~/.coot</samp> file:
</p>
<p><code>(set-do-probe-dots-on-rotamers-and-chis 1)</code>
</p>
<p>To get a semi-static view (dots are regenerated in the region of zone
after a &quot;Real Space Refinement&quot;):
</p>
<p><code>(set-do-probe-dots-post-refine 1)</code>
</p>

<hr>
<a name="GLN-and-ASN-B_002dfactor-Outliers"></a>
<div class="header">
<p>
Next: <a href="#Validation-Graphs" accesskey="n" rel="next">Validation Graphs</a>, Previous: <a href="#Molprobity-Tools-Interface" accesskey="p" rel="previous">Molprobity Tools Interface</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="GLN-and-ASN-B_002dfactor-Outliers-1"></a>
<h3 class="section">7.8 GLN and ASN B-factor Outliers</h3>
<a name="index-GLN-and-ASN-B_002dfactor-Outliers"></a>

<p>It is often difficult to detect by eye the correct orientation of the
amino-carbonylo group of GLN and ASNs.  However, we can use (properly
refined) temperature factors to detect outliers.  We take the Z value as
half the difference between the B-factor of the NE2 and OE1 divided by
the standard deviation of the B-factors of the rest of the residue.  An
analysis of GLNs and ASNs of high resolutions structures indicates that
a Z value of greater than 2.25 indicates a potential (if not probable)
flip.  A &ldquo;Fix&rdquo; button is provided in the resultant dialog make this
easy to do.
</p>
<p>This analysis was added after discussions with Atsushi Nakagawa and so
is called &ldquo;Nakagawa&rsquo;s Bees&rdquo;.
</p>
<p>The analysis does not check residues with multiple conformations.
</p>
<hr>
<a name="Validation-Graphs"></a>
<div class="header">
<p>
Previous: <a href="#GLN-and-ASN-B_002dfactor-Outliers" accesskey="p" rel="previous">GLN and ASN B-factor Outliers</a>, Up: <a href="#Validation" accesskey="u" rel="up">Validation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Validation-Graphs-1"></a>
<h3 class="section">7.9 Validation Graphs</h3>
<a name="index-Validation-Graphs"></a>

<p>Coot provides several graphs that are useful for model validation (on
a residue by residue basis): residue density fit, geometry distortion,
temperature factor variance, peptide distortion and rotamer analysis.
</p>
<a name="Residue-Density-Fit"></a>
<h4 class="subsection">7.9.1 Residue Density Fit</h4>

<p>The density fit graph shows the density fit for residues.  The score
is the average electron density level at the atom centres of the atoms
in the residue.  The height of the blocks is inversely proportional to
the density average.
</p>
<p>The residue density fit is by default scaled to a map that is
calculated on the absolute scale.  Sometimes you might be using a map
with density levels considerably different to this, which makes the
residue density fit graph less useful.  To correct for this you can
use the scripting function:
</p>
<p><code>(set-residue-density-fit-scale-factor <em>factor</em>)</code>
</p>
<p>where <code><em>factor</em></code> would be <em>1/(4* rmd_{map})</em> (as a rule
of thumb).
</p>
<p><code>(residue-density-fit-scale-factor)</code> returns the current scale
factor (default 1.0).
</p>
<p>There is also a GUI to this: 
</p>
<p><code>Extensions -&gt; Refine&hellip; -&gt; Set Density Fit Graph Weight&hellip;</code>
</p>
<a name="Rotamer-Analysis"></a>
<h4 class="subsection">7.9.2 Rotamer Analysis</h4>
<p>Residue rotamers are scored according to the prior likelihood.  Note
that when CD1 and CD2 of a PHE residue are exchanged (simply a
nomenclature error) this can lead to large red blocks in the graph
(apparently due to very unlikely rotamers).  There are several other
residues that can have nomenclature errors like this.  To fix these
problems use 
</p>
<p><code>(fix-nomenclature-errors <em>imol</em>)</code>
</p>
<a name="Temperature-Factor-Variance"></a>
<h4 class="subsection">7.9.3 Temperature Factor Variance</h4>

<p>This idea is from Eleanor Dodson, who liked to use the standard
deviation of a residue&rsquo;s temperature factors to highlight regions of
questionable structure.
</p>
<p>Note that Hydrogens are ignored in this analysis.
</p>
<a name="Peptide-Omega-Angle-Distortion"></a>
<h4 class="subsection">7.9.4 Peptide Omega Angle Distortion</h4>

<p>Some variability of the <em>\omega</em> is to be expected in the peptide
bond.  But not too much. Anything more than 13 degrees is suspicicous.
Unexpected peptide bonds show up red by default.  If cis peptides
<em>are</em> to be expected, and should not marked as bad, then you can
tell this to Coot using:
</p>
<p><code>Edit -&gt; Preferences -&gt; Geometry -&gt; Cis-Peptides -&gt; No</code>
</p>








<hr>
<a name="Representation"></a>
<div class="header">
<p>
Next: <a href="#Hints-and-Usage-Tips" accesskey="n" rel="next">Hints and Usage Tips</a>, Previous: <a href="#Validation" accesskey="p" rel="previous">Validation</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Representation-1"></a>
<h2 class="chapter">8 Representation</h2>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Surfaces" accesskey="1">Surfaces</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Surfaces"></a>
<div class="header">
<p>
Up: <a href="#Representation" accesskey="u" rel="up">Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Surfaces-1"></a>
<h3 class="section">8.1 Surfaces</h3>

<p>Coot uses the surface code from Gruber and Noble (2004).
</p>
<p>Coot uses the partial charges of the atoms (the <var>partial_charge</var>
field in the <var>_chem_comp_atom</var> block) from the charge dictionary item
in the refmac (or other) cif dictionary.  However, partial charges are
only used under certain conditions
</p>
<p>1) the molecule consists of less than 100 atoms
</p>
<p>or 
</p>
<p>2) the number of atoms in the molecule that are hydrogens is at least
15% of the total number of atoms in the molecule
</p>
<p>If partial charges are not used, then the fall-back is to use charges
from side-chains charged at physiological pH (Arg, Lys, Asp, Glu).
</p>

<hr>
<a name="Hints-and-Usage-Tips"></a>
<div class="header">
<p>
Next: <a href="#Other-Programs" accesskey="n" rel="next">Other Programs</a>, Previous: <a href="#Representation" accesskey="p" rel="previous">Representation</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Hints-and-Usage-Tips-1"></a>
<h2 class="chapter">9 Hints and Usage Tips</h2>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Documentation" accesskey="1">Documentation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Low-Resolution" accesskey="2">Low Resolution</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Coot-Droppings" accesskey="3">Coot Droppings</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Clearing-Backups" accesskey="4">Clearing Backups</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Getting-out-of-_0060_0060Translate_0027_0027-Mode" accesskey="5">Getting out of ``Translate'' Mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Getting-out-of-_0060_0060Continuous-Rotation_0027_0027-Mode" accesskey="6">Getting out of ``Continuous Rotation'' Mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Label-Atom-Only-Mode" accesskey="7">Label Atom Only Mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Button-Labels" accesskey="8">Button Labels</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sec_005fpicking" accesskey="9">sec_picking</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Resizing-View">Resizing View</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scroll_002dwheel">Scroll-wheel</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Slow-Computer-Configuration">Slow Computer Configuration</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Documentation"></a>
<div class="header">
<p>
Next: <a href="#Low-Resolution" accesskey="n" rel="next">Low Resolution</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Documentation-1"></a>
<h3 class="section">9.1 Documentation</h3>

<p>This manual is on the web where it can be searched:
</p><ul>
<li> <a href="http://www.biop.ox.ac.uk/coot/doc/user-manual.html">http://www.biop.ox.ac.uk/coot/doc/user-manual.html</a> monolithic version
</li><li> <a href="http://www.biop.ox.ac.uk/coot/doc/chapters/user-manual_toc.html">http://www.biop.ox.ac.uk/coot/doc/chapters/user-manual_toc.html</a> which is 
        split into sections
</li></ul>

<p>In the Menu item &ldquo;About&rdquo;, under &ldquo;Online Docs URL...&rdquo; there is a
entry bar that can be used to search the Coot documentation via Google.
The results are returned as a web page in web browser. The browser type
can be specified as in this example:
</p>
<p><code>(set-browser-interface &quot;firefox&quot;)</code>
</p>
<p>Example usage can be found in <samp>xxx/share/coot/scheme/group-settings.scm</samp>
</p>

<hr>
<a name="Low-Resolution"></a>
<div class="header">
<p>
Next: <a href="#Coot-Droppings" accesskey="n" rel="next">Coot Droppings</a>, Previous: <a href="#Documentation" accesskey="p" rel="previous">Documentation</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Low-Resolution-1"></a>
<h3 class="section">9.2 Low Resolution</h3>

<p>Building structures using low resolution data is a pain.  We hope to
make it less of a pain in future, but there are some things that you can
do now.
</p>
<ul>
<li> [Add Planar Peptide Restraints] Add restraints via scripting command
</li><li> [Use Secondary Structure Restraints] where appropriate under Refinement Control
</li><li> [Check Chirals] Check Chiral Volumes regularly
</li><li> [Change the Weighing Scheme] <code>(set-matrix 20.0)</code>
[Default is 60, the lower the number the more the geometry is idealised]
</li></ul>

<hr>
<a name="Coot-Droppings"></a>
<div class="header">
<p>
Next: <a href="#Clearing-Backups" accesskey="n" rel="next">Clearing Backups</a>, Previous: <a href="#Low-Resolution" accesskey="p" rel="previous">Low Resolution</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Coot-Droppings-1"></a>
<h3 class="section">9.3 Coot Droppings</h3>
<a name="index-coot-droppings"></a>
<a name="index-droppings"></a>

<p>This describes the files and directory that coot leaves behind after it
has been fed (sorry, I mean &ldquo;used&rdquo;).  Everything except the
<tt>0-coot.state.scm</tt> state file can comfortably be deleted if needed
after coot has finished.
</p>
<a name="index-guano"></a>
<p>You can stop the state and history files being written if you start coot
with the <code>--no-guano</code> option.
</p>
<ul>
<li> <tt>0-coot.state.scm</tt>  The most important file.  This contains the state of 
        coot when you last exited. It contains things like which molecules  
        were read, the maps, the colours of the molecules and map, the screen 
        centre, map size and so on.  When restarting a coot session, this file
        should usually be used.  

</li><li> <tt>0-coot-history.scm</tt> The history of coot commands you used in your last 
        coot session in scheme format.  Incomplete history.  One day this will be 
        a complete history of the session suitable for uploading into a database 
        describing the model modification.

</li><li> <tt>0-coot-history.py</tt> The history of coot commands you used in your last 
        coot session in python format.

</li><li> <tt>coot-download</tt> directory where the files downloaded from the network (e.g. 
        from the EBI and EDS) go.

</li><li> <tt>coot-backup</tt> Each model modification generates the saving of coordinates 
        as a pdb file in this directory.

</li><li> <tt>coot-refmac</tt> When running REFMAC using the Coot interface, the input to 
        refmac and the output go in this directory.

</li><li> <tt>coot-molprobity</tt> When running Molprobity&rsquo;s Probe and Reduce using the Coot
         interface, the input and output go in this directory.

</li></ul>

<hr>
<a name="Clearing-Backups"></a>
<div class="header">
<p>
Next: <a href="#Getting-out-of-_0060_0060Translate_0027_0027-Mode" accesskey="n" rel="next">Getting out of ``Translate'' Mode</a>, Previous: <a href="#Coot-Droppings" accesskey="p" rel="previous">Coot Droppings</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Clearing-Backups-1"></a>
<h3 class="section">9.4 Clearing Backups</h3>
<a name="index-clearing-backups"></a>
<a name="index-backups_002c-clearing"></a>

<p>Coot will occasionally ask you to clear up the <samp>coot-backup</samp>
directory.  You can adjust the behaviour in a number of ways:
</p>
<ul>
<li> <code>(define *clear-out-backup-run-n-days* 3)</code> will run the backup clearance every 3 days (the default is every 7).
</li><li> <code>(define *clear-out-backup-old-days* 1)</code> will clear out files older then 1 day (rather than the default 7 days).
</li><li> You can create your own version of the function that is run on exiting Coot:
   <code>(clear-backups-maybe)</code>
</li></ul>

<p>So, if you wanted to clear out everything more than 1 day old, every
time, without Coot asking you about it:
</p>
<div class="smalllisp">
<pre class="smalllisp">(define *clear-out-backup-run-n-days* 0)
(define *clear-out-backup-old-days* 1)
(define (clear-backups-maybe) 
  (delete-coot-backup-files 'delete)
  (coot-real-exit 0))
</pre></div>


<hr>
<a name="Getting-out-of-_0060_0060Translate_0027_0027-Mode"></a>
<div class="header">
<p>
Next: <a href="#Getting-out-of-_0060_0060Continuous-Rotation_0027_0027-Mode" accesskey="n" rel="next">Getting out of ``Continuous Rotation'' Mode</a>, Previous: <a href="#Clearing-Backups" accesskey="p" rel="previous">Clearing Backups</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Getting-out-of-_0060_0060Translate_0027_0027-Mode-1"></a>
<h3 class="section">9.5 Getting out of &ldquo;Translate&rdquo; Mode</h3>
<p>If you get stuck in &quot;translate&quot; mode in the GL canvas (<em>i.e.</em> mouse
does not rotate the view as you would expect) simply press and release
the Ctrl key to return to &quot;rotate&quot; mode.
</p>
<hr>
<a name="Getting-out-of-_0060_0060Continuous-Rotation_0027_0027-Mode"></a>
<div class="header">
<p>
Next: <a href="#Label-Atom-Only-Mode" accesskey="n" rel="next">Label Atom Only Mode</a>, Previous: <a href="#Getting-out-of-_0060_0060Translate_0027_0027-Mode" accesskey="p" rel="previous">Getting out of ``Translate'' Mode</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Getting-out-of-_0060_0060Continuous-Rotation_0027_0027-Mode-1"></a>
<h3 class="section">9.6 Getting out of &ldquo;Continuous Rotation&rdquo; Mode</h3>

<p>The keyboard <tt class="key">I</tt> key toggles the &ldquo;continuous rotation&rdquo; mode.  The
menu item <code>Draw -&gt; Spin View On/Off</code> does the same thing.
</p>
<hr>
<a name="Label-Atom-Only-Mode"></a>
<div class="header">
<p>
Next: <a href="#Button-Labels" accesskey="n" rel="next">Button Labels</a>, Previous: <a href="#Getting-out-of-_0060_0060Continuous-Rotation_0027_0027-Mode" accesskey="p" rel="previous">Getting out of ``Continuous Rotation'' Mode</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Getting-out-of-_0060_0060Label-Atom-Only_0027_0027-Mode"></a>
<h3 class="section">9.7 Getting out of &ldquo;Label Atom Only&rdquo; Mode</h3>
<p>Similarly, if you are stuck in a mode where the &ldquo;Model/Fit/Refine&rdquo;
buttons don&rsquo;t work (the atoms are not selected, only the atom gets
labelled), press and release the Shift key.
</p>
<hr>
<a name="Button-Labels"></a>
<div class="header">
<p>
Next: <a href="#sec_005fpicking" accesskey="n" rel="next">sec_picking</a>, Previous: <a href="#Label-Atom-Only-Mode" accesskey="p" rel="previous">Label Atom Only Mode</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Button-Labels-1"></a>
<h3 class="section">9.8 Button Labels</h3>
<p>Button labels ending in &ldquo;&hellip;&rdquo; mean that a new dialog will pop-up
when this button is pressed.
</p>
<hr>
<a name="sec_005fpicking"></a>
<div class="header">
<p>
Next: <a href="#Resizing-View" accesskey="n" rel="next">Resizing View</a>, Previous: <a href="#Button-Labels" accesskey="p" rel="previous">Button Labels</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Picking"></a>
<h3 class="section">9.9 Picking</h3>
<a name="index-picking"></a>
<p>Note that left-mouse in the 
graphics window is used for both atom picking and rotating the view,
so try not to click over an atom when trying to rotate the view when
in atom selection mode.  
</p>
<hr>
<a name="Resizing-View"></a>
<div class="header">
<p>
Next: <a href="#Scroll_002dwheel" accesskey="n" rel="next">Scroll-wheel</a>, Previous: <a href="#sec_005fpicking" accesskey="p" rel="previous">sec_picking</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Resizing-View-1"></a>
<h3 class="section">9.10 Resizing View</h3>
<a name="index-resizing-view"></a>
<a name="index-zoom"></a>
<p>Click and drag using right-mouse (up and down or left and right) to zoom in and out.
</p>

<hr>
<a name="Scroll_002dwheel"></a>
<div class="header">
<p>
Next: <a href="#Slow-Computer-Configuration" accesskey="n" rel="next">Slow Computer Configuration</a>, Previous: <a href="#Resizing-View" accesskey="p" rel="previous">Resizing View</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Scroll_002dwheel-1"></a>
<h3 class="section">9.11 Scroll-wheel</h3>
<a name="index-scroll-wheel_002c-map-for"></a>
<p>To change the map to which the scroll-wheel is attached, use the scroll
check button in the Display Manager or use <code>HID -&gt; Scrollwheel
-&gt; Attach Scrollwheel to which map?</code>
</p> 
<hr>
<a name="Slow-Computer-Configuration"></a>
<div class="header">
<p>
Previous: <a href="#Scroll_002dwheel" accesskey="p" rel="previous">Scroll-wheel</a>, Up: <a href="#Hints-and-Usage-Tips" accesskey="u" rel="up">Hints and Usage Tips</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Slow-Computer-Configuration-1"></a>
<h3 class="section">9.12 Slow Computer Configuration</h3>
<a name="index-slow-computer"></a>
<p>Several of the parameters of Coot are chosen
because they are reasonable on my &ldquo;middle-ground&rdquo; development
machine.  However, these parameters can be tweaked so that slower
computers perform better:
</p>
<ul>
<li> <code>(set-use-stroke-characters 1) ; default is to use bitmap characters </code>
</li><li> <code>(set-smooth-scroll-steps 8) ; default 40 </code>
</li><li> <code>(set-smooth-scroll-limit 30) ; Angstroms</code>
</li><li> <code>(set-residue-selection-flash-frames-number 3);</code>
</li><li> <code>(set-skeleton-box-size 20.0) ; A (default 40).</code>
</li><li> <code>(set-active-map-drag-flag 0) ; turn off recontouring every step</code>
</li><li> <code>(set-idle-function-rotate-angle 1.5) ; continuous spin speed</code>
</li></ul>

<hr>
<a name="Other-Programs"></a>
<div class="header">
<p>
Next: <a href="#Scripting-Functions" accesskey="n" rel="next">Scripting Functions</a>, Previous: <a href="#Hints-and-Usage-Tips" accesskey="p" rel="previous">Hints and Usage Tips</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Other-Programs-1"></a>
<h2 class="chapter">10 Other Programs</h2>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#findligand" accesskey="1">findligand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="findligand"></a>
<div class="header">
<p>
Up: <a href="#Other-Programs" accesskey="u" rel="up">Other Programs</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="findligand-1"></a>
<h3 class="section">10.1 findligand</h3>
<a name="index-findligand"></a>

<p><code>findligand</code> is a stand-alone command-line program that uses the
libraries of Coot.
</p>
<p><code>findligand</code> provides a number of command line arguments for
increased flexibility:
</p>
<ul>
<li> <code>--pdbin <em>pdb-in-filename</em></code>

<p>where <code><em>pdb-in-filename</em></code> is the protein (typically)
 </p></li><li> <code>--hklin <em>mtz-filename</em></code>
 </li><li> <code>--f <em>f_col_label</em></code>
 </li><li> <code>--phi <em>phi_col_label</em></code>
 </li><li> <code>--clusters <em>nclust</em></code>

<p>where <code><em>nclust</em></code> is the number of density clusters 
        (potential ligand sites) to search for
 </p></li><li> <code>--sigma <em>sigma-level</em></code>

<p>where <code><em>sigma-level</em></code> the density level (in sigma) above 
        which the map is searched for ligands
 </p></li><li> <code>--fit-fraction <em>frac</em></code>

<p>where <code><em>frac</em></code> is the minimum fraction of atoms in 
        density allowed after fit [default 0.75]
 </p></li><li> <code>--flexible</code>

<p>means use torsional conformation ligand search
 </p></li><li> <code>--samples <em>nsamples</em></code>

<p><code><em>nsamples</em></code> is the number of flexible conformation samples [default 30]
 </p></li><li> <code>--dictionary <em>cif-dictionary-name</em></code>

<p>the file containing the CIF ligand dictionary description
</p></li></ul>

<p>One uses <code>findligand</code> like this:
</p>
<table class="cartouche" border="1"><tr><td>
<p><code>$ findligand <em>various-args ligand-pdb-file-name(s)</em></code>
</p></td></tr></table>

<p><em>i.e.</em> the example ligand pdb files that you wish to search for
are given at the end of the command line.
</p>
<hr>
<a name="Scripting-Functions"></a>
<div class="header">
<p>
Next: <a href="#More-Scripting-Functions" accesskey="n" rel="next">More Scripting Functions</a>, Previous: <a href="#Other-Programs" accesskey="p" rel="previous">Other Programs</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Scripting-Functions-1"></a>
<h2 class="chapter">11 Scripting Functions</h2>


<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#Startup-Functions" accesskey="1">Startup Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#File-System-Functions" accesskey="2">File System Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Widget-Utilities" accesskey="3">Widget Utilities</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#MTZ-and-data-handling-utilities" accesskey="4">MTZ and data handling utilities</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Molecule-Info-Functions" accesskey="5">Molecule Info Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Library-and-Utility-Functions" accesskey="6">Library and Utility Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Graphics-Utility-Functions" accesskey="7">Graphics Utility Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Interface-Preferences" accesskey="8">Interface Preferences</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mouse-Buttons" accesskey="9">Mouse Buttons</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Cursor-Function">Cursor Function</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Model_002fFit_002fRefine-Functions">Model/Fit/Refine Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Backup-Functions">Backup Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Recover-Session-Function">Recover Session Function</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Functions">Map Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Density-Increment">Density Increment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Density-Functions">Density Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Parameters-from-map">Parameters from map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#PDB-Functions">PDB Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Info-Dialog">Info Dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Refmac-Functions">Refmac Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Symmetry-Functions">Symmetry Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#History-Functions">History Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#State-Functions">State Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#The-Virtual-Trackball">The Virtual Trackball</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Clipping-Functions">Clipping Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Unit-Cell-interface">Unit Cell interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Colour">Colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-colour">Map colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Anisotropic-Atoms-Interface">Anisotropic Atoms Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Display-Functions">Display Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Smooth-Scrolling">Smooth Scrolling</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Font-Parameters">Font Parameters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotation-Centre">Rotation Centre</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Selection-Utilities">Atom Selection Utilities</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Skeletonization-Interface">Skeletonization Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Save-Coordinates">Save Coordinates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Read-Phases-File-Functions">Read Phases File Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Graphics-Move">Graphics Move</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Go-To-Atom-Widget-Functions">Go To Atom Widget Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-and-Molecule-Control">Map and Molecule Control</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Align-and-Mutate">Align and Mutate</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Renumber-Residue-Range">Renumber Residue Range</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Scripting-Interface">Scripting Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Monomer">Monomer</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Regularization-and-Refinement">Regularization and Refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Simplex-Refinement-Interface">Simplex Refinement Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Nomenclature-Errors">Nomenclature Errors</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Info-Interface">Atom Info  Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Residue-Info">Residue Info</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Residue-Environment-Functions">Residue Environment Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pointer-Position-Function">Pointer Position Function</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pointer-Functions">Pointer Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Zoom-Functions">Zoom Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#CNS-Data-Functions">CNS Data Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mmCIF-Functions">mmCIF Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#SHELXL-Functions">SHELXL Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Validation-Functions">Validation Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Ramachandran-Plot-Functions">Ramachandran Plot Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Sequence-View-Interface">Sequence View Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Labelling">Atom Labelling</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Screen-Rotation">Screen Rotation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Screen-Translation">Screen Translation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Views-Interface">Views Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Background-Colour">Background Colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Ligand-Fitting-Functions">Ligand Fitting Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Water-Fitting-Functions">Water Fitting Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Bond-Representation">Bond Representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Dots-Representation">Dots Representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pep_002dflip-Interface">Pep-flip Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rigid-Body-Refinement-Interface">Rigid Body Refinement Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Add-Terminal-Residue-Functions">Add Terminal Residue Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Delete-Residues">Delete Residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mainchain-Building-Functions">Mainchain Building Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Close-Molecule-Functions">Close Molecule Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotamer-Functions">Rotamer Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#g_t180-Flip-Side-chain">180 Flip Side chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Mutate-Functions">Mutate Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Alternative-Conformation">Alternative Conformation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Pointer-Atom-Functions">Pointer Atom Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Baton-Build-Interface-Functions">Baton Build Interface Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Crosshairs-Interface">Crosshairs  Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Edit-Chi-Angles">Edit Chi Angles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Masks">Masks</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#check-Waters-Interface">check Waters Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Least_002dSquares-matching">Least-Squares matching</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Trim">Trim</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#External-Ray_002dTracing">External Ray-Tracing</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Superposition-_0028SSM_0029">Superposition (SSM)</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#NCS">NCS</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Helices-and-Strands">Helices and Strands</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Nucleotides">Nucleotides</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#New-Molecule-by-Section-Interface">New Molecule by Section Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#RNA_002fDNA">RNA/DNA</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Sequence-_0028Assignment_0029">Sequence (Assignment)</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Surface-Interface">Surface Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#FFFearing">FFFearing</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Remote-Control">Remote Control</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Display-Lists-for-Maps">Display Lists for Maps</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Browser-Interface">Browser Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Molprobity-Interface">Molprobity Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Sharpening-Interface">Map Sharpening Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Intermediate-Atom-Manipulation-Interface">Intermediate Atom Manipulation Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Marking-Fixed-Atom-Interface">Marking Fixed Atom Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Partial-Charges">Partial Charges</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#EM-interface">EM interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#CCP4mg-Interface">CCP4mg Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Dipoles">Dipoles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Aux-functions">Aux functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#SMILES">SMILES</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#PHENIX-Support">PHENIX Support</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Graphics-Text">Graphics Text</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#PISA-Interaction">PISA Interaction</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Jiggle-Fit">Jiggle Fit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#SBase-interface">SBase interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#FLE_002dView">FLE-View</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#LSQ_002dimprove">LSQ-improve</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#single_002dmodel-view">single-model view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics-2D-ligand-view">graphics 2D ligand view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="Startup-Functions"></a>
<div class="header">
<p>
Next: <a href="#File-System-Functions" accesskey="n" rel="next">File System Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Startup-Functions-1"></a>
<h3 class="section">11.1 Startup Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dprefer_002dpython" accesskey="1">set-prefer-python</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#prefer_002dpython" accesskey="2">prefer-python</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dprefer_002dpython"></a>
<div class="header">
<p>
Next: <a href="#prefer_002dpython" accesskey="n" rel="next">prefer-python</a>, Up: <a href="#Startup-Functions" accesskey="u" rel="up">Startup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dprefer_002dpython-1"></a>
<h4 class="subsection">11.1.1 set-prefer-python</h4>
<dl>
<dt><a name="index-set_002dprefer_002dpython"></a>function: <strong>set-prefer-python</strong></dt>
<dd>
<p>tell coot that you prefer to run python scripts if/when there is an option to do so. 
</p>
</dd></dl>

<hr>
<a name="prefer_002dpython"></a>
<div class="header">
<p>
Previous: <a href="#set_002dprefer_002dpython" accesskey="p" rel="previous">set-prefer-python</a>, Up: <a href="#Startup-Functions" accesskey="u" rel="up">Startup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="prefer_002dpython-1"></a>
<h4 class="subsection">11.1.2 prefer-python</h4>
<dl>
<dt><a name="index-prefer_002dpython"></a>function: <strong>prefer-python</strong></dt>
<dd>
<p>the python-prefered mode. 
</p>
<p>This is available so that the scripting functions know whether on not to put themselves onto in as menu items.
</p>
<p>If you consider using this, consider in preference use_gui_qm == 2, which is used elsewhere to stop python functions adding to the gui, when guile-gtk functions have alread done so. We should clean up this (rather obscure) interface at some stage.
</p>
<p>return 1 for python is prefered, 0 for not. 
</p>
</dd></dl>

<hr>
<a name="File-System-Functions"></a>
<div class="header">
<p>
Next: <a href="#Widget-Utilities" accesskey="n" rel="next">Widget Utilities</a>, Previous: <a href="#Startup-Functions" accesskey="p" rel="previous">Startup Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="File-System-Functions-1"></a>
<h3 class="section">11.2 File System Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#make_002ddirectory_002dmaybe" accesskey="1">make-directory-maybe</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dpaths_002din_002ddisplay_002dmanager" accesskey="2">set-show-paths-in-display-manager</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dpaths_002din_002ddisplay_002dmanager_002dstate" accesskey="3">show-paths-in-display-manager-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dcoordinates_002dglob_002dextension" accesskey="4">add-coordinates-glob-extension</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002ddata_002dglob_002dextension" accesskey="5">add-data-glob-extension</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002ddictionary_002dglob_002dextension" accesskey="6">add-dictionary-glob-extension</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dmap_002dglob_002dextension" accesskey="7">add-map-glob-extension</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002dcoordinates_002dglob_002dextension" accesskey="8">remove-coordinates-glob-extension</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002ddata_002dglob_002dextension" accesskey="9">remove-data-glob-extension</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002ddictionary_002dglob_002dextension">remove-dictionary-glob-extension</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002dmap_002dglob_002dextension">remove-map-glob-extension</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsticky_002dsort_002dby_002ddate">set-sticky-sort-by-date</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#unset_002dsticky_002dsort_002dby_002ddate">unset-sticky-sort-by-date</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfilter_002dfileselection_002dfilenames">set-filter-fileselection-filenames</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#filter_002dfileselection_002dfilenames_002dstate">filter-fileselection-filenames-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#file_002dtype_002dcoords">file-type-coords</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#open_002dcoords_002ddialog">open-coords-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfile_002dchooser_002dselector">set-file-chooser-selector</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="make_002ddirectory_002dmaybe"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002dpaths_002din_002ddisplay_002dmanager" accesskey="n" rel="next">set-show-paths-in-display-manager</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002ddirectory_002dmaybe-1"></a>
<h4 class="subsection">11.2.1 make-directory-maybe</h4>
<dl>
<dt><a name="index-make_002ddirectory_002dmaybe"></a>function: <strong>make-directory-maybe</strong> <em>dir</em></dt>
<dd>
<p>Where <em>dir</em> is a string
</p>
<p>make a directory dir (if it doesn&rsquo;t exist) and return error code 
</p>
<p>If it can be created, create the directory dir, return the success status like mkdir: mkdir
</p>
<p>Returns: zero on success, or -1 if an error occurred. If dir already exists as a directory, return 0 of course. 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002dpaths_002din_002ddisplay_002dmanager"></a>
<div class="header">
<p>
Next: <a href="#show_002dpaths_002din_002ddisplay_002dmanager_002dstate" accesskey="n" rel="next">show-paths-in-display-manager-state</a>, Previous: <a href="#make_002ddirectory_002dmaybe" accesskey="p" rel="previous">make-directory-maybe</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dpaths_002din_002ddisplay_002dmanager-1"></a>
<h4 class="subsection">11.2.2 set-show-paths-in-display-manager</h4>
<dl>
<dt><a name="index-set_002dshow_002dpaths_002din_002ddisplay_002dmanager"></a>function: <strong>set-show-paths-in-display-manager</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>Show Paths in Display Manager? 
</p>
<p>Some people don&rsquo;t like to see the full path names in the display manager here is the way to turn them off, with an argument of 1. 
</p>
</dd></dl>

<hr>
<a name="show_002dpaths_002din_002ddisplay_002dmanager_002dstate"></a>
<div class="header">
<p>
Next: <a href="#add_002dcoordinates_002dglob_002dextension" accesskey="n" rel="next">add-coordinates-glob-extension</a>, Previous: <a href="#set_002dshow_002dpaths_002din_002ddisplay_002dmanager" accesskey="p" rel="previous">set-show-paths-in-display-manager</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dpaths_002din_002ddisplay_002dmanager_002dstate-1"></a>
<h4 class="subsection">11.2.3 show-paths-in-display-manager-state</h4>
<dl>
<dt><a name="index-show_002dpaths_002din_002ddisplay_002dmanager_002dstate"></a>function: <strong>show-paths-in-display-manager-state</strong></dt>
<dd>
<p>return the internal state 
</p>
<p>What is the internal flag?
</p>
<p>Returns: 1 for &quot;yes, display paths&quot; , 0 for not 
</p>
</dd></dl>

<hr>
<a name="add_002dcoordinates_002dglob_002dextension"></a>
<div class="header">
<p>
Next: <a href="#add_002ddata_002dglob_002dextension" accesskey="n" rel="next">add-data-glob-extension</a>, Previous: <a href="#show_002dpaths_002din_002ddisplay_002dmanager_002dstate" accesskey="p" rel="previous">show-paths-in-display-manager-state</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dcoordinates_002dglob_002dextension-1"></a>
<h4 class="subsection">11.2.4 add-coordinates-glob-extension</h4>
<dl>
<dt><a name="index-add_002dcoordinates_002dglob_002dextension"></a>function: <strong>add-coordinates-glob-extension</strong> <em>ext</em></dt>
<dd>
<p>Where <em>ext</em> is a string
</p>
<p>add an extension to be treated as coordinate files 
</p>
</dd></dl>

<hr>
<a name="add_002ddata_002dglob_002dextension"></a>
<div class="header">
<p>
Next: <a href="#add_002ddictionary_002dglob_002dextension" accesskey="n" rel="next">add-dictionary-glob-extension</a>, Previous: <a href="#add_002dcoordinates_002dglob_002dextension" accesskey="p" rel="previous">add-coordinates-glob-extension</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002ddata_002dglob_002dextension-1"></a>
<h4 class="subsection">11.2.5 add-data-glob-extension</h4>
<dl>
<dt><a name="index-add_002ddata_002dglob_002dextension"></a>function: <strong>add-data-glob-extension</strong> <em>ext</em></dt>
<dd>
<p>Where <em>ext</em> is a string
</p>
<p>add an extension to be treated as data (reflection) files 
</p>
</dd></dl>

<hr>
<a name="add_002ddictionary_002dglob_002dextension"></a>
<div class="header">
<p>
Next: <a href="#add_002dmap_002dglob_002dextension" accesskey="n" rel="next">add-map-glob-extension</a>, Previous: <a href="#add_002ddata_002dglob_002dextension" accesskey="p" rel="previous">add-data-glob-extension</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002ddictionary_002dglob_002dextension-1"></a>
<h4 class="subsection">11.2.6 add-dictionary-glob-extension</h4>
<dl>
<dt><a name="index-add_002ddictionary_002dglob_002dextension"></a>function: <strong>add-dictionary-glob-extension</strong> <em>ext</em></dt>
<dd>
<p>Where <em>ext</em> is a string
</p>
<p>add an extension to be treated as geometry dictionary files 
</p>
</dd></dl>

<hr>
<a name="add_002dmap_002dglob_002dextension"></a>
<div class="header">
<p>
Next: <a href="#remove_002dcoordinates_002dglob_002dextension" accesskey="n" rel="next">remove-coordinates-glob-extension</a>, Previous: <a href="#add_002ddictionary_002dglob_002dextension" accesskey="p" rel="previous">add-dictionary-glob-extension</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dmap_002dglob_002dextension-1"></a>
<h4 class="subsection">11.2.7 add-map-glob-extension</h4>
<dl>
<dt><a name="index-add_002dmap_002dglob_002dextension"></a>function: <strong>add-map-glob-extension</strong> <em>ext</em></dt>
<dd>
<p>Where <em>ext</em> is a string
</p>
<p>add an extension to be treated as geometry map files 
</p>
</dd></dl>

<hr>
<a name="remove_002dcoordinates_002dglob_002dextension"></a>
<div class="header">
<p>
Next: <a href="#remove_002ddata_002dglob_002dextension" accesskey="n" rel="next">remove-data-glob-extension</a>, Previous: <a href="#add_002dmap_002dglob_002dextension" accesskey="p" rel="previous">add-map-glob-extension</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002dcoordinates_002dglob_002dextension-1"></a>
<h4 class="subsection">11.2.8 remove-coordinates-glob-extension</h4>
<dl>
<dt><a name="index-remove_002dcoordinates_002dglob_002dextension"></a>function: <strong>remove-coordinates-glob-extension</strong> <em>ext</em></dt>
<dd>
<p>Where <em>ext</em> is a string
</p>
<p>remove an extension to be treated as coordinate files 
</p>
</dd></dl>

<hr>
<a name="remove_002ddata_002dglob_002dextension"></a>
<div class="header">
<p>
Next: <a href="#remove_002ddictionary_002dglob_002dextension" accesskey="n" rel="next">remove-dictionary-glob-extension</a>, Previous: <a href="#remove_002dcoordinates_002dglob_002dextension" accesskey="p" rel="previous">remove-coordinates-glob-extension</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002ddata_002dglob_002dextension-1"></a>
<h4 class="subsection">11.2.9 remove-data-glob-extension</h4>
<dl>
<dt><a name="index-remove_002ddata_002dglob_002dextension"></a>function: <strong>remove-data-glob-extension</strong> <em>ext</em></dt>
<dd>
<p>Where <em>ext</em> is a string
</p>
<p>remove an extension to be treated as data (reflection) files 
</p>
</dd></dl>

<hr>
<a name="remove_002ddictionary_002dglob_002dextension"></a>
<div class="header">
<p>
Next: <a href="#remove_002dmap_002dglob_002dextension" accesskey="n" rel="next">remove-map-glob-extension</a>, Previous: <a href="#remove_002ddata_002dglob_002dextension" accesskey="p" rel="previous">remove-data-glob-extension</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002ddictionary_002dglob_002dextension-1"></a>
<h4 class="subsection">11.2.10 remove-dictionary-glob-extension</h4>
<dl>
<dt><a name="index-remove_002ddictionary_002dglob_002dextension"></a>function: <strong>remove-dictionary-glob-extension</strong> <em>ext</em></dt>
<dd>
<p>Where <em>ext</em> is a string
</p>
<p>remove an extension to be treated as geometry dictionary files 
</p>
</dd></dl>

<hr>
<a name="remove_002dmap_002dglob_002dextension"></a>
<div class="header">
<p>
Next: <a href="#set_002dsticky_002dsort_002dby_002ddate" accesskey="n" rel="next">set-sticky-sort-by-date</a>, Previous: <a href="#remove_002ddictionary_002dglob_002dextension" accesskey="p" rel="previous">remove-dictionary-glob-extension</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002dmap_002dglob_002dextension-1"></a>
<h4 class="subsection">11.2.11 remove-map-glob-extension</h4>
<dl>
<dt><a name="index-remove_002dmap_002dglob_002dextension"></a>function: <strong>remove-map-glob-extension</strong> <em>ext</em></dt>
<dd>
<p>Where <em>ext</em> is a string
</p>
<p>remove an extension to be treated as geometry map files 
</p>
</dd></dl>

<hr>
<a name="set_002dsticky_002dsort_002dby_002ddate"></a>
<div class="header">
<p>
Next: <a href="#unset_002dsticky_002dsort_002dby_002ddate" accesskey="n" rel="next">unset-sticky-sort-by-date</a>, Previous: <a href="#remove_002dmap_002dglob_002dextension" accesskey="p" rel="previous">remove-map-glob-extension</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsticky_002dsort_002dby_002ddate-1"></a>
<h4 class="subsection">11.2.12 set-sticky-sort-by-date</h4>
<dl>
<dt><a name="index-set_002dsticky_002dsort_002dby_002ddate"></a>function: <strong>set-sticky-sort-by-date</strong></dt>
<dd>
<p>sort files in the file selection by date? 
</p>
<p>some people like to have their files sorted by date by default 
</p>
</dd></dl>

<hr>
<a name="unset_002dsticky_002dsort_002dby_002ddate"></a>
<div class="header">
<p>
Next: <a href="#set_002dfilter_002dfileselection_002dfilenames" accesskey="n" rel="next">set-filter-fileselection-filenames</a>, Previous: <a href="#set_002dsticky_002dsort_002dby_002ddate" accesskey="p" rel="previous">set-sticky-sort-by-date</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="unset_002dsticky_002dsort_002dby_002ddate-1"></a>
<h4 class="subsection">11.2.13 unset-sticky-sort-by-date</h4>
<dl>
<dt><a name="index-unset_002dsticky_002dsort_002dby_002ddate"></a>function: <strong>unset-sticky-sort-by-date</strong></dt>
<dd>
<p>do not sort files in the file selection by date? 
</p>
<p>removes the sorting of files by date 
</p>
</dd></dl>

<hr>
<a name="set_002dfilter_002dfileselection_002dfilenames"></a>
<div class="header">
<p>
Next: <a href="#filter_002dfileselection_002dfilenames_002dstate" accesskey="n" rel="next">filter-fileselection-filenames-state</a>, Previous: <a href="#unset_002dsticky_002dsort_002dby_002ddate" accesskey="p" rel="previous">unset-sticky-sort-by-date</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfilter_002dfileselection_002dfilenames-1"></a>
<h4 class="subsection">11.2.14 set-filter-fileselection-filenames</h4>
<dl>
<dt><a name="index-set_002dfilter_002dfileselection_002dfilenames"></a>function: <strong>set-filter-fileselection-filenames</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>on opening a file selection dialog, pre-filter the files. 
</p>
<p>set to 1 to pre-filter, [0 (off, non-pre-filtering) is the default 
</p>
</dd></dl>

<hr>
<a name="filter_002dfileselection_002dfilenames_002dstate"></a>
<div class="header">
<p>
Next: <a href="#file_002dtype_002dcoords" accesskey="n" rel="next">file-type-coords</a>, Previous: <a href="#set_002dfilter_002dfileselection_002dfilenames" accesskey="p" rel="previous">set-filter-fileselection-filenames</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="filter_002dfileselection_002dfilenames_002dstate-1"></a>
<h4 class="subsection">11.2.15 filter-fileselection-filenames-state</h4>
<dl>
<dt><a name="index-filter_002dfileselection_002dfilenames_002dstate"></a>function: <strong>filter-fileselection-filenames-state</strong></dt>
<dd>
<p>, return the state of the above variable 
</p>
</dd></dl>

<hr>
<a name="file_002dtype_002dcoords"></a>
<div class="header">
<p>
Next: <a href="#open_002dcoords_002ddialog" accesskey="n" rel="next">open-coords-dialog</a>, Previous: <a href="#filter_002dfileselection_002dfilenames_002dstate" accesskey="p" rel="previous">filter-fileselection-filenames-state</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="file_002dtype_002dcoords-1"></a>
<h4 class="subsection">11.2.16 file-type-coords</h4>
<dl>
<dt><a name="index-file_002dtype_002dcoords"></a>function: <strong>file-type-coords</strong> <em>file_name</em></dt>
<dd>
<p>Where <em>file_name</em> is a string
</p>
<p>is the given file name suitable to be read as coordinates? 
</p>
</dd></dl>

<hr>
<a name="open_002dcoords_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#set_002dfile_002dchooser_002dselector" accesskey="n" rel="next">set-file-chooser-selector</a>, Previous: <a href="#file_002dtype_002dcoords" accesskey="p" rel="previous">file-type-coords</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="open_002dcoords_002ddialog-1"></a>
<h4 class="subsection">11.2.17 open-coords-dialog</h4>
<dl>
<dt><a name="index-open_002dcoords_002ddialog"></a>function: <strong>open-coords-dialog</strong></dt>
<dd>
<p>display the open coordinates dialog 
</p>
</dd></dl>

<hr>
<a name="set_002dfile_002dchooser_002dselector"></a>
<div class="header">
<p>
Previous: <a href="#open_002dcoords_002ddialog" accesskey="p" rel="previous">open-coords-dialog</a>, Up: <a href="#File-System-Functions" accesskey="u" rel="up">File System Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfile_002dchooser_002dselector-1"></a>
<h4 class="subsection">11.2.18 set-file-chooser-selector</h4>
<dl>
<dt><a name="index-set_002dfile_002dchooser_002dselector"></a>function: <strong>set-file-chooser-selector</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>this flag set chooser as default for windows, otherwise use selector 0 is selector 1 is chooser 
</p>
</dd></dl>

<hr>
<a name="Widget-Utilities"></a>
<div class="header">
<p>
Next: <a href="#MTZ-and-data-handling-utilities" accesskey="n" rel="next">MTZ and data handling utilities</a>, Previous: <a href="#File-System-Functions" accesskey="p" rel="previous">File System Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Widget-Utilities-1"></a>
<h3 class="section">11.3 Widget Utilities</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dmain_002dwindow_002dtitle" accesskey="1">set-main-window-title</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dmain_002dwindow_002dtitle"></a>
<div class="header">
<p>
Up: <a href="#Widget-Utilities" accesskey="u" rel="up">Widget Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmain_002dwindow_002dtitle-1"></a>
<h4 class="subsection">11.3.1 set-main-window-title</h4>
<dl>
<dt><a name="index-set_002dmain_002dwindow_002dtitle"></a>function: <strong>set-main-window-title</strong> <em>s</em></dt>
<dd>
<p>Where <em>s</em> is a string
</p>
<p>set the main window title. 
</p>
<p>function added for Lothar Esser 
</p>
</dd></dl>

<hr>
<a name="MTZ-and-data-handling-utilities"></a>
<div class="header">
<p>
Next: <a href="#Molecule-Info-Functions" accesskey="n" rel="next">Molecule Info Functions</a>, Previous: <a href="#Widget-Utilities" accesskey="p" rel="previous">Widget Utilities</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="MTZ-and-data-handling-utilities-1"></a>
<h3 class="section">11.4 MTZ and data handling utilities</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#manage_002dcolumn_002dselector" accesskey="1">manage-column-selector</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="manage_002dcolumn_002dselector"></a>
<div class="header">
<p>
Up: <a href="#MTZ-and-data-handling-utilities" accesskey="u" rel="up">MTZ and data handling utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="manage_002dcolumn_002dselector-1"></a>
<h4 class="subsection">11.4.1 manage-column-selector</h4>
<dl>
<dt><a name="index-manage_002dcolumn_002dselector"></a>function: <strong>manage-column-selector</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>given a filename, try to read it as a data file 
</p>
<p>We try as .phs and .cif files first 
</p>
</dd></dl>

<hr>
<a name="Molecule-Info-Functions"></a>
<div class="header">
<p>
Next: <a href="#Library-and-Utility-Functions" accesskey="n" rel="next">Library and Utility Functions</a>, Previous: <a href="#MTZ-and-data-handling-utilities" accesskey="p" rel="previous">MTZ and data handling utilities</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Molecule-Info-Functions-1"></a>
<h3 class="section">11.5 Molecule Info Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#chain_002dn_002dresidues" accesskey="1">chain-n-residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#molecule_002dcentre_002dinternal" accesskey="2">molecule-centre-internal</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#seqnum_002dfrom_002dserial_002dnumber" accesskey="3">seqnum-from-serial-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#insertion_002dcode_002dfrom_002dserial_002dnumber" accesskey="4">insertion-code-from-serial-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#chain_002did_002dscm" accesskey="5">chain-id-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#n_002dmodels" accesskey="6">n-models</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#n_002dchains" accesskey="7">n-chains</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#is_002dsolvent_002dchain_002dp" accesskey="8">is-solvent-chain-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#is_002dprotein_002dchain_002dp" accesskey="9">is-protein-chain-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#is_002dnucleotide_002dchain_002dp">is-nucleotide-chain-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#n_002dresidues">n-residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#n_002datoms">n-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remarks_002dscm">remarks-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sort_002dchains">sort-chains</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sort_002dresidues">sort-residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remarks_002ddialog">remarks-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#print_002dheader_002dsecondary_002dstructure_002dinfo">print-header-secondary-structure-info</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dheader_002dsecondary_002dstructure_002dinfo">add-header-secondary-structure-info</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#copy_002dmolecule">copy-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule">add-ligand-delete-residue-copy-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#exchange_002dchain_002dids_002dfor_002dseg_002dids">exchange-chain-ids-for-seg-ids</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dremarks_002dbrowswer">show-remarks-browswer</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="chain_002dn_002dresidues"></a>
<div class="header">
<p>
Next: <a href="#molecule_002dcentre_002dinternal" accesskey="n" rel="next">molecule-centre-internal</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="chain_002dn_002dresidues-1"></a>
<h4 class="subsection">11.5.1 chain-n-residues</h4>
<dl>
<dt><a name="index-chain_002dn_002dresidues"></a>function: <strong>chain-n-residues</strong> <em>chain_id imol</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>chain_id</em> is a string
</li><li> <em>imol</em> is an integer number
</li></ul>

<p>the number of residues in chain chain_id and molecule number imol 
</p>
<p>Returns: the number of residues 
</p>
</dd></dl>

<hr>
<a name="molecule_002dcentre_002dinternal"></a>
<div class="header">
<p>
Next: <a href="#seqnum_002dfrom_002dserial_002dnumber" accesskey="n" rel="next">seqnum-from-serial-number</a>, Previous: <a href="#chain_002dn_002dresidues" accesskey="p" rel="previous">chain-n-residues</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="molecule_002dcentre_002dinternal-1"></a>
<h4 class="subsection">11.5.2 molecule-centre-internal</h4>
<dl>
<dt><a name="index-molecule_002dcentre_002dinternal"></a>function: <strong>molecule-centre-internal</strong> <em>imol iaxis</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>iaxis</em> is an integer number
</li></ul>

<p>internal function for molecule centre 
</p>
<p>Returns: status, less than -9999 is for failure (eg. bad imol); 
</p>
</dd></dl>

<hr>
<a name="seqnum_002dfrom_002dserial_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#insertion_002dcode_002dfrom_002dserial_002dnumber" accesskey="n" rel="next">insertion-code-from-serial-number</a>, Previous: <a href="#molecule_002dcentre_002dinternal" accesskey="p" rel="previous">molecule-centre-internal</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="seqnum_002dfrom_002dserial_002dnumber-1"></a>
<h4 class="subsection">11.5.3 seqnum-from-serial-number</h4>
<dl>
<dt><a name="index-seqnum_002dfrom_002dserial_002dnumber"></a>function: <strong>seqnum-from-serial-number</strong> <em>imol chain_id serial_num</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>serial_num</em> is an integer number
</li></ul>

<p>a residue seqnum (normal residue number) from a residue serial number 
</p>
<p>Returns: &lt; -9999 on failure 
</p>
</dd></dl>

<hr>
<a name="insertion_002dcode_002dfrom_002dserial_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#chain_002did_002dscm" accesskey="n" rel="next">chain-id-scm</a>, Previous: <a href="#seqnum_002dfrom_002dserial_002dnumber" accesskey="p" rel="previous">seqnum-from-serial-number</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="insertion_002dcode_002dfrom_002dserial_002dnumber-1"></a>
<h4 class="subsection">11.5.4 insertion-code-from-serial-number</h4>
<dl>
<dt><a name="index-insertion_002dcode_002dfrom_002dserial_002dnumber"></a>function: <strong>insertion-code-from-serial-number</strong> <em>imol chain_id serial_num</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>serial_num</em> is an integer number
</li></ul>

<p>the insertion code of the residue. 
</p>
<p>Returns: NULL (scheme False) on failure. 
</p>
</dd></dl>

<hr>
<a name="chain_002did_002dscm"></a>
<div class="header">
<p>
Next: <a href="#n_002dmodels" accesskey="n" rel="next">n-models</a>, Previous: <a href="#insertion_002dcode_002dfrom_002dserial_002dnumber" accesskey="p" rel="previous">insertion-code-from-serial-number</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="chain_002did_002dscm-1"></a>
<h4 class="subsection">11.5.5 chain-id-scm</h4>
<dl>
<dt><a name="index-chain_002did_002dscm"></a>function: <strong>chain-id-scm</strong> <em>imol ichain</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>ichain</em> is an integer number
</li></ul>

<p>the chain_id (string) of the ichain-th chain molecule number imol 
</p>
<p>Returns: the chain-id 
</p>
</dd></dl>

<hr>
<a name="n_002dmodels"></a>
<div class="header">
<p>
Next: <a href="#n_002dchains" accesskey="n" rel="next">n-chains</a>, Previous: <a href="#chain_002did_002dscm" accesskey="p" rel="previous">chain-id-scm</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="n_002dmodels-1"></a>
<h4 class="subsection">11.5.6 n-models</h4>
<dl>
<dt><a name="index-n_002dmodels"></a>function: <strong>n-models</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the number of models in molecule number imol 
</p>
<p>useful for NMR or other such multi-model molecules.
</p>
<p>return the number of models or -1 if there was a problem with the given molecule. 
</p>
</dd></dl>

<hr>
<a name="n_002dchains"></a>
<div class="header">
<p>
Next: <a href="#is_002dsolvent_002dchain_002dp" accesskey="n" rel="next">is-solvent-chain-p</a>, Previous: <a href="#n_002dmodels" accesskey="p" rel="previous">n-models</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="n_002dchains-1"></a>
<h4 class="subsection">11.5.7 n-chains</h4>
<dl>
<dt><a name="index-n_002dchains"></a>function: <strong>n-chains</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>number of chains in molecule number imol 
</p>
<p>Returns: the number of chains 
</p>
</dd></dl>

<hr>
<a name="is_002dsolvent_002dchain_002dp"></a>
<div class="header">
<p>
Next: <a href="#is_002dprotein_002dchain_002dp" accesskey="n" rel="next">is-protein-chain-p</a>, Previous: <a href="#n_002dchains" accesskey="p" rel="previous">n-chains</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="is_002dsolvent_002dchain_002dp-1"></a>
<h4 class="subsection">11.5.8 is-solvent-chain-p</h4>
<dl>
<dt><a name="index-is_002dsolvent_002dchain_002dp"></a>function: <strong>is-solvent-chain-p</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li></ul>

<p>is this a solvent chain? [Raw function] 
</p>
<p>This is a raw interface function, you should generally not use this, but instead use (is-solvent-chain? imol chain-id)
</p>

<p>This wraps the mmdb function isSolventChain(). 
</p>
<p>Returns: -1 on error, 0 for no, 1 for is &quot;a solvent chain&quot;. We wouldn&rsquo;t want to be doing rotamer searches and the like on such a chain.
</p>
</dd></dl>

<hr>
<a name="is_002dprotein_002dchain_002dp"></a>
<div class="header">
<p>
Next: <a href="#is_002dnucleotide_002dchain_002dp" accesskey="n" rel="next">is-nucleotide-chain-p</a>, Previous: <a href="#is_002dsolvent_002dchain_002dp" accesskey="p" rel="previous">is-solvent-chain-p</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="is_002dprotein_002dchain_002dp-1"></a>
<h4 class="subsection">11.5.9 is-protein-chain-p</h4>
<dl>
<dt><a name="index-is_002dprotein_002dchain_002dp"></a>function: <strong>is-protein-chain-p</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li></ul>

<p>is this a protein chain? [Raw function] 
</p>
<p>This is a raw interface function, you should generally not use this, but instead use (is-protein-chain? imol chain-id)
</p>

<p>This wraps the mmdb function isAminoacidChain(). 
</p>
<p>Returns: -1 on error, 0 for no, 1 for is &quot;a protein chain&quot;. We wouldn&rsquo;t want to be doing rotamer searches and the like on such a chain.
</p>
</dd></dl>

<hr>
<a name="is_002dnucleotide_002dchain_002dp"></a>
<div class="header">
<p>
Next: <a href="#n_002dresidues" accesskey="n" rel="next">n-residues</a>, Previous: <a href="#is_002dprotein_002dchain_002dp" accesskey="p" rel="previous">is-protein-chain-p</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="is_002dnucleotide_002dchain_002dp-1"></a>
<h4 class="subsection">11.5.10 is-nucleotide-chain-p</h4>
<dl>
<dt><a name="index-is_002dnucleotide_002dchain_002dp"></a>function: <strong>is-nucleotide-chain-p</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li></ul>

<p>is this a nucleic acid chain? [Raw function] 
</p>
<p>This is a raw interface function, you should generally not use this, but instead use (is-nucleicacid-chain? imol chain-id)
</p>

<p>This wraps the mmdb function isNucleotideChain(). For completeness. 
</p>
<p>Returns: -1 on error, 0 for no, 1 for is &quot;a nucleicacid chain&quot;. We wouldn&rsquo;t want to be doing rotamer searches and the like on such a chain.
</p>
</dd></dl>

<hr>
<a name="n_002dresidues"></a>
<div class="header">
<p>
Next: <a href="#n_002datoms" accesskey="n" rel="next">n-atoms</a>, Previous: <a href="#is_002dnucleotide_002dchain_002dp" accesskey="p" rel="previous">is-nucleotide-chain-p</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="n_002dresidues-1"></a>
<h4 class="subsection">11.5.11 n-residues</h4>
<dl>
<dt><a name="index-n_002dresidues"></a>function: <strong>n-residues</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the number of residues in the molecule, 
</p>
<p>return -1 if this is a map or closed. 
</p>
</dd></dl>

<hr>
<a name="n_002datoms"></a>
<div class="header">
<p>
Next: <a href="#remarks_002dscm" accesskey="n" rel="next">remarks-scm</a>, Previous: <a href="#n_002dresidues" accesskey="p" rel="previous">n-residues</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="n_002datoms-1"></a>
<h4 class="subsection">11.5.12 n-atoms</h4>
<dl>
<dt><a name="index-n_002datoms"></a>function: <strong>n-atoms</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the atoms of residues in the molecule, 
</p>
<p>return -1 if this is a map or closed. 
</p>
</dd></dl>

<hr>
<a name="remarks_002dscm"></a>
<div class="header">
<p>
Next: <a href="#sort_002dchains" accesskey="n" rel="next">sort-chains</a>, Previous: <a href="#n_002datoms" accesskey="p" rel="previous">n-atoms</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remarks_002dscm-1"></a>
<h4 class="subsection">11.5.13 remarks-scm</h4>
<dl>
<dt><a name="index-remarks_002dscm"></a>function: <strong>remarks-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return a list of the remarks of hte molecule number imol 
</p>
</dd></dl>

<hr>
<a name="sort_002dchains"></a>
<div class="header">
<p>
Next: <a href="#sort_002dresidues" accesskey="n" rel="next">sort-residues</a>, Previous: <a href="#remarks_002dscm" accesskey="p" rel="previous">remarks-scm</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="sort_002dchains-1"></a>
<h4 class="subsection">11.5.14 sort-chains</h4>
<dl>
<dt><a name="index-sort_002dchains"></a>function: <strong>sort-chains</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>sort the chain ids of the imol-th molecule in lexographical order 
</p>
</dd></dl>

<hr>
<a name="sort_002dresidues"></a>
<div class="header">
<p>
Next: <a href="#remarks_002ddialog" accesskey="n" rel="next">remarks-dialog</a>, Previous: <a href="#sort_002dchains" accesskey="p" rel="previous">sort-chains</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="sort_002dresidues-1"></a>
<h4 class="subsection">11.5.15 sort-residues</h4>
<dl>
<dt><a name="index-sort_002dresidues"></a>function: <strong>sort-residues</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>sort the residues of the imol-th molecule 
</p>
</dd></dl>

<hr>
<a name="remarks_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#print_002dheader_002dsecondary_002dstructure_002dinfo" accesskey="n" rel="next">print-header-secondary-structure-info</a>, Previous: <a href="#sort_002dresidues" accesskey="p" rel="previous">sort-residues</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remarks_002ddialog-1"></a>
<h4 class="subsection">11.5.16 remarks-dialog</h4>
<dl>
<dt><a name="index-remarks_002ddialog"></a>function: <strong>remarks-dialog</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>a gui dialog showing remarks header info (for a model molecule). 
</p>
</dd></dl>

<hr>
<a name="print_002dheader_002dsecondary_002dstructure_002dinfo"></a>
<div class="header">
<p>
Next: <a href="#add_002dheader_002dsecondary_002dstructure_002dinfo" accesskey="n" rel="next">add-header-secondary-structure-info</a>, Previous: <a href="#remarks_002ddialog" accesskey="p" rel="previous">remarks-dialog</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="print_002dheader_002dsecondary_002dstructure_002dinfo-1"></a>
<h4 class="subsection">11.5.17 print-header-secondary-structure-info</h4>
<dl>
<dt><a name="index-print_002dheader_002dsecondary_002dstructure_002dinfo"></a>function: <strong>print-header-secondary-structure-info</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>simply print secondary structure info to the terminal/console. In future, this could/should return the info. 
</p>
</dd></dl>

<hr>
<a name="add_002dheader_002dsecondary_002dstructure_002dinfo"></a>
<div class="header">
<p>
Next: <a href="#copy_002dmolecule" accesskey="n" rel="next">copy-molecule</a>, Previous: <a href="#print_002dheader_002dsecondary_002dstructure_002dinfo" accesskey="p" rel="previous">print-header-secondary-structure-info</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dheader_002dsecondary_002dstructure_002dinfo-1"></a>
<h4 class="subsection">11.5.18 add-header-secondary-structure-info</h4>
<dl>
<dt><a name="index-add_002dheader_002dsecondary_002dstructure_002dinfo"></a>function: <strong>add-header-secondary-structure-info</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>add secondary structure info to the internal representation of the model 
</p>
</dd></dl>

<hr>
<a name="copy_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule" accesskey="n" rel="next">add-ligand-delete-residue-copy-molecule</a>, Previous: <a href="#add_002dheader_002dsecondary_002dstructure_002dinfo" accesskey="p" rel="previous">add-header-secondary-structure-info</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="copy_002dmolecule-1"></a>
<h4 class="subsection">11.5.19 copy-molecule</h4>
<dl>
<dt><a name="index-copy_002dmolecule"></a>function: <strong>copy-molecule</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>copy molecule imol 
</p>
<p>Returns: the new molecule number. Return -1 on failure to copy molecule (out of range, or molecule is closed) 
</p>
</dd></dl>

<hr>
<a name="add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#exchange_002dchain_002dids_002dfor_002dseg_002dids" accesskey="n" rel="next">exchange-chain-ids-for-seg-ids</a>, Previous: <a href="#copy_002dmolecule" accesskey="p" rel="previous">copy-molecule</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule-1"></a>
<h4 class="subsection">11.5.20 add-ligand-delete-residue-copy-molecule</h4>
<dl>
<dt><a name="index-add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule"></a>function: <strong>add-ligand-delete-residue-copy-molecule</strong> <em>imol_ligand_new chain_id_ligand_new resno_ligand_new imol_current chain_id_ligand_current resno_ligand_current</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_ligand_new</em> is an integer number
</li><li> <em>chain_id_ligand_new</em> is a string
</li><li> <em>resno_ligand_new</em> is an integer number
</li><li> <em>imol_current</em> is an integer number
</li><li> <em>chain_id_ligand_current</em> is a string
</li><li> <em>resno_ligand_current</em> is an integer number
</li></ul>

<p>Copy a molecule with addition of a ligand and a deletion of current ligand. 
</p>
<p>This function is used when adding a new (modified) ligand to a structure. It creates a new molecule that is a copy of the current molecule except that the new ligand is added and the current ligand/residue is deleted. 
</p>
</dd></dl>

<hr>
<a name="exchange_002dchain_002dids_002dfor_002dseg_002dids"></a>
<div class="header">
<p>
Next: <a href="#show_002dremarks_002dbrowswer" accesskey="n" rel="next">show-remarks-browswer</a>, Previous: <a href="#add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule" accesskey="p" rel="previous">add-ligand-delete-residue-copy-molecule</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="exchange_002dchain_002dids_002dfor_002dseg_002dids-1"></a>
<h4 class="subsection">11.5.21 exchange-chain-ids-for-seg-ids</h4>
<dl>
<dt><a name="index-exchange_002dchain_002dids_002dfor_002dseg_002dids"></a>function: <strong>exchange-chain-ids-for-seg-ids</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Experimental interface for Ribosome People. 
</p>
<p>Ribosome People have many chains in their pdb file, they prefer segids to chainids (chainids are only 1 character). But coot uses the concept of chain ids and not seg-ids. mmdb allow us to use more than one char in the chainid, so after we read in a pdb, let&rsquo;s replace the chain ids with the segids. Will that help? 
</p>
</dd></dl>

<hr>
<a name="show_002dremarks_002dbrowswer"></a>
<div class="header">
<p>
Previous: <a href="#exchange_002dchain_002dids_002dfor_002dseg_002dids" accesskey="p" rel="previous">exchange-chain-ids-for-seg-ids</a>, Up: <a href="#Molecule-Info-Functions" accesskey="u" rel="up">Molecule Info Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dremarks_002dbrowswer-1"></a>
<h4 class="subsection">11.5.22 show-remarks-browswer</h4>
<dl>
<dt><a name="index-show_002dremarks_002dbrowswer"></a>function: <strong>show-remarks-browswer</strong></dt>
<dd>
<p>show the remarks browser 
</p>
</dd></dl>

<hr>
<a name="Library-and-Utility-Functions"></a>
<div class="header">
<p>
Next: <a href="#Graphics-Utility-Functions" accesskey="n" rel="next">Graphics Utility Functions</a>, Previous: <a href="#Molecule-Info-Functions" accesskey="p" rel="previous">Molecule Info Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Library-and-Utility-Functions-1"></a>
<h3 class="section">11.6 Library and Utility Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#git_002drevision_002dcount" accesskey="1">git-revision-count</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#svn_002drevision" accesskey="2">svn-revision</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#molecule_002dname" accesskey="3">molecule-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#molecule_002dname_002dstub_002dscm" accesskey="4">molecule-name-stub-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#molecule_002dname_002dstub_002dpy" accesskey="5">molecule-name-stub-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmolecule_002dname" accesskey="6">set-molecule-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot_002dreal_002dexit" accesskey="7">coot-real-exit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot_002dno_002dstate_002dreal_002dexit" accesskey="8">coot-no-state-real-exit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot_002dclear_002dbackup_002dor_002dreal_002dexit" accesskey="9">coot-clear-backup-or-real-exit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot_002dsave_002dstate_002dand_002dexit">coot-save-state-and-exit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#run_002dclear_002dbackups">run-clear-backups</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#first_002dcoords_002dimol">first-coords-imol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#first_002dsmall_002dcoords_002dimol">first-small-coords-imol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#first_002dunsaved_002dcoords_002dimol">first-unsaved-coords-imol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mmcif_002dsfs_002dto_002dmtz">mmcif-sfs-to-mtz</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="git_002drevision_002dcount"></a>
<div class="header">
<p>
Next: <a href="#svn_002drevision" accesskey="n" rel="next">svn-revision</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="git_002drevision_002dcount-1"></a>
<h4 class="subsection">11.6.1 git-revision-count</h4>
<dl>
<dt><a name="index-git_002drevision_002dcount"></a>function: <strong>git-revision-count</strong></dt>
<dd>
<p>return the git revision count for for this build. 
</p>
</dd></dl>

<hr>
<a name="svn_002drevision"></a>
<div class="header">
<p>
Next: <a href="#molecule_002dname" accesskey="n" rel="next">molecule-name</a>, Previous: <a href="#git_002drevision_002dcount" accesskey="p" rel="previous">git-revision-count</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="svn_002drevision-1"></a>
<h4 class="subsection">11.6.2 svn-revision</h4>
<dl>
<dt><a name="index-svn_002drevision"></a>function: <strong>svn-revision</strong></dt>
<dd>
<p>an alias to git_revision_count() for backwards compatibility 
</p>
</dd></dl>

<hr>
<a name="molecule_002dname"></a>
<div class="header">
<p>
Next: <a href="#molecule_002dname_002dstub_002dscm" accesskey="n" rel="next">molecule-name-stub-scm</a>, Previous: <a href="#svn_002drevision" accesskey="p" rel="previous">svn-revision</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="molecule_002dname-1"></a>
<h4 class="subsection">11.6.3 molecule-name</h4>
<dl>
<dt><a name="index-molecule_002dname"></a>function: <strong>molecule-name</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the name of molecule number imol 
</p>
<p>Returns: 0 if not a valid name ( -&gt; False in scheme) e.g. &quot;/a/b/c.pdb&quot; for &quot;d/e/f.mtz FWT PHWT&quot; 
</p>
</dd></dl>

<hr>
<a name="molecule_002dname_002dstub_002dscm"></a>
<div class="header">
<p>
Next: <a href="#molecule_002dname_002dstub_002dpy" accesskey="n" rel="next">molecule-name-stub-py</a>, Previous: <a href="#molecule_002dname" accesskey="p" rel="previous">molecule-name</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="molecule_002dname_002dstub_002dscm-1"></a>
<h4 class="subsection">11.6.4 molecule-name-stub-scm</h4>
<dl>
<dt><a name="index-molecule_002dname_002dstub_002dscm"></a>function: <strong>molecule-name-stub-scm</strong> <em>imol include_path_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>include_path_flag</em> is an integer number
</li></ul>

<p>return the molecule name without file extension 
</p>
</dd></dl>

<hr>
<a name="molecule_002dname_002dstub_002dpy"></a>
<div class="header">
<p>
Next: <a href="#set_002dmolecule_002dname" accesskey="n" rel="next">set-molecule-name</a>, Previous: <a href="#molecule_002dname_002dstub_002dscm" accesskey="p" rel="previous">molecule-name-stub-scm</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="molecule_002dname_002dstub_002dpy-1"></a>
<h4 class="subsection">11.6.5 molecule-name-stub-py</h4>
<dl>
<dt><a name="index-molecule_002dname_002dstub_002dpy"></a>function: <strong>molecule-name-stub-py</strong> <em>imol include_path_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>include_path_flag</em> is an integer number
</li></ul>

<p>return the molecule name without file extension 
</p>
</dd></dl>

<hr>
<a name="set_002dmolecule_002dname"></a>
<div class="header">
<p>
Next: <a href="#coot_002dreal_002dexit" accesskey="n" rel="next">coot-real-exit</a>, Previous: <a href="#molecule_002dname_002dstub_002dpy" accesskey="p" rel="previous">molecule-name-stub-py</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmolecule_002dname-1"></a>
<h4 class="subsection">11.6.6 set-molecule-name</h4>
<dl>
<dt><a name="index-set_002dmolecule_002dname"></a>function: <strong>set-molecule-name</strong> <em>imol new_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>new_name</em> is a string
</li></ul>

<p>set the molecule name of the imol-th molecule 
</p>
</dd></dl>

<hr>
<a name="coot_002dreal_002dexit"></a>
<div class="header">
<p>
Next: <a href="#coot_002dno_002dstate_002dreal_002dexit" accesskey="n" rel="next">coot-no-state-real-exit</a>, Previous: <a href="#set_002dmolecule_002dname" accesskey="p" rel="previous">set-molecule-name</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot_002dreal_002dexit-1"></a>
<h4 class="subsection">11.6.7 coot-real-exit</h4>
<dl>
<dt><a name="index-coot_002dreal_002dexit"></a>function: <strong>coot-real-exit</strong> <em>retval</em></dt>
<dd>
<p>Where <em>retval</em> is an integer number
</p>
<p>exit from coot, give return value retval back to invoking process. 
</p>
</dd></dl>

<hr>
<a name="coot_002dno_002dstate_002dreal_002dexit"></a>
<div class="header">
<p>
Next: <a href="#coot_002dclear_002dbackup_002dor_002dreal_002dexit" accesskey="n" rel="next">coot-clear-backup-or-real-exit</a>, Previous: <a href="#coot_002dreal_002dexit" accesskey="p" rel="previous">coot-real-exit</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot_002dno_002dstate_002dreal_002dexit-1"></a>
<h4 class="subsection">11.6.8 coot-no-state-real-exit</h4>
<dl>
<dt><a name="index-coot_002dno_002dstate_002dreal_002dexit"></a>function: <strong>coot-no-state-real-exit</strong> <em>retval</em></dt>
<dd>
<p>Where <em>retval</em> is an integer number
</p>
<p>exit without writing a state file 
</p>
</dd></dl>

<hr>
<a name="coot_002dclear_002dbackup_002dor_002dreal_002dexit"></a>
<div class="header">
<p>
Next: <a href="#coot_002dsave_002dstate_002dand_002dexit" accesskey="n" rel="next">coot-save-state-and-exit</a>, Previous: <a href="#coot_002dno_002dstate_002dreal_002dexit" accesskey="p" rel="previous">coot-no-state-real-exit</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot_002dclear_002dbackup_002dor_002dreal_002dexit-1"></a>
<h4 class="subsection">11.6.9 coot-clear-backup-or-real-exit</h4>
<dl>
<dt><a name="index-coot_002dclear_002dbackup_002dor_002dreal_002dexit"></a>function: <strong>coot-clear-backup-or-real-exit</strong> <em>retval</em></dt>
<dd>
<p>Where <em>retval</em> is an integer number
</p>
<p>exit coot doing clear-backup maybe 
</p>
</dd></dl>

<hr>
<a name="coot_002dsave_002dstate_002dand_002dexit"></a>
<div class="header">
<p>
Next: <a href="#run_002dclear_002dbackups" accesskey="n" rel="next">run-clear-backups</a>, Previous: <a href="#coot_002dclear_002dbackup_002dor_002dreal_002dexit" accesskey="p" rel="previous">coot-clear-backup-or-real-exit</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot_002dsave_002dstate_002dand_002dexit-1"></a>
<h4 class="subsection">11.6.10 coot-save-state-and-exit</h4>
<dl>
<dt><a name="index-coot_002dsave_002dstate_002dand_002dexit"></a>function: <strong>coot-save-state-and-exit</strong> <em>retval save_state_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>retval</em> is an integer number
</li><li> <em>save_state_flag</em> is an integer number
</li></ul>

<p>exit coot, write a state file 
</p>
</dd></dl>

<hr>
<a name="run_002dclear_002dbackups"></a>
<div class="header">
<p>
Next: <a href="#first_002dcoords_002dimol" accesskey="n" rel="next">first-coords-imol</a>, Previous: <a href="#coot_002dsave_002dstate_002dand_002dexit" accesskey="p" rel="previous">coot-save-state-and-exit</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="run_002dclear_002dbackups-1"></a>
<h4 class="subsection">11.6.11 run-clear-backups</h4>
<dl>
<dt><a name="index-run_002dclear_002dbackups"></a>function: <strong>run-clear-backups</strong> <em>retval</em></dt>
<dd>
<p>Where <em>retval</em> is an integer number
</p>
<p>run clear-backups 
</p>
</dd></dl>

<hr>
<a name="first_002dcoords_002dimol"></a>
<div class="header">
<p>
Next: <a href="#first_002dsmall_002dcoords_002dimol" accesskey="n" rel="next">first-small-coords-imol</a>, Previous: <a href="#run_002dclear_002dbackups" accesskey="p" rel="previous">run-clear-backups</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="first_002dcoords_002dimol-1"></a>
<h4 class="subsection">11.6.12 first-coords-imol</h4>
<dl>
<dt><a name="index-first_002dcoords_002dimol"></a>function: <strong>first-coords-imol</strong></dt>
<dd>
<p>What is the molecule number of first coordinates molecule? 
</p>
<p>return -1 when there is none. 
</p>
</dd></dl>

<hr>
<a name="first_002dsmall_002dcoords_002dimol"></a>
<div class="header">
<p>
Next: <a href="#first_002dunsaved_002dcoords_002dimol" accesskey="n" rel="next">first-unsaved-coords-imol</a>, Previous: <a href="#first_002dcoords_002dimol" accesskey="p" rel="previous">first-coords-imol</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="first_002dsmall_002dcoords_002dimol-1"></a>
<h4 class="subsection">11.6.13 first-small-coords-imol</h4>
<dl>
<dt><a name="index-first_002dsmall_002dcoords_002dimol"></a>function: <strong>first-small-coords-imol</strong></dt>
<dd>
<p>molecule number of first small (&lt;400 atoms) molecule. 
</p>
<p>return -1 on no such molecule 
</p>
</dd></dl>

<hr>
<a name="first_002dunsaved_002dcoords_002dimol"></a>
<div class="header">
<p>
Next: <a href="#mmcif_002dsfs_002dto_002dmtz" accesskey="n" rel="next">mmcif-sfs-to-mtz</a>, Previous: <a href="#first_002dsmall_002dcoords_002dimol" accesskey="p" rel="previous">first-small-coords-imol</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="first_002dunsaved_002dcoords_002dimol-1"></a>
<h4 class="subsection">11.6.14 first-unsaved-coords-imol</h4>
<dl>
<dt><a name="index-first_002dunsaved_002dcoords_002dimol"></a>function: <strong>first-unsaved-coords-imol</strong></dt>
<dd>
<p>What is the molecule number of first unsaved coordinates molecule? 
</p>
<p>return -1 when there is none. 
</p>
</dd></dl>

<hr>
<a name="mmcif_002dsfs_002dto_002dmtz"></a>
<div class="header">
<p>
Previous: <a href="#first_002dunsaved_002dcoords_002dimol" accesskey="p" rel="previous">first-unsaved-coords-imol</a>, Up: <a href="#Library-and-Utility-Functions" accesskey="u" rel="up">Library and Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mmcif_002dsfs_002dto_002dmtz-1"></a>
<h4 class="subsection">11.6.15 mmcif-sfs-to-mtz</h4>
<dl>
<dt><a name="index-mmcif_002dsfs_002dto_002dmtz"></a>function: <strong>mmcif-sfs-to-mtz</strong> <em>cif_file_name mtz_file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>cif_file_name</em> is a string
</li><li> <em>mtz_file_name</em> is a string
</li></ul>

<p>convert the structure factors in cif_file_name to an mtz file. 
</p>
<p>Return 1 on success. Return 0 on a file without Rfree, return -1 on complete failure to write a file. 
</p>
</dd></dl>

<hr>
<a name="Graphics-Utility-Functions"></a>
<div class="header">
<p>
Next: <a href="#Interface-Preferences" accesskey="n" rel="next">Interface Preferences</a>, Previous: <a href="#Library-and-Utility-Functions" accesskey="p" rel="previous">Library and Utility Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Graphics-Utility-Functions-1"></a>
<h3 class="section">11.7 Graphics Utility Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002ddo_002danti_002daliasing" accesskey="1">set-do-anti-aliasing</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#do_002danti_002daliasing_002dstate" accesskey="2">do-anti-aliasing-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddo_002dGL_002dlighting" accesskey="3">set-do-GL-lighting</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#do_002dGL_002dlighting_002dstate" accesskey="4">do-GL-lighting-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#use_002dgraphics_002dinterface_002dstate" accesskey="5">use-graphics-interface-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#python_002dat_002dprompt_002dat_002dstartup_002dstate" accesskey="6">python-at-prompt-at-startup-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#start_002dgraphics_002dinterface" accesskey="7">start-graphics-interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#reset_002dview" accesskey="8">reset-view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dn_002dmolecules" accesskey="9">graphics-n-molecules</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#toggle_002didle_002dspin_002dfunction">toggle-idle-spin-function</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#toggle_002didle_002drock_002dfunction">toggle-idle-rock-function</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drocking_002dfactors">set-rocking-factors</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002didle_002dfunction_002drotate_002dangle">set-idle-function-rotate-angle</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#idle_002dfunction_002drotate_002dangle">idle-function-rotate-angle</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#handle_002dread_002ddraw_002dmolecule">handle-read-draw-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dupdating_002dmodel_002dmolecule">make-updating-model-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#allow_002dduplicate_002dsequence_002dnumbers">allow-duplicate-sequence-numbers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dconvert_002dto_002dv2_002datom_002dnames">set-convert-to-v2-atom-names</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre">handle-read-draw-molecule-with-recentre</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere">handle-read-draw-molecule-and-move-molecule-here</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#read_002dpdb">read-pdb</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#assign_002dhetatms">assign-hetatms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#hetify_002dresidue">hetify-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residue_002dhas_002dhetatms">residue-has-hetatms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#het_002dgroup_002dresidues_002dscm">het-group-residues-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#het_002dgroup_002dresidues_002dpy">het-group-residues-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#het_002dgroup_002dn_002datoms">het-group-n-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#replace_002dfragment">replace-fragment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#copy_002dresidue_002drange">copy-residue-range</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#replace_002dresidues_002dfrom_002dmol_002dscm">replace-residues-from-mol-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile">clear-and-update-model-molecule-from-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#screendump_002dimage">screendump-image</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#check_002dfor_002ddark_002dblue_002ddensity">check-for-dark-blue-density</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dsolid_002ddensity_002dsurface">set-draw-solid-density-surface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dmap_002dstandard_002dlines">set-draw-map-standard-lines</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsolid_002ddensity_002dsurface_002dopacity">set-solid-density-surface-opacity</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface">set-flat-shading-for-solid-density-surface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002ddo_002danti_002daliasing"></a>
<div class="header">
<p>
Next: <a href="#do_002danti_002daliasing_002dstate" accesskey="n" rel="next">do-anti-aliasing-state</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddo_002danti_002daliasing-1"></a>
<h4 class="subsection">11.7.1 set-do-anti-aliasing</h4>
<dl>
<dt><a name="index-set_002ddo_002danti_002daliasing"></a>function: <strong>set-do-anti-aliasing</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set the bond lines to be antialiased 
</p>
</dd></dl>

<hr>
<a name="do_002danti_002daliasing_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002ddo_002dGL_002dlighting" accesskey="n" rel="next">set-do-GL-lighting</a>, Previous: <a href="#set_002ddo_002danti_002daliasing" accesskey="p" rel="previous">set-do-anti-aliasing</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002danti_002daliasing_002dstate-1"></a>
<h4 class="subsection">11.7.2 do-anti-aliasing-state</h4>
<dl>
<dt><a name="index-do_002danti_002daliasing_002dstate"></a>function: <strong>do-anti-aliasing-state</strong></dt>
<dd>
<p>return the flag for antialiasing the bond lines 
</p>
</dd></dl>

<hr>
<a name="set_002ddo_002dGL_002dlighting"></a>
<div class="header">
<p>
Next: <a href="#do_002dGL_002dlighting_002dstate" accesskey="n" rel="next">do-GL-lighting-state</a>, Previous: <a href="#do_002danti_002daliasing_002dstate" accesskey="p" rel="previous">do-anti-aliasing-state</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddo_002dGL_002dlighting-1"></a>
<h4 class="subsection">11.7.3 set-do-GL-lighting</h4>
<dl>
<dt><a name="index-set_002ddo_002dGL_002dlighting"></a>function: <strong>set-do-GL-lighting</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>turn the GL lighting on (state = 1) or off (state = 0) 
</p>
<p>slows down the display of simple lines 
</p>
</dd></dl>

<hr>
<a name="do_002dGL_002dlighting_002dstate"></a>
<div class="header">
<p>
Next: <a href="#use_002dgraphics_002dinterface_002dstate" accesskey="n" rel="next">use-graphics-interface-state</a>, Previous: <a href="#set_002ddo_002dGL_002dlighting" accesskey="p" rel="previous">set-do-GL-lighting</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dGL_002dlighting_002dstate-1"></a>
<h4 class="subsection">11.7.4 do-GL-lighting-state</h4>
<dl>
<dt><a name="index-do_002dGL_002dlighting_002dstate"></a>function: <strong>do-GL-lighting-state</strong></dt>
<dd>
<p>return the flag for GL lighting 
</p>
</dd></dl>

<hr>
<a name="use_002dgraphics_002dinterface_002dstate"></a>
<div class="header">
<p>
Next: <a href="#python_002dat_002dprompt_002dat_002dstartup_002dstate" accesskey="n" rel="next">python-at-prompt-at-startup-state</a>, Previous: <a href="#do_002dGL_002dlighting_002dstate" accesskey="p" rel="previous">do-GL-lighting-state</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="use_002dgraphics_002dinterface_002dstate-1"></a>
<h4 class="subsection">11.7.5 use-graphics-interface-state</h4>
<dl>
<dt><a name="index-use_002dgraphics_002dinterface_002dstate"></a>function: <strong>use-graphics-interface-state</strong></dt>
<dd>
<p>shall we start up the Gtk and the graphics window? 
</p>
<p>if passed the command line argument 
</p>
<p>no-graphics, coot will not start up gtk itself.
</p>
<p>An interface function for Ralf. 
</p>
</dd></dl>

<hr>
<a name="python_002dat_002dprompt_002dat_002dstartup_002dstate"></a>
<div class="header">
<p>
Next: <a href="#start_002dgraphics_002dinterface" accesskey="n" rel="next">start-graphics-interface</a>, Previous: <a href="#use_002dgraphics_002dinterface_002dstate" accesskey="p" rel="previous">use-graphics-interface-state</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="python_002dat_002dprompt_002dat_002dstartup_002dstate-1"></a>
<h4 class="subsection">11.7.6 python-at-prompt-at-startup-state</h4>
<dl>
<dt><a name="index-python_002dat_002dprompt_002dat_002dstartup_002dstate"></a>function: <strong>python-at-prompt-at-startup-state</strong></dt>
<dd>
<p>is the python interpreter at the prompt? 
</p>
<p>Returns: 1 for yes, 0 for no. 
</p>
</dd></dl>

<hr>
<a name="start_002dgraphics_002dinterface"></a>
<div class="header">
<p>
Next: <a href="#reset_002dview" accesskey="n" rel="next">reset-view</a>, Previous: <a href="#python_002dat_002dprompt_002dat_002dstartup_002dstate" accesskey="p" rel="previous">python-at-prompt-at-startup-state</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="start_002dgraphics_002dinterface-1"></a>
<h4 class="subsection">11.7.7 start-graphics-interface</h4>
<dl>
<dt><a name="index-start_002dgraphics_002dinterface"></a>function: <strong>start-graphics-interface</strong></dt>
<dd>
<p>start Gtk (and graphics) 
</p>
<p>This function is useful if it was not started already (which can be achieved by using the command line argument 
</p>
<p>no-graphics).
</p>
<p>An interface for Ralf 
</p>
</dd></dl>

<hr>
<a name="reset_002dview"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dn_002dmolecules" accesskey="n" rel="next">graphics-n-molecules</a>, Previous: <a href="#start_002dgraphics_002dinterface" accesskey="p" rel="previous">start-graphics-interface</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="reset_002dview-1"></a>
<h4 class="subsection">11.7.8 reset-view</h4>
<dl>
<dt><a name="index-reset_002dview"></a>function: <strong>reset-view</strong></dt>
<dd>
<p>&quot;Reset&quot; the view 
</p>
<p>return 1 if we moved, else return 0.
</p>
<p>centre on last-read molecule with zoom 100. If we are there, then go to the previous molecule, if we are there, then go to the origin. 
</p>
</dd></dl>

<hr>
<a name="graphics_002dn_002dmolecules"></a>
<div class="header">
<p>
Next: <a href="#toggle_002didle_002dspin_002dfunction" accesskey="n" rel="next">toggle-idle-spin-function</a>, Previous: <a href="#reset_002dview" accesskey="p" rel="previous">reset-view</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dn_002dmolecules-1"></a>
<h4 class="subsection">11.7.9 graphics-n-molecules</h4>
<dl>
<dt><a name="index-graphics_002dn_002dmolecules"></a>function: <strong>graphics-n-molecules</strong></dt>
<dd>
<p>return the number of molecules (coordinates molecules and map molecules combined) that are currently in coot 
</p>
<p>Returns: the number of molecules (closed molecules are not counted) 
</p>
</dd></dl>

<hr>
<a name="toggle_002didle_002dspin_002dfunction"></a>
<div class="header">
<p>
Next: <a href="#toggle_002didle_002drock_002dfunction" accesskey="n" rel="next">toggle-idle-rock-function</a>, Previous: <a href="#graphics_002dn_002dmolecules" accesskey="p" rel="previous">graphics-n-molecules</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="toggle_002didle_002dspin_002dfunction-1"></a>
<h4 class="subsection">11.7.10 toggle-idle-spin-function</h4>
<dl>
<dt><a name="index-toggle_002didle_002dspin_002dfunction"></a>function: <strong>toggle-idle-spin-function</strong></dt>
<dd>
<p>Spin spin spin (or not) 
</p>
</dd></dl>

<hr>
<a name="toggle_002didle_002drock_002dfunction"></a>
<div class="header">
<p>
Next: <a href="#set_002drocking_002dfactors" accesskey="n" rel="next">set-rocking-factors</a>, Previous: <a href="#toggle_002didle_002dspin_002dfunction" accesskey="p" rel="previous">toggle-idle-spin-function</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="toggle_002didle_002drock_002dfunction-1"></a>
<h4 class="subsection">11.7.11 toggle-idle-rock-function</h4>
<dl>
<dt><a name="index-toggle_002didle_002drock_002dfunction"></a>function: <strong>toggle-idle-rock-function</strong></dt>
<dd>
<p>Rock (not roll) (self-timed) 
</p>
</dd></dl>

<hr>
<a name="set_002drocking_002dfactors"></a>
<div class="header">
<p>
Next: <a href="#set_002didle_002dfunction_002drotate_002dangle" accesskey="n" rel="next">set-idle-function-rotate-angle</a>, Previous: <a href="#toggle_002didle_002drock_002dfunction" accesskey="p" rel="previous">toggle-idle-rock-function</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drocking_002dfactors-1"></a>
<h4 class="subsection">11.7.12 set-rocking-factors</h4>
<dl>
<dt><a name="index-set_002drocking_002dfactors"></a>function: <strong>set-rocking-factors</strong> <em>width_scale frequency_scale</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>width_scale</em> is a number
</li><li> <em>frequency_scale</em> is a number
</li></ul>

<p>Settings for the inevitable discontents who dislike the default rocking rates (defaults 1 and 1) 
</p>
</dd></dl>

<hr>
<a name="set_002didle_002dfunction_002drotate_002dangle"></a>
<div class="header">
<p>
Next: <a href="#idle_002dfunction_002drotate_002dangle" accesskey="n" rel="next">idle-function-rotate-angle</a>, Previous: <a href="#set_002drocking_002dfactors" accesskey="p" rel="previous">set-rocking-factors</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002didle_002dfunction_002drotate_002dangle-1"></a>
<h4 class="subsection">11.7.13 set-idle-function-rotate-angle</h4>
<dl>
<dt><a name="index-set_002didle_002dfunction_002drotate_002dangle"></a>function: <strong>set-idle-function-rotate-angle</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>how far should we rotate when (auto) spinning? Fast computer? set this to 0.1 
</p>
</dd></dl>

<hr>
<a name="idle_002dfunction_002drotate_002dangle"></a>
<div class="header">
<p>
Next: <a href="#handle_002dread_002ddraw_002dmolecule" accesskey="n" rel="next">handle-read-draw-molecule</a>, Previous: <a href="#set_002didle_002dfunction_002drotate_002dangle" accesskey="p" rel="previous">set-idle-function-rotate-angle</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="idle_002dfunction_002drotate_002dangle-1"></a>
<h4 class="subsection">11.7.14 idle-function-rotate-angle</h4>
<dl>
<dt><a name="index-idle_002dfunction_002drotate_002dangle"></a>function: <strong>idle-function-rotate-angle</strong></dt>
<dd>
<p>what is the idle function rotation angle? 
</p>
</dd></dl>

<hr>
<a name="handle_002dread_002ddraw_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#make_002dupdating_002dmodel_002dmolecule" accesskey="n" rel="next">make-updating-model-molecule</a>, Previous: <a href="#idle_002dfunction_002drotate_002dangle" accesskey="p" rel="previous">idle-function-rotate-angle</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002dread_002ddraw_002dmolecule-1"></a>
<h4 class="subsection">11.7.15 handle-read-draw-molecule</h4>
<dl>
<dt><a name="index-handle_002dread_002ddraw_002dmolecule"></a>function: <strong>handle-read-draw-molecule</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>a synonym for read-pdb. Read the coordinates from filename (can be pdb, cif or shelx format) 
</p>
</dd></dl>

<hr>
<a name="make_002dupdating_002dmodel_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#allow_002dduplicate_002dsequence_002dnumbers" accesskey="n" rel="next">allow-duplicate-sequence-numbers</a>, Previous: <a href="#handle_002dread_002ddraw_002dmolecule" accesskey="p" rel="previous">handle-read-draw-molecule</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dupdating_002dmodel_002dmolecule-1"></a>
<h4 class="subsection">11.7.16 make-updating-model-molecule</h4>
<dl>
<dt><a name="index-make_002dupdating_002dmodel_002dmolecule"></a>function: <strong>make-updating-model-molecule</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>make a model molecule from the give file name. 
</p>
<p>If the file updates, then the model will be updated. 
</p>
</dd></dl>

<hr>
<a name="allow_002dduplicate_002dsequence_002dnumbers"></a>
<div class="header">
<p>
Next: <a href="#set_002dconvert_002dto_002dv2_002datom_002dnames" accesskey="n" rel="next">set-convert-to-v2-atom-names</a>, Previous: <a href="#make_002dupdating_002dmodel_002dmolecule" accesskey="p" rel="previous">make-updating-model-molecule</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="allow_002dduplicate_002dsequence_002dnumbers-1"></a>
<h4 class="subsection">11.7.17 allow-duplicate-sequence-numbers</h4>
<dl>
<dt><a name="index-allow_002dduplicate_002dsequence_002dnumbers"></a>function: <strong>allow-duplicate-sequence-numbers</strong></dt>
<dd>
<p>enable reading PDB/pdbx files with duplicate sequence numbers 
</p>
</dd></dl>

<hr>
<a name="set_002dconvert_002dto_002dv2_002datom_002dnames"></a>
<div class="header">
<p>
Next: <a href="#handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre" accesskey="n" rel="next">handle-read-draw-molecule-with-recentre</a>, Previous: <a href="#allow_002dduplicate_002dsequence_002dnumbers" accesskey="p" rel="previous">allow-duplicate-sequence-numbers</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dconvert_002dto_002dv2_002datom_002dnames-1"></a>
<h4 class="subsection">11.7.18 set-convert-to-v2-atom-names</h4>
<dl>
<dt><a name="index-set_002dconvert_002dto_002dv2_002datom_002dnames"></a>function: <strong>set-convert-to-v2-atom-names</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>shall we convert nucleotides to match the old dictionary names? 
</p>
<p>Usually (after 2006 or so) we do not want to do this (given current Coot architecture). Coot should handle the residue synonyms transparently.
</p>
<p>default off (0). 
</p>
</dd></dl>

<hr>
<a name="handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre"></a>
<div class="header">
<p>
Next: <a href="#handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere" accesskey="n" rel="next">handle-read-draw-molecule-and-move-molecule-here</a>, Previous: <a href="#set_002dconvert_002dto_002dv2_002datom_002dnames" accesskey="p" rel="previous">set-convert-to-v2-atom-names</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre-1"></a>
<h4 class="subsection">11.7.19 handle-read-draw-molecule-with-recentre</h4>
<dl>
<dt><a name="index-handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre"></a>function: <strong>handle-read-draw-molecule-with-recentre</strong> <em>filename recentre_on_read_pdb_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>filename</em> is a string
</li><li> <em>recentre_on_read_pdb_flag</em> is an integer number
</li></ul>

<p>read coordinates from filename with option to not recentre. 
</p>
<p>set recentre_on_read_pdb_flag to 0 if you don&rsquo;t want the view to recentre on the new coordinates. 
</p>
</dd></dl>

<hr>
<a name="handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere"></a>
<div class="header">
<p>
Next: <a href="#read_002dpdb" accesskey="n" rel="next">read-pdb</a>, Previous: <a href="#handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre" accesskey="p" rel="previous">handle-read-draw-molecule-with-recentre</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere-1"></a>
<h4 class="subsection">11.7.20 handle-read-draw-molecule-and-move-molecule-here</h4>
<dl>
<dt><a name="index-handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere"></a>function: <strong>handle-read-draw-molecule-and-move-molecule-here</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>read coordinates from filename and recentre the new molecule at the screen rotation centre. 
</p>
</dd></dl>

<hr>
<a name="read_002dpdb"></a>
<div class="header">
<p>
Next: <a href="#assign_002dhetatms" accesskey="n" rel="next">assign-hetatms</a>, Previous: <a href="#handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere" accesskey="p" rel="previous">handle-read-draw-molecule-and-move-molecule-here</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="read_002dpdb-1"></a>
<h4 class="subsection">11.7.21 read-pdb</h4>
<dl>
<dt><a name="index-read_002dpdb"></a>function: <strong>read-pdb</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>read coordinates from filename 
</p>
</dd></dl>

<hr>
<a name="assign_002dhetatms"></a>
<div class="header">
<p>
Next: <a href="#hetify_002dresidue" accesskey="n" rel="next">hetify-residue</a>, Previous: <a href="#read_002dpdb" accesskey="p" rel="previous">read-pdb</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="assign_002dhetatms-1"></a>
<h4 class="subsection">11.7.22 assign-hetatms</h4>
<dl>
<dt><a name="index-assign_002dhetatms"></a>function: <strong>assign-hetatms</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>some programs produce PDB files with ATOMs where there should be HETATMs. This is a function to assign HETATMs as per the PDB definition. 
</p>
</dd></dl>

<hr>
<a name="hetify_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#residue_002dhas_002dhetatms" accesskey="n" rel="next">residue-has-hetatms</a>, Previous: <a href="#assign_002dhetatms" accesskey="p" rel="previous">assign-hetatms</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="hetify_002dresidue-1"></a>
<h4 class="subsection">11.7.23 hetify-residue</h4>
<dl>
<dt><a name="index-hetify_002dresidue"></a>function: <strong>hetify-residue</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>if this is not a standard group, then turn the atoms to HETATMs. 
</p>
<p>Return 1 on atoms changes, 0 on not. Return -1 if residue not found. 
</p>
</dd></dl>

<hr>
<a name="residue_002dhas_002dhetatms"></a>
<div class="header">
<p>
Next: <a href="#het_002dgroup_002dresidues_002dscm" accesskey="n" rel="next">het-group-residues-scm</a>, Previous: <a href="#hetify_002dresidue" accesskey="p" rel="previous">hetify-residue</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residue_002dhas_002dhetatms-1"></a>
<h4 class="subsection">11.7.24 residue-has-hetatms</h4>
<dl>
<dt><a name="index-residue_002dhas_002dhetatms"></a>function: <strong>residue-has-hetatms</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>residue has HETATMs? 
</p>
<p>return 1 if all atoms of the specified residue are HETATMs, else, return 0. If residue not found, return -1. 
</p>
</dd></dl>

<hr>
<a name="het_002dgroup_002dresidues_002dscm"></a>
<div class="header">
<p>
Next: <a href="#het_002dgroup_002dresidues_002dpy" accesskey="n" rel="next">het-group-residues-py</a>, Previous: <a href="#residue_002dhas_002dhetatms" accesskey="p" rel="previous">residue-has-hetatms</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="het_002dgroup_002dresidues_002dscm-1"></a>
<h4 class="subsection">11.7.25 het-group-residues-scm</h4>
<dl>
<dt><a name="index-het_002dgroup_002dresidues_002dscm"></a>function: <strong>het-group-residues-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>get the specs for hetgroups - waters are not counted as het-groups. 
</p>
</dd></dl>

<hr>
<a name="het_002dgroup_002dresidues_002dpy"></a>
<div class="header">
<p>
Next: <a href="#het_002dgroup_002dn_002datoms" accesskey="n" rel="next">het-group-n-atoms</a>, Previous: <a href="#het_002dgroup_002dresidues_002dscm" accesskey="p" rel="previous">het-group-residues-scm</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="het_002dgroup_002dresidues_002dpy-1"></a>
<h4 class="subsection">11.7.26 het-group-residues-py</h4>
<dl>
<dt><a name="index-het_002dgroup_002dresidues_002dpy"></a>function: <strong>het-group-residues-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>get the specs for hetgroups - waters are not counted as het-groups. 
</p>
</dd></dl>

<hr>
<a name="het_002dgroup_002dn_002datoms"></a>
<div class="header">
<p>
Next: <a href="#replace_002dfragment" accesskey="n" rel="next">replace-fragment</a>, Previous: <a href="#het_002dgroup_002dresidues_002dpy" accesskey="p" rel="previous">het-group-residues-py</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="het_002dgroup_002dn_002datoms-1"></a>
<h4 class="subsection">11.7.27 het-group-n-atoms</h4>
<dl>
<dt><a name="index-het_002dgroup_002dn_002datoms"></a>function: <strong>het-group-n-atoms</strong> <em>comp_id</em></dt>
<dd>
<p>Where <em>comp_id</em> is a string
</p>
<p>return the number of non-hydrogen atoms in the given het-group (comp-id). 
</p>
<p>Return -1 on comp-id not found in dictionary. 
</p>
</dd></dl>

<hr>
<a name="replace_002dfragment"></a>
<div class="header">
<p>
Next: <a href="#copy_002dresidue_002drange" accesskey="n" rel="next">copy-residue-range</a>, Previous: <a href="#het_002dgroup_002dn_002datoms" accesskey="p" rel="previous">het-group-n-atoms</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="replace_002dfragment-1"></a>
<h4 class="subsection">11.7.28 replace-fragment</h4>
<dl>
<dt><a name="index-replace_002dfragment"></a>function: <strong>replace-fragment</strong> <em>imol_target imol_fragment atom_selection</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_target</em> is an integer number
</li><li> <em>imol_fragment</em> is an integer number
</li><li> <em>atom_selection</em> is a string
</li></ul>

<p>replace the parts of molecule number imol that are duplicated in molecule number imol_frag 
</p>
</dd></dl>

<hr>
<a name="copy_002dresidue_002drange"></a>
<div class="header">
<p>
Next: <a href="#replace_002dresidues_002dfrom_002dmol_002dscm" accesskey="n" rel="next">replace-residues-from-mol-scm</a>, Previous: <a href="#replace_002dfragment" accesskey="p" rel="previous">replace-fragment</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="copy_002dresidue_002drange-1"></a>
<h4 class="subsection">11.7.29 copy-residue-range</h4>
<dl>
<dt><a name="index-copy_002dresidue_002drange"></a>function: <strong>copy-residue-range</strong> <em>imol_target chain_id_target imol_reference chain_id_reference resno_range_start resno_range_end</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_target</em> is an integer number
</li><li> <em>chain_id_target</em> is a string
</li><li> <em>imol_reference</em> is an integer number
</li><li> <em>chain_id_reference</em> is a string
</li><li> <em>resno_range_start</em> is an integer number
</li><li> <em>resno_range_end</em> is an integer number
</li></ul>

<p>copy the given residue range from the reference chain to the target chain 
</p>
<p>resno_range_start and resno_range_end are inclusive. 
</p>
</dd></dl>

<hr>
<a name="replace_002dresidues_002dfrom_002dmol_002dscm"></a>
<div class="header">
<p>
Next: <a href="#clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile" accesskey="n" rel="next">clear-and-update-model-molecule-from-file</a>, Previous: <a href="#copy_002dresidue_002drange" accesskey="p" rel="previous">copy-residue-range</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="replace_002dresidues_002dfrom_002dmol_002dscm-1"></a>
<h4 class="subsection">11.7.30 replace-residues-from-mol-scm</h4>
<dl>
<dt><a name="index-replace_002dresidues_002dfrom_002dmol_002dscm"></a>function: <strong>replace-residues-from-mol-scm</strong> <em>imol_target imol_ref residue_specs_list_ref_scm</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_target</em> is an integer number
</li><li> <em>imol_ref</em> is an integer number
</li><li> <em>residue_specs_list_ref_scm</em> is a SCM
</li></ul>

<p>replace the given residues from the reference molecule to the target molecule 
</p>
</dd></dl>

<hr>
<a name="clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile"></a>
<div class="header">
<p>
Next: <a href="#screendump_002dimage" accesskey="n" rel="next">screendump-image</a>, Previous: <a href="#replace_002dresidues_002dfrom_002dmol_002dscm" accesskey="p" rel="previous">replace-residues-from-mol-scm</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile-1"></a>
<h4 class="subsection">11.7.31 clear-and-update-model-molecule-from-file</h4>
<dl>
<dt><a name="index-clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile"></a>function: <strong>clear-and-update-model-molecule-from-file</strong> <em>molecule_number file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>molecule_number</em> is an integer number
</li><li> <em>file_name</em> is a string
</li></ul>

<p>replace pdb. Fail if molecule_number is not a valid model molecule. Return -1 on failure. Else return molecule_number 
</p>
</dd></dl>

<hr>
<a name="screendump_002dimage"></a>
<div class="header">
<p>
Next: <a href="#check_002dfor_002ddark_002dblue_002ddensity" accesskey="n" rel="next">check-for-dark-blue-density</a>, Previous: <a href="#clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile" accesskey="p" rel="previous">clear-and-update-model-molecule-from-file</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="screendump_002dimage-1"></a>
<h4 class="subsection">11.7.32 screendump-image</h4>
<dl>
<dt><a name="index-screendump_002dimage"></a>function: <strong>screendump-image</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>dump the current screen image to a file. Format ppm 
</p>
<p>You can use this, in conjunction with spinning and view moving functions to make movies 
</p>
</dd></dl>

<hr>
<a name="check_002dfor_002ddark_002dblue_002ddensity"></a>
<div class="header">
<p>
Next: <a href="#set_002ddraw_002dsolid_002ddensity_002dsurface" accesskey="n" rel="next">set-draw-solid-density-surface</a>, Previous: <a href="#screendump_002dimage" accesskey="p" rel="previous">screendump-image</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="check_002dfor_002ddark_002dblue_002ddensity-1"></a>
<h4 class="subsection">11.7.33 check-for-dark-blue-density</h4>
<dl>
<dt><a name="index-check_002dfor_002ddark_002dblue_002ddensity"></a>function: <strong>check-for-dark-blue-density</strong></dt>
<dd>
<p>give a warning dialog if density it too dark (blue) 
</p>
</dd></dl>

<hr>
<a name="set_002ddraw_002dsolid_002ddensity_002dsurface"></a>
<div class="header">
<p>
Next: <a href="#set_002ddraw_002dmap_002dstandard_002dlines" accesskey="n" rel="next">set-draw-map-standard-lines</a>, Previous: <a href="#check_002dfor_002ddark_002dblue_002ddensity" accesskey="p" rel="previous">check-for-dark-blue-density</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dsolid_002ddensity_002dsurface-1"></a>
<h4 class="subsection">11.7.34 set-draw-solid-density-surface</h4>
<dl>
<dt><a name="index-set_002ddraw_002dsolid_002ddensity_002dsurface"></a>function: <strong>set-draw-solid-density-surface</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>sets the density map of the given molecule to be drawn as a (transparent) solid surface. 
</p>
</dd></dl>

<hr>
<a name="set_002ddraw_002dmap_002dstandard_002dlines"></a>
<div class="header">
<p>
Next: <a href="#set_002dsolid_002ddensity_002dsurface_002dopacity" accesskey="n" rel="next">set-solid-density-surface-opacity</a>, Previous: <a href="#set_002ddraw_002dsolid_002ddensity_002dsurface" accesskey="p" rel="previous">set-draw-solid-density-surface</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dmap_002dstandard_002dlines-1"></a>
<h4 class="subsection">11.7.35 set-draw-map-standard-lines</h4>
<dl>
<dt><a name="index-set_002ddraw_002dmap_002dstandard_002dlines"></a>function: <strong>set-draw-map-standard-lines</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>toggle for standard lines representation of map. 
</p>
<p>This turns off/on standard lines representation of map. transparent surface is another representation type.
</p>
<p>If you want to just turn off a map, don&rsquo;t use this, use 
</p>
<p>. 
</p>
</dd></dl>

<hr>
<a name="set_002dsolid_002ddensity_002dsurface_002dopacity"></a>
<div class="header">
<p>
Next: <a href="#set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface" accesskey="n" rel="next">set-flat-shading-for-solid-density-surface</a>, Previous: <a href="#set_002ddraw_002dmap_002dstandard_002dlines" accesskey="p" rel="previous">set-draw-map-standard-lines</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsolid_002ddensity_002dsurface_002dopacity-1"></a>
<h4 class="subsection">11.7.36 set-solid-density-surface-opacity</h4>
<dl>
<dt><a name="index-set_002dsolid_002ddensity_002dsurface_002dopacity"></a>function: <strong>set-solid-density-surface-opacity</strong> <em>imol opacity</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>opacity</em> is a number
</li></ul>

<p>set the opacity of density surface representation of the given map. 
</p>
<p>0.0 is totally transparent, 1.0 is completely opaque and (because the objects are no longer depth sorted) considerably faster to render. 0.3 is a reasonable number. 
</p>
</dd></dl>

<hr>
<a name="set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface"></a>
<div class="header">
<p>
Previous: <a href="#set_002dsolid_002ddensity_002dsurface_002dopacity" accesskey="p" rel="previous">set-solid-density-surface-opacity</a>, Up: <a href="#Graphics-Utility-Functions" accesskey="u" rel="up">Graphics Utility Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface-1"></a>
<h4 class="subsection">11.7.37 set-flat-shading-for-solid-density-surface</h4>
<dl>
<dt><a name="index-set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface"></a>function: <strong>set-flat-shading-for-solid-density-surface</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set the flag to do flat shading rather than smooth shading for solid density surface. 
</p>
<p>Default is 1 (on. 
</p>
</dd></dl>

<hr>
<a name="Interface-Preferences"></a>
<div class="header">
<p>
Next: <a href="#Mouse-Buttons" accesskey="n" rel="next">Mouse Buttons</a>, Previous: <a href="#Graphics-Utility-Functions" accesskey="p" rel="previous">Graphics Utility Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Interface-Preferences-1"></a>
<h3 class="section">11.8 Interface Preferences</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dscroll_002dby_002dwheel_002dmouse" accesskey="1">set-scroll-by-wheel-mouse</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#scroll_002dby_002dwheel_002dmouse_002dstate" accesskey="2">scroll-by-wheel-mouse-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dauto_002drecontour_002dmap" accesskey="3">set-auto-recontour-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dauto_002drecontour_002dmap" accesskey="4">get-auto-recontour-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap" accesskey="5">set-default-initial-contour-level-for-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap" accesskey="6">set-default-initial-contour-level-for-difference-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#print_002dview_002dmatrix" accesskey="7">print-view-matrix</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dview_002dquaternion_002dinternal" accesskey="8">get-view-quaternion-internal</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dview_002dquaternion" accesskey="9">set-view-quaternion</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#apply_002dncs_002dto_002dview_002dorientation">apply-ncs-to-view-orientation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre">apply-ncs-to-view-orientation-and-screen-centre</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfps_002dflag">set-fps-flag</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dfps_002dflag">get-fps-flag</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dorigin_002dmarker">set-show-origin-marker</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dorigin_002dmarker_002dstate">show-origin-marker-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#hide_002dmodelling_002dtoolbar">hide-modelling-toolbar</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dmodelling_002dtoolbar">show-modelling-toolbar</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#hide_002dmain_002dtoolbar">hide-main-toolbar</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dmain_002dtoolbar">show-main-toolbar</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dmodel_002dtoolbar_002dall_002dicons">show-model-toolbar-all-icons</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dmodel_002dtoolbar_002dmain_002dicons">show-model-toolbar-main-icons</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#reattach_002dmodelling_002dtoolbar">reattach-modelling-toolbar</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmodel_002dtoolbar_002ddocked_002dposition">set-model-toolbar-docked-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#suck_002dmodel_002dfit_002ddialog">suck-model-fit-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dstatus_002dbar_002dtext">add-status-bar-text</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop">set-model-fit-refine-dialog-stays-on-top</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate">model-fit-refine-dialog-stays-on-top-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#accept_002dreject_002ddialog_002ddocked_002dstate">accept-reject-dialog-docked-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002daccept_002dreject_002ddialog_002ddocked_002dshow">set-accept-reject-dialog-docked-show</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#accept_002dreject_002ddialog_002ddocked_002dshow_002dstate">accept-reject-dialog-docked-show-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dscroll_002dby_002dwheel_002dmouse"></a>
<div class="header">
<p>
Next: <a href="#scroll_002dby_002dwheel_002dmouse_002dstate" accesskey="n" rel="next">scroll-by-wheel-mouse-state</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dscroll_002dby_002dwheel_002dmouse-1"></a>
<h4 class="subsection">11.8.1 set-scroll-by-wheel-mouse</h4>
<dl>
<dt><a name="index-set_002dscroll_002dby_002dwheel_002dmouse"></a>function: <strong>set-scroll-by-wheel-mouse</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>Some people (like Phil Evans) don&rsquo;t want to scroll their map with the mouse-wheel. 
</p>
<p>To turn off mouse wheel recontouring call this with istate value of 0 
</p>
</dd></dl>

<hr>
<a name="scroll_002dby_002dwheel_002dmouse_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dauto_002drecontour_002dmap" accesskey="n" rel="next">set-auto-recontour-map</a>, Previous: <a href="#set_002dscroll_002dby_002dwheel_002dmouse" accesskey="p" rel="previous">set-scroll-by-wheel-mouse</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="scroll_002dby_002dwheel_002dmouse_002dstate-1"></a>
<h4 class="subsection">11.8.2 scroll-by-wheel-mouse-state</h4>
<dl>
<dt><a name="index-scroll_002dby_002dwheel_002dmouse_002dstate"></a>function: <strong>scroll-by-wheel-mouse-state</strong></dt>
<dd>
<p>return the internal state of the scroll-wheel map contouring 
</p>
</dd></dl>

<hr>
<a name="set_002dauto_002drecontour_002dmap"></a>
<div class="header">
<p>
Next: <a href="#get_002dauto_002drecontour_002dmap" accesskey="n" rel="next">get-auto-recontour-map</a>, Previous: <a href="#scroll_002dby_002dwheel_002dmouse_002dstate" accesskey="p" rel="previous">scroll-by-wheel-mouse-state</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dauto_002drecontour_002dmap-1"></a>
<h4 class="subsection">11.8.3 set-auto-recontour-map</h4>
<dl>
<dt><a name="index-set_002dauto_002drecontour_002dmap"></a>function: <strong>set-auto-recontour-map</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>turn off (0) or on (1) auto recontouring (on screen centre change) (default it on) 
</p>
</dd></dl>

<hr>
<a name="get_002dauto_002drecontour_002dmap"></a>
<div class="header">
<p>
Next: <a href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap" accesskey="n" rel="next">set-default-initial-contour-level-for-map</a>, Previous: <a href="#set_002dauto_002drecontour_002dmap" accesskey="p" rel="previous">set-auto-recontour-map</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dauto_002drecontour_002dmap-1"></a>
<h4 class="subsection">11.8.4 get-auto-recontour-map</h4>
<dl>
<dt><a name="index-get_002dauto_002drecontour_002dmap"></a>function: <strong>get-auto-recontour-map</strong></dt>
<dd>
<p>return the auto-recontour state 
</p>
</dd></dl>

<hr>
<a name="set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap"></a>
<div class="header">
<p>
Next: <a href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap" accesskey="n" rel="next">set-default-initial-contour-level-for-difference-map</a>, Previous: <a href="#get_002dauto_002drecontour_002dmap" accesskey="p" rel="previous">get-auto-recontour-map</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap-1"></a>
<h4 class="subsection">11.8.5 set-default-initial-contour-level-for-map</h4>
<dl>
<dt><a name="index-set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap"></a>function: <strong>set-default-initial-contour-level-for-map</strong> <em>n_sigma</em></dt>
<dd>
<p>Where <em>n_sigma</em> is a number
</p>
<p>set the default inital contour for 2FoFc-style map 
</p>
<p>in sigma 
</p>
</dd></dl>

<hr>
<a name="set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap"></a>
<div class="header">
<p>
Next: <a href="#print_002dview_002dmatrix" accesskey="n" rel="next">print-view-matrix</a>, Previous: <a href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap" accesskey="p" rel="previous">set-default-initial-contour-level-for-map</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap-1"></a>
<h4 class="subsection">11.8.6 set-default-initial-contour-level-for-difference-map</h4>
<dl>
<dt><a name="index-set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap"></a>function: <strong>set-default-initial-contour-level-for-difference-map</strong> <em>n_sigma</em></dt>
<dd>
<p>Where <em>n_sigma</em> is a number
</p>
<p>set the default inital contour for FoFc-style map 
</p>
<p>in sigma 
</p>
</dd></dl>

<hr>
<a name="print_002dview_002dmatrix"></a>
<div class="header">
<p>
Next: <a href="#get_002dview_002dquaternion_002dinternal" accesskey="n" rel="next">get-view-quaternion-internal</a>, Previous: <a href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap" accesskey="p" rel="previous">set-default-initial-contour-level-for-difference-map</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="print_002dview_002dmatrix-1"></a>
<h4 class="subsection">11.8.7 print-view-matrix</h4>
<dl>
<dt><a name="index-print_002dview_002dmatrix"></a>function: <strong>print-view-matrix</strong></dt>
<dd>
<p>print the view matrix to the console, useful for molscript, perhaps 
</p>
</dd></dl>

<hr>
<a name="get_002dview_002dquaternion_002dinternal"></a>
<div class="header">
<p>
Next: <a href="#set_002dview_002dquaternion" accesskey="n" rel="next">set-view-quaternion</a>, Previous: <a href="#print_002dview_002dmatrix" accesskey="p" rel="previous">print-view-matrix</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dview_002dquaternion_002dinternal-1"></a>
<h4 class="subsection">11.8.8 get-view-quaternion-internal</h4>
<dl>
<dt><a name="index-get_002dview_002dquaternion_002dinternal"></a>function: <strong>get-view-quaternion-internal</strong> <em>element</em></dt>
<dd>
<p>Where <em>element</em> is an integer number
</p>
<p>internal function to get an element of the view quaternion. The whole quaternion is returned by the scheme function view-quaternion 
</p>
</dd></dl>

<hr>
<a name="set_002dview_002dquaternion"></a>
<div class="header">
<p>
Next: <a href="#apply_002dncs_002dto_002dview_002dorientation" accesskey="n" rel="next">apply-ncs-to-view-orientation</a>, Previous: <a href="#get_002dview_002dquaternion_002dinternal" accesskey="p" rel="previous">get-view-quaternion-internal</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dview_002dquaternion-1"></a>
<h4 class="subsection">11.8.9 set-view-quaternion</h4>
<dl>
<dt><a name="index-set_002dview_002dquaternion"></a>function: <strong>set-view-quaternion</strong> <em>i j k l</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>i</em> is a number
</li><li> <em>j</em> is a number
</li><li> <em>k</em> is a number
</li><li> <em>l</em> is a number
</li></ul>

<p>Set the view quaternion. 
</p>
</dd></dl>

<hr>
<a name="apply_002dncs_002dto_002dview_002dorientation"></a>
<div class="header">
<p>
Next: <a href="#apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre" accesskey="n" rel="next">apply-ncs-to-view-orientation-and-screen-centre</a>, Previous: <a href="#set_002dview_002dquaternion" accesskey="p" rel="previous">set-view-quaternion</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="apply_002dncs_002dto_002dview_002dorientation-1"></a>
<h4 class="subsection">11.8.10 apply-ncs-to-view-orientation</h4>
<dl>
<dt><a name="index-apply_002dncs_002dto_002dview_002dorientation"></a>function: <strong>apply-ncs-to-view-orientation</strong> <em>imol current_chain next_ncs_chain</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>current_chain</em> is a string
</li><li> <em>next_ncs_chain</em> is a string
</li></ul>

<p>Given that we are in chain current_chain, apply the NCS operator that maps current_chain on to next_ncs_chain, so that the relative view is preserved. For NCS skipping. 
</p>
</dd></dl>

<hr>
<a name="apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre"></a>
<div class="header">
<p>
Next: <a href="#set_002dfps_002dflag" accesskey="n" rel="next">set-fps-flag</a>, Previous: <a href="#apply_002dncs_002dto_002dview_002dorientation" accesskey="p" rel="previous">apply-ncs-to-view-orientation</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre-1"></a>
<h4 class="subsection">11.8.11 apply-ncs-to-view-orientation-and-screen-centre</h4>
<dl>
<dt><a name="index-apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre"></a>function: <strong>apply-ncs-to-view-orientation-and-screen-centre</strong> <em>imol current_chain next_ncs_chain forward_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>current_chain</em> is a string
</li><li> <em>next_ncs_chain</em> is a string
</li><li> <em>forward_flag</em> is an integer number
</li></ul>

<p>as above, but shift the screen centre also. 
</p>
</dd></dl>

<hr>
<a name="set_002dfps_002dflag"></a>
<div class="header">
<p>
Next: <a href="#get_002dfps_002dflag" accesskey="n" rel="next">get-fps-flag</a>, Previous: <a href="#apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre" accesskey="p" rel="previous">apply-ncs-to-view-orientation-and-screen-centre</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfps_002dflag-1"></a>
<h4 class="subsection">11.8.12 set-fps-flag</h4>
<dl>
<dt><a name="index-set_002dfps_002dflag"></a>function: <strong>set-fps-flag</strong> <em>t</em></dt>
<dd>
<p>Where <em>t</em> is an integer number
</p>
<p>set show frame-per-second flag 
</p>
</dd></dl>

<hr>
<a name="get_002dfps_002dflag"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002dorigin_002dmarker" accesskey="n" rel="next">set-show-origin-marker</a>, Previous: <a href="#set_002dfps_002dflag" accesskey="p" rel="previous">set-fps-flag</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dfps_002dflag-1"></a>
<h4 class="subsection">11.8.13 get-fps-flag</h4>
<dl>
<dt><a name="index-get_002dfps_002dflag"></a>function: <strong>get-fps-flag</strong></dt>
<dd>
<p>set the state of show frames-per-second flag 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002dorigin_002dmarker"></a>
<div class="header">
<p>
Next: <a href="#show_002dorigin_002dmarker_002dstate" accesskey="n" rel="next">show-origin-marker-state</a>, Previous: <a href="#get_002dfps_002dflag" accesskey="p" rel="previous">get-fps-flag</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dorigin_002dmarker-1"></a>
<h4 class="subsection">11.8.14 set-show-origin-marker</h4>
<dl>
<dt><a name="index-set_002dshow_002dorigin_002dmarker"></a>function: <strong>set-show-origin-marker</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>set a flag: is the origin marker to be shown? 1 for yes, 0 for no. 
</p>
</dd></dl>

<hr>
<a name="show_002dorigin_002dmarker_002dstate"></a>
<div class="header">
<p>
Next: <a href="#hide_002dmodelling_002dtoolbar" accesskey="n" rel="next">hide-modelling-toolbar</a>, Previous: <a href="#set_002dshow_002dorigin_002dmarker" accesskey="p" rel="previous">set-show-origin-marker</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dorigin_002dmarker_002dstate-1"></a>
<h4 class="subsection">11.8.15 show-origin-marker-state</h4>
<dl>
<dt><a name="index-show_002dorigin_002dmarker_002dstate"></a>function: <strong>show-origin-marker-state</strong></dt>
<dd>
<p>return the origin marker shown? state 
</p>
</dd></dl>

<hr>
<a name="hide_002dmodelling_002dtoolbar"></a>
<div class="header">
<p>
Next: <a href="#show_002dmodelling_002dtoolbar" accesskey="n" rel="next">show-modelling-toolbar</a>, Previous: <a href="#show_002dorigin_002dmarker_002dstate" accesskey="p" rel="previous">show-origin-marker-state</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="hide_002dmodelling_002dtoolbar-1"></a>
<h4 class="subsection">11.8.16 hide-modelling-toolbar</h4>
<dl>
<dt><a name="index-hide_002dmodelling_002dtoolbar"></a>function: <strong>hide-modelling-toolbar</strong></dt>
<dd>
<p>hide the vertical modelling toolbar in the GTK2 version 
</p>
</dd></dl>

<hr>
<a name="show_002dmodelling_002dtoolbar"></a>
<div class="header">
<p>
Next: <a href="#hide_002dmain_002dtoolbar" accesskey="n" rel="next">hide-main-toolbar</a>, Previous: <a href="#hide_002dmodelling_002dtoolbar" accesskey="p" rel="previous">hide-modelling-toolbar</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dmodelling_002dtoolbar-1"></a>
<h4 class="subsection">11.8.17 show-modelling-toolbar</h4>
<dl>
<dt><a name="index-show_002dmodelling_002dtoolbar"></a>function: <strong>show-modelling-toolbar</strong></dt>
<dd>
<p>show the vertical modelling toolbar in the GTK2 version (the toolbar is shown by default) 
</p>
</dd></dl>

<hr>
<a name="hide_002dmain_002dtoolbar"></a>
<div class="header">
<p>
Next: <a href="#show_002dmain_002dtoolbar" accesskey="n" rel="next">show-main-toolbar</a>, Previous: <a href="#show_002dmodelling_002dtoolbar" accesskey="p" rel="previous">show-modelling-toolbar</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="hide_002dmain_002dtoolbar-1"></a>
<h4 class="subsection">11.8.18 hide-main-toolbar</h4>
<dl>
<dt><a name="index-hide_002dmain_002dtoolbar"></a>function: <strong>hide-main-toolbar</strong></dt>
<dd>
<p>hide the horizontal main toolbar in the GTK2 version 
</p>
</dd></dl>

<hr>
<a name="show_002dmain_002dtoolbar"></a>
<div class="header">
<p>
Next: <a href="#show_002dmodel_002dtoolbar_002dall_002dicons" accesskey="n" rel="next">show-model-toolbar-all-icons</a>, Previous: <a href="#hide_002dmain_002dtoolbar" accesskey="p" rel="previous">hide-main-toolbar</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dmain_002dtoolbar-1"></a>
<h4 class="subsection">11.8.19 show-main-toolbar</h4>
<dl>
<dt><a name="index-show_002dmain_002dtoolbar"></a>function: <strong>show-main-toolbar</strong></dt>
<dd>
<p>show the horizontal main toolbar in the GTK2 version (the toolbar is shown by default) 
</p>
</dd></dl>

<hr>
<a name="show_002dmodel_002dtoolbar_002dall_002dicons"></a>
<div class="header">
<p>
Next: <a href="#show_002dmodel_002dtoolbar_002dmain_002dicons" accesskey="n" rel="next">show-model-toolbar-main-icons</a>, Previous: <a href="#show_002dmain_002dtoolbar" accesskey="p" rel="previous">show-main-toolbar</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dmodel_002dtoolbar_002dall_002dicons-1"></a>
<h4 class="subsection">11.8.20 show-model-toolbar-all-icons</h4>
<dl>
<dt><a name="index-show_002dmodel_002dtoolbar_002dall_002dicons"></a>function: <strong>show-model-toolbar-all-icons</strong></dt>
<dd>
<p>show all available icons in the modelling toolbar (same as MFR dialog) 
</p>
</dd></dl>

<hr>
<a name="show_002dmodel_002dtoolbar_002dmain_002dicons"></a>
<div class="header">
<p>
Next: <a href="#reattach_002dmodelling_002dtoolbar" accesskey="n" rel="next">reattach-modelling-toolbar</a>, Previous: <a href="#show_002dmodel_002dtoolbar_002dall_002dicons" accesskey="p" rel="previous">show-model-toolbar-all-icons</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dmodel_002dtoolbar_002dmain_002dicons-1"></a>
<h4 class="subsection">11.8.21 show-model-toolbar-main-icons</h4>
<dl>
<dt><a name="index-show_002dmodel_002dtoolbar_002dmain_002dicons"></a>function: <strong>show-model-toolbar-main-icons</strong></dt>
<dd>
<p>show only a selection of icons in the modelling toolbar 
</p>
</dd></dl>

<hr>
<a name="reattach_002dmodelling_002dtoolbar"></a>
<div class="header">
<p>
Next: <a href="#set_002dmodel_002dtoolbar_002ddocked_002dposition" accesskey="n" rel="next">set-model-toolbar-docked-position</a>, Previous: <a href="#show_002dmodel_002dtoolbar_002dmain_002dicons" accesskey="p" rel="previous">show-model-toolbar-main-icons</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="reattach_002dmodelling_002dtoolbar-1"></a>
<h4 class="subsection">11.8.22 reattach-modelling-toolbar</h4>
<dl>
<dt><a name="index-reattach_002dmodelling_002dtoolbar"></a>function: <strong>reattach-modelling-toolbar</strong></dt>
<dd>
<p>reattach the modelling toolbar to the last attached position 
</p>
</dd></dl>

<hr>
<a name="set_002dmodel_002dtoolbar_002ddocked_002dposition"></a>
<div class="header">
<p>
Next: <a href="#suck_002dmodel_002dfit_002ddialog" accesskey="n" rel="next">suck-model-fit-dialog</a>, Previous: <a href="#reattach_002dmodelling_002dtoolbar" accesskey="p" rel="previous">reattach-modelling-toolbar</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmodel_002dtoolbar_002ddocked_002dposition-1"></a>
<h4 class="subsection">11.8.23 set-model-toolbar-docked-position</h4>
<dl>
<dt><a name="index-set_002dmodel_002dtoolbar_002ddocked_002dposition"></a>function: <strong>set-model-toolbar-docked-position</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>to swap sides of the Model/Fit/Refine toolbar 0 (default) is right, 1 is left, 2 is top, 3 is bottom 
</p>
</dd></dl>

<hr>
<a name="suck_002dmodel_002dfit_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#add_002dstatus_002dbar_002dtext" accesskey="n" rel="next">add-status-bar-text</a>, Previous: <a href="#set_002dmodel_002dtoolbar_002ddocked_002dposition" accesskey="p" rel="previous">set-model-toolbar-docked-position</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="suck_002dmodel_002dfit_002ddialog-1"></a>
<h4 class="subsection">11.8.24 suck-model-fit-dialog</h4>
<dl>
<dt><a name="index-suck_002dmodel_002dfit_002ddialog"></a>function: <strong>suck-model-fit-dialog</strong></dt>
<dd>
<p>reparent the Model/Fit/Refine dialog so that it becomes part of the main window, next to the GL graphics context 
</p>
</dd></dl>

<hr>
<a name="add_002dstatus_002dbar_002dtext"></a>
<div class="header">
<p>
Next: <a href="#set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop" accesskey="n" rel="next">set-model-fit-refine-dialog-stays-on-top</a>, Previous: <a href="#suck_002dmodel_002dfit_002ddialog" accesskey="p" rel="previous">suck-model-fit-dialog</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dstatus_002dbar_002dtext-1"></a>
<h4 class="subsection">11.8.25 add-status-bar-text</h4>
<dl>
<dt><a name="index-add_002dstatus_002dbar_002dtext"></a>function: <strong>add-status-bar-text</strong> <em>s</em></dt>
<dd>
<p>Where <em>s</em> is a string
</p>
<p>Put text s into the status bar. 
</p>
<p>use this to put info for the user in the statusbar (less intrusive than popup). 
</p>
</dd></dl>

<hr>
<a name="set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop"></a>
<div class="header">
<p>
Next: <a href="#model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate" accesskey="n" rel="next">model-fit-refine-dialog-stays-on-top-state</a>, Previous: <a href="#add_002dstatus_002dbar_002dtext" accesskey="p" rel="previous">add-status-bar-text</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop-1"></a>
<h4 class="subsection">11.8.26 set-model-fit-refine-dialog-stays-on-top</h4>
<dl>
<dt><a name="index-set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop"></a>function: <strong>set-model-fit-refine-dialog-stays-on-top</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>model-fit-refine dialog stays on top 
</p>
</dd></dl>

<hr>
<a name="model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate"></a>
<div class="header">
<p>
Next: <a href="#accept_002dreject_002ddialog_002ddocked_002dstate" accesskey="n" rel="next">accept-reject-dialog-docked-state</a>, Previous: <a href="#set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop" accesskey="p" rel="previous">set-model-fit-refine-dialog-stays-on-top</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate-1"></a>
<h4 class="subsection">11.8.27 model-fit-refine-dialog-stays-on-top-state</h4>
<dl>
<dt><a name="index-model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate"></a>function: <strong>model-fit-refine-dialog-stays-on-top-state</strong></dt>
<dd>
<p>return the state model-fit-refine dialog stays on top 
</p>
</dd></dl>

<hr>
<a name="accept_002dreject_002ddialog_002ddocked_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002daccept_002dreject_002ddialog_002ddocked_002dshow" accesskey="n" rel="next">set-accept-reject-dialog-docked-show</a>, Previous: <a href="#model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate" accesskey="p" rel="previous">model-fit-refine-dialog-stays-on-top-state</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="accept_002dreject_002ddialog_002ddocked_002dstate-1"></a>
<h4 class="subsection">11.8.28 accept-reject-dialog-docked-state</h4>
<dl>
<dt><a name="index-accept_002dreject_002ddialog_002ddocked_002dstate"></a>function: <strong>accept-reject-dialog-docked-state</strong></dt>
<dd>
<p>the accept/reject dialog docked state 
</p>
</dd></dl>

<hr>
<a name="set_002daccept_002dreject_002ddialog_002ddocked_002dshow"></a>
<div class="header">
<p>
Next: <a href="#accept_002dreject_002ddialog_002ddocked_002dshow_002dstate" accesskey="n" rel="next">accept-reject-dialog-docked-show-state</a>, Previous: <a href="#accept_002dreject_002ddialog_002ddocked_002dstate" accesskey="p" rel="previous">accept-reject-dialog-docked-state</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002daccept_002dreject_002ddialog_002ddocked_002dshow-1"></a>
<h4 class="subsection">11.8.29 set-accept-reject-dialog-docked-show</h4>
<dl>
<dt><a name="index-set_002daccept_002dreject_002ddialog_002ddocked_002dshow"></a>function: <strong>set-accept-reject-dialog-docked-show</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set the accept/reject dialog docked show state 
</p>
</dd></dl>

<hr>
<a name="accept_002dreject_002ddialog_002ddocked_002dshow_002dstate"></a>
<div class="header">
<p>
Previous: <a href="#set_002daccept_002dreject_002ddialog_002ddocked_002dshow" accesskey="p" rel="previous">set-accept-reject-dialog-docked-show</a>, Up: <a href="#Interface-Preferences" accesskey="u" rel="up">Interface Preferences</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="accept_002dreject_002ddialog_002ddocked_002dshow_002dstate-1"></a>
<h4 class="subsection">11.8.30 accept-reject-dialog-docked-show-state</h4>
<dl>
<dt><a name="index-accept_002dreject_002ddialog_002ddocked_002dshow_002dstate"></a>function: <strong>accept-reject-dialog-docked-show-state</strong></dt>
<dd>
<p>what is the accept/reject dialog docked show state? 
</p>
</dd></dl>

<hr>
<a name="Mouse-Buttons"></a>
<div class="header">
<p>
Next: <a href="#Cursor-Function" accesskey="n" rel="next">Cursor Function</a>, Previous: <a href="#Interface-Preferences" accesskey="p" rel="previous">Interface Preferences</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Mouse-Buttons-1"></a>
<h3 class="section">11.9 Mouse Buttons</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#quanta_002dbuttons" accesskey="1">quanta-buttons</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#quanta_002dlike_002dzoom" accesskey="2">quanta-like-zoom</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcontrol_002dkey_002dfor_002drotate" accesskey="3">set-control-key-for-rotate</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#control_002dkey_002dfor_002drotate_002dstate" accesskey="4">control-key-for-rotate-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#blob_002dunder_002dpointer_002dto_002dscreen_002dcentre" accesskey="5">blob-under-pointer-to-screen-centre</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#select_002datom_002dunder_002dpointer_002dscm" accesskey="6">select-atom-under-pointer-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#select_002datom_002dunder_002dpointer_002dpy" accesskey="7">select-atom-under-pointer-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="quanta_002dbuttons"></a>
<div class="header">
<p>
Next: <a href="#quanta_002dlike_002dzoom" accesskey="n" rel="next">quanta-like-zoom</a>, Up: <a href="#Mouse-Buttons" accesskey="u" rel="up">Mouse Buttons</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="quanta_002dbuttons-1"></a>
<h4 class="subsection">11.9.1 quanta-buttons</h4>
<dl>
<dt><a name="index-quanta_002dbuttons"></a>function: <strong>quanta-buttons</strong></dt>
<dd>
<p>quanta-like buttons 
</p>
<p>Note, when you have set these, there is no way to turn them of again (other than restarting). 
</p>
</dd></dl>

<hr>
<a name="quanta_002dlike_002dzoom"></a>
<div class="header">
<p>
Next: <a href="#set_002dcontrol_002dkey_002dfor_002drotate" accesskey="n" rel="next">set-control-key-for-rotate</a>, Previous: <a href="#quanta_002dbuttons" accesskey="p" rel="previous">quanta-buttons</a>, Up: <a href="#Mouse-Buttons" accesskey="u" rel="up">Mouse Buttons</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="quanta_002dlike_002dzoom-1"></a>
<h4 class="subsection">11.9.2 quanta-like-zoom</h4>
<dl>
<dt><a name="index-quanta_002dlike_002dzoom"></a>function: <strong>quanta-like-zoom</strong></dt>
<dd>
<p>quanta-like zoom buttons 
</p>
<p>Note, when you have set these, there is no way to turn them of again (other than restarting). 
</p>
</dd></dl>

<hr>
<a name="set_002dcontrol_002dkey_002dfor_002drotate"></a>
<div class="header">
<p>
Next: <a href="#control_002dkey_002dfor_002drotate_002dstate" accesskey="n" rel="next">control-key-for-rotate-state</a>, Previous: <a href="#quanta_002dlike_002dzoom" accesskey="p" rel="previous">quanta-like-zoom</a>, Up: <a href="#Mouse-Buttons" accesskey="u" rel="up">Mouse Buttons</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcontrol_002dkey_002dfor_002drotate-1"></a>
<h4 class="subsection">11.9.3 set-control-key-for-rotate</h4>
<dl>
<dt><a name="index-set_002dcontrol_002dkey_002dfor_002drotate"></a>function: <strong>set-control-key-for-rotate</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>Alternate mode for rotation. 
</p>
<p>Prefered by some, including Dirk Kostrewa. I don&rsquo;t think this mode works properly yet 
</p>
</dd></dl>

<hr>
<a name="control_002dkey_002dfor_002drotate_002dstate"></a>
<div class="header">
<p>
Next: <a href="#blob_002dunder_002dpointer_002dto_002dscreen_002dcentre" accesskey="n" rel="next">blob-under-pointer-to-screen-centre</a>, Previous: <a href="#set_002dcontrol_002dkey_002dfor_002drotate" accesskey="p" rel="previous">set-control-key-for-rotate</a>, Up: <a href="#Mouse-Buttons" accesskey="u" rel="up">Mouse Buttons</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="control_002dkey_002dfor_002drotate_002dstate-1"></a>
<h4 class="subsection">11.9.4 control-key-for-rotate-state</h4>
<dl>
<dt><a name="index-control_002dkey_002dfor_002drotate_002dstate"></a>function: <strong>control-key-for-rotate-state</strong></dt>
<dd>
<p>return the control key rotate state 
</p>
</dd></dl>

<hr>
<a name="blob_002dunder_002dpointer_002dto_002dscreen_002dcentre"></a>
<div class="header">
<p>
Next: <a href="#select_002datom_002dunder_002dpointer_002dscm" accesskey="n" rel="next">select-atom-under-pointer-scm</a>, Previous: <a href="#control_002dkey_002dfor_002drotate_002dstate" accesskey="p" rel="previous">control-key-for-rotate-state</a>, Up: <a href="#Mouse-Buttons" accesskey="u" rel="up">Mouse Buttons</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="blob_002dunder_002dpointer_002dto_002dscreen_002dcentre-1"></a>
<h4 class="subsection">11.9.5 blob-under-pointer-to-screen-centre</h4>
<dl>
<dt><a name="index-blob_002dunder_002dpointer_002dto_002dscreen_002dcentre"></a>function: <strong>blob-under-pointer-to-screen-centre</strong></dt>
<dd>
<p>Put the blob under the cursor to the screen centre. Check only positive blobs. Useful function if bound to a key. 
</p>
<p>The refinement map must be set. (We can&rsquo;t check all maps because they are not (or may not be) on the same scale).
</p>
<p>Returns: 1 if successfully found a blob and moved there. return 0 if no move. 
</p>
</dd></dl>

<hr>
<a name="select_002datom_002dunder_002dpointer_002dscm"></a>
<div class="header">
<p>
Next: <a href="#select_002datom_002dunder_002dpointer_002dpy" accesskey="n" rel="next">select-atom-under-pointer-py</a>, Previous: <a href="#blob_002dunder_002dpointer_002dto_002dscreen_002dcentre" accesskey="p" rel="previous">blob-under-pointer-to-screen-centre</a>, Up: <a href="#Mouse-Buttons" accesskey="u" rel="up">Mouse Buttons</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="select_002datom_002dunder_002dpointer_002dscm-1"></a>
<h4 class="subsection">11.9.6 select-atom-under-pointer-scm</h4>
<dl>
<dt><a name="index-select_002datom_002dunder_002dpointer_002dscm"></a>function: <strong>select-atom-under-pointer-scm</strong></dt>
<dd>
<p>return scheme false or a list of molecule number and an atom spec 
</p>
</dd></dl>

<hr>
<a name="select_002datom_002dunder_002dpointer_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#select_002datom_002dunder_002dpointer_002dscm" accesskey="p" rel="previous">select-atom-under-pointer-scm</a>, Up: <a href="#Mouse-Buttons" accesskey="u" rel="up">Mouse Buttons</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="select_002datom_002dunder_002dpointer_002dpy-1"></a>
<h4 class="subsection">11.9.7 select-atom-under-pointer-py</h4>
<dl>
<dt><a name="index-select_002datom_002dunder_002dpointer_002dpy"></a>function: <strong>select-atom-under-pointer-py</strong></dt>
<dd>
<p>return Python false or a list of molecule number and an atom spec 
</p>
</dd></dl>

<hr>
<a name="Cursor-Function"></a>
<div class="header">
<p>
Next: <a href="#Model_002fFit_002fRefine-Functions" accesskey="n" rel="next">Model/Fit/Refine Functions</a>, Previous: <a href="#Mouse-Buttons" accesskey="p" rel="previous">Mouse Buttons</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Cursor-Function-1"></a>
<h3 class="section">11.10 Cursor Function</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#normal_002dcursor" accesskey="1">normal-cursor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fleur_002dcursor" accesskey="2">fleur-cursor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#pick_002dcursor_002dmaybe" accesskey="3">pick-cursor-maybe</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#rotate_002dcursor" accesskey="4">rotate-cursor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dpick_002dcursor_002dindex" accesskey="5">set-pick-cursor-index</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="normal_002dcursor"></a>
<div class="header">
<p>
Next: <a href="#fleur_002dcursor" accesskey="n" rel="next">fleur-cursor</a>, Up: <a href="#Cursor-Function" accesskey="u" rel="up">Cursor Function</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="normal_002dcursor-1"></a>
<h4 class="subsection">11.10.1 normal-cursor</h4>
<dl>
<dt><a name="index-normal_002dcursor"></a>function: <strong>normal-cursor</strong></dt>
<dd>
<p>normal cursor 
</p>
</dd></dl>

<hr>
<a name="fleur_002dcursor"></a>
<div class="header">
<p>
Next: <a href="#pick_002dcursor_002dmaybe" accesskey="n" rel="next">pick-cursor-maybe</a>, Previous: <a href="#normal_002dcursor" accesskey="p" rel="previous">normal-cursor</a>, Up: <a href="#Cursor-Function" accesskey="u" rel="up">Cursor Function</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fleur_002dcursor-1"></a>
<h4 class="subsection">11.10.2 fleur-cursor</h4>
<dl>
<dt><a name="index-fleur_002dcursor"></a>function: <strong>fleur-cursor</strong></dt>
<dd>
<p>fleur cursor 
</p>
</dd></dl>

<hr>
<a name="pick_002dcursor_002dmaybe"></a>
<div class="header">
<p>
Next: <a href="#rotate_002dcursor" accesskey="n" rel="next">rotate-cursor</a>, Previous: <a href="#fleur_002dcursor" accesskey="p" rel="previous">fleur-cursor</a>, Up: <a href="#Cursor-Function" accesskey="u" rel="up">Cursor Function</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="pick_002dcursor_002dmaybe-1"></a>
<h4 class="subsection">11.10.3 pick-cursor-maybe</h4>
<dl>
<dt><a name="index-pick_002dcursor_002dmaybe"></a>function: <strong>pick-cursor-maybe</strong></dt>
<dd>
<p>pick cursor maybe 
</p>
</dd></dl>

<hr>
<a name="rotate_002dcursor"></a>
<div class="header">
<p>
Next: <a href="#set_002dpick_002dcursor_002dindex" accesskey="n" rel="next">set-pick-cursor-index</a>, Previous: <a href="#pick_002dcursor_002dmaybe" accesskey="p" rel="previous">pick-cursor-maybe</a>, Up: <a href="#Cursor-Function" accesskey="u" rel="up">Cursor Function</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rotate_002dcursor-1"></a>
<h4 class="subsection">11.10.4 rotate-cursor</h4>
<dl>
<dt><a name="index-rotate_002dcursor"></a>function: <strong>rotate-cursor</strong></dt>
<dd>
<p>rotate cursor 
</p>
</dd></dl>

<hr>
<a name="set_002dpick_002dcursor_002dindex"></a>
<div class="header">
<p>
Previous: <a href="#rotate_002dcursor" accesskey="p" rel="previous">rotate-cursor</a>, Up: <a href="#Cursor-Function" accesskey="u" rel="up">Cursor Function</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dpick_002dcursor_002dindex-1"></a>
<h4 class="subsection">11.10.5 set-pick-cursor-index</h4>
<dl>
<dt><a name="index-set_002dpick_002dcursor_002dindex"></a>function: <strong>set-pick-cursor-index</strong> <em>icursor_index</em></dt>
<dd>
<p>Where <em>icursor_index</em> is an integer number
</p>
<p>let the user have a different pick cursor 
</p>
<p>sometimes (the default) GDK_CROSSHAIR is hard to see, let the user set their own 
</p>
</dd></dl>

<hr>
<a name="Model_002fFit_002fRefine-Functions"></a>
<div class="header">
<p>
Next: <a href="#Backup-Functions" accesskey="n" rel="next">Backup Functions</a>, Previous: <a href="#Cursor-Function" accesskey="p" rel="previous">Cursor Function</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Model_002fFit_002fRefine-Functions-1"></a>
<h3 class="section">11.11 Model/Fit/Refine Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#post_002dmodel_002dfit_002drefine_002ddialog" accesskey="1">post-model-fit-refine-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#unset_002dmodel_002dfit_002drefine_002ddialog" accesskey="2">unset-model-fit-refine-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#unset_002drefine_002dparams_002ddialog" accesskey="3">unset-refine-params-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dselect_002dmap_002ddialog" accesskey="4">show-select-map-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel" accesskey="5">set-model-fit-refine-rotate-translate-zone-label</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel" accesskey="6">set-model-fit-refine-place-atom-at-pointer-label</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#post_002dother_002dmodelling_002dtools_002ddialog" accesskey="7">post-other-modelling-tools-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy" accesskey="8">set-refinement-move-atoms-with-zero-occupancy</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate" accesskey="9">refinement-move-atoms-with-zero-occupancy-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="post_002dmodel_002dfit_002drefine_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#unset_002dmodel_002dfit_002drefine_002ddialog" accesskey="n" rel="next">unset-model-fit-refine-dialog</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="post_002dmodel_002dfit_002drefine_002ddialog-1"></a>
<h4 class="subsection">11.11.1 post-model-fit-refine-dialog</h4>
<dl>
<dt><a name="index-post_002dmodel_002dfit_002drefine_002ddialog"></a>function: <strong>post-model-fit-refine-dialog</strong></dt>
<dd>
<p>display the Model/Fit/Refine dialog 
</p>
</dd></dl>

<hr>
<a name="unset_002dmodel_002dfit_002drefine_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#unset_002drefine_002dparams_002ddialog" accesskey="n" rel="next">unset-refine-params-dialog</a>, Previous: <a href="#post_002dmodel_002dfit_002drefine_002ddialog" accesskey="p" rel="previous">post-model-fit-refine-dialog</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="unset_002dmodel_002dfit_002drefine_002ddialog-1"></a>
<h4 class="subsection">11.11.2 unset-model-fit-refine-dialog</h4>
<dl>
<dt><a name="index-unset_002dmodel_002dfit_002drefine_002ddialog"></a>function: <strong>unset-model-fit-refine-dialog</strong></dt>
<dd>
<p>unset model/fit/refine dialog 
</p>
</dd></dl>

<hr>
<a name="unset_002drefine_002dparams_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#show_002dselect_002dmap_002ddialog" accesskey="n" rel="next">show-select-map-dialog</a>, Previous: <a href="#unset_002dmodel_002dfit_002drefine_002ddialog" accesskey="p" rel="previous">unset-model-fit-refine-dialog</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="unset_002drefine_002dparams_002ddialog-1"></a>
<h4 class="subsection">11.11.3 unset-refine-params-dialog</h4>
<dl>
<dt><a name="index-unset_002drefine_002dparams_002ddialog"></a>function: <strong>unset-refine-params-dialog</strong></dt>
<dd>
<p>unset refine params dialog 
</p>
</dd></dl>

<hr>
<a name="show_002dselect_002dmap_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel" accesskey="n" rel="next">set-model-fit-refine-rotate-translate-zone-label</a>, Previous: <a href="#unset_002drefine_002dparams_002ddialog" accesskey="p" rel="previous">unset-refine-params-dialog</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dselect_002dmap_002ddialog-1"></a>
<h4 class="subsection">11.11.4 show-select-map-dialog</h4>
<dl>
<dt><a name="index-show_002dselect_002dmap_002ddialog"></a>function: <strong>show-select-map-dialog</strong></dt>
<dd>
<p>display the Display Manager dialog 
</p>
</dd></dl>

<hr>
<a name="set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel"></a>
<div class="header">
<p>
Next: <a href="#set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel" accesskey="n" rel="next">set-model-fit-refine-place-atom-at-pointer-label</a>, Previous: <a href="#show_002dselect_002dmap_002ddialog" accesskey="p" rel="previous">show-select-map-dialog</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel-1"></a>
<h4 class="subsection">11.11.5 set-model-fit-refine-rotate-translate-zone-label</h4>
<dl>
<dt><a name="index-set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel"></a>function: <strong>set-model-fit-refine-rotate-translate-zone-label</strong> <em>txt</em></dt>
<dd>
<p>Where <em>txt</em> is a string
</p>
<p>Allow the changing of Model/Fit/Refine button label from &quot;Rotate/Translate Zone&quot;. 
</p>
</dd></dl>

<hr>
<a name="set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel"></a>
<div class="header">
<p>
Next: <a href="#post_002dother_002dmodelling_002dtools_002ddialog" accesskey="n" rel="next">post-other-modelling-tools-dialog</a>, Previous: <a href="#set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel" accesskey="p" rel="previous">set-model-fit-refine-rotate-translate-zone-label</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel-1"></a>
<h4 class="subsection">11.11.6 set-model-fit-refine-place-atom-at-pointer-label</h4>
<dl>
<dt><a name="index-set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel"></a>function: <strong>set-model-fit-refine-place-atom-at-pointer-label</strong> <em>txt</em></dt>
<dd>
<p>Where <em>txt</em> is a string
</p>
<p>Allow the changing of Model/Fit/Refine button label from &quot;Place Atom at Pointer&quot;. 
</p>
</dd></dl>

<hr>
<a name="post_002dother_002dmodelling_002dtools_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy" accesskey="n" rel="next">set-refinement-move-atoms-with-zero-occupancy</a>, Previous: <a href="#set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel" accesskey="p" rel="previous">set-model-fit-refine-place-atom-at-pointer-label</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="post_002dother_002dmodelling_002dtools_002ddialog-1"></a>
<h4 class="subsection">11.11.7 post-other-modelling-tools-dialog</h4>
<dl>
<dt><a name="index-post_002dother_002dmodelling_002dtools_002ddialog"></a>function: <strong>post-other-modelling-tools-dialog</strong></dt>
<dd>
<p>display the Other Modelling Tools dialog 
</p>
</dd></dl>

<hr>
<a name="set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy"></a>
<div class="header">
<p>
Next: <a href="#refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate" accesskey="n" rel="next">refinement-move-atoms-with-zero-occupancy-state</a>, Previous: <a href="#post_002dother_002dmodelling_002dtools_002ddialog" accesskey="p" rel="previous">post-other-modelling-tools-dialog</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy-1"></a>
<h4 class="subsection">11.11.8 set-refinement-move-atoms-with-zero-occupancy</h4>
<dl>
<dt><a name="index-set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy"></a>function: <strong>set-refinement-move-atoms-with-zero-occupancy</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>shall atoms with zero occupancy be moved when refining? (default 1, yes) 
</p>
</dd></dl>

<hr>
<a name="refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate"></a>
<div class="header">
<p>
Previous: <a href="#set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy" accesskey="p" rel="previous">set-refinement-move-atoms-with-zero-occupancy</a>, Up: <a href="#Model_002fFit_002fRefine-Functions" accesskey="u" rel="up">Model/Fit/Refine Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate-1"></a>
<h4 class="subsection">11.11.9 refinement-move-atoms-with-zero-occupancy-state</h4>
<dl>
<dt><a name="index-refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate"></a>function: <strong>refinement-move-atoms-with-zero-occupancy-state</strong></dt>
<dd>
<p>return the state of &quot;shall atoms with zero occupancy be moved
when refining?&quot; 
</p>
</dd></dl>

<hr>
<a name="Backup-Functions"></a>
<div class="header">
<p>
Next: <a href="#Recover-Session-Function" accesskey="n" rel="next">Recover Session Function</a>, Previous: <a href="#Model_002fFit_002fRefine-Functions" accesskey="p" rel="previous">Model/Fit/Refine Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Backup-Functions-1"></a>
<h3 class="section">11.12 Backup Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#make_002dbackup" accesskey="1">make-backup</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#turn_002doff_002dbackup" accesskey="2">turn-off-backup</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#turn_002don_002dbackup" accesskey="3">turn-on-backup</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#backup_002dstate" accesskey="4">backup-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#apply_002dundo" accesskey="5">apply-undo</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#apply_002dredo" accesskey="6">apply-redo</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dhave_002dunsaved_002dchanges" accesskey="7">set-have-unsaved-changes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#have_002dunsaved_002dchanges_002dp" accesskey="8">have-unsaved-changes-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dundo_002dmolecule" accesskey="9">set-undo-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dset_002dundo_002dmolecule_002dchooser">show-set-undo-molecule-chooser</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dunpathed_002dbackup_002dfile_002dnames">set-unpathed-backup-file-names</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#unpathed_002dbackup_002dfile_002dnames_002dstate">unpathed-backup-file-names-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddecoloned_002dbackup_002dfile_002dnames">set-decoloned-backup-file-names</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#decoloned_002dbackup_002dfile_002dnames_002dstate">decoloned-backup-file-names-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#backup_002dcompress_002dfiles_002dstate">backup-compress-files-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dbackup_002dcompress_002dfiles">set-backup-compress-files</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="make_002dbackup"></a>
<div class="header">
<p>
Next: <a href="#turn_002doff_002dbackup" accesskey="n" rel="next">turn-off-backup</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dbackup-1"></a>
<h4 class="subsection">11.12.1 make-backup</h4>
<dl>
<dt><a name="index-make_002dbackup"></a>function: <strong>make-backup</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>make backup for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="turn_002doff_002dbackup"></a>
<div class="header">
<p>
Next: <a href="#turn_002don_002dbackup" accesskey="n" rel="next">turn-on-backup</a>, Previous: <a href="#make_002dbackup" accesskey="p" rel="previous">make-backup</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="turn_002doff_002dbackup-1"></a>
<h4 class="subsection">11.12.2 turn-off-backup</h4>
<dl>
<dt><a name="index-turn_002doff_002dbackup"></a>function: <strong>turn-off-backup</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>turn off backups for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="turn_002don_002dbackup"></a>
<div class="header">
<p>
Next: <a href="#backup_002dstate" accesskey="n" rel="next">backup-state</a>, Previous: <a href="#turn_002doff_002dbackup" accesskey="p" rel="previous">turn-off-backup</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="turn_002don_002dbackup-1"></a>
<h4 class="subsection">11.12.3 turn-on-backup</h4>
<dl>
<dt><a name="index-turn_002don_002dbackup"></a>function: <strong>turn-on-backup</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>turn on backups for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="backup_002dstate"></a>
<div class="header">
<p>
Next: <a href="#apply_002dundo" accesskey="n" rel="next">apply-undo</a>, Previous: <a href="#turn_002don_002dbackup" accesskey="p" rel="previous">turn-on-backup</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="backup_002dstate-1"></a>
<h4 class="subsection">11.12.4 backup-state</h4>
<dl>
<dt><a name="index-backup_002dstate"></a>function: <strong>backup-state</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the backup state for molecule number imol 
</p>
<p>return 0 for backups off, 1 for backups on, -1 for unknown 
</p>
</dd></dl>

<hr>
<a name="apply_002dundo"></a>
<div class="header">
<p>
Next: <a href="#apply_002dredo" accesskey="n" rel="next">apply-redo</a>, Previous: <a href="#backup_002dstate" accesskey="p" rel="previous">backup-state</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="apply_002dundo-1"></a>
<h4 class="subsection">11.12.5 apply-undo</h4>
<dl>
<dt><a name="index-apply_002dundo"></a>function: <strong>apply-undo</strong></dt>
<dd>
<p>apply undo - the &quot;Undo&quot; button callback 
</p>
</dd></dl>

<hr>
<a name="apply_002dredo"></a>
<div class="header">
<p>
Next: <a href="#set_002dhave_002dunsaved_002dchanges" accesskey="n" rel="next">set-have-unsaved-changes</a>, Previous: <a href="#apply_002dundo" accesskey="p" rel="previous">apply-undo</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="apply_002dredo-1"></a>
<h4 class="subsection">11.12.6 apply-redo</h4>
<dl>
<dt><a name="index-apply_002dredo"></a>function: <strong>apply-redo</strong></dt>
<dd>
<p>apply redo - the &quot;Redo&quot; button callback 
</p>
</dd></dl>

<hr>
<a name="set_002dhave_002dunsaved_002dchanges"></a>
<div class="header">
<p>
Next: <a href="#have_002dunsaved_002dchanges_002dp" accesskey="n" rel="next">have-unsaved-changes-p</a>, Previous: <a href="#apply_002dredo" accesskey="p" rel="previous">apply-redo</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dhave_002dunsaved_002dchanges-1"></a>
<h4 class="subsection">11.12.7 set-have-unsaved-changes</h4>
<dl>
<dt><a name="index-set_002dhave_002dunsaved_002dchanges"></a>function: <strong>set-have-unsaved-changes</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>set the molecule number imol to be marked as having unsaved changes 
</p>
</dd></dl>

<hr>
<a name="have_002dunsaved_002dchanges_002dp"></a>
<div class="header">
<p>
Next: <a href="#set_002dundo_002dmolecule" accesskey="n" rel="next">set-undo-molecule</a>, Previous: <a href="#set_002dhave_002dunsaved_002dchanges" accesskey="p" rel="previous">set-have-unsaved-changes</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="have_002dunsaved_002dchanges_002dp-1"></a>
<h4 class="subsection">11.12.8 have-unsaved-changes-p</h4>
<dl>
<dt><a name="index-have_002dunsaved_002dchanges_002dp"></a>function: <strong>have-unsaved-changes-p</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>does molecule number imol have unsaved changes? 
</p>
<p>Returns: -1 on bad imol, 0 on no unsaved changes, 1 on has unsaved changes 
</p>
</dd></dl>

<hr>
<a name="set_002dundo_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#show_002dset_002dundo_002dmolecule_002dchooser" accesskey="n" rel="next">show-set-undo-molecule-chooser</a>, Previous: <a href="#have_002dunsaved_002dchanges_002dp" accesskey="p" rel="previous">have-unsaved-changes-p</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dundo_002dmolecule-1"></a>
<h4 class="subsection">11.12.9 set-undo-molecule</h4>
<dl>
<dt><a name="index-set_002dundo_002dmolecule"></a>function: <strong>set-undo-molecule</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>set the molecule to which undo operations are done to molecule number imol 
</p>
</dd></dl>

<hr>
<a name="show_002dset_002dundo_002dmolecule_002dchooser"></a>
<div class="header">
<p>
Next: <a href="#set_002dunpathed_002dbackup_002dfile_002dnames" accesskey="n" rel="next">set-unpathed-backup-file-names</a>, Previous: <a href="#set_002dundo_002dmolecule" accesskey="p" rel="previous">set-undo-molecule</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dset_002dundo_002dmolecule_002dchooser-1"></a>
<h4 class="subsection">11.12.10 show-set-undo-molecule-chooser</h4>
<dl>
<dt><a name="index-show_002dset_002dundo_002dmolecule_002dchooser"></a>function: <strong>show-set-undo-molecule-chooser</strong></dt>
<dd>
<p>show the Undo Molecule chooser - i.e. choose the molecule to which the &quot;Undo&quot; button applies. 
</p>
</dd></dl>

<hr>
<a name="set_002dunpathed_002dbackup_002dfile_002dnames"></a>
<div class="header">
<p>
Next: <a href="#unpathed_002dbackup_002dfile_002dnames_002dstate" accesskey="n" rel="next">unpathed-backup-file-names-state</a>, Previous: <a href="#show_002dset_002dundo_002dmolecule_002dchooser" accesskey="p" rel="previous">show-set-undo-molecule-chooser</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dunpathed_002dbackup_002dfile_002dnames-1"></a>
<h4 class="subsection">11.12.11 set-unpathed-backup-file-names</h4>
<dl>
<dt><a name="index-set_002dunpathed_002dbackup_002dfile_002dnames"></a>function: <strong>set-unpathed-backup-file-names</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set the state for adding paths to backup file names 
</p>
<p>by default directories names are added into the filename for backup (with / to _ mapping). call this with state=1 to turn off directory names 
</p>
</dd></dl>

<hr>
<a name="unpathed_002dbackup_002dfile_002dnames_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002ddecoloned_002dbackup_002dfile_002dnames" accesskey="n" rel="next">set-decoloned-backup-file-names</a>, Previous: <a href="#set_002dunpathed_002dbackup_002dfile_002dnames" accesskey="p" rel="previous">set-unpathed-backup-file-names</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="unpathed_002dbackup_002dfile_002dnames_002dstate-1"></a>
<h4 class="subsection">11.12.12 unpathed-backup-file-names-state</h4>
<dl>
<dt><a name="index-unpathed_002dbackup_002dfile_002dnames_002dstate"></a>function: <strong>unpathed-backup-file-names-state</strong></dt>
<dd>
<p>return the state for adding paths to backup file names 
</p>
</dd></dl>

<hr>
<a name="set_002ddecoloned_002dbackup_002dfile_002dnames"></a>
<div class="header">
<p>
Next: <a href="#decoloned_002dbackup_002dfile_002dnames_002dstate" accesskey="n" rel="next">decoloned-backup-file-names-state</a>, Previous: <a href="#unpathed_002dbackup_002dfile_002dnames_002dstate" accesskey="p" rel="previous">unpathed-backup-file-names-state</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddecoloned_002dbackup_002dfile_002dnames-1"></a>
<h4 class="subsection">11.12.13 set-decoloned-backup-file-names</h4>
<dl>
<dt><a name="index-set_002ddecoloned_002dbackup_002dfile_002dnames"></a>function: <strong>set-decoloned-backup-file-names</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set the state for adding paths to backup file names 
</p>
<p>by default directories names are added into the filename for backup (with / to _ mapping). call this with state=1 to turn off directory names 
</p>
</dd></dl>

<hr>
<a name="decoloned_002dbackup_002dfile_002dnames_002dstate"></a>
<div class="header">
<p>
Next: <a href="#backup_002dcompress_002dfiles_002dstate" accesskey="n" rel="next">backup-compress-files-state</a>, Previous: <a href="#set_002ddecoloned_002dbackup_002dfile_002dnames" accesskey="p" rel="previous">set-decoloned-backup-file-names</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="decoloned_002dbackup_002dfile_002dnames_002dstate-1"></a>
<h4 class="subsection">11.12.14 decoloned-backup-file-names-state</h4>
<dl>
<dt><a name="index-decoloned_002dbackup_002dfile_002dnames_002dstate"></a>function: <strong>decoloned-backup-file-names-state</strong></dt>
<dd>
<p>return the state for adding paths to backup file names 
</p>
</dd></dl>

<hr>
<a name="backup_002dcompress_002dfiles_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dbackup_002dcompress_002dfiles" accesskey="n" rel="next">set-backup-compress-files</a>, Previous: <a href="#decoloned_002dbackup_002dfile_002dnames_002dstate" accesskey="p" rel="previous">decoloned-backup-file-names-state</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="backup_002dcompress_002dfiles_002dstate-1"></a>
<h4 class="subsection">11.12.15 backup-compress-files-state</h4>
<dl>
<dt><a name="index-backup_002dcompress_002dfiles_002dstate"></a>function: <strong>backup-compress-files-state</strong></dt>
<dd>
<p>return the state for compression of backup files 
</p>
</dd></dl>

<hr>
<a name="set_002dbackup_002dcompress_002dfiles"></a>
<div class="header">
<p>
Previous: <a href="#backup_002dcompress_002dfiles_002dstate" accesskey="p" rel="previous">backup-compress-files-state</a>, Up: <a href="#Backup-Functions" accesskey="u" rel="up">Backup Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbackup_002dcompress_002dfiles-1"></a>
<h4 class="subsection">11.12.16 set-backup-compress-files</h4>
<dl>
<dt><a name="index-set_002dbackup_002dcompress_002dfiles"></a>function: <strong>set-backup-compress-files</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set if backup files will be compressed or not using gzip 
</p>
</dd></dl>

<hr>
<a name="Recover-Session-Function"></a>
<div class="header">
<p>
Next: <a href="#Map-Functions" accesskey="n" rel="next">Map Functions</a>, Previous: <a href="#Backup-Functions" accesskey="p" rel="previous">Backup Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Recover-Session-Function-1"></a>
<h3 class="section">11.13 Recover Session Function</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#recover_002dsession" accesskey="1">recover-session</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="recover_002dsession"></a>
<div class="header">
<p>
Up: <a href="#Recover-Session-Function" accesskey="u" rel="up">Recover Session Function</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="recover_002dsession-1"></a>
<h4 class="subsection">11.13.1 recover-session</h4>
<dl>
<dt><a name="index-recover_002dsession"></a>function: <strong>recover-session</strong></dt>
<dd>
<p>recover session 
</p>
<p>After a crash, we provide this convenient interface to restore the session. It runs through all the molecules with models and looks at the coot backup directory looking for related backup files that are more recent that the read file. (Not very good, because you need to remember which files you read in before the crash - should be improved.) 
</p>
</dd></dl>

<hr>
<a name="Map-Functions"></a>
<div class="header">
<p>
Next: <a href="#Density-Increment" accesskey="n" rel="next">Density Increment</a>, Previous: <a href="#Recover-Session-Function" accesskey="p" rel="previous">Recover Session Function</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Functions-1"></a>
<h3 class="section">11.14 Map Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#calc_002dphases_002dgeneric" accesskey="1">calc-phases-generic</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases" accesskey="2">map-from-mtz-by-refmac-calc-phases</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dfrom_002dmtz_002dby_002dcalc_002dphases" accesskey="3">map-from-mtz-by-calc-phases</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sfcalc_002dgenmap" accesskey="4">sfcalc-genmap</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dauto_002dupdating_002dsfcalc_002dgenmap" accesskey="5">set-auto-updating-sfcalc-genmap</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dscroll_002dwheel_002dmap" accesskey="6">set-scroll-wheel-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dscrollable_002dmap" accesskey="7">set-scrollable-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#scroll_002dwheel_002dmap" accesskey="8">scroll-wheel-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#save_002dprevious_002dmap_002dcolour" accesskey="9">save-previous-map-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#restore_002dprevious_002dmap_002dcolour">restore-previous-map-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dactive_002dmap_002ddrag_002dflag">set-active-map-drag-flag</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dactive_002dmap_002ddrag_002dflag">get-active-map-drag-flag</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dlast_002dmap_002dcolour">set-last-map-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dcolour">set-map-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcontour_002dlevel_002dabsolute">set-contour-level-absolute</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcontour_002dlevel_002din_002dsigma">set-contour-level-in-sigma</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dcontour_002dlevel_002dabsolute">get-contour-level-absolute</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dcontour_002dlevel_002din_002dsigma">get-contour-level-in-sigma</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dlast_002dmap_002dsigma_002dstep">set-last-map-sigma-step</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol">set-contour-by-sigma-step-by-mol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#data_002dresolution">data-resolution</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#model_002dresolution">model-resolution</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#export_002dmap">export-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#export_002dmap_002dfragment">export-map-fragment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#export_002dmap_002dfragment_002dwith_002dorigin_002dshift">export-map-fragment-with-origin-shift</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#export_002dmap_002dfragment_002dto_002dplain_002dfile">export-map-fragment-to-plain-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#difference_002dmap">difference-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#reinterp_002dmap">reinterp-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#smooth_002dmap">smooth-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#average_002dmap_002dscm">average-map-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#average_002dmap_002dpy">average-map-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="calc_002dphases_002dgeneric"></a>
<div class="header">
<p>
Next: <a href="#map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases" accesskey="n" rel="next">map-from-mtz-by-refmac-calc-phases</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="calc_002dphases_002dgeneric-1"></a>
<h4 class="subsection">11.14.1 calc-phases-generic</h4>
<dl>
<dt><a name="index-calc_002dphases_002dgeneric"></a>function: <strong>calc-phases-generic</strong> <em>mtz_file_name</em></dt>
<dd>
<p>Where <em>mtz_file_name</em> is a string
</p>
<p>fire up a GUI, which asks us which model molecule we want to calc phases from. On &quot;OK&quot; button there, we call map_from_mtz_by_refmac_calc_phases()
</p>
</dd></dl>

<hr>
<a name="map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases"></a>
<div class="header">
<p>
Next: <a href="#map_002dfrom_002dmtz_002dby_002dcalc_002dphases" accesskey="n" rel="next">map-from-mtz-by-calc-phases</a>, Previous: <a href="#calc_002dphases_002dgeneric" accesskey="p" rel="previous">calc-phases-generic</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases-1"></a>
<h4 class="subsection">11.14.2 map-from-mtz-by-refmac-calc-phases</h4>
<dl>
<dt><a name="index-map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases"></a>function: <strong>map-from-mtz-by-refmac-calc-phases</strong> <em>mtz_file_name f_col sigf_col imol_coords</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mtz_file_name</em> is a string
</li><li> <em>f_col</em> is a string
</li><li> <em>sigf_col</em> is a string
</li><li> <em>imol_coords</em> is an integer number
</li></ul>

<p>Calculate SFs (using refmac optionally) from an MTZ file and generate a map. Get F and SIGF automatically (first of their type) from the mtz file. 
</p>
<p>Returns: the new molecule number, -1 on a problem. 
</p>
</dd></dl>

<hr>
<a name="map_002dfrom_002dmtz_002dby_002dcalc_002dphases"></a>
<div class="header">
<p>
Next: <a href="#sfcalc_002dgenmap" accesskey="n" rel="next">sfcalc-genmap</a>, Previous: <a href="#map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases" accesskey="p" rel="previous">map-from-mtz-by-refmac-calc-phases</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dfrom_002dmtz_002dby_002dcalc_002dphases-1"></a>
<h4 class="subsection">11.14.3 map-from-mtz-by-calc-phases</h4>
<dl>
<dt><a name="index-map_002dfrom_002dmtz_002dby_002dcalc_002dphases"></a>function: <strong>map-from-mtz-by-calc-phases</strong> <em>mtz_file_name f_col sigf_col imol_coords</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mtz_file_name</em> is a string
</li><li> <em>f_col</em> is a string
</li><li> <em>sigf_col</em> is a string
</li><li> <em>imol_coords</em> is an integer number
</li></ul>

<p>Calculate SFs from an MTZ file and generate a map. 
</p>
<p>Returns: the new molecule number. 
</p>
</dd></dl>

<hr>
<a name="sfcalc_002dgenmap"></a>
<div class="header">
<p>
Next: <a href="#set_002dauto_002dupdating_002dsfcalc_002dgenmap" accesskey="n" rel="next">set-auto-updating-sfcalc-genmap</a>, Previous: <a href="#map_002dfrom_002dmtz_002dby_002dcalc_002dphases" accesskey="p" rel="previous">map-from-mtz-by-calc-phases</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="sfcalc_002dgenmap-1"></a>
<h4 class="subsection">11.14.4 sfcalc-genmap</h4>
<dl>
<dt><a name="index-sfcalc_002dgenmap"></a>function: <strong>sfcalc-genmap</strong> <em>imol_model imol_map_with_data_attached imol_updating_difference_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_model</em> is an integer number
</li><li> <em>imol_map_with_data_attached</em> is an integer number
</li><li> <em>imol_updating_difference_map</em> is an integer number
</li></ul>

<p>Calculate structure factors from the model and update the given difference map accordingly. 
</p>
</dd></dl>

<hr>
<a name="set_002dauto_002dupdating_002dsfcalc_002dgenmap"></a>
<div class="header">
<p>
Next: <a href="#set_002dscroll_002dwheel_002dmap" accesskey="n" rel="next">set-scroll-wheel-map</a>, Previous: <a href="#sfcalc_002dgenmap" accesskey="p" rel="previous">sfcalc-genmap</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dauto_002dupdating_002dsfcalc_002dgenmap-1"></a>
<h4 class="subsection">11.14.5 set-auto-updating-sfcalc-genmap</h4>
<dl>
<dt><a name="index-set_002dauto_002dupdating_002dsfcalc_002dgenmap"></a>function: <strong>set-auto-updating-sfcalc-genmap</strong> <em>imol_model imol_map_with_data_attached imol_updating_difference_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_model</em> is an integer number
</li><li> <em>imol_map_with_data_attached</em> is an integer number
</li><li> <em>imol_updating_difference_map</em> is an integer number
</li></ul>

<p>As above, calculate structure factors from the model and update the given difference map accordingly - but difference map gets updated automatically on modification of the imol_model molecule. 
</p>
</dd></dl>

<hr>
<a name="set_002dscroll_002dwheel_002dmap"></a>
<div class="header">
<p>
Next: <a href="#set_002dscrollable_002dmap" accesskey="n" rel="next">set-scrollable-map</a>, Previous: <a href="#set_002dauto_002dupdating_002dsfcalc_002dgenmap" accesskey="p" rel="previous">set-auto-updating-sfcalc-genmap</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dscroll_002dwheel_002dmap-1"></a>
<h4 class="subsection">11.14.6 set-scroll-wheel-map</h4>
<dl>
<dt><a name="index-set_002dscroll_002dwheel_002dmap"></a>function: <strong>set-scroll-wheel-map</strong> <em>imap</em></dt>
<dd>
<p>Where <em>imap</em> is an integer number
</p>
<p>set the map that is moved by changing the scroll wheel and change_contour_level(). 
</p>
</dd></dl>

<hr>
<a name="set_002dscrollable_002dmap"></a>
<div class="header">
<p>
Next: <a href="#scroll_002dwheel_002dmap" accesskey="n" rel="next">scroll-wheel-map</a>, Previous: <a href="#set_002dscroll_002dwheel_002dmap" accesskey="p" rel="previous">set-scroll-wheel-map</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dscrollable_002dmap-1"></a>
<h4 class="subsection">11.14.7 set-scrollable-map</h4>
<dl>
<dt><a name="index-set_002dscrollable_002dmap"></a>function: <strong>set-scrollable-map</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the molecule number to which the mouse scroll wheel is attached 
</p>
<p>set the map that has its contour level changed by the scrolling the mouse wheel to molecule number imol (same as 
</p>
<p>). 
</p>
</dd></dl>

<hr>
<a name="scroll_002dwheel_002dmap"></a>
<div class="header">
<p>
Next: <a href="#save_002dprevious_002dmap_002dcolour" accesskey="n" rel="next">save-previous-map-colour</a>, Previous: <a href="#set_002dscrollable_002dmap" accesskey="p" rel="previous">set-scrollable-map</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="scroll_002dwheel_002dmap-1"></a>
<h4 class="subsection">11.14.8 scroll-wheel-map</h4>
<dl>
<dt><a name="index-scroll_002dwheel_002dmap"></a>function: <strong>scroll-wheel-map</strong></dt>
<dd>
<p>the contouring of which map is altered when the scroll wheel changes? 
</p>
</dd></dl>

<hr>
<a name="save_002dprevious_002dmap_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#restore_002dprevious_002dmap_002dcolour" accesskey="n" rel="next">restore-previous-map-colour</a>, Previous: <a href="#scroll_002dwheel_002dmap" accesskey="p" rel="previous">scroll-wheel-map</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dprevious_002dmap_002dcolour-1"></a>
<h4 class="subsection">11.14.9 save-previous-map-colour</h4>
<dl>
<dt><a name="index-save_002dprevious_002dmap_002dcolour"></a>function: <strong>save-previous-map-colour</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>save previous colour map for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="restore_002dprevious_002dmap_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#set_002dactive_002dmap_002ddrag_002dflag" accesskey="n" rel="next">set-active-map-drag-flag</a>, Previous: <a href="#save_002dprevious_002dmap_002dcolour" accesskey="p" rel="previous">save-previous-map-colour</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="restore_002dprevious_002dmap_002dcolour-1"></a>
<h4 class="subsection">11.14.10 restore-previous-map-colour</h4>
<dl>
<dt><a name="index-restore_002dprevious_002dmap_002dcolour"></a>function: <strong>restore-previous-map-colour</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>restore previous colour map for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="set_002dactive_002dmap_002ddrag_002dflag"></a>
<div class="header">
<p>
Next: <a href="#get_002dactive_002dmap_002ddrag_002dflag" accesskey="n" rel="next">get-active-map-drag-flag</a>, Previous: <a href="#restore_002dprevious_002dmap_002dcolour" accesskey="p" rel="previous">restore-previous-map-colour</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dactive_002dmap_002ddrag_002dflag-1"></a>
<h4 class="subsection">11.14.11 set-active-map-drag-flag</h4>
<dl>
<dt><a name="index-set_002dactive_002dmap_002ddrag_002dflag"></a>function: <strong>set-active-map-drag-flag</strong> <em>t</em></dt>
<dd>
<p>Where <em>t</em> is an integer number
</p>
<p>set the state of immediate map upate on map drag. 
</p>
<p>By default, it is on (t=1). On slower computers it might be better to set t=0. 
</p>
</dd></dl>

<hr>
<a name="get_002dactive_002dmap_002ddrag_002dflag"></a>
<div class="header">
<p>
Next: <a href="#set_002dlast_002dmap_002dcolour" accesskey="n" rel="next">set-last-map-colour</a>, Previous: <a href="#set_002dactive_002dmap_002ddrag_002dflag" accesskey="p" rel="previous">set-active-map-drag-flag</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dactive_002dmap_002ddrag_002dflag-1"></a>
<h4 class="subsection">11.14.12 get-active-map-drag-flag</h4>
<dl>
<dt><a name="index-get_002dactive_002dmap_002ddrag_002dflag"></a>function: <strong>get-active-map-drag-flag</strong></dt>
<dd>
<p>return the state of the dragged map flag 
</p>
</dd></dl>

<hr>
<a name="set_002dlast_002dmap_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002dcolour" accesskey="n" rel="next">set-map-colour</a>, Previous: <a href="#get_002dactive_002dmap_002ddrag_002dflag" accesskey="p" rel="previous">get-active-map-drag-flag</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dlast_002dmap_002dcolour-1"></a>
<h4 class="subsection">11.14.13 set-last-map-colour</h4>
<dl>
<dt><a name="index-set_002dlast_002dmap_002dcolour"></a>function: <strong>set-last-map-colour</strong> <em>f1 f2 f3</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>f1</em> is a number
</li><li> <em>f2</em> is a number
</li><li> <em>f3</em> is a number
</li></ul>

<p>set the colour of the last (highest molecule number) map 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#set_002dcontour_002dlevel_002dabsolute" accesskey="n" rel="next">set-contour-level-absolute</a>, Previous: <a href="#set_002dlast_002dmap_002dcolour" accesskey="p" rel="previous">set-last-map-colour</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dcolour-1"></a>
<h4 class="subsection">11.14.14 set-map-colour</h4>
<dl>
<dt><a name="index-set_002dmap_002dcolour"></a>function: <strong>set-map-colour</strong> <em>imol red green blue</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>red</em> is a number
</li><li> <em>green</em> is a number
</li><li> <em>blue</em> is a number
</li></ul>

<p>set the colour of the imolth map 
</p>
</dd></dl>

<hr>
<a name="set_002dcontour_002dlevel_002dabsolute"></a>
<div class="header">
<p>
Next: <a href="#set_002dcontour_002dlevel_002din_002dsigma" accesskey="n" rel="next">set-contour-level-in-sigma</a>, Previous: <a href="#set_002dmap_002dcolour" accesskey="p" rel="previous">set-map-colour</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcontour_002dlevel_002dabsolute-1"></a>
<h4 class="subsection">11.14.15 set-contour-level-absolute</h4>
<dl>
<dt><a name="index-set_002dcontour_002dlevel_002dabsolute"></a>function: <strong>set-contour-level-absolute</strong> <em>imol_map level</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>level</em> is a number
</li></ul>

<p>set the contour level, direct control 
</p>
</dd></dl>

<hr>
<a name="set_002dcontour_002dlevel_002din_002dsigma"></a>
<div class="header">
<p>
Next: <a href="#get_002dcontour_002dlevel_002dabsolute" accesskey="n" rel="next">get-contour-level-absolute</a>, Previous: <a href="#set_002dcontour_002dlevel_002dabsolute" accesskey="p" rel="previous">set-contour-level-absolute</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcontour_002dlevel_002din_002dsigma-1"></a>
<h4 class="subsection">11.14.16 set-contour-level-in-sigma</h4>
<dl>
<dt><a name="index-set_002dcontour_002dlevel_002din_002dsigma"></a>function: <strong>set-contour-level-in-sigma</strong> <em>imol_map level</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>level</em> is a number
</li></ul>

<p>set the contour level, direct control in r.m.s.d. (if you like that sort of thing) 
</p>
</dd></dl>

<hr>
<a name="get_002dcontour_002dlevel_002dabsolute"></a>
<div class="header">
<p>
Next: <a href="#get_002dcontour_002dlevel_002din_002dsigma" accesskey="n" rel="next">get-contour-level-in-sigma</a>, Previous: <a href="#set_002dcontour_002dlevel_002din_002dsigma" accesskey="p" rel="previous">set-contour-level-in-sigma</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dcontour_002dlevel_002dabsolute-1"></a>
<h4 class="subsection">11.14.17 get-contour-level-absolute</h4>
<dl>
<dt><a name="index-get_002dcontour_002dlevel_002dabsolute"></a>function: <strong>get-contour-level-absolute</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>get the contour level 
</p>
</dd></dl>

<hr>
<a name="get_002dcontour_002dlevel_002din_002dsigma"></a>
<div class="header">
<p>
Next: <a href="#set_002dlast_002dmap_002dsigma_002dstep" accesskey="n" rel="next">set-last-map-sigma-step</a>, Previous: <a href="#get_002dcontour_002dlevel_002dabsolute" accesskey="p" rel="previous">get-contour-level-absolute</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dcontour_002dlevel_002din_002dsigma-1"></a>
<h4 class="subsection">11.14.18 get-contour-level-in-sigma</h4>
<dl>
<dt><a name="index-get_002dcontour_002dlevel_002din_002dsigma"></a>function: <strong>get-contour-level-in-sigma</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>get the contour level in rmd above 0. 
</p>
</dd></dl>

<hr>
<a name="set_002dlast_002dmap_002dsigma_002dstep"></a>
<div class="header">
<p>
Next: <a href="#set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol" accesskey="n" rel="next">set-contour-by-sigma-step-by-mol</a>, Previous: <a href="#get_002dcontour_002dlevel_002din_002dsigma" accesskey="p" rel="previous">get-contour-level-in-sigma</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dlast_002dmap_002dsigma_002dstep-1"></a>
<h4 class="subsection">11.14.19 set-last-map-sigma-step</h4>
<dl>
<dt><a name="index-set_002dlast_002dmap_002dsigma_002dstep"></a>function: <strong>set-last-map-sigma-step</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the sigma step of the last map to f sigma 
</p>
</dd></dl>

<hr>
<a name="set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol"></a>
<div class="header">
<p>
Next: <a href="#data_002dresolution" accesskey="n" rel="next">data-resolution</a>, Previous: <a href="#set_002dlast_002dmap_002dsigma_002dstep" accesskey="p" rel="previous">set-last-map-sigma-step</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol-1"></a>
<h4 class="subsection">11.14.20 set-contour-by-sigma-step-by-mol</h4>
<dl>
<dt><a name="index-set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol"></a>function: <strong>set-contour-by-sigma-step-by-mol</strong> <em>f state imol</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>f</em> is a number
</li><li> <em>state</em> is an integer number
</li><li> <em>imol</em> is an integer number
</li></ul>

<p>set the contour level step 
</p>
<p>set the contour level step of molecule number imol to f and variable state (setting state to 0 turns off contouring by sigma level) 
</p>
</dd></dl>

<hr>
<a name="data_002dresolution"></a>
<div class="header">
<p>
Next: <a href="#model_002dresolution" accesskey="n" rel="next">model-resolution</a>, Previous: <a href="#set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol" accesskey="p" rel="previous">set-contour-by-sigma-step-by-mol</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="data_002dresolution-1"></a>
<h4 class="subsection">11.14.21 data-resolution</h4>
<dl>
<dt><a name="index-data_002dresolution"></a>function: <strong>data-resolution</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the resolution of the data for molecule number imol. Return negative number on error, otherwise resolution in A (eg. 2.0) 
</p>
</dd></dl>

<hr>
<a name="model_002dresolution"></a>
<div class="header">
<p>
Next: <a href="#export_002dmap" accesskey="n" rel="next">export-map</a>, Previous: <a href="#data_002dresolution" accesskey="p" rel="previous">data-resolution</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="model_002dresolution-1"></a>
<h4 class="subsection">11.14.22 model-resolution</h4>
<dl>
<dt><a name="index-model_002dresolution"></a>function: <strong>model-resolution</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the resolution set in the header of the model/coordinates file. If this number is not available, return a number less than 0. 
</p>
</dd></dl>

<hr>
<a name="export_002dmap"></a>
<div class="header">
<p>
Next: <a href="#export_002dmap_002dfragment" accesskey="n" rel="next">export-map-fragment</a>, Previous: <a href="#model_002dresolution" accesskey="p" rel="previous">model-resolution</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="export_002dmap-1"></a>
<h4 class="subsection">11.14.23 export-map</h4>
<dl>
<dt><a name="index-export_002dmap"></a>function: <strong>export-map</strong> <em>imol filename</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>filename</em> is a string
</li></ul>

<p>export (write to disk) the map of molecule number imol to filename. 
</p>
<p>Return 0 on failure, 1 on success. 
</p>
</dd></dl>

<hr>
<a name="export_002dmap_002dfragment"></a>
<div class="header">
<p>
Next: <a href="#export_002dmap_002dfragment_002dwith_002dorigin_002dshift" accesskey="n" rel="next">export-map-fragment-with-origin-shift</a>, Previous: <a href="#export_002dmap" accesskey="p" rel="previous">export-map</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="export_002dmap_002dfragment-1"></a>
<h4 class="subsection">11.14.24 export-map-fragment</h4>
<dl>
<dt><a name="index-export_002dmap_002dfragment"></a>function: <strong>export-map-fragment</strong> <em>imol x y z radius filename</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li><li> <em>radius</em> is a number
</li><li> <em>filename</em> is a string
</li></ul>

<p>export a fragment of the map about (x,y,z) 
</p>
</dd></dl>

<hr>
<a name="export_002dmap_002dfragment_002dwith_002dorigin_002dshift"></a>
<div class="header">
<p>
Next: <a href="#export_002dmap_002dfragment_002dto_002dplain_002dfile" accesskey="n" rel="next">export-map-fragment-to-plain-file</a>, Previous: <a href="#export_002dmap_002dfragment" accesskey="p" rel="previous">export-map-fragment</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="export_002dmap_002dfragment_002dwith_002dorigin_002dshift-1"></a>
<h4 class="subsection">11.14.25 export-map-fragment-with-origin-shift</h4>
<dl>
<dt><a name="index-export_002dmap_002dfragment_002dwith_002dorigin_002dshift"></a>function: <strong>export-map-fragment-with-origin-shift</strong> <em>imol x y z radius filename</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li><li> <em>radius</em> is a number
</li><li> <em>filename</em> is a string
</li></ul>

<p>export a fragment of the map about (x,y,z) 
</p>
</dd></dl>

<hr>
<a name="export_002dmap_002dfragment_002dto_002dplain_002dfile"></a>
<div class="header">
<p>
Next: <a href="#difference_002dmap" accesskey="n" rel="next">difference-map</a>, Previous: <a href="#export_002dmap_002dfragment_002dwith_002dorigin_002dshift" accesskey="p" rel="previous">export-map-fragment-with-origin-shift</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="export_002dmap_002dfragment_002dto_002dplain_002dfile-1"></a>
<h4 class="subsection">11.14.26 export-map-fragment-to-plain-file</h4>
<dl>
<dt><a name="index-export_002dmap_002dfragment_002dto_002dplain_002dfile"></a>function: <strong>export-map-fragment-to-plain-file</strong> <em>imol x y z radius filename</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li><li> <em>radius</em> is a number
</li><li> <em>filename</em> is a string
</li></ul>

<p>tmp interface for Hamish 
</p>
</dd></dl>

<hr>
<a name="difference_002dmap"></a>
<div class="header">
<p>
Next: <a href="#reinterp_002dmap" accesskey="n" rel="next">reinterp-map</a>, Previous: <a href="#export_002dmap_002dfragment_002dto_002dplain_002dfile" accesskey="p" rel="previous">export-map-fragment-to-plain-file</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="difference_002dmap-1"></a>
<h4 class="subsection">11.14.27 difference-map</h4>
<dl>
<dt><a name="index-difference_002dmap"></a>function: <strong>difference-map</strong> <em>imol1 imol2 map_scale</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol1</em> is an integer number
</li><li> <em>imol2</em> is an integer number
</li><li> <em>map_scale</em> is a number
</li></ul>

<p>make a difference map, taking map_scale * imap2 from imap1, on the grid of imap1. Return the new molecule number. Return -1 on failure. 
</p>
</dd></dl>

<hr>
<a name="reinterp_002dmap"></a>
<div class="header">
<p>
Next: <a href="#smooth_002dmap" accesskey="n" rel="next">smooth-map</a>, Previous: <a href="#difference_002dmap" accesskey="p" rel="previous">difference-map</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="reinterp_002dmap-1"></a>
<h4 class="subsection">11.14.28 reinterp-map</h4>
<dl>
<dt><a name="index-reinterp_002dmap"></a>function: <strong>reinterp-map</strong> <em>map_no reference_map_no</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>map_no</em> is an integer number
</li><li> <em>reference_map_no</em> is an integer number
</li></ul>

<p>make a new map (a copy of map_no) that is in the cell, spacegroup and gridding of the map in reference_map_no. 
</p>
<p>Return the new map molecule number - return -1 on failure 
</p>
</dd></dl>

<hr>
<a name="smooth_002dmap"></a>
<div class="header">
<p>
Next: <a href="#average_002dmap_002dscm" accesskey="n" rel="next">average-map-scm</a>, Previous: <a href="#reinterp_002dmap" accesskey="p" rel="previous">reinterp-map</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="smooth_002dmap-1"></a>
<h4 class="subsection">11.14.29 smooth-map</h4>
<dl>
<dt><a name="index-smooth_002dmap"></a>function: <strong>smooth-map</strong> <em>map_no sampling_multiplier</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>map_no</em> is an integer number
</li><li> <em>sampling_multiplier</em> is a number
</li></ul>

<p>make a new map (a copy of map_no) that is in the cell, spacegroup and a multiple of the sampling of the input map (a sampling factor of more than 1 makes the output maps smoother) 
</p>
</dd></dl>

<hr>
<a name="average_002dmap_002dscm"></a>
<div class="header">
<p>
Next: <a href="#average_002dmap_002dpy" accesskey="n" rel="next">average-map-py</a>, Previous: <a href="#smooth_002dmap" accesskey="p" rel="previous">smooth-map</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="average_002dmap_002dscm-1"></a>
<h4 class="subsection">11.14.30 average-map-scm</h4>
<dl>
<dt><a name="index-average_002dmap_002dscm"></a>function: <strong>average-map-scm</strong> <em>map_number_and_scales</em></dt>
<dd>
<p>Where <em>map_number_and_scales</em> is a SCM
</p>
<p>make an average map from the map_number_and_scales (which is a list of pairs (list map-number scale-factor)) (the scale factors are typically 1.0 of course). The output map is in the same grid as the first (valid) map. Return -1 on failure to make an averaged map, otherwise return the new map molecule number. 
</p>
</dd></dl>

<hr>
<a name="average_002dmap_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#average_002dmap_002dscm" accesskey="p" rel="previous">average-map-scm</a>, Up: <a href="#Map-Functions" accesskey="u" rel="up">Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="average_002dmap_002dpy-1"></a>
<h4 class="subsection">11.14.31 average-map-py</h4>
<dl>
<dt><a name="index-average_002dmap_002dpy"></a>function: <strong>average-map-py</strong> <em>map_number_and_scales</em></dt>
<dd>
<p>Where <em>map_number_and_scales</em> is a PyObject
</p>
<p>make an average map from the map_number_and_scales (which is a list of pairs map_number, scale_factor. The output map is in the same grid as the first (valid) map. Return -1 on failure to make an averaged map, otherwise return the new map molecule number. 
</p>
</dd></dl>

<hr>
<a name="Density-Increment"></a>
<div class="header">
<p>
Next: <a href="#Density-Functions" accesskey="n" rel="next">Density Functions</a>, Previous: <a href="#Map-Functions" accesskey="p" rel="previous">Map Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Density-Increment-1"></a>
<h3 class="section">11.15 Density Increment</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002diso_002dlevel_002dincrement" accesskey="1">set-iso-level-increment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement" accesskey="2">set-diff-map-iso-level-increment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002ddiff_002dmap_002diso_002dlevel_002dincrement" accesskey="3">get-diff-map-iso-level-increment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext" accesskey="4">set-diff-map-iso-level-increment-from-text</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dsampling_002drate_002dtext" accesskey="5">set-map-sampling-rate-text</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dsampling_002drate" accesskey="6">set-map-sampling-rate</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dmap_002dsampling_002drate" accesskey="7">get-map-sampling-rate</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#change_002dcontour_002dlevel" accesskey="8">change-contour-level</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dlast_002dmap_002dcontour_002dlevel" accesskey="9">set-last-map-contour-level</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma">set-last-map-contour-level-by-sigma</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dstop_002dscroll_002ddiff_002dmap">set-stop-scroll-diff-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dstop_002dscroll_002diso_002dmap">set-stop-scroll-iso-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dstop_002dscroll_002diso_002dmap_002dlevel">set-stop-scroll-iso-map-level</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dstop_002dscroll_002ddiff_002dmap_002dlevel">set-stop-scroll-diff-map-level</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dresidue_002ddensity_002dfit_002dscale_002dfactor">set-residue-density-fit-scale-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002diso_002dlevel_002dincrement"></a>
<div class="header">
<p>
Next: <a href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement" accesskey="n" rel="next">set-diff-map-iso-level-increment</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002diso_002dlevel_002dincrement-1"></a>
<h4 class="subsection">11.15.1 set-iso-level-increment</h4>
<dl>
<dt><a name="index-set_002diso_002dlevel_002dincrement"></a>function: <strong>set-iso-level-increment</strong> <em>val</em></dt>
<dd>
<p>Where <em>val</em> is a number
</p>
<p>set the contour scroll step (in absolute e/A3) for 2Fo-Fc-style maps to val 
</p>
<p>The is only activated when scrolling by sigma is turned off 
</p>
</dd></dl>

<hr>
<a name="set_002ddiff_002dmap_002diso_002dlevel_002dincrement"></a>
<div class="header">
<p>
Next: <a href="#get_002ddiff_002dmap_002diso_002dlevel_002dincrement" accesskey="n" rel="next">get-diff-map-iso-level-increment</a>, Previous: <a href="#set_002diso_002dlevel_002dincrement" accesskey="p" rel="previous">set-iso-level-increment</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddiff_002dmap_002diso_002dlevel_002dincrement-1"></a>
<h4 class="subsection">11.15.2 set-diff-map-iso-level-increment</h4>
<dl>
<dt><a name="index-set_002ddiff_002dmap_002diso_002dlevel_002dincrement"></a>function: <strong>set-diff-map-iso-level-increment</strong> <em>val</em></dt>
<dd>
<p>Where <em>val</em> is a number
</p>
<p>set the contour scroll step for difference map (in absolute e/A3) to val 
</p>
<p>The is only activated when scrolling by sigma is turned off 
</p>
</dd></dl>

<hr>
<a name="get_002ddiff_002dmap_002diso_002dlevel_002dincrement"></a>
<div class="header">
<p>
Next: <a href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext" accesskey="n" rel="next">set-diff-map-iso-level-increment-from-text</a>, Previous: <a href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement" accesskey="p" rel="previous">set-diff-map-iso-level-increment</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002ddiff_002dmap_002diso_002dlevel_002dincrement-1"></a>
<h4 class="subsection">11.15.3 get-diff-map-iso-level-increment</h4>
<dl>
<dt><a name="index-get_002ddiff_002dmap_002diso_002dlevel_002dincrement"></a>function: <strong>get-diff-map-iso-level-increment</strong></dt>
<dd>
<p>return difference maps iso-map level increment 
</p>
</dd></dl>

<hr>
<a name="set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002dsampling_002drate_002dtext" accesskey="n" rel="next">set-map-sampling-rate-text</a>, Previous: <a href="#get_002ddiff_002dmap_002diso_002dlevel_002dincrement" accesskey="p" rel="previous">get-diff-map-iso-level-increment</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext-1"></a>
<h4 class="subsection">11.15.4 set-diff-map-iso-level-increment-from-text</h4>
<dl>
<dt><a name="index-set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext"></a>function: <strong>set-diff-map-iso-level-increment-from-text</strong> <em>text imol</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>text</em> is a string
</li><li> <em>imol</em> is an integer number
</li></ul>

<p>set the difference maps iso-map level increment 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002dsampling_002drate_002dtext"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002dsampling_002drate" accesskey="n" rel="next">set-map-sampling-rate</a>, Previous: <a href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext" accesskey="p" rel="previous">set-diff-map-iso-level-increment-from-text</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dsampling_002drate_002dtext-1"></a>
<h4 class="subsection">11.15.5 set-map-sampling-rate-text</h4>
<dl>
<dt><a name="index-set_002dmap_002dsampling_002drate_002dtext"></a>function: <strong>set-map-sampling-rate-text</strong> <em>text</em></dt>
<dd>
<p>Where <em>text</em> is a string
</p>
<p>sampling rate 
</p>
<p>find the molecule for which the single map dialog applies and set the contour level and redraw 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002dsampling_002drate"></a>
<div class="header">
<p>
Next: <a href="#get_002dmap_002dsampling_002drate" accesskey="n" rel="next">get-map-sampling-rate</a>, Previous: <a href="#set_002dmap_002dsampling_002drate_002dtext" accesskey="p" rel="previous">set-map-sampling-rate-text</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dsampling_002drate-1"></a>
<h4 class="subsection">11.15.6 set-map-sampling-rate</h4>
<dl>
<dt><a name="index-set_002dmap_002dsampling_002drate"></a>function: <strong>set-map-sampling-rate</strong> <em>r</em></dt>
<dd>
<p>Where <em>r</em> is a number
</p>
<p>set the map sampling rate (default 1.5) 
</p>
<p>Set to something like 2.0 or 2.5 for more finely sampled maps. Useful for baton-building low resolution maps. 
</p>
</dd></dl>

<hr>
<a name="get_002dmap_002dsampling_002drate"></a>
<div class="header">
<p>
Next: <a href="#change_002dcontour_002dlevel" accesskey="n" rel="next">change-contour-level</a>, Previous: <a href="#set_002dmap_002dsampling_002drate" accesskey="p" rel="previous">set-map-sampling-rate</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dmap_002dsampling_002drate-1"></a>
<h4 class="subsection">11.15.7 get-map-sampling-rate</h4>
<dl>
<dt><a name="index-get_002dmap_002dsampling_002drate"></a>function: <strong>get-map-sampling-rate</strong></dt>
<dd>
<p>return the map sampling rate 
</p>
</dd></dl>

<hr>
<a name="change_002dcontour_002dlevel"></a>
<div class="header">
<p>
Next: <a href="#set_002dlast_002dmap_002dcontour_002dlevel" accesskey="n" rel="next">set-last-map-contour-level</a>, Previous: <a href="#get_002dmap_002dsampling_002drate" accesskey="p" rel="previous">get-map-sampling-rate</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="change_002dcontour_002dlevel-1"></a>
<h4 class="subsection">11.15.8 change-contour-level</h4>
<dl>
<dt><a name="index-change_002dcontour_002dlevel"></a>function: <strong>change-contour-level</strong> <em>is_increment</em></dt>
<dd>
<p>Where <em>is_increment</em> is an integer number
</p>
<p>change the contour level of the current map by a step 
</p>
<p>if is_increment=1 the contour level is increased. If is_increment=0 the map contour level is decreased. 
</p>
</dd></dl>

<hr>
<a name="set_002dlast_002dmap_002dcontour_002dlevel"></a>
<div class="header">
<p>
Next: <a href="#set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma" accesskey="n" rel="next">set-last-map-contour-level-by-sigma</a>, Previous: <a href="#change_002dcontour_002dlevel" accesskey="p" rel="previous">change-contour-level</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dlast_002dmap_002dcontour_002dlevel-1"></a>
<h4 class="subsection">11.15.9 set-last-map-contour-level</h4>
<dl>
<dt><a name="index-set_002dlast_002dmap_002dcontour_002dlevel"></a>function: <strong>set-last-map-contour-level</strong> <em>level</em></dt>
<dd>
<p>Where <em>level</em> is a number
</p>
<p>set the contour level of the map with the highest molecule number to level 
</p>
</dd></dl>

<hr>
<a name="set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma"></a>
<div class="header">
<p>
Next: <a href="#set_002dstop_002dscroll_002ddiff_002dmap" accesskey="n" rel="next">set-stop-scroll-diff-map</a>, Previous: <a href="#set_002dlast_002dmap_002dcontour_002dlevel" accesskey="p" rel="previous">set-last-map-contour-level</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma-1"></a>
<h4 class="subsection">11.15.10 set-last-map-contour-level-by-sigma</h4>
<dl>
<dt><a name="index-set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma"></a>function: <strong>set-last-map-contour-level-by-sigma</strong> <em>n_sigma</em></dt>
<dd>
<p>Where <em>n_sigma</em> is a number
</p>
<p>set the contour level of the map with the highest molecule number to n_sigma sigma 
</p>
</dd></dl>

<hr>
<a name="set_002dstop_002dscroll_002ddiff_002dmap"></a>
<div class="header">
<p>
Next: <a href="#set_002dstop_002dscroll_002diso_002dmap" accesskey="n" rel="next">set-stop-scroll-iso-map</a>, Previous: <a href="#set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma" accesskey="p" rel="previous">set-last-map-contour-level-by-sigma</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dstop_002dscroll_002ddiff_002dmap-1"></a>
<h4 class="subsection">11.15.11 set-stop-scroll-diff-map</h4>
<dl>
<dt><a name="index-set_002dstop_002dscroll_002ddiff_002dmap"></a>function: <strong>set-stop-scroll-diff-map</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>create a lower limit to the &quot;Fo-Fc-style&quot; map contour level changing 
</p>
<p>(default 1 on) 
</p>
</dd></dl>

<hr>
<a name="set_002dstop_002dscroll_002diso_002dmap"></a>
<div class="header">
<p>
Next: <a href="#set_002dstop_002dscroll_002diso_002dmap_002dlevel" accesskey="n" rel="next">set-stop-scroll-iso-map-level</a>, Previous: <a href="#set_002dstop_002dscroll_002ddiff_002dmap" accesskey="p" rel="previous">set-stop-scroll-diff-map</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dstop_002dscroll_002diso_002dmap-1"></a>
<h4 class="subsection">11.15.12 set-stop-scroll-iso-map</h4>
<dl>
<dt><a name="index-set_002dstop_002dscroll_002diso_002dmap"></a>function: <strong>set-stop-scroll-iso-map</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>create a lower limit to the &quot;2Fo-Fc-style&quot; map contour level changing 
</p>
<p>(default 1 on) 
</p>
</dd></dl>

<hr>
<a name="set_002dstop_002dscroll_002diso_002dmap_002dlevel"></a>
<div class="header">
<p>
Next: <a href="#set_002dstop_002dscroll_002ddiff_002dmap_002dlevel" accesskey="n" rel="next">set-stop-scroll-diff-map-level</a>, Previous: <a href="#set_002dstop_002dscroll_002diso_002dmap" accesskey="p" rel="previous">set-stop-scroll-iso-map</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dstop_002dscroll_002diso_002dmap_002dlevel-1"></a>
<h4 class="subsection">11.15.13 set-stop-scroll-iso-map-level</h4>
<dl>
<dt><a name="index-set_002dstop_002dscroll_002diso_002dmap_002dlevel"></a>function: <strong>set-stop-scroll-iso-map-level</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the actual map level changing limit 
</p>
<p>(default 0.0) 
</p>
</dd></dl>

<hr>
<a name="set_002dstop_002dscroll_002ddiff_002dmap_002dlevel"></a>
<div class="header">
<p>
Next: <a href="#set_002dresidue_002ddensity_002dfit_002dscale_002dfactor" accesskey="n" rel="next">set-residue-density-fit-scale-factor</a>, Previous: <a href="#set_002dstop_002dscroll_002diso_002dmap_002dlevel" accesskey="p" rel="previous">set-stop-scroll-iso-map-level</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dstop_002dscroll_002ddiff_002dmap_002dlevel-1"></a>
<h4 class="subsection">11.15.14 set-stop-scroll-diff-map-level</h4>
<dl>
<dt><a name="index-set_002dstop_002dscroll_002ddiff_002dmap_002dlevel"></a>function: <strong>set-stop-scroll-diff-map-level</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the actual difference map level changing limit 
</p>
<p>(default 0.0) 
</p>
</dd></dl>

<hr>
<a name="set_002dresidue_002ddensity_002dfit_002dscale_002dfactor"></a>
<div class="header">
<p>
Previous: <a href="#set_002dstop_002dscroll_002ddiff_002dmap_002dlevel" accesskey="p" rel="previous">set-stop-scroll-diff-map-level</a>, Up: <a href="#Density-Increment" accesskey="u" rel="up">Density Increment</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dresidue_002ddensity_002dfit_002dscale_002dfactor-1"></a>
<h4 class="subsection">11.15.15 set-residue-density-fit-scale-factor</h4>
<dl>
<dt><a name="index-set_002dresidue_002ddensity_002dfit_002dscale_002dfactor"></a>function: <strong>set-residue-density-fit-scale-factor</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the scale factor for the Residue Density fit analysis 
</p>
</dd></dl>

<hr>
<a name="Density-Functions"></a>
<div class="header">
<p>
Next: <a href="#Parameters-from-map" accesskey="n" rel="next">Parameters from map</a>, Previous: <a href="#Density-Increment" accesskey="p" rel="previous">Density Increment</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Density-Functions-1"></a>
<h3 class="section">11.16 Density Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dline_002dwidth" accesskey="1">set-map-line-width</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dline_002dwidth_002dstate" accesskey="2">map-line-width-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dand_002ddraw_002dmap" accesskey="3">make-and-draw-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams" accesskey="4">make-and-draw-map-with-refmac-params</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams" accesskey="5">make-and-draw-map-with-reso-with-refmac-params</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dupdating_002dmap" accesskey="6">make-updating-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#valid_002dlabels" accesskey="7">valid-labels</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mtz_002dfile_002dhas_002dphases_002dp" accesskey="8">mtz-file-has-phases-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#is_002dmtz_002dfile_002dp" accesskey="9">is-mtz-file-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#cns_002dfile_002dhas_002dphases_002dp">cns-file-has-phases-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate">auto-read-do-difference-map-too-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dauto_002dread_002dcolumn_002dlabels">set-auto-read-column-labels</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dradius">set-map-radius</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dradius_002dem">set-map-radius-em</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddensity_002dsize">set-density-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddisplay_002dintro_002dstring">set-display-intro-string</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dmap_002dradius">get-map-radius</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002desoteric_002ddepth_002dcue">set-esoteric-depth-cue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#esoteric_002ddepth_002dcue_002dstate">esoteric-depth-cue-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dswap_002ddifference_002dmap_002dcolours">set-swap-difference-map-colours</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dis_002ddifference_002dmap">set-map-is-difference-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dis_002ddifference_002dmap">map-is-difference-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#another_002dlevel">another-level</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber">another-level-from-map-molecule-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residue_002ddensity_002dfit_002dscale_002dfactor">residue-density-fit-scale-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#density_002dat_002dpoint">density-at-point</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dmap_002dline_002dwidth"></a>
<div class="header">
<p>
Next: <a href="#map_002dline_002dwidth_002dstate" accesskey="n" rel="next">map-line-width-state</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dline_002dwidth-1"></a>
<h4 class="subsection">11.16.1 set-map-line-width</h4>
<dl>
<dt><a name="index-set_002dmap_002dline_002dwidth"></a>function: <strong>set-map-line-width</strong> <em>w</em></dt>
<dd>
<p>Where <em>w</em> is an integer number
</p>
<p>draw the lines of the chickenwire density in width w 
</p>
</dd></dl>

<hr>
<a name="map_002dline_002dwidth_002dstate"></a>
<div class="header">
<p>
Next: <a href="#make_002dand_002ddraw_002dmap" accesskey="n" rel="next">make-and-draw-map</a>, Previous: <a href="#set_002dmap_002dline_002dwidth" accesskey="p" rel="previous">set-map-line-width</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dline_002dwidth_002dstate-1"></a>
<h4 class="subsection">11.16.2 map-line-width-state</h4>
<dl>
<dt><a name="index-map_002dline_002dwidth_002dstate"></a>function: <strong>map-line-width-state</strong></dt>
<dd>
<p>return the width in which density contours are drawn 
</p>
</dd></dl>

<hr>
<a name="make_002dand_002ddraw_002dmap"></a>
<div class="header">
<p>
Next: <a href="#make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams" accesskey="n" rel="next">make-and-draw-map-with-refmac-params</a>, Previous: <a href="#map_002dline_002dwidth_002dstate" accesskey="p" rel="previous">map-line-width-state</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dand_002ddraw_002dmap-1"></a>
<h4 class="subsection">11.16.3 make-and-draw-map</h4>
<dl>
<dt><a name="index-make_002dand_002ddraw_002dmap"></a>function: <strong>make-and-draw-map</strong> <em>mtz_file_name f_col phi_col weight use_weights is_diff_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mtz_file_name</em> is a string
</li><li> <em>f_col</em> is a string
</li><li> <em>phi_col</em> is a string
</li><li> <em>weight</em> is a string
</li><li> <em>use_weights</em> is an integer number
</li><li> <em>is_diff_map</em> is an integer number
</li></ul>

<p>make a map from an mtz file (simple interface) 
</p>
<p>given mtz file mtz_file_name and F column f_col and phases column phi_col and optional weight column weight_col (pass use_weights=0 if weights are not to be used). Also mark the map as a difference map (is_diff_map=1) or not (is_diff_map=0) because they are handled differently inside coot.
</p>
<p>Returns: -1 on error, else return the new molecule number 
</p>
</dd></dl>

<hr>
<a name="make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams"></a>
<div class="header">
<p>
Next: <a href="#make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams" accesskey="n" rel="next">make-and-draw-map-with-reso-with-refmac-params</a>, Previous: <a href="#make_002dand_002ddraw_002dmap" accesskey="p" rel="previous">make-and-draw-map</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams-1"></a>
<h4 class="subsection">11.16.4 make-and-draw-map-with-refmac-params</h4>
<dl>
<dt><a name="index-make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams"></a>function: <strong>make-and-draw-map-with-refmac-params</strong> <em>mtz_file_name a b weight use_weights is_diff_map have_refmac_params fobs_col sigfobs_col r_free_col sensible_f_free_col</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mtz_file_name</em> is a string
</li><li> <em>a</em> is a string
</li><li> <em>b</em> is a string
</li><li> <em>weight</em> is a string
</li><li> <em>use_weights</em> is an integer number
</li><li> <em>is_diff_map</em> is an integer number
</li><li> <em>have_refmac_params</em> is an integer number
</li><li> <em>fobs_col</em> is a string
</li><li> <em>sigfobs_col</em> is a string
</li><li> <em>r_free_col</em> is a string
</li><li> <em>sensible_f_free_col</em> is an integer number
</li></ul>

<p>as the above function, execpt set refmac parameters too 
</p>
<p>pass along the refmac column labels for storage (not used in the creation of the map)
</p>
<p>Returns: -1 on error, else return imol 
</p>
</dd></dl>

<hr>
<a name="make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams"></a>
<div class="header">
<p>
Next: <a href="#make_002dupdating_002dmap" accesskey="n" rel="next">make-updating-map</a>, Previous: <a href="#make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams" accesskey="p" rel="previous">make-and-draw-map-with-refmac-params</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams-1"></a>
<h4 class="subsection">11.16.5 make-and-draw-map-with-reso-with-refmac-params</h4>
<dl>
<dt><a name="index-make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams"></a>function: <strong>make-and-draw-map-with-reso-with-refmac-params</strong> <em>mtz_file_name a b weight use_weights is_diff_map have_refmac_params fobs_col sigfobs_col r_free_col sensible_f_free_col is_anomalous use_reso_limits low_reso_limit high_reso_lim</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mtz_file_name</em> is a string
</li><li> <em>a</em> is a string
</li><li> <em>b</em> is a string
</li><li> <em>weight</em> is a string
</li><li> <em>use_weights</em> is an integer number
</li><li> <em>is_diff_map</em> is an integer number
</li><li> <em>have_refmac_params</em> is an integer number
</li><li> <em>fobs_col</em> is a string
</li><li> <em>sigfobs_col</em> is a string
</li><li> <em>r_free_col</em> is a string
</li><li> <em>sensible_f_free_col</em> is an integer number
</li><li> <em>is_anomalous</em> is an integer number
</li><li> <em>use_reso_limits</em> is an integer number
</li><li> <em>low_reso_limit</em> is a number
</li><li> <em>high_reso_lim</em> is a number
</li></ul>

<p>as the above function, except set expert options too. 
</p>
</dd></dl>

<hr>
<a name="make_002dupdating_002dmap"></a>
<div class="header">
<p>
Next: <a href="#valid_002dlabels" accesskey="n" rel="next">valid-labels</a>, Previous: <a href="#make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams" accesskey="p" rel="previous">make-and-draw-map-with-reso-with-refmac-params</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dupdating_002dmap-1"></a>
<h4 class="subsection">11.16.6 make-updating-map</h4>
<dl>
<dt><a name="index-make_002dupdating_002dmap"></a>function: <strong>make-updating-map</strong> <em>mtz_file_name f_col phi_col weight use_weights is_diff_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mtz_file_name</em> is a string
</li><li> <em>f_col</em> is a string
</li><li> <em>phi_col</em> is a string
</li><li> <em>weight</em> is a string
</li><li> <em>use_weights</em> is an integer number
</li><li> <em>is_diff_map</em> is an integer number
</li></ul>

<p>make a map molecule from the give file name. 
</p>
<p>If the file updates, then the map will be updated. 
</p>
</dd></dl>

<hr>
<a name="valid_002dlabels"></a>
<div class="header">
<p>
Next: <a href="#mtz_002dfile_002dhas_002dphases_002dp" accesskey="n" rel="next">mtz-file-has-phases-p</a>, Previous: <a href="#make_002dupdating_002dmap" accesskey="p" rel="previous">make-updating-map</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="valid_002dlabels-1"></a>
<h4 class="subsection">11.16.7 valid-labels</h4>
<dl>
<dt><a name="index-valid_002dlabels"></a>function: <strong>valid-labels</strong> <em>mtz_file_name f_col phi_col weight_col use_weights</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mtz_file_name</em> is a string
</li><li> <em>f_col</em> is a string
</li><li> <em>phi_col</em> is a string
</li><li> <em>weight_col</em> is a string
</li><li> <em>use_weights</em> is an integer number
</li></ul>

<p>does the mtz file have the columms that we want it to have? 
</p>
</dd></dl>

<hr>
<a name="mtz_002dfile_002dhas_002dphases_002dp"></a>
<div class="header">
<p>
Next: <a href="#is_002dmtz_002dfile_002dp" accesskey="n" rel="next">is-mtz-file-p</a>, Previous: <a href="#valid_002dlabels" accesskey="p" rel="previous">valid-labels</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mtz_002dfile_002dhas_002dphases_002dp-1"></a>
<h4 class="subsection">11.16.8 mtz-file-has-phases-p</h4>
<dl>
<dt><a name="index-mtz_002dfile_002dhas_002dphases_002dp"></a>function: <strong>mtz-file-has-phases-p</strong> <em>mtz_file_name</em></dt>
<dd>
<p>Where <em>mtz_file_name</em> is a string
</p>
<p>does the mtz file have phases? 
</p>
</dd></dl>

<hr>
<a name="is_002dmtz_002dfile_002dp"></a>
<div class="header">
<p>
Next: <a href="#cns_002dfile_002dhas_002dphases_002dp" accesskey="n" rel="next">cns-file-has-phases-p</a>, Previous: <a href="#mtz_002dfile_002dhas_002dphases_002dp" accesskey="p" rel="previous">mtz-file-has-phases-p</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="is_002dmtz_002dfile_002dp-1"></a>
<h4 class="subsection">11.16.9 is-mtz-file-p</h4>
<dl>
<dt><a name="index-is_002dmtz_002dfile_002dp"></a>function: <strong>is-mtz-file-p</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>is the given filename an mtz file? 
</p>
</dd></dl>

<hr>
<a name="cns_002dfile_002dhas_002dphases_002dp"></a>
<div class="header">
<p>
Next: <a href="#auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate" accesskey="n" rel="next">auto-read-do-difference-map-too-state</a>, Previous: <a href="#is_002dmtz_002dfile_002dp" accesskey="p" rel="previous">is-mtz-file-p</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="cns_002dfile_002dhas_002dphases_002dp-1"></a>
<h4 class="subsection">11.16.10 cns-file-has-phases-p</h4>
<dl>
<dt><a name="index-cns_002dfile_002dhas_002dphases_002dp"></a>function: <strong>cns-file-has-phases-p</strong> <em>cns_file_name</em></dt>
<dd>
<p>Where <em>cns_file_name</em> is a string
</p>
<p>does the given file have cns phases? 
</p>
</dd></dl>

<hr>
<a name="auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dauto_002dread_002dcolumn_002dlabels" accesskey="n" rel="next">set-auto-read-column-labels</a>, Previous: <a href="#cns_002dfile_002dhas_002dphases_002dp" accesskey="p" rel="previous">cns-file-has-phases-p</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate-1"></a>
<h4 class="subsection">11.16.11 auto-read-do-difference-map-too-state</h4>
<dl>
<dt><a name="index-auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate"></a>function: <strong>auto-read-do-difference-map-too-state</strong></dt>
<dd>
<p>return the flag to do a difference map (too) on auto-read MTZ 
</p>
<p>Returns: 0 means no, 1 means yes. 
</p>
</dd></dl>

<hr>
<a name="set_002dauto_002dread_002dcolumn_002dlabels"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002dradius" accesskey="n" rel="next">set-map-radius</a>, Previous: <a href="#auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate" accesskey="p" rel="previous">auto-read-do-difference-map-too-state</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dauto_002dread_002dcolumn_002dlabels-1"></a>
<h4 class="subsection">11.16.12 set-auto-read-column-labels</h4>
<dl>
<dt><a name="index-set_002dauto_002dread_002dcolumn_002dlabels"></a>function: <strong>set-auto-read-column-labels</strong> <em>fwt phwt is_for_diff_map_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>fwt</em> is a string
</li><li> <em>phwt</em> is a string
</li><li> <em>is_for_diff_map_flag</em> is an integer number
</li></ul>

<p>set the expected MTZ columns for Auto-reading MTZ file. 
</p>
<p>Not every program uses the default refmac labels (&quot;FWT&quot;/&quot;PHWT&quot;) for its MTZ file. Here we can tell coot to expect other labels so that coot can &quot;Auto-open&quot; such MTZ files.
</p>
<p>e.g. (set-auto-read-column-labels &quot;2FOFCWT&quot; &quot;PH2FOFCWT&quot; 0) 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002dradius"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002dradius_002dem" accesskey="n" rel="next">set-map-radius-em</a>, Previous: <a href="#set_002dauto_002dread_002dcolumn_002dlabels" accesskey="p" rel="previous">set-auto-read-column-labels</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dradius-1"></a>
<h4 class="subsection">11.16.13 set-map-radius</h4>
<dl>
<dt><a name="index-set_002dmap_002dradius"></a>function: <strong>set-map-radius</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the extent of the box/radius of electron density contours for x-ray maps 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002dradius_002dem"></a>
<div class="header">
<p>
Next: <a href="#set_002ddensity_002dsize" accesskey="n" rel="next">set-density-size</a>, Previous: <a href="#set_002dmap_002dradius" accesskey="p" rel="previous">set-map-radius</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dradius_002dem-1"></a>
<h4 class="subsection">11.16.14 set-map-radius-em</h4>
<dl>
<dt><a name="index-set_002dmap_002dradius_002dem"></a>function: <strong>set-map-radius-em</strong> <em>radius</em></dt>
<dd>
<p>Where <em>radius</em> is a number
</p>
<p>set the extent of the box/radius of electron density contours for EM map 
</p>
</dd></dl>

<hr>
<a name="set_002ddensity_002dsize"></a>
<div class="header">
<p>
Next: <a href="#set_002ddisplay_002dintro_002dstring" accesskey="n" rel="next">set-display-intro-string</a>, Previous: <a href="#set_002dmap_002dradius_002dem" accesskey="p" rel="previous">set-map-radius-em</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddensity_002dsize-1"></a>
<h4 class="subsection">11.16.15 set-density-size</h4>
<dl>
<dt><a name="index-set_002ddensity_002dsize"></a>function: <strong>set-density-size</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>another (old) way of setting the radius of the map 
</p>
</dd></dl>

<hr>
<a name="set_002ddisplay_002dintro_002dstring"></a>
<div class="header">
<p>
Next: <a href="#get_002dmap_002dradius" accesskey="n" rel="next">get-map-radius</a>, Previous: <a href="#set_002ddensity_002dsize" accesskey="p" rel="previous">set-density-size</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddisplay_002dintro_002dstring-1"></a>
<h4 class="subsection">11.16.16 set-display-intro-string</h4>
<dl>
<dt><a name="index-set_002ddisplay_002dintro_002dstring"></a>function: <strong>set-display-intro-string</strong> <em>str</em></dt>
<dd>
<p>Where <em>str</em> is a string
</p>
<p>Give me this nice message str when I start coot. 
</p>
</dd></dl>

<hr>
<a name="get_002dmap_002dradius"></a>
<div class="header">
<p>
Next: <a href="#set_002desoteric_002ddepth_002dcue" accesskey="n" rel="next">set-esoteric-depth-cue</a>, Previous: <a href="#set_002ddisplay_002dintro_002dstring" accesskey="p" rel="previous">set-display-intro-string</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dmap_002dradius-1"></a>
<h4 class="subsection">11.16.17 get-map-radius</h4>
<dl>
<dt><a name="index-get_002dmap_002dradius"></a>function: <strong>get-map-radius</strong></dt>
<dd>
<p>return the extent of the box/radius of electron density contours 
</p>
</dd></dl>

<hr>
<a name="set_002desoteric_002ddepth_002dcue"></a>
<div class="header">
<p>
Next: <a href="#esoteric_002ddepth_002dcue_002dstate" accesskey="n" rel="next">esoteric-depth-cue-state</a>, Previous: <a href="#get_002dmap_002dradius" accesskey="p" rel="previous">get-map-radius</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002desoteric_002ddepth_002dcue-1"></a>
<h4 class="subsection">11.16.18 set-esoteric-depth-cue</h4>
<dl>
<dt><a name="index-set_002desoteric_002ddepth_002dcue"></a>function: <strong>set-esoteric-depth-cue</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>not everone likes coot&rsquo;s esoteric depth cueing system 
</p>
<p>Pass an argument istate=1 to turn it off
</p>
<p>(this function is currently disabled). 
</p>
</dd></dl>

<hr>
<a name="esoteric_002ddepth_002dcue_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dswap_002ddifference_002dmap_002dcolours" accesskey="n" rel="next">set-swap-difference-map-colours</a>, Previous: <a href="#set_002desoteric_002ddepth_002dcue" accesskey="p" rel="previous">set-esoteric-depth-cue</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="esoteric_002ddepth_002dcue_002dstate-1"></a>
<h4 class="subsection">11.16.19 esoteric-depth-cue-state</h4>
<dl>
<dt><a name="index-esoteric_002ddepth_002dcue_002dstate"></a>function: <strong>esoteric-depth-cue-state</strong></dt>
<dd>
<p>native depth cueing system 
</p>
<p>return the state of the esoteric depth cueing flag 
</p>
</dd></dl>

<hr>
<a name="set_002dswap_002ddifference_002dmap_002dcolours"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002dis_002ddifference_002dmap" accesskey="n" rel="next">set-map-is-difference-map</a>, Previous: <a href="#esoteric_002ddepth_002dcue_002dstate" accesskey="p" rel="previous">esoteric-depth-cue-state</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dswap_002ddifference_002dmap_002dcolours-1"></a>
<h4 class="subsection">11.16.20 set-swap-difference-map-colours</h4>
<dl>
<dt><a name="index-set_002dswap_002ddifference_002dmap_002dcolours"></a>function: <strong>set-swap-difference-map-colours</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>not everone likes coot&rsquo;s default difference map colouring. 
</p>
<p>Pass an argument i=1 to swap the difference map colouring so that red is positive and green is negative. 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002dis_002ddifference_002dmap"></a>
<div class="header">
<p>
Next: <a href="#map_002dis_002ddifference_002dmap" accesskey="n" rel="next">map-is-difference-map</a>, Previous: <a href="#set_002dswap_002ddifference_002dmap_002dcolours" accesskey="p" rel="previous">set-swap-difference-map-colours</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dis_002ddifference_002dmap-1"></a>
<h4 class="subsection">11.16.21 set-map-is-difference-map</h4>
<dl>
<dt><a name="index-set_002dmap_002dis_002ddifference_002dmap"></a>function: <strong>set-map-is-difference-map</strong> <em>imol bool_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>bool_flag</em> is an integer number
</li></ul>

<p>post-hoc set the map of molecule number imol to be a difference map 
</p>
<p>Returns: success status, 0 -&gt; failure (imol does not have a map) 
</p>
</dd></dl>

<hr>
<a name="map_002dis_002ddifference_002dmap"></a>
<div class="header">
<p>
Next: <a href="#another_002dlevel" accesskey="n" rel="next">another-level</a>, Previous: <a href="#set_002dmap_002dis_002ddifference_002dmap" accesskey="p" rel="previous">set-map-is-difference-map</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dis_002ddifference_002dmap-1"></a>
<h4 class="subsection">11.16.22 map-is-difference-map</h4>
<dl>
<dt><a name="index-map_002dis_002ddifference_002dmap"></a>function: <strong>map-is-difference-map</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>map is difference map? 
</p>
</dd></dl>

<hr>
<a name="another_002dlevel"></a>
<div class="header">
<p>
Next: <a href="#another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber" accesskey="n" rel="next">another-level-from-map-molecule-number</a>, Previous: <a href="#map_002dis_002ddifference_002dmap" accesskey="p" rel="previous">map-is-difference-map</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="another_002dlevel-1"></a>
<h4 class="subsection">11.16.23 another-level</h4>
<dl>
<dt><a name="index-another_002dlevel"></a>function: <strong>another-level</strong></dt>
<dd>
<p>Add another contour level for the last added map. 
</p>
<p>Currently, the map must have been generated from an MTZ file. 
</p>
<p>Returns: the molecule number of the new molecule or -1 on failure 
</p>
</dd></dl>

<hr>
<a name="another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#residue_002ddensity_002dfit_002dscale_002dfactor" accesskey="n" rel="next">residue-density-fit-scale-factor</a>, Previous: <a href="#another_002dlevel" accesskey="p" rel="previous">another-level</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber-1"></a>
<h4 class="subsection">11.16.24 another-level-from-map-molecule-number</h4>
<dl>
<dt><a name="index-another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber"></a>function: <strong>another-level-from-map-molecule-number</strong> <em>imap</em></dt>
<dd>
<p>Where <em>imap</em> is an integer number
</p>
<p>Add another contour level for the given map. 
</p>
<p>Currently, the map must have been generated from an MTZ file. 
</p>
<p>Returns: the molecule number of the new molecule or -1 on failure 
</p>
</dd></dl>

<hr>
<a name="residue_002ddensity_002dfit_002dscale_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#density_002dat_002dpoint" accesskey="n" rel="next">density-at-point</a>, Previous: <a href="#another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber" accesskey="p" rel="previous">another-level-from-map-molecule-number</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residue_002ddensity_002dfit_002dscale_002dfactor-1"></a>
<h4 class="subsection">11.16.25 residue-density-fit-scale-factor</h4>
<dl>
<dt><a name="index-residue_002ddensity_002dfit_002dscale_002dfactor"></a>function: <strong>residue-density-fit-scale-factor</strong></dt>
<dd>
<p>return the scale factor for the Residue Density fit analysis 
</p>
</dd></dl>

<hr>
<a name="density_002dat_002dpoint"></a>
<div class="header">
<p>
Previous: <a href="#residue_002ddensity_002dfit_002dscale_002dfactor" accesskey="p" rel="previous">residue-density-fit-scale-factor</a>, Up: <a href="#Density-Functions" accesskey="u" rel="up">Density Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="density_002dat_002dpoint-1"></a>
<h4 class="subsection">11.16.26 density-at-point</h4>
<dl>
<dt><a name="index-density_002dat_002dpoint"></a>function: <strong>density-at-point</strong> <em>imol_map x y z</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li></ul>

<p>return the density at the given point for the given map. Return 0 for bad imol 
</p>
</dd></dl>

<hr>
<a name="Parameters-from-map"></a>
<div class="header">
<p>
Next: <a href="#PDB-Functions" accesskey="n" rel="next">PDB Functions</a>, Previous: <a href="#Density-Functions" accesskey="p" rel="previous">Density Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Parameters-from-map-1"></a>
<h3 class="section">11.17 Parameters from map</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#mtz_002dhklin_002dfor_002dmap" accesskey="1">mtz-hklin-for-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mtz_002dfp_002dfor_002dmap" accesskey="2">mtz-fp-for-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mtz_002dphi_002dfor_002dmap" accesskey="3">mtz-phi-for-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mtz_002dweight_002dfor_002dmap" accesskey="4">mtz-weight-for-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mtz_002duse_002dweight_002dfor_002dmap" accesskey="5">mtz-use-weight-for-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dparameters_002dscm" accesskey="6">map-parameters-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#cell_002dscm" accesskey="7">cell-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dparameters_002dpy" accesskey="8">map-parameters-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#cell_002dpy" accesskey="9">cell-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="mtz_002dhklin_002dfor_002dmap"></a>
<div class="header">
<p>
Next: <a href="#mtz_002dfp_002dfor_002dmap" accesskey="n" rel="next">mtz-fp-for-map</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mtz_002dhklin_002dfor_002dmap-1"></a>
<h4 class="subsection">11.17.1 mtz-hklin-for-map</h4>
<dl>
<dt><a name="index-mtz_002dhklin_002dfor_002dmap"></a>function: <strong>mtz-hklin-for-map</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>return the mtz file that was use to generate the map 
</p>
<p>return 0 when there is no mtz file associated with that map (it was generated from a CCP4 map file say). 
</p>
</dd></dl>

<hr>
<a name="mtz_002dfp_002dfor_002dmap"></a>
<div class="header">
<p>
Next: <a href="#mtz_002dphi_002dfor_002dmap" accesskey="n" rel="next">mtz-phi-for-map</a>, Previous: <a href="#mtz_002dhklin_002dfor_002dmap" accesskey="p" rel="previous">mtz-hklin-for-map</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mtz_002dfp_002dfor_002dmap-1"></a>
<h4 class="subsection">11.17.2 mtz-fp-for-map</h4>
<dl>
<dt><a name="index-mtz_002dfp_002dfor_002dmap"></a>function: <strong>mtz-fp-for-map</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>return the FP column in the file that was use to generate the map 
</p>
<p>return 0 when there is no mtz file associated with that map (it was generated from a CCP4 map file say).
</p>
<p>Caller should dispose of returned pointer. 
</p>
</dd></dl>

<hr>
<a name="mtz_002dphi_002dfor_002dmap"></a>
<div class="header">
<p>
Next: <a href="#mtz_002dweight_002dfor_002dmap" accesskey="n" rel="next">mtz-weight-for-map</a>, Previous: <a href="#mtz_002dfp_002dfor_002dmap" accesskey="p" rel="previous">mtz-fp-for-map</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mtz_002dphi_002dfor_002dmap-1"></a>
<h4 class="subsection">11.17.3 mtz-phi-for-map</h4>
<dl>
<dt><a name="index-mtz_002dphi_002dfor_002dmap"></a>function: <strong>mtz-phi-for-map</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>return the phases column in mtz file that was use to generate the map 
</p>
<p>return 0 when there is no mtz file associated with that map (it was generated from a CCP4 map file say). Caller should dispose of returned pointer. 
</p>
</dd></dl>

<hr>
<a name="mtz_002dweight_002dfor_002dmap"></a>
<div class="header">
<p>
Next: <a href="#mtz_002duse_002dweight_002dfor_002dmap" accesskey="n" rel="next">mtz-use-weight-for-map</a>, Previous: <a href="#mtz_002dphi_002dfor_002dmap" accesskey="p" rel="previous">mtz-phi-for-map</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mtz_002dweight_002dfor_002dmap-1"></a>
<h4 class="subsection">11.17.4 mtz-weight-for-map</h4>
<dl>
<dt><a name="index-mtz_002dweight_002dfor_002dmap"></a>function: <strong>mtz-weight-for-map</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>return the weight column in the mtz file that was use to generate the map 
</p>
<p>return 0 when there is no mtz file associated with that map (it was generated from a CCP4 map file say) or no weights were used. Caller should dispose of returned pointer. 
</p>
</dd></dl>

<hr>
<a name="mtz_002duse_002dweight_002dfor_002dmap"></a>
<div class="header">
<p>
Next: <a href="#map_002dparameters_002dscm" accesskey="n" rel="next">map-parameters-scm</a>, Previous: <a href="#mtz_002dweight_002dfor_002dmap" accesskey="p" rel="previous">mtz-weight-for-map</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mtz_002duse_002dweight_002dfor_002dmap-1"></a>
<h4 class="subsection">11.17.5 mtz-use-weight-for-map</h4>
<dl>
<dt><a name="index-mtz_002duse_002dweight_002dfor_002dmap"></a>function: <strong>mtz-use-weight-for-map</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>return flag for whether weights were used that was use to generate the map 
</p>
<p>return 0 when no weights were used or there is no mtz file associated with that map. 
</p>
</dd></dl>

<hr>
<a name="map_002dparameters_002dscm"></a>
<div class="header">
<p>
Next: <a href="#cell_002dscm" accesskey="n" rel="next">cell-scm</a>, Previous: <a href="#mtz_002duse_002dweight_002dfor_002dmap" accesskey="p" rel="previous">mtz-use-weight-for-map</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dparameters_002dscm-1"></a>
<h4 class="subsection">11.17.6 map-parameters-scm</h4>
<dl>
<dt><a name="index-map_002dparameters_002dscm"></a>function: <strong>map-parameters-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the parameter that made the map, 
</p>
<p>Returns: false or a string like (&quot;xxx.mtz&quot; &quot;FPH&quot; &quot;PHWT&quot; &quot;&quot; False) 
</p>
</dd></dl>

<hr>
<a name="cell_002dscm"></a>
<div class="header">
<p>
Next: <a href="#map_002dparameters_002dpy" accesskey="n" rel="next">map-parameters-py</a>, Previous: <a href="#map_002dparameters_002dscm" accesskey="p" rel="previous">map-parameters-scm</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="cell_002dscm-1"></a>
<h4 class="subsection">11.17.7 cell-scm</h4>
<dl>
<dt><a name="index-cell_002dscm"></a>function: <strong>cell-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the parameter that made the map, 
</p>
<p>Returns: false or a list like (45 46 47 90 90 120), angles in degress 
</p>
</dd></dl>

<hr>
<a name="map_002dparameters_002dpy"></a>
<div class="header">
<p>
Next: <a href="#cell_002dpy" accesskey="n" rel="next">cell-py</a>, Previous: <a href="#cell_002dscm" accesskey="p" rel="previous">cell-scm</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dparameters_002dpy-1"></a>
<h4 class="subsection">11.17.8 map-parameters-py</h4>
<dl>
<dt><a name="index-map_002dparameters_002dpy"></a>function: <strong>map-parameters-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the parameter of the molecule, something like (45 46 47 90 90 120), angles in degress 
</p>
<p>return the parameter that made the map,
</p>
<p>Returns: False or something like [&quot;xxx.mtz&quot;, &quot;FPH&quot;, &quot;PHWT&quot;, &quot;&quot;, False] 
</p>
</dd></dl>

<hr>
<a name="cell_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#map_002dparameters_002dpy" accesskey="p" rel="previous">map-parameters-py</a>, Up: <a href="#Parameters-from-map" accesskey="u" rel="up">Parameters from map</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="cell_002dpy-1"></a>
<h4 class="subsection">11.17.9 cell-py</h4>
<dl>
<dt><a name="index-cell_002dpy"></a>function: <strong>cell-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the parameter that made the map, 
</p>
<p>Returns: False or something like [45, 46, 47, 90, 90, 120], angles in degress 
</p>
</dd></dl>

<hr>
<a name="PDB-Functions"></a>
<div class="header">
<p>
Next: <a href="#Info-Dialog" accesskey="n" rel="next">Info Dialog</a>, Previous: <a href="#Parameters-from-map" accesskey="p" rel="previous">Parameters from map</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="PDB-Functions-1"></a>
<h3 class="section">11.18 PDB Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#write_002dpdb_002dfile" accesskey="1">write-pdb-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#write_002dcif_002dfile" accesskey="2">write-cif-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#write_002dresidue_002drange_002dto_002dpdb_002dfile" accesskey="3">write-residue-range-to-pdb-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#quick_002dsave" accesskey="4">quick-save</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="write_002dpdb_002dfile"></a>
<div class="header">
<p>
Next: <a href="#write_002dcif_002dfile" accesskey="n" rel="next">write-cif-file</a>, Up: <a href="#PDB-Functions" accesskey="u" rel="up">PDB Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002dpdb_002dfile-1"></a>
<h4 class="subsection">11.18.1 write-pdb-file</h4>
<dl>
<dt><a name="index-write_002dpdb_002dfile"></a>function: <strong>write-pdb-file</strong> <em>imol file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>file_name</em> is a string
</li></ul>

<p>write molecule number imol as a PDB to file file_name 
</p>
</dd></dl>

<hr>
<a name="write_002dcif_002dfile"></a>
<div class="header">
<p>
Next: <a href="#write_002dresidue_002drange_002dto_002dpdb_002dfile" accesskey="n" rel="next">write-residue-range-to-pdb-file</a>, Previous: <a href="#write_002dpdb_002dfile" accesskey="p" rel="previous">write-pdb-file</a>, Up: <a href="#PDB-Functions" accesskey="u" rel="up">PDB Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002dcif_002dfile-1"></a>
<h4 class="subsection">11.18.2 write-cif-file</h4>
<dl>
<dt><a name="index-write_002dcif_002dfile"></a>function: <strong>write-cif-file</strong> <em>imol file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>file_name</em> is a string
</li></ul>

<p>write molecule number imol as a mmCIF to file file_name 
</p>
</dd></dl>

<hr>
<a name="write_002dresidue_002drange_002dto_002dpdb_002dfile"></a>
<div class="header">
<p>
Next: <a href="#quick_002dsave" accesskey="n" rel="next">quick-save</a>, Previous: <a href="#write_002dcif_002dfile" accesskey="p" rel="previous">write-cif-file</a>, Up: <a href="#PDB-Functions" accesskey="u" rel="up">PDB Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002dresidue_002drange_002dto_002dpdb_002dfile-1"></a>
<h4 class="subsection">11.18.3 write-residue-range-to-pdb-file</h4>
<dl>
<dt><a name="index-write_002dresidue_002drange_002dto_002dpdb_002dfile"></a>function: <strong>write-residue-range-to-pdb-file</strong> <em>imol chainid resno_start resno_end filename</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chainid</em> is a string
</li><li> <em>resno_start</em> is an integer number
</li><li> <em>resno_end</em> is an integer number
</li><li> <em>filename</em> is a string
</li></ul>

<p>write molecule number imol&rsquo;s residue range as a PDB to file file_name 
</p>
</dd></dl>

<hr>
<a name="quick_002dsave"></a>
<div class="header">
<p>
Previous: <a href="#write_002dresidue_002drange_002dto_002dpdb_002dfile" accesskey="p" rel="previous">write-residue-range-to-pdb-file</a>, Up: <a href="#PDB-Functions" accesskey="u" rel="up">PDB Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="quick_002dsave-1"></a>
<h4 class="subsection">11.18.4 quick-save</h4>
<dl>
<dt><a name="index-quick_002dsave"></a>function: <strong>quick-save</strong></dt>
<dd>
<p>save all modified coordinates molecules to the default names and save the state too. 
</p>
</dd></dl>

<hr>
<a name="Info-Dialog"></a>
<div class="header">
<p>
Next: <a href="#Refmac-Functions" accesskey="n" rel="next">Refmac Functions</a>, Previous: <a href="#PDB-Functions" accesskey="p" rel="previous">PDB Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Info-Dialog-1"></a>
<h3 class="section">11.19 Info Dialog</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#info_002ddialog" accesskey="1">info-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#info_002ddialog_002dand_002dtext" accesskey="2">info-dialog-and-text</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#info_002ddialog_002dwith_002dmarkup" accesskey="3">info-dialog-with-markup</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="info_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#info_002ddialog_002dand_002dtext" accesskey="n" rel="next">info-dialog-and-text</a>, Up: <a href="#Info-Dialog" accesskey="u" rel="up">Info Dialog</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="info_002ddialog-1"></a>
<h4 class="subsection">11.19.1 info-dialog</h4>
<dl>
<dt><a name="index-info_002ddialog"></a>function: <strong>info-dialog</strong> <em>txt</em></dt>
<dd>
<p>Where <em>txt</em> is a string
</p>
<p>create a dialog with information 
</p>
<p>create a dialog with information string txt. User has to click to dismiss it, but it is not modal (nothing in coot is modal). 
</p>
</dd></dl>

<hr>
<a name="info_002ddialog_002dand_002dtext"></a>
<div class="header">
<p>
Next: <a href="#info_002ddialog_002dwith_002dmarkup" accesskey="n" rel="next">info-dialog-with-markup</a>, Previous: <a href="#info_002ddialog" accesskey="p" rel="previous">info-dialog</a>, Up: <a href="#Info-Dialog" accesskey="u" rel="up">Info Dialog</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="info_002ddialog_002dand_002dtext-1"></a>
<h4 class="subsection">11.19.2 info-dialog-and-text</h4>
<dl>
<dt><a name="index-info_002ddialog_002dand_002dtext"></a>function: <strong>info-dialog-and-text</strong> <em>txt</em></dt>
<dd>
<p>Where <em>txt</em> is a string
</p>
<p>create a dialog with information and print to console 
</p>
<p>as info_dialog but print to console as well. 
</p>
</dd></dl>

<hr>
<a name="info_002ddialog_002dwith_002dmarkup"></a>
<div class="header">
<p>
Previous: <a href="#info_002ddialog_002dand_002dtext" accesskey="p" rel="previous">info-dialog-and-text</a>, Up: <a href="#Info-Dialog" accesskey="u" rel="up">Info Dialog</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="info_002ddialog_002dwith_002dmarkup-1"></a>
<h4 class="subsection">11.19.3 info-dialog-with-markup</h4>
<dl>
<dt><a name="index-info_002ddialog_002dwith_002dmarkup"></a>function: <strong>info-dialog-with-markup</strong> <em>txt</em></dt>
<dd>
<p>Where <em>txt</em> is a string
</p>
<p>as above, create a dialog with information 
</p>
<p>This dialog is left-justified and can use markup such as angled bracketted tt or i 
</p>
</dd></dl>

<hr>
<a name="Refmac-Functions"></a>
<div class="header">
<p>
Next: <a href="#Symmetry-Functions" accesskey="n" rel="next">Symmetry Functions</a>, Previous: <a href="#Info-Dialog" accesskey="p" rel="previous">Info Dialog</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Refmac-Functions-1"></a>
<h3 class="section">11.20 Refmac Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002drefmac_002dcounter" accesskey="1">set-refmac-counter</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#swap_002dmap_002dcolours" accesskey="2">swap-map-colours</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dkeep_002dmap_002dcolour_002dafter_002drefmac" accesskey="3">set-keep-map-colour-after-refmac</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#keep_002dmap_002dcolour_002dafter_002drefmac_002dstate" accesskey="4">keep-map-colour-after-refmac-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002drefmac_002dcounter"></a>
<div class="header">
<p>
Next: <a href="#swap_002dmap_002dcolours" accesskey="n" rel="next">swap-map-colours</a>, Up: <a href="#Refmac-Functions" accesskey="u" rel="up">Refmac Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefmac_002dcounter-1"></a>
<h4 class="subsection">11.20.1 set-refmac-counter</h4>
<dl>
<dt><a name="index-set_002drefmac_002dcounter"></a>function: <strong>set-refmac-counter</strong> <em>imol refmac_count</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>refmac_count</em> is an integer number
</li></ul>

<p>set counter for runs of refmac so that this can be used to construct a unique filename for new output 
</p>
</dd></dl>

<hr>
<a name="swap_002dmap_002dcolours"></a>
<div class="header">
<p>
Next: <a href="#set_002dkeep_002dmap_002dcolour_002dafter_002drefmac" accesskey="n" rel="next">set-keep-map-colour-after-refmac</a>, Previous: <a href="#set_002drefmac_002dcounter" accesskey="p" rel="previous">set-refmac-counter</a>, Up: <a href="#Refmac-Functions" accesskey="u" rel="up">Refmac Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="swap_002dmap_002dcolours-1"></a>
<h4 class="subsection">11.20.2 swap-map-colours</h4>
<dl>
<dt><a name="index-swap_002dmap_002dcolours"></a>function: <strong>swap-map-colours</strong> <em>imol1 imol2</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol1</em> is an integer number
</li><li> <em>imol2</em> is an integer number
</li></ul>

<p>swap the colours of maps 
</p>
<p>swap the colour of maps imol1 and imol2. Useful to some after running refmac, so that the map to be build into is always the same colour 
</p>
</dd></dl>

<hr>
<a name="set_002dkeep_002dmap_002dcolour_002dafter_002drefmac"></a>
<div class="header">
<p>
Next: <a href="#keep_002dmap_002dcolour_002dafter_002drefmac_002dstate" accesskey="n" rel="next">keep-map-colour-after-refmac-state</a>, Previous: <a href="#swap_002dmap_002dcolours" accesskey="p" rel="previous">swap-map-colours</a>, Up: <a href="#Refmac-Functions" accesskey="u" rel="up">Refmac Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dkeep_002dmap_002dcolour_002dafter_002drefmac-1"></a>
<h4 class="subsection">11.20.3 set-keep-map-colour-after-refmac</h4>
<dl>
<dt><a name="index-set_002dkeep_002dmap_002dcolour_002dafter_002drefmac"></a>function: <strong>set-keep-map-colour-after-refmac</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>flag to enable above 
</p>
<p>call this with istate=1 
</p>
</dd></dl>

<hr>
<a name="keep_002dmap_002dcolour_002dafter_002drefmac_002dstate"></a>
<div class="header">
<p>
Previous: <a href="#set_002dkeep_002dmap_002dcolour_002dafter_002drefmac" accesskey="p" rel="previous">set-keep-map-colour-after-refmac</a>, Up: <a href="#Refmac-Functions" accesskey="u" rel="up">Refmac Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="keep_002dmap_002dcolour_002dafter_002drefmac_002dstate-1"></a>
<h4 class="subsection">11.20.4 keep-map-colour-after-refmac-state</h4>
<dl>
<dt><a name="index-keep_002dmap_002dcolour_002dafter_002drefmac_002dstate"></a>function: <strong>keep-map-colour-after-refmac-state</strong></dt>
<dd>
<p>the keep-map-colour-after-refmac internal state 
</p>
<p>Returns: 1 for &quot;yes&quot;, 0 for &quot;no&quot; 
</p>
</dd></dl>

<hr>
<a name="Symmetry-Functions"></a>
<div class="header">
<p>
Next: <a href="#History-Functions" accesskey="n" rel="next">History Functions</a>, Previous: <a href="#Refmac-Functions" accesskey="p" rel="previous">Refmac Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Symmetry-Functions-1"></a>
<h3 class="section">11.21 Symmetry Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dsymmetry_002dsize" accesskey="1">set-symmetry-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dshow_002dsymmetry" accesskey="2">get-show-symmetry</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dsymmetry_002dmaster" accesskey="3">set-show-symmetry-master</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dsymmetry_002dmolecule" accesskey="4">set-show-symmetry-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#symmetry_002das_002dcalphas" accesskey="5">symmetry-as-calphas</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dsymmetry_002das_002dcalphas_002dstate" accesskey="6">get-symmetry-as-calphas-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap" accesskey="7">set-symmetry-molecule-rotate-colour-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate" accesskey="8">symmetry-molecule-rotate-colour-map-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsymmetry_002dcolour_002dby_002dsymop" accesskey="9">set-symmetry-colour-by-symop</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsymmetry_002dwhole_002dchain">set-symmetry-whole-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsymmetry_002datom_002dlabels_002dexpanded">set-symmetry-atom-labels-expanded</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#has_002dunit_002dcell_002dstate">has-unit-cell-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#undo_002dsymmetry_002dview">undo-symmetry-view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#first_002dmolecule_002dwith_002dsymmetry_002ddisplayed">first-molecule-with-symmetry-displayed</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#save_002dsymmetry_002dcoords">save-symmetry-coords</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#new_002dmolecule_002dby_002dsymmetry">new-molecule-by-symmetry</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection">new-molecule-by-symmetry-with-atom-selection</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#new_002dmolecule_002dby_002dsymop">new-molecule-by-symop</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#n_002dsymops">n-symops</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#origin_002dpre_002dshift_002dscm">origin-pre-shift-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#origin_002dpre_002dshift_002dpy">origin-pre-shift-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dspace_002dgroup">set-space-group</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dunit_002dcell_002dand_002dspace_002dgroup">set-unit-cell-and-space-group</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule">set-unit-cell-and-space-group-using-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsymmetry_002dshift_002dsearch_002dsize">set-symmetry-shift-search-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dsymmetry_002dsize"></a>
<div class="header">
<p>
Next: <a href="#get_002dshow_002dsymmetry" accesskey="n" rel="next">get-show-symmetry</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsymmetry_002dsize-1"></a>
<h4 class="subsection">11.21.1 set-symmetry-size</h4>
<dl>
<dt><a name="index-set_002dsymmetry_002dsize"></a>function: <strong>set-symmetry-size</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the size of the displayed symmetry 
</p>
</dd></dl>

<hr>
<a name="get_002dshow_002dsymmetry"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002dsymmetry_002dmaster" accesskey="n" rel="next">set-show-symmetry-master</a>, Previous: <a href="#set_002dsymmetry_002dsize" accesskey="p" rel="previous">set-symmetry-size</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dshow_002dsymmetry-1"></a>
<h4 class="subsection">11.21.2 get-show-symmetry</h4>
<dl>
<dt><a name="index-get_002dshow_002dsymmetry"></a>function: <strong>get-show-symmetry</strong></dt>
<dd>
<p>is symmetry master display control on? 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002dsymmetry_002dmaster"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002dsymmetry_002dmolecule" accesskey="n" rel="next">set-show-symmetry-molecule</a>, Previous: <a href="#get_002dshow_002dsymmetry" accesskey="p" rel="previous">get-show-symmetry</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dsymmetry_002dmaster-1"></a>
<h4 class="subsection">11.21.3 set-show-symmetry-master</h4>
<dl>
<dt><a name="index-set_002dshow_002dsymmetry_002dmaster"></a>function: <strong>set-show-symmetry-master</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set display symmetry, master controller 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002dsymmetry_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#symmetry_002das_002dcalphas" accesskey="n" rel="next">symmetry-as-calphas</a>, Previous: <a href="#set_002dshow_002dsymmetry_002dmaster" accesskey="p" rel="previous">set-show-symmetry-master</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dsymmetry_002dmolecule-1"></a>
<h4 class="subsection">11.21.4 set-show-symmetry-molecule</h4>
<dl>
<dt><a name="index-set_002dshow_002dsymmetry_002dmolecule"></a>function: <strong>set-show-symmetry-molecule</strong> <em>mol_no state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mol_no</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>set display symmetry for molecule number mol_no 
</p>
<p>pass with state=0 for off, state=1 for on 
</p>
</dd></dl>

<hr>
<a name="symmetry_002das_002dcalphas"></a>
<div class="header">
<p>
Next: <a href="#get_002dsymmetry_002das_002dcalphas_002dstate" accesskey="n" rel="next">get-symmetry-as-calphas-state</a>, Previous: <a href="#set_002dshow_002dsymmetry_002dmolecule" accesskey="p" rel="previous">set-show-symmetry-molecule</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="symmetry_002das_002dcalphas-1"></a>
<h4 class="subsection">11.21.5 symmetry-as-calphas</h4>
<dl>
<dt><a name="index-symmetry_002das_002dcalphas"></a>function: <strong>symmetry-as-calphas</strong> <em>mol_no state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mol_no</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>display symmetry as CAs? 
</p>
<p>pass with state=0 for off, state=1 for on 
</p>
</dd></dl>

<hr>
<a name="get_002dsymmetry_002das_002dcalphas_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap" accesskey="n" rel="next">set-symmetry-molecule-rotate-colour-map</a>, Previous: <a href="#symmetry_002das_002dcalphas" accesskey="p" rel="previous">symmetry-as-calphas</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dsymmetry_002das_002dcalphas_002dstate-1"></a>
<h4 class="subsection">11.21.6 get-symmetry-as-calphas-state</h4>
<dl>
<dt><a name="index-get_002dsymmetry_002das_002dcalphas_002dstate"></a>function: <strong>get-symmetry-as-calphas-state</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>what is state of display CAs for molecule number mol_no? 
</p>
<p>return state=0 for off, state=1 for on 
</p>
</dd></dl>

<hr>
<a name="set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap"></a>
<div class="header">
<p>
Next: <a href="#symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate" accesskey="n" rel="next">symmetry-molecule-rotate-colour-map-state</a>, Previous: <a href="#get_002dsymmetry_002das_002dcalphas_002dstate" accesskey="p" rel="previous">get-symmetry-as-calphas-state</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap-1"></a>
<h4 class="subsection">11.21.7 set-symmetry-molecule-rotate-colour-map</h4>
<dl>
<dt><a name="index-set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap"></a>function: <strong>set-symmetry-molecule-rotate-colour-map</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>set the colour map rotation (i.e. the hue) for the symmetry atoms of molecule number imol 
</p>
</dd></dl>

<hr>
<a name="symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dsymmetry_002dcolour_002dby_002dsymop" accesskey="n" rel="next">set-symmetry-colour-by-symop</a>, Previous: <a href="#set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap" accesskey="p" rel="previous">set-symmetry-molecule-rotate-colour-map</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate-1"></a>
<h4 class="subsection">11.21.8 symmetry-molecule-rotate-colour-map-state</h4>
<dl>
<dt><a name="index-symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate"></a>function: <strong>symmetry-molecule-rotate-colour-map-state</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>should there be colour map rotation (i.e. the hue) change for the symmetry atoms of molecule number imol? 
</p>
<p>return state=0 for off, state=1 for on 
</p>
</dd></dl>

<hr>
<a name="set_002dsymmetry_002dcolour_002dby_002dsymop"></a>
<div class="header">
<p>
Next: <a href="#set_002dsymmetry_002dwhole_002dchain" accesskey="n" rel="next">set-symmetry-whole-chain</a>, Previous: <a href="#symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate" accesskey="p" rel="previous">symmetry-molecule-rotate-colour-map-state</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsymmetry_002dcolour_002dby_002dsymop-1"></a>
<h4 class="subsection">11.21.9 set-symmetry-colour-by-symop</h4>
<dl>
<dt><a name="index-set_002dsymmetry_002dcolour_002dby_002dsymop"></a>function: <strong>set-symmetry-colour-by-symop</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>set symmetry colour by symop mode 
</p>
</dd></dl>

<hr>
<a name="set_002dsymmetry_002dwhole_002dchain"></a>
<div class="header">
<p>
Next: <a href="#set_002dsymmetry_002datom_002dlabels_002dexpanded" accesskey="n" rel="next">set-symmetry-atom-labels-expanded</a>, Previous: <a href="#set_002dsymmetry_002dcolour_002dby_002dsymop" accesskey="p" rel="previous">set-symmetry-colour-by-symop</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsymmetry_002dwhole_002dchain-1"></a>
<h4 class="subsection">11.21.10 set-symmetry-whole-chain</h4>
<dl>
<dt><a name="index-set_002dsymmetry_002dwhole_002dchain"></a>function: <strong>set-symmetry-whole-chain</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>set symmetry colour for the chain 
</p>
</dd></dl>

<hr>
<a name="set_002dsymmetry_002datom_002dlabels_002dexpanded"></a>
<div class="header">
<p>
Next: <a href="#has_002dunit_002dcell_002dstate" accesskey="n" rel="next">has-unit-cell-state</a>, Previous: <a href="#set_002dsymmetry_002dwhole_002dchain" accesskey="p" rel="previous">set-symmetry-whole-chain</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsymmetry_002datom_002dlabels_002dexpanded-1"></a>
<h4 class="subsection">11.21.11 set-symmetry-atom-labels-expanded</h4>
<dl>
<dt><a name="index-set_002dsymmetry_002datom_002dlabels_002dexpanded"></a>function: <strong>set-symmetry-atom-labels-expanded</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set use expanded symmetry atom labels 
</p>
</dd></dl>

<hr>
<a name="has_002dunit_002dcell_002dstate"></a>
<div class="header">
<p>
Next: <a href="#undo_002dsymmetry_002dview" accesskey="n" rel="next">undo-symmetry-view</a>, Previous: <a href="#set_002dsymmetry_002datom_002dlabels_002dexpanded" accesskey="p" rel="previous">set-symmetry-atom-labels-expanded</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="has_002dunit_002dcell_002dstate-1"></a>
<h4 class="subsection">11.21.12 has-unit-cell-state</h4>
<dl>
<dt><a name="index-has_002dunit_002dcell_002dstate"></a>function: <strong>has-unit-cell-state</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>molecule number imol has a unit cell? 
</p>
<p>Returns: 1 on &quot;yes, it has a cell&quot;, 0 for &quot;no&quot; 
</p>
</dd></dl>

<hr>
<a name="undo_002dsymmetry_002dview"></a>
<div class="header">
<p>
Next: <a href="#first_002dmolecule_002dwith_002dsymmetry_002ddisplayed" accesskey="n" rel="next">first-molecule-with-symmetry-displayed</a>, Previous: <a href="#has_002dunit_002dcell_002dstate" accesskey="p" rel="previous">has-unit-cell-state</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="undo_002dsymmetry_002dview-1"></a>
<h4 class="subsection">11.21.13 undo-symmetry-view</h4>
<dl>
<dt><a name="index-undo_002dsymmetry_002dview"></a>function: <strong>undo-symmetry-view</strong></dt>
<dd>
<p>Undo symmetry view. Translate back to main molecule from this symmetry position. 
</p>
</dd></dl>

<hr>
<a name="first_002dmolecule_002dwith_002dsymmetry_002ddisplayed"></a>
<div class="header">
<p>
Next: <a href="#save_002dsymmetry_002dcoords" accesskey="n" rel="next">save-symmetry-coords</a>, Previous: <a href="#undo_002dsymmetry_002dview" accesskey="p" rel="previous">undo-symmetry-view</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="first_002dmolecule_002dwith_002dsymmetry_002ddisplayed-1"></a>
<h4 class="subsection">11.21.14 first-molecule-with-symmetry-displayed</h4>
<dl>
<dt><a name="index-first_002dmolecule_002dwith_002dsymmetry_002ddisplayed"></a>function: <strong>first-molecule-with-symmetry-displayed</strong></dt>
<dd>
<p>return the molecule number. 
</p>
<p>Returns: -1 if there is no molecule with symmetry displayed. 
</p>
</dd></dl>

<hr>
<a name="save_002dsymmetry_002dcoords"></a>
<div class="header">
<p>
Next: <a href="#new_002dmolecule_002dby_002dsymmetry" accesskey="n" rel="next">new-molecule-by-symmetry</a>, Previous: <a href="#first_002dmolecule_002dwith_002dsymmetry_002ddisplayed" accesskey="p" rel="previous">first-molecule-with-symmetry-displayed</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dsymmetry_002dcoords-1"></a>
<h4 class="subsection">11.21.15 save-symmetry-coords</h4>
<dl>
<dt><a name="index-save_002dsymmetry_002dcoords"></a>function: <strong>save-symmetry-coords</strong> <em>imol filename symop_no shift_a shift_b shift_c pre_shift_to_origin_na pre_shift_to_origin_nb pre_shift_to_origin_nc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>filename</em> is a string
</li><li> <em>symop_no</em> is an integer number
</li><li> <em>shift_a</em> is an integer number
</li><li> <em>shift_b</em> is an integer number
</li><li> <em>shift_c</em> is an integer number
</li><li> <em>pre_shift_to_origin_na</em> is an integer number
</li><li> <em>pre_shift_to_origin_nb</em> is an integer number
</li><li> <em>pre_shift_to_origin_nc</em> is an integer number
</li></ul>

<p>save the symmetry coordinates of molecule number imol to filename 
</p>
<p>Allow a shift of the coordinates to the origin before symmetry expansion is apllied (this is how symmetry works in Coot internals). 
</p>
</dd></dl>

<hr>
<a name="new_002dmolecule_002dby_002dsymmetry"></a>
<div class="header">
<p>
Next: <a href="#new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection" accesskey="n" rel="next">new-molecule-by-symmetry-with-atom-selection</a>, Previous: <a href="#save_002dsymmetry_002dcoords" accesskey="p" rel="previous">save-symmetry-coords</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="new_002dmolecule_002dby_002dsymmetry-1"></a>
<h4 class="subsection">11.21.16 new-molecule-by-symmetry</h4>
<dl>
<dt><a name="index-new_002dmolecule_002dby_002dsymmetry"></a>function: <strong>new-molecule-by-symmetry</strong> <em>imol name m11 m12 m13 m21 m22 m23 m31 m32 m33 tx ty tz pre_shift_to_origin_na pre_shift_to_origin_nb pre_shift_to_origin_nc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>name</em> is a string
</li><li> <em>m11</em> is a number
</li><li> <em>m12</em> is a number
</li><li> <em>m13</em> is a number
</li><li> <em>m21</em> is a number
</li><li> <em>m22</em> is a number
</li><li> <em>m23</em> is a number
</li><li> <em>m31</em> is a number
</li><li> <em>m32</em> is a number
</li><li> <em>m33</em> is a number
</li><li> <em>tx</em> is a number
</li><li> <em>ty</em> is a number
</li><li> <em>tz</em> is a number
</li><li> <em>pre_shift_to_origin_na</em> is an integer number
</li><li> <em>pre_shift_to_origin_nb</em> is an integer number
</li><li> <em>pre_shift_to_origin_nc</em> is an integer number
</li></ul>

<p>create a new molecule (molecule number is the return value) from imol. 
</p>
<p>The rotation/translation matrix components are given in 
</p>
<p>coordinates.
</p>
<p>Allow a shift of the coordinates to the origin before symmetry expansion is aplied.
</p>
<p>Pass &quot;&quot; as the name-in and a name will be constructed for you.
</p>
<p>Return -1 on failure. 
</p>
</dd></dl>

<hr>
<a name="new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection"></a>
<div class="header">
<p>
Next: <a href="#new_002dmolecule_002dby_002dsymop" accesskey="n" rel="next">new-molecule-by-symop</a>, Previous: <a href="#new_002dmolecule_002dby_002dsymmetry" accesskey="p" rel="previous">new-molecule-by-symmetry</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection-1"></a>
<h4 class="subsection">11.21.17 new-molecule-by-symmetry-with-atom-selection</h4>
<dl>
<dt><a name="index-new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection"></a>function: <strong>new-molecule-by-symmetry-with-atom-selection</strong> <em>imol name mmdb_atom_selection_string m11 m12 m13 m21 m22 m23 m31 m32 m33 tx ty tz pre_shift_to_origin_na pre_shift_to_origin_nb pre_shift_to_origin_nc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>name</em> is a string
</li><li> <em>mmdb_atom_selection_string</em> is a string
</li><li> <em>m11</em> is a number
</li><li> <em>m12</em> is a number
</li><li> <em>m13</em> is a number
</li><li> <em>m21</em> is a number
</li><li> <em>m22</em> is a number
</li><li> <em>m23</em> is a number
</li><li> <em>m31</em> is a number
</li><li> <em>m32</em> is a number
</li><li> <em>m33</em> is a number
</li><li> <em>tx</em> is a number
</li><li> <em>ty</em> is a number
</li><li> <em>tz</em> is a number
</li><li> <em>pre_shift_to_origin_na</em> is an integer number
</li><li> <em>pre_shift_to_origin_nb</em> is an integer number
</li><li> <em>pre_shift_to_origin_nc</em> is an integer number
</li></ul>

<p>create a new molecule (molecule number is the return value) from imol, but only for atom that match the mmdb_atom_selection_string. 
</p>
<p>The rotation/translation matrix components are given in 
</p>
<p>coordinates.
</p>
<p>Allow a shift of the coordinates to the origin before symmetry expansion is aplied.
</p>
<p>Pass &quot;&quot; as the name-in and a name will be constructed for you.
</p>
<p>Return -1 on failure. 
</p>
</dd></dl>

<hr>
<a name="new_002dmolecule_002dby_002dsymop"></a>
<div class="header">
<p>
Next: <a href="#n_002dsymops" accesskey="n" rel="next">n-symops</a>, Previous: <a href="#new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection" accesskey="p" rel="previous">new-molecule-by-symmetry-with-atom-selection</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="new_002dmolecule_002dby_002dsymop-1"></a>
<h4 class="subsection">11.21.18 new-molecule-by-symop</h4>
<dl>
<dt><a name="index-new_002dmolecule_002dby_002dsymop"></a>function: <strong>new-molecule-by-symop</strong> <em>imol symop_string pre_shift_to_origin_na pre_shift_to_origin_nb pre_shift_to_origin_nc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>symop_string</em> is a string
</li><li> <em>pre_shift_to_origin_na</em> is an integer number
</li><li> <em>pre_shift_to_origin_nb</em> is an integer number
</li><li> <em>pre_shift_to_origin_nc</em> is an integer number
</li></ul>

<p>create a new molecule (molecule number is the return value) from imol. 
</p>
</dd></dl>

<hr>
<a name="n_002dsymops"></a>
<div class="header">
<p>
Next: <a href="#origin_002dpre_002dshift_002dscm" accesskey="n" rel="next">origin-pre-shift-scm</a>, Previous: <a href="#new_002dmolecule_002dby_002dsymop" accesskey="p" rel="previous">new-molecule-by-symop</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="n_002dsymops-1"></a>
<h4 class="subsection">11.21.19 n-symops</h4>
<dl>
<dt><a name="index-n_002dsymops"></a>function: <strong>n-symops</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the number of symmetry operators for the given molecule 
</p>
<p>return -1 on no-symmetry for molecule or inappropriate imol number 
</p>
</dd></dl>

<hr>
<a name="origin_002dpre_002dshift_002dscm"></a>
<div class="header">
<p>
Next: <a href="#origin_002dpre_002dshift_002dpy" accesskey="n" rel="next">origin-pre-shift-py</a>, Previous: <a href="#n_002dsymops" accesskey="p" rel="previous">n-symops</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="origin_002dpre_002dshift_002dscm-1"></a>
<h4 class="subsection">11.21.20 origin-pre-shift-scm</h4>
<dl>
<dt><a name="index-origin_002dpre_002dshift_002dscm"></a>function: <strong>origin-pre-shift-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the pre-shift (the shift that translates the centre of the molecule as close as possible to the origin) as a list of ints or scheme false on failure 
</p>
</dd></dl>

<hr>
<a name="origin_002dpre_002dshift_002dpy"></a>
<div class="header">
<p>
Next: <a href="#set_002dspace_002dgroup" accesskey="n" rel="next">set-space-group</a>, Previous: <a href="#origin_002dpre_002dshift_002dscm" accesskey="p" rel="previous">origin-pre-shift-scm</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="origin_002dpre_002dshift_002dpy-1"></a>
<h4 class="subsection">11.21.21 origin-pre-shift-py</h4>
<dl>
<dt><a name="index-origin_002dpre_002dshift_002dpy"></a>function: <strong>origin-pre-shift-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the pre-shift (the shift that translates the centre of the molecule as close as possible to the origin) as a list of ints or Python false on failure 
</p>
</dd></dl>

<hr>
<a name="set_002dspace_002dgroup"></a>
<div class="header">
<p>
Next: <a href="#set_002dunit_002dcell_002dand_002dspace_002dgroup" accesskey="n" rel="next">set-unit-cell-and-space-group</a>, Previous: <a href="#origin_002dpre_002dshift_002dpy" accesskey="p" rel="previous">origin-pre-shift-py</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dspace_002dgroup-1"></a>
<h4 class="subsection">11.21.22 set-space-group</h4>
<dl>
<dt><a name="index-set_002dspace_002dgroup"></a>function: <strong>set-space-group</strong> <em>imol spg</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>spg</em> is a string
</li></ul>

<p>set the space group for a coordinates molecule 
</p>
<p>for shelx FA pdb files, there is no space group. So allow the user to set it. This can be initted with a HM symbol or a symm list for clipper.
</p>
<p>This will only work on model molecules.
</p>
<p>Returns: the success status of the setting (1 good, 0 fail). 
</p>
</dd></dl>

<hr>
<a name="set_002dunit_002dcell_002dand_002dspace_002dgroup"></a>
<div class="header">
<p>
Next: <a href="#set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule" accesskey="n" rel="next">set-unit-cell-and-space-group-using-molecule</a>, Previous: <a href="#set_002dspace_002dgroup" accesskey="p" rel="previous">set-space-group</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dunit_002dcell_002dand_002dspace_002dgroup-1"></a>
<h4 class="subsection">11.21.23 set-unit-cell-and-space-group</h4>
<dl>
<dt><a name="index-set_002dunit_002dcell_002dand_002dspace_002dgroup"></a>function: <strong>set-unit-cell-and-space-group</strong> <em>imol a b c alpha beta gamma space_group</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>a</em> is a number
</li><li> <em>b</em> is a number
</li><li> <em>c</em> is a number
</li><li> <em>alpha</em> is a number
</li><li> <em>beta</em> is a number
</li><li> <em>gamma</em> is a number
</li><li> <em>space_group</em> is a string
</li></ul>

<p>set the unit cell for a given model molecule 
</p>
<p>Angles in degress, cell lengths in Angstroms.
</p>
<p>Returns: the success status of the setting (1 good, 0 fail). 
</p>
</dd></dl>

<hr>
<a name="set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#set_002dsymmetry_002dshift_002dsearch_002dsize" accesskey="n" rel="next">set-symmetry-shift-search-size</a>, Previous: <a href="#set_002dunit_002dcell_002dand_002dspace_002dgroup" accesskey="p" rel="previous">set-unit-cell-and-space-group</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule-1"></a>
<h4 class="subsection">11.21.24 set-unit-cell-and-space-group-using-molecule</h4>
<dl>
<dt><a name="index-set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule"></a>function: <strong>set-unit-cell-and-space-group-using-molecule</strong> <em>imol imol_from</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>imol_from</em> is an integer number
</li></ul>

<p>set the unit cell for a given model molecule using the cell of moecule imol_from 
</p>
<p>This will only work on model molecules. 
</p>
<p>Returns: the success status of the setting (1 good, 0 fail). 
</p>
</dd></dl>

<hr>
<a name="set_002dsymmetry_002dshift_002dsearch_002dsize"></a>
<div class="header">
<p>
Previous: <a href="#set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule" accesskey="p" rel="previous">set-unit-cell-and-space-group-using-molecule</a>, Up: <a href="#Symmetry-Functions" accesskey="u" rel="up">Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsymmetry_002dshift_002dsearch_002dsize-1"></a>
<h4 class="subsection">11.21.25 set-symmetry-shift-search-size</h4>
<dl>
<dt><a name="index-set_002dsymmetry_002dshift_002dsearch_002dsize"></a>function: <strong>set-symmetry-shift-search-size</strong> <em>shift</em></dt>
<dd>
<p>Where <em>shift</em> is an integer number
</p>
<p>set the cell shift search size for symmetry searching. 
</p>
<p>When the coordinates for one (or some) symmetry operator are missing (which happens sometimes, but rarely), try changing setting this to 2 (default is 1). It slows symmetry searching, which is why it is not set to 2 by default. 
</p>
</dd></dl>

<hr>
<a name="History-Functions"></a>
<div class="header">
<p>
Next: <a href="#State-Functions" accesskey="n" rel="next">State Functions</a>, Previous: <a href="#Symmetry-Functions" accesskey="p" rel="previous">Symmetry Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="History-Functions-1"></a>
<h3 class="section">11.22 History Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#print_002dall_002dhistory_002din_002dscheme" accesskey="1">print-all-history-in-scheme</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#print_002dall_002dhistory_002din_002dpython" accesskey="2">print-all-history-in-python</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dconsole_002ddisplay_002dcommands_002dstate" accesskey="3">set-console-display-commands-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dconsole_002ddisplay_002dcommands_002dhilights" accesskey="4">set-console-display-commands-hilights</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="print_002dall_002dhistory_002din_002dscheme"></a>
<div class="header">
<p>
Next: <a href="#print_002dall_002dhistory_002din_002dpython" accesskey="n" rel="next">print-all-history-in-python</a>, Up: <a href="#History-Functions" accesskey="u" rel="up">History Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="print_002dall_002dhistory_002din_002dscheme-1"></a>
<h4 class="subsection">11.22.1 print-all-history-in-scheme</h4>
<dl>
<dt><a name="index-print_002dall_002dhistory_002din_002dscheme"></a>function: <strong>print-all-history-in-scheme</strong></dt>
<dd>
<p>print the history in scheme format 
</p>
</dd></dl>

<hr>
<a name="print_002dall_002dhistory_002din_002dpython"></a>
<div class="header">
<p>
Next: <a href="#set_002dconsole_002ddisplay_002dcommands_002dstate" accesskey="n" rel="next">set-console-display-commands-state</a>, Previous: <a href="#print_002dall_002dhistory_002din_002dscheme" accesskey="p" rel="previous">print-all-history-in-scheme</a>, Up: <a href="#History-Functions" accesskey="u" rel="up">History Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="print_002dall_002dhistory_002din_002dpython-1"></a>
<h4 class="subsection">11.22.2 print-all-history-in-python</h4>
<dl>
<dt><a name="index-print_002dall_002dhistory_002din_002dpython"></a>function: <strong>print-all-history-in-python</strong></dt>
<dd>
<p>print the history in python format 
</p>
</dd></dl>

<hr>
<a name="set_002dconsole_002ddisplay_002dcommands_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dconsole_002ddisplay_002dcommands_002dhilights" accesskey="n" rel="next">set-console-display-commands-hilights</a>, Previous: <a href="#print_002dall_002dhistory_002din_002dpython" accesskey="p" rel="previous">print-all-history-in-python</a>, Up: <a href="#History-Functions" accesskey="u" rel="up">History Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dconsole_002ddisplay_002dcommands_002dstate-1"></a>
<h4 class="subsection">11.22.3 set-console-display-commands-state</h4>
<dl>
<dt><a name="index-set_002dconsole_002ddisplay_002dcommands_002dstate"></a>function: <strong>set-console-display-commands-state</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>set a flag to show the text command equivalent of gui commands in the console as they happen. 
</p>
<p>1 for on, 0 for off. 
</p>
</dd></dl>

<hr>
<a name="set_002dconsole_002ddisplay_002dcommands_002dhilights"></a>
<div class="header">
<p>
Previous: <a href="#set_002dconsole_002ddisplay_002dcommands_002dstate" accesskey="p" rel="previous">set-console-display-commands-state</a>, Up: <a href="#History-Functions" accesskey="u" rel="up">History Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dconsole_002ddisplay_002dcommands_002dhilights-1"></a>
<h4 class="subsection">11.22.4 set-console-display-commands-hilights</h4>
<dl>
<dt><a name="index-set_002dconsole_002ddisplay_002dcommands_002dhilights"></a>function: <strong>set-console-display-commands-hilights</strong> <em>bold_flag colour_flag colour_index</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>bold_flag</em> is an integer number
</li><li> <em>colour_flag</em> is an integer number
</li><li> <em>colour_index</em> is an integer number
</li></ul>

<p>set a flag to show the text command equivalent of gui commands in the console as they happen in bold and colours. 
</p>
<p>colour_flag: pass 1 for on, 0 for off.
</p>
<p>colour_index 0 to 7 inclusive for various different colourings. 
</p>
</dd></dl>

<hr>
<a name="State-Functions"></a>
<div class="header">
<p>
Next: <a href="#The-Virtual-Trackball" accesskey="n" rel="next">The Virtual Trackball</a>, Previous: <a href="#History-Functions" accesskey="p" rel="previous">History Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="State-Functions-1"></a>
<h3 class="section">11.23 State Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#save_002dstate" accesskey="1">save-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#save_002dstate_002dfile" accesskey="2">save-state-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#save_002dstate_002dfile_002dpy" accesskey="3">save-state-file-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsave_002dstate_002dfile_002dname" accesskey="4">set-save-state-file-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#save_002dstate_002dfile_002dname_002dscm" accesskey="5">save-state-file-name-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#save_002dstate_002dfile_002dname_002dpy" accesskey="6">save-state-file-name-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drun_002dstate_002dfile_002dstatus" accesskey="7">set-run-state-file-status</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#run_002dstate_002dfile" accesskey="8">run-state-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#run_002dstate_002dfile_002dmaybe" accesskey="9">run-state-file-maybe</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="save_002dstate"></a>
<div class="header">
<p>
Next: <a href="#save_002dstate_002dfile" accesskey="n" rel="next">save-state-file</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dstate-1"></a>
<h4 class="subsection">11.23.1 save-state</h4>
<dl>
<dt><a name="index-save_002dstate"></a>function: <strong>save-state</strong></dt>
<dd>
<p>save the current state to the default filename 
</p>
</dd></dl>

<hr>
<a name="save_002dstate_002dfile"></a>
<div class="header">
<p>
Next: <a href="#save_002dstate_002dfile_002dpy" accesskey="n" rel="next">save-state-file-py</a>, Previous: <a href="#save_002dstate" accesskey="p" rel="previous">save-state</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dstate_002dfile-1"></a>
<h4 class="subsection">11.23.2 save-state-file</h4>
<dl>
<dt><a name="index-save_002dstate_002dfile"></a>function: <strong>save-state-file</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>save the current state to file filename 
</p>
</dd></dl>

<hr>
<a name="save_002dstate_002dfile_002dpy"></a>
<div class="header">
<p>
Next: <a href="#set_002dsave_002dstate_002dfile_002dname" accesskey="n" rel="next">set-save-state-file-name</a>, Previous: <a href="#save_002dstate_002dfile" accesskey="p" rel="previous">save-state-file</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dstate_002dfile_002dpy-1"></a>
<h4 class="subsection">11.23.3 save-state-file-py</h4>
<dl>
<dt><a name="index-save_002dstate_002dfile_002dpy"></a>function: <strong>save-state-file-py</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>save the current state to file filename 
</p>
</dd></dl>

<hr>
<a name="set_002dsave_002dstate_002dfile_002dname"></a>
<div class="header">
<p>
Next: <a href="#save_002dstate_002dfile_002dname_002dscm" accesskey="n" rel="next">save-state-file-name-scm</a>, Previous: <a href="#save_002dstate_002dfile_002dpy" accesskey="p" rel="previous">save-state-file-py</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsave_002dstate_002dfile_002dname-1"></a>
<h4 class="subsection">11.23.4 set-save-state-file-name</h4>
<dl>
<dt><a name="index-set_002dsave_002dstate_002dfile_002dname"></a>function: <strong>set-save-state-file-name</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>set the default state file name (default 0-coot.state.scm) 
</p>
</dd></dl>

<hr>
<a name="save_002dstate_002dfile_002dname_002dscm"></a>
<div class="header">
<p>
Next: <a href="#save_002dstate_002dfile_002dname_002dpy" accesskey="n" rel="next">save-state-file-name-py</a>, Previous: <a href="#set_002dsave_002dstate_002dfile_002dname" accesskey="p" rel="previous">set-save-state-file-name</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dstate_002dfile_002dname_002dscm-1"></a>
<h4 class="subsection">11.23.5 save-state-file-name-scm</h4>
<dl>
<dt><a name="index-save_002dstate_002dfile_002dname_002dscm"></a>function: <strong>save-state-file-name-scm</strong></dt>
<dd>
<p>the save state file name 
</p>
<p>Returns: the save state file name 
</p>
</dd></dl>

<hr>
<a name="save_002dstate_002dfile_002dname_002dpy"></a>
<div class="header">
<p>
Next: <a href="#set_002drun_002dstate_002dfile_002dstatus" accesskey="n" rel="next">set-run-state-file-status</a>, Previous: <a href="#save_002dstate_002dfile_002dname_002dscm" accesskey="p" rel="previous">save-state-file-name-scm</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dstate_002dfile_002dname_002dpy-1"></a>
<h4 class="subsection">11.23.6 save-state-file-name-py</h4>
<dl>
<dt><a name="index-save_002dstate_002dfile_002dname_002dpy"></a>function: <strong>save-state-file-name-py</strong></dt>
<dd>
<p>the save state file name 
</p>
<p>Returns: the save state file name 
</p>
</dd></dl>

<hr>
<a name="set_002drun_002dstate_002dfile_002dstatus"></a>
<div class="header">
<p>
Next: <a href="#run_002dstate_002dfile" accesskey="n" rel="next">run-state-file</a>, Previous: <a href="#save_002dstate_002dfile_002dname_002dpy" accesskey="p" rel="previous">save-state-file-name-py</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drun_002dstate_002dfile_002dstatus-1"></a>
<h4 class="subsection">11.23.7 set-run-state-file-status</h4>
<dl>
<dt><a name="index-set_002drun_002dstate_002dfile_002dstatus"></a>function: <strong>set-run-state-file-status</strong> <em>istat</em></dt>
<dd>
<p>Where <em>istat</em> is an integer number
</p>
<p>set run state file status 
</p>
<p>0: never run it 1: ask to run it 2: run it, no questions 
</p>
</dd></dl>

<hr>
<a name="run_002dstate_002dfile"></a>
<div class="header">
<p>
Next: <a href="#run_002dstate_002dfile_002dmaybe" accesskey="n" rel="next">run-state-file-maybe</a>, Previous: <a href="#set_002drun_002dstate_002dfile_002dstatus" accesskey="p" rel="previous">set-run-state-file-status</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="run_002dstate_002dfile-1"></a>
<h4 class="subsection">11.23.8 run-state-file</h4>
<dl>
<dt><a name="index-run_002dstate_002dfile"></a>function: <strong>run-state-file</strong></dt>
<dd>
<p>run the state file (reading from default filenname) 
</p>
</dd></dl>

<hr>
<a name="run_002dstate_002dfile_002dmaybe"></a>
<div class="header">
<p>
Previous: <a href="#run_002dstate_002dfile" accesskey="p" rel="previous">run-state-file</a>, Up: <a href="#State-Functions" accesskey="u" rel="up">State Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="run_002dstate_002dfile_002dmaybe-1"></a>
<h4 class="subsection">11.23.9 run-state-file-maybe</h4>
<dl>
<dt><a name="index-run_002dstate_002dfile_002dmaybe"></a>function: <strong>run-state-file-maybe</strong></dt>
<dd>
<p>run the state file depending on the state variables 
</p>
</dd></dl>

<hr>
<a name="The-Virtual-Trackball"></a>
<div class="header">
<p>
Next: <a href="#Clipping-Functions" accesskey="n" rel="next">Clipping Functions</a>, Previous: <a href="#State-Functions" accesskey="p" rel="previous">State Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="The-Virtual-Trackball-1"></a>
<h3 class="section">11.24 The Virtual Trackball</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#vt_002dsurface" accesskey="1">vt-surface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#vt_002dsurface_002dstatus" accesskey="2">vt-surface-status</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="vt_002dsurface"></a>
<div class="header">
<p>
Next: <a href="#vt_002dsurface_002dstatus" accesskey="n" rel="next">vt-surface-status</a>, Up: <a href="#The-Virtual-Trackball" accesskey="u" rel="up">The Virtual Trackball</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="vt_002dsurface-1"></a>
<h4 class="subsection">11.24.1 vt-surface</h4>
<dl>
<dt><a name="index-vt_002dsurface"></a>function: <strong>vt-surface</strong> <em>mode</em></dt>
<dd>
<p>Where <em>mode</em> is an integer number
</p>
<p>How should the mouse move the view? 
</p>
<p>mode=1 for &quot;Flat&quot;, mode=2 for &quot;Spherical Surface&quot; 
</p>
</dd></dl>

<hr>
<a name="vt_002dsurface_002dstatus"></a>
<div class="header">
<p>
Previous: <a href="#vt_002dsurface" accesskey="p" rel="previous">vt-surface</a>, Up: <a href="#The-Virtual-Trackball" accesskey="u" rel="up">The Virtual Trackball</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="vt_002dsurface_002dstatus-1"></a>
<h4 class="subsection">11.24.2 vt-surface-status</h4>
<dl>
<dt><a name="index-vt_002dsurface_002dstatus"></a>function: <strong>vt-surface-status</strong></dt>
<dd>
<p>return the mouse view status mode 
</p>
<p>mode=1 for &quot;Flat&quot;, mode=2 for &quot;Spherical Surface&quot; 
</p>
</dd></dl>

<hr>
<a name="Clipping-Functions"></a>
<div class="header">
<p>
Next: <a href="#Unit-Cell-interface" accesskey="n" rel="next">Unit Cell interface</a>, Previous: <a href="#The-Virtual-Trackball" accesskey="p" rel="previous">The Virtual Trackball</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Clipping-Functions-1"></a>
<h3 class="section">11.25 Clipping Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dclipping_002dback" accesskey="1">set-clipping-back</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dclipping_002dfront" accesskey="2">set-clipping-front</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dclipping_002dback"></a>
<div class="header">
<p>
Next: <a href="#set_002dclipping_002dfront" accesskey="n" rel="next">set-clipping-front</a>, Up: <a href="#Clipping-Functions" accesskey="u" rel="up">Clipping Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dclipping_002dback-1"></a>
<h4 class="subsection">11.25.1 set-clipping-back</h4>
<dl>
<dt><a name="index-set_002dclipping_002dback"></a>function: <strong>set-clipping-back</strong> <em>v</em></dt>
<dd>
<p>Where <em>v</em> is a number
</p>
<p>set clipping plane back 
</p>
</dd></dl>

<hr>
<a name="set_002dclipping_002dfront"></a>
<div class="header">
<p>
Previous: <a href="#set_002dclipping_002dback" accesskey="p" rel="previous">set-clipping-back</a>, Up: <a href="#Clipping-Functions" accesskey="u" rel="up">Clipping Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dclipping_002dfront-1"></a>
<h4 class="subsection">11.25.2 set-clipping-front</h4>
<dl>
<dt><a name="index-set_002dclipping_002dfront"></a>function: <strong>set-clipping-front</strong> <em>v</em></dt>
<dd>
<p>Where <em>v</em> is a number
</p>
<p>set clipping plane front 
</p>
</dd></dl>

<hr>
<a name="Unit-Cell-interface"></a>
<div class="header">
<p>
Next: <a href="#Colour" accesskey="n" rel="next">Colour</a>, Previous: <a href="#Clipping-Functions" accesskey="p" rel="previous">Clipping Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Unit-Cell-interface-1"></a>
<h3 class="section">11.26 Unit Cell interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#get_002dshow_002dunit_002dcell" accesskey="1">get-show-unit-cell</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dunit_002dcells_002dall" accesskey="2">set-show-unit-cells-all</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dunit_002dcell" accesskey="3">set-show-unit-cell</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="get_002dshow_002dunit_002dcell"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002dunit_002dcells_002dall" accesskey="n" rel="next">set-show-unit-cells-all</a>, Up: <a href="#Unit-Cell-interface" accesskey="u" rel="up">Unit Cell interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dshow_002dunit_002dcell-1"></a>
<h4 class="subsection">11.26.1 get-show-unit-cell</h4>
<dl>
<dt><a name="index-get_002dshow_002dunit_002dcell"></a>function: <strong>get-show-unit-cell</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the stage of show unit cell for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002dunit_002dcells_002dall"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002dunit_002dcell" accesskey="n" rel="next">set-show-unit-cell</a>, Previous: <a href="#get_002dshow_002dunit_002dcell" accesskey="p" rel="previous">get-show-unit-cell</a>, Up: <a href="#Unit-Cell-interface" accesskey="u" rel="up">Unit Cell interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dunit_002dcells_002dall-1"></a>
<h4 class="subsection">11.26.2 set-show-unit-cells-all</h4>
<dl>
<dt><a name="index-set_002dshow_002dunit_002dcells_002dall"></a>function: <strong>set-show-unit-cells-all</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>set the state of show unit cell for all molecules 
</p>
<p>1 for displayed 0 for undisplayed 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002dunit_002dcell"></a>
<div class="header">
<p>
Previous: <a href="#set_002dshow_002dunit_002dcells_002dall" accesskey="p" rel="previous">set-show-unit-cells-all</a>, Up: <a href="#Unit-Cell-interface" accesskey="u" rel="up">Unit Cell interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dunit_002dcell-1"></a>
<h4 class="subsection">11.26.3 set-show-unit-cell</h4>
<dl>
<dt><a name="index-set_002dshow_002dunit_002dcell"></a>function: <strong>set-show-unit-cell</strong> <em>imol istate</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>istate</em> is an integer number
</li></ul>

<p>set the state of show unit cell for the particular molecule number imol 
</p>
<p>1 for displayed 0 for undisplayed 
</p>
</dd></dl>

<hr>
<a name="Colour"></a>
<div class="header">
<p>
Next: <a href="#Map-colour" accesskey="n" rel="next">Map colour</a>, Previous: <a href="#Unit-Cell-interface" accesskey="p" rel="previous">Unit Cell interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Colour-1"></a>
<h3 class="section">11.27 Colour</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb" accesskey="1">set-colour-map-rotation-on-read-pdb</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag" accesskey="2">set-colour-map-rotation-on-read-pdb-flag</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag" accesskey="3">set-colour-map-rotation-on-read-pdb-c-only-flag</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcolour_002dby_002dchain" accesskey="4">set-colour-by-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcolour_002dby_002dchain_002dgoodsell_002dmode" accesskey="5">set-colour-by-chain-goodsell-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dcolour_002dby_002dmolecule" accesskey="6">set-colour-by-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsymmetry_002dcolour" accesskey="7">set-symmetry-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb"></a>
<div class="header">
<p>
Next: <a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag" accesskey="n" rel="next">set-colour-map-rotation-on-read-pdb-flag</a>, Up: <a href="#Colour" accesskey="u" rel="up">Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb-1"></a>
<h4 class="subsection">11.27.1 set-colour-map-rotation-on-read-pdb</h4>
<dl>
<dt><a name="index-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb"></a>function: <strong>set-colour-map-rotation-on-read-pdb</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the hue change step on reading a new molecule 
</p>
</dd></dl>

<hr>
<a name="set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag"></a>
<div class="header">
<p>
Next: <a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag" accesskey="n" rel="next">set-colour-map-rotation-on-read-pdb-c-only-flag</a>, Previous: <a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb" accesskey="p" rel="previous">set-colour-map-rotation-on-read-pdb</a>, Up: <a href="#Colour" accesskey="u" rel="up">Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag-1"></a>
<h4 class="subsection">11.27.2 set-colour-map-rotation-on-read-pdb-flag</h4>
<dl>
<dt><a name="index-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag"></a>function: <strong>set-colour-map-rotation-on-read-pdb-flag</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>shall the hue change step be used? 
</p>
</dd></dl>

<hr>
<a name="set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag"></a>
<div class="header">
<p>
Next: <a href="#set_002dcolour_002dby_002dchain" accesskey="n" rel="next">set-colour-by-chain</a>, Previous: <a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag" accesskey="p" rel="previous">set-colour-map-rotation-on-read-pdb-flag</a>, Up: <a href="#Colour" accesskey="u" rel="up">Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag-1"></a>
<h4 class="subsection">11.27.3 set-colour-map-rotation-on-read-pdb-c-only-flag</h4>
<dl>
<dt><a name="index-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag"></a>function: <strong>set-colour-map-rotation-on-read-pdb-c-only-flag</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>shall the colour map rotation apply only to C atoms? 
</p>
</dd></dl>

<hr>
<a name="set_002dcolour_002dby_002dchain"></a>
<div class="header">
<p>
Next: <a href="#set_002dcolour_002dby_002dchain_002dgoodsell_002dmode" accesskey="n" rel="next">set-colour-by-chain-goodsell-mode</a>, Previous: <a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag" accesskey="p" rel="previous">set-colour-map-rotation-on-read-pdb-c-only-flag</a>, Up: <a href="#Colour" accesskey="u" rel="up">Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcolour_002dby_002dchain-1"></a>
<h4 class="subsection">11.27.4 set-colour-by-chain</h4>
<dl>
<dt><a name="index-set_002dcolour_002dby_002dchain"></a>function: <strong>set-colour-by-chain</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>colour molecule number imol by chain type 
</p>
</dd></dl>

<hr>
<a name="set_002dcolour_002dby_002dchain_002dgoodsell_002dmode"></a>
<div class="header">
<p>
Next: <a href="#set_002dcolour_002dby_002dmolecule" accesskey="n" rel="next">set-colour-by-molecule</a>, Previous: <a href="#set_002dcolour_002dby_002dchain" accesskey="p" rel="previous">set-colour-by-chain</a>, Up: <a href="#Colour" accesskey="u" rel="up">Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcolour_002dby_002dchain_002dgoodsell_002dmode-1"></a>
<h4 class="subsection">11.27.5 set-colour-by-chain-goodsell-mode</h4>
<dl>
<dt><a name="index-set_002dcolour_002dby_002dchain_002dgoodsell_002dmode"></a>function: <strong>set-colour-by-chain-goodsell-mode</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>colour molecule number imol by chain type, goodsell-like colour scheme 
</p>
</dd></dl>

<hr>
<a name="set_002dcolour_002dby_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#set_002dsymmetry_002dcolour" accesskey="n" rel="next">set-symmetry-colour</a>, Previous: <a href="#set_002dcolour_002dby_002dchain_002dgoodsell_002dmode" accesskey="p" rel="previous">set-colour-by-chain-goodsell-mode</a>, Up: <a href="#Colour" accesskey="u" rel="up">Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcolour_002dby_002dmolecule-1"></a>
<h4 class="subsection">11.27.6 set-colour-by-molecule</h4>
<dl>
<dt><a name="index-set_002dcolour_002dby_002dmolecule"></a>function: <strong>set-colour-by-molecule</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>colour molecule number imol by molecule 
</p>
</dd></dl>

<hr>
<a name="set_002dsymmetry_002dcolour"></a>
<div class="header">
<p>
Previous: <a href="#set_002dcolour_002dby_002dmolecule" accesskey="p" rel="previous">set-colour-by-molecule</a>, Up: <a href="#Colour" accesskey="u" rel="up">Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsymmetry_002dcolour-1"></a>
<h4 class="subsection">11.27.7 set-symmetry-colour</h4>
<dl>
<dt><a name="index-set_002dsymmetry_002dcolour"></a>function: <strong>set-symmetry-colour</strong> <em>r g b</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>r</em> is a number
</li><li> <em>g</em> is a number
</li><li> <em>b</em> is a number
</li></ul>

<p>set the symmetry colour base 
</p>
</dd></dl>

<hr>
<a name="Map-colour"></a>
<div class="header">
<p>
Next: <a href="#Anisotropic-Atoms-Interface" accesskey="n" rel="next">Anisotropic Atoms Interface</a>, Previous: <a href="#Colour" accesskey="p" rel="previous">Colour</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-colour-1"></a>
<h3 class="section">11.28 Map colour</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dcolour_002dmap_002drotation_002dfor_002dmap" accesskey="1">set-colour-map-rotation-for-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation" accesskey="2">set-molecule-bonds-colour-map-rotation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation" accesskey="3">get-molecule-bonds-colour-map-rotation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dcolour_002dmap_002drotation_002dfor_002dmap"></a>
<div class="header">
<p>
Next: <a href="#set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation" accesskey="n" rel="next">set-molecule-bonds-colour-map-rotation</a>, Up: <a href="#Map-colour" accesskey="u" rel="up">Map colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dcolour_002dmap_002drotation_002dfor_002dmap-1"></a>
<h4 class="subsection">11.28.1 set-colour-map-rotation-for-map</h4>
<dl>
<dt><a name="index-set_002dcolour_002dmap_002drotation_002dfor_002dmap"></a>function: <strong>set-colour-map-rotation-for-map</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the colour map rotation (hue change) for maps 
</p>
<p>default: for maps is 14 degrees. 
</p>
</dd></dl>

<hr>
<a name="set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation"></a>
<div class="header">
<p>
Next: <a href="#get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation" accesskey="n" rel="next">get-molecule-bonds-colour-map-rotation</a>, Previous: <a href="#set_002dcolour_002dmap_002drotation_002dfor_002dmap" accesskey="p" rel="previous">set-colour-map-rotation-for-map</a>, Up: <a href="#Map-colour" accesskey="u" rel="up">Map colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation-1"></a>
<h4 class="subsection">11.28.2 set-molecule-bonds-colour-map-rotation</h4>
<dl>
<dt><a name="index-set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation"></a>function: <strong>set-molecule-bonds-colour-map-rotation</strong> <em>imol theta</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>theta</em> is a number
</li></ul>

<p>set the colour map rotation for molecule number imol 
</p>
<p>theta is in degrees 
</p>
</dd></dl>

<hr>
<a name="get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation"></a>
<div class="header">
<p>
Previous: <a href="#set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation" accesskey="p" rel="previous">set-molecule-bonds-colour-map-rotation</a>, Up: <a href="#Map-colour" accesskey="u" rel="up">Map colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation-1"></a>
<h4 class="subsection">11.28.3 get-molecule-bonds-colour-map-rotation</h4>
<dl>
<dt><a name="index-get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation"></a>function: <strong>get-molecule-bonds-colour-map-rotation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Get the colour map rotation for molecule number imol. 
</p>
</dd></dl>

<hr>
<a name="Anisotropic-Atoms-Interface"></a>
<div class="header">
<p>
Next: <a href="#Display-Functions" accesskey="n" rel="next">Display Functions</a>, Previous: <a href="#Map-colour" accesskey="p" rel="previous">Map colour</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Anisotropic-Atoms-Interface-1"></a>
<h3 class="section">11.29 Anisotropic Atoms Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#get_002dlimit_002daniso" accesskey="1">get-limit-aniso</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dshow_002dlimit_002daniso" accesskey="2">get-show-limit-aniso</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dshow_002daniso" accesskey="3">get-show-aniso</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dlimit_002daniso" accesskey="4">set-limit-aniso</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002daniso" accesskey="5">set-show-aniso</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002daniso_002dprobability" accesskey="6">set-aniso-probability</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002daniso_002dprobability" accesskey="7">get-aniso-probability</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="get_002dlimit_002daniso"></a>
<div class="header">
<p>
Next: <a href="#get_002dshow_002dlimit_002daniso" accesskey="n" rel="next">get-show-limit-aniso</a>, Up: <a href="#Anisotropic-Atoms-Interface" accesskey="u" rel="up">Anisotropic Atoms Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dlimit_002daniso-1"></a>
<h4 class="subsection">11.29.1 get-limit-aniso</h4>
<dl>
<dt><a name="index-get_002dlimit_002daniso"></a>function: <strong>get-limit-aniso</strong></dt>
<dd>
<p>get the aniso radius limit 
</p>
</dd></dl>

<hr>
<a name="get_002dshow_002dlimit_002daniso"></a>
<div class="header">
<p>
Next: <a href="#get_002dshow_002daniso" accesskey="n" rel="next">get-show-aniso</a>, Previous: <a href="#get_002dlimit_002daniso" accesskey="p" rel="previous">get-limit-aniso</a>, Up: <a href="#Anisotropic-Atoms-Interface" accesskey="u" rel="up">Anisotropic Atoms Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dshow_002dlimit_002daniso-1"></a>
<h4 class="subsection">11.29.2 get-show-limit-aniso</h4>
<dl>
<dt><a name="index-get_002dshow_002dlimit_002daniso"></a>function: <strong>get-show-limit-aniso</strong></dt>
<dd>
<p>get show the aniso limit 
</p>
</dd></dl>

<hr>
<a name="get_002dshow_002daniso"></a>
<div class="header">
<p>
Next: <a href="#set_002dlimit_002daniso" accesskey="n" rel="next">set-limit-aniso</a>, Previous: <a href="#get_002dshow_002dlimit_002daniso" accesskey="p" rel="previous">get-show-limit-aniso</a>, Up: <a href="#Anisotropic-Atoms-Interface" accesskey="u" rel="up">Anisotropic Atoms Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dshow_002daniso-1"></a>
<h4 class="subsection">11.29.3 get-show-aniso</h4>
<dl>
<dt><a name="index-get_002dshow_002daniso"></a>function: <strong>get-show-aniso</strong></dt>
<dd>
<p>return show-aniso-atoms state 
</p>
</dd></dl>

<hr>
<a name="set_002dlimit_002daniso"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002daniso" accesskey="n" rel="next">set-show-aniso</a>, Previous: <a href="#get_002dshow_002daniso" accesskey="p" rel="previous">get-show-aniso</a>, Up: <a href="#Anisotropic-Atoms-Interface" accesskey="u" rel="up">Anisotropic Atoms Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dlimit_002daniso-1"></a>
<h4 class="subsection">11.29.4 set-limit-aniso</h4>
<dl>
<dt><a name="index-set_002dlimit_002daniso"></a>function: <strong>set-limit-aniso</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set the aniso atom limit 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002daniso"></a>
<div class="header">
<p>
Next: <a href="#set_002daniso_002dprobability" accesskey="n" rel="next">set-aniso-probability</a>, Previous: <a href="#set_002dlimit_002daniso" accesskey="p" rel="previous">set-limit-aniso</a>, Up: <a href="#Anisotropic-Atoms-Interface" accesskey="u" rel="up">Anisotropic Atoms Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002daniso-1"></a>
<h4 class="subsection">11.29.5 set-show-aniso</h4>
<dl>
<dt><a name="index-set_002dshow_002daniso"></a>function: <strong>set-show-aniso</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set show aniso atoms 
</p>
</dd></dl>

<hr>
<a name="set_002daniso_002dprobability"></a>
<div class="header">
<p>
Next: <a href="#get_002daniso_002dprobability" accesskey="n" rel="next">get-aniso-probability</a>, Previous: <a href="#set_002dshow_002daniso" accesskey="p" rel="previous">set-show-aniso</a>, Up: <a href="#Anisotropic-Atoms-Interface" accesskey="u" rel="up">Anisotropic Atoms Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002daniso_002dprobability-1"></a>
<h4 class="subsection">11.29.6 set-aniso-probability</h4>
<dl>
<dt><a name="index-set_002daniso_002dprobability"></a>function: <strong>set-aniso-probability</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set aniso probability 
</p>
</dd></dl>

<hr>
<a name="get_002daniso_002dprobability"></a>
<div class="header">
<p>
Previous: <a href="#set_002daniso_002dprobability" accesskey="p" rel="previous">set-aniso-probability</a>, Up: <a href="#Anisotropic-Atoms-Interface" accesskey="u" rel="up">Anisotropic Atoms Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002daniso_002dprobability-1"></a>
<h4 class="subsection">11.29.7 get-aniso-probability</h4>
<dl>
<dt><a name="index-get_002daniso_002dprobability"></a>function: <strong>get-aniso-probability</strong></dt>
<dd>
<p>get aniso probability 
</p>
</dd></dl>

<hr>
<a name="Display-Functions"></a>
<div class="header">
<p>
Next: <a href="#Smooth-Scrolling" accesskey="n" rel="next">Smooth Scrolling</a>, Previous: <a href="#Anisotropic-Atoms-Interface" accesskey="p" rel="previous">Anisotropic Atoms Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Display-Functions-1"></a>
<h3 class="section">11.30 Display Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dgraphics_002dwindow_002dsize" accesskey="1">set-graphics-window-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dgraphics_002dwindow_002dposition" accesskey="2">set-graphics-window-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#store_002dgraphics_002dwindow_002dposition" accesskey="3">store-graphics-window-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences" accesskey="4">graphics-window-size-and-position-to-preferences</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002ddraw" accesskey="5">graphics-draw</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#zalman_002dstereo_002dmode" accesskey="6">zalman-stereo-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#hardware_002dstereo_002dmode" accesskey="7">hardware-stereo-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#stereo_002dmode_002dstate" accesskey="8">stereo-mode-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mono_002dmode" accesskey="9">mono-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#side_002dby_002dside_002dstereo_002dmode">side-by-side-stereo-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dhardware_002dstereo_002dangle_002dfactor">set-hardware-stereo-angle-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#hardware_002dstereo_002dangle_002dfactor_002dstate">hardware-stereo-angle-factor-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmodel_002dfit_002drefine_002ddialog_002dposition">set-model-fit-refine-dialog-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddisplay_002dcontrol_002ddialog_002dposition">set-display-control-dialog-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dgo_002dto_002datom_002dwindow_002dposition">set-go-to-atom-window-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddelete_002ddialog_002dposition">set-delete-dialog-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drotate_002dtranslate_002ddialog_002dposition">set-rotate-translate-dialog-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002daccept_002dreject_002ddialog_002dposition">set-accept-reject-dialog-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dramachandran_002dplot_002ddialog_002dposition">set-ramachandran-plot-dialog-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dedit_002dchi_002dangles_002ddialog_002dposition">set-edit-chi-angles-dialog-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drotamer_002dselection_002ddialog_002dposition">set-rotamer-selection-dialog-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dgraphics_002dwindow_002dsize"></a>
<div class="header">
<p>
Next: <a href="#set_002dgraphics_002dwindow_002dposition" accesskey="n" rel="next">set-graphics-window-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgraphics_002dwindow_002dsize-1"></a>
<h4 class="subsection">11.30.1 set-graphics-window-size</h4>
<dl>
<dt><a name="index-set_002dgraphics_002dwindow_002dsize"></a>function: <strong>set-graphics-window-size</strong> <em>x_size y_size</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_size</em> is an integer number
</li><li> <em>y_size</em> is an integer number
</li></ul>

<p>set the window size 
</p>
</dd></dl>

<hr>
<a name="set_002dgraphics_002dwindow_002dposition"></a>
<div class="header">
<p>
Next: <a href="#store_002dgraphics_002dwindow_002dposition" accesskey="n" rel="next">store-graphics-window-position</a>, Previous: <a href="#set_002dgraphics_002dwindow_002dsize" accesskey="p" rel="previous">set-graphics-window-size</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgraphics_002dwindow_002dposition-1"></a>
<h4 class="subsection">11.30.2 set-graphics-window-position</h4>
<dl>
<dt><a name="index-set_002dgraphics_002dwindow_002dposition"></a>function: <strong>set-graphics-window-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set the graphics window position 
</p>
</dd></dl>

<hr>
<a name="store_002dgraphics_002dwindow_002dposition"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences" accesskey="n" rel="next">graphics-window-size-and-position-to-preferences</a>, Previous: <a href="#set_002dgraphics_002dwindow_002dposition" accesskey="p" rel="previous">set-graphics-window-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="store_002dgraphics_002dwindow_002dposition-1"></a>
<h4 class="subsection">11.30.3 store-graphics-window-position</h4>
<dl>
<dt><a name="index-store_002dgraphics_002dwindow_002dposition"></a>function: <strong>store-graphics-window-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>store the graphics window position 
</p>
</dd></dl>

<hr>
<a name="graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences"></a>
<div class="header">
<p>
Next: <a href="#graphics_002ddraw" accesskey="n" rel="next">graphics-draw</a>, Previous: <a href="#store_002dgraphics_002dwindow_002dposition" accesskey="p" rel="previous">store-graphics-window-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences-1"></a>
<h4 class="subsection">11.30.4 graphics-window-size-and-position-to-preferences</h4>
<dl>
<dt><a name="index-graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences"></a>function: <strong>graphics-window-size-and-position-to-preferences</strong></dt>
<dd>
<p>store the graphics window position and size to zenops-graphics-window-size-and-postion.scm in the preferences directory. 
</p>
</dd></dl>

<hr>
<a name="graphics_002ddraw"></a>
<div class="header">
<p>
Next: <a href="#zalman_002dstereo_002dmode" accesskey="n" rel="next">zalman-stereo-mode</a>, Previous: <a href="#graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences" accesskey="p" rel="previous">graphics-window-size-and-position-to-preferences</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002ddraw-1"></a>
<h4 class="subsection">11.30.5 graphics-draw</h4>
<dl>
<dt><a name="index-graphics_002ddraw"></a>function: <strong>graphics-draw</strong></dt>
<dd>
<p>draw a frame 
</p>
</dd></dl>

<hr>
<a name="zalman_002dstereo_002dmode"></a>
<div class="header">
<p>
Next: <a href="#hardware_002dstereo_002dmode" accesskey="n" rel="next">hardware-stereo-mode</a>, Previous: <a href="#graphics_002ddraw" accesskey="p" rel="previous">graphics-draw</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="zalman_002dstereo_002dmode-1"></a>
<h4 class="subsection">11.30.6 zalman-stereo-mode</h4>
<dl>
<dt><a name="index-zalman_002dstereo_002dmode"></a>function: <strong>zalman-stereo-mode</strong></dt>
<dd>
<p>try to turn on Zalman stereo mode 
</p>
</dd></dl>

<hr>
<a name="hardware_002dstereo_002dmode"></a>
<div class="header">
<p>
Next: <a href="#stereo_002dmode_002dstate" accesskey="n" rel="next">stereo-mode-state</a>, Previous: <a href="#zalman_002dstereo_002dmode" accesskey="p" rel="previous">zalman-stereo-mode</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="hardware_002dstereo_002dmode-1"></a>
<h4 class="subsection">11.30.7 hardware-stereo-mode</h4>
<dl>
<dt><a name="index-hardware_002dstereo_002dmode"></a>function: <strong>hardware-stereo-mode</strong></dt>
<dd>
<p>try to turn on stereo mode 
</p>
</dd></dl>

<hr>
<a name="stereo_002dmode_002dstate"></a>
<div class="header">
<p>
Next: <a href="#mono_002dmode" accesskey="n" rel="next">mono-mode</a>, Previous: <a href="#hardware_002dstereo_002dmode" accesskey="p" rel="previous">hardware-stereo-mode</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="stereo_002dmode_002dstate-1"></a>
<h4 class="subsection">11.30.8 stereo-mode-state</h4>
<dl>
<dt><a name="index-stereo_002dmode_002dstate"></a>function: <strong>stereo-mode-state</strong></dt>
<dd>
<p>what is the stero state? 
</p>
<p>Returns: 1 for in hardware stereo, 2 for side by side stereo, else return 0. 
</p>
</dd></dl>

<hr>
<a name="mono_002dmode"></a>
<div class="header">
<p>
Next: <a href="#side_002dby_002dside_002dstereo_002dmode" accesskey="n" rel="next">side-by-side-stereo-mode</a>, Previous: <a href="#stereo_002dmode_002dstate" accesskey="p" rel="previous">stereo-mode-state</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mono_002dmode-1"></a>
<h4 class="subsection">11.30.9 mono-mode</h4>
<dl>
<dt><a name="index-mono_002dmode"></a>function: <strong>mono-mode</strong></dt>
<dd>
<p>try to turn on mono mode 
</p>
</dd></dl>

<hr>
<a name="side_002dby_002dside_002dstereo_002dmode"></a>
<div class="header">
<p>
Next: <a href="#set_002dhardware_002dstereo_002dangle_002dfactor" accesskey="n" rel="next">set-hardware-stereo-angle-factor</a>, Previous: <a href="#mono_002dmode" accesskey="p" rel="previous">mono-mode</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="side_002dby_002dside_002dstereo_002dmode-1"></a>
<h4 class="subsection">11.30.10 side-by-side-stereo-mode</h4>
<dl>
<dt><a name="index-side_002dby_002dside_002dstereo_002dmode"></a>function: <strong>side-by-side-stereo-mode</strong> <em>use_wall_eye_mode</em></dt>
<dd>
<p>Where <em>use_wall_eye_mode</em> is an integer number
</p>
<p>turn on side bye side stereo mode 
</p>
</dd></dl>

<hr>
<a name="set_002dhardware_002dstereo_002dangle_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#hardware_002dstereo_002dangle_002dfactor_002dstate" accesskey="n" rel="next">hardware-stereo-angle-factor-state</a>, Previous: <a href="#side_002dby_002dside_002dstereo_002dmode" accesskey="p" rel="previous">side-by-side-stereo-mode</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dhardware_002dstereo_002dangle_002dfactor-1"></a>
<h4 class="subsection">11.30.11 set-hardware-stereo-angle-factor</h4>
<dl>
<dt><a name="index-set_002dhardware_002dstereo_002dangle_002dfactor"></a>function: <strong>set-hardware-stereo-angle-factor</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>how much should the eyes be separated in stereo mode? 
</p>
</dd></dl>

<hr>
<a name="hardware_002dstereo_002dangle_002dfactor_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dmodel_002dfit_002drefine_002ddialog_002dposition" accesskey="n" rel="next">set-model-fit-refine-dialog-position</a>, Previous: <a href="#set_002dhardware_002dstereo_002dangle_002dfactor" accesskey="p" rel="previous">set-hardware-stereo-angle-factor</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="hardware_002dstereo_002dangle_002dfactor_002dstate-1"></a>
<h4 class="subsection">11.30.12 hardware-stereo-angle-factor-state</h4>
<dl>
<dt><a name="index-hardware_002dstereo_002dangle_002dfactor_002dstate"></a>function: <strong>hardware-stereo-angle-factor-state</strong></dt>
<dd>
<p>return the hardware stereo angle factor 
</p>
</dd></dl>

<hr>
<a name="set_002dmodel_002dfit_002drefine_002ddialog_002dposition"></a>
<div class="header">
<p>
Next: <a href="#set_002ddisplay_002dcontrol_002ddialog_002dposition" accesskey="n" rel="next">set-display-control-dialog-position</a>, Previous: <a href="#hardware_002dstereo_002dangle_002dfactor_002dstate" accesskey="p" rel="previous">hardware-stereo-angle-factor-state</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmodel_002dfit_002drefine_002ddialog_002dposition-1"></a>
<h4 class="subsection">11.30.13 set-model-fit-refine-dialog-position</h4>
<dl>
<dt><a name="index-set_002dmodel_002dfit_002drefine_002ddialog_002dposition"></a>function: <strong>set-model-fit-refine-dialog-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set position of Model/Fit/Refine dialog 
</p>
</dd></dl>

<hr>
<a name="set_002ddisplay_002dcontrol_002ddialog_002dposition"></a>
<div class="header">
<p>
Next: <a href="#set_002dgo_002dto_002datom_002dwindow_002dposition" accesskey="n" rel="next">set-go-to-atom-window-position</a>, Previous: <a href="#set_002dmodel_002dfit_002drefine_002ddialog_002dposition" accesskey="p" rel="previous">set-model-fit-refine-dialog-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddisplay_002dcontrol_002ddialog_002dposition-1"></a>
<h4 class="subsection">11.30.14 set-display-control-dialog-position</h4>
<dl>
<dt><a name="index-set_002ddisplay_002dcontrol_002ddialog_002dposition"></a>function: <strong>set-display-control-dialog-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set position of Display Control dialog 
</p>
</dd></dl>

<hr>
<a name="set_002dgo_002dto_002datom_002dwindow_002dposition"></a>
<div class="header">
<p>
Next: <a href="#set_002ddelete_002ddialog_002dposition" accesskey="n" rel="next">set-delete-dialog-position</a>, Previous: <a href="#set_002ddisplay_002dcontrol_002ddialog_002dposition" accesskey="p" rel="previous">set-display-control-dialog-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgo_002dto_002datom_002dwindow_002dposition-1"></a>
<h4 class="subsection">11.30.15 set-go-to-atom-window-position</h4>
<dl>
<dt><a name="index-set_002dgo_002dto_002datom_002dwindow_002dposition"></a>function: <strong>set-go-to-atom-window-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set position of Go To Atom dialog 
</p>
</dd></dl>

<hr>
<a name="set_002ddelete_002ddialog_002dposition"></a>
<div class="header">
<p>
Next: <a href="#set_002drotate_002dtranslate_002ddialog_002dposition" accesskey="n" rel="next">set-rotate-translate-dialog-position</a>, Previous: <a href="#set_002dgo_002dto_002datom_002dwindow_002dposition" accesskey="p" rel="previous">set-go-to-atom-window-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddelete_002ddialog_002dposition-1"></a>
<h4 class="subsection">11.30.16 set-delete-dialog-position</h4>
<dl>
<dt><a name="index-set_002ddelete_002ddialog_002dposition"></a>function: <strong>set-delete-dialog-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set position of Delete dialog 
</p>
</dd></dl>

<hr>
<a name="set_002drotate_002dtranslate_002ddialog_002dposition"></a>
<div class="header">
<p>
Next: <a href="#set_002daccept_002dreject_002ddialog_002dposition" accesskey="n" rel="next">set-accept-reject-dialog-position</a>, Previous: <a href="#set_002ddelete_002ddialog_002dposition" accesskey="p" rel="previous">set-delete-dialog-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drotate_002dtranslate_002ddialog_002dposition-1"></a>
<h4 class="subsection">11.30.17 set-rotate-translate-dialog-position</h4>
<dl>
<dt><a name="index-set_002drotate_002dtranslate_002ddialog_002dposition"></a>function: <strong>set-rotate-translate-dialog-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set position of the Rotate/Translate Residue Range dialog 
</p>
</dd></dl>

<hr>
<a name="set_002daccept_002dreject_002ddialog_002dposition"></a>
<div class="header">
<p>
Next: <a href="#set_002dramachandran_002dplot_002ddialog_002dposition" accesskey="n" rel="next">set-ramachandran-plot-dialog-position</a>, Previous: <a href="#set_002drotate_002dtranslate_002ddialog_002dposition" accesskey="p" rel="previous">set-rotate-translate-dialog-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002daccept_002dreject_002ddialog_002dposition-1"></a>
<h4 class="subsection">11.30.18 set-accept-reject-dialog-position</h4>
<dl>
<dt><a name="index-set_002daccept_002dreject_002ddialog_002dposition"></a>function: <strong>set-accept-reject-dialog-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set position of the Accept/Reject dialog 
</p>
</dd></dl>

<hr>
<a name="set_002dramachandran_002dplot_002ddialog_002dposition"></a>
<div class="header">
<p>
Next: <a href="#set_002dedit_002dchi_002dangles_002ddialog_002dposition" accesskey="n" rel="next">set-edit-chi-angles-dialog-position</a>, Previous: <a href="#set_002daccept_002dreject_002ddialog_002dposition" accesskey="p" rel="previous">set-accept-reject-dialog-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dramachandran_002dplot_002ddialog_002dposition-1"></a>
<h4 class="subsection">11.30.19 set-ramachandran-plot-dialog-position</h4>
<dl>
<dt><a name="index-set_002dramachandran_002dplot_002ddialog_002dposition"></a>function: <strong>set-ramachandran-plot-dialog-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set position of the Ramachadran Plot dialog 
</p>
</dd></dl>

<hr>
<a name="set_002dedit_002dchi_002dangles_002ddialog_002dposition"></a>
<div class="header">
<p>
Next: <a href="#set_002drotamer_002dselection_002ddialog_002dposition" accesskey="n" rel="next">set-rotamer-selection-dialog-position</a>, Previous: <a href="#set_002dramachandran_002dplot_002ddialog_002dposition" accesskey="p" rel="previous">set-ramachandran-plot-dialog-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dedit_002dchi_002dangles_002ddialog_002dposition-1"></a>
<h4 class="subsection">11.30.20 set-edit-chi-angles-dialog-position</h4>
<dl>
<dt><a name="index-set_002dedit_002dchi_002dangles_002ddialog_002dposition"></a>function: <strong>set-edit-chi-angles-dialog-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set edit chi angles dialog position 
</p>
</dd></dl>

<hr>
<a name="set_002drotamer_002dselection_002ddialog_002dposition"></a>
<div class="header">
<p>
Previous: <a href="#set_002dedit_002dchi_002dangles_002ddialog_002dposition" accesskey="p" rel="previous">set-edit-chi-angles-dialog-position</a>, Up: <a href="#Display-Functions" accesskey="u" rel="up">Display Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drotamer_002dselection_002ddialog_002dposition-1"></a>
<h4 class="subsection">11.30.21 set-rotamer-selection-dialog-position</h4>
<dl>
<dt><a name="index-set_002drotamer_002dselection_002ddialog_002dposition"></a>function: <strong>set-rotamer-selection-dialog-position</strong> <em>x_pos y_pos</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x_pos</em> is an integer number
</li><li> <em>y_pos</em> is an integer number
</li></ul>

<p>set rotamer selection dialog position 
</p>
</dd></dl>

<hr>
<a name="Smooth-Scrolling"></a>
<div class="header">
<p>
Next: <a href="#Font-Parameters" accesskey="n" rel="next">Font Parameters</a>, Previous: <a href="#Display-Functions" accesskey="p" rel="previous">Display Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Smooth-Scrolling-1"></a>
<h3 class="section">11.31 Smooth Scrolling</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dsmooth_002dscroll_002dflag" accesskey="1">set-smooth-scroll-flag</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dsmooth_002dscroll" accesskey="2">get-smooth-scroll</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsmooth_002dscroll_002dsteps" accesskey="3">set-smooth-scroll-steps</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsmooth_002dscroll_002dlimit" accesskey="4">set-smooth-scroll-limit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dsmooth_002dscroll_002dflag"></a>
<div class="header">
<p>
Next: <a href="#get_002dsmooth_002dscroll" accesskey="n" rel="next">get-smooth-scroll</a>, Up: <a href="#Smooth-Scrolling" accesskey="u" rel="up">Smooth Scrolling</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsmooth_002dscroll_002dflag-1"></a>
<h4 class="subsection">11.31.1 set-smooth-scroll-flag</h4>
<dl>
<dt><a name="index-set_002dsmooth_002dscroll_002dflag"></a>function: <strong>set-smooth-scroll-flag</strong> <em>v</em></dt>
<dd>
<p>Where <em>v</em> is an integer number
</p>
<p>set smooth scrolling 
</p>
</dd></dl>

<hr>
<a name="get_002dsmooth_002dscroll"></a>
<div class="header">
<p>
Next: <a href="#set_002dsmooth_002dscroll_002dsteps" accesskey="n" rel="next">set-smooth-scroll-steps</a>, Previous: <a href="#set_002dsmooth_002dscroll_002dflag" accesskey="p" rel="previous">set-smooth-scroll-flag</a>, Up: <a href="#Smooth-Scrolling" accesskey="u" rel="up">Smooth Scrolling</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dsmooth_002dscroll-1"></a>
<h4 class="subsection">11.31.2 get-smooth-scroll</h4>
<dl>
<dt><a name="index-get_002dsmooth_002dscroll"></a>function: <strong>get-smooth-scroll</strong></dt>
<dd>
<p>return the smooth scrolling state 
</p>
</dd></dl>

<hr>
<a name="set_002dsmooth_002dscroll_002dsteps"></a>
<div class="header">
<p>
Next: <a href="#set_002dsmooth_002dscroll_002dlimit" accesskey="n" rel="next">set-smooth-scroll-limit</a>, Previous: <a href="#get_002dsmooth_002dscroll" accesskey="p" rel="previous">get-smooth-scroll</a>, Up: <a href="#Smooth-Scrolling" accesskey="u" rel="up">Smooth Scrolling</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsmooth_002dscroll_002dsteps-1"></a>
<h4 class="subsection">11.31.3 set-smooth-scroll-steps</h4>
<dl>
<dt><a name="index-set_002dsmooth_002dscroll_002dsteps"></a>function: <strong>set-smooth-scroll-steps</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set the number of steps in the smooth scroll 
</p>
<p>Set more steps (e.g. 50) for more smoothness (default 10). 
</p>
</dd></dl>

<hr>
<a name="set_002dsmooth_002dscroll_002dlimit"></a>
<div class="header">
<p>
Previous: <a href="#set_002dsmooth_002dscroll_002dsteps" accesskey="p" rel="previous">set-smooth-scroll-steps</a>, Up: <a href="#Smooth-Scrolling" accesskey="u" rel="up">Smooth Scrolling</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsmooth_002dscroll_002dlimit-1"></a>
<h4 class="subsection">11.31.4 set-smooth-scroll-limit</h4>
<dl>
<dt><a name="index-set_002dsmooth_002dscroll_002dlimit"></a>function: <strong>set-smooth-scroll-limit</strong> <em>lim</em></dt>
<dd>
<p>Where <em>lim</em> is a number
</p>
<p>do not scroll for distances greater this limit 
</p>
</dd></dl>

<hr>
<a name="Font-Parameters"></a>
<div class="header">
<p>
Next: <a href="#Rotation-Centre" accesskey="n" rel="next">Rotation Centre</a>, Previous: <a href="#Smooth-Scrolling" accesskey="p" rel="previous">Smooth Scrolling</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Font-Parameters-1"></a>
<h3 class="section">11.32 Font Parameters</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dfont_002dsize" accesskey="1">set-font-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dfont_002dsize" accesskey="2">get-font-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfont_002dcolour" accesskey="3">set-font-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002duse_002dstroke_002dcharacters" accesskey="4">set-use-stroke-characters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dfont_002dsize"></a>
<div class="header">
<p>
Next: <a href="#get_002dfont_002dsize" accesskey="n" rel="next">get-font-size</a>, Up: <a href="#Font-Parameters" accesskey="u" rel="up">Font Parameters</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfont_002dsize-1"></a>
<h4 class="subsection">11.32.1 set-font-size</h4>
<dl>
<dt><a name="index-set_002dfont_002dsize"></a>function: <strong>set-font-size</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set the font size 
</p>
</dd></dl>

<hr>
<a name="get_002dfont_002dsize"></a>
<div class="header">
<p>
Next: <a href="#set_002dfont_002dcolour" accesskey="n" rel="next">set-font-colour</a>, Previous: <a href="#set_002dfont_002dsize" accesskey="p" rel="previous">set-font-size</a>, Up: <a href="#Font-Parameters" accesskey="u" rel="up">Font Parameters</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dfont_002dsize-1"></a>
<h4 class="subsection">11.32.2 get-font-size</h4>
<dl>
<dt><a name="index-get_002dfont_002dsize"></a>function: <strong>get-font-size</strong></dt>
<dd>
<p>return the font size 
</p>
<p>Returns: 1 (small) 2 (medium, default) 3 (large) 
</p>
</dd></dl>

<hr>
<a name="set_002dfont_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#set_002duse_002dstroke_002dcharacters" accesskey="n" rel="next">set-use-stroke-characters</a>, Previous: <a href="#get_002dfont_002dsize" accesskey="p" rel="previous">get-font-size</a>, Up: <a href="#Font-Parameters" accesskey="u" rel="up">Font Parameters</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfont_002dcolour-1"></a>
<h4 class="subsection">11.32.3 set-font-colour</h4>
<dl>
<dt><a name="index-set_002dfont_002dcolour"></a>function: <strong>set-font-colour</strong> <em>red green blue</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>red</em> is a number
</li><li> <em>green</em> is a number
</li><li> <em>blue</em> is a number
</li></ul>

<p>set the colour of the atom label font - the arguments are in the range 0-&gt;1 
</p>
</dd></dl>

<hr>
<a name="set_002duse_002dstroke_002dcharacters"></a>
<div class="header">
<p>
Previous: <a href="#set_002dfont_002dcolour" accesskey="p" rel="previous">set-font-colour</a>, Up: <a href="#Font-Parameters" accesskey="u" rel="up">Font Parameters</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002duse_002dstroke_002dcharacters-1"></a>
<h4 class="subsection">11.32.4 set-use-stroke-characters</h4>
<dl>
<dt><a name="index-set_002duse_002dstroke_002dcharacters"></a>function: <strong>set-use-stroke-characters</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set use stroke characters 
</p>
</dd></dl>

<hr>
<a name="Rotation-Centre"></a>
<div class="header">
<p>
Next: <a href="#Atom-Selection-Utilities" accesskey="n" rel="next">Atom Selection Utilities</a>, Previous: <a href="#Font-Parameters" accesskey="p" rel="previous">Font Parameters</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rotation-Centre-1"></a>
<h3 class="section">11.33 Rotation Centre</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002drotation_002dcentre_002dsize" accesskey="1">set-rotation-centre-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#recentre_002don_002dread_002dpdb" accesskey="2">recentre-on-read-pdb</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drecentre_002don_002dread_002dpdb" accesskey="3">set-recentre-on-read-pdb</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drotation_002dcentre" accesskey="4">set-rotation-centre</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#go_002dto_002dligand" accesskey="5">go-to-ligand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit" accesskey="6">set-go-to-ligand-n-atoms-limit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dreorienting_002dnext_002dresidue_002dmode" accesskey="7">set-reorienting-next-residue-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002drotation_002dcentre_002dsize"></a>
<div class="header">
<p>
Next: <a href="#recentre_002don_002dread_002dpdb" accesskey="n" rel="next">recentre-on-read-pdb</a>, Up: <a href="#Rotation-Centre" accesskey="u" rel="up">Rotation Centre</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drotation_002dcentre_002dsize-1"></a>
<h4 class="subsection">11.33.1 set-rotation-centre-size</h4>
<dl>
<dt><a name="index-set_002drotation_002dcentre_002dsize"></a>function: <strong>set-rotation-centre-size</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set rotoation centre marker size 
</p>
</dd></dl>

<hr>
<a name="recentre_002don_002dread_002dpdb"></a>
<div class="header">
<p>
Next: <a href="#set_002drecentre_002don_002dread_002dpdb" accesskey="n" rel="next">set-recentre-on-read-pdb</a>, Previous: <a href="#set_002drotation_002dcentre_002dsize" accesskey="p" rel="previous">set-rotation-centre-size</a>, Up: <a href="#Rotation-Centre" accesskey="u" rel="up">Rotation Centre</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="recentre_002don_002dread_002dpdb-1"></a>
<h4 class="subsection">11.33.2 recentre-on-read-pdb</h4>
<dl>
<dt><a name="index-recentre_002don_002dread_002dpdb"></a>function: <strong>recentre-on-read-pdb</strong></dt>
<dd>
<p>return the recentre-on-pdb state 
</p>
</dd></dl>

<hr>
<a name="set_002drecentre_002don_002dread_002dpdb"></a>
<div class="header">
<p>
Next: <a href="#set_002drotation_002dcentre" accesskey="n" rel="next">set-rotation-centre</a>, Previous: <a href="#recentre_002don_002dread_002dpdb" accesskey="p" rel="previous">recentre-on-read-pdb</a>, Up: <a href="#Rotation-Centre" accesskey="u" rel="up">Rotation Centre</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drecentre_002don_002dread_002dpdb-1"></a>
<h4 class="subsection">11.33.3 set-recentre-on-read-pdb</h4>
<dl>
<dt><a name="index-set_002drecentre_002don_002dread_002dpdb"></a>function: <strong>set-recentre-on-read-pdb</strong> <em>int</em></dt>
<dd>
<p>Where <em>int</em> is a short
</p>
<p>set the recentre-on-pdb state 
</p>
</dd></dl>

<hr>
<a name="set_002drotation_002dcentre"></a>
<div class="header">
<p>
Next: <a href="#go_002dto_002dligand" accesskey="n" rel="next">go-to-ligand</a>, Previous: <a href="#set_002drecentre_002don_002dread_002dpdb" accesskey="p" rel="previous">set-recentre-on-read-pdb</a>, Up: <a href="#Rotation-Centre" accesskey="u" rel="up">Rotation Centre</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drotation_002dcentre-1"></a>
<h4 class="subsection">11.33.4 set-rotation-centre</h4>
<dl>
<dt><a name="index-set_002drotation_002dcentre-1"></a>function: <strong>set-rotation-centre</strong> <em>x y z</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li></ul>

<p>set the rotation centre 
</p>
</dd></dl>

<hr>
<a name="go_002dto_002dligand"></a>
<div class="header">
<p>
Next: <a href="#set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit" accesskey="n" rel="next">set-go-to-ligand-n-atoms-limit</a>, Previous: <a href="#set_002drotation_002dcentre" accesskey="p" rel="previous">set-rotation-centre</a>, Up: <a href="#Rotation-Centre" accesskey="u" rel="up">Rotation Centre</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="go_002dto_002dligand-1"></a>
<h4 class="subsection">11.33.5 go-to-ligand</h4>
<dl>
<dt><a name="index-go_002dto_002dligand"></a>function: <strong>go-to-ligand</strong></dt>
<dd>
<p>centre on the ligand of the &quot;active molecule&quot;, if we are already there, centre on the next hetgroup (etc) 
</p>
</dd></dl>

<hr>
<a name="set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit"></a>
<div class="header">
<p>
Next: <a href="#set_002dreorienting_002dnext_002dresidue_002dmode" accesskey="n" rel="next">set-reorienting-next-residue-mode</a>, Previous: <a href="#go_002dto_002dligand" accesskey="p" rel="previous">go-to-ligand</a>, Up: <a href="#Rotation-Centre" accesskey="u" rel="up">Rotation Centre</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit-1"></a>
<h4 class="subsection">11.33.6 set-go-to-ligand-n-atoms-limit</h4>
<dl>
<dt><a name="index-set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit"></a>function: <strong>set-go-to-ligand-n-atoms-limit</strong> <em>n_atom_min</em></dt>
<dd>
<p>Where <em>n_atom_min</em> is an integer number
</p>
<p>go to the ligand that has more than n_atom_min atoms 
</p>
</dd></dl>

<hr>
<a name="set_002dreorienting_002dnext_002dresidue_002dmode"></a>
<div class="header">
<p>
Previous: <a href="#set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit" accesskey="p" rel="previous">set-go-to-ligand-n-atoms-limit</a>, Up: <a href="#Rotation-Centre" accesskey="u" rel="up">Rotation Centre</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dreorienting_002dnext_002dresidue_002dmode-1"></a>
<h4 class="subsection">11.33.7 set-reorienting-next-residue-mode</h4>
<dl>
<dt><a name="index-set_002dreorienting_002dnext_002dresidue_002dmode"></a>function: <strong>set-reorienting-next-residue-mode</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>rotate the view so that the next main-chain atoms are oriented in the same direction as the previous - hence side-chain always seems to be &quot;up&quot; - set this mode to 1 for reorientation-mode - and 0 for off (standard translation) 
</p>
</dd></dl>

<hr>
<a name="Atom-Selection-Utilities"></a>
<div class="header">
<p>
Next: <a href="#Skeletonization-Interface" accesskey="n" rel="next">Skeletonization Interface</a>, Previous: <a href="#Rotation-Centre" accesskey="p" rel="previous">Rotation Centre</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Atom-Selection-Utilities-1"></a>
<h3 class="section">11.34 Atom Selection Utilities</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#median_002dtemperature_002dfactor" accesskey="1">median-temperature-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#average_002dtemperature_002dfactor" accesskey="2">average-temperature-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#standard_002ddeviation_002dtemperature_002dfactor" accesskey="3">standard-deviation-temperature-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clear_002dpending_002dpicks" accesskey="4">clear-pending-picks</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms" accesskey="5">set-default-temperature-factor-for-new-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#default_002dnew_002datoms_002db_002dfactor" accesskey="6">default-new-atoms-b-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dreset_002db_002dfactor_002dmoved_002datoms" accesskey="7">set-reset-b-factor-moved-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate" accesskey="8">get-reset-b-factor-moved-atoms-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002datom_002dattribute" accesskey="9">set-atom-attribute</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002datom_002dstring_002dattribute">set-atom-string-attribute</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002datom_002dattributes">set-atom-attributes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dresidue_002dname">set-residue-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="median_002dtemperature_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#average_002dtemperature_002dfactor" accesskey="n" rel="next">average-temperature-factor</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="median_002dtemperature_002dfactor-1"></a>
<h4 class="subsection">11.34.1 median-temperature-factor</h4>
<dl>
<dt><a name="index-median_002dtemperature_002dfactor"></a>function: <strong>median-temperature-factor</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the median temperature factor for imol 
</p>
</dd></dl>

<hr>
<a name="average_002dtemperature_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#standard_002ddeviation_002dtemperature_002dfactor" accesskey="n" rel="next">standard-deviation-temperature-factor</a>, Previous: <a href="#median_002dtemperature_002dfactor" accesskey="p" rel="previous">median-temperature-factor</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="average_002dtemperature_002dfactor-1"></a>
<h4 class="subsection">11.34.2 average-temperature-factor</h4>
<dl>
<dt><a name="index-average_002dtemperature_002dfactor"></a>function: <strong>average-temperature-factor</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the average temperature factor for the atoms in imol 
</p>
</dd></dl>

<hr>
<a name="standard_002ddeviation_002dtemperature_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#clear_002dpending_002dpicks" accesskey="n" rel="next">clear-pending-picks</a>, Previous: <a href="#average_002dtemperature_002dfactor" accesskey="p" rel="previous">average-temperature-factor</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="standard_002ddeviation_002dtemperature_002dfactor-1"></a>
<h4 class="subsection">11.34.3 standard-deviation-temperature-factor</h4>
<dl>
<dt><a name="index-standard_002ddeviation_002dtemperature_002dfactor"></a>function: <strong>standard-deviation-temperature-factor</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the standard deviation of the atom temperature factors for imol 
</p>
</dd></dl>

<hr>
<a name="clear_002dpending_002dpicks"></a>
<div class="header">
<p>
Next: <a href="#set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms" accesskey="n" rel="next">set-default-temperature-factor-for-new-atoms</a>, Previous: <a href="#standard_002ddeviation_002dtemperature_002dfactor" accesskey="p" rel="previous">standard-deviation-temperature-factor</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002dpending_002dpicks-1"></a>
<h4 class="subsection">11.34.4 clear-pending-picks</h4>
<dl>
<dt><a name="index-clear_002dpending_002dpicks"></a>function: <strong>clear-pending-picks</strong></dt>
<dd>
<p>clear pending picks (stop coot thinking that the user is about to pick an atom). 
</p>
</dd></dl>

<hr>
<a name="set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms"></a>
<div class="header">
<p>
Next: <a href="#default_002dnew_002datoms_002db_002dfactor" accesskey="n" rel="next">default-new-atoms-b-factor</a>, Previous: <a href="#clear_002dpending_002dpicks" accesskey="p" rel="previous">clear-pending-picks</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms-1"></a>
<h4 class="subsection">11.34.5 set-default-temperature-factor-for-new-atoms</h4>
<dl>
<dt><a name="index-set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms"></a>function: <strong>set-default-temperature-factor-for-new-atoms</strong> <em>new_b</em></dt>
<dd>
<p>Where <em>new_b</em> is a number
</p>
<p>set the default temperature factor for newly created atoms (initial default 20) 
</p>
</dd></dl>

<hr>
<a name="default_002dnew_002datoms_002db_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#set_002dreset_002db_002dfactor_002dmoved_002datoms" accesskey="n" rel="next">set-reset-b-factor-moved-atoms</a>, Previous: <a href="#set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms" accesskey="p" rel="previous">set-default-temperature-factor-for-new-atoms</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="default_002dnew_002datoms_002db_002dfactor-1"></a>
<h4 class="subsection">11.34.6 default-new-atoms-b-factor</h4>
<dl>
<dt><a name="index-default_002dnew_002datoms_002db_002dfactor"></a>function: <strong>default-new-atoms-b-factor</strong></dt>
<dd>
<p>return the default temperature factor for newly created atoms 
</p>
</dd></dl>

<hr>
<a name="set_002dreset_002db_002dfactor_002dmoved_002datoms"></a>
<div class="header">
<p>
Next: <a href="#get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate" accesskey="n" rel="next">get-reset-b-factor-moved-atoms-state</a>, Previous: <a href="#default_002dnew_002datoms_002db_002dfactor" accesskey="p" rel="previous">default-new-atoms-b-factor</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dreset_002db_002dfactor_002dmoved_002datoms-1"></a>
<h4 class="subsection">11.34.7 set-reset-b-factor-moved-atoms</h4>
<dl>
<dt><a name="index-set_002dreset_002db_002dfactor_002dmoved_002datoms"></a>function: <strong>set-reset-b-factor-moved-atoms</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>reset temperature factor for all moved atoms to the default for new atoms (usually 30) 
</p>
</dd></dl>

<hr>
<a name="get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002datom_002dattribute" accesskey="n" rel="next">set-atom-attribute</a>, Previous: <a href="#set_002dreset_002db_002dfactor_002dmoved_002datoms" accesskey="p" rel="previous">set-reset-b-factor-moved-atoms</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate-1"></a>
<h4 class="subsection">11.34.8 get-reset-b-factor-moved-atoms-state</h4>
<dl>
<dt><a name="index-get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate"></a>function: <strong>get-reset-b-factor-moved-atoms-state</strong></dt>
<dd>
<p>return the state if temperature factors shoudl be reset for moved atoms 
</p>
</dd></dl>

<hr>
<a name="set_002datom_002dattribute"></a>
<div class="header">
<p>
Next: <a href="#set_002datom_002dstring_002dattribute" accesskey="n" rel="next">set-atom-string-attribute</a>, Previous: <a href="#get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate" accesskey="p" rel="previous">get-reset-b-factor-moved-atoms-state</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002datom_002dattribute-1"></a>
<h4 class="subsection">11.34.9 set-atom-attribute</h4>
<dl>
<dt><a name="index-set_002datom_002dattribute"></a>function: <strong>set-atom-attribute</strong> <em>imol chain_id resno ins_code atom_name alt_conf attribute_name val</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>atom_name</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>attribute_name</em> is a string
</li><li> <em>val</em> is a number
</li></ul>

<p>set a numberical attibute to the atom with the given specifier. 
</p>
<p>Attributes can be &quot;x&quot;, &quot;y&quot;,&quot;z&quot;, &quot;B&quot;, &quot;occ&quot; and the attribute val is a floating point number 
</p>
</dd></dl>

<hr>
<a name="set_002datom_002dstring_002dattribute"></a>
<div class="header">
<p>
Next: <a href="#set_002datom_002dattributes" accesskey="n" rel="next">set-atom-attributes</a>, Previous: <a href="#set_002datom_002dattribute" accesskey="p" rel="previous">set-atom-attribute</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002datom_002dstring_002dattribute-1"></a>
<h4 class="subsection">11.34.10 set-atom-string-attribute</h4>
<dl>
<dt><a name="index-set_002datom_002dstring_002dattribute"></a>function: <strong>set-atom-string-attribute</strong> <em>imol chain_id resno ins_code atom_name alt_conf attribute_name attribute_value</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>atom_name</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>attribute_name</em> is a string
</li><li> <em>attribute_value</em> is a string
</li></ul>

<p>set a string attibute to the atom with the given specifier. 
</p>
<p>Attributes can be &quot;atom-name&quot;, &quot;alt-conf&quot;, &quot;element&quot; or &quot;segid&quot;. 
</p>
</dd></dl>

<hr>
<a name="set_002datom_002dattributes"></a>
<div class="header">
<p>
Next: <a href="#set_002dresidue_002dname" accesskey="n" rel="next">set-residue-name</a>, Previous: <a href="#set_002datom_002dstring_002dattribute" accesskey="p" rel="previous">set-atom-string-attribute</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002datom_002dattributes-1"></a>
<h4 class="subsection">11.34.11 set-atom-attributes</h4>
<dl>
<dt><a name="index-set_002datom_002dattributes"></a>function: <strong>set-atom-attributes</strong> <em>attribute_expression_list</em></dt>
<dd>
<p>Where <em>attribute_expression_list</em> is a SCM
</p>
<p>set lots of atom attributes at once by-passing the rebonding and redrawing of the above 2 functions 
</p>
</dd></dl>

<hr>
<a name="set_002dresidue_002dname"></a>
<div class="header">
<p>
Previous: <a href="#set_002datom_002dattributes" accesskey="p" rel="previous">set-atom-attributes</a>, Up: <a href="#Atom-Selection-Utilities" accesskey="u" rel="up">Atom Selection Utilities</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dresidue_002dname-1"></a>
<h4 class="subsection">11.34.12 set-residue-name</h4>
<dl>
<dt><a name="index-set_002dresidue_002dname"></a>function: <strong>set-residue-name</strong> <em>imol chain_id res_no ins_code new_residue_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>new_residue_name</em> is a string
</li></ul>

<p>set the residue name of the specified residue 
</p>
</dd></dl>

<hr>
<a name="Skeletonization-Interface"></a>
<div class="header">
<p>
Next: <a href="#Save-Coordinates" accesskey="n" rel="next">Save Coordinates</a>, Previous: <a href="#Atom-Selection-Utilities" accesskey="p" rel="previous">Atom Selection Utilities</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Skeletonization-Interface-1"></a>
<h3 class="section">11.35 Skeletonization Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#skeletonize_002dmap" accesskey="1">skeletonize-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#unskeletonize_002dmap" accesskey="2">unskeletonize-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmax_002dskeleton_002dsearch_002ddepth" accesskey="3">set-max-skeleton-search-depth</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dskeleton_002dbox_002dsize" accesskey="4">set-skeleton-box-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="skeletonize_002dmap"></a>
<div class="header">
<p>
Next: <a href="#unskeletonize_002dmap" accesskey="n" rel="next">unskeletonize-map</a>, Up: <a href="#Skeletonization-Interface" accesskey="u" rel="up">Skeletonization Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="skeletonize_002dmap-1"></a>
<h4 class="subsection">11.35.1 skeletonize-map</h4>
<dl>
<dt><a name="index-skeletonize_002dmap"></a>function: <strong>skeletonize-map</strong> <em>imol prune_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>prune_flag</em> is an integer number
</li></ul>

<p>skeletonize molecule number imol 
</p>
<p>the prune_flag should almost always be 0.
</p>
<p>NOTE:: The arguments to have been reversed for coot 0.8.3 and later (now the molecule number comes first). 
</p>
</dd></dl>

<hr>
<a name="unskeletonize_002dmap"></a>
<div class="header">
<p>
Next: <a href="#set_002dmax_002dskeleton_002dsearch_002ddepth" accesskey="n" rel="next">set-max-skeleton-search-depth</a>, Previous: <a href="#skeletonize_002dmap" accesskey="p" rel="previous">skeletonize-map</a>, Up: <a href="#Skeletonization-Interface" accesskey="u" rel="up">Skeletonization Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="unskeletonize_002dmap-1"></a>
<h4 class="subsection">11.35.2 unskeletonize-map</h4>
<dl>
<dt><a name="index-unskeletonize_002dmap"></a>function: <strong>unskeletonize-map</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>undisplay the skeleton on molecule number imol 
</p>
</dd></dl>

<hr>
<a name="set_002dmax_002dskeleton_002dsearch_002ddepth"></a>
<div class="header">
<p>
Next: <a href="#set_002dskeleton_002dbox_002dsize" accesskey="n" rel="next">set-skeleton-box-size</a>, Previous: <a href="#unskeletonize_002dmap" accesskey="p" rel="previous">unskeletonize-map</a>, Up: <a href="#Skeletonization-Interface" accesskey="u" rel="up">Skeletonization Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmax_002dskeleton_002dsearch_002ddepth-1"></a>
<h4 class="subsection">11.35.3 set-max-skeleton-search-depth</h4>
<dl>
<dt><a name="index-set_002dmax_002dskeleton_002dsearch_002ddepth"></a>function: <strong>set-max-skeleton-search-depth</strong> <em>v</em></dt>
<dd>
<p>Where <em>v</em> is an integer number
</p>
<p>set the skeleton search depth, used in baton building 
</p>
<p>For high resolution maps, you need to search deeper down the skeleton tree. This limit needs to be increased to 20 or so for high res maps (it is 10 by default) 
</p>
</dd></dl>

<hr>
<a name="set_002dskeleton_002dbox_002dsize"></a>
<div class="header">
<p>
Previous: <a href="#set_002dmax_002dskeleton_002dsearch_002ddepth" accesskey="p" rel="previous">set-max-skeleton-search-depth</a>, Up: <a href="#Skeletonization-Interface" accesskey="u" rel="up">Skeletonization Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dskeleton_002dbox_002dsize-1"></a>
<h4 class="subsection">11.35.4 set-skeleton-box-size</h4>
<dl>
<dt><a name="index-set_002dskeleton_002dbox_002dsize"></a>function: <strong>set-skeleton-box-size</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>the box size (in Angstroms) for which the skeleton is displayed 
</p>
</dd></dl>

<hr>
<a name="Save-Coordinates"></a>
<div class="header">
<p>
Next: <a href="#Read-Phases-File-Functions" accesskey="n" rel="next">Read Phases File Functions</a>, Previous: <a href="#Skeletonization-Interface" accesskey="p" rel="previous">Skeletonization Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Save-Coordinates-2"></a>
<h3 class="section">11.36 Save Coordinates</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#save_002dcoordinates" accesskey="1">save-coordinates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory" accesskey="2">set-save-coordinates-in-original-directory</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="save_002dcoordinates"></a>
<div class="header">
<p>
Next: <a href="#set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory" accesskey="n" rel="next">set-save-coordinates-in-original-directory</a>, Up: <a href="#Save-Coordinates" accesskey="u" rel="up">Save Coordinates</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dcoordinates-1"></a>
<h4 class="subsection">11.36.1 save-coordinates</h4>
<dl>
<dt><a name="index-save_002dcoordinates"></a>function: <strong>save-coordinates</strong> <em>imol filename</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>filename</em> is a string
</li></ul>

<p>save coordinates of molecule number imol in filename 
</p>
<p>Returns: status 1 is good (success), 0 is fail. 
</p>
</dd></dl>

<hr>
<a name="set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory"></a>
<div class="header">
<p>
Previous: <a href="#save_002dcoordinates" accesskey="p" rel="previous">save-coordinates</a>, Up: <a href="#Save-Coordinates" accesskey="u" rel="up">Save Coordinates</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory-1"></a>
<h4 class="subsection">11.36.2 set-save-coordinates-in-original-directory</h4>
<dl>
<dt><a name="index-set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory"></a>function: <strong>set-save-coordinates-in-original-directory</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set save coordinates in the starting directory 
</p>
</dd></dl>

<hr>
<a name="Read-Phases-File-Functions"></a>
<div class="header">
<p>
Next: <a href="#Graphics-Move" accesskey="n" rel="next">Graphics Move</a>, Previous: <a href="#Save-Coordinates" accesskey="p" rel="previous">Save Coordinates</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Read-Phases-File-Functions-1"></a>
<h3 class="section">11.37 Read Phases File Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap" accesskey="1">read-phs-and-coords-and-make-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol" accesskey="2">read-phs-and-make-map-using-cell-symm-from-previous-mol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol" accesskey="3">read-phs-and-make-map-using-cell-symm-from-mol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm" accesskey="4">read-phs-and-make-map-using-cell-symm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits" accesskey="5">read-phs-and-make-map-with-reso-limits</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap"></a>
<div class="header">
<p>
Next: <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol" accesskey="n" rel="next">read-phs-and-make-map-using-cell-symm-from-previous-mol</a>, Up: <a href="#Read-Phases-File-Functions" accesskey="u" rel="up">Read Phases File Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap-1"></a>
<h4 class="subsection">11.37.1 read-phs-and-coords-and-make-map</h4>
<dl>
<dt><a name="index-read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap"></a>function: <strong>read-phs-and-coords-and-make-map</strong> <em>pdb_filename</em></dt>
<dd>
<p>Where <em>pdb_filename</em> is a string
</p>
<p>read phs file use coords to get cell and symm to make map 
</p>
<p>uses pending data to make the map. 
</p>
</dd></dl>

<hr>
<a name="read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol"></a>
<div class="header">
<p>
Next: <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol" accesskey="n" rel="next">read-phs-and-make-map-using-cell-symm-from-mol</a>, Previous: <a href="#read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap" accesskey="p" rel="previous">read-phs-and-coords-and-make-map</a>, Up: <a href="#Read-Phases-File-Functions" accesskey="u" rel="up">Read Phases File Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol-1"></a>
<h4 class="subsection">11.37.2 read-phs-and-make-map-using-cell-symm-from-previous-mol</h4>
<dl>
<dt><a name="index-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol"></a>function: <strong>read-phs-and-make-map-using-cell-symm-from-previous-mol</strong> <em>phs_filename</em></dt>
<dd>
<p>Where <em>phs_filename</em> is a string
</p>
<p>read a phs file, the cell and symm information is from previously read (most recently read) coordinates file 
</p>
<p>For use with phs data filename provided on the command line 
</p>
</dd></dl>

<hr>
<a name="read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol"></a>
<div class="header">
<p>
Next: <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm" accesskey="n" rel="next">read-phs-and-make-map-using-cell-symm</a>, Previous: <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol" accesskey="p" rel="previous">read-phs-and-make-map-using-cell-symm-from-previous-mol</a>, Up: <a href="#Read-Phases-File-Functions" accesskey="u" rel="up">Read Phases File Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol-1"></a>
<h4 class="subsection">11.37.3 read-phs-and-make-map-using-cell-symm-from-mol</h4>
<dl>
<dt><a name="index-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol"></a>function: <strong>read-phs-and-make-map-using-cell-symm-from-mol</strong> <em>phs_filename imol</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>phs_filename</em> is a string
</li><li> <em>imol</em> is an integer number
</li></ul>

<p>read phs file and use a previously read molecule to provide the cell and symmetry information 
</p>
<p>Returns: the new molecule number, return -1 if problem creating the map (e.g. not phs data, file not found etc). 
</p>
</dd></dl>

<hr>
<a name="read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm"></a>
<div class="header">
<p>
Next: <a href="#read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits" accesskey="n" rel="next">read-phs-and-make-map-with-reso-limits</a>, Previous: <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol" accesskey="p" rel="previous">read-phs-and-make-map-using-cell-symm-from-mol</a>, Up: <a href="#Read-Phases-File-Functions" accesskey="u" rel="up">Read Phases File Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm-1"></a>
<h4 class="subsection">11.37.4 read-phs-and-make-map-using-cell-symm</h4>
<dl>
<dt><a name="index-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm"></a>function: <strong>read-phs-and-make-map-using-cell-symm</strong> <em>phs_file_name hm_spacegroup a b c alpha beta gamma</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>phs_file_name</em> is a string
</li><li> <em>hm_spacegroup</em> is a string
</li><li> <em>a</em> is a number
</li><li> <em>b</em> is a number
</li><li> <em>c</em> is a number
</li><li> <em>alpha</em> is a number
</li><li> <em>beta</em> is a number
</li><li> <em>gamma</em> is a number
</li></ul>

<p>read phs file use coords to use cell and symm to make map 
</p>
<p>in degrees 
</p>
</dd></dl>

<hr>
<a name="read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits"></a>
<div class="header">
<p>
Previous: <a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm" accesskey="p" rel="previous">read-phs-and-make-map-using-cell-symm</a>, Up: <a href="#Read-Phases-File-Functions" accesskey="u" rel="up">Read Phases File Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits-1"></a>
<h4 class="subsection">11.37.5 read-phs-and-make-map-with-reso-limits</h4>
<dl>
<dt><a name="index-read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits"></a>function: <strong>read-phs-and-make-map-with-reso-limits</strong> <em>imol phs_file_name reso_lim_low reso_lim_high</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>phs_file_name</em> is a string
</li><li> <em>reso_lim_low</em> is a number
</li><li> <em>reso_lim_high</em> is a number
</li></ul>

<p>read a phs file and use the cell and symm in molecule number imol and use the resolution limits reso_lim_high (in Angstroems). 
</p>
</dd></dl>

<hr>
<a name="Graphics-Move"></a>
<div class="header">
<p>
Next: <a href="#Go-To-Atom-Widget-Functions" accesskey="n" rel="next">Go To Atom Widget Functions</a>, Previous: <a href="#Read-Phases-File-Functions" accesskey="p" rel="previous">Read Phases File Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Graphics-Move-1"></a>
<h3 class="section">11.38 Graphics Move</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#undo_002dlast_002dmove" accesskey="1">undo-last-move</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#translate_002dmolecule_002dby" accesskey="2">translate-molecule-by</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#transform_002dmolecule_002dby" accesskey="3">transform-molecule-by</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#transform_002dzone" accesskey="4">transform-zone</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="undo_002dlast_002dmove"></a>
<div class="header">
<p>
Next: <a href="#translate_002dmolecule_002dby" accesskey="n" rel="next">translate-molecule-by</a>, Up: <a href="#Graphics-Move" accesskey="u" rel="up">Graphics Move</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="undo_002dlast_002dmove-1"></a>
<h4 class="subsection">11.38.1 undo-last-move</h4>
<dl>
<dt><a name="index-undo_002dlast_002dmove"></a>function: <strong>undo-last-move</strong></dt>
<dd>
<p>undo last move 
</p>
</dd></dl>

<hr>
<a name="translate_002dmolecule_002dby"></a>
<div class="header">
<p>
Next: <a href="#transform_002dmolecule_002dby" accesskey="n" rel="next">transform-molecule-by</a>, Previous: <a href="#undo_002dlast_002dmove" accesskey="p" rel="previous">undo-last-move</a>, Up: <a href="#Graphics-Move" accesskey="u" rel="up">Graphics Move</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="translate_002dmolecule_002dby-1"></a>
<h4 class="subsection">11.38.2 translate-molecule-by</h4>
<dl>
<dt><a name="index-translate_002dmolecule_002dby"></a>function: <strong>translate-molecule-by</strong> <em>imol x y z</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li></ul>

<p>translate molecule number imol by (x,y,z) in Angstroms 
</p>
</dd></dl>

<hr>
<a name="transform_002dmolecule_002dby"></a>
<div class="header">
<p>
Next: <a href="#transform_002dzone" accesskey="n" rel="next">transform-zone</a>, Previous: <a href="#translate_002dmolecule_002dby" accesskey="p" rel="previous">translate-molecule-by</a>, Up: <a href="#Graphics-Move" accesskey="u" rel="up">Graphics Move</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="transform_002dmolecule_002dby-1"></a>
<h4 class="subsection">11.38.3 transform-molecule-by</h4>
<dl>
<dt><a name="index-transform_002dmolecule_002dby"></a>function: <strong>transform-molecule-by</strong> <em>imol m11 m12 m13 m21 m22 m23 m31 m32 m33 x y z</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>m11</em> is a number
</li><li> <em>m12</em> is a number
</li><li> <em>m13</em> is a number
</li><li> <em>m21</em> is a number
</li><li> <em>m22</em> is a number
</li><li> <em>m23</em> is a number
</li><li> <em>m31</em> is a number
</li><li> <em>m32</em> is a number
</li><li> <em>m33</em> is a number
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li></ul>

<p>transform molecule number imol by the given rotation matrix, then translate by (x,y,z) in Angstroms 
</p>
</dd></dl>

<hr>
<a name="transform_002dzone"></a>
<div class="header">
<p>
Previous: <a href="#transform_002dmolecule_002dby" accesskey="p" rel="previous">transform-molecule-by</a>, Up: <a href="#Graphics-Move" accesskey="u" rel="up">Graphics Move</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="transform_002dzone-1"></a>
<h4 class="subsection">11.38.4 transform-zone</h4>
<dl>
<dt><a name="index-transform_002dzone"></a>function: <strong>transform-zone</strong> <em>imol chain_id resno_start resno_end ins_code m11 m12 m13 m21 m22 m23 m31 m32 m33 x y z</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno_start</em> is an integer number
</li><li> <em>resno_end</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>m11</em> is a number
</li><li> <em>m12</em> is a number
</li><li> <em>m13</em> is a number
</li><li> <em>m21</em> is a number
</li><li> <em>m22</em> is a number
</li><li> <em>m23</em> is a number
</li><li> <em>m31</em> is a number
</li><li> <em>m32</em> is a number
</li><li> <em>m33</em> is a number
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li></ul>

<p>transform fragment of molecule number imol by the given rotation matrix, then translate by (x,y,z) in Angstroms 
</p>
</dd></dl>

<hr>
<a name="Go-To-Atom-Widget-Functions"></a>
<div class="header">
<p>
Next: <a href="#Map-and-Molecule-Control" accesskey="n" rel="next">Map and Molecule Control</a>, Previous: <a href="#Graphics-Move" accesskey="p" rel="previous">Graphics Move</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Go-To-Atom-Widget-Functions-1"></a>
<h3 class="section">11.39 Go To Atom Widget Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#post_002dgo_002dto_002datom_002dwindow" accesskey="1">post-go-to-atom-window</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#go_002dto_002datom_002dmolecule_002dnumber" accesskey="2">go-to-atom-molecule-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#go_002dto_002datom_002dchain_002did" accesskey="3">go-to-atom-chain-id</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#go_002dto_002datom_002datom_002dname" accesskey="4">go-to-atom-atom-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#go_002dto_002datom_002dresidue_002dnumber" accesskey="5">go-to-atom-residue-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#go_002dto_002datom_002dins_002dcode" accesskey="6">go-to-atom-ins-code</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#go_002dto_002datom_002dalt_002dconf" accesskey="7">go-to-atom-alt-conf</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname" accesskey="8">set-go-to-atom-chain-residue-atom-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull" accesskey="9">set-go-to-atom-chain-residue-atom-name-full</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw">set-go-to-atom-chain-residue-atom-name-no-redraw</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition">update-go-to-atom-from-current-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#atom_002dspec_002dto_002datom_002dindex">atom-spec-to-atom-index</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#full_002datom_002dspec_002dto_002datom_002dindex">full-atom-spec-to-atom-index</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol">update-go-to-atom-window-on-changed-mol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol">update-go-to-atom-window-on-new-mol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dgo_002dto_002datom_002dmolecule">set-go-to-atom-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="post_002dgo_002dto_002datom_002dwindow"></a>
<div class="header">
<p>
Next: <a href="#go_002dto_002datom_002dmolecule_002dnumber" accesskey="n" rel="next">go-to-atom-molecule-number</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="post_002dgo_002dto_002datom_002dwindow-1"></a>
<h4 class="subsection">11.39.1 post-go-to-atom-window</h4>
<dl>
<dt><a name="index-post_002dgo_002dto_002datom_002dwindow"></a>function: <strong>post-go-to-atom-window</strong></dt>
<dd>
<p>Post the Go To Atom Window. 
</p>
</dd></dl>

<hr>
<a name="go_002dto_002datom_002dmolecule_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#go_002dto_002datom_002dchain_002did" accesskey="n" rel="next">go-to-atom-chain-id</a>, Previous: <a href="#post_002dgo_002dto_002datom_002dwindow" accesskey="p" rel="previous">post-go-to-atom-window</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="go_002dto_002datom_002dmolecule_002dnumber-1"></a>
<h4 class="subsection">11.39.2 go-to-atom-molecule-number</h4>
<dl>
<dt><a name="index-go_002dto_002datom_002dmolecule_002dnumber"></a>function: <strong>go-to-atom-molecule-number</strong></dt>
<dd>
<p>the go-to-atom molecule number 
</p>
</dd></dl>

<hr>
<a name="go_002dto_002datom_002dchain_002did"></a>
<div class="header">
<p>
Next: <a href="#go_002dto_002datom_002datom_002dname" accesskey="n" rel="next">go-to-atom-atom-name</a>, Previous: <a href="#go_002dto_002datom_002dmolecule_002dnumber" accesskey="p" rel="previous">go-to-atom-molecule-number</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="go_002dto_002datom_002dchain_002did-1"></a>
<h4 class="subsection">11.39.3 go-to-atom-chain-id</h4>
<dl>
<dt><a name="index-go_002dto_002datom_002dchain_002did"></a>function: <strong>go-to-atom-chain-id</strong></dt>
<dd>
<p>the go-to-atom chain-id 
</p>
</dd></dl>

<hr>
<a name="go_002dto_002datom_002datom_002dname"></a>
<div class="header">
<p>
Next: <a href="#go_002dto_002datom_002dresidue_002dnumber" accesskey="n" rel="next">go-to-atom-residue-number</a>, Previous: <a href="#go_002dto_002datom_002dchain_002did" accesskey="p" rel="previous">go-to-atom-chain-id</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="go_002dto_002datom_002datom_002dname-1"></a>
<h4 class="subsection">11.39.4 go-to-atom-atom-name</h4>
<dl>
<dt><a name="index-go_002dto_002datom_002datom_002dname"></a>function: <strong>go-to-atom-atom-name</strong></dt>
<dd>
<p>the go-to-atom atom name 
</p>
</dd></dl>

<hr>
<a name="go_002dto_002datom_002dresidue_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#go_002dto_002datom_002dins_002dcode" accesskey="n" rel="next">go-to-atom-ins-code</a>, Previous: <a href="#go_002dto_002datom_002datom_002dname" accesskey="p" rel="previous">go-to-atom-atom-name</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="go_002dto_002datom_002dresidue_002dnumber-1"></a>
<h4 class="subsection">11.39.5 go-to-atom-residue-number</h4>
<dl>
<dt><a name="index-go_002dto_002datom_002dresidue_002dnumber"></a>function: <strong>go-to-atom-residue-number</strong></dt>
<dd>
<p>the go-to-atom residue number 
</p>
</dd></dl>

<hr>
<a name="go_002dto_002datom_002dins_002dcode"></a>
<div class="header">
<p>
Next: <a href="#go_002dto_002datom_002dalt_002dconf" accesskey="n" rel="next">go-to-atom-alt-conf</a>, Previous: <a href="#go_002dto_002datom_002dresidue_002dnumber" accesskey="p" rel="previous">go-to-atom-residue-number</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="go_002dto_002datom_002dins_002dcode-1"></a>
<h4 class="subsection">11.39.6 go-to-atom-ins-code</h4>
<dl>
<dt><a name="index-go_002dto_002datom_002dins_002dcode"></a>function: <strong>go-to-atom-ins-code</strong></dt>
<dd>
<p>the go-to-atom insertion code 
</p>
</dd></dl>

<hr>
<a name="go_002dto_002datom_002dalt_002dconf"></a>
<div class="header">
<p>
Next: <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname" accesskey="n" rel="next">set-go-to-atom-chain-residue-atom-name</a>, Previous: <a href="#go_002dto_002datom_002dins_002dcode" accesskey="p" rel="previous">go-to-atom-ins-code</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="go_002dto_002datom_002dalt_002dconf-1"></a>
<h4 class="subsection">11.39.7 go-to-atom-alt-conf</h4>
<dl>
<dt><a name="index-go_002dto_002datom_002dalt_002dconf"></a>function: <strong>go-to-atom-alt-conf</strong></dt>
<dd>
<p>the go-to-atom alt conf 
</p>
</dd></dl>

<hr>
<a name="set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname"></a>
<div class="header">
<p>
Next: <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull" accesskey="n" rel="next">set-go-to-atom-chain-residue-atom-name-full</a>, Previous: <a href="#go_002dto_002datom_002dalt_002dconf" accesskey="p" rel="previous">go-to-atom-alt-conf</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname-1"></a>
<h4 class="subsection">11.39.8 set-go-to-atom-chain-residue-atom-name</h4>
<dl>
<dt><a name="index-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname"></a>function: <strong>set-go-to-atom-chain-residue-atom-name</strong> <em>t1_chain_id iresno t3_atom_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>t1_chain_id</em> is a string
</li><li> <em>iresno</em> is an integer number
</li><li> <em>t3_atom_name</em> is a string
</li></ul>

<p>set the go to atom specification 
</p>
<p>It seems important for swig that the char * arguments are const char *, not const gchar * (or else we get wrong type of argument error on (say) &quot;A&quot;
</p>
<p>Returns: the success status of the go to. 0 for fail, 1 for success. 
</p>
</dd></dl>

<hr>
<a name="set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull"></a>
<div class="header">
<p>
Next: <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw" accesskey="n" rel="next">set-go-to-atom-chain-residue-atom-name-no-redraw</a>, Previous: <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname" accesskey="p" rel="previous">set-go-to-atom-chain-residue-atom-name</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull-1"></a>
<h4 class="subsection">11.39.9 set-go-to-atom-chain-residue-atom-name-full</h4>
<dl>
<dt><a name="index-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull"></a>function: <strong>set-go-to-atom-chain-residue-atom-name-full</strong> <em>chain_id resno ins_code atom_name alt_conf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>atom_name</em> is a string
</li><li> <em>alt_conf</em> is a string
</li></ul>

<p>set the go to (full) atom specification 
</p>
<p>It seems important for swig that the char * arguments are const char *, not const gchar * (or else we get wrong type of argument error on (say) &quot;A&quot;
</p>
<p>Returns: the success status of the go to. 0 for fail, 1 for success. 
</p>
</dd></dl>

<hr>
<a name="set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw"></a>
<div class="header">
<p>
Next: <a href="#update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition" accesskey="n" rel="next">update-go-to-atom-from-current-position</a>, Previous: <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull" accesskey="p" rel="previous">set-go-to-atom-chain-residue-atom-name-full</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw-1"></a>
<h4 class="subsection">11.39.10 set-go-to-atom-chain-residue-atom-name-no-redraw</h4>
<dl>
<dt><a name="index-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw"></a>function: <strong>set-go-to-atom-chain-residue-atom-name-no-redraw</strong> <em>t1 iresno t3 make_the_move_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>t1</em> is a string
</li><li> <em>iresno</em> is an integer number
</li><li> <em>t3</em> is a string
</li><li> <em>make_the_move_flag</em> is an integer number
</li></ul>

<p>set go to atom but don&rsquo;t redraw 
</p>
</dd></dl>

<hr>
<a name="update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition"></a>
<div class="header">
<p>
Next: <a href="#atom_002dspec_002dto_002datom_002dindex" accesskey="n" rel="next">atom-spec-to-atom-index</a>, Previous: <a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw" accesskey="p" rel="previous">set-go-to-atom-chain-residue-atom-name-no-redraw</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition-1"></a>
<h4 class="subsection">11.39.11 update-go-to-atom-from-current-position</h4>
<dl>
<dt><a name="index-update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition"></a>function: <strong>update-go-to-atom-from-current-position</strong></dt>
<dd>
<p>update the Go To Atom widget entries to atom closest to screen centre. 
</p>
</dd></dl>

<hr>
<a name="atom_002dspec_002dto_002datom_002dindex"></a>
<div class="header">
<p>
Next: <a href="#full_002datom_002dspec_002dto_002datom_002dindex" accesskey="n" rel="next">full-atom-spec-to-atom-index</a>, Previous: <a href="#update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition" accesskey="p" rel="previous">update-go-to-atom-from-current-position</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="atom_002dspec_002dto_002datom_002dindex-1"></a>
<h4 class="subsection">11.39.12 atom-spec-to-atom-index</h4>
<dl>
<dt><a name="index-atom_002dspec_002dto_002datom_002dindex"></a>function: <strong>atom-spec-to-atom-index</strong> <em>mol chain resno atom_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>mol</em> is an integer number
</li><li> <em>chain</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>atom_name</em> is a string
</li></ul>

<p>what is the atom index of the given atom? 
</p>
</dd></dl>

<hr>
<a name="full_002datom_002dspec_002dto_002datom_002dindex"></a>
<div class="header">
<p>
Next: <a href="#update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol" accesskey="n" rel="next">update-go-to-atom-window-on-changed-mol</a>, Previous: <a href="#atom_002dspec_002dto_002datom_002dindex" accesskey="p" rel="previous">atom-spec-to-atom-index</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="full_002datom_002dspec_002dto_002datom_002dindex-1"></a>
<h4 class="subsection">11.39.13 full-atom-spec-to-atom-index</h4>
<dl>
<dt><a name="index-full_002datom_002dspec_002dto_002datom_002dindex"></a>function: <strong>full-atom-spec-to-atom-index</strong> <em>imol chain resno inscode atom_name altloc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>inscode</em> is a string
</li><li> <em>atom_name</em> is a string
</li><li> <em>altloc</em> is a string
</li></ul>

<p>what is the atom index of the given atom? 
</p>
</dd></dl>

<hr>
<a name="update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol"></a>
<div class="header">
<p>
Next: <a href="#update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol" accesskey="n" rel="next">update-go-to-atom-window-on-new-mol</a>, Previous: <a href="#full_002datom_002dspec_002dto_002datom_002dindex" accesskey="p" rel="previous">full-atom-spec-to-atom-index</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol-1"></a>
<h4 class="subsection">11.39.14 update-go-to-atom-window-on-changed-mol</h4>
<dl>
<dt><a name="index-update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol"></a>function: <strong>update-go-to-atom-window-on-changed-mol</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>update the Go To Atom window 
</p>
</dd></dl>

<hr>
<a name="update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol"></a>
<div class="header">
<p>
Next: <a href="#set_002dgo_002dto_002datom_002dmolecule" accesskey="n" rel="next">set-go-to-atom-molecule</a>, Previous: <a href="#update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol" accesskey="p" rel="previous">update-go-to-atom-window-on-changed-mol</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol-1"></a>
<h4 class="subsection">11.39.15 update-go-to-atom-window-on-new-mol</h4>
<dl>
<dt><a name="index-update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol"></a>function: <strong>update-go-to-atom-window-on-new-mol</strong></dt>
<dd>
<p>update the Go To Atom window. This updates the option menu for the molecules. 
</p>
</dd></dl>

<hr>
<a name="set_002dgo_002dto_002datom_002dmolecule"></a>
<div class="header">
<p>
Previous: <a href="#update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol" accesskey="p" rel="previous">update-go-to-atom-window-on-new-mol</a>, Up: <a href="#Go-To-Atom-Widget-Functions" accesskey="u" rel="up">Go To Atom Widget Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgo_002dto_002datom_002dmolecule-1"></a>
<h4 class="subsection">11.39.16 set-go-to-atom-molecule</h4>
<dl>
<dt><a name="index-set_002dgo_002dto_002datom_002dmolecule"></a>function: <strong>set-go-to-atom-molecule</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>set the molecule for the Go To Atom 
</p>
<p>For dynarama callback sake. The widget/class knows which molecule that it was generated from, so in order to go to the molecule from dynarama, we first need to the the molecule - because 
</p>
<p>does not mention the molecule (see &quot;Next/Previous Residue&quot; for reasons for that). This function simply calls the graphics_info_t function of the same name.
</p>
<p>Also used in scripting, where go-to-atom-chain-residue-atom-name does not mention the molecule number.
</p>
<p>20090914-PE set-go-to-atom-molecule can be used in a script and it should change the go-to-atom-molecule in the Go To Atom dialog (if it is being displayed). This does mean, of course that using the ramachandran plot to centre on atoms will change the Go To Atom dialog. Maybe that is surprising (maybe not). 
</p>
</dd></dl>

<hr>
<a name="Map-and-Molecule-Control"></a>
<div class="header">
<p>
Next: <a href="#Align-and-Mutate" accesskey="n" rel="next">Align and Mutate</a>, Previous: <a href="#Go-To-Atom-Widget-Functions" accesskey="p" rel="previous">Go To Atom Widget Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-and-Molecule-Control-1"></a>
<h3 class="section">11.40 Map and Molecule Control</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#post_002ddisplay_002dcontrol_002dwindow" accesskey="1">post-display-control-window</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002ddisplayed" accesskey="2">set-map-displayed</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmol_002ddisplayed" accesskey="3">set-mol-displayed</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddisplay_002donly_002dmodel_002dmol" accesskey="4">set-display-only-model-mol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmol_002dactive" accesskey="5">set-mol-active</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#display_002dmaps_002dscm" accesskey="6">display-maps-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mol_002dis_002ddisplayed" accesskey="7">mol-is-displayed</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mol_002dis_002dactive" accesskey="8">mol-is-active</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dis_002ddisplayed" accesskey="9">map-is-displayed</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dall_002dmaps_002ddisplayed">set-all-maps-displayed</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dall_002dmodels_002ddisplayed_002dand_002dactive">set-all-models-displayed-and-active</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed">set-only-last-model-molecule-displayed</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#display_002donly_002dactive">display-only-active</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#space_002dgroup_002dscm">space-group-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dspacegroup">show-spacegroup</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#symmetry_002doperators_002dscm">symmetry-operators-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#symmetry_002doperators_002dpy">symmetry-operators-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="post_002ddisplay_002dcontrol_002dwindow"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002ddisplayed" accesskey="n" rel="next">set-map-displayed</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="post_002ddisplay_002dcontrol_002dwindow-1"></a>
<h4 class="subsection">11.40.1 post-display-control-window</h4>
<dl>
<dt><a name="index-post_002ddisplay_002dcontrol_002dwindow"></a>function: <strong>post-display-control-window</strong></dt>
<dd>
<p>display the Display Constrol window 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002ddisplayed"></a>
<div class="header">
<p>
Next: <a href="#set_002dmol_002ddisplayed" accesskey="n" rel="next">set-mol-displayed</a>, Previous: <a href="#post_002ddisplay_002dcontrol_002dwindow" accesskey="p" rel="previous">post-display-control-window</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002ddisplayed-1"></a>
<h4 class="subsection">11.40.2 set-map-displayed</h4>
<dl>
<dt><a name="index-set_002dmap_002ddisplayed"></a>function: <strong>set-map-displayed</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>make the map displayed/undisplayed, 0 for off, 1 for on 
</p>
</dd></dl>

<hr>
<a name="set_002dmol_002ddisplayed"></a>
<div class="header">
<p>
Next: <a href="#set_002ddisplay_002donly_002dmodel_002dmol" accesskey="n" rel="next">set-display-only-model-mol</a>, Previous: <a href="#set_002dmap_002ddisplayed" accesskey="p" rel="previous">set-map-displayed</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmol_002ddisplayed-1"></a>
<h4 class="subsection">11.40.3 set-mol-displayed</h4>
<dl>
<dt><a name="index-set_002dmol_002ddisplayed"></a>function: <strong>set-mol-displayed</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>make the coordinates molecule displayed/undisplayed, 0 for off, 1 for on 
</p>
</dd></dl>

<hr>
<a name="set_002ddisplay_002donly_002dmodel_002dmol"></a>
<div class="header">
<p>
Next: <a href="#set_002dmol_002dactive" accesskey="n" rel="next">set-mol-active</a>, Previous: <a href="#set_002dmol_002ddisplayed" accesskey="p" rel="previous">set-mol-displayed</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddisplay_002donly_002dmodel_002dmol-1"></a>
<h4 class="subsection">11.40.4 set-display-only-model-mol</h4>
<dl>
<dt><a name="index-set_002ddisplay_002donly_002dmodel_002dmol"></a>function: <strong>set-display-only-model-mol</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>from all the model molecules, display only imol 
</p>
<p>This stops flashing/delayed animations with many molecules 
</p>
</dd></dl>

<hr>
<a name="set_002dmol_002dactive"></a>
<div class="header">
<p>
Next: <a href="#display_002dmaps_002dscm" accesskey="n" rel="next">display-maps-scm</a>, Previous: <a href="#set_002ddisplay_002donly_002dmodel_002dmol" accesskey="p" rel="previous">set-display-only-model-mol</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmol_002dactive-1"></a>
<h4 class="subsection">11.40.5 set-mol-active</h4>
<dl>
<dt><a name="index-set_002dmol_002dactive"></a>function: <strong>set-mol-active</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>make the coordinates molecule active/inactve (clickable), 0 for off, 1 for on 
</p>
</dd></dl>

<hr>
<a name="display_002dmaps_002dscm"></a>
<div class="header">
<p>
Next: <a href="#mol_002dis_002ddisplayed" accesskey="n" rel="next">mol-is-displayed</a>, Previous: <a href="#set_002dmol_002dactive" accesskey="p" rel="previous">set-mol-active</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="display_002dmaps_002dscm-1"></a>
<h4 class="subsection">11.40.6 display-maps-scm</h4>
<dl>
<dt><a name="index-display_002dmaps_002dscm"></a>function: <strong>display-maps-scm</strong> <em>maps_list</em></dt>
<dd>
<p>Where <em>maps_list</em> is a SCM
</p>
<p>maps_list is a list of integers (map molecule numbers). 
</p>
<p>This interface is uses so that we don&rsquo;t get flashing when a map is turned off (using set_mol_displayed). 
</p>
</dd></dl>

<hr>
<a name="mol_002dis_002ddisplayed"></a>
<div class="header">
<p>
Next: <a href="#mol_002dis_002dactive" accesskey="n" rel="next">mol-is-active</a>, Previous: <a href="#display_002dmaps_002dscm" accesskey="p" rel="previous">display-maps-scm</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mol_002dis_002ddisplayed-1"></a>
<h4 class="subsection">11.40.7 mol-is-displayed</h4>
<dl>
<dt><a name="index-mol_002dis_002ddisplayed"></a>function: <strong>mol-is-displayed</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the display state of molecule number imol 
</p>
<p>Returns: 1 for on, 0 for off 
</p>
</dd></dl>

<hr>
<a name="mol_002dis_002dactive"></a>
<div class="header">
<p>
Next: <a href="#map_002dis_002ddisplayed" accesskey="n" rel="next">map-is-displayed</a>, Previous: <a href="#mol_002dis_002ddisplayed" accesskey="p" rel="previous">mol-is-displayed</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mol_002dis_002dactive-1"></a>
<h4 class="subsection">11.40.8 mol-is-active</h4>
<dl>
<dt><a name="index-mol_002dis_002dactive"></a>function: <strong>mol-is-active</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the active state of molecule number imol 
</p>
<p>Returns: 1 for on, 0 for off 
</p>
</dd></dl>

<hr>
<a name="map_002dis_002ddisplayed"></a>
<div class="header">
<p>
Next: <a href="#set_002dall_002dmaps_002ddisplayed" accesskey="n" rel="next">set-all-maps-displayed</a>, Previous: <a href="#mol_002dis_002dactive" accesskey="p" rel="previous">mol-is-active</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dis_002ddisplayed-1"></a>
<h4 class="subsection">11.40.9 map-is-displayed</h4>
<dl>
<dt><a name="index-map_002dis_002ddisplayed"></a>function: <strong>map-is-displayed</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the display state of molecule number imol 
</p>
<p>Returns: 1 for on, 0 for off 
</p>
</dd></dl>

<hr>
<a name="set_002dall_002dmaps_002ddisplayed"></a>
<div class="header">
<p>
Next: <a href="#set_002dall_002dmodels_002ddisplayed_002dand_002dactive" accesskey="n" rel="next">set-all-models-displayed-and-active</a>, Previous: <a href="#map_002dis_002ddisplayed" accesskey="p" rel="previous">map-is-displayed</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dall_002dmaps_002ddisplayed-1"></a>
<h4 class="subsection">11.40.10 set-all-maps-displayed</h4>
<dl>
<dt><a name="index-set_002dall_002dmaps_002ddisplayed"></a>function: <strong>set-all-maps-displayed</strong> <em>on_or_off</em></dt>
<dd>
<p>Where <em>on_or_off</em> is an integer number
</p>
<p>if on_or_off is 0 turn off all maps displayed, for other values of on_or_off turn on all maps 
</p>
</dd></dl>

<hr>
<a name="set_002dall_002dmodels_002ddisplayed_002dand_002dactive"></a>
<div class="header">
<p>
Next: <a href="#set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed" accesskey="n" rel="next">set-only-last-model-molecule-displayed</a>, Previous: <a href="#set_002dall_002dmaps_002ddisplayed" accesskey="p" rel="previous">set-all-maps-displayed</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dall_002dmodels_002ddisplayed_002dand_002dactive-1"></a>
<h4 class="subsection">11.40.11 set-all-models-displayed-and-active</h4>
<dl>
<dt><a name="index-set_002dall_002dmodels_002ddisplayed_002dand_002dactive"></a>function: <strong>set-all-models-displayed-and-active</strong> <em>on_or_off</em></dt>
<dd>
<p>Where <em>on_or_off</em> is an integer number
</p>
<p>if on_or_off is 0 turn off all models displayed and active, for other values of on_or_off turn on all models. 
</p>
</dd></dl>

<hr>
<a name="set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed"></a>
<div class="header">
<p>
Next: <a href="#display_002donly_002dactive" accesskey="n" rel="next">display-only-active</a>, Previous: <a href="#set_002dall_002dmodels_002ddisplayed_002dand_002dactive" accesskey="p" rel="previous">set-all-models-displayed-and-active</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed-1"></a>
<h4 class="subsection">11.40.12 set-only-last-model-molecule-displayed</h4>
<dl>
<dt><a name="index-set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed"></a>function: <strong>set-only-last-model-molecule-displayed</strong></dt>
<dd>
<p>only display the last model molecule 
</p>
</dd></dl>

<hr>
<a name="display_002donly_002dactive"></a>
<div class="header">
<p>
Next: <a href="#space_002dgroup_002dscm" accesskey="n" rel="next">space-group-scm</a>, Previous: <a href="#set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed" accesskey="p" rel="previous">set-only-last-model-molecule-displayed</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="display_002donly_002dactive-1"></a>
<h4 class="subsection">11.40.13 display-only-active</h4>
<dl>
<dt><a name="index-display_002donly_002dactive"></a>function: <strong>display-only-active</strong></dt>
<dd>
<p>display only the active mol and the refinement map 
</p>
</dd></dl>

<hr>
<a name="space_002dgroup_002dscm"></a>
<div class="header">
<p>
Next: <a href="#show_002dspacegroup" accesskey="n" rel="next">show-spacegroup</a>, Previous: <a href="#display_002donly_002dactive" accesskey="p" rel="previous">display-only-active</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="space_002dgroup_002dscm-1"></a>
<h4 class="subsection">11.40.14 space-group-scm</h4>
<dl>
<dt><a name="index-space_002dgroup_002dscm"></a>function: <strong>space-group-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the spacegroup as a string, return scheme false if unable to do so. 
</p>
</dd></dl>

<hr>
<a name="show_002dspacegroup"></a>
<div class="header">
<p>
Next: <a href="#symmetry_002doperators_002dscm" accesskey="n" rel="next">symmetry-operators-scm</a>, Previous: <a href="#space_002dgroup_002dscm" accesskey="p" rel="previous">space-group-scm</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dspacegroup-1"></a>
<h4 class="subsection">11.40.15 show-spacegroup</h4>
<dl>
<dt><a name="index-show_002dspacegroup"></a>function: <strong>show-spacegroup</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the spacegroup of molecule number imol . Deprecated. 
</p>
<p>Returns: &quot;No Spacegroup&quot; when the spacegroup of a molecule has not been set. 
</p>
</dd></dl>

<hr>
<a name="symmetry_002doperators_002dscm"></a>
<div class="header">
<p>
Next: <a href="#symmetry_002doperators_002dpy" accesskey="n" rel="next">symmetry-operators-py</a>, Previous: <a href="#show_002dspacegroup" accesskey="p" rel="previous">show-spacegroup</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="symmetry_002doperators_002dscm-1"></a>
<h4 class="subsection">11.40.16 symmetry-operators-scm</h4>
<dl>
<dt><a name="index-symmetry_002doperators_002dscm"></a>function: <strong>symmetry-operators-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return a list of symmetry operators as strings - or scheme false if that is not possible. 
</p>
</dd></dl>

<hr>
<a name="symmetry_002doperators_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#symmetry_002doperators_002dscm" accesskey="p" rel="previous">symmetry-operators-scm</a>, Up: <a href="#Map-and-Molecule-Control" accesskey="u" rel="up">Map and Molecule Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="symmetry_002doperators_002dpy-1"></a>
<h4 class="subsection">11.40.17 symmetry-operators-py</h4>
<dl>
<dt><a name="index-symmetry_002doperators_002dpy"></a>function: <strong>symmetry-operators-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return a list of symmetry operators as strings - or Python False if that is not possible. 
</p>
</dd></dl>

<hr>
<a name="Align-and-Mutate"></a>
<div class="header">
<p>
Next: <a href="#Renumber-Residue-Range" accesskey="n" rel="next">Renumber Residue Range</a>, Previous: <a href="#Map-and-Molecule-Control" accesskey="p" rel="previous">Map and Molecule Control</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Align-and-Mutate-1"></a>
<h3 class="section">11.41 Align and Mutate</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#align_002dand_002dmutate" accesskey="1">align-and-mutate</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dalignment_002dgap_002dand_002dspace_002dpenalty" accesskey="2">set-alignment-gap-and-space-penalty</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#nearest_002dresidue_002dby_002dsequence_002dscm" accesskey="3">nearest-residue-by-sequence-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#nearest_002dresidue_002dby_002dsequence_002dpy" accesskey="4">nearest-residue-by-sequence-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="align_002dand_002dmutate"></a>
<div class="header">
<p>
Next: <a href="#set_002dalignment_002dgap_002dand_002dspace_002dpenalty" accesskey="n" rel="next">set-alignment-gap-and-space-penalty</a>, Up: <a href="#Align-and-Mutate" accesskey="u" rel="up">Align and Mutate</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="align_002dand_002dmutate-1"></a>
<h4 class="subsection">11.41.1 align-and-mutate</h4>
<dl>
<dt><a name="index-align_002dand_002dmutate"></a>function: <strong>align-and-mutate</strong> <em>imol chain_id fasta_maybe renumber_residues_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>fasta_maybe</em> is a string
</li><li> <em>renumber_residues_flag</em> is an integer number
</li></ul>

<p>align and mutate the given chain to the given sequence 
</p>
</dd></dl>

<hr>
<a name="set_002dalignment_002dgap_002dand_002dspace_002dpenalty"></a>
<div class="header">
<p>
Next: <a href="#nearest_002dresidue_002dby_002dsequence_002dscm" accesskey="n" rel="next">nearest-residue-by-sequence-scm</a>, Previous: <a href="#align_002dand_002dmutate" accesskey="p" rel="previous">align-and-mutate</a>, Up: <a href="#Align-and-Mutate" accesskey="u" rel="up">Align and Mutate</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dalignment_002dgap_002dand_002dspace_002dpenalty-1"></a>
<h4 class="subsection">11.41.2 set-alignment-gap-and-space-penalty</h4>
<dl>
<dt><a name="index-set_002dalignment_002dgap_002dand_002dspace_002dpenalty"></a>function: <strong>set-alignment-gap-and-space-penalty</strong> <em>wgap wspace</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>wgap</em> is a number
</li><li> <em>wspace</em> is a number
</li></ul>

<p>set the penalty for affine gap and space when aligning, defaults -3.0 and -0.4 
</p>
</dd></dl>

<hr>
<a name="nearest_002dresidue_002dby_002dsequence_002dscm"></a>
<div class="header">
<p>
Next: <a href="#nearest_002dresidue_002dby_002dsequence_002dpy" accesskey="n" rel="next">nearest-residue-by-sequence-py</a>, Previous: <a href="#set_002dalignment_002dgap_002dand_002dspace_002dpenalty" accesskey="p" rel="previous">set-alignment-gap-and-space-penalty</a>, Up: <a href="#Align-and-Mutate" accesskey="u" rel="up">Align and Mutate</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="nearest_002dresidue_002dby_002dsequence_002dscm-1"></a>
<h4 class="subsection">11.41.3 nearest-residue-by-sequence-scm</h4>
<dl>
<dt><a name="index-nearest_002dresidue_002dby_002dsequence_002dscm"></a>function: <strong>nearest-residue-by-sequence-scm</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>return the residue spec of the nearest residue by sequence numbering. 
</p>
<p>Returns: scheme false if not possible 
</p>
</dd></dl>

<hr>
<a name="nearest_002dresidue_002dby_002dsequence_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#nearest_002dresidue_002dby_002dsequence_002dscm" accesskey="p" rel="previous">nearest-residue-by-sequence-scm</a>, Up: <a href="#Align-and-Mutate" accesskey="u" rel="up">Align and Mutate</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="nearest_002dresidue_002dby_002dsequence_002dpy-1"></a>
<h4 class="subsection">11.41.4 nearest-residue-by-sequence-py</h4>
<dl>
<dt><a name="index-nearest_002dresidue_002dby_002dsequence_002dpy"></a>function: <strong>nearest-residue-by-sequence-py</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>return the residue spec of the nearest residue by sequence numbering. Return Python False if not possible 
</p>
</dd></dl>

<hr>
<a name="Renumber-Residue-Range"></a>
<div class="header">
<p>
Next: <a href="#Scripting-Interface" accesskey="n" rel="next">Scripting Interface</a>, Previous: <a href="#Align-and-Mutate" accesskey="p" rel="previous">Align and Mutate</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Renumber-Residue-Range-1"></a>
<h3 class="section">11.42 Renumber Residue Range</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#renumber_002dresidue_002drange" accesskey="1">renumber-residue-range</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#change_002dresidue_002dnumber" accesskey="2">change-residue-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="renumber_002dresidue_002drange"></a>
<div class="header">
<p>
Next: <a href="#change_002dresidue_002dnumber" accesskey="n" rel="next">change-residue-number</a>, Up: <a href="#Renumber-Residue-Range" accesskey="u" rel="up">Renumber Residue Range</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="renumber_002dresidue_002drange-1"></a>
<h4 class="subsection">11.42.1 renumber-residue-range</h4>
<dl>
<dt><a name="index-renumber_002dresidue_002drange"></a>function: <strong>renumber-residue-range</strong> <em>imol chain_id start_res last_res offset</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>start_res</em> is an integer number
</li><li> <em>last_res</em> is an integer number
</li><li> <em>offset</em> is an integer number
</li></ul>

<p>renumber the given residue range by offset residues 
</p>
</dd></dl>

<hr>
<a name="change_002dresidue_002dnumber"></a>
<div class="header">
<p>
Previous: <a href="#renumber_002dresidue_002drange" accesskey="p" rel="previous">renumber-residue-range</a>, Up: <a href="#Renumber-Residue-Range" accesskey="u" rel="up">Renumber Residue Range</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="change_002dresidue_002dnumber-1"></a>
<h4 class="subsection">11.42.2 change-residue-number</h4>
<dl>
<dt><a name="index-change_002dresidue_002dnumber"></a>function: <strong>change-residue-number</strong> <em>imol chain_id current_resno current_inscode new_resno new_inscode</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>current_resno</em> is an integer number
</li><li> <em>current_inscode</em> is a string
</li><li> <em>new_resno</em> is an integer number
</li><li> <em>new_inscode</em> is a string
</li></ul>

<p>change chain id, residue number or insertion code for given residue 
</p>
</dd></dl>

<hr>
<a name="Scripting-Interface"></a>
<div class="header">
<p>
Next: <a href="#Monomer" accesskey="n" rel="next">Monomer</a>, Previous: <a href="#Renumber-Residue-Range" accesskey="p" rel="previous">Renumber Residue Range</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Scripting-Interface-1"></a>
<h3 class="section">11.43 Scripting Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#probe_002davailable_002dp" accesskey="1">probe-available-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#post_002dscripting_002dwindow" accesskey="2">post-scripting-window</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#post_002dscheme_002dscripting_002dwindow" accesskey="3">post-scheme-scripting-window</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#post_002dpython_002dscripting_002dwindow" accesskey="4">post-python-scripting-window</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="probe_002davailable_002dp"></a>
<div class="header">
<p>
Next: <a href="#post_002dscripting_002dwindow" accesskey="n" rel="next">post-scripting-window</a>, Up: <a href="#Scripting-Interface" accesskey="u" rel="up">Scripting Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="probe_002davailable_002dp-1"></a>
<h4 class="subsection">11.43.1 probe-available-p</h4>
<dl>
<dt><a name="index-probe_002davailable_002dp"></a>function: <strong>probe-available-p</strong></dt>
<dd>
<p>Can we run probe (was the executable variable set properly?) (predicate). 
</p>
<p>Returns: 1 for yes, 2 for no 
</p>
</dd></dl>

<hr>
<a name="post_002dscripting_002dwindow"></a>
<div class="header">
<p>
Next: <a href="#post_002dscheme_002dscripting_002dwindow" accesskey="n" rel="next">post-scheme-scripting-window</a>, Previous: <a href="#probe_002davailable_002dp" accesskey="p" rel="previous">probe-available-p</a>, Up: <a href="#Scripting-Interface" accesskey="u" rel="up">Scripting Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="post_002dscripting_002dwindow-1"></a>
<h4 class="subsection">11.43.2 post-scripting-window</h4>
<dl>
<dt><a name="index-post_002dscripting_002dwindow"></a>function: <strong>post-scripting-window</strong></dt>
<dd>
<p>do nothing - compatibility function 
</p>
</dd></dl>

<hr>
<a name="post_002dscheme_002dscripting_002dwindow"></a>
<div class="header">
<p>
Next: <a href="#post_002dpython_002dscripting_002dwindow" accesskey="n" rel="next">post-python-scripting-window</a>, Previous: <a href="#post_002dscripting_002dwindow" accesskey="p" rel="previous">post-scripting-window</a>, Up: <a href="#Scripting-Interface" accesskey="u" rel="up">Scripting Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="post_002dscheme_002dscripting_002dwindow-1"></a>
<h4 class="subsection">11.43.3 post-scheme-scripting-window</h4>
<dl>
<dt><a name="index-post_002dscheme_002dscripting_002dwindow"></a>function: <strong>post-scheme-scripting-window</strong></dt>
<dd>
<p>pop-up a scripting window for scheming 
</p>
</dd></dl>

<hr>
<a name="post_002dpython_002dscripting_002dwindow"></a>
<div class="header">
<p>
Previous: <a href="#post_002dscheme_002dscripting_002dwindow" accesskey="p" rel="previous">post-scheme-scripting-window</a>, Up: <a href="#Scripting-Interface" accesskey="u" rel="up">Scripting Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="post_002dpython_002dscripting_002dwindow-1"></a>
<h4 class="subsection">11.43.4 post-python-scripting-window</h4>
<dl>
<dt><a name="index-post_002dpython_002dscripting_002dwindow"></a>function: <strong>post-python-scripting-window</strong></dt>
<dd>
<p>pop-up a scripting window for pythoning 
</p>
</dd></dl>

<hr>
<a name="Monomer"></a>
<div class="header">
<p>
Next: <a href="#Regularization-and-Refinement" accesskey="n" rel="next">Regularization and Refinement</a>, Previous: <a href="#Scripting-Interface" accesskey="p" rel="previous">Scripting Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Monomer-1"></a>
<h3 class="section">11.44 Monomer</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#run_002dscript" accesskey="1">run-script</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#run_002dguile_002dscript" accesskey="2">run-guile-script</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#run_002dpython_002dscript" accesskey="3">run-python-script</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#import_002dpython_002dmodule" accesskey="4">import-python-module</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm" accesskey="5">matching-compound-names-from-dictionary-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#comp_002did_002dto_002dname_002dscm" accesskey="6">comp-id-to-name-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#auto_002dload_002ddictionary" accesskey="7">auto-load-dictionary</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#reload_002ddictionary" accesskey="8">reload-dictionary</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dnon_002dauto_002dload_002dresidue_002dname" accesskey="9">add-non-auto-load-residue-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002dnon_002dauto_002dload_002dresidue_002dname">remove-non-auto-load-residue-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy">matching-compound-names-from-dictionary-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#comp_002did_002dto_002dname_002dpy">comp-id-to-name-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="run_002dscript"></a>
<div class="header">
<p>
Next: <a href="#run_002dguile_002dscript" accesskey="n" rel="next">run-guile-script</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="run_002dscript-1"></a>
<h4 class="subsection">11.44.1 run-script</h4>
<dl>
<dt><a name="index-run_002dscript"></a>function: <strong>run-script</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>run script file 
</p>
</dd></dl>

<hr>
<a name="run_002dguile_002dscript"></a>
<div class="header">
<p>
Next: <a href="#run_002dpython_002dscript" accesskey="n" rel="next">run-python-script</a>, Previous: <a href="#run_002dscript" accesskey="p" rel="previous">run-script</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="run_002dguile_002dscript-1"></a>
<h4 class="subsection">11.44.2 run-guile-script</h4>
<dl>
<dt><a name="index-run_002dguile_002dscript"></a>function: <strong>run-guile-script</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>guile run script file 
</p>
</dd></dl>

<hr>
<a name="run_002dpython_002dscript"></a>
<div class="header">
<p>
Next: <a href="#import_002dpython_002dmodule" accesskey="n" rel="next">import-python-module</a>, Previous: <a href="#run_002dguile_002dscript" accesskey="p" rel="previous">run-guile-script</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="run_002dpython_002dscript-1"></a>
<h4 class="subsection">11.44.3 run-python-script</h4>
<dl>
<dt><a name="index-run_002dpython_002dscript"></a>function: <strong>run-python-script</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>run python script file 
</p>
</dd></dl>

<hr>
<a name="import_002dpython_002dmodule"></a>
<div class="header">
<p>
Next: <a href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm" accesskey="n" rel="next">matching-compound-names-from-dictionary-scm</a>, Previous: <a href="#run_002dpython_002dscript" accesskey="p" rel="previous">run-python-script</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="import_002dpython_002dmodule-1"></a>
<h4 class="subsection">11.44.4 import-python-module</h4>
<dl>
<dt><a name="index-import_002dpython_002dmodule"></a>function: <strong>import-python-module</strong> <em>module_name use_namespace</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>module_name</em> is a string
</li><li> <em>use_namespace</em> is an integer number
</li></ul>

<p>import python module 
</p>
</dd></dl>

<hr>
<a name="matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm"></a>
<div class="header">
<p>
Next: <a href="#comp_002did_002dto_002dname_002dscm" accesskey="n" rel="next">comp-id-to-name-scm</a>, Previous: <a href="#import_002dpython_002dmodule" accesskey="p" rel="previous">import-python-module</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm-1"></a>
<h4 class="subsection">11.44.5 matching-compound-names-from-dictionary-scm</h4>
<dl>
<dt><a name="index-matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm"></a>function: <strong>matching-compound-names-from-dictionary-scm</strong> <em>compound_name_fragment allow_minimal_descriptions_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>compound_name_fragment</em> is a string
</li><li> <em>allow_minimal_descriptions_flag</em> is an integer number
</li></ul>

<p>return a list of compoundIDs in the dictionary of which the given string is a substring of the compound name 
</p>
</dd></dl>

<hr>
<a name="comp_002did_002dto_002dname_002dscm"></a>
<div class="header">
<p>
Next: <a href="#auto_002dload_002ddictionary" accesskey="n" rel="next">auto-load-dictionary</a>, Previous: <a href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm" accesskey="p" rel="previous">matching-compound-names-from-dictionary-scm</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="comp_002did_002dto_002dname_002dscm-1"></a>
<h4 class="subsection">11.44.6 comp-id-to-name-scm</h4>
<dl>
<dt><a name="index-comp_002did_002dto_002dname_002dscm"></a>function: <strong>comp-id-to-name-scm</strong> <em>comp_id</em></dt>
<dd>
<p>Where <em>comp_id</em> is a string
</p>
<p>return the monomer name 
</p>
<p>return scheme false if not found 
</p>
</dd></dl>

<hr>
<a name="auto_002dload_002ddictionary"></a>
<div class="header">
<p>
Next: <a href="#reload_002ddictionary" accesskey="n" rel="next">reload-dictionary</a>, Previous: <a href="#comp_002did_002dto_002dname_002dscm" accesskey="p" rel="previous">comp-id-to-name-scm</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="auto_002dload_002ddictionary-1"></a>
<h4 class="subsection">11.44.7 auto-load-dictionary</h4>
<dl>
<dt><a name="index-auto_002dload_002ddictionary"></a>function: <strong>auto-load-dictionary</strong> <em>comp_id</em></dt>
<dd>
<p>Where <em>comp_id</em> is a string
</p>
<p>try to auto-load the dictionary for comp_id from the refmac monomer library. 
</p>
<p>return 0 on failure. return 1 on successful auto-load. return 2 on already-read. 
</p>
</dd></dl>

<hr>
<a name="reload_002ddictionary"></a>
<div class="header">
<p>
Next: <a href="#add_002dnon_002dauto_002dload_002dresidue_002dname" accesskey="n" rel="next">add-non-auto-load-residue-name</a>, Previous: <a href="#auto_002dload_002ddictionary" accesskey="p" rel="previous">auto-load-dictionary</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="reload_002ddictionary-1"></a>
<h4 class="subsection">11.44.8 reload-dictionary</h4>
<dl>
<dt><a name="index-reload_002ddictionary"></a>function: <strong>reload-dictionary</strong> <em>comp_id</em></dt>
<dd>
<p>Where <em>comp_id</em> is a string
</p>
<p>as above, but dictionary is cleared and re-read if it already exists 
</p>
</dd></dl>

<hr>
<a name="add_002dnon_002dauto_002dload_002dresidue_002dname"></a>
<div class="header">
<p>
Next: <a href="#remove_002dnon_002dauto_002dload_002dresidue_002dname" accesskey="n" rel="next">remove-non-auto-load-residue-name</a>, Previous: <a href="#reload_002ddictionary" accesskey="p" rel="previous">reload-dictionary</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dnon_002dauto_002dload_002dresidue_002dname-1"></a>
<h4 class="subsection">11.44.9 add-non-auto-load-residue-name</h4>
<dl>
<dt><a name="index-add_002dnon_002dauto_002dload_002dresidue_002dname"></a>function: <strong>add-non-auto-load-residue-name</strong> <em>s</em></dt>
<dd>
<p>Where <em>s</em> is a string
</p>
<p>add residue name to the list of residue names that don&rsquo;t get auto-loaded from the Refmac dictionary. 
</p>
</dd></dl>

<hr>
<a name="remove_002dnon_002dauto_002dload_002dresidue_002dname"></a>
<div class="header">
<p>
Next: <a href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy" accesskey="n" rel="next">matching-compound-names-from-dictionary-py</a>, Previous: <a href="#add_002dnon_002dauto_002dload_002dresidue_002dname" accesskey="p" rel="previous">add-non-auto-load-residue-name</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002dnon_002dauto_002dload_002dresidue_002dname-1"></a>
<h4 class="subsection">11.44.10 remove-non-auto-load-residue-name</h4>
<dl>
<dt><a name="index-remove_002dnon_002dauto_002dload_002dresidue_002dname"></a>function: <strong>remove-non-auto-load-residue-name</strong> <em>s</em></dt>
<dd>
<p>Where <em>s</em> is a string
</p>
<p>remove residue name from the list of residue names that don&rsquo;t get auto-loaded from the Refmac dictionary. 
</p>
</dd></dl>

<hr>
<a name="matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy"></a>
<div class="header">
<p>
Next: <a href="#comp_002did_002dto_002dname_002dpy" accesskey="n" rel="next">comp-id-to-name-py</a>, Previous: <a href="#remove_002dnon_002dauto_002dload_002dresidue_002dname" accesskey="p" rel="previous">remove-non-auto-load-residue-name</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy-1"></a>
<h4 class="subsection">11.44.11 matching-compound-names-from-dictionary-py</h4>
<dl>
<dt><a name="index-matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy"></a>function: <strong>matching-compound-names-from-dictionary-py</strong> <em>compound_name_fragment allow_minimal_descriptions_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>compound_name_fragment</em> is a string
</li><li> <em>allow_minimal_descriptions_flag</em> is an integer number
</li></ul>

<p>return a list of compoundIDs in the dictionary which the given string is a substring of the compound name 
</p>
</dd></dl>

<hr>
<a name="comp_002did_002dto_002dname_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy" accesskey="p" rel="previous">matching-compound-names-from-dictionary-py</a>, Up: <a href="#Monomer" accesskey="u" rel="up">Monomer</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="comp_002did_002dto_002dname_002dpy-1"></a>
<h4 class="subsection">11.44.12 comp-id-to-name-py</h4>
<dl>
<dt><a name="index-comp_002did_002dto_002dname_002dpy"></a>function: <strong>comp-id-to-name-py</strong> <em>comp_id</em></dt>
<dd>
<p>Where <em>comp_id</em> is a string
</p>
<p>return the monomer name 
</p>
<p>return python false if not found 
</p>
</dd></dl>

<hr>
<a name="Regularization-and-Refinement"></a>
<div class="header">
<p>
Next: <a href="#Simplex-Refinement-Interface" accesskey="n" rel="next">Simplex Refinement Interface</a>, Previous: <a href="#Monomer" accesskey="p" rel="previous">Monomer</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Regularization-and-Refinement-1"></a>
<h3 class="section">11.45 Regularization and Refinement</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#add_002dplanar_002dpeptide_002drestraints" accesskey="1">add-planar-peptide-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002dplanar_002dpeptide_002drestraints" accesskey="2">remove-planar-peptide-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dtight_002dplanar_002dpeptide_002drestraints" accesskey="3">make-tight-planar-peptide-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002duse_002dtrans_002dpeptide_002drestraints" accesskey="4">set-use-trans-peptide-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002domega_002dtorsion_002drestriants" accesskey="5">add-omega-torsion-restriants</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002domega_002dtorsion_002drestriants" accesskey="6">remove-omega-torsion-restriants</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dauto_002dh_002dbond_002drestraints" accesskey="7">set-auto-h-bond-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefinement_002dimmediate_002dreplacement" accesskey="8">set-refinement-immediate-replacement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refinement_002dimmediate_002dreplacement_002dstate" accesskey="9">refinement-immediate-replacement-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dresidue_002dselection_002dflash_002dframes_002dnumber">set-residue-selection-flash-frames-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#accept_002dmoving_002datoms">accept-moving-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#accept_002dregularizement">accept-regularizement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refinement_002dalready_002dongoing_002dp">refinement-already-ongoing-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refine_002dresidues_002dscm">refine-residues-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refine_002dresidues_002dpy">refine-residues-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#shiftfield_002db_002dfactor_002drefinement">shiftfield-b-factor-refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#shiftfield_002dxyz_002dfactor_002drefinement">shiftfield-xyz-factor-refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefine_002dwith_002dtorsion_002drestraints">set-refine-with-torsion-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refine_002dwith_002dtorsion_002drestraints_002dstate">refine-with-torsion-restraints-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmatrix">set-matrix</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#matrix_002dstate">matrix-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dmap_002dweight">get-map-weight</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefine_002dauto_002drange_002dstep">set-refine-auto-range-step</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefine_002dmax_002dresidues">set-refine-max-residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refine_002dzone_002datom_002dindex_002ddefine">refine-zone-atom-index-define</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refine_002dzone">refine-zone</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#repeat_002drefine_002dzone">repeat-refine-zone</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refine_002dauto_002drange">refine-auto-range</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#regularize_002dzone">regularize-zone</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddragged_002drefinement_002dsteps_002dper_002dframe">set-dragged-refinement-steps-per-frame</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#dragged_002drefinement_002dsteps_002dper_002dframe">dragged-refinement-steps-per-frame</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefinement_002drefine_002dper_002dframe">set-refinement-refine-per-frame</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refinement_002drefine_002dper_002dframe_002dstate">refinement-refine-per-frame-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefinement_002ddrag_002delasticity">set-refinement-drag-elasticity</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefine_002dramachandran_002dangles">set-refine-ramachandran-angles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefine_002dramachandran_002drestraints_002dtype">set-refine-ramachandran-restraints-type</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefine_002dramachandran_002drestraints_002dweight">set-refine-ramachandran-restraints-weight</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement">set-fix-chiral-volumes-before-refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#check_002dchiral_002dvolumes">check-chiral-volumes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dchiral_002dvolume_002derrors_002ddialog">set-show-chiral-volume-errors-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsecondary_002dstructure_002drestraints_002dtype">set-secondary-structure-restraints-type</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#secondary_002dstructure_002drestraints_002dtype">secondary-structure-restraints-type</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#imol_002drefinement_002dmap">imol-refinement-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dimol_002drefinement_002dmap">set-imol-refinement-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#does_002dresidue_002dexist_002dp">does-residue-exist-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002drestraints">delete-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dextra_002dbond_002drestraint">add-extra-bond-restraint</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dextra_002dgeman_002dmcclure_002drestraint">add-extra-geman-mcclure-restraint</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dall_002dextra_002drestraints">delete-all-extra-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dextra_002drestraints_002dfor_002dresidue">delete-extra-restraints-for-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits">set-extra-restraints-prosmart-sigma-limits</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#generate_002dlocal_002dself_002drestraints">generate-local-self-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#generate_002dself_002drestraints">generate-self-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm">generate-local-self-restraints-by-residues-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#write_002dinterpolated_002dextra_002drestraints">write-interpolated-extra-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints">write-interpolated-models-and-extra-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions">set-use-only-extra-torsion-restraints-for-torsions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate">use-only-extra-torsion-restraints-for-torsions-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002drestraints_002deditor">show-restraints-editor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002drestraints_002deditor_002dby_002dindex">show-restraints-editor-by-index</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#write_002drestraints_002dcif_002ddictionary">write-restraints-cif-dictionary</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="add_002dplanar_002dpeptide_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#remove_002dplanar_002dpeptide_002drestraints" accesskey="n" rel="next">remove-planar-peptide-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dplanar_002dpeptide_002drestraints-1"></a>
<h4 class="subsection">11.45.1 add-planar-peptide-restraints</h4>
<dl>
<dt><a name="index-add_002dplanar_002dpeptide_002drestraints"></a>function: <strong>add-planar-peptide-restraints</strong></dt>
<dd>
<p>add a restraint on peptides to make them planar 
</p>
<p>This adds a 5 atom restraint that includes both CA atoms of the peptide. Use this rather than editting the mon_lib_list.cif file. 
</p>
</dd></dl>

<hr>
<a name="remove_002dplanar_002dpeptide_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#make_002dtight_002dplanar_002dpeptide_002drestraints" accesskey="n" rel="next">make-tight-planar-peptide-restraints</a>, Previous: <a href="#add_002dplanar_002dpeptide_002drestraints" accesskey="p" rel="previous">add-planar-peptide-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002dplanar_002dpeptide_002drestraints-1"></a>
<h4 class="subsection">11.45.2 remove-planar-peptide-restraints</h4>
<dl>
<dt><a name="index-remove_002dplanar_002dpeptide_002drestraints"></a>function: <strong>remove-planar-peptide-restraints</strong></dt>
<dd>
<p>remove restraints on peptides to make them planar. 
</p>
</dd></dl>

<hr>
<a name="make_002dtight_002dplanar_002dpeptide_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#set_002duse_002dtrans_002dpeptide_002drestraints" accesskey="n" rel="next">set-use-trans-peptide-restraints</a>, Previous: <a href="#remove_002dplanar_002dpeptide_002drestraints" accesskey="p" rel="previous">remove-planar-peptide-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dtight_002dplanar_002dpeptide_002drestraints-1"></a>
<h4 class="subsection">11.45.3 make-tight-planar-peptide-restraints</h4>
<dl>
<dt><a name="index-make_002dtight_002dplanar_002dpeptide_002drestraints"></a>function: <strong>make-tight-planar-peptide-restraints</strong></dt>
<dd>
<p>make the planar peptide restraints tight 
</p>
<p>Useful when refining models with cryo-EM maps 
</p>
</dd></dl>

<hr>
<a name="set_002duse_002dtrans_002dpeptide_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#add_002domega_002dtorsion_002drestriants" accesskey="n" rel="next">add-omega-torsion-restriants</a>, Previous: <a href="#make_002dtight_002dplanar_002dpeptide_002drestraints" accesskey="p" rel="previous">make-tight-planar-peptide-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002duse_002dtrans_002dpeptide_002drestraints-1"></a>
<h4 class="subsection">11.45.4 set-use-trans-peptide-restraints</h4>
<dl>
<dt><a name="index-set_002duse_002dtrans_002dpeptide_002drestraints"></a>function: <strong>set-use-trans-peptide-restraints</strong> <em>on_off_state</em></dt>
<dd>
<p>Where <em>on_off_state</em> is an integer number
</p>
<p>add a restraint on peptides to keep trans peptides trans 
</p>
<p>i.e. omega in trans-peptides is restraints to 180 degrees. 
</p>
</dd></dl>

<hr>
<a name="add_002domega_002dtorsion_002drestriants"></a>
<div class="header">
<p>
Next: <a href="#remove_002domega_002dtorsion_002drestriants" accesskey="n" rel="next">remove-omega-torsion-restriants</a>, Previous: <a href="#set_002duse_002dtrans_002dpeptide_002drestraints" accesskey="p" rel="previous">set-use-trans-peptide-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002domega_002dtorsion_002drestriants-1"></a>
<h4 class="subsection">11.45.5 add-omega-torsion-restriants</h4>
<dl>
<dt><a name="index-add_002domega_002dtorsion_002drestriants"></a>function: <strong>add-omega-torsion-restriants</strong></dt>
<dd>
<p>add restraints on the omega angle of the peptides 
</p>
<p>(that is the torsion round the peptide bond). Omega angles that are closer to 0 than to 180 will be refined as cis peptides (and of course if omega is greater than 90 then the peptide will be refined as a trans peptide (this is the normal case). 
</p>
</dd></dl>

<hr>
<a name="remove_002domega_002dtorsion_002drestriants"></a>
<div class="header">
<p>
Next: <a href="#set_002dauto_002dh_002dbond_002drestraints" accesskey="n" rel="next">set-auto-h-bond-restraints</a>, Previous: <a href="#add_002domega_002dtorsion_002drestriants" accesskey="p" rel="previous">add-omega-torsion-restriants</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002domega_002dtorsion_002drestriants-1"></a>
<h4 class="subsection">11.45.6 remove-omega-torsion-restriants</h4>
<dl>
<dt><a name="index-remove_002domega_002dtorsion_002drestriants"></a>function: <strong>remove-omega-torsion-restriants</strong></dt>
<dd>
<p>remove omega restraints on CIS and TRANS linked residues. 
</p>
</dd></dl>

<hr>
<a name="set_002dauto_002dh_002dbond_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#set_002drefinement_002dimmediate_002dreplacement" accesskey="n" rel="next">set-refinement-immediate-replacement</a>, Previous: <a href="#remove_002domega_002dtorsion_002drestriants" accesskey="p" rel="previous">remove-omega-torsion-restriants</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dauto_002dh_002dbond_002drestraints-1"></a>
<h4 class="subsection">11.45.7 set-auto-h-bond-restraints</h4>
<dl>
<dt><a name="index-set_002dauto_002dh_002dbond_002drestraints"></a>function: <strong>set-auto-h-bond-restraints</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>add or remove auto H-bond restraints 
</p>
</dd></dl>

<hr>
<a name="set_002drefinement_002dimmediate_002dreplacement"></a>
<div class="header">
<p>
Next: <a href="#refinement_002dimmediate_002dreplacement_002dstate" accesskey="n" rel="next">refinement-immediate-replacement-state</a>, Previous: <a href="#set_002dauto_002dh_002dbond_002drestraints" accesskey="p" rel="previous">set-auto-h-bond-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefinement_002dimmediate_002dreplacement-1"></a>
<h4 class="subsection">11.45.8 set-refinement-immediate-replacement</h4>
<dl>
<dt><a name="index-set_002drefinement_002dimmediate_002dreplacement"></a>function: <strong>set-refinement-immediate-replacement</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>set immediate replacement mode for refinement and regularization. You need this (call with istate=1) if you are scripting refinement/regularization 
</p>
</dd></dl>

<hr>
<a name="refinement_002dimmediate_002dreplacement_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dresidue_002dselection_002dflash_002dframes_002dnumber" accesskey="n" rel="next">set-residue-selection-flash-frames-number</a>, Previous: <a href="#set_002drefinement_002dimmediate_002dreplacement" accesskey="p" rel="previous">set-refinement-immediate-replacement</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refinement_002dimmediate_002dreplacement_002dstate-1"></a>
<h4 class="subsection">11.45.9 refinement-immediate-replacement-state</h4>
<dl>
<dt><a name="index-refinement_002dimmediate_002dreplacement_002dstate"></a>function: <strong>refinement-immediate-replacement-state</strong></dt>
<dd>
<p>query the state of the immediate replacement mode 
</p>
</dd></dl>

<hr>
<a name="set_002dresidue_002dselection_002dflash_002dframes_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#accept_002dmoving_002datoms" accesskey="n" rel="next">accept-moving-atoms</a>, Previous: <a href="#refinement_002dimmediate_002dreplacement_002dstate" accesskey="p" rel="previous">refinement-immediate-replacement-state</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dresidue_002dselection_002dflash_002dframes_002dnumber-1"></a>
<h4 class="subsection">11.45.10 set-residue-selection-flash-frames-number</h4>
<dl>
<dt><a name="index-set_002dresidue_002dselection_002dflash_002dframes_002dnumber"></a>function: <strong>set-residue-selection-flash-frames-number</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set the number of frames for which the selected residue range flashes 
</p>
<p>On fast computers, this can be set to higher than the default for more aesthetic appeal. 
</p>
</dd></dl>

<hr>
<a name="accept_002dmoving_002datoms"></a>
<div class="header">
<p>
Next: <a href="#accept_002dregularizement" accesskey="n" rel="next">accept-regularizement</a>, Previous: <a href="#set_002dresidue_002dselection_002dflash_002dframes_002dnumber" accesskey="p" rel="previous">set-residue-selection-flash-frames-number</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="accept_002dmoving_002datoms-1"></a>
<h4 class="subsection">11.45.11 accept-moving-atoms</h4>
<dl>
<dt><a name="index-accept_002dmoving_002datoms"></a>function: <strong>accept-moving-atoms</strong></dt>
<dd>
<p>accept the new positions of the regularized or refined residues 
</p>
<p>If you are scripting refinement and/or regularization, this is the function that you need to call after refine-zone or regularize-zone. 
</p>
</dd></dl>

<hr>
<a name="accept_002dregularizement"></a>
<div class="header">
<p>
Next: <a href="#refinement_002dalready_002dongoing_002dp" accesskey="n" rel="next">refinement-already-ongoing-p</a>, Previous: <a href="#accept_002dmoving_002datoms" accesskey="p" rel="previous">accept-moving-atoms</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="accept_002dregularizement-1"></a>
<h4 class="subsection">11.45.12 accept-regularizement</h4>
<dl>
<dt><a name="index-accept_002dregularizement"></a>function: <strong>accept-regularizement</strong></dt>
<dd>
<p>a hideous alias for the above 
</p>
</dd></dl>

<hr>
<a name="refinement_002dalready_002dongoing_002dp"></a>
<div class="header">
<p>
Next: <a href="#refine_002dresidues_002dscm" accesskey="n" rel="next">refine-residues-scm</a>, Previous: <a href="#accept_002dregularizement" accesskey="p" rel="previous">accept-regularizement</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refinement_002dalready_002dongoing_002dp-1"></a>
<h4 class="subsection">11.45.13 refinement-already-ongoing-p</h4>
<dl>
<dt><a name="index-refinement_002dalready_002dongoing_002dp"></a>function: <strong>refinement-already-ongoing-p</strong></dt>
<dd>
<p>If there is a refinement on-going already, we don&rsquo;t want to start a new one. 
</p>
<p>The is the means to ask if that is the case. This needs a scheme wrapper to provide refinement-already-ongoing? The question is translated to &quot;are the intermediate atoms being displayed?&quot; so that might be a more accurate function name than the current one.
</p>
<p>Returns: 1 for yes, 0 for no. 
</p>
</dd></dl>

<hr>
<a name="refine_002dresidues_002dscm"></a>
<div class="header">
<p>
Next: <a href="#refine_002dresidues_002dpy" accesskey="n" rel="next">refine-residues-py</a>, Previous: <a href="#refinement_002dalready_002dongoing_002dp" accesskey="p" rel="previous">refinement-already-ongoing-p</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refine_002dresidues_002dscm-1"></a>
<h4 class="subsection">11.45.14 refine-residues-scm</h4>
<dl>
<dt><a name="index-refine_002dresidues_002dscm"></a>function: <strong>refine-residues-scm</strong> <em>imol r</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>r</em> is a SCM
</li></ul>

<p>refine residues, r is a list of residue specs. 
</p>

<p>1 - an information string (in case of error)
</p>
<p>2 - the progress variable (from GSL)
</p>
<p>3 - refinement results for each particular geometry type (bonds, angles etc.) 
</p>
<p>Returns: refinement results, which consists of
</p>
</dd></dl>

<hr>
<a name="refine_002dresidues_002dpy"></a>
<div class="header">
<p>
Next: <a href="#shiftfield_002db_002dfactor_002drefinement" accesskey="n" rel="next">shiftfield-b-factor-refinement</a>, Previous: <a href="#refine_002dresidues_002dscm" accesskey="p" rel="previous">refine-residues-scm</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refine_002dresidues_002dpy-1"></a>
<h4 class="subsection">11.45.15 refine-residues-py</h4>
<dl>
<dt><a name="index-refine_002dresidues_002dpy"></a>function: <strong>refine-residues-py</strong> <em>imol r</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>r</em> is a PyObject
</li></ul>

<p>refine the residues in the given residue spec list 
</p>
<p>Returns: the refinement summary statistics 
</p>
</dd></dl>

<hr>
<a name="shiftfield_002db_002dfactor_002drefinement"></a>
<div class="header">
<p>
Next: <a href="#shiftfield_002dxyz_002dfactor_002drefinement" accesskey="n" rel="next">shiftfield-xyz-factor-refinement</a>, Previous: <a href="#refine_002dresidues_002dpy" accesskey="p" rel="previous">refine-residues-py</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="shiftfield_002db_002dfactor_002drefinement-1"></a>
<h4 class="subsection">11.45.16 shiftfield-b-factor-refinement</h4>
<dl>
<dt><a name="index-shiftfield_002db_002dfactor_002drefinement"></a>function: <strong>shiftfield-b-factor-refinement</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>shiftfield B-factor refinement 
</p>
</dd></dl>

<hr>
<a name="shiftfield_002dxyz_002dfactor_002drefinement"></a>
<div class="header">
<p>
Next: <a href="#set_002drefine_002dwith_002dtorsion_002drestraints" accesskey="n" rel="next">set-refine-with-torsion-restraints</a>, Previous: <a href="#shiftfield_002db_002dfactor_002drefinement" accesskey="p" rel="previous">shiftfield-b-factor-refinement</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="shiftfield_002dxyz_002dfactor_002drefinement-1"></a>
<h4 class="subsection">11.45.17 shiftfield-xyz-factor-refinement</h4>
<dl>
<dt><a name="index-shiftfield_002dxyz_002dfactor_002drefinement"></a>function: <strong>shiftfield-xyz-factor-refinement</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>shiftfield xyz refinement 
</p>
</dd></dl>

<hr>
<a name="set_002drefine_002dwith_002dtorsion_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#refine_002dwith_002dtorsion_002drestraints_002dstate" accesskey="n" rel="next">refine-with-torsion-restraints-state</a>, Previous: <a href="#shiftfield_002dxyz_002dfactor_002drefinement" accesskey="p" rel="previous">shiftfield-xyz-factor-refinement</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefine_002dwith_002dtorsion_002drestraints-1"></a>
<h4 class="subsection">11.45.18 set-refine-with-torsion-restraints</h4>
<dl>
<dt><a name="index-set_002drefine_002dwith_002dtorsion_002drestraints"></a>function: <strong>set-refine-with-torsion-restraints</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>turn on (or off) torsion restraints 
</p>
<p>Pass with istate=1 for on, istate=0 for off. 
</p>
</dd></dl>

<hr>
<a name="refine_002dwith_002dtorsion_002drestraints_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dmatrix" accesskey="n" rel="next">set-matrix</a>, Previous: <a href="#set_002drefine_002dwith_002dtorsion_002drestraints" accesskey="p" rel="previous">set-refine-with-torsion-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refine_002dwith_002dtorsion_002drestraints_002dstate-1"></a>
<h4 class="subsection">11.45.19 refine-with-torsion-restraints-state</h4>
<dl>
<dt><a name="index-refine_002dwith_002dtorsion_002drestraints_002dstate"></a>function: <strong>refine-with-torsion-restraints-state</strong></dt>
<dd>
<p>return the state of above 
</p>
</dd></dl>

<hr>
<a name="set_002dmatrix"></a>
<div class="header">
<p>
Next: <a href="#matrix_002dstate" accesskey="n" rel="next">matrix-state</a>, Previous: <a href="#refine_002dwith_002dtorsion_002drestraints_002dstate" accesskey="p" rel="previous">refine-with-torsion-restraints-state</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmatrix-1"></a>
<h4 class="subsection">11.45.20 set-matrix</h4>
<dl>
<dt><a name="index-set_002dmatrix-1"></a>function: <strong>set-matrix</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the relative weight of the geometric terms to the map terms 
</p>
<p>The default is 60.
</p>
<p>The higher the number the more weight that is given to the map terms but the resulting chi squared values are higher). This will be needed for maps generated from data not on (or close to) the absolute scale or maps that have been scaled (for example so that the sigma level has been scaled to 1.0). 
</p>
</dd></dl>

<hr>
<a name="matrix_002dstate"></a>
<div class="header">
<p>
Next: <a href="#get_002dmap_002dweight" accesskey="n" rel="next">get-map-weight</a>, Previous: <a href="#set_002dmatrix" accesskey="p" rel="previous">set-matrix</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="matrix_002dstate-1"></a>
<h4 class="subsection">11.45.21 matrix-state</h4>
<dl>
<dt><a name="index-matrix_002dstate"></a>function: <strong>matrix-state</strong></dt>
<dd>
<p>return the relative weight of the geometric terms to the map terms. 
</p>
</dd></dl>

<hr>
<a name="get_002dmap_002dweight"></a>
<div class="header">
<p>
Next: <a href="#set_002drefine_002dauto_002drange_002dstep" accesskey="n" rel="next">set-refine-auto-range-step</a>, Previous: <a href="#matrix_002dstate" accesskey="p" rel="previous">matrix-state</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dmap_002dweight-1"></a>
<h4 class="subsection">11.45.22 get-map-weight</h4>
<dl>
<dt><a name="index-get_002dmap_002dweight"></a>function: <strong>get-map-weight</strong></dt>
<dd>
<p>return the relative weight of the geometric terms to the map terms. 
</p>
<p>A more sensible name for the 
</p>
<p>function) 
</p>
</dd></dl>

<hr>
<a name="set_002drefine_002dauto_002drange_002dstep"></a>
<div class="header">
<p>
Next: <a href="#set_002drefine_002dmax_002dresidues" accesskey="n" rel="next">set-refine-max-residues</a>, Previous: <a href="#get_002dmap_002dweight" accesskey="p" rel="previous">get-map-weight</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefine_002dauto_002drange_002dstep-1"></a>
<h4 class="subsection">11.45.23 set-refine-auto-range-step</h4>
<dl>
<dt><a name="index-set_002drefine_002dauto_002drange_002dstep"></a>function: <strong>set-refine-auto-range-step</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>change the +/- step for autoranging (default is 1) 
</p>
<p>Auto-ranging alow you to select a range from one button press, this allows you to set the number of residues either side of the clicked residue that becomes the selected zone 
</p>
</dd></dl>

<hr>
<a name="set_002drefine_002dmax_002dresidues"></a>
<div class="header">
<p>
Next: <a href="#refine_002dzone_002datom_002dindex_002ddefine" accesskey="n" rel="next">refine-zone-atom-index-define</a>, Previous: <a href="#set_002drefine_002dauto_002drange_002dstep" accesskey="p" rel="previous">set-refine-auto-range-step</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefine_002dmax_002dresidues-1"></a>
<h4 class="subsection">11.45.24 set-refine-max-residues</h4>
<dl>
<dt><a name="index-set_002drefine_002dmax_002dresidues"></a>function: <strong>set-refine-max-residues</strong> <em>n</em></dt>
<dd>
<p>Where <em>n</em> is an integer number
</p>
<p>set the heuristic fencepost for the maximum number of residues in the refinement/regularization residue range 
</p>
<p>Default is 20 
</p>
</dd></dl>

<hr>
<a name="refine_002dzone_002datom_002dindex_002ddefine"></a>
<div class="header">
<p>
Next: <a href="#refine_002dzone" accesskey="n" rel="next">refine-zone</a>, Previous: <a href="#set_002drefine_002dmax_002dresidues" accesskey="p" rel="previous">set-refine-max-residues</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refine_002dzone_002datom_002dindex_002ddefine-1"></a>
<h4 class="subsection">11.45.25 refine-zone-atom-index-define</h4>
<dl>
<dt><a name="index-refine_002dzone_002datom_002dindex_002ddefine"></a>function: <strong>refine-zone-atom-index-define</strong> <em>imol ind1 ind2</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>ind1</em> is an integer number
</li><li> <em>ind2</em> is an integer number
</li></ul>

<p>refine a zone based on atom indexing 
</p>
</dd></dl>

<hr>
<a name="refine_002dzone"></a>
<div class="header">
<p>
Next: <a href="#repeat_002drefine_002dzone" accesskey="n" rel="next">repeat-refine-zone</a>, Previous: <a href="#refine_002dzone_002datom_002dindex_002ddefine" accesskey="p" rel="previous">refine-zone-atom-index-define</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refine_002dzone-1"></a>
<h4 class="subsection">11.45.26 refine-zone</h4>
<dl>
<dt><a name="index-refine_002dzone"></a>function: <strong>refine-zone</strong> <em>imol chain_id resno1 resno2 altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno1</em> is an integer number
</li><li> <em>resno2</em> is an integer number
</li><li> <em>altconf</em> is a string
</li></ul>

<p>refine a zone 
</p>
<p>presumes that imol_Refinement_Map has been set 
</p>
</dd></dl>

<hr>
<a name="repeat_002drefine_002dzone"></a>
<div class="header">
<p>
Next: <a href="#refine_002dauto_002drange" accesskey="n" rel="next">refine-auto-range</a>, Previous: <a href="#refine_002dzone" accesskey="p" rel="previous">refine-zone</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="repeat_002drefine_002dzone-1"></a>
<h4 class="subsection">11.45.27 repeat-refine-zone</h4>
<dl>
<dt><a name="index-repeat_002drefine_002dzone"></a>function: <strong>repeat-refine-zone</strong></dt>
<dd>
<p>repeat the previous (user-selected) refine zone 
</p>
</dd></dl>

<hr>
<a name="refine_002dauto_002drange"></a>
<div class="header">
<p>
Next: <a href="#regularize_002dzone" accesskey="n" rel="next">regularize-zone</a>, Previous: <a href="#repeat_002drefine_002dzone" accesskey="p" rel="previous">repeat-refine-zone</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refine_002dauto_002drange-1"></a>
<h4 class="subsection">11.45.28 refine-auto-range</h4>
<dl>
<dt><a name="index-refine_002dauto_002drange"></a>function: <strong>refine-auto-range</strong> <em>imol chain_id resno1 altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno1</em> is an integer number
</li><li> <em>altconf</em> is a string
</li></ul>

<p>refine a zone using auto-range 
</p>
<p>presumes that imol_Refinement_Map has been set 
</p>
</dd></dl>

<hr>
<a name="regularize_002dzone"></a>
<div class="header">
<p>
Next: <a href="#set_002ddragged_002drefinement_002dsteps_002dper_002dframe" accesskey="n" rel="next">set-dragged-refinement-steps-per-frame</a>, Previous: <a href="#refine_002dauto_002drange" accesskey="p" rel="previous">refine-auto-range</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="regularize_002dzone-1"></a>
<h4 class="subsection">11.45.29 regularize-zone</h4>
<dl>
<dt><a name="index-regularize_002dzone"></a>function: <strong>regularize-zone</strong> <em>imol chain_id resno1 resno2 altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno1</em> is an integer number
</li><li> <em>resno2</em> is an integer number
</li><li> <em>altconf</em> is a string
</li></ul>

<p>regularize a zone 
</p>
<p>Returns: a status, whether the regularisation was done or not. 0 for no, 1 for yes. 
</p>
</dd></dl>

<hr>
<a name="set_002ddragged_002drefinement_002dsteps_002dper_002dframe"></a>
<div class="header">
<p>
Next: <a href="#dragged_002drefinement_002dsteps_002dper_002dframe" accesskey="n" rel="next">dragged-refinement-steps-per-frame</a>, Previous: <a href="#regularize_002dzone" accesskey="p" rel="previous">regularize-zone</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddragged_002drefinement_002dsteps_002dper_002dframe-1"></a>
<h4 class="subsection">11.45.30 set-dragged-refinement-steps-per-frame</h4>
<dl>
<dt><a name="index-set_002ddragged_002drefinement_002dsteps_002dper_002dframe"></a>function: <strong>set-dragged-refinement-steps-per-frame</strong> <em>v</em></dt>
<dd>
<p>Where <em>v</em> is an integer number
</p>
<p>set the number of refinement steps applied to the intermediate atoms each frame of graphics. 
</p>
<p>smaller numbers make the movement of the intermediate atoms slower, smoother, more elegant.
</p>
<p>Default: 80. 
</p>
</dd></dl>

<hr>
<a name="dragged_002drefinement_002dsteps_002dper_002dframe"></a>
<div class="header">
<p>
Next: <a href="#set_002drefinement_002drefine_002dper_002dframe" accesskey="n" rel="next">set-refinement-refine-per-frame</a>, Previous: <a href="#set_002ddragged_002drefinement_002dsteps_002dper_002dframe" accesskey="p" rel="previous">set-dragged-refinement-steps-per-frame</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="dragged_002drefinement_002dsteps_002dper_002dframe-1"></a>
<h4 class="subsection">11.45.31 dragged-refinement-steps-per-frame</h4>
<dl>
<dt><a name="index-dragged_002drefinement_002dsteps_002dper_002dframe"></a>function: <strong>dragged-refinement-steps-per-frame</strong></dt>
<dd>
<p>return the number of steps per frame in dragged refinement 
</p>
</dd></dl>

<hr>
<a name="set_002drefinement_002drefine_002dper_002dframe"></a>
<div class="header">
<p>
Next: <a href="#refinement_002drefine_002dper_002dframe_002dstate" accesskey="n" rel="next">refinement-refine-per-frame-state</a>, Previous: <a href="#dragged_002drefinement_002dsteps_002dper_002dframe" accesskey="p" rel="previous">dragged-refinement-steps-per-frame</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefinement_002drefine_002dper_002dframe-1"></a>
<h4 class="subsection">11.45.32 set-refinement-refine-per-frame</h4>
<dl>
<dt><a name="index-set_002drefinement_002drefine_002dper_002dframe"></a>function: <strong>set-refinement-refine-per-frame</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>allow refinement of intermediate atoms after dragging, before displaying (default: 0, off). 
</p>
<p>An attempt to do something like xfit does, at the request of Frank von Delft.
</p>
<p>Pass with istate=1 to enable this option. 
</p>
</dd></dl>

<hr>
<a name="refinement_002drefine_002dper_002dframe_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002drefinement_002ddrag_002delasticity" accesskey="n" rel="next">set-refinement-drag-elasticity</a>, Previous: <a href="#set_002drefinement_002drefine_002dper_002dframe" accesskey="p" rel="previous">set-refinement-refine-per-frame</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refinement_002drefine_002dper_002dframe_002dstate-1"></a>
<h4 class="subsection">11.45.33 refinement-refine-per-frame-state</h4>
<dl>
<dt><a name="index-refinement_002drefine_002dper_002dframe_002dstate"></a>function: <strong>refinement-refine-per-frame-state</strong></dt>
<dd>
<p>query the state of the above option 
</p>
</dd></dl>

<hr>
<a name="set_002drefinement_002ddrag_002delasticity"></a>
<div class="header">
<p>
Next: <a href="#set_002drefine_002dramachandran_002dangles" accesskey="n" rel="next">set-refine-ramachandran-angles</a>, Previous: <a href="#refinement_002drefine_002dper_002dframe_002dstate" accesskey="p" rel="previous">refinement-refine-per-frame-state</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefinement_002ddrag_002delasticity-1"></a>
<h4 class="subsection">11.45.34 set-refinement-drag-elasticity</h4>
<dl>
<dt><a name="index-set_002drefinement_002ddrag_002delasticity"></a>function: <strong>set-refinement-drag-elasticity</strong> <em>e</em></dt>
<dd>
<p>Where <em>e</em> is a number
</p>
<p>the elasticity of the dragged atom in refinement mode. 
</p>
<p>Default 0.33
</p>
<p>Bigger numbers mean bigger movement of the other atoms. 
</p>
</dd></dl>

<hr>
<a name="set_002drefine_002dramachandran_002dangles"></a>
<div class="header">
<p>
Next: <a href="#set_002drefine_002dramachandran_002drestraints_002dtype" accesskey="n" rel="next">set-refine-ramachandran-restraints-type</a>, Previous: <a href="#set_002drefinement_002ddrag_002delasticity" accesskey="p" rel="previous">set-refinement-drag-elasticity</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefine_002dramachandran_002dangles-1"></a>
<h4 class="subsection">11.45.35 set-refine-ramachandran-angles</h4>
<dl>
<dt><a name="index-set_002drefine_002dramachandran_002dangles"></a>function: <strong>set-refine-ramachandran-angles</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>turn on Ramachandran angles refinement in refinement and regularization 
</p>
<p>name consistent with 
</p>
<p>!? 
</p>
</dd></dl>

<hr>
<a name="set_002drefine_002dramachandran_002drestraints_002dtype"></a>
<div class="header">
<p>
Next: <a href="#set_002drefine_002dramachandran_002drestraints_002dweight" accesskey="n" rel="next">set-refine-ramachandran-restraints-weight</a>, Previous: <a href="#set_002drefine_002dramachandran_002dangles" accesskey="p" rel="previous">set-refine-ramachandran-angles</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefine_002dramachandran_002drestraints_002dtype-1"></a>
<h4 class="subsection">11.45.36 set-refine-ramachandran-restraints-type</h4>
<dl>
<dt><a name="index-set_002drefine_002dramachandran_002drestraints_002dtype"></a>function: <strong>set-refine-ramachandran-restraints-type</strong> <em>type</em></dt>
<dd>
<p>Where <em>type</em> is an integer number
</p>
<p>change the target function type 
</p>
</dd></dl>

<hr>
<a name="set_002drefine_002dramachandran_002drestraints_002dweight"></a>
<div class="header">
<p>
Next: <a href="#set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement" accesskey="n" rel="next">set-fix-chiral-volumes-before-refinement</a>, Previous: <a href="#set_002drefine_002dramachandran_002drestraints_002dtype" accesskey="p" rel="previous">set-refine-ramachandran-restraints-type</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefine_002dramachandran_002drestraints_002dweight-1"></a>
<h4 class="subsection">11.45.37 set-refine-ramachandran-restraints-weight</h4>
<dl>
<dt><a name="index-set_002drefine_002dramachandran_002drestraints_002dweight"></a>function: <strong>set-refine-ramachandran-restraints-weight</strong> <em>w</em></dt>
<dd>
<p>Where <em>w</em> is a number
</p>
<p>change the target function weight 
</p>
<p>a big number means bad things 
</p>
</dd></dl>

<hr>
<a name="set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement"></a>
<div class="header">
<p>
Next: <a href="#check_002dchiral_002dvolumes" accesskey="n" rel="next">check-chiral-volumes</a>, Previous: <a href="#set_002drefine_002dramachandran_002drestraints_002dweight" accesskey="p" rel="previous">set-refine-ramachandran-restraints-weight</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement-1"></a>
<h4 class="subsection">11.45.38 set-fix-chiral-volumes-before-refinement</h4>
<dl>
<dt><a name="index-set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement"></a>function: <strong>set-fix-chiral-volumes-before-refinement</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>correct the sign of chiral volumes before commencing refinement? 
</p>
<p>Do we want to fix chiral volumes (by moving the chiral atom to the other side of the chiral plane if necessary). Default yes (1). Note: doesn&rsquo;t work currently. 
</p>
</dd></dl>

<hr>
<a name="check_002dchiral_002dvolumes"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002dchiral_002dvolume_002derrors_002ddialog" accesskey="n" rel="next">set-show-chiral-volume-errors-dialog</a>, Previous: <a href="#set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement" accesskey="p" rel="previous">set-fix-chiral-volumes-before-refinement</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="check_002dchiral_002dvolumes-1"></a>
<h4 class="subsection">11.45.39 check-chiral-volumes</h4>
<dl>
<dt><a name="index-check_002dchiral_002dvolumes"></a>function: <strong>check-chiral-volumes</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>query the state of the above option 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002dchiral_002dvolume_002derrors_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#set_002dsecondary_002dstructure_002drestraints_002dtype" accesskey="n" rel="next">set-secondary-structure-restraints-type</a>, Previous: <a href="#check_002dchiral_002dvolumes" accesskey="p" rel="previous">check-chiral-volumes</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dchiral_002dvolume_002derrors_002ddialog-1"></a>
<h4 class="subsection">11.45.40 set-show-chiral-volume-errors-dialog</h4>
<dl>
<dt><a name="index-set_002dshow_002dchiral_002dvolume_002derrors_002ddialog"></a>function: <strong>set-show-chiral-volume-errors-dialog</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>For experienced Cooters who don&rsquo;t like Coot nannying about chiral volumes during refinement. 
</p>
</dd></dl>

<hr>
<a name="set_002dsecondary_002dstructure_002drestraints_002dtype"></a>
<div class="header">
<p>
Next: <a href="#secondary_002dstructure_002drestraints_002dtype" accesskey="n" rel="next">secondary-structure-restraints-type</a>, Previous: <a href="#set_002dshow_002dchiral_002dvolume_002derrors_002ddialog" accesskey="p" rel="previous">set-show-chiral-volume-errors-dialog</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsecondary_002dstructure_002drestraints_002dtype-1"></a>
<h4 class="subsection">11.45.41 set-secondary-structure-restraints-type</h4>
<dl>
<dt><a name="index-set_002dsecondary_002dstructure_002drestraints_002dtype"></a>function: <strong>set-secondary-structure-restraints-type</strong> <em>itype</em></dt>
<dd>
<p>Where <em>itype</em> is an integer number
</p>
<p>set the type of secondary structure restraints 
</p>
<p>0 no sec str restraints
</p>
<p>1 alpha helix restraints
</p>
<p>2 beta strand restraints 
</p>
</dd></dl>

<hr>
<a name="secondary_002dstructure_002drestraints_002dtype"></a>
<div class="header">
<p>
Next: <a href="#imol_002drefinement_002dmap" accesskey="n" rel="next">imol-refinement-map</a>, Previous: <a href="#set_002dsecondary_002dstructure_002drestraints_002dtype" accesskey="p" rel="previous">set-secondary-structure-restraints-type</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="secondary_002dstructure_002drestraints_002dtype-1"></a>
<h4 class="subsection">11.45.42 secondary-structure-restraints-type</h4>
<dl>
<dt><a name="index-secondary_002dstructure_002drestraints_002dtype"></a>function: <strong>secondary-structure-restraints-type</strong></dt>
<dd>
<p>return the secondary structure restraints type 
</p>
</dd></dl>

<hr>
<a name="imol_002drefinement_002dmap"></a>
<div class="header">
<p>
Next: <a href="#set_002dimol_002drefinement_002dmap" accesskey="n" rel="next">set-imol-refinement-map</a>, Previous: <a href="#secondary_002dstructure_002drestraints_002dtype" accesskey="p" rel="previous">secondary-structure-restraints-type</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="imol_002drefinement_002dmap-1"></a>
<h4 class="subsection">11.45.43 imol-refinement-map</h4>
<dl>
<dt><a name="index-imol_002drefinement_002dmap"></a>function: <strong>imol-refinement-map</strong></dt>
<dd>
<p>the molecule number of the map used for refinement 
</p>
<p>Returns: the map number, if it has been set or there is only one map, return -1 on no map set (ambiguous) or no maps. 
</p>
</dd></dl>

<hr>
<a name="set_002dimol_002drefinement_002dmap"></a>
<div class="header">
<p>
Next: <a href="#does_002dresidue_002dexist_002dp" accesskey="n" rel="next">does-residue-exist-p</a>, Previous: <a href="#imol_002drefinement_002dmap" accesskey="p" rel="previous">imol-refinement-map</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dimol_002drefinement_002dmap-1"></a>
<h4 class="subsection">11.45.44 set-imol-refinement-map</h4>
<dl>
<dt><a name="index-set_002dimol_002drefinement_002dmap"></a>function: <strong>set-imol-refinement-map</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>set the molecule number of the map to be used for refinement/fitting. 
</p>
<p>Returns: imol on success, -1 on failure 
</p>
</dd></dl>

<hr>
<a name="does_002dresidue_002dexist_002dp"></a>
<div class="header">
<p>
Next: <a href="#delete_002drestraints" accesskey="n" rel="next">delete-restraints</a>, Previous: <a href="#set_002dimol_002drefinement_002dmap" accesskey="p" rel="previous">set-imol-refinement-map</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="does_002dresidue_002dexist_002dp-1"></a>
<h4 class="subsection">11.45.45 does-residue-exist-p</h4>
<dl>
<dt><a name="index-does_002dresidue_002dexist_002dp"></a>function: <strong>does-residue-exist-p</strong> <em>imol chain_id resno inscode</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>inscode</em> is a string
</li></ul>

<p>Does the residue exist? (Raw function) 
</p>
<p>Returns: 0 on not-exist, 1 on does exist. 
</p>
</dd></dl>

<hr>
<a name="delete_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#add_002dextra_002dbond_002drestraint" accesskey="n" rel="next">add-extra-bond-restraint</a>, Previous: <a href="#does_002dresidue_002dexist_002dp" accesskey="p" rel="previous">does-residue-exist-p</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002drestraints-1"></a>
<h4 class="subsection">11.45.46 delete-restraints</h4>
<dl>
<dt><a name="index-delete_002drestraints"></a>function: <strong>delete-restraints</strong> <em>comp_id</em></dt>
<dd>
<p>Where <em>comp_id</em> is a string
</p>
<p>delete the restraints for the given comp_id (i.e. residue name) 
</p>
<p>Returns: success status (0 is failed, 1 is success) 
</p>
</dd></dl>

<hr>
<a name="add_002dextra_002dbond_002drestraint"></a>
<div class="header">
<p>
Next: <a href="#add_002dextra_002dgeman_002dmcclure_002drestraint" accesskey="n" rel="next">add-extra-geman-mcclure-restraint</a>, Previous: <a href="#delete_002drestraints" accesskey="p" rel="previous">delete-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dextra_002dbond_002drestraint-1"></a>
<h4 class="subsection">11.45.47 add-extra-bond-restraint</h4>
<dl>
<dt><a name="index-add_002dextra_002dbond_002drestraint"></a>function: <strong>add-extra-bond-restraint</strong> <em>imol chain_id_1 res_no_1 ins_code_1 atom_name_1 alt_conf_1 chain_id_2 res_no_2 ins_code_2 atom_name_2 alt_conf_2 bond_dist esd</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id_1</em> is a string
</li><li> <em>res_no_1</em> is an integer number
</li><li> <em>ins_code_1</em> is a string
</li><li> <em>atom_name_1</em> is a string
</li><li> <em>alt_conf_1</em> is a string
</li><li> <em>chain_id_2</em> is a string
</li><li> <em>res_no_2</em> is an integer number
</li><li> <em>ins_code_2</em> is a string
</li><li> <em>atom_name_2</em> is a string
</li><li> <em>alt_conf_2</em> is a string
</li><li> <em>bond_dist</em> is a number
</li><li> <em>esd</em> is a number
</li></ul>

<p>add a user-define bond restraint 
</p>
<p>this extra restraint is used when the given atoms are selected in refinement or regularization.
</p>
<p>Returns: the index of the new restraint.
</p>
<p>Returns: -1 when the atoms were not found and no extra bond restraint was stored. 
</p>
</dd></dl>

<hr>
<a name="add_002dextra_002dgeman_002dmcclure_002drestraint"></a>
<div class="header">
<p>
Next: <a href="#delete_002dall_002dextra_002drestraints" accesskey="n" rel="next">delete-all-extra-restraints</a>, Previous: <a href="#add_002dextra_002dbond_002drestraint" accesskey="p" rel="previous">add-extra-bond-restraint</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dextra_002dgeman_002dmcclure_002drestraint-1"></a>
<h4 class="subsection">11.45.48 add-extra-geman-mcclure-restraint</h4>
<dl>
<dt><a name="index-add_002dextra_002dgeman_002dmcclure_002drestraint"></a>function: <strong>add-extra-geman-mcclure-restraint</strong> <em>imol chain_id_1 res_no_1 ins_code_1 atom_name_1 alt_conf_1 chain_id_2 res_no_2 ins_code_2 atom_name_2 alt_conf_2 bond_dist esd</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id_1</em> is a string
</li><li> <em>res_no_1</em> is an integer number
</li><li> <em>ins_code_1</em> is a string
</li><li> <em>atom_name_1</em> is a string
</li><li> <em>alt_conf_1</em> is a string
</li><li> <em>chain_id_2</em> is a string
</li><li> <em>res_no_2</em> is an integer number
</li><li> <em>ins_code_2</em> is a string
</li><li> <em>atom_name_2</em> is a string
</li><li> <em>alt_conf_2</em> is a string
</li><li> <em>bond_dist</em> is a number
</li><li> <em>esd</em> is a number
</li></ul>

<p>add a user-define GM distance restraint 
</p>
<p>this extra restraint is used when the given atoms are selected in refinement or regularization.
</p>
<p>Returns: the index of the new restraint.
</p>
<p>Returns: -1 when the atoms were not found and no extra bond restraint was stored. 
</p>
</dd></dl>

<hr>
<a name="delete_002dall_002dextra_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#delete_002dextra_002drestraints_002dfor_002dresidue" accesskey="n" rel="next">delete-extra-restraints-for-residue</a>, Previous: <a href="#add_002dextra_002dgeman_002dmcclure_002drestraint" accesskey="p" rel="previous">add-extra-geman-mcclure-restraint</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dall_002dextra_002drestraints-1"></a>
<h4 class="subsection">11.45.49 delete-all-extra-restraints</h4>
<dl>
<dt><a name="index-delete_002dall_002dextra_002drestraints"></a>function: <strong>delete-all-extra-restraints</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>clear out all the extra/user-defined restraints for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="delete_002dextra_002drestraints_002dfor_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits" accesskey="n" rel="next">set-extra-restraints-prosmart-sigma-limits</a>, Previous: <a href="#delete_002dall_002dextra_002drestraints" accesskey="p" rel="previous">delete-all-extra-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dextra_002drestraints_002dfor_002dresidue-1"></a>
<h4 class="subsection">11.45.50 delete-extra-restraints-for-residue</h4>
<dl>
<dt><a name="index-delete_002dextra_002drestraints_002dfor_002dresidue"></a>function: <strong>delete-extra-restraints-for-residue</strong> <em>imol chain_id res_no ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>clear out all the extra/user-defined restraints for this residue in molecule number imol 
</p>
</dd></dl>

<hr>
<a name="set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits"></a>
<div class="header">
<p>
Next: <a href="#generate_002dlocal_002dself_002drestraints" accesskey="n" rel="next">generate-local-self-restraints</a>, Previous: <a href="#delete_002dextra_002drestraints_002dfor_002dresidue" accesskey="p" rel="previous">delete-extra-restraints-for-residue</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits-1"></a>
<h4 class="subsection">11.45.51 set-extra-restraints-prosmart-sigma-limits</h4>
<dl>
<dt><a name="index-set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits"></a>function: <strong>set-extra-restraints-prosmart-sigma-limits</strong> <em>imol limit_high limit_low</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>limit_high</em> is a number
</li><li> <em>limit_low</em> is a number
</li></ul>

<p>often we don&rsquo;t want to see all prosmart restraints, just the (big) violations 
</p>
</dd></dl>

<hr>
<a name="generate_002dlocal_002dself_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#generate_002dself_002drestraints" accesskey="n" rel="next">generate-self-restraints</a>, Previous: <a href="#set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits" accesskey="p" rel="previous">set-extra-restraints-prosmart-sigma-limits</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="generate_002dlocal_002dself_002drestraints-1"></a>
<h4 class="subsection">11.45.52 generate-local-self-restraints</h4>
<dl>
<dt><a name="index-generate_002dlocal_002dself_002drestraints"></a>function: <strong>generate-local-self-restraints</strong> <em>imol chain_id local_dist_max</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>local_dist_max</em> is a number
</li></ul>

<p>generate external distance local self restraints 
</p>
</dd></dl>

<hr>
<a name="generate_002dself_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm" accesskey="n" rel="next">generate-local-self-restraints-by-residues-scm</a>, Previous: <a href="#generate_002dlocal_002dself_002drestraints" accesskey="p" rel="previous">generate-local-self-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="generate_002dself_002drestraints-1"></a>
<h4 class="subsection">11.45.53 generate-self-restraints</h4>
<dl>
<dt><a name="index-generate_002dself_002drestraints"></a>function: <strong>generate-self-restraints</strong> <em>imol local_dist_max</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>local_dist_max</em> is a number
</li></ul>

<p>generate external distance all-molecule self restraints 
</p>
</dd></dl>

<hr>
<a name="generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm"></a>
<div class="header">
<p>
Next: <a href="#write_002dinterpolated_002dextra_002drestraints" accesskey="n" rel="next">write-interpolated-extra-restraints</a>, Previous: <a href="#generate_002dself_002drestraints" accesskey="p" rel="previous">generate-self-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm-1"></a>
<h4 class="subsection">11.45.54 generate-local-self-restraints-by-residues-scm</h4>
<dl>
<dt><a name="index-generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm"></a>function: <strong>generate-local-self-restraints-by-residues-scm</strong> <em>imol residue_specs local_dist_max</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a SCM
</li><li> <em>local_dist_max</em> is a number
</li></ul>

<p>generate external distance self restraints for selected residues 
</p>
</dd></dl>

<hr>
<a name="write_002dinterpolated_002dextra_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints" accesskey="n" rel="next">write-interpolated-models-and-extra-restraints</a>, Previous: <a href="#generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm" accesskey="p" rel="previous">generate-local-self-restraints-by-residues-scm</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002dinterpolated_002dextra_002drestraints-1"></a>
<h4 class="subsection">11.45.55 write-interpolated-extra-restraints</h4>
<dl>
<dt><a name="index-write_002dinterpolated_002dextra_002drestraints"></a>function: <strong>write-interpolated-extra-restraints</strong> <em>imol_1 imol_2 n_steps file_name_stub</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_1</em> is an integer number
</li><li> <em>imol_2</em> is an integer number
</li><li> <em>n_steps</em> is an integer number
</li><li> <em>file_name_stub</em> is a string
</li></ul>

<p>proSMART interpolated restraints for model morphing 
</p>
</dd></dl>

<hr>
<a name="write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions" accesskey="n" rel="next">set-use-only-extra-torsion-restraints-for-torsions</a>, Previous: <a href="#write_002dinterpolated_002dextra_002drestraints" accesskey="p" rel="previous">write-interpolated-extra-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints-1"></a>
<h4 class="subsection">11.45.56 write-interpolated-models-and-extra-restraints</h4>
<dl>
<dt><a name="index-write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints"></a>function: <strong>write-interpolated-models-and-extra-restraints</strong> <em>imol_1 imol_2 n_steps file_name_stub interpolation_mode</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_1</em> is an integer number
</li><li> <em>imol_2</em> is an integer number
</li><li> <em>n_steps</em> is an integer number
</li><li> <em>file_name_stub</em> is a string
</li><li> <em>interpolation_mode</em> is an integer number
</li></ul>

<p>proSMART interpolated restraints for model morphing and write interpolated model 
</p>
<p>interpolation_mode is currently dummy - in due course I will addd torion angle interpolation. 
</p>
</dd></dl>

<hr>
<a name="set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions"></a>
<div class="header">
<p>
Next: <a href="#use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate" accesskey="n" rel="next">use-only-extra-torsion-restraints-for-torsions-state</a>, Previous: <a href="#write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints" accesskey="p" rel="previous">write-interpolated-models-and-extra-restraints</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions-1"></a>
<h4 class="subsection">11.45.57 set-use-only-extra-torsion-restraints-for-torsions</h4>
<dl>
<dt><a name="index-set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions"></a>function: <strong>set-use-only-extra-torsion-restraints-for-torsions</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set use only extra torsion restraints for torsions 
</p>
</dd></dl>

<hr>
<a name="use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate"></a>
<div class="header">
<p>
Next: <a href="#show_002drestraints_002deditor" accesskey="n" rel="next">show-restraints-editor</a>, Previous: <a href="#set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions" accesskey="p" rel="previous">set-use-only-extra-torsion-restraints-for-torsions</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate-1"></a>
<h4 class="subsection">11.45.58 use-only-extra-torsion-restraints-for-torsions-state</h4>
<dl>
<dt><a name="index-use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate"></a>function: <strong>use-only-extra-torsion-restraints-for-torsions-state</strong></dt>
<dd>
<p>return only-use-extra-torsion-restraints-for-torsions state 
</p>
</dd></dl>

<hr>
<a name="show_002drestraints_002deditor"></a>
<div class="header">
<p>
Next: <a href="#show_002drestraints_002deditor_002dby_002dindex" accesskey="n" rel="next">show-restraints-editor-by-index</a>, Previous: <a href="#use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate" accesskey="p" rel="previous">use-only-extra-torsion-restraints-for-torsions-state</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002drestraints_002deditor-1"></a>
<h4 class="subsection">11.45.59 show-restraints-editor</h4>
<dl>
<dt><a name="index-show_002drestraints_002deditor"></a>function: <strong>show-restraints-editor</strong> <em>monomer_type</em></dt>
<dd>
<p>Where <em>monomer_type</em> is a string
</p>
<p>show restraints editor 
</p>
</dd></dl>

<hr>
<a name="show_002drestraints_002deditor_002dby_002dindex"></a>
<div class="header">
<p>
Next: <a href="#write_002drestraints_002dcif_002ddictionary" accesskey="n" rel="next">write-restraints-cif-dictionary</a>, Previous: <a href="#show_002drestraints_002deditor" accesskey="p" rel="previous">show-restraints-editor</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002drestraints_002deditor_002dby_002dindex-1"></a>
<h4 class="subsection">11.45.60 show-restraints-editor-by-index</h4>
<dl>
<dt><a name="index-show_002drestraints_002deditor_002dby_002dindex"></a>function: <strong>show-restraints-editor-by-index</strong> <em>menu_item_index</em></dt>
<dd>
<p>Where <em>menu_item_index</em> is an integer number
</p>
<p>show restraints editor using residue type index 
</p>
</dd></dl>

<hr>
<a name="write_002drestraints_002dcif_002ddictionary"></a>
<div class="header">
<p>
Previous: <a href="#show_002drestraints_002deditor_002dby_002dindex" accesskey="p" rel="previous">show-restraints-editor-by-index</a>, Up: <a href="#Regularization-and-Refinement" accesskey="u" rel="up">Regularization and Refinement</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002drestraints_002dcif_002ddictionary-1"></a>
<h4 class="subsection">11.45.61 write-restraints-cif-dictionary</h4>
<dl>
<dt><a name="index-write_002drestraints_002dcif_002ddictionary"></a>function: <strong>write-restraints-cif-dictionary</strong> <em>monomer_type file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>monomer_type</em> is a string
</li><li> <em>file_name</em> is a string
</li></ul>

<p>write cif restraints for monomer 
</p>
</dd></dl>

<hr>
<a name="Simplex-Refinement-Interface"></a>
<div class="header">
<p>
Next: <a href="#Nomenclature-Errors" accesskey="n" rel="next">Nomenclature Errors</a>, Previous: <a href="#Regularization-and-Refinement" accesskey="p" rel="previous">Regularization and Refinement</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Simplex-Refinement-Interface-1"></a>
<h3 class="section">11.46 Simplex Refinement Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex" accesskey="1">fit-residue-range-to-map-by-simplex</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#score_002dresidue_002drange_002dfit_002dto_002dmap" accesskey="2">score-residue-range-fit-to-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex"></a>
<div class="header">
<p>
Next: <a href="#score_002dresidue_002drange_002dfit_002dto_002dmap" accesskey="n" rel="next">score-residue-range-fit-to-map</a>, Up: <a href="#Simplex-Refinement-Interface" accesskey="u" rel="up">Simplex Refinement Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex-1"></a>
<h4 class="subsection">11.46.1 fit-residue-range-to-map-by-simplex</h4>
<dl>
<dt><a name="index-fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex"></a>function: <strong>fit-residue-range-to-map-by-simplex</strong> <em>res1 res2 altloc chain_id imol imol_for_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>res1</em> is an integer number
</li><li> <em>res2</em> is an integer number
</li><li> <em>altloc</em> is a string
</li><li> <em>chain_id</em> is a string
</li><li> <em>imol</em> is an integer number
</li><li> <em>imol_for_map</em> is an integer number
</li></ul>

<p>refine residue range using simplex optimization 
</p>
</dd></dl>

<hr>
<a name="score_002dresidue_002drange_002dfit_002dto_002dmap"></a>
<div class="header">
<p>
Previous: <a href="#fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex" accesskey="p" rel="previous">fit-residue-range-to-map-by-simplex</a>, Up: <a href="#Simplex-Refinement-Interface" accesskey="u" rel="up">Simplex Refinement Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="score_002dresidue_002drange_002dfit_002dto_002dmap-1"></a>
<h4 class="subsection">11.46.2 score-residue-range-fit-to-map</h4>
<dl>
<dt><a name="index-score_002dresidue_002drange_002dfit_002dto_002dmap"></a>function: <strong>score-residue-range-fit-to-map</strong> <em>res1 res2 altloc chain_id imol imol_for_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>res1</em> is an integer number
</li><li> <em>res2</em> is an integer number
</li><li> <em>altloc</em> is a string
</li><li> <em>chain_id</em> is a string
</li><li> <em>imol</em> is an integer number
</li><li> <em>imol_for_map</em> is an integer number
</li></ul>

<p>simply score the residue range fit to map 
</p>
</dd></dl>

<hr>
<a name="Nomenclature-Errors"></a>
<div class="header">
<p>
Next: <a href="#Atom-Info-Interface" accesskey="n" rel="next">Atom Info  Interface</a>, Previous: <a href="#Simplex-Refinement-Interface" accesskey="p" rel="previous">Simplex Refinement Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Nomenclature-Errors-1"></a>
<h3 class="section">11.47 Nomenclature Errors</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#fix_002dnomenclature_002derrors" accesskey="1">fix-nomenclature-errors</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dnomenclature_002derrors_002don_002dread" accesskey="2">set-nomenclature-errors-on-read</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="fix_002dnomenclature_002derrors"></a>
<div class="header">
<p>
Next: <a href="#set_002dnomenclature_002derrors_002don_002dread" accesskey="n" rel="next">set-nomenclature-errors-on-read</a>, Up: <a href="#Nomenclature-Errors" accesskey="u" rel="up">Nomenclature Errors</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fix_002dnomenclature_002derrors-1"></a>
<h4 class="subsection">11.47.1 fix-nomenclature-errors</h4>
<dl>
<dt><a name="index-fix_002dnomenclature_002derrors"></a>function: <strong>fix-nomenclature-errors</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>fix nomenclature errors in molecule number imol 
</p>
<p>Returns: the number of resides altered. 
</p>
</dd></dl>

<hr>
<a name="set_002dnomenclature_002derrors_002don_002dread"></a>
<div class="header">
<p>
Previous: <a href="#fix_002dnomenclature_002derrors" accesskey="p" rel="previous">fix-nomenclature-errors</a>, Up: <a href="#Nomenclature-Errors" accesskey="u" rel="up">Nomenclature Errors</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dnomenclature_002derrors_002don_002dread-1"></a>
<h4 class="subsection">11.47.2 set-nomenclature-errors-on-read</h4>
<dl>
<dt><a name="index-set_002dnomenclature_002derrors_002don_002dread"></a>function: <strong>set-nomenclature-errors-on-read</strong> <em>mode</em></dt>
<dd>
<p>Where <em>mode</em> is a string
</p>
<p>set way nomenclature errors should be handled on reading coordinates. 
</p>
<p>mode should be &quot;auto-correct&quot;, &quot;ignore&quot;, &quot;prompt&quot;. The default is &quot;prompt&quot; 
</p>
</dd></dl>

<hr>
<a name="Atom-Info-Interface"></a>
<div class="header">
<p>
Next: <a href="#Residue-Info" accesskey="n" rel="next">Residue Info</a>, Previous: <a href="#Nomenclature-Errors" accesskey="p" rel="previous">Nomenclature Errors</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Atom-Info-Interface-1"></a>
<h3 class="section">11.48 Atom Info  Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#output_002datom_002dinfo_002das_002dtext" accesskey="1">output-atom-info-as-text</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="output_002datom_002dinfo_002das_002dtext"></a>
<div class="header">
<p>
Up: <a href="#Atom-Info-Interface" accesskey="u" rel="up">Atom Info  Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="output_002datom_002dinfo_002das_002dtext-1"></a>
<h4 class="subsection">11.48.1 output-atom-info-as-text</h4>
<dl>
<dt><a name="index-output_002datom_002dinfo_002das_002dtext"></a>function: <strong>output-atom-info-as-text</strong> <em>imol chain_id resno ins_code atname altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>atname</em> is a string
</li><li> <em>altconf</em> is a string
</li></ul>

<p>output to the terminal the Atom Info for the give atom specs 
</p>
</dd></dl>

<hr>
<a name="Residue-Info"></a>
<div class="header">
<p>
Next: <a href="#Residue-Environment-Functions" accesskey="n" rel="next">Residue Environment Functions</a>, Previous: <a href="#Atom-Info-Interface" accesskey="p" rel="previous">Atom Info  Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Residue-Info-1"></a>
<h3 class="section">11.49 Residue Info</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#residue_002dinfo_002ddialog" accesskey="1">residue-info-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="residue_002dinfo_002ddialog"></a>
<div class="header">
<p>
Up: <a href="#Residue-Info" accesskey="u" rel="up">Residue Info</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residue_002dinfo_002ddialog-1"></a>
<h4 class="subsection">11.49.1 residue-info-dialog</h4>
<dl>
<dt><a name="index-residue_002dinfo_002ddialog"></a>function: <strong>residue-info-dialog</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>show residue info dialog for given residue 
</p>
</dd></dl>

<hr>
<a name="Residue-Environment-Functions"></a>
<div class="header">
<p>
Next: <a href="#Pointer-Position-Function" accesskey="n" rel="next">Pointer Position Function</a>, Previous: <a href="#Residue-Info" accesskey="p" rel="previous">Residue Info</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Residue-Environment-Functions-1"></a>
<h3 class="section">11.50 Residue Environment Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002denvironment_002ddistances" accesskey="1">set-show-environment-distances</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002denvironment_002ddistances_002dbumps" accesskey="2">set-show-environment-distances-bumps</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002denvironment_002ddistances_002dh_002dbonds" accesskey="3">set-show-environment-distances-h-bonds</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002denvironment_002ddistances_002dstate" accesskey="4">show-environment-distances-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002denvironment_002ddistances_002ddistance_002dlimits" accesskey="5">set-environment-distances-distance-limits</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002denvironment_002ddistances_002das_002dsolid" accesskey="6">set-show-environment-distances-as-solid</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002denvironment_002ddistances_002dlabel_002datom" accesskey="7">set-environment-distances-label-atom</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#label_002dneighbours" accesskey="8">label-neighbours</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#label_002datoms_002din_002dresidue" accesskey="9">label-atoms-in-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dgeometry_002ddistance">add-geometry-distance</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dshow_002denvironment_002ddistances"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002denvironment_002ddistances_002dbumps" accesskey="n" rel="next">set-show-environment-distances-bumps</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002denvironment_002ddistances-1"></a>
<h4 class="subsection">11.50.1 set-show-environment-distances</h4>
<dl>
<dt><a name="index-set_002dshow_002denvironment_002ddistances"></a>function: <strong>set-show-environment-distances</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>show environment distances. If state is 0, distances are turned off, otherwise distances are turned on. 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002denvironment_002ddistances_002dbumps"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002denvironment_002ddistances_002dh_002dbonds" accesskey="n" rel="next">set-show-environment-distances-h-bonds</a>, Previous: <a href="#set_002dshow_002denvironment_002ddistances" accesskey="p" rel="previous">set-show-environment-distances</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002denvironment_002ddistances_002dbumps-1"></a>
<h4 class="subsection">11.50.2 set-show-environment-distances-bumps</h4>
<dl>
<dt><a name="index-set_002dshow_002denvironment_002ddistances_002dbumps"></a>function: <strong>set-show-environment-distances-bumps</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>show bumps environment distances. If state is 0, bump distances are turned off, otherwise bump distances are turned on. 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002denvironment_002ddistances_002dh_002dbonds"></a>
<div class="header">
<p>
Next: <a href="#show_002denvironment_002ddistances_002dstate" accesskey="n" rel="next">show-environment-distances-state</a>, Previous: <a href="#set_002dshow_002denvironment_002ddistances_002dbumps" accesskey="p" rel="previous">set-show-environment-distances-bumps</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002denvironment_002ddistances_002dh_002dbonds-1"></a>
<h4 class="subsection">11.50.3 set-show-environment-distances-h-bonds</h4>
<dl>
<dt><a name="index-set_002dshow_002denvironment_002ddistances_002dh_002dbonds"></a>function: <strong>set-show-environment-distances-h-bonds</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>show H-bond environment distances. If state is 0, bump distances are turned off, otherwise H-bond distances are turned on. 
</p>
</dd></dl>

<hr>
<a name="show_002denvironment_002ddistances_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002denvironment_002ddistances_002ddistance_002dlimits" accesskey="n" rel="next">set-environment-distances-distance-limits</a>, Previous: <a href="#set_002dshow_002denvironment_002ddistances_002dh_002dbonds" accesskey="p" rel="previous">set-show-environment-distances-h-bonds</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002denvironment_002ddistances_002dstate-1"></a>
<h4 class="subsection">11.50.4 show-environment-distances-state</h4>
<dl>
<dt><a name="index-show_002denvironment_002ddistances_002dstate"></a>function: <strong>show-environment-distances-state</strong></dt>
<dd>
<p>show the state of display of the environment distances 
</p>
</dd></dl>

<hr>
<a name="set_002denvironment_002ddistances_002ddistance_002dlimits"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002denvironment_002ddistances_002das_002dsolid" accesskey="n" rel="next">set-show-environment-distances-as-solid</a>, Previous: <a href="#show_002denvironment_002ddistances_002dstate" accesskey="p" rel="previous">show-environment-distances-state</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002denvironment_002ddistances_002ddistance_002dlimits-1"></a>
<h4 class="subsection">11.50.5 set-environment-distances-distance-limits</h4>
<dl>
<dt><a name="index-set_002denvironment_002ddistances_002ddistance_002dlimits"></a>function: <strong>set-environment-distances-distance-limits</strong> <em>min_dist max_dist</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>min_dist</em> is a number
</li><li> <em>max_dist</em> is a number
</li></ul>

<p>min and max distances for the environment distances 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002denvironment_002ddistances_002das_002dsolid"></a>
<div class="header">
<p>
Next: <a href="#set_002denvironment_002ddistances_002dlabel_002datom" accesskey="n" rel="next">set-environment-distances-label-atom</a>, Previous: <a href="#set_002denvironment_002ddistances_002ddistance_002dlimits" accesskey="p" rel="previous">set-environment-distances-distance-limits</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002denvironment_002ddistances_002das_002dsolid-1"></a>
<h4 class="subsection">11.50.6 set-show-environment-distances-as-solid</h4>
<dl>
<dt><a name="index-set_002dshow_002denvironment_002ddistances_002das_002dsolid"></a>function: <strong>set-show-environment-distances-as-solid</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>show the environment distances with solid modelling 
</p>
</dd></dl>

<hr>
<a name="set_002denvironment_002ddistances_002dlabel_002datom"></a>
<div class="header">
<p>
Next: <a href="#label_002dneighbours" accesskey="n" rel="next">label-neighbours</a>, Previous: <a href="#set_002dshow_002denvironment_002ddistances_002das_002dsolid" accesskey="p" rel="previous">set-show-environment-distances-as-solid</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002denvironment_002ddistances_002dlabel_002datom-1"></a>
<h4 class="subsection">11.50.7 set-environment-distances-label-atom</h4>
<dl>
<dt><a name="index-set_002denvironment_002ddistances_002dlabel_002datom"></a>function: <strong>set-environment-distances-label-atom</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>Label the atom on Environment Distances start/change. 
</p>
</dd></dl>

<hr>
<a name="label_002dneighbours"></a>
<div class="header">
<p>
Next: <a href="#label_002datoms_002din_002dresidue" accesskey="n" rel="next">label-atoms-in-residue</a>, Previous: <a href="#set_002denvironment_002ddistances_002dlabel_002datom" accesskey="p" rel="previous">set-environment-distances-label-atom</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="label_002dneighbours-1"></a>
<h4 class="subsection">11.50.8 label-neighbours</h4>
<dl>
<dt><a name="index-label_002dneighbours"></a>function: <strong>label-neighbours</strong></dt>
<dd>
<p>Label the atoms in the residues around the central residue. 
</p>
</dd></dl>

<hr>
<a name="label_002datoms_002din_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#add_002dgeometry_002ddistance" accesskey="n" rel="next">add-geometry-distance</a>, Previous: <a href="#label_002dneighbours" accesskey="p" rel="previous">label-neighbours</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="label_002datoms_002din_002dresidue-1"></a>
<h4 class="subsection">11.50.9 label-atoms-in-residue</h4>
<dl>
<dt><a name="index-label_002datoms_002din_002dresidue"></a>function: <strong>label-atoms-in-residue</strong></dt>
<dd>
<p>Label the atoms in the central residue. 
</p>
</dd></dl>

<hr>
<a name="add_002dgeometry_002ddistance"></a>
<div class="header">
<p>
Previous: <a href="#label_002datoms_002din_002dresidue" accesskey="p" rel="previous">label-atoms-in-residue</a>, Up: <a href="#Residue-Environment-Functions" accesskey="u" rel="up">Residue Environment Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dgeometry_002ddistance-1"></a>
<h4 class="subsection">11.50.10 add-geometry-distance</h4>
<dl>
<dt><a name="index-add_002dgeometry_002ddistance"></a>function: <strong>add-geometry-distance</strong> <em>imol_1 x_1 y_1 z_1 imol_2 x_2 y_2 z_2</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_1</em> is an integer number
</li><li> <em>x_1</em> is a number
</li><li> <em>y_1</em> is a number
</li><li> <em>z_1</em> is a number
</li><li> <em>imol_2</em> is an integer number
</li><li> <em>x_2</em> is a number
</li><li> <em>y_2</em> is a number
</li><li> <em>z_2</em> is a number
</li></ul>

<p>Add a geometry distance between points in a given molecule. 
</p>
<p>Returns: the distance between the points 
</p>
</dd></dl>

<hr>
<a name="Pointer-Position-Function"></a>
<div class="header">
<p>
Next: <a href="#Pointer-Functions" accesskey="n" rel="next">Pointer Functions</a>, Previous: <a href="#Residue-Environment-Functions" accesskey="p" rel="previous">Residue Environment Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Pointer-Position-Function-1"></a>
<h3 class="section">11.51 Pointer Position Function</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#get_002dpointer_002dposition_002dfrac_002dpy" accesskey="1">get-pointer-position-frac-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="get_002dpointer_002dposition_002dfrac_002dpy"></a>
<div class="header">
<p>
Up: <a href="#Pointer-Position-Function" accesskey="u" rel="up">Pointer Position Function</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dpointer_002dposition_002dfrac_002dpy-1"></a>
<h4 class="subsection">11.51.1 get-pointer-position-frac-py</h4>
<dl>
<dt><a name="index-get_002dpointer_002dposition_002dfrac_002dpy"></a>function: <strong>get-pointer-position-frac-py</strong></dt>
<dd>
<p>return the [x,y] position of the pointer in fractional coordinates. 
</p>
<p>the origin is top-left. may return false if pointer is not available 
</p>
</dd></dl>

<hr>
<a name="Pointer-Functions"></a>
<div class="header">
<p>
Next: <a href="#Zoom-Functions" accesskey="n" rel="next">Zoom Functions</a>, Previous: <a href="#Pointer-Position-Function" accesskey="p" rel="previous">Pointer Position Function</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Pointer-Functions-1"></a>
<h3 class="section">11.52 Pointer Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dpointer_002ddistances" accesskey="1">set-show-pointer-distances</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dpointer_002ddistances_002dstate" accesskey="2">show-pointer-distances-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dshow_002dpointer_002ddistances"></a>
<div class="header">
<p>
Next: <a href="#show_002dpointer_002ddistances_002dstate" accesskey="n" rel="next">show-pointer-distances-state</a>, Up: <a href="#Pointer-Functions" accesskey="u" rel="up">Pointer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dpointer_002ddistances-1"></a>
<h4 class="subsection">11.52.1 set-show-pointer-distances</h4>
<dl>
<dt><a name="index-set_002dshow_002dpointer_002ddistances"></a>function: <strong>set-show-pointer-distances</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>turn on (or off) the pointer distance by passing 1 (or 0). 
</p>
</dd></dl>

<hr>
<a name="show_002dpointer_002ddistances_002dstate"></a>
<div class="header">
<p>
Previous: <a href="#set_002dshow_002dpointer_002ddistances" accesskey="p" rel="previous">set-show-pointer-distances</a>, Up: <a href="#Pointer-Functions" accesskey="u" rel="up">Pointer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dpointer_002ddistances_002dstate-1"></a>
<h4 class="subsection">11.52.2 show-pointer-distances-state</h4>
<dl>
<dt><a name="index-show_002dpointer_002ddistances_002dstate"></a>function: <strong>show-pointer-distances-state</strong></dt>
<dd>
<p>show the state of display of the pointer distances 
</p>
</dd></dl>

<hr>
<a name="Zoom-Functions"></a>
<div class="header">
<p>
Next: <a href="#CNS-Data-Functions" accesskey="n" rel="next">CNS Data Functions</a>, Previous: <a href="#Pointer-Functions" accesskey="p" rel="previous">Pointer Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Zoom-Functions-1"></a>
<h3 class="section">11.53 Zoom Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#scale_002dzoom" accesskey="1">scale-zoom</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#zoom_002dfactor" accesskey="2">zoom-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsmooth_002dscroll_002ddo_002dzoom" accesskey="3">set-smooth-scroll-do-zoom</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#smooth_002dscroll_002ddo_002dzoom" accesskey="4">smooth-scroll-do-zoom</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="scale_002dzoom"></a>
<div class="header">
<p>
Next: <a href="#zoom_002dfactor" accesskey="n" rel="next">zoom-factor</a>, Up: <a href="#Zoom-Functions" accesskey="u" rel="up">Zoom Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="scale_002dzoom-1"></a>
<h4 class="subsection">11.53.1 scale-zoom</h4>
<dl>
<dt><a name="index-scale_002dzoom"></a>function: <strong>scale-zoom</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>scale the view by f 
</p>
<p>external (scripting) interface (with redraw) 
</p>
</dd></dl>

<hr>
<a name="zoom_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#set_002dsmooth_002dscroll_002ddo_002dzoom" accesskey="n" rel="next">set-smooth-scroll-do-zoom</a>, Previous: <a href="#scale_002dzoom" accesskey="p" rel="previous">scale-zoom</a>, Up: <a href="#Zoom-Functions" accesskey="u" rel="up">Zoom Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="zoom_002dfactor-1"></a>
<h4 class="subsection">11.53.2 zoom-factor</h4>
<dl>
<dt><a name="index-zoom_002dfactor"></a>function: <strong>zoom-factor</strong></dt>
<dd>
<p>return the current zoom factor 
</p>
</dd></dl>

<hr>
<a name="set_002dsmooth_002dscroll_002ddo_002dzoom"></a>
<div class="header">
<p>
Next: <a href="#smooth_002dscroll_002ddo_002dzoom" accesskey="n" rel="next">smooth-scroll-do-zoom</a>, Previous: <a href="#zoom_002dfactor" accesskey="p" rel="previous">zoom-factor</a>, Up: <a href="#Zoom-Functions" accesskey="u" rel="up">Zoom Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsmooth_002dscroll_002ddo_002dzoom-1"></a>
<h4 class="subsection">11.53.3 set-smooth-scroll-do-zoom</h4>
<dl>
<dt><a name="index-set_002dsmooth_002dscroll_002ddo_002dzoom"></a>function: <strong>set-smooth-scroll-do-zoom</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set smooth scroll with zoom 
</p>
</dd></dl>

<hr>
<a name="smooth_002dscroll_002ddo_002dzoom"></a>
<div class="header">
<p>
Previous: <a href="#set_002dsmooth_002dscroll_002ddo_002dzoom" accesskey="p" rel="previous">set-smooth-scroll-do-zoom</a>, Up: <a href="#Zoom-Functions" accesskey="u" rel="up">Zoom Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="smooth_002dscroll_002ddo_002dzoom-1"></a>
<h4 class="subsection">11.53.4 smooth-scroll-do-zoom</h4>
<dl>
<dt><a name="index-smooth_002dscroll_002ddo_002dzoom"></a>function: <strong>smooth-scroll-do-zoom</strong></dt>
<dd>
<p>return the state of the above system 
</p>
</dd></dl>

<hr>
<a name="CNS-Data-Functions"></a>
<div class="header">
<p>
Next: <a href="#mmCIF-Functions" accesskey="n" rel="next">mmCIF Functions</a>, Previous: <a href="#Zoom-Functions" accesskey="p" rel="previous">Zoom Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="CNS-Data-Functions-1"></a>
<h3 class="section">11.54 CNS Data Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#handle_002dcns_002ddata_002dfile" accesskey="1">handle-cns-data-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#handle_002dcns_002ddata_002dfile_002dwith_002dcell" accesskey="2">handle-cns-data-file-with-cell</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="handle_002dcns_002ddata_002dfile"></a>
<div class="header">
<p>
Next: <a href="#handle_002dcns_002ddata_002dfile_002dwith_002dcell" accesskey="n" rel="next">handle-cns-data-file-with-cell</a>, Up: <a href="#CNS-Data-Functions" accesskey="u" rel="up">CNS Data Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002dcns_002ddata_002dfile-1"></a>
<h4 class="subsection">11.54.1 handle-cns-data-file</h4>
<dl>
<dt><a name="index-handle_002dcns_002ddata_002dfile"></a>function: <strong>handle-cns-data-file</strong> <em>filename imol</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>filename</em> is a string
</li><li> <em>imol</em> is an integer number
</li></ul>

<p>read CNS data (currently only a placeholder) 
</p>
</dd></dl>

<hr>
<a name="handle_002dcns_002ddata_002dfile_002dwith_002dcell"></a>
<div class="header">
<p>
Previous: <a href="#handle_002dcns_002ddata_002dfile" accesskey="p" rel="previous">handle-cns-data-file</a>, Up: <a href="#CNS-Data-Functions" accesskey="u" rel="up">CNS Data Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002dcns_002ddata_002dfile_002dwith_002dcell-1"></a>
<h4 class="subsection">11.54.2 handle-cns-data-file-with-cell</h4>
<dl>
<dt><a name="index-handle_002dcns_002ddata_002dfile_002dwith_002dcell"></a>function: <strong>handle-cns-data-file-with-cell</strong> <em>filename imol a b c alpha beta gamma spg_info</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>filename</em> is a string
</li><li> <em>imol</em> is an integer number
</li><li> <em>a</em> is a number
</li><li> <em>b</em> is a number
</li><li> <em>c</em> is a number
</li><li> <em>alpha</em> is a number
</li><li> <em>beta</em> is a number
</li><li> <em>gamma</em> is a number
</li><li> <em>spg_info</em> is a string
</li></ul>

<p>read CNS data (currently only a placeholder) 
</p>
<p>a, b,c are in Angstroems. alpha, beta, gamma are in degrees. spg is the space group info, either ;-delimited symmetry operators or the space group name 
</p>
</dd></dl>

<hr>
<a name="mmCIF-Functions"></a>
<div class="header">
<p>
Next: <a href="#SHELXL-Functions" accesskey="n" rel="next">SHELXL Functions</a>, Previous: <a href="#CNS-Data-Functions" accesskey="p" rel="previous">CNS Data Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mmCIF-Functions-1"></a>
<h3 class="section">11.55 mmCIF Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#open_002dcif_002ddictionary_002dfile_002dselector_002ddialog" accesskey="1">open-cif-dictionary-file-selector-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#non_002dstandard_002dresidue_002dnames_002dscm" accesskey="2">non-standard-residue-names-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#non_002dstandard_002dresidue_002dnames_002dpy" accesskey="3">non-standard-residue-names-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="open_002dcif_002ddictionary_002dfile_002dselector_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#non_002dstandard_002dresidue_002dnames_002dscm" accesskey="n" rel="next">non-standard-residue-names-scm</a>, Up: <a href="#mmCIF-Functions" accesskey="u" rel="up">mmCIF Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="open_002dcif_002ddictionary_002dfile_002dselector_002ddialog-1"></a>
<h4 class="subsection">11.55.1 open-cif-dictionary-file-selector-dialog</h4>
<dl>
<dt><a name="index-open_002dcif_002ddictionary_002dfile_002dselector_002ddialog"></a>function: <strong>open-cif-dictionary-file-selector-dialog</strong></dt>
<dd>
<p>open the cif dictionary file selector dialog 
</p>
</dd></dl>

<hr>
<a name="non_002dstandard_002dresidue_002dnames_002dscm"></a>
<div class="header">
<p>
Next: <a href="#non_002dstandard_002dresidue_002dnames_002dpy" accesskey="n" rel="next">non-standard-residue-names-py</a>, Previous: <a href="#open_002dcif_002ddictionary_002dfile_002dselector_002ddialog" accesskey="p" rel="previous">open-cif-dictionary-file-selector-dialog</a>, Up: <a href="#mmCIF-Functions" accesskey="u" rel="up">mmCIF Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="non_002dstandard_002dresidue_002dnames_002dscm-1"></a>
<h4 class="subsection">11.55.2 non-standard-residue-names-scm</h4>
<dl>
<dt><a name="index-non_002dstandard_002dresidue_002dnames_002dscm"></a>function: <strong>non-standard-residue-names-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>non-standard residue/monomer names (note HOH is not non-standard). 
</p>
</dd></dl>

<hr>
<a name="non_002dstandard_002dresidue_002dnames_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#non_002dstandard_002dresidue_002dnames_002dscm" accesskey="p" rel="previous">non-standard-residue-names-scm</a>, Up: <a href="#mmCIF-Functions" accesskey="u" rel="up">mmCIF Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="non_002dstandard_002dresidue_002dnames_002dpy-1"></a>
<h4 class="subsection">11.55.3 non-standard-residue-names-py</h4>
<dl>
<dt><a name="index-non_002dstandard_002dresidue_002dnames_002dpy"></a>function: <strong>non-standard-residue-names-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>non-standard residue/monomer names (note HOH is not non-standard). 
</p>
</dd></dl>

<hr>
<a name="SHELXL-Functions"></a>
<div class="header">
<p>
Next: <a href="#Validation-Functions" accesskey="n" rel="next">Validation Functions</a>, Previous: <a href="#mmCIF-Functions" accesskey="p" rel="previous">mmCIF Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="SHELXL-Functions-1"></a>
<h3 class="section">11.56 SHELXL Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#read_002dshelx_002dins_002dfile" accesskey="1">read-shelx-ins-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#write_002dshelx_002dins_002dfile" accesskey="2">write-shelx-ins-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="read_002dshelx_002dins_002dfile"></a>
<div class="header">
<p>
Next: <a href="#write_002dshelx_002dins_002dfile" accesskey="n" rel="next">write-shelx-ins-file</a>, Up: <a href="#SHELXL-Functions" accesskey="u" rel="up">SHELXL Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="read_002dshelx_002dins_002dfile-1"></a>
<h4 class="subsection">11.56.1 read-shelx-ins-file</h4>
<dl>
<dt><a name="index-read_002dshelx_002dins_002dfile"></a>function: <strong>read-shelx-ins-file</strong> <em>filename recentre_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>filename</em> is a string
</li><li> <em>recentre_flag</em> is an integer number
</li></ul>

<p>read a SHELXL .ins file 
</p>
</dd></dl>

<hr>
<a name="write_002dshelx_002dins_002dfile"></a>
<div class="header">
<p>
Previous: <a href="#read_002dshelx_002dins_002dfile" accesskey="p" rel="previous">read-shelx-ins-file</a>, Up: <a href="#SHELXL-Functions" accesskey="u" rel="up">SHELXL Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002dshelx_002dins_002dfile-1"></a>
<h4 class="subsection">11.56.2 write-shelx-ins-file</h4>
<dl>
<dt><a name="index-write_002dshelx_002dins_002dfile"></a>function: <strong>write-shelx-ins-file</strong> <em>imol filename</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>filename</em> is a string
</li></ul>

<p>write a SHELXL .ins file for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="Validation-Functions"></a>
<div class="header">
<p>
Next: <a href="#Ramachandran-Plot-Functions" accesskey="n" rel="next">Ramachandran Plot Functions</a>, Previous: <a href="#SHELXL-Functions" accesskey="p" rel="previous">SHELXL Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Validation-Functions-1"></a>
<h3 class="section">11.57 Validation Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#difference_002dmap_002dpeaks" accesskey="1">difference-map-peaks</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#gln_002dasn_002db_002dfactor_002doutliers" accesskey="2">gln-asn-b-factor-outliers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dpeaks_002dpy" accesskey="3">map-peaks-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dpeaks_002dscm" accesskey="4">map-peaks-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dtorsion_002dscm" accesskey="5">set-torsion-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#multi_002dresidue_002dtorsion_002dscm" accesskey="6">multi-residue-torsion-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dtorsion_002dpy" accesskey="7">set-torsion-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#multi_002dresidue_002dtorsion_002dpy" accesskey="8">multi-residue-torsion-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="difference_002dmap_002dpeaks"></a>
<div class="header">
<p>
Next: <a href="#gln_002dasn_002db_002dfactor_002doutliers" accesskey="n" rel="next">gln-asn-b-factor-outliers</a>, Up: <a href="#Validation-Functions" accesskey="u" rel="up">Validation Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="difference_002dmap_002dpeaks-1"></a>
<h4 class="subsection">11.57.1 difference-map-peaks</h4>
<dl>
<dt><a name="index-difference_002dmap_002dpeaks"></a>function: <strong>difference-map-peaks</strong> <em>imol imol_coords level max_closeness do_positive_level_flag do_negative_level_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>imol_coords</em> is an integer number
</li><li> <em>level</em> is a number
</li><li> <em>max_closeness</em> is a number
</li><li> <em>do_positive_level_flag</em> is an integer number
</li><li> <em>do_negative_level_flag</em> is an integer number
</li></ul>

<p>generate a list of difference map peaks 
</p>
<p>peaks within max_closeness (2.0 A typically) of a larger peak are not listed. 
</p>
</dd></dl>

<hr>
<a name="gln_002dasn_002db_002dfactor_002doutliers"></a>
<div class="header">
<p>
Next: <a href="#map_002dpeaks_002dpy" accesskey="n" rel="next">map-peaks-py</a>, Previous: <a href="#difference_002dmap_002dpeaks" accesskey="p" rel="previous">difference-map-peaks</a>, Up: <a href="#Validation-Functions" accesskey="u" rel="up">Validation Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="gln_002dasn_002db_002dfactor_002doutliers-1"></a>
<h4 class="subsection">11.57.2 gln-asn-b-factor-outliers</h4>
<dl>
<dt><a name="index-gln_002dasn_002db_002dfactor_002doutliers"></a>function: <strong>gln-asn-b-factor-outliers</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Make a gui for GLN adn ASN B-factor outiers, compairing the O and N temperatur factors difference to the distribution of temperature factors from the other atoms. 
</p>
</dd></dl>

<hr>
<a name="map_002dpeaks_002dpy"></a>
<div class="header">
<p>
Next: <a href="#map_002dpeaks_002dscm" accesskey="n" rel="next">map-peaks-scm</a>, Previous: <a href="#gln_002dasn_002db_002dfactor_002doutliers" accesskey="p" rel="previous">gln-asn-b-factor-outliers</a>, Up: <a href="#Validation-Functions" accesskey="u" rel="up">Validation Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dpeaks_002dpy-1"></a>
<h4 class="subsection">11.57.3 map-peaks-py</h4>
<dl>
<dt><a name="index-map_002dpeaks_002dpy"></a>function: <strong>map-peaks-py</strong> <em>imol_map n_sigma</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>n_sigma</em> is a number
</li></ul>

<p>return a list of map peaks of molecule number imol_map above n_sigma. There will be cluster filtering of the map peaks. Return a list of 3d cartestian coordinates or Python False if imol_map is not suitable for peak picking. 
</p>
</dd></dl>

<hr>
<a name="map_002dpeaks_002dscm"></a>
<div class="header">
<p>
Next: <a href="#set_002dtorsion_002dscm" accesskey="n" rel="next">set-torsion-scm</a>, Previous: <a href="#map_002dpeaks_002dpy" accesskey="p" rel="previous">map-peaks-py</a>, Up: <a href="#Validation-Functions" accesskey="u" rel="up">Validation Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dpeaks_002dscm-1"></a>
<h4 class="subsection">11.57.4 map-peaks-scm</h4>
<dl>
<dt><a name="index-map_002dpeaks_002dscm"></a>function: <strong>map-peaks-scm</strong> <em>imol_map n_sigma</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>n_sigma</em> is a number
</li></ul>

<p>return a list of map peaks of molecule number imol_map above n_sigma. There will be cluster filtering of the map peaks. Return a list of 3d cartestian coordinates or scheme false if imol_map is not suitable for peak picking. 
</p>
</dd></dl>

<hr>
<a name="set_002dtorsion_002dscm"></a>
<div class="header">
<p>
Next: <a href="#multi_002dresidue_002dtorsion_002dscm" accesskey="n" rel="next">multi-residue-torsion-scm</a>, Previous: <a href="#map_002dpeaks_002dscm" accesskey="p" rel="previous">map-peaks-scm</a>, Up: <a href="#Validation-Functions" accesskey="u" rel="up">Validation Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dtorsion_002dscm-1"></a>
<h4 class="subsection">11.57.5 set-torsion-scm</h4>
<dl>
<dt><a name="index-set_002dtorsion_002dscm"></a>function: <strong>set-torsion-scm</strong> <em>imol chain_id res_no insertion_code alt_conf atom_name_1 atom_name_2 atom_name_3 atom_name_4 tors</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>insertion_code</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>atom_name_1</em> is a string
</li><li> <em>atom_name_2</em> is a string
</li><li> <em>atom_name_3</em> is a string
</li><li> <em>atom_name_4</em> is a string
</li><li> <em>tors</em> is a number
</li></ul>

<p>set the given torsion the given residue. tors is in degrees. Return the resulting torsion (also in degrees). 
</p>
</dd></dl>

<hr>
<a name="multi_002dresidue_002dtorsion_002dscm"></a>
<div class="header">
<p>
Next: <a href="#set_002dtorsion_002dpy" accesskey="n" rel="next">set-torsion-py</a>, Previous: <a href="#set_002dtorsion_002dscm" accesskey="p" rel="previous">set-torsion-scm</a>, Up: <a href="#Validation-Functions" accesskey="u" rel="up">Validation Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="multi_002dresidue_002dtorsion_002dscm-1"></a>
<h4 class="subsection">11.57.6 multi-residue-torsion-scm</h4>
<dl>
<dt><a name="index-multi_002dresidue_002dtorsion_002dscm"></a>function: <strong>multi-residue-torsion-scm</strong> <em>imol residues_specs_scm</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residues_specs_scm</em> is a SCM
</li></ul>

<p>create a multi-residue torsion dialog (user manipulation of torsions) 
</p>
</dd></dl>

<hr>
<a name="set_002dtorsion_002dpy"></a>
<div class="header">
<p>
Next: <a href="#multi_002dresidue_002dtorsion_002dpy" accesskey="n" rel="next">multi-residue-torsion-py</a>, Previous: <a href="#multi_002dresidue_002dtorsion_002dscm" accesskey="p" rel="previous">multi-residue-torsion-scm</a>, Up: <a href="#Validation-Functions" accesskey="u" rel="up">Validation Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dtorsion_002dpy-1"></a>
<h4 class="subsection">11.57.7 set-torsion-py</h4>
<dl>
<dt><a name="index-set_002dtorsion_002dpy"></a>function: <strong>set-torsion-py</strong> <em>imol chain_id res_no insertion_code alt_conf atom_name_1 atom_name_2 atom_name_3 atom_name_4 tors</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>insertion_code</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>atom_name_1</em> is a string
</li><li> <em>atom_name_2</em> is a string
</li><li> <em>atom_name_3</em> is a string
</li><li> <em>atom_name_4</em> is a string
</li><li> <em>tors</em> is a number
</li></ul>

<p>set the given torsion the given residue. tors is in degrees. Return the resulting torsion (also in degrees). 
</p>
</dd></dl>

<hr>
<a name="multi_002dresidue_002dtorsion_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#set_002dtorsion_002dpy" accesskey="p" rel="previous">set-torsion-py</a>, Up: <a href="#Validation-Functions" accesskey="u" rel="up">Validation Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="multi_002dresidue_002dtorsion_002dpy-1"></a>
<h4 class="subsection">11.57.8 multi-residue-torsion-py</h4>
<dl>
<dt><a name="index-multi_002dresidue_002dtorsion_002dpy"></a>function: <strong>multi-residue-torsion-py</strong> <em>imol residues_specs_py</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residues_specs_py</em> is a PyObject
</li></ul>

<p>create a multi-residue torsion dialog (user manipulation of torsions) 
</p>
</dd></dl>

<hr>
<a name="Ramachandran-Plot-Functions"></a>
<div class="header">
<p>
Next: <a href="#Sequence-View-Interface" accesskey="n" rel="next">Sequence View Interface</a>, Previous: <a href="#Validation-Functions" accesskey="p" rel="previous">Validation Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Ramachandran-Plot-Functions-1"></a>
<h3 class="section">11.58 Ramachandran Plot Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#do_002dramachandran_002dplot" accesskey="1">do-ramachandran-plot</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dkleywegt_002dplot_002dn_002ddiffs" accesskey="2">set-kleywegt-plot-n-diffs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dramachandran_002dplot_002dcontour_002dlevels" accesskey="3">set-ramachandran-plot-contour-levels</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dramachandran_002dplot_002dbackground_002dblock_002dsize" accesskey="4">set-ramachandran-plot-background-block-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dramachandran_002dpsi_002daxis_002dmode" accesskey="5">set-ramachandran-psi-axis-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ramachandran_002dplot_002ddifferences" accesskey="6">ramachandran-plot-differences</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ramachandran_002dplot_002ddifferences_002dby_002dchain" accesskey="7">ramachandran-plot-differences-by-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="do_002dramachandran_002dplot"></a>
<div class="header">
<p>
Next: <a href="#set_002dkleywegt_002dplot_002dn_002ddiffs" accesskey="n" rel="next">set-kleywegt-plot-n-diffs</a>, Up: <a href="#Ramachandran-Plot-Functions" accesskey="u" rel="up">Ramachandran Plot Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dramachandran_002dplot-1"></a>
<h4 class="subsection">11.58.1 do-ramachandran-plot</h4>
<dl>
<dt><a name="index-do_002dramachandran_002dplot"></a>function: <strong>do-ramachandran-plot</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Ramachandran plot for molecule number imol. 
</p>
</dd></dl>

<hr>
<a name="set_002dkleywegt_002dplot_002dn_002ddiffs"></a>
<div class="header">
<p>
Next: <a href="#set_002dramachandran_002dplot_002dcontour_002dlevels" accesskey="n" rel="next">set-ramachandran-plot-contour-levels</a>, Previous: <a href="#do_002dramachandran_002dplot" accesskey="p" rel="previous">do-ramachandran-plot</a>, Up: <a href="#Ramachandran-Plot-Functions" accesskey="u" rel="up">Ramachandran Plot Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dkleywegt_002dplot_002dn_002ddiffs-1"></a>
<h4 class="subsection">11.58.2 set-kleywegt-plot-n-diffs</h4>
<dl>
<dt><a name="index-set_002dkleywegt_002dplot_002dn_002ddiffs"></a>function: <strong>set-kleywegt-plot-n-diffs</strong> <em>n_diffs</em></dt>
<dd>
<p>Where <em>n_diffs</em> is an integer number
</p>
<p>set the number of biggest difference arrows on the Kleywegt plot. 
</p>
</dd></dl>

<hr>
<a name="set_002dramachandran_002dplot_002dcontour_002dlevels"></a>
<div class="header">
<p>
Next: <a href="#set_002dramachandran_002dplot_002dbackground_002dblock_002dsize" accesskey="n" rel="next">set-ramachandran-plot-background-block-size</a>, Previous: <a href="#set_002dkleywegt_002dplot_002dn_002ddiffs" accesskey="p" rel="previous">set-kleywegt-plot-n-diffs</a>, Up: <a href="#Ramachandran-Plot-Functions" accesskey="u" rel="up">Ramachandran Plot Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dramachandran_002dplot_002dcontour_002dlevels-1"></a>
<h4 class="subsection">11.58.3 set-ramachandran-plot-contour-levels</h4>
<dl>
<dt><a name="index-set_002dramachandran_002dplot_002dcontour_002dlevels"></a>function: <strong>set-ramachandran-plot-contour-levels</strong> <em>level_prefered level_allowed</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>level_prefered</em> is a number
</li><li> <em>level_allowed</em> is a number
</li></ul>

<p>set the contour levels for the ramachandran plot, default values are 0.02 (prefered) 0.002 (allowed) 
</p>
</dd></dl>

<hr>
<a name="set_002dramachandran_002dplot_002dbackground_002dblock_002dsize"></a>
<div class="header">
<p>
Next: <a href="#set_002dramachandran_002dpsi_002daxis_002dmode" accesskey="n" rel="next">set-ramachandran-psi-axis-mode</a>, Previous: <a href="#set_002dramachandran_002dplot_002dcontour_002dlevels" accesskey="p" rel="previous">set-ramachandran-plot-contour-levels</a>, Up: <a href="#Ramachandran-Plot-Functions" accesskey="u" rel="up">Ramachandran Plot Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dramachandran_002dplot_002dbackground_002dblock_002dsize-1"></a>
<h4 class="subsection">11.58.4 set-ramachandran-plot-background-block-size</h4>
<dl>
<dt><a name="index-set_002dramachandran_002dplot_002dbackground_002dblock_002dsize"></a>function: <strong>set-ramachandran-plot-background-block-size</strong> <em>blocksize</em></dt>
<dd>
<p>Where <em>blocksize</em> is a number
</p>
<p>set the ramachandran plot background block size. 
</p>
<p>Smaller is smoother but slower. Should be divisible exactly into
</p>
</dd></dl>

<hr>
<a name="set_002dramachandran_002dpsi_002daxis_002dmode"></a>
<div class="header">
<p>
Next: <a href="#ramachandran_002dplot_002ddifferences" accesskey="n" rel="next">ramachandran-plot-differences</a>, Previous: <a href="#set_002dramachandran_002dplot_002dbackground_002dblock_002dsize" accesskey="p" rel="previous">set-ramachandran-plot-background-block-size</a>, Up: <a href="#Ramachandran-Plot-Functions" accesskey="u" rel="up">Ramachandran Plot Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dramachandran_002dpsi_002daxis_002dmode-1"></a>
<h4 class="subsection">11.58.5 set-ramachandran-psi-axis-mode</h4>
<dl>
<dt><a name="index-set_002dramachandran_002dpsi_002daxis_002dmode"></a>function: <strong>set-ramachandran-psi-axis-mode</strong> <em>mode</em></dt>
<dd>
<p>Where <em>mode</em> is an integer number
</p>
<p>set the psi axis for the ramachandran plot. Default (0) from -180 to 180. Alternative (1) from -120 to 240. 
</p>
</dd></dl>

<hr>
<a name="ramachandran_002dplot_002ddifferences"></a>
<div class="header">
<p>
Next: <a href="#ramachandran_002dplot_002ddifferences_002dby_002dchain" accesskey="n" rel="next">ramachandran-plot-differences-by-chain</a>, Previous: <a href="#set_002dramachandran_002dpsi_002daxis_002dmode" accesskey="p" rel="previous">set-ramachandran-psi-axis-mode</a>, Up: <a href="#Ramachandran-Plot-Functions" accesskey="u" rel="up">Ramachandran Plot Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ramachandran_002dplot_002ddifferences-1"></a>
<h4 class="subsection">11.58.6 ramachandran-plot-differences</h4>
<dl>
<dt><a name="index-ramachandran_002dplot_002ddifferences"></a>function: <strong>ramachandran-plot-differences</strong> <em>imol1 imol2</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol1</em> is an integer number
</li><li> <em>imol2</em> is an integer number
</li></ul>

<p>2 molecule ramachandran plot (NCS differences) a.k.a. A Kleywegt Plot. 
</p>
</dd></dl>

<hr>
<a name="ramachandran_002dplot_002ddifferences_002dby_002dchain"></a>
<div class="header">
<p>
Previous: <a href="#ramachandran_002dplot_002ddifferences" accesskey="p" rel="previous">ramachandran-plot-differences</a>, Up: <a href="#Ramachandran-Plot-Functions" accesskey="u" rel="up">Ramachandran Plot Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ramachandran_002dplot_002ddifferences_002dby_002dchain-1"></a>
<h4 class="subsection">11.58.7 ramachandran-plot-differences-by-chain</h4>
<dl>
<dt><a name="index-ramachandran_002dplot_002ddifferences_002dby_002dchain"></a>function: <strong>ramachandran-plot-differences-by-chain</strong> <em>imol1 imol2 a_chain b_chain</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol1</em> is an integer number
</li><li> <em>imol2</em> is an integer number
</li><li> <em>a_chain</em> is a string
</li><li> <em>b_chain</em> is a string
</li></ul>

<p>A chain-specific Kleywegt Plot. 
</p>
</dd></dl>

<hr>
<a name="Sequence-View-Interface"></a>
<div class="header">
<p>
Next: <a href="#Atom-Labelling" accesskey="n" rel="next">Atom Labelling</a>, Previous: <a href="#Ramachandran-Plot-Functions" accesskey="p" rel="previous">Ramachandran Plot Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Sequence-View-Interface-1"></a>
<h3 class="section">11.59 Sequence View Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#do_002dsequence_002dview" accesskey="1">do-sequence-view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="do_002dsequence_002dview"></a>
<div class="header">
<p>
Up: <a href="#Sequence-View-Interface" accesskey="u" rel="up">Sequence View Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dsequence_002dview-1"></a>
<h4 class="subsection">11.59.1 do-sequence-view</h4>
<dl>
<dt><a name="index-do_002dsequence_002dview"></a>function: <strong>do-sequence-view</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>display the sequence view dialog for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="Atom-Labelling"></a>
<div class="header">
<p>
Next: <a href="#Screen-Rotation" accesskey="n" rel="next">Screen Rotation</a>, Previous: <a href="#Sequence-View-Interface" accesskey="p" rel="previous">Sequence View Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Atom-Labelling-1"></a>
<h3 class="section">11.60 Atom Labelling</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dbrief_002datom_002dlabels" accesskey="1">set-brief-atom-labels</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#brief_002datom_002dlabels_002dstate" accesskey="2">brief-atom-labels-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dseg_002dids_002din_002datom_002dlabels" accesskey="3">set-seg-ids-in-atom-labels</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dbrief_002datom_002dlabels"></a>
<div class="header">
<p>
Next: <a href="#brief_002datom_002dlabels_002dstate" accesskey="n" rel="next">brief-atom-labels-state</a>, Up: <a href="#Atom-Labelling" accesskey="u" rel="up">Atom Labelling</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbrief_002datom_002dlabels-1"></a>
<h4 class="subsection">11.60.1 set-brief-atom-labels</h4>
<dl>
<dt><a name="index-set_002dbrief_002datom_002dlabels"></a>function: <strong>set-brief-atom-labels</strong> <em>istat</em></dt>
<dd>
<p>Where <em>istat</em> is an integer number
</p>
<p>use brief atom names for on-screen labels 
</p>
<p>call with istat=1 to use brief labels, istat=0 for normal labels 
</p>
</dd></dl>

<hr>
<a name="brief_002datom_002dlabels_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dseg_002dids_002din_002datom_002dlabels" accesskey="n" rel="next">set-seg-ids-in-atom-labels</a>, Previous: <a href="#set_002dbrief_002datom_002dlabels" accesskey="p" rel="previous">set-brief-atom-labels</a>, Up: <a href="#Atom-Labelling" accesskey="u" rel="up">Atom Labelling</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="brief_002datom_002dlabels_002dstate-1"></a>
<h4 class="subsection">11.60.2 brief-atom-labels-state</h4>
<dl>
<dt><a name="index-brief_002datom_002dlabels_002dstate"></a>function: <strong>brief-atom-labels-state</strong></dt>
<dd>
<p>the brief atom label state 
</p>
</dd></dl>

<hr>
<a name="set_002dseg_002dids_002din_002datom_002dlabels"></a>
<div class="header">
<p>
Previous: <a href="#brief_002datom_002dlabels_002dstate" accesskey="p" rel="previous">brief-atom-labels-state</a>, Up: <a href="#Atom-Labelling" accesskey="u" rel="up">Atom Labelling</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dseg_002dids_002din_002datom_002dlabels-1"></a>
<h4 class="subsection">11.60.3 set-seg-ids-in-atom-labels</h4>
<dl>
<dt><a name="index-set_002dseg_002dids_002din_002datom_002dlabels"></a>function: <strong>set-seg-ids-in-atom-labels</strong> <em>istat</em></dt>
<dd>
<p>Where <em>istat</em> is an integer number
</p>
<p>set if brief atom labels should have seg-ids also 
</p>
</dd></dl>

<hr>
<a name="Screen-Rotation"></a>
<div class="header">
<p>
Next: <a href="#Screen-Translation" accesskey="n" rel="next">Screen Translation</a>, Previous: <a href="#Atom-Labelling" accesskey="p" rel="previous">Atom Labelling</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Screen-Rotation-1"></a>
<h3 class="section">11.61 Screen Rotation</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#rotate_002dy_002dscene" accesskey="1">rotate-y-scene</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#rotate_002dx_002dscene" accesskey="2">rotate-x-scene</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#rotate_002dz_002dscene" accesskey="3">rotate-z-scene</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#spin_002dzoom_002dtrans" accesskey="4">spin-zoom-trans</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="rotate_002dy_002dscene"></a>
<div class="header">
<p>
Next: <a href="#rotate_002dx_002dscene" accesskey="n" rel="next">rotate-x-scene</a>, Up: <a href="#Screen-Rotation" accesskey="u" rel="up">Screen Rotation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rotate_002dy_002dscene-1"></a>
<h4 class="subsection">11.61.1 rotate-y-scene</h4>
<dl>
<dt><a name="index-rotate_002dy_002dscene"></a>function: <strong>rotate-y-scene</strong> <em>nsteps stepsize</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>nsteps</em> is an integer number
</li><li> <em>stepsize</em> is a number
</li></ul>

<p>rotate view round y axis stepsize degrees for nstep such steps 
</p>
</dd></dl>

<hr>
<a name="rotate_002dx_002dscene"></a>
<div class="header">
<p>
Next: <a href="#rotate_002dz_002dscene" accesskey="n" rel="next">rotate-z-scene</a>, Previous: <a href="#rotate_002dy_002dscene" accesskey="p" rel="previous">rotate-y-scene</a>, Up: <a href="#Screen-Rotation" accesskey="u" rel="up">Screen Rotation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rotate_002dx_002dscene-1"></a>
<h4 class="subsection">11.61.2 rotate-x-scene</h4>
<dl>
<dt><a name="index-rotate_002dx_002dscene"></a>function: <strong>rotate-x-scene</strong> <em>nsteps stepsize</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>nsteps</em> is an integer number
</li><li> <em>stepsize</em> is a number
</li></ul>

<p>rotate view round x axis stepsize degrees for nstep such steps 
</p>
</dd></dl>

<hr>
<a name="rotate_002dz_002dscene"></a>
<div class="header">
<p>
Next: <a href="#spin_002dzoom_002dtrans" accesskey="n" rel="next">spin-zoom-trans</a>, Previous: <a href="#rotate_002dx_002dscene" accesskey="p" rel="previous">rotate-x-scene</a>, Up: <a href="#Screen-Rotation" accesskey="u" rel="up">Screen Rotation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rotate_002dz_002dscene-1"></a>
<h4 class="subsection">11.61.3 rotate-z-scene</h4>
<dl>
<dt><a name="index-rotate_002dz_002dscene"></a>function: <strong>rotate-z-scene</strong> <em>nsteps stepsize</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>nsteps</em> is an integer number
</li><li> <em>stepsize</em> is a number
</li></ul>

<p>rotate view round z axis stepsize degrees for nstep such steps 
</p>
</dd></dl>

<hr>
<a name="spin_002dzoom_002dtrans"></a>
<div class="header">
<p>
Previous: <a href="#rotate_002dz_002dscene" accesskey="p" rel="previous">rotate-z-scene</a>, Up: <a href="#Screen-Rotation" accesskey="u" rel="up">Screen Rotation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="spin_002dzoom_002dtrans-1"></a>
<h4 class="subsection">11.61.4 spin-zoom-trans</h4>
<dl>
<dt><a name="index-spin_002dzoom_002dtrans"></a>function: <strong>spin-zoom-trans</strong> <em>axis nstep stepsize zoom_by x_rel y_rel z_rel</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>axis</em> is an integer number
</li><li> <em>nstep</em> is an integer number
</li><li> <em>stepsize</em> is a number
</li><li> <em>zoom_by</em> is a number
</li><li> <em>x_rel</em> is a number
</li><li> <em>y_rel</em> is a number
</li><li> <em>z_rel</em> is a number
</li></ul>

<p>Bells and whistles rotation. 
</p>
<p>spin, zoom and translate.
</p>
<p>where axis is either x,y or z, stepsize is in degrees, zoom_by and x_rel etc are how much zoom, x,y,z should have changed by after nstep steps. 
</p>
</dd></dl>

<hr>
<a name="Screen-Translation"></a>
<div class="header">
<p>
Next: <a href="#Views-Interface" accesskey="n" rel="next">Views Interface</a>, Previous: <a href="#Screen-Rotation" accesskey="p" rel="previous">Screen Rotation</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Screen-Translation-1"></a>
<h3 class="section">11.62 Screen Translation</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#translate_002dscene_002dx" accesskey="1">translate-scene-x</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#translate_002dscene_002dy" accesskey="2">translate-scene-y</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#translate_002dscene_002dz" accesskey="3">translate-scene-z</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="translate_002dscene_002dx"></a>
<div class="header">
<p>
Next: <a href="#translate_002dscene_002dy" accesskey="n" rel="next">translate-scene-y</a>, Up: <a href="#Screen-Translation" accesskey="u" rel="up">Screen Translation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="translate_002dscene_002dx-1"></a>
<h4 class="subsection">11.62.1 translate-scene-x</h4>
<dl>
<dt><a name="index-translate_002dscene_002dx"></a>function: <strong>translate-scene-x</strong> <em>nsteps</em></dt>
<dd>
<p>Where <em>nsteps</em> is an integer number
</p>
<p>translate rotation centre relative to screen axes for nsteps 
</p>
</dd></dl>

<hr>
<a name="translate_002dscene_002dy"></a>
<div class="header">
<p>
Next: <a href="#translate_002dscene_002dz" accesskey="n" rel="next">translate-scene-z</a>, Previous: <a href="#translate_002dscene_002dx" accesskey="p" rel="previous">translate-scene-x</a>, Up: <a href="#Screen-Translation" accesskey="u" rel="up">Screen Translation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="translate_002dscene_002dy-1"></a>
<h4 class="subsection">11.62.2 translate-scene-y</h4>
<dl>
<dt><a name="index-translate_002dscene_002dy"></a>function: <strong>translate-scene-y</strong> <em>nsteps</em></dt>
<dd>
<p>Where <em>nsteps</em> is an integer number
</p>
<p>translate rotation centre relative to screen axes for nsteps 
</p>
</dd></dl>

<hr>
<a name="translate_002dscene_002dz"></a>
<div class="header">
<p>
Previous: <a href="#translate_002dscene_002dy" accesskey="p" rel="previous">translate-scene-y</a>, Up: <a href="#Screen-Translation" accesskey="u" rel="up">Screen Translation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="translate_002dscene_002dz-1"></a>
<h4 class="subsection">11.62.3 translate-scene-z</h4>
<dl>
<dt><a name="index-translate_002dscene_002dz"></a>function: <strong>translate-scene-z</strong> <em>nsteps</em></dt>
<dd>
<p>Where <em>nsteps</em> is an integer number
</p>
<p>translate rotation centre relative to screen axes for nsteps 
</p>
</dd></dl>

<hr>
<a name="Views-Interface"></a>
<div class="header">
<p>
Next: <a href="#Background-Colour" accesskey="n" rel="next">Background Colour</a>, Previous: <a href="#Screen-Translation" accesskey="p" rel="previous">Screen Translation</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Views-Interface-1"></a>
<h3 class="section">11.63 Views Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#add_002dview_002dhere" accesskey="1">add-view-here</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dview_002draw" accesskey="2">add-view-raw</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002dnamed_002dview" accesskey="3">remove-named-view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002dview" accesskey="4">remove-view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dview_002ddescription" accesskey="5">add-view-description</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002daction_002dview" accesskey="6">add-action-view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#insert_002daction_002dview_002dafter_002dview" accesskey="7">insert-action-view-after-view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#save_002dviews" accesskey="8">save-views</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#view_002dname" accesskey="9">view-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#view_002dname_002dpy">view-name-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clear_002dall_002dviews">clear-all-views</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="add_002dview_002dhere"></a>
<div class="header">
<p>
Next: <a href="#add_002dview_002draw" accesskey="n" rel="next">add-view-raw</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dview_002dhere-1"></a>
<h4 class="subsection">11.63.1 add-view-here</h4>
<dl>
<dt><a name="index-add_002dview_002dhere"></a>function: <strong>add-view-here</strong> <em>view_name</em></dt>
<dd>
<p>Where <em>view_name</em> is a string
</p>
<p>return the view number 
</p>
</dd></dl>

<hr>
<a name="add_002dview_002draw"></a>
<div class="header">
<p>
Next: <a href="#remove_002dnamed_002dview" accesskey="n" rel="next">remove-named-view</a>, Previous: <a href="#add_002dview_002dhere" accesskey="p" rel="previous">add-view-here</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dview_002draw-1"></a>
<h4 class="subsection">11.63.2 add-view-raw</h4>
<dl>
<dt><a name="index-add_002dview_002draw"></a>function: <strong>add-view-raw</strong> <em>rcx rcy rcz quat1 quat2 quat3 quat4 zoom view_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>rcx</em> is a number
</li><li> <em>rcy</em> is a number
</li><li> <em>rcz</em> is a number
</li><li> <em>quat1</em> is a number
</li><li> <em>quat2</em> is a number
</li><li> <em>quat3</em> is a number
</li><li> <em>quat4</em> is a number
</li><li> <em>zoom</em> is a number
</li><li> <em>view_name</em> is a string
</li></ul>

<p>return the view number 
</p>
</dd></dl>

<hr>
<a name="remove_002dnamed_002dview"></a>
<div class="header">
<p>
Next: <a href="#remove_002dview" accesskey="n" rel="next">remove-view</a>, Previous: <a href="#add_002dview_002draw" accesskey="p" rel="previous">add-view-raw</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002dnamed_002dview-1"></a>
<h4 class="subsection">11.63.3 remove-named-view</h4>
<dl>
<dt><a name="index-remove_002dnamed_002dview"></a>function: <strong>remove-named-view</strong> <em>view_name</em></dt>
<dd>
<p>Where <em>view_name</em> is a string
</p>
<p>the view with the given name 
</p>
</dd></dl>

<hr>
<a name="remove_002dview"></a>
<div class="header">
<p>
Next: <a href="#add_002dview_002ddescription" accesskey="n" rel="next">add-view-description</a>, Previous: <a href="#remove_002dnamed_002dview" accesskey="p" rel="previous">remove-named-view</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002dview-1"></a>
<h4 class="subsection">11.63.4 remove-view</h4>
<dl>
<dt><a name="index-remove_002dview"></a>function: <strong>remove-view</strong> <em>view_number</em></dt>
<dd>
<p>Where <em>view_number</em> is an integer number
</p>
<p>the given view number 
</p>
</dd></dl>

<hr>
<a name="add_002dview_002ddescription"></a>
<div class="header">
<p>
Next: <a href="#add_002daction_002dview" accesskey="n" rel="next">add-action-view</a>, Previous: <a href="#remove_002dview" accesskey="p" rel="previous">remove-view</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dview_002ddescription-1"></a>
<h4 class="subsection">11.63.5 add-view-description</h4>
<dl>
<dt><a name="index-add_002dview_002ddescription"></a>function: <strong>add-view-description</strong> <em>view_number description</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>view_number</em> is an integer number
</li><li> <em>description</em> is a string
</li></ul>

<p>Add a view description/annotation to the give view number. 
</p>
</dd></dl>

<hr>
<a name="add_002daction_002dview"></a>
<div class="header">
<p>
Next: <a href="#insert_002daction_002dview_002dafter_002dview" accesskey="n" rel="next">insert-action-view-after-view</a>, Previous: <a href="#add_002dview_002ddescription" accesskey="p" rel="previous">add-view-description</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002daction_002dview-1"></a>
<h4 class="subsection">11.63.6 add-action-view</h4>
<dl>
<dt><a name="index-add_002daction_002dview"></a>function: <strong>add-action-view</strong> <em>view_name action_function</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>view_name</em> is a string
</li><li> <em>action_function</em> is a string
</li></ul>

<p>add a view (not add to an existing view) that does something (e.g. displays or undisplays a molecule) rather than move the graphics. 
</p>
<p>Returns: the view number for this (new) view. 
</p>
</dd></dl>

<hr>
<a name="insert_002daction_002dview_002dafter_002dview"></a>
<div class="header">
<p>
Next: <a href="#save_002dviews" accesskey="n" rel="next">save-views</a>, Previous: <a href="#add_002daction_002dview" accesskey="p" rel="previous">add-action-view</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="insert_002daction_002dview_002dafter_002dview-1"></a>
<h4 class="subsection">11.63.7 insert-action-view-after-view</h4>
<dl>
<dt><a name="index-insert_002daction_002dview_002dafter_002dview"></a>function: <strong>insert-action-view-after-view</strong> <em>view_number view_name action_function</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>view_number</em> is an integer number
</li><li> <em>view_name</em> is a string
</li><li> <em>action_function</em> is a string
</li></ul>

<p>add an action view after the view of the given view number 
</p>
<p>Returns: the view number for this (new) view. 
</p>
</dd></dl>

<hr>
<a name="save_002dviews"></a>
<div class="header">
<p>
Next: <a href="#view_002dname" accesskey="n" rel="next">view-name</a>, Previous: <a href="#insert_002daction_002dview_002dafter_002dview" accesskey="p" rel="previous">insert-action-view-after-view</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="save_002dviews-1"></a>
<h4 class="subsection">11.63.8 save-views</h4>
<dl>
<dt><a name="index-save_002dviews"></a>function: <strong>save-views</strong> <em>view_file_name</em></dt>
<dd>
<p>Where <em>view_file_name</em> is a string
</p>
<p>save views to view_file_name 
</p>
</dd></dl>

<hr>
<a name="view_002dname"></a>
<div class="header">
<p>
Next: <a href="#view_002dname_002dpy" accesskey="n" rel="next">view-name-py</a>, Previous: <a href="#save_002dviews" accesskey="p" rel="previous">save-views</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="view_002dname-1"></a>
<h4 class="subsection">11.63.9 view-name</h4>
<dl>
<dt><a name="index-view_002dname"></a>function: <strong>view-name</strong> <em>view_number</em></dt>
<dd>
<p>Where <em>view_number</em> is an integer number
</p>
<p>return the name of the given view, if view_number does not specify a view return scheme value False 
</p>
</dd></dl>

<hr>
<a name="view_002dname_002dpy"></a>
<div class="header">
<p>
Next: <a href="#clear_002dall_002dviews" accesskey="n" rel="next">clear-all-views</a>, Previous: <a href="#view_002dname" accesskey="p" rel="previous">view-name</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="view_002dname_002dpy-1"></a>
<h4 class="subsection">11.63.10 view-name-py</h4>
<dl>
<dt><a name="index-view_002dname_002dpy"></a>function: <strong>view-name-py</strong> <em>view_number</em></dt>
<dd>
<p>Where <em>view_number</em> is an integer number
</p>
<p>return the name of the given view, if view_number does not specify a view return Python value False 
</p>
</dd></dl>

<hr>
<a name="clear_002dall_002dviews"></a>
<div class="header">
<p>
Previous: <a href="#view_002dname_002dpy" accesskey="p" rel="previous">view-name-py</a>, Up: <a href="#Views-Interface" accesskey="u" rel="up">Views Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002dall_002dviews-1"></a>
<h4 class="subsection">11.63.11 clear-all-views</h4>
<dl>
<dt><a name="index-clear_002dall_002dviews"></a>function: <strong>clear-all-views</strong></dt>
<dd>
<p>Clear the view list. 
</p>
</dd></dl>

<hr>
<a name="Background-Colour"></a>
<div class="header">
<p>
Next: <a href="#Ligand-Fitting-Functions" accesskey="n" rel="next">Ligand Fitting Functions</a>, Previous: <a href="#Views-Interface" accesskey="p" rel="previous">Views Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Background-Colour-1"></a>
<h3 class="section">11.64 Background Colour</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dbackground_002dcolour" accesskey="1">set-background-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#redraw_002dbackground" accesskey="2">redraw-background</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#background_002dis_002dblack_002dp" accesskey="3">background-is-black-p</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dbackground_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#redraw_002dbackground" accesskey="n" rel="next">redraw-background</a>, Up: <a href="#Background-Colour" accesskey="u" rel="up">Background Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbackground_002dcolour-1"></a>
<h4 class="subsection">11.64.1 set-background-colour</h4>
<dl>
<dt><a name="index-set_002dbackground_002dcolour"></a>function: <strong>set-background-colour</strong> <em>red green blue</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>red</em> is a number
</li><li> <em>green</em> is a number
</li><li> <em>blue</em> is a number
</li></ul>

<p>set the background colour 
</p>
<p>red, green and blue are numbers between 0.0 and 1.0 
</p>
</dd></dl>

<hr>
<a name="redraw_002dbackground"></a>
<div class="header">
<p>
Next: <a href="#background_002dis_002dblack_002dp" accesskey="n" rel="next">background-is-black-p</a>, Previous: <a href="#set_002dbackground_002dcolour" accesskey="p" rel="previous">set-background-colour</a>, Up: <a href="#Background-Colour" accesskey="u" rel="up">Background Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="redraw_002dbackground-1"></a>
<h4 class="subsection">11.64.2 redraw-background</h4>
<dl>
<dt><a name="index-redraw_002dbackground"></a>function: <strong>redraw-background</strong></dt>
<dd>
<p>re draw the background colour when switching between mono and stereo 
</p>
</dd></dl>

<hr>
<a name="background_002dis_002dblack_002dp"></a>
<div class="header">
<p>
Previous: <a href="#redraw_002dbackground" accesskey="p" rel="previous">redraw-background</a>, Up: <a href="#Background-Colour" accesskey="u" rel="up">Background Colour</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="background_002dis_002dblack_002dp-1"></a>
<h4 class="subsection">11.64.3 background-is-black-p</h4>
<dl>
<dt><a name="index-background_002dis_002dblack_002dp"></a>function: <strong>background-is-black-p</strong></dt>
<dd>
<p>is the background black (or nearly black)? 
</p>
<p>Returns: 1 if the background is black (or nearly black), else return 0. 
</p>
</dd></dl>

<hr>
<a name="Ligand-Fitting-Functions"></a>
<div class="header">
<p>
Next: <a href="#Water-Fitting-Functions" accesskey="n" rel="next">Water Fitting Functions</a>, Previous: <a href="#Background-Colour" accesskey="p" rel="previous">Background Colour</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Ligand-Fitting-Functions-1"></a>
<h3 class="section">11.65 Ligand Fitting Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dligand_002dacceptable_002dfit_002dfraction" accesskey="1">set-ligand-acceptable-fit-fraction</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dligand_002dcluster_002dsigma_002dlevel" accesskey="2">set-ligand-cluster-sigma-level</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dligand_002dflexible_002dligand_002dn_002dsamples" accesskey="3">set-ligand-flexible-ligand-n-samples</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfind_002dligand_002dn_002dtop_002dligands" accesskey="4">set-find-ligand-n-top-ligands</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster" accesskey="5">set-find-ligand-multi-solutions-per-cluster</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfind_002dligand_002dmask_002dwaters" accesskey="6">set-find-ligand-mask-waters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dligand_002dsearch_002dprotein_002dmolecule" accesskey="7">set-ligand-search-protein-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dligand_002dsearch_002dmap_002dmolecule" accesskey="8">set-ligand-search-map-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dligand_002dsearch_002dligand_002dmolecule" accesskey="9">add-ligand-search-ligand-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule">add-ligand-search-wiggly-ligand-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfind_002dligand_002dhere_002dcluster">set-find-ligand-here-cluster</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ligand_002dexpert">ligand-expert</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#do_002dfind_002dligands_002ddialog">do-find-ligands-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#overlap_002dligands">overlap-ligands</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#match_002dligand_002datom_002dnames">match-ligand-atom-names</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#match_002dligand_002datom_002dnames_002dto_002dcomp_002did">match-ligand-atom-names-to-comp-id</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#flip_002dligand">flip-ligand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dligand_002dacceptable_002dfit_002dfraction"></a>
<div class="header">
<p>
Next: <a href="#set_002dligand_002dcluster_002dsigma_002dlevel" accesskey="n" rel="next">set-ligand-cluster-sigma-level</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dligand_002dacceptable_002dfit_002dfraction-1"></a>
<h4 class="subsection">11.65.1 set-ligand-acceptable-fit-fraction</h4>
<dl>
<dt><a name="index-set_002dligand_002dacceptable_002dfit_002dfraction"></a>function: <strong>set-ligand-acceptable-fit-fraction</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the fraction of atoms which must be in positive density after a ligand fit 
</p>
</dd></dl>

<hr>
<a name="set_002dligand_002dcluster_002dsigma_002dlevel"></a>
<div class="header">
<p>
Next: <a href="#set_002dligand_002dflexible_002dligand_002dn_002dsamples" accesskey="n" rel="next">set-ligand-flexible-ligand-n-samples</a>, Previous: <a href="#set_002dligand_002dacceptable_002dfit_002dfraction" accesskey="p" rel="previous">set-ligand-acceptable-fit-fraction</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dligand_002dcluster_002dsigma_002dlevel-1"></a>
<h4 class="subsection">11.65.2 set-ligand-cluster-sigma-level</h4>
<dl>
<dt><a name="index-set_002dligand_002dcluster_002dsigma_002dlevel"></a>function: <strong>set-ligand-cluster-sigma-level</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the default sigma level that the map is searched to find potential ligand sites 
</p>
</dd></dl>

<hr>
<a name="set_002dligand_002dflexible_002dligand_002dn_002dsamples"></a>
<div class="header">
<p>
Next: <a href="#set_002dfind_002dligand_002dn_002dtop_002dligands" accesskey="n" rel="next">set-find-ligand-n-top-ligands</a>, Previous: <a href="#set_002dligand_002dcluster_002dsigma_002dlevel" accesskey="p" rel="previous">set-ligand-cluster-sigma-level</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dligand_002dflexible_002dligand_002dn_002dsamples-1"></a>
<h4 class="subsection">11.65.3 set-ligand-flexible-ligand-n-samples</h4>
<dl>
<dt><a name="index-set_002dligand_002dflexible_002dligand_002dn_002dsamples"></a>function: <strong>set-ligand-flexible-ligand-n-samples</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set the number of conformation samples 
</p>
<p>big ligands require more samples. Default 10. 
</p>
</dd></dl>

<hr>
<a name="set_002dfind_002dligand_002dn_002dtop_002dligands"></a>
<div class="header">
<p>
Next: <a href="#set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster" accesskey="n" rel="next">set-find-ligand-multi-solutions-per-cluster</a>, Previous: <a href="#set_002dligand_002dflexible_002dligand_002dn_002dsamples" accesskey="p" rel="previous">set-ligand-flexible-ligand-n-samples</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfind_002dligand_002dn_002dtop_002dligands-1"></a>
<h4 class="subsection">11.65.4 set-find-ligand-n-top-ligands</h4>
<dl>
<dt><a name="index-set_002dfind_002dligand_002dn_002dtop_002dligands"></a>function: <strong>set-find-ligand-n-top-ligands</strong> <em>n</em></dt>
<dd>
<p>Where <em>n</em> is an integer number
</p>
<p>search the top n sites for ligands. 
</p>
<p>Default 10. 
</p>
</dd></dl>

<hr>
<a name="set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster"></a>
<div class="header">
<p>
Next: <a href="#set_002dfind_002dligand_002dmask_002dwaters" accesskey="n" rel="next">set-find-ligand-mask-waters</a>, Previous: <a href="#set_002dfind_002dligand_002dn_002dtop_002dligands" accesskey="p" rel="previous">set-find-ligand-n-top-ligands</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster-1"></a>
<h4 class="subsection">11.65.5 set-find-ligand-multi-solutions-per-cluster</h4>
<dl>
<dt><a name="index-set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster"></a>function: <strong>set-find-ligand-multi-solutions-per-cluster</strong> <em>lim_1 lim_2</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>lim_1</em> is a number
</li><li> <em>lim_2</em> is a number
</li></ul>

<p>allow multiple ligand solutions per cluster. 
</p>
<p>The first limit is the fraction of the top scored positions that go on to correlation scoring (closer to 1 means less and faster - default 0.7).
</p>
<p>The second limit is the fraction of the top correlation score that is considered interesting. Limits the number of solutions displayed to user. Default 0.9.
</p>
<p>There is currently no chi-angle set redundancy filtering - I suspect that there should be.
</p>
<p>Nino-mode. 
</p>
</dd></dl>

<hr>
<a name="set_002dfind_002dligand_002dmask_002dwaters"></a>
<div class="header">
<p>
Next: <a href="#set_002dligand_002dsearch_002dprotein_002dmolecule" accesskey="n" rel="next">set-ligand-search-protein-molecule</a>, Previous: <a href="#set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster" accesskey="p" rel="previous">set-find-ligand-multi-solutions-per-cluster</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfind_002dligand_002dmask_002dwaters-1"></a>
<h4 class="subsection">11.65.6 set-find-ligand-mask-waters</h4>
<dl>
<dt><a name="index-set_002dfind_002dligand_002dmask_002dwaters"></a>function: <strong>set-find-ligand-mask-waters</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>how shall we treat the waters during ligand fitting? 
</p>
<p>pass with istate=1 for waters to mask the map in the same way that protein atoms do. 
</p>
</dd></dl>

<hr>
<a name="set_002dligand_002dsearch_002dprotein_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#set_002dligand_002dsearch_002dmap_002dmolecule" accesskey="n" rel="next">set-ligand-search-map-molecule</a>, Previous: <a href="#set_002dfind_002dligand_002dmask_002dwaters" accesskey="p" rel="previous">set-find-ligand-mask-waters</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dligand_002dsearch_002dprotein_002dmolecule-1"></a>
<h4 class="subsection">11.65.7 set-ligand-search-protein-molecule</h4>
<dl>
<dt><a name="index-set_002dligand_002dsearch_002dprotein_002dmolecule"></a>function: <strong>set-ligand-search-protein-molecule</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>set the protein molecule for ligand searching 
</p>
</dd></dl>

<hr>
<a name="set_002dligand_002dsearch_002dmap_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#add_002dligand_002dsearch_002dligand_002dmolecule" accesskey="n" rel="next">add-ligand-search-ligand-molecule</a>, Previous: <a href="#set_002dligand_002dsearch_002dprotein_002dmolecule" accesskey="p" rel="previous">set-ligand-search-protein-molecule</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dligand_002dsearch_002dmap_002dmolecule-1"></a>
<h4 class="subsection">11.65.8 set-ligand-search-map-molecule</h4>
<dl>
<dt><a name="index-set_002dligand_002dsearch_002dmap_002dmolecule"></a>function: <strong>set-ligand-search-map-molecule</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>set the map molecule for ligand searching 
</p>
</dd></dl>

<hr>
<a name="add_002dligand_002dsearch_002dligand_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule" accesskey="n" rel="next">add-ligand-search-wiggly-ligand-molecule</a>, Previous: <a href="#set_002dligand_002dsearch_002dmap_002dmolecule" accesskey="p" rel="previous">set-ligand-search-map-molecule</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dligand_002dsearch_002dligand_002dmolecule-1"></a>
<h4 class="subsection">11.65.9 add-ligand-search-ligand-molecule</h4>
<dl>
<dt><a name="index-add_002dligand_002dsearch_002dligand_002dmolecule"></a>function: <strong>add-ligand-search-ligand-molecule</strong> <em>imol_ligand</em></dt>
<dd>
<p>Where <em>imol_ligand</em> is an integer number
</p>
<p>add a rigid ligand molecule to the list of ligands to search for in ligand searching 
</p>
</dd></dl>

<hr>
<a name="add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#set_002dfind_002dligand_002dhere_002dcluster" accesskey="n" rel="next">set-find-ligand-here-cluster</a>, Previous: <a href="#add_002dligand_002dsearch_002dligand_002dmolecule" accesskey="p" rel="previous">add-ligand-search-ligand-molecule</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule-1"></a>
<h4 class="subsection">11.65.10 add-ligand-search-wiggly-ligand-molecule</h4>
<dl>
<dt><a name="index-add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule"></a>function: <strong>add-ligand-search-wiggly-ligand-molecule</strong> <em>imol_ligand</em></dt>
<dd>
<p>Where <em>imol_ligand</em> is an integer number
</p>
<p>add a flexible ligand molecule to the list of ligands to search for in ligand searching 
</p>
</dd></dl>

<hr>
<a name="set_002dfind_002dligand_002dhere_002dcluster"></a>
<div class="header">
<p>
Next: <a href="#ligand_002dexpert" accesskey="n" rel="next">ligand-expert</a>, Previous: <a href="#add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule" accesskey="p" rel="previous">add-ligand-search-wiggly-ligand-molecule</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfind_002dligand_002dhere_002dcluster-1"></a>
<h4 class="subsection">11.65.11 set-find-ligand-here-cluster</h4>
<dl>
<dt><a name="index-set_002dfind_002dligand_002dhere_002dcluster"></a>function: <strong>set-find-ligand-here-cluster</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>Allow the user a scripting means to find ligand at the rotation centre. 
</p>
</dd></dl>

<hr>
<a name="ligand_002dexpert"></a>
<div class="header">
<p>
Next: <a href="#do_002dfind_002dligands_002ddialog" accesskey="n" rel="next">do-find-ligands-dialog</a>, Previous: <a href="#set_002dfind_002dligand_002dhere_002dcluster" accesskey="p" rel="previous">set-find-ligand-here-cluster</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ligand_002dexpert-1"></a>
<h4 class="subsection">11.65.12 ligand-expert</h4>
<dl>
<dt><a name="index-ligand_002dexpert"></a>function: <strong>ligand-expert</strong></dt>
<dd>
<p>this sets the flag to have expert option ligand entries in the Ligand Searching dialog 
</p>
</dd></dl>

<hr>
<a name="do_002dfind_002dligands_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#overlap_002dligands" accesskey="n" rel="next">overlap-ligands</a>, Previous: <a href="#ligand_002dexpert" accesskey="p" rel="previous">ligand-expert</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dfind_002dligands_002ddialog-1"></a>
<h4 class="subsection">11.65.13 do-find-ligands-dialog</h4>
<dl>
<dt><a name="index-do_002dfind_002dligands_002ddialog"></a>function: <strong>do-find-ligands-dialog</strong></dt>
<dd>
<p>display the find ligands dialog 
</p>
<p>if maps, coords and ligands are available, that is. 
</p>
</dd></dl>

<hr>
<a name="overlap_002dligands"></a>
<div class="header">
<p>
Next: <a href="#match_002dligand_002datom_002dnames" accesskey="n" rel="next">match-ligand-atom-names</a>, Previous: <a href="#do_002dfind_002dligands_002ddialog" accesskey="p" rel="previous">do-find-ligands-dialog</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="overlap_002dligands-1"></a>
<h4 class="subsection">11.65.14 overlap-ligands</h4>
<dl>
<dt><a name="index-overlap_002dligands"></a>function: <strong>overlap-ligands</strong> <em>imol_ligand imol_ref chain_id_ref resno_ref</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_ligand</em> is an integer number
</li><li> <em>imol_ref</em> is an integer number
</li><li> <em>chain_id_ref</em> is a string
</li><li> <em>resno_ref</em> is an integer number
</li></ul>

<p>Overlap residue with &quot;template&quot;-based matching. 
</p>
<p>Overlap the first residue in imol_ligand onto the residue specified by the reference parameters. Use graph matching, not atom names.
</p>
<p>Returns: success status, False = failed to find residue in either imol_ligand or imo_ref. If success, return the RT operator. 
</p>
</dd></dl>

<hr>
<a name="match_002dligand_002datom_002dnames"></a>
<div class="header">
<p>
Next: <a href="#match_002dligand_002datom_002dnames_002dto_002dcomp_002did" accesskey="n" rel="next">match-ligand-atom-names-to-comp-id</a>, Previous: <a href="#overlap_002dligands" accesskey="p" rel="previous">overlap-ligands</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="match_002dligand_002datom_002dnames-1"></a>
<h4 class="subsection">11.65.15 match-ligand-atom-names</h4>
<dl>
<dt><a name="index-match_002dligand_002datom_002dnames"></a>function: <strong>match-ligand-atom-names</strong> <em>imol_ligand chain_id_ligand resno_ligand ins_code_ligand imol_reference chain_id_reference resno_reference ins_code_reference</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_ligand</em> is an integer number
</li><li> <em>chain_id_ligand</em> is a string
</li><li> <em>resno_ligand</em> is an integer number
</li><li> <em>ins_code_ligand</em> is a string
</li><li> <em>imol_reference</em> is an integer number
</li><li> <em>chain_id_reference</em> is a string
</li><li> <em>resno_reference</em> is an integer number
</li><li> <em>ins_code_reference</em> is a string
</li></ul>

<p>Match ligand atom names. 
</p>
<p>By using graph matching, make the names of the atoms of the given ligand/residue match those of the reference residue/ligand as closely as possible - where there would be an atom name clash, invent a new atom name. 
</p>
</dd></dl>

<hr>
<a name="match_002dligand_002datom_002dnames_002dto_002dcomp_002did"></a>
<div class="header">
<p>
Next: <a href="#flip_002dligand" accesskey="n" rel="next">flip-ligand</a>, Previous: <a href="#match_002dligand_002datom_002dnames" accesskey="p" rel="previous">match-ligand-atom-names</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="match_002dligand_002datom_002dnames_002dto_002dcomp_002did-1"></a>
<h4 class="subsection">11.65.16 match-ligand-atom-names-to-comp-id</h4>
<dl>
<dt><a name="index-match_002dligand_002datom_002dnames_002dto_002dcomp_002did"></a>function: <strong>match-ligand-atom-names-to-comp-id</strong> <em>imol_ligand chain_id_ligand resno_ligand ins_code_ligand comp_id_ref</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_ligand</em> is an integer number
</li><li> <em>chain_id_ligand</em> is a string
</li><li> <em>resno_ligand</em> is an integer number
</li><li> <em>ins_code_ligand</em> is a string
</li><li> <em>comp_id_ref</em> is a string
</li></ul>

<p>Match ligand atom names to a reference ligand type (comp_id) 
</p>
<p>By using graph matching, make the names of the atoms of the given ligand/residue match those of the reference ligand from the geometry store as closely as possible. Where there would be an atom name clash, invent a new atom name.
</p>
<p>This doesn&rsquo;t create a new dictionary for the selected ligand - and that&rsquo;s a big problem (see match_residue_and_dictionary). 
</p>
</dd></dl>

<hr>
<a name="flip_002dligand"></a>
<div class="header">
<p>
Previous: <a href="#match_002dligand_002datom_002dnames_002dto_002dcomp_002did" accesskey="p" rel="previous">match-ligand-atom-names-to-comp-id</a>, Up: <a href="#Ligand-Fitting-Functions" accesskey="u" rel="up">Ligand Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="flip_002dligand-1"></a>
<h4 class="subsection">11.65.17 flip-ligand</h4>
<dl>
<dt><a name="index-flip_002dligand"></a>function: <strong>flip-ligand</strong> <em>imol chain_id resno</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li></ul>

<p>flip the ligand (usually active residue) around its eigen vectors to the next flip number. Immediate replacement (like flip peptide). 
</p>
</dd></dl>

<hr>
<a name="Water-Fitting-Functions"></a>
<div class="header">
<p>
Next: <a href="#Bond-Representation" accesskey="n" rel="next">Bond Representation</a>, Previous: <a href="#Ligand-Fitting-Functions" accesskey="p" rel="previous">Ligand Fitting Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Water-Fitting-Functions-1"></a>
<h3 class="section">11.66 Water Fitting Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#wrapped_002dcreate_002dfind_002dwaters_002ddialog" accesskey="1">wrapped-create-find-waters-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#renumber_002dwaters" accesskey="2">renumber-waters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#execute_002dfind_002dwaters_002dreal" accesskey="3">execute-find-waters-real</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#move_002dwaters_002dto_002daround_002dprotein" accesskey="4">move-waters-to-around-protein</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#move_002dhetgroups_002dto_002daround_002dprotein" accesskey="5">move-hetgroups-to-around-protein</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#max_002dwater_002ddistance" accesskey="6">max-water-distance</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit" accesskey="7">set-water-check-spherical-variance-limit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits" accesskey="8">set-ligand-water-to-protein-distance-limits</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dligand_002dwater_002dn_002dcycles" accesskey="9">set-ligand-water-n-cycles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#execute_002dfind_002dblobs">execute-find-blobs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#split_002dwater">split-water</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="wrapped_002dcreate_002dfind_002dwaters_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#renumber_002dwaters" accesskey="n" rel="next">renumber-waters</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="wrapped_002dcreate_002dfind_002dwaters_002ddialog-1"></a>
<h4 class="subsection">11.66.1 wrapped-create-find-waters-dialog</h4>
<dl>
<dt><a name="index-wrapped_002dcreate_002dfind_002dwaters_002ddialog"></a>function: <strong>wrapped-create-find-waters-dialog</strong></dt>
<dd>
<p>create a dialog for water fitting 
</p>
</dd></dl>

<hr>
<a name="renumber_002dwaters"></a>
<div class="header">
<p>
Next: <a href="#execute_002dfind_002dwaters_002dreal" accesskey="n" rel="next">execute-find-waters-real</a>, Previous: <a href="#wrapped_002dcreate_002dfind_002dwaters_002ddialog" accesskey="p" rel="previous">wrapped-create-find-waters-dialog</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="renumber_002dwaters-1"></a>
<h4 class="subsection">11.66.2 renumber-waters</h4>
<dl>
<dt><a name="index-renumber_002dwaters"></a>function: <strong>renumber-waters</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Renumber the waters of molecule number imol with consecutive numbering. 
</p>
</dd></dl>

<hr>
<a name="execute_002dfind_002dwaters_002dreal"></a>
<div class="header">
<p>
Next: <a href="#move_002dwaters_002dto_002daround_002dprotein" accesskey="n" rel="next">move-waters-to-around-protein</a>, Previous: <a href="#renumber_002dwaters" accesskey="p" rel="previous">renumber-waters</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="execute_002dfind_002dwaters_002dreal-1"></a>
<h4 class="subsection">11.66.3 execute-find-waters-real</h4>
<dl>
<dt><a name="index-execute_002dfind_002dwaters_002dreal"></a>function: <strong>execute-find-waters-real</strong> <em>imol_for_map imol_for_protein new_waters_mol_flag rmsd_cut_off</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_for_map</em> is an integer number
</li><li> <em>imol_for_protein</em> is an integer number
</li><li> <em>new_waters_mol_flag</em> is an integer number
</li><li> <em>rmsd_cut_off</em> is a number
</li></ul>

<p>find waters 
</p>
</dd></dl>

<hr>
<a name="move_002dwaters_002dto_002daround_002dprotein"></a>
<div class="header">
<p>
Next: <a href="#move_002dhetgroups_002dto_002daround_002dprotein" accesskey="n" rel="next">move-hetgroups-to-around-protein</a>, Previous: <a href="#execute_002dfind_002dwaters_002dreal" accesskey="p" rel="previous">execute-find-waters-real</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="move_002dwaters_002dto_002daround_002dprotein-1"></a>
<h4 class="subsection">11.66.4 move-waters-to-around-protein</h4>
<dl>
<dt><a name="index-move_002dwaters_002dto_002daround_002dprotein"></a>function: <strong>move-waters-to-around-protein</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>move waters of molecule number imol so that they are around the protein. 
</p>
<p>Returns: the number of moved waters. 
</p>
</dd></dl>

<hr>
<a name="move_002dhetgroups_002dto_002daround_002dprotein"></a>
<div class="header">
<p>
Next: <a href="#max_002dwater_002ddistance" accesskey="n" rel="next">max-water-distance</a>, Previous: <a href="#move_002dwaters_002dto_002daround_002dprotein" accesskey="p" rel="previous">move-waters-to-around-protein</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="move_002dhetgroups_002dto_002daround_002dprotein-1"></a>
<h4 class="subsection">11.66.5 move-hetgroups-to-around-protein</h4>
<dl>
<dt><a name="index-move_002dhetgroups_002dto_002daround_002dprotein"></a>function: <strong>move-hetgroups-to-around-protein</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>move all hetgroups (including waters) of molecule number imol so that they are around the protein. 
</p>
</dd></dl>

<hr>
<a name="max_002dwater_002ddistance"></a>
<div class="header">
<p>
Next: <a href="#set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit" accesskey="n" rel="next">set-water-check-spherical-variance-limit</a>, Previous: <a href="#move_002dhetgroups_002dto_002daround_002dprotein" accesskey="p" rel="previous">move-hetgroups-to-around-protein</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="max_002dwater_002ddistance-1"></a>
<h4 class="subsection">11.66.6 max-water-distance</h4>
<dl>
<dt><a name="index-max_002dwater_002ddistance"></a>function: <strong>max-water-distance</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the maximum minimum distance of any water atom to any protein atom - used in validation of move_waters_to_around_protein() funtion. 
</p>
</dd></dl>

<hr>
<a name="set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit"></a>
<div class="header">
<p>
Next: <a href="#set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits" accesskey="n" rel="next">set-ligand-water-to-protein-distance-limits</a>, Previous: <a href="#max_002dwater_002ddistance" accesskey="p" rel="previous">max-water-distance</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit-1"></a>
<h4 class="subsection">11.66.7 set-water-check-spherical-variance-limit</h4>
<dl>
<dt><a name="index-set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit"></a>function: <strong>set-water-check-spherical-variance-limit</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the limit of interesting variance, above which waters are listed (otherwise ignored) 
</p>
<p>default 0.12. 
</p>
</dd></dl>

<hr>
<a name="set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits"></a>
<div class="header">
<p>
Next: <a href="#set_002dligand_002dwater_002dn_002dcycles" accesskey="n" rel="next">set-ligand-water-n-cycles</a>, Previous: <a href="#set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit" accesskey="p" rel="previous">set-water-check-spherical-variance-limit</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits-1"></a>
<h4 class="subsection">11.66.8 set-ligand-water-to-protein-distance-limits</h4>
<dl>
<dt><a name="index-set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits"></a>function: <strong>set-ligand-water-to-protein-distance-limits</strong> <em>f1 f2</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>f1</em> is a number
</li><li> <em>f2</em> is a number
</li></ul>

<p>set ligand to protein distance limits 
</p>
<p>f1 is the minimum distance, f2 is the maximum distance 
</p>
</dd></dl>

<hr>
<a name="set_002dligand_002dwater_002dn_002dcycles"></a>
<div class="header">
<p>
Next: <a href="#execute_002dfind_002dblobs" accesskey="n" rel="next">execute-find-blobs</a>, Previous: <a href="#set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits" accesskey="p" rel="previous">set-ligand-water-to-protein-distance-limits</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dligand_002dwater_002dn_002dcycles-1"></a>
<h4 class="subsection">11.66.9 set-ligand-water-n-cycles</h4>
<dl>
<dt><a name="index-set_002dligand_002dwater_002dn_002dcycles"></a>function: <strong>set-ligand-water-n-cycles</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set the number of cycles of water searching 
</p>
</dd></dl>

<hr>
<a name="execute_002dfind_002dblobs"></a>
<div class="header">
<p>
Next: <a href="#split_002dwater" accesskey="n" rel="next">split-water</a>, Previous: <a href="#set_002dligand_002dwater_002dn_002dcycles" accesskey="p" rel="previous">set-ligand-water-n-cycles</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="execute_002dfind_002dblobs-1"></a>
<h4 class="subsection">11.66.10 execute-find-blobs</h4>
<dl>
<dt><a name="index-execute_002dfind_002dblobs"></a>function: <strong>execute-find-blobs</strong> <em>imol_model imol_for_map cut_off interactive_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_model</em> is an integer number
</li><li> <em>imol_for_map</em> is an integer number
</li><li> <em>cut_off</em> is a number
</li><li> <em>interactive_flag</em> is an integer number
</li></ul>

<p>find blobs 
</p>
</dd></dl>

<hr>
<a name="split_002dwater"></a>
<div class="header">
<p>
Previous: <a href="#execute_002dfind_002dblobs" accesskey="p" rel="previous">execute-find-blobs</a>, Up: <a href="#Water-Fitting-Functions" accesskey="u" rel="up">Water Fitting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="split_002dwater-1"></a>
<h4 class="subsection">11.66.11 split-water</h4>
<dl>
<dt><a name="index-split_002dwater"></a>function: <strong>split-water</strong> <em>imol chain_id res_no ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>split the given water and fit to map. 
</p>
<p>If refinement map is not defined, don&rsquo;t do anything.
</p>
<p>If there is more than one atom in the specified resiue, don&rsquo;t do anything.
</p>
<p>If the given atom does not have an alt conf of &quot;&quot;, don&rsquo;t do anything. 
</p>
</dd></dl>

<hr>
<a name="Bond-Representation"></a>
<div class="header">
<p>
Next: <a href="#Dots-Representation" accesskey="n" rel="next">Dots Representation</a>, Previous: <a href="#Water-Fitting-Functions" accesskey="p" rel="previous">Water Fitting Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Bond-Representation-1"></a>
<h3 class="section">11.67 Bond Representation</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002ddefault_002dbond_002dthickness" accesskey="1">set-default-bond-thickness</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dbond_002dthickness" accesskey="2">set-bond-thickness</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dbond_002dthickness_002dintermediate_002datoms" accesskey="3">set-bond-thickness-intermediate-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002duse_002dvariable_002dbond_002dthickness" accesskey="4">set-use-variable-bond-thickness</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dbond_002dcolour_002drotation_002dfor_002dmolecule" accesskey="5">set-bond-colour-rotation-for-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dstick_002dmode_002datoms_002ddefault" accesskey="6">set-draw-stick-mode-atoms-default</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dbond_002dcolour_002drotation_002dfor_002dmolecule" accesskey="7">get-bond-colour-rotation-for-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddefault_002drepresentation_002dtype" accesskey="8">set-default-representation-type</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002ddefault_002dbond_002dthickness" accesskey="9">get-default-bond-thickness</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dzero_002docc_002dmarkers">set-draw-zero-occ-markers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dcis_002dpeptide_002dmarkups">set-draw-cis-peptide-markups</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dhydrogens">set-draw-hydrogens</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#draw_002dhydrogens_002dstate">draw-hydrogens-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dstick_002dmode_002datoms">set-draw-stick-mode-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dmissing_002dresidues_002dloops">set-draw-missing-residues-loops</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002dca_002drepresentation">graphics-to-ca-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002dca_002dplus_002dligands_002drepresentation">graphics-to-ca-plus-ligands-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation">graphics-to-ca-plus-ligands-and-sidechains-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation">graphics-to-bonds-no-waters-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002dbonds_002drepresentation">graphics-to-bonds-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation">graphics-to-ca-plus-ligands-sec-struct-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation">graphics-to-sec-struct-bonds-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002drainbow_002drepresentation">graphics-to-rainbow-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002db_002dfactor_002drepresentation">graphics-to-b-factor-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002db_002dfactor_002dcas_002drepresentation">graphics-to-b-factor-cas-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002doccupancy_002drepresentation">graphics-to-occupancy-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation">graphics-to-user-defined-atom-colours-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation">graphics-to-user-defined-atom-colours-all-atoms-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#graphics_002dmolecule_002dbond_002dtype">graphics-molecule-bond-type</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002db_002dfactor_002dbonds_002dscale_002dfactor">set-b-factor-bonds-scale-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#change_002dmodel_002dmolecule_002drepresentation_002dmode">change-model-molecule-representation-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule">set-use-grey-carbons-for-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dgrey_002dcarbon_002dcolour">set-grey-carbon-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dball_002dand_002dstick">make-ball-and-stick</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clear_002dball_002dand_002dstick">clear-ball-and-stick</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#additional_002drepresentation_002dby_002dstring">additional-representation-by-string</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#additional_002drepresentation_002dby_002dattributes">additional-representation-by-attributes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002ddefault_002dbond_002dthickness"></a>
<div class="header">
<p>
Next: <a href="#set_002dbond_002dthickness" accesskey="n" rel="next">set-bond-thickness</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddefault_002dbond_002dthickness-1"></a>
<h4 class="subsection">11.67.1 set-default-bond-thickness</h4>
<dl>
<dt><a name="index-set_002ddefault_002dbond_002dthickness"></a>function: <strong>set-default-bond-thickness</strong> <em>t</em></dt>
<dd>
<p>Where <em>t</em> is an integer number
</p>
<p>set the default thickness for bonds (e.g. in ~/.coot) 
</p>
</dd></dl>

<hr>
<a name="set_002dbond_002dthickness"></a>
<div class="header">
<p>
Next: <a href="#set_002dbond_002dthickness_002dintermediate_002datoms" accesskey="n" rel="next">set-bond-thickness-intermediate-atoms</a>, Previous: <a href="#set_002ddefault_002dbond_002dthickness" accesskey="p" rel="previous">set-default-bond-thickness</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbond_002dthickness-1"></a>
<h4 class="subsection">11.67.2 set-bond-thickness</h4>
<dl>
<dt><a name="index-set_002dbond_002dthickness"></a>function: <strong>set-bond-thickness</strong> <em>imol t</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>t</em> is a number
</li></ul>

<p>set the thickness of the bonds in molecule number imol to t pixels 
</p>
</dd></dl>

<hr>
<a name="set_002dbond_002dthickness_002dintermediate_002datoms"></a>
<div class="header">
<p>
Next: <a href="#set_002duse_002dvariable_002dbond_002dthickness" accesskey="n" rel="next">set-use-variable-bond-thickness</a>, Previous: <a href="#set_002dbond_002dthickness" accesskey="p" rel="previous">set-bond-thickness</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbond_002dthickness_002dintermediate_002datoms-1"></a>
<h4 class="subsection">11.67.3 set-bond-thickness-intermediate-atoms</h4>
<dl>
<dt><a name="index-set_002dbond_002dthickness_002dintermediate_002datoms"></a>function: <strong>set-bond-thickness-intermediate-atoms</strong> <em>t</em></dt>
<dd>
<p>Where <em>t</em> is a number
</p>
<p>set the thickness of the bonds of the intermediate atoms to t pixels 
</p>
</dd></dl>

<hr>
<a name="set_002duse_002dvariable_002dbond_002dthickness"></a>
<div class="header">
<p>
Next: <a href="#set_002dbond_002dcolour_002drotation_002dfor_002dmolecule" accesskey="n" rel="next">set-bond-colour-rotation-for-molecule</a>, Previous: <a href="#set_002dbond_002dthickness_002dintermediate_002datoms" accesskey="p" rel="previous">set-bond-thickness-intermediate-atoms</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002duse_002dvariable_002dbond_002dthickness-1"></a>
<h4 class="subsection">11.67.4 set-use-variable-bond-thickness</h4>
<dl>
<dt><a name="index-set_002duse_002dvariable_002dbond_002dthickness"></a>function: <strong>set-use-variable-bond-thickness</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>allow lines that are further away to be thinner 
</p>
</dd></dl>

<hr>
<a name="set_002dbond_002dcolour_002drotation_002dfor_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#set_002ddraw_002dstick_002dmode_002datoms_002ddefault" accesskey="n" rel="next">set-draw-stick-mode-atoms-default</a>, Previous: <a href="#set_002duse_002dvariable_002dbond_002dthickness" accesskey="p" rel="previous">set-use-variable-bond-thickness</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbond_002dcolour_002drotation_002dfor_002dmolecule-1"></a>
<h4 class="subsection">11.67.5 set-bond-colour-rotation-for-molecule</h4>
<dl>
<dt><a name="index-set_002dbond_002dcolour_002drotation_002dfor_002dmolecule"></a>function: <strong>set-bond-colour-rotation-for-molecule</strong> <em>imol f</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>f</em> is a number
</li></ul>

<p>set bond colour for molecule 
</p>
</dd></dl>

<hr>
<a name="set_002ddraw_002dstick_002dmode_002datoms_002ddefault"></a>
<div class="header">
<p>
Next: <a href="#get_002dbond_002dcolour_002drotation_002dfor_002dmolecule" accesskey="n" rel="next">get-bond-colour-rotation-for-molecule</a>, Previous: <a href="#set_002dbond_002dcolour_002drotation_002dfor_002dmolecule" accesskey="p" rel="previous">set-bond-colour-rotation-for-molecule</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dstick_002dmode_002datoms_002ddefault-1"></a>
<h4 class="subsection">11.67.6 set-draw-stick-mode-atoms-default</h4>
<dl>
<dt><a name="index-set_002ddraw_002dstick_002dmode_002datoms_002ddefault"></a>function: <strong>set-draw-stick-mode-atoms-default</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set default for the drawing of atoms in stick mode (default is on (1)) 
</p>
</dd></dl>

<hr>
<a name="get_002dbond_002dcolour_002drotation_002dfor_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#set_002ddefault_002drepresentation_002dtype" accesskey="n" rel="next">set-default-representation-type</a>, Previous: <a href="#set_002ddraw_002dstick_002dmode_002datoms_002ddefault" accesskey="p" rel="previous">set-draw-stick-mode-atoms-default</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dbond_002dcolour_002drotation_002dfor_002dmolecule-1"></a>
<h4 class="subsection">11.67.7 get-bond-colour-rotation-for-molecule</h4>
<dl>
<dt><a name="index-get_002dbond_002dcolour_002drotation_002dfor_002dmolecule"></a>function: <strong>get-bond-colour-rotation-for-molecule</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>get the bond colour for molecule. 
</p>
<p>Return -1 on err (bad molecule number) 
</p>
</dd></dl>

<hr>
<a name="set_002ddefault_002drepresentation_002dtype"></a>
<div class="header">
<p>
Next: <a href="#get_002ddefault_002dbond_002dthickness" accesskey="n" rel="next">get-default-bond-thickness</a>, Previous: <a href="#get_002dbond_002dcolour_002drotation_002dfor_002dmolecule" accesskey="p" rel="previous">get-bond-colour-rotation-for-molecule</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddefault_002drepresentation_002dtype-1"></a>
<h4 class="subsection">11.67.8 set-default-representation-type</h4>
<dl>
<dt><a name="index-set_002ddefault_002drepresentation_002dtype"></a>function: <strong>set-default-representation-type</strong> <em>type</em></dt>
<dd>
<p>Where <em>type</em> is an integer number
</p>
<p>set the default representation type (default 1). 
</p>
</dd></dl>

<hr>
<a name="get_002ddefault_002dbond_002dthickness"></a>
<div class="header">
<p>
Next: <a href="#set_002ddraw_002dzero_002docc_002dmarkers" accesskey="n" rel="next">set-draw-zero-occ-markers</a>, Previous: <a href="#set_002ddefault_002drepresentation_002dtype" accesskey="p" rel="previous">set-default-representation-type</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002ddefault_002dbond_002dthickness-1"></a>
<h4 class="subsection">11.67.9 get-default-bond-thickness</h4>
<dl>
<dt><a name="index-get_002ddefault_002dbond_002dthickness"></a>function: <strong>get-default-bond-thickness</strong></dt>
<dd>
<p>get the default thickness for bonds 
</p>
</dd></dl>

<hr>
<a name="set_002ddraw_002dzero_002docc_002dmarkers"></a>
<div class="header">
<p>
Next: <a href="#set_002ddraw_002dcis_002dpeptide_002dmarkups" accesskey="n" rel="next">set-draw-cis-peptide-markups</a>, Previous: <a href="#get_002ddefault_002dbond_002dthickness" accesskey="p" rel="previous">get-default-bond-thickness</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dzero_002docc_002dmarkers-1"></a>
<h4 class="subsection">11.67.10 set-draw-zero-occ-markers</h4>
<dl>
<dt><a name="index-set_002ddraw_002dzero_002docc_002dmarkers"></a>function: <strong>set-draw-zero-occ-markers</strong> <em>status</em></dt>
<dd>
<p>Where <em>status</em> is an integer number
</p>
<p>set status of drawing zero occupancy markers. 
</p>
<p>default status is 1. 
</p>
</dd></dl>

<hr>
<a name="set_002ddraw_002dcis_002dpeptide_002dmarkups"></a>
<div class="header">
<p>
Next: <a href="#set_002ddraw_002dhydrogens" accesskey="n" rel="next">set-draw-hydrogens</a>, Previous: <a href="#set_002ddraw_002dzero_002docc_002dmarkers" accesskey="p" rel="previous">set-draw-zero-occ-markers</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dcis_002dpeptide_002dmarkups-1"></a>
<h4 class="subsection">11.67.11 set-draw-cis-peptide-markups</h4>
<dl>
<dt><a name="index-set_002ddraw_002dcis_002dpeptide_002dmarkups"></a>function: <strong>set-draw-cis-peptide-markups</strong> <em>status</em></dt>
<dd>
<p>Where <em>status</em> is an integer number
</p>
<p>set status of drawing cis-peptide markups 
</p>
<p>default status is 1. 
</p>
</dd></dl>

<hr>
<a name="set_002ddraw_002dhydrogens"></a>
<div class="header">
<p>
Next: <a href="#draw_002dhydrogens_002dstate" accesskey="n" rel="next">draw-hydrogens-state</a>, Previous: <a href="#set_002ddraw_002dcis_002dpeptide_002dmarkups" accesskey="p" rel="previous">set-draw-cis-peptide-markups</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dhydrogens-1"></a>
<h4 class="subsection">11.67.12 set-draw-hydrogens</h4>
<dl>
<dt><a name="index-set_002ddraw_002dhydrogens"></a>function: <strong>set-draw-hydrogens</strong> <em>imol istat</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>istat</em> is an integer number
</li></ul>

<p>set the hydrogen drawing state. istat = 0 is hydrogens off, istat = 1: show hydrogens 
</p>
</dd></dl>

<hr>
<a name="draw_002dhydrogens_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002ddraw_002dstick_002dmode_002datoms" accesskey="n" rel="next">set-draw-stick-mode-atoms</a>, Previous: <a href="#set_002ddraw_002dhydrogens" accesskey="p" rel="previous">set-draw-hydrogens</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="draw_002dhydrogens_002dstate-1"></a>
<h4 class="subsection">11.67.13 draw-hydrogens-state</h4>
<dl>
<dt><a name="index-draw_002dhydrogens_002dstate"></a>function: <strong>draw-hydrogens-state</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>the state of draw hydrogens for molecule number imol. 
</p>
<p>return -1 on bad imol. 
</p>
</dd></dl>

<hr>
<a name="set_002ddraw_002dstick_002dmode_002datoms"></a>
<div class="header">
<p>
Next: <a href="#set_002ddraw_002dmissing_002dresidues_002dloops" accesskey="n" rel="next">set-draw-missing-residues-loops</a>, Previous: <a href="#draw_002dhydrogens_002dstate" accesskey="p" rel="previous">draw-hydrogens-state</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dstick_002dmode_002datoms-1"></a>
<h4 class="subsection">11.67.14 set-draw-stick-mode-atoms</h4>
<dl>
<dt><a name="index-set_002ddraw_002dstick_002dmode_002datoms"></a>function: <strong>set-draw-stick-mode-atoms</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>draw little coloured balls on atoms 
</p>
<p>turn off with state = 0
</p>
<p>turn on with state = 1 
</p>
</dd></dl>

<hr>
<a name="set_002ddraw_002dmissing_002dresidues_002dloops"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002dca_002drepresentation" accesskey="n" rel="next">graphics-to-ca-representation</a>, Previous: <a href="#set_002ddraw_002dstick_002dmode_002datoms" accesskey="p" rel="previous">set-draw-stick-mode-atoms</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dmissing_002dresidues_002dloops-1"></a>
<h4 class="subsection">11.67.15 set-draw-missing-residues-loops</h4>
<dl>
<dt><a name="index-set_002ddraw_002dmissing_002dresidues_002dloops"></a>function: <strong>set-draw-missing-residues-loops</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set the state for drawing missing resiude loops 
</p>
<p>For taking screenshots, we often don&rsquo;t want to see them. 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002dca_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002dca_002dplus_002dligands_002drepresentation" accesskey="n" rel="next">graphics-to-ca-plus-ligands-representation</a>, Previous: <a href="#set_002ddraw_002dmissing_002dresidues_002dloops" accesskey="p" rel="previous">set-draw-missing-residues-loops</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002dca_002drepresentation-1"></a>
<h4 class="subsection">11.67.16 graphics-to-ca-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002dca_002drepresentation"></a>function: <strong>graphics-to-ca-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol as CAs 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002dca_002dplus_002dligands_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation" accesskey="n" rel="next">graphics-to-ca-plus-ligands-and-sidechains-representation</a>, Previous: <a href="#graphics_002dto_002dca_002drepresentation" accesskey="p" rel="previous">graphics-to-ca-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002dca_002dplus_002dligands_002drepresentation-1"></a>
<h4 class="subsection">11.67.17 graphics-to-ca-plus-ligands-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002dca_002dplus_002dligands_002drepresentation"></a>function: <strong>graphics-to-ca-plus-ligands-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol as CA + ligands 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation" accesskey="n" rel="next">graphics-to-bonds-no-waters-representation</a>, Previous: <a href="#graphics_002dto_002dca_002dplus_002dligands_002drepresentation" accesskey="p" rel="previous">graphics-to-ca-plus-ligands-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation-1"></a>
<h4 class="subsection">11.67.18 graphics-to-ca-plus-ligands-and-sidechains-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation"></a>function: <strong>graphics-to-ca-plus-ligands-and-sidechains-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol as CA + ligands + sidechains 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002dbonds_002drepresentation" accesskey="n" rel="next">graphics-to-bonds-representation</a>, Previous: <a href="#graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation" accesskey="p" rel="previous">graphics-to-ca-plus-ligands-and-sidechains-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation-1"></a>
<h4 class="subsection">11.67.19 graphics-to-bonds-no-waters-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation"></a>function: <strong>graphics-to-bonds-no-waters-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol with no waters 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002dbonds_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation" accesskey="n" rel="next">graphics-to-ca-plus-ligands-sec-struct-representation</a>, Previous: <a href="#graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation" accesskey="p" rel="previous">graphics-to-bonds-no-waters-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002dbonds_002drepresentation-1"></a>
<h4 class="subsection">11.67.20 graphics-to-bonds-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002dbonds_002drepresentation"></a>function: <strong>graphics-to-bonds-representation</strong> <em>mol</em></dt>
<dd>
<p>Where <em>mol</em> is an integer number
</p>
<p>draw molecule number imol with normal bonds 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation" accesskey="n" rel="next">graphics-to-sec-struct-bonds-representation</a>, Previous: <a href="#graphics_002dto_002dbonds_002drepresentation" accesskey="p" rel="previous">graphics-to-bonds-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation-1"></a>
<h4 class="subsection">11.67.21 graphics-to-ca-plus-ligands-sec-struct-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation"></a>function: <strong>graphics-to-ca-plus-ligands-sec-struct-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol with CA bonds in secondary structure representation and ligands 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002drainbow_002drepresentation" accesskey="n" rel="next">graphics-to-rainbow-representation</a>, Previous: <a href="#graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation" accesskey="p" rel="previous">graphics-to-ca-plus-ligands-sec-struct-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation-1"></a>
<h4 class="subsection">11.67.22 graphics-to-sec-struct-bonds-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation"></a>function: <strong>graphics-to-sec-struct-bonds-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol with bonds in secondary structure representation 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002drainbow_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002db_002dfactor_002drepresentation" accesskey="n" rel="next">graphics-to-b-factor-representation</a>, Previous: <a href="#graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation" accesskey="p" rel="previous">graphics-to-sec-struct-bonds-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002drainbow_002drepresentation-1"></a>
<h4 class="subsection">11.67.23 graphics-to-rainbow-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002drainbow_002drepresentation"></a>function: <strong>graphics-to-rainbow-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol in Jones&rsquo; Rainbow 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002db_002dfactor_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002db_002dfactor_002dcas_002drepresentation" accesskey="n" rel="next">graphics-to-b-factor-cas-representation</a>, Previous: <a href="#graphics_002dto_002drainbow_002drepresentation" accesskey="p" rel="previous">graphics-to-rainbow-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002db_002dfactor_002drepresentation-1"></a>
<h4 class="subsection">11.67.24 graphics-to-b-factor-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002db_002dfactor_002drepresentation"></a>function: <strong>graphics-to-b-factor-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol coloured by B-factor 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002db_002dfactor_002dcas_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002doccupancy_002drepresentation" accesskey="n" rel="next">graphics-to-occupancy-representation</a>, Previous: <a href="#graphics_002dto_002db_002dfactor_002drepresentation" accesskey="p" rel="previous">graphics-to-b-factor-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002db_002dfactor_002dcas_002drepresentation-1"></a>
<h4 class="subsection">11.67.25 graphics-to-b-factor-cas-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002db_002dfactor_002dcas_002drepresentation"></a>function: <strong>graphics-to-b-factor-cas-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol coloured by B-factor, CA + ligands 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002doccupancy_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation" accesskey="n" rel="next">graphics-to-user-defined-atom-colours-representation</a>, Previous: <a href="#graphics_002dto_002db_002dfactor_002dcas_002drepresentation" accesskey="p" rel="previous">graphics-to-b-factor-cas-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002doccupancy_002drepresentation-1"></a>
<h4 class="subsection">11.67.26 graphics-to-occupancy-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002doccupancy_002drepresentation"></a>function: <strong>graphics-to-occupancy-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol coloured by occupancy 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation" accesskey="n" rel="next">graphics-to-user-defined-atom-colours-all-atoms-representation</a>, Previous: <a href="#graphics_002dto_002doccupancy_002drepresentation" accesskey="p" rel="previous">graphics-to-occupancy-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation-1"></a>
<h4 class="subsection">11.67.27 graphics-to-user-defined-atom-colours-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation"></a>function: <strong>graphics-to-user-defined-atom-colours-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol in CA+Ligands mode coloured by user-defined atom colours 
</p>
</dd></dl>

<hr>
<a name="graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#graphics_002dmolecule_002dbond_002dtype" accesskey="n" rel="next">graphics-molecule-bond-type</a>, Previous: <a href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation" accesskey="p" rel="previous">graphics-to-user-defined-atom-colours-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation-1"></a>
<h4 class="subsection">11.67.28 graphics-to-user-defined-atom-colours-all-atoms-representation</h4>
<dl>
<dt><a name="index-graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation"></a>function: <strong>graphics-to-user-defined-atom-colours-all-atoms-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>draw molecule number imol all atoms coloured by user-defined atom colours 
</p>
</dd></dl>

<hr>
<a name="graphics_002dmolecule_002dbond_002dtype"></a>
<div class="header">
<p>
Next: <a href="#set_002db_002dfactor_002dbonds_002dscale_002dfactor" accesskey="n" rel="next">set-b-factor-bonds-scale-factor</a>, Previous: <a href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation" accesskey="p" rel="previous">graphics-to-user-defined-atom-colours-all-atoms-representation</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics_002dmolecule_002dbond_002dtype-1"></a>
<h4 class="subsection">11.67.29 graphics-molecule-bond-type</h4>
<dl>
<dt><a name="index-graphics_002dmolecule_002dbond_002dtype"></a>function: <strong>graphics-molecule-bond-type</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>what is the bond drawing state of molecule number imol 
</p>
</dd></dl>

<hr>
<a name="set_002db_002dfactor_002dbonds_002dscale_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#change_002dmodel_002dmolecule_002drepresentation_002dmode" accesskey="n" rel="next">change-model-molecule-representation-mode</a>, Previous: <a href="#graphics_002dmolecule_002dbond_002dtype" accesskey="p" rel="previous">graphics-molecule-bond-type</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002db_002dfactor_002dbonds_002dscale_002dfactor-1"></a>
<h4 class="subsection">11.67.30 set-b-factor-bonds-scale-factor</h4>
<dl>
<dt><a name="index-set_002db_002dfactor_002dbonds_002dscale_002dfactor"></a>function: <strong>set-b-factor-bonds-scale-factor</strong> <em>imol f</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>f</em> is a number
</li></ul>

<p>scale the colours for colour by b factor representation 
</p>
</dd></dl>

<hr>
<a name="change_002dmodel_002dmolecule_002drepresentation_002dmode"></a>
<div class="header">
<p>
Next: <a href="#set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule" accesskey="n" rel="next">set-use-grey-carbons-for-molecule</a>, Previous: <a href="#set_002db_002dfactor_002dbonds_002dscale_002dfactor" accesskey="p" rel="previous">set-b-factor-bonds-scale-factor</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="change_002dmodel_002dmolecule_002drepresentation_002dmode-1"></a>
<h4 class="subsection">11.67.31 change-model-molecule-representation-mode</h4>
<dl>
<dt><a name="index-change_002dmodel_002dmolecule_002drepresentation_002dmode"></a>function: <strong>change-model-molecule-representation-mode</strong> <em>up_or_down</em></dt>
<dd>
<p>Where <em>up_or_down</em> is an integer number
</p>
<p>change the representation of the model molecule closest to the centre of the screen 
</p>
</dd></dl>

<hr>
<a name="set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#set_002dgrey_002dcarbon_002dcolour" accesskey="n" rel="next">set-grey-carbon-colour</a>, Previous: <a href="#change_002dmodel_002dmolecule_002drepresentation_002dmode" accesskey="p" rel="previous">change-model-molecule-representation-mode</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule-1"></a>
<h4 class="subsection">11.67.32 set-use-grey-carbons-for-molecule</h4>
<dl>
<dt><a name="index-set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule"></a>function: <strong>set-use-grey-carbons-for-molecule</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>make the carbon atoms for molecule imol be grey 
</p>
</dd></dl>

<hr>
<a name="set_002dgrey_002dcarbon_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#make_002dball_002dand_002dstick" accesskey="n" rel="next">make-ball-and-stick</a>, Previous: <a href="#set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule" accesskey="p" rel="previous">set-use-grey-carbons-for-molecule</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dgrey_002dcarbon_002dcolour-1"></a>
<h4 class="subsection">11.67.33 set-grey-carbon-colour</h4>
<dl>
<dt><a name="index-set_002dgrey_002dcarbon_002dcolour"></a>function: <strong>set-grey-carbon-colour</strong> <em>imol r g b</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>r</em> is a number
</li><li> <em>g</em> is a number
</li><li> <em>b</em> is a number
</li></ul>

<p>set the colour for the carbon atoms 
</p>
<p>can be not grey if you desire, r, g, b in the range 0 to 1. 
</p>
</dd></dl>

<hr>
<a name="make_002dball_002dand_002dstick"></a>
<div class="header">
<p>
Next: <a href="#clear_002dball_002dand_002dstick" accesskey="n" rel="next">clear-ball-and-stick</a>, Previous: <a href="#set_002dgrey_002dcarbon_002dcolour" accesskey="p" rel="previous">set-grey-carbon-colour</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dball_002dand_002dstick-1"></a>
<h4 class="subsection">11.67.34 make-ball-and-stick</h4>
<dl>
<dt><a name="index-make_002dball_002dand_002dstick"></a>function: <strong>make-ball-and-stick</strong> <em>imol atom_selection_str bond_thickness sphere_size do_spheres_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>atom_selection_str</em> is a string
</li><li> <em>bond_thickness</em> is a number
</li><li> <em>sphere_size</em> is a number
</li><li> <em>do_spheres_flag</em> is an integer number
</li></ul>

<p>make a ball and stick representation of imol given atom selection 
</p>
<p>e.g. (make-ball-and-stick 0 &quot;/1&quot; 0.15 0.25 1) 
</p>
</dd></dl>

<hr>
<a name="clear_002dball_002dand_002dstick"></a>
<div class="header">
<p>
Next: <a href="#additional_002drepresentation_002dby_002dstring" accesskey="n" rel="next">additional-representation-by-string</a>, Previous: <a href="#make_002dball_002dand_002dstick" accesskey="p" rel="previous">make-ball-and-stick</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002dball_002dand_002dstick-1"></a>
<h4 class="subsection">11.67.35 clear-ball-and-stick</h4>
<dl>
<dt><a name="index-clear_002dball_002dand_002dstick"></a>function: <strong>clear-ball-and-stick</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>clear ball and stick representation of molecule number imol 
</p>
</dd></dl>

<hr>
<a name="additional_002drepresentation_002dby_002dstring"></a>
<div class="header">
<p>
Next: <a href="#additional_002drepresentation_002dby_002dattributes" accesskey="n" rel="next">additional-representation-by-attributes</a>, Previous: <a href="#clear_002dball_002dand_002dstick" accesskey="p" rel="previous">clear-ball-and-stick</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="additional_002drepresentation_002dby_002dstring-1"></a>
<h4 class="subsection">11.67.36 additional-representation-by-string</h4>
<dl>
<dt><a name="index-additional_002drepresentation_002dby_002dstring"></a>function: <strong>additional-representation-by-string</strong> <em>imol atom_selection representation_type bonds_box_type bond_width draw_hydrogens_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>atom_selection</em> is a string
</li><li> <em>representation_type</em> is an integer number
</li><li> <em>bonds_box_type</em> is an integer number
</li><li> <em>bond_width</em> is a number
</li><li> <em>draw_hydrogens_flag</em> is an integer number
</li></ul>

<p>return the index of the additional representation. Return -1 on error 
</p>
</dd></dl>

<hr>
<a name="additional_002drepresentation_002dby_002dattributes"></a>
<div class="header">
<p>
Previous: <a href="#additional_002drepresentation_002dby_002dstring" accesskey="p" rel="previous">additional-representation-by-string</a>, Up: <a href="#Bond-Representation" accesskey="u" rel="up">Bond Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="additional_002drepresentation_002dby_002dattributes-1"></a>
<h4 class="subsection">11.67.37 additional-representation-by-attributes</h4>
<dl>
<dt><a name="index-additional_002drepresentation_002dby_002dattributes"></a>function: <strong>additional-representation-by-attributes</strong> <em>imol chain_id resno_start resno_end ins_code representation_type bonds_box_type bond_width draw_hydrogens_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno_start</em> is an integer number
</li><li> <em>resno_end</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>representation_type</em> is an integer number
</li><li> <em>bonds_box_type</em> is an integer number
</li><li> <em>bond_width</em> is a number
</li><li> <em>draw_hydrogens_flag</em> is an integer number
</li></ul>

<p>return the index of the additional representation. 
</p>
<p>Returns: -1 on error. 
</p>
</dd></dl>

<hr>
<a name="Dots-Representation"></a>
<div class="header">
<p>
Next: <a href="#Pep_002dflip-Interface" accesskey="n" rel="next">Pep-flip Interface</a>, Previous: <a href="#Bond-Representation" accesskey="p" rel="previous">Bond Representation</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Dots-Representation-1"></a>
<h3 class="section">11.68 Dots Representation</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#dots" accesskey="1">dots</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddots_002dcolour" accesskey="2">set-dots-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#unset_002ddots_002dcolour" accesskey="3">unset-dots-colour</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clear_002ddots" accesskey="4">clear-dots</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clear_002ddots_002dby_002dname" accesskey="5">clear-dots-by-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#n_002ddots_002dsets" accesskey="6">n-dots-sets</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="dots"></a>
<div class="header">
<p>
Next: <a href="#set_002ddots_002dcolour" accesskey="n" rel="next">set-dots-colour</a>, Up: <a href="#Dots-Representation" accesskey="u" rel="up">Dots Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="dots-1"></a>
<h4 class="subsection">11.68.1 dots</h4>
<dl>
<dt><a name="index-dots"></a>function: <strong>dots</strong> <em>imol atom_selection_str dots_object_name dot_density sphere_size_scale</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>atom_selection_str</em> is a string
</li><li> <em>dots_object_name</em> is a string
</li><li> <em>dot_density</em> is a number
</li><li> <em>sphere_size_scale</em> is a number
</li></ul>

<p>display dotted surface 
</p>
<p>return a generic objects handle (which can be used to remove later) 
</p>
</dd></dl>

<hr>
<a name="set_002ddots_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#unset_002ddots_002dcolour" accesskey="n" rel="next">unset-dots-colour</a>, Previous: <a href="#dots" accesskey="p" rel="previous">dots</a>, Up: <a href="#Dots-Representation" accesskey="u" rel="up">Dots Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddots_002dcolour-1"></a>
<h4 class="subsection">11.68.2 set-dots-colour</h4>
<dl>
<dt><a name="index-set_002ddots_002dcolour"></a>function: <strong>set-dots-colour</strong> <em>imol r g b</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>r</em> is a number
</li><li> <em>g</em> is a number
</li><li> <em>b</em> is a number
</li></ul>

<p>set the colour of the surface dots of the imol-th molecule to be the given single colour 
</p>
<p>r,g,b are values between 0.0 and 1.0 
</p>
</dd></dl>

<hr>
<a name="unset_002ddots_002dcolour"></a>
<div class="header">
<p>
Next: <a href="#clear_002ddots" accesskey="n" rel="next">clear-dots</a>, Previous: <a href="#set_002ddots_002dcolour" accesskey="p" rel="previous">set-dots-colour</a>, Up: <a href="#Dots-Representation" accesskey="u" rel="up">Dots Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="unset_002ddots_002dcolour-1"></a>
<h4 class="subsection">11.68.3 unset-dots-colour</h4>
<dl>
<dt><a name="index-unset_002ddots_002dcolour"></a>function: <strong>unset-dots-colour</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>no longer set the dots of molecule imol to a single colour 
</p>
<p>i.e. go back to element-based colours. 
</p>
</dd></dl>

<hr>
<a name="clear_002ddots"></a>
<div class="header">
<p>
Next: <a href="#clear_002ddots_002dby_002dname" accesskey="n" rel="next">clear-dots-by-name</a>, Previous: <a href="#unset_002ddots_002dcolour" accesskey="p" rel="previous">unset-dots-colour</a>, Up: <a href="#Dots-Representation" accesskey="u" rel="up">Dots Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002ddots-1"></a>
<h4 class="subsection">11.68.4 clear-dots</h4>
<dl>
<dt><a name="index-clear_002ddots"></a>function: <strong>clear-dots</strong> <em>imol dots_handle</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>dots_handle</em> is an integer number
</li></ul>

<p>clear dots in imol with dots_handle 
</p>
</dd></dl>

<hr>
<a name="clear_002ddots_002dby_002dname"></a>
<div class="header">
<p>
Next: <a href="#n_002ddots_002dsets" accesskey="n" rel="next">n-dots-sets</a>, Previous: <a href="#clear_002ddots" accesskey="p" rel="previous">clear-dots</a>, Up: <a href="#Dots-Representation" accesskey="u" rel="up">Dots Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002ddots_002dby_002dname-1"></a>
<h4 class="subsection">11.68.5 clear-dots-by-name</h4>
<dl>
<dt><a name="index-clear_002ddots_002dby_002dname"></a>function: <strong>clear-dots-by-name</strong> <em>imol dots_object_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>dots_object_name</em> is a string
</li></ul>

<p>clear the first dots object for imol with given name 
</p>
</dd></dl>

<hr>
<a name="n_002ddots_002dsets"></a>
<div class="header">
<p>
Previous: <a href="#clear_002ddots_002dby_002dname" accesskey="p" rel="previous">clear-dots-by-name</a>, Up: <a href="#Dots-Representation" accesskey="u" rel="up">Dots Representation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="n_002ddots_002dsets-1"></a>
<h4 class="subsection">11.68.6 n-dots-sets</h4>
<dl>
<dt><a name="index-n_002ddots_002dsets"></a>function: <strong>n-dots-sets</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the number of dots sets for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="Pep_002dflip-Interface"></a>
<div class="header">
<p>
Next: <a href="#Rigid-Body-Refinement-Interface" accesskey="n" rel="next">Rigid Body Refinement Interface</a>, Previous: <a href="#Dots-Representation" accesskey="p" rel="previous">Dots Representation</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Pep_002dflip-Interface-1"></a>
<h3 class="section">11.69 Pep-flip Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#pepflip" accesskey="1">pepflip</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="pepflip"></a>
<div class="header">
<p>
Up: <a href="#Pep_002dflip-Interface" accesskey="u" rel="up">Pep-flip Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="pepflip-1"></a>
<h4 class="subsection">11.69.1 pepflip</h4>
<dl>
<dt><a name="index-pepflip-1"></a>function: <strong>pepflip</strong> <em>imol chain_id resno inscode altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>inscode</em> is a string
</li><li> <em>altconf</em> is a string
</li></ul>

<p>pepflip the given residue 
</p>
</dd></dl>

<hr>
<a name="Rigid-Body-Refinement-Interface"></a>
<div class="header">
<p>
Next: <a href="#Add-Terminal-Residue-Functions" accesskey="n" rel="next">Add Terminal Residue Functions</a>, Previous: <a href="#Pep_002dflip-Interface" accesskey="p" rel="previous">Pep-flip Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rigid-Body-Refinement-Interface-1"></a>
<h3 class="section">11.70 Rigid Body Refinement Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#rigid_002dbody_002drefine_002dzone" accesskey="1">rigid-body-refine-zone</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm" accesskey="2">rigid-body-refine-by-residue-ranges-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy" accesskey="3">rigid-body-refine-by-residue-ranges-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction" accesskey="4">set-rigid-body-fit-acceptable-fit-fraction</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="rigid_002dbody_002drefine_002dzone"></a>
<div class="header">
<p>
Next: <a href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm" accesskey="n" rel="next">rigid-body-refine-by-residue-ranges-scm</a>, Up: <a href="#Rigid-Body-Refinement-Interface" accesskey="u" rel="up">Rigid Body Refinement Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rigid_002dbody_002drefine_002dzone-1"></a>
<h4 class="subsection">11.70.1 rigid-body-refine-zone</h4>
<dl>
<dt><a name="index-rigid_002dbody_002drefine_002dzone"></a>function: <strong>rigid-body-refine-zone</strong> <em>reso_start resno_end chain_id imol</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>reso_start</em> is an integer number
</li><li> <em>resno_end</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>imol</em> is an integer number
</li></ul>

<p>setup rigid body refine zone 
</p>
<p>where we set the atom selection holders according to the arguments and then call execute_rigid_body_refine() 
</p>
</dd></dl>

<hr>
<a name="rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm"></a>
<div class="header">
<p>
Next: <a href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy" accesskey="n" rel="next">rigid-body-refine-by-residue-ranges-py</a>, Previous: <a href="#rigid_002dbody_002drefine_002dzone" accesskey="p" rel="previous">rigid-body-refine-zone</a>, Up: <a href="#Rigid-Body-Refinement-Interface" accesskey="u" rel="up">Rigid Body Refinement Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm-1"></a>
<h4 class="subsection">11.70.2 rigid-body-refine-by-residue-ranges-scm</h4>
<dl>
<dt><a name="index-rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm"></a>function: <strong>rigid-body-refine-by-residue-ranges-scm</strong> <em>imol residue_ranges</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_ranges</em> is a SCM
</li></ul>

<p>rigid body refine using residue ranges. residue_ranges is a list of residue ranges. A residue range is (list chain-id resno-start resno-end). 
</p>
</dd></dl>

<hr>
<a name="rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy"></a>
<div class="header">
<p>
Next: <a href="#set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction" accesskey="n" rel="next">set-rigid-body-fit-acceptable-fit-fraction</a>, Previous: <a href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm" accesskey="p" rel="previous">rigid-body-refine-by-residue-ranges-scm</a>, Up: <a href="#Rigid-Body-Refinement-Interface" accesskey="u" rel="up">Rigid Body Refinement Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy-1"></a>
<h4 class="subsection">11.70.3 rigid-body-refine-by-residue-ranges-py</h4>
<dl>
<dt><a name="index-rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy"></a>function: <strong>rigid-body-refine-by-residue-ranges-py</strong> <em>imol residue_ranges</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_ranges</em> is a PyObject
</li></ul>

<p>rigid body refine using residue ranges. residue_ranges is a list of residue ranges. A residue range is [chain_id, resno_start, resno_end]. 
</p>
</dd></dl>

<hr>
<a name="set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction"></a>
<div class="header">
<p>
Previous: <a href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy" accesskey="p" rel="previous">rigid-body-refine-by-residue-ranges-py</a>, Up: <a href="#Rigid-Body-Refinement-Interface" accesskey="u" rel="up">Rigid Body Refinement Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction-1"></a>
<h4 class="subsection">11.70.4 set-rigid-body-fit-acceptable-fit-fraction</h4>
<dl>
<dt><a name="index-set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction"></a>function: <strong>set-rigid-body-fit-acceptable-fit-fraction</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set rigid body fraction of atoms in positive density 
</p>
</dd></dl>

<hr>
<a name="Add-Terminal-Residue-Functions"></a>
<div class="header">
<p>
Next: <a href="#Delete-Residues" accesskey="n" rel="next">Delete Residues</a>, Previous: <a href="#Rigid-Body-Refinement-Interface" accesskey="p" rel="previous">Rigid Body Refinement Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Add-Terminal-Residue-Functions-1"></a>
<h3 class="section">11.71 Add Terminal Residue Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition" accesskey="1">set-add-terminal-residue-immediate-addition</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dterminal_002dresidue" accesskey="2">add-terminal-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dnucleotide" accesskey="3">add-nucleotide</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dterminal_002dresidue_002dusing_002dphi_002dpsi" accesskey="4">add-terminal-residue-using-phi-psi</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype" accesskey="5">set-add-terminal-residue-default-residue-type</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine" accesskey="6">set-add-terminal-residue-do-post-refine</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate" accesskey="7">add-terminal-residue-do-post-refine-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition"></a>
<div class="header">
<p>
Next: <a href="#add_002dterminal_002dresidue" accesskey="n" rel="next">add-terminal-residue</a>, Up: <a href="#Add-Terminal-Residue-Functions" accesskey="u" rel="up">Add Terminal Residue Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition-1"></a>
<h4 class="subsection">11.71.1 set-add-terminal-residue-immediate-addition</h4>
<dl>
<dt><a name="index-set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition"></a>function: <strong>set-add-terminal-residue-immediate-addition</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set immediate addition of terminal residue 
</p>
<p>call with i=1 for immediate addtion 
</p>
</dd></dl>

<hr>
<a name="add_002dterminal_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#add_002dnucleotide" accesskey="n" rel="next">add-nucleotide</a>, Previous: <a href="#set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition" accesskey="p" rel="previous">set-add-terminal-residue-immediate-addition</a>, Up: <a href="#Add-Terminal-Residue-Functions" accesskey="u" rel="up">Add Terminal Residue Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dterminal_002dresidue-1"></a>
<h4 class="subsection">11.71.2 add-terminal-residue</h4>
<dl>
<dt><a name="index-add_002dterminal_002dresidue"></a>function: <strong>add-terminal-residue</strong> <em>imol chain_id residue_number residue_type immediate_add</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>residue_number</em> is an integer number
</li><li> <em>residue_type</em> is a string
</li><li> <em>immediate_add</em> is an integer number
</li></ul>

<p>Add a terminal residue. 
</p>
<p>residue type can be &quot;auto&quot; and immediate_add is recommended to be 1.
</p>
<p>Returns: 0 on failure, 1 on success 
</p>
</dd></dl>

<hr>
<a name="add_002dnucleotide"></a>
<div class="header">
<p>
Next: <a href="#add_002dterminal_002dresidue_002dusing_002dphi_002dpsi" accesskey="n" rel="next">add-terminal-residue-using-phi-psi</a>, Previous: <a href="#add_002dterminal_002dresidue" accesskey="p" rel="previous">add-terminal-residue</a>, Up: <a href="#Add-Terminal-Residue-Functions" accesskey="u" rel="up">Add Terminal Residue Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dnucleotide-1"></a>
<h4 class="subsection">11.71.3 add-nucleotide</h4>
<dl>
<dt><a name="index-add_002dnucleotide"></a>function: <strong>add-nucleotide</strong> <em>imol chain_id res_no</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li></ul>

<p>Add a terminal nucleotide. 
</p>
<p>No fitting is done 
</p>
</dd></dl>

<hr>
<a name="add_002dterminal_002dresidue_002dusing_002dphi_002dpsi"></a>
<div class="header">
<p>
Next: <a href="#set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype" accesskey="n" rel="next">set-add-terminal-residue-default-residue-type</a>, Previous: <a href="#add_002dnucleotide" accesskey="p" rel="previous">add-nucleotide</a>, Up: <a href="#Add-Terminal-Residue-Functions" accesskey="u" rel="up">Add Terminal Residue Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dterminal_002dresidue_002dusing_002dphi_002dpsi-1"></a>
<h4 class="subsection">11.71.4 add-terminal-residue-using-phi-psi</h4>
<dl>
<dt><a name="index-add_002dterminal_002dresidue_002dusing_002dphi_002dpsi"></a>function: <strong>add-terminal-residue-using-phi-psi</strong> <em>imol chain_id res_no residue_type phi psi</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>residue_type</em> is a string
</li><li> <em>phi</em> is a number
</li><li> <em>psi</em> is a number
</li></ul>

<p>Add a terminal residue using given phi and psi angles. 
</p>
<p>Returns: the success status, 0 on failure, 1 on success 
</p>
</dd></dl>

<hr>
<a name="set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype"></a>
<div class="header">
<p>
Next: <a href="#set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine" accesskey="n" rel="next">set-add-terminal-residue-do-post-refine</a>, Previous: <a href="#add_002dterminal_002dresidue_002dusing_002dphi_002dpsi" accesskey="p" rel="previous">add-terminal-residue-using-phi-psi</a>, Up: <a href="#Add-Terminal-Residue-Functions" accesskey="u" rel="up">Add Terminal Residue Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype-1"></a>
<h4 class="subsection">11.71.5 set-add-terminal-residue-default-residue-type</h4>
<dl>
<dt><a name="index-set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype"></a>function: <strong>set-add-terminal-residue-default-residue-type</strong> <em>type</em></dt>
<dd>
<p>Where <em>type</em> is a string
</p>
<p>set the residue type of an added terminal residue. 
</p>
</dd></dl>

<hr>
<a name="set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine"></a>
<div class="header">
<p>
Next: <a href="#add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate" accesskey="n" rel="next">add-terminal-residue-do-post-refine-state</a>, Previous: <a href="#set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype" accesskey="p" rel="previous">set-add-terminal-residue-default-residue-type</a>, Up: <a href="#Add-Terminal-Residue-Functions" accesskey="u" rel="up">Add Terminal Residue Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine-1"></a>
<h4 class="subsection">11.71.6 set-add-terminal-residue-do-post-refine</h4>
<dl>
<dt><a name="index-set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine"></a>function: <strong>set-add-terminal-residue-do-post-refine</strong> <em>istat</em></dt>
<dd>
<p>Where <em>istat</em> is an integer number
</p>
<p>set a flag to run refine zone on terminal residues after an addition. 
</p>
</dd></dl>

<hr>
<a name="add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate"></a>
<div class="header">
<p>
Previous: <a href="#set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine" accesskey="p" rel="previous">set-add-terminal-residue-do-post-refine</a>, Up: <a href="#Add-Terminal-Residue-Functions" accesskey="u" rel="up">Add Terminal Residue Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate-1"></a>
<h4 class="subsection">11.71.7 add-terminal-residue-do-post-refine-state</h4>
<dl>
<dt><a name="index-add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate"></a>function: <strong>add-terminal-residue-do-post-refine-state</strong></dt>
<dd>
<p>what is the value of the previous flag? 
</p>
</dd></dl>

<hr>
<a name="Delete-Residues"></a>
<div class="header">
<p>
Next: <a href="#Mainchain-Building-Functions" accesskey="n" rel="next">Mainchain Building Functions</a>, Previous: <a href="#Add-Terminal-Residue-Functions" accesskey="p" rel="previous">Add Terminal Residue Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Delete-Residues-1"></a>
<h3 class="section">11.72 Delete Residues</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#delete_002dresidue_002drange" accesskey="1">delete-residue-range</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dresidue" accesskey="2">delete-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dresidue_002dwith_002dfull_002dspec" accesskey="3">delete-residue-with-full-spec</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dresidues_002dscm" accesskey="4">delete-residues-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dresidues_002dpy" accesskey="5">delete-residues-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dresidue_002dhydrogens" accesskey="6">delete-residue-hydrogens</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002datom" accesskey="7">delete-atom</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dresidue_002dsidechain" accesskey="8">delete-residue-sidechain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dhydrogen_002datoms" accesskey="9">delete-hydrogen-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dhydrogens">delete-hydrogens</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dwaters">delete-waters</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dchain">delete-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dsidechains_002dfor_002dchain">delete-sidechains-for-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="delete_002dresidue_002drange"></a>
<div class="header">
<p>
Next: <a href="#delete_002dresidue" accesskey="n" rel="next">delete-residue</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dresidue_002drange-1"></a>
<h4 class="subsection">11.72.1 delete-residue-range</h4>
<dl>
<dt><a name="index-delete_002dresidue_002drange"></a>function: <strong>delete-residue-range</strong> <em>imol chain_id resno_start end_resno</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno_start</em> is an integer number
</li><li> <em>end_resno</em> is an integer number
</li></ul>

<p>delete residue range 
</p>
</dd></dl>

<hr>
<a name="delete_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#delete_002dresidue_002dwith_002dfull_002dspec" accesskey="n" rel="next">delete-residue-with-full-spec</a>, Previous: <a href="#delete_002dresidue_002drange" accesskey="p" rel="previous">delete-residue-range</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dresidue-1"></a>
<h4 class="subsection">11.72.2 delete-residue</h4>
<dl>
<dt><a name="index-delete_002dresidue"></a>function: <strong>delete-residue</strong> <em>imol chain_id resno inscode</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>inscode</em> is a string
</li></ul>

<p>delete residue 
</p>
</dd></dl>

<hr>
<a name="delete_002dresidue_002dwith_002dfull_002dspec"></a>
<div class="header">
<p>
Next: <a href="#delete_002dresidues_002dscm" accesskey="n" rel="next">delete-residues-scm</a>, Previous: <a href="#delete_002dresidue" accesskey="p" rel="previous">delete-residue</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dresidue_002dwith_002dfull_002dspec-1"></a>
<h4 class="subsection">11.72.3 delete-residue-with-full-spec</h4>
<dl>
<dt><a name="index-delete_002dresidue_002dwith_002dfull_002dspec"></a>function: <strong>delete-residue-with-full-spec</strong> <em>imol imodel chain_id resno inscode altloc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>imodel</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>inscode</em> is a string
</li><li> <em>altloc</em> is a string
</li></ul>

<p>delete residue with altconf 
</p>
</dd></dl>

<hr>
<a name="delete_002dresidues_002dscm"></a>
<div class="header">
<p>
Next: <a href="#delete_002dresidues_002dpy" accesskey="n" rel="next">delete-residues-py</a>, Previous: <a href="#delete_002dresidue_002dwith_002dfull_002dspec" accesskey="p" rel="previous">delete-residue-with-full-spec</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dresidues_002dscm-1"></a>
<h4 class="subsection">11.72.4 delete-residues-scm</h4>
<dl>
<dt><a name="index-delete_002dresidues_002dscm"></a>function: <strong>delete-residues-scm</strong> <em>imol residue_specs_scm</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs_scm</em> is a SCM
</li></ul>

<p>delete residues in the residue spec list 
</p>
</dd></dl>

<hr>
<a name="delete_002dresidues_002dpy"></a>
<div class="header">
<p>
Next: <a href="#delete_002dresidue_002dhydrogens" accesskey="n" rel="next">delete-residue-hydrogens</a>, Previous: <a href="#delete_002dresidues_002dscm" accesskey="p" rel="previous">delete-residues-scm</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dresidues_002dpy-1"></a>
<h4 class="subsection">11.72.5 delete-residues-py</h4>
<dl>
<dt><a name="index-delete_002dresidues_002dpy"></a>function: <strong>delete-residues-py</strong> <em>imol residue_specs_py</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs_py</em> is a PyObject
</li></ul>

<p>delete residues in the residue spec list 
</p>
</dd></dl>

<hr>
<a name="delete_002dresidue_002dhydrogens"></a>
<div class="header">
<p>
Next: <a href="#delete_002datom" accesskey="n" rel="next">delete-atom</a>, Previous: <a href="#delete_002dresidues_002dpy" accesskey="p" rel="previous">delete-residues-py</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dresidue_002dhydrogens-1"></a>
<h4 class="subsection">11.72.6 delete-residue-hydrogens</h4>
<dl>
<dt><a name="index-delete_002dresidue_002dhydrogens"></a>function: <strong>delete-residue-hydrogens</strong> <em>imol chain_id resno inscode altloc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>inscode</em> is a string
</li><li> <em>altloc</em> is a string
</li></ul>

<p>delete hydrogen atoms in residue 
</p>
</dd></dl>

<hr>
<a name="delete_002datom"></a>
<div class="header">
<p>
Next: <a href="#delete_002dresidue_002dsidechain" accesskey="n" rel="next">delete-residue-sidechain</a>, Previous: <a href="#delete_002dresidue_002dhydrogens" accesskey="p" rel="previous">delete-residue-hydrogens</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002datom-1"></a>
<h4 class="subsection">11.72.7 delete-atom</h4>
<dl>
<dt><a name="index-delete_002datom"></a>function: <strong>delete-atom</strong> <em>imol chain_id resno ins_code at_name altloc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>at_name</em> is a string
</li><li> <em>altloc</em> is a string
</li></ul>

<p>delete atom in residue 
</p>
</dd></dl>

<hr>
<a name="delete_002dresidue_002dsidechain"></a>
<div class="header">
<p>
Next: <a href="#delete_002dhydrogen_002datoms" accesskey="n" rel="next">delete-hydrogen-atoms</a>, Previous: <a href="#delete_002datom" accesskey="p" rel="previous">delete-atom</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dresidue_002dsidechain-1"></a>
<h4 class="subsection">11.72.8 delete-residue-sidechain</h4>
<dl>
<dt><a name="index-delete_002dresidue_002dsidechain"></a>function: <strong>delete-residue-sidechain</strong> <em>imol chain_id resno ins_code do_delete_dialog</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>do_delete_dialog</em> is an integer number
</li></ul>

<p>delete all atoms in residue that are not main chain or CB 
</p>
</dd></dl>

<hr>
<a name="delete_002dhydrogen_002datoms"></a>
<div class="header">
<p>
Next: <a href="#delete_002dhydrogens" accesskey="n" rel="next">delete-hydrogens</a>, Previous: <a href="#delete_002dresidue_002dsidechain" accesskey="p" rel="previous">delete-residue-sidechain</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dhydrogen_002datoms-1"></a>
<h4 class="subsection">11.72.9 delete-hydrogen-atoms</h4>
<dl>
<dt><a name="index-delete_002dhydrogen_002datoms"></a>function: <strong>delete-hydrogen-atoms</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>delete all hydrogens in molecule, 
</p>
<p>Returns: number of hydrogens deleted. 
</p>
</dd></dl>

<hr>
<a name="delete_002dhydrogens"></a>
<div class="header">
<p>
Next: <a href="#delete_002dwaters" accesskey="n" rel="next">delete-waters</a>, Previous: <a href="#delete_002dhydrogen_002datoms" accesskey="p" rel="previous">delete-hydrogen-atoms</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dhydrogens-1"></a>
<h4 class="subsection">11.72.10 delete-hydrogens</h4>
<dl>
<dt><a name="index-delete_002dhydrogens"></a>function: <strong>delete-hydrogens</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>delete all hydrogens in molecule, 
</p>
<p>Returns: number of hydrogens deleted. 
</p>
</dd></dl>

<hr>
<a name="delete_002dwaters"></a>
<div class="header">
<p>
Next: <a href="#delete_002dchain" accesskey="n" rel="next">delete-chain</a>, Previous: <a href="#delete_002dhydrogens" accesskey="p" rel="previous">delete-hydrogens</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dwaters-1"></a>
<h4 class="subsection">11.72.11 delete-waters</h4>
<dl>
<dt><a name="index-delete_002dwaters"></a>function: <strong>delete-waters</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>delete all waters in molecule, 
</p>
<p>Returns: number of waters deleted. 
</p>
</dd></dl>

<hr>
<a name="delete_002dchain"></a>
<div class="header">
<p>
Next: <a href="#delete_002dsidechains_002dfor_002dchain" accesskey="n" rel="next">delete-sidechains-for-chain</a>, Previous: <a href="#delete_002dwaters" accesskey="p" rel="previous">delete-waters</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dchain-1"></a>
<h4 class="subsection">11.72.12 delete-chain</h4>
<dl>
<dt><a name="index-delete_002dchain"></a>function: <strong>delete-chain</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li></ul>

<p>delete the chain 
</p>
</dd></dl>

<hr>
<a name="delete_002dsidechains_002dfor_002dchain"></a>
<div class="header">
<p>
Previous: <a href="#delete_002dchain" accesskey="p" rel="previous">delete-chain</a>, Up: <a href="#Delete-Residues" accesskey="u" rel="up">Delete Residues</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dsidechains_002dfor_002dchain-1"></a>
<h4 class="subsection">11.72.13 delete-sidechains-for-chain</h4>
<dl>
<dt><a name="index-delete_002dsidechains_002dfor_002dchain"></a>function: <strong>delete-sidechains-for-chain</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li></ul>

<p>delete the side chains in the chain 
</p>
</dd></dl>

<hr>
<a name="Mainchain-Building-Functions"></a>
<div class="header">
<p>
Next: <a href="#Close-Molecule-Functions" accesskey="n" rel="next">Close Molecule Functions</a>, Previous: <a href="#Delete-Residues" accesskey="p" rel="previous">Delete Residues</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Mainchain-Building-Functions-1"></a>
<h3 class="section">11.73 Mainchain Building Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#db_002dmainchain" accesskey="1">db-mainchain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#db_002dmainchains_002dfragment" accesskey="2">db-mainchains-fragment</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="db_002dmainchain"></a>
<div class="header">
<p>
Next: <a href="#db_002dmainchains_002dfragment" accesskey="n" rel="next">db-mainchains-fragment</a>, Up: <a href="#Mainchain-Building-Functions" accesskey="u" rel="up">Mainchain Building Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="db_002dmainchain-1"></a>
<h4 class="subsection">11.73.1 db-mainchain</h4>
<dl>
<dt><a name="index-db_002dmainchain"></a>function: <strong>db-mainchain</strong> <em>imol chain_id iresno_start iresno_end direction</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>iresno_start</em> is an integer number
</li><li> <em>iresno_end</em> is an integer number
</li><li> <em>direction</em> is a string
</li></ul>

<p>CA -&gt; mainchain conversion. 
</p>
<p>direction is either &quot;forwards&quot; or &quot;backwards&quot;
</p>
<p>See also the function below.
</p>
<p>return the new molecule number 
</p>
</dd></dl>

<hr>
<a name="db_002dmainchains_002dfragment"></a>
<div class="header">
<p>
Previous: <a href="#db_002dmainchain" accesskey="p" rel="previous">db-mainchain</a>, Up: <a href="#Mainchain-Building-Functions" accesskey="u" rel="up">Mainchain Building Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="db_002dmainchains_002dfragment-1"></a>
<h4 class="subsection">11.73.2 db-mainchains-fragment</h4>
<dl>
<dt><a name="index-db_002dmainchains_002dfragment"></a>function: <strong>db-mainchains-fragment</strong> <em>imol chain_id res_no</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li></ul>

<p>CA-Zone to Mainchain for a fragment based on the given residue. 
</p>
<p>Both directions are built. This is the modern interface. 
</p>
</dd></dl>

<hr>
<a name="Close-Molecule-Functions"></a>
<div class="header">
<p>
Next: <a href="#Rotamer-Functions" accesskey="n" rel="next">Rotamer Functions</a>, Previous: <a href="#Mainchain-Building-Functions" accesskey="p" rel="previous">Mainchain Building Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Close-Molecule-Functions-1"></a>
<h3 class="section">11.74 Close Molecule Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#close_002dmolecule" accesskey="1">close-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="close_002dmolecule"></a>
<div class="header">
<p>
Up: <a href="#Close-Molecule-Functions" accesskey="u" rel="up">Close Molecule Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="close_002dmolecule-1"></a>
<h4 class="subsection">11.74.1 close-molecule</h4>
<dl>
<dt><a name="index-close_002dmolecule"></a>function: <strong>close-molecule</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>close the molecule 
</p>
</dd></dl>

<hr>
<a name="Rotamer-Functions"></a>
<div class="header">
<p>
Next: <a href="#g_t180-Flip-Side-chain" accesskey="n" rel="next">180 Flip Side chain</a>, Previous: <a href="#Close-Molecule-Functions" accesskey="p" rel="previous">Close Molecule Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rotamer-Functions-1"></a>
<h3 class="section">11.75 Rotamer Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002drotamer_002dsearch_002dmode" accesskey="1">set-rotamer-search-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drotamer_002dlowest_002dprobability" accesskey="2">set-rotamer-lowest-probability</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drotamer_002dcheck_002dclashes" accesskey="3">set-rotamer-check-clashes</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#auto_002dfit_002dbest_002drotamer" accesskey="4">auto-fit-best-rotamer</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag" accesskey="5">set-auto-fit-best-rotamer-clash-flag</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#n_002drotamers" accesskey="6">n-rotamers</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dresidue_002dto_002drotamer_002dnumber" accesskey="7">set-residue-to-rotamer-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dresidue_002dto_002drotamer_002dname" accesskey="8">set-residue-to-rotamer-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fill_002dpartial_002dresidues" accesskey="9">fill-partial-residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#simple_002dfill_002dpartial_002dresidues">simple-fill-partial-residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#rotamer_002dgraphs">rotamer-graphs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#rotamer_002dgraphs_002dpy">rotamer-graphs-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002drotamer_002dsearch_002dmode"></a>
<div class="header">
<p>
Next: <a href="#set_002drotamer_002dlowest_002dprobability" accesskey="n" rel="next">set-rotamer-lowest-probability</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drotamer_002dsearch_002dmode-1"></a>
<h4 class="subsection">11.75.1 set-rotamer-search-mode</h4>
<dl>
<dt><a name="index-set_002drotamer_002dsearch_002dmode"></a>function: <strong>set-rotamer-search-mode</strong> <em>mode</em></dt>
<dd>
<p>Where <em>mode</em> is an integer number
</p>
<p>set the mode of rotamer search, options are (ROTAMERSEARCHAUTOMATIC), (ROTAMERSEARCHLOWRES) (aka. &quot;backrub rotamers&quot;), (ROTAMERSEARCHHIGHRES) (with rigid body fitting) 
</p>
</dd></dl>

<hr>
<a name="set_002drotamer_002dlowest_002dprobability"></a>
<div class="header">
<p>
Next: <a href="#set_002drotamer_002dcheck_002dclashes" accesskey="n" rel="next">set-rotamer-check-clashes</a>, Previous: <a href="#set_002drotamer_002dsearch_002dmode" accesskey="p" rel="previous">set-rotamer-search-mode</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drotamer_002dlowest_002dprobability-1"></a>
<h4 class="subsection">11.75.2 set-rotamer-lowest-probability</h4>
<dl>
<dt><a name="index-set_002drotamer_002dlowest_002dprobability"></a>function: <strong>set-rotamer-lowest-probability</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>For Dunbrack rotamers, set the lowest probability to be considered. Set as a percentage i.e. 1.00 is quite low. For Richardson Rotamers, this has no effect. 
</p>
</dd></dl>

<hr>
<a name="set_002drotamer_002dcheck_002dclashes"></a>
<div class="header">
<p>
Next: <a href="#auto_002dfit_002dbest_002drotamer" accesskey="n" rel="next">auto-fit-best-rotamer</a>, Previous: <a href="#set_002drotamer_002dlowest_002dprobability" accesskey="p" rel="previous">set-rotamer-lowest-probability</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drotamer_002dcheck_002dclashes-1"></a>
<h4 class="subsection">11.75.3 set-rotamer-check-clashes</h4>
<dl>
<dt><a name="index-set_002drotamer_002dcheck_002dclashes"></a>function: <strong>set-rotamer-check-clashes</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set a flag: 0 is off, 1 is on 
</p>
</dd></dl>

<hr>
<a name="auto_002dfit_002dbest_002drotamer"></a>
<div class="header">
<p>
Next: <a href="#set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag" accesskey="n" rel="next">set-auto-fit-best-rotamer-clash-flag</a>, Previous: <a href="#set_002drotamer_002dcheck_002dclashes" accesskey="p" rel="previous">set-rotamer-check-clashes</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="auto_002dfit_002dbest_002drotamer-1"></a>
<h4 class="subsection">11.75.4 auto-fit-best-rotamer</h4>
<dl>
<dt><a name="index-auto_002dfit_002dbest_002drotamer"></a>function: <strong>auto-fit-best-rotamer</strong> <em>resno altloc insertion_code chain_id imol_coords imol_map clash_flag lowest_probability</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>resno</em> is an integer number
</li><li> <em>altloc</em> is a string
</li><li> <em>insertion_code</em> is a string
</li><li> <em>chain_id</em> is a string
</li><li> <em>imol_coords</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li><li> <em>clash_flag</em> is an integer number
</li><li> <em>lowest_probability</em> is a number
</li></ul>

<p>auto fit by rotamer search. 
</p>
<p>return the score, for some not very good reason. clash_flag determines if we use clashes with other residues in the score for this rotamer (or not). It would be cool to call this from a script that went residue by residue along a (newly-built) chain (now available). 
</p>
</dd></dl>

<hr>
<a name="set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag"></a>
<div class="header">
<p>
Next: <a href="#n_002drotamers" accesskey="n" rel="next">n-rotamers</a>, Previous: <a href="#auto_002dfit_002dbest_002drotamer" accesskey="p" rel="previous">auto-fit-best-rotamer</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag-1"></a>
<h4 class="subsection">11.75.5 set-auto-fit-best-rotamer-clash-flag</h4>
<dl>
<dt><a name="index-set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag"></a>function: <strong>set-auto-fit-best-rotamer-clash-flag</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>set the clash flag for rotamer search 
</p>
<p>And this functions for [pre-setting] the variables for auto_fit_best_rotamer called interactively (using a graphics_info_t function). 0 off, 1 on. 
</p>
</dd></dl>

<hr>
<a name="n_002drotamers"></a>
<div class="header">
<p>
Next: <a href="#set_002dresidue_002dto_002drotamer_002dnumber" accesskey="n" rel="next">set-residue-to-rotamer-number</a>, Previous: <a href="#set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag" accesskey="p" rel="previous">set-auto-fit-best-rotamer-clash-flag</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="n_002drotamers-1"></a>
<h4 class="subsection">11.75.6 n-rotamers</h4>
<dl>
<dt><a name="index-n_002drotamers"></a>function: <strong>n-rotamers</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>return the number of rotamers for this residue - return -1 on no residue found. 
</p>
</dd></dl>

<hr>
<a name="set_002dresidue_002dto_002drotamer_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#set_002dresidue_002dto_002drotamer_002dname" accesskey="n" rel="next">set-residue-to-rotamer-name</a>, Previous: <a href="#n_002drotamers" accesskey="p" rel="previous">n-rotamers</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dresidue_002dto_002drotamer_002dnumber-1"></a>
<h4 class="subsection">11.75.7 set-residue-to-rotamer-number</h4>
<dl>
<dt><a name="index-set_002dresidue_002dto_002drotamer_002dnumber"></a>function: <strong>set-residue-to-rotamer-number</strong> <em>imol chain_id resno ins_code alt_conf rotamer_number</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>rotamer_number</em> is an integer number
</li></ul>

<p>set the residue specified to the rotamer number specifed. 
</p>
</dd></dl>

<hr>
<a name="set_002dresidue_002dto_002drotamer_002dname"></a>
<div class="header">
<p>
Next: <a href="#fill_002dpartial_002dresidues" accesskey="n" rel="next">fill-partial-residues</a>, Previous: <a href="#set_002dresidue_002dto_002drotamer_002dnumber" accesskey="p" rel="previous">set-residue-to-rotamer-number</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dresidue_002dto_002drotamer_002dname-1"></a>
<h4 class="subsection">11.75.8 set-residue-to-rotamer-name</h4>
<dl>
<dt><a name="index-set_002dresidue_002dto_002drotamer_002dname"></a>function: <strong>set-residue-to-rotamer-name</strong> <em>imol chain_id resno ins_code alt_conf rotamer_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>rotamer_name</em> is a string
</li></ul>

<p>set the residue specified to the rotamer name specified. 
</p>
<p>(rotamer names are the Richardson rotamer names.)
</p>
<p>return value is 0 if atoms were not moved (e.g. because rotamer-name was not know) 
</p>
</dd></dl>

<hr>
<a name="fill_002dpartial_002dresidues"></a>
<div class="header">
<p>
Next: <a href="#simple_002dfill_002dpartial_002dresidues" accesskey="n" rel="next">simple-fill-partial-residues</a>, Previous: <a href="#set_002dresidue_002dto_002drotamer_002dname" accesskey="p" rel="previous">set-residue-to-rotamer-name</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fill_002dpartial_002dresidues-1"></a>
<h4 class="subsection">11.75.9 fill-partial-residues</h4>
<dl>
<dt><a name="index-fill_002dpartial_002dresidues"></a>function: <strong>fill-partial-residues</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>fill all the residues of molecule number imol that have missing atoms. 
</p>
<p>To be used to remove the effects of chainsaw. 
</p>
</dd></dl>

<hr>
<a name="simple_002dfill_002dpartial_002dresidues"></a>
<div class="header">
<p>
Next: <a href="#rotamer_002dgraphs" accesskey="n" rel="next">rotamer-graphs</a>, Previous: <a href="#fill_002dpartial_002dresidues" accesskey="p" rel="previous">fill-partial-residues</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="simple_002dfill_002dpartial_002dresidues-1"></a>
<h4 class="subsection">11.75.10 simple-fill-partial-residues</h4>
<dl>
<dt><a name="index-simple_002dfill_002dpartial_002dresidues"></a>function: <strong>simple-fill-partial-residues</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Fill amino acid residues. 
</p>
<p>do backrub rotamer search for residues, but don&rsquo;t do refinement 
</p>
</dd></dl>

<hr>
<a name="rotamer_002dgraphs"></a>
<div class="header">
<p>
Next: <a href="#rotamer_002dgraphs_002dpy" accesskey="n" rel="next">rotamer-graphs-py</a>, Previous: <a href="#simple_002dfill_002dpartial_002dresidues" accesskey="p" rel="previous">simple-fill-partial-residues</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rotamer_002dgraphs-1"></a>
<h4 class="subsection">11.75.11 rotamer-graphs</h4>
<dl>
<dt><a name="index-rotamer_002dgraphs"></a>function: <strong>rotamer-graphs</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Activate rotamer graph analysis for molecule number imol. 
</p>
<p>Return rotamer info - function used in testing. 
</p>
</dd></dl>

<hr>
<a name="rotamer_002dgraphs_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#rotamer_002dgraphs" accesskey="p" rel="previous">rotamer-graphs</a>, Up: <a href="#Rotamer-Functions" accesskey="u" rel="up">Rotamer Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rotamer_002dgraphs_002dpy-1"></a>
<h4 class="subsection">11.75.12 rotamer-graphs-py</h4>
<dl>
<dt><a name="index-rotamer_002dgraphs_002dpy"></a>function: <strong>rotamer-graphs-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Activate rotamer graph analysis for molecule number imol. 
</p>
<p>Return rotamer info - function used in testing. 
</p>
</dd></dl>

<hr>
<a name="g_t180-Flip-Side-chain"></a>
<div class="header">
<p>
Next: <a href="#Mutate-Functions" accesskey="n" rel="next">Mutate Functions</a>, Previous: <a href="#Rotamer-Functions" accesskey="p" rel="previous">Rotamer Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="g_t180-Flip-Side-chain-1"></a>
<h3 class="section">11.76 180 Flip Side chain</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#do_002d180_002ddegree_002dside_002dchain_002dflip" accesskey="1">do-180-degree-side-chain-flip</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="do_002d180_002ddegree_002dside_002dchain_002dflip"></a>
<div class="header">
<p>
Up: <a href="#g_t180-Flip-Side-chain" accesskey="u" rel="up">180 Flip Side chain</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002d180_002ddegree_002dside_002dchain_002dflip-1"></a>
<h4 class="subsection">11.76.1 do-180-degree-side-chain-flip</h4>
<dl>
<dt><a name="index-do_002d180_002ddegree_002dside_002dchain_002dflip"></a>function: <strong>do-180-degree-side-chain-flip</strong> <em>imol chain_id resno inscode altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>inscode</em> is a string
</li><li> <em>altconf</em> is a string
</li></ul>

<p>rotate 180 degrees round the last chi angle 
</p>
</dd></dl>

<hr>
<a name="Mutate-Functions"></a>
<div class="header">
<p>
Next: <a href="#Alternative-Conformation" accesskey="n" rel="next">Alternative Conformation</a>, Previous: <a href="#g_t180-Flip-Side-chain" accesskey="p" rel="previous">180 Flip Side chain</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Mutate-Functions-1"></a>
<h3 class="section">11.77 Mutate Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#setup_002dmutate_002dauto_002dfit" accesskey="1">setup-mutate-auto-fit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mutate_002dactive_002dresidue" accesskey="2">mutate-active-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mutate" accesskey="3">mutate</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mutate_002dbase" accesskey="4">mutate-base</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine" accesskey="5">set-mutate-auto-fit-do-post-refine</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate" accesskey="6">mutate-auto-fit-do-post-refine-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine" accesskey="7">set-rotamer-auto-fit-do-post-refine</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate" accesskey="8">rotamer-auto-fit-do-post-refine-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber" accesskey="9">mutate-single-residue-by-serial-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dresidue_002dtype_002dchooser_002dstub_002dstate">set-residue-type-chooser-stub-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="setup_002dmutate_002dauto_002dfit"></a>
<div class="header">
<p>
Next: <a href="#mutate_002dactive_002dresidue" accesskey="n" rel="next">mutate-active-residue</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="setup_002dmutate_002dauto_002dfit-1"></a>
<h4 class="subsection">11.77.1 setup-mutate-auto-fit</h4>
<dl>
<dt><a name="index-setup_002dmutate_002dauto_002dfit"></a>function: <strong>setup-mutate-auto-fit</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>Mutate then fit to map. 
</p>
<p>that we have a map define is checked first 
</p>
</dd></dl>

<hr>
<a name="mutate_002dactive_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#mutate" accesskey="n" rel="next">mutate</a>, Previous: <a href="#setup_002dmutate_002dauto_002dfit" accesskey="p" rel="previous">setup-mutate-auto-fit</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mutate_002dactive_002dresidue-1"></a>
<h4 class="subsection">11.77.2 mutate-active-residue</h4>
<dl>
<dt><a name="index-mutate_002dactive_002dresidue"></a>function: <strong>mutate-active-residue</strong></dt>
<dd>
<p>display a dialog that allows the choice of residue type to which to mutate 
</p>
</dd></dl>

<hr>
<a name="mutate"></a>
<div class="header">
<p>
Next: <a href="#mutate_002dbase" accesskey="n" rel="next">mutate-base</a>, Previous: <a href="#mutate_002dactive_002dresidue" accesskey="p" rel="previous">mutate-active-residue</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mutate-1"></a>
<h4 class="subsection">11.77.3 mutate</h4>
<dl>
<dt><a name="index-mutate-1"></a>function: <strong>mutate</strong> <em>imol chain_id ires inscode target_res_type</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>ires</em> is an integer number
</li><li> <em>inscode</em> is a string
</li><li> <em>target_res_type</em> is a string
</li></ul>

<p>mutate a given residue 
</p>
<p>target_res_type is a three-letter-code.
</p>
<p>Return 1 on a good mutate. 
</p>
</dd></dl>

<hr>
<a name="mutate_002dbase"></a>
<div class="header">
<p>
Next: <a href="#set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine" accesskey="n" rel="next">set-mutate-auto-fit-do-post-refine</a>, Previous: <a href="#mutate" accesskey="p" rel="previous">mutate</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mutate_002dbase-1"></a>
<h4 class="subsection">11.77.4 mutate-base</h4>
<dl>
<dt><a name="index-mutate_002dbase"></a>function: <strong>mutate-base</strong> <em>imol chain_id res_no ins_code res_type</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>res_type</em> is a string
</li></ul>

<p>mutate a base. return success status, 1 for a good mutate. 
</p>
</dd></dl>

<hr>
<a name="set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine"></a>
<div class="header">
<p>
Next: <a href="#mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate" accesskey="n" rel="next">mutate-auto-fit-do-post-refine-state</a>, Previous: <a href="#mutate_002dbase" accesskey="p" rel="previous">mutate-base</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine-1"></a>
<h4 class="subsection">11.77.5 set-mutate-auto-fit-do-post-refine</h4>
<dl>
<dt><a name="index-set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine"></a>function: <strong>set-mutate-auto-fit-do-post-refine</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>Do you want Coot to automatically run a refinement after every mutate and autofit? 
</p>
<p>1 for yes, 0 for no. 
</p>
</dd></dl>

<hr>
<a name="mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine" accesskey="n" rel="next">set-rotamer-auto-fit-do-post-refine</a>, Previous: <a href="#set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine" accesskey="p" rel="previous">set-mutate-auto-fit-do-post-refine</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate-1"></a>
<h4 class="subsection">11.77.6 mutate-auto-fit-do-post-refine-state</h4>
<dl>
<dt><a name="index-mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate"></a>function: <strong>mutate-auto-fit-do-post-refine-state</strong></dt>
<dd>
<p>what is the value of the previous flag? 
</p>
</dd></dl>

<hr>
<a name="set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine"></a>
<div class="header">
<p>
Next: <a href="#rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate" accesskey="n" rel="next">rotamer-auto-fit-do-post-refine-state</a>, Previous: <a href="#mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate" accesskey="p" rel="previous">mutate-auto-fit-do-post-refine-state</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine-1"></a>
<h4 class="subsection">11.77.7 set-rotamer-auto-fit-do-post-refine</h4>
<dl>
<dt><a name="index-set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine"></a>function: <strong>set-rotamer-auto-fit-do-post-refine</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>Do you want Coot to automatically run a refinement after every rotamer autofit? 
</p>
<p>1 for yes, 0 for no. 
</p>
</dd></dl>

<hr>
<a name="rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate"></a>
<div class="header">
<p>
Next: <a href="#mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber" accesskey="n" rel="next">mutate-single-residue-by-serial-number</a>, Previous: <a href="#set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine" accesskey="p" rel="previous">set-rotamer-auto-fit-do-post-refine</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate-1"></a>
<h4 class="subsection">11.77.8 rotamer-auto-fit-do-post-refine-state</h4>
<dl>
<dt><a name="index-rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate"></a>function: <strong>rotamer-auto-fit-do-post-refine-state</strong></dt>
<dd>
<p>what is the value of the previous flag? 
</p>
</dd></dl>

<hr>
<a name="mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#set_002dresidue_002dtype_002dchooser_002dstub_002dstate" accesskey="n" rel="next">set-residue-type-chooser-stub-state</a>, Previous: <a href="#rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate" accesskey="p" rel="previous">rotamer-auto-fit-do-post-refine-state</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber-1"></a>
<h4 class="subsection">11.77.9 mutate-single-residue-by-serial-number</h4>
<dl>
<dt><a name="index-mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber"></a>function: <strong>mutate-single-residue-by-serial-number</strong> <em>ires_ser chain_id imol target_res_type</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>ires_ser</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>imol</em> is an integer number
</li><li> <em>target_res_type</em> is a character
</li></ul>

<p>an alternate interface to mutation of a singe residue. 
</p>

<p>ires-ser is the serial number of the residue, not the seqnum There 2 functions don&rsquo;t make backups, but 
</p>
<p>does - CHECKME Hence 
</p>
<p>is for use as a &quot;one-by-one&quot; type and the following 2 by wrappers that muate either a residue range or a whole chain
</p>
<p>Note that the target_res_type is a char, not a string (or a char *). So from the scheme interface you&rsquo;d use (for example) hash backslash A for ALA. 
</p>
<p>Returns: 1 on success, 0 on failure
</p>
</dd></dl>

<hr>
<a name="set_002dresidue_002dtype_002dchooser_002dstub_002dstate"></a>
<div class="header">
<p>
Previous: <a href="#mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber" accesskey="p" rel="previous">mutate-single-residue-by-serial-number</a>, Up: <a href="#Mutate-Functions" accesskey="u" rel="up">Mutate Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dresidue_002dtype_002dchooser_002dstub_002dstate-1"></a>
<h4 class="subsection">11.77.10 set-residue-type-chooser-stub-state</h4>
<dl>
<dt><a name="index-set_002dresidue_002dtype_002dchooser_002dstub_002dstate"></a>function: <strong>set-residue-type-chooser-stub-state</strong> <em>istat</em></dt>
<dd>
<p>Where <em>istat</em> is an integer number
</p>
<p>set a flag saying that the residue chosen by mutate or auto-fit mutate should only be added as a stub (mainchain + CB) 
</p>
</dd></dl>

<hr>
<a name="Alternative-Conformation"></a>
<div class="header">
<p>
Next: <a href="#Pointer-Atom-Functions" accesskey="n" rel="next">Pointer Atom Functions</a>, Previous: <a href="#Mutate-Functions" accesskey="p" rel="previous">Mutate Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Alternative-Conformation-1"></a>
<h3 class="section">11.78 Alternative Conformation</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#add_002dalt_002dconf_002dscm" accesskey="1">add-alt-conf-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dalt_002dconf_002dpy" accesskey="2">add-alt-conf-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="add_002dalt_002dconf_002dscm"></a>
<div class="header">
<p>
Next: <a href="#add_002dalt_002dconf_002dpy" accesskey="n" rel="next">add-alt-conf-py</a>, Up: <a href="#Alternative-Conformation" accesskey="u" rel="up">Alternative Conformation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dalt_002dconf_002dscm-1"></a>
<h4 class="subsection">11.78.1 add-alt-conf-scm</h4>
<dl>
<dt><a name="index-add_002dalt_002dconf_002dscm"></a>function: <strong>add-alt-conf-scm</strong> <em>imol chain_id res_no ins_code alt_conf rotamer_number</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>rotamer_number</em> is an integer number
</li></ul>

<p>add an alternative conformer to a residue. Add it in conformation rotamer number rotamer_number. 
</p>
<p>Return the new alt_conf chain_id on sucess, scheme false on fail 
</p>
</dd></dl>

<hr>
<a name="add_002dalt_002dconf_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#add_002dalt_002dconf_002dscm" accesskey="p" rel="previous">add-alt-conf-scm</a>, Up: <a href="#Alternative-Conformation" accesskey="u" rel="up">Alternative Conformation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dalt_002dconf_002dpy-1"></a>
<h4 class="subsection">11.78.2 add-alt-conf-py</h4>
<dl>
<dt><a name="index-add_002dalt_002dconf_002dpy"></a>function: <strong>add-alt-conf-py</strong> <em>imol chain_id res_no ins_code alt_conf rotamer_number</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>rotamer_number</em> is an integer number
</li></ul>

<p>add an alternative conformer to a residue. Add it in conformation rotamer number rotamer_number. 
</p>
<p>Return the new alt_conf chain_id on sucess, python False on fail 
</p>
</dd></dl>

<hr>
<a name="Pointer-Atom-Functions"></a>
<div class="header">
<p>
Next: <a href="#Baton-Build-Interface-Functions" accesskey="n" rel="next">Baton Build Interface Functions</a>, Previous: <a href="#Alternative-Conformation" accesskey="p" rel="previous">Alternative Conformation</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Pointer-Atom-Functions-1"></a>
<h3 class="section">11.79 Pointer Atom Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#create_002dpointer_002datom_002dmolecule_002dmaybe" accesskey="1">create-pointer-atom-molecule-maybe</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#pointer_002datom_002dmolecule" accesskey="2">pointer-atom-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="create_002dpointer_002datom_002dmolecule_002dmaybe"></a>
<div class="header">
<p>
Next: <a href="#pointer_002datom_002dmolecule" accesskey="n" rel="next">pointer-atom-molecule</a>, Up: <a href="#Pointer-Atom-Functions" accesskey="u" rel="up">Pointer Atom Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="create_002dpointer_002datom_002dmolecule_002dmaybe-1"></a>
<h4 class="subsection">11.79.1 create-pointer-atom-molecule-maybe</h4>
<dl>
<dt><a name="index-create_002dpointer_002datom_002dmolecule_002dmaybe"></a>function: <strong>create-pointer-atom-molecule-maybe</strong></dt>
<dd>
<p>Return the current pointer atom molecule, create a pointer atom molecule if necessary (i.e. when the user has not set it). 
</p>
</dd></dl>

<hr>
<a name="pointer_002datom_002dmolecule"></a>
<div class="header">
<p>
Previous: <a href="#create_002dpointer_002datom_002dmolecule_002dmaybe" accesskey="p" rel="previous">create-pointer-atom-molecule-maybe</a>, Up: <a href="#Pointer-Atom-Functions" accesskey="u" rel="up">Pointer Atom Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="pointer_002datom_002dmolecule-1"></a>
<h4 class="subsection">11.79.2 pointer-atom-molecule</h4>
<dl>
<dt><a name="index-pointer_002datom_002dmolecule"></a>function: <strong>pointer-atom-molecule</strong></dt>
<dd>
<p>Return the current pointer atom molecule. 
</p>
</dd></dl>

<hr>
<a name="Baton-Build-Interface-Functions"></a>
<div class="header">
<p>
Next: <a href="#Crosshairs-Interface" accesskey="n" rel="next">Crosshairs  Interface</a>, Previous: <a href="#Pointer-Atom-Functions" accesskey="p" rel="previous">Pointer Atom Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Baton-Build-Interface-Functions-1"></a>
<h3 class="section">11.80 Baton Build Interface Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dbaton_002dmode" accesskey="1">set-baton-mode</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#try_002dset_002ddraw_002dbaton" accesskey="2">try-set-draw-baton</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#accept_002dbaton_002dposition" accesskey="3">accept-baton-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#baton_002dtip_002dtry_002danother" accesskey="4">baton-tip-try-another</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#baton_002dtip_002dprevious" accesskey="5">baton-tip-previous</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#shorten_002dbaton" accesskey="6">shorten-baton</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#lengthen_002dbaton" accesskey="7">lengthen-baton</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#baton_002dbuild_002ddelete_002dlast_002dresidue" accesskey="8">baton-build-delete-last-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dbaton_002dbuild_002dparams" accesskey="9">set-baton-build-params</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dbaton_002dmode"></a>
<div class="header">
<p>
Next: <a href="#try_002dset_002ddraw_002dbaton" accesskey="n" rel="next">try-set-draw-baton</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbaton_002dmode-1"></a>
<h4 class="subsection">11.80.1 set-baton-mode</h4>
<dl>
<dt><a name="index-set_002dbaton_002dmode"></a>function: <strong>set-baton-mode</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>toggle so that mouse movement moves the baton not rotates the view. 
</p>
</dd></dl>

<hr>
<a name="try_002dset_002ddraw_002dbaton"></a>
<div class="header">
<p>
Next: <a href="#accept_002dbaton_002dposition" accesskey="n" rel="next">accept-baton-position</a>, Previous: <a href="#set_002dbaton_002dmode" accesskey="p" rel="previous">set-baton-mode</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="try_002dset_002ddraw_002dbaton-1"></a>
<h4 class="subsection">11.80.2 try-set-draw-baton</h4>
<dl>
<dt><a name="index-try_002dset_002ddraw_002dbaton"></a>function: <strong>try-set-draw-baton</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>draw the baton or not 
</p>
</dd></dl>

<hr>
<a name="accept_002dbaton_002dposition"></a>
<div class="header">
<p>
Next: <a href="#baton_002dtip_002dtry_002danother" accesskey="n" rel="next">baton-tip-try-another</a>, Previous: <a href="#try_002dset_002ddraw_002dbaton" accesskey="p" rel="previous">try-set-draw-baton</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="accept_002dbaton_002dposition-1"></a>
<h4 class="subsection">11.80.3 accept-baton-position</h4>
<dl>
<dt><a name="index-accept_002dbaton_002dposition"></a>function: <strong>accept-baton-position</strong></dt>
<dd>
<p>accept the baton tip position - a prime candidate for a key binding 
</p>
</dd></dl>

<hr>
<a name="baton_002dtip_002dtry_002danother"></a>
<div class="header">
<p>
Next: <a href="#baton_002dtip_002dprevious" accesskey="n" rel="next">baton-tip-previous</a>, Previous: <a href="#accept_002dbaton_002dposition" accesskey="p" rel="previous">accept-baton-position</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="baton_002dtip_002dtry_002danother-1"></a>
<h4 class="subsection">11.80.4 baton-tip-try-another</h4>
<dl>
<dt><a name="index-baton_002dtip_002dtry_002danother"></a>function: <strong>baton-tip-try-another</strong></dt>
<dd>
<p>move the baton tip position - another prime candidate for a key binding 
</p>
</dd></dl>

<hr>
<a name="baton_002dtip_002dprevious"></a>
<div class="header">
<p>
Next: <a href="#shorten_002dbaton" accesskey="n" rel="next">shorten-baton</a>, Previous: <a href="#baton_002dtip_002dtry_002danother" accesskey="p" rel="previous">baton-tip-try-another</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="baton_002dtip_002dprevious-1"></a>
<h4 class="subsection">11.80.5 baton-tip-previous</h4>
<dl>
<dt><a name="index-baton_002dtip_002dprevious"></a>function: <strong>baton-tip-previous</strong></dt>
<dd>
<p>move the baton tip to the previous position 
</p>
</dd></dl>

<hr>
<a name="shorten_002dbaton"></a>
<div class="header">
<p>
Next: <a href="#lengthen_002dbaton" accesskey="n" rel="next">lengthen-baton</a>, Previous: <a href="#baton_002dtip_002dprevious" accesskey="p" rel="previous">baton-tip-previous</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="shorten_002dbaton-1"></a>
<h4 class="subsection">11.80.6 shorten-baton</h4>
<dl>
<dt><a name="index-shorten_002dbaton"></a>function: <strong>shorten-baton</strong></dt>
<dd>
<p>shorten the baton length 
</p>
</dd></dl>

<hr>
<a name="lengthen_002dbaton"></a>
<div class="header">
<p>
Next: <a href="#baton_002dbuild_002ddelete_002dlast_002dresidue" accesskey="n" rel="next">baton-build-delete-last-residue</a>, Previous: <a href="#shorten_002dbaton" accesskey="p" rel="previous">shorten-baton</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="lengthen_002dbaton-1"></a>
<h4 class="subsection">11.80.7 lengthen-baton</h4>
<dl>
<dt><a name="index-lengthen_002dbaton"></a>function: <strong>lengthen-baton</strong></dt>
<dd>
<p>lengthen the baton 
</p>
</dd></dl>

<hr>
<a name="baton_002dbuild_002ddelete_002dlast_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#set_002dbaton_002dbuild_002dparams" accesskey="n" rel="next">set-baton-build-params</a>, Previous: <a href="#lengthen_002dbaton" accesskey="p" rel="previous">lengthen-baton</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="baton_002dbuild_002ddelete_002dlast_002dresidue-1"></a>
<h4 class="subsection">11.80.8 baton-build-delete-last-residue</h4>
<dl>
<dt><a name="index-baton_002dbuild_002ddelete_002dlast_002dresidue"></a>function: <strong>baton-build-delete-last-residue</strong></dt>
<dd>
<p>delete the most recently build CA position 
</p>
</dd></dl>

<hr>
<a name="set_002dbaton_002dbuild_002dparams"></a>
<div class="header">
<p>
Previous: <a href="#baton_002dbuild_002ddelete_002dlast_002dresidue" accesskey="p" rel="previous">baton-build-delete-last-residue</a>, Up: <a href="#Baton-Build-Interface-Functions" accesskey="u" rel="up">Baton Build Interface Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbaton_002dbuild_002dparams-1"></a>
<h4 class="subsection">11.80.9 set-baton-build-params</h4>
<dl>
<dt><a name="index-set_002dbaton_002dbuild_002dparams"></a>function: <strong>set-baton-build-params</strong> <em>istart_resno chain_id direction</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>istart_resno</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>direction</em> is a string
</li></ul>

<p>set the parameters for the start of a new baton-built fragment. direction can either be &quot;forwards&quot; or &quot;backwards&quot; 
</p>
</dd></dl>

<hr>
<a name="Crosshairs-Interface"></a>
<div class="header">
<p>
Next: <a href="#Edit-Chi-Angles" accesskey="n" rel="next">Edit Chi Angles</a>, Previous: <a href="#Baton-Build-Interface-Functions" accesskey="p" rel="previous">Baton Build Interface Functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Crosshairs-Interface-1"></a>
<h3 class="section">11.81 Crosshairs  Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dcrosshairs" accesskey="1">set-draw-crosshairs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002ddraw_002dcrosshairs"></a>
<div class="header">
<p>
Up: <a href="#Crosshairs-Interface" accesskey="u" rel="up">Crosshairs  Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dcrosshairs-1"></a>
<h4 class="subsection">11.81.1 set-draw-crosshairs</h4>
<dl>
<dt><a name="index-set_002ddraw_002dcrosshairs"></a>function: <strong>set-draw-crosshairs</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>draw the distance crosshairs, 0 for off, 1 for on. 
</p>
</dd></dl>

<hr>
<a name="Edit-Chi-Angles"></a>
<div class="header">
<p>
Next: <a href="#Masks" accesskey="n" rel="next">Masks</a>, Previous: <a href="#Crosshairs-Interface" accesskey="p" rel="previous">Crosshairs  Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Edit-Chi-Angles-1"></a>
<h3 class="section">11.82 Edit Chi Angles</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dfind_002dhydrogen_002dtorsions" accesskey="1">set-find-hydrogen-torsions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#edit_002dchi_002dangles" accesskey="2">edit-chi-angles</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#setup_002dtorsion_002dgeneral" accesskey="3">setup-torsion-general</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dfind_002dhydrogen_002dtorsions"></a>
<div class="header">
<p>
Next: <a href="#edit_002dchi_002dangles" accesskey="n" rel="next">edit-chi-angles</a>, Up: <a href="#Edit-Chi-Angles" accesskey="u" rel="up">Edit Chi Angles</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfind_002dhydrogen_002dtorsions-1"></a>
<h4 class="subsection">11.82.1 set-find-hydrogen-torsions</h4>
<dl>
<dt><a name="index-set_002dfind_002dhydrogen_002dtorsions"></a>function: <strong>set-find-hydrogen-torsions</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>show torsions that rotate hydrogens in the torsion angle manipulation dialog. Note that this may be needed if, in the dictionary cif file torsion which have as a 4th atom both a hydrogen and a heavier atom bonding to the 3rd atom, but list the 4th atom as a hydrogen (not a heavier atom). 
</p>
</dd></dl>

<hr>
<a name="edit_002dchi_002dangles"></a>
<div class="header">
<p>
Next: <a href="#setup_002dtorsion_002dgeneral" accesskey="n" rel="next">setup-torsion-general</a>, Previous: <a href="#set_002dfind_002dhydrogen_002dtorsions" accesskey="p" rel="previous">set-find-hydrogen-torsions</a>, Up: <a href="#Edit-Chi-Angles" accesskey="u" rel="up">Edit Chi Angles</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="edit_002dchi_002dangles-1"></a>
<h4 class="subsection">11.82.2 edit-chi-angles</h4>
<dl>
<dt><a name="index-edit_002dchi_002dangles"></a>function: <strong>edit-chi-angles</strong> <em>imol chain_id resno ins_code altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>altconf</em> is a string
</li></ul>

<p>display the edit chi angles gui for the given residue 
</p>
<p>return a status of 0 if it failed to fined the residue, return a value of 1 if it worked. 
</p>
</dd></dl>

<hr>
<a name="setup_002dtorsion_002dgeneral"></a>
<div class="header">
<p>
Previous: <a href="#edit_002dchi_002dangles" accesskey="p" rel="previous">edit-chi-angles</a>, Up: <a href="#Edit-Chi-Angles" accesskey="u" rel="up">Edit Chi Angles</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="setup_002dtorsion_002dgeneral-1"></a>
<h4 class="subsection">11.82.3 setup-torsion-general</h4>
<dl>
<dt><a name="index-setup_002dtorsion_002dgeneral"></a>function: <strong>setup-torsion-general</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>beloved torsion general at last makes an entrance onto the Coot scene... 
</p>
</dd></dl>

<hr>
<a name="Masks"></a>
<div class="header">
<p>
Next: <a href="#check-Waters-Interface" accesskey="n" rel="next">check Waters Interface</a>, Previous: <a href="#Edit-Chi-Angles" accesskey="p" rel="previous">Edit Chi Angles</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Masks-2"></a>
<h3 class="section">11.83 Masks</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#mask_002dmap_002dby_002dmolecule" accesskey="1">mask-map-by-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mask_002dmap_002dby_002datom_002dselection" accesskey="2">mask-map-by-atom-selection</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dmasked_002dmaps_002dsplit_002dby_002dchain" accesskey="3">make-masked-maps-split-by-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dmask_002datom_002dradius" accesskey="4">set-map-mask-atom-radius</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dmask_002datom_002dradius" accesskey="5">map-mask-atom-radius</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="mask_002dmap_002dby_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#mask_002dmap_002dby_002datom_002dselection" accesskey="n" rel="next">mask-map-by-atom-selection</a>, Up: <a href="#Masks" accesskey="u" rel="up">Masks</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mask_002dmap_002dby_002dmolecule-1"></a>
<h4 class="subsection">11.83.1 mask-map-by-molecule</h4>
<dl>
<dt><a name="index-mask_002dmap_002dby_002dmolecule"></a>function: <strong>mask-map-by-molecule</strong> <em>map_mol_no coord_mol_no invert_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>map_mol_no</em> is an integer number
</li><li> <em>coord_mol_no</em> is an integer number
</li><li> <em>invert_flag</em> is an integer number
</li></ul>

<p>generate a new map that has been masked by some coordinates 
</p>
<p>(mask-map-by-molecule map-no mol-no invert?) creates and displays a masked map, cuts down density where the coordinates are (invert is 0). If invert? is 1, cut the density down where there are no atoms atoms. 
</p>
</dd></dl>

<hr>
<a name="mask_002dmap_002dby_002datom_002dselection"></a>
<div class="header">
<p>
Next: <a href="#make_002dmasked_002dmaps_002dsplit_002dby_002dchain" accesskey="n" rel="next">make-masked-maps-split-by-chain</a>, Previous: <a href="#mask_002dmap_002dby_002dmolecule" accesskey="p" rel="previous">mask-map-by-molecule</a>, Up: <a href="#Masks" accesskey="u" rel="up">Masks</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mask_002dmap_002dby_002datom_002dselection-1"></a>
<h4 class="subsection">11.83.2 mask-map-by-atom-selection</h4>
<dl>
<dt><a name="index-mask_002dmap_002dby_002datom_002dselection"></a>function: <strong>mask-map-by-atom-selection</strong> <em>map_mol_no coords_mol_no mmdb_atom_selection invert_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>map_mol_no</em> is an integer number
</li><li> <em>coords_mol_no</em> is an integer number
</li><li> <em>mmdb_atom_selection</em> is a string
</li><li> <em>invert_flag</em> is an integer number
</li></ul>

<p>mask map by atom selection 
</p>
</dd></dl>

<hr>
<a name="make_002dmasked_002dmaps_002dsplit_002dby_002dchain"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002dmask_002datom_002dradius" accesskey="n" rel="next">set-map-mask-atom-radius</a>, Previous: <a href="#mask_002dmap_002dby_002datom_002dselection" accesskey="p" rel="previous">mask-map-by-atom-selection</a>, Up: <a href="#Masks" accesskey="u" rel="up">Masks</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dmasked_002dmaps_002dsplit_002dby_002dchain-1"></a>
<h4 class="subsection">11.83.3 make-masked-maps-split-by-chain</h4>
<dl>
<dt><a name="index-make_002dmasked_002dmaps_002dsplit_002dby_002dchain"></a>function: <strong>make-masked-maps-split-by-chain</strong> <em>imol imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>make chain masked maps 
</p>
<p>needs to return a list of values 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002dmask_002datom_002dradius"></a>
<div class="header">
<p>
Next: <a href="#map_002dmask_002datom_002dradius" accesskey="n" rel="next">map-mask-atom-radius</a>, Previous: <a href="#make_002dmasked_002dmaps_002dsplit_002dby_002dchain" accesskey="p" rel="previous">make-masked-maps-split-by-chain</a>, Up: <a href="#Masks" accesskey="u" rel="up">Masks</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dmask_002datom_002dradius-1"></a>
<h4 class="subsection">11.83.4 set-map-mask-atom-radius</h4>
<dl>
<dt><a name="index-set_002dmap_002dmask_002datom_002dradius"></a>function: <strong>set-map-mask-atom-radius</strong> <em>rad</em></dt>
<dd>
<p>Where <em>rad</em> is a number
</p>
<p>set the atom radius for map masking 
</p>
</dd></dl>

<hr>
<a name="map_002dmask_002datom_002dradius"></a>
<div class="header">
<p>
Previous: <a href="#set_002dmap_002dmask_002datom_002dradius" accesskey="p" rel="previous">set-map-mask-atom-radius</a>, Up: <a href="#Masks" accesskey="u" rel="up">Masks</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dmask_002datom_002dradius-1"></a>
<h4 class="subsection">11.83.5 map-mask-atom-radius</h4>
<dl>
<dt><a name="index-map_002dmask_002datom_002dradius"></a>function: <strong>map-mask-atom-radius</strong></dt>
<dd>
<p>get the atom radius for map masking 
</p>
</dd></dl>

<hr>
<a name="check-Waters-Interface"></a>
<div class="header">
<p>
Next: <a href="#Least_002dSquares-matching" accesskey="n" rel="next">Least-Squares matching</a>, Previous: <a href="#Masks" accesskey="p" rel="previous">Masks</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="check-Waters-Interface-1"></a>
<h3 class="section">11.84 check Waters Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#delete_002dchecked_002dwaters_002dbaddies" accesskey="1">delete-checked-waters-baddies</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#highly_002dcoordinated_002dwaters_002dscm" accesskey="2">highly-coordinated-waters-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#highly_002dcoordinated_002dwaters_002dpy" accesskey="3">highly-coordinated-waters-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="delete_002dchecked_002dwaters_002dbaddies"></a>
<div class="header">
<p>
Next: <a href="#highly_002dcoordinated_002dwaters_002dscm" accesskey="n" rel="next">highly-coordinated-waters-scm</a>, Up: <a href="#check-Waters-Interface" accesskey="u" rel="up">check Waters Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dchecked_002dwaters_002dbaddies-1"></a>
<h4 class="subsection">11.84.1 delete-checked-waters-baddies</h4>
<dl>
<dt><a name="index-delete_002dchecked_002dwaters_002dbaddies"></a>function: <strong>delete-checked-waters-baddies</strong> <em>imol b_factor_lim map_sigma_lim min_dist max_dist part_occ_contact_flag zero_occ_flag logical_operator_and_or_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>b_factor_lim</em> is a number
</li><li> <em>map_sigma_lim</em> is a number
</li><li> <em>min_dist</em> is a number
</li><li> <em>max_dist</em> is a number
</li><li> <em>part_occ_contact_flag</em> is an integer number
</li><li> <em>zero_occ_flag</em> is an integer number
</li><li> <em>logical_operator_and_or_flag</em> is an integer number
</li></ul>

<p>Delete waters that are fail to meet the given criteria. 
</p>
</dd></dl>

<hr>
<a name="highly_002dcoordinated_002dwaters_002dscm"></a>
<div class="header">
<p>
Next: <a href="#highly_002dcoordinated_002dwaters_002dpy" accesskey="n" rel="next">highly-coordinated-waters-py</a>, Previous: <a href="#delete_002dchecked_002dwaters_002dbaddies" accesskey="p" rel="previous">delete-checked-waters-baddies</a>, Up: <a href="#check-Waters-Interface" accesskey="u" rel="up">check Waters Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="highly_002dcoordinated_002dwaters_002dscm-1"></a>
<h4 class="subsection">11.84.2 highly-coordinated-waters-scm</h4>
<dl>
<dt><a name="index-highly_002dcoordinated_002dwaters_002dscm"></a>function: <strong>highly-coordinated-waters-scm</strong> <em>imol coordination_number dist_max</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>coordination_number</em> is an integer number
</li><li> <em>dist_max</em> is a number
</li></ul>

<p>return an improper list first is list of metals, second is list of waters that are coordinated with at least coordination_number of other atoms at distances less than or equal to dist_max. Return scheme false on not able to make a list, otherwise a list of atoms and neighbours. Can return scheme false if imol is not a valid molecule. 
</p>
</dd></dl>

<hr>
<a name="highly_002dcoordinated_002dwaters_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#highly_002dcoordinated_002dwaters_002dscm" accesskey="p" rel="previous">highly-coordinated-waters-scm</a>, Up: <a href="#check-Waters-Interface" accesskey="u" rel="up">check Waters Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="highly_002dcoordinated_002dwaters_002dpy-1"></a>
<h4 class="subsection">11.84.3 highly-coordinated-waters-py</h4>
<dl>
<dt><a name="index-highly_002dcoordinated_002dwaters_002dpy"></a>function: <strong>highly-coordinated-waters-py</strong> <em>imol coordination_number dist_max</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>coordination_number</em> is an integer number
</li><li> <em>dist_max</em> is a number
</li></ul>

<p>return a list first of waters, second metals that are coordinated with at least coordination_number of other atoms at distances less than or equal to dist_max. Return Python false on not able to make a list, otherwise a list of atoms and neighours. can return Python False if imol is not a valid molecule. 
</p>
</dd></dl>

<hr>
<a name="Least_002dSquares-matching"></a>
<div class="header">
<p>
Next: <a href="#Trim" accesskey="n" rel="next">Trim</a>, Previous: <a href="#check-Waters-Interface" accesskey="p" rel="previous">check Waters Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Least_002dSquares-matching-1"></a>
<h3 class="section">11.85 Least-Squares matching</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#apply_002dlsq_002dmatches" accesskey="1">apply-lsq-matches</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="apply_002dlsq_002dmatches"></a>
<div class="header">
<p>
Up: <a href="#Least_002dSquares-matching" accesskey="u" rel="up">Least-Squares matching</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="apply_002dlsq_002dmatches-1"></a>
<h4 class="subsection">11.85.1 apply-lsq-matches</h4>
<dl>
<dt><a name="index-apply_002dlsq_002dmatches"></a>function: <strong>apply-lsq-matches</strong> <em>imol_reference imol_moving</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_reference</em> is an integer number
</li><li> <em>imol_moving</em> is an integer number
</li></ul>

<p>apply the LSQ matches 
</p>
<p>Returns: an rtop pair (proper list) on good match, else false 
</p>
</dd></dl>

<hr>
<a name="Trim"></a>
<div class="header">
<p>
Next: <a href="#External-Ray_002dTracing" accesskey="n" rel="next">External Ray-Tracing</a>, Previous: <a href="#Least_002dSquares-matching" accesskey="p" rel="previous">Least-Squares matching</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Trim-1"></a>
<h3 class="section">11.86 Trim</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#trim_002dmolecule_002dby_002dmap" accesskey="1">trim-molecule-by-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="trim_002dmolecule_002dby_002dmap"></a>
<div class="header">
<p>
Up: <a href="#Trim" accesskey="u" rel="up">Trim</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="trim_002dmolecule_002dby_002dmap-1"></a>
<h4 class="subsection">11.86.1 trim-molecule-by-map</h4>
<dl>
<dt><a name="index-trim_002dmolecule_002dby_002dmap"></a>function: <strong>trim-molecule-by-map</strong> <em>imol_coords imol_map map_level delete_or_zero_occ_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_coords</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li><li> <em>map_level</em> is a number
</li><li> <em>delete_or_zero_occ_flag</em> is an integer number
</li></ul>

<p>cut off (delete or give zero occupancy) atoms in the given molecule if they are below the given map (absolute) level. 
</p>
</dd></dl>

<hr>
<a name="External-Ray_002dTracing"></a>
<div class="header">
<p>
Next: <a href="#Superposition-_0028SSM_0029" accesskey="n" rel="next">Superposition (SSM)</a>, Previous: <a href="#Trim" accesskey="p" rel="previous">Trim</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="External-Ray_002dTracing-1"></a>
<h3 class="section">11.87 External Ray-Tracing</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#raster3d" accesskey="1">raster3d</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002draster3d_002dbond_002dthickness" accesskey="2">set-raster3d-bond-thickness</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002draster3d_002datom_002dradius" accesskey="3">set-raster3d-atom-radius</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002draster3d_002ddensity_002dthickness" accesskey="4">set-raster3d-density-thickness</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drenderer_002dshow_002datoms" accesskey="5">set-renderer-show-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002draster3d_002dbone_002dthickness" accesskey="6">set-raster3d-bone-thickness</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002draster3d_002dshadows_002denabled" accesskey="7">set-raster3d-shadows-enabled</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002draster3d_002dwater_002dsphere" accesskey="8">set-raster3d-water-sphere</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002draster3d_002dfont_002dsize" accesskey="9">set-raster3d-font-size</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#raster_002dscreen_002dshot">raster-screen-shot</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="raster3d"></a>
<div class="header">
<p>
Next: <a href="#set_002draster3d_002dbond_002dthickness" accesskey="n" rel="next">set-raster3d-bond-thickness</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="raster3d-1"></a>
<h4 class="subsection">11.87.1 raster3d</h4>
<dl>
<dt><a name="index-raster3d"></a>function: <strong>raster3d</strong> <em>rd3_filename</em></dt>
<dd>
<p>Where <em>rd3_filename</em> is a string
</p>
<p>create a r3d file for the current view 
</p>
</dd></dl>

<hr>
<a name="set_002draster3d_002dbond_002dthickness"></a>
<div class="header">
<p>
Next: <a href="#set_002draster3d_002datom_002dradius" accesskey="n" rel="next">set-raster3d-atom-radius</a>, Previous: <a href="#raster3d" accesskey="p" rel="previous">raster3d</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002draster3d_002dbond_002dthickness-1"></a>
<h4 class="subsection">11.87.2 set-raster3d-bond-thickness</h4>
<dl>
<dt><a name="index-set_002draster3d_002dbond_002dthickness"></a>function: <strong>set-raster3d-bond-thickness</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the bond thickness for the Raster3D representation 
</p>
</dd></dl>

<hr>
<a name="set_002draster3d_002datom_002dradius"></a>
<div class="header">
<p>
Next: <a href="#set_002draster3d_002ddensity_002dthickness" accesskey="n" rel="next">set-raster3d-density-thickness</a>, Previous: <a href="#set_002draster3d_002dbond_002dthickness" accesskey="p" rel="previous">set-raster3d-bond-thickness</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002draster3d_002datom_002dradius-1"></a>
<h4 class="subsection">11.87.3 set-raster3d-atom-radius</h4>
<dl>
<dt><a name="index-set_002draster3d_002datom_002dradius"></a>function: <strong>set-raster3d-atom-radius</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the atom radius for the Raster3D representation 
</p>
</dd></dl>

<hr>
<a name="set_002draster3d_002ddensity_002dthickness"></a>
<div class="header">
<p>
Next: <a href="#set_002drenderer_002dshow_002datoms" accesskey="n" rel="next">set-renderer-show-atoms</a>, Previous: <a href="#set_002draster3d_002datom_002dradius" accesskey="p" rel="previous">set-raster3d-atom-radius</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002draster3d_002ddensity_002dthickness-1"></a>
<h4 class="subsection">11.87.4 set-raster3d-density-thickness</h4>
<dl>
<dt><a name="index-set_002draster3d_002ddensity_002dthickness"></a>function: <strong>set-raster3d-density-thickness</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the density line thickness for the Raster3D representation 
</p>
</dd></dl>

<hr>
<a name="set_002drenderer_002dshow_002datoms"></a>
<div class="header">
<p>
Next: <a href="#set_002draster3d_002dbone_002dthickness" accesskey="n" rel="next">set-raster3d-bone-thickness</a>, Previous: <a href="#set_002draster3d_002ddensity_002dthickness" accesskey="p" rel="previous">set-raster3d-density-thickness</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drenderer_002dshow_002datoms-1"></a>
<h4 class="subsection">11.87.5 set-renderer-show-atoms</h4>
<dl>
<dt><a name="index-set_002drenderer_002dshow_002datoms"></a>function: <strong>set-renderer-show-atoms</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>set the flag to show atoms for the Raster3D representation 
</p>
</dd></dl>

<hr>
<a name="set_002draster3d_002dbone_002dthickness"></a>
<div class="header">
<p>
Next: <a href="#set_002draster3d_002dshadows_002denabled" accesskey="n" rel="next">set-raster3d-shadows-enabled</a>, Previous: <a href="#set_002drenderer_002dshow_002datoms" accesskey="p" rel="previous">set-renderer-show-atoms</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002draster3d_002dbone_002dthickness-1"></a>
<h4 class="subsection">11.87.6 set-raster3d-bone-thickness</h4>
<dl>
<dt><a name="index-set_002draster3d_002dbone_002dthickness"></a>function: <strong>set-raster3d-bone-thickness</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the bone (skeleton) thickness for the Raster3D representation 
</p>
</dd></dl>

<hr>
<a name="set_002draster3d_002dshadows_002denabled"></a>
<div class="header">
<p>
Next: <a href="#set_002draster3d_002dwater_002dsphere" accesskey="n" rel="next">set-raster3d-water-sphere</a>, Previous: <a href="#set_002draster3d_002dbone_002dthickness" accesskey="p" rel="previous">set-raster3d-bone-thickness</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002draster3d_002dshadows_002denabled-1"></a>
<h4 class="subsection">11.87.7 set-raster3d-shadows-enabled</h4>
<dl>
<dt><a name="index-set_002draster3d_002dshadows_002denabled"></a>function: <strong>set-raster3d-shadows-enabled</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>turn off shadows for raster3d output - give argument 0 to turn off 
</p>
</dd></dl>

<hr>
<a name="set_002draster3d_002dwater_002dsphere"></a>
<div class="header">
<p>
Next: <a href="#set_002draster3d_002dfont_002dsize" accesskey="n" rel="next">set-raster3d-font-size</a>, Previous: <a href="#set_002draster3d_002dshadows_002denabled" accesskey="p" rel="previous">set-raster3d-shadows-enabled</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002draster3d_002dwater_002dsphere-1"></a>
<h4 class="subsection">11.87.8 set-raster3d-water-sphere</h4>
<dl>
<dt><a name="index-set_002draster3d_002dwater_002dsphere"></a>function: <strong>set-raster3d-water-sphere</strong> <em>istate</em></dt>
<dd>
<p>Where <em>istate</em> is an integer number
</p>
<p>set the flag to show waters as spheres for the Raster3D representation. 1 show as spheres, 0 the usual stars. 
</p>
</dd></dl>

<hr>
<a name="set_002draster3d_002dfont_002dsize"></a>
<div class="header">
<p>
Next: <a href="#raster_002dscreen_002dshot" accesskey="n" rel="next">raster-screen-shot</a>, Previous: <a href="#set_002draster3d_002dwater_002dsphere" accesskey="p" rel="previous">set-raster3d-water-sphere</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002draster3d_002dfont_002dsize-1"></a>
<h4 class="subsection">11.87.9 set-raster3d-font-size</h4>
<dl>
<dt><a name="index-set_002draster3d_002dfont_002dsize"></a>function: <strong>set-raster3d-font-size</strong> <em>size_in</em></dt>
<dd>
<p>Where <em>size_in</em> is a string
</p>
<p>set the font size (as a string) for raster3d 
</p>
</dd></dl>

<hr>
<a name="raster_002dscreen_002dshot"></a>
<div class="header">
<p>
Previous: <a href="#set_002draster3d_002dfont_002dsize" accesskey="p" rel="previous">set-raster3d-font-size</a>, Up: <a href="#External-Ray_002dTracing" accesskey="u" rel="up">External Ray-Tracing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="raster_002dscreen_002dshot-1"></a>
<h4 class="subsection">11.87.10 raster-screen-shot</h4>
<dl>
<dt><a name="index-raster_002dscreen_002dshot"></a>function: <strong>raster-screen-shot</strong></dt>
<dd>
<p>run raster3d and display the resulting image. 
</p>
</dd></dl>

<hr>
<a name="Superposition-_0028SSM_0029"></a>
<div class="header">
<p>
Next: <a href="#NCS" accesskey="n" rel="next">NCS</a>, Previous: <a href="#External-Ray_002dTracing" accesskey="p" rel="previous">External Ray-Tracing</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Superposition-_0028SSM_0029-1"></a>
<h3 class="section">11.88 Superposition (SSM)</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#superpose" accesskey="1">superpose</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#superpose_002dwith_002dchain_002dselection" accesskey="2">superpose-with-chain-selection</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#superpose_002dwith_002datom_002dselection" accesskey="3">superpose-with-atom-selection</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="superpose"></a>
<div class="header">
<p>
Next: <a href="#superpose_002dwith_002dchain_002dselection" accesskey="n" rel="next">superpose-with-chain-selection</a>, Up: <a href="#Superposition-_0028SSM_0029" accesskey="u" rel="up">Superposition (SSM)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="superpose-1"></a>
<h4 class="subsection">11.88.1 superpose</h4>
<dl>
<dt><a name="index-superpose"></a>function: <strong>superpose</strong> <em>imol1 imol2 move_imol2_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol1</em> is an integer number
</li><li> <em>imol2</em> is an integer number
</li><li> <em>move_imol2_flag</em> is an integer number
</li></ul>

<p>simple interface to superposition. 
</p>
<p>Superpose all residues of imol2 onto imol1. imol1 is reference, we can either move imol2 or copy it to generate a new molecule depending on the vaule of move_imol2_flag (1 for copy 0 for move). 
</p>
</dd></dl>

<hr>
<a name="superpose_002dwith_002dchain_002dselection"></a>
<div class="header">
<p>
Next: <a href="#superpose_002dwith_002datom_002dselection" accesskey="n" rel="next">superpose-with-atom-selection</a>, Previous: <a href="#superpose" accesskey="p" rel="previous">superpose</a>, Up: <a href="#Superposition-_0028SSM_0029" accesskey="u" rel="up">Superposition (SSM)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="superpose_002dwith_002dchain_002dselection-1"></a>
<h4 class="subsection">11.88.2 superpose-with-chain-selection</h4>
<dl>
<dt><a name="index-superpose_002dwith_002dchain_002dselection"></a>function: <strong>superpose-with-chain-selection</strong> <em>imol1 imol2 chain_imol1 chain_imol2 chain_used_flag_imol1 chain_used_flag_imol2 move_imol2_copy_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol1</em> is an integer number
</li><li> <em>imol2</em> is an integer number
</li><li> <em>chain_imol1</em> is a string
</li><li> <em>chain_imol2</em> is a string
</li><li> <em>chain_used_flag_imol1</em> is an integer number
</li><li> <em>chain_used_flag_imol2</em> is an integer number
</li><li> <em>move_imol2_copy_flag</em> is an integer number
</li></ul>

<p>chain-based interface to superposition. 
</p>
<p>Superpose the given chains of imol2 onto imol1. imol1 is reference, we can either move imol2 or copy it to generate a new molecule depending on the vaule of move_imol2_flag (1 for move 0 for copy). 
</p>
</dd></dl>

<hr>
<a name="superpose_002dwith_002datom_002dselection"></a>
<div class="header">
<p>
Previous: <a href="#superpose_002dwith_002dchain_002dselection" accesskey="p" rel="previous">superpose-with-chain-selection</a>, Up: <a href="#Superposition-_0028SSM_0029" accesskey="u" rel="up">Superposition (SSM)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="superpose_002dwith_002datom_002dselection-1"></a>
<h4 class="subsection">11.88.3 superpose-with-atom-selection</h4>
<dl>
<dt><a name="index-superpose_002dwith_002datom_002dselection"></a>function: <strong>superpose-with-atom-selection</strong> <em>imol1 imol2 mmdb_atom_sel_str_1 mmdb_atom_sel_str_2 move_imol2_copy_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol1</em> is an integer number
</li><li> <em>imol2</em> is an integer number
</li><li> <em>mmdb_atom_sel_str_1</em> is a string
</li><li> <em>mmdb_atom_sel_str_2</em> is a string
</li><li> <em>move_imol2_copy_flag</em> is an integer number
</li></ul>

<p>detailed interface to superposition. 
</p>
<p>Superpose the given atom selection (specified by the mmdb atom selection strings) of imol2 onto imol1. imol1 is reference, we can either move imol2 or copy it to generate a new molecule depending on the vaule of move_imol2_flag (1 for move 0 for copy).
</p>
<p>Returns: the index of the superposed molecule - which could either be a new molecule (if move_imol2_flag was 1) or the imol2 or -1 (signifying failure to do the SMM superposition). 
</p>
</dd></dl>

<hr>
<a name="NCS"></a>
<div class="header">
<p>
Next: <a href="#Helices-and-Strands" accesskey="n" rel="next">Helices and Strands</a>, Previous: <a href="#Superposition-_0028SSM_0029" accesskey="p" rel="previous">Superposition (SSM)</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="NCS-1"></a>
<h3 class="section">11.89 NCS</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002ddraw_002dncs_002dghosts" accesskey="1">set-draw-ncs-ghosts</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#draw_002dncs_002dghosts_002dstate" accesskey="2">draw-ncs-ghosts-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dncs_002dghost_002dbond_002dthickness" accesskey="3">set-ncs-ghost-bond-thickness</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dupdate_002dghosts" accesskey="4">ncs-update-ghosts</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002ddynamically_002dtransformed_002dncs_002dmaps" accesskey="5">make-dynamically-transformed-ncs-maps</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dncs_002dmatrix" accesskey="6">add-ncs-matrix</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dstrict_002dncs_002dmatrix" accesskey="7">add-strict-ncs-matrix</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#show_002dstrict_002dncs_002dstate" accesskey="8">show-strict-ncs-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dstrict_002dncs" accesskey="9">set-show-strict-ncs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dncs_002dhomology_002dlevel">set-ncs-homology-level</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#copy_002dchain">copy-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#copy_002dfrom_002dncs_002dmaster_002dto_002dothers">copy-from-ncs-master-to-others</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers">copy-residue-range-from-ncs-master-to-others</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dmaster_002dchains_002dscm">ncs-master-chains-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm">copy-from-ncs-master-to-chains-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain">ncs-control-change-ncs-master-to-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did">ncs-control-change-ncs-master-to-chain-id</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dcontrol_002ddisplay_002dchain">ncs-control-display-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dchain_002dids_002dscm">ncs-chain-ids-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dchain_002dids_002dpy">ncs-chain-ids-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dghosts_002dscm">ncs-ghosts-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs_002dghosts_002dpy">ncs-ghosts-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002ddraw_002dncs_002dghosts"></a>
<div class="header">
<p>
Next: <a href="#draw_002dncs_002dghosts_002dstate" accesskey="n" rel="next">draw-ncs-ghosts-state</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddraw_002dncs_002dghosts-1"></a>
<h4 class="subsection">11.89.1 set-draw-ncs-ghosts</h4>
<dl>
<dt><a name="index-set_002ddraw_002dncs_002dghosts"></a>function: <strong>set-draw-ncs-ghosts</strong> <em>imol istate</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>istate</em> is an integer number
</li></ul>

<p>set drawing state of NCS ghosts for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="draw_002dncs_002dghosts_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dncs_002dghost_002dbond_002dthickness" accesskey="n" rel="next">set-ncs-ghost-bond-thickness</a>, Previous: <a href="#set_002ddraw_002dncs_002dghosts" accesskey="p" rel="previous">set-draw-ncs-ghosts</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="draw_002dncs_002dghosts_002dstate-1"></a>
<h4 class="subsection">11.89.2 draw-ncs-ghosts-state</h4>
<dl>
<dt><a name="index-draw_002dncs_002dghosts_002dstate"></a>function: <strong>draw-ncs-ghosts-state</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the drawing state of NCS ghosts for molecule number imol. Return -1 on imol is a bad molecule or no ghosts. 
</p>
</dd></dl>

<hr>
<a name="set_002dncs_002dghost_002dbond_002dthickness"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dupdate_002dghosts" accesskey="n" rel="next">ncs-update-ghosts</a>, Previous: <a href="#draw_002dncs_002dghosts_002dstate" accesskey="p" rel="previous">draw-ncs-ghosts-state</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dncs_002dghost_002dbond_002dthickness-1"></a>
<h4 class="subsection">11.89.3 set-ncs-ghost-bond-thickness</h4>
<dl>
<dt><a name="index-set_002dncs_002dghost_002dbond_002dthickness"></a>function: <strong>set-ncs-ghost-bond-thickness</strong> <em>imol f</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>f</em> is a number
</li></ul>

<p>set bond thickness of NCS ghosts for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="ncs_002dupdate_002dghosts"></a>
<div class="header">
<p>
Next: <a href="#make_002ddynamically_002dtransformed_002dncs_002dmaps" accesskey="n" rel="next">make-dynamically-transformed-ncs-maps</a>, Previous: <a href="#set_002dncs_002dghost_002dbond_002dthickness" accesskey="p" rel="previous">set-ncs-ghost-bond-thickness</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dupdate_002dghosts-1"></a>
<h4 class="subsection">11.89.4 ncs-update-ghosts</h4>
<dl>
<dt><a name="index-ncs_002dupdate_002dghosts"></a>function: <strong>ncs-update-ghosts</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>update ghosts for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="make_002ddynamically_002dtransformed_002dncs_002dmaps"></a>
<div class="header">
<p>
Next: <a href="#add_002dncs_002dmatrix" accesskey="n" rel="next">add-ncs-matrix</a>, Previous: <a href="#ncs_002dupdate_002dghosts" accesskey="p" rel="previous">ncs-update-ghosts</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002ddynamically_002dtransformed_002dncs_002dmaps-1"></a>
<h4 class="subsection">11.89.5 make-dynamically-transformed-ncs-maps</h4>
<dl>
<dt><a name="index-make_002ddynamically_002dtransformed_002dncs_002dmaps"></a>function: <strong>make-dynamically-transformed-ncs-maps</strong> <em>imol_model imol_map overwrite_maps_of_same_name_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_model</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li><li> <em>overwrite_maps_of_same_name_flag</em> is an integer number
</li></ul>

<p>make NCS map 
</p>
</dd></dl>

<hr>
<a name="add_002dncs_002dmatrix"></a>
<div class="header">
<p>
Next: <a href="#add_002dstrict_002dncs_002dmatrix" accesskey="n" rel="next">add-strict-ncs-matrix</a>, Previous: <a href="#make_002ddynamically_002dtransformed_002dncs_002dmaps" accesskey="p" rel="previous">make-dynamically-transformed-ncs-maps</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dncs_002dmatrix-1"></a>
<h4 class="subsection">11.89.6 add-ncs-matrix</h4>
<dl>
<dt><a name="index-add_002dncs_002dmatrix"></a>function: <strong>add-ncs-matrix</strong> <em>imol this_chain_id target_chain_id m11 m12 m13 m21 m22 m23 m31 m32 m33 t1 t2 t3</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>this_chain_id</em> is a string
</li><li> <em>target_chain_id</em> is a string
</li><li> <em>m11</em> is a number
</li><li> <em>m12</em> is a number
</li><li> <em>m13</em> is a number
</li><li> <em>m21</em> is a number
</li><li> <em>m22</em> is a number
</li><li> <em>m23</em> is a number
</li><li> <em>m31</em> is a number
</li><li> <em>m32</em> is a number
</li><li> <em>m33</em> is a number
</li><li> <em>t1</em> is a number
</li><li> <em>t2</em> is a number
</li><li> <em>t3</em> is a number
</li></ul>

<p>Add NCS matrix. 
</p>
</dd></dl>

<hr>
<a name="add_002dstrict_002dncs_002dmatrix"></a>
<div class="header">
<p>
Next: <a href="#show_002dstrict_002dncs_002dstate" accesskey="n" rel="next">show-strict-ncs-state</a>, Previous: <a href="#add_002dncs_002dmatrix" accesskey="p" rel="previous">add-ncs-matrix</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dstrict_002dncs_002dmatrix-1"></a>
<h4 class="subsection">11.89.7 add-strict-ncs-matrix</h4>
<dl>
<dt><a name="index-add_002dstrict_002dncs_002dmatrix"></a>function: <strong>add-strict-ncs-matrix</strong> <em>imol this_chain_id target_chain_id m11 m12 m13 m21 m22 m23 m31 m32 m33 t1 t2 t3</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>this_chain_id</em> is a string
</li><li> <em>target_chain_id</em> is a string
</li><li> <em>m11</em> is a number
</li><li> <em>m12</em> is a number
</li><li> <em>m13</em> is a number
</li><li> <em>m21</em> is a number
</li><li> <em>m22</em> is a number
</li><li> <em>m23</em> is a number
</li><li> <em>m31</em> is a number
</li><li> <em>m32</em> is a number
</li><li> <em>m33</em> is a number
</li><li> <em>t1</em> is a number
</li><li> <em>t2</em> is a number
</li><li> <em>t3</em> is a number
</li></ul>

<p>add an NCS matrix for strict NCS molecule representation 
</p>
<p>for CNS strict NCS usage: expand like normal symmetry does 
</p>
</dd></dl>

<hr>
<a name="show_002dstrict_002dncs_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002dshow_002dstrict_002dncs" accesskey="n" rel="next">set-show-strict-ncs</a>, Previous: <a href="#add_002dstrict_002dncs_002dmatrix" accesskey="p" rel="previous">add-strict-ncs-matrix</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dstrict_002dncs_002dstate-1"></a>
<h4 class="subsection">11.89.8 show-strict-ncs-state</h4>
<dl>
<dt><a name="index-show_002dstrict_002dncs_002dstate"></a>function: <strong>show-strict-ncs-state</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the state of NCS ghost molecules for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="set_002dshow_002dstrict_002dncs"></a>
<div class="header">
<p>
Next: <a href="#set_002dncs_002dhomology_002dlevel" accesskey="n" rel="next">set-ncs-homology-level</a>, Previous: <a href="#show_002dstrict_002dncs_002dstate" accesskey="p" rel="previous">show-strict-ncs-state</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dstrict_002dncs-1"></a>
<h4 class="subsection">11.89.9 set-show-strict-ncs</h4>
<dl>
<dt><a name="index-set_002dshow_002dstrict_002dncs"></a>function: <strong>set-show-strict-ncs</strong> <em>imol state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>set display state of NCS ghost molecules for molecule number imol 
</p>
</dd></dl>

<hr>
<a name="set_002dncs_002dhomology_002dlevel"></a>
<div class="header">
<p>
Next: <a href="#copy_002dchain" accesskey="n" rel="next">copy-chain</a>, Previous: <a href="#set_002dshow_002dstrict_002dncs" accesskey="p" rel="previous">set-show-strict-ncs</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dncs_002dhomology_002dlevel-1"></a>
<h4 class="subsection">11.89.10 set-ncs-homology-level</h4>
<dl>
<dt><a name="index-set_002dncs_002dhomology_002dlevel"></a>function: <strong>set-ncs-homology-level</strong> <em>flev</em></dt>
<dd>
<p>Where <em>flev</em> is a number
</p>
<p>At what level of homology should we say that we can&rsquo;t see homology for NCS calculation? (default 0.8) 
</p>
</dd></dl>

<hr>
<a name="copy_002dchain"></a>
<div class="header">
<p>
Next: <a href="#copy_002dfrom_002dncs_002dmaster_002dto_002dothers" accesskey="n" rel="next">copy-from-ncs-master-to-others</a>, Previous: <a href="#set_002dncs_002dhomology_002dlevel" accesskey="p" rel="previous">set-ncs-homology-level</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="copy_002dchain-1"></a>
<h4 class="subsection">11.89.11 copy-chain</h4>
<dl>
<dt><a name="index-copy_002dchain"></a>function: <strong>copy-chain</strong> <em>imol from_chain to_chain</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>from_chain</em> is a string
</li><li> <em>to_chain</em> is a string
</li></ul>

<p>Copy single NCS chain. 
</p>
</dd></dl>

<hr>
<a name="copy_002dfrom_002dncs_002dmaster_002dto_002dothers"></a>
<div class="header">
<p>
Next: <a href="#copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers" accesskey="n" rel="next">copy-residue-range-from-ncs-master-to-others</a>, Previous: <a href="#copy_002dchain" accesskey="p" rel="previous">copy-chain</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="copy_002dfrom_002dncs_002dmaster_002dto_002dothers-1"></a>
<h4 class="subsection">11.89.12 copy-from-ncs-master-to-others</h4>
<dl>
<dt><a name="index-copy_002dfrom_002dncs_002dmaster_002dto_002dothers"></a>function: <strong>copy-from-ncs-master-to-others</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li></ul>

<p>Copy chain from master to all related NCS chains. 
</p>
</dd></dl>

<hr>
<a name="copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dmaster_002dchains_002dscm" accesskey="n" rel="next">ncs-master-chains-scm</a>, Previous: <a href="#copy_002dfrom_002dncs_002dmaster_002dto_002dothers" accesskey="p" rel="previous">copy-from-ncs-master-to-others</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers-1"></a>
<h4 class="subsection">11.89.13 copy-residue-range-from-ncs-master-to-others</h4>
<dl>
<dt><a name="index-copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers"></a>function: <strong>copy-residue-range-from-ncs-master-to-others</strong> <em>imol master_chain_id residue_range_start residue_range_end</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>master_chain_id</em> is a string
</li><li> <em>residue_range_start</em> is an integer number
</li><li> <em>residue_range_end</em> is an integer number
</li></ul>

<p>Copy residue range to all related NCS chains. 
</p>
<p>If the target residues do not exist in the peer chains, then create them. 
</p>
</dd></dl>

<hr>
<a name="ncs_002dmaster_002dchains_002dscm"></a>
<div class="header">
<p>
Next: <a href="#copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm" accesskey="n" rel="next">copy-from-ncs-master-to-chains-scm</a>, Previous: <a href="#copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers" accesskey="p" rel="previous">copy-residue-range-from-ncs-master-to-others</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dmaster_002dchains_002dscm-1"></a>
<h4 class="subsection">11.89.14 ncs-master-chains-scm</h4>
<dl>
<dt><a name="index-ncs_002dmaster_002dchains_002dscm"></a>function: <strong>ncs-master-chains-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Copy residue range to selected NCS chains. 
</p>
<p>If the target residues do not exist in the peer chains, then create them.
</p>
<p>return a list of NCS masters or scheme false 
</p>
</dd></dl>

<hr>
<a name="copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain" accesskey="n" rel="next">ncs-control-change-ncs-master-to-chain</a>, Previous: <a href="#ncs_002dmaster_002dchains_002dscm" accesskey="p" rel="previous">ncs-master-chains-scm</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm-1"></a>
<h4 class="subsection">11.89.15 copy-from-ncs-master-to-chains-scm</h4>
<dl>
<dt><a name="index-copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm"></a>function: <strong>copy-from-ncs-master-to-chains-scm</strong> <em>imol master_chain_id chain_id_list</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>master_chain_id</em> is a string
</li><li> <em>chain_id_list</em> is a SCM
</li></ul>

<p>Copy chain from master to a list of NCS chains. 
</p>
</dd></dl>

<hr>
<a name="ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did" accesskey="n" rel="next">ncs-control-change-ncs-master-to-chain-id</a>, Previous: <a href="#copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm" accesskey="p" rel="previous">copy-from-ncs-master-to-chains-scm</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain-1"></a>
<h4 class="subsection">11.89.16 ncs-control-change-ncs-master-to-chain</h4>
<dl>
<dt><a name="index-ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain"></a>function: <strong>ncs-control-change-ncs-master-to-chain</strong> <em>imol ichain</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>ichain</em> is an integer number
</li></ul>

<p>change the NCS master chain (by number) 
</p>
</dd></dl>

<hr>
<a name="ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dcontrol_002ddisplay_002dchain" accesskey="n" rel="next">ncs-control-display-chain</a>, Previous: <a href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain" accesskey="p" rel="previous">ncs-control-change-ncs-master-to-chain</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did-1"></a>
<h4 class="subsection">11.89.17 ncs-control-change-ncs-master-to-chain-id</h4>
<dl>
<dt><a name="index-ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did"></a>function: <strong>ncs-control-change-ncs-master-to-chain-id</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li></ul>

<p>change the NCS master chain (by chain_id) 
</p>
</dd></dl>

<hr>
<a name="ncs_002dcontrol_002ddisplay_002dchain"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dchain_002dids_002dscm" accesskey="n" rel="next">ncs-chain-ids-scm</a>, Previous: <a href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did" accesskey="p" rel="previous">ncs-control-change-ncs-master-to-chain-id</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dcontrol_002ddisplay_002dchain-1"></a>
<h4 class="subsection">11.89.18 ncs-control-display-chain</h4>
<dl>
<dt><a name="index-ncs_002dcontrol_002ddisplay_002dchain"></a>function: <strong>ncs-control-display-chain</strong> <em>imol ichain state</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>ichain</em> is an integer number
</li><li> <em>state</em> is an integer number
</li></ul>

<p>display the NCS master chain 
</p>
</dd></dl>

<hr>
<a name="ncs_002dchain_002dids_002dscm"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dchain_002dids_002dpy" accesskey="n" rel="next">ncs-chain-ids-py</a>, Previous: <a href="#ncs_002dcontrol_002ddisplay_002dchain" accesskey="p" rel="previous">ncs-control-display-chain</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dchain_002dids_002dscm-1"></a>
<h4 class="subsection">11.89.19 ncs-chain-ids-scm</h4>
<dl>
<dt><a name="index-ncs_002dchain_002dids_002dscm"></a>function: <strong>ncs-chain-ids-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Return the ncs chains id for the given molecule. 
</p>
<p>return something like: &rsquo;((&quot;A&quot; &quot;B&quot;)) or &rsquo;((&quot;A&quot; &quot;C&quot; &quot;E&quot;) (&quot;B&quot; &quot;D&quot; &quot;F&quot;)). The master chain goes in first.
</p>
<p>If imol does not have NCS ghosts, return scheme false. 
</p>
</dd></dl>

<hr>
<a name="ncs_002dchain_002dids_002dpy"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dghosts_002dscm" accesskey="n" rel="next">ncs-ghosts-scm</a>, Previous: <a href="#ncs_002dchain_002dids_002dscm" accesskey="p" rel="previous">ncs-chain-ids-scm</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dchain_002dids_002dpy-1"></a>
<h4 class="subsection">11.89.20 ncs-chain-ids-py</h4>
<dl>
<dt><a name="index-ncs_002dchain_002dids_002dpy"></a>function: <strong>ncs-chain-ids-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Return the ncs chains id for the given molecule. 
</p>
<p>return something like: [[&quot;A&quot;, &quot;B&quot;]] or [[&quot;A&quot;, &quot;C&quot;, &quot;E&quot;], [&quot;B&quot;, &quot;D&quot;, &quot;F&quot;]]. The master chain goes in first.
</p>
<p>If imol does not have NCS ghosts, return python False. 
</p>
</dd></dl>

<hr>
<a name="ncs_002dghosts_002dscm"></a>
<div class="header">
<p>
Next: <a href="#ncs_002dghosts_002dpy" accesskey="n" rel="next">ncs-ghosts-py</a>, Previous: <a href="#ncs_002dchain_002dids_002dpy" accesskey="p" rel="previous">ncs-chain-ids-py</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dghosts_002dscm-1"></a>
<h4 class="subsection">11.89.21 ncs-ghosts-scm</h4>
<dl>
<dt><a name="index-ncs_002dghosts_002dscm"></a>function: <strong>ncs-ghosts-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>get the NCS ghost description 
</p>
<p>Returns: false on bad imol or a list of ghosts on good imol. Can include NCS rtops if they are available, else the rtops are False 
</p>
</dd></dl>

<hr>
<a name="ncs_002dghosts_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#ncs_002dghosts_002dscm" accesskey="p" rel="previous">ncs-ghosts-scm</a>, Up: <a href="#NCS" accesskey="u" rel="up">NCS</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs_002dghosts_002dpy-1"></a>
<h4 class="subsection">11.89.22 ncs-ghosts-py</h4>
<dl>
<dt><a name="index-ncs_002dghosts_002dpy"></a>function: <strong>ncs-ghosts-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Get the NCS ghosts description. 
</p>
<p>Returns: False on bad imol or a list of ghosts on good imol. Can include NCS rtops if they are available, else the rtops are False 
</p>
</dd></dl>

<hr>
<a name="Helices-and-Strands"></a>
<div class="header">
<p>
Next: <a href="#Nucleotides" accesskey="n" rel="next">Nucleotides</a>, Previous: <a href="#NCS" accesskey="p" rel="previous">NCS</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Helices-and-Strands-1"></a>
<h3 class="section">11.90 Helices and Strands</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#place_002dhelix_002dhere" accesskey="1">place-helix-here</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#place_002dstrand_002dhere" accesskey="2">place-strand-here</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#place_002dstrand_002dhere_002ddialog" accesskey="3">place-strand-here-dialog</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#find_002dhelices" accesskey="4">find-helices</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#find_002dstrands" accesskey="5">find-strands</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#find_002dsecondary_002dstructure" accesskey="6">find-secondary-structure</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#find_002dsecondary_002dstructure_002dlocal" accesskey="7">find-secondary-structure-local</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="place_002dhelix_002dhere"></a>
<div class="header">
<p>
Next: <a href="#place_002dstrand_002dhere" accesskey="n" rel="next">place-strand-here</a>, Up: <a href="#Helices-and-Strands" accesskey="u" rel="up">Helices and Strands</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="place_002dhelix_002dhere-1"></a>
<h4 class="subsection">11.90.1 place-helix-here</h4>
<dl>
<dt><a name="index-place_002dhelix_002dhere"></a>function: <strong>place-helix-here</strong></dt>
<dd>
<p>add a helix 
</p>
<p>Add a helix somewhere close to this point in the map, try to fit the orientation. Add to a molecule called &quot;Helix&quot;, create it if needed. Create another moecule called &quot;Reverse Helix&quot; if the helix orientation isn&rsquo;t completely unequivocal.
</p>
<p>Returns: the index of the new molecule. 
</p>
</dd></dl>

<hr>
<a name="place_002dstrand_002dhere"></a>
<div class="header">
<p>
Next: <a href="#place_002dstrand_002dhere_002ddialog" accesskey="n" rel="next">place-strand-here-dialog</a>, Previous: <a href="#place_002dhelix_002dhere" accesskey="p" rel="previous">place-helix-here</a>, Up: <a href="#Helices-and-Strands" accesskey="u" rel="up">Helices and Strands</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="place_002dstrand_002dhere-1"></a>
<h4 class="subsection">11.90.2 place-strand-here</h4>
<dl>
<dt><a name="index-place_002dstrand_002dhere"></a>function: <strong>place-strand-here</strong> <em>n_residues n_sample_strands</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>n_residues</em> is an integer number
</li><li> <em>n_sample_strands</em> is an integer number
</li></ul>

<p>add a strands 
</p>
<p>Add a strand close to this point in the map, try to fit the orientation. Add to a molecule called &quot;Strand&quot;, create it if needed. n_residues is the estimated number of residues in the strand.
</p>
<p>n_sample_strands is the number of strands from the database tested to fit into this strand density. 8 is a suggested number. 20 for a more rigourous search, but it will be slower.
</p>
<p>Returns: the index of the new molecule. 
</p>
</dd></dl>

<hr>
<a name="place_002dstrand_002dhere_002ddialog"></a>
<div class="header">
<p>
Next: <a href="#find_002dhelices" accesskey="n" rel="next">find-helices</a>, Previous: <a href="#place_002dstrand_002dhere" accesskey="p" rel="previous">place-strand-here</a>, Up: <a href="#Helices-and-Strands" accesskey="u" rel="up">Helices and Strands</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="place_002dstrand_002dhere_002ddialog-1"></a>
<h4 class="subsection">11.90.3 place-strand-here-dialog</h4>
<dl>
<dt><a name="index-place_002dstrand_002dhere_002ddialog"></a>function: <strong>place-strand-here-dialog</strong></dt>
<dd>
<p>show the strand placement gui. 
</p>
<p>Choose the python version in there, if needed. Call scripting function, display it in place, don&rsquo;t return a widget. 
</p>
</dd></dl>

<hr>
<a name="find_002dhelices"></a>
<div class="header">
<p>
Next: <a href="#find_002dstrands" accesskey="n" rel="next">find-strands</a>, Previous: <a href="#place_002dstrand_002dhere_002ddialog" accesskey="p" rel="previous">place-strand-here-dialog</a>, Up: <a href="#Helices-and-Strands" accesskey="u" rel="up">Helices and Strands</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="find_002dhelices-1"></a>
<h4 class="subsection">11.90.4 find-helices</h4>
<dl>
<dt><a name="index-find_002dhelices"></a>function: <strong>find-helices</strong></dt>
<dd>
<p>autobuild helices 
</p>
<p>Find secondary structure in the current map. Add to a molecule called &quot;Helices&quot;, create it if needed.
</p>
<p>Returns: the index of the new molecule. 
</p>
</dd></dl>

<hr>
<a name="find_002dstrands"></a>
<div class="header">
<p>
Next: <a href="#find_002dsecondary_002dstructure" accesskey="n" rel="next">find-secondary-structure</a>, Previous: <a href="#find_002dhelices" accesskey="p" rel="previous">find-helices</a>, Up: <a href="#Helices-and-Strands" accesskey="u" rel="up">Helices and Strands</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="find_002dstrands-1"></a>
<h4 class="subsection">11.90.5 find-strands</h4>
<dl>
<dt><a name="index-find_002dstrands"></a>function: <strong>find-strands</strong></dt>
<dd>
<p>autobuild strands 
</p>
<p>Find secondary structure in the current map. Add to a molecule called &quot;Strands&quot;, create it if needed.
</p>
<p>Returns: the index of the new molecule. 
</p>
</dd></dl>

<hr>
<a name="find_002dsecondary_002dstructure"></a>
<div class="header">
<p>
Next: <a href="#find_002dsecondary_002dstructure_002dlocal" accesskey="n" rel="next">find-secondary-structure-local</a>, Previous: <a href="#find_002dstrands" accesskey="p" rel="previous">find-strands</a>, Up: <a href="#Helices-and-Strands" accesskey="u" rel="up">Helices and Strands</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="find_002dsecondary_002dstructure-1"></a>
<h4 class="subsection">11.90.6 find-secondary-structure</h4>
<dl>
<dt><a name="index-find_002dsecondary_002dstructure"></a>function: <strong>find-secondary-structure</strong> <em>use_helix helix_length helix_target use_strand strand_length strand_target</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>use_helix</em> is an integer number
</li><li> <em>helix_length</em> is an integer number
</li><li> <em>helix_target</em> is an integer number
</li><li> <em>use_strand</em> is an integer number
</li><li> <em>strand_length</em> is an integer number
</li><li> <em>strand_target</em> is an integer number
</li></ul>

<p>autobuild secondary structure 
</p>
<p>Find secondary structure in the current map. Add to a molecule called &quot;SecStruc&quot;, create it if needed.
</p>
<p>Returns: the index of the new molecule. 
</p>
</dd></dl>

<hr>
<a name="find_002dsecondary_002dstructure_002dlocal"></a>
<div class="header">
<p>
Previous: <a href="#find_002dsecondary_002dstructure" accesskey="p" rel="previous">find-secondary-structure</a>, Up: <a href="#Helices-and-Strands" accesskey="u" rel="up">Helices and Strands</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="find_002dsecondary_002dstructure_002dlocal-1"></a>
<h4 class="subsection">11.90.7 find-secondary-structure-local</h4>
<dl>
<dt><a name="index-find_002dsecondary_002dstructure_002dlocal"></a>function: <strong>find-secondary-structure-local</strong> <em>use_helix helix_length helix_target use_strand strand_length strand_target radius</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>use_helix</em> is an integer number
</li><li> <em>helix_length</em> is an integer number
</li><li> <em>helix_target</em> is an integer number
</li><li> <em>use_strand</em> is an integer number
</li><li> <em>strand_length</em> is an integer number
</li><li> <em>strand_target</em> is an integer number
</li><li> <em>radius</em> is a number
</li></ul>

<p>autobuild secondary structure 
</p>
<p>Find secondary structure local to current view in the current map. Add to a molecule called &quot;SecStruc&quot;, create it if needed.
</p>
<p>Returns: the index of the new molecule. 
</p>
</dd></dl>

<hr>
<a name="Nucleotides"></a>
<div class="header">
<p>
Next: <a href="#New-Molecule-by-Section-Interface" accesskey="n" rel="next">New Molecule by Section Interface</a>, Previous: <a href="#Helices-and-Strands" accesskey="p" rel="previous">Helices and Strands</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Nucleotides-1"></a>
<h3 class="section">11.91 Nucleotides</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#find_002dnucleic_002dacids_002dlocal" accesskey="1">find-nucleic-acids-local</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="find_002dnucleic_002dacids_002dlocal"></a>
<div class="header">
<p>
Up: <a href="#Nucleotides" accesskey="u" rel="up">Nucleotides</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="find_002dnucleic_002dacids_002dlocal-1"></a>
<h4 class="subsection">11.91.1 find-nucleic-acids-local</h4>
<dl>
<dt><a name="index-find_002dnucleic_002dacids_002dlocal"></a>function: <strong>find-nucleic-acids-local</strong> <em>radius</em></dt>
<dd>
<p>Where <em>radius</em> is a number
</p>
<p>autobuild nucleic acid chains 
</p>
<p>Find secondary structure local to current view in the current map. Add to a molecule called &quot;NuclAcid&quot;, create it if needed.
</p>
<p>Returns: the index of the new molecule. 
</p>
</dd></dl>

<hr>
<a name="New-Molecule-by-Section-Interface"></a>
<div class="header">
<p>
Next: <a href="#RNA_002fDNA" accesskey="n" rel="next">RNA/DNA</a>, Previous: <a href="#Nucleotides" accesskey="p" rel="previous">Nucleotides</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="New-Molecule-by-Section-Interface-1"></a>
<h3 class="section">11.92 New Molecule by Section Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#new_002dmolecule_002dby_002dresidue_002dtype_002dselection" accesskey="1">new-molecule-by-residue-type-selection</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#new_002dmolecule_002dby_002datom_002dselection" accesskey="2">new-molecule-by-atom-selection</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#new_002dmolecule_002dby_002dsphere_002dselection" accesskey="3">new-molecule-by-sphere-selection</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dpy" accesskey="4">new-molecule-by-residue-specs-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dscm" accesskey="5">new-molecule-by-residue-specs-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="new_002dmolecule_002dby_002dresidue_002dtype_002dselection"></a>
<div class="header">
<p>
Next: <a href="#new_002dmolecule_002dby_002datom_002dselection" accesskey="n" rel="next">new-molecule-by-atom-selection</a>, Up: <a href="#New-Molecule-by-Section-Interface" accesskey="u" rel="up">New Molecule by Section Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="new_002dmolecule_002dby_002dresidue_002dtype_002dselection-1"></a>
<h4 class="subsection">11.92.1 new-molecule-by-residue-type-selection</h4>
<dl>
<dt><a name="index-new_002dmolecule_002dby_002dresidue_002dtype_002dselection"></a>function: <strong>new-molecule-by-residue-type-selection</strong> <em>imol residue_type</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_type</em> is a string
</li></ul>

<p>create a new molecule that consists of only the residue of type residue_type in molecule number imol 
</p>
<p>Returns: the new molecule number, -1 means an error. 
</p>
</dd></dl>

<hr>
<a name="new_002dmolecule_002dby_002datom_002dselection"></a>
<div class="header">
<p>
Next: <a href="#new_002dmolecule_002dby_002dsphere_002dselection" accesskey="n" rel="next">new-molecule-by-sphere-selection</a>, Previous: <a href="#new_002dmolecule_002dby_002dresidue_002dtype_002dselection" accesskey="p" rel="previous">new-molecule-by-residue-type-selection</a>, Up: <a href="#New-Molecule-by-Section-Interface" accesskey="u" rel="up">New Molecule by Section Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="new_002dmolecule_002dby_002datom_002dselection-1"></a>
<h4 class="subsection">11.92.2 new-molecule-by-atom-selection</h4>
<dl>
<dt><a name="index-new_002dmolecule_002dby_002datom_002dselection"></a>function: <strong>new-molecule-by-atom-selection</strong> <em>imol atom_selection</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>atom_selection</em> is a string
</li></ul>

<p>create a new molecule that consists of only the atoms specified by the mmdb atoms selection string in molecule number imol 
</p>
<p>Returns: the new molecule number, -1 means an error. 
</p>
</dd></dl>

<hr>
<a name="new_002dmolecule_002dby_002dsphere_002dselection"></a>
<div class="header">
<p>
Next: <a href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dpy" accesskey="n" rel="next">new-molecule-by-residue-specs-py</a>, Previous: <a href="#new_002dmolecule_002dby_002datom_002dselection" accesskey="p" rel="previous">new-molecule-by-atom-selection</a>, Up: <a href="#New-Molecule-by-Section-Interface" accesskey="u" rel="up">New Molecule by Section Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="new_002dmolecule_002dby_002dsphere_002dselection-1"></a>
<h4 class="subsection">11.92.3 new-molecule-by-sphere-selection</h4>
<dl>
<dt><a name="index-new_002dmolecule_002dby_002dsphere_002dselection"></a>function: <strong>new-molecule-by-sphere-selection</strong> <em>imol x y z r allow_partial_residues</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li><li> <em>r</em> is a number
</li><li> <em>allow_partial_residues</em> is an integer number
</li></ul>

<p>create a new molecule that consists of only the atoms within the given radius (r) of the given position. 
</p>
<p>Returns: the new molecule number, -1 means an error. 
</p>
</dd></dl>

<hr>
<a name="new_002dmolecule_002dby_002dresidue_002dspecs_002dpy"></a>
<div class="header">
<p>
Next: <a href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dscm" accesskey="n" rel="next">new-molecule-by-residue-specs-scm</a>, Previous: <a href="#new_002dmolecule_002dby_002dsphere_002dselection" accesskey="p" rel="previous">new-molecule-by-sphere-selection</a>, Up: <a href="#New-Molecule-by-Section-Interface" accesskey="u" rel="up">New Molecule by Section Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="new_002dmolecule_002dby_002dresidue_002dspecs_002dpy-1"></a>
<h4 class="subsection">11.92.4 new-molecule-by-residue-specs-py</h4>
<dl>
<dt><a name="index-new_002dmolecule_002dby_002dresidue_002dspecs_002dpy"></a>function: <strong>new-molecule-by-residue-specs-py</strong> <em>imol residue_spec_list_py</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec_list_py</em> is a PyObject
</li></ul>

<p>create a new molecule that consists of only the atoms of the specified list of residues 
</p>
<p>Returns: the new molecule number, -1 means an error. 
</p>
</dd></dl>

<hr>
<a name="new_002dmolecule_002dby_002dresidue_002dspecs_002dscm"></a>
<div class="header">
<p>
Previous: <a href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dpy" accesskey="p" rel="previous">new-molecule-by-residue-specs-py</a>, Up: <a href="#New-Molecule-by-Section-Interface" accesskey="u" rel="up">New Molecule by Section Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="new_002dmolecule_002dby_002dresidue_002dspecs_002dscm-1"></a>
<h4 class="subsection">11.92.5 new-molecule-by-residue-specs-scm</h4>
<dl>
<dt><a name="index-new_002dmolecule_002dby_002dresidue_002dspecs_002dscm"></a>function: <strong>new-molecule-by-residue-specs-scm</strong> <em>imol residue_spec_list_scm</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec_list_scm</em> is a SCM
</li></ul>

<p>create a new molecule that consists of only the atoms of the specified list of residues 
</p>
<p>Returns: the new molecule number, -1 means an error. 
</p>
</dd></dl>

<hr>
<a name="RNA_002fDNA"></a>
<div class="header">
<p>
Next: <a href="#Sequence-_0028Assignment_0029" accesskey="n" rel="next">Sequence (Assignment)</a>, Previous: <a href="#New-Molecule-by-Section-Interface" accesskey="p" rel="previous">New Molecule by Section Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="RNA_002fDNA-1"></a>
<h3 class="section">11.93 RNA/DNA</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#ideal_002dnucleic_002dacid" accesskey="1">ideal-nucleic-acid</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#pucker_002dinfo_002dscm" accesskey="2">pucker-info-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#pucker_002dinfo_002dpy" accesskey="3">pucker-info-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#watson_002dcrick_002dpair" accesskey="4">watson-crick-pair</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#watson_002dcrick_002dpair_002dfor_002dresidue_002drange" accesskey="5">watson-crick-pair-for-residue-range</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="ideal_002dnucleic_002dacid"></a>
<div class="header">
<p>
Next: <a href="#pucker_002dinfo_002dscm" accesskey="n" rel="next">pucker-info-scm</a>, Up: <a href="#RNA_002fDNA" accesskey="u" rel="up">RNA/DNA</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ideal_002dnucleic_002dacid-1"></a>
<h4 class="subsection">11.93.1 ideal-nucleic-acid</h4>
<dl>
<dt><a name="index-ideal_002dnucleic_002dacid"></a>function: <strong>ideal-nucleic-acid</strong> <em>RNA_or_DNA form single_stranged_flag sequence</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>RNA_or_DNA</em> is a string
</li><li> <em>form</em> is a string
</li><li> <em>single_stranged_flag</em> is an integer number
</li><li> <em>sequence</em> is a string
</li></ul>

<p>create a molecule of idea nucleotides 
</p>
<p>use the given sequence (single letter code)
</p>
<p>RNA_or_DNA is either &quot;RNA&quot; or &quot;DNA&quot;
</p>
<p>form is either &quot;A&quot; or &quot;B&quot;
</p>
<p>Returns: the new molecule number or -1 if a problem 
</p>
</dd></dl>

<hr>
<a name="pucker_002dinfo_002dscm"></a>
<div class="header">
<p>
Next: <a href="#pucker_002dinfo_002dpy" accesskey="n" rel="next">pucker-info-py</a>, Previous: <a href="#ideal_002dnucleic_002dacid" accesskey="p" rel="previous">ideal-nucleic-acid</a>, Up: <a href="#RNA_002fDNA" accesskey="u" rel="up">RNA/DNA</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="pucker_002dinfo_002dscm-1"></a>
<h4 class="subsection">11.93.2 pucker-info-scm</h4>
<dl>
<dt><a name="index-pucker_002dinfo_002dscm"></a>function: <strong>pucker-info-scm</strong> <em>imol residue_spec do_pukka_pucker_check</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec</em> is a SCM
</li><li> <em>do_pukka_pucker_check</em> is an integer number
</li></ul>

<p>get the pucker info for the specified residue 
</p>

<p>(where plane-distortion is for the other 4 atoms in the plane (I think)).
</p>
<p>and if there is no following residue, then the phosphate distance cannot be calculated, so the (inner) list is null (not filled). 
</p>
<p>Returns: scheme false if residue not found, otherwise (list (list phosphate-distance puckered-atom out-of-plane-distance plane-distortion) chain-id resno ins-code)
</p>
</dd></dl>

<hr>
<a name="pucker_002dinfo_002dpy"></a>
<div class="header">
<p>
Next: <a href="#watson_002dcrick_002dpair" accesskey="n" rel="next">watson-crick-pair</a>, Previous: <a href="#pucker_002dinfo_002dscm" accesskey="p" rel="previous">pucker-info-scm</a>, Up: <a href="#RNA_002fDNA" accesskey="u" rel="up">RNA/DNA</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="pucker_002dinfo_002dpy-1"></a>
<h4 class="subsection">11.93.3 pucker-info-py</h4>
<dl>
<dt><a name="index-pucker_002dinfo_002dpy"></a>function: <strong>pucker-info-py</strong> <em>imol residue_spec do_pukka_pucker_check</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec</em> is a PyObject
</li><li> <em>do_pukka_pucker_check</em> is an integer number
</li></ul>

<p>return False if residue not found, otherwise [[phosphate_distance, puckered_atom, out_of_plane_distance, plane_distortion], chain_id, resno, ins_code] 
</p>
<p>(where plane_distortion is for the other 4 atoms in the plane (I think)).
</p>
<p>and if there is no following residue, then the phosphate distance cannot be calculated, so the (inner) list is null (not filled). 
</p>
</dd></dl>

<hr>
<a name="watson_002dcrick_002dpair"></a>
<div class="header">
<p>
Next: <a href="#watson_002dcrick_002dpair_002dfor_002dresidue_002drange" accesskey="n" rel="next">watson-crick-pair-for-residue-range</a>, Previous: <a href="#pucker_002dinfo_002dpy" accesskey="p" rel="previous">pucker-info-py</a>, Up: <a href="#RNA_002fDNA" accesskey="u" rel="up">RNA/DNA</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="watson_002dcrick_002dpair-1"></a>
<h4 class="subsection">11.93.4 watson-crick-pair</h4>
<dl>
<dt><a name="index-watson_002dcrick_002dpair"></a>function: <strong>watson-crick-pair</strong> <em>imol chain_id resno</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li></ul>

<p>Return a molecule that contains a residue that is the WC pair partner of the clicked/picked/selected residue. 
</p>
</dd></dl>

<hr>
<a name="watson_002dcrick_002dpair_002dfor_002dresidue_002drange"></a>
<div class="header">
<p>
Previous: <a href="#watson_002dcrick_002dpair" accesskey="p" rel="previous">watson-crick-pair</a>, Up: <a href="#RNA_002fDNA" accesskey="u" rel="up">RNA/DNA</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="watson_002dcrick_002dpair_002dfor_002dresidue_002drange-1"></a>
<h4 class="subsection">11.93.5 watson-crick-pair-for-residue-range</h4>
<dl>
<dt><a name="index-watson_002dcrick_002dpair_002dfor_002dresidue_002drange"></a>function: <strong>watson-crick-pair-for-residue-range</strong> <em>imol chain_id resno_start resno_end</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno_start</em> is an integer number
</li><li> <em>resno_end</em> is an integer number
</li></ul>

<p>add base pairs for the given residue range, modify molecule imol by creating a new chain 
</p>
</dd></dl>

<hr>
<a name="Sequence-_0028Assignment_0029"></a>
<div class="header">
<p>
Next: <a href="#Surface-Interface" accesskey="n" rel="next">Surface Interface</a>, Previous: <a href="#RNA_002fDNA" accesskey="p" rel="previous">RNA/DNA</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Sequence-_0028Assignment_0029-1"></a>
<h3 class="section">11.94 Sequence (Assignment)</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#print_002dsequence_002dchain" accesskey="1">print-sequence-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#print_002dsequence_002dchain_002dgeneral" accesskey="2">print-sequence-chain-general</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#assign_002dfasta_002dsequence" accesskey="3">assign-fasta-sequence</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#assign_002dpir_002dsequence" accesskey="4">assign-pir-sequence</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#assign_002dsequence_002dfrom_002dfile" accesskey="5">assign-sequence-from-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#assign_002dsequence_002dfrom_002dstring" accesskey="6">assign-sequence-from-string</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dall_002dsequences_002dfrom_002dmolecule" accesskey="7">delete-all-sequences-from-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#delete_002dsequence_002dby_002dchain_002did" accesskey="8">delete-sequence-by-chain-id</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sequence_002dinfo" accesskey="9">sequence-info</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#alignment_002dmismatches_002dscm">alignment-mismatches-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sequence_002dinfo_002dpy">sequence-info-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#alignment_002dmismatches_002dpy">alignment-mismatches-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="print_002dsequence_002dchain"></a>
<div class="header">
<p>
Next: <a href="#print_002dsequence_002dchain_002dgeneral" accesskey="n" rel="next">print-sequence-chain-general</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="print_002dsequence_002dchain-1"></a>
<h4 class="subsection">11.94.1 print-sequence-chain</h4>
<dl>
<dt><a name="index-print_002dsequence_002dchain"></a>function: <strong>print-sequence-chain</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li></ul>

<p>Print the sequence to the console of the given molecule. 
</p>
</dd></dl>

<hr>
<a name="print_002dsequence_002dchain_002dgeneral"></a>
<div class="header">
<p>
Next: <a href="#assign_002dfasta_002dsequence" accesskey="n" rel="next">assign-fasta-sequence</a>, Previous: <a href="#print_002dsequence_002dchain" accesskey="p" rel="previous">print-sequence-chain</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="print_002dsequence_002dchain_002dgeneral-1"></a>
<h4 class="subsection">11.94.2 print-sequence-chain-general</h4>
<dl>
<dt><a name="index-print_002dsequence_002dchain_002dgeneral"></a>function: <strong>print-sequence-chain-general</strong> <em>imol chain_id pir_format file_output file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>pir_format</em> is an integer number
</li><li> <em>file_output</em> is an integer number
</li><li> <em>file_name</em> is a string
</li></ul>

<p>optionally write the sequence to the file for the given molecule, optionally in PIR format 
</p>
</dd></dl>

<hr>
<a name="assign_002dfasta_002dsequence"></a>
<div class="header">
<p>
Next: <a href="#assign_002dpir_002dsequence" accesskey="n" rel="next">assign-pir-sequence</a>, Previous: <a href="#print_002dsequence_002dchain_002dgeneral" accesskey="p" rel="previous">print-sequence-chain-general</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="assign_002dfasta_002dsequence-1"></a>
<h4 class="subsection">11.94.3 assign-fasta-sequence</h4>
<dl>
<dt><a name="index-assign_002dfasta_002dsequence"></a>function: <strong>assign-fasta-sequence</strong> <em>imol chain_id_in seq</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id_in</em> is a string
</li><li> <em>seq</em> is a string
</li></ul>

<p>Assign a FASTA sequence to a given chain in the molecule. 
</p>
</dd></dl>

<hr>
<a name="assign_002dpir_002dsequence"></a>
<div class="header">
<p>
Next: <a href="#assign_002dsequence_002dfrom_002dfile" accesskey="n" rel="next">assign-sequence-from-file</a>, Previous: <a href="#assign_002dfasta_002dsequence" accesskey="p" rel="previous">assign-fasta-sequence</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="assign_002dpir_002dsequence-1"></a>
<h4 class="subsection">11.94.4 assign-pir-sequence</h4>
<dl>
<dt><a name="index-assign_002dpir_002dsequence"></a>function: <strong>assign-pir-sequence</strong> <em>imol chain_id_in seq</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id_in</em> is a string
</li><li> <em>seq</em> is a string
</li></ul>

<p>Assign a PIR sequence to a given chain in the molecule. If the chain of the molecule already had a chain assigned to it, then this will overwrite that old assignment with the new one. 
</p>
</dd></dl>

<hr>
<a name="assign_002dsequence_002dfrom_002dfile"></a>
<div class="header">
<p>
Next: <a href="#assign_002dsequence_002dfrom_002dstring" accesskey="n" rel="next">assign-sequence-from-string</a>, Previous: <a href="#assign_002dpir_002dsequence" accesskey="p" rel="previous">assign-pir-sequence</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="assign_002dsequence_002dfrom_002dfile-1"></a>
<h4 class="subsection">11.94.5 assign-sequence-from-file</h4>
<dl>
<dt><a name="index-assign_002dsequence_002dfrom_002dfile"></a>function: <strong>assign-sequence-from-file</strong> <em>imol file</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>file</em> is a string
</li></ul>

<p>Assign a sequence to a given molecule from (whatever) sequence file. 
</p>
</dd></dl>

<hr>
<a name="assign_002dsequence_002dfrom_002dstring"></a>
<div class="header">
<p>
Next: <a href="#delete_002dall_002dsequences_002dfrom_002dmolecule" accesskey="n" rel="next">delete-all-sequences-from-molecule</a>, Previous: <a href="#assign_002dsequence_002dfrom_002dfile" accesskey="p" rel="previous">assign-sequence-from-file</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="assign_002dsequence_002dfrom_002dstring-1"></a>
<h4 class="subsection">11.94.6 assign-sequence-from-string</h4>
<dl>
<dt><a name="index-assign_002dsequence_002dfrom_002dstring"></a>function: <strong>assign-sequence-from-string</strong> <em>imol chain_id_in seq</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id_in</em> is a string
</li><li> <em>seq</em> is a string
</li></ul>

<p>Assign a sequence to a given molecule from a simple string. 
</p>
</dd></dl>

<hr>
<a name="delete_002dall_002dsequences_002dfrom_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#delete_002dsequence_002dby_002dchain_002did" accesskey="n" rel="next">delete-sequence-by-chain-id</a>, Previous: <a href="#assign_002dsequence_002dfrom_002dstring" accesskey="p" rel="previous">assign-sequence-from-string</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dall_002dsequences_002dfrom_002dmolecule-1"></a>
<h4 class="subsection">11.94.7 delete-all-sequences-from-molecule</h4>
<dl>
<dt><a name="index-delete_002dall_002dsequences_002dfrom_002dmolecule"></a>function: <strong>delete-all-sequences-from-molecule</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Delete all the sequences from a given molecule. 
</p>
</dd></dl>

<hr>
<a name="delete_002dsequence_002dby_002dchain_002did"></a>
<div class="header">
<p>
Next: <a href="#sequence_002dinfo" accesskey="n" rel="next">sequence-info</a>, Previous: <a href="#delete_002dall_002dsequences_002dfrom_002dmolecule" accesskey="p" rel="previous">delete-all-sequences-from-molecule</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="delete_002dsequence_002dby_002dchain_002did-1"></a>
<h4 class="subsection">11.94.8 delete-sequence-by-chain-id</h4>
<dl>
<dt><a name="index-delete_002dsequence_002dby_002dchain_002did"></a>function: <strong>delete-sequence-by-chain-id</strong> <em>imol chain_id_in</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id_in</em> is a string
</li></ul>

<p>Delete the sequence for a given chain_id from a given molecule. 
</p>
</dd></dl>

<hr>
<a name="sequence_002dinfo"></a>
<div class="header">
<p>
Next: <a href="#alignment_002dmismatches_002dscm" accesskey="n" rel="next">alignment-mismatches-scm</a>, Previous: <a href="#delete_002dsequence_002dby_002dchain_002did" accesskey="p" rel="previous">delete-sequence-by-chain-id</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="sequence_002dinfo-1"></a>
<h4 class="subsection">11.94.9 sequence-info</h4>
<dl>
<dt><a name="index-sequence_002dinfo"></a>function: <strong>sequence-info</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the sequence info that has been assigned to molecule number imol. return as a list of dotted pairs (list (cons chain-id seq)). To be used in constructing the cootaneer gui. Return Scheme False when no sequence has been assigned to imol. 
</p>
</dd></dl>

<hr>
<a name="alignment_002dmismatches_002dscm"></a>
<div class="header">
<p>
Next: <a href="#sequence_002dinfo_002dpy" accesskey="n" rel="next">sequence-info-py</a>, Previous: <a href="#sequence_002dinfo" accesskey="p" rel="previous">sequence-info</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="alignment_002dmismatches_002dscm-1"></a>
<h4 class="subsection">11.94.10 alignment-mismatches-scm</h4>
<dl>
<dt><a name="index-alignment_002dmismatches_002dscm"></a>function: <strong>alignment-mismatches-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>do a internal alignment of all the assigned sequences, return a list of mismatches that need to be made to model number imol to match the input sequence. 
</p>
<p>Return a list of mutations deletions insetions. Return scheme false on failure to align (e.g. not assigned sequence) and the empty list on no alignment mismatches. 
</p>
</dd></dl>

<hr>
<a name="sequence_002dinfo_002dpy"></a>
<div class="header">
<p>
Next: <a href="#alignment_002dmismatches_002dpy" accesskey="n" rel="next">alignment-mismatches-py</a>, Previous: <a href="#alignment_002dmismatches_002dscm" accesskey="p" rel="previous">alignment-mismatches-scm</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="sequence_002dinfo_002dpy-1"></a>
<h4 class="subsection">11.94.11 sequence-info-py</h4>
<dl>
<dt><a name="index-sequence_002dinfo_002dpy"></a>function: <strong>sequence-info-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the sequence info that has been assigned to molecule number imol. return as a list of dotted pairs [[chain-id, seq]]. To be used in constructing the cootaneer gui. Return False when no sequence has been assigned. 
</p>
</dd></dl>

<hr>
<a name="alignment_002dmismatches_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#sequence_002dinfo_002dpy" accesskey="p" rel="previous">sequence-info-py</a>, Up: <a href="#Sequence-_0028Assignment_0029" accesskey="u" rel="up">Sequence (Assignment)</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="alignment_002dmismatches_002dpy-1"></a>
<h4 class="subsection">11.94.12 alignment-mismatches-py</h4>
<dl>
<dt><a name="index-alignment_002dmismatches_002dpy"></a>function: <strong>alignment-mismatches-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>do a internal alignment of all the assigned sequences, return a list of mismatches that need to be made to model number imol to match the input sequence. 
</p>
<p>Return a list of mutations deletions insetions. Return False on failure to align (e.g. not assigned sequence) and the empty list on no alignment mismatches. 
</p>
</dd></dl>

<hr>
<a name="Surface-Interface"></a>
<div class="header">
<p>
Next: <a href="#FFFearing" accesskey="n" rel="next">FFFearing</a>, Previous: <a href="#Sequence-_0028Assignment_0029" accesskey="p" rel="previous">Sequence (Assignment)</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Surface-Interface-1"></a>
<h3 class="section">11.95 Surface Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#do_002dsurface" accesskey="1">do-surface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#do_002dclipped_002dsurface_002dscm" accesskey="2">do-clipped-surface-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#do_002dclipped_002dsurface_002dpy" accesskey="3">do-clipped-surface-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dtransparent_002delectrostatic_002dsurface" accesskey="4">set-transparent-electrostatic-surface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002delectrostatic_002dsurface_002dopacity" accesskey="5">get-electrostatic-surface-opacity</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="do_002dsurface"></a>
<div class="header">
<p>
Next: <a href="#do_002dclipped_002dsurface_002dscm" accesskey="n" rel="next">do-clipped-surface-scm</a>, Up: <a href="#Surface-Interface" accesskey="u" rel="up">Surface Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dsurface-1"></a>
<h4 class="subsection">11.95.1 do-surface</h4>
<dl>
<dt><a name="index-do_002dsurface"></a>function: <strong>do-surface</strong> <em>imol istate</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>istate</em> is an integer number
</li></ul>

<p>draw surface of molecule number imol 
</p>
<p>if state = 1 draw the surface (normal representation goes away)
</p>
<p>if state = 0 don&rsquo;t draw surface 
</p>
</dd></dl>

<hr>
<a name="do_002dclipped_002dsurface_002dscm"></a>
<div class="header">
<p>
Next: <a href="#do_002dclipped_002dsurface_002dpy" accesskey="n" rel="next">do-clipped-surface-py</a>, Previous: <a href="#do_002dsurface" accesskey="p" rel="previous">do-surface</a>, Up: <a href="#Surface-Interface" accesskey="u" rel="up">Surface Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dclipped_002dsurface_002dscm-1"></a>
<h4 class="subsection">11.95.2 do-clipped-surface-scm</h4>
<dl>
<dt><a name="index-do_002dclipped_002dsurface_002dscm"></a>function: <strong>do-clipped-surface-scm</strong> <em>imol residue_specs</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a SCM
</li></ul>

<p>draw the surface of the imolth molecule clipped to the residues given by residue_specs. 
</p>
<p>residue_specs must not contain spec for waters (you wouldn&rsquo;t want to surface over waters anyway). 
</p>
</dd></dl>

<hr>
<a name="do_002dclipped_002dsurface_002dpy"></a>
<div class="header">
<p>
Next: <a href="#set_002dtransparent_002delectrostatic_002dsurface" accesskey="n" rel="next">set-transparent-electrostatic-surface</a>, Previous: <a href="#do_002dclipped_002dsurface_002dscm" accesskey="p" rel="previous">do-clipped-surface-scm</a>, Up: <a href="#Surface-Interface" accesskey="u" rel="up">Surface Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dclipped_002dsurface_002dpy-1"></a>
<h4 class="subsection">11.95.3 do-clipped-surface-py</h4>
<dl>
<dt><a name="index-do_002dclipped_002dsurface_002dpy"></a>function: <strong>do-clipped-surface-py</strong> <em>imol residue_specs</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a PyObject
</li></ul>

<p>draw the surface of the imolth molecule clipped to the residues given by residue_specs 
</p>
</dd></dl>

<hr>
<a name="set_002dtransparent_002delectrostatic_002dsurface"></a>
<div class="header">
<p>
Next: <a href="#get_002delectrostatic_002dsurface_002dopacity" accesskey="n" rel="next">get-electrostatic-surface-opacity</a>, Previous: <a href="#do_002dclipped_002dsurface_002dpy" accesskey="p" rel="previous">do-clipped-surface-py</a>, Up: <a href="#Surface-Interface" accesskey="u" rel="up">Surface Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dtransparent_002delectrostatic_002dsurface-1"></a>
<h4 class="subsection">11.95.4 set-transparent-electrostatic-surface</h4>
<dl>
<dt><a name="index-set_002dtransparent_002delectrostatic_002dsurface"></a>function: <strong>set-transparent-electrostatic-surface</strong> <em>imol opacity</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>opacity</em> is a number
</li></ul>

<p>simple on/off screendoor transparency at the moment, an opacity &gt; 0.0 will turn on screendoor transparency (stippling). 
</p>
</dd></dl>

<hr>
<a name="get_002delectrostatic_002dsurface_002dopacity"></a>
<div class="header">
<p>
Previous: <a href="#set_002dtransparent_002delectrostatic_002dsurface" accesskey="p" rel="previous">set-transparent-electrostatic-surface</a>, Up: <a href="#Surface-Interface" accesskey="u" rel="up">Surface Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002delectrostatic_002dsurface_002dopacity-1"></a>
<h4 class="subsection">11.95.5 get-electrostatic-surface-opacity</h4>
<dl>
<dt><a name="index-get_002delectrostatic_002dsurface_002dopacity"></a>function: <strong>get-electrostatic-surface-opacity</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return 1.0 for non transparent and 0.5 if screendoor transparency has been turned on. 
</p>
</dd></dl>

<hr>
<a name="FFFearing"></a>
<div class="header">
<p>
Next: <a href="#Remote-Control" accesskey="n" rel="next">Remote Control</a>, Previous: <a href="#Surface-Interface" accesskey="p" rel="previous">Surface Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="FFFearing-1"></a>
<h3 class="section">11.96 FFFearing</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#fffear_002dsearch" accesskey="1">fffear-search</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dfffear_002dangular_002dresolution" accesskey="2">set-fffear-angular-resolution</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fffear_002dangular_002dresolution" accesskey="3">fffear-angular-resolution</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="fffear_002dsearch"></a>
<div class="header">
<p>
Next: <a href="#set_002dfffear_002dangular_002dresolution" accesskey="n" rel="next">set-fffear-angular-resolution</a>, Up: <a href="#FFFearing" accesskey="u" rel="up">FFFearing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fffear_002dsearch-1"></a>
<h4 class="subsection">11.96.1 fffear-search</h4>
<dl>
<dt><a name="index-fffear_002dsearch"></a>function: <strong>fffear-search</strong> <em>imol_model imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_model</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>fffear search model in molecule number imol_model in map number imol_map 
</p>
</dd></dl>

<hr>
<a name="set_002dfffear_002dangular_002dresolution"></a>
<div class="header">
<p>
Next: <a href="#fffear_002dangular_002dresolution" accesskey="n" rel="next">fffear-angular-resolution</a>, Previous: <a href="#fffear_002dsearch" accesskey="p" rel="previous">fffear-search</a>, Up: <a href="#FFFearing" accesskey="u" rel="up">FFFearing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dfffear_002dangular_002dresolution-1"></a>
<h4 class="subsection">11.96.2 set-fffear-angular-resolution</h4>
<dl>
<dt><a name="index-set_002dfffear_002dangular_002dresolution"></a>function: <strong>set-fffear-angular-resolution</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set and return the fffear angular resolution in degrees 
</p>
</dd></dl>

<hr>
<a name="fffear_002dangular_002dresolution"></a>
<div class="header">
<p>
Previous: <a href="#set_002dfffear_002dangular_002dresolution" accesskey="p" rel="previous">set-fffear-angular-resolution</a>, Up: <a href="#FFFearing" accesskey="u" rel="up">FFFearing</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fffear_002dangular_002dresolution-1"></a>
<h4 class="subsection">11.96.3 fffear-angular-resolution</h4>
<dl>
<dt><a name="index-fffear_002dangular_002dresolution"></a>function: <strong>fffear-angular-resolution</strong></dt>
<dd>
<p>return the fffear angular resolution in degrees 
</p>
</dd></dl>

<hr>
<a name="Remote-Control"></a>
<div class="header">
<p>
Next: <a href="#Display-Lists-for-Maps" accesskey="n" rel="next">Display Lists for Maps</a>, Previous: <a href="#FFFearing" accesskey="p" rel="previous">FFFearing</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Remote-Control-1"></a>
<h3 class="section">11.97 Remote Control</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#make_002dsocket_002dlistener_002dmaybe" accesskey="1">make-socket-listener-maybe</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsocket_002dstring_002dwaiting" accesskey="2">set-socket-string-waiting</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dsocket_002dpython_002dstring_002dwaiting" accesskey="3">set-socket-python-string-waiting</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="make_002dsocket_002dlistener_002dmaybe"></a>
<div class="header">
<p>
Next: <a href="#set_002dsocket_002dstring_002dwaiting" accesskey="n" rel="next">set-socket-string-waiting</a>, Up: <a href="#Remote-Control" accesskey="u" rel="up">Remote Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dsocket_002dlistener_002dmaybe-1"></a>
<h4 class="subsection">11.97.1 make-socket-listener-maybe</h4>
<dl>
<dt><a name="index-make_002dsocket_002dlistener_002dmaybe"></a>function: <strong>make-socket-listener-maybe</strong></dt>
<dd>
<p>try to make socket listener 
</p>
</dd></dl>

<hr>
<a name="set_002dsocket_002dstring_002dwaiting"></a>
<div class="header">
<p>
Next: <a href="#set_002dsocket_002dpython_002dstring_002dwaiting" accesskey="n" rel="next">set-socket-python-string-waiting</a>, Previous: <a href="#make_002dsocket_002dlistener_002dmaybe" accesskey="p" rel="previous">make-socket-listener-maybe</a>, Up: <a href="#Remote-Control" accesskey="u" rel="up">Remote Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsocket_002dstring_002dwaiting-1"></a>
<h4 class="subsection">11.97.2 set-socket-string-waiting</h4>
<dl>
<dt><a name="index-set_002dsocket_002dstring_002dwaiting"></a>function: <strong>set-socket-string-waiting</strong> <em>s</em></dt>
<dd>
<p>Where <em>s</em> is a string
</p>
<p>feed the main thread a scheme script to evaluate 
</p>
</dd></dl>

<hr>
<a name="set_002dsocket_002dpython_002dstring_002dwaiting"></a>
<div class="header">
<p>
Previous: <a href="#set_002dsocket_002dstring_002dwaiting" accesskey="p" rel="previous">set-socket-string-waiting</a>, Up: <a href="#Remote-Control" accesskey="u" rel="up">Remote Control</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dsocket_002dpython_002dstring_002dwaiting-1"></a>
<h4 class="subsection">11.97.3 set-socket-python-string-waiting</h4>
<dl>
<dt><a name="index-set_002dsocket_002dpython_002dstring_002dwaiting"></a>function: <strong>set-socket-python-string-waiting</strong> <em>s</em></dt>
<dd>
<p>Where <em>s</em> is a string
</p>
<p>feed the main thread a python script to evaluate 
</p>
</dd></dl>

<hr>
<a name="Display-Lists-for-Maps"></a>
<div class="header">
<p>
Next: <a href="#Browser-Interface" accesskey="n" rel="next">Browser Interface</a>, Previous: <a href="#Remote-Control" accesskey="p" rel="previous">Remote Control</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Display-Lists-for-Maps-1"></a>
<h3 class="section">11.98 Display Lists for Maps</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002ddisplay_002dlists_002dfor_002dmaps" accesskey="1">set-display-lists-for-maps</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#display_002dlists_002dfor_002dmaps_002dstate" accesskey="2">display-lists-for-maps-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002ddisplay_002dlists_002dfor_002dmaps"></a>
<div class="header">
<p>
Next: <a href="#display_002dlists_002dfor_002dmaps_002dstate" accesskey="n" rel="next">display-lists-for-maps-state</a>, Up: <a href="#Display-Lists-for-Maps" accesskey="u" rel="up">Display Lists for Maps</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddisplay_002dlists_002dfor_002dmaps-1"></a>
<h4 class="subsection">11.98.1 set-display-lists-for-maps</h4>
<dl>
<dt><a name="index-set_002ddisplay_002dlists_002dfor_002dmaps"></a>function: <strong>set-display-lists-for-maps</strong> <em>i</em></dt>
<dd>
<p>Where <em>i</em> is an integer number
</p>
<p>Should display lists be used for maps? It may speed things up if these are turned on (or off) - depends on graphics card and drivers. Pass 1 for on, 0 for off. 
</p>
</dd></dl>

<hr>
<a name="display_002dlists_002dfor_002dmaps_002dstate"></a>
<div class="header">
<p>
Previous: <a href="#set_002ddisplay_002dlists_002dfor_002dmaps" accesskey="p" rel="previous">set-display-lists-for-maps</a>, Up: <a href="#Display-Lists-for-Maps" accesskey="u" rel="up">Display Lists for Maps</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="display_002dlists_002dfor_002dmaps_002dstate-1"></a>
<h4 class="subsection">11.98.2 display-lists-for-maps-state</h4>
<dl>
<dt><a name="index-display_002dlists_002dfor_002dmaps_002dstate"></a>function: <strong>display-lists-for-maps-state</strong></dt>
<dd>
<p>return the state of display_lists_for_maps. 
</p>
</dd></dl>

<hr>
<a name="Browser-Interface"></a>
<div class="header">
<p>
Next: <a href="#Molprobity-Interface" accesskey="n" rel="next">Molprobity Interface</a>, Previous: <a href="#Display-Lists-for-Maps" accesskey="p" rel="previous">Display Lists for Maps</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Browser-Interface-1"></a>
<h3 class="section">11.99 Browser Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#browser_002durl" accesskey="1">browser-url</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dbrowser_002dinterface" accesskey="2">set-browser-interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#handle_002donline_002dcoot_002dsearch_002drequest" accesskey="3">handle-online-coot-search-request</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="browser_002durl"></a>
<div class="header">
<p>
Next: <a href="#set_002dbrowser_002dinterface" accesskey="n" rel="next">set-browser-interface</a>, Up: <a href="#Browser-Interface" accesskey="u" rel="up">Browser Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="browser_002durl-1"></a>
<h4 class="subsection">11.99.1 browser-url</h4>
<dl>
<dt><a name="index-browser_002durl"></a>function: <strong>browser-url</strong> <em>url</em></dt>
<dd>
<p>Where <em>url</em> is a string
</p>
<p>try to open given url in Web browser 
</p>
</dd></dl>

<hr>
<a name="set_002dbrowser_002dinterface"></a>
<div class="header">
<p>
Next: <a href="#handle_002donline_002dcoot_002dsearch_002drequest" accesskey="n" rel="next">handle-online-coot-search-request</a>, Previous: <a href="#browser_002durl" accesskey="p" rel="previous">browser-url</a>, Up: <a href="#Browser-Interface" accesskey="u" rel="up">Browser Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbrowser_002dinterface-1"></a>
<h4 class="subsection">11.99.2 set-browser-interface</h4>
<dl>
<dt><a name="index-set_002dbrowser_002dinterface"></a>function: <strong>set-browser-interface</strong> <em>browser</em></dt>
<dd>
<p>Where <em>browser</em> is a string
</p>
<p>set command to open the web browser, 
</p>
<p>examples are &quot;open&quot; or &quot;mozilla&quot; 
</p>
</dd></dl>

<hr>
<a name="handle_002donline_002dcoot_002dsearch_002drequest"></a>
<div class="header">
<p>
Previous: <a href="#set_002dbrowser_002dinterface" accesskey="p" rel="previous">set-browser-interface</a>, Up: <a href="#Browser-Interface" accesskey="u" rel="up">Browser Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002donline_002dcoot_002dsearch_002drequest-1"></a>
<h4 class="subsection">11.99.3 handle-online-coot-search-request</h4>
<dl>
<dt><a name="index-handle_002donline_002dcoot_002dsearch_002drequest"></a>function: <strong>handle-online-coot-search-request</strong> <em>entry_text</em></dt>
<dd>
<p>Where <em>entry_text</em> is a string
</p>
<p>the search interface 
</p>
<p>find words, construct a url and open it. 
</p>
</dd></dl>

<hr>
<a name="Molprobity-Interface"></a>
<div class="header">
<p>
Next: <a href="#Map-Sharpening-Interface" accesskey="n" rel="next">Map Sharpening Interface</a>, Previous: <a href="#Browser-Interface" accesskey="p" rel="previous">Browser Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Molprobity-Interface-1"></a>
<h3 class="section">11.100 Molprobity Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#handle_002dread_002ddraw_002dprobe_002ddots" accesskey="1">handle-read-draw-probe-dots</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted" accesskey="2">handle-read-draw-probe-dots-unformatted</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis" accesskey="3">set-do-probe-dots-on-rotamers-and-chis</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate" accesskey="4">do-probe-dots-on-rotamers-and-chis-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002ddo_002dprobe_002ddots_002dpost_002drefine" accesskey="5">set-do-probe-dots-post-refine</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#do_002dprobe_002ddots_002dpost_002drefine_002dstate" accesskey="6">do-probe-dots-post-refine-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#unmangle_002dhydrogen_002dname" accesskey="7">unmangle-hydrogen-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius" accesskey="8">set-interactive-probe-dots-molprobity-radius</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#interactive_002dprobe_002ddots_002dmolprobity_002dradius" accesskey="9">interactive-probe-dots-molprobity-radius</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#user_002dmods_002dscm">user-mods-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#user_002dmods_002dpy">user-mods-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="handle_002dread_002ddraw_002dprobe_002ddots"></a>
<div class="header">
<p>
Next: <a href="#handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted" accesskey="n" rel="next">handle-read-draw-probe-dots-unformatted</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002dread_002ddraw_002dprobe_002ddots-1"></a>
<h4 class="subsection">11.100.1 handle-read-draw-probe-dots</h4>
<dl>
<dt><a name="index-handle_002dread_002ddraw_002dprobe_002ddots"></a>function: <strong>handle-read-draw-probe-dots</strong> <em>dots_file</em></dt>
<dd>
<p>Where <em>dots_file</em> is a string
</p>
<p>pass a filename that contains molprobity&rsquo;s probe output in XtalView format 
</p>
</dd></dl>

<hr>
<a name="handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted"></a>
<div class="header">
<p>
Next: <a href="#set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis" accesskey="n" rel="next">set-do-probe-dots-on-rotamers-and-chis</a>, Previous: <a href="#handle_002dread_002ddraw_002dprobe_002ddots" accesskey="p" rel="previous">handle-read-draw-probe-dots</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted-1"></a>
<h4 class="subsection">11.100.2 handle-read-draw-probe-dots-unformatted</h4>
<dl>
<dt><a name="index-handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted"></a>function: <strong>handle-read-draw-probe-dots-unformatted</strong> <em>dots_file imol show_clash_gui_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>dots_file</em> is a string
</li><li> <em>imol</em> is an integer number
</li><li> <em>show_clash_gui_flag</em> is an integer number
</li></ul>

<p>pass a filename that contains molprobity&rsquo;s probe output in unformatted format 
</p>
</dd></dl>

<hr>
<a name="set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis"></a>
<div class="header">
<p>
Next: <a href="#do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate" accesskey="n" rel="next">do-probe-dots-on-rotamers-and-chis-state</a>, Previous: <a href="#handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted" accesskey="p" rel="previous">handle-read-draw-probe-dots-unformatted</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis-1"></a>
<h4 class="subsection">11.100.3 set-do-probe-dots-on-rotamers-and-chis</h4>
<dl>
<dt><a name="index-set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis"></a>function: <strong>set-do-probe-dots-on-rotamers-and-chis</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>shall we run molprobity for on edit chi angles intermediate atoms? 
</p>
</dd></dl>

<hr>
<a name="do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate"></a>
<div class="header">
<p>
Next: <a href="#set_002ddo_002dprobe_002ddots_002dpost_002drefine" accesskey="n" rel="next">set-do-probe-dots-post-refine</a>, Previous: <a href="#set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis" accesskey="p" rel="previous">set-do-probe-dots-on-rotamers-and-chis</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate-1"></a>
<h4 class="subsection">11.100.4 do-probe-dots-on-rotamers-and-chis-state</h4>
<dl>
<dt><a name="index-do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate"></a>function: <strong>do-probe-dots-on-rotamers-and-chis-state</strong></dt>
<dd>
<p>return the state of if run molprobity for on edit chi angles intermediate atoms? 
</p>
</dd></dl>

<hr>
<a name="set_002ddo_002dprobe_002ddots_002dpost_002drefine"></a>
<div class="header">
<p>
Next: <a href="#do_002dprobe_002ddots_002dpost_002drefine_002dstate" accesskey="n" rel="next">do-probe-dots-post-refine-state</a>, Previous: <a href="#do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate" accesskey="p" rel="previous">do-probe-dots-on-rotamers-and-chis-state</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002ddo_002dprobe_002ddots_002dpost_002drefine-1"></a>
<h4 class="subsection">11.100.5 set-do-probe-dots-post-refine</h4>
<dl>
<dt><a name="index-set_002ddo_002dprobe_002ddots_002dpost_002drefine"></a>function: <strong>set-do-probe-dots-post-refine</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>shall we run molprobity after a refinement has happened? 
</p>
</dd></dl>

<hr>
<a name="do_002dprobe_002ddots_002dpost_002drefine_002dstate"></a>
<div class="header">
<p>
Next: <a href="#unmangle_002dhydrogen_002dname" accesskey="n" rel="next">unmangle-hydrogen-name</a>, Previous: <a href="#set_002ddo_002dprobe_002ddots_002dpost_002drefine" accesskey="p" rel="previous">set-do-probe-dots-post-refine</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dprobe_002ddots_002dpost_002drefine_002dstate-1"></a>
<h4 class="subsection">11.100.6 do-probe-dots-post-refine-state</h4>
<dl>
<dt><a name="index-do_002dprobe_002ddots_002dpost_002drefine_002dstate"></a>function: <strong>do-probe-dots-post-refine-state</strong></dt>
<dd>
<p>show the state of shall we run molprobity after a refinement has happened? 
</p>
</dd></dl>

<hr>
<a name="unmangle_002dhydrogen_002dname"></a>
<div class="header">
<p>
Next: <a href="#set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius" accesskey="n" rel="next">set-interactive-probe-dots-molprobity-radius</a>, Previous: <a href="#do_002dprobe_002ddots_002dpost_002drefine_002dstate" accesskey="p" rel="previous">do-probe-dots-post-refine-state</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="unmangle_002dhydrogen_002dname-1"></a>
<h4 class="subsection">11.100.7 unmangle-hydrogen-name</h4>
<dl>
<dt><a name="index-unmangle_002dhydrogen_002dname"></a>function: <strong>unmangle-hydrogen-name</strong> <em>pdb_hydrogen_name</em></dt>
<dd>
<p>Where <em>pdb_hydrogen_name</em> is a string
</p>
<p>make an attempt to convert pdb hydrogen name to the name used in Coot (and the refmac dictionary, perhaps). 
</p>
</dd></dl>

<hr>
<a name="set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius"></a>
<div class="header">
<p>
Next: <a href="#interactive_002dprobe_002ddots_002dmolprobity_002dradius" accesskey="n" rel="next">interactive-probe-dots-molprobity-radius</a>, Previous: <a href="#unmangle_002dhydrogen_002dname" accesskey="p" rel="previous">unmangle-hydrogen-name</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius-1"></a>
<h4 class="subsection">11.100.8 set-interactive-probe-dots-molprobity-radius</h4>
<dl>
<dt><a name="index-set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius"></a>function: <strong>set-interactive-probe-dots-molprobity-radius</strong> <em>r</em></dt>
<dd>
<p>Where <em>r</em> is a number
</p>
<p>set the radius over which we can run interactive probe, bigger is better but slower. 
</p>
<p>default is 6.0 
</p>
</dd></dl>

<hr>
<a name="interactive_002dprobe_002ddots_002dmolprobity_002dradius"></a>
<div class="header">
<p>
Next: <a href="#user_002dmods_002dscm" accesskey="n" rel="next">user-mods-scm</a>, Previous: <a href="#set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius" accesskey="p" rel="previous">set-interactive-probe-dots-molprobity-radius</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="interactive_002dprobe_002ddots_002dmolprobity_002dradius-1"></a>
<h4 class="subsection">11.100.9 interactive-probe-dots-molprobity-radius</h4>
<dl>
<dt><a name="index-interactive_002dprobe_002ddots_002dmolprobity_002dradius"></a>function: <strong>interactive-probe-dots-molprobity-radius</strong></dt>
<dd>
<p>return the radius over which we can run interactive probe. 
</p>
</dd></dl>

<hr>
<a name="user_002dmods_002dscm"></a>
<div class="header">
<p>
Next: <a href="#user_002dmods_002dpy" accesskey="n" rel="next">user-mods-py</a>, Previous: <a href="#interactive_002dprobe_002ddots_002dmolprobity_002dradius" accesskey="p" rel="previous">interactive-probe-dots-molprobity-radius</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="user_002dmods_002dscm-1"></a>
<h4 class="subsection">11.100.10 user-mods-scm</h4>
<dl>
<dt><a name="index-user_002dmods_002dscm"></a>function: <strong>user-mods-scm</strong> <em>file_name</em></dt>
<dd>
<p>Where <em>file_name</em> is a string
</p>
<p>return the parsed user mod fields from the PDB file file_name (output by reduce most likely) 
</p>
</dd></dl>

<hr>
<a name="user_002dmods_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#user_002dmods_002dscm" accesskey="p" rel="previous">user-mods-scm</a>, Up: <a href="#Molprobity-Interface" accesskey="u" rel="up">Molprobity Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="user_002dmods_002dpy-1"></a>
<h4 class="subsection">11.100.11 user-mods-py</h4>
<dl>
<dt><a name="index-user_002dmods_002dpy"></a>function: <strong>user-mods-py</strong> <em>file_name</em></dt>
<dd>
<p>Where <em>file_name</em> is a string
</p>
<p>return the parsed user mod fields from the PDB file file_name (output by reduce most likely) 
</p>
</dd></dl>

<hr>
<a name="Map-Sharpening-Interface"></a>
<div class="header">
<p>
Next: <a href="#Intermediate-Atom-Manipulation-Interface" accesskey="n" rel="next">Intermediate Atom Manipulation Interface</a>, Previous: <a href="#Molprobity-Interface" accesskey="p" rel="previous">Molprobity Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Sharpening-Interface-1"></a>
<h3 class="section">11.101 Map Sharpening Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#sharpen" accesskey="1">sharpen</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dsharpening_002dscale_002dlimit" accesskey="2">set-map-sharpening-scale-limit</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="sharpen"></a>
<div class="header">
<p>
Next: <a href="#set_002dmap_002dsharpening_002dscale_002dlimit" accesskey="n" rel="next">set-map-sharpening-scale-limit</a>, Up: <a href="#Map-Sharpening-Interface" accesskey="u" rel="up">Map Sharpening Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="sharpen-1"></a>
<h4 class="subsection">11.101.1 sharpen</h4>
<dl>
<dt><a name="index-sharpen"></a>function: <strong>sharpen</strong> <em>imol b_factor</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>b_factor</em> is a number
</li></ul>

<p>Sharpen map imol by b_factor (note (of course) that positive numbers blur the map). 
</p>
</dd></dl>

<hr>
<a name="set_002dmap_002dsharpening_002dscale_002dlimit"></a>
<div class="header">
<p>
Previous: <a href="#sharpen" accesskey="p" rel="previous">sharpen</a>, Up: <a href="#Map-Sharpening-Interface" accesskey="u" rel="up">Map Sharpening Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dsharpening_002dscale_002dlimit-1"></a>
<h4 class="subsection">11.101.2 set-map-sharpening-scale-limit</h4>
<dl>
<dt><a name="index-set_002dmap_002dsharpening_002dscale_002dlimit"></a>function: <strong>set-map-sharpening-scale-limit</strong> <em>f</em></dt>
<dd>
<p>Where <em>f</em> is a number
</p>
<p>set the limit of the b-factor map sharpening slider (default 30) 
</p>
</dd></dl>

<hr>
<a name="Intermediate-Atom-Manipulation-Interface"></a>
<div class="header">
<p>
Next: <a href="#Marking-Fixed-Atom-Interface" accesskey="n" rel="next">Marking Fixed Atom Interface</a>, Previous: <a href="#Map-Sharpening-Interface" accesskey="p" rel="previous">Map Sharpening Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Intermediate-Atom-Manipulation-Interface-1"></a>
<h3 class="section">11.102 Intermediate Atom Manipulation Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#add_002dtarget_002dposition_002drestraint_002dfor_002dintermediate_002datom_002dpy" accesskey="1">add-target-position-restraint-for-intermediate-atom-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="add_002dtarget_002dposition_002drestraint_002dfor_002dintermediate_002datom_002dpy"></a>
<div class="header">
<p>
Up: <a href="#Intermediate-Atom-Manipulation-Interface" accesskey="u" rel="up">Intermediate Atom Manipulation Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dtarget_002dposition_002drestraint_002dfor_002dintermediate_002datom_002dpy-1"></a>
<h4 class="subsection">11.102.1 add-target-position-restraint-for-intermediate-atom-py</h4>
<dl>
<dt><a name="index-add_002dtarget_002dposition_002drestraint_002dfor_002dintermediate_002datom_002dpy"></a>function: <strong>add-target-position-restraint-for-intermediate-atom-py</strong> <em>atom_spec position</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>atom_spec</em> is a PyObject
</li><li> <em>position</em> is a PyObject
</li></ul>

<p>add a target position for an intermediate atom and refine 
</p>
</dd></dl>

<hr>
<a name="Marking-Fixed-Atom-Interface"></a>
<div class="header">
<p>
Next: <a href="#Partial-Charges" accesskey="n" rel="next">Partial Charges</a>, Previous: <a href="#Intermediate-Atom-Manipulation-Interface" accesskey="p" rel="previous">Intermediate Atom Manipulation Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Marking-Fixed-Atom-Interface-1"></a>
<h3 class="section">11.103 Marking Fixed Atom Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#clear_002dall_002dfixed_002datoms" accesskey="1">clear-all-fixed-atoms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="clear_002dall_002dfixed_002datoms"></a>
<div class="header">
<p>
Up: <a href="#Marking-Fixed-Atom-Interface" accesskey="u" rel="up">Marking Fixed Atom Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002dall_002dfixed_002datoms-1"></a>
<h4 class="subsection">11.103.1 clear-all-fixed-atoms</h4>
<dl>
<dt><a name="index-clear_002dall_002dfixed_002datoms"></a>function: <strong>clear-all-fixed-atoms</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>clear all fixed atoms 
</p>
</dd></dl>

<hr>
<a name="Partial-Charges"></a>
<div class="header">
<p>
Next: <a href="#EM-interface" accesskey="n" rel="next">EM interface</a>, Previous: <a href="#Marking-Fixed-Atom-Interface" accesskey="p" rel="previous">Marking Fixed Atom Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Partial-Charges-1"></a>
<h3 class="section">11.104 Partial Charges</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#show_002dpartial_002dcharge_002dinfo" accesskey="1">show-partial-charge-info</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="show_002dpartial_002dcharge_002dinfo"></a>
<div class="header">
<p>
Up: <a href="#Partial-Charges" accesskey="u" rel="up">Partial Charges</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="show_002dpartial_002dcharge_002dinfo-1"></a>
<h4 class="subsection">11.104.1 show-partial-charge-info</h4>
<dl>
<dt><a name="index-show_002dpartial_002dcharge_002dinfo"></a>function: <strong>show-partial-charge-info</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>show the partial charges for the residue of the given specs (charges are read from the dictionary) 
</p>
</dd></dl>

<hr>
<a name="EM-interface"></a>
<div class="header">
<p>
Next: <a href="#CCP4mg-Interface" accesskey="n" rel="next">CCP4mg Interface</a>, Previous: <a href="#Partial-Charges" accesskey="p" rel="previous">Partial Charges</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="EM-interface-1"></a>
<h3 class="section">11.105 EM interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#scale_002dcell" accesskey="1">scale-cell</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="scale_002dcell"></a>
<div class="header">
<p>
Up: <a href="#EM-interface" accesskey="u" rel="up">EM interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="scale_002dcell-1"></a>
<h4 class="subsection">11.105.1 scale-cell</h4>
<dl>
<dt><a name="index-scale_002dcell"></a>function: <strong>scale-cell</strong> <em>imol_map fac_u fac_v fac_w</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>fac_u</em> is a number
</li><li> <em>fac_v</em> is a number
</li><li> <em>fac_w</em> is a number
</li></ul>

<p>Scale the cell, for use with EM maps, where the cell needs to be adjusted. Use like: (scale-cell 2 1.012 1.012 1.012). Return error status, 1 means it worked, 0 means it did not work. 
</p>
</dd></dl>

<hr>
<a name="CCP4mg-Interface"></a>
<div class="header">
<p>
Next: <a href="#Dipoles" accesskey="n" rel="next">Dipoles</a>, Previous: <a href="#EM-interface" accesskey="p" rel="previous">EM interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="CCP4mg-Interface-1"></a>
<h3 class="section">11.106 CCP4mg Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#ccp4i_002dprojects_002dscm" accesskey="1">ccp4i-projects-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ccp4i_002dprojects_002dpy" accesskey="2">ccp4i-projects-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs" accesskey="3">set-add-ccp4i-projects-to-file-dialogs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#write_002dccp4mg_002dpicture_002ddescription" accesskey="4">write-ccp4mg-picture-description</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002datom_002dcolour_002dfrom_002dmol_002dno" accesskey="5">get-atom-colour-from-mol-no</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="ccp4i_002dprojects_002dscm"></a>
<div class="header">
<p>
Next: <a href="#ccp4i_002dprojects_002dpy" accesskey="n" rel="next">ccp4i-projects-py</a>, Up: <a href="#CCP4mg-Interface" accesskey="u" rel="up">CCP4mg Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ccp4i_002dprojects_002dscm-1"></a>
<h4 class="subsection">11.106.1 ccp4i-projects-scm</h4>
<dl>
<dt><a name="index-ccp4i_002dprojects_002dscm"></a>function: <strong>ccp4i-projects-scm</strong></dt>
<dd>
<p>return a list of pairs of strings, the project names and the directory. Include aliases. 
</p>
</dd></dl>

<hr>
<a name="ccp4i_002dprojects_002dpy"></a>
<div class="header">
<p>
Next: <a href="#set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs" accesskey="n" rel="next">set-add-ccp4i-projects-to-file-dialogs</a>, Previous: <a href="#ccp4i_002dprojects_002dscm" accesskey="p" rel="previous">ccp4i-projects-scm</a>, Up: <a href="#CCP4mg-Interface" accesskey="u" rel="up">CCP4mg Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ccp4i_002dprojects_002dpy-1"></a>
<h4 class="subsection">11.106.2 ccp4i-projects-py</h4>
<dl>
<dt><a name="index-ccp4i_002dprojects_002dpy"></a>function: <strong>ccp4i-projects-py</strong></dt>
<dd>
<p>return a list of pairs of strings, the project names and the directory. Include aliases. 
</p>
</dd></dl>

<hr>
<a name="set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs"></a>
<div class="header">
<p>
Next: <a href="#write_002dccp4mg_002dpicture_002ddescription" accesskey="n" rel="next">write-ccp4mg-picture-description</a>, Previous: <a href="#ccp4i_002dprojects_002dpy" accesskey="p" rel="previous">ccp4i-projects-py</a>, Up: <a href="#CCP4mg-Interface" accesskey="u" rel="up">CCP4mg Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs-1"></a>
<h4 class="subsection">11.106.3 set-add-ccp4i-projects-to-file-dialogs</h4>
<dl>
<dt><a name="index-set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs"></a>function: <strong>set-add-ccp4i-projects-to-file-dialogs</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>allow the user to not add ccp4i directories to the file choosers 
</p>
<p>use state=0 to turn it off 
</p>
</dd></dl>

<hr>
<a name="write_002dccp4mg_002dpicture_002ddescription"></a>
<div class="header">
<p>
Next: <a href="#get_002datom_002dcolour_002dfrom_002dmol_002dno" accesskey="n" rel="next">get-atom-colour-from-mol-no</a>, Previous: <a href="#set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs" accesskey="p" rel="previous">set-add-ccp4i-projects-to-file-dialogs</a>, Up: <a href="#CCP4mg-Interface" accesskey="u" rel="up">CCP4mg Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002dccp4mg_002dpicture_002ddescription-1"></a>
<h4 class="subsection">11.106.4 write-ccp4mg-picture-description</h4>
<dl>
<dt><a name="index-write_002dccp4mg_002dpicture_002ddescription"></a>function: <strong>write-ccp4mg-picture-description</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>write a ccp4mg picture description file 
</p>
</dd></dl>

<hr>
<a name="get_002datom_002dcolour_002dfrom_002dmol_002dno"></a>
<div class="header">
<p>
Previous: <a href="#write_002dccp4mg_002dpicture_002ddescription" accesskey="p" rel="previous">write-ccp4mg-picture-description</a>, Up: <a href="#CCP4mg-Interface" accesskey="u" rel="up">CCP4mg Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002datom_002dcolour_002dfrom_002dmol_002dno-1"></a>
<h4 class="subsection">11.106.5 get-atom-colour-from-mol-no</h4>
<dl>
<dt><a name="index-get_002datom_002dcolour_002dfrom_002dmol_002dno"></a>function: <strong>get-atom-colour-from-mol-no</strong> <em>imol element</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>element</em> is a string
</li></ul>

<p>get element colour for imol as Python formatted list char 
</p>
</dd></dl>

<hr>
<a name="Dipoles"></a>
<div class="header">
<p>
Next: <a href="#Aux-functions" accesskey="n" rel="next">Aux functions</a>, Previous: <a href="#CCP4mg-Interface" accesskey="p" rel="previous">CCP4mg Interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Dipoles-1"></a>
<h3 class="section">11.107 Dipoles</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#add_002ddipole_002dfor_002dresidues_002dscm" accesskey="1">add-dipole-for-residues-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002ddipole_002dscm" accesskey="2">add-dipole-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002ddipole_002dpy" accesskey="3">add-dipole-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002ddipole_002dfor_002dresidues_002dpy" accesskey="4">add-dipole-for-residues-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="add_002ddipole_002dfor_002dresidues_002dscm"></a>
<div class="header">
<p>
Next: <a href="#add_002ddipole_002dscm" accesskey="n" rel="next">add-dipole-scm</a>, Up: <a href="#Dipoles" accesskey="u" rel="up">Dipoles</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002ddipole_002dfor_002dresidues_002dscm-1"></a>
<h4 class="subsection">11.107.1 add-dipole-for-residues-scm</h4>
<dl>
<dt><a name="index-add_002ddipole_002dfor_002dresidues_002dscm"></a>function: <strong>add-dipole-for-residues-scm</strong> <em>imol residue_specs</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a SCM
</li></ul>

<p>generate a dipole from all atoms in the given residues. Return the dipole description 
</p>
</dd></dl>

<hr>
<a name="add_002ddipole_002dscm"></a>
<div class="header">
<p>
Next: <a href="#add_002ddipole_002dpy" accesskey="n" rel="next">add-dipole-py</a>, Previous: <a href="#add_002ddipole_002dfor_002dresidues_002dscm" accesskey="p" rel="previous">add-dipole-for-residues-scm</a>, Up: <a href="#Dipoles" accesskey="u" rel="up">Dipoles</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002ddipole_002dscm-1"></a>
<h4 class="subsection">11.107.2 add-dipole-scm</h4>
<dl>
<dt><a name="index-add_002ddipole_002dscm"></a>function: <strong>add-dipole-scm</strong> <em>imol chain_id res_no ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>return the dipole number 
</p>
</dd></dl>

<hr>
<a name="add_002ddipole_002dpy"></a>
<div class="header">
<p>
Next: <a href="#add_002ddipole_002dfor_002dresidues_002dpy" accesskey="n" rel="next">add-dipole-for-residues-py</a>, Previous: <a href="#add_002ddipole_002dscm" accesskey="p" rel="previous">add-dipole-scm</a>, Up: <a href="#Dipoles" accesskey="u" rel="up">Dipoles</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002ddipole_002dpy-1"></a>
<h4 class="subsection">11.107.3 add-dipole-py</h4>
<dl>
<dt><a name="index-add_002ddipole_002dpy"></a>function: <strong>add-dipole-py</strong> <em>imol chain_id res_no ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>generate a dipole from all atoms in the given residues. 
</p>
</dd></dl>

<hr>
<a name="add_002ddipole_002dfor_002dresidues_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#add_002ddipole_002dpy" accesskey="p" rel="previous">add-dipole-py</a>, Up: <a href="#Dipoles" accesskey="u" rel="up">Dipoles</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002ddipole_002dfor_002dresidues_002dpy-1"></a>
<h4 class="subsection">11.107.4 add-dipole-for-residues-py</h4>
<dl>
<dt><a name="index-add_002ddipole_002dfor_002dresidues_002dpy"></a>function: <strong>add-dipole-for-residues-py</strong> <em>imol residue_specs</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a PyObject
</li></ul>

<p>add a dipole given a set of residues. Return a dipole description. 
</p>
</dd></dl>

<hr>
<a name="Aux-functions"></a>
<div class="header">
<p>
Next: <a href="#SMILES" accesskey="n" rel="next">SMILES</a>, Previous: <a href="#Dipoles" accesskey="p" rel="previous">Dipoles</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Aux-functions-1"></a>
<h3 class="section">11.108 Aux functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#laplacian" accesskey="1">laplacian</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="laplacian"></a>
<div class="header">
<p>
Up: <a href="#Aux-functions" accesskey="u" rel="up">Aux functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="laplacian-1"></a>
<h4 class="subsection">11.108.1 laplacian</h4>
<dl>
<dt><a name="index-laplacian"></a>function: <strong>laplacian</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Create the &quot;Laplacian&quot; (-ve second derivative) of the given map. 
</p>
</dd></dl>

<hr>
<a name="SMILES"></a>
<div class="header">
<p>
Next: <a href="#PHENIX-Support" accesskey="n" rel="next">PHENIX Support</a>, Previous: <a href="#Aux-functions" accesskey="p" rel="previous">Aux functions</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="SMILES-1"></a>
<h3 class="section">11.109 SMILES</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#do_002dsmiles_002dgui" accesskey="1">do-smiles-gui</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="do_002dsmiles_002dgui"></a>
<div class="header">
<p>
Up: <a href="#SMILES" accesskey="u" rel="up">SMILES</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="do_002dsmiles_002dgui-1"></a>
<h4 class="subsection">11.109.1 do-smiles-gui</h4>
<dl>
<dt><a name="index-do_002dsmiles_002dgui"></a>function: <strong>do-smiles-gui</strong></dt>
<dd>
<p>display the SMILES string dialog 
</p>
</dd></dl>

<hr>
<a name="PHENIX-Support"></a>
<div class="header">
<p>
Next: <a href="#Graphics-Text" accesskey="n" rel="next">Graphics Text</a>, Previous: <a href="#SMILES" accesskey="p" rel="previous">SMILES</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="PHENIX-Support-1"></a>
<h3 class="section">11.110 PHENIX Support</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dbutton_002dlabel_002dfor_002dexternal_002drefinement" accesskey="1">set-button-label-for-external-refinement</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dbutton_002dlabel_002dfor_002dexternal_002drefinement"></a>
<div class="header">
<p>
Up: <a href="#PHENIX-Support" accesskey="u" rel="up">PHENIX Support</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dbutton_002dlabel_002dfor_002dexternal_002drefinement-1"></a>
<h4 class="subsection">11.110.1 set-button-label-for-external-refinement</h4>
<dl>
<dt><a name="index-set_002dbutton_002dlabel_002dfor_002dexternal_002drefinement"></a>function: <strong>set-button-label-for-external-refinement</strong> <em>button_label</em></dt>
<dd>
<p>Where <em>button_label</em> is a string
</p>
<p>set the button label of the external Refinement program 
</p>
</dd></dl>

<hr>
<a name="Graphics-Text"></a>
<div class="header">
<p>
Next: <a href="#PISA-Interaction" accesskey="n" rel="next">PISA Interaction</a>, Previous: <a href="#PHENIX-Support" accesskey="p" rel="previous">PHENIX Support</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Graphics-Text-1"></a>
<h3 class="section">11.111 Graphics Text</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#place_002dtext" accesskey="1">place-text</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remove_002dtext" accesskey="2">remove-text</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#text_002dindex_002dnear_002dposition" accesskey="3">text-index-near-position</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="place_002dtext"></a>
<div class="header">
<p>
Next: <a href="#remove_002dtext" accesskey="n" rel="next">remove-text</a>, Up: <a href="#Graphics-Text" accesskey="u" rel="up">Graphics Text</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="place_002dtext-1"></a>
<h4 class="subsection">11.111.1 place-text</h4>
<dl>
<dt><a name="index-place_002dtext"></a>function: <strong>place-text</strong> <em>text x y z size</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>text</em> is a string
</li><li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li><li> <em>size</em> is an integer number
</li></ul>

<p>Put text at x,y,z. 
</p>

<p>size variable is currently ignored. 
</p>
<p>Returns: a text handle
</p>
</dd></dl>

<hr>
<a name="remove_002dtext"></a>
<div class="header">
<p>
Next: <a href="#text_002dindex_002dnear_002dposition" accesskey="n" rel="next">text-index-near-position</a>, Previous: <a href="#place_002dtext" accesskey="p" rel="previous">place-text</a>, Up: <a href="#Graphics-Text" accesskey="u" rel="up">Graphics Text</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remove_002dtext-1"></a>
<h4 class="subsection">11.111.2 remove-text</h4>
<dl>
<dt><a name="index-remove_002dtext"></a>function: <strong>remove-text</strong> <em>text_handle</em></dt>
<dd>
<p>Where <em>text_handle</em> is an integer number
</p>
<p>Remove &quot;3d&quot; text item. 
</p>
</dd></dl>

<hr>
<a name="text_002dindex_002dnear_002dposition"></a>
<div class="header">
<p>
Previous: <a href="#remove_002dtext" accesskey="p" rel="previous">remove-text</a>, Up: <a href="#Graphics-Text" accesskey="u" rel="up">Graphics Text</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="text_002dindex_002dnear_002dposition-1"></a>
<h4 class="subsection">11.111.3 text-index-near-position</h4>
<dl>
<dt><a name="index-text_002dindex_002dnear_002dposition"></a>function: <strong>text-index-near-position</strong> <em>x y z r</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>x</em> is a number
</li><li> <em>y</em> is a number
</li><li> <em>z</em> is a number
</li><li> <em>r</em> is a number
</li></ul>

<p>return the closest text that is with r A of the given position. If no text item is close, then return -1 
</p>
</dd></dl>

<hr>
<a name="PISA-Interaction"></a>
<div class="header">
<p>
Next: <a href="#Jiggle-Fit" accesskey="n" rel="next">Jiggle Fit</a>, Previous: <a href="#Graphics-Text" accesskey="p" rel="previous">Graphics Text</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="PISA-Interaction-1"></a>
<h3 class="section">11.112 PISA Interaction</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#pisa_002dinteraction" accesskey="1">pisa-interaction</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="pisa_002dinteraction"></a>
<div class="header">
<p>
Up: <a href="#PISA-Interaction" accesskey="u" rel="up">PISA Interaction</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="pisa_002dinteraction-1"></a>
<h4 class="subsection">11.112.1 pisa-interaction</h4>
<dl>
<dt><a name="index-pisa_002dinteraction"></a>function: <strong>pisa-interaction</strong> <em>imol_1 imol_2</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_1</em> is an integer number
</li><li> <em>imol_2</em> is an integer number
</li></ul>

<p>return the molecule number of the interacting residues. Return -1 if no new model was created. Old, not very useful. 
</p>
</dd></dl>

<hr>
<a name="Jiggle-Fit"></a>
<div class="header">
<p>
Next: <a href="#SBase-interface" accesskey="n" rel="next">SBase interface</a>, Previous: <a href="#PISA-Interaction" accesskey="p" rel="previous">PISA Interaction</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Jiggle-Fit-1"></a>
<h3 class="section">11.113 Jiggle Fit</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#fit_002dto_002dmap_002dby_002drandom_002djiggle" accesskey="1">fit-to-map-by-random-jiggle</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle" accesskey="2">fit-molecule-to-map-by-random-jiggle</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur" accesskey="3">fit-molecule-to-map-by-random-jiggle-and-blur</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle" accesskey="4">fit-chain-to-map-by-random-jiggle</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur" accesskey="5">fit-chain-to-map-by-random-jiggle-and-blur</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="fit_002dto_002dmap_002dby_002drandom_002djiggle"></a>
<div class="header">
<p>
Next: <a href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle" accesskey="n" rel="next">fit-molecule-to-map-by-random-jiggle</a>, Up: <a href="#Jiggle-Fit" accesskey="u" rel="up">Jiggle Fit</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fit_002dto_002dmap_002dby_002drandom_002djiggle-1"></a>
<h4 class="subsection">11.113.1 fit-to-map-by-random-jiggle</h4>
<dl>
<dt><a name="index-fit_002dto_002dmap_002dby_002drandom_002djiggle"></a>function: <strong>fit-to-map-by-random-jiggle</strong> <em>imol chain_id resno ins_code n_trials jiggle_scale_factor</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>n_trials</em> is an integer number
</li><li> <em>jiggle_scale_factor</em> is a number
</li></ul>

<p>jiggle fit to the current refinment map. return &lt; -100 if not possible, else return the new best fit for this residue. 
</p>
</dd></dl>

<hr>
<a name="fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle"></a>
<div class="header">
<p>
Next: <a href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur" accesskey="n" rel="next">fit-molecule-to-map-by-random-jiggle-and-blur</a>, Previous: <a href="#fit_002dto_002dmap_002dby_002drandom_002djiggle" accesskey="p" rel="previous">fit-to-map-by-random-jiggle</a>, Up: <a href="#Jiggle-Fit" accesskey="u" rel="up">Jiggle Fit</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle-1"></a>
<h4 class="subsection">11.113.2 fit-molecule-to-map-by-random-jiggle</h4>
<dl>
<dt><a name="index-fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle"></a>function: <strong>fit-molecule-to-map-by-random-jiggle</strong> <em>imol n_trials jiggle_scale_factor</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>n_trials</em> is an integer number
</li><li> <em>jiggle_scale_factor</em> is a number
</li></ul>

<p>jiggle fit the molecule to the current refinment map. return &lt; -100 if not possible, else return the new best fit for this molecule. 
</p>
</dd></dl>

<hr>
<a name="fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur"></a>
<div class="header">
<p>
Next: <a href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle" accesskey="n" rel="next">fit-chain-to-map-by-random-jiggle</a>, Previous: <a href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle" accesskey="p" rel="previous">fit-molecule-to-map-by-random-jiggle</a>, Up: <a href="#Jiggle-Fit" accesskey="u" rel="up">Jiggle Fit</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur-1"></a>
<h4 class="subsection">11.113.3 fit-molecule-to-map-by-random-jiggle-and-blur</h4>
<dl>
<dt><a name="index-fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur"></a>function: <strong>fit-molecule-to-map-by-random-jiggle-and-blur</strong> <em>imol n_trials jiggle_scale_factor map_blur_factor</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>n_trials</em> is an integer number
</li><li> <em>jiggle_scale_factor</em> is a number
</li><li> <em>map_blur_factor</em> is a number
</li></ul>

<p>jiggle fit the molecule to the current refinment map. return &lt; -100 if not possible, else return the new best fit for this molecule - create a map that is blurred by the given factor for fitting 
</p>
</dd></dl>

<hr>
<a name="fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle"></a>
<div class="header">
<p>
Next: <a href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur" accesskey="n" rel="next">fit-chain-to-map-by-random-jiggle-and-blur</a>, Previous: <a href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur" accesskey="p" rel="previous">fit-molecule-to-map-by-random-jiggle-and-blur</a>, Up: <a href="#Jiggle-Fit" accesskey="u" rel="up">Jiggle Fit</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle-1"></a>
<h4 class="subsection">11.113.4 fit-chain-to-map-by-random-jiggle</h4>
<dl>
<dt><a name="index-fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle"></a>function: <strong>fit-chain-to-map-by-random-jiggle</strong> <em>imol chain_id n_trials jiggle_scale_factor</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>n_trials</em> is an integer number
</li><li> <em>jiggle_scale_factor</em> is a number
</li></ul>

<p>jiggle fit the chain to the current refinment map. return &lt; -100 if not possible, else return the new best fit for this chain. 
</p>
</dd></dl>

<hr>
<a name="fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur"></a>
<div class="header">
<p>
Previous: <a href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle" accesskey="p" rel="previous">fit-chain-to-map-by-random-jiggle</a>, Up: <a href="#Jiggle-Fit" accesskey="u" rel="up">Jiggle Fit</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur-1"></a>
<h4 class="subsection">11.113.5 fit-chain-to-map-by-random-jiggle-and-blur</h4>
<dl>
<dt><a name="index-fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur"></a>function: <strong>fit-chain-to-map-by-random-jiggle-and-blur</strong> <em>imol chain_id n_trials jiggle_scale_factor map_blur_factor</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>n_trials</em> is an integer number
</li><li> <em>jiggle_scale_factor</em> is a number
</li><li> <em>map_blur_factor</em> is a number
</li></ul>

<p>jiggle fit the chain to the current refinment map 
</p>
<p>Use a map that is blurred by the give factor for fitting. 
</p>
<p>Returns: &lt; -100 if not possible, else return the new best fit for this chain. 
</p>
</dd></dl>

<hr>
<a name="SBase-interface"></a>
<div class="header">
<p>
Next: <a href="#FLE_002dView" accesskey="n" rel="next">FLE-View</a>, Previous: <a href="#Jiggle-Fit" accesskey="p" rel="previous">Jiggle Fit</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="SBase-interface-1"></a>
<h3 class="section">11.114 SBase interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm" accesskey="1">matching-compound-names-from-sbase-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy" accesskey="2">matching-compound-names-from-sbase-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dccp4srs_002dmonomer_002dand_002ddictionary" accesskey="3">get-ccp4srs-monomer-and-dictionary</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dsbase_002dmonomer" accesskey="4">get-sbase-monomer</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm"></a>
<div class="header">
<p>
Next: <a href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy" accesskey="n" rel="next">matching-compound-names-from-sbase-py</a>, Up: <a href="#SBase-interface" accesskey="u" rel="up">SBase interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm-1"></a>
<h4 class="subsection">11.114.1 matching-compound-names-from-sbase-scm</h4>
<dl>
<dt><a name="index-matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm"></a>function: <strong>matching-compound-names-from-sbase-scm</strong> <em>compound_name_fragment</em></dt>
<dd>
<p>Where <em>compound_name_fragment</em> is a string
</p>
<p>return a list of compoundIDs of in SBase of which the given string is a substring of the compound name 
</p>
</dd></dl>

<hr>
<a name="matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy"></a>
<div class="header">
<p>
Next: <a href="#get_002dccp4srs_002dmonomer_002dand_002ddictionary" accesskey="n" rel="next">get-ccp4srs-monomer-and-dictionary</a>, Previous: <a href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm" accesskey="p" rel="previous">matching-compound-names-from-sbase-scm</a>, Up: <a href="#SBase-interface" accesskey="u" rel="up">SBase interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy-1"></a>
<h4 class="subsection">11.114.2 matching-compound-names-from-sbase-py</h4>
<dl>
<dt><a name="index-matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy"></a>function: <strong>matching-compound-names-from-sbase-py</strong> <em>compound_name_fragment</em></dt>
<dd>
<p>Where <em>compound_name_fragment</em> is a string
</p>
<p>return a list of compoundIDs of in SBase of which the given string is a substring of the compound name 
</p>
</dd></dl>

<hr>
<a name="get_002dccp4srs_002dmonomer_002dand_002ddictionary"></a>
<div class="header">
<p>
Next: <a href="#get_002dsbase_002dmonomer" accesskey="n" rel="next">get-sbase-monomer</a>, Previous: <a href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy" accesskey="p" rel="previous">matching-compound-names-from-sbase-py</a>, Up: <a href="#SBase-interface" accesskey="u" rel="up">SBase interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dccp4srs_002dmonomer_002dand_002ddictionary-1"></a>
<h4 class="subsection">11.114.3 get-ccp4srs-monomer-and-dictionary</h4>
<dl>
<dt><a name="index-get_002dccp4srs_002dmonomer_002dand_002ddictionary"></a>function: <strong>get-ccp4srs-monomer-and-dictionary</strong> <em>comp_id</em></dt>
<dd>
<p>Where <em>comp_id</em> is a string
</p>
<p>return the new molecule number of the monomer. 
</p>
<p>The monomer will have chainid &quot;A&quot; and residue number 1.
</p>
<p>Return -1 on failure to get monomer. 
</p>
</dd></dl>

<hr>
<a name="get_002dsbase_002dmonomer"></a>
<div class="header">
<p>
Previous: <a href="#get_002dccp4srs_002dmonomer_002dand_002ddictionary" accesskey="p" rel="previous">get-ccp4srs-monomer-and-dictionary</a>, Up: <a href="#SBase-interface" accesskey="u" rel="up">SBase interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dsbase_002dmonomer-1"></a>
<h4 class="subsection">11.114.4 get-sbase-monomer</h4>
<dl>
<dt><a name="index-get_002dsbase_002dmonomer"></a>function: <strong>get-sbase-monomer</strong> <em>comp_id</em></dt>
<dd>
<p>Where <em>comp_id</em> is a string
</p>
<p>same as above but using old name for back-compatibility 
</p>
</dd></dl>

<hr>
<a name="FLE_002dView"></a>
<div class="header">
<p>
Next: <a href="#LSQ_002dimprove" accesskey="n" rel="next">LSQ-improve</a>, Previous: <a href="#SBase-interface" accesskey="p" rel="previous">SBase interface</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="FLE_002dView-1"></a>
<h3 class="section">11.115 FLE-View</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#fle_002dview_002dset_002dwater_002ddist_002dmax" accesskey="1">fle-view-set-water-dist-max</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fle_002dview_002dset_002dh_002dbond_002ddist_002dmax" accesskey="2">fle-view-set-h-bond-dist-max</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="fle_002dview_002dset_002dwater_002ddist_002dmax"></a>
<div class="header">
<p>
Next: <a href="#fle_002dview_002dset_002dh_002dbond_002ddist_002dmax" accesskey="n" rel="next">fle-view-set-h-bond-dist-max</a>, Up: <a href="#FLE_002dView" accesskey="u" rel="up">FLE-View</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fle_002dview_002dset_002dwater_002ddist_002dmax-1"></a>
<h4 class="subsection">11.115.1 fle-view-set-water-dist-max</h4>
<dl>
<dt><a name="index-fle_002dview_002dset_002dwater_002ddist_002dmax"></a>function: <strong>fle-view-set-water-dist-max</strong> <em>dist_max</em></dt>
<dd>
<p>Where <em>dist_max</em> is a number
</p>
<p>set the maximum considered distance to water 
</p>
<p>default 3.25 A. 
</p>
</dd></dl>

<hr>
<a name="fle_002dview_002dset_002dh_002dbond_002ddist_002dmax"></a>
<div class="header">
<p>
Previous: <a href="#fle_002dview_002dset_002dwater_002ddist_002dmax" accesskey="p" rel="previous">fle-view-set-water-dist-max</a>, Up: <a href="#FLE_002dView" accesskey="u" rel="up">FLE-View</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fle_002dview_002dset_002dh_002dbond_002ddist_002dmax-1"></a>
<h4 class="subsection">11.115.2 fle-view-set-h-bond-dist-max</h4>
<dl>
<dt><a name="index-fle_002dview_002dset_002dh_002dbond_002ddist_002dmax"></a>function: <strong>fle-view-set-h-bond-dist-max</strong> <em>h_bond_dist_max</em></dt>
<dd>
<p>Where <em>h_bond_dist_max</em> is a number
</p>
<p>set the maximum considered hydrogen bond distance 
</p>
<p>default 3.9 A. 
</p>
</dd></dl>

<hr>
<a name="LSQ_002dimprove"></a>
<div class="header">
<p>
Next: <a href="#single_002dmodel-view" accesskey="n" rel="next">single-model view</a>, Previous: <a href="#FLE_002dView" accesskey="p" rel="previous">FLE-View</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="LSQ_002dimprove-1"></a>
<h3 class="section">11.116 LSQ-improve</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#lsq_002dimprove" accesskey="1">lsq-improve</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="lsq_002dimprove"></a>
<div class="header">
<p>
Up: <a href="#LSQ_002dimprove" accesskey="u" rel="up">LSQ-improve</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="lsq_002dimprove-1"></a>
<h4 class="subsection">11.116.1 lsq-improve</h4>
<dl>
<dt><a name="index-lsq_002dimprove"></a>function: <strong>lsq-improve</strong> <em>imol_ref ref_selection imol_moving moving_selection n_res dist_crit</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_ref</em> is an integer number
</li><li> <em>ref_selection</em> is a string
</li><li> <em>imol_moving</em> is an integer number
</li><li> <em>moving_selection</em> is a string
</li><li> <em>n_res</em> is an integer number
</li><li> <em>dist_crit</em> is a number
</li></ul>

<p>an slightly-modified implementation of the &quot;lsq_improve&quot; algorithm of Kleywegt and Jones (1997). 
</p>
<p>Note that if a residue selection is specified in the residue selection(s), then the first residue of the given range must exist in the molecule (if not, then mmdb will not select any atoms from that molecule).
</p>
<p>Kleywegt and Jones set n_res to 4 and dist_crit to 6.0. 
</p>
</dd></dl>

<hr>
<a name="single_002dmodel-view"></a>
<div class="header">
<p>
Next: <a href="#graphics-2D-ligand-view" accesskey="n" rel="next">graphics 2D ligand view</a>, Previous: <a href="#LSQ_002dimprove" accesskey="p" rel="previous">LSQ-improve</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="single_002dmodel-view-1"></a>
<h3 class="section">11.117 single-model view</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#single_002dmodel_002dview_002dmodel_002dnumber" accesskey="1">single-model-view-model-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#single_002dmodel_002dview_002dthis_002dmodel_002dnumber" accesskey="2">single-model-view-this-model-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#single_002dmodel_002dview_002dnext_002dmodel_002dnumber" accesskey="3">single-model-view-next-model-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#single_002dmodel_002dview_002dprev_002dmodel_002dnumber" accesskey="4">single-model-view-prev-model-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="single_002dmodel_002dview_002dmodel_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#single_002dmodel_002dview_002dthis_002dmodel_002dnumber" accesskey="n" rel="next">single-model-view-this-model-number</a>, Up: <a href="#single_002dmodel-view" accesskey="u" rel="up">single-model view</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="single_002dmodel_002dview_002dmodel_002dnumber-1"></a>
<h4 class="subsection">11.117.1 single-model-view-model-number</h4>
<dl>
<dt><a name="index-single_002dmodel_002dview_002dmodel_002dnumber"></a>function: <strong>single-model-view-model-number</strong> <em>imol imodel</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>imodel</em> is an integer number
</li></ul>

<p>put molecule number imol to display only model number imodel 
</p>
</dd></dl>

<hr>
<a name="single_002dmodel_002dview_002dthis_002dmodel_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#single_002dmodel_002dview_002dnext_002dmodel_002dnumber" accesskey="n" rel="next">single-model-view-next-model-number</a>, Previous: <a href="#single_002dmodel_002dview_002dmodel_002dnumber" accesskey="p" rel="previous">single-model-view-model-number</a>, Up: <a href="#single_002dmodel-view" accesskey="u" rel="up">single-model view</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="single_002dmodel_002dview_002dthis_002dmodel_002dnumber-1"></a>
<h4 class="subsection">11.117.2 single-model-view-this-model-number</h4>
<dl>
<dt><a name="index-single_002dmodel_002dview_002dthis_002dmodel_002dnumber"></a>function: <strong>single-model-view-this-model-number</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>the current model number being displayed 
</p>
<p>return 0 on non-multimodel-molecule. 
</p>
</dd></dl>

<hr>
<a name="single_002dmodel_002dview_002dnext_002dmodel_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#single_002dmodel_002dview_002dprev_002dmodel_002dnumber" accesskey="n" rel="next">single-model-view-prev-model-number</a>, Previous: <a href="#single_002dmodel_002dview_002dthis_002dmodel_002dnumber" accesskey="p" rel="previous">single-model-view-this-model-number</a>, Up: <a href="#single_002dmodel-view" accesskey="u" rel="up">single-model view</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="single_002dmodel_002dview_002dnext_002dmodel_002dnumber-1"></a>
<h4 class="subsection">11.117.3 single-model-view-next-model-number</h4>
<dl>
<dt><a name="index-single_002dmodel_002dview_002dnext_002dmodel_002dnumber"></a>function: <strong>single-model-view-next-model-number</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>change the representation to the next model number to be displayed 
</p>
<p>return 0 on non-multimodel-molecule. 
</p>
</dd></dl>

<hr>
<a name="single_002dmodel_002dview_002dprev_002dmodel_002dnumber"></a>
<div class="header">
<p>
Previous: <a href="#single_002dmodel_002dview_002dnext_002dmodel_002dnumber" accesskey="p" rel="previous">single-model-view-next-model-number</a>, Up: <a href="#single_002dmodel-view" accesskey="u" rel="up">single-model view</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="single_002dmodel_002dview_002dprev_002dmodel_002dnumber-1"></a>
<h4 class="subsection">11.117.4 single-model-view-prev-model-number</h4>
<dl>
<dt><a name="index-single_002dmodel_002dview_002dprev_002dmodel_002dnumber"></a>function: <strong>single-model-view-prev-model-number</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>change the representation to the previous model number to be displayed 
</p>
<p>return 0 on non-multimodel-molecule. 
</p>
</dd></dl>

<hr>
<a name="graphics-2D-ligand-view"></a>
<div class="header">
<p>
Previous: <a href="#single_002dmodel-view" accesskey="p" rel="previous">single-model view</a>, Up: <a href="#Scripting-Functions" accesskey="u" rel="up">Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="graphics-2D-ligand-view-1"></a>
<h3 class="section">11.118 graphics 2D ligand view</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dshow_002dgraphics_002dligand_002dview" accesskey="1">set-show-graphics-ligand-view</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dshow_002dgraphics_002dligand_002dview"></a>
<div class="header">
<p>
Up: <a href="#graphics-2D-ligand-view" accesskey="u" rel="up">graphics 2D ligand view</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dshow_002dgraphics_002dligand_002dview-1"></a>
<h4 class="subsection">11.118.1 set-show-graphics-ligand-view</h4>
<dl>
<dt><a name="index-set_002dshow_002dgraphics_002dligand_002dview"></a>function: <strong>set-show-graphics-ligand-view</strong> <em>state</em></dt>
<dd>
<p>Where <em>state</em> is an integer number
</p>
<p>set the graphics ligand view state 
</p>
<p>(default is 1 (on)). 
</p>
</dd></dl>

<hr>
<a name="More-Scripting-Functions"></a>
<div class="header">
<p>
Next: <a href="#More-Refinement-Scripting-Functions" accesskey="n" rel="next">More Refinement Scripting Functions</a>, Previous: <a href="#Scripting-Functions" accesskey="p" rel="previous">Scripting Functions</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="More-Scripting-Functions-1"></a>
<h2 class="chapter">12 More Scripting Functions</h2>
<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#More-Symmetry-Functions" accesskey="1">More Symmetry Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Extra-Map-Functions" accesskey="2">Extra Map Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Multi_002dResidue-Torsion" accesskey="3">Multi-Residue Torsion</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Merge-Fragments" accesskey="4">Merge Fragments</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Execute-Refmac" accesskey="5">Execute Refmac</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Dictionary-Functions" accesskey="6">Dictionary Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Restraints-Interface" accesskey="7">Restraints Interface</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Atom-Information-functions" accesskey="8">Atom Information functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Using-S_002dexpression-molecules" accesskey="9">Using S-expression molecules</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Refinement-with-specs">Refinement with specs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Water-Chain-Functions">Water Chain Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Glyco-Tools">Glyco Tools</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Spin-Search-Functions">Spin Search Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Rotamer-Scoring">Rotamer Scoring</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#protein_002ddb">protein-db</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Coot_0027s-Hole-implementation">Coot's Hole implementation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Drag-and-Drop-Functions">Drag and Drop Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-Contouring-Functions">Map Contouring Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Map-to-Model-Correlation">Map to Model Correlation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#Get-Sequence">Get Sequence</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="More-Symmetry-Functions"></a>
<div class="header">
<p>
Next: <a href="#Extra-Map-Functions" accesskey="n" rel="next">Extra Map Functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="More-Symmetry-Functions-1"></a>
<h3 class="section">12.1 More Symmetry Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#get_002dsymmetry" accesskey="1">get-symmetry</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clashes_002dwith_002dsymmetry" accesskey="2">clashes-with-symmetry</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="get_002dsymmetry"></a>
<div class="header">
<p>
Next: <a href="#clashes_002dwith_002dsymmetry" accesskey="n" rel="next">clashes-with-symmetry</a>, Up: <a href="#More-Symmetry-Functions" accesskey="u" rel="up">More Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dsymmetry-1"></a>
<h4 class="subsection">12.1.1 get-symmetry</h4>
<dl>
<dt><a name="index-get_002dsymmetry"></a>function: <strong>get-symmetry</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the symmetry of the imolth molecule 
</p>
<p>Return as a list of strings the symmetry operators of the given molecule. If imol is a not a valid molecule, return an empty list. 
</p>
</dd></dl>

<hr>
<a name="clashes_002dwith_002dsymmetry"></a>
<div class="header">
<p>
Previous: <a href="#get_002dsymmetry" accesskey="p" rel="previous">get-symmetry</a>, Up: <a href="#More-Symmetry-Functions" accesskey="u" rel="up">More Symmetry Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clashes_002dwith_002dsymmetry-1"></a>
<h4 class="subsection">12.1.2 clashes-with-symmetry</h4>
<dl>
<dt><a name="index-clashes_002dwith_002dsymmetry"></a>function: <strong>clashes-with-symmetry</strong> <em>imol chain_id res_no ins_code clash_dist</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>clash_dist</em> is a number
</li></ul>

<p>return 1 if this residue clashes with the symmetry-related atoms of the same molecule. 
</p>
<p>0 means that it did not clash, -1 means that the residue or molecule could not be found or that there was no cell and symmetry. 
</p>
</dd></dl>

<hr>
<a name="Extra-Map-Functions"></a>
<div class="header">
<p>
Next: <a href="#Multi_002dResidue-Torsion" accesskey="n" rel="next">Multi-Residue Torsion</a>, Previous: <a href="#More-Symmetry-Functions" accesskey="p" rel="previous">More Symmetry Functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Extra-Map-Functions-1"></a>
<h3 class="section">12.2 Extra Map Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps" accesskey="1">auto-read-make-and-draw-maps</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz" accesskey="2">auto-read-make-and-draw-maps-from-mtz</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sharpen_002dblur_002dmap" accesskey="3">sharpen-blur-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#sharpen_002dblur_002dmap_002dwith_002dresampling" accesskey="4">sharpen-blur-map-with-resampling</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#multi_002dsharpen_002dblur_002dmap_002dscm" accesskey="5">multi-sharpen-blur-map-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#multi_002dsharpen_002dblur_002dmap_002dpy" accesskey="6">multi-sharpen-blur-map-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#flip_002dhand" accesskey="7">flip-hand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#go_002dto_002dmap_002dmolecule_002dcentre" accesskey="8">go-to-map-molecule-centre</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#b_002dfactor_002dfrom_002dmap" accesskey="9">b-factor-from-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dcolour_002dcomponents">map-colour-components</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dcolour_002dcomponents_002dpy">map-colour-components-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#handle_002dread_002dccp4_002dmap">handle-read-ccp4-map</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="auto_002dread_002dmake_002dand_002ddraw_002dmaps"></a>
<div class="header">
<p>
Next: <a href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz" accesskey="n" rel="next">auto-read-make-and-draw-maps-from-mtz</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="auto_002dread_002dmake_002dand_002ddraw_002dmaps-1"></a>
<h4 class="subsection">12.2.1 auto-read-make-and-draw-maps</h4>
<dl>
<dt><a name="index-auto_002dread_002dmake_002dand_002ddraw_002dmaps"></a>function: <strong>auto-read-make-and-draw-maps</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>read MTZ file filename and from it try to make maps 
</p>
<p>Useful for reading the output of refmac. The default labels (FWT/PHWT and DELFWT/PHDELFWT) can be changed using ...[something]
</p>
<p>Returns: a list of molecule numbers for the new maps 
</p>
</dd></dl>

<hr>
<a name="auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz"></a>
<div class="header">
<p>
Next: <a href="#sharpen_002dblur_002dmap" accesskey="n" rel="next">sharpen-blur-map</a>, Previous: <a href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps" accesskey="p" rel="previous">auto-read-make-and-draw-maps</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz-1"></a>
<h4 class="subsection">12.2.2 auto-read-make-and-draw-maps-from-mtz</h4>
<dl>
<dt><a name="index-auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz"></a>function: <strong>auto-read-make-and-draw-maps-from-mtz</strong> <em>filename</em></dt>
<dd>
<p>Where <em>filename</em> is a string
</p>
<p>set the flag to do a difference map (too) on auto-read MTZ 
</p>
</dd></dl>

<hr>
<a name="sharpen_002dblur_002dmap"></a>
<div class="header">
<p>
Next: <a href="#sharpen_002dblur_002dmap_002dwith_002dresampling" accesskey="n" rel="next">sharpen-blur-map-with-resampling</a>, Previous: <a href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz" accesskey="p" rel="previous">auto-read-make-and-draw-maps-from-mtz</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="sharpen_002dblur_002dmap-1"></a>
<h4 class="subsection">12.2.3 sharpen-blur-map</h4>
<dl>
<dt><a name="index-sharpen_002dblur_002dmap"></a>function: <strong>sharpen-blur-map</strong> <em>imol_map b_factor</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>b_factor</em> is a number
</li></ul>

<p>make a sharpened or blurred map 
</p>
<p>blurred maps are generated by using a positive value of b_factor.
</p>
<p>Returns: the index of the map created by applying a b-factor to the given map. Return -1 on failure. 
</p>
</dd></dl>

<hr>
<a name="sharpen_002dblur_002dmap_002dwith_002dresampling"></a>
<div class="header">
<p>
Next: <a href="#multi_002dsharpen_002dblur_002dmap_002dscm" accesskey="n" rel="next">multi-sharpen-blur-map-scm</a>, Previous: <a href="#sharpen_002dblur_002dmap" accesskey="p" rel="previous">sharpen-blur-map</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="sharpen_002dblur_002dmap_002dwith_002dresampling-1"></a>
<h4 class="subsection">12.2.4 sharpen-blur-map-with-resampling</h4>
<dl>
<dt><a name="index-sharpen_002dblur_002dmap_002dwith_002dresampling"></a>function: <strong>sharpen-blur-map-with-resampling</strong> <em>imol_map b_factor resample_factor</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>b_factor</em> is a number
</li><li> <em>resample_factor</em> is a number
</li></ul>

<p>make a sharpened or blurred map with resampling 
</p>
<p>resampling factor might typically be 1.3
</p>
<p>blurred maps are generated by using a positive value of b_factor.
</p>
<p>Returns: the index of the map created by applying a b-factor to the given map. Return -1 on failure. 
</p>
</dd></dl>

<hr>
<a name="multi_002dsharpen_002dblur_002dmap_002dscm"></a>
<div class="header">
<p>
Next: <a href="#multi_002dsharpen_002dblur_002dmap_002dpy" accesskey="n" rel="next">multi-sharpen-blur-map-py</a>, Previous: <a href="#sharpen_002dblur_002dmap_002dwith_002dresampling" accesskey="p" rel="previous">sharpen-blur-map-with-resampling</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="multi_002dsharpen_002dblur_002dmap_002dscm-1"></a>
<h4 class="subsection">12.2.5 multi-sharpen-blur-map-scm</h4>
<dl>
<dt><a name="index-multi_002dsharpen_002dblur_002dmap_002dscm"></a>function: <strong>multi-sharpen-blur-map-scm</strong> <em>imol_map b_factors_list</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>b_factors_list</em> is a SCM
</li></ul>

<p>make many sharpened or blurred maps 
</p>
<p>blurred maps are generated by using a positive value of b_factor. 
</p>
</dd></dl>

<hr>
<a name="multi_002dsharpen_002dblur_002dmap_002dpy"></a>
<div class="header">
<p>
Next: <a href="#flip_002dhand" accesskey="n" rel="next">flip-hand</a>, Previous: <a href="#multi_002dsharpen_002dblur_002dmap_002dscm" accesskey="p" rel="previous">multi-sharpen-blur-map-scm</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="multi_002dsharpen_002dblur_002dmap_002dpy-1"></a>
<h4 class="subsection">12.2.6 multi-sharpen-blur-map-py</h4>
<dl>
<dt><a name="index-multi_002dsharpen_002dblur_002dmap_002dpy"></a>function: <strong>multi-sharpen-blur-map-py</strong> <em>imol_map b_factors_list</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>b_factors_list</em> is a PyObject
</li></ul>

<p>make many sharpened or blurred maps 
</p>
<p>blurred maps are generated by using a positive value of b_factor. 
</p>
</dd></dl>

<hr>
<a name="flip_002dhand"></a>
<div class="header">
<p>
Next: <a href="#go_002dto_002dmap_002dmolecule_002dcentre" accesskey="n" rel="next">go-to-map-molecule-centre</a>, Previous: <a href="#multi_002dsharpen_002dblur_002dmap_002dpy" accesskey="p" rel="previous">multi-sharpen-blur-map-py</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="flip_002dhand-1"></a>
<h4 class="subsection">12.2.7 flip-hand</h4>
<dl>
<dt><a name="index-flip_002dhand"></a>function: <strong>flip-hand</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>Flip the hand of the map. 
</p>
<p>in case it was accidentally generated on the wrong one. 
</p>
<p>Returns: the molecule number of the flipped map. 
</p>
</dd></dl>

<hr>
<a name="go_002dto_002dmap_002dmolecule_002dcentre"></a>
<div class="header">
<p>
Next: <a href="#b_002dfactor_002dfrom_002dmap" accesskey="n" rel="next">b-factor-from-map</a>, Previous: <a href="#flip_002dhand" accesskey="p" rel="previous">flip-hand</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="go_002dto_002dmap_002dmolecule_002dcentre-1"></a>
<h4 class="subsection">12.2.8 go-to-map-molecule-centre</h4>
<dl>
<dt><a name="index-go_002dto_002dmap_002dmolecule_002dcentre"></a>function: <strong>go-to-map-molecule-centre</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>Go to the centre of the molecule - for Cryo-EM Molecules. 
</p>
<p>and recontour at a sensible value. 
</p>
</dd></dl>

<hr>
<a name="b_002dfactor_002dfrom_002dmap"></a>
<div class="header">
<p>
Next: <a href="#map_002dcolour_002dcomponents" accesskey="n" rel="next">map-colour-components</a>, Previous: <a href="#go_002dto_002dmap_002dmolecule_002dcentre" accesskey="p" rel="previous">go-to-map-molecule-centre</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="b_002dfactor_002dfrom_002dmap-1"></a>
<h4 class="subsection">12.2.9 b-factor-from-map</h4>
<dl>
<dt><a name="index-b_002dfactor_002dfrom_002dmap"></a>function: <strong>b-factor-from-map</strong> <em>imol_map</em></dt>
<dd>
<p>Where <em>imol_map</em> is an integer number
</p>
<p>b-factor from map 
</p>
<p>calculate structure factors and use the amplitudes to estimate the B-factor of the data using a wilson plot using a low resolution limit of 4.5A. 
</p>
<p>Returns: -1 when given a bad map or there were no data beyond 4.5A 
</p>
</dd></dl>

<hr>
<a name="map_002dcolour_002dcomponents"></a>
<div class="header">
<p>
Next: <a href="#map_002dcolour_002dcomponents_002dpy" accesskey="n" rel="next">map-colour-components-py</a>, Previous: <a href="#b_002dfactor_002dfrom_002dmap" accesskey="p" rel="previous">b-factor-from-map</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dcolour_002dcomponents-1"></a>
<h4 class="subsection">12.2.10 map-colour-components</h4>
<dl>
<dt><a name="index-map_002dcolour_002dcomponents"></a>function: <strong>map-colour-components</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the colour triple of the imolth map 
</p>
<p>(e.g.: (list 0.4 0.6 0.8). If invalid imol return scheme false. 
</p>
</dd></dl>

<hr>
<a name="map_002dcolour_002dcomponents_002dpy"></a>
<div class="header">
<p>
Next: <a href="#handle_002dread_002dccp4_002dmap" accesskey="n" rel="next">handle-read-ccp4-map</a>, Previous: <a href="#map_002dcolour_002dcomponents" accesskey="p" rel="previous">map-colour-components</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dcolour_002dcomponents_002dpy-1"></a>
<h4 class="subsection">12.2.11 map-colour-components-py</h4>
<dl>
<dt><a name="index-map_002dcolour_002dcomponents_002dpy"></a>function: <strong>map-colour-components-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the colour triple of the imolth map 
</p>
<p>e.g.: [0.4, 0.6, 0.8]. If invalid imol return Py_False. 
</p>
</dd></dl>

<hr>
<a name="handle_002dread_002dccp4_002dmap"></a>
<div class="header">
<p>
Previous: <a href="#map_002dcolour_002dcomponents_002dpy" accesskey="p" rel="previous">map-colour-components-py</a>, Up: <a href="#Extra-Map-Functions" accesskey="u" rel="up">Extra Map Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002dread_002dccp4_002dmap-1"></a>
<h4 class="subsection">12.2.12 handle-read-ccp4-map</h4>
<dl>
<dt><a name="index-handle_002dread_002dccp4_002dmap"></a>function: <strong>handle-read-ccp4-map</strong> <em>filename is_diff_map_flag</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>filename</em> is a const std::string &amp;
</li><li> <em>is_diff_map_flag</em> is an integer number
</li></ul>

<p>read a CCP4 map or a CNS map (despite the name). 
</p>
</dd></dl>

<hr>
<a name="Multi_002dResidue-Torsion"></a>
<div class="header">
<p>
Next: <a href="#Merge-Fragments" accesskey="n" rel="next">Merge Fragments</a>, Previous: <a href="#Extra-Map-Functions" accesskey="p" rel="previous">Extra Map Functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Multi_002dResidue-Torsion-1"></a>
<h3 class="section">12.3 Multi-Residue Torsion</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#multi_002dresidue_002dtorsion_002dfit_002dscm" accesskey="1">multi-residue-torsion-fit-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#multi_002dresidue_002dtorsion_002dfit_002dpy" accesskey="2">multi-residue-torsion-fit-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="multi_002dresidue_002dtorsion_002dfit_002dscm"></a>
<div class="header">
<p>
Next: <a href="#multi_002dresidue_002dtorsion_002dfit_002dpy" accesskey="n" rel="next">multi-residue-torsion-fit-py</a>, Up: <a href="#Multi_002dResidue-Torsion" accesskey="u" rel="up">Multi-Residue Torsion</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="multi_002dresidue_002dtorsion_002dfit_002dscm-1"></a>
<h4 class="subsection">12.3.1 multi-residue-torsion-fit-scm</h4>
<dl>
<dt><a name="index-multi_002dresidue_002dtorsion_002dfit_002dscm"></a>function: <strong>multi-residue-torsion-fit-scm</strong> <em>imol residues_specs_scm n_trials</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residues_specs_scm</em> is a SCM
</li><li> <em>n_trials</em> is an integer number
</li></ul>

<p>fit residues 
</p>
<p>(note: fit to the current-refinement map) 
</p>
</dd></dl>

<hr>
<a name="multi_002dresidue_002dtorsion_002dfit_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#multi_002dresidue_002dtorsion_002dfit_002dscm" accesskey="p" rel="previous">multi-residue-torsion-fit-scm</a>, Up: <a href="#Multi_002dResidue-Torsion" accesskey="u" rel="up">Multi-Residue Torsion</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="multi_002dresidue_002dtorsion_002dfit_002dpy-1"></a>
<h4 class="subsection">12.3.2 multi-residue-torsion-fit-py</h4>
<dl>
<dt><a name="index-multi_002dresidue_002dtorsion_002dfit_002dpy"></a>function: <strong>multi-residue-torsion-fit-py</strong> <em>imol residues_specs_py n_trials</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residues_specs_py</em> is a PyObject
</li><li> <em>n_trials</em> is an integer number
</li></ul>

<p>fit residues 
</p>
<p>(note: fit to the current-refinement map) 
</p>
</dd></dl>

<hr>
<a name="Merge-Fragments"></a>
<div class="header">
<p>
Next: <a href="#Execute-Refmac" accesskey="n" rel="next">Execute Refmac</a>, Previous: <a href="#Multi_002dResidue-Torsion" accesskey="p" rel="previous">Multi-Residue Torsion</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Merge-Fragments-1"></a>
<h3 class="section">12.4 Merge Fragments</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#merge_002dfragments" accesskey="1">merge-fragments</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="merge_002dfragments"></a>
<div class="header">
<p>
Up: <a href="#Merge-Fragments" accesskey="u" rel="up">Merge Fragments</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="merge_002dfragments-1"></a>
<h4 class="subsection">12.4.1 merge-fragments</h4>
<dl>
<dt><a name="index-merge_002dfragments"></a>function: <strong>merge-fragments</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>each fragment is presumed to be in its own chain. 
</p>
</dd></dl>

<hr>
<a name="Execute-Refmac"></a>
<div class="header">
<p>
Next: <a href="#Dictionary-Functions" accesskey="n" rel="next">Dictionary Functions</a>, Previous: <a href="#Merge-Fragments" accesskey="p" rel="previous">Merge Fragments</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Execute-Refmac-1"></a>
<h3 class="section">12.5 Execute Refmac</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#execute_002drefmac_002dreal" accesskey="1">execute-refmac-real</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refmac_002dname" accesskey="2">refmac-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="execute_002drefmac_002dreal"></a>
<div class="header">
<p>
Next: <a href="#refmac_002dname" accesskey="n" rel="next">refmac-name</a>, Up: <a href="#Execute-Refmac" accesskey="u" rel="up">Execute Refmac</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="execute_002drefmac_002dreal-1"></a>
<h4 class="subsection">12.5.1 execute-refmac-real</h4>
<dl>
<dt><a name="index-execute_002drefmac_002dreal"></a>function: <strong>execute-refmac-real</strong> <em>pdb_in_filename pdb_out_filename mtz_in_filename mtz_out_filename cif_lib_filename fobs_col_name sigfobs_col_name r_free_col_name have_sensible_free_r_flag make_molecules_flag refmac_count_string swap_map_colours_post_refmac_flag imol_refmac_map diff_map_flag phase_combine_flag phib_string fom_string ccp4i_project_dir</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>pdb_in_filename</em> is a std::string
</li><li> <em>pdb_out_filename</em> is a std::string
</li><li> <em>mtz_in_filename</em> is a std::string
</li><li> <em>mtz_out_filename</em> is a std::string
</li><li> <em>cif_lib_filename</em> is a std::string
</li><li> <em>fobs_col_name</em> is a std::string
</li><li> <em>sigfobs_col_name</em> is a std::string
</li><li> <em>r_free_col_name</em> is a std::string
</li><li> <em>have_sensible_free_r_flag</em> is an integer number
</li><li> <em>make_molecules_flag</em> is an integer number
</li><li> <em>refmac_count_string</em> is a std::string
</li><li> <em>swap_map_colours_post_refmac_flag</em> is an integer number
</li><li> <em>imol_refmac_map</em> is an integer number
</li><li> <em>diff_map_flag</em> is an integer number
</li><li> <em>phase_combine_flag</em> is an integer number
</li><li> <em>phib_string</em> is a std::string
</li><li> <em>fom_string</em> is a std::string
</li><li> <em>ccp4i_project_dir</em> is a std::string
</li></ul>

<p>if swap_map_colours_post_refmac_flag is not 1 thenn imol_refmac_map is ignored. 
</p>
</dd></dl>

<hr>
<a name="refmac_002dname"></a>
<div class="header">
<p>
Previous: <a href="#execute_002drefmac_002dreal" accesskey="p" rel="previous">execute-refmac-real</a>, Up: <a href="#Execute-Refmac" accesskey="u" rel="up">Execute Refmac</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refmac_002dname-1"></a>
<h4 class="subsection">12.5.2 refmac-name</h4>
<dl>
<dt><a name="index-refmac_002dname"></a>function: <strong>refmac-name</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>the name for refmac 
</p>
<p>Returns: a stub name used in the construction of filename for refmac output 
</p>
</dd></dl>

<hr>
<a name="Dictionary-Functions"></a>
<div class="header">
<p>
Next: <a href="#Restraints-Interface" accesskey="n" rel="next">Restraints Interface</a>, Previous: <a href="#Execute-Refmac" accesskey="p" rel="previous">Execute Refmac</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Dictionary-Functions-1"></a>
<h3 class="section">12.6 Dictionary Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#dictionaries_002dread" accesskey="1">dictionaries-read</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="dictionaries_002dread"></a>
<div class="header">
<p>
Up: <a href="#Dictionary-Functions" accesskey="u" rel="up">Dictionary Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="dictionaries_002dread-1"></a>
<h4 class="subsection">12.6.1 dictionaries-read</h4>
<dl>
<dt><a name="index-dictionaries_002dread"></a>function: <strong>dictionaries-read</strong></dt>
<dd>
<p>return a list of all the dictionaries read 
</p>
</dd></dl>

<hr>
<a name="Restraints-Interface"></a>
<div class="header">
<p>
Next: <a href="#Atom-Information-functions" accesskey="n" rel="next">Atom Information functions</a>, Previous: <a href="#Dictionary-Functions" accesskey="p" rel="previous">Dictionary Functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Restraints-Interface-1"></a>
<h3 class="section">12.7 Restraints Interface</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dmonomer_002drestraints" accesskey="1">set-monomer-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dmonomer_002drestraints"></a>
<div class="header">
<p>
Up: <a href="#Restraints-Interface" accesskey="u" rel="up">Restraints Interface</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmonomer_002drestraints-1"></a>
<h4 class="subsection">12.7.1 set-monomer-restraints</h4>
<dl>
<dt><a name="index-set_002dmonomer_002drestraints"></a>function: <strong>set-monomer-restraints</strong> <em>monomer_type restraints</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>monomer_type</em> is a string
</li><li> <em>restraints</em> is a SCM
</li></ul>

<p>set the monomer restraints of the given monomer_type 
</p>
<p>Returns: scheme false or true for success or failure to set the restrains for monomer_type 
</p>
</dd></dl>

<hr>
<a name="Atom-Information-functions"></a>
<div class="header">
<p>
Next: <a href="#Using-S_002dexpression-molecules" accesskey="n" rel="next">Using S-expression molecules</a>, Previous: <a href="#Restraints-Interface" accesskey="p" rel="previous">Restraints Interface</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Atom-Information-functions-1"></a>
<h3 class="section">12.8 Atom Information functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#resname_002dfrom_002dserial_002dnumber" accesskey="1">resname-from-serial-number</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residue_002dname" accesskey="2">residue-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#serial_002dnumber_002dfrom_002dresidue_002dspecs" accesskey="3">serial-number-from-residue-specs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residue_002dinfo" accesskey="4">residue-info</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#chain_002dfragments_002dscm" accesskey="5">chain-fragments-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dmolecule" accesskey="6">add-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clear_002dand_002dupdate_002dmolecule" accesskey="7">clear-and-update-molecule</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#active_002dresidue" accesskey="8">active-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#closest_002datom_002dsimple_002dscm" accesskey="9">closest-atom-simple-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#closest_002datom">closest-atom</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#closest_002datom_002draw_002dscm">closest-atom-raw-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residues_002dnear_002dresidue">residues-near-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residues_002dnear_002dresidues_002dscm">residues-near-residues-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residues_002dnear_002dposition_002dscm">residues-near-position-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm">label-closest-atoms-in-neighbour-residues-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#hydrogenate_002dregion">hydrogenate-region</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dhydrogens_002dfrom_002dfile">add-hydrogens-from-file</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#atom_002dinfo_002dstring_002dpy">atom-info-string-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#molecule_002dto_002dpdb_002dstring_002dpy">molecule-to-pdb-string-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residue_002dinfo_002dpy">residue-info-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="resname_002dfrom_002dserial_002dnumber"></a>
<div class="header">
<p>
Next: <a href="#residue_002dname" accesskey="n" rel="next">residue-name</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="resname_002dfrom_002dserial_002dnumber-1"></a>
<h4 class="subsection">12.8.1 resname-from-serial-number</h4>
<dl>
<dt><a name="index-resname_002dfrom_002dserial_002dnumber"></a>function: <strong>resname-from-serial-number</strong> <em>imol chain_id serial_num</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>serial_num</em> is an integer number
</li></ul>

<p>return the rename from a residue serial number 
</p>
<p>Returns: blank (&quot;&quot;) on failure. 
</p>
</dd></dl>

<hr>
<a name="residue_002dname"></a>
<div class="header">
<p>
Next: <a href="#serial_002dnumber_002dfrom_002dresidue_002dspecs" accesskey="n" rel="next">serial-number-from-residue-specs</a>, Previous: <a href="#resname_002dfrom_002dserial_002dnumber" accesskey="p" rel="previous">resname-from-serial-number</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residue_002dname-1"></a>
<h4 class="subsection">12.8.2 residue-name</h4>
<dl>
<dt><a name="index-residue_002dname"></a>function: <strong>residue-name</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a const std::string &amp;
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a const std::string &amp;
</li></ul>

<p>return the residue name of the specified residue 
</p>
</dd></dl>

<hr>
<a name="serial_002dnumber_002dfrom_002dresidue_002dspecs"></a>
<div class="header">
<p>
Next: <a href="#residue_002dinfo" accesskey="n" rel="next">residue-info</a>, Previous: <a href="#residue_002dname" accesskey="p" rel="previous">residue-name</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="serial_002dnumber_002dfrom_002dresidue_002dspecs-1"></a>
<h4 class="subsection">12.8.3 serial-number-from-residue-specs</h4>
<dl>
<dt><a name="index-serial_002dnumber_002dfrom_002dresidue_002dspecs"></a>function: <strong>serial-number-from-residue-specs</strong> <em>imol chain_id res_no ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a const std::string &amp;
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a const std::string &amp;
</li></ul>

<p>return the serial number of the specified residue 
</p>
<p>Returns: -1 on failure to find the residue 
</p>
</dd></dl>

<hr>
<a name="residue_002dinfo"></a>
<div class="header">
<p>
Next: <a href="#chain_002dfragments_002dscm" accesskey="n" rel="next">chain-fragments-scm</a>, Previous: <a href="#serial_002dnumber_002dfrom_002dresidue_002dspecs" accesskey="p" rel="previous">serial-number-from-residue-specs</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residue_002dinfo-1"></a>
<h4 class="subsection">12.8.4 residue-info</h4>
<dl>
<dt><a name="index-residue_002dinfo"></a>function: <strong>residue-info</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>Return a list of atom info for each atom in the specified residue. 
</p>
<p>output is like this: (list (list (list atom-name alt-conf) (list occ temp-fact element) (list x y z)))
</p>
<p>occ can be a single number or a list of seven numbers of which the first is the isotropic B. 
</p>
</dd></dl>

<hr>
<a name="chain_002dfragments_002dscm"></a>
<div class="header">
<p>
Next: <a href="#add_002dmolecule" accesskey="n" rel="next">add-molecule</a>, Previous: <a href="#residue_002dinfo" accesskey="p" rel="previous">residue-info</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="chain_002dfragments_002dscm-1"></a>
<h4 class="subsection">12.8.5 chain-fragments-scm</h4>
<dl>
<dt><a name="index-chain_002dfragments_002dscm"></a>function: <strong>chain-fragments-scm</strong> <em>imol screen_output_also</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>screen_output_also</em> is an integer number
</li></ul>

<p>chain fragments 
</p>
</dd></dl>

<hr>
<a name="add_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#clear_002dand_002dupdate_002dmolecule" accesskey="n" rel="next">clear-and-update-molecule</a>, Previous: <a href="#chain_002dfragments_002dscm" accesskey="p" rel="previous">chain-fragments-scm</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dmolecule-1"></a>
<h4 class="subsection">12.8.6 add-molecule</h4>
<dl>
<dt><a name="index-add_002dmolecule"></a>function: <strong>add-molecule</strong> <em>molecule_expression name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>molecule_expression</em> is a SCM
</li><li> <em>name</em> is a string
</li></ul>

<p>generate a molecule from an s-expression 
</p>
<p>return a molecule number, -1 on error 
</p>
</dd></dl>

<hr>
<a name="clear_002dand_002dupdate_002dmolecule"></a>
<div class="header">
<p>
Next: <a href="#active_002dresidue" accesskey="n" rel="next">active-residue</a>, Previous: <a href="#add_002dmolecule" accesskey="p" rel="previous">add-molecule</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002dand_002dupdate_002dmolecule-1"></a>
<h4 class="subsection">12.8.7 clear-and-update-molecule</h4>
<dl>
<dt><a name="index-clear_002dand_002dupdate_002dmolecule"></a>function: <strong>clear-and-update-molecule</strong> <em>molecule_number molecule_expression</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>molecule_number</em> is an integer number
</li><li> <em>molecule_expression</em> is a SCM
</li></ul>

<p>update a molecule from a s-expression 
</p>
<p>And going the other way, given an s-expression, update molecule_number by the given molecule. Clear what&rsquo;s currently there first though. 
</p>
</dd></dl>

<hr>
<a name="active_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#closest_002datom_002dsimple_002dscm" accesskey="n" rel="next">closest-atom-simple-scm</a>, Previous: <a href="#clear_002dand_002dupdate_002dmolecule" accesskey="p" rel="previous">clear-and-update-molecule</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="active_002dresidue-1"></a>
<h4 class="subsection">12.8.8 active-residue</h4>
<dl>
<dt><a name="index-active_002dresidue"></a>function: <strong>active-residue</strong></dt>
<dd>
<p>return specs of the atom close to screen centre 
</p>
<p>Return a list of (list imol chain-id resno ins-code atom-name alt-conf) for atom that is closest to the screen centre in any displayed molecule. If there are multiple models with the same coordinates at the screen centre, return the attributes of the atom in the highest number molecule number.
</p>
<p>return scheme false if no active residue 
</p>
</dd></dl>

<hr>
<a name="closest_002datom_002dsimple_002dscm"></a>
<div class="header">
<p>
Next: <a href="#closest_002datom" accesskey="n" rel="next">closest-atom</a>, Previous: <a href="#active_002dresidue" accesskey="p" rel="previous">active-residue</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="closest_002datom_002dsimple_002dscm-1"></a>
<h4 class="subsection">12.8.9 closest-atom-simple-scm</h4>
<dl>
<dt><a name="index-closest_002datom_002dsimple_002dscm"></a>function: <strong>closest-atom-simple-scm</strong></dt>
<dd>
<p>return the specs of the closest displayed atom 
</p>
<p>Return a list of (list imol chain-id resno ins-code atom-name alt-conf (list x y z)) for atom that is closest to the screen centre in the given molecule (unlike active-residue, potential CA substition is not performed). If there is no atom, or if imol is not a valid model molecule, return scheme false. 
</p>
</dd></dl>

<hr>
<a name="closest_002datom"></a>
<div class="header">
<p>
Next: <a href="#closest_002datom_002draw_002dscm" accesskey="n" rel="next">closest-atom-raw-scm</a>, Previous: <a href="#closest_002datom_002dsimple_002dscm" accesskey="p" rel="previous">closest-atom-simple-scm</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="closest_002datom-1"></a>
<h4 class="subsection">12.8.10 closest-atom</h4>
<dl>
<dt><a name="index-closest_002datom"></a>function: <strong>closest-atom</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the specs of the closest atom in imolth molecule 
</p>
<p>Return a list of (list imol chain-id resno ins-code atom-name alt-conf (list x y z)) for atom that is closest to the screen centre in the given molecule (unlike active-residue, no account is taken of the displayed state of the molecule). If there is no atom, or if imol is not a valid model molecule, return scheme false. 
</p>
</dd></dl>

<hr>
<a name="closest_002datom_002draw_002dscm"></a>
<div class="header">
<p>
Next: <a href="#residues_002dnear_002dresidue" accesskey="n" rel="next">residues-near-residue</a>, Previous: <a href="#closest_002datom" accesskey="p" rel="previous">closest-atom</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="closest_002datom_002draw_002dscm-1"></a>
<h4 class="subsection">12.8.11 closest-atom-raw-scm</h4>
<dl>
<dt><a name="index-closest_002datom_002draw_002dscm"></a>function: <strong>closest-atom-raw-scm</strong></dt>
<dd>
<p>return the specs of the closest atom to the centre of the screen 
</p>
<p>Return a list of (list imol chain-id resno ins-code atom-name alt-conf (list x y z)) for atom that is closest to the screen for displayed molecules. If there is no atom, return scheme false. Don&rsquo;t choose the CA of the residue if there is a CA in the residue of the closest atom. 201602015-PE: I add this now, but I have a feeling that I&rsquo;ve done this before. 
</p>
</dd></dl>

<hr>
<a name="residues_002dnear_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#residues_002dnear_002dresidues_002dscm" accesskey="n" rel="next">residues-near-residues-scm</a>, Previous: <a href="#closest_002datom_002draw_002dscm" accesskey="p" rel="previous">closest-atom-raw-scm</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residues_002dnear_002dresidue-1"></a>
<h4 class="subsection">12.8.12 residues-near-residue</h4>
<dl>
<dt><a name="index-residues_002dnear_002dresidue"></a>function: <strong>residues-near-residue</strong> <em>imol residue_in_scm radius</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_in_scm</em> is a SCM
</li><li> <em>radius</em> is a number
</li></ul>

<p>return residues near residue 
</p>
<p>Return residue specs for residues that have atoms that are closer than radius Angstroems to any atom in the residue specified by res_in. 
</p>
</dd></dl>

<hr>
<a name="residues_002dnear_002dresidues_002dscm"></a>
<div class="header">
<p>
Next: <a href="#residues_002dnear_002dposition_002dscm" accesskey="n" rel="next">residues-near-position-scm</a>, Previous: <a href="#residues_002dnear_002dresidue" accesskey="p" rel="previous">residues-near-residue</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residues_002dnear_002dresidues_002dscm-1"></a>
<h4 class="subsection">12.8.13 residues-near-residues-scm</h4>
<dl>
<dt><a name="index-residues_002dnear_002dresidues_002dscm"></a>function: <strong>residues-near-residues-scm</strong> <em>imol residues_in radius</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residues_in</em> is a SCM
</li><li> <em>radius</em> is a number
</li></ul>

<p>return residues near the given residues 
</p>
<p>Return residue specs for residues that have atoms that are closer than radius Angstroems to any atom in the residue specified by res_in. 
</p>
</dd></dl>

<hr>
<a name="residues_002dnear_002dposition_002dscm"></a>
<div class="header">
<p>
Next: <a href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm" accesskey="n" rel="next">label-closest-atoms-in-neighbour-residues-scm</a>, Previous: <a href="#residues_002dnear_002dresidues_002dscm" accesskey="p" rel="previous">residues-near-residues-scm</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residues_002dnear_002dposition_002dscm-1"></a>
<h4 class="subsection">12.8.14 residues-near-position-scm</h4>
<dl>
<dt><a name="index-residues_002dnear_002dposition_002dscm"></a>function: <strong>residues-near-position-scm</strong> <em>imol pos radius</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>pos</em> is a SCM
</li><li> <em>radius</em> is a number
</li></ul>

<p>residues near residue 
</p>

<p>Return a list of pairs of (imol, residue_spec). pos is a list of 3 numbers. (get imol from active-atom) 
</p>
<p>Returns: residues within radius of pos (x,y,z) position
</p>
</dd></dl>

<hr>
<a name="label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm"></a>
<div class="header">
<p>
Next: <a href="#hydrogenate_002dregion" accesskey="n" rel="next">hydrogenate-region</a>, Previous: <a href="#residues_002dnear_002dposition_002dscm" accesskey="p" rel="previous">residues-near-position-scm</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm-1"></a>
<h4 class="subsection">12.8.15 label-closest-atoms-in-neighbour-residues-scm</h4>
<dl>
<dt><a name="index-label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm"></a>function: <strong>label-closest-atoms-in-neighbour-residues-scm</strong> <em>imol residue_spec_scm radius</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec_scm</em> is a SCM
</li><li> <em>radius</em> is a number
</li></ul>

<p>label the closest atoms in the residues that neighbour residue_spec 
</p>
</dd></dl>

<hr>
<a name="hydrogenate_002dregion"></a>
<div class="header">
<p>
Next: <a href="#add_002dhydrogens_002dfrom_002dfile" accesskey="n" rel="next">add-hydrogens-from-file</a>, Previous: <a href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm" accesskey="p" rel="previous">label-closest-atoms-in-neighbour-residues-scm</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="hydrogenate_002dregion-1"></a>
<h4 class="subsection">12.8.16 hydrogenate-region</h4>
<dl>
<dt><a name="index-hydrogenate_002dregion"></a>function: <strong>hydrogenate-region</strong> <em>radius</em></dt>
<dd>
<p>Where <em>radius</em> is a number
</p>
<p>find the active residue, find the near residues (within radius) create a new molecule, run reduce on that, import hydrogens from the result and apply them to the molecule of the active residue. 
</p>
</dd></dl>

<hr>
<a name="add_002dhydrogens_002dfrom_002dfile"></a>
<div class="header">
<p>
Next: <a href="#atom_002dinfo_002dstring_002dpy" accesskey="n" rel="next">atom-info-string-py</a>, Previous: <a href="#hydrogenate_002dregion" accesskey="p" rel="previous">hydrogenate-region</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dhydrogens_002dfrom_002dfile-1"></a>
<h4 class="subsection">12.8.17 add-hydrogens-from-file</h4>
<dl>
<dt><a name="index-add_002dhydrogens_002dfrom_002dfile"></a>function: <strong>add-hydrogens-from-file</strong> <em>imol pdb_with_Hs_file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>pdb_with_Hs_file_name</em> is a std::string
</li></ul>

<p>Add hydrogens to imol from the given pdb file. 
</p>
</dd></dl>

<hr>
<a name="atom_002dinfo_002dstring_002dpy"></a>
<div class="header">
<p>
Next: <a href="#molecule_002dto_002dpdb_002dstring_002dpy" accesskey="n" rel="next">molecule-to-pdb-string-py</a>, Previous: <a href="#add_002dhydrogens_002dfrom_002dfile" accesskey="p" rel="previous">add-hydrogens-from-file</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="atom_002dinfo_002dstring_002dpy-1"></a>
<h4 class="subsection">12.8.18 atom-info-string-py</h4>
<dl>
<dt><a name="index-atom_002dinfo_002dstring_002dpy"></a>function: <strong>atom-info-string-py</strong> <em>imol chain_id resno ins_code atname altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>atname</em> is a string
</li><li> <em>altconf</em> is a string
</li></ul>

<p>output atom info in a python list for use in scripting: 
</p>
<p>in this format [occ, temp_factor, element, x, y, z]. Return empty list if atom not found. 
</p>
</dd></dl>

<hr>
<a name="molecule_002dto_002dpdb_002dstring_002dpy"></a>
<div class="header">
<p>
Next: <a href="#residue_002dinfo_002dpy" accesskey="n" rel="next">residue-info-py</a>, Previous: <a href="#atom_002dinfo_002dstring_002dpy" accesskey="p" rel="previous">atom-info-string-py</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="molecule_002dto_002dpdb_002dstring_002dpy-1"></a>
<h4 class="subsection">12.8.19 molecule-to-pdb-string-py</h4>
<dl>
<dt><a name="index-molecule_002dto_002dpdb_002dstring_002dpy"></a>function: <strong>molecule-to-pdb-string-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>Return the molecule as a PDB string. 
</p>
</dd></dl>

<hr>
<a name="residue_002dinfo_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#molecule_002dto_002dpdb_002dstring_002dpy" accesskey="p" rel="previous">molecule-to-pdb-string-py</a>, Up: <a href="#Atom-Information-functions" accesskey="u" rel="up">Atom Information functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residue_002dinfo_002dpy-1"></a>
<h4 class="subsection">12.8.20 residue-info-py</h4>
<dl>
<dt><a name="index-residue_002dinfo_002dpy"></a>function: <strong>residue-info-py</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li></ul>

<p>Return a list of atom info for each atom in the specified residue: 
</p>
<p>output is like this: [ [[atom-name,alt-conf] [occ,temp_fact,element] [x,y,z]]] 
</p>
</dd></dl>

<hr>
<a name="Using-S_002dexpression-molecules"></a>
<div class="header">
<p>
Next: <a href="#Refinement-with-specs" accesskey="n" rel="next">Refinement with specs</a>, Previous: <a href="#Atom-Information-functions" accesskey="p" rel="previous">Atom Information functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Using-S_002dexpression-molecules-1"></a>
<h3 class="section">12.9 Using S-expression molecules</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#active_002dresidue_002dpy" accesskey="1">active-residue-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#closest_002datom_002dsimple_002dpy" accesskey="2">closest-atom-simple-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#closest_002datom_002dpy" accesskey="3">closest-atom-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#closest_002datom_002draw_002dpy" accesskey="4">closest-atom-raw-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#residues_002dnear_002dposition_002dpy" accesskey="5">residues-near-position-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy" accesskey="6">label-closest-atoms-in-neighbour-residues-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dbonds_002drepresentation" accesskey="7">get-bonds-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002ddictionary_002dradii" accesskey="8">get-dictionary-radii</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002denvironment_002ddistances_002drepresentation_002dpy" accesskey="9">get-environment-distances-representation-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dintermediate_002datoms_002dbonds_002drepresentation">get-intermediate-atoms-bonds-representation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate">get-continue-updating-refinement-atoms-state</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="active_002dresidue_002dpy"></a>
<div class="header">
<p>
Next: <a href="#closest_002datom_002dsimple_002dpy" accesskey="n" rel="next">closest-atom-simple-py</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="active_002dresidue_002dpy-1"></a>
<h4 class="subsection">12.9.1 active-residue-py</h4>
<dl>
<dt><a name="index-active_002dresidue_002dpy"></a>function: <strong>active-residue-py</strong></dt>
<dd>
<p>Return a list of [imol, chain-id, resno, ins-code, atom-name, alt-conf] for atom that is closest to the screen centre. If there are multiple models with the same coordinates at the screen centre, return the attributes of the atom in the highest number molecule number. 
</p>
<p>return False if no active residue 
</p>
</dd></dl>

<hr>
<a name="closest_002datom_002dsimple_002dpy"></a>
<div class="header">
<p>
Next: <a href="#closest_002datom_002dpy" accesskey="n" rel="next">closest-atom-py</a>, Previous: <a href="#active_002dresidue_002dpy" accesskey="p" rel="previous">active-residue-py</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="closest_002datom_002dsimple_002dpy-1"></a>
<h4 class="subsection">12.9.2 closest-atom-simple-py</h4>
<dl>
<dt><a name="index-closest_002datom_002dsimple_002dpy"></a>function: <strong>closest-atom-simple-py</strong></dt>
<dd>
<p>return the spec of the closest displayed atom 
</p>
<p>Return a list of [imol, chain-id, resno, ins-code, atom-name, alt-conf, [x, y, z]] for atom that is closest to the screen centre in the given molecule (unlike active-residue, potential CA substition is not performed). If there is no atom, or if imol is not a valid model molecule, return False. 
</p>
</dd></dl>

<hr>
<a name="closest_002datom_002dpy"></a>
<div class="header">
<p>
Next: <a href="#closest_002datom_002draw_002dpy" accesskey="n" rel="next">closest-atom-raw-py</a>, Previous: <a href="#closest_002datom_002dsimple_002dpy" accesskey="p" rel="previous">closest-atom-simple-py</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="closest_002datom_002dpy-1"></a>
<h4 class="subsection">12.9.3 closest-atom-py</h4>
<dl>
<dt><a name="index-closest_002datom_002dpy"></a>function: <strong>closest-atom-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return closest atom in imolth molecule 
</p>
<p>Return a list of [imol, chain-id, resno, ins-code, atom-name, alt-conf, [x, y, z]] for atom that is closest to the screen centre in the given molecule (unlike active-residue, no account is taken of the displayed state of the molecule). If there is no atom, or if imol is not a valid model molecule, return False. 
</p>
</dd></dl>

<hr>
<a name="closest_002datom_002draw_002dpy"></a>
<div class="header">
<p>
Next: <a href="#residues_002dnear_002dposition_002dpy" accesskey="n" rel="next">residues-near-position-py</a>, Previous: <a href="#closest_002datom_002dpy" accesskey="p" rel="previous">closest-atom-py</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="closest_002datom_002draw_002dpy-1"></a>
<h4 class="subsection">12.9.4 closest-atom-raw-py</h4>
<dl>
<dt><a name="index-closest_002datom_002draw_002dpy"></a>function: <strong>closest-atom-raw-py</strong></dt>
<dd>
<p>return the specs of the closest atom to the centre of the screen 
</p>
<p>Return a list of (list imol chain-id resno ins-code atom-name alt-conf (list x y z)) for atom that is closest to the screen for displayed molecules. If there is no atom, return scheme false. Don&rsquo;t choose the CA of the residue if there is a CA in the residue of the closest atom 
</p>
</dd></dl>

<hr>
<a name="residues_002dnear_002dposition_002dpy"></a>
<div class="header">
<p>
Next: <a href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy" accesskey="n" rel="next">label-closest-atoms-in-neighbour-residues-py</a>, Previous: <a href="#closest_002datom_002draw_002dpy" accesskey="p" rel="previous">closest-atom-raw-py</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="residues_002dnear_002dposition_002dpy-1"></a>
<h4 class="subsection">12.9.5 residues-near-position-py</h4>
<dl>
<dt><a name="index-residues_002dnear_002dposition_002dpy"></a>function: <strong>residues-near-position-py</strong> <em>imol pos_in radius</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>pos_in</em> is a PyObject
</li><li> <em>radius</em> is a number
</li></ul>

<p>Return residue specs for residues that have atoms that are closer than radius Angstroems to the given position. 
</p>
</dd></dl>

<hr>
<a name="label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy"></a>
<div class="header">
<p>
Next: <a href="#get_002dbonds_002drepresentation" accesskey="n" rel="next">get-bonds-representation</a>, Previous: <a href="#residues_002dnear_002dposition_002dpy" accesskey="p" rel="previous">residues-near-position-py</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy-1"></a>
<h4 class="subsection">12.9.6 label-closest-atoms-in-neighbour-residues-py</h4>
<dl>
<dt><a name="index-label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy"></a>function: <strong>label-closest-atoms-in-neighbour-residues-py</strong> <em>imol residue_spec_py radius</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec_py</em> is a PyObject
</li><li> <em>radius</em> is a number
</li></ul>

<p>label the closest atoms in the residues that neighbour residue_spec 
</p>
</dd></dl>

<hr>
<a name="get_002dbonds_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#get_002ddictionary_002dradii" accesskey="n" rel="next">get-dictionary-radii</a>, Previous: <a href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy" accesskey="p" rel="previous">label-closest-atoms-in-neighbour-residues-py</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dbonds_002drepresentation-1"></a>
<h4 class="subsection">12.9.7 get-bonds-representation</h4>
<dl>
<dt><a name="index-get_002dbonds_002drepresentation"></a>function: <strong>get-bonds-representation</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return a Python object for the bonds 
</p>
</dd></dl>

<hr>
<a name="get_002ddictionary_002dradii"></a>
<div class="header">
<p>
Next: <a href="#get_002denvironment_002ddistances_002drepresentation_002dpy" accesskey="n" rel="next">get-environment-distances-representation-py</a>, Previous: <a href="#get_002dbonds_002drepresentation" accesskey="p" rel="previous">get-bonds-representation</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002ddictionary_002dradii-1"></a>
<h4 class="subsection">12.9.8 get-dictionary-radii</h4>
<dl>
<dt><a name="index-get_002ddictionary_002dradii"></a>function: <strong>get-dictionary-radii</strong></dt>
<dd>
<p>return a Python object for the radii of the atoms in the dictionary 
</p>
</dd></dl>

<hr>
<a name="get_002denvironment_002ddistances_002drepresentation_002dpy"></a>
<div class="header">
<p>
Next: <a href="#get_002dintermediate_002datoms_002dbonds_002drepresentation" accesskey="n" rel="next">get-intermediate-atoms-bonds-representation</a>, Previous: <a href="#get_002ddictionary_002dradii" accesskey="p" rel="previous">get-dictionary-radii</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002denvironment_002ddistances_002drepresentation_002dpy-1"></a>
<h4 class="subsection">12.9.9 get-environment-distances-representation-py</h4>
<dl>
<dt><a name="index-get_002denvironment_002ddistances_002drepresentation_002dpy"></a>function: <strong>get-environment-distances-representation-py</strong> <em>imol residue_spec_py</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec_py</em> is a PyObject
</li></ul>

<p>return a Python object for the representation of bump and hydrogen bonds of 
</p>
</dd></dl>

<hr>
<a name="get_002dintermediate_002datoms_002dbonds_002drepresentation"></a>
<div class="header">
<p>
Next: <a href="#get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate" accesskey="n" rel="next">get-continue-updating-refinement-atoms-state</a>, Previous: <a href="#get_002denvironment_002ddistances_002drepresentation_002dpy" accesskey="p" rel="previous">get-environment-distances-representation-py</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dintermediate_002datoms_002dbonds_002drepresentation-1"></a>
<h4 class="subsection">12.9.10 get-intermediate-atoms-bonds-representation</h4>
<dl>
<dt><a name="index-get_002dintermediate_002datoms_002dbonds_002drepresentation"></a>function: <strong>get-intermediate-atoms-bonds-representation</strong></dt>
<dd>
<p>return a Python object for the intermediate atoms bonds 
</p>
</dd></dl>

<hr>
<a name="get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate"></a>
<div class="header">
<p>
Previous: <a href="#get_002dintermediate_002datoms_002dbonds_002drepresentation" accesskey="p" rel="previous">get-intermediate-atoms-bonds-representation</a>, Up: <a href="#Using-S_002dexpression-molecules" accesskey="u" rel="up">Using S-expression molecules</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate-1"></a>
<h4 class="subsection">12.9.11 get-continue-updating-refinement-atoms-state</h4>
<dl>
<dt><a name="index-get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate"></a>function: <strong>get-continue-updating-refinement-atoms-state</strong></dt>
<dd>
<p>return the continue-updating-refinement-atoms state 
</p>
</dd></dl>

<hr>
<a name="Refinement-with-specs"></a>
<div class="header">
<p>
Next: <a href="#Water-Chain-Functions" accesskey="n" rel="next">Water Chain Functions</a>, Previous: <a href="#Using-S_002dexpression-molecules" accesskey="p" rel="previous">Using S-expression molecules</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Refinement-with-specs-1"></a>
<h3 class="section">12.10 Refinement with specs</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#all_002dresidues_002dwith_002dserial_002dnumbers_002dscm" accesskey="1">all-residues-with-serial-numbers-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#regularize_002dresidues" accesskey="2">regularize-residues</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mtz_002dfile_002dname" accesskey="3">mtz-file-name</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm" accesskey="4">refine-zone-with-full-residue-spec-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="all_002dresidues_002dwith_002dserial_002dnumbers_002dscm"></a>
<div class="header">
<p>
Next: <a href="#regularize_002dresidues" accesskey="n" rel="next">regularize-residues</a>, Up: <a href="#Refinement-with-specs" accesskey="u" rel="up">Refinement with specs</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="all_002dresidues_002dwith_002dserial_002dnumbers_002dscm-1"></a>
<h4 class="subsection">12.10.1 all-residues-with-serial-numbers-scm</h4>
<dl>
<dt><a name="index-all_002dresidues_002dwith_002dserial_002dnumbers_002dscm"></a>function: <strong>all-residues-with-serial-numbers-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>a utility to return the specs of all the residues, each spec prefixed by the serial number 
</p>
</dd></dl>

<hr>
<a name="regularize_002dresidues"></a>
<div class="header">
<p>
Next: <a href="#mtz_002dfile_002dname" accesskey="n" rel="next">mtz-file-name</a>, Previous: <a href="#all_002dresidues_002dwith_002dserial_002dnumbers_002dscm" accesskey="p" rel="previous">all-residues-with-serial-numbers-scm</a>, Up: <a href="#Refinement-with-specs" accesskey="u" rel="up">Refinement with specs</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="regularize_002dresidues-1"></a>
<h4 class="subsection">12.10.2 regularize-residues</h4>
<dl>
<dt><a name="index-regularize_002dresidues"></a>function: <strong>regularize-residues</strong> <em>imol residues</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residues</em> is a const std::vector&lt; coot::residue_spec_t &gt; &amp;
</li></ul>

<p>regularize the given residues 
</p>
</dd></dl>

<hr>
<a name="mtz_002dfile_002dname"></a>
<div class="header">
<p>
Next: <a href="#refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm" accesskey="n" rel="next">refine-zone-with-full-residue-spec-scm</a>, Previous: <a href="#regularize_002dresidues" accesskey="p" rel="previous">regularize-residues</a>, Up: <a href="#Refinement-with-specs" accesskey="u" rel="up">Refinement with specs</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mtz_002dfile_002dname-1"></a>
<h4 class="subsection">12.10.3 mtz-file-name</h4>
<dl>
<dt><a name="index-mtz_002dfile_002dname"></a>function: <strong>mtz-file-name</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>presumes that imol_Refinement_Map has been set 
</p>
</dd></dl>

<hr>
<a name="refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm"></a>
<div class="header">
<p>
Previous: <a href="#mtz_002dfile_002dname" accesskey="p" rel="previous">mtz-file-name</a>, Up: <a href="#Refinement-with-specs" accesskey="u" rel="up">Refinement with specs</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm-1"></a>
<h4 class="subsection">12.10.4 refine-zone-with-full-residue-spec-scm</h4>
<dl>
<dt><a name="index-refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm"></a>function: <strong>refine-zone-with-full-residue-spec-scm</strong> <em>imol chain_id resno1 inscode_1 resno2 inscode_2 altconf</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno1</em> is an integer number
</li><li> <em>inscode_1</em> is a string
</li><li> <em>resno2</em> is an integer number
</li><li> <em>inscode_2</em> is a string
</li><li> <em>altconf</em> is a string
</li></ul>

<p>Refine the given residue range. 
</p>
</dd></dl>

<hr>
<a name="Water-Chain-Functions"></a>
<div class="header">
<p>
Next: <a href="#Glyco-Tools" accesskey="n" rel="next">Glyco Tools</a>, Previous: <a href="#Refinement-with-specs" accesskey="p" rel="previous">Refinement with specs</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Water-Chain-Functions-1"></a>
<h3 class="section">12.11 Water Chain Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#water_002dchain_002dfrom_002dshelx_002dins_002dscm" accesskey="1">water-chain-from-shelx-ins-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#water_002dchain_002dscm" accesskey="2">water-chain-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#water_002dchain_002dpy" accesskey="3">water-chain-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="water_002dchain_002dfrom_002dshelx_002dins_002dscm"></a>
<div class="header">
<p>
Next: <a href="#water_002dchain_002dscm" accesskey="n" rel="next">water-chain-scm</a>, Up: <a href="#Water-Chain-Functions" accesskey="u" rel="up">Water Chain Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="water_002dchain_002dfrom_002dshelx_002dins_002dscm-1"></a>
<h4 class="subsection">12.11.1 water-chain-from-shelx-ins-scm</h4>
<dl>
<dt><a name="index-water_002dchain_002dfrom_002dshelx_002dins_002dscm"></a>function: <strong>water-chain-from-shelx-ins-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the chain id of the water chain from a shelx molecule. Raw interface 
</p>
<p>Returns: scheme false if no chain or bad imol 
</p>
</dd></dl>

<hr>
<a name="water_002dchain_002dscm"></a>
<div class="header">
<p>
Next: <a href="#water_002dchain_002dpy" accesskey="n" rel="next">water-chain-py</a>, Previous: <a href="#water_002dchain_002dfrom_002dshelx_002dins_002dscm" accesskey="p" rel="previous">water-chain-from-shelx-ins-scm</a>, Up: <a href="#Water-Chain-Functions" accesskey="u" rel="up">Water Chain Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="water_002dchain_002dscm-1"></a>
<h4 class="subsection">12.11.2 water-chain-scm</h4>
<dl>
<dt><a name="index-water_002dchain_002dscm"></a>function: <strong>water-chain-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the chain id of the water chain. Raw interface 
</p>
</dd></dl>

<hr>
<a name="water_002dchain_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#water_002dchain_002dscm" accesskey="p" rel="previous">water-chain-scm</a>, Up: <a href="#Water-Chain-Functions" accesskey="u" rel="up">Water Chain Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="water_002dchain_002dpy-1"></a>
<h4 class="subsection">12.11.3 water-chain-py</h4>
<dl>
<dt><a name="index-water_002dchain_002dpy"></a>function: <strong>water-chain-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return the chain id of the water chain. Raw interface 
</p>
</dd></dl>

<hr>
<a name="Glyco-Tools"></a>
<div class="header">
<p>
Next: <a href="#Spin-Search-Functions" accesskey="n" rel="next">Spin Search Functions</a>, Previous: <a href="#Water-Chain-Functions" accesskey="p" rel="previous">Water Chain Functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Glyco-Tools-1"></a>
<h3 class="section">12.12 Glyco Tools</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#print_002dglyco_002dtree" accesskey="1">print-glyco-tree</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="print_002dglyco_002dtree"></a>
<div class="header">
<p>
Up: <a href="#Glyco-Tools" accesskey="u" rel="up">Glyco Tools</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="print_002dglyco_002dtree-1"></a>
<h4 class="subsection">12.12.1 print-glyco-tree</h4>
<dl>
<dt><a name="index-print_002dglyco_002dtree"></a>function: <strong>print-glyco-tree</strong> <em>imol chain_id resno ins_code</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a const std::string &amp;
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a const std::string &amp;
</li></ul>

<p>print the glycosylation tree that contains the specified residue 
</p>
</dd></dl>

<hr>
<a name="Spin-Search-Functions"></a>
<div class="header">
<p>
Next: <a href="#Rotamer-Scoring" accesskey="n" rel="next">Rotamer Scoring</a>, Previous: <a href="#Glyco-Tools" accesskey="p" rel="previous">Glyco Tools</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Spin-Search-Functions-1"></a>
<h3 class="section">12.13 Spin Search Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#spin_002dsearch" accesskey="1">spin-search</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#spin_002dN_002dscm" accesskey="2">spin-N-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#CG_002dspin_002dsearch_002dscm" accesskey="3">CG-spin-search-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#spin_002dsearch_002dpy" accesskey="4">spin-search-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#spin_002dN_002dpy" accesskey="5">spin-N-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#CG_002dspin_002dsearch_002dpy" accesskey="6">CG-spin-search-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="spin_002dsearch"></a>
<div class="header">
<p>
Next: <a href="#spin_002dN_002dscm" accesskey="n" rel="next">spin-N-scm</a>, Up: <a href="#Spin-Search-Functions" accesskey="u" rel="up">Spin Search Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="spin_002dsearch-1"></a>
<h4 class="subsection">12.13.1 spin-search</h4>
<dl>
<dt><a name="index-spin_002dsearch"></a>function: <strong>spin-search</strong> <em>imol_map imol chain_id resno ins_code direction_atoms_list moving_atoms_list</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>direction_atoms_list</em> is a SCM
</li><li> <em>moving_atoms_list</em> is a SCM
</li></ul>

<p>for the given residue, spin the atoms in moving_atom_list around the bond defined by direction_atoms_list looking for the best fit to density of imol_map map of the first atom in moving_atom_list. Works (only) with atoms in altconf &quot;&quot; 
</p>
</dd></dl>

<hr>
<a name="spin_002dN_002dscm"></a>
<div class="header">
<p>
Next: <a href="#CG_002dspin_002dsearch_002dscm" accesskey="n" rel="next">CG-spin-search-scm</a>, Previous: <a href="#spin_002dsearch" accesskey="p" rel="previous">spin-search</a>, Up: <a href="#Spin-Search-Functions" accesskey="u" rel="up">Spin Search Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="spin_002dN_002dscm-1"></a>
<h4 class="subsection">12.13.2 spin-N-scm</h4>
<dl>
<dt><a name="index-spin_002dN_002dscm"></a>function: <strong>spin-N-scm</strong> <em>imol residue_spec_scm angle</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec_scm</em> is a SCM
</li><li> <em>angle</em> is a number
</li></ul>

<p>Spin N and CB (and the rest of the side chain if extant) 
</p>
<p>Sometime on N-terminal addition, then N ends up pointing the wrong way. The allows us to (more or less) interchange the positions of the CB and the N. angle is in degrees. 
</p>
</dd></dl>

<hr>
<a name="CG_002dspin_002dsearch_002dscm"></a>
<div class="header">
<p>
Next: <a href="#spin_002dsearch_002dpy" accesskey="n" rel="next">spin-search-py</a>, Previous: <a href="#spin_002dN_002dscm" accesskey="p" rel="previous">spin-N-scm</a>, Up: <a href="#Spin-Search-Functions" accesskey="u" rel="up">Spin Search Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="CG_002dspin_002dsearch_002dscm-1"></a>
<h4 class="subsection">12.13.3 CG-spin-search-scm</h4>
<dl>
<dt><a name="index-CG_002dspin_002dsearch_002dscm"></a>function: <strong>CG-spin-search-scm</strong> <em>imol_model imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_model</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>Spin search the density based on possible positions of CG of a side-chain. 
</p>
<p>c.f. EM-Ringer 
</p>
</dd></dl>

<hr>
<a name="spin_002dsearch_002dpy"></a>
<div class="header">
<p>
Next: <a href="#spin_002dN_002dpy" accesskey="n" rel="next">spin-N-py</a>, Previous: <a href="#CG_002dspin_002dsearch_002dscm" accesskey="p" rel="previous">CG-spin-search-scm</a>, Up: <a href="#Spin-Search-Functions" accesskey="u" rel="up">Spin Search Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="spin_002dsearch_002dpy-1"></a>
<h4 class="subsection">12.13.4 spin-search-py</h4>
<dl>
<dt><a name="index-spin_002dsearch_002dpy"></a>function: <strong>spin-search-py</strong> <em>imol_map imol chain_id resno ins_code direction_atoms_list moving_atoms_list</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_map</em> is an integer number
</li><li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>resno</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>direction_atoms_list</em> is a PyObject
</li><li> <em>moving_atoms_list</em> is a PyObject
</li></ul>

<p>for the given residue, spin the atoms in moving_atom_list... 
</p>
<p>around the bond defined by direction_atoms_list looking for the best fit to density of imom_map map of the first atom in moving_atom_list. Works (only) with atoms in altconf &quot;&quot; 
</p>
</dd></dl>

<hr>
<a name="spin_002dN_002dpy"></a>
<div class="header">
<p>
Next: <a href="#CG_002dspin_002dsearch_002dpy" accesskey="n" rel="next">CG-spin-search-py</a>, Previous: <a href="#spin_002dsearch_002dpy" accesskey="p" rel="previous">spin-search-py</a>, Up: <a href="#Spin-Search-Functions" accesskey="u" rel="up">Spin Search Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="spin_002dN_002dpy-1"></a>
<h4 class="subsection">12.13.5 spin-N-py</h4>
<dl>
<dt><a name="index-spin_002dN_002dpy"></a>function: <strong>spin-N-py</strong> <em>imol residue_spec angle</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec</em> is a PyObject
</li><li> <em>angle</em> is a number
</li></ul>

<p>Spin N and CB (and the rest of the side chain if extant) 
</p>
<p>Sometime on N-terminal addition, then N ends up pointing the wrong way. The allows us to (more or less) interchange the positions of the CB and the N. angle is in degrees. 
</p>
</dd></dl>

<hr>
<a name="CG_002dspin_002dsearch_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#spin_002dN_002dpy" accesskey="p" rel="previous">spin-N-py</a>, Up: <a href="#Spin-Search-Functions" accesskey="u" rel="up">Spin Search Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="CG_002dspin_002dsearch_002dpy-1"></a>
<h4 class="subsection">12.13.6 CG-spin-search-py</h4>
<dl>
<dt><a name="index-CG_002dspin_002dsearch_002dpy"></a>function: <strong>CG-spin-search-py</strong> <em>imol_model imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_model</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>Spin search the density based on possible positions of CG of a side-chain. 
</p>
</dd></dl>

<hr>
<a name="Rotamer-Scoring"></a>
<div class="header">
<p>
Next: <a href="#protein_002ddb" accesskey="n" rel="next">protein-db</a>, Previous: <a href="#Spin-Search-Functions" accesskey="p" rel="previous">Spin Search Functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Rotamer-Scoring-1"></a>
<h3 class="section">12.14 Rotamer Scoring</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#score_002drotamers_002dscm" accesskey="1">score-rotamers-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="score_002drotamers_002dscm"></a>
<div class="header">
<p>
Up: <a href="#Rotamer-Scoring" accesskey="u" rel="up">Rotamer Scoring</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="score_002drotamers_002dscm-1"></a>
<h4 class="subsection">12.14.1 score-rotamers-scm</h4>
<dl>
<dt><a name="index-score_002drotamers_002dscm"></a>function: <strong>score-rotamers-scm</strong> <em>imol chain_id res_no ins_code alt_conf imol_map clash_flag lowest_probability</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>alt_conf</em> is a string
</li><li> <em>imol_map</em> is an integer number
</li><li> <em>clash_flag</em> is an integer number
</li><li> <em>lowest_probability</em> is a number
</li></ul>

<p>return the scores of the rotamers for this residue. 
</p>
</dd></dl>

<hr>
<a name="protein_002ddb"></a>
<div class="header">
<p>
Next: <a href="#Coot_0027s-Hole-implementation" accesskey="n" rel="next">Coot's Hole implementation</a>, Previous: <a href="#Rotamer-Scoring" accesskey="p" rel="previous">Rotamer Scoring</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="protein_002ddb-1"></a>
<h3 class="section">12.15 protein-db</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#protein_002ddb_002dloops" accesskey="1">protein-db-loops</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="protein_002ddb_002dloops"></a>
<div class="header">
<p>
Up: <a href="#protein_002ddb" accesskey="u" rel="up">protein-db</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="protein_002ddb_002dloops-1"></a>
<h4 class="subsection">12.15.1 protein-db-loops</h4>
<dl>
<dt><a name="index-protein_002ddb_002dloops"></a>function: <strong>protein-db-loops</strong> <em>imol_coords residue_specs imol_map nfrags preserve_residue_names</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol_coords</em> is an integer number
</li><li> <em>residue_specs</em> is a const std::vector&lt; coot::residue_spec_t &gt; &amp;
</li><li> <em>imol_map</em> is an integer number
</li><li> <em>nfrags</em> is an integer number
</li><li> <em>preserve_residue_names</em> is a bool
</li></ul>

<p>Cowtan&rsquo;s protein_db loops. 
</p>
</dd></dl>

<hr>
<a name="Coot_0027s-Hole-implementation"></a>
<div class="header">
<p>
Next: <a href="#Drag-and-Drop-Functions" accesskey="n" rel="next">Drag and Drop Functions</a>, Previous: <a href="#protein_002ddb" accesskey="p" rel="previous">protein-db</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Coot_0027s-Hole-implementation-1"></a>
<h3 class="section">12.16 Coot&rsquo;s Hole implementation</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#hole" accesskey="1">hole</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#make_002dlink" accesskey="2">make-link</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="hole"></a>
<div class="header">
<p>
Next: <a href="#make_002dlink" accesskey="n" rel="next">make-link</a>, Up: <a href="#Coot_0027s-Hole-implementation" accesskey="u" rel="up">Coot's Hole implementation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="hole-1"></a>
<h4 class="subsection">12.16.1 hole</h4>
<dl>
<dt><a name="index-hole"></a>function: <strong>hole</strong> <em>imol start_x start_y start_z end_x end_y end_z colour_map_multiplier colour_map_offset n_runs show_probe_radius_graph_flag export_surface_dots_file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>start_x</em> is a number
</li><li> <em>start_y</em> is a number
</li><li> <em>start_z</em> is a number
</li><li> <em>end_x</em> is a number
</li><li> <em>end_y</em> is a number
</li><li> <em>end_z</em> is a number
</li><li> <em>colour_map_multiplier</em> is a number
</li><li> <em>colour_map_offset</em> is a number
</li><li> <em>n_runs</em> is an integer number
</li><li> <em>show_probe_radius_graph_flag</em> is a bool
</li><li> <em>export_surface_dots_file_name</em> is a std::string
</li></ul>

<p>starting piont and end point, colour map multiplier and shall the probe radius graph be shown (dummy value currently). 
</p>
<p>if export_dots_file_name string length is zero, then don&rsquo;t try to export the surface dots. 
</p>
</dd></dl>

<hr>
<a name="make_002dlink"></a>
<div class="header">
<p>
Previous: <a href="#hole" accesskey="p" rel="previous">hole</a>, Up: <a href="#Coot_0027s-Hole-implementation" accesskey="u" rel="up">Coot's Hole implementation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="make_002dlink-1"></a>
<h4 class="subsection">12.16.2 make-link</h4>
<dl>
<dt><a name="index-make_002dlink"></a>function: <strong>make-link</strong> <em>imol spec_1 spec_2 link_name length</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>spec_1</em> is a coot::atom_spec_t &amp;
</li><li> <em>spec_2</em> is a coot::atom_spec_t &amp;
</li><li> <em>link_name</em> is a const std::string &amp;
</li><li> <em>length</em> is a number
</li></ul>

<p>make a link between the specified atoms 
</p>
</dd></dl>

<hr>
<a name="Drag-and-Drop-Functions"></a>
<div class="header">
<p>
Next: <a href="#Map-Contouring-Functions" accesskey="n" rel="next">Map Contouring Functions</a>, Previous: <a href="#Coot_0027s-Hole-implementation" accesskey="p" rel="previous">Coot's Hole implementation</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Drag-and-Drop-Functions-1"></a>
<h3 class="section">12.17 Drag and Drop Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#handle_002ddrag_002dand_002ddrop_002dstring" accesskey="1">handle-drag-and-drop-string</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="handle_002ddrag_002dand_002ddrop_002dstring"></a>
<div class="header">
<p>
Up: <a href="#Drag-and-Drop-Functions" accesskey="u" rel="up">Drag and Drop Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="handle_002ddrag_002dand_002ddrop_002dstring-1"></a>
<h4 class="subsection">12.17.1 handle-drag-and-drop-string</h4>
<dl>
<dt><a name="index-handle_002ddrag_002dand_002ddrop_002dstring"></a>function: <strong>handle-drag-and-drop-string</strong> <em>uri</em></dt>
<dd>
<p>Where <em>uri</em> is a const std::string &amp;
</p>
<p>handle the string that get when a file or URL is dropped. 
</p>
</dd></dl>

<hr>
<a name="Map-Contouring-Functions"></a>
<div class="header">
<p>
Next: <a href="#Map-to-Model-Correlation" accesskey="n" rel="next">Map to Model Correlation</a>, Previous: <a href="#Drag-and-Drop-Functions" accesskey="p" rel="previous">Drag and Drop Functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-Contouring-Functions-1"></a>
<h3 class="section">12.18 Map Contouring Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#map_002dcontours" accesskey="1">map-contours</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="map_002dcontours"></a>
<div class="header">
<p>
Up: <a href="#Map-Contouring-Functions" accesskey="u" rel="up">Map Contouring Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dcontours-1"></a>
<h4 class="subsection">12.18.1 map-contours</h4>
<dl>
<dt><a name="index-map_002dcontours"></a>function: <strong>map-contours</strong> <em>imol contour_level</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>contour_level</em> is a number
</li></ul>

<p>return two lists: a list of vertices and a list of indices for connection 
</p>
</dd></dl>

<hr>
<a name="Map-to-Model-Correlation"></a>
<div class="header">
<p>
Next: <a href="#Get-Sequence" accesskey="n" rel="next">Get Sequence</a>, Previous: <a href="#Map-Contouring-Functions" accesskey="p" rel="previous">Map Contouring Functions</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Map-to-Model-Correlation-1"></a>
<h3 class="section">12.19 Map to Model Correlation</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#set_002dmap_002dcorrelation_002datom_002dradius" accesskey="1">set-map-correlation-atom-radius</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dto_002dmodel_002dcorrelation_002dscm" accesskey="2">map-to-model-correlation-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dto_002dmodel_002dcorrelation" accesskey="3">map-to-model-correlation</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dto_002dmodel_002dcorrelation_002dstats" accesskey="4">map-to-model-correlation-stats</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue" accesskey="5">map-to-model-correlation-per-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue" accesskey="6">map-to-model-correlation-stats-per-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm" accesskey="7">map-to-model-correlation-per-residue-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm" accesskey="8">map-to-model-correlation-stats-per-residue-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#qq_002dplot_002dmap_002dand_002dmodel_002dscm" accesskey="9">qq-plot-map-and-model-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#density_002dscore_002dresidue">density-score-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dmean_002dscm">map-mean-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dstatistics_002dscm">map-statistics-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#map_002dmean_002dpy">map-mean-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="set_002dmap_002dcorrelation_002datom_002dradius"></a>
<div class="header">
<p>
Next: <a href="#map_002dto_002dmodel_002dcorrelation_002dscm" accesskey="n" rel="next">map-to-model-correlation-scm</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dmap_002dcorrelation_002datom_002dradius-1"></a>
<h4 class="subsection">12.19.1 set-map-correlation-atom-radius</h4>
<dl>
<dt><a name="index-set_002dmap_002dcorrelation_002datom_002dradius"></a>function: <strong>set-map-correlation-atom-radius</strong> <em>r</em></dt>
<dd>
<p>Where <em>r</em> is a number
</p>
<p>The atom radius is not passed as a parameter to correlation. 
</p>
</dd></dl>

<hr>
<a name="map_002dto_002dmodel_002dcorrelation_002dscm"></a>
<div class="header">
<p>
Next: <a href="#map_002dto_002dmodel_002dcorrelation" accesskey="n" rel="next">map-to-model-correlation</a>, Previous: <a href="#set_002dmap_002dcorrelation_002datom_002dradius" accesskey="p" rel="previous">set-map-correlation-atom-radius</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dto_002dmodel_002dcorrelation_002dscm-1"></a>
<h4 class="subsection">12.19.2 map-to-model-correlation-scm</h4>
<dl>
<dt><a name="index-map_002dto_002dmodel_002dcorrelation_002dscm"></a>function: <strong>map-to-model-correlation-scm</strong> <em>imol residue_specs neighb_residue_specs atom_mask_mode imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a SCM
</li><li> <em>neighb_residue_specs</em> is a SCM
</li><li> <em>atom_mask_mode</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>atom-mask-mode is as follows: 
</p>
</dd></dl>

<hr>
<a name="map_002dto_002dmodel_002dcorrelation"></a>
<div class="header">
<p>
Next: <a href="#map_002dto_002dmodel_002dcorrelation_002dstats" accesskey="n" rel="next">map-to-model-correlation-stats</a>, Previous: <a href="#map_002dto_002dmodel_002dcorrelation_002dscm" accesskey="p" rel="previous">map-to-model-correlation-scm</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dto_002dmodel_002dcorrelation-1"></a>
<h4 class="subsection">12.19.3 map-to-model-correlation</h4>
<dl>
<dt><a name="index-map_002dto_002dmodel_002dcorrelation"></a>function: <strong>map-to-model-correlation</strong> <em>imol residue_specs neigh_residue_specs atom_mask_mode imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a const std::vector&lt; coot::residue_spec_t &gt; &amp;
</li><li> <em>neigh_residue_specs</em> is a const std::vector&lt; coot::residue_spec_t &gt; &amp;
</li><li> <em>atom_mask_mode</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>atom-mask-mode is as follows: 
</p>
</dd></dl>

<hr>
<a name="map_002dto_002dmodel_002dcorrelation_002dstats"></a>
<div class="header">
<p>
Next: <a href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue" accesskey="n" rel="next">map-to-model-correlation-per-residue</a>, Previous: <a href="#map_002dto_002dmodel_002dcorrelation" accesskey="p" rel="previous">map-to-model-correlation</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dto_002dmodel_002dcorrelation_002dstats-1"></a>
<h4 class="subsection">12.19.4 map-to-model-correlation-stats</h4>
<dl>
<dt><a name="index-map_002dto_002dmodel_002dcorrelation_002dstats"></a>function: <strong>map-to-model-correlation-stats</strong> <em>imol residue_specs neigh_residue_specs atom_mask_mode imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a const std::vector&lt; coot::residue_spec_t &gt; &amp;
</li><li> <em>neigh_residue_specs</em> is a const std::vector&lt; coot::residue_spec_t &gt; &amp;
</li><li> <em>atom_mask_mode</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>map to model density correlation stats 
</p>
<p>atom-mask-mode is as follows: 
</p>
</dd></dl>

<hr>
<a name="map_002dto_002dmodel_002dcorrelation_002dper_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue" accesskey="n" rel="next">map-to-model-correlation-stats-per-residue</a>, Previous: <a href="#map_002dto_002dmodel_002dcorrelation_002dstats" accesskey="p" rel="previous">map-to-model-correlation-stats</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dto_002dmodel_002dcorrelation_002dper_002dresidue-1"></a>
<h4 class="subsection">12.19.5 map-to-model-correlation-per-residue</h4>
<dl>
<dt><a name="index-map_002dto_002dmodel_002dcorrelation_002dper_002dresidue"></a>function: <strong>map-to-model-correlation-per-residue</strong> <em>imol specs atom_mask_mode imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>specs</em> is a const std::vector&lt; coot::residue_spec_t &gt; &amp;
</li><li> <em>atom_mask_mode</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>map to model density correlation, reported per residue 
</p>
<p>atom-mask-mode is as follows: 
</p>
</dd></dl>

<hr>
<a name="map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm" accesskey="n" rel="next">map-to-model-correlation-per-residue-scm</a>, Previous: <a href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue" accesskey="p" rel="previous">map-to-model-correlation-per-residue</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue-1"></a>
<h4 class="subsection">12.19.6 map-to-model-correlation-stats-per-residue</h4>
<dl>
<dt><a name="index-map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue"></a>function: <strong>map-to-model-correlation-stats-per-residue</strong> <em>imol residue_specs atom_mask_mode atom_radius_for_masking imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a const std::vector&lt; coot::residue_spec_t &gt; &amp;
</li><li> <em>atom_mask_mode</em> is an integer number
</li><li> <em>atom_radius_for_masking</em> is a number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>map to model density statistics, reported per residue 
</p>
</dd></dl>

<hr>
<a name="map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm"></a>
<div class="header">
<p>
Next: <a href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm" accesskey="n" rel="next">map-to-model-correlation-stats-per-residue-scm</a>, Previous: <a href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue" accesskey="p" rel="previous">map-to-model-correlation-stats-per-residue</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm-1"></a>
<h4 class="subsection">12.19.7 map-to-model-correlation-per-residue-scm</h4>
<dl>
<dt><a name="index-map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm"></a>function: <strong>map-to-model-correlation-per-residue-scm</strong> <em>imol residue_specs atom_mask_mode imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs</em> is a SCM
</li><li> <em>atom_mask_mode</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>map to model correlation 
</p>
</dd></dl>

<hr>
<a name="map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm"></a>
<div class="header">
<p>
Next: <a href="#qq_002dplot_002dmap_002dand_002dmodel_002dscm" accesskey="n" rel="next">qq-plot-map-and-model-scm</a>, Previous: <a href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm" accesskey="p" rel="previous">map-to-model-correlation-per-residue-scm</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm-1"></a>
<h4 class="subsection">12.19.8 map-to-model-correlation-stats-per-residue-scm</h4>
<dl>
<dt><a name="index-map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm"></a>function: <strong>map-to-model-correlation-stats-per-residue-scm</strong> <em>imol residue_specs_scm atom_mask_mode atom_radius_for_masking imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs_scm</em> is a SCM
</li><li> <em>atom_mask_mode</em> is an integer number
</li><li> <em>atom_radius_for_masking</em> is a number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>map to model stats 
</p>
</dd></dl>

<hr>
<a name="qq_002dplot_002dmap_002dand_002dmodel_002dscm"></a>
<div class="header">
<p>
Next: <a href="#density_002dscore_002dresidue" accesskey="n" rel="next">density-score-residue</a>, Previous: <a href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm" accesskey="p" rel="previous">map-to-model-correlation-stats-per-residue-scm</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="qq_002dplot_002dmap_002dand_002dmodel_002dscm-1"></a>
<h4 class="subsection">12.19.9 qq-plot-map-and-model-scm</h4>
<dl>
<dt><a name="index-qq_002dplot_002dmap_002dand_002dmodel_002dscm"></a>function: <strong>qq-plot-map-and-model-scm</strong> <em>imol residue_specs_scm neigh_residue_specs_scm atom_mask_mode imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_specs_scm</em> is a SCM
</li><li> <em>neigh_residue_specs_scm</em> is a SCM
</li><li> <em>atom_mask_mode</em> is an integer number
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>QQ plot of the model density correlation, reported per residue. 
</p>
<p>atom-mask-mode is as follows: 
</p>
</dd></dl>

<hr>
<a name="density_002dscore_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#map_002dmean_002dscm" accesskey="n" rel="next">map-mean-scm</a>, Previous: <a href="#qq_002dplot_002dmap_002dand_002dmodel_002dscm" accesskey="p" rel="previous">qq-plot-map-and-model-scm</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="density_002dscore_002dresidue-1"></a>
<h4 class="subsection">12.19.10 density-score-residue</h4>
<dl>
<dt><a name="index-density_002dscore_002dresidue"></a>function: <strong>density-score-residue</strong> <em>imol chain_id res_no ins_code imol_map</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a string
</li><li> <em>res_no</em> is an integer number
</li><li> <em>ins_code</em> is a string
</li><li> <em>imol_map</em> is an integer number
</li></ul>

<p>simple density score for given residue (over-ridden by scripting function) 
</p>
</dd></dl>

<hr>
<a name="map_002dmean_002dscm"></a>
<div class="header">
<p>
Next: <a href="#map_002dstatistics_002dscm" accesskey="n" rel="next">map-statistics-scm</a>, Previous: <a href="#density_002dscore_002dresidue" accesskey="p" rel="previous">density-score-residue</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dmean_002dscm-1"></a>
<h4 class="subsection">12.19.11 map-mean-scm</h4>
<dl>
<dt><a name="index-map_002dmean_002dscm"></a>function: <strong>map-mean-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return sigma for the given map. Return scheme False if not a valid map molecule number. 
</p>
</dd></dl>

<hr>
<a name="map_002dstatistics_002dscm"></a>
<div class="header">
<p>
Next: <a href="#map_002dmean_002dpy" accesskey="n" rel="next">map-mean-py</a>, Previous: <a href="#map_002dmean_002dscm" accesskey="p" rel="previous">map-mean-scm</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dstatistics_002dscm-1"></a>
<h4 class="subsection">12.19.12 map-statistics-scm</h4>
<dl>
<dt><a name="index-map_002dstatistics_002dscm"></a>function: <strong>map-statistics-scm</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return either scheme false on non-a-map or list (mean, standard-deviation, skew, kurtosis) 
</p>
</dd></dl>

<hr>
<a name="map_002dmean_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#map_002dstatistics_002dscm" accesskey="p" rel="previous">map-statistics-scm</a>, Up: <a href="#Map-to-Model-Correlation" accesskey="u" rel="up">Map to Model Correlation</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="map_002dmean_002dpy-1"></a>
<h4 class="subsection">12.19.13 map-mean-py</h4>
<dl>
<dt><a name="index-map_002dmean_002dpy"></a>function: <strong>map-mean-py</strong> <em>imol</em></dt>
<dd>
<p>Where <em>imol</em> is an integer number
</p>
<p>return sigma for the given map. Return Python False if not a valid map molecule number. 
</p>
</dd></dl>

<hr>
<a name="Get-Sequence"></a>
<div class="header">
<p>
Previous: <a href="#Map-to-Model-Correlation" accesskey="p" rel="previous">Map to Model Correlation</a>, Up: <a href="#More-Scripting-Functions" accesskey="u" rel="up">More Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Get-Sequence-1"></a>
<h3 class="section">12.20 Get Sequence</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#get_002dsequence_002das_002dfasta_002dfor_002dchain" accesskey="1">get-sequence-as-fasta-for-chain</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#write_002dsequence" accesskey="2">write-sequence</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="get_002dsequence_002das_002dfasta_002dfor_002dchain"></a>
<div class="header">
<p>
Next: <a href="#write_002dsequence" accesskey="n" rel="next">write-sequence</a>, Up: <a href="#Get-Sequence" accesskey="u" rel="up">Get Sequence</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002dsequence_002das_002dfasta_002dfor_002dchain-1"></a>
<h4 class="subsection">12.20.1 get-sequence-as-fasta-for-chain</h4>
<dl>
<dt><a name="index-get_002dsequence_002das_002dfasta_002dfor_002dchain"></a>function: <strong>get-sequence-as-fasta-for-chain</strong> <em>imol chain_id</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>chain_id</em> is a const std::string &amp;
</li></ul>

<p>get the sequence for chain_id in imol 
</p>
</dd></dl>

<hr>
<a name="write_002dsequence"></a>
<div class="header">
<p>
Previous: <a href="#get_002dsequence_002das_002dfasta_002dfor_002dchain" accesskey="p" rel="previous">get-sequence-as-fasta-for-chain</a>, Up: <a href="#Get-Sequence" accesskey="u" rel="up">Get Sequence</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="write_002dsequence-1"></a>
<h4 class="subsection">12.20.2 write-sequence</h4>
<dl>
<dt><a name="index-write_002dsequence"></a>function: <strong>write-sequence</strong> <em>imol file_name</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>file_name</em> is a const std::string &amp;
</li></ul>

<p>write the sequence for imol as fasta 
</p>
</dd></dl>

<hr>
<a name="More-Refinement-Scripting-Functions"></a>
<div class="header">
<p>
Next: <a href="#Scheme-Scripting-Functions" accesskey="n" rel="next">Scheme Scripting Functions</a>, Previous: <a href="#More-Scripting-Functions" accesskey="p" rel="previous">More Scripting Functions</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="More-Refinement-Scripting-Functions-1"></a>
<h2 class="chapter">13 More Refinement Scripting Functions</h2>
<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#More-Refinement-Functions" accesskey="1">More Refinement Functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="More-Refinement-Functions"></a>
<div class="header">
<p>
Up: <a href="#More-Refinement-Scripting-Functions" accesskey="u" rel="up">More Refinement Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="More-Refinement-Functions-1"></a>
<h3 class="section">13.1 More Refinement Functions</h3>

<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#use_002dunimodal_002dring_002dtorsion_002drestraints" accesskey="1">use-unimodal-ring-torsion-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue" accesskey="2">use-unimodal-ring-torsion-restraints-for-residue</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefinement_002dgeman_002dmcclure_002dalpha" accesskey="3">set-refinement-geman-mcclure-alpha</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002drefinement_002dgeman_002dmcclure_002dalpha" accesskey="4">get-refinement-geman-mcclure-alpha</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002drefinement_002dlennard_002djones_002depsilon" accesskey="5">set-refinement-lennard-jones-epsilon</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor" accesskey="6">set-log-cosh-target-distance-scale-factor</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#crankshaft_002dpeptide_002drotation_002doptimization_002dscm" accesskey="7">crankshaft-peptide-rotation-optimization-scm</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#crankshaft_002dpeptide_002drotation_002doptimization_002dpy" accesskey="8">crankshaft-peptide-rotation-optimization-py</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>

<hr>
<a name="use_002dunimodal_002dring_002dtorsion_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue" accesskey="n" rel="next">use-unimodal-ring-torsion-restraints-for-residue</a>, Up: <a href="#More-Refinement-Functions" accesskey="u" rel="up">More Refinement Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="use_002dunimodal_002dring_002dtorsion_002drestraints-1"></a>
<h4 class="subsection">13.1.1 use-unimodal-ring-torsion-restraints</h4>
<dl>
<dt><a name="index-use_002dunimodal_002dring_002dtorsion_002drestraints"></a>function: <strong>use-unimodal-ring-torsion-restraints</strong> <em>res_name</em></dt>
<dd>
<p>Where <em>res_name</em> is a const std::string &amp;
</p>
<p>use unimodal ring torsion restraints (e.g. for carbohydrate pyranose) 
</p>
</dd></dl>

<hr>
<a name="use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue"></a>
<div class="header">
<p>
Next: <a href="#set_002drefinement_002dgeman_002dmcclure_002dalpha" accesskey="n" rel="next">set-refinement-geman-mcclure-alpha</a>, Previous: <a href="#use_002dunimodal_002dring_002dtorsion_002drestraints" accesskey="p" rel="previous">use-unimodal-ring-torsion-restraints</a>, Up: <a href="#More-Refinement-Functions" accesskey="u" rel="up">More Refinement Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue-1"></a>
<h4 class="subsection">13.1.2 use-unimodal-ring-torsion-restraints-for-residue</h4>
<dl>
<dt><a name="index-use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue"></a>function: <strong>use-unimodal-ring-torsion-restraints-for-residue</strong> <em>res_name torsions_info_list</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>res_name</em> is a const std::string &amp;
</li><li> <em>torsions_info_list</em> is a PyObject
</li></ul>

<p>use unimodal ring torsion restraints (e.g. for carbohydrate pyranose) 
</p>
</dd></dl>

<hr>
<a name="set_002drefinement_002dgeman_002dmcclure_002dalpha"></a>
<div class="header">
<p>
Next: <a href="#get_002drefinement_002dgeman_002dmcclure_002dalpha" accesskey="n" rel="next">get-refinement-geman-mcclure-alpha</a>, Previous: <a href="#use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue" accesskey="p" rel="previous">use-unimodal-ring-torsion-restraints-for-residue</a>, Up: <a href="#More-Refinement-Functions" accesskey="u" rel="up">More Refinement Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefinement_002dgeman_002dmcclure_002dalpha-1"></a>
<h4 class="subsection">13.1.3 set-refinement-geman-mcclure-alpha</h4>
<dl>
<dt><a name="index-set_002drefinement_002dgeman_002dmcclure_002dalpha"></a>function: <strong>set-refinement-geman-mcclure-alpha</strong> <em>alpha</em></dt>
<dd>
<p>Where <em>alpha</em> is a number
</p>
<p>set the Geman-McClure distance alpha value (weight) 
</p>
</dd></dl>

<hr>
<a name="get_002drefinement_002dgeman_002dmcclure_002dalpha"></a>
<div class="header">
<p>
Next: <a href="#set_002drefinement_002dlennard_002djones_002depsilon" accesskey="n" rel="next">set-refinement-lennard-jones-epsilon</a>, Previous: <a href="#set_002drefinement_002dgeman_002dmcclure_002dalpha" accesskey="p" rel="previous">set-refinement-geman-mcclure-alpha</a>, Up: <a href="#More-Refinement-Functions" accesskey="u" rel="up">More Refinement Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002drefinement_002dgeman_002dmcclure_002dalpha-1"></a>
<h4 class="subsection">13.1.4 get-refinement-geman-mcclure-alpha</h4>
<dl>
<dt><a name="index-get_002drefinement_002dgeman_002dmcclure_002dalpha"></a>function: <strong>get-refinement-geman-mcclure-alpha</strong></dt>
<dd>
<p>get the Geman-McClure distance alpha value (weight) 
</p>
</dd></dl>

<hr>
<a name="set_002drefinement_002dlennard_002djones_002depsilon"></a>
<div class="header">
<p>
Next: <a href="#set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor" accesskey="n" rel="next">set-log-cosh-target-distance-scale-factor</a>, Previous: <a href="#get_002drefinement_002dgeman_002dmcclure_002dalpha" accesskey="p" rel="previous">get-refinement-geman-mcclure-alpha</a>, Up: <a href="#More-Refinement-Functions" accesskey="u" rel="up">More Refinement Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002drefinement_002dlennard_002djones_002depsilon-1"></a>
<h4 class="subsection">13.1.5 set-refinement-lennard-jones-epsilon</h4>
<dl>
<dt><a name="index-set_002drefinement_002dlennard_002djones_002depsilon"></a>function: <strong>set-refinement-lennard-jones-epsilon</strong> <em>epsilon</em></dt>
<dd>
<p>Where <em>epsilon</em> is a number
</p>
<p>set the Lennard Jones epsilon parameter 
</p>
</dd></dl>

<hr>
<a name="set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor"></a>
<div class="header">
<p>
Next: <a href="#crankshaft_002dpeptide_002drotation_002doptimization_002dscm" accesskey="n" rel="next">crankshaft-peptide-rotation-optimization-scm</a>, Previous: <a href="#set_002drefinement_002dlennard_002djones_002depsilon" accesskey="p" rel="previous">set-refinement-lennard-jones-epsilon</a>, Up: <a href="#More-Refinement-Functions" accesskey="u" rel="up">More Refinement Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor-1"></a>
<h4 class="subsection">13.1.6 set-log-cosh-target-distance-scale-factor</h4>
<dl>
<dt><a name="index-set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor"></a>function: <strong>set-log-cosh-target-distance-scale-factor</strong> <em>sf</em></dt>
<dd>
<p>Where <em>sf</em> is a number
</p>
<p>set the log cosh scale factor for target position restraints 
</p>
</dd></dl>

<hr>
<a name="crankshaft_002dpeptide_002drotation_002doptimization_002dscm"></a>
<div class="header">
<p>
Next: <a href="#crankshaft_002dpeptide_002drotation_002doptimization_002dpy" accesskey="n" rel="next">crankshaft-peptide-rotation-optimization-py</a>, Previous: <a href="#set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor" accesskey="p" rel="previous">set-log-cosh-target-distance-scale-factor</a>, Up: <a href="#More-Refinement-Functions" accesskey="u" rel="up">More Refinement Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="crankshaft_002dpeptide_002drotation_002doptimization_002dscm-1"></a>
<h4 class="subsection">13.1.7 crankshaft-peptide-rotation-optimization-scm</h4>
<dl>
<dt><a name="index-crankshaft_002dpeptide_002drotation_002doptimization_002dscm"></a>function: <strong>crankshaft-peptide-rotation-optimization-scm</strong> <em>imol residue_spec_smc</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec_smc</em> is a SCM
</li></ul>

<p>Apply crankshaft peptide rotation optimization to the specified residue. 
</p>
</dd></dl>

<hr>
<a name="crankshaft_002dpeptide_002drotation_002doptimization_002dpy"></a>
<div class="header">
<p>
Previous: <a href="#crankshaft_002dpeptide_002drotation_002doptimization_002dscm" accesskey="p" rel="previous">crankshaft-peptide-rotation-optimization-scm</a>, Up: <a href="#More-Refinement-Functions" accesskey="u" rel="up">More Refinement Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="crankshaft_002dpeptide_002drotation_002doptimization_002dpy-1"></a>
<h4 class="subsection">13.1.8 crankshaft-peptide-rotation-optimization-py</h4>
<dl>
<dt><a name="index-crankshaft_002dpeptide_002drotation_002doptimization_002dpy"></a>function: <strong>crankshaft-peptide-rotation-optimization-py</strong> <em>imol residue_spec_py</em></dt>
<dd>
<p>Where: 
</p><ul>
<li> <em>imol</em> is an integer number
</li><li> <em>residue_spec_py</em> is a PyObject
</li></ul>

<p>Apply crankshaft peptide rotation optimization to the specified residue. 
</p>
</dd></dl>





<hr>
<a name="Scheme-Scripting-Functions"></a>
<div class="header">
<p>
Next: <a href="#Concept-Index" accesskey="n" rel="next">Concept Index</a>, Previous: <a href="#More-Refinement-Scripting-Functions" accesskey="p" rel="previous">More Refinement Scripting Functions</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Scheme-Scripting-Functions-1"></a>
<h2 class="chapter">14 Scheme Scripting Functions</h2>


<table class="menu" border="0" cellspacing="0">
<tr><td align="left" valign="top">&bull; <a href="#shelx" accesskey="1">shelx</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#run_002dmogul" accesskey="2">run-mogul</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#remote_002dcontrol" accesskey="3">remote-control</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#tips" accesskey="4">tips</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refmac" accesskey="5">refmac</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#refmac_002dproblems" accesskey="6">refmac-problems</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#redefine_002dfunctions" accesskey="7">redefine-functions</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#tips_002dgui" accesskey="8">tips-gui</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#raster3d_002dfrom_002dscheme" accesskey="9">raster3d-from-scheme</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#quat_002dconvert">quat-convert</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#gui_002dprosmart">gui-prosmart</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mutate_002dfrom_002dscheme">mutate-from-scheme</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#parse_002dpisa_002dxml">parse-pisa-xml</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#add_002dlinked_002dcho">add-linked-cho</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ncs">ncs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#acedrg_002dlink">acedrg-link</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#ligand_002dcheck">ligand-check</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#what_002dcheck">what-check</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#user_002ddefine_002drestraints">user-define-restraints</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#jligand_002dgui">jligand-gui</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#gui_002dligand_002dsliders">gui-ligand-sliders</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#mutate_002din_002dscheme">mutate-in-scheme</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#entry_002bdo_002dbutton">entry+do-button</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#dictionary_002dgenerators">dictionary-generators</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#extra_002dtop_002dlevel">extra-top-level</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot_002dlsq">coot-lsq</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002debi">get-ebi</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot_002dgui">coot-gui</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#hello">hello</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot_002dutils">coot-utils</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#contact_002dscore_002disolated_002dligand">contact-score-isolated-ligand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#libcheck">libcheck</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#cns2coot">cns2coot</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#clear_002dbackup">clear-backup</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#generic_002dobjects">generic-objects</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#exercise_002dscm_002dmol">exercise-scm-mol</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#check_002dfor_002dupdates">check-for-updates</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#brute_002dlsqman">brute-lsqman</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fitting">fitting</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#group_002dsettings">group-settings</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#background_002ddemo">background-demo</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#americanisms">americanisms</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#filter">filter</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#a_002drapper_002dgui">a-rapper-gui</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#fascinating_002dthings">fascinating-things</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#get_002drecent_002dpdbe">get-recent-pdbe</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#g_t3d_002dgenerator_002dimport">3d-generator-import</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#coot">coot</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
<tr><td align="left" valign="top">&bull; <a href="#jligand">jligand</a>:</td><td>&nbsp;&nbsp;</td><td align="left" valign="top">
</td></tr>
</table>
<hr>
<a name="shelx"></a>
<div class="header">
<p>
Next: <a href="#run_002dmogul" accesskey="n" rel="next">run-mogul</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="shelx-1"></a>
<h3 class="section">14.1 shelx</h3>

<dl>
<dt><a name="index-shelxl_002drefine"></a>procedure: <strong>shelxl-refine</strong></dt>
</dl>

<dl>
<dt><a name="index-shelxl_002drefine_002dprimitive"></a>procedure: <strong>shelxl-refine-primitive</strong></dt>
<dd><p>hkl-file-in can be null &rsquo;() or (list astring).
</p></dd></dl>

<dl>
<dt><a name="index-read_002dshelx_002dlst_002dfile"></a>procedure: <strong>read-shelx-lst-file</strong></dt>
<dd><p>ie. create a interesting-things GUI for split (and other things?)
in a shelx .lst file.
</p>
</dd></dl>
<hr>
<a name="run_002dmogul"></a>
<div class="header">
<p>
Next: <a href="#remote_002dcontrol" accesskey="n" rel="next">remote-control</a>, Previous: <a href="#shelx" accesskey="p" rel="previous">shelx</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="run_002dmogul-1"></a>
<h3 class="section">14.2 run-mogul</h3>

<dl>
<dt><a name="index-run_002dmogul"></a>procedure: <strong>run-mogul</strong> <em>mode imol chain-id res-no ins-code prefix-str use-cache?</em></dt>
<dd><p>prefix-str can be blank
</p>
<p>return:
  a string that is the mogul-out file-name
  a symbol: something went wrong
</p>
</dd></dl>

<dl>
<dt><a name="index-run_002dmogul_002dshow_002dresults"></a>procedure: <strong>run-mogul-show-results</strong> <em>imol code mode chain-id res-no ins-code</em></dt>
<dd><p>return a string that is the mogul results file name that gets
parsed to make results (return #f on failure)
mode is &rsquo;all or &rsquo;bonds-and-angles
</p>
</dd></dl>
<hr>
<a name="remote_002dcontrol"></a>
<div class="header">
<p>
Next: <a href="#tips" accesskey="n" rel="next">tips</a>, Previous: <a href="#run_002dmogul" accesskey="p" rel="previous">run-mogul</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="remote_002dcontrol-1"></a>
<h3 class="section">14.3 remote-control</h3>

<dl>
<dt><a name="index-_0025coot_002dlistener_002dsocket"></a>procedure: <strong>%coot-listener-socket</strong></dt>
<dd><p>a place-holder for the coot listener socket
</p></dd></dl>

<dl>
<dt><a name="index-open_002dcoot_002dlistener_002dsocket"></a>procedure: <strong>open-coot-listener-socket</strong> <em>port-number host-name</em></dt>
<dd><p>Open a coot listener socket, return nothing much, but do set! 
%coot-listener-socket.
</p>
<p>Hmmm... why make a side-effect like that?  Why not return
%coot-listener-socket so that the caller can set it?  There may be
a reason...
</p>
<p>And the reason is that I can then call
coot-listener-idle-function-proc without having to use a c++
variable.
</p>
</dd></dl>

<dl>
<dt><a name="index-open_002dcoot_002dlistener_002dsocket_002dwith_002dtimeout"></a>procedure: <strong>open-coot-listener-socket-with-timeout</strong> <em>port-number host-name</em></dt>
<dd><p>yet another go to make a coot port reader work.  This time, we use
a gtk-timer to read stuff from the socket.  
</p>
<p>The gtk-timer function must return 1 to be called again.  When we
want to close the socket reader, simply make the function return 0.
</p>
</dd></dl>

<dl>
<dt><a name="index-coot_002dsocket_002dtimeout_002dfunc"></a>procedure: <strong>coot-socket-timeout-func</strong></dt>
<dd><p>based on coot-listener-idle-function-proc
</p>
</dd></dl>

<dl>
<dt><a name="index-coot_002dlistener_002didle_002dfunction_002dproc"></a>procedure: <strong>coot-listener-idle-function-proc</strong></dt>
<dd><p>Do this thing when idle
</p>
<p>currently set to listen to the %coot-listener-socket
</p>
</dd></dl>

<dl>
<dt><a name="index-eval_002dsocket_002dstring"></a>procedure: <strong>eval-socket-string</strong> <em>s</em></dt>
<dd><p>the function to run from the main thread to evaluate a string:
</p></dd></dl>
<hr>
<a name="tips"></a>
<div class="header">
<p>
Next: <a href="#refmac" accesskey="n" rel="next">refmac</a>, Previous: <a href="#remote_002dcontrol" accesskey="p" rel="previous">remote-control</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="tips-1"></a>
<h3 class="section">14.4 tips</h3>

<dl>
<dt><a name="index-tip_002dlist"></a>procedure: <strong>tip-list</strong></dt>
<dd><p>a list of tips for Coot
</p></dd></dl>

<dl>
<dt><a name="index-no_002dcoot_002dtips"></a>procedure: <strong>no-coot-tips</strong></dt>
<dd><p>Function to turn off coot tips at start
</p></dd></dl>
<hr>
<a name="refmac"></a>
<div class="header">
<p>
Next: <a href="#refmac_002dproblems" accesskey="n" rel="next">refmac-problems</a>, Previous: <a href="#tips" accesskey="p" rel="previous">tips</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refmac-1"></a>
<h3 class="section">14.5 refmac</h3>

<dl>
<dt><a name="index-refmac_002dexe"></a>procedure: <strong>refmac-exe</strong></dt>
<dd><p>This is the default refmac version, it is presumed to be in the
path.  It can be overridden using a re-definition either at the
scripting interface or in one&rsquo;s ~/.coot file. E.g.:
(define refmac-exec &quot;/y/programs/xtal/refmac-latest/bin/refmac5-3-dec-2004&quot;)
</p></dd></dl>

<dl>
<dt><a name="index-refmac_002dextra_002dparams"></a>procedure: <strong>refmac-extra-params</strong></dt>
<dd><p>Set this to a list of parameter strings:
</p>
<p>If refmac-extra-params is a list of strings, it is used in
preference to the &quot;refmac-extra-params&quot; file (should it exist).
e.g. (set! refmac-extra-params (list &quot;WEIGHT 0.2&quot; 
                                     &quot;NCYC 10&quot; 
                                     &quot;REFI BREF ISO&quot;
                                     &quot;REFI METH CGMAT&quot;
                                     &quot;REFI TYPE REST RESO 20 1.64&quot;))
</p>
</dd></dl>

<dl>
<dt><a name="index-run_002drefmac_002dby_002dfilename"></a>procedure: <strong>run-refmac-by-filename</strong> <em>pdb-in-filename pdb-out-filename mtz-in-filename mtz-out-filename extra-cif-lib-filename imol-refmac-count swap-map-colours-post-refmac? imol-mtz-molecule show-diff-map-flag phase-combine-flag phib-fom-pair force-n-cycles make-molecules-flag ccp4i-project-dir f-col sig-f-col . r-free-col</em></dt>
</dl>

<dl>
<dt><a name="index-extra_002dparams_002dinclude_002dweight_003f"></a>procedure: <strong>extra-params-include-weight?</strong> <em>params-list</em></dt>
<dd><p>Return #t if the list of strings <var>params-list</var> contains a
string beginning with &quot;WEIGHT&quot;.  If not return #f
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002drefmac_002dextra_002dparams"></a>procedure: <strong>get-refmac-extra-params</strong></dt>
<dd><p>If refmac-extra-params is defined (as a list of strings), then
return that, else read the file &quot;refmac-extra-params&quot;.
</p>
<p>Return a list a list of strings.
</p>
</dd></dl>

<dl>
<dt><a name="index-run_002drefmac_002dfor_002dphases"></a>procedure: <strong>run-refmac-for-phases</strong></dt>
<dd><p>this is not run as a sub-thread, no useful return value.
</p>
</dd></dl>
<hr>
<a name="refmac_002dproblems"></a>
<div class="header">
<p>
Next: <a href="#redefine_002dfunctions" accesskey="n" rel="next">redefine-functions</a>, Previous: <a href="#refmac" accesskey="p" rel="previous">refmac</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="refmac_002dproblems-1"></a>
<h3 class="section">14.6 refmac-problems</h3>

<dl>
<dt><a name="index-refmac_002dproblems_002dgui"></a>procedure: <strong>refmac-problems-gui</strong> <em>imol problem-list</em></dt>
<dd><p>The gui for the refmac problems
</p>
</dd></dl>
<hr>
<a name="redefine_002dfunctions"></a>
<div class="header">
<p>
Next: <a href="#tips_002dgui" accesskey="n" rel="next">tips-gui</a>, Previous: <a href="#refmac_002dproblems" accesskey="p" rel="previous">refmac-problems</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="redefine_002dfunctions-1"></a>
<h3 class="section">14.7 redefine-functions</h3>

<dl>
<dt><a name="index-enhanced_002dligand_002dcoot_003f"></a>procedure: <strong>enhanced-ligand-coot?</strong></dt>
<dd><p>scm aliases.  
</p>
</dd></dl>

<dl>
<dt><a name="index-set_002dfind_002dhydrogen_002dtorsion"></a>procedure: <strong>set-find-hydrogen-torsion</strong></dt>
<dd><p>fix typo of set-find-hydrogen-torsions (backward compatibility in case anyone was using that)
</p></dd></dl>

<dl>
<dt><a name="index-cif_002dfile_002dfor_002dcomp_002did"></a>procedure: <strong>cif-file-for-comp-id</strong></dt>
<dd><p>(define residue-&gt;sdf-file residue-to-sdf-file)
</p></dd></dl>

<dl>
<dt><a name="index-toggle_002didle_002dligand_002dinteractions"></a>procedure: <strong>toggle-idle-ligand-interactions</strong></dt>
<dd><p>I changed the function name - save those (just a few) with scripts that I&rsquo;ve handed out
</p></dd></dl>
<hr>
<a name="tips_002dgui"></a>
<div class="header">
<p>
Next: <a href="#raster3d_002dfrom_002dscheme" accesskey="n" rel="next">raster3d-from-scheme</a>, Previous: <a href="#redefine_002dfunctions" accesskey="p" rel="previous">redefine-functions</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="tips_002dgui-1"></a>
<h3 class="section">14.8 tips-gui</h3>

<dl>
<dt><a name="index-show_002dcoot_002dtip_002dfrom_002dlist"></a>procedure: <strong>show-coot-tip-from-list</strong> <em>n text</em></dt>
<dd><p>given a number and a gtk text widget <var>text</var>, put tip number
<var>n</var> into the widget.
</p></dd></dl>

<dl>
<dt><a name="index-increment_002dcoot_002dtip_002dnumber"></a>procedure: <strong>increment-coot-tip-number</strong></dt>
<dd><p>increment the tip number when the user sees a tip
</p></dd></dl>

<dl>
<dt><a name="index-decrease_002dcoot_002dtip_002dnumber"></a>procedure: <strong>decrease-coot-tip-number</strong></dt>
<dd><p>decrement the tip number when the user sees a tip
</p></dd></dl>

<dl>
<dt><a name="index-tips_002dgui"></a>procedure: <strong>tips-gui</strong></dt>
<dd><p>run the tips gui.
</p></dd></dl>
<hr>
<a name="raster3d_002dfrom_002dscheme"></a>
<div class="header">
<p>
Next: <a href="#quat_002dconvert" accesskey="n" rel="next">quat-convert</a>, Previous: <a href="#tips_002dgui" accesskey="p" rel="previous">tips-gui</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="raster3d_002dfrom_002dscheme-1"></a>
<h3 class="section">14.9 raster3d-from-scheme</h3>

<dl>
<dt><a name="index-render_002dimage"></a>procedure: <strong>render-image</strong></dt>
<dd><p>run raster3d
</p></dd></dl>

<dl>
<dt><a name="index-raytrace"></a>procedure: <strong>raytrace</strong></dt>
<dd><p>Run either raster3d or povray
</p>
</dd></dl>
<hr>
<a name="quat_002dconvert"></a>
<div class="header">
<p>
Next: <a href="#gui_002dprosmart" accesskey="n" rel="next">gui-prosmart</a>, Previous: <a href="#raster3d_002dfrom_002dscheme" accesskey="p" rel="previous">raster3d-from-scheme</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="quat_002dconvert-1"></a>
<h3 class="section">14.10 quat-convert</h3>

<dl>
<dt><a name="index-matrix_002d_003equaternion"></a>procedure: <strong>matrix-&gt;quaternion</strong></dt>
<dd><p>convert a view matrix to a view quaternion to set Coot view internals.
</p>
</dd></dl>

<dl>
<dt><a name="index-set_002dview_002dmatrix"></a>procedure: <strong>set-view-matrix</strong> <em>m00 m10 m20 m01 m11 m21 m02 m12 m22</em></dt>
<dd><p>Set the view matrix
</p>
</dd></dl>
<hr>
<a name="gui_002dprosmart"></a>
<div class="header">
<p>
Next: <a href="#mutate_002dfrom_002dscheme" accesskey="n" rel="next">mutate-from-scheme</a>, Previous: <a href="#quat_002dconvert" accesskey="p" rel="previous">quat-convert</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="gui_002dprosmart-1"></a>
<h3 class="section">14.11 gui-prosmart</h3>

<dl>
<dt><a name="index-run_002dprosmart"></a>procedure: <strong>run-prosmart</strong> <em>imol-target imol-ref include-side-chains?</em></dt>
<dd><p>target is my molecule, ref is the homologous (high-res) model
</p>
</dd></dl>
<hr>
<a name="mutate_002dfrom_002dscheme"></a>
<div class="header">
<p>
Next: <a href="#parse_002dpisa_002dxml" accesskey="n" rel="next">parse-pisa-xml</a>, Previous: <a href="#gui_002dprosmart" accesskey="p" rel="previous">gui-prosmart</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mutate_002dfrom_002dscheme-1"></a>
<h3 class="section">14.12 mutate-from-scheme</h3>

<dl>
<dt><a name="index-povray_002dversion"></a>procedure: <strong>povray-version</strong></dt>
<dd><p>this should negate the need for Bill&rsquo;s patch here.
</p></dd></dl>

<dl>
<dt><a name="index-povray_002dargs"></a>procedure: <strong>povray-args</strong></dt>
<dd><p>args not including the output filename 
</p>
</dd></dl>

<dl>
<dt><a name="index-povray_002dimage"></a>procedure: <strong>povray-image</strong></dt>
<dd><p>Run provray using current displayed image and write .pov file to
default filename
</p>
</dd></dl>
<hr>
<a name="parse_002dpisa_002dxml"></a>
<div class="header">
<p>
Next: <a href="#add_002dlinked_002dcho" accesskey="n" rel="next">add-linked-cho</a>, Previous: <a href="#mutate_002dfrom_002dscheme" accesskey="p" rel="previous">mutate-from-scheme</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="parse_002dpisa_002dxml-1"></a>
<h3 class="section">14.13 parse-pisa-xml</h3>

<dl>
<dt><a name="index-_002apisa_002dcommand_002a"></a>procedure: <strong>*pisa-command*</strong></dt>
<dd><p>(define *pisa-command* &quot;/home/paule/ccp4/ccp4-6.1.2/bin/pisa&quot;)
</p></dd></dl>

<dl>
<dt><a name="index-pisa_002dassemblies_002dxml"></a>procedure: <strong>pisa-assemblies-xml</strong> <em>imol file-name</em></dt>
<dd><p>it calls parse-pisa-assemblies which does the work
</p>
</dd></dl>

<dl>
<dt><a name="index-pisa_002dhandle_002dsxml_002dmolecule"></a>procedure: <strong>pisa-handle-sxml-molecule</strong> <em>imol molecule pisa-results-type</em></dt>
<dd><p>a interface-molecule record contains information about pvalue and residues.
</p>
</dd></dl>

<dl>
<dt><a name="index-parse_002dpisa_002dassemblies"></a>procedure: <strong>parse-pisa-assemblies</strong></dt>
<dd><p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-
                     pisa assemblies:
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;-
</p>
<p>pisa_results
   name
   status
   total_asm
   asm_set
      ser_no
      assembly
         id
         size
         mmsize
         diss_energy
         asa
         bas
         entropy
         diss_area
         int_energy
         n_uc
         n_diss
         symNumber
         molecule
            chain_id
            rxx
            rxy
            rxz
            tx
            ryx
            ryy
            ryz
            ty
            rzx
            rzy
            rzz
            tz
            rxx-f
            rxy-f
            rxz-f
            tx-f
            ryx-f
            ryy-f
            ryz-f
            ty-f
            rzx-f
            rzy-f
            rzz-f
            tz-f
</p>
</dd></dl>

<dl>
<dt><a name="index-prep_002dfor_002dpisa"></a>procedure: <strong>prep-for-pisa</strong> <em>mode imol</em></dt>
<dd><p>20100213 prep-for-pisa needs to make directory, config file, write
the pdb file and thre return value should be #f if there was a
problem or some value where we can check that pisa -analyse ran
(probably a directory).  It is not clear right now where the output
is going.  config files has PISA_WORK_ROOT coot-pisa but things
seems to be written to DATA_ROOT
/home/emsley/ccp4/ccp4-6.1.3/share/pisa which seems like a bug (or
something like it) in pisa.  Needs rechecking.
</p>
</dd></dl>

<dl>
<dt><a name="index-cached_002dpisa_002danalysis"></a>procedure: <strong>cached-pisa-analysis</strong> <em>dir</em></dt>
<dd><p>needs fleshing out (see notes for prep-for-pisa).
</p>
</dd></dl>

<dl>
<dt><a name="index-parse_002dpisa_002dinterfaces"></a>procedure: <strong>parse-pisa-interfaces</strong> <em>imol sxml-entity</em></dt>
<dd><p>pdb_entry
   pdb_code
   status
   n_interfaces
   interface
      id
      type
      n_occ
      int_area
      int_solv_en
      pvalue
      stab_en
      css
      overlap
      x-rel
      fixed
      h-bonds
         n_bonds
         bond
            chain-1
            res-1
            seqnum-1
            inscode-1
            atname-1
            chain-2
            res-2
            seqnum-2
            inscode-2
            atname-2
            dist
      salt-bridges
         n-bonds
         bond
            chain-1
            res-1
            seqnum-1
            inscode-1
            atname-1
            chain-2
            res-2
            seqnum-2
            inscode-2
            atname-2
            dist
      ss-bonds
         n-bonds
         bond
            chain-1
            res-1
            seqnum-1
            inscode-1
            atname-1
            chain-2
            res-2
            seqnum-2
            inscode-2
            atname-2
            dist
      cov-bonds
         n-bonds
         bond
            chain-1
            res-1
            seqnum-1
            inscode-1
            atname-1
            chain-2
            res-2
            seqnum-2
            inscode-2
            atname-2
            dist
      molecule
         id
         chain_id
         class
         symop
         symop_no
         cell_i
         cell_j
         cell_k
         rxx
         rxy
         rxz
         tx
         ryx
         ryy
         ryz
         ty
         rzx
         rzy
         rzz
         tz
         int_natoms
         int_nres
         int_area
         int_solv_en
         pvalue
         residues
            residue
               ser_no
               name
               seq_num
               ins_code
               bonds
               asa
               bsa
               solv_en
</p>               
</dd></dl>
<hr>
<a name="add_002dlinked_002dcho"></a>
<div class="header">
<p>
Next: <a href="#ncs" accesskey="n" rel="next">ncs</a>, Previous: <a href="#parse_002dpisa_002dxml" accesskey="p" rel="previous">parse-pisa-xml</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="add_002dlinked_002dcho-1"></a>
<h3 class="section">14.14 add-linked-cho</h3>

<dl>
<dt><a name="index-new_002dmolecule_002dfrom_002dthis_002dglyco_002dtree"></a>procedure: <strong>new-molecule-from-this-glyco-tree</strong></dt>
<dd><p>return the new molecule number
</p></dd></dl>

<dl>
<dt><a name="index-complex_002dmammal_002dtree"></a>procedure: <strong>complex-mammal-tree</strong></dt>
<dd><p>complex mammal
bianntennary mammal
</p></dd></dl>

<dl>
<dt><a name="index-complex_002dplant_002dtree"></a>procedure: <strong>complex-plant-tree</strong></dt>
<dd><p>complex plant
plant bianntennary
</p></dd></dl>

<dl>
<dt><a name="index-_002aadd_002dlinked_002dresidue_002dtree_002dcorrelation_002dcut_002doff_002a"></a>procedure: <strong>*add-linked-residue-tree-correlation-cut-off*</strong></dt>
<dd><p>now users can set this
</p></dd></dl>
<hr>
<a name="ncs"></a>
<div class="header">
<p>
Next: <a href="#acedrg_002dlink" accesskey="n" rel="next">acedrg-link</a>, Previous: <a href="#add_002dlinked_002dcho" accesskey="p" rel="previous">add-linked-cho</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ncs-1"></a>
<h3 class="section">14.15 ncs</h3>

<dl>
<dt><a name="index-skip_002dto_002dnext_002dncs_002dchain"></a>procedure: <strong>skip-to-next-ncs-chain</strong> <em>direction</em></dt>
<dd><p>Skip the residue in the next chain (typically of a molecule with
NCS) with the same residue number.  If on the last chain, then wrap
to beginning.  If it can&rsquo;t find anything then don&rsquo;t move (and put a
message in the status bar)
</p>
</dd></dl>

<dl>
<dt><a name="index-single_002dmanual_002dncs_002dghost"></a>procedure: <strong>single-manual-ncs-ghost</strong> <em>imol resno-start resno-end ref-chain peer-chain</em></dt>
<dd><p>We can only see one peer at a time with this (each time we do a
clear-ncs-ghost-matrices).
</p>
</dd></dl>

<dl>
<dt><a name="index-manual_002dncs_002dghosts"></a>procedure: <strong>manual-ncs-ghosts</strong> <em>imol resno-start resno-end chain-id-list</em></dt>
<dd><p>chain-id-list is (list &quot;A&quot; &quot;B&quot; &quot;C&quot; &quot;D&quot;)), i.e. the
reference/target/master chain-id first and then the peers.  This
allows us to add many peers at the same time (unlike above
function).
</p>
</dd></dl>

<dl>
<dt><a name="index-update_002dncs_002dghosts_002dby_002dlocal_002dsphere"></a>procedure: <strong>update-ncs-ghosts-by-local-sphere</strong></dt>
<dd><p>Update NCS ghosts based on local environment (residues within 6A of
(and including) the active residue).
</p>
<p>Typically one would bind this function to a key.
</p>
</dd></dl>

<dl>
<dt><a name="index-ncs_002dmaster_002dchain_002did"></a>procedure: <strong>ncs-master-chain-id</strong> <em>imol</em></dt>
<dd><p>Return the first master chain id (usually there is only one of course) or #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-ncs_002dligand"></a>procedure: <strong>ncs-ligand</strong> <em>imol-protein ncs-master-chain-id imol-ligand chain-id-ligand resno-ligand-start resno-ligand-stop</em></dt>
<dd><p>This was designed to create an NCS copy of a ligand (or range of
residues) in the active site of one chain to the as yet unoccupied
active site of another, i.e. it makes a NCS ligand &quot;D&quot;1 that is a NCS
copy of ligand &quot;C&quot;1 using an NCS operator that maps protein chain &quot;A&quot;
onto chain &quot;B&quot;.
</p>
</dd></dl>
<hr>
<a name="acedrg_002dlink"></a>
<div class="header">
<p>
Next: <a href="#ligand_002dcheck" accesskey="n" rel="next">ligand-check</a>, Previous: <a href="#ncs" accesskey="p" rel="previous">ncs</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="acedrg_002dlink-1"></a>
<h3 class="section">14.16 acedrg-link</h3>

<dl>
<dt><a name="index-hack_002dlink"></a>procedure: <strong>hack-link</strong> <em>fn</em></dt>
<dd><p>return the new file name
</p></dd></dl>
<hr>
<a name="ligand_002dcheck"></a>
<div class="header">
<p>
Next: <a href="#what_002dcheck" accesskey="n" rel="next">what-check</a>, Previous: <a href="#acedrg_002dlink" accesskey="p" rel="previous">acedrg-link</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="ligand_002dcheck-1"></a>
<h3 class="section">14.17 ligand-check</h3>

<dl>
<dt><a name="index-get_002dmetrics_002dfor_002dligand"></a>procedure: <strong>get-metrics-for-ligand</strong> <em>imol chain-id res-no ins-code refmac-input-mtz-file-name fobs-col sig-fobs-col rfree-col refmac-dir</em></dt>
<dd><p>test:
(using-active-atom (get-metrics-for-ligand aa-imol aa-chain-id aa-res-no aa-ins-code &quot;rnasa-1.8-all_refmac1.mtz&quot; &quot;FGMP18&quot; &quot;SIGFGMP18&quot; &quot;FreeR_flag&quot; &quot;coot-refmac&quot;))
</p>
<p>refmac-dir: the dir where input/output refmac files should be written.
</p>
<p>return (list correlation mogul bumps difference-map-stats b-factor-info)
where:
    mogul is an improper pair (mogul-z-worst . mogul-out-file-name) or an error status
    difference-map-stats:
      the output of map-to-model-correlation-stats-scm
      (list correlation var_x var_y n sum_x sum_y D D2 (based on mean/sd of the map at the ligand) 
            map_mean map_mean_at_ligand map_sd map_sd_at_ligand SCM_UNDEFINED)
and b-factor-info
      list of (median-ratio median-ligand median-env ks-test-result)
</p>
</dd></dl>

<dl>
<dt><a name="index-_002aligand_002dcheck_002dresolution_002dlimit_002a"></a>procedure: <strong>*ligand-check-resolution-limit*</strong></dt>
<dd><p>only look at ligands in maps with resolution worse than this:
(to look at everything, set it to 0.1)
</p>
</dd></dl>

<dl>
<dt><a name="index-filter_002dout_002dwaters"></a>procedure: <strong>filter-out-waters</strong> <em>imol env-residues</em></dt>
<dd><p>remove residues that are waters from env-residues
</p>
</dd></dl>
<hr>
<a name="what_002dcheck"></a>
<div class="header">
<p>
Next: <a href="#user_002ddefine_002drestraints" accesskey="n" rel="next">user-define-restraints</a>, Previous: <a href="#ligand_002dcheck" accesskey="p" rel="previous">ligand-check</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="what_002dcheck-1"></a>
<h3 class="section">14.18 what-check</h3>

<dl>
<dt><a name="index-strip_002dleading_002dspaces"></a>procedure: <strong>strip-leading-spaces</strong> <em>str</em></dt>
<dd><p>&quot;  53&quot; -&gt; &quot;53&quot;, &quot; &quot; -&gt; &quot;&quot;
</p></dd></dl>

<dl>
<dt><a name="index-go_002dto_002dresidue_002dby_002dspec"></a>procedure: <strong>go-to-residue-by-spec</strong> <em>imol spec</em></dt>
<dd><p>consider for coot-utils
</p></dd></dl>

<dl>
<dt><a name="index-residue_002dspec_002d_003estring"></a>procedure: <strong>residue-spec-&gt;string</strong> <em>spec</em></dt>
<dd><p>consider for coot-utils
</p></dd></dl>

<dl>
<dt><a name="index-get_002dflip_002dresidue"></a>procedure: <strong>get-flip-residue</strong> <em>line</em></dt>
<dd><p>make this internal 
</p>
<p>where line is something like  &quot;  53 HIS  (  53 ) A12&quot;
                                  ignore   xxxxI Cyy
xxxx is 4char resno, I insertion code, C chain id, yy is model number
</p>
<p>return a residue spec, or #f
</p>
</dd></dl>

<dl>
<dt><a name="index-problem_002dresidues_002d_003edialog"></a>procedure: <strong>problem-residues-&gt;dialog</strong> <em>imol problemed-res-list-list</em></dt>
<dd><p>A problemed-res-list is improper pair where the car is a pair of
string, one describing the problem, and the other being the &quot;fix
it&quot; button label.  The cdr is a list of residue specs.
</p>
<p>A problemed-flip-list-list is a list of those things.
</p>
</dd></dl>

<dl>
<dt><a name="index-parse_002dcheck_002ddb"></a>procedure: <strong>parse-check-db</strong> <em>imol file-name action</em></dt>
<dd><p>action is either &rsquo;gui (then we get a gui) or &rsquo;apply-actions, then the model
modifications are automatically applied.
</p>
</dd></dl>
<hr>
<a name="user_002ddefine_002drestraints"></a>
<div class="header">
<p>
Next: <a href="#jligand_002dgui" accesskey="n" rel="next">jligand-gui</a>, Previous: <a href="#what_002dcheck" accesskey="p" rel="previous">what-check</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="user_002ddefine_002drestraints-1"></a>
<h3 class="section">14.19 user-define-restraints</h3>

<dl>
<dt><a name="index-add_002dbase_002drestraint"></a>procedure: <strong>add-base-restraint</strong> <em>imol spec-1 spec-2 atom-name-1 atom-name-2 dist</em></dt>
<dd><p>spec-1 and spec-2 are 7-element atom-specs
</p>
</dd></dl>

<dl>
<dt><a name="index-write_002drefmac_002dparallel_002dplane_002drestraint"></a>procedure: <strong>write-refmac-parallel-plane-restraint</strong> <em>file-name res-spec-0 res-spec-1 atom-list-0 atom-list-1</em></dt>
<dd><p>Example of a stack restraint: &quot;exte stac plan 1  firs resi 99 chai A atoms substitute-open-brace CB CG CD1 CD2 CE1 CE2 CZ OH substitute-close-brace  plan 2  firs resi 61 chai B atoms substitute-open-brace CB CG CD1 CD2 CE1 CE2 CZ substitute-close-brace  dist 3.4 sddi 0.2  sdan 6.0 type 1&quot;
</p>
</dd></dl>
<hr>
<a name="jligand_002dgui"></a>
<div class="header">
<p>
Next: <a href="#gui_002dligand_002dsliders" accesskey="n" rel="next">gui-ligand-sliders</a>, Previous: <a href="#user_002ddefine_002drestraints" accesskey="p" rel="previous">user-define-restraints</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="jligand_002dgui-1"></a>
<h3 class="section">14.20 jligand-gui</h3>

<dl>
<dt><a name="index-launch_002djligand_002dfunction"></a>procedure: <strong>launch-jligand-function</strong></dt>
<dd><p>This happens when user clicks on the &quot;Launch JLigand&quot; button.
It starts a jligand and puts it in the background.
</p>
</dd></dl>

<dl>
<dt><a name="index-click_002dselect_002dresidues_002dfor_002djligand"></a>procedure: <strong>click-select-residues-for-jligand</strong></dt>
<dd><p>This happens when user clicks on the &quot;Select Residues for JLigand&quot;
(or some such) button.  It expects the user to click on atoms of
the two residues involved in the link.
</p>
</dd></dl>
<hr>
<a name="gui_002dligand_002dsliders"></a>
<div class="header">
<p>
Next: <a href="#snarf_002dcoot_002ddocs" accesskey="n" rel="next">snarf-coot-docs</a>, Previous: <a href="#jligand_002dgui" accesskey="p" rel="previous">jligand-gui</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="gui_002dligand_002dsliders-1"></a>
<h3 class="section">14.21 gui-ligand-sliders</h3>

<dl>
<dt><a name="index-mtz_002dfile_002dname_002d_003erefinement_002dcolumn_002dlabels"></a>procedure: <strong>mtz-file-name-&gt;refinement-column-labels</strong> <em>file-name</em></dt>
<dd><p>return 3 values 
</p>
</dd></dl>
<hr>
<a name="snarf_002dcoot_002ddocs"></a>
<div class="header">
<p>
Next: <a href="#mutate_002din_002dscheme" accesskey="n" rel="next">mutate-in-scheme</a>, Previous: <a href="#gui_002dligand_002dsliders" accesskey="p" rel="previous">gui-ligand-sliders</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="snarf_002dcoot_002ddocs-1"></a>
<h3 class="section">14.22 snarf-coot-docs</h3>

<dl>
<dt><a name="index-directory_002das_002dfile_002dname"></a>procedure: <strong>directory-as-file-name</strong> <em>dir</em></dt>
<dd><p>remove any trailing /s
</p>
</dd></dl>

<dl>
<dt><a name="index-file_002dname_002dfile"></a>procedure: <strong>file-name-file</strong> <em>file-name</em></dt>
<dd><p>return the file component of file-name
&quot;x/y&quot; -&gt; &quot;y&quot; 
&quot;x&quot; -&gt; &quot;x&quot;
</p></dd></dl>

<dl>
<dt><a name="index-useful_002dglob"></a>procedure: <strong>useful-glob</strong> <em>dir pat</em></dt>
<dd><p>prepend the files with the directory dir
</p>
<p>(note that the args are reversed to &quot;natural&quot; order (which is the
reverse of glob args)
</p>
</dd></dl>

<dl>
<dt><a name="index-glob"></a>procedure: <strong>glob</strong> <em>pat dir</em></dt>
<dd><p>return a list of file names that match pattern pat in directory dir.
</p></dd></dl>

<dl>
<dt><a name="index-strip_002dextension"></a>procedure: <strong>strip-extension</strong> <em>s</em></dt>
<dd><p>&quot;a.b.res&quot; -&gt; &quot;a.b&quot;
file-name-sans-extension
</p>
</dd></dl>

<dl>
<dt><a name="index-command_002din_002dpath_003f"></a>procedure: <strong>command-in-path?</strong> <em>cmd</em></dt>
<dd><p>Return #t or #f:
</p></dd></dl>

<dl>
<dt><a name="index-goosh_002dcommand"></a>procedure: <strong>goosh-command</strong> <em>cmd args data-list log-file-name screen-output-also?</em></dt>
<dd><p>Where cmd is e.g. &quot;refmac&quot; 
      args is (list &quot;HKLIN&quot; &quot;thing.mtz&quot;)
      log-file-name is &quot;refmac.log&quot;      
      data-list is (list &quot;HEAD&quot; &quot;END&quot;)
</p>
<p>Return the exist status e.g. 0 or 1.
</p>
</dd></dl>

<dl>
<dt><a name="index-run_002dcommand_002fstrings"></a>procedure: <strong>run-command/strings</strong> <em>cmd args data-list</em></dt>
<dd><p>Return the strings screen output of cmd or #f if command was not found
</p>
</dd></dl>

<dl>
<dt><a name="index-string_002d_003elist_002dof_002dstrings"></a>procedure: <strong>string-&gt;list-of-strings</strong> <em>str</em></dt>
<dd><p>Return a list if str is a string, else return &rsquo;()
</p>
</dd></dl>

<dl>
<dt><a name="index-directory_002dfiles"></a>procedure: <strong>directory-files</strong> <em>dir</em></dt>
<dd><p>The following functions from PLEAC (guile version thereof of course).
</p>
<p>or define a utility function for this
</p></dd></dl>

<dl>
<dt><a name="index-add_002dto_002dlist_002dsection_002dtexis"></a>procedure: <strong>add-to-list-section-texis</strong></dt>
</dl>

<dl>
<dt><a name="index-delete_002dsection_002dtexi_002dfiles"></a>procedure: <strong>delete-section-texi-files</strong></dt>
</dl>
<hr>
<a name="mutate_002din_002dscheme"></a>
<div class="header">
<p>
Next: <a href="#entry_002bdo_002dbutton" accesskey="n" rel="next">entry+do-button</a>, Previous: <a href="#snarf_002dcoot_002ddocs" accesskey="p" rel="previous">snarf-coot-docs</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="mutate_002din_002dscheme-1"></a>
<h3 class="section">14.23 mutate-in-scheme</h3>

<dl>
<dt><a name="index-mutate_002dchain"></a>procedure: <strong>mutate-chain</strong> <em>imol chain-id sequence</em></dt>
<dd><p>Mutate chain-id of molecule number imol given sequence.  This
presumes a protein sequence.
</p>
<p>The number of residues in chain-id must match the length of sequence.
</p>
</dd></dl>

<dl>
<dt><a name="index-multi_002dmutate-1"></a>procedure: <strong>multi-mutate</strong></dt>
<dd><p>An internal function of mutate.  This presumes a protein sequence.
</p></dd></dl>

<dl>
<dt><a name="index-mutate_002dresidue_002drange"></a>procedure: <strong>mutate-residue-range</strong> <em>imol chain-id start-res-no stop-res-no sequence</em></dt>
<dd><p>The stop-res-no is inclusive, so usage e.g. (mutate-residue-range 0 &quot;A&quot; 1 2 &quot;AC&quot;)
</p>
<p>This presumes a protein sequence (not nucleic acid).
</p>
</dd></dl>

<dl>
<dt><a name="index-mutate_002dand_002dautofit_002dresidue_002drange"></a>procedure: <strong>mutate-and-autofit-residue-range</strong> <em>imol chain-id start-res-no stop-res-no sequence</em></dt>
<dd><p>mutate and auto fit a residue range.
</p>
<p>This presumes a protein sequence (not nucleic acid).
</p>
<p>The sequence is a string of one letter codes
</p>
</dd></dl>

<dl>
<dt><a name="index-mutate_002dand_002dauto_002dfit"></a>procedure: <strong>mutate-and-auto-fit</strong> <em>residue-number chain-id mol mol-for-map residue-type</em></dt>
<dd><p>mutate and autofit whole chain
</p>
<p>This presumes a protein sequence (not nucleic acid).
</p>
</dd></dl>

<dl>
<dt><a name="index-maf"></a>procedure: <strong>maf</strong></dt>
<dd><p>a short-hand for mutate-and-auto-fit
</p>
</dd></dl>

<dl>
<dt><a name="index-3_002dletter_002dcode_002d_003esingle_002dletter"></a>procedure: <strong>3-letter-code-&gt;single-letter</strong> <em>residue-type</em></dt>
<dd><p>return a char, return #\A for unknown residue-type
</p>
</dd></dl>

<dl>
<dt><a name="index-mutate_002dresidue_002dredundant"></a>procedure: <strong>mutate-residue-redundant</strong></dt>
<dd><p>a wrapper for mutate-single-residue-by-seqno (which uses slightly
inconvenient single letter code)
</p>
<p>Here residue-type is the 3-letter code
</p>
</dd></dl>

<dl>
<dt><a name="index-delete_002dsidechain_002drange"></a>procedure: <strong>delete-sidechain-range</strong></dt>
<dd><p>Delete (back to the CB stub) the side change in the range
resno-start to resno-end
</p>
</dd></dl>
<hr>
<a name="entry_002bdo_002dbutton"></a>
<div class="header">
<p>
Next: <a href="#dictionary_002dgenerators" accesskey="n" rel="next">dictionary-generators</a>, Previous: <a href="#mutate_002din_002dscheme" accesskey="p" rel="previous">mutate-in-scheme</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="entry_002bdo_002dbutton-1"></a>
<h3 class="section">14.24 entry+do-button</h3>

<dl>
<dt><a name="index-teb"></a>procedure: <strong>teb</strong></dt>
<dd><p>test that function:
</p></dd></dl>

<dl>
<dt><a name="index-teb2"></a>procedure: <strong>teb2</strong></dt>
<dd><p>test that function:
</p></dd></dl>
<hr>
<a name="dictionary_002dgenerators"></a>
<div class="header">
<p>
Next: <a href="#extra_002dtop_002dlevel" accesskey="n" rel="next">extra-top-level</a>, Previous: <a href="#entry_002bdo_002dbutton" accesskey="p" rel="previous">entry+do-button</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="dictionary_002dgenerators-1"></a>
<h3 class="section">14.25 dictionary-generators</h3>

<dl>
<dt><a name="index-generate_002dmolecule_002dfrom_002dmmcif_002dby_002ddict_002dgen"></a>procedure: <strong>generate-molecule-from-mmcif-by-dict-gen</strong> <em>comp-id mmcif-file-name</em></dt>
<dd><p>return a molecule number.  Return -1 on fail.
</p>
</dd></dl>
<hr>
<a name="extra_002dtop_002dlevel"></a>
<div class="header">
<p>
Next: <a href="#coot_002dlsq" accesskey="n" rel="next">coot-lsq</a>, Previous: <a href="#dictionary_002dgenerators" accesskey="p" rel="previous">dictionary-generators</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="extra_002dtop_002dlevel-1"></a>
<h3 class="section">14.26 extra-top-level</h3>

<dl>
<dt><a name="index-my_002dbutton_002dcallback"></a>procedure: <strong>my-button-callback</strong></dt>
<dd><p>This is what to do when the button is pressed.
It can be any guile or coot function.
</p>
</dd></dl>

<dl>
<dt><a name="index-my_002dtop_002dlevel"></a>procedure: <strong>my-top-level</strong></dt>
<dd><p>define a simple window and put a button in it
</p>
</dd></dl>
<hr>
<a name="coot_002dlsq"></a>
<div class="header">
<p>
Next: <a href="#get_002debi" accesskey="n" rel="next">get-ebi</a>, Previous: <a href="#extra_002dtop_002dlevel" accesskey="p" rel="previous">extra-top-level</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot_002dlsq-1"></a>
<h3 class="section">14.27 coot-lsq</h3>

<dl>
<dt><a name="index-lsq_002dmatch_002dtype_002dsymbol_002d_003enumber"></a>procedure: <strong>lsq-match-type-symbol-&gt;number</strong> <em>match-type-in</em></dt>
<dd><p>Internal type conversion for LSQ fitting.  Return a number
according to the symbol match-type-in
</p>
</dd></dl>

<dl>
<dt><a name="index-set_002dmatch_002delement"></a>procedure: <strong>set-match-element</strong></dt>
<dd><p>Create matchers, 7 elements: 
  (list ref-start-resno ref-end-resno ref-chain-id imol-ref
	   mov-start-resno mov-end-resno mov-chain-id imol-mov
	   match-type)
</p>
</dd></dl>

<dl>
<dt><a name="index-lsq_002dmatch"></a>procedure: <strong>lsq-match</strong> <em>imol-ref imol-moving match-list</em></dt>
<dd><p>The scripting interface to LSQ matching.  Pass molecule numbers for
the reference (imol-ref) and moving (imol-moving) molecules and a
match list.  The match list format is described in the manual.
</p>
</dd></dl>

<dl>
<dt><a name="index-simple_002dlsq_002dmatch"></a>procedure: <strong>simple-lsq-match</strong> <em>ref-start-resno ref-end-resno ref-chain-id imol-ref mov-start-resno mov-end-resno mov-chain-id imol-mov match-type</em></dt>
<dd><p>Simple interface to LSQ fitting.  More often than not this is what
you will want, I imagine, e.g. (simple-lsq-match 940 950 &quot;A&quot; 0 940
950 &quot;A&quot; 1 &rsquo;main)
</p>
</dd></dl>
<hr>
<a name="get_002debi"></a>
<div class="header">
<p>
Next: <a href="#coot_002dgui" accesskey="n" rel="next">coot-gui</a>, Previous: <a href="#coot_002dlsq" accesskey="p" rel="previous">coot-lsq</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002debi-1"></a>
<h3 class="section">14.28 get-ebi</h3>

<dl>
<dt><a name="index-net_002dget_002durl"></a>procedure: <strong>net-get-url</strong> <em>my-url file-name</em></dt>
<dd><p>return 0 on success
</p></dd></dl>

<dl>
<dt><a name="index-check_002ddir_002dand_002dget_002durl"></a>procedure: <strong>check-dir-and-get-url</strong></dt>
<dd><p>check the directory and get url url-string.
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002durl_002dstr"></a>procedure: <strong>get-url-str</strong></dt>
<dd><p>get url-string for data type &rsquo;pdb or &rsquo;sfs
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002debi_002dpdb"></a>procedure: <strong>get-ebi-pdb</strong> <em>id</em></dt>
<dd><p>Return a molecule number on success
or not a number (#f) or -1 on error.
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002deds_002dpdb_002dand_002dmtz"></a>procedure: <strong>get-eds-pdb-and-mtz</strong> <em>id</em></dt>
<dd><p>Return a list of molecules (i.e. the model molecule and the 2 maps).
</p>
<p>or, if it didn&rsquo;t work then return #f
</p>
</dd></dl>
<hr>
<a name="coot_002dgui"></a>
<div class="header">
<p>
Next: <a href="#hello" accesskey="n" rel="next">hello</a>, Previous: <a href="#get_002debi" accesskey="p" rel="previous">get-ebi</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot_002dgui-1"></a>
<h3 class="section">14.29 coot-gui</h3>

<dl>
<dt><a name="index-run_002dgtk_002dpending_002devents"></a>procedure: <strong>run-gtk-pending-events</strong></dt>
</dl>

<dl>
<dt><a name="index-coot_002dgui"></a>procedure: <strong>coot-gui</strong></dt>
<dd><p>Fire up the coot scripting gui.  This function is called from the
main C++ code of coot.  Not much use if you don&rsquo;t have a gui to
type functions in to start with.
</p>
</dd></dl>

<dl>
<dt><a name="index-handle_002dsmiles_002dgo"></a>procedure: <strong>handle-smiles-go</strong> <em>tlc-entry smiles-entry</em></dt>
<dd><p>The callback from pressing the Go button in the smiles widget, an
interface to run libcheck.
</p>
</dd></dl>

<dl>
<dt><a name="index-smiles_002dgui"></a>procedure: <strong>smiles-gui</strong></dt>
<dd><p>smiles GUI
</p></dd></dl>

<dl>
<dt><a name="index-generic_002dsingle_002dentry"></a>procedure: <strong>generic-single-entry</strong> <em>function-label entry-1-default-text go-button-label handle-go-function</em></dt>
<dd><p>Generic single entry widget
</p>
<p>Pass the hint labels of the entries and a function that gets called
when user hits &quot;Go&quot;.  The handle-go-function accepts one argument
that is the entry text when the go button is pressed.
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002ddouble_002dentry"></a>procedure: <strong>generic-double-entry</strong> <em>label-1 label-2 entry-1-default-text entry-2-default-text check-button-label handle-check-button-function go-button-label handle-go-function</em></dt>
<dd><p>handle-go-function takes 3 arguments, the third of which is the
state of the check button.
</p>
<p>if check-button-label not a string, then we don&rsquo;t display (or
create, even) the check-button.  If it *is* a string, create a
check button and add the callback handle-check-button-function
which takes as an argument the active-state of the the checkbutton.
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002dmultiple_002dentries_002dwith_002dcheck_002dbutton"></a>procedure: <strong>generic-multiple-entries-with-check-button</strong> <em>entry-info-list check-button-info go-button-label handle-go-function</em></dt>
<dd><p>generic double entry widget, now with a check button
</p>


<p>OLD:
</p>
<p>pass a the hint labels of the entries and a function
(handle-go-function) that gets called when user hits &quot;Go&quot; (which
takes two string aguments and the active-state of the check button
(either #t of #f).
</p>
<p>if check-button-label not a string, then we don&rsquo;t display (or
create, even) the check-button.  If it *is* a string, create a
check button and add the callback handle-check-button-function
which takes as an argument the active-state of the the checkbutton.
</p>
</dd></dl>

<dl>
<dt><a name="index-molecule_002dcentres_002dgui"></a>procedure: <strong>molecule-centres-gui</strong></dt>
<dd><p>A demo gui to move about to molecules.
</p>
</dd></dl>

<dl>
<dt><a name="index-old_002dcoot_003f"></a>procedure: <strong>old-coot?</strong></dt>
<dd><p>old coot test
</p></dd></dl>

<dl>
<dt><a name="index-interesting_002dthings_002dgui"></a>procedure: <strong>interesting-things-gui</strong> <em>title baddie-list</em></dt>
<dd><p>We can either go to a place (in which case the element is a list of
button label (string) and 3 numbers that represent the x y z
coordinates) or an atom (in which case the element is a list of a
button label (string) followed by the molecule-number chain-id
residue-number ins-code atom-name altconf)
</p>
<p>e.g. (interesting-things-gui 
      &quot;Bad things by Analysis X&quot; 
      (list (list &quot;Bad Chiral&quot; 0 &quot;A&quot; 23 &quot;&quot; &quot;CA&quot; &quot;A&quot;) 
            (list &quot;Bad Density Fit&quot; 0 &quot;B&quot; 65 &quot;&quot; &quot;CA&quot; &quot;&quot;) 
            (list &quot;Interesting blob&quot; 45.6 46.7 87.5)))
</p>
</dd></dl>

<dl>
<dt><a name="index-interesting_002dthings_002dwith_002dfix_002dmaybe"></a>procedure: <strong>interesting-things-with-fix-maybe</strong> <em>title baddie-list</em></dt>
<dd><p>In this case, each baddie can have a function at the end which is
called when the fix button is clicked.
</p>
</dd></dl>

<dl>
<dt><a name="index-fill_002doption_002dmenu_002dwith_002dmol_002doptions"></a>procedure: <strong>fill-option-menu-with-mol-options</strong> <em>menu filter-function</em></dt>
<dd><p>Fill an option menu with the &quot;right type&quot; of molecules.  If
filter-function returns #t then add it.  Typical value of
filter-function is valid-model-molecule?
</p>
</dd></dl>

<dl>
<dt><a name="index-fill_002doption_002dmenu_002dwith_002dcoordinates_002dmol_002doptions"></a>procedure: <strong>fill-option-menu-with-coordinates-mol-options</strong> <em>menu</em></dt>
<dd><p>Helper function for molecule chooser.  Not really for users.
</p>
<p>Return a list of models, corresponding to the menu items of the
option menu.
</p>
<p>The returned list will not contain references to map or closed
molecules.
</p>
</dd></dl>

<dl>
<dt><a name="index-fill_002doption_002dmenu_002dwith_002dnumber_002doptions"></a>procedure: <strong>fill-option-menu-with-number-options</strong> <em>menu number-list default-option-value</em></dt>
</dl>

<dl>
<dt><a name="index-get_002doption_002dmenu_002dactive_002dmolecule"></a>procedure: <strong>get-option-menu-active-molecule</strong> <em>option-menu model-mol-list</em></dt>
<dd><p>Helper function for molecule chooser.  Not really for users.
</p>
<p>return the molecule number of the active item in the option menu,
or return #f if there was a problem (e.g. closed molecule)
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002doption_002dmenu_002dactive_002ditem"></a>procedure: <strong>get-option-menu-active-item</strong> <em>option-menu item-list</em></dt>
<dd><p>Here we return the active item in an option menu of generic items
</p>
</dd></dl>

<dl>
<dt><a name="index-molecule_002dchooser_002dgui_002dgeneric"></a>procedure: <strong>molecule-chooser-gui-generic</strong> <em>chooser-label callback-function option-menu-fill-function</em></dt>
<dd><p>Typically option-menu-fill-function is fill-option-menu-with-coordinates-mol-options
</p>
</dd></dl>

<dl>
<dt><a name="index-molecule_002dchooser_002dgui"></a>procedure: <strong>molecule-chooser-gui</strong></dt>
<dd><p>Fire up a coordinates/model molecule chooser dialog, with a given
label and on OK we call the call-back-fuction with an argument of
the chosen molecule number.
</p>
<p>chooser-label is a directive to the user such as &quot;Choose a Molecule&quot;
</p>
<p>callback-function is a function that takes a molecule number as an
argument.
</p>
</dd></dl>

<dl>
<dt><a name="index-map_002dmolecule_002dchooser_002dgui"></a>procedure: <strong>map-molecule-chooser-gui</strong></dt>
<dd>
<p>Fire up a map molecule chooser dialog, with a given label and on OK we
call the call-back-fuction with an argument of the chosen molecule
number. 
</p>
<p>chooser-label is a directive to the user such as &quot;Choose a Molecule&quot;
</p>
<p>callback-function is a function that takes a molecule number as an
argument.
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002dchooser_002dand_002dentry"></a>procedure: <strong>generic-chooser-and-entry</strong> <em>chooser-label entry-hint-text defaut-entry-text callback-function . always-dismiss-on-ok-clicked?</em></dt>
<dd><p>if always-dismiss-on-ok-clicked? is false then the dialog is not dismissed if 
callback-function returns #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002dchooser_002dand_002dentry_002dand_002dcheckbutton"></a>procedure: <strong>generic-chooser-and-entry-and-checkbutton</strong> <em>chooser-label entry-hint-text defaut-entry-text check-button-label callback-function . always-dismiss-on-ok-clicked?</em></dt>
<dd><p>as above, plus we also have a check-button and an additional argument in the callback
If check-button-label is false, then don&rsquo;t create a check-button.
</p></dd></dl>

<dl>
<dt><a name="index-generic_002dchooser_002dand_002dfile_002dselector"></a>procedure: <strong>generic-chooser-and-file-selector</strong> <em>chooser-label chooser-filter file-selector-hint default-file-name callback-function</em></dt>
<dd><p>Create a window.
</p>
<p>Return a pair of widgets, a molecule chooser and an entry.
callback-function is a function that takes a molecule number and a
file-name
</p>
<p>chooser-filter is typically valid-map-molecule? or valid-model-molecule?
</p>
</dd></dl>

<dl>
<dt><a name="index-coot_002dmenubar_002dmenu"></a>procedure: <strong>coot-menubar-menu</strong> <em>menu-label</em></dt>
<dd><p>If a menu with label menu-label is not found in the coot main
menubar, then create it and return it. 
If it does exist, simply return it.
</p> 
</dd></dl>

<dl>
<dt><a name="index-add_002dsimple_002dcoot_002dmenu_002dmenuitem"></a>procedure: <strong>add-simple-coot-menu-menuitem</strong> <em>menu menu-item-label activate-function</em></dt>
<dd><p>Given that we have a menu (e.g. one called &quot;Extensions&quot;) provide a
cleaner interface to adding something to it:
</p>
<p>activate-function is a thunk.
</p>
</dd></dl>

<dl>
<dt><a name="index-missing_002datoms_002dgui"></a>procedure: <strong>missing-atoms-gui</strong> <em>imol</em></dt>
<dd><p>Make an interesting things GUI for residues with missing atoms
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002dbuttons_002ddialog"></a>procedure: <strong>generic-buttons-dialog</strong> <em>dialog-name button-list</em></dt>
<dd><p>button-list is a list of pairs (improper list) the first item of
which is the button label text the second item is a lambda
function, what to do when the button is pressed.
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002dinteresting_002dthings"></a>procedure: <strong>generic-interesting-things</strong> <em>imol gui-title-string residue-test-func</em></dt>
<dd><p>Generic interesting things gui: user passes a function that takes 4
args: the chain-id, resno, inscode and residue-serial-number
(should it be needed) and returns either #f or something
interesting (e.g. a label/value).  It is the residue-test-func of
the residue-matching-criteria function.
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002dnumber_002dchooser"></a>procedure: <strong>generic-number-chooser</strong> <em>number-list default-option-value hint-text go-button-label go-function</em></dt>
<dd><p>A gui that makes a generic number chooser the go function is a
lambda function that takes the value of the active menu item - as a
number.
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002dmolecule_002dchooser"></a>procedure: <strong>generic-molecule-chooser</strong> <em>hbox hint-text</em></dt>
<dd><p>pack a hint text and a molecule chooser option menu into the given vbox.
</p>
<p>return the option-menu and model molecule list:
</p></dd></dl>

<dl>
<dt><a name="index-file_002dselector_002dentry"></a>procedure: <strong>file-selector-entry</strong> <em>hbox hint-text</em></dt>
<dd><p>Return an entry, the widget is inserted into the hbox passed to
this function.
</p>
</dd></dl>

<dl>
<dt><a name="index-cootaneer_002dgui"></a>procedure: <strong>cootaneer-gui</strong> <em>imol</em></dt>
<dd><p>Cootaneer gui
</p></dd></dl>

<dl>
<dt><a name="index-view_002dsaver_002dgui"></a>procedure: <strong>view-saver-gui</strong></dt>
<dd><p>The gui for saving views
</p></dd></dl>

<dl>
<dt><a name="index-add_002dview_002dto_002dviews_002dpanel"></a>procedure: <strong>add-view-to-views-panel</strong> <em>view-name view-number</em></dt>
</dl>

<dl>
<dt><a name="index-dialog_002dbox_002dof_002dbuttons_002dwith_002dcheck_002dbutton"></a>procedure: <strong>dialog-box-of-buttons-with-check-button</strong> <em>window-name geometry buttons close-button-label check-button-label check-button-func check-button-is-initially-on-flag</em></dt>
<dd><p>If check-button-label is #f, don&rsquo;t make one, otherwise create with with
the given label and &quot;on&quot; state.
</p>
</dd></dl>

<dl>
<dt><a name="index-dialog_002dbox_002dof_002dpairs_002dof_002dbuttons"></a>procedure: <strong>dialog-box-of-pairs-of-buttons</strong> <em>imol window-name geometry buttons close-button-label</em></dt>
<dd><p>geometry is an improper list of ints
buttons is a list of: (list (list button-1-label button-1-action
                                  button-2-label button-2-action))
The button-1-action function takes as an argument the imol
The button-2-action function takes as an argument the imol
</p>
</dd></dl>

<dl>
<dt><a name="index-views_002dpanel_002dgui"></a>procedure: <strong>views-panel-gui</strong></dt>
<dd><p>A gui showing views:
</p></dd></dl>

<dl>
<dt><a name="index-nudge_002dscreen_002dcentre_002dgui"></a>procedure: <strong>nudge-screen-centre-gui</strong></dt>
<dd><p>nudge screen centre box.  Useful when Ctrl left-mouse has been
taken over by another function.
</p>
</dd></dl>

<dl>
<dt><a name="index-make_002ddifference_002dmap_002dgui"></a>procedure: <strong>make-difference-map-gui</strong></dt>
<dd><p>A gui to make a difference map (from arbitrarily gridded maps
(that&rsquo;s it&rsquo;s advantage))
</p>
</dd></dl>

<dl>
<dt><a name="index-cis_002dpeptides_002dgui"></a>procedure: <strong>cis-peptides-gui</strong> <em>imol</em></dt>
<dd><p>A GUI to display all the CIS peptides and navigate to them.
</p>
</dd></dl>

<dl>
<dt><a name="index-associate_002dpir_002dwith_002dmolecule_002dgui"></a>procedure: <strong>associate-pir-with-molecule-gui</strong> <em>do-alignment?</em></dt>
<dd><p>Associate the contents of a PIR file with a molecule.  Select file from a GUI.
</p>
</dd></dl>

<dl>
<dt><a name="index-alignment_002dmismatches_002dgui"></a>procedure: <strong>alignment-mismatches-gui</strong> <em>imol</em></dt>
<dd><p>Make a box-of-buttons GUI for the various modifications that need
to be made to match the model sequence to the assigned sequence(s).
</p>
<p>Call this when the associated sequence(s) have been read in already.
</p>
</dd></dl>

<dl>
<dt><a name="index-wrapper_002dalignment_002dmismatches_002dgui"></a>procedure: <strong>wrapper-alignment-mismatches-gui</strong> <em>imol</em></dt>
<dd><p>Wrapper in that we test if there have been sequence(s) assigned to
imol before we look for the sequence mismatches
</p>
</dd></dl>

<dl>
<dt><a name="index-run_002dclustalw_002dalignment"></a>procedure: <strong>run-clustalw-alignment</strong> <em>imol chain-id target-sequence-pir-file</em></dt>
<dd><p>Use clustalw to do the alignment. Then mutate using that alignement.
</p>
</dd></dl>

<dl>
<dt><a name="index-residue_002drange_002dgui"></a>procedure: <strong>residue-range-gui</strong></dt>
<dd><p>(Notice that we are not dealing with insertion codes).
</p>
</dd></dl>

<dl>
<dt><a name="index-user_002dmods_002dgui"></a>procedure: <strong>user-mods-gui</strong> <em>imol pdb-file-name</em></dt>
<dd><p>USER MODS gui
</p>
</dd></dl>

<dl>
<dt><a name="index-rename_002dresidue_002dgui"></a>procedure: <strong>rename-residue-gui</strong></dt>
<dd><p>simple rename residue GUI
</p></dd></dl>

<dl>
<dt><a name="index-water_002dcoordination_002dgui"></a>procedure: <strong>water-coordination-gui</strong></dt>
<dd><p>close-button
</p>
</dd></dl>

<dl>
<dt><a name="index-pepflips_002dby_002ddifference_002dmap_002dgui"></a>procedure: <strong>pepflips-by-difference-map-gui</strong></dt>
<dd><p>interface to the difference-map based pepflip finder
</p></dd></dl>

<dl>
<dt><a name="index-min_002dmax_002dresidues_002dfrom_002datom_002dspecs"></a>procedure: <strong>min-max-residues-from-atom-specs</strong> <em>specs</em></dt>
<dd><p>return a list, or #f (e.g. if not in same chain and molecule)
</p>
</dd></dl>

<dl>
<dt><a name="index-_002adb_002dloop_002dpreserve_002dresidue_002dnames_002a"></a>procedure: <strong>*db-loop-preserve-residue-names*</strong></dt>
<dd><p>by default, rename loop residues to UNK.  If scheme true, then
leave them as the residue names found in the database.
</p></dd></dl>
<hr>
<a name="hello"></a>
<div class="header">
<p>
Next: <a href="#coot_002dcrash_002dcatcher" accesskey="n" rel="next">coot-crash-catcher</a>, Previous: <a href="#coot_002dgui" accesskey="p" rel="previous">coot-gui</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="hello-1"></a>
<h3 class="section">14.30 hello</h3>

<dl>
<dt><a name="index-first_002dnon_002dtrivial_002dname"></a>procedure: <strong>first-non-trivial-name</strong></dt>
<dd><p>Primarily for Indian Names.
</p>
<p>Say we are given str: (list &quot;M.&quot; &quot;D.&quot; &quot;Albert&quot; &quot;Dorkins&quot;).  We want
to return &quot;&quot;Albert&quot; not &quot;M.&quot;)  We reject names that are one
character long and names that are 2 characters long that end in
&quot;.&quot;.  So, &quot;M.&quot; is rejected, but &quot;Ma&quot; is not.
</p>
<p>An excercise for the committed is to also reject run-together
dotted initials such as &quot;D.K.&quot;.  I was sufficiently committed.
</p>
</dd></dl>
<hr>
<a name="coot_002dcrash_002dcatcher"></a>
<div class="header">
<p>
Next: <a href="#coot_002dutils" accesskey="n" rel="next">coot-utils</a>, Previous: <a href="#hello" accesskey="p" rel="previous">hello</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot_002dcrash_002dcatcher-1"></a>
<h3 class="section">14.31 coot-crash-catcher</h3>

<dl>
<dt><a name="index-filter"></a>procedure: <strong>filter</strong> <em>fn ls</em></dt>
<dd><p>Basic scheme function, filter the objects in list ls by function
fn.  e.g. (filter even? (list 0 1 2 3) -&gt; &rsquo;(0 2)
</p>
</dd></dl>

<dl>
<dt><a name="index-string_002dconcatenate"></a>procedure: <strong>string-concatenate</strong> <em>ls</em></dt>
<dd><p>simple scheme functions to concat the strings in ls (ls must contain
only strings)
</p>
</dd></dl>

<dl>
<dt><a name="index-directory_002dfiles-1"></a>procedure: <strong>directory-files</strong> <em>dir</em></dt>
<dd><p>The following functions from PLEAC (guile version thereof of course).
</p>
<p>or define a utility function for this
</p></dd></dl>

<dl>
<dt><a name="index-glob-1"></a>procedure: <strong>glob</strong> <em>pat dir</em></dt>
<dd><p>return a list of file names that match pattern pat in directory dir.
</p></dd></dl>

<dl>
<dt><a name="index-run_002dcommand_002fstrings-1"></a>procedure: <strong>run-command/strings</strong> <em>cmd args data-list</em></dt>
<dd><p>Return the strings screen output of cmd (reversed) or #f if command
was not found
</p>
</dd></dl>

<dl>
<dt><a name="index-command_002din_002dpath_003f-1"></a>procedure: <strong>command-in-path?</strong> <em>cmd</em></dt>
<dd><p>Return #t or #f:
</p></dd></dl>

<dl>
<dt><a name="index-string_002d_003elist_002dof_002dstrings-1"></a>procedure: <strong>string-&gt;list-of-strings</strong> <em>str</em></dt>
<dd><p>Return a list if str is a string, else return &rsquo;()
</p>
</dd></dl>

<dl>
<dt><a name="index-shell_002dcommand_002dto_002dstring"></a>procedure: <strong>shell-command-to-string</strong> <em>cmd</em></dt>
<dd><p>code from thi &lt;ttn at mingle.glug.org&gt;
</p>
<p>run command and put the output into a string and return
it. (c.f. <code>run-command/strings</code>)
</p>
</dd></dl>

<dl>
<dt><a name="index-make_002dgdb_002dscript"></a>procedure: <strong>make-gdb-script</strong></dt>
</dl>

<dl>
<dt><a name="index-get_002dgdb_002dstrings"></a>procedure: <strong>get-gdb-strings</strong></dt>
<dd><p>return #f or list of strings
</p></dd></dl>

<dl>
<dt><a name="index-make_002dgui"></a>procedure: <strong>make-gui</strong></dt>
<dd><p>gui
</p></dd></dl>
<hr>
<a name="coot_002dutils"></a>
<div class="header">
<p>
Next: <a href="#contact_002dscore_002disolated_002dligand" accesskey="n" rel="next">contact-score-isolated-ligand</a>, Previous: <a href="#coot_002dcrash_002dcatcher" accesskey="p" rel="previous">coot-crash-catcher</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot_002dutils-1"></a>
<h3 class="section">14.32 coot-utils</h3>

<dl>
<dt><a name="index-_002aannotations_002a"></a>procedure: <strong>*annotations*</strong></dt>
<dd><p>3D annotations - a bit of a hack currently
</p></dd></dl>

<dl>
<dt><a name="index-_002adefault_002dball_002dand_002dstick_002dselection_002a"></a>procedure: <strong>*default-ball-and-stick-selection*</strong></dt>
<dd><p>used in Extensions -&gt; Representation -&gt; Ball &amp; Stick
</p></dd></dl>

<dl>
<dt><a name="index-rigid_002dbody_002drefine_002dby_002dresidue_002dranges"></a>procedure: <strong>rigid-body-refine-by-residue-ranges</strong></dt>
<dd><p>rigid body refine using residue ranges.  Takes 2 arguments, the
first is the molecule number, the second is a list of
residue-ranges.  A residue range is 
<code>(list <em>chain-id resno-start resno-end</em>)</code>.
</p>
</dd></dl>

<dl>
<dt><a name="index-find_002daligned_002dresidue_002dtype"></a>procedure: <strong>find-aligned-residue-type</strong></dt>
<dd><p>add terminal residue is the normal thing we do with an aligned
sequence, but also we can try ton find the residue type of a
residue in the middle of the chain that is modelled as an ALA, say.
</p>
</dd></dl>

<dl>
<dt><a name="index-all_002dtrue_003f"></a>procedure: <strong>all-true?</strong> <em>ls</em></dt>
<dd><p>not tail-recursive
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002ddirectory"></a>procedure: <strong>get-directory</strong> <em>dir-name</em></dt>
<dd><p>make the directory and return the directory name.  If you cant make
it directly in this directory try to make it in $HOME.  Return #f if
complete failure.  e.g. coot-ccp4 or coot-backup
</p>
</dd></dl>

<dl>
<dt><a name="index-molecule_002dhas_002dhydrogens_003f"></a>procedure: <strong>molecule-has-hydrogens?</strong> <em>imol</em></dt>
<dd><p>schemify function
</p>
</dd></dl>

<dl>
<dt><a name="index-add_002dhydrogens_002dusing_002drefmac_002dinner"></a>procedure: <strong>add-hydrogens-using-refmac-inner</strong> <em>imol in-file-name out-file-name</em></dt>
<dd><p>out-file-name has been written, in-file-name needs to be read.
</p>
</dd></dl>

<dl>
<dt><a name="index-post_002dmanipulation_002dhook-1"></a>procedure: <strong>post-manipulation-hook</strong></dt>
</dl>

<dl>
<dt><a name="index-post_002dset_002drotation_002dcentre_002dhook"></a>procedure: <strong>post-set-rotation-centre-hook</strong></dt>
<dd><p>do something based on the active residue (presumably)
</p></dd></dl>

<dl>
<dt><a name="index-pre_002drelease_003f"></a>procedure: <strong>pre-release?</strong></dt>
<dd><p>Return a boolean
</p>
</dd></dl>

<dl>
<dt><a name="index-molecule_002dnumber_002dlist"></a>procedure: <strong>molecule-number-list</strong></dt>
<dd><p>Return a list of molecule numbers (closed and open) The elements of
the returned list need to be tested against
is-valid-model-molecule?
</p>
</dd></dl>

<dl>
<dt><a name="index-undisplay_002dall_002dmaps_002dexcept"></a>procedure: <strong>undisplay-all-maps-except</strong> <em>imol-map</em></dt>
<dd><p>c.f. graphics_info_t::undisplay_all_model_molecules_except(int imol)
</p></dd></dl>

<dl>
<dt><a name="index-just_002done_002dor_002dnext_002dmap"></a>procedure: <strong>just-one-or-next-map</strong></dt>
<dd><p>is isn&rsquo;t quite because one of the options is &quot;all&quot;
</p></dd></dl>

<dl>
<dt><a name="index-first_002dn"></a>procedure: <strong>first-n</strong> <em>n ls</em></dt>
<dd><p>first n fields of ls. if length ls is less than n, return ls.
if ls is not a list, return ls.  If n is negative, return ls.
</p>
</dd></dl>

<dl>
<dt><a name="index-directory_002dis_002dmodifiable_003f"></a>procedure: <strong>directory-is-modifiable?</strong> <em>prefix-dir</em></dt>
<dd><p>Test for prefix-dir (1) being a string (2) existing (3) being a
directory (4) modifiable by user (ie. u+rwx).  prefix-dir must be a
string.
</p>
<p>Return #t or #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-absolutify"></a>procedure: <strong>absolutify</strong> <em>file-name</em></dt>
<dd><p>return an absolute file-name for file-name or #f
</p>
</dd></dl>

<dl>
<dt><a name="index-file_002dname_002ddirectory"></a>procedure: <strong>file-name-directory</strong> <em>file-name</em></dt>
<dd><p>return the directory component of file-name, leave &quot;/&quot; on, if it&rsquo;s
there. Note &quot;x&quot;, &quot;&quot;, &quot;/&quot; -&gt; &quot;&quot;
</p>
</dd></dl>

<dl>
<dt><a name="index-file_002dname_002dfile-1"></a>procedure: <strong>file-name-file</strong> <em>file-name</em></dt>
<dd><p>return the file component of file-name
&quot;x/y&quot; -&gt; &quot;y&quot; 
&quot;x&quot; -&gt; &quot;x&quot;
</p></dd></dl>

<dl>
<dt><a name="index-most_002drecently_002dcreated_002dfile"></a>procedure: <strong>most-recently-created-file</strong></dt>
<dd><p>return #f on no-such-file
</p>
</dd></dl>

<dl>
<dt><a name="index-residue_002dspec_002d_003eatom_002dselection_002dstring"></a>procedure: <strong>residue-spec-&gt;atom-selection-string</strong> <em>centre-residue-spec</em></dt>
<dd><p>Convert a residue-spec to an mmdb atom selection string.
</p>
</dd></dl>

<dl>
<dt><a name="index-residue_002datom_002d_003eatom_002dname"></a>procedure: <strong>residue-atom-&gt;atom-name</strong> <em>ra</em></dt>
<dd><p>residue-info atom
</p></dd></dl>

<dl>
<dt><a name="index-residue_002datom_002d_003ealt_002dconf"></a>procedure: <strong>residue-atom-&gt;alt-conf</strong> <em>ra</em></dt>
<dd><p>residue-info atom
</p></dd></dl>

<dl>
<dt><a name="index-residue_002datom_002d_003eoccupancy"></a>procedure: <strong>residue-atom-&gt;occupancy</strong> <em>ra</em></dt>
<dd><p>residue-info atom
</p></dd></dl>

<dl>
<dt><a name="index-residue_002datom_002d_003eposition"></a>procedure: <strong>residue-atom-&gt;position</strong> <em>ra</em></dt>
<dd><p>residue-info atom
</p></dd></dl>

<dl>
<dt><a name="index-residue_002datom_002d_003eatom_002dspec"></a>procedure: <strong>residue-atom-&gt;atom-spec</strong> <em>ra chain-id res-no ins-code</em></dt>
<dd><p>residue-info atom needs other parameters to make a spec for an atom
</p></dd></dl>

<dl>
<dt><a name="index-residue_002dspec_002d_003echain_002did"></a>procedure: <strong>residue-spec-&gt;chain-id</strong> <em>rs</em></dt>
<dd><p>residue spec (e.g. from residue-near-residue)
</p></dd></dl>

<dl>
<dt><a name="index-residue_002dspec_002dless_002dthan"></a>procedure: <strong>residue-spec-less-than</strong> <em>spec-1 spec-2</em></dt>
<dd><p>for sorting residue specs
</p></dd></dl>

<dl>
<dt><a name="index-map_002dmolecule_002dlist"></a>procedure: <strong>map-molecule-list</strong></dt>
<dd><p>Return a list of molecules that are maps
</p>
</dd></dl>

<dl>
<dt><a name="index-model_002dmolecule_002dlist"></a>procedure: <strong>model-molecule-list</strong></dt>
<dd><p>Return a list of molecules that are maps
</p>
</dd></dl>

<dl>
<dt><a name="index-shelx_002dmolecule_003f"></a>procedure: <strong>shelx-molecule?</strong> <em>imol</em></dt>
<dd><p>Return #t (#f) if <var>imol</var> is (isn&rsquo;t) a shelx molecule.
</p>
</dd></dl>

<dl>
<dt><a name="index-set_002dvirtual_002dtrackball_002dtype"></a>procedure: <strong>set-virtual-trackball-type</strong> <em>type</em></dt>
<dd><p>Set the virtual trackball behaviour.
</p>
<p>trackball <var>type</var> is a symbol: either &rsquo;flat or &rsquo;spherical-surface.
</p>
</dd></dl>

<dl>
<dt><a name="index-list_002dof_002dstrings_003f"></a>procedure: <strong>list-of-strings?</strong> <em>ls</em></dt>
<dd><p>Is <var>ls</var> a list of strings?  Return #t or #f
</p>
</dd></dl>

<dl>
<dt><a name="index-coot_002dreplace_002dstring"></a>procedure: <strong>coot-replace-string</strong> <em>string-in target replacing-str</em></dt>
<dd><p>(coot-replace-string &quot;one two three&quot; &quot; &quot; &quot;_&quot;) -&gt; &quot;one_two_three&quot;
</p>
</dd></dl>

<dl>
<dt><a name="index-string_002dappend_002dwith_002dspaces"></a>procedure: <strong>string-append-with-spaces</strong> <em>ls</em></dt>
<dd><p>string concat with spaces, <var>ls</var> must be a list of strings.
</p>
</dd></dl>

<dl>
<dt><a name="index-rotation_002dcentre"></a>procedure: <strong>rotation-centre</strong></dt>
<dd><p>The screen centre.
</p>
<p>return the rotation centre as a 3 membered list of numbers
</p>
</dd></dl>

<dl>
<dt><a name="index-number_002dlist"></a>procedure: <strong>number-list</strong> <em>a b</em></dt>
<dd><p>Make list of integers, <var>a</var> to <var>b</var>: eg (number-list 2 5) -&gt;
(2 3 4 5)
</p>
</dd></dl>

<dl>
<dt><a name="index-string_002dmember_003f"></a>procedure: <strong>string-member?</strong> <em>atom ls</em></dt>
<dd><p>ls must be a list of strings, atom must be a string.
</p>
<p>return either #t or #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-member_003f"></a>procedure: <strong>member?</strong> <em>atom ls</em></dt>
</dl>

<dl>
<dt><a name="index-range"></a>procedure: <strong>range</strong> <em>first . second</em></dt>
<dd><p>range: works like the eponymous python function
e.g. (range 3) -&gt; &rsquo;(0, 1, 2)
e.g. (range 1 3) -&gt; &rsquo;(1, 2)
</p>
</dd></dl>

<dl>
<dt><a name="index-shell_002dcommand_002dto_002dstring-1"></a>procedure: <strong>shell-command-to-string</strong> <em>cmd</em></dt>
<dd><p>code from thi &lt;ttn at mingle.glug.org&gt;
</p>
<p>run command and put the output into a string and return
it. (c.f. <code>run-command/strings</code>)
</p>
</dd></dl>

<dl>
<dt><a name="index-shell_002dcommand_002dto_002dfile_002dwith_002ddata"></a>procedure: <strong>shell-command-to-file-with-data</strong> <em>cmd file-name data-list</em></dt>
<dd><p>run <var>cmd</var> putting output to <var>file-name</var> and reading
commands data from the list of strings <var>data-list</var>.
</p>
</dd></dl>

<dl>
<dt><a name="index-command_002din_002dpath_003f-2"></a>procedure: <strong>command-in-path?</strong> <em>f</em></dt>
<dd><p>Is the command <var>f</var> in the path?
return #t or #f.
</p>
<p>Thank you for this, rixed at happyleptic.org
</p></dd></dl>

<dl>
<dt><a name="index-command_002din_002dpath_002dor_002dabsolute_003f"></a>procedure: <strong>command-in-path-or-absolute?</strong> <em>cmd</em></dt>
<dd><p>Return #t or #f
</p>
</dd></dl>

<dl>
<dt><a name="index-goosh_002dcommand-1"></a>procedure: <strong>goosh-command</strong> <em>cmd args data-list log-file-name screen-output-also?</em></dt>
<dd><p>Where cmd is e.g. &quot;refmac&quot; 
      args is (list &quot;HKLIN&quot; &quot;thing.mtz&quot;)
      log-file-name is &quot;refmac.log&quot;      
      data-list is (list &quot;HEAD&quot; &quot;END&quot;)
</p>
<p>Return the exist status e.g. 0 or 1.
</p>
</dd></dl>

<dl>
<dt><a name="index-goosh_002dcommand_002dwith_002dfile_002dinput"></a>procedure: <strong>goosh-command-with-file-input</strong> <em>cmd args input-file log-file-name</em></dt>
<dd><p>run commands from an input file.
</p>
</dd></dl>

<dl>
<dt><a name="index-run_002dcommand_002fstrings-2"></a>procedure: <strong>run-command/strings</strong> <em>cmd args data-list</em></dt>
<dd><p>Return the strings screen output of cmd or #f if command was not found
</p>
</dd></dl>

<dl>
<dt><a name="index-close_002dfloat_003f"></a>procedure: <strong>close-float?</strong> <em>x1 x2</em></dt>
<dd><p>Crude test to see of 2 floats are the same (more or less).
Used in a greg test after setting the atom position.
</p>
</dd></dl>

<dl>
<dt><a name="index-strip_002dspaces"></a>procedure: <strong>strip-spaces</strong> <em>str</em></dt>
<dd><p>if passed a string, return a string with no spaces,
else return #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-strip_002dleading_002dspaces-1"></a>procedure: <strong>strip-leading-spaces</strong> <em>str</em></dt>
<dd><p>&quot;  53&quot; -&gt; &quot;53&quot;, &quot; &quot; -&gt; &quot;&quot;
</p></dd></dl>

<dl>
<dt><a name="index-string_002dappend_002dwith_002dstring"></a>procedure: <strong>string-append-with-string</strong> <em>str-ls tag-str</em></dt>
<dd><p>Append strings with tag-str between them
</p>
</dd></dl>

<dl>
<dt><a name="index-strip_002dextension-1"></a>procedure: <strong>strip-extension</strong> <em>s</em></dt>
<dd><p>&quot;a.b.res&quot; -&gt; &quot;a.b&quot;
file-name-sans-extension
</p>
</dd></dl>

<dl>
<dt><a name="index-file_002dname_002dextension"></a>procedure: <strong>file-name-extension</strong></dt>
<dd><p>What is the extension of file-name?
</p>
<p>&quot;a.pdb&quot; -&gt; &quot;pdb&quot;
&quot;&quot; -&gt; &quot;&quot;
</p>
</dd></dl>

<dl>
<dt><a name="index-add_002dtmp_002dextension_002dto"></a>procedure: <strong>add-tmp-extension-to</strong></dt>
<dd><p>e.g. &quot;a.pdb&quot; -&gt; &quot;a-tmp.pdb&quot;
</p>
</dd></dl>

<dl>
<dt><a name="index-file_002dname_002dsans_002dextension"></a>procedure: <strong>file-name-sans-extension</strong></dt>
<dd><p>Same function as strip-extension, different name, as per scsh, in fact.
</p>
</dd></dl>

<dl>
<dt><a name="index-strip_002dpath"></a>procedure: <strong>strip-path</strong> <em>s</em></dt>
<dd><p>/a/b.t -&gt; b.t   d/e.ext -&gt; e.ext
file-name-sans-path
</p>
</dd></dl>

<dl>
<dt><a name="index-slash_002dstart_003f"></a>procedure: <strong>slash-start?</strong> <em>s</em></dt>
<dd><p>does s start with a &quot;/&quot; ?
return #t or #f
</p>
</dd></dl>

<dl>
<dt><a name="index-string_002dconcatenate-1"></a>procedure: <strong>string-concatenate</strong> <em>ls</em></dt>
<dd><p>simple scheme functions to concat the strings in ls (ls must contain
only strings)
</p>
</dd></dl>

<dl>
<dt><a name="index-unique_002ddate_002ftime_002dstr"></a>procedure: <strong>unique-date/time-str</strong></dt>
<dd><p>return a string that contains the date/time
e.g.	&quot;2006-01-02_2216.03&quot;
</p>
</dd></dl>

<dl>
<dt><a name="index-every_002dnth"></a>procedure: <strong>every-nth</strong> <em>ls n</em></dt>
<dd><p>return a list that has only every-nth members; 
e.g. <code>(every-nth '(0 1 2 3 4 5 6 7 8) 2)</code> -&gt; &rsquo;(0 2 3 6 8)
     <code>(every-nth '(0 1 2 3 4 5 6 7 8) 3)</code> -&gt; &rsquo;(0 3 6)
</p>
<p><var>n</var> must be positive
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002datom_002dfrom_002dresidue"></a>procedure: <strong>get-atom-from-residue</strong> <em>atom-name residue-atoms alt-conf</em></dt>
<dd><p>now useful in testing
</p>
<p>residue-atoms must be a list
</p></dd></dl>

<dl>
<dt><a name="index-get_002datom"></a>procedure: <strong>get-atom</strong> <em>imol chain-id resno ins-code atom-name . alt-conf</em></dt>
<dd><p>Return an atom info or #f (if atom not found).
</p>
</dd></dl>

<dl>
<dt><a name="index-multi_002dread_002dpdb"></a>procedure: <strong>multi-read-pdb</strong> <em>glob-pattern dir</em></dt>
<dd><p>multi-read-pdb reads all the files matching
<code><em>glob-pattern</em></code> in
directory <code><em>dir</em></code>.  Typical usage of this might be:
<code>(multi-read-pdb &quot;a*.pdb&quot; &quot;.&quot;)</code>
</p> 
</dd></dl>

<dl>
<dt><a name="index-read_002dpdb_002dall"></a>procedure: <strong>read-pdb-all</strong></dt>
<dd><p>read-pdb-all reads all the &quot;*.pdb&quot; files in the current directory.
</p>
</dd></dl>

<dl>
<dt><a name="index-string_002d_003elist_002dof_002dstrings-2"></a>procedure: <strong>string-&gt;list-of-strings</strong> <em>str</em></dt>
<dd><p>Return a list if str is a string, else return &rsquo;()
</p>
</dd></dl>

<dl>
<dt><a name="index-append_002ddir_002dfile"></a>procedure: <strong>append-dir-file</strong> <em>dir-name file-name</em></dt>
<dd><p>In a laughable attempt to minimise system dependence.
</p>
</dd></dl>

<dl>
<dt><a name="index-append_002ddir_002ddir"></a>procedure: <strong>append-dir-dir</strong> <em>dir-name sub-dir-name</em></dt>
<dd><p>Similarly attempting to minimise system dependence.
</p>
</dd></dl>

<dl>
<dt><a name="index-join_002ddir_002dfile"></a>procedure: <strong>join-dir-file</strong> <em>path-elements</em></dt>
<dd><p>the first elements are directories, the last is the file name
c.f. os.path.join
</p>
</dd></dl>

<dl>
<dt><a name="index-directory_002das_002dfile_002dname-1"></a>procedure: <strong>directory-as-file-name</strong> <em>dir</em></dt>
<dd><p>remove any trailing /s
</p>
</dd></dl>

<dl>
<dt><a name="index-is_002ddirectory_003f"></a>procedure: <strong>is-directory?</strong> <em>file-name</em></dt>
<dd><p>return #t or #f depending on if file-name (which must be a string)
is a directory.
</p>
</dd></dl>

<dl>
<dt><a name="index-coot_002dmkdir"></a>procedure: <strong>coot-mkdir</strong> <em>dir-name</em></dt>
<dd><p>return the dir-name on success.
</p>
<p>return #f if dir-name is a file or we can&rsquo;t do the mkdir.
</p>
</dd></dl>

<dl>
<dt><a name="index-directory_002dfiles-2"></a>procedure: <strong>directory-files</strong> <em>dir</em></dt>
<dd><p>The following functions from PLEAC (guile version thereof of course).
</p>
<p>Return: a list of files in the given directory
</p>
</dd></dl>

<dl>
<dt><a name="index-glob-2"></a>procedure: <strong>glob</strong> <em>pat dir</em></dt>
<dd><p>return a list of file names that match pattern pat in directory dir.
</p>
</dd></dl>

<dl>
<dt><a name="index-useful_002dglob-1"></a>procedure: <strong>useful-glob</strong> <em>dir pat</em></dt>
<dd><p>prepend the files with the directory dir
</p>
<p>(note that the args are reversed to &quot;natural&quot; order (which is the
reverse of glob args)
</p>
</dd></dl>

<dl>
<dt><a name="index-view_002dmatrix"></a>procedure: <strong>view-matrix</strong></dt>
<dd><p>return the view matrix (useful for molscript, perhaps).
</p>
</dd></dl>

<dl>
<dt><a name="index-view_002dquaternion"></a>procedure: <strong>view-quaternion</strong></dt>
<dd><p>return the view quaternion
</p>
</dd></dl>

<dl>
<dt><a name="index-add_002dview"></a>procedure: <strong>add-view</strong> <em>position quaternion zoom view-name</em></dt>
<dd><p>Return the view number 
</p>
</dd></dl>

<dl>
<dt><a name="index-matrix_002d_003equaternion-1"></a>procedure: <strong>matrix-&gt;quaternion</strong></dt>
<dd><p>Convert a view matrix to a view quaternion to set Coot view internals.
</p>
</dd></dl>

<dl>
<dt><a name="index-set_002dview_002dmatrix-1"></a>procedure: <strong>set-view-matrix</strong> <em>m00 m10 m20 m01 m11 m21 m02 m12 m22</em></dt>
<dd><p>Set the view matrix using matrix-&gt;quaternion. 
</p>
<p>Useful for using a view matrix from another program, perhaps.
</p>
</dd></dl>

<dl>
<dt><a name="index-molecule_002dcentre"></a>procedure: <strong>molecule-centre</strong> <em>imol</em></dt>
<dd><p>Return the molecule centre as a list of 3 numbers.
</p>
<p>Note: mol-cen could contain values less than -9999.  
</p>
</dd></dl>

<dl>
<dt><a name="index-move_002dmolecule_002dto_002dscreen_002dcentre"></a>procedure: <strong>move-molecule-to-screen-centre</strong> <em>imol</em></dt>
<dd><p>Move the centre of molecule number imol to the current screen centre
</p>
</dd></dl>

<dl>
<dt><a name="index-move_002dmolecule_002dhere"></a>procedure: <strong>move-molecule-here</strong></dt>
<dd><p>This is a short name for the above.
</p>
</dd></dl>

<dl>
<dt><a name="index-move_002dmolecule_002dto_002dscreen_002dcenter"></a>procedure: <strong>move-molecule-to-screen-center</strong></dt>
<dd><p>this is an americanism
</p>
</dd></dl>

<dl>
<dt><a name="index-identity_002dmatrix"></a>procedure: <strong>identity-matrix</strong></dt>
<dd><p>Return a nine-membered list of numbers.
</p>
</dd></dl>

<dl>
<dt><a name="index-translation"></a>procedure: <strong>translation</strong> <em>axis length</em></dt>
<dd><p>e.g. (translation &rsquo;x 2)
 -&gt; &rsquo;(2 0 0)
Return: &quot;scheme false&quot; on error
</p>
</dd></dl>

<dl>
<dt><a name="index-rotate_002dabout_002dscreen_002daxis"></a>procedure: <strong>rotate-about-screen-axis</strong> <em>axis degrees</em></dt>
<dd><p>Rotate degrees about screen axis, where axis is either &rsquo;x, &rsquo;y or &rsquo;z.
</p>
</dd></dl>

<dl>
<dt><a name="index-toggle_002ddisplay_002dmap"></a>procedure: <strong>toggle-display-map</strong> <em>imol idummy</em></dt>
<dd><p>Support for old toggle functions.  (consider instead the raw
functions use the direct set_displayed functions).
</p>
</dd></dl>

<dl>
<dt><a name="index-toggle_002ddisplay_002dmol"></a>procedure: <strong>toggle-display-mol</strong> <em>imol</em></dt>
<dd><p>toggle the display of imol
</p>
</dd></dl>

<dl>
<dt><a name="index-toggle_002dactive_002dmol"></a>procedure: <strong>toggle-active-mol</strong> <em>imol</em></dt>
<dd><p>toggle the active state (clickability) of imol
</p>
</dd></dl>

<dl>
<dt><a name="index-scheme_002drepresentation"></a>procedure: <strong>scheme-representation</strong> <em>imol</em></dt>
<dd><p>return a scheme representation of molecule imol, or #f if we can&rsquo;t
do it (imol is a map, say).
</p>
</dd></dl>

<dl>
<dt><a name="index-reorder_002dchains"></a>procedure: <strong>reorder-chains</strong> <em>imol</em></dt>
<dd><p>reorder chains
</p>
</dd></dl>

<dl>
<dt><a name="index-transform_002dcoords_002dmolecule"></a>procedure: <strong>transform-coords-molecule</strong> <em>imol rtop</em></dt>
<dd><p>transform a coordinates molecule by a coot-rtop (which is a SCM
expression of a clipper::RTop), i.e. a list of a 9-element list and
a 3 element list. e.g. (list (list 1 0 0 0 1 0 0 0 1) (list 4.5 0.4
1.2)).
</p>
</dd></dl>

<dl>
<dt><a name="index-transform_002dmap"></a>procedure: <strong>transform-map</strong></dt>
<dd><p><code>(transform-map imol mat trans about-pt radius space-group cell)</code>
</p>
<p>where space-group is a HM-symbol and cell is a list of 6
parameters, where the cell angles are in degrees.
</p>
<p>or <code>(transform-map imol trans about-pt radius)</code> for a simple translation
</p>
<p>or <code>(transform-map imol trans radius)</code> when using the default
rotation-centre as the about-pt
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002dfirst_002dncs_002dmaster_002dchain"></a>procedure: <strong>get-first-ncs-master-chain</strong></dt>
<dd><p>return the NCS master of the first molecule that has NCS.
</p>
<p>return &quot;&quot; on fail to find an ncs chain
</p>
</dd></dl>

<dl>
<dt><a name="index-transform_002dmap_002dusing_002dlsq_002dmatrix"></a>procedure: <strong>transform-map-using-lsq-matrix</strong> <em>imol-ref ref-chain ref-resno-start ref-resno-end imol-mov mov-chain mov-resno-start mov-resno-end imol-map about-pt radius</em></dt>
<dd><p>Remember, that now the about-pt is the &quot;to&quot; point, i.e. the maps are brought from 
somewhere else and generated about the about-pt.
</p>
</dd></dl>

<dl>
<dt><a name="index-brighten_002dmap"></a>procedure: <strong>brighten-map</strong> <em>imol scale-factor</em></dt>
<dd><p>Make the imol-th map brighter.
</p>
</dd></dl>

<dl>
<dt><a name="index-brighten_002dmaps"></a>procedure: <strong>brighten-maps</strong></dt>
<dd><p>Make all maps brighter
</p>
</dd></dl>

<dl>
<dt><a name="index-darken_002dmaps"></a>procedure: <strong>darken-maps</strong></dt>
<dd><p>Make all maps darker.
</p>
</dd></dl>

<dl>
<dt><a name="index-chain_002dids"></a>procedure: <strong>chain-ids</strong> <em>imol</em></dt>
<dd><p>Return a list of chain ids for given molecule number <var>imol</var>.
return empty list on error
</p>
</dd></dl>

<dl>
<dt><a name="index-is_002dsolvent_002dchain_003f"></a>procedure: <strong>is-solvent-chain?</strong> <em>imol chain-id</em></dt>
<dd><p>convert from interface name to schemish name
</p>
<p>Return #t or #f. 
</p>
</dd></dl>

<dl>
<dt><a name="index-valid_002dmodel_002dmolecule_003f"></a>procedure: <strong>valid-model-molecule?</strong> <em>imol</em></dt>
<dd><p>schemey interface to eponymous scripting interface function.
Return scheme true or false
</p>
</dd></dl>

<dl>
<dt><a name="index-valid_002dmap_002dmolecule_003f"></a>procedure: <strong>valid-map-molecule?</strong> <em>imol</em></dt>
<dd><p>schemey interface to eponymous scripting interface function.
Return scheme true or false.
</p>
</dd></dl>

<dl>
<dt><a name="index-valid_002drefinement_002dmap_003f"></a>procedure: <strong>valid-refinement-map?</strong></dt>
<dd><p>Return #t or #f
</p>
</dd></dl>

<dl>
<dt><a name="index-shelx_002dmolecule_003f-1"></a>procedure: <strong>shelx-molecule?</strong> <em>imol</em></dt>
<dd><p>schemey interface to shelx molecule test
</p>
<p>Return #t or #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-is_002ddifference_002dmap_003f"></a>procedure: <strong>is-difference-map?</strong> <em>imol-map</em></dt>
<dd><p>Return #t or #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-residue_002dexists_003f"></a>procedure: <strong>residue-exists?</strong> <em>imol chain-id resno ins-code</em></dt>
<dd><p>Does residue resno with insertion code ins-code of chain chain-id
and in molecule number imol exist?  
</p>
<p>Return #t or #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-residue_002dhas_002dhetatms_003f"></a>procedure: <strong>residue-has-hetatms?</strong> <em>imol chain-id res-no ins-code</em></dt>
<dd><p>Does the residue contain hetatoms? 
Return #t or #f.
</p>
</dd></dl>

<dl>
<dt><a name="index-atom_002dspecs"></a>procedure: <strong>atom-specs</strong> <em>imol chain-id resno ins-code atom-name alt-conf</em></dt>
<dd><p>Return as a list the occupancy temperature-factor element x y z coordinates
of the given atom.
(the x,y,z are in Cartesian Angstroms).
</p>
<p>on error (e.g. atom not found) return #f
</p>
</dd></dl>

<dl>
<dt><a name="index-guess_002drefinement_002dmap"></a>procedure: <strong>guess-refinement-map</strong></dt>
<dd><p>return a guess at the map to be refined (usually called after
imol-refinement-map returns -1)
</p>
</dd></dl>

<dl>
<dt><a name="index-target_002dauto_002dweighting_002dvalue"></a>procedure: <strong>target-auto-weighting-value</strong></dt>
<dd><p>Ian Tickle says (as far as I can understand) that the target rmsd
should be 0.25 or thereabouts.  You can over-ride it now.
</p>
</dd></dl>

<dl>
<dt><a name="index-auto_002dweight_002dfor_002drefinement"></a>procedure: <strong>auto-weight-for-refinement</strong></dt>
<dd><p>Set the refinement weight (matrix) by iterating the refinement and
varying the weight until the chi squares (not including the
non-bonded terms) reach 1.0 =/- 10%.  It uses sphere refinement.
The refinement map must be set.  At the end show the new weight in
the status bar.  Seems to take about 5 rounds.
</p>
</dd></dl>

<dl>
<dt><a name="index-print_002dsequence"></a>procedure: <strong>print-sequence</strong> <em>imol</em></dt>
<dd><p>Print the sequence of molecule number <var>imol</var>
</p>
<p>This is not really a util, perhaps it should be somewhere else?
</p>
</dd></dl>

<dl>
<dt><a name="index-pir_002dfile_002dname_002d_003epir_002dsequence"></a>procedure: <strong>pir-file-name-&gt;pir-sequence</strong> <em>pir-file-name</em></dt>
<dd><p>simple utility function to return the contents of a file as a string.
</p>
</dd></dl>

<dl>
<dt><a name="index-associate_002dpir_002dfile"></a>procedure: <strong>associate-pir-file</strong> <em>imol chain-id pir-file-name</em></dt>
<dd><p>Associate the contents of a PIR file with a molecule.
</p>
</dd></dl>

<dl>
<dt><a name="index-graphics_002dcomma_002dkey_002dpressed_002dhook"></a>procedure: <strong>graphics-comma-key-pressed-hook</strong></dt>
<dd><p>comma key hook
</p></dd></dl>

<dl>
<dt><a name="index-graphics_002ddot_002dkey_002dpressed_002dhook"></a>procedure: <strong>graphics-dot-key-pressed-hook</strong></dt>
<dd><p>dot key hook
</p></dd></dl>

<dl>
<dt><a name="index-_002akey_002dbindings_002a"></a>procedure: <strong>*key-bindings*</strong></dt>
<dd><p>a list of (code key name thunk) 
e.g. &rsquo;(103 &quot;g&quot; &quot;Goto Blob&quot; (blob-under-pointer-to-screen-centre))
</p></dd></dl>

<dl>
<dt><a name="index-add_002dkey_002dbinding"></a>procedure: <strong>add-key-binding</strong> <em>name key thunk</em></dt>
<dd><p>Add a key binding 
</p>
<p>with a given name, key (e.g. &quot;x&quot; or &quot;S&quot;) and the function to run
(a thunk) when that key is pressed.
</p>
</dd></dl>

<dl>
<dt><a name="index-graphics_002dgeneral_002dkey_002dpress_002dhook"></a>procedure: <strong>graphics-general-key-press-hook</strong> <em>key</em></dt>
<dd><p>general key press hook, not for public use.
</p>
</dd></dl>

<dl>
<dt><a name="index-read_002dvu_002dfile"></a>procedure: <strong>read-vu-file</strong></dt>
<dd><p>Function requested by Mark White.
</p>
<p>read XtalView (and maybe other) .vu files and convert them into generic 
objects.  
</p>
<p>Pass the filename and an object name e.g.
(read-vu-file &quot;axes.vu&quot; &quot;axes&quot;)
</p>
<p>Returns: nothing interesting.
</p>
</dd></dl>

<dl>
<dt><a name="index-residues_002dmatching_002dcriteria"></a>procedure: <strong>residues-matching-criteria</strong> <em>imol residue-test-func</em></dt>
<dd><p>Return a list of residues, each of which has a return value at the
start, ie. (list return-value chain-id res-no ins-code)
</p>
</dd></dl>

<dl>
<dt><a name="index-residues_002din_002dchain"></a>procedure: <strong>residues-in-chain</strong> <em>imol chain-id-in</em></dt>
<dd><p>Return a list of all the residues in the chain
</p>
</dd></dl>

<dl>
<dt><a name="index-residues_002dwith_002dalt_002dconfs"></a>procedure: <strong>residues-with-alt-confs</strong> <em>imol</em></dt>
<dd><p>Return a list of all residues that have alt confs: where a residue
is specified thusly: (list chain-id resno ins-code)
</p>
</dd></dl>

<dl>
<dt><a name="index-residue_002dalt_002dconfs"></a>procedure: <strong>residue-alt-confs</strong> <em>imol chain-id res-no ins-code</em></dt>
<dd><p>Return a list of all the altconfs in the residue. 
Typically this will return (list &quot;&quot;) or (list &quot;A&quot; &quot;B&quot;)
</p>
</dd></dl>

<dl>
<dt><a name="index-atom_002dspec_002d_003echain_002did"></a>procedure: <strong>atom-spec-&gt;chain-id</strong> <em>atom-spec</em></dt>
<dd><p>extraction function
</p></dd></dl>

<dl>
<dt><a name="index-atom_002dspec_002d_003eres_002dno"></a>procedure: <strong>atom-spec-&gt;res-no</strong> <em>atom-spec</em></dt>
<dd><p>extraction function
</p></dd></dl>

<dl>
<dt><a name="index-atom_002dspec_002d_003eins_002dcode"></a>procedure: <strong>atom-spec-&gt;ins-code</strong> <em>atom-spec</em></dt>
<dd><p>extraction function
</p></dd></dl>

<dl>
<dt><a name="index-atom_002dspec_002d_003eatom_002dname"></a>procedure: <strong>atom-spec-&gt;atom-name</strong> <em>atom-spec</em></dt>
<dd><p>extraction function
</p></dd></dl>

<dl>
<dt><a name="index-atom_002dspec_002d_003ealt_002dloc"></a>procedure: <strong>atom-spec-&gt;alt-loc</strong> <em>atom-spec</em></dt>
<dd><p>extraction function
</p></dd></dl>

<dl>
<dt><a name="index-res_002dspec_002d_003echain_002did"></a>procedure: <strong>res-spec-&gt;chain-id</strong> <em>res-spec</em></dt>
<dd><p>simple extraction function
</p></dd></dl>

<dl>
<dt><a name="index-res_002dspec_002d_003eres_002dno"></a>procedure: <strong>res-spec-&gt;res-no</strong> <em>res-spec</em></dt>
<dd><p>simple extraction function 
</p></dd></dl>

<dl>
<dt><a name="index-res_002dspec_002d_003eins_002dcode"></a>procedure: <strong>res-spec-&gt;ins-code</strong> <em>res-spec</em></dt>
<dd><p>simple extraction function 
</p></dd></dl>

<dl>
<dt><a name="index-residue_002dspec_002d_003eatom_002dfor_002dcentre"></a>procedure: <strong>residue-spec-&gt;atom-for-centre</strong> <em>imol chain-id res-no ins-code</em></dt>
<dd><p>Return #f if no atom can be found given the spec else return a list
consisting of the atom name and alt-conf specifier.  
</p>
<p>Choose an atom that is called &quot; CA &quot;.  Failing that choose the
first atom.
</p>
</dd></dl>

<dl>
<dt><a name="index-update_002dgo_002dto_002datom_002dfrom_002dcurrent_002datom"></a>procedure: <strong>update-go-to-atom-from-current-atom</strong></dt>
</dl>

<dl>
<dt><a name="index-flip_002dactive_002dligand"></a>procedure: <strong>flip-active-ligand</strong></dt>
</dl>

<dl>
<dt><a name="index-delete_002datom_002dby_002dactive_002dresidue"></a>procedure: <strong>delete-atom-by-active-residue</strong></dt>
<dd><p>Typically one might want to use this on a water, but it deletes the
nearest CA currently...  Needs a re-think.  Should active-atom just
return the nearest atom and not be clever about returning a CA.
</p>
</dd></dl>

<dl>
<dt><a name="index-merge_002dsolvent_002dchains"></a>procedure: <strong>merge-solvent-chains</strong> <em>imol</em></dt>
</dl>

<dl>
<dt><a name="index-mutate_002dby_002doverlap"></a>procedure: <strong>mutate-by-overlap</strong> <em>imol chain-id-in resno tlc</em></dt>
<dd><p>change chain ids with residue range for the PTY
</p>
</dd></dl>

<dl>
<dt><a name="index-phosphorylate_002dactive_002dresidue"></a>procedure: <strong>phosphorylate-active-residue</strong></dt>
<dd><p>A bit of fun 
</p>
</dd></dl>

<dl>
<dt><a name="index-overlay_002dmy_002dligands"></a>procedure: <strong>overlay-my-ligands</strong> <em>imol-mov chain-id-mov resno-mov imol-ref chain-id-ref resno-ref</em></dt>
<dd><p>A function for Overlaying ligands.  The transformation is applied
to all the atoms of the molecule that contains the moving ligand.
</p>
</dd></dl>

<dl>
<dt><a name="index-label_002dall_002dCAs"></a>procedure: <strong>label-all-CAs</strong> <em>imol</em></dt>
</dl>

<dl>
<dt><a name="index-label_002dall_002datoms_002din_002dresidue"></a>procedure: <strong>label-all-atoms-in-residue</strong> <em>imol chain-id resno inscode</em></dt>
</dl>

<dl>
<dt><a name="index-label_002dall_002dactive_002dresidue_002datoms"></a>procedure: <strong>label-all-active-residue-atoms</strong></dt>
</dl>

<dl>
<dt><a name="index-sanitise_002dalt_002dconfs"></a>procedure: <strong>sanitise-alt-confs</strong> <em>atom-info atom-ls</em></dt>
<dd><p>Resets alt confs and occupancies of atoms in residue that have
orphan alt-loc attributes.
</p>
</dd></dl>

<dl>
<dt><a name="index-sanitise_002dalt_002dconfs_002din_002dresidue"></a>procedure: <strong>sanitise-alt-confs-in-residue</strong> <em>imol chain-id resno inscode</em></dt>
</dl>

<dl>
<dt><a name="index-sanitise_002dalt_002dconfs_002dactive_002dresidue"></a>procedure: <strong>sanitise-alt-confs-active-residue</strong></dt>
<dd><p>Resets alt confs and occupancies of atoms in residue that have
orphan alt-loc attributes.  Use the active-residue.
</p>
</dd></dl>

<dl>
<dt><a name="index-print_002dmolecule_002dnames"></a>procedure: <strong>print-molecule-names</strong></dt>
</dl>

<dl>
<dt><a name="index-save_002ddialog_002dpositions_002dto_002dinit_002dfile"></a>procedure: <strong>save-dialog-positions-to-init-file</strong></dt>
</dl>

<dl>
<dt><a name="index-multi_002dchicken"></a>procedure: <strong>multi-chicken</strong> <em>imol . n-colours</em></dt>
<dd><p>multiple maps of varying colour from a given map.
</p>
</dd></dl>

<dl>
<dt><a name="index-BALL_005fAND_005fSTICK"></a>procedure: <strong>BALL_AND_STICK</strong></dt>
<dd><p>simple enumeration
</p>
</dd></dl>

<dl>
<dt><a name="index-hilight_002dbinding_002dsite"></a>procedure: <strong>hilight-binding-site</strong> <em>imol centre-residue-spec hilight-colour radius</em></dt>
<dd><p>hilight-colour is specified in degrees (round the colour wheel -
starting at yellow (e.g. 230 is purple))
</p>
</dd></dl>

<dl>
<dt><a name="index-pukka_002dpuckers_003f"></a>procedure: <strong>pukka-puckers?</strong> <em>imol</em></dt>
<dd><p>To paraphrase:
The distance of the plane of the base to the following phosphate
is highly correlated to the pucker of the ribose. 
</p>
<p>An analysis of the structures in RNADB2005 shows that a critical
distance of 3.3A provides a partition function to separate C2&rsquo; from
C3&rsquo; endo puckering.  Not all ribose follow this rule.  There may be
some errors in the models comprising RNADB2005. So we check the
distance of the following phosphate to the plane of the ribose and
record the riboses that are inconsitent.  We also report puckers
that are not C2&rsquo; or C3&rsquo;.  The puckers are determined by the most
out-of-plane atom of the ribose (the rms deviation of the 4 atoms
in the plane is calculated, but not used to determine the
puckering atom).
</p>
</dd></dl>

<dl>
<dt><a name="index-_002ause_002dmogul_002a"></a>procedure: <strong>*use-mogul*</strong></dt>
<dd><p>if you don&rsquo;t have mogul, set this to #f
</p></dd></dl>

<dl>
<dt><a name="index-prodrg_002dify"></a>procedure: <strong>prodrg-ify</strong> <em>imol chain-id res-no ins-code</em></dt>
<dd><p>Generate restraints from the residue at the centre of the screen
using PRODRG. Delete hydrogens from the residue because PRODRG has
anomalous hydrogens.
</p>
</dd></dl>

<dl>
<dt><a name="index-add_002dannotation_002dhere"></a>procedure: <strong>add-annotation-here</strong> <em>text</em></dt>
</dl>

<dl>
<dt><a name="index-save_002dannotations"></a>procedure: <strong>save-annotations</strong> <em>file-name</em></dt>
</dl>

<dl>
<dt><a name="index-load_002dannotations"></a>procedure: <strong>load-annotations</strong> <em>file-name</em></dt>
</dl>

<dl>
<dt><a name="index-make_002dlatest_002dversion_002durl"></a>procedure: <strong>make-latest-version-url</strong></dt>
<dd><p>Here we construct the url that contains the latest (pre)
release info adding in &quot;pre-release&quot; if this binary is a
pre-release.  args ends up as something like: (&quot;-s&quot;
&quot;xxx/phone-home.scm&quot; &quot;pre-release&quot; &quot;binary&quot;
&quot;Linux-1386-fedora-10-python-gtk2&quot; &quot;command-line&quot;
&quot;/home/xx/coot/bin/coot&quot;)
</p>
</dd></dl>

<dl>
<dt><a name="index-run_002ddownload_002dbinary_002dcurl"></a>procedure: <strong>run-download-binary-curl</strong> <em>revision version-string pending-install-in-place-func set-file-name-func</em></dt>
<dd><p>Get the binary (i.e. the action that happens when the download
button is pressed).  This is run in a thread, so it can&rsquo;t do
any graphics stuff. 
</p>
<p>return #t if tar file was successfully downloaded and untared
and #f if not.
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002drevision_002dfrom_002dstring"></a>procedure: <strong>get-revision-from-string</strong> <em>str</em></dt>
<dd><p>(used in downloading new version)
</p>
</dd></dl>

<dl>
<dt><a name="index-coot_002dsplit_002dversion_002dstring"></a>procedure: <strong>coot-split-version-string</strong> <em>str</em></dt>
<dd><p>e.g. input:  &quot;coot-0.6.2-pre-1-revision-2765\n&quot;
     output: &quot;coot-0.6.2-pre-1-revision-2765&quot;
</p> 
</dd></dl>

<dl>
<dt><a name="index-load_002ddefault_002dsequence"></a>procedure: <strong>load-default-sequence</strong></dt>
<dd><p>In the first case the sequence is assigned to the closest match
(model sequence to target sequence), subsequently only chains
without a sequence associated with them are candidates for
matching.  The protein sequence has to have at least 95% sequence
identity with the target sequence in &quot;default.seq&quot;
</p>
</dd></dl>

<dl>
<dt><a name="index-coot_002dupdates_002derror_002dhandler"></a>procedure: <strong>coot-updates-error-handler</strong> <em>key . args</em></dt>
<dd><p>not really for public manipulation.
</p>
</dd></dl>

<dl>
<dt><a name="index-update_002dself"></a>procedure: <strong>update-self</strong></dt>
<dd><p>keep a copy of the old directories around in a directory named
after expiration time.
</p>
</dd></dl>

<dl>
<dt><a name="index-chiral_002dcentre_002dinverter_002dscm_002dversion"></a>procedure: <strong>chiral-centre-inverter-scm-version</strong></dt>
<dd><p>This should almost all be c++ code so that Bernie doesn&rsquo;t have to
redo it.  This is temporary then.
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002ddrug_002dvia_002dwikipedia"></a>procedure: <strong>get-drug-via-wikipedia</strong> <em>drug-name-in</em></dt>
<dd><p>return a string &quot;DB00xxxx.mol&quot; or some such - this is the file name
of the mdl mol file from drugbank. Or #f/undefined on fail.  
Test result with string?.
</p>
<p>If you are debugging this function, you probably want to be looking
at handle-rev-string or drugbox-&gt;drugitem.
</p>
</dd></dl>

<dl>
<dt><a name="index-coot_002dhas_002dpygtk_003f"></a>procedure: <strong>coot-has-pygtk?</strong></dt>
<dd><p>to determine if we have pygtk
</p>
</dd></dl>
<hr>
<a name="contact_002dscore_002disolated_002dligand"></a>
<div class="header">
<p>
Next: <a href="#libcheck" accesskey="n" rel="next">libcheck</a>, Previous: <a href="#coot_002dutils" accesskey="p" rel="previous">coot-utils</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="contact_002dscore_002disolated_002dligand-1"></a>
<h3 class="section">14.33 contact-score-isolated-ligand</h3>

<dl>
<dt><a name="index-contact_002dscore_002dligand"></a>procedure: <strong>contact-score-ligand</strong> <em>imol res-spec</em></dt>
<dd><p>This will hydrogenate the active residue, not imol
</p>
</dd></dl>
<hr>
<a name="libcheck"></a>
<div class="header">
<p>
Next: <a href="#cns2coot" accesskey="n" rel="next">cns2coot</a>, Previous: <a href="#contact_002dscore_002disolated_002dligand" accesskey="p" rel="previous">contact-score-isolated-ligand</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="libcheck-1"></a>
<h3 class="section">14.34 libcheck</h3>

<dl>
<dt><a name="index-libcheck_002dexe"></a>procedure: <strong>libcheck-exe</strong></dt>
<dd><p>this is override-able by the user in their .coot file (for example).
</p></dd></dl>

<dl>
<dt><a name="index-monomer_002dmolecule_002dfrom_002d3_002dlet_002dcode"></a>procedure: <strong>monomer-molecule-from-3-let-code</strong> <em>code dict-cif-libin . ccp4i-project-dir</em></dt>
<dd><p>Return -2 on <var>code</var> is not a string
Return -3 on libcheck failure
Return <var>imol</var> on success
Return <code>handle-read-draw-molecule</code> error code on failure to 
      read resultant pdb file
</p>
<p>Actually, it would be nice to know if this code represented a full
description - or a minimal one... perhaps we can parse the log
file and call a pop-up that will tell us.
</p>
<p><var>dict-cif-libin should be a string</var>.  If it is &quot;&quot; then it is
ignored.  If it is not &quot;&quot; then it is used to create input to
libcheck (not a command line argument) so that bespoke dictionary
libraries can produce coords using &quot;Get Monomer&quot;.
</p>
</dd></dl>
<hr>
<a name="cns2coot"></a>
<div class="header">
<p>
Next: <a href="#clear_002dbackup" accesskey="n" rel="next">clear-backup</a>, Previous: <a href="#libcheck" accesskey="p" rel="previous">libcheck</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="cns2coot-1"></a>
<h3 class="section">14.35 cns2coot</h3>

<dl>
<dt><a name="index-cns_002d_003ecoot"></a>procedure: <strong>cns-&gt;coot</strong> <em>2fofc-coeffs fofc-coeffs model-pdb</em></dt>
<dd><p>Read in cns coeff-data (given filenames) and a pdb molecule filename to make
maps.
</p>
</dd></dl>
<hr>
<a name="clear_002dbackup"></a>
<div class="header">
<p>
Next: <a href="#generic_002dobjects" accesskey="n" rel="next">generic-objects</a>, Previous: <a href="#cns2coot" accesskey="p" rel="previous">cns2coot</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="clear_002dbackup-1"></a>
<h3 class="section">14.36 clear-backup</h3>

<dl>
<dt><a name="index-delete_002dcoot_002dbackup_002dfiles"></a>procedure: <strong>delete-coot-backup-files</strong> <em>action-type</em></dt>
</dl>

<dl>
<dt><a name="index-clear_002dbackup_002dgui"></a>procedure: <strong>clear-backup-gui</strong></dt>
<dd><p>Return #t or #f depending on if the GUI dialog was shown (it isn&rsquo;t
show if there are no files to delete).
</p>
</dd></dl>

<dl>
<dt><a name="index-clear_002dbackups_002dmaybe"></a>procedure: <strong>clear-backups-maybe</strong></dt>
<dd><p>Return a status, #f or #t, did the GUI run?
</p>
<p>Note that clear-backup-gui returns either #t or #f too.
</p>
<p>If this function returns #f, then coot_real_exit() just exits with
coot_real_exit().  Otherwise we wait for the GUI.
</p>
</dd></dl>
<hr>
<a name="generic_002dobjects"></a>
<div class="header">
<p>
Next: <a href="#exercise_002dscm_002dmol" accesskey="n" rel="next">exercise-scm-mol</a>, Previous: <a href="#clear_002dbackup" accesskey="p" rel="previous">clear-backup</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="generic_002dobjects-1"></a>
<h3 class="section">14.37 generic-objects</h3>

<dl>
<dt><a name="index-generic_002dobject_002dis_002ddisplayed_003f"></a>procedure: <strong>generic-object-is-displayed?</strong></dt>
<dd><p>map to scheme names:
</p></dd></dl>

<dl>
<dt><a name="index-is_002dclosed_002dgeneric_002dobject_003f"></a>procedure: <strong>is-closed-generic-object?</strong></dt>
<dd><p>map to scheme names:
</p></dd></dl>

<dl>
<dt><a name="index-generic_002dobject_002dwith_002dname"></a>procedure: <strong>generic-object-with-name</strong></dt>
<dd><p>return a new generic object number for the given object obj-name.
If there is no such object with name obj-name, then create a new
one.  (Maybe you want the generic object to be cleaned if it exists
before returning, this function does not do that).
</p>
</dd></dl>

<dl>
<dt><a name="index-generic_002dobjects_002dgui"></a>procedure: <strong>generic-objects-gui</strong></dt>
<dd><p>display a GUI for generic objects
</p>
</dd></dl>

<dl>
<dt><a name="index-reduce_002don_002dpdb_002dfile"></a>procedure: <strong>reduce-on-pdb-file</strong> <em>imol pdb-in pdb-out</em></dt>
<dd><p>return status.
</p>
</dd></dl>

<dl>
<dt><a name="index-reduce_002don_002dpdb_002dfile_002dno_002dflip"></a>procedure: <strong>reduce-on-pdb-file-no-flip</strong> <em>imol pdb-in pdb-out</em></dt>
<dd><p>return status.
</p>
</dd></dl>

<dl>
<dt><a name="index-reduce_002don_002dpdb_002dfile_002dgeneric"></a>procedure: <strong>reduce-on-pdb-file-generic</strong> <em>imol no-flip-or-build pdb-in pdb-out</em></dt>
<dd><p>return status.
</p>
</dd></dl>

<dl>
<dt><a name="index-probe-1"></a>procedure: <strong>probe</strong></dt>
<dd><p>run molprobity (well reduce and probe) to make generic objects (and
display the generic objects gui)
</p>
</dd></dl>

<dl>
<dt><a name="index-write_002dreduce_002dhet_002ddict"></a>procedure: <strong>write-reduce-het-dict</strong> <em>imol reduce-het-dict-file-name</em></dt>
<dd><p>Write the connectivity for the non-standard (non-water) residues in
the given molecule for which we have the dictionary.
</p>
<p>Don&rsquo;t return anything interesting.  
</p>
</dd></dl>

<dl>
<dt><a name="index-_002ainteractive_002dprobe_002dis_002dOK_003f_002a"></a>procedure: <strong>*interactive-probe-is-OK?*</strong></dt>
<dd><p>gets set the first time we run interactive-probe.  Takes values
unset (initial value) &rsquo;yes and &rsquo;no)
</p>			
</dd></dl>

<dl>
<dt><a name="index-interactive_002dprobe"></a>procedure: <strong>interactive-probe</strong></dt>
<dd><p>run &quot;probe&quot; interactively, 
which in the current implementation, means that this function 
can run during a edit-chi angles manipulation, or after
a real space refine zone.
</p>
<p>Thus function presumes that there are 2 pdb files in the current 
directory, one of which is the reference pdb file and the other
is a pdb file containing the tmp/moving atom set.
</p>
<p>The function takes arguments for the centre of the probe dots
and the radius of the probe dots sphere.  The chain id and 
residue number are also needed to pass as arguments to probe.
</p>
</dd></dl>

<dl>
<dt><a name="index-probe_002dlocal_002dsphere"></a>procedure: <strong>probe-local-sphere</strong> <em>imol radius</em></dt>
<dd><p>Update the generic objects probe dots from residues within radius
of the screen centre.
</p>
<p>Return nothing interesting.
</p>
</dd></dl>
<hr>
<a name="exercise_002dscm_002dmol"></a>
<div class="header">
<p>
Next: <a href="#check_002dfor_002dupdates" accesskey="n" rel="next">check-for-updates</a>, Previous: <a href="#generic_002dobjects" accesskey="p" rel="previous">generic-objects</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="exercise_002dscm_002dmol-1"></a>
<h3 class="section">14.38 exercise-scm-mol</h3>

<dl>
<dt><a name="index-jiggled_002dmol"></a>procedure: <strong>jiggled-mol</strong> <em>reference-mol current-mol traj-frac</em></dt>
</dl>

<dl>
<dt><a name="index-disrupt"></a>procedure: <strong>disrupt</strong> <em>reference-mol biggness</em></dt>
</dl>
<hr>
<a name="check_002dfor_002dupdates"></a>
<div class="header">
<p>
Next: <a href="#brute_002dlsqman" accesskey="n" rel="next">brute-lsqman</a>, Previous: <a href="#exercise_002dscm_002dmol" accesskey="p" rel="previous">exercise-scm-mol</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="check_002dfor_002dupdates-1"></a>
<h3 class="section">14.39 check-for-updates</h3>

<dl>
<dt><a name="index-get_002dstable_002drelease_002dfrom_002dserver_002dstring"></a>procedure: <strong>get-stable-release-from-server-string</strong> <em>str</em></dt>
<dd><p>Is this true?
</p></dd></dl>

<dl>
<dt><a name="index-get_002dstable_002drelease_002dfrom_002dcoot_002dversion"></a>procedure: <strong>get-stable-release-from-coot-version</strong></dt>
<dd><p>Needs testing.
</p></dd></dl>

<dl>
<dt><a name="index-notify_002dof_002dnew_002dversion"></a>procedure: <strong>notify-of-new-version</strong> <em>str</em></dt>
<dd><p>show the dialog
</p>
</dd></dl>

<dl>
<dt><a name="index-download_002dbinary_002ddialog"></a>procedure: <strong>download-binary-dialog</strong></dt>
<dd><p>version-string is something like: &quot;coot-0.6-pre-1-revision-2060&quot;
</p>
</dd></dl>

<dl>
<dt><a name="index-directory_002dis_002dmodifiable_003f-1"></a>procedure: <strong>directory-is-modifiable?</strong> <em>prefix-dir</em></dt>
<dd><p>Test for prefix-dir 1) existing 2) being a directory 3) modifiable by user (ie. u+rwx)
</p>
<p>Return #t or #f.
</p>
</dd></dl>
<hr>
<a name="brute_002dlsqman"></a>
<div class="header">
<p>
Next: <a href="#fitting" accesskey="n" rel="next">fitting</a>, Previous: <a href="#check_002dfor_002dupdates" accesskey="p" rel="previous">check-for-updates</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="brute_002dlsqman-1"></a>
<h3 class="section">14.40 brute-lsqman</h3>

<dl>
<dt><a name="index-lsqman_002dcount"></a>procedure: <strong>lsqman-count</strong></dt>
<dd><p>brute-lsqman - run lsqman on chains A of two pdb-files and read in
the result to coot. Charlie Bond 2003.
Can keep a count of the number of successful runs if necessary
</p></dd></dl>
<hr>
<a name="fitting"></a>
<div class="header">
<p>
Next: <a href="#group_002dsettings" accesskey="n" rel="next">group-settings</a>, Previous: <a href="#brute_002dlsqman" accesskey="p" rel="previous">brute-lsqman</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fitting-1"></a>
<h3 class="section">14.41 fitting</h3>

<dl>
<dt><a name="index-fit_002dprotein"></a>procedure: <strong>fit-protein</strong> <em>imol</em></dt>
<dd><p>Note that residue with alt confs do not undergo auto-fit-rotamer.
This is because that autofit-rotamer then refine will tend to put
both rotamers into the same place.  Not good.  It seems a
reasonable expectation that residues with an alternate conformation
are already resonably well-fitted.  So residues with alternate
conformations undergo only real space refinement.
</p>
<p>This is simple-minded and outdated now we have the interruptible
version (below).
</p>
</dd></dl>

<dl>
<dt><a name="index-_002acontinue_002dmulti_002drefine_002a"></a>procedure: <strong>*continue-multi-refine*</strong></dt>
<dd><p>These 2 variables are used by multi-refine function(s), called by
idle functions to refine just one residue.
</p>
</dd></dl>

<dl>
<dt><a name="index-fit_002dprotein_002dmake_002dspecs"></a>procedure: <strong>fit-protein-make-specs</strong> <em>imol chain-specifier</em></dt>
<dd><p>chain-specifier can be a string, where it is the chain of interest.
or &rsquo;all-chains, where all chains are chosen.
</p>
</dd></dl>

<dl>
<dt><a name="index-interruptible_002dfit_002dprotein"></a>procedure: <strong>interruptible-fit-protein</strong> <em>imol func</em></dt>
<dd><p>func is a refinement function that takes 2 args, one a residue
spec, the other the imol-refinement-map.  e.g. fit-protein-fit-function
</p>
</dd></dl>

<dl>
<dt><a name="index-fit_002dchain"></a>procedure: <strong>fit-chain</strong> <em>imol chain-id</em></dt>
<dd><p>For each residue in chain chain-id of molecule number imol, do a
rotamer fit and real space refinement of each residue.  Don&rsquo;t
update the graphics while this is happening (which makes it faster
than fit-protein, but much less interesting to look at).
</p>
</dd></dl>

<dl>
<dt><a name="index-fit_002dwaters"></a>procedure: <strong>fit-waters</strong> <em>imol . animate?</em></dt>
<dd><p>For each residue in the solvent chains of molecule number
<var>imol</var>, do a rigid body fit of the water to the density.
</p>
</dd></dl>

<dl>
<dt><a name="index-stepped_002drefine_002dprotein"></a>procedure: <strong>stepped-refine-protein</strong> <em>imol . res-step</em></dt>
<dd><p>Step through the residues of molecule number imol and at each step
do a residue range refinement (unlike fit-protein for example,
which does real-space refinement for every residue).
</p>
<p>The step is set internally to 2.
</p>
</dd></dl>

<dl>
<dt><a name="index-stepped_002drefine_002dprotein_002dfor_002drama"></a>procedure: <strong>stepped-refine-protein-for-rama</strong> <em>imol</em></dt>
<dd><p>refine each residue with ramachandran restraints
</p>
</dd></dl>

<dl>
<dt><a name="index-stepped_002drefine_002dprotein_002dwith_002drefine_002dfunc"></a>procedure: <strong>stepped-refine-protein-with-refine-func</strong> <em>imol refine-func . res-step</em></dt>
</dl>

<dl>
<dt><a name="index-post_002dligand_002dfit_002dgui"></a>procedure: <strong>post-ligand-fit-gui</strong></dt>
<dd><p>The GUI that you see after ligand finding. 
</p>
</dd></dl>

<dl>
<dt><a name="index-molecules_002dmatching_002dcriteria"></a>procedure: <strong>molecules-matching-criteria</strong> <em>test-func</em></dt>
<dd><p>test-func is a function given one argument (a molecule number) that
returns either #f if the condition is not satisfied or something
else if it is.  And that &quot;something else&quot; can be a list like
(list label x y z)
or 
(list &quot;Bad Chiral&quot; 0 &quot;A&quot; 23 &quot;&quot; &quot;CA&quot; &quot;A&quot;)
</p>
<p>It is used in the create a button label and &quot;what to do when the
button is pressed&quot;.
</p>
</dd></dl>

<dl>
<dt><a name="index-refine_002dactive_002dresidue_002dgeneric"></a>procedure: <strong>refine-active-residue-generic</strong> <em>side-residue-offset</em></dt>
<dd><p>This totally ignores insertion codes.  A clever algorithm would
need a re-write, I think.  Well, we&rsquo;d have at this end a function
that took a chain-id res-no-1 ins-code-1 res-no-2 ins-code-2 
</p>
<p>And refine-zone would need to be re-written too, of course.  So
let&rsquo;s save that for a rainy day (days... (weeks)).
</p>
</dd></dl>

<dl>
<dt><a name="index-refine_002dactive_002dresidue"></a>procedure: <strong>refine-active-residue</strong></dt>
<dd><p>refine active residue
</p>
</dd></dl>

<dl>
<dt><a name="index-refine_002dactive_002dresidue_002dtriple"></a>procedure: <strong>refine-active-residue-triple</strong></dt>
<dd><p>refine active residue triple
</p>
</dd></dl>

<dl>
<dt><a name="index-manual_002drefine_002dresidues"></a>procedure: <strong>manual-refine-residues</strong> <em>side-residue-offset</em></dt>
<dd><p>For just one (this) residue, side-residue-offset is 0.
</p>
</dd></dl>

<dl>
<dt><a name="index-pepflip_002dactive_002dresidue"></a>procedure: <strong>pepflip-active-residue</strong></dt>
<dd><p>Pepflip the active residue - needs a key binding.
</p>
</dd></dl>

<dl>
<dt><a name="index-auto_002dfit_002drotamer_002dactive_002dresidue"></a>procedure: <strong>auto-fit-rotamer-active-residue</strong></dt>
<dd><p>Auto-fit rotamer on active residues
</p>
</dd></dl>

<dl>
<dt><a name="index-add_002dextra_002drestraints_002dto_002dother_002dmolecule"></a>procedure: <strong>add-extra-restraints-to-other-molecule</strong> <em>imol chain-id resno-range-start resno-range-end atom-sel-type imol-ref</em></dt>
<dd><p>atom-sel-type is either &rsquo;all &rsquo;main-chain or &rsquo;ca
</p>
</dd></dl>
<hr>
<a name="group_002dsettings"></a>
<div class="header">
<p>
Next: <a href="#background_002ddemo" accesskey="n" rel="next">background-demo</a>, Previous: <a href="#fitting" accesskey="p" rel="previous">fitting</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="group_002dsettings-1"></a>
<h3 class="section">14.42 group-settings</h3>

<dl>
<dt><a name="index-rapper_002ddir"></a>procedure: <strong>rapper-dir</strong></dt>
<dd><p>The rapper installation dir.  This is just a guess, that it is
installed in the users home directory.
</p>
</dd></dl>
<hr>
<a name="background_002ddemo"></a>
<div class="header">
<p>
Next: <a href="#americanisms" accesskey="n" rel="next">americanisms</a>, Previous: <a href="#group_002dsettings" accesskey="p" rel="previous">group-settings</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="background_002ddemo-1"></a>
<h3 class="section">14.43 background-demo</h3>

<dl>
<dt><a name="index-background_002ddemo"></a>procedure: <strong>background-demo</strong></dt>
<dd><p>flash the background different colours in some uninteresting way.
</p></dd></dl>
<hr>
<a name="americanisms"></a>
<div class="header">
<p>
Next: <a href="#filter" accesskey="n" rel="next">filter</a>, Previous: <a href="#background_002ddemo" accesskey="p" rel="previous">background-demo</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="americanisms-1"></a>
<h3 class="section">14.44 americanisms</h3>

<dl>
<dt><a name="index-set_002drotation_002dcenter"></a>procedure: <strong>set-rotation-center</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-handle_002dread_002ddraw_002dmolecule_002dwith_002drecenter"></a>procedure: <strong>handle-read-draw-molecule-with-recenter</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-set_002drotation_002dcenter_002dsize"></a>procedure: <strong>set-rotation-center-size</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-center_002datom_002dlabel_002dstatus"></a>procedure: <strong>center-atom-label-status</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-set_002dlast_002dmap_002dcolor"></a>procedure: <strong>set-last-map-color</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-center_002dof_002dmass"></a>procedure: <strong>center-of-mass</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-set_002dswap_002ddifference_002dmap_002dcolors"></a>procedure: <strong>set-swap-difference-map-colors</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-set_002dfont_002dcolor"></a>procedure: <strong>set-font-color</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-set_002ddots_002dcolor"></a>procedure: <strong>set-dots-color</strong></dt>
<dd><p>an americanism
</p></dd></dl>

<dl>
<dt><a name="index-invert_002dchiral_002dcenter"></a>procedure: <strong>invert-chiral-center</strong></dt>
<dd><p>an americanism
</p></dd></dl>
<hr>
<a name="filter"></a>
<div class="header">
<p>
Next: <a href="#a_002drapper_002dgui" accesskey="n" rel="next">a-rapper-gui</a>, Previous: <a href="#americanisms" accesskey="p" rel="previous">americanisms</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="filter-1"></a>
<h3 class="section">14.45 filter</h3>

<dl>
<dt><a name="index-filter-1"></a>procedure: <strong>filter</strong> <em>fn ls</em></dt>
<dd><p>Basic scheme function, filter the objects in list ls by function
fn.  e.g. (filter even? (list 0 1 2 3) -&gt; &rsquo;(0 2)
</p>
</dd></dl>
<hr>
<a name="a_002drapper_002dgui"></a>
<div class="header">
<p>
Next: <a href="#fascinating_002dthings" accesskey="n" rel="next">fascinating-things</a>, Previous: <a href="#filter" accesskey="p" rel="previous">filter</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="a_002drapper_002dgui-1"></a>
<h3 class="section">14.46 a-rapper-gui</h3>

<dl>
<dt><a name="index-rapper_002ddir-1"></a>procedure: <strong>rapper-dir</strong></dt>
<dd><p>something that the user sets:
</p></dd></dl>

<dl>
<dt><a name="index-a_002drapper_002dgui"></a>procedure: <strong>a-rapper-gui</strong> <em>loop-building-tool</em></dt>
<dd><p>loop-building-tool is either &rsquo;rapper or &rsquo;ARP/wARP 
</p></dd></dl>
<hr>
<a name="fascinating_002dthings"></a>
<div class="header">
<p>
Next: <a href="#get_002drecent_002dpdbe" accesskey="n" rel="next">get-recent-pdbe</a>, Previous: <a href="#a_002drapper_002dgui" accesskey="p" rel="previous">a-rapper-gui</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="fascinating_002dthings-1"></a>
<h3 class="section">14.47 fascinating-things</h3>

<dl>
<dt><a name="index-fascinating_002dclusters_002dgui"></a>procedure: <strong>fascinating-clusters-gui</strong> <em>window-name sorting-options cluster-list</em></dt>
<dd><p>(list cluster-name-string
      cluster-center-go-button-label-string
      ccgb-x ccgb-y ccgb-z
      ; now a list of specific items
      (list
        (list specific-button-label-string button-red button-green button-blue
              specific-x specific-y specific-z)
        (list specific-button-label-string button-red button-green button-blue
              specific-x specific-y specific-z)))))
</p>
</dd></dl>
<hr>
<a name="get_002drecent_002dpdbe"></a>
<div class="header">
<p>
Next: <a href="#g_t3d_002dgenerator_002dimport" accesskey="n" rel="next">3d-generator-import</a>, Previous: <a href="#fascinating_002dthings" accesskey="p" rel="previous">fascinating-things</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="get_002drecent_002dpdbe-1"></a>
<h3 class="section">14.48 get-recent-pdbe</h3>

<dl>
<dt><a name="index-coot_002dthread_002ddispatcher"></a>procedure: <strong>coot-thread-dispatcher</strong></dt>
<dd><p>(coot-thread-dispatcher image-name thunk)
</p>
</dd></dl>

<dl>
<dt><a name="index-dialog_002dbox_002dof_002dbuttons_002dwith_002dasync_002dligands"></a>procedure: <strong>dialog-box-of-buttons-with-async-ligands</strong> <em>window-name geometry buttons close-button-label</em></dt>
<dd><p>If check-button-label is #f, don&rsquo;t make one, otherwise create with with
the given label and &quot;on&quot; state.
</p>
</dd></dl>

<dl>
<dt><a name="index-cache_002dor_002dnet_002dget_002dimage"></a>procedure: <strong>cache-or-net-get-image</strong> <em>image-url image-name func</em></dt>
<dd><p>Get image-name (caller doesn&rsquo;t care how) and when it is in place run thunk func.
</p>
<p>This is a generally useful function, so it has been promoted outside of dig-table.
</p>
</dd></dl>

<dl>
<dt><a name="index-refmac_002dcalc_002dsfs_002dmake_002dmtz_002dwith_002dcolumns"></a>procedure: <strong>refmac-calc-sfs-make-mtz-with-columns</strong> <em>pdb-in-file-name mtz-file-name mtz-refmaced-file-name f-col sigf-col r-free-col</em></dt>
<dd><p>return refmac-result or #f 
</p>
</dd></dl>

<dl>
<dt><a name="index-pdbe_002dget_002dpdb_002dand_002dsfs_002dcif"></a>procedure: <strong>pdbe-get-pdb-and-sfs-cif</strong> <em>include-get-sfs-flag entry-id</em></dt>
<dd><p>Use progress bars
</p>
<p>include-get-sfs-flag is either &rsquo;no-sfs or &rsquo;include-sfs
</p>
</dd></dl>

<dl>
<dt><a name="index-pdbe_002dlatest_002dreleases_002dgui"></a>procedure: <strong>pdbe-latest-releases-gui</strong></dt>
<dd><p>Sameer Velankar says to get this file for the latest releases
&quot;http://www.ebi.ac.uk/pdbe-apps/jsonizer/latest/released/&quot; (note the end &quot;/&quot;).
</p>
</dd></dl>
<hr>
<a name="g_t3d_002dgenerator_002dimport"></a>
<div class="header">
<p>
Next: <a href="#coot" accesskey="n" rel="next">coot</a>, Previous: <a href="#get_002drecent_002dpdbe" accesskey="p" rel="previous">get-recent-pdbe</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="g_t3d_002dgenerator_002dimport-1"></a>
<h3 class="section">14.49 3d-generator-import</h3>

<dl>
<dt><a name="index-prodrg_002dxyzin"></a>procedure: <strong>prodrg-xyzin</strong></dt>
<dd><p>if there is a prodrg-xyzin set the current-time to its mtime, else #f
</p>
</dd></dl>

<dl>
<dt><a name="index-import_002dfrom_002d3d_002dgenerator_002dfrom_002dmdl"></a>procedure: <strong>import-from-3d-generator-from-mdl</strong> <em>mdl-file-name comp-id</em></dt>
<dd><p>This function can be overwritten by your favourite 3d conformer and restraints generator.
</p>
</dd></dl>

<dl>
<dt><a name="index-new_002dmolecule_002dby_002dsmiles_002dstring"></a>procedure: <strong>new-molecule-by-smiles-string</strong> <em>tlc-text smiles-text</em></dt>
<dd><p>Run libcheck to convert from SMILES string
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002dmdl_002dlatest_002dtime"></a>procedure: <strong>get-mdl-latest-time</strong></dt>
<dd><p>not needed?
</p></dd></dl>

<dl>
<dt><a name="index-prodrg_002dflat"></a>procedure: <strong>prodrg-flat</strong> <em>imol-in chain-id-in res-no-in</em></dt>
<dd><p>return #f (if fail) or a list of: the molecule number of the
selected residue, the prodrg output mol file-name, the prodrg
output pdb file-name
</p>
</dd></dl>

<dl>
<dt><a name="index-get_002dccp4srs_002dmonomer_002dand_002doverlay"></a>procedure: <strong>get-ccp4srs-monomer-and-overlay</strong> <em>comp-id</em></dt>
<dd><p>import from SBASE, callback using sbase_import_function
</p>
</dd></dl>
<hr>
<a name="coot"></a>
<div class="header">
<p>
Next: <a href="#jligand" accesskey="n" rel="next">jligand</a>, Previous: <a href="#g_t3d_002dgenerator_002dimport" accesskey="p" rel="previous">3d-generator-import</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="coot-1"></a>
<h3 class="section">14.50 coot</h3>

<dl>
<dt><a name="index-_002aprobe_002dcommand_002a"></a>procedure: <strong>*probe-command*</strong></dt>
<dd><p>This is full pathname of molprobity&rsquo;s probe program
</p></dd></dl>

<dl>
<dt><a name="index-_002areduce_002dcommand_002a"></a>procedure: <strong>*reduce-command*</strong></dt>
<dd><p>This is full pathname of molprobity&rsquo;s reduce program
</p></dd></dl>

<dl>
<dt><a name="index-_002ado_002dcoot_002dtips_002dflag_002a"></a>procedure: <strong>*do-coot-tips-flag*</strong></dt>
<dd><p>This has be be here (in a general place) because tips-gui (where it
used to be is conditionally loaded).
(default to tips-gui displayed is true).
</p></dd></dl>

<dl>
<dt><a name="index-load_002dall_002dscheme"></a>procedure: <strong>load-all-scheme</strong></dt>
<dd><p>Note the position of coot-gui is important.  It seem that if there
are too many files in front of it (even blank ones!) coot barfs
when it gets to coot-gui.scm.
</p>
<p>20060203 I have now enabled coot in scripting mode (no graphics
(&ndash;no-graphics command line option)).  In that case, we need to not
load up scheme files which load up gtk (currently coot-gui and
tips-gui).
</p>
</dd></dl>
<hr>
<a name="jligand"></a>
<div class="header">
<p>
Previous: <a href="#coot" accesskey="p" rel="previous">coot</a>, Up: <a href="#Scheme-Scripting-Functions" accesskey="u" rel="up">Scheme Scripting Functions</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="jligand-1"></a>
<h3 class="section">14.51 jligand</h3>

<dl>
<dt><a name="index-_002ato_002djligand_002dsecret_002dfile_002dname_002a"></a>procedure: <strong>*to-jligand-secret-file-name*</strong></dt>
<dd><p>JLigand sends Coot a file that contains filenames for the link and
new dictionary/restraints.
</p>
</dd></dl>

<dl>
<dt><a name="index-_002aimol_002djligand_002dlink_002a"></a>procedure: <strong>*imol-jligand-link*</strong></dt>
<dd><p>jligand internal parameter
</p></dd></dl>

<dl>
<dt><a name="index-get_002dfile_002dmtime"></a>procedure: <strong>get-file-mtime</strong> <em>file-name</em></dt>
<dd><p>this could be in utils
</p></dd></dl>

<dl>
<dt><a name="index-start_002djligand_002dlistener"></a>procedure: <strong>start-jligand-listener</strong></dt>
<dd><p>every fraction of a second look to see if
*from-jligand-secret-link-file-name* has been updated.  If so, then
run the handle-read-from-jligand-file function.
</p>
</dd></dl>




<hr>
<a name="Concept-Index"></a>
<div class="header">
<p>
Next: <a href="#Function-Index" accesskey="n" rel="next">Function Index</a>, Previous: <a href="#Scheme-Scripting-Functions" accesskey="p" rel="previous">Scheme Scripting Functions</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Concept-Index-1"></a>
<h2 class="unnumbered">Concept Index</h2>
     
<table><tr><th valign="top">Jump to: &nbsp; </th><td><a class="summary-letter" href="#Concept-Index_cp_symbol-1"><b>.</b></a>
 &nbsp; 
<br>
<a class="summary-letter" href="#Concept-Index_cp_letter-A"><b>A</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-B"><b>B</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-C"><b>C</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-D"><b>D</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-E"><b>E</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-F"><b>F</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-G"><b>G</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-H"><b>H</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-I"><b>I</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-L"><b>L</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-M"><b>M</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-N"><b>N</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-O"><b>O</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-P"><b>P</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-R"><b>R</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-S"><b>S</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-T"><b>T</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-U"><b>U</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-V"><b>V</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-W"><b>W</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-Y"><b>Y</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-Z"><b>Z</b></a>
 &nbsp; 
</td></tr></table>
<table class="index-cp" border="0">
<tr><td></td><th align="left">Index Entry</th><td>&nbsp;</td><th align="left"> Section</th></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_symbol-1">.</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002ecoot"><code>.coot</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scheme">Scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002ecoot_002epy"><code>.coot.py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#Python">Python</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-A">A</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-alternate-conformation">alternate conformation</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Adding-Alternative-Conformations">Adding Alternative Conformations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-anisotropic-atoms">anisotropic atoms</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Anisotropic-Atoms">Anisotropic Atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-annotations">annotations</a>:</td><td>&nbsp;</td><td valign="top"><a href="#g_t3D-Annotations">3D Annotations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom-colouring">atom colouring</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Manager">Display Manager</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom-colouring-1">atom colouring</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atom-Colouring">Atom Colouring</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom-info">atom info</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atom-Info">Atom Info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom-label_002c-brief">atom label, brief</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atom-Labeling">Atom Labeling</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom-labeling">atom labeling</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atom-Labeling">Atom Labeling</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom-picking">atom picking</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Clear-Pending-Picks">Clear Pending Picks</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atomic-dots">atomic dots</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atomic-Dots">Atomic Dots</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-auto_002dfit-rotamer">auto-fit rotamer</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rotamers">Rotamers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-average-map">average map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Make-an-Averaged-Map">Make an Averaged Map</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-B">B</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-B_002dFactor_002c-New-Atoms">B-Factor, New Atoms</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Temperature-Factor-for-New-Atoms">Temperature Factor for New Atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-background-colour">background colour</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Background-colour">Background colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-backrub-rotamers">backrub rotamers</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rotamers">Rotamers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-backups">backups</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Backups-and-Undo">Backups and Undo</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-backups_002c-clearing">backups, clearing</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Clearing-Backups">Clearing Backups</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ball-and-stick">ball and stick</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Ball-and-Stick-Representation">Ball and Stick Representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-baton-build">baton build</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Baton-Building">Baton Building</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-baton-mode">baton mode</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Baton-Building">Baton Building</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-big-maps">big maps</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Dynamic-Map-Sampling-and-Display-Size">Dynamic Map Sampling and Display Size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-blobs">blobs</a>:</td><td>&nbsp;</td><td valign="top"><a href="#sec_005fblobs">sec_blobs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-bond-thickness">bond thickness</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Bond-Parameters">Bond Parameters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-bones">bones</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Skeletonization">Skeletonization</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-C">C</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-C_002dterminus">C-terminus</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Add-OXT-Atom-to-Residue">Add OXT Atom to Residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-carbohydrates">carbohydrates</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Refining-Carbohydrates">Refining Carbohydrates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-chain-id">chain id</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Changing-Chain-IDs">Changing Chain IDs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-change-contour-level">change contour level</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Maps-in-General">Maps in General</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-changing-chain-ids">changing chain ids</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Changing-Chain-IDs">Changing Chain IDs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-changing-the-Refinement-Map">changing the Refinement Map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Changing-the-Map-for-Building_002fRefinement">Changing the Map for Building/Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-chi-angles">chi angles</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Editing-Chi-Angles">Editing Chi Angles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cif-dictionary">cif dictionary</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Dictionary">Dictionary</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-citing-coot">citing coot</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Citing-Coot">Citing Coot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clashing-residues">clashing residues</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rotamers">Rotamers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Clear-Pending-Picks">Clear Pending Picks</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Clear-Pending-Picks">Clear Pending Picks</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clearing-backups">clearing backups</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Clearing-Backups">Clearing Backups</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clipping">clipping</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Clipping-Manipulation">Clipping Manipulation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-colour-by-chain">colour by chain</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Manager">Display Manager</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-colouring_002c-atoms">colouring, atoms</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atom-Colouring">Atom Colouring</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-colouring_002c-map">colouring, map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-colouring">Map colouring</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-command-line-arguments">command line arguments</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Command-Line-Arguments">Command Line Arguments</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-contouring_002c-map">contouring, map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scrollwheel">Scrollwheel</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-contouring_002c-map-1">contouring, map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Contouring">Map Contouring</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coordinates-format">coordinates format</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Reading-coordinates">Reading coordinates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot-droppings">coot droppings</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Coot-Droppings">Coot Droppings</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-crash">crash</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Crash">Crash</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-crash-recovery">crash recovery</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Crash">Crash</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-crosshairs">crosshairs</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Crosshairs">Crosshairs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-C_005calpha-representation">C<em>\alpha</em> representation</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Manager">Display Manager</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-C_005calpha-symmetry">C<em>\alpha</em> symmetry</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Symmetry">Symmetry</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-D">D</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-debugger">debugger</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Crash">Crash</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-default-refmac-executable">default refmac executable</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Running-Refmac">Running Refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete">delete</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Delete">Delete</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-density-line-thickness">density line thickness</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Line-Width">Map Line Width</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-depth_002dcueing">depth-cueing</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Issues">Display Issues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dictionary_002c-cif">dictionary, cif</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Environment-Variables">Environment Variables</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dictionary_002c-cif-1">dictionary, cif</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Find-Ligands">Find Ligands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dictionary_002c-ligands">dictionary, ligands</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Environment-Variables">Environment Variables</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-difference-map">difference map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Contouring">Map Contouring</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-difference-map-1">difference map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Make-a-Difference-Map">Make a Difference Map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-difference-map-colours">difference map colours</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Contouring">Map Contouring</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-directory-for-save-coordinates">directory for save coordinates</a>:</td><td>&nbsp;</td><td valign="top"><a href="#The-file-selector">The file selector</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Display-Manager">Display Manager</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Manager">Display Manager</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-DNA_002c-ideal">DNA, ideal</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Building-Ideal-DNA-and-RNA">Building Ideal DNA and RNA</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-DNA_002c-mutating">DNA, mutating</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mutation">Mutation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-docking-sidechains">docking sidechains</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Docking-Sidechains">Docking Sidechains</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-droppings">droppings</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Coot-Droppings">Coot Droppings</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dual-conformations">dual conformations</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Adding-Alternative-Conformations">Adding Alternative Conformations</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-E">E</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-edit-B_002dfactors">edit B-factors</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atom-Info">Atom Info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-edit-occupancy">edit occupancy</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atom-Info">Atom Info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-edit-_005cchi-angles">edit <em>\chi</em> angles</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Editing-Chi-Angles">Editing Chi Angles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Electron-Density-Server">Electron Density Server</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Get-Coordinates-and-Map-from-EDS">Get Coordinates and Map from EDS</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-executing-commands">executing commands</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scripting">Scripting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-export-map">export map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Export-Map">Export Map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-EYC_002dTIED">EYC-TIED</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Maps-in-General">Maps in General</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-F">F</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-file_002dname-filtering">file-name filtering</a>:</td><td>&nbsp;</td><td valign="top"><a href="#The-file-selector">The file selector</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fill-partial-residues">fill partial residues</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Fill-Partial-Residues">Fill Partial Residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-findligand">findligand</a>:</td><td>&nbsp;</td><td valign="top"><a href="#findligand">findligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-flip-peptide">flip peptide</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Pep_002dflip">Pep-flip</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fragment-direction-change">fragment direction change</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Reversing-Direction-of-Fragment">Reversing Direction of Fragment</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-frame-rate">frame rate</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Frame-Rate">Frame Rate</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-G">G</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-gdb">gdb</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Crash">Crash</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-GLN-and-ASN-B_002dfactor-Outliers">GLN and ASN B-factor Outliers</a>:</td><td>&nbsp;</td><td valign="top"><a href="#GLN-and-ASN-B_002dfactor-Outliers">GLN and ASN B-factor Outliers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-goose">goose</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Download-coordinates">Download coordinates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-guano">guano</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Coot-Droppings">Coot Droppings</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-guile">guile</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scheme">Scheme</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-H">H</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-helix-placement">helix placement</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Place-Helix">Place Helix</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hydrogens">hydrogens</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Bond-Parameters">Bond Parameters</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-I">I</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-initialization-file">initialization file</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scheme">Scheme</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-L">L</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Least-Squares-Fitting">Least Squares Fitting</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Least_002dSquares-Fitting">Least-Squares Fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ligand-orientation">ligand orientation</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Flip-Ligand">Flip Ligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ligand-torsion-angles">ligand torsion angles</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Torsion-General">Torsion General</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ligand_002c-overlay">ligand, overlay</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Ligand-Overlaying">Ligand Overlaying</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ligands">ligands</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Importing-Ligands_002fMonomers">Importing Ligands/Monomers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ligands-1">ligands</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Find-Ligands">Find Ligands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ligands_002c-flexible">ligands, flexible</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Find-Ligands">Find Ligands</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-M">M</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-mainchain">mainchain</a>:</td><td>&nbsp;</td><td valign="top"><a href="#C_002dalpha-_002d_003e-Mainchain">C-alpha -&gt; Mainchain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mainchain-torsions">mainchain torsions</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Backbone-Torsion-Angles">Backbone Torsion Angles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map-box-radius">map box radius</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Extent">Map Extent</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map-changing-_0028for-refinement_002fbuilding_0029">map changing (for refinement/building)</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Changing-the-Map-for-Building_002fRefinement">Changing the Map for Building/Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map-extent">map extent</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Extent">Map Extent</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map-line-width">map line width</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Line-Width">Map Line Width</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map-re_002dInterpolation">map re-Interpolation</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Re_002dInterpolation">Map Re-Interpolation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map-scrolling">map scrolling</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scrollwheel">Scrollwheel</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map-transformation">map transformation</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Transformation">Map Transformation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-masks">masks</a>:</td><td>&nbsp;</td><td valign="top"><a href="#masks">masks</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mean-B_002dfactor">mean B-factor</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mean-and-Median-Temperature-Factors">Mean and Median Temperature Factors</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-median-B_002dfactor">median B-factor</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mean-and-Median-Temperature-Factors">Mean and Median Temperature Factors</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-merge-molecules">merge molecules</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Merge-Molecules">Merge Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-missing-symmetry">missing symmetry</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Symmetry">Symmetry</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mmCIF-dictionary">mmCIF dictionary</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Dictionary">Dictionary</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-modelling-toolbar">modelling toolbar</a>:</td><td>&nbsp;</td><td valign="top"><a href="#The-Modelling-Toolbar">The Modelling Toolbar</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-modified-labels">modified labels</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Modifying-the-Labels-on-the-Model_002fFit_002fRefine-dialog">Modifying the Labels on the Model/Fit/Refine dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule-centre">molecule centre</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Moving-Molecules_002fLigands">Moving Molecules/Ligands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule-number">molecule number</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Molecule-Number">Molecule Number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molprobity-Tools">Molprobity Tools</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Molprobity-Tools-Interface">Molprobity Tools Interface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-monomers">monomers</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Importing-Ligands_002fMonomers">Importing Ligands/Monomers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mouse">mouse</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Hardware-Requirements">Hardware Requirements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mouse-buttons">mouse buttons</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mousing-and-Keyboarding">Mousing and Keyboarding</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-moving-ligands">moving ligands</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Moving-Molecules_002fLigands">Moving Molecules/Ligands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-moving-molecules">moving molecules</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Moving-Molecules_002fLigands">Moving Molecules/Ligands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Moving-Zero-Occupancy-Atoms">Moving Zero Occupancy Atoms</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Moving-Zero-Occupancy-Atoms">Moving Zero Occupancy Atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dmutate">multi-mutate</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mutation">Mutation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multiple-coordinates-files">multiple coordinates files</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Reading-coordinates">Reading coordinates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate">mutate</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mutation">Mutation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutating-DNA">mutating DNA</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mutation">Mutation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutating-RNA">mutating RNA</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mutation">Mutation</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-N">N</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-NCS">NCS</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Bond-Parameters">Bond Parameters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-NCS-averaging">NCS averaging</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Bond-Parameters">Bond Parameters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-NCS-edits">NCS edits</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Applying-NCS-Edits">Applying NCS Edits</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-negative-contour-levels">negative contour levels</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Contour-_0060_0060Scrolling_0027_0027-Limits">Map Contour ``Scrolling'' Limits</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-O">O</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-OCA">OCA</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Download-coordinates">Download coordinates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-OpenGL">OpenGL</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Issues">Display Issues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Orientation-Axes">Orientation Axes</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Orientation-Axes">Orientation Axes</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-origin-marker">origin marker</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Issues">Display Issues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-output">output</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Program-Output">Program Output</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-overlaying-ligands">overlaying ligands</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Ligand-Overlaying">Ligand Overlaying</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-OXT-atom">OXT atom</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Add-OXT-Atom-to-Residue">Add OXT Atom to Residue</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-P">P</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-packing-diagram">packing diagram</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Symmetry">Symmetry</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Patterson">Patterson</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Pattersons">Pattersons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pepflip">pepflip</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Pep_002dflip">Pep-flip</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-peptide-restraints_002c-planar">peptide restraints, planar</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Planar-Peptide-Restraints">Planar Peptide Restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-PHS-data">PHS data</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Create-a-Map">Create a Map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-PHS-data-format">PHS data format</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Create-a-Map">Create a Map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-picking">picking</a>:</td><td>&nbsp;</td><td valign="top"><a href="#sec_005fpicking">sec_picking</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pink-pointer">pink pointer</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rotation-Centre-Pointer">Rotation Centre Pointer</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-planar-peptide-restraints">planar peptide restraints</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Planar-Peptide-Restraints">Planar Peptide Restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-planes">planes</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-polynucleotides">polynucleotides</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Building-Ideal-DNA-and-RNA">Building Ideal DNA and RNA</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dmanipulation_002dhook">post-manipulation-hook</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Post_002dmanipulation_002dhook">Post-manipulation-hook</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Print-Sequence">Print Sequence</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Print-Sequence">Print Sequence</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-probe">probe</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Molprobity-Tools-Interface">Molprobity Tools Interface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-python">python</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Python">Python</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-R">R</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Ramachandran-plot">Ramachandran plot</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Ramachandran-Plots">Ramachandran Plots</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Raster3D">Raster3D</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Raster3D">Raster3D</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reading-multiple-pdb-files">reading multiple pdb files</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Reading-coordinates">Reading coordinates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-recentring-view">recentring view</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Recentring-View">Recentring View</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-recover-session">recover session</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Crash">Crash</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-redo">redo</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Backups-and-Undo">Backups and Undo</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reduce">reduce</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Molprobity-Tools-Interface">Molprobity Tools Interface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reference-manual">reference manual</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Introduction">Introduction</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reference-structures">reference structures</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Environment-Variables">Environment Variables</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine-single-click">refine single click</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refinement">refinement</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refinement-weight">refinement weight</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refinement_002c-rigid-body">refinement, rigid body</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rigid-Body-Refinement">Rigid Body Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refinement_002c-simplex">refinement, simplex</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Simplex-Refinement">Simplex Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refining-residues">refining residues</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Refining-Specific-Residues">Refining Specific Residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac">refmac</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Running-Refmac">Running Refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac-map-colour">refmac map colour</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Running-Refmac">Running Refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac-parameters">refmac parameters</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Running-Refmac">Running Refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac_002c-default">refmac, default</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Running-Refmac">Running Refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-regularization">regularization</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-render">render</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Raster3D">Raster3D</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-renumbering-residues">renumbering residues</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mutation">Mutation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue-info">residue info</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Atom-Info">Atom Info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-resizing-view">resizing view</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Resizing-View">Resizing View</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-restore-after-crash">restore after crash</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Backups-and-Undo">Backups and Undo</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reverse-direction">reverse direction</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Reversing-Direction-of-Fragment">Reversing Direction of Fragment</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rigid-body-fit">rigid body fit</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rigid-Body-Refinement">Rigid Body Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-RNA_002c-ideal">RNA, ideal</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Building-Ideal-DNA-and-RNA">Building Ideal DNA and RNA</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-RNA_002c-mutating">RNA, mutating</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mutation">Mutation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotamers">rotamers</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rotamers">Rotamers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotate_002ftranslate_002c-manual">rotate/translate, manual</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rotate_002fTranslate-Zone">Rotate/Translate Zone</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotation-centre">rotation centre</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Recentring-View">Recentring View</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotation-centre-pointer">rotation centre pointer</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Rotation-Centre-Pointer">Rotation Centre Pointer</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-running-refmac">running refmac</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Running-Refmac">Running Refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-running-SHELXL">running SHELXL</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Running-SHELXL">Running SHELXL</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-S">S</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-save-coordinates-directory">save coordinates directory</a>:</td><td>&nbsp;</td><td valign="top"><a href="#The-file-selector">The file selector</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scheme">scheme</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scheme">Scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-screenshot_002e">screenshot.</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Screenshot">Screenshot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scripting">scripting</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scripting">Scripting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Scroll">Scroll</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Manager">Display Manager</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scroll-wheel_002c-map-for">scroll wheel, map for</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scroll_002dwheel">Scroll-wheel</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scroll_002c-map-contour-change-by">scroll, map contour change by</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scrollwheel">Scrollwheel</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scroll_002dwheel">scroll-wheel</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scrollwheel">Scrollwheel</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Secondary-Structure-Matching-_0028SSM_0029">Secondary Structure Matching (SSM)</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Secondary-Structure-Matching-_0028SSM_0029">Secondary Structure Matching (SSM)</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sequence-view">sequence view</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Sequence-View">Sequence View</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmatrix"><code>set-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotation_002dcentre">set-rotation-centre</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Recentring-View">Recentring View</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-setting-space-group">setting space group</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Setting-the-Space-Group">Setting the Space Group</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sharpening_002c-map">sharpening, map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Sharpening">Map Sharpening</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-SHELX-_002eins">SHELX .ins</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Reading-coordinates">Reading coordinates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-SHELXL">SHELXL</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Running-SHELXL">Running SHELXL</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sidechains">sidechains</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Docking-Sidechains">Docking Sidechains</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-simplex-refinement">simplex refinement</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Simplex-Refinement">Simplex Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-single-click-refine">single click refine</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-skeleton-updating">skeleton updating</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Skeletonization">Skeletonization</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-skeleton_002c-missing">skeleton, missing</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Baton-Building">Baton Building</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-skeletonization">skeletonization</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Skeletonization">Skeletonization</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-slab">slab</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Clipping-Manipulation">Clipping Manipulation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sliding">sliding</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Recentring-View">Recentring View</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-slow-computer">slow computer</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Slow-Computer-Configuration">Slow Computer Configuration</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-SMILES-strings">SMILES strings</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Ligand-from-SMILES-strings">Ligand from SMILES strings</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-space-group-names">space group names</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Reading-coordinates">Reading coordinates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-space-group-operators">space group operators</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Space-Group-and-Symmetry">Space Group and Symmetry</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sphere-refinement">sphere refinement</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Sphere-Refinement">Sphere Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-startup-dialog-_0028state_0029">startup dialog (state)</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Coot-State">Coot State</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-startup-settings-_0028python_0029">startup settings (python)</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Python">Python</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-startup-settings-_0028scheme_0029">startup settings (scheme)</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Scheme">Scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-state">state</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Coot-State">Coot State</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-superposition">superposition</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Secondary-Structure-Matching-_0028SSM_0029">Secondary Structure Matching (SSM)</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-symmetry">symmetry</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Symmetry">Symmetry</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-symmetry-operators">symmetry operators</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Space-Group-and-Symmetry">Space Group and Symmetry</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-T">T</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-terminal-oxygen">terminal oxygen</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Add-OXT-Atom-to-Residue">Add OXT Atom to Residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-terminal-residue">terminal residue</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Add-Terminal-Residue">Add Terminal Residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-thickness-of-density-lines">thickness of density lines</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Map-Line-Width">Map Line Width</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-tooltips">tooltips</a>:</td><td>&nbsp;</td><td valign="top"><a href="#General-Features">General Features</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-torsion-angles_002c-ligand">torsion angles, ligand</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Torsion-General">Torsion General</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-torsion-general">torsion general</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Torsion-General">Torsion General</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-torsion-restraints">torsion restraints</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-torsions">torsions</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Editing-Chi-Angles">Editing Chi Angles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-trackball_002c-virtual">trackball, virtual</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Virtual-Trackball">Virtual Trackball</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-traffic-lights">traffic lights</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-translate-molecule">translate molecule</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Moving-Molecules_002fLigands">Moving Molecules/Ligands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-translation_002c-keyboard">translation, keyboard</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Keyboard-Z-Translation">Keyboard Z Translation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-translation_002c-mouse">translation, mouse</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Mouse-Z-Translation">Mouse Z Translation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-trimming-atoms">trimming atoms</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Trimming-Atoms">Trimming Atoms</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-U">U</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-undo">undo</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Backups-and-Undo">Backups and Undo</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unexplained-density">unexplained density</a>:</td><td>&nbsp;</td><td valign="top"><a href="#sec_005fblobs">sec_blobs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unit-cell">unit cell</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Unit-Cell">Unit Cell</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-UNK-residue"><code>UNK</code> residue</a>:</td><td>&nbsp;</td><td valign="top"><a href="#The-UNK-residue-type">The UNK residue type</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unmodelled-density">unmodelled density</a>:</td><td>&nbsp;</td><td valign="top"><a href="#sec_005fblobs">sec_blobs</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-V">V</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Validation-Graphs">Validation Graphs</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Validation-Graphs">Validation Graphs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-version-number">version number</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Version-number">Version number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-view-matrix">view matrix</a>:</td><td>&nbsp;</td><td valign="top"><a href="#View-Matrix">View Matrix</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-W">W</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-waters_002c-finding">waters, finding</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Find-Waters">Find Waters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-web-page">web page</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Web-Page">Web Page</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-weight_002c-real-space-refinement">weight, real space refinement</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Regularization-and-Real-Space-Refinement">Regularization and Real Space Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-width_002c-bonds">width, bonds</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Bond-Parameters">Bond Parameters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write-map">write map</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Export-Map">Export Map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-writing-PDBs">writing PDBs</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Writing-PDB-files">Writing PDB files</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-Y">Y</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-yellow-box">yellow box</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Display-Issues">Display Issues</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Concept-Index_cp_letter-Z">Z</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-z_002drotation">z-rotation</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Virtual-Trackball">Virtual Trackball</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-zero-occupancy">zero occupancy</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Zero-Occupancy-Marker">Zero Occupancy Marker</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-zoom">zoom</a>:</td><td>&nbsp;</td><td valign="top"><a href="#Resizing-View">Resizing View</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-zoom_002c-slider">zoom, slider</a>:</td><td>&nbsp;</td><td valign="top"><a href="#more-on-zooming">more on zooming</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
</table>
<table><tr><th valign="top">Jump to: &nbsp; </th><td><a class="summary-letter" href="#Concept-Index_cp_symbol-1"><b>.</b></a>
 &nbsp; 
<br>
<a class="summary-letter" href="#Concept-Index_cp_letter-A"><b>A</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-B"><b>B</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-C"><b>C</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-D"><b>D</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-E"><b>E</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-F"><b>F</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-G"><b>G</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-H"><b>H</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-I"><b>I</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-L"><b>L</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-M"><b>M</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-N"><b>N</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-O"><b>O</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-P"><b>P</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-R"><b>R</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-S"><b>S</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-T"><b>T</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-U"><b>U</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-V"><b>V</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-W"><b>W</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-Y"><b>Y</b></a>
 &nbsp; 
<a class="summary-letter" href="#Concept-Index_cp_letter-Z"><b>Z</b></a>
 &nbsp; 
</td></tr></table>

<hr>
<a name="Function-Index"></a>
<div class="header">
<p>
Previous: <a href="#Concept-Index" accesskey="p" rel="previous">Concept Index</a>, Up: <a href="#Top" accesskey="u" rel="up">Top</a> &nbsp; [<a href="#SEC_Contents" title="Table of contents" rel="contents">Contents</a>][<a href="#Concept-Index" title="Index" rel="index">Index</a>]</p>
</div>
<a name="Function-Index-1"></a>
<h2 class="unnumbered">Function Index</h2>
<table><tr><th valign="top">Jump to: &nbsp; </th><td><a class="summary-letter" href="#Function-Index_fn_symbol-1"><b>%</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_symbol-2"><b>*</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_symbol-3"><b>3</b></a>
 &nbsp; 
<br>
<a class="summary-letter" href="#Function-Index_fn_letter-A"><b>A</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-B"><b>B</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-C"><b>C</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-D"><b>D</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-E"><b>E</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-F"><b>F</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-G"><b>G</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-H"><b>H</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-I"><b>I</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-J"><b>J</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-K"><b>K</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-L"><b>L</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-M"><b>M</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-N"><b>N</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-O"><b>O</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-P"><b>P</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-Q"><b>Q</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-R"><b>R</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-S"><b>S</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-T"><b>T</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-U"><b>U</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-V"><b>V</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-W"><b>W</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-Z"><b>Z</b></a>
 &nbsp; 
</td></tr></table>
<table class="index-fn" border="0">
<tr><td></td><th align="left">Index Entry</th><td>&nbsp;</td><th align="left"> Section</th></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_symbol-1">%</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-_0025coot_002dlistener_002dsocket"><code>%coot-listener-socket</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remote_002dcontrol">remote-control</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_symbol-2">*</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002aadd_002dlinked_002dresidue_002dtree_002dcorrelation_002dcut_002doff_002a"><code>*add-linked-residue-tree-correlation-cut-off*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dlinked_002dcho">add-linked-cho</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002aannotations_002a"><code>*annotations*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002acontinue_002dmulti_002drefine_002a"><code>*continue-multi-refine*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002adb_002dloop_002dpreserve_002dresidue_002dnames_002a"><code>*db-loop-preserve-residue-names*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002adefault_002dball_002dand_002dstick_002dselection_002a"><code>*default-ball-and-stick-selection*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002ado_002dcoot_002dtips_002dflag_002a"><code>*do-coot-tips-flag*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot">coot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002aimol_002djligand_002dlink_002a"><code>*imol-jligand-link*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#jligand">jligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002ainteractive_002dprobe_002dis_002dOK_003f_002a"><code>*interactive-probe-is-OK?*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002akey_002dbindings_002a"><code>*key-bindings*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002aligand_002dcheck_002dresolution_002dlimit_002a"><code>*ligand-check-resolution-limit*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ligand_002dcheck">ligand-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002apisa_002dcommand_002a"><code>*pisa-command*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#parse_002dpisa_002dxml">parse-pisa-xml</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002aprobe_002dcommand_002a"><code>*probe-command*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot">coot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002areduce_002dcommand_002a"><code>*reduce-command*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot">coot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002ato_002djligand_002dsecret_002dfile_002dname_002a"><code>*to-jligand-secret-file-name*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#jligand">jligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-_002ause_002dmogul_002a"><code>*use-mogul*</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_symbol-3">3</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-3_002dletter_002dcode_002d_003esingle_002dletter"><code>3-letter-code-&gt;single-letter</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-A">A</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-a_002drapper_002dgui"><code>a-rapper-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#a_002drapper_002dgui">a-rapper-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-absolutify"><code>absolutify</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-accept_002dbaton_002dposition"><code>accept-baton-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#accept_002dbaton_002dposition">accept-baton-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-accept_002dmoving_002datoms"><code>accept-moving-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#accept_002dmoving_002datoms">accept-moving-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-accept_002dregularizement"><code>accept-regularizement</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#accept_002dregularizement">accept-regularizement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-accept_002dreject_002ddialog_002ddocked_002dshow_002dstate"><code>accept-reject-dialog-docked-show-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#accept_002dreject_002ddialog_002ddocked_002dshow_002dstate">accept-reject-dialog-docked-show-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-accept_002dreject_002ddialog_002ddocked_002dstate"><code>accept-reject-dialog-docked-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#accept_002dreject_002ddialog_002ddocked_002dstate">accept-reject-dialog-docked-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-active_002dresidue"><code>active-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#active_002dresidue">active-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-active_002dresidue_002dpy"><code>active-residue-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#active_002dresidue_002dpy">active-residue-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002daction_002dview"><code>add-action-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002daction_002dview">add-action-view</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dalt_002dconf_002dpy"><code>add-alt-conf-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dalt_002dconf_002dpy">add-alt-conf-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dalt_002dconf_002dscm"><code>add-alt-conf-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dalt_002dconf_002dscm">add-alt-conf-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dannotation_002dhere"><code>add-annotation-here</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dbase_002drestraint"><code>add-base-restraint</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#user_002ddefine_002drestraints">user-define-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dcoordinates_002dglob_002dextension"><code>add-coordinates-glob-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dcoordinates_002dglob_002dextension">add-coordinates-glob-extension</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002ddata_002dglob_002dextension"><code>add-data-glob-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002ddata_002dglob_002dextension">add-data-glob-extension</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002ddictionary_002dglob_002dextension"><code>add-dictionary-glob-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002ddictionary_002dglob_002dextension">add-dictionary-glob-extension</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002ddipole_002dfor_002dresidues_002dpy"><code>add-dipole-for-residues-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002ddipole_002dfor_002dresidues_002dpy">add-dipole-for-residues-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002ddipole_002dfor_002dresidues_002dscm"><code>add-dipole-for-residues-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002ddipole_002dfor_002dresidues_002dscm">add-dipole-for-residues-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002ddipole_002dpy"><code>add-dipole-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002ddipole_002dpy">add-dipole-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002ddipole_002dscm"><code>add-dipole-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002ddipole_002dscm">add-dipole-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dextra_002dbond_002drestraint"><code>add-extra-bond-restraint</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dextra_002dbond_002drestraint">add-extra-bond-restraint</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dextra_002dgeman_002dmcclure_002drestraint"><code>add-extra-geman-mcclure-restraint</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dextra_002dgeman_002dmcclure_002drestraint">add-extra-geman-mcclure-restraint</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dextra_002drestraints_002dto_002dother_002dmolecule"><code>add-extra-restraints-to-other-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dgeometry_002ddistance"><code>add-geometry-distance</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dgeometry_002ddistance">add-geometry-distance</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dheader_002dsecondary_002dstructure_002dinfo"><code>add-header-secondary-structure-info</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dheader_002dsecondary_002dstructure_002dinfo">add-header-secondary-structure-info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dhydrogens_002dfrom_002dfile"><code>add-hydrogens-from-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dhydrogens_002dfrom_002dfile">add-hydrogens-from-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dhydrogens_002dusing_002drefmac_002dinner"><code>add-hydrogens-using-refmac-inner</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dkey_002dbinding"><code>add-key-binding</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule"><code>add-ligand-delete-residue-copy-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dligand_002ddelete_002dresidue_002dcopy_002dmolecule">add-ligand-delete-residue-copy-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dligand_002dsearch_002dligand_002dmolecule"><code>add-ligand-search-ligand-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dligand_002dsearch_002dligand_002dmolecule">add-ligand-search-ligand-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule"><code>add-ligand-search-wiggly-ligand-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dligand_002dsearch_002dwiggly_002dligand_002dmolecule">add-ligand-search-wiggly-ligand-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dmap_002dglob_002dextension"><code>add-map-glob-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dmap_002dglob_002dextension">add-map-glob-extension</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dmolecule"><code>add-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dmolecule">add-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dncs_002dmatrix"><code>add-ncs-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dncs_002dmatrix">add-ncs-matrix</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dnon_002dauto_002dload_002dresidue_002dname"><code>add-non-auto-load-residue-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dnon_002dauto_002dload_002dresidue_002dname">add-non-auto-load-residue-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dnucleotide"><code>add-nucleotide</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dnucleotide">add-nucleotide</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002domega_002dtorsion_002drestriants"><code>add-omega-torsion-restriants</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002domega_002dtorsion_002drestriants">add-omega-torsion-restriants</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dplanar_002dpeptide_002drestraints"><code>add-planar-peptide-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dplanar_002dpeptide_002drestraints">add-planar-peptide-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dsimple_002dcoot_002dmenu_002dmenuitem"><code>add-simple-coot-menu-menuitem</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dstatus_002dbar_002dtext"><code>add-status-bar-text</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dstatus_002dbar_002dtext">add-status-bar-text</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dstrict_002dncs_002dmatrix"><code>add-strict-ncs-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dstrict_002dncs_002dmatrix">add-strict-ncs-matrix</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dtarget_002dposition_002drestraint_002dfor_002dintermediate_002datom_002dpy"><code>add-target-position-restraint-for-intermediate-atom-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dtarget_002dposition_002drestraint_002dfor_002dintermediate_002datom_002dpy">add-target-position-restraint-for-intermediate-atom-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dterminal_002dresidue"><code>add-terminal-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dterminal_002dresidue">add-terminal-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate"><code>add-terminal-residue-do-post-refine-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dterminal_002dresidue_002ddo_002dpost_002drefine_002dstate">add-terminal-residue-do-post-refine-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dterminal_002dresidue_002dusing_002dphi_002dpsi"><code>add-terminal-residue-using-phi-psi</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dterminal_002dresidue_002dusing_002dphi_002dpsi">add-terminal-residue-using-phi-psi</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dtmp_002dextension_002dto"><code>add-tmp-extension-to</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dto_002dlist_002dsection_002dtexis"><code>add-to-list-section-texis</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dview"><code>add-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dview_002ddescription"><code>add-view-description</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dview_002ddescription">add-view-description</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dview_002dhere"><code>add-view-here</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dview_002dhere">add-view-here</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dview_002draw"><code>add-view-raw</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dview_002draw">add-view-raw</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-add_002dview_002dto_002dviews_002dpanel"><code>add-view-to-views-panel</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-additional_002drepresentation_002dby_002dattributes"><code>additional-representation-by-attributes</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#additional_002drepresentation_002dby_002dattributes">additional-representation-by-attributes</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-additional_002drepresentation_002dby_002dstring"><code>additional-representation-by-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#additional_002drepresentation_002dby_002dstring">additional-representation-by-string</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-align_002dand_002dmutate"><code>align-and-mutate</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#align_002dand_002dmutate">align-and-mutate</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-alignment_002dmismatches_002dgui"><code>alignment-mismatches-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-alignment_002dmismatches_002dpy"><code>alignment-mismatches-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#alignment_002dmismatches_002dpy">alignment-mismatches-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-alignment_002dmismatches_002dscm"><code>alignment-mismatches-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#alignment_002dmismatches_002dscm">alignment-mismatches-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-all_002dresidues_002dwith_002dserial_002dnumbers_002dscm"><code>all-residues-with-serial-numbers-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#all_002dresidues_002dwith_002dserial_002dnumbers_002dscm">all-residues-with-serial-numbers-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-all_002dtrue_003f"><code>all-true?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-allow_002dduplicate_002dsequence_002dnumbers"><code>allow-duplicate-sequence-numbers</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#allow_002dduplicate_002dsequence_002dnumbers">allow-duplicate-sequence-numbers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-another_002dlevel"><code>another-level</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#another_002dlevel">another-level</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber"><code>another-level-from-map-molecule-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#another_002dlevel_002dfrom_002dmap_002dmolecule_002dnumber">another-level-from-map-molecule-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-append_002ddir_002ddir"><code>append-dir-dir</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-append_002ddir_002dfile"><code>append-dir-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-apply_002dlsq_002dmatches"><code>apply-lsq-matches</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#apply_002dlsq_002dmatches">apply-lsq-matches</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-apply_002dncs_002dto_002dview_002dorientation"><code>apply-ncs-to-view-orientation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#apply_002dncs_002dto_002dview_002dorientation">apply-ncs-to-view-orientation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre"><code>apply-ncs-to-view-orientation-and-screen-centre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#apply_002dncs_002dto_002dview_002dorientation_002dand_002dscreen_002dcentre">apply-ncs-to-view-orientation-and-screen-centre</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-apply_002dredo"><code>apply-redo</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#apply_002dredo">apply-redo</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-apply_002dundo"><code>apply-undo</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#apply_002dundo">apply-undo</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-assign_002dfasta_002dsequence"><code>assign-fasta-sequence</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#assign_002dfasta_002dsequence">assign-fasta-sequence</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-assign_002dhetatms"><code>assign-hetatms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#assign_002dhetatms">assign-hetatms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-assign_002dpir_002dsequence"><code>assign-pir-sequence</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#assign_002dpir_002dsequence">assign-pir-sequence</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-assign_002dsequence_002dfrom_002dfile"><code>assign-sequence-from-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#assign_002dsequence_002dfrom_002dfile">assign-sequence-from-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-assign_002dsequence_002dfrom_002dstring"><code>assign-sequence-from-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#assign_002dsequence_002dfrom_002dstring">assign-sequence-from-string</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-associate_002dpir_002dfile"><code>associate-pir-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-associate_002dpir_002dwith_002dmolecule_002dgui"><code>associate-pir-with-molecule-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom_002dinfo_002dstring_002dpy"><code>atom-info-string-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#atom_002dinfo_002dstring_002dpy">atom-info-string-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom_002dspec_002d_003ealt_002dloc"><code>atom-spec-&gt;alt-loc</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom_002dspec_002d_003eatom_002dname"><code>atom-spec-&gt;atom-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom_002dspec_002d_003echain_002did"><code>atom-spec-&gt;chain-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom_002dspec_002d_003eins_002dcode"><code>atom-spec-&gt;ins-code</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom_002dspec_002d_003eres_002dno"><code>atom-spec-&gt;res-no</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom_002dspec_002dto_002datom_002dindex"><code>atom-spec-to-atom-index</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#atom_002dspec_002dto_002datom_002dindex">atom-spec-to-atom-index</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-atom_002dspecs"><code>atom-specs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-auto_002dfit_002dbest_002drotamer"><code>auto-fit-best-rotamer</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#auto_002dfit_002dbest_002drotamer">auto-fit-best-rotamer</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-auto_002dfit_002drotamer_002dactive_002dresidue"><code>auto-fit-rotamer-active-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-auto_002dload_002ddictionary"><code>auto-load-dictionary</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#auto_002dload_002ddictionary">auto-load-dictionary</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate"><code>auto-read-do-difference-map-too-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#auto_002dread_002ddo_002ddifference_002dmap_002dtoo_002dstate">auto-read-do-difference-map-too-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-auto_002dread_002dmake_002dand_002ddraw_002dmaps"><code>auto-read-make-and-draw-maps</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps">auto-read-make-and-draw-maps</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz"><code>auto-read-make-and-draw-maps-from-mtz</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#auto_002dread_002dmake_002dand_002ddraw_002dmaps_002dfrom_002dmtz">auto-read-make-and-draw-maps-from-mtz</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-auto_002dweight_002dfor_002drefinement"><code>auto-weight-for-refinement</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-average_002dmap_002dpy"><code>average-map-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#average_002dmap_002dpy">average-map-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-average_002dmap_002dscm"><code>average-map-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#average_002dmap_002dscm">average-map-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-average_002dtemperature_002dfactor"><code>average-temperature-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#average_002dtemperature_002dfactor">average-temperature-factor</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-B">B</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-b_002dfactor_002dfrom_002dmap"><code>b-factor-from-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#b_002dfactor_002dfrom_002dmap">b-factor-from-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-background_002ddemo"><code>background-demo</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#background_002ddemo">background-demo</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-background_002dis_002dblack_002dp"><code>background-is-black-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#background_002dis_002dblack_002dp">background-is-black-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-backup_002dcompress_002dfiles_002dstate"><code>backup-compress-files-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#backup_002dcompress_002dfiles_002dstate">backup-compress-files-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-backup_002dstate"><code>backup-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#backup_002dstate">backup-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-BALL_005fAND_005fSTICK"><code>BALL_AND_STICK</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-baton_002dbuild_002ddelete_002dlast_002dresidue"><code>baton-build-delete-last-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#baton_002dbuild_002ddelete_002dlast_002dresidue">baton-build-delete-last-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-baton_002dtip_002dprevious"><code>baton-tip-previous</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#baton_002dtip_002dprevious">baton-tip-previous</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-baton_002dtip_002dtry_002danother"><code>baton-tip-try-another</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#baton_002dtip_002dtry_002danother">baton-tip-try-another</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-blob_002dunder_002dpointer_002dto_002dscreen_002dcentre"><code>blob-under-pointer-to-screen-centre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#blob_002dunder_002dpointer_002dto_002dscreen_002dcentre">blob-under-pointer-to-screen-centre</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-brief_002datom_002dlabels_002dstate"><code>brief-atom-labels-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#brief_002datom_002dlabels_002dstate">brief-atom-labels-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-brighten_002dmap"><code>brighten-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-brighten_002dmaps"><code>brighten-maps</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-browser_002durl"><code>browser-url</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#browser_002durl">browser-url</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-C">C</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-cache_002dor_002dnet_002dget_002dimage"><code>cache-or-net-get-image</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002drecent_002dpdbe">get-recent-pdbe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cached_002dpisa_002danalysis"><code>cached-pisa-analysis</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#parse_002dpisa_002dxml">parse-pisa-xml</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-calc_002dphases_002dgeneric"><code>calc-phases-generic</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#calc_002dphases_002dgeneric">calc-phases-generic</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ccp4i_002dprojects_002dpy"><code>ccp4i-projects-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ccp4i_002dprojects_002dpy">ccp4i-projects-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ccp4i_002dprojects_002dscm"><code>ccp4i-projects-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ccp4i_002dprojects_002dscm">ccp4i-projects-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cell_002dpy"><code>cell-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#cell_002dpy">cell-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cell_002dscm"><code>cell-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#cell_002dscm">cell-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-center_002datom_002dlabel_002dstatus"><code>center-atom-label-status</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-center_002dof_002dmass"><code>center-of-mass</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-CG_002dspin_002dsearch_002dpy"><code>CG-spin-search-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#CG_002dspin_002dsearch_002dpy">CG-spin-search-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-CG_002dspin_002dsearch_002dscm"><code>CG-spin-search-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#CG_002dspin_002dsearch_002dscm">CG-spin-search-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-chain_002dfragments_002dscm"><code>chain-fragments-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#chain_002dfragments_002dscm">chain-fragments-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-chain_002did_002dscm"><code>chain-id-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#chain_002did_002dscm">chain-id-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-chain_002dids"><code>chain-ids</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-chain_002dn_002dresidues"><code>chain-n-residues</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#chain_002dn_002dresidues">chain-n-residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-change_002dcontour_002dlevel"><code>change-contour-level</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#change_002dcontour_002dlevel">change-contour-level</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-change_002dmodel_002dmolecule_002drepresentation_002dmode"><code>change-model-molecule-representation-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#change_002dmodel_002dmolecule_002drepresentation_002dmode">change-model-molecule-representation-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-change_002dresidue_002dnumber"><code>change-residue-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#change_002dresidue_002dnumber">change-residue-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-check_002dchiral_002dvolumes"><code>check-chiral-volumes</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#check_002dchiral_002dvolumes">check-chiral-volumes</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-check_002ddir_002dand_002dget_002durl"><code>check-dir-and-get-url</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002debi">get-ebi</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-check_002dfor_002ddark_002dblue_002ddensity"><code>check-for-dark-blue-density</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#check_002dfor_002ddark_002dblue_002ddensity">check-for-dark-blue-density</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-chiral_002dcentre_002dinverter_002dscm_002dversion"><code>chiral-centre-inverter-scm-version</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cif_002dfile_002dfor_002dcomp_002did"><code>cif-file-for-comp-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#redefine_002dfunctions">redefine-functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cis_002dpeptides_002dgui"><code>cis-peptides-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clashes_002dwith_002dsymmetry"><code>clashes-with-symmetry</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clashes_002dwith_002dsymmetry">clashes-with-symmetry</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002dall_002dfixed_002datoms"><code>clear-all-fixed-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dall_002dfixed_002datoms">clear-all-fixed-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002dall_002dviews"><code>clear-all-views</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dall_002dviews">clear-all-views</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile"><code>clear-and-update-model-molecule-from-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dand_002dupdate_002dmodel_002dmolecule_002dfrom_002dfile">clear-and-update-model-molecule-from-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002dand_002dupdate_002dmolecule"><code>clear-and-update-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dand_002dupdate_002dmolecule">clear-and-update-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002dbackup_002dgui"><code>clear-backup-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dbackup">clear-backup</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002dbackups_002dmaybe"><code>clear-backups-maybe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dbackup">clear-backup</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002dball_002dand_002dstick"><code>clear-ball-and-stick</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dball_002dand_002dstick">clear-ball-and-stick</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002ddots"><code>clear-dots</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002ddots">clear-dots</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002ddots_002dby_002dname"><code>clear-dots-by-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002ddots_002dby_002dname">clear-dots-by-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-clear_002dpending_002dpicks"><code>clear-pending-picks</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dpending_002dpicks">clear-pending-picks</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-click_002dselect_002dresidues_002dfor_002djligand"><code>click-select-residues-for-jligand</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#jligand_002dgui">jligand-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-close_002dfloat_003f"><code>close-float?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-close_002dmolecule"><code>close-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#close_002dmolecule">close-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-closest_002datom"><code>closest-atom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#closest_002datom">closest-atom</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-closest_002datom_002dpy"><code>closest-atom-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#closest_002datom_002dpy">closest-atom-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-closest_002datom_002draw_002dpy"><code>closest-atom-raw-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#closest_002datom_002draw_002dpy">closest-atom-raw-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-closest_002datom_002draw_002dscm"><code>closest-atom-raw-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#closest_002datom_002draw_002dscm">closest-atom-raw-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-closest_002datom_002dsimple_002dpy"><code>closest-atom-simple-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#closest_002datom_002dsimple_002dpy">closest-atom-simple-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-closest_002datom_002dsimple_002dscm"><code>closest-atom-simple-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#closest_002datom_002dsimple_002dscm">closest-atom-simple-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cns_002d_003ecoot"><code>cns-&gt;coot</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#cns2coot">cns2coot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cns_002dfile_002dhas_002dphases_002dp"><code>cns-file-has-phases-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#cns_002dfile_002dhas_002dphases_002dp">cns-file-has-phases-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-command_002din_002dpath_002dor_002dabsolute_003f"><code>command-in-path-or-absolute?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-command_002din_002dpath_003f"><code>command-in-path?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-command_002din_002dpath_003f-1"><code>command-in-path?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-command_002din_002dpath_003f-2"><code>command-in-path?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-comp_002did_002dto_002dname_002dpy"><code>comp-id-to-name-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#comp_002did_002dto_002dname_002dpy">comp-id-to-name-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-comp_002did_002dto_002dname_002dscm"><code>comp-id-to-name-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#comp_002did_002dto_002dname_002dscm">comp-id-to-name-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-complex_002dmammal_002dtree"><code>complex-mammal-tree</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dlinked_002dcho">add-linked-cho</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-complex_002dplant_002dtree"><code>complex-plant-tree</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dlinked_002dcho">add-linked-cho</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-contact_002dscore_002dligand"><code>contact-score-ligand</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#contact_002dscore_002disolated_002dligand">contact-score-isolated-ligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-control_002dkey_002dfor_002drotate_002dstate"><code>control-key-for-rotate-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#control_002dkey_002dfor_002drotate_002dstate">control-key-for-rotate-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dclear_002dbackup_002dor_002dreal_002dexit"><code>coot-clear-backup-or-real-exit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dclear_002dbackup_002dor_002dreal_002dexit">coot-clear-backup-or-real-exit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dgui"><code>coot-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dhas_002dpygtk_003f"><code>coot-has-pygtk?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dlistener_002didle_002dfunction_002dproc"><code>coot-listener-idle-function-proc</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remote_002dcontrol">remote-control</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dmenubar_002dmenu"><code>coot-menubar-menu</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dmkdir"><code>coot-mkdir</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dno_002dstate_002dreal_002dexit"><code>coot-no-state-real-exit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dno_002dstate_002dreal_002dexit">coot-no-state-real-exit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dreal_002dexit"><code>coot-real-exit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dreal_002dexit">coot-real-exit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dreplace_002dstring"><code>coot-replace-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dsave_002dstate_002dand_002dexit"><code>coot-save-state-and-exit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dsave_002dstate_002dand_002dexit">coot-save-state-and-exit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dsocket_002dtimeout_002dfunc"><code>coot-socket-timeout-func</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remote_002dcontrol">remote-control</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dsplit_002dversion_002dstring"><code>coot-split-version-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dthread_002ddispatcher"><code>coot-thread-dispatcher</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002drecent_002dpdbe">get-recent-pdbe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-coot_002dupdates_002derror_002dhandler"><code>coot-updates-error-handler</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cootaneer_002dgui"><code>cootaneer-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-copy_002dchain"><code>copy-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#copy_002dchain">copy-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm"><code>copy-from-ncs-master-to-chains-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#copy_002dfrom_002dncs_002dmaster_002dto_002dchains_002dscm">copy-from-ncs-master-to-chains-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-copy_002dfrom_002dncs_002dmaster_002dto_002dothers"><code>copy-from-ncs-master-to-others</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#copy_002dfrom_002dncs_002dmaster_002dto_002dothers">copy-from-ncs-master-to-others</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-copy_002dmolecule"><code>copy-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#copy_002dmolecule">copy-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-copy_002dresidue_002drange"><code>copy-residue-range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#copy_002dresidue_002drange">copy-residue-range</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers"><code>copy-residue-range-from-ncs-master-to-others</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#copy_002dresidue_002drange_002dfrom_002dncs_002dmaster_002dto_002dothers">copy-residue-range-from-ncs-master-to-others</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-crankshaft_002dpeptide_002drotation_002doptimization_002dpy"><code>crankshaft-peptide-rotation-optimization-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#crankshaft_002dpeptide_002drotation_002doptimization_002dpy">crankshaft-peptide-rotation-optimization-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-crankshaft_002dpeptide_002drotation_002doptimization_002dscm"><code>crankshaft-peptide-rotation-optimization-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#crankshaft_002dpeptide_002drotation_002doptimization_002dscm">crankshaft-peptide-rotation-optimization-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-create_002dpointer_002datom_002dmolecule_002dmaybe"><code>create-pointer-atom-molecule-maybe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#create_002dpointer_002datom_002dmolecule_002dmaybe">create-pointer-atom-molecule-maybe</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-D">D</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-darken_002dmaps"><code>darken-maps</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-data_002dresolution"><code>data-resolution</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#data_002dresolution">data-resolution</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-db_002dmainchain"><code>db-mainchain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#db_002dmainchain">db-mainchain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-db_002dmainchains_002dfragment"><code>db-mainchains-fragment</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#db_002dmainchains_002dfragment">db-mainchains-fragment</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-decoloned_002dbackup_002dfile_002dnames_002dstate"><code>decoloned-backup-file-names-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#decoloned_002dbackup_002dfile_002dnames_002dstate">decoloned-backup-file-names-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-decrease_002dcoot_002dtip_002dnumber"><code>decrease-coot-tip-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#tips_002dgui">tips-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-default_002dnew_002datoms_002db_002dfactor"><code>default-new-atoms-b-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#default_002dnew_002datoms_002db_002dfactor">default-new-atoms-b-factor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dall_002dextra_002drestraints"><code>delete-all-extra-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dall_002dextra_002drestraints">delete-all-extra-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dall_002dsequences_002dfrom_002dmolecule"><code>delete-all-sequences-from-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dall_002dsequences_002dfrom_002dmolecule">delete-all-sequences-from-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002datom"><code>delete-atom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002datom">delete-atom</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002datom_002dby_002dactive_002dresidue"><code>delete-atom-by-active-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dchain"><code>delete-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dchain">delete-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dchecked_002dwaters_002dbaddies"><code>delete-checked-waters-baddies</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dchecked_002dwaters_002dbaddies">delete-checked-waters-baddies</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dcoot_002dbackup_002dfiles"><code>delete-coot-backup-files</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#clear_002dbackup">clear-backup</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dextra_002drestraints_002dfor_002dresidue"><code>delete-extra-restraints-for-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dextra_002drestraints_002dfor_002dresidue">delete-extra-restraints-for-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dhydrogen_002datoms"><code>delete-hydrogen-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dhydrogen_002datoms">delete-hydrogen-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dhydrogens"><code>delete-hydrogens</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dhydrogens">delete-hydrogens</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dresidue"><code>delete-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dresidue">delete-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dresidue_002dhydrogens"><code>delete-residue-hydrogens</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dresidue_002dhydrogens">delete-residue-hydrogens</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dresidue_002drange"><code>delete-residue-range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dresidue_002drange">delete-residue-range</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dresidue_002dsidechain"><code>delete-residue-sidechain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dresidue_002dsidechain">delete-residue-sidechain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dresidue_002dwith_002dfull_002dspec"><code>delete-residue-with-full-spec</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dresidue_002dwith_002dfull_002dspec">delete-residue-with-full-spec</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dresidues_002dpy"><code>delete-residues-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dresidues_002dpy">delete-residues-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dresidues_002dscm"><code>delete-residues-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dresidues_002dscm">delete-residues-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002drestraints"><code>delete-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002drestraints">delete-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dsection_002dtexi_002dfiles"><code>delete-section-texi-files</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dsequence_002dby_002dchain_002did"><code>delete-sequence-by-chain-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dsequence_002dby_002dchain_002did">delete-sequence-by-chain-id</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dsidechain_002drange"><code>delete-sidechain-range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dsidechains_002dfor_002dchain"><code>delete-sidechains-for-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dsidechains_002dfor_002dchain">delete-sidechains-for-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-delete_002dwaters"><code>delete-waters</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#delete_002dwaters">delete-waters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-density_002dat_002dpoint"><code>density-at-point</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#density_002dat_002dpoint">density-at-point</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-density_002dscore_002dresidue"><code>density-score-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#density_002dscore_002dresidue">density-score-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dialog_002dbox_002dof_002dbuttons_002dwith_002dasync_002dligands"><code>dialog-box-of-buttons-with-async-ligands</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002drecent_002dpdbe">get-recent-pdbe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dialog_002dbox_002dof_002dbuttons_002dwith_002dcheck_002dbutton"><code>dialog-box-of-buttons-with-check-button</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dialog_002dbox_002dof_002dpairs_002dof_002dbuttons"><code>dialog-box-of-pairs-of-buttons</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dictionaries_002dread"><code>dictionaries-read</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#dictionaries_002dread">dictionaries-read</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-difference_002dmap"><code>difference-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#difference_002dmap">difference-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-difference_002dmap_002dpeaks"><code>difference-map-peaks</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#difference_002dmap_002dpeaks">difference-map-peaks</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-directory_002das_002dfile_002dname"><code>directory-as-file-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-directory_002das_002dfile_002dname-1"><code>directory-as-file-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-directory_002dfiles"><code>directory-files</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-directory_002dfiles-1"><code>directory-files</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-directory_002dfiles-2"><code>directory-files</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-directory_002dis_002dmodifiable_003f"><code>directory-is-modifiable?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-directory_002dis_002dmodifiable_003f-1"><code>directory-is-modifiable?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#check_002dfor_002dupdates">check-for-updates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-display_002dlists_002dfor_002dmaps_002dstate"><code>display-lists-for-maps-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#display_002dlists_002dfor_002dmaps_002dstate">display-lists-for-maps-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-display_002dmaps_002dscm"><code>display-maps-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#display_002dmaps_002dscm">display-maps-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-display_002donly_002dactive"><code>display-only-active</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#display_002donly_002dactive">display-only-active</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-disrupt"><code>disrupt</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#exercise_002dscm_002dmol">exercise-scm-mol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002d180_002ddegree_002dside_002dchain_002dflip"><code>do-180-degree-side-chain-flip</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002d180_002ddegree_002dside_002dchain_002dflip">do-180-degree-side-chain-flip</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002danti_002daliasing_002dstate"><code>do-anti-aliasing-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002danti_002daliasing_002dstate">do-anti-aliasing-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dclipped_002dsurface_002dpy"><code>do-clipped-surface-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dclipped_002dsurface_002dpy">do-clipped-surface-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dclipped_002dsurface_002dscm"><code>do-clipped-surface-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dclipped_002dsurface_002dscm">do-clipped-surface-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dfind_002dligands_002ddialog"><code>do-find-ligands-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dfind_002dligands_002ddialog">do-find-ligands-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dGL_002dlighting_002dstate"><code>do-GL-lighting-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dGL_002dlighting_002dstate">do-GL-lighting-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate"><code>do-probe-dots-on-rotamers-and-chis-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dprobe_002ddots_002don_002drotamers_002dand_002dchis_002dstate">do-probe-dots-on-rotamers-and-chis-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dprobe_002ddots_002dpost_002drefine_002dstate"><code>do-probe-dots-post-refine-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dprobe_002ddots_002dpost_002drefine_002dstate">do-probe-dots-post-refine-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dramachandran_002dplot"><code>do-ramachandran-plot</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dramachandran_002dplot">do-ramachandran-plot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dsequence_002dview"><code>do-sequence-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dsequence_002dview">do-sequence-view</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dsmiles_002dgui"><code>do-smiles-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dsmiles_002dgui">do-smiles-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-do_002dsurface"><code>do-surface</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#do_002dsurface">do-surface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-does_002dresidue_002dexist_002dp"><code>does-residue-exist-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#does_002dresidue_002dexist_002dp">does-residue-exist-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dots"><code>dots</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#dots">dots</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-download_002dbinary_002ddialog"><code>download-binary-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#check_002dfor_002dupdates">check-for-updates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-dragged_002drefinement_002dsteps_002dper_002dframe"><code>dragged-refinement-steps-per-frame</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#dragged_002drefinement_002dsteps_002dper_002dframe">dragged-refinement-steps-per-frame</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-draw_002dhydrogens_002dstate"><code>draw-hydrogens-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#draw_002dhydrogens_002dstate">draw-hydrogens-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-draw_002dncs_002dghosts_002dstate"><code>draw-ncs-ghosts-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#draw_002dncs_002dghosts_002dstate">draw-ncs-ghosts-state</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-E">E</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-edit_002dchi_002dangles"><code>edit-chi-angles</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#edit_002dchi_002dangles">edit-chi-angles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-enhanced_002dligand_002dcoot_003f"><code>enhanced-ligand-coot?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#redefine_002dfunctions">redefine-functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-esoteric_002ddepth_002dcue_002dstate"><code>esoteric-depth-cue-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#esoteric_002ddepth_002dcue_002dstate">esoteric-depth-cue-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-eval_002dsocket_002dstring"><code>eval-socket-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remote_002dcontrol">remote-control</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-every_002dnth"><code>every-nth</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-exchange_002dchain_002dids_002dfor_002dseg_002dids"><code>exchange-chain-ids-for-seg-ids</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#exchange_002dchain_002dids_002dfor_002dseg_002dids">exchange-chain-ids-for-seg-ids</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-execute_002dfind_002dblobs"><code>execute-find-blobs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#execute_002dfind_002dblobs">execute-find-blobs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-execute_002dfind_002dwaters_002dreal"><code>execute-find-waters-real</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#execute_002dfind_002dwaters_002dreal">execute-find-waters-real</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-execute_002drefmac_002dreal"><code>execute-refmac-real</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#execute_002drefmac_002dreal">execute-refmac-real</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-export_002dmap"><code>export-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#export_002dmap">export-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-export_002dmap_002dfragment"><code>export-map-fragment</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#export_002dmap_002dfragment">export-map-fragment</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-export_002dmap_002dfragment_002dto_002dplain_002dfile"><code>export-map-fragment-to-plain-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#export_002dmap_002dfragment_002dto_002dplain_002dfile">export-map-fragment-to-plain-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-export_002dmap_002dfragment_002dwith_002dorigin_002dshift"><code>export-map-fragment-with-origin-shift</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#export_002dmap_002dfragment_002dwith_002dorigin_002dshift">export-map-fragment-with-origin-shift</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-extra_002dparams_002dinclude_002dweight_003f"><code>extra-params-include-weight?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refmac">refmac</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-F">F</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-fascinating_002dclusters_002dgui"><code>fascinating-clusters-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fascinating_002dthings">fascinating-things</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fffear_002dangular_002dresolution"><code>fffear-angular-resolution</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fffear_002dangular_002dresolution">fffear-angular-resolution</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fffear_002dsearch"><code>fffear-search</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fffear_002dsearch">fffear-search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-file_002dname_002ddirectory"><code>file-name-directory</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-file_002dname_002dextension"><code>file-name-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-file_002dname_002dfile"><code>file-name-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-file_002dname_002dfile-1"><code>file-name-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-file_002dname_002dsans_002dextension"><code>file-name-sans-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-file_002dselector_002dentry"><code>file-selector-entry</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-file_002dtype_002dcoords"><code>file-type-coords</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#file_002dtype_002dcoords">file-type-coords</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fill_002doption_002dmenu_002dwith_002dcoordinates_002dmol_002doptions"><code>fill-option-menu-with-coordinates-mol-options</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fill_002doption_002dmenu_002dwith_002dmol_002doptions"><code>fill-option-menu-with-mol-options</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fill_002doption_002dmenu_002dwith_002dnumber_002doptions"><code>fill-option-menu-with-number-options</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fill_002dpartial_002dresidues"><code>fill-partial-residues</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fill_002dpartial_002dresidues">fill-partial-residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-filter"><code>filter</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-filter-1"><code>filter</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#filter">filter</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-filter_002dfileselection_002dfilenames_002dstate"><code>filter-fileselection-filenames-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#filter_002dfileselection_002dfilenames_002dstate">filter-fileselection-filenames-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-filter_002dout_002dwaters"><code>filter-out-waters</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ligand_002dcheck">ligand-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-find_002daligned_002dresidue_002dtype"><code>find-aligned-residue-type</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-find_002dhelices"><code>find-helices</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#find_002dhelices">find-helices</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-find_002dnucleic_002dacids_002dlocal"><code>find-nucleic-acids-local</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#find_002dnucleic_002dacids_002dlocal">find-nucleic-acids-local</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-find_002dsecondary_002dstructure"><code>find-secondary-structure</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#find_002dsecondary_002dstructure">find-secondary-structure</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-find_002dsecondary_002dstructure_002dlocal"><code>find-secondary-structure-local</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#find_002dsecondary_002dstructure_002dlocal">find-secondary-structure-local</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-find_002dstrands"><code>find-strands</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#find_002dstrands">find-strands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-first_002dcoords_002dimol"><code>first-coords-imol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#first_002dcoords_002dimol">first-coords-imol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-first_002dmolecule_002dwith_002dsymmetry_002ddisplayed"><code>first-molecule-with-symmetry-displayed</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#first_002dmolecule_002dwith_002dsymmetry_002ddisplayed">first-molecule-with-symmetry-displayed</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-first_002dn"><code>first-n</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-first_002dnon_002dtrivial_002dname"><code>first-non-trivial-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#hello">hello</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-first_002dsmall_002dcoords_002dimol"><code>first-small-coords-imol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#first_002dsmall_002dcoords_002dimol">first-small-coords-imol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-first_002dunsaved_002dcoords_002dimol"><code>first-unsaved-coords-imol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#first_002dunsaved_002dcoords_002dimol">first-unsaved-coords-imol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dchain"><code>fit-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle"><code>fit-chain-to-map-by-random-jiggle</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle">fit-chain-to-map-by-random-jiggle</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur"><code>fit-chain-to-map-by-random-jiggle-and-blur</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fit_002dchain_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur">fit-chain-to-map-by-random-jiggle-and-blur</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle"><code>fit-molecule-to-map-by-random-jiggle</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle">fit-molecule-to-map-by-random-jiggle</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur"><code>fit-molecule-to-map-by-random-jiggle-and-blur</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fit_002dmolecule_002dto_002dmap_002dby_002drandom_002djiggle_002dand_002dblur">fit-molecule-to-map-by-random-jiggle-and-blur</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dprotein"><code>fit-protein</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dprotein_002dmake_002dspecs"><code>fit-protein-make-specs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex"><code>fit-residue-range-to-map-by-simplex</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fit_002dresidue_002drange_002dto_002dmap_002dby_002dsimplex">fit-residue-range-to-map-by-simplex</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dto_002dmap_002dby_002drandom_002djiggle"><code>fit-to-map-by-random-jiggle</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fit_002dto_002dmap_002dby_002drandom_002djiggle">fit-to-map-by-random-jiggle</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fit_002dwaters"><code>fit-waters</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fix_002dnomenclature_002derrors"><code>fix-nomenclature-errors</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fix_002dnomenclature_002derrors">fix-nomenclature-errors</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fle_002dview_002dset_002dh_002dbond_002ddist_002dmax"><code>fle-view-set-h-bond-dist-max</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fle_002dview_002dset_002dh_002dbond_002ddist_002dmax">fle-view-set-h-bond-dist-max</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fle_002dview_002dset_002dwater_002ddist_002dmax"><code>fle-view-set-water-dist-max</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fle_002dview_002dset_002dwater_002ddist_002dmax">fle-view-set-water-dist-max</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-fleur_002dcursor"><code>fleur-cursor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fleur_002dcursor">fleur-cursor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-flip_002dactive_002dligand"><code>flip-active-ligand</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-flip_002dhand"><code>flip-hand</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#flip_002dhand">flip-hand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-flip_002dligand"><code>flip-ligand</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#flip_002dligand">flip-ligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-full_002datom_002dspec_002dto_002datom_002dindex"><code>full-atom-spec-to-atom-index</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#full_002datom_002dspec_002dto_002datom_002dindex">full-atom-spec-to-atom-index</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-G">G</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-generate_002dlocal_002dself_002drestraints"><code>generate-local-self-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generate_002dlocal_002dself_002drestraints">generate-local-self-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm"><code>generate-local-self-restraints-by-residues-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generate_002dlocal_002dself_002drestraints_002dby_002dresidues_002dscm">generate-local-self-restraints-by-residues-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generate_002dmolecule_002dfrom_002dmmcif_002dby_002ddict_002dgen"><code>generate-molecule-from-mmcif-by-dict-gen</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#dictionary_002dgenerators">dictionary-generators</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generate_002dself_002drestraints"><code>generate-self-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generate_002dself_002drestraints">generate-self-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dbuttons_002ddialog"><code>generic-buttons-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dchooser_002dand_002dentry"><code>generic-chooser-and-entry</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dchooser_002dand_002dentry_002dand_002dcheckbutton"><code>generic-chooser-and-entry-and-checkbutton</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dchooser_002dand_002dfile_002dselector"><code>generic-chooser-and-file-selector</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002ddouble_002dentry"><code>generic-double-entry</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dinteresting_002dthings"><code>generic-interesting-things</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dmolecule_002dchooser"><code>generic-molecule-chooser</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dmultiple_002dentries_002dwith_002dcheck_002dbutton"><code>generic-multiple-entries-with-check-button</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dnumber_002dchooser"><code>generic-number-chooser</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dobject_002dis_002ddisplayed_003f"><code>generic-object-is-displayed?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dobject_002dwith_002dname"><code>generic-object-with-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dobjects_002dgui"><code>generic-objects-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-generic_002dsingle_002dentry"><code>generic-single-entry</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dactive_002dmap_002ddrag_002dflag"><code>get-active-map-drag-flag</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dactive_002dmap_002ddrag_002dflag">get-active-map-drag-flag</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002daniso_002dprobability"><code>get-aniso-probability</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002daniso_002dprobability">get-aniso-probability</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002datom"><code>get-atom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002datom_002dcolour_002dfrom_002dmol_002dno"><code>get-atom-colour-from-mol-no</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002datom_002dcolour_002dfrom_002dmol_002dno">get-atom-colour-from-mol-no</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002datom_002dfrom_002dresidue"><code>get-atom-from-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dauto_002drecontour_002dmap"><code>get-auto-recontour-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dauto_002drecontour_002dmap">get-auto-recontour-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dbond_002dcolour_002drotation_002dfor_002dmolecule"><code>get-bond-colour-rotation-for-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dbond_002dcolour_002drotation_002dfor_002dmolecule">get-bond-colour-rotation-for-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dbonds_002drepresentation"><code>get-bonds-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dbonds_002drepresentation">get-bonds-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dccp4srs_002dmonomer_002dand_002ddictionary"><code>get-ccp4srs-monomer-and-dictionary</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dccp4srs_002dmonomer_002dand_002ddictionary">get-ccp4srs-monomer-and-dictionary</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dccp4srs_002dmonomer_002dand_002doverlay"><code>get-ccp4srs-monomer-and-overlay</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#g_t3d_002dgenerator_002dimport">3d-generator-import</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate"><code>get-continue-updating-refinement-atoms-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dcontinue_002dupdating_002drefinement_002datoms_002dstate">get-continue-updating-refinement-atoms-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dcontour_002dlevel_002dabsolute"><code>get-contour-level-absolute</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dcontour_002dlevel_002dabsolute">get-contour-level-absolute</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dcontour_002dlevel_002din_002dsigma"><code>get-contour-level-in-sigma</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dcontour_002dlevel_002din_002dsigma">get-contour-level-in-sigma</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002ddefault_002dbond_002dthickness"><code>get-default-bond-thickness</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002ddefault_002dbond_002dthickness">get-default-bond-thickness</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002ddictionary_002dradii"><code>get-dictionary-radii</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002ddictionary_002dradii">get-dictionary-radii</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002ddiff_002dmap_002diso_002dlevel_002dincrement"><code>get-diff-map-iso-level-increment</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002ddiff_002dmap_002diso_002dlevel_002dincrement">get-diff-map-iso-level-increment</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002ddirectory"><code>get-directory</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002ddrug_002dvia_002dwikipedia"><code>get-drug-via-wikipedia</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002debi_002dpdb"><code>get-ebi-pdb</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002debi">get-ebi</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002deds_002dpdb_002dand_002dmtz"><code>get-eds-pdb-and-mtz</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002debi">get-ebi</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002delectrostatic_002dsurface_002dopacity"><code>get-electrostatic-surface-opacity</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002delectrostatic_002dsurface_002dopacity">get-electrostatic-surface-opacity</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002denvironment_002ddistances_002drepresentation_002dpy"><code>get-environment-distances-representation-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002denvironment_002ddistances_002drepresentation_002dpy">get-environment-distances-representation-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dfile_002dmtime"><code>get-file-mtime</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#jligand">jligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dfirst_002dncs_002dmaster_002dchain"><code>get-first-ncs-master-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dflip_002dresidue"><code>get-flip-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#what_002dcheck">what-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dfont_002dsize"><code>get-font-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dfont_002dsize">get-font-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dfps_002dflag"><code>get-fps-flag</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dfps_002dflag">get-fps-flag</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dgdb_002dstrings"><code>get-gdb-strings</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dintermediate_002datoms_002dbonds_002drepresentation"><code>get-intermediate-atoms-bonds-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dintermediate_002datoms_002dbonds_002drepresentation">get-intermediate-atoms-bonds-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dlimit_002daniso"><code>get-limit-aniso</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dlimit_002daniso">get-limit-aniso</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dmap_002dradius"><code>get-map-radius</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dmap_002dradius">get-map-radius</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dmap_002dsampling_002drate"><code>get-map-sampling-rate</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dmap_002dsampling_002drate">get-map-sampling-rate</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dmap_002dweight"><code>get-map-weight</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dmap_002dweight">get-map-weight</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dmdl_002dlatest_002dtime"><code>get-mdl-latest-time</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#g_t3d_002dgenerator_002dimport">3d-generator-import</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dmetrics_002dfor_002dligand"><code>get-metrics-for-ligand</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ligand_002dcheck">ligand-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation"><code>get-molecule-bonds-colour-map-rotation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dmolecule_002dbonds_002dcolour_002dmap_002drotation">get-molecule-bonds-colour-map-rotation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002doption_002dmenu_002dactive_002ditem"><code>get-option-menu-active-item</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002doption_002dmenu_002dactive_002dmolecule"><code>get-option-menu-active-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dpointer_002dposition_002dfrac_002dpy"><code>get-pointer-position-frac-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dpointer_002dposition_002dfrac_002dpy">get-pointer-position-frac-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002drefinement_002dgeman_002dmcclure_002dalpha"><code>get-refinement-geman-mcclure-alpha</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002drefinement_002dgeman_002dmcclure_002dalpha">get-refinement-geman-mcclure-alpha</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002drefmac_002dextra_002dparams"><code>get-refmac-extra-params</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refmac">refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate"><code>get-reset-b-factor-moved-atoms-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dreset_002db_002dfactor_002dmoved_002datoms_002dstate">get-reset-b-factor-moved-atoms-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002drevision_002dfrom_002dstring"><code>get-revision-from-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dsbase_002dmonomer"><code>get-sbase-monomer</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dsbase_002dmonomer">get-sbase-monomer</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dsequence_002das_002dfasta_002dfor_002dchain"><code>get-sequence-as-fasta-for-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dsequence_002das_002dfasta_002dfor_002dchain">get-sequence-as-fasta-for-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dshow_002daniso"><code>get-show-aniso</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dshow_002daniso">get-show-aniso</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dshow_002dlimit_002daniso"><code>get-show-limit-aniso</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dshow_002dlimit_002daniso">get-show-limit-aniso</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dshow_002dsymmetry"><code>get-show-symmetry</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dshow_002dsymmetry">get-show-symmetry</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dshow_002dunit_002dcell"><code>get-show-unit-cell</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dshow_002dunit_002dcell">get-show-unit-cell</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dsmooth_002dscroll"><code>get-smooth-scroll</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dsmooth_002dscroll">get-smooth-scroll</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dstable_002drelease_002dfrom_002dcoot_002dversion"><code>get-stable-release-from-coot-version</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#check_002dfor_002dupdates">check-for-updates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dstable_002drelease_002dfrom_002dserver_002dstring"><code>get-stable-release-from-server-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#check_002dfor_002dupdates">check-for-updates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dsymmetry"><code>get-symmetry</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dsymmetry">get-symmetry</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dsymmetry_002das_002dcalphas_002dstate"><code>get-symmetry-as-calphas-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dsymmetry_002das_002dcalphas_002dstate">get-symmetry-as-calphas-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002durl_002dstr"><code>get-url-str</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002debi">get-ebi</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-get_002dview_002dquaternion_002dinternal"><code>get-view-quaternion-internal</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002dview_002dquaternion_002dinternal">get-view-quaternion-internal</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-git_002drevision_002dcount"><code>git-revision-count</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#git_002drevision_002dcount">git-revision-count</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-gln_002dasn_002db_002dfactor_002doutliers"><code>gln-asn-b-factor-outliers</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#gln_002dasn_002db_002dfactor_002doutliers">gln-asn-b-factor-outliers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-glob"><code>glob</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-glob-1"><code>glob</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-glob-2"><code>glob</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002datom_002dalt_002dconf"><code>go-to-atom-alt-conf</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#go_002dto_002datom_002dalt_002dconf">go-to-atom-alt-conf</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002datom_002datom_002dname"><code>go-to-atom-atom-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#go_002dto_002datom_002datom_002dname">go-to-atom-atom-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002datom_002dchain_002did"><code>go-to-atom-chain-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#go_002dto_002datom_002dchain_002did">go-to-atom-chain-id</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002datom_002dins_002dcode"><code>go-to-atom-ins-code</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#go_002dto_002datom_002dins_002dcode">go-to-atom-ins-code</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002datom_002dmolecule_002dnumber"><code>go-to-atom-molecule-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#go_002dto_002datom_002dmolecule_002dnumber">go-to-atom-molecule-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002datom_002dresidue_002dnumber"><code>go-to-atom-residue-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#go_002dto_002datom_002dresidue_002dnumber">go-to-atom-residue-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002dligand"><code>go-to-ligand</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#go_002dto_002dligand">go-to-ligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002dmap_002dmolecule_002dcentre"><code>go-to-map-molecule-centre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#go_002dto_002dmap_002dmolecule_002dcentre">go-to-map-molecule-centre</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-go_002dto_002dresidue_002dby_002dspec"><code>go-to-residue-by-spec</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#what_002dcheck">what-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-goosh_002dcommand"><code>goosh-command</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-goosh_002dcommand-1"><code>goosh-command</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-goosh_002dcommand_002dwith_002dfile_002dinput"><code>goosh-command-with-file-input</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dcomma_002dkey_002dpressed_002dhook"><code>graphics-comma-key-pressed-hook</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002ddot_002dkey_002dpressed_002dhook"><code>graphics-dot-key-pressed-hook</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002ddraw"><code>graphics-draw</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002ddraw">graphics-draw</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dgeneral_002dkey_002dpress_002dhook"><code>graphics-general-key-press-hook</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dmolecule_002dbond_002dtype"><code>graphics-molecule-bond-type</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dmolecule_002dbond_002dtype">graphics-molecule-bond-type</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dn_002dmolecules"><code>graphics-n-molecules</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dn_002dmolecules">graphics-n-molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002db_002dfactor_002dcas_002drepresentation"><code>graphics-to-b-factor-cas-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002db_002dfactor_002dcas_002drepresentation">graphics-to-b-factor-cas-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002db_002dfactor_002drepresentation"><code>graphics-to-b-factor-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002db_002dfactor_002drepresentation">graphics-to-b-factor-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation"><code>graphics-to-bonds-no-waters-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002dbonds_002dno_002dwaters_002drepresentation">graphics-to-bonds-no-waters-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002dbonds_002drepresentation"><code>graphics-to-bonds-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002dbonds_002drepresentation">graphics-to-bonds-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation"><code>graphics-to-ca-plus-ligands-and-sidechains-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002dca_002dplus_002dligands_002dand_002dsidechains_002drepresentation">graphics-to-ca-plus-ligands-and-sidechains-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002dca_002dplus_002dligands_002drepresentation"><code>graphics-to-ca-plus-ligands-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002dca_002dplus_002dligands_002drepresentation">graphics-to-ca-plus-ligands-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation"><code>graphics-to-ca-plus-ligands-sec-struct-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002dca_002dplus_002dligands_002dsec_002dstruct_002drepresentation">graphics-to-ca-plus-ligands-sec-struct-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002dca_002drepresentation"><code>graphics-to-ca-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002dca_002drepresentation">graphics-to-ca-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002doccupancy_002drepresentation"><code>graphics-to-occupancy-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002doccupancy_002drepresentation">graphics-to-occupancy-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002drainbow_002drepresentation"><code>graphics-to-rainbow-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002drainbow_002drepresentation">graphics-to-rainbow-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation"><code>graphics-to-sec-struct-bonds-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002dsec_002dstruct_002dbonds_002drepresentation">graphics-to-sec-struct-bonds-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation"><code>graphics-to-user-defined-atom-colours-all-atoms-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002dall_002datoms_002drepresentation">graphics-to-user-defined-atom-colours-all-atoms-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation"><code>graphics-to-user-defined-atom-colours-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dto_002duser_002ddefined_002datom_002dcolours_002drepresentation">graphics-to-user-defined-atom-colours-representation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences"><code>graphics-window-size-and-position-to-preferences</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#graphics_002dwindow_002dsize_002dand_002dposition_002dto_002dpreferences">graphics-window-size-and-position-to-preferences</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-guess_002drefinement_002dmap"><code>guess-refinement-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-H">H</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-hack_002dlink"><code>hack-link</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#acedrg_002dlink">acedrg-link</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dcns_002ddata_002dfile"><code>handle-cns-data-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002dcns_002ddata_002dfile">handle-cns-data-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dcns_002ddata_002dfile_002dwith_002dcell"><code>handle-cns-data-file-with-cell</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002dcns_002ddata_002dfile_002dwith_002dcell">handle-cns-data-file-with-cell</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002ddrag_002dand_002ddrop_002dstring"><code>handle-drag-and-drop-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002ddrag_002dand_002ddrop_002dstring">handle-drag-and-drop-string</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002donline_002dcoot_002dsearch_002drequest"><code>handle-online-coot-search-request</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002donline_002dcoot_002dsearch_002drequest">handle-online-coot-search-request</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dread_002dccp4_002dmap"><code>handle-read-ccp4-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002dread_002dccp4_002dmap">handle-read-ccp4-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dread_002ddraw_002dmolecule"><code>handle-read-draw-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002dread_002ddraw_002dmolecule">handle-read-draw-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere"><code>handle-read-draw-molecule-and-move-molecule-here</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002dread_002ddraw_002dmolecule_002dand_002dmove_002dmolecule_002dhere">handle-read-draw-molecule-and-move-molecule-here</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dread_002ddraw_002dmolecule_002dwith_002drecenter"><code>handle-read-draw-molecule-with-recenter</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre"><code>handle-read-draw-molecule-with-recentre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002dread_002ddraw_002dmolecule_002dwith_002drecentre">handle-read-draw-molecule-with-recentre</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dread_002ddraw_002dprobe_002ddots"><code>handle-read-draw-probe-dots</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002dread_002ddraw_002dprobe_002ddots">handle-read-draw-probe-dots</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted"><code>handle-read-draw-probe-dots-unformatted</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#handle_002dread_002ddraw_002dprobe_002ddots_002dunformatted">handle-read-draw-probe-dots-unformatted</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-handle_002dsmiles_002dgo"><code>handle-smiles-go</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hardware_002dstereo_002dangle_002dfactor_002dstate"><code>hardware-stereo-angle-factor-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#hardware_002dstereo_002dangle_002dfactor_002dstate">hardware-stereo-angle-factor-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hardware_002dstereo_002dmode"><code>hardware-stereo-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#hardware_002dstereo_002dmode">hardware-stereo-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-has_002dunit_002dcell_002dstate"><code>has-unit-cell-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#has_002dunit_002dcell_002dstate">has-unit-cell-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-have_002dunsaved_002dchanges_002dp"><code>have-unsaved-changes-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#have_002dunsaved_002dchanges_002dp">have-unsaved-changes-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-het_002dgroup_002dn_002datoms"><code>het-group-n-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#het_002dgroup_002dn_002datoms">het-group-n-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-het_002dgroup_002dresidues_002dpy"><code>het-group-residues-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#het_002dgroup_002dresidues_002dpy">het-group-residues-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-het_002dgroup_002dresidues_002dscm"><code>het-group-residues-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#het_002dgroup_002dresidues_002dscm">het-group-residues-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hetify_002dresidue"><code>hetify-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#hetify_002dresidue">hetify-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hide_002dmain_002dtoolbar"><code>hide-main-toolbar</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#hide_002dmain_002dtoolbar">hide-main-toolbar</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hide_002dmodelling_002dtoolbar"><code>hide-modelling-toolbar</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#hide_002dmodelling_002dtoolbar">hide-modelling-toolbar</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-highly_002dcoordinated_002dwaters_002dpy"><code>highly-coordinated-waters-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#highly_002dcoordinated_002dwaters_002dpy">highly-coordinated-waters-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-highly_002dcoordinated_002dwaters_002dscm"><code>highly-coordinated-waters-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#highly_002dcoordinated_002dwaters_002dscm">highly-coordinated-waters-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hilight_002dbinding_002dsite"><code>hilight-binding-site</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hole"><code>hole</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#hole">hole</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-hydrogenate_002dregion"><code>hydrogenate-region</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#hydrogenate_002dregion">hydrogenate-region</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-I">I</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-ideal_002dnucleic_002dacid"><code>ideal-nucleic-acid</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ideal_002dnucleic_002dacid">ideal-nucleic-acid</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-identity_002dmatrix"><code>identity-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-idle_002dfunction_002drotate_002dangle"><code>idle-function-rotate-angle</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#idle_002dfunction_002drotate_002dangle">idle-function-rotate-angle</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-imol_002drefinement_002dmap"><code>imol-refinement-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#imol_002drefinement_002dmap">imol-refinement-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-import_002dfrom_002d3d_002dgenerator_002dfrom_002dmdl"><code>import-from-3d-generator-from-mdl</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#g_t3d_002dgenerator_002dimport">3d-generator-import</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-import_002dpython_002dmodule"><code>import-python-module</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#import_002dpython_002dmodule">import-python-module</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-increment_002dcoot_002dtip_002dnumber"><code>increment-coot-tip-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#tips_002dgui">tips-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-info_002ddialog"><code>info-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#info_002ddialog">info-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-info_002ddialog_002dand_002dtext"><code>info-dialog-and-text</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#info_002ddialog_002dand_002dtext">info-dialog-and-text</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-info_002ddialog_002dwith_002dmarkup"><code>info-dialog-with-markup</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#info_002ddialog_002dwith_002dmarkup">info-dialog-with-markup</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-insert_002daction_002dview_002dafter_002dview"><code>insert-action-view-after-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#insert_002daction_002dview_002dafter_002dview">insert-action-view-after-view</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-insertion_002dcode_002dfrom_002dserial_002dnumber"><code>insertion-code-from-serial-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#insertion_002dcode_002dfrom_002dserial_002dnumber">insertion-code-from-serial-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-interactive_002dprobe"><code>interactive-probe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-interactive_002dprobe_002ddots_002dmolprobity_002dradius"><code>interactive-probe-dots-molprobity-radius</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#interactive_002dprobe_002ddots_002dmolprobity_002dradius">interactive-probe-dots-molprobity-radius</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-interesting_002dthings_002dgui"><code>interesting-things-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-interesting_002dthings_002dwith_002dfix_002dmaybe"><code>interesting-things-with-fix-maybe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-interruptible_002dfit_002dprotein"><code>interruptible-fit-protein</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-invert_002dchiral_002dcenter"><code>invert-chiral-center</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-is_002dclosed_002dgeneric_002dobject_003f"><code>is-closed-generic-object?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-is_002ddifference_002dmap_003f"><code>is-difference-map?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-is_002ddirectory_003f"><code>is-directory?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-is_002dmtz_002dfile_002dp"><code>is-mtz-file-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#is_002dmtz_002dfile_002dp">is-mtz-file-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-is_002dnucleotide_002dchain_002dp"><code>is-nucleotide-chain-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#is_002dnucleotide_002dchain_002dp">is-nucleotide-chain-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-is_002dprotein_002dchain_002dp"><code>is-protein-chain-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#is_002dprotein_002dchain_002dp">is-protein-chain-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-is_002dsolvent_002dchain_002dp"><code>is-solvent-chain-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#is_002dsolvent_002dchain_002dp">is-solvent-chain-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-is_002dsolvent_002dchain_003f"><code>is-solvent-chain?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-J">J</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-jiggled_002dmol"><code>jiggled-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#exercise_002dscm_002dmol">exercise-scm-mol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-join_002ddir_002dfile"><code>join-dir-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-just_002done_002dor_002dnext_002dmap"><code>just-one-or-next-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-K">K</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-keep_002dmap_002dcolour_002dafter_002drefmac_002dstate"><code>keep-map-colour-after-refmac-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#keep_002dmap_002dcolour_002dafter_002drefmac_002dstate">keep-map-colour-after-refmac-state</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-L">L</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-label_002dall_002dactive_002dresidue_002datoms"><code>label-all-active-residue-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-label_002dall_002datoms_002din_002dresidue"><code>label-all-atoms-in-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-label_002dall_002dCAs"><code>label-all-CAs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-label_002datoms_002din_002dresidue"><code>label-atoms-in-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#label_002datoms_002din_002dresidue">label-atoms-in-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy"><code>label-closest-atoms-in-neighbour-residues-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dpy">label-closest-atoms-in-neighbour-residues-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm"><code>label-closest-atoms-in-neighbour-residues-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#label_002dclosest_002datoms_002din_002dneighbour_002dresidues_002dscm">label-closest-atoms-in-neighbour-residues-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-label_002dneighbours"><code>label-neighbours</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#label_002dneighbours">label-neighbours</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-laplacian"><code>laplacian</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#laplacian">laplacian</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-launch_002djligand_002dfunction"><code>launch-jligand-function</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#jligand_002dgui">jligand-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-lengthen_002dbaton"><code>lengthen-baton</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#lengthen_002dbaton">lengthen-baton</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-libcheck_002dexe"><code>libcheck-exe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#libcheck">libcheck</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ligand_002dexpert"><code>ligand-expert</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ligand_002dexpert">ligand-expert</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-list_002dof_002dstrings_003f"><code>list-of-strings?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-load_002dall_002dscheme"><code>load-all-scheme</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot">coot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-load_002dannotations"><code>load-annotations</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-load_002ddefault_002dsequence"><code>load-default-sequence</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-lsq_002dimprove"><code>lsq-improve</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#lsq_002dimprove">lsq-improve</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-lsq_002dmatch"><code>lsq-match</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dlsq">coot-lsq</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-lsq_002dmatch_002dtype_002dsymbol_002d_003enumber"><code>lsq-match-type-symbol-&gt;number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dlsq">coot-lsq</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-lsqman_002dcount"><code>lsqman-count</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#brute_002dlsqman">brute-lsqman</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-M">M</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-maf"><code>maf</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dand_002ddraw_002dmap"><code>make-and-draw-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dand_002ddraw_002dmap">make-and-draw-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams"><code>make-and-draw-map-with-refmac-params</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dand_002ddraw_002dmap_002dwith_002drefmac_002dparams">make-and-draw-map-with-refmac-params</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams"><code>make-and-draw-map-with-reso-with-refmac-params</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dand_002ddraw_002dmap_002dwith_002dreso_002dwith_002drefmac_002dparams">make-and-draw-map-with-reso-with-refmac-params</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dbackup"><code>make-backup</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dbackup">make-backup</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dball_002dand_002dstick"><code>make-ball-and-stick</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dball_002dand_002dstick">make-ball-and-stick</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002ddifference_002dmap_002dgui"><code>make-difference-map-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002ddirectory_002dmaybe"><code>make-directory-maybe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002ddirectory_002dmaybe">make-directory-maybe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002ddynamically_002dtransformed_002dncs_002dmaps"><code>make-dynamically-transformed-ncs-maps</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002ddynamically_002dtransformed_002dncs_002dmaps">make-dynamically-transformed-ncs-maps</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dgdb_002dscript"><code>make-gdb-script</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dgui"><code>make-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dlatest_002dversion_002durl"><code>make-latest-version-url</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dlink"><code>make-link</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dlink">make-link</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dmasked_002dmaps_002dsplit_002dby_002dchain"><code>make-masked-maps-split-by-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dmasked_002dmaps_002dsplit_002dby_002dchain">make-masked-maps-split-by-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dsocket_002dlistener_002dmaybe"><code>make-socket-listener-maybe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dsocket_002dlistener_002dmaybe">make-socket-listener-maybe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dtight_002dplanar_002dpeptide_002drestraints"><code>make-tight-planar-peptide-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dtight_002dplanar_002dpeptide_002drestraints">make-tight-planar-peptide-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dupdating_002dmap"><code>make-updating-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dupdating_002dmap">make-updating-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-make_002dupdating_002dmodel_002dmolecule"><code>make-updating-model-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#make_002dupdating_002dmodel_002dmolecule">make-updating-model-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-manage_002dcolumn_002dselector"><code>manage-column-selector</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#manage_002dcolumn_002dselector">manage-column-selector</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-manual_002dncs_002dghosts"><code>manual-ncs-ghosts</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs">ncs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-manual_002drefine_002dresidues"><code>manual-refine-residues</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dcolour_002dcomponents"><code>map-colour-components</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dcolour_002dcomponents">map-colour-components</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dcolour_002dcomponents_002dpy"><code>map-colour-components-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dcolour_002dcomponents_002dpy">map-colour-components-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dcontours"><code>map-contours</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dcontours">map-contours</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dfrom_002dmtz_002dby_002dcalc_002dphases"><code>map-from-mtz-by-calc-phases</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dfrom_002dmtz_002dby_002dcalc_002dphases">map-from-mtz-by-calc-phases</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases"><code>map-from-mtz-by-refmac-calc-phases</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dfrom_002dmtz_002dby_002drefmac_002dcalc_002dphases">map-from-mtz-by-refmac-calc-phases</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dis_002ddifference_002dmap"><code>map-is-difference-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dis_002ddifference_002dmap">map-is-difference-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dis_002ddisplayed"><code>map-is-displayed</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dis_002ddisplayed">map-is-displayed</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dline_002dwidth_002dstate"><code>map-line-width-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dline_002dwidth_002dstate">map-line-width-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dmask_002datom_002dradius"><code>map-mask-atom-radius</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dmask_002datom_002dradius">map-mask-atom-radius</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dmean_002dpy"><code>map-mean-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dmean_002dpy">map-mean-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dmean_002dscm"><code>map-mean-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dmean_002dscm">map-mean-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dmolecule_002dchooser_002dgui"><code>map-molecule-chooser-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dmolecule_002dlist"><code>map-molecule-list</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dparameters_002dpy"><code>map-parameters-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dparameters_002dpy">map-parameters-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dparameters_002dscm"><code>map-parameters-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dparameters_002dscm">map-parameters-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dpeaks_002dpy"><code>map-peaks-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dpeaks_002dpy">map-peaks-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dpeaks_002dscm"><code>map-peaks-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dpeaks_002dscm">map-peaks-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dstatistics_002dscm"><code>map-statistics-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dstatistics_002dscm">map-statistics-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dto_002dmodel_002dcorrelation"><code>map-to-model-correlation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dto_002dmodel_002dcorrelation">map-to-model-correlation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dto_002dmodel_002dcorrelation_002dper_002dresidue"><code>map-to-model-correlation-per-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue">map-to-model-correlation-per-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm"><code>map-to-model-correlation-per-residue-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dto_002dmodel_002dcorrelation_002dper_002dresidue_002dscm">map-to-model-correlation-per-residue-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dto_002dmodel_002dcorrelation_002dscm"><code>map-to-model-correlation-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dto_002dmodel_002dcorrelation_002dscm">map-to-model-correlation-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dto_002dmodel_002dcorrelation_002dstats"><code>map-to-model-correlation-stats</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dto_002dmodel_002dcorrelation_002dstats">map-to-model-correlation-stats</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue"><code>map-to-model-correlation-stats-per-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue">map-to-model-correlation-stats-per-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm"><code>map-to-model-correlation-stats-per-residue-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#map_002dto_002dmodel_002dcorrelation_002dstats_002dper_002dresidue_002dscm">map-to-model-correlation-stats-per-residue-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mask_002dmap_002dby_002datom_002dselection"><code>mask-map-by-atom-selection</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mask_002dmap_002dby_002datom_002dselection">mask-map-by-atom-selection</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mask_002dmap_002dby_002dmolecule"><code>mask-map-by-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mask_002dmap_002dby_002dmolecule">mask-map-by-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-match_002dligand_002datom_002dnames"><code>match-ligand-atom-names</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#match_002dligand_002datom_002dnames">match-ligand-atom-names</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-match_002dligand_002datom_002dnames_002dto_002dcomp_002did"><code>match-ligand-atom-names-to-comp-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#match_002dligand_002datom_002dnames_002dto_002dcomp_002did">match-ligand-atom-names-to-comp-id</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy"><code>matching-compound-names-from-dictionary-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dpy">matching-compound-names-from-dictionary-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm"><code>matching-compound-names-from-dictionary-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#matching_002dcompound_002dnames_002dfrom_002ddictionary_002dscm">matching-compound-names-from-dictionary-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy"><code>matching-compound-names-from-sbase-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dpy">matching-compound-names-from-sbase-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm"><code>matching-compound-names-from-sbase-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#matching_002dcompound_002dnames_002dfrom_002dsbase_002dscm">matching-compound-names-from-sbase-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-matrix_002d_003equaternion"><code>matrix-&gt;quaternion</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#quat_002dconvert">quat-convert</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-matrix_002d_003equaternion-1"><code>matrix-&gt;quaternion</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-matrix_002dstate"><code>matrix-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#matrix_002dstate">matrix-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-max_002dwater_002ddistance"><code>max-water-distance</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#max_002dwater_002ddistance">max-water-distance</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-median_002dtemperature_002dfactor"><code>median-temperature-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#median_002dtemperature_002dfactor">median-temperature-factor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-member_003f"><code>member?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-merge_002dfragments"><code>merge-fragments</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#merge_002dfragments">merge-fragments</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-merge_002dsolvent_002dchains"><code>merge-solvent-chains</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-min_002dmax_002dresidues_002dfrom_002datom_002dspecs"><code>min-max-residues-from-atom-specs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-missing_002datoms_002dgui"><code>missing-atoms-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mmcif_002dsfs_002dto_002dmtz"><code>mmcif-sfs-to-mtz</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mmcif_002dsfs_002dto_002dmtz">mmcif-sfs-to-mtz</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate"><code>model-fit-refine-dialog-stays-on-top-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#model_002dfit_002drefine_002ddialog_002dstays_002don_002dtop_002dstate">model-fit-refine-dialog-stays-on-top-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-model_002dmolecule_002dlist"><code>model-molecule-list</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-model_002dresolution"><code>model-resolution</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#model_002dresolution">model-resolution</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mol_002dis_002dactive"><code>mol-is-active</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mol_002dis_002dactive">mol-is-active</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mol_002dis_002ddisplayed"><code>mol-is-displayed</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mol_002dis_002ddisplayed">mol-is-displayed</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dcentre"><code>molecule-centre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dcentre_002dinternal"><code>molecule-centre-internal</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#molecule_002dcentre_002dinternal">molecule-centre-internal</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dcentres_002dgui"><code>molecule-centres-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dchooser_002dgui"><code>molecule-chooser-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dchooser_002dgui_002dgeneric"><code>molecule-chooser-gui-generic</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dhas_002dhydrogens_003f"><code>molecule-has-hydrogens?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dname"><code>molecule-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#molecule_002dname">molecule-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dname_002dstub_002dpy"><code>molecule-name-stub-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#molecule_002dname_002dstub_002dpy">molecule-name-stub-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dname_002dstub_002dscm"><code>molecule-name-stub-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#molecule_002dname_002dstub_002dscm">molecule-name-stub-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dnumber_002dlist"><code>molecule-number-list</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecule_002dto_002dpdb_002dstring_002dpy"><code>molecule-to-pdb-string-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#molecule_002dto_002dpdb_002dstring_002dpy">molecule-to-pdb-string-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-molecules_002dmatching_002dcriteria"><code>molecules-matching-criteria</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mono_002dmode"><code>mono-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mono_002dmode">mono-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-monomer_002dmolecule_002dfrom_002d3_002dlet_002dcode"><code>monomer-molecule-from-3-let-code</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#libcheck">libcheck</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-most_002drecently_002dcreated_002dfile"><code>most-recently-created-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-move_002dhetgroups_002dto_002daround_002dprotein"><code>move-hetgroups-to-around-protein</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#move_002dhetgroups_002dto_002daround_002dprotein">move-hetgroups-to-around-protein</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-move_002dmolecule_002dhere"><code>move-molecule-here</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-move_002dmolecule_002dto_002dscreen_002dcenter"><code>move-molecule-to-screen-center</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-move_002dmolecule_002dto_002dscreen_002dcentre"><code>move-molecule-to-screen-centre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-move_002dwaters_002dto_002daround_002dprotein"><code>move-waters-to-around-protein</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#move_002dwaters_002dto_002daround_002dprotein">move-waters-to-around-protein</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mtz_002dfile_002dhas_002dphases_002dp"><code>mtz-file-has-phases-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mtz_002dfile_002dhas_002dphases_002dp">mtz-file-has-phases-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mtz_002dfile_002dname"><code>mtz-file-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mtz_002dfile_002dname">mtz-file-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mtz_002dfile_002dname_002d_003erefinement_002dcolumn_002dlabels"><code>mtz-file-name-&gt;refinement-column-labels</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#gui_002dligand_002dsliders">gui-ligand-sliders</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mtz_002dfp_002dfor_002dmap"><code>mtz-fp-for-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mtz_002dfp_002dfor_002dmap">mtz-fp-for-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mtz_002dhklin_002dfor_002dmap"><code>mtz-hklin-for-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mtz_002dhklin_002dfor_002dmap">mtz-hklin-for-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mtz_002dphi_002dfor_002dmap"><code>mtz-phi-for-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mtz_002dphi_002dfor_002dmap">mtz-phi-for-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mtz_002duse_002dweight_002dfor_002dmap"><code>mtz-use-weight-for-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mtz_002duse_002dweight_002dfor_002dmap">mtz-use-weight-for-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mtz_002dweight_002dfor_002dmap"><code>mtz-weight-for-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mtz_002dweight_002dfor_002dmap">mtz-weight-for-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dchicken"><code>multi-chicken</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dmutate-1"><code>multi-mutate</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dread_002dpdb"><code>multi-read-pdb</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dresidue_002dtorsion_002dfit_002dpy"><code>multi-residue-torsion-fit-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#multi_002dresidue_002dtorsion_002dfit_002dpy">multi-residue-torsion-fit-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dresidue_002dtorsion_002dfit_002dscm"><code>multi-residue-torsion-fit-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#multi_002dresidue_002dtorsion_002dfit_002dscm">multi-residue-torsion-fit-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dresidue_002dtorsion_002dpy"><code>multi-residue-torsion-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#multi_002dresidue_002dtorsion_002dpy">multi-residue-torsion-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dresidue_002dtorsion_002dscm"><code>multi-residue-torsion-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#multi_002dresidue_002dtorsion_002dscm">multi-residue-torsion-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dsharpen_002dblur_002dmap_002dpy"><code>multi-sharpen-blur-map-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#multi_002dsharpen_002dblur_002dmap_002dpy">multi-sharpen-blur-map-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-multi_002dsharpen_002dblur_002dmap_002dscm"><code>multi-sharpen-blur-map-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#multi_002dsharpen_002dblur_002dmap_002dscm">multi-sharpen-blur-map-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate-1"><code>mutate</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate">mutate</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dactive_002dresidue"><code>mutate-active-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002dactive_002dresidue">mutate-active-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dand_002dauto_002dfit"><code>mutate-and-auto-fit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dand_002dautofit_002dresidue_002drange"><code>mutate-and-autofit-residue-range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate"><code>mutate-auto-fit-do-post-refine-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate">mutate-auto-fit-do-post-refine-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dbase"><code>mutate-base</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002dbase">mutate-base</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dby_002doverlap"><code>mutate-by-overlap</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dchain"><code>mutate-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dresidue_002drange"><code>mutate-residue-range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dresidue_002dredundant"><code>mutate-residue-redundant</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002din_002dscheme">mutate-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber"><code>mutate-single-residue-by-serial-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002dsingle_002dresidue_002dby_002dserial_002dnumber">mutate-single-residue-by-serial-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-my_002dbutton_002dcallback"><code>my-button-callback</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#extra_002dtop_002dlevel">extra-top-level</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-my_002dtop_002dlevel"><code>my-top-level</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#extra_002dtop_002dlevel">extra-top-level</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-N">N</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-n_002datoms"><code>n-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#n_002datoms">n-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-n_002dchains"><code>n-chains</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#n_002dchains">n-chains</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-n_002ddots_002dsets"><code>n-dots-sets</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#n_002ddots_002dsets">n-dots-sets</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-n_002dmodels"><code>n-models</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#n_002dmodels">n-models</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-n_002dresidues"><code>n-residues</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#n_002dresidues">n-residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-n_002drotamers"><code>n-rotamers</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#n_002drotamers">n-rotamers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-n_002dsymops"><code>n-symops</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#n_002dsymops">n-symops</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dchain_002dids_002dpy"><code>ncs-chain-ids-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dchain_002dids_002dpy">ncs-chain-ids-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dchain_002dids_002dscm"><code>ncs-chain-ids-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dchain_002dids_002dscm">ncs-chain-ids-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain"><code>ncs-control-change-ncs-master-to-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain">ncs-control-change-ncs-master-to-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did"><code>ncs-control-change-ncs-master-to-chain-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dcontrol_002dchange_002dncs_002dmaster_002dto_002dchain_002did">ncs-control-change-ncs-master-to-chain-id</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dcontrol_002ddisplay_002dchain"><code>ncs-control-display-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dcontrol_002ddisplay_002dchain">ncs-control-display-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dghosts_002dpy"><code>ncs-ghosts-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dghosts_002dpy">ncs-ghosts-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dghosts_002dscm"><code>ncs-ghosts-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dghosts_002dscm">ncs-ghosts-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dligand"><code>ncs-ligand</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs">ncs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dmaster_002dchain_002did"><code>ncs-master-chain-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs">ncs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dmaster_002dchains_002dscm"><code>ncs-master-chains-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dmaster_002dchains_002dscm">ncs-master-chains-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ncs_002dupdate_002dghosts"><code>ncs-update-ghosts</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs_002dupdate_002dghosts">ncs-update-ghosts</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-nearest_002dresidue_002dby_002dsequence_002dpy"><code>nearest-residue-by-sequence-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#nearest_002dresidue_002dby_002dsequence_002dpy">nearest-residue-by-sequence-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-nearest_002dresidue_002dby_002dsequence_002dscm"><code>nearest-residue-by-sequence-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#nearest_002dresidue_002dby_002dsequence_002dscm">nearest-residue-by-sequence-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-net_002dget_002durl"><code>net-get-url</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002debi">get-ebi</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002datom_002dselection"><code>new-molecule-by-atom-selection</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#new_002dmolecule_002dby_002datom_002dselection">new-molecule-by-atom-selection</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002dresidue_002dspecs_002dpy"><code>new-molecule-by-residue-specs-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dpy">new-molecule-by-residue-specs-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002dresidue_002dspecs_002dscm"><code>new-molecule-by-residue-specs-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#new_002dmolecule_002dby_002dresidue_002dspecs_002dscm">new-molecule-by-residue-specs-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002dresidue_002dtype_002dselection"><code>new-molecule-by-residue-type-selection</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#new_002dmolecule_002dby_002dresidue_002dtype_002dselection">new-molecule-by-residue-type-selection</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002dsmiles_002dstring"><code>new-molecule-by-smiles-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#g_t3d_002dgenerator_002dimport">3d-generator-import</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002dsphere_002dselection"><code>new-molecule-by-sphere-selection</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#new_002dmolecule_002dby_002dsphere_002dselection">new-molecule-by-sphere-selection</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002dsymmetry"><code>new-molecule-by-symmetry</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#new_002dmolecule_002dby_002dsymmetry">new-molecule-by-symmetry</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection"><code>new-molecule-by-symmetry-with-atom-selection</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#new_002dmolecule_002dby_002dsymmetry_002dwith_002datom_002dselection">new-molecule-by-symmetry-with-atom-selection</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dby_002dsymop"><code>new-molecule-by-symop</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#new_002dmolecule_002dby_002dsymop">new-molecule-by-symop</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-new_002dmolecule_002dfrom_002dthis_002dglyco_002dtree"><code>new-molecule-from-this-glyco-tree</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#add_002dlinked_002dcho">add-linked-cho</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-no_002dcoot_002dtips"><code>no-coot-tips</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#tips">tips</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-non_002dstandard_002dresidue_002dnames_002dpy"><code>non-standard-residue-names-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#non_002dstandard_002dresidue_002dnames_002dpy">non-standard-residue-names-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-non_002dstandard_002dresidue_002dnames_002dscm"><code>non-standard-residue-names-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#non_002dstandard_002dresidue_002dnames_002dscm">non-standard-residue-names-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-normal_002dcursor"><code>normal-cursor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#normal_002dcursor">normal-cursor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-notify_002dof_002dnew_002dversion"><code>notify-of-new-version</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#check_002dfor_002dupdates">check-for-updates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-nudge_002dscreen_002dcentre_002dgui"><code>nudge-screen-centre-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-number_002dlist"><code>number-list</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-O">O</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-old_002dcoot_003f"><code>old-coot?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-open_002dcif_002ddictionary_002dfile_002dselector_002ddialog"><code>open-cif-dictionary-file-selector-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#open_002dcif_002ddictionary_002dfile_002dselector_002ddialog">open-cif-dictionary-file-selector-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-open_002dcoords_002ddialog"><code>open-coords-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#open_002dcoords_002ddialog">open-coords-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-open_002dcoot_002dlistener_002dsocket"><code>open-coot-listener-socket</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remote_002dcontrol">remote-control</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-open_002dcoot_002dlistener_002dsocket_002dwith_002dtimeout"><code>open-coot-listener-socket-with-timeout</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remote_002dcontrol">remote-control</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-origin_002dpre_002dshift_002dpy"><code>origin-pre-shift-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#origin_002dpre_002dshift_002dpy">origin-pre-shift-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-origin_002dpre_002dshift_002dscm"><code>origin-pre-shift-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#origin_002dpre_002dshift_002dscm">origin-pre-shift-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-output_002datom_002dinfo_002das_002dtext"><code>output-atom-info-as-text</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#output_002datom_002dinfo_002das_002dtext">output-atom-info-as-text</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-overlap_002dligands"><code>overlap-ligands</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#overlap_002dligands">overlap-ligands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-overlay_002dmy_002dligands"><code>overlay-my-ligands</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-P">P</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-parse_002dcheck_002ddb"><code>parse-check-db</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#what_002dcheck">what-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-parse_002dpisa_002dassemblies"><code>parse-pisa-assemblies</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#parse_002dpisa_002dxml">parse-pisa-xml</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-parse_002dpisa_002dinterfaces"><code>parse-pisa-interfaces</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#parse_002dpisa_002dxml">parse-pisa-xml</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pdbe_002dget_002dpdb_002dand_002dsfs_002dcif"><code>pdbe-get-pdb-and-sfs-cif</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002drecent_002dpdbe">get-recent-pdbe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pdbe_002dlatest_002dreleases_002dgui"><code>pdbe-latest-releases-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002drecent_002dpdbe">get-recent-pdbe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pepflip-1"><code>pepflip</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#pepflip">pepflip</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pepflip_002dactive_002dresidue"><code>pepflip-active-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pepflips_002dby_002ddifference_002dmap_002dgui"><code>pepflips-by-difference-map-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-phosphorylate_002dactive_002dresidue"><code>phosphorylate-active-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pick_002dcursor_002dmaybe"><code>pick-cursor-maybe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#pick_002dcursor_002dmaybe">pick-cursor-maybe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pir_002dfile_002dname_002d_003epir_002dsequence"><code>pir-file-name-&gt;pir-sequence</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pisa_002dassemblies_002dxml"><code>pisa-assemblies-xml</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#parse_002dpisa_002dxml">parse-pisa-xml</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pisa_002dhandle_002dsxml_002dmolecule"><code>pisa-handle-sxml-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#parse_002dpisa_002dxml">parse-pisa-xml</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pisa_002dinteraction"><code>pisa-interaction</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#pisa_002dinteraction">pisa-interaction</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-place_002dhelix_002dhere"><code>place-helix-here</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#place_002dhelix_002dhere">place-helix-here</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-place_002dstrand_002dhere"><code>place-strand-here</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#place_002dstrand_002dhere">place-strand-here</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-place_002dstrand_002dhere_002ddialog"><code>place-strand-here-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#place_002dstrand_002dhere_002ddialog">place-strand-here-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-place_002dtext"><code>place-text</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#place_002dtext">place-text</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pointer_002datom_002dmolecule"><code>pointer-atom-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#pointer_002datom_002dmolecule">pointer-atom-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002ddisplay_002dcontrol_002dwindow"><code>post-display-control-window</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#post_002ddisplay_002dcontrol_002dwindow">post-display-control-window</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dgo_002dto_002datom_002dwindow"><code>post-go-to-atom-window</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#post_002dgo_002dto_002datom_002dwindow">post-go-to-atom-window</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dligand_002dfit_002dgui"><code>post-ligand-fit-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dmanipulation_002dhook-1"><code>post-manipulation-hook</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dmodel_002dfit_002drefine_002ddialog"><code>post-model-fit-refine-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#post_002dmodel_002dfit_002drefine_002ddialog">post-model-fit-refine-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dother_002dmodelling_002dtools_002ddialog"><code>post-other-modelling-tools-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#post_002dother_002dmodelling_002dtools_002ddialog">post-other-modelling-tools-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dpython_002dscripting_002dwindow"><code>post-python-scripting-window</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#post_002dpython_002dscripting_002dwindow">post-python-scripting-window</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dscheme_002dscripting_002dwindow"><code>post-scheme-scripting-window</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#post_002dscheme_002dscripting_002dwindow">post-scheme-scripting-window</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dscripting_002dwindow"><code>post-scripting-window</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#post_002dscripting_002dwindow">post-scripting-window</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-post_002dset_002drotation_002dcentre_002dhook"><code>post-set-rotation-centre-hook</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-povray_002dargs"><code>povray-args</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002dfrom_002dscheme">mutate-from-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-povray_002dimage"><code>povray-image</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002dfrom_002dscheme">mutate-from-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-povray_002dversion"><code>povray-version</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#mutate_002dfrom_002dscheme">mutate-from-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pre_002drelease_003f"><code>pre-release?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-prefer_002dpython"><code>prefer-python</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#prefer_002dpython">prefer-python</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-prep_002dfor_002dpisa"><code>prep-for-pisa</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#parse_002dpisa_002dxml">parse-pisa-xml</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dall_002dhistory_002din_002dpython"><code>print-all-history-in-python</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#print_002dall_002dhistory_002din_002dpython">print-all-history-in-python</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dall_002dhistory_002din_002dscheme"><code>print-all-history-in-scheme</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#print_002dall_002dhistory_002din_002dscheme">print-all-history-in-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dglyco_002dtree"><code>print-glyco-tree</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#print_002dglyco_002dtree">print-glyco-tree</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dheader_002dsecondary_002dstructure_002dinfo"><code>print-header-secondary-structure-info</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#print_002dheader_002dsecondary_002dstructure_002dinfo">print-header-secondary-structure-info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dmolecule_002dnames"><code>print-molecule-names</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dsequence"><code>print-sequence</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dsequence_002dchain"><code>print-sequence-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#print_002dsequence_002dchain">print-sequence-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dsequence_002dchain_002dgeneral"><code>print-sequence-chain-general</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#print_002dsequence_002dchain_002dgeneral">print-sequence-chain-general</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-print_002dview_002dmatrix"><code>print-view-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#print_002dview_002dmatrix">print-view-matrix</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-probe-1"><code>probe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-probe_002davailable_002dp"><code>probe-available-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#probe_002davailable_002dp">probe-available-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-probe_002dlocal_002dsphere"><code>probe-local-sphere</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-problem_002dresidues_002d_003edialog"><code>problem-residues-&gt;dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#what_002dcheck">what-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-prodrg_002dflat"><code>prodrg-flat</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#g_t3d_002dgenerator_002dimport">3d-generator-import</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-prodrg_002dify"><code>prodrg-ify</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-prodrg_002dxyzin"><code>prodrg-xyzin</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#g_t3d_002dgenerator_002dimport">3d-generator-import</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-protein_002ddb_002dloops"><code>protein-db-loops</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#protein_002ddb_002dloops">protein-db-loops</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pucker_002dinfo_002dpy"><code>pucker-info-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#pucker_002dinfo_002dpy">pucker-info-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pucker_002dinfo_002dscm"><code>pucker-info-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#pucker_002dinfo_002dscm">pucker-info-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-pukka_002dpuckers_003f"><code>pukka-puckers?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-python_002dat_002dprompt_002dat_002dstartup_002dstate"><code>python-at-prompt-at-startup-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#python_002dat_002dprompt_002dat_002dstartup_002dstate">python-at-prompt-at-startup-state</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-Q">Q</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-qq_002dplot_002dmap_002dand_002dmodel_002dscm"><code>qq-plot-map-and-model-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#qq_002dplot_002dmap_002dand_002dmodel_002dscm">qq-plot-map-and-model-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-quanta_002dbuttons"><code>quanta-buttons</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#quanta_002dbuttons">quanta-buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-quanta_002dlike_002dzoom"><code>quanta-like-zoom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#quanta_002dlike_002dzoom">quanta-like-zoom</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-quick_002dsave"><code>quick-save</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#quick_002dsave">quick-save</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-R">R</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-ramachandran_002dplot_002ddifferences"><code>ramachandran-plot-differences</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ramachandran_002dplot_002ddifferences">ramachandran-plot-differences</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ramachandran_002dplot_002ddifferences_002dby_002dchain"><code>ramachandran-plot-differences-by-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ramachandran_002dplot_002ddifferences_002dby_002dchain">ramachandran-plot-differences-by-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-range"><code>range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rapper_002ddir"><code>rapper-dir</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#group_002dsettings">group-settings</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rapper_002ddir-1"><code>rapper-dir</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#a_002drapper_002dgui">a-rapper-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-raster_002dscreen_002dshot"><code>raster-screen-shot</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#raster_002dscreen_002dshot">raster-screen-shot</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-raster3d"><code>raster3d</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#raster3d">raster3d</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-raytrace"><code>raytrace</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#raster3d_002dfrom_002dscheme">raster3d-from-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dpdb"><code>read-pdb</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#read_002dpdb">read-pdb</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dpdb_002dall"><code>read-pdb-all</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap"><code>read-phs-and-coords-and-make-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#read_002dphs_002dand_002dcoords_002dand_002dmake_002dmap">read-phs-and-coords-and-make-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm"><code>read-phs-and-make-map-using-cell-symm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm">read-phs-and-make-map-using-cell-symm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol"><code>read-phs-and-make-map-using-cell-symm-from-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dmol">read-phs-and-make-map-using-cell-symm-from-mol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol"><code>read-phs-and-make-map-using-cell-symm-from-previous-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#read_002dphs_002dand_002dmake_002dmap_002dusing_002dcell_002dsymm_002dfrom_002dprevious_002dmol">read-phs-and-make-map-using-cell-symm-from-previous-mol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits"><code>read-phs-and-make-map-with-reso-limits</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#read_002dphs_002dand_002dmake_002dmap_002dwith_002dreso_002dlimits">read-phs-and-make-map-with-reso-limits</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dshelx_002dins_002dfile"><code>read-shelx-ins-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#read_002dshelx_002dins_002dfile">read-shelx-ins-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dshelx_002dlst_002dfile"><code>read-shelx-lst-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#shelx">shelx</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-read_002dvu_002dfile"><code>read-vu-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reattach_002dmodelling_002dtoolbar"><code>reattach-modelling-toolbar</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#reattach_002dmodelling_002dtoolbar">reattach-modelling-toolbar</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-recentre_002don_002dread_002dpdb"><code>recentre-on-read-pdb</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#recentre_002don_002dread_002dpdb">recentre-on-read-pdb</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-recover_002dsession"><code>recover-session</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#recover_002dsession">recover-session</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-redraw_002dbackground"><code>redraw-background</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#redraw_002dbackground">redraw-background</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reduce_002don_002dpdb_002dfile"><code>reduce-on-pdb-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reduce_002don_002dpdb_002dfile_002dgeneric"><code>reduce-on-pdb-file-generic</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reduce_002don_002dpdb_002dfile_002dno_002dflip"><code>reduce-on-pdb-file-no-flip</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dactive_002dresidue"><code>refine-active-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dactive_002dresidue_002dgeneric"><code>refine-active-residue-generic</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dactive_002dresidue_002dtriple"><code>refine-active-residue-triple</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dauto_002drange"><code>refine-auto-range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refine_002dauto_002drange">refine-auto-range</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dresidues_002dpy"><code>refine-residues-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refine_002dresidues_002dpy">refine-residues-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dresidues_002dscm"><code>refine-residues-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refine_002dresidues_002dscm">refine-residues-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dwith_002dtorsion_002drestraints_002dstate"><code>refine-with-torsion-restraints-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refine_002dwith_002dtorsion_002drestraints_002dstate">refine-with-torsion-restraints-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dzone"><code>refine-zone</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refine_002dzone">refine-zone</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dzone_002datom_002dindex_002ddefine"><code>refine-zone-atom-index-define</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refine_002dzone_002datom_002dindex_002ddefine">refine-zone-atom-index-define</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm"><code>refine-zone-with-full-residue-spec-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refine_002dzone_002dwith_002dfull_002dresidue_002dspec_002dscm">refine-zone-with-full-residue-spec-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refinement_002dalready_002dongoing_002dp"><code>refinement-already-ongoing-p</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refinement_002dalready_002dongoing_002dp">refinement-already-ongoing-p</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refinement_002dimmediate_002dreplacement_002dstate"><code>refinement-immediate-replacement-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refinement_002dimmediate_002dreplacement_002dstate">refinement-immediate-replacement-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate"><code>refinement-move-atoms-with-zero-occupancy-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refinement_002dmove_002datoms_002dwith_002dzero_002doccupancy_002dstate">refinement-move-atoms-with-zero-occupancy-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refinement_002drefine_002dper_002dframe_002dstate"><code>refinement-refine-per-frame-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refinement_002drefine_002dper_002dframe_002dstate">refinement-refine-per-frame-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac_002dcalc_002dsfs_002dmake_002dmtz_002dwith_002dcolumns"><code>refmac-calc-sfs-make-mtz-with-columns</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#get_002drecent_002dpdbe">get-recent-pdbe</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac_002dexe"><code>refmac-exe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refmac">refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac_002dextra_002dparams"><code>refmac-extra-params</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refmac">refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac_002dname"><code>refmac-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refmac_002dname">refmac-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-refmac_002dproblems_002dgui"><code>refmac-problems-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refmac_002dproblems">refmac-problems</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-regularize_002dresidues"><code>regularize-residues</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#regularize_002dresidues">regularize-residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-regularize_002dzone"><code>regularize-zone</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#regularize_002dzone">regularize-zone</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reinterp_002dmap"><code>reinterp-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#reinterp_002dmap">reinterp-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reload_002ddictionary"><code>reload-dictionary</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#reload_002ddictionary">reload-dictionary</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remarks_002ddialog"><code>remarks-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remarks_002ddialog">remarks-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remarks_002dscm"><code>remarks-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remarks_002dscm">remarks-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002dcoordinates_002dglob_002dextension"><code>remove-coordinates-glob-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002dcoordinates_002dglob_002dextension">remove-coordinates-glob-extension</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002ddata_002dglob_002dextension"><code>remove-data-glob-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002ddata_002dglob_002dextension">remove-data-glob-extension</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002ddictionary_002dglob_002dextension"><code>remove-dictionary-glob-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002ddictionary_002dglob_002dextension">remove-dictionary-glob-extension</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002dmap_002dglob_002dextension"><code>remove-map-glob-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002dmap_002dglob_002dextension">remove-map-glob-extension</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002dnamed_002dview"><code>remove-named-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002dnamed_002dview">remove-named-view</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002dnon_002dauto_002dload_002dresidue_002dname"><code>remove-non-auto-load-residue-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002dnon_002dauto_002dload_002dresidue_002dname">remove-non-auto-load-residue-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002domega_002dtorsion_002drestriants"><code>remove-omega-torsion-restriants</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002domega_002dtorsion_002drestriants">remove-omega-torsion-restriants</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002dplanar_002dpeptide_002drestraints"><code>remove-planar-peptide-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002dplanar_002dpeptide_002drestraints">remove-planar-peptide-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002dtext"><code>remove-text</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002dtext">remove-text</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-remove_002dview"><code>remove-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#remove_002dview">remove-view</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rename_002dresidue_002dgui"><code>rename-residue-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-render_002dimage"><code>render-image</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#raster3d_002dfrom_002dscheme">raster3d-from-scheme</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-renumber_002dresidue_002drange"><code>renumber-residue-range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#renumber_002dresidue_002drange">renumber-residue-range</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-renumber_002dwaters"><code>renumber-waters</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#renumber_002dwaters">renumber-waters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reorder_002dchains"><code>reorder-chains</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-repeat_002drefine_002dzone"><code>repeat-refine-zone</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#repeat_002drefine_002dzone">repeat-refine-zone</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-replace_002dfragment"><code>replace-fragment</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#replace_002dfragment">replace-fragment</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-replace_002dresidues_002dfrom_002dmol_002dscm"><code>replace-residues-from-mol-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#replace_002dresidues_002dfrom_002dmol_002dscm">replace-residues-from-mol-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-res_002dspec_002d_003echain_002did"><code>res-spec-&gt;chain-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-res_002dspec_002d_003eins_002dcode"><code>res-spec-&gt;ins-code</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-res_002dspec_002d_003eres_002dno"><code>res-spec-&gt;res-no</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-reset_002dview"><code>reset-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#reset_002dview">reset-view</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dalt_002dconfs"><code>residue-alt-confs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002datom_002d_003ealt_002dconf"><code>residue-atom-&gt;alt-conf</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002datom_002d_003eatom_002dname"><code>residue-atom-&gt;atom-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002datom_002d_003eatom_002dspec"><code>residue-atom-&gt;atom-spec</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002datom_002d_003eoccupancy"><code>residue-atom-&gt;occupancy</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002datom_002d_003eposition"><code>residue-atom-&gt;position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002ddensity_002dfit_002dscale_002dfactor"><code>residue-density-fit-scale-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residue_002ddensity_002dfit_002dscale_002dfactor">residue-density-fit-scale-factor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dexists_003f"><code>residue-exists?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dhas_002dhetatms"><code>residue-has-hetatms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residue_002dhas_002dhetatms">residue-has-hetatms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dhas_002dhetatms_003f"><code>residue-has-hetatms?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dinfo"><code>residue-info</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residue_002dinfo">residue-info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dinfo_002ddialog"><code>residue-info-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residue_002dinfo_002ddialog">residue-info-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dinfo_002dpy"><code>residue-info-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residue_002dinfo_002dpy">residue-info-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dname"><code>residue-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residue_002dname">residue-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002drange_002dgui"><code>residue-range-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dspec_002d_003eatom_002dfor_002dcentre"><code>residue-spec-&gt;atom-for-centre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dspec_002d_003eatom_002dselection_002dstring"><code>residue-spec-&gt;atom-selection-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dspec_002d_003echain_002did"><code>residue-spec-&gt;chain-id</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dspec_002d_003estring"><code>residue-spec-&gt;string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#what_002dcheck">what-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residue_002dspec_002dless_002dthan"><code>residue-spec-less-than</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residues_002din_002dchain"><code>residues-in-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residues_002dmatching_002dcriteria"><code>residues-matching-criteria</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residues_002dnear_002dposition_002dpy"><code>residues-near-position-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residues_002dnear_002dposition_002dpy">residues-near-position-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residues_002dnear_002dposition_002dscm"><code>residues-near-position-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residues_002dnear_002dposition_002dscm">residues-near-position-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residues_002dnear_002dresidue"><code>residues-near-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residues_002dnear_002dresidue">residues-near-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residues_002dnear_002dresidues_002dscm"><code>residues-near-residues-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#residues_002dnear_002dresidues_002dscm">residues-near-residues-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-residues_002dwith_002dalt_002dconfs"><code>residues-with-alt-confs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-resname_002dfrom_002dserial_002dnumber"><code>resname-from-serial-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#resname_002dfrom_002dserial_002dnumber">resname-from-serial-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-restore_002dprevious_002dmap_002dcolour"><code>restore-previous-map-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#restore_002dprevious_002dmap_002dcolour">restore-previous-map-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rigid_002dbody_002drefine_002dby_002dresidue_002dranges"><code>rigid-body-refine-by-residue-ranges</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy"><code>rigid-body-refine-by-residue-ranges-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dpy">rigid-body-refine-by-residue-ranges-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm"><code>rigid-body-refine-by-residue-ranges-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rigid_002dbody_002drefine_002dby_002dresidue_002dranges_002dscm">rigid-body-refine-by-residue-ranges-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rigid_002dbody_002drefine_002dzone"><code>rigid-body-refine-zone</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rigid_002dbody_002drefine_002dzone">rigid-body-refine-zone</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate"><code>rotamer-auto-fit-do-post-refine-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rotamer_002dauto_002dfit_002ddo_002dpost_002drefine_002dstate">rotamer-auto-fit-do-post-refine-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotamer_002dgraphs"><code>rotamer-graphs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rotamer_002dgraphs">rotamer-graphs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotamer_002dgraphs_002dpy"><code>rotamer-graphs-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rotamer_002dgraphs_002dpy">rotamer-graphs-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotate_002dabout_002dscreen_002daxis"><code>rotate-about-screen-axis</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotate_002dcursor"><code>rotate-cursor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rotate_002dcursor">rotate-cursor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotate_002dx_002dscene"><code>rotate-x-scene</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rotate_002dx_002dscene">rotate-x-scene</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotate_002dy_002dscene"><code>rotate-y-scene</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rotate_002dy_002dscene">rotate-y-scene</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotate_002dz_002dscene"><code>rotate-z-scene</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#rotate_002dz_002dscene">rotate-z-scene</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-rotation_002dcentre"><code>rotation-centre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dclear_002dbackups"><code>run-clear-backups</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#run_002dclear_002dbackups">run-clear-backups</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dclustalw_002dalignment"><code>run-clustalw-alignment</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dcommand_002fstrings"><code>run-command/strings</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dcommand_002fstrings-1"><code>run-command/strings</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dcommand_002fstrings-2"><code>run-command/strings</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002ddownload_002dbinary_002dcurl"><code>run-download-binary-curl</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dgtk_002dpending_002devents"><code>run-gtk-pending-events</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dguile_002dscript"><code>run-guile-script</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#run_002dguile_002dscript">run-guile-script</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dmogul"><code>run-mogul</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#run_002dmogul">run-mogul</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dmogul_002dshow_002dresults"><code>run-mogul-show-results</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#run_002dmogul">run-mogul</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dprosmart"><code>run-prosmart</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#gui_002dprosmart">gui-prosmart</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dpython_002dscript"><code>run-python-script</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#run_002dpython_002dscript">run-python-script</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002drefmac_002dby_002dfilename"><code>run-refmac-by-filename</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refmac">refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002drefmac_002dfor_002dphases"><code>run-refmac-for-phases</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#refmac">refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dscript"><code>run-script</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#run_002dscript">run-script</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dstate_002dfile"><code>run-state-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#run_002dstate_002dfile">run-state-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-run_002dstate_002dfile_002dmaybe"><code>run-state-file-maybe</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#run_002dstate_002dfile_002dmaybe">run-state-file-maybe</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-S">S</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-sanitise_002dalt_002dconfs"><code>sanitise-alt-confs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sanitise_002dalt_002dconfs_002dactive_002dresidue"><code>sanitise-alt-confs-active-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sanitise_002dalt_002dconfs_002din_002dresidue"><code>sanitise-alt-confs-in-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dannotations"><code>save-annotations</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dcoordinates"><code>save-coordinates</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dcoordinates">save-coordinates</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002ddialog_002dpositions_002dto_002dinit_002dfile"><code>save-dialog-positions-to-init-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dprevious_002dmap_002dcolour"><code>save-previous-map-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dprevious_002dmap_002dcolour">save-previous-map-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dstate"><code>save-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dstate">save-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dstate_002dfile"><code>save-state-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dstate_002dfile">save-state-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dstate_002dfile_002dname_002dpy"><code>save-state-file-name-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dstate_002dfile_002dname_002dpy">save-state-file-name-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dstate_002dfile_002dname_002dscm"><code>save-state-file-name-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dstate_002dfile_002dname_002dscm">save-state-file-name-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dstate_002dfile_002dpy"><code>save-state-file-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dstate_002dfile_002dpy">save-state-file-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dsymmetry_002dcoords"><code>save-symmetry-coords</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dsymmetry_002dcoords">save-symmetry-coords</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-save_002dviews"><code>save-views</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#save_002dviews">save-views</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scale_002dcell"><code>scale-cell</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#scale_002dcell">scale-cell</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scale_002dzoom"><code>scale-zoom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#scale_002dzoom">scale-zoom</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scheme_002drepresentation"><code>scheme-representation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-score_002dresidue_002drange_002dfit_002dto_002dmap"><code>score-residue-range-fit-to-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#score_002dresidue_002drange_002dfit_002dto_002dmap">score-residue-range-fit-to-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-score_002drotamers_002dscm"><code>score-rotamers-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#score_002drotamers_002dscm">score-rotamers-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-screendump_002dimage"><code>screendump-image</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#screendump_002dimage">screendump-image</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scroll_002dby_002dwheel_002dmouse_002dstate"><code>scroll-by-wheel-mouse-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#scroll_002dby_002dwheel_002dmouse_002dstate">scroll-by-wheel-mouse-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-scroll_002dwheel_002dmap"><code>scroll-wheel-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#scroll_002dwheel_002dmap">scroll-wheel-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-secondary_002dstructure_002drestraints_002dtype"><code>secondary-structure-restraints-type</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#secondary_002dstructure_002drestraints_002dtype">secondary-structure-restraints-type</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-select_002datom_002dunder_002dpointer_002dpy"><code>select-atom-under-pointer-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#select_002datom_002dunder_002dpointer_002dpy">select-atom-under-pointer-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-select_002datom_002dunder_002dpointer_002dscm"><code>select-atom-under-pointer-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#select_002datom_002dunder_002dpointer_002dscm">select-atom-under-pointer-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-seqnum_002dfrom_002dserial_002dnumber"><code>seqnum-from-serial-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#seqnum_002dfrom_002dserial_002dnumber">seqnum-from-serial-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sequence_002dinfo"><code>sequence-info</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#sequence_002dinfo">sequence-info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sequence_002dinfo_002dpy"><code>sequence-info-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#sequence_002dinfo_002dpy">sequence-info-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-serial_002dnumber_002dfrom_002dresidue_002dspecs"><code>serial-number-from-residue-specs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#serial_002dnumber_002dfrom_002dresidue_002dspecs">serial-number-from-residue-specs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002daccept_002dreject_002ddialog_002ddocked_002dshow"><code>set-accept-reject-dialog-docked-show</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002daccept_002dreject_002ddialog_002ddocked_002dshow">set-accept-reject-dialog-docked-show</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002daccept_002dreject_002ddialog_002dposition"><code>set-accept-reject-dialog-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002daccept_002dreject_002ddialog_002dposition">set-accept-reject-dialog-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dactive_002dmap_002ddrag_002dflag"><code>set-active-map-drag-flag</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dactive_002dmap_002ddrag_002dflag">set-active-map-drag-flag</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs"><code>set-add-ccp4i-projects-to-file-dialogs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dadd_002dccp4i_002dprojects_002dto_002dfile_002ddialogs">set-add-ccp4i-projects-to-file-dialogs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype"><code>set-add-terminal-residue-default-residue-type</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dadd_002dterminal_002dresidue_002ddefault_002dresidue_002dtype">set-add-terminal-residue-default-residue-type</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine"><code>set-add-terminal-residue-do-post-refine</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dadd_002dterminal_002dresidue_002ddo_002dpost_002drefine">set-add-terminal-residue-do-post-refine</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition"><code>set-add-terminal-residue-immediate-addition</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dadd_002dterminal_002dresidue_002dimmediate_002daddition">set-add-terminal-residue-immediate-addition</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dalignment_002dgap_002dand_002dspace_002dpenalty"><code>set-alignment-gap-and-space-penalty</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dalignment_002dgap_002dand_002dspace_002dpenalty">set-alignment-gap-and-space-penalty</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dall_002dmaps_002ddisplayed"><code>set-all-maps-displayed</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dall_002dmaps_002ddisplayed">set-all-maps-displayed</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dall_002dmodels_002ddisplayed_002dand_002dactive"><code>set-all-models-displayed-and-active</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dall_002dmodels_002ddisplayed_002dand_002dactive">set-all-models-displayed-and-active</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002daniso_002dprobability"><code>set-aniso-probability</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002daniso_002dprobability">set-aniso-probability</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002datom_002dattribute"><code>set-atom-attribute</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002datom_002dattribute">set-atom-attribute</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002datom_002dattributes"><code>set-atom-attributes</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002datom_002dattributes">set-atom-attributes</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002datom_002dstring_002dattribute"><code>set-atom-string-attribute</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002datom_002dstring_002dattribute">set-atom-string-attribute</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag"><code>set-auto-fit-best-rotamer-clash-flag</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dauto_002dfit_002dbest_002drotamer_002dclash_002dflag">set-auto-fit-best-rotamer-clash-flag</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dauto_002dh_002dbond_002drestraints"><code>set-auto-h-bond-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dauto_002dh_002dbond_002drestraints">set-auto-h-bond-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dauto_002dread_002dcolumn_002dlabels"><code>set-auto-read-column-labels</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dauto_002dread_002dcolumn_002dlabels">set-auto-read-column-labels</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dauto_002drecontour_002dmap"><code>set-auto-recontour-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dauto_002drecontour_002dmap">set-auto-recontour-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dauto_002dupdating_002dsfcalc_002dgenmap"><code>set-auto-updating-sfcalc-genmap</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dauto_002dupdating_002dsfcalc_002dgenmap">set-auto-updating-sfcalc-genmap</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002db_002dfactor_002dbonds_002dscale_002dfactor"><code>set-b-factor-bonds-scale-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002db_002dfactor_002dbonds_002dscale_002dfactor">set-b-factor-bonds-scale-factor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbackground_002dcolour"><code>set-background-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbackground_002dcolour">set-background-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbackup_002dcompress_002dfiles"><code>set-backup-compress-files</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbackup_002dcompress_002dfiles">set-backup-compress-files</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbaton_002dbuild_002dparams"><code>set-baton-build-params</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbaton_002dbuild_002dparams">set-baton-build-params</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbaton_002dmode"><code>set-baton-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbaton_002dmode">set-baton-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbond_002dcolour_002drotation_002dfor_002dmolecule"><code>set-bond-colour-rotation-for-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbond_002dcolour_002drotation_002dfor_002dmolecule">set-bond-colour-rotation-for-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbond_002dthickness"><code>set-bond-thickness</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbond_002dthickness">set-bond-thickness</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbond_002dthickness_002dintermediate_002datoms"><code>set-bond-thickness-intermediate-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbond_002dthickness_002dintermediate_002datoms">set-bond-thickness-intermediate-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbrief_002datom_002dlabels"><code>set-brief-atom-labels</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbrief_002datom_002dlabels">set-brief-atom-labels</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbrowser_002dinterface"><code>set-browser-interface</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbrowser_002dinterface">set-browser-interface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dbutton_002dlabel_002dfor_002dexternal_002drefinement"><code>set-button-label-for-external-refinement</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dbutton_002dlabel_002dfor_002dexternal_002drefinement">set-button-label-for-external-refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dclipping_002dback"><code>set-clipping-back</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dclipping_002dback">set-clipping-back</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dclipping_002dfront"><code>set-clipping-front</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dclipping_002dfront">set-clipping-front</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcolour_002dby_002dchain"><code>set-colour-by-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcolour_002dby_002dchain">set-colour-by-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcolour_002dby_002dchain_002dgoodsell_002dmode"><code>set-colour-by-chain-goodsell-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcolour_002dby_002dchain_002dgoodsell_002dmode">set-colour-by-chain-goodsell-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcolour_002dby_002dmolecule"><code>set-colour-by-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcolour_002dby_002dmolecule">set-colour-by-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcolour_002dmap_002drotation_002dfor_002dmap"><code>set-colour-map-rotation-for-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcolour_002dmap_002drotation_002dfor_002dmap">set-colour-map-rotation-for-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb"><code>set-colour-map-rotation-on-read-pdb</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb">set-colour-map-rotation-on-read-pdb</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag"><code>set-colour-map-rotation-on-read-pdb-c-only-flag</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dc_002donly_002dflag">set-colour-map-rotation-on-read-pdb-c-only-flag</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag"><code>set-colour-map-rotation-on-read-pdb-flag</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcolour_002dmap_002drotation_002don_002dread_002dpdb_002dflag">set-colour-map-rotation-on-read-pdb-flag</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dconsole_002ddisplay_002dcommands_002dhilights"><code>set-console-display-commands-hilights</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dconsole_002ddisplay_002dcommands_002dhilights">set-console-display-commands-hilights</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dconsole_002ddisplay_002dcommands_002dstate"><code>set-console-display-commands-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dconsole_002ddisplay_002dcommands_002dstate">set-console-display-commands-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol"><code>set-contour-by-sigma-step-by-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcontour_002dby_002dsigma_002dstep_002dby_002dmol">set-contour-by-sigma-step-by-mol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcontour_002dlevel_002dabsolute"><code>set-contour-level-absolute</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcontour_002dlevel_002dabsolute">set-contour-level-absolute</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcontour_002dlevel_002din_002dsigma"><code>set-contour-level-in-sigma</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcontour_002dlevel_002din_002dsigma">set-contour-level-in-sigma</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dcontrol_002dkey_002dfor_002drotate"><code>set-control-key-for-rotate</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dcontrol_002dkey_002dfor_002drotate">set-control-key-for-rotate</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dconvert_002dto_002dv2_002datom_002dnames"><code>set-convert-to-v2-atom-names</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dconvert_002dto_002dv2_002datom_002dnames">set-convert-to-v2-atom-names</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddecoloned_002dbackup_002dfile_002dnames"><code>set-decoloned-backup-file-names</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddecoloned_002dbackup_002dfile_002dnames">set-decoloned-backup-file-names</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddefault_002dbond_002dthickness"><code>set-default-bond-thickness</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddefault_002dbond_002dthickness">set-default-bond-thickness</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap"><code>set-default-initial-contour-level-for-difference-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002ddifference_002dmap">set-default-initial-contour-level-for-difference-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap"><code>set-default-initial-contour-level-for-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddefault_002dinitial_002dcontour_002dlevel_002dfor_002dmap">set-default-initial-contour-level-for-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddefault_002drepresentation_002dtype"><code>set-default-representation-type</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddefault_002drepresentation_002dtype">set-default-representation-type</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms"><code>set-default-temperature-factor-for-new-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddefault_002dtemperature_002dfactor_002dfor_002dnew_002datoms">set-default-temperature-factor-for-new-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddelete_002ddialog_002dposition"><code>set-delete-dialog-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddelete_002ddialog_002dposition">set-delete-dialog-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddensity_002dsize"><code>set-density-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddensity_002dsize">set-density-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddiff_002dmap_002diso_002dlevel_002dincrement"><code>set-diff-map-iso-level-increment</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement">set-diff-map-iso-level-increment</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext"><code>set-diff-map-iso-level-increment-from-text</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddiff_002dmap_002diso_002dlevel_002dincrement_002dfrom_002dtext">set-diff-map-iso-level-increment-from-text</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddisplay_002dcontrol_002ddialog_002dposition"><code>set-display-control-dialog-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddisplay_002dcontrol_002ddialog_002dposition">set-display-control-dialog-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddisplay_002dintro_002dstring"><code>set-display-intro-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddisplay_002dintro_002dstring">set-display-intro-string</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddisplay_002dlists_002dfor_002dmaps"><code>set-display-lists-for-maps</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddisplay_002dlists_002dfor_002dmaps">set-display-lists-for-maps</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddisplay_002donly_002dmodel_002dmol"><code>set-display-only-model-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddisplay_002donly_002dmodel_002dmol">set-display-only-model-mol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddo_002danti_002daliasing"><code>set-do-anti-aliasing</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddo_002danti_002daliasing">set-do-anti-aliasing</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddo_002dGL_002dlighting"><code>set-do-GL-lighting</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddo_002dGL_002dlighting">set-do-GL-lighting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis"><code>set-do-probe-dots-on-rotamers-and-chis</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddo_002dprobe_002ddots_002don_002drotamers_002dand_002dchis">set-do-probe-dots-on-rotamers-and-chis</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddo_002dprobe_002ddots_002dpost_002drefine"><code>set-do-probe-dots-post-refine</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddo_002dprobe_002ddots_002dpost_002drefine">set-do-probe-dots-post-refine</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddots_002dcolor"><code>set-dots-color</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddots_002dcolour"><code>set-dots-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddots_002dcolour">set-dots-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddragged_002drefinement_002dsteps_002dper_002dframe"><code>set-dragged-refinement-steps-per-frame</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddragged_002drefinement_002dsteps_002dper_002dframe">set-dragged-refinement-steps-per-frame</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dcis_002dpeptide_002dmarkups"><code>set-draw-cis-peptide-markups</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dcis_002dpeptide_002dmarkups">set-draw-cis-peptide-markups</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dcrosshairs"><code>set-draw-crosshairs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dcrosshairs">set-draw-crosshairs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dhydrogens"><code>set-draw-hydrogens</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dhydrogens">set-draw-hydrogens</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dmap_002dstandard_002dlines"><code>set-draw-map-standard-lines</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dmap_002dstandard_002dlines">set-draw-map-standard-lines</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dmissing_002dresidues_002dloops"><code>set-draw-missing-residues-loops</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dmissing_002dresidues_002dloops">set-draw-missing-residues-loops</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dncs_002dghosts"><code>set-draw-ncs-ghosts</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dncs_002dghosts">set-draw-ncs-ghosts</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dsolid_002ddensity_002dsurface"><code>set-draw-solid-density-surface</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dsolid_002ddensity_002dsurface">set-draw-solid-density-surface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dstick_002dmode_002datoms"><code>set-draw-stick-mode-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dstick_002dmode_002datoms">set-draw-stick-mode-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dstick_002dmode_002datoms_002ddefault"><code>set-draw-stick-mode-atoms-default</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dstick_002dmode_002datoms_002ddefault">set-draw-stick-mode-atoms-default</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002ddraw_002dzero_002docc_002dmarkers"><code>set-draw-zero-occ-markers</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002ddraw_002dzero_002docc_002dmarkers">set-draw-zero-occ-markers</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dedit_002dchi_002dangles_002ddialog_002dposition"><code>set-edit-chi-angles-dialog-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dedit_002dchi_002dangles_002ddialog_002dposition">set-edit-chi-angles-dialog-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002denvironment_002ddistances_002ddistance_002dlimits"><code>set-environment-distances-distance-limits</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002denvironment_002ddistances_002ddistance_002dlimits">set-environment-distances-distance-limits</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002denvironment_002ddistances_002dlabel_002datom"><code>set-environment-distances-label-atom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002denvironment_002ddistances_002dlabel_002datom">set-environment-distances-label-atom</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002desoteric_002ddepth_002dcue"><code>set-esoteric-depth-cue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002desoteric_002ddepth_002dcue">set-esoteric-depth-cue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits"><code>set-extra-restraints-prosmart-sigma-limits</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dextra_002drestraints_002dprosmart_002dsigma_002dlimits">set-extra-restraints-prosmart-sigma-limits</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfffear_002dangular_002dresolution"><code>set-fffear-angular-resolution</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfffear_002dangular_002dresolution">set-fffear-angular-resolution</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfile_002dchooser_002dselector"><code>set-file-chooser-selector</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfile_002dchooser_002dselector">set-file-chooser-selector</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfilter_002dfileselection_002dfilenames"><code>set-filter-fileselection-filenames</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfilter_002dfileselection_002dfilenames">set-filter-fileselection-filenames</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfind_002dhydrogen_002dtorsion"><code>set-find-hydrogen-torsion</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#redefine_002dfunctions">redefine-functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfind_002dhydrogen_002dtorsions"><code>set-find-hydrogen-torsions</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfind_002dhydrogen_002dtorsions">set-find-hydrogen-torsions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfind_002dligand_002dhere_002dcluster"><code>set-find-ligand-here-cluster</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfind_002dligand_002dhere_002dcluster">set-find-ligand-here-cluster</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfind_002dligand_002dmask_002dwaters"><code>set-find-ligand-mask-waters</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfind_002dligand_002dmask_002dwaters">set-find-ligand-mask-waters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster"><code>set-find-ligand-multi-solutions-per-cluster</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfind_002dligand_002dmulti_002dsolutions_002dper_002dcluster">set-find-ligand-multi-solutions-per-cluster</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfind_002dligand_002dn_002dtop_002dligands"><code>set-find-ligand-n-top-ligands</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfind_002dligand_002dn_002dtop_002dligands">set-find-ligand-n-top-ligands</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement"><code>set-fix-chiral-volumes-before-refinement</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfix_002dchiral_002dvolumes_002dbefore_002drefinement">set-fix-chiral-volumes-before-refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface"><code>set-flat-shading-for-solid-density-surface</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dflat_002dshading_002dfor_002dsolid_002ddensity_002dsurface">set-flat-shading-for-solid-density-surface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfont_002dcolor"><code>set-font-color</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfont_002dcolour"><code>set-font-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfont_002dcolour">set-font-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfont_002dsize"><code>set-font-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfont_002dsize">set-font-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dfps_002dflag"><code>set-fps-flag</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dfps_002dflag">set-fps-flag</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname"><code>set-go-to-atom-chain-residue-atom-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname">set-go-to-atom-chain-residue-atom-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull"><code>set-go-to-atom-chain-residue-atom-name-full</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dfull">set-go-to-atom-chain-residue-atom-name-full</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw"><code>set-go-to-atom-chain-residue-atom-name-no-redraw</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgo_002dto_002datom_002dchain_002dresidue_002datom_002dname_002dno_002dredraw">set-go-to-atom-chain-residue-atom-name-no-redraw</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgo_002dto_002datom_002dmolecule"><code>set-go-to-atom-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgo_002dto_002datom_002dmolecule">set-go-to-atom-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgo_002dto_002datom_002dwindow_002dposition"><code>set-go-to-atom-window-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgo_002dto_002datom_002dwindow_002dposition">set-go-to-atom-window-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit"><code>set-go-to-ligand-n-atoms-limit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgo_002dto_002dligand_002dn_002datoms_002dlimit">set-go-to-ligand-n-atoms-limit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgraphics_002dwindow_002dposition"><code>set-graphics-window-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgraphics_002dwindow_002dposition">set-graphics-window-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgraphics_002dwindow_002dsize"><code>set-graphics-window-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgraphics_002dwindow_002dsize">set-graphics-window-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dgrey_002dcarbon_002dcolour"><code>set-grey-carbon-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dgrey_002dcarbon_002dcolour">set-grey-carbon-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dhardware_002dstereo_002dangle_002dfactor"><code>set-hardware-stereo-angle-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dhardware_002dstereo_002dangle_002dfactor">set-hardware-stereo-angle-factor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dhave_002dunsaved_002dchanges"><code>set-have-unsaved-changes</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dhave_002dunsaved_002dchanges">set-have-unsaved-changes</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002didle_002dfunction_002drotate_002dangle"><code>set-idle-function-rotate-angle</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002didle_002dfunction_002drotate_002dangle">set-idle-function-rotate-angle</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dimol_002drefinement_002dmap"><code>set-imol-refinement-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dimol_002drefinement_002dmap">set-imol-refinement-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius"><code>set-interactive-probe-dots-molprobity-radius</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dinteractive_002dprobe_002ddots_002dmolprobity_002dradius">set-interactive-probe-dots-molprobity-radius</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002diso_002dlevel_002dincrement"><code>set-iso-level-increment</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002diso_002dlevel_002dincrement">set-iso-level-increment</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dkeep_002dmap_002dcolour_002dafter_002drefmac"><code>set-keep-map-colour-after-refmac</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dkeep_002dmap_002dcolour_002dafter_002drefmac">set-keep-map-colour-after-refmac</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dkleywegt_002dplot_002dn_002ddiffs"><code>set-kleywegt-plot-n-diffs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dkleywegt_002dplot_002dn_002ddiffs">set-kleywegt-plot-n-diffs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dlast_002dmap_002dcolor"><code>set-last-map-color</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dlast_002dmap_002dcolour"><code>set-last-map-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dlast_002dmap_002dcolour">set-last-map-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dlast_002dmap_002dcontour_002dlevel"><code>set-last-map-contour-level</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dlast_002dmap_002dcontour_002dlevel">set-last-map-contour-level</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma"><code>set-last-map-contour-level-by-sigma</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dlast_002dmap_002dcontour_002dlevel_002dby_002dsigma">set-last-map-contour-level-by-sigma</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dlast_002dmap_002dsigma_002dstep"><code>set-last-map-sigma-step</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dlast_002dmap_002dsigma_002dstep">set-last-map-sigma-step</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dligand_002dacceptable_002dfit_002dfraction"><code>set-ligand-acceptable-fit-fraction</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dligand_002dacceptable_002dfit_002dfraction">set-ligand-acceptable-fit-fraction</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dligand_002dcluster_002dsigma_002dlevel"><code>set-ligand-cluster-sigma-level</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dligand_002dcluster_002dsigma_002dlevel">set-ligand-cluster-sigma-level</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dligand_002dflexible_002dligand_002dn_002dsamples"><code>set-ligand-flexible-ligand-n-samples</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dligand_002dflexible_002dligand_002dn_002dsamples">set-ligand-flexible-ligand-n-samples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dligand_002dsearch_002dmap_002dmolecule"><code>set-ligand-search-map-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dligand_002dsearch_002dmap_002dmolecule">set-ligand-search-map-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dligand_002dsearch_002dprotein_002dmolecule"><code>set-ligand-search-protein-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dligand_002dsearch_002dprotein_002dmolecule">set-ligand-search-protein-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dligand_002dwater_002dn_002dcycles"><code>set-ligand-water-n-cycles</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dligand_002dwater_002dn_002dcycles">set-ligand-water-n-cycles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits"><code>set-ligand-water-to-protein-distance-limits</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dligand_002dwater_002dto_002dprotein_002ddistance_002dlimits">set-ligand-water-to-protein-distance-limits</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dlimit_002daniso"><code>set-limit-aniso</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dlimit_002daniso">set-limit-aniso</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor"><code>set-log-cosh-target-distance-scale-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dlog_002dcosh_002dtarget_002ddistance_002dscale_002dfactor">set-log-cosh-target-distance-scale-factor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmain_002dwindow_002dtitle"><code>set-main-window-title</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmain_002dwindow_002dtitle">set-main-window-title</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dcolour"><code>set-map-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dcolour">set-map-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dcorrelation_002datom_002dradius"><code>set-map-correlation-atom-radius</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dcorrelation_002datom_002dradius">set-map-correlation-atom-radius</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002ddisplayed"><code>set-map-displayed</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002ddisplayed">set-map-displayed</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dis_002ddifference_002dmap"><code>set-map-is-difference-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dis_002ddifference_002dmap">set-map-is-difference-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dline_002dwidth"><code>set-map-line-width</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dline_002dwidth">set-map-line-width</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dmask_002datom_002dradius"><code>set-map-mask-atom-radius</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dmask_002datom_002dradius">set-map-mask-atom-radius</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dradius"><code>set-map-radius</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dradius">set-map-radius</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dradius_002dem"><code>set-map-radius-em</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dradius_002dem">set-map-radius-em</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dsampling_002drate"><code>set-map-sampling-rate</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dsampling_002drate">set-map-sampling-rate</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dsampling_002drate_002dtext"><code>set-map-sampling-rate-text</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dsampling_002drate_002dtext">set-map-sampling-rate-text</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmap_002dsharpening_002dscale_002dlimit"><code>set-map-sharpening-scale-limit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmap_002dsharpening_002dscale_002dlimit">set-map-sharpening-scale-limit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmatch_002delement"><code>set-match-element</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dlsq">coot-lsq</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmatrix-1"><code>set-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmatrix">set-matrix</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmax_002dskeleton_002dsearch_002ddepth"><code>set-max-skeleton-search-depth</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmax_002dskeleton_002dsearch_002ddepth">set-max-skeleton-search-depth</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmodel_002dfit_002drefine_002ddialog_002dposition"><code>set-model-fit-refine-dialog-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmodel_002dfit_002drefine_002ddialog_002dposition">set-model-fit-refine-dialog-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop"><code>set-model-fit-refine-dialog-stays-on-top</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmodel_002dfit_002drefine_002ddialog_002dstays_002don_002dtop">set-model-fit-refine-dialog-stays-on-top</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel"><code>set-model-fit-refine-place-atom-at-pointer-label</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmodel_002dfit_002drefine_002dplace_002datom_002dat_002dpointer_002dlabel">set-model-fit-refine-place-atom-at-pointer-label</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel"><code>set-model-fit-refine-rotate-translate-zone-label</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmodel_002dfit_002drefine_002drotate_002dtranslate_002dzone_002dlabel">set-model-fit-refine-rotate-translate-zone-label</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmodel_002dtoolbar_002ddocked_002dposition"><code>set-model-toolbar-docked-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmodel_002dtoolbar_002ddocked_002dposition">set-model-toolbar-docked-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmol_002dactive"><code>set-mol-active</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmol_002dactive">set-mol-active</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmol_002ddisplayed"><code>set-mol-displayed</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmol_002ddisplayed">set-mol-displayed</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation"><code>set-molecule-bonds-colour-map-rotation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmolecule_002dbonds_002dcolour_002dmap_002drotation">set-molecule-bonds-colour-map-rotation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmolecule_002dname"><code>set-molecule-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmolecule_002dname">set-molecule-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmonomer_002drestraints"><code>set-monomer-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmonomer_002drestraints">set-monomer-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine"><code>set-mutate-auto-fit-do-post-refine</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dmutate_002dauto_002dfit_002ddo_002dpost_002drefine">set-mutate-auto-fit-do-post-refine</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dncs_002dghost_002dbond_002dthickness"><code>set-ncs-ghost-bond-thickness</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dncs_002dghost_002dbond_002dthickness">set-ncs-ghost-bond-thickness</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dncs_002dhomology_002dlevel"><code>set-ncs-homology-level</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dncs_002dhomology_002dlevel">set-ncs-homology-level</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dnomenclature_002derrors_002don_002dread"><code>set-nomenclature-errors-on-read</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dnomenclature_002derrors_002don_002dread">set-nomenclature-errors-on-read</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed"><code>set-only-last-model-molecule-displayed</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002donly_002dlast_002dmodel_002dmolecule_002ddisplayed">set-only-last-model-molecule-displayed</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dpick_002dcursor_002dindex"><code>set-pick-cursor-index</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dpick_002dcursor_002dindex">set-pick-cursor-index</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dprefer_002dpython"><code>set-prefer-python</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dprefer_002dpython">set-prefer-python</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dramachandran_002dplot_002dbackground_002dblock_002dsize"><code>set-ramachandran-plot-background-block-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dramachandran_002dplot_002dbackground_002dblock_002dsize">set-ramachandran-plot-background-block-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dramachandran_002dplot_002dcontour_002dlevels"><code>set-ramachandran-plot-contour-levels</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dramachandran_002dplot_002dcontour_002dlevels">set-ramachandran-plot-contour-levels</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dramachandran_002dplot_002ddialog_002dposition"><code>set-ramachandran-plot-dialog-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dramachandran_002dplot_002ddialog_002dposition">set-ramachandran-plot-dialog-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dramachandran_002dpsi_002daxis_002dmode"><code>set-ramachandran-psi-axis-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dramachandran_002dpsi_002daxis_002dmode">set-ramachandran-psi-axis-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002draster3d_002datom_002dradius"><code>set-raster3d-atom-radius</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002draster3d_002datom_002dradius">set-raster3d-atom-radius</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002draster3d_002dbond_002dthickness"><code>set-raster3d-bond-thickness</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002draster3d_002dbond_002dthickness">set-raster3d-bond-thickness</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002draster3d_002dbone_002dthickness"><code>set-raster3d-bone-thickness</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002draster3d_002dbone_002dthickness">set-raster3d-bone-thickness</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002draster3d_002ddensity_002dthickness"><code>set-raster3d-density-thickness</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002draster3d_002ddensity_002dthickness">set-raster3d-density-thickness</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002draster3d_002dfont_002dsize"><code>set-raster3d-font-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002draster3d_002dfont_002dsize">set-raster3d-font-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002draster3d_002dshadows_002denabled"><code>set-raster3d-shadows-enabled</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002draster3d_002dshadows_002denabled">set-raster3d-shadows-enabled</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002draster3d_002dwater_002dsphere"><code>set-raster3d-water-sphere</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002draster3d_002dwater_002dsphere">set-raster3d-water-sphere</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drecentre_002don_002dread_002dpdb"><code>set-recentre-on-read-pdb</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drecentre_002don_002dread_002dpdb">set-recentre-on-read-pdb</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefine_002dauto_002drange_002dstep"><code>set-refine-auto-range-step</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefine_002dauto_002drange_002dstep">set-refine-auto-range-step</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefine_002dmax_002dresidues"><code>set-refine-max-residues</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefine_002dmax_002dresidues">set-refine-max-residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefine_002dramachandran_002dangles"><code>set-refine-ramachandran-angles</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefine_002dramachandran_002dangles">set-refine-ramachandran-angles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefine_002dramachandran_002drestraints_002dtype"><code>set-refine-ramachandran-restraints-type</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefine_002dramachandran_002drestraints_002dtype">set-refine-ramachandran-restraints-type</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefine_002dramachandran_002drestraints_002dweight"><code>set-refine-ramachandran-restraints-weight</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefine_002dramachandran_002drestraints_002dweight">set-refine-ramachandran-restraints-weight</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefine_002dwith_002dtorsion_002drestraints"><code>set-refine-with-torsion-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefine_002dwith_002dtorsion_002drestraints">set-refine-with-torsion-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefinement_002ddrag_002delasticity"><code>set-refinement-drag-elasticity</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefinement_002ddrag_002delasticity">set-refinement-drag-elasticity</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefinement_002dgeman_002dmcclure_002dalpha"><code>set-refinement-geman-mcclure-alpha</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefinement_002dgeman_002dmcclure_002dalpha">set-refinement-geman-mcclure-alpha</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefinement_002dimmediate_002dreplacement"><code>set-refinement-immediate-replacement</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefinement_002dimmediate_002dreplacement">set-refinement-immediate-replacement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefinement_002dlennard_002djones_002depsilon"><code>set-refinement-lennard-jones-epsilon</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefinement_002dlennard_002djones_002depsilon">set-refinement-lennard-jones-epsilon</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy"><code>set-refinement-move-atoms-with-zero-occupancy</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefinement_002dmove_002datoms_002dwith_002dzero_002doccupancy">set-refinement-move-atoms-with-zero-occupancy</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefinement_002drefine_002dper_002dframe"><code>set-refinement-refine-per-frame</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefinement_002drefine_002dper_002dframe">set-refinement-refine-per-frame</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drefmac_002dcounter"><code>set-refmac-counter</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drefmac_002dcounter">set-refmac-counter</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drenderer_002dshow_002datoms"><code>set-renderer-show-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drenderer_002dshow_002datoms">set-renderer-show-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dreorienting_002dnext_002dresidue_002dmode"><code>set-reorienting-next-residue-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dreorienting_002dnext_002dresidue_002dmode">set-reorienting-next-residue-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dreset_002db_002dfactor_002dmoved_002datoms"><code>set-reset-b-factor-moved-atoms</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dreset_002db_002dfactor_002dmoved_002datoms">set-reset-b-factor-moved-atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dresidue_002ddensity_002dfit_002dscale_002dfactor"><code>set-residue-density-fit-scale-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dresidue_002ddensity_002dfit_002dscale_002dfactor">set-residue-density-fit-scale-factor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dresidue_002dname"><code>set-residue-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dresidue_002dname">set-residue-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dresidue_002dselection_002dflash_002dframes_002dnumber"><code>set-residue-selection-flash-frames-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dresidue_002dselection_002dflash_002dframes_002dnumber">set-residue-selection-flash-frames-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dresidue_002dto_002drotamer_002dname"><code>set-residue-to-rotamer-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dresidue_002dto_002drotamer_002dname">set-residue-to-rotamer-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dresidue_002dto_002drotamer_002dnumber"><code>set-residue-to-rotamer-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dresidue_002dto_002drotamer_002dnumber">set-residue-to-rotamer-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dresidue_002dtype_002dchooser_002dstub_002dstate"><code>set-residue-type-chooser-stub-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dresidue_002dtype_002dchooser_002dstub_002dstate">set-residue-type-chooser-stub-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction"><code>set-rigid-body-fit-acceptable-fit-fraction</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drigid_002dbody_002dfit_002dacceptable_002dfit_002dfraction">set-rigid-body-fit-acceptable-fit-fraction</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drocking_002dfactors"><code>set-rocking-factors</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drocking_002dfactors">set-rocking-factors</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine"><code>set-rotamer-auto-fit-do-post-refine</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drotamer_002dauto_002dfit_002ddo_002dpost_002drefine">set-rotamer-auto-fit-do-post-refine</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotamer_002dcheck_002dclashes"><code>set-rotamer-check-clashes</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drotamer_002dcheck_002dclashes">set-rotamer-check-clashes</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotamer_002dlowest_002dprobability"><code>set-rotamer-lowest-probability</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drotamer_002dlowest_002dprobability">set-rotamer-lowest-probability</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotamer_002dsearch_002dmode"><code>set-rotamer-search-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drotamer_002dsearch_002dmode">set-rotamer-search-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotamer_002dselection_002ddialog_002dposition"><code>set-rotamer-selection-dialog-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drotamer_002dselection_002ddialog_002dposition">set-rotamer-selection-dialog-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotate_002dtranslate_002ddialog_002dposition"><code>set-rotate-translate-dialog-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drotate_002dtranslate_002ddialog_002dposition">set-rotate-translate-dialog-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotation_002dcenter"><code>set-rotation-center</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotation_002dcenter_002dsize"><code>set-rotation-center-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotation_002dcentre-1"><code>set-rotation-centre</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drotation_002dcentre">set-rotation-centre</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drotation_002dcentre_002dsize"><code>set-rotation-centre-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drotation_002dcentre_002dsize">set-rotation-centre-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002drun_002dstate_002dfile_002dstatus"><code>set-run-state-file-status</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002drun_002dstate_002dfile_002dstatus">set-run-state-file-status</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory"><code>set-save-coordinates-in-original-directory</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsave_002dcoordinates_002din_002doriginal_002ddirectory">set-save-coordinates-in-original-directory</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsave_002dstate_002dfile_002dname"><code>set-save-state-file-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsave_002dstate_002dfile_002dname">set-save-state-file-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dscroll_002dby_002dwheel_002dmouse"><code>set-scroll-by-wheel-mouse</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dscroll_002dby_002dwheel_002dmouse">set-scroll-by-wheel-mouse</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dscroll_002dwheel_002dmap"><code>set-scroll-wheel-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dscroll_002dwheel_002dmap">set-scroll-wheel-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dscrollable_002dmap"><code>set-scrollable-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dscrollable_002dmap">set-scrollable-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsecondary_002dstructure_002drestraints_002dtype"><code>set-secondary-structure-restraints-type</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsecondary_002dstructure_002drestraints_002dtype">set-secondary-structure-restraints-type</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dseg_002dids_002din_002datom_002dlabels"><code>set-seg-ids-in-atom-labels</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dseg_002dids_002din_002datom_002dlabels">set-seg-ids-in-atom-labels</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002daniso"><code>set-show-aniso</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002daniso">set-show-aniso</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dchiral_002dvolume_002derrors_002ddialog"><code>set-show-chiral-volume-errors-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dchiral_002dvolume_002derrors_002ddialog">set-show-chiral-volume-errors-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002denvironment_002ddistances"><code>set-show-environment-distances</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002denvironment_002ddistances">set-show-environment-distances</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002denvironment_002ddistances_002das_002dsolid"><code>set-show-environment-distances-as-solid</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002denvironment_002ddistances_002das_002dsolid">set-show-environment-distances-as-solid</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002denvironment_002ddistances_002dbumps"><code>set-show-environment-distances-bumps</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002denvironment_002ddistances_002dbumps">set-show-environment-distances-bumps</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002denvironment_002ddistances_002dh_002dbonds"><code>set-show-environment-distances-h-bonds</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002denvironment_002ddistances_002dh_002dbonds">set-show-environment-distances-h-bonds</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dgraphics_002dligand_002dview"><code>set-show-graphics-ligand-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dgraphics_002dligand_002dview">set-show-graphics-ligand-view</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dorigin_002dmarker"><code>set-show-origin-marker</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dorigin_002dmarker">set-show-origin-marker</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dpaths_002din_002ddisplay_002dmanager"><code>set-show-paths-in-display-manager</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dpaths_002din_002ddisplay_002dmanager">set-show-paths-in-display-manager</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dpointer_002ddistances"><code>set-show-pointer-distances</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dpointer_002ddistances">set-show-pointer-distances</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dstrict_002dncs"><code>set-show-strict-ncs</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dstrict_002dncs">set-show-strict-ncs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dsymmetry_002dmaster"><code>set-show-symmetry-master</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dsymmetry_002dmaster">set-show-symmetry-master</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dsymmetry_002dmolecule"><code>set-show-symmetry-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dsymmetry_002dmolecule">set-show-symmetry-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dunit_002dcell"><code>set-show-unit-cell</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dunit_002dcell">set-show-unit-cell</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dshow_002dunit_002dcells_002dall"><code>set-show-unit-cells-all</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dshow_002dunit_002dcells_002dall">set-show-unit-cells-all</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dskeleton_002dbox_002dsize"><code>set-skeleton-box-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dskeleton_002dbox_002dsize">set-skeleton-box-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsmooth_002dscroll_002ddo_002dzoom"><code>set-smooth-scroll-do-zoom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsmooth_002dscroll_002ddo_002dzoom">set-smooth-scroll-do-zoom</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsmooth_002dscroll_002dflag"><code>set-smooth-scroll-flag</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsmooth_002dscroll_002dflag">set-smooth-scroll-flag</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsmooth_002dscroll_002dlimit"><code>set-smooth-scroll-limit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsmooth_002dscroll_002dlimit">set-smooth-scroll-limit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsmooth_002dscroll_002dsteps"><code>set-smooth-scroll-steps</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsmooth_002dscroll_002dsteps">set-smooth-scroll-steps</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsocket_002dpython_002dstring_002dwaiting"><code>set-socket-python-string-waiting</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsocket_002dpython_002dstring_002dwaiting">set-socket-python-string-waiting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsocket_002dstring_002dwaiting"><code>set-socket-string-waiting</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsocket_002dstring_002dwaiting">set-socket-string-waiting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsolid_002ddensity_002dsurface_002dopacity"><code>set-solid-density-surface-opacity</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsolid_002ddensity_002dsurface_002dopacity">set-solid-density-surface-opacity</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dspace_002dgroup"><code>set-space-group</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dspace_002dgroup">set-space-group</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsticky_002dsort_002dby_002ddate"><code>set-sticky-sort-by-date</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsticky_002dsort_002dby_002ddate">set-sticky-sort-by-date</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dstop_002dscroll_002ddiff_002dmap"><code>set-stop-scroll-diff-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dstop_002dscroll_002ddiff_002dmap">set-stop-scroll-diff-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dstop_002dscroll_002ddiff_002dmap_002dlevel"><code>set-stop-scroll-diff-map-level</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dstop_002dscroll_002ddiff_002dmap_002dlevel">set-stop-scroll-diff-map-level</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dstop_002dscroll_002diso_002dmap"><code>set-stop-scroll-iso-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dstop_002dscroll_002diso_002dmap">set-stop-scroll-iso-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dstop_002dscroll_002diso_002dmap_002dlevel"><code>set-stop-scroll-iso-map-level</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dstop_002dscroll_002diso_002dmap_002dlevel">set-stop-scroll-iso-map-level</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dswap_002ddifference_002dmap_002dcolors"><code>set-swap-difference-map-colors</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#americanisms">americanisms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dswap_002ddifference_002dmap_002dcolours"><code>set-swap-difference-map-colours</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dswap_002ddifference_002dmap_002dcolours">set-swap-difference-map-colours</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsymmetry_002datom_002dlabels_002dexpanded"><code>set-symmetry-atom-labels-expanded</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsymmetry_002datom_002dlabels_002dexpanded">set-symmetry-atom-labels-expanded</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsymmetry_002dcolour"><code>set-symmetry-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsymmetry_002dcolour">set-symmetry-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsymmetry_002dcolour_002dby_002dsymop"><code>set-symmetry-colour-by-symop</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsymmetry_002dcolour_002dby_002dsymop">set-symmetry-colour-by-symop</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap"><code>set-symmetry-molecule-rotate-colour-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsymmetry_002dmolecule_002drotate_002dcolour_002dmap">set-symmetry-molecule-rotate-colour-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsymmetry_002dshift_002dsearch_002dsize"><code>set-symmetry-shift-search-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsymmetry_002dshift_002dsearch_002dsize">set-symmetry-shift-search-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsymmetry_002dsize"><code>set-symmetry-size</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsymmetry_002dsize">set-symmetry-size</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dsymmetry_002dwhole_002dchain"><code>set-symmetry-whole-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dsymmetry_002dwhole_002dchain">set-symmetry-whole-chain</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dtorsion_002dpy"><code>set-torsion-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dtorsion_002dpy">set-torsion-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dtorsion_002dscm"><code>set-torsion-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dtorsion_002dscm">set-torsion-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dtransparent_002delectrostatic_002dsurface"><code>set-transparent-electrostatic-surface</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dtransparent_002delectrostatic_002dsurface">set-transparent-electrostatic-surface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dundo_002dmolecule"><code>set-undo-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dundo_002dmolecule">set-undo-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dunit_002dcell_002dand_002dspace_002dgroup"><code>set-unit-cell-and-space-group</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dunit_002dcell_002dand_002dspace_002dgroup">set-unit-cell-and-space-group</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule"><code>set-unit-cell-and-space-group-using-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dunit_002dcell_002dand_002dspace_002dgroup_002dusing_002dmolecule">set-unit-cell-and-space-group-using-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dunpathed_002dbackup_002dfile_002dnames"><code>set-unpathed-backup-file-names</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dunpathed_002dbackup_002dfile_002dnames">set-unpathed-backup-file-names</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule"><code>set-use-grey-carbons-for-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002duse_002dgrey_002dcarbons_002dfor_002dmolecule">set-use-grey-carbons-for-molecule</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions"><code>set-use-only-extra-torsion-restraints-for-torsions</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002duse_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions">set-use-only-extra-torsion-restraints-for-torsions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002duse_002dstroke_002dcharacters"><code>set-use-stroke-characters</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002duse_002dstroke_002dcharacters">set-use-stroke-characters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002duse_002dtrans_002dpeptide_002drestraints"><code>set-use-trans-peptide-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002duse_002dtrans_002dpeptide_002drestraints">set-use-trans-peptide-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002duse_002dvariable_002dbond_002dthickness"><code>set-use-variable-bond-thickness</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002duse_002dvariable_002dbond_002dthickness">set-use-variable-bond-thickness</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dview_002dmatrix"><code>set-view-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#quat_002dconvert">quat-convert</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dview_002dmatrix-1"><code>set-view-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dview_002dquaternion"><code>set-view-quaternion</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dview_002dquaternion">set-view-quaternion</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dvirtual_002dtrackball_002dtype"><code>set-virtual-trackball-type</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit"><code>set-water-check-spherical-variance-limit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#set_002dwater_002dcheck_002dspherical_002dvariance_002dlimit">set-water-check-spherical-variance-limit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-setup_002dmutate_002dauto_002dfit"><code>setup-mutate-auto-fit</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#setup_002dmutate_002dauto_002dfit">setup-mutate-auto-fit</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-setup_002dtorsion_002dgeneral"><code>setup-torsion-general</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#setup_002dtorsion_002dgeneral">setup-torsion-general</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sfcalc_002dgenmap"><code>sfcalc-genmap</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#sfcalc_002dgenmap">sfcalc-genmap</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sharpen"><code>sharpen</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#sharpen">sharpen</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sharpen_002dblur_002dmap"><code>sharpen-blur-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#sharpen_002dblur_002dmap">sharpen-blur-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sharpen_002dblur_002dmap_002dwith_002dresampling"><code>sharpen-blur-map-with-resampling</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#sharpen_002dblur_002dmap_002dwith_002dresampling">sharpen-blur-map-with-resampling</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shell_002dcommand_002dto_002dfile_002dwith_002ddata"><code>shell-command-to-file-with-data</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shell_002dcommand_002dto_002dstring"><code>shell-command-to-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shell_002dcommand_002dto_002dstring-1"><code>shell-command-to-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shelx_002dmolecule_003f"><code>shelx-molecule?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shelx_002dmolecule_003f-1"><code>shelx-molecule?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shelxl_002drefine"><code>shelxl-refine</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#shelx">shelx</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shelxl_002drefine_002dprimitive"><code>shelxl-refine-primitive</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#shelx">shelx</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shiftfield_002db_002dfactor_002drefinement"><code>shiftfield-b-factor-refinement</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#shiftfield_002db_002dfactor_002drefinement">shiftfield-b-factor-refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shiftfield_002dxyz_002dfactor_002drefinement"><code>shiftfield-xyz-factor-refinement</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#shiftfield_002dxyz_002dfactor_002drefinement">shiftfield-xyz-factor-refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-shorten_002dbaton"><code>shorten-baton</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#shorten_002dbaton">shorten-baton</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dcoot_002dtip_002dfrom_002dlist"><code>show-coot-tip-from-list</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#tips_002dgui">tips-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002denvironment_002ddistances_002dstate"><code>show-environment-distances-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002denvironment_002ddistances_002dstate">show-environment-distances-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dmain_002dtoolbar"><code>show-main-toolbar</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dmain_002dtoolbar">show-main-toolbar</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dmodel_002dtoolbar_002dall_002dicons"><code>show-model-toolbar-all-icons</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dmodel_002dtoolbar_002dall_002dicons">show-model-toolbar-all-icons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dmodel_002dtoolbar_002dmain_002dicons"><code>show-model-toolbar-main-icons</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dmodel_002dtoolbar_002dmain_002dicons">show-model-toolbar-main-icons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dmodelling_002dtoolbar"><code>show-modelling-toolbar</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dmodelling_002dtoolbar">show-modelling-toolbar</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dorigin_002dmarker_002dstate"><code>show-origin-marker-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dorigin_002dmarker_002dstate">show-origin-marker-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dpartial_002dcharge_002dinfo"><code>show-partial-charge-info</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dpartial_002dcharge_002dinfo">show-partial-charge-info</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dpaths_002din_002ddisplay_002dmanager_002dstate"><code>show-paths-in-display-manager-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dpaths_002din_002ddisplay_002dmanager_002dstate">show-paths-in-display-manager-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dpointer_002ddistances_002dstate"><code>show-pointer-distances-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dpointer_002ddistances_002dstate">show-pointer-distances-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dremarks_002dbrowswer"><code>show-remarks-browswer</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dremarks_002dbrowswer">show-remarks-browswer</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002drestraints_002deditor"><code>show-restraints-editor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002drestraints_002deditor">show-restraints-editor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002drestraints_002deditor_002dby_002dindex"><code>show-restraints-editor-by-index</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002drestraints_002deditor_002dby_002dindex">show-restraints-editor-by-index</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dselect_002dmap_002ddialog"><code>show-select-map-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dselect_002dmap_002ddialog">show-select-map-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dset_002dundo_002dmolecule_002dchooser"><code>show-set-undo-molecule-chooser</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dset_002dundo_002dmolecule_002dchooser">show-set-undo-molecule-chooser</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dspacegroup"><code>show-spacegroup</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dspacegroup">show-spacegroup</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-show_002dstrict_002dncs_002dstate"><code>show-strict-ncs-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#show_002dstrict_002dncs_002dstate">show-strict-ncs-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-side_002dby_002dside_002dstereo_002dmode"><code>side-by-side-stereo-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#side_002dby_002dside_002dstereo_002dmode">side-by-side-stereo-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-simple_002dfill_002dpartial_002dresidues"><code>simple-fill-partial-residues</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#simple_002dfill_002dpartial_002dresidues">simple-fill-partial-residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-simple_002dlsq_002dmatch"><code>simple-lsq-match</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dlsq">coot-lsq</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-single_002dmanual_002dncs_002dghost"><code>single-manual-ncs-ghost</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs">ncs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-single_002dmodel_002dview_002dmodel_002dnumber"><code>single-model-view-model-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#single_002dmodel_002dview_002dmodel_002dnumber">single-model-view-model-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-single_002dmodel_002dview_002dnext_002dmodel_002dnumber"><code>single-model-view-next-model-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#single_002dmodel_002dview_002dnext_002dmodel_002dnumber">single-model-view-next-model-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-single_002dmodel_002dview_002dprev_002dmodel_002dnumber"><code>single-model-view-prev-model-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#single_002dmodel_002dview_002dprev_002dmodel_002dnumber">single-model-view-prev-model-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-single_002dmodel_002dview_002dthis_002dmodel_002dnumber"><code>single-model-view-this-model-number</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#single_002dmodel_002dview_002dthis_002dmodel_002dnumber">single-model-view-this-model-number</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-skeletonize_002dmap"><code>skeletonize-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#skeletonize_002dmap">skeletonize-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-skip_002dto_002dnext_002dncs_002dchain"><code>skip-to-next-ncs-chain</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs">ncs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-slash_002dstart_003f"><code>slash-start?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-smiles_002dgui"><code>smiles-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-smooth_002dmap"><code>smooth-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#smooth_002dmap">smooth-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-smooth_002dscroll_002ddo_002dzoom"><code>smooth-scroll-do-zoom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#smooth_002dscroll_002ddo_002dzoom">smooth-scroll-do-zoom</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sort_002dchains"><code>sort-chains</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#sort_002dchains">sort-chains</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-sort_002dresidues"><code>sort-residues</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#sort_002dresidues">sort-residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-space_002dgroup_002dscm"><code>space-group-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#space_002dgroup_002dscm">space-group-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-spin_002dN_002dpy"><code>spin-N-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#spin_002dN_002dpy">spin-N-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-spin_002dN_002dscm"><code>spin-N-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#spin_002dN_002dscm">spin-N-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-spin_002dsearch"><code>spin-search</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#spin_002dsearch">spin-search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-spin_002dsearch_002dpy"><code>spin-search-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#spin_002dsearch_002dpy">spin-search-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-spin_002dzoom_002dtrans"><code>spin-zoom-trans</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#spin_002dzoom_002dtrans">spin-zoom-trans</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-split_002dwater"><code>split-water</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#split_002dwater">split-water</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-standard_002ddeviation_002dtemperature_002dfactor"><code>standard-deviation-temperature-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#standard_002ddeviation_002dtemperature_002dfactor">standard-deviation-temperature-factor</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-start_002dgraphics_002dinterface"><code>start-graphics-interface</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#start_002dgraphics_002dinterface">start-graphics-interface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-start_002djligand_002dlistener"><code>start-jligand-listener</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#jligand">jligand</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-stepped_002drefine_002dprotein"><code>stepped-refine-protein</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-stepped_002drefine_002dprotein_002dfor_002drama"><code>stepped-refine-protein-for-rama</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-stepped_002drefine_002dprotein_002dwith_002drefine_002dfunc"><code>stepped-refine-protein-with-refine-func</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#fitting">fitting</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-stereo_002dmode_002dstate"><code>stereo-mode-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#stereo_002dmode_002dstate">stereo-mode-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-store_002dgraphics_002dwindow_002dposition"><code>store-graphics-window-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#store_002dgraphics_002dwindow_002dposition">store-graphics-window-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-string_002d_003elist_002dof_002dstrings"><code>string-&gt;list-of-strings</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-string_002d_003elist_002dof_002dstrings-1"><code>string-&gt;list-of-strings</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-string_002d_003elist_002dof_002dstrings-2"><code>string-&gt;list-of-strings</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-string_002dappend_002dwith_002dspaces"><code>string-append-with-spaces</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-string_002dappend_002dwith_002dstring"><code>string-append-with-string</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-string_002dconcatenate"><code>string-concatenate</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dcrash_002dcatcher">coot-crash-catcher</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-string_002dconcatenate-1"><code>string-concatenate</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-string_002dmember_003f"><code>string-member?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-strip_002dextension"><code>strip-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-strip_002dextension-1"><code>strip-extension</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-strip_002dleading_002dspaces"><code>strip-leading-spaces</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#what_002dcheck">what-check</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-strip_002dleading_002dspaces-1"><code>strip-leading-spaces</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-strip_002dpath"><code>strip-path</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-strip_002dspaces"><code>strip-spaces</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-suck_002dmodel_002dfit_002ddialog"><code>suck-model-fit-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#suck_002dmodel_002dfit_002ddialog">suck-model-fit-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-superpose"><code>superpose</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#superpose">superpose</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-superpose_002dwith_002datom_002dselection"><code>superpose-with-atom-selection</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#superpose_002dwith_002datom_002dselection">superpose-with-atom-selection</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-superpose_002dwith_002dchain_002dselection"><code>superpose-with-chain-selection</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#superpose_002dwith_002dchain_002dselection">superpose-with-chain-selection</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-svn_002drevision"><code>svn-revision</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#svn_002drevision">svn-revision</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-swap_002dmap_002dcolours"><code>swap-map-colours</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#swap_002dmap_002dcolours">swap-map-colours</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-symmetry_002das_002dcalphas"><code>symmetry-as-calphas</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#symmetry_002das_002dcalphas">symmetry-as-calphas</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate"><code>symmetry-molecule-rotate-colour-map-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#symmetry_002dmolecule_002drotate_002dcolour_002dmap_002dstate">symmetry-molecule-rotate-colour-map-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-symmetry_002doperators_002dpy"><code>symmetry-operators-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#symmetry_002doperators_002dpy">symmetry-operators-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-symmetry_002doperators_002dscm"><code>symmetry-operators-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#symmetry_002doperators_002dscm">symmetry-operators-scm</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-T">T</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-target_002dauto_002dweighting_002dvalue"><code>target-auto-weighting-value</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-teb"><code>teb</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#entry_002bdo_002dbutton">entry+do-button</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-teb2"><code>teb2</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#entry_002bdo_002dbutton">entry+do-button</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-text_002dindex_002dnear_002dposition"><code>text-index-near-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#text_002dindex_002dnear_002dposition">text-index-near-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-tip_002dlist"><code>tip-list</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#tips">tips</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-tips_002dgui"><code>tips-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#tips_002dgui">tips-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-toggle_002dactive_002dmol"><code>toggle-active-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-toggle_002ddisplay_002dmap"><code>toggle-display-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-toggle_002ddisplay_002dmol"><code>toggle-display-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-toggle_002didle_002dligand_002dinteractions"><code>toggle-idle-ligand-interactions</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#redefine_002dfunctions">redefine-functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-toggle_002didle_002drock_002dfunction"><code>toggle-idle-rock-function</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#toggle_002didle_002drock_002dfunction">toggle-idle-rock-function</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-toggle_002didle_002dspin_002dfunction"><code>toggle-idle-spin-function</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#toggle_002didle_002dspin_002dfunction">toggle-idle-spin-function</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-transform_002dcoords_002dmolecule"><code>transform-coords-molecule</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-transform_002dmap"><code>transform-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-transform_002dmap_002dusing_002dlsq_002dmatrix"><code>transform-map-using-lsq-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-transform_002dmolecule_002dby"><code>transform-molecule-by</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#transform_002dmolecule_002dby">transform-molecule-by</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-transform_002dzone"><code>transform-zone</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#transform_002dzone">transform-zone</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-translate_002dmolecule_002dby"><code>translate-molecule-by</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#translate_002dmolecule_002dby">translate-molecule-by</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-translate_002dscene_002dx"><code>translate-scene-x</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#translate_002dscene_002dx">translate-scene-x</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-translate_002dscene_002dy"><code>translate-scene-y</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#translate_002dscene_002dy">translate-scene-y</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-translate_002dscene_002dz"><code>translate-scene-z</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#translate_002dscene_002dz">translate-scene-z</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-translation"><code>translation</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-trim_002dmolecule_002dby_002dmap"><code>trim-molecule-by-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#trim_002dmolecule_002dby_002dmap">trim-molecule-by-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-try_002dset_002ddraw_002dbaton"><code>try-set-draw-baton</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#try_002dset_002ddraw_002dbaton">try-set-draw-baton</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-turn_002doff_002dbackup"><code>turn-off-backup</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#turn_002doff_002dbackup">turn-off-backup</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-turn_002don_002dbackup"><code>turn-on-backup</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#turn_002don_002dbackup">turn-on-backup</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-U">U</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-undisplay_002dall_002dmaps_002dexcept"><code>undisplay-all-maps-except</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-undo_002dlast_002dmove"><code>undo-last-move</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#undo_002dlast_002dmove">undo-last-move</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-undo_002dsymmetry_002dview"><code>undo-symmetry-view</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#undo_002dsymmetry_002dview">undo-symmetry-view</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unique_002ddate_002ftime_002dstr"><code>unique-date/time-str</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unmangle_002dhydrogen_002dname"><code>unmangle-hydrogen-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#unmangle_002dhydrogen_002dname">unmangle-hydrogen-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unpathed_002dbackup_002dfile_002dnames_002dstate"><code>unpathed-backup-file-names-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#unpathed_002dbackup_002dfile_002dnames_002dstate">unpathed-backup-file-names-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unset_002ddots_002dcolour"><code>unset-dots-colour</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#unset_002ddots_002dcolour">unset-dots-colour</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unset_002dmodel_002dfit_002drefine_002ddialog"><code>unset-model-fit-refine-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#unset_002dmodel_002dfit_002drefine_002ddialog">unset-model-fit-refine-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unset_002drefine_002dparams_002ddialog"><code>unset-refine-params-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#unset_002drefine_002dparams_002ddialog">unset-refine-params-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unset_002dsticky_002dsort_002dby_002ddate"><code>unset-sticky-sort-by-date</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#unset_002dsticky_002dsort_002dby_002ddate">unset-sticky-sort-by-date</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-unskeletonize_002dmap"><code>unskeletonize-map</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#unskeletonize_002dmap">unskeletonize-map</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-update_002dgo_002dto_002datom_002dfrom_002dcurrent_002datom"><code>update-go-to-atom-from-current-atom</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition"><code>update-go-to-atom-from-current-position</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#update_002dgo_002dto_002datom_002dfrom_002dcurrent_002dposition">update-go-to-atom-from-current-position</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol"><code>update-go-to-atom-window-on-changed-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#update_002dgo_002dto_002datom_002dwindow_002don_002dchanged_002dmol">update-go-to-atom-window-on-changed-mol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol"><code>update-go-to-atom-window-on-new-mol</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#update_002dgo_002dto_002datom_002dwindow_002don_002dnew_002dmol">update-go-to-atom-window-on-new-mol</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-update_002dncs_002dghosts_002dby_002dlocal_002dsphere"><code>update-ncs-ghosts-by-local-sphere</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#ncs">ncs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-update_002dself"><code>update-self</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-use_002dgraphics_002dinterface_002dstate"><code>use-graphics-interface-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#use_002dgraphics_002dinterface_002dstate">use-graphics-interface-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate"><code>use-only-extra-torsion-restraints-for-torsions-state</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#use_002donly_002dextra_002dtorsion_002drestraints_002dfor_002dtorsions_002dstate">use-only-extra-torsion-restraints-for-torsions-state</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-use_002dunimodal_002dring_002dtorsion_002drestraints"><code>use-unimodal-ring-torsion-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#use_002dunimodal_002dring_002dtorsion_002drestraints">use-unimodal-ring-torsion-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue"><code>use-unimodal-ring-torsion-restraints-for-residue</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#use_002dunimodal_002dring_002dtorsion_002drestraints_002dfor_002dresidue">use-unimodal-ring-torsion-restraints-for-residue</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-useful_002dglob"><code>useful-glob</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#snarf_002dcoot_002ddocs">snarf-coot-docs</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-useful_002dglob-1"><code>useful-glob</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-user_002dmods_002dgui"><code>user-mods-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-user_002dmods_002dpy"><code>user-mods-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#user_002dmods_002dpy">user-mods-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-user_002dmods_002dscm"><code>user-mods-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#user_002dmods_002dscm">user-mods-scm</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-V">V</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-valid_002dlabels"><code>valid-labels</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#valid_002dlabels">valid-labels</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-valid_002dmap_002dmolecule_003f"><code>valid-map-molecule?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-valid_002dmodel_002dmolecule_003f"><code>valid-model-molecule?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-valid_002drefinement_002dmap_003f"><code>valid-refinement-map?</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-view_002dmatrix"><code>view-matrix</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-view_002dname"><code>view-name</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#view_002dname">view-name</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-view_002dname_002dpy"><code>view-name-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#view_002dname_002dpy">view-name-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-view_002dquaternion"><code>view-quaternion</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dutils">coot-utils</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-view_002dsaver_002dgui"><code>view-saver-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-views_002dpanel_002dgui"><code>views-panel-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-vt_002dsurface"><code>vt-surface</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#vt_002dsurface">vt-surface</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-vt_002dsurface_002dstatus"><code>vt-surface-status</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#vt_002dsurface_002dstatus">vt-surface-status</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-W">W</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-water_002dchain_002dfrom_002dshelx_002dins_002dscm"><code>water-chain-from-shelx-ins-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#water_002dchain_002dfrom_002dshelx_002dins_002dscm">water-chain-from-shelx-ins-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-water_002dchain_002dpy"><code>water-chain-py</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#water_002dchain_002dpy">water-chain-py</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-water_002dchain_002dscm"><code>water-chain-scm</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#water_002dchain_002dscm">water-chain-scm</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-water_002dcoordination_002dgui"><code>water-coordination-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-watson_002dcrick_002dpair"><code>watson-crick-pair</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#watson_002dcrick_002dpair">watson-crick-pair</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-watson_002dcrick_002dpair_002dfor_002dresidue_002drange"><code>watson-crick-pair-for-residue-range</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#watson_002dcrick_002dpair_002dfor_002dresidue_002drange">watson-crick-pair-for-residue-range</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-wrapped_002dcreate_002dfind_002dwaters_002ddialog"><code>wrapped-create-find-waters-dialog</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#wrapped_002dcreate_002dfind_002dwaters_002ddialog">wrapped-create-find-waters-dialog</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-wrapper_002dalignment_002dmismatches_002dgui"><code>wrapper-alignment-mismatches-gui</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#coot_002dgui">coot-gui</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dccp4mg_002dpicture_002ddescription"><code>write-ccp4mg-picture-description</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002dccp4mg_002dpicture_002ddescription">write-ccp4mg-picture-description</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dcif_002dfile"><code>write-cif-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002dcif_002dfile">write-cif-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dinterpolated_002dextra_002drestraints"><code>write-interpolated-extra-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002dinterpolated_002dextra_002drestraints">write-interpolated-extra-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints"><code>write-interpolated-models-and-extra-restraints</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002dinterpolated_002dmodels_002dand_002dextra_002drestraints">write-interpolated-models-and-extra-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dpdb_002dfile"><code>write-pdb-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002dpdb_002dfile">write-pdb-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dreduce_002dhet_002ddict"><code>write-reduce-het-dict</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#generic_002dobjects">generic-objects</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002drefmac_002dparallel_002dplane_002drestraint"><code>write-refmac-parallel-plane-restraint</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#user_002ddefine_002drestraints">user-define-restraints</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dresidue_002drange_002dto_002dpdb_002dfile"><code>write-residue-range-to-pdb-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002dresidue_002drange_002dto_002dpdb_002dfile">write-residue-range-to-pdb-file</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002drestraints_002dcif_002ddictionary"><code>write-restraints-cif-dictionary</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002drestraints_002dcif_002ddictionary">write-restraints-cif-dictionary</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dsequence"><code>write-sequence</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002dsequence">write-sequence</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-write_002dshelx_002dins_002dfile"><code>write-shelx-ins-file</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#write_002dshelx_002dins_002dfile">write-shelx-ins-file</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
<tr><th><a name="Function-Index_fn_letter-Z">Z</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-zalman_002dstereo_002dmode"><code>zalman-stereo-mode</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#zalman_002dstereo_002dmode">zalman-stereo-mode</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-zoom_002dfactor"><code>zoom-factor</code></a>:</td><td>&nbsp;</td><td valign="top"><a href="#zoom_002dfactor">zoom-factor</a></td></tr>
<tr><td colspan="4"> <hr></td></tr>
</table>
<table><tr><th valign="top">Jump to: &nbsp; </th><td><a class="summary-letter" href="#Function-Index_fn_symbol-1"><b>%</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_symbol-2"><b>*</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_symbol-3"><b>3</b></a>
 &nbsp; 
<br>
<a class="summary-letter" href="#Function-Index_fn_letter-A"><b>A</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-B"><b>B</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-C"><b>C</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-D"><b>D</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-E"><b>E</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-F"><b>F</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-G"><b>G</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-H"><b>H</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-I"><b>I</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-J"><b>J</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-K"><b>K</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-L"><b>L</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-M"><b>M</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-N"><b>N</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-O"><b>O</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-P"><b>P</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-Q"><b>Q</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-R"><b>R</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-S"><b>S</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-T"><b>T</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-U"><b>U</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-V"><b>V</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-W"><b>W</b></a>
 &nbsp; 
<a class="summary-letter" href="#Function-Index_fn_letter-Z"><b>Z</b></a>
 &nbsp; 
</td></tr></table>
 
<div class="footnote">
<hr>
<h4 class="footnotes-heading">Footnotes</h4>

<h3><a name="FOOT1" href="#DOCF1">(1)</a></h3>
<p>The official CCP4 graphics program (which contains parts of Coot
    (and Coot contains parts of CCP4MG)), CCP4MG is under the
    direct control of Liz Potterton and Stuart McNicholas.</p>
<h3><a name="FOOT2" href="#DOCF2">(2)</a></h3>
<p>although it does have a
    local refinement algorithm it is no substitute for refmac
    (a wrapper for refmac is available).</p>
<h3><a name="FOOT3" href="#DOCF3">(3)</a></h3>
<p>that&rsquo;s the job of the
     CCP4 Program Suite.</p>
<h3><a name="FOOT4" href="#DOCF4">(4)</a></h3>
<p>it&rsquo;s Free Software after all and I could give you a
  hand.</p>
<h3><a name="FOOT5" href="#DOCF5">(5)</a></h3>
<p>I
  can get by with a one button Macintosh - but it&rsquo;s not ideal.</p>
<h3><a name="FOOT6" href="#DOCF6">(6)</a></h3>
<p>as it is known in Clipper.</p>
<h3><a name="FOOT7" href="#DOCF7">(7)</a></h3>
<p>coot
  has several influences and dependencies, but these will not be
  discussed here in the User Manual.</p>
<h3><a name="FOOT8" href="#DOCF8">(8)</a></h3>
<p>COOT_BACKUP_DIR is used
  in preference if set</p>
<h3><a name="FOOT9" href="#DOCF9">(9)</a></h3>
<p>The map-reading
  problem (documented in Section <a href="#Maps-in-General">Maps in General</a>) is already
  known.</p>
<h3><a name="FOOT10" href="#DOCF10">(10)</a></h3>
<p>now
there&rsquo;s a hostage to fortune.</p>
<h3><a name="FOOT11" href="#DOCF11">(11)</a></h3>
<p>to do so, please send me the output of the following:
  <tt>$ gdb `which coot-real` <em>corefile</em></tt> and then at the
  <tt>(gdb)</tt> prompt type: <code>where</code>, where
  <tt><em>corefile</em></tt> is the core dump file, <samp>core</samp> or
  <samp>core.4536</samp> or some such.</p>
<h3><a name="FOOT12" href="#DOCF12">(12)</a></h3>
<p>See also <a href="#more-on-zooming">more on zooming</a></p>
<h3><a name="FOOT13" href="#DOCF13">(13)</a></h3>
<p>Mouse movement in
  &ldquo;Spherical Surface&rdquo; mode generates a component of (often
  undesirable) screen z-rotation, particularly noticeable when the
  mouse is at the edge of the screen.</p>
<h3><a name="FOOT14" href="#DOCF14">(14)</a></h3>
<p><code>(vt-surface 0)</code> to turn it back to &ldquo;Spherical&rdquo;
  mode.</p>
<h3><a name="FOOT15" href="#DOCF15">(15)</a></h3>
<p>this is off by default
  because I find it annoying.</p>
<h3><a name="FOOT16" href="#DOCF16">(16)</a></h3>
<p>Put your mouse over a widget for a couple of seconds,
if that widget has a tooltip, it will pop-up in a yellow box (or a grey
box for some reason if you are using Macintosh).</p>
<h3><a name="FOOT17" href="#DOCF17">(17)</a></h3>
<p>that works for me, at least.</p>
<h3><a name="FOOT18" href="#DOCF18">(18)</a></h3>
<p>Povray support is only semi-working, there is a problem
 with the orientation of the image.</p>
<h3><a name="FOOT19" href="#DOCF19">(19)</a></h3>
<p>to a limited
  extent.</p>
<h3><a name="FOOT20" href="#DOCF20">(20)</a></h3>
<p>the substantial
  majority of the time you will want your the buttons to be both
  either depressed or undepressed, rarely one but not the other.</p>
<h3><a name="FOOT21" href="#DOCF21">(21)</a></h3>
<p>British
modelling, of course</p>
<h3><a name="FOOT22" href="#DOCF22">(22)</a></h3>
<p>which
causes the evaluation of the command.</p>
<h3><a name="FOOT23" href="#DOCF23">(23)</a></h3>
<p>such as the Coot
state file (Section <a href="#Coot-State">Coot State</a>).</p>
<h3><a name="FOOT24" href="#DOCF24">(24)</a></h3>
<p>in that particular directory.</p>
<h3><a name="FOOT25" href="#DOCF25">(25)</a></h3>
<p>funny that</p>
<h3><a name="FOOT26" href="#DOCF26">(26)</a></h3>
<p>this might be initially surprising
since this could chew up a lot of disk space.  However, disk space is
cheap compared to losing you molecule.</p>
<h3><a name="FOOT27" href="#DOCF27">(27)</a></h3>
<p>The coordinates are written in
pdb format - that&rsquo;s OK, isn&rsquo;t it?.</p>
<h3><a name="FOOT28" href="#DOCF28">(28)</a></h3>
<p>unless you tell it not to, of course -
use (<em>e.g.</em>)  <code>(turn-off-backup 0)</code> to turn off the backup
(for molecule 0 in this case).</p>
<h3><a name="FOOT29" href="#DOCF29">(29)</a></h3>
<p>or system manager.</p>
<h3><a name="FOOT30" href="#DOCF30">(30)</a></h3>
<p>It works like the &ldquo;Forwards&rdquo; buttons in a web
browser - which is not available immediately after viewing a new
page.</p>
<h3><a name="FOOT31" href="#DOCF31">(31)</a></h3>
<p>for example, if coot crashes.</p>
<h3><a name="FOOT32" href="#DOCF32">(32)</a></h3>
<p>if no space group has been assigned it
  returns <code>``No spacegroup for this molecule''</code></p>
<h3><a name="FOOT33" href="#DOCF33">(33)</a></h3>
<p>I find a clipping
  level of about 3.5 to 4 comfortable for viewing electron density
  maps - it is a little &ldquo;thinner&rdquo; than the default startup
  thickness.</p>
<h3><a name="FOOT34" href="#DOCF34">(34)</a></h3>
<p>Inspired by PyMol?  Yep&hellip; sure was!</p>
<h3><a name="FOOT35" href="#DOCF35">(35)</a></h3>
<p>and <tt class="key">C</tt> again to toggle them off.</p>
<h3><a name="FOOT36" href="#DOCF36">(36)</a></h3>
<p>compared to some other one.</p>
<h3><a name="FOOT37" href="#DOCF37">(37)</a></h3>
<p>prompted by changing the contour
  level.</p>
<h3><a name="FOOT38" href="#DOCF38">(38)</a></h3>
<p><em>i.e.</em> the terminal
  in which you started Coot.</p>
<h3><a name="FOOT39" href="#DOCF39">(39)</a></h3>
<p>Hue Saturation Value (Intensity).</p>
<h3><a name="FOOT40" href="#DOCF40">(40)</a></h3>
<p>they
  can be redisplayed using <code>(set-draw-hydrogens <em>mol-no</em> 1)</code>.</p>
<h3><a name="FOOT41" href="#DOCF41">(41)</a></h3>
<p>the
reference chain is, by default, the first chain of that type in the
coordinates file.  The reference (master) chain can be changed using
the NCS Ghosts Control dialog.</p>
<h3><a name="FOOT42" href="#DOCF42">(42)</a></h3>
<p>that also only makes sense in the region of the reference
  chain.</p>
<h3><a name="FOOT43" href="#DOCF43">(43)</a></h3>
<p>OCA is &ldquo;goose&rdquo; in Spanish (and Italian)</p>
<h3><a name="FOOT44" href="#DOCF44">(44)</a></h3>
<p>the default is the
  Weizmann Institute - which for reasons I won&rsquo;t go into here is
  currently much faster than the EBI server.</p>
<h3><a name="FOOT45" href="#DOCF45">(45)</a></h3>
<p>which (currently) only a fraction are.</p>
<h3><a name="FOOT46" href="#DOCF46">(46)</a></h3>
<p>using thermal ellipsoids</p>
<h3><a name="FOOT47" href="#DOCF47">(47)</a></h3>
<p>in the case
  of isotropic atoms, ellipsoids are spherical, of course.</p>
<h3><a name="FOOT48" href="#DOCF48">(48)</a></h3>
<p>symmetry labels are in pale blue/purple and also
  provide the symmetry operator number and the translations along the
  a, b and c axes.</p>
<h3><a name="FOOT49" href="#DOCF49">(49)</a></h3>
<p>contacts involving two hydrogens
or at least one carbon atom are yellow, denoting &rsquo;bump&rsquo;.  Hydrogen
contacts display cut-off are based on the user-defined maximum
distance, but shortened by 0.5&Aring; per hydrogen atom.</p>
<h3><a name="FOOT50" href="#DOCF50">(50)</a></h3>
<p>Use <code>Angle</code> for an angle, of
  course.</p>
<h3><a name="FOOT51" href="#DOCF51">(51)</a></h3>
<p><em>e.g.</em> this
molecule was a map or a closed molecule.</p>
<h3><a name="FOOT52" href="#DOCF52">(52)</a></h3>
<p>the same one as in the CCP4 program
SUPERPOSE</p>
<h3><a name="FOOT53" href="#DOCF53">(53)</a></h3>
<p>given a half-decent window
  manager</p>
<h3><a name="FOOT54" href="#DOCF54">(54)</a></h3>
<p>Diamond, R. (1971). A Real-Space Refinement Procedure
  for Proteins. <em>Acta Crystallographica</em> <b>A</b>27, 436-452.
  </p>
<h3><a name="FOOT55" href="#DOCF55">(55)</a></h3>
<p>but the resulting <em>\chi^2</em> values are higher.</p>
<h3><a name="FOOT56" href="#DOCF56">(56)</a></h3>
<p>And a few extras, such as
  phosphate</p>
<h3><a name="FOOT57" href="#DOCF57">(57)</a></h3>
<p>unless you are using Buccaneer, of course</p>
<h3><a name="FOOT58" href="#DOCF58">(58)</a></h3>
<p>if you
  want to move only one residue, then click the same atom twice.</p>
<h3><a name="FOOT59" href="#DOCF59">(59)</a></h3>
<p>like Refinement and Regularization</p>
<h3><a name="FOOT60" href="#DOCF60">(60)</a></h3>
<p>like Regularization and Refinement.</p>
<h3><a name="FOOT61" href="#DOCF61">(61)</a></h3>
<p>T. J.  Oldfield &amp; R. E. Hubbard (1994).  &ldquo;Analysis of
C<em>\alpha</em> Geometry in Protein Structures&rdquo; <em>Proteins-Structure
Function and Genetics</em> <b>18(4)</b> 324 &ndash; 337.</p>
<h3><a name="FOOT62" href="#DOCF62">(62)</a></h3>
<p>which is quite likely at first since
  coot has no knowledge of where the chain has been and cannot score
  according to geometric criteria.</p>
<h3><a name="FOOT63" href="#DOCF63">(63)</a></h3>
<p>&ldquo;b&rdquo; again toggles the mode off.</p>
<h3><a name="FOOT64" href="#DOCF64">(64)</a></h3>
<p>high-resolution map
  interpretation is planned.</p>
<h3><a name="FOOT65" href="#DOCF65">(65)</a></h3>
<p><em>i.e.</em> one less than the starting residue in the
    forward direction (defaults to 1).</p>
<h3><a name="FOOT66" href="#DOCF66">(66)</a></h3>
<p>R. M. Esnouf &ldquo;Polyalanine Reconstruction from
  C<em>\alpha</em> Positions Using the Program <em>CALPHA</em> Can Aid Initial
  Phasing of Data by Molecular Replacement Procedures&rdquo; <em>Acta
    Cryst. </em>, D<b>53</b>, 666-672 (1997).</p>
<h3><a name="FOOT67" href="#DOCF67">(67)</a></h3>
<p>T.A.  Jones &amp; S. Thirup &ldquo;Using known
  substructures in protein model building and crystallography&rdquo;
  <em>EMBO J.</em> <b>5</b>, 819&ndash;822 (1986).</p>
<h3><a name="FOOT68" href="#DOCF68">(68)</a></h3>
<p>and
  high resolution</p>
<h3><a name="FOOT69" href="#DOCF69">(69)</a></h3>
<p><em>i.e.</em>  there are
  severely misplaced atoms</p>
<h3><a name="FOOT70" href="#DOCF70">(70)</a></h3>
<p>as for
Rotate/Translate Zone (Section <a href="#Rotate_002fTranslate-Zone">Rotate/Translate Zone</a>).</p>
<h3><a name="FOOT71" href="#DOCF71">(71)</a></h3>
<p>since version 0.4</p>
<h3><a name="FOOT72" href="#DOCF72">(72)</a></h3>
<p>SC Lovell, JM Word, JS Richardson and DC Richardson (2000)
&quot;The Penultimate Rotamer Library&quot; Proteins: Structure Function and
Genetics 40: 389-408.  You can get the paper from
<a href="http://kinemage.biochem.duke.edu/databases/rotamer.php">http://kinemage.biochem.duke.edu/databases/rotamer.php</a></p>
<h3><a name="FOOT73" href="#DOCF73">(73)</a></h3>
<p>&quot;The Backrub Motion: How Protein Backbone Shrugs
     When a Sidechain Dances&quot; <em>Structure</em>, Volume <b>14</b>, Issue 2, Pages
     265-274 I. Davis, W. Bryan Arendall III, D. Richardson, J. Richardson</p>
<h3><a name="FOOT74" href="#DOCF74">(74)</a></h3>
<p>for example, in preparation of
  a model for molecular replacement</p>
<h3><a name="FOOT75" href="#DOCF75">(75)</a></h3>
<p>atoms of residues
  that are too close to each other</p>
<h3><a name="FOOT76" href="#DOCF76">(76)</a></h3>
<p>that&rsquo;s
  &ldquo;zero&rdquo;.</p>
<h3><a name="FOOT77" href="#DOCF77">(77)</a></h3>
<p>Note that selecting a residue type that matches the
  residue in the graphics will also result in a mutation</p>
<h3><a name="FOOT78" href="#DOCF78">(78)</a></h3>
<p><em>i.e.</em> the current residue type matches the
  residue type to which you wish to mutate the residue</p>
<h3><a name="FOOT79" href="#DOCF79">(79)</a></h3>
<p><em>e.g.</em> after using Ca
  -&gt; Mainchain.</p>
<h3><a name="FOOT80" href="#DOCF80">(80)</a></h3>
<p><em>i.e.</em> the residue atoms will remain untouched</p>
<h3><a name="FOOT81" href="#DOCF81">(81)</a></h3>
<p>this is a wrapper round LIBCHECK, so you must
have CCP4 suite to installed for this function to work</p>
<h3><a name="FOOT82" href="#DOCF82">(82)</a></h3>
<p>rather than running LIBCHECK again</p>
<h3><a name="FOOT83" href="#DOCF83">(83)</a></h3>
<p>you can search for many different
ligand types.</p>
<h3><a name="FOOT84" href="#DOCF84">(84)</a></h3>
<p>usually if this still fails after
two repetitions then it never seems to work.</p>
<h3><a name="FOOT85" href="#DOCF85">(85)</a></h3>
<p>The
map is selected using &ldquo;Refine/Regularize Control&rdquo;</p>
<h3><a name="FOOT86" href="#DOCF86">(86)</a></h3>
<p><em>e.g.</em> <code>(add-OXT-to-residue 0 428 &quot;&quot; &quot;A&quot;)</code></p>
<h3><a name="FOOT87" href="#DOCF87">(87)</a></h3>
<p>including
  sulfate or phosphate ions (in such a case, it is probably useful to
do a &ldquo;Rigid Body Fit Zone&rdquo; on that new residue).</p>
<h3><a name="FOOT88" href="#DOCF88">(88)</a></h3>
<p>For example,
  after a ligand search has been performed.</p>
<h3><a name="FOOT89" href="#DOCF89">(89)</a></h3>
<p><em>i.e.</em> Coot is idle and ignores all
  input.</p>
<h3><a name="FOOT90" href="#DOCF90">(90)</a></h3>
<p>This is not an ideal
  feature, of course and will be addressed in future<small class="enddots">...</small>. Digressive
  Musing: If only computers were fast enough to run Refmac
  interactively&hellip;</p>
<h3><a name="FOOT91" href="#DOCF91">(91)</a></h3>
<p>such that Coot would subsequently expect an atom
  selection &ldquo;pick&rdquo; in the graphics window.</p>
<h3><a name="FOOT92" href="#DOCF92">(92)</a></h3>
<p>the current single function doesn&rsquo;t always perform very well
in tests</p>
<h3><a name="FOOT93" href="#DOCF93">(93)</a></h3>
<p>R.W.W. Hooft, G. Vriend, C. Sander, E.E. Abola, Errors in
protein structures.  <em>Nature</em> (1996) <strong>381</strong>, 272-272.</p>
<h3><a name="FOOT94" href="#DOCF94">(94)</a></h3>
<p><em>e.g.</em> it&rsquo;s a directory or a
coordinate filename.</p>
<h3><a name="FOOT95" href="#DOCF95">(95)</a></h3>
<p>like I don&rsquo;t on my Mac.</p>
<h3><a name="FOOT96" href="#DOCF96">(96)</a></h3>
<p>actually, it&rsquo;s a box.</p>
<h3><a name="FOOT97" href="#DOCF97">(97)</a></h3>
<p>in a coot
  difference map you will get to see the negative level contoured at
  the inverted level of the positive level, what I mean is that you
  don&rsquo;t want to see the &ldquo;positive&rdquo; level going less than 0.</p>
<h3><a name="FOOT98" href="#DOCF98">(98)</a></h3>
<p>takes you right back
  to the good old Frodo days, no?</p>
<h3><a name="FOOT99" href="#DOCF99">(99)</a></h3>
<p>Jan Dohnalek,
  for instance.</p>
<h3><a name="FOOT100" href="#DOCF100">(100)</a></h3>
<p>a value of 2.5 is often
  sufficient.</p>
<h3><a name="FOOT101" href="#DOCF101">(101)</a></h3>
<p>If you&rsquo;re living in Sweden<small class="enddots">...</small> or Captain Kirk, that
  is.</p>
<h3><a name="FOOT102" href="#DOCF102">(102)</a></h3>
<p>you may think it strange that a box has a
  radius, this is an idiosyncrasy of Coot.</p>
<h3><a name="FOOT103" href="#DOCF103">(103)</a></h3>
<p>or set their occupancy to
  zero</p>
<h3><a name="FOOT104" href="#DOCF104">(104)</a></h3>
<p>the not-GLY-or-PRO is the most familiar
  Ramachandran plot.</p>
<h3><a name="FOOT105" href="#DOCF105">(105)</a></h3>
<p>prolines have a grey outline rather than a black
  one, triangles are glycines.</p>
<h3><a name="FOOT106" href="#DOCF106">(106)</a></h3>
<p>but that&rsquo;s partly the idea, I suppose.</p>
<h3><a name="FOOT107" href="#DOCF107">(107)</a></h3>
<p>and therefore most interesting</p>
<h3><a name="FOOT108" href="#DOCF108">(108)</a></h3>
<p>or the water should be more properly modelled as
  anistrotropic or a split partial site</p>
<h3><a name="FOOT109" href="#DOCF109">(109)</a></h3>
<p>and remember to check the difference map button in the &ldquo;Run
Refmac&rdquo; dialog</p>
</div>
<hr>



</body>
</html>
